From ad3a04e80d0489fdeedacfe79e24944e87443389 Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Thu, 15 Mar 2018 11:37:53 +0100
Subject: [PATCH] bigobj

---
 src/libOpenImageIO/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libOpenImageIO/CMakeLists.txt b/src/libOpenImageIO/CMakeLists.txt
index 98d399f..aafb427 100644
--- a/src/libOpenImageIO/CMakeLists.txt
+++ b/src/libOpenImageIO/CMakeLists.txt
@@ -149,6 +149,10 @@ endif ()
 
 if (MSVC)
     set_property (SOURCE imagebufalgo_pixelmath.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " /bigobj ")
+    set_property (SOURCE imagebufalgo_addsub.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " /bigobj ")
+    set_property (SOURCE imagebufalgo_muldiv.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " /bigobj ")
+    set_property (SOURCE imagebufalgo_copy.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " /bigobj ")
+    set_property (SOURCE imagebuf.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " /bigobj ")
 endif ()
 
 if (WIN32)
-- 
2.5.3

