--- setup.py	2022-02-07 01:50:01.000000000 +0000
+++ setup.py	2022-02-07 01:50:24.000000000 +0000
@@ -8,8 +8,8 @@
 
 requirements = ["torch"]
 
-with open("./requirements.txt", "r", encoding="utf-8") as fh:
-    install_requires = fh.read()
+# with open("./requirements.txt", "r", encoding="utf-8") as fh:
+#     install_requires = fh.read()
 
 with open("./README.md", "r", encoding="utf-8") as fh:
     long_description = fh.read()
@@ -17,7 +17,7 @@
 
 
 setup(
-    install_requires=install_requires,
+#     install_requires=install_requires,
     name="spikingjelly",
     version="0.0.0.0.8",
     author="PKU MLG, PCL, and other contributors",
