--- /dev/null	2023-10-31 08:18:41.070325754 -0500
+++ b/pyproject.toml	2023-11-02 14:18:34.877316586 -0500
@@ -0,0 +1,12 @@
+[project]
+name = "aioresponses"  
+version = "0.7.4"  
+description = "Mock out requests made by ClientSession from aiohttp package"  
+readme = "README.rst" 
+requires-python = ">=3.6"
+
+license = {file = "LICENSE"}
+
+authors = [
+  {name = "Pawel Nuckowski", email = "p.nuckowski@gmail.com" }
+]
