diff --git a/PKG-INFO b/PKG-INFO
index 0ef6895..88bb5ce 100644
--- a/PKG-INFO
+++ b/PKG-INFO
@@ -10,7 +10,7 @@ Classifier: License :: OSI Approved :: BSD License
 Classifier: Operating System :: OS Independent
 Classifier: Programming Language :: Python :: 3
 Requires-Python: >=3.8
-Requires-Dist: cbor
+Requires-Dist: cbor2
 Requires-Dist: pyzmq
 Description-Content-Type: text/markdown
 
diff --git a/pyproject.toml b/pyproject.toml
index 4b5eae7..64b7501 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -18,7 +18,7 @@ classifiers = [
 ]
 dependencies = [
     'pyzmq',
-    'cbor',
+    'cbor2',
 ]
 
 [project.urls]
