diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS -x autom4te.cache mpi-serial-MPIserial_2.3.0.orig/Makefile mpi-serial-MPIserial_2.3.0/Makefile
--- mpi-serial-MPIserial_2.3.0.orig/Makefile	2023-05-11 16:29:07.781842054 +0200
+++ mpi-serial-MPIserial_2.3.0/Makefile	2023-05-11 16:29:26.412931834 +0200
@@ -83,7 +83,7 @@
 tests:
 	cd tests; make
 
-install: lib
+install: $(LIB)
 	$(MKINSTALLDIRS) $(libdir) $(includedir)
 	$(INSTALL) lib$(MODULE).a -m 644 $(libdir)
 	$(INSTALL) mpi.h -m 644 $(includedir)
