diff --git a/src/napari_aideveloper/_widget.py b/src/napari_aideveloper/_widget.py
index 0ab9ec8..d8aaa4b 100644
--- a/src/napari_aideveloper/_widget.py
+++ b/src/napari_aideveloper/_widget.py
@@ -14,6 +14,7 @@ import platform
 import sys
 import time
 import traceback
+import warnings
 from stat import S_IREAD, S_IRGRP, S_IROTH, S_IWGRP, S_IWOTH, S_IWRITE
 
 import cv2
@@ -31,7 +32,7 @@ from tensorflow.keras.utils import to_categorical
 
 from . import aid_bin, aid_dl, aid_img, aid_start, model_zoo
 
-np.warnings.filterwarnings('ignore', category=np.VisibleDeprecationWarning)
+warnings.filterwarnings('ignore', category=np.VisibleDeprecationWarning)
 
 
 VERSION = "0.3.0" #Python 3.9.9 Version
