--- drivers.list.orig	2015-05-08 15:52:28.182154417 -0400
+++ drivers.list	2015-05-08 15:53:51.101288817 -0400
@@ -37,14 +37,14 @@
 ! NEDRIV 0 /NEXT      Computers running NeXTstep operating system
   NUDRIV 0 /NULL      Null device (no output)				Std F77
 ! PGDRIV 0 /PGMF      PGPLOT metafile (new format, experimental)        Std F77
-! PNDRIV 1 /PNG       Portable Network Graphics file                    C
-! PNDRIV 2 /TPNG      Portable Network Graphics file - transparent background C
+  PNDRIV 1 /PNG       Portable Network Graphics file                    C
+  PNDRIV 2 /TPNG      Portable Network Graphics file - transparent background C
 ! PPDRIV 1 /PPM       Portable Pixel Map file, landscape
 ! PPDRIV 2 /VPPM      Portable PIxel Map file, portrait
-! PSDRIV 1 /PS        PostScript printers, monochrome, landscape	Std F77
-! PSDRIV 2 /VPS       Postscript printers, monochrome, portrait		Std F77
-! PSDRIV 3 /CPS       PostScript printers, color, landscape		Std F77
-! PSDRIV 4 /VCPS      PostScript printers, color, portrait		Std F77
+  PSDRIV 1 /PS        PostScript printers, monochrome, landscape	Std F77
+  PSDRIV 2 /VPS       Postscript printers, monochrome, portrait		Std F77
+  PSDRIV 3 /CPS       PostScript printers, color, landscape		Std F77
+  PSDRIV 4 /VCPS      PostScript printers, color, portrait		Std F77
 ! PXDRIV 0 /PRINTRONI Printronix P300 or P600 dot-matrix printer
 ! QMDRIV 1 /QMS       QUIC devices (QMS and Talaris), landscape		Std F77
 ! QMDRIV 2 /VQMS      QUIC devices (QMS and Talaris), portrait		Std F77
@@ -68,8 +68,8 @@
 ! WDDRIV 2 /VWD       X Window dump file, portrait
 ! WSDRIV 0 /WS        VAX workstations running VWS software		VMS
 ! X2DRIV 0 /XDISP     PGDISP or FIGDISP server for X workstations	C
-! XWDRIV 1 /XWINDOW   Workstations running X Window System		C
-! XWDRIV 2 /XSERVE    Persistent window on X Window System		C
+  XWDRIV 1 /XWINDOW   Workstations running X Window System		C
+  XWDRIV 2 /XSERVE    Persistent window on X Window System		C
 ! ZEDRIV 0 /ZETA      Zeta 8 Digital Plotter
 !
 ! The following drivers can only be used in PGPLOT installations on MS-DOS
