--- sys_linux/gfortran_gcc.conf.orig	2015-05-08 15:56:11.249515977 -0400
+++ sys_linux/gfortran_gcc.conf	2015-05-08 15:57:48.302673286 -0400
@@ -0,0 +1,24 @@
+XINCL=""
+MOTIF_INCL="$XINCL"
+ATHENA_INCL="$XINCL"
+TK_INCL="$XINCL -I$PREFIX/include"
+RV_INCL=""
+FCOMPL="gfortran"
+FFLAGC="-ffixed-form -ffixed-line-length-none -u -Wall -fPIC -O"
+FFLAGD="-fno-backslash"
+CCOMPL="gcc"
+CFLAGC="-Wall -fPIC -DPG_PPU -O -I$PREFIX/include -DCONDA_DATADIR=\\\"$PREFIX/share/pgplot/\\\""
+CFLAGD="-Wall -O -I$PREFIX"
+PGBIND_FLAGS="bsd"
+LIBS="-L$PREFIX/lib -lX11"
+MOTIF_LIBS="-lXm -lXt $LIBS"
+ATHENA_LIBS="-lXaw -lXt -lXmu -lXext $LIBS"
+TK_INCL=
+TK_LIBS=
+RANLIB="ranlib"
+SHARED_LIB="libpgplot.so"
+SHARED_LD="gcc -shared -o $SHARED_LIB"
+SHARED_LIB_LIBS="-L$PREFIX/lib -lpng -lgfortran"
+MCOMPL=""
+MFLAGC=""
+SYSDIR="$SYSDIR"
