--- drivers/pndriv.c.orig	2015-05-08 16:14:52.974334131 -0400
+++ drivers/pndriv.c	2015-05-08 16:15:18.892376140 -0400
@@ -222,7 +222,7 @@
 	return;
   }
 
-  if (setjmp(png_ptr->jmpbuf)) { /* not really sure what I'm doing here... */
+  if (setjmp(png_jmpbuf(png_ptr))) { /* not really sure what I'm doing here... */
 	fprintf(stderr,"%s: error in libpng while writing file %s, plotting disabled\n", png_ident, filename);
 	png_destroy_write_struct(&png_ptr,&info_ptr);
 	dev->error = true;
