diff --git a/build.sh b/build.sh
index bc7243e0..fffef9f9 100644
--- a/build.sh
+++ b/build.sh
@@ -349,10 +349,9 @@ common_args="$common_args -Wno-format-nonliteral"
 link_args=""
 
 os_type=`uname`
-
+c_compiler=$CC
+cpp_compiler=$CXX
 if [ "$os_type" = "Linux" ]; then
-   c_compiler=gcc
-   cpp_compiler=g++
 
    # try moving some of these g++ specific warnings into both_args if clang eventually supports them
    both_args="$both_args -Wlogical-op"
@@ -504,8 +503,6 @@ elif [ "$os_type" = "Darwin" ]; then
    # reference on rpath & install_name: https://www.mikeash.com/pyblog/friday-qa-2009-11-06-linking-and-install-names.html
 
    # try moving some of these clang specific warnings into both_args if g++ eventually supports them
-   c_compiler=clang
-   cpp_compiler=clang++
    both_args="$both_args -Wnull-dereference"
    both_args="$both_args -Wgnu-zero-variadic-macro-arguments"
 
