diff --git a/setup.py b/setup.py
--- a/setup.py	(revision 63d09100066380201ba50aada3da6b72b8e48d24)
+++ b/setup.py	(date 1690372051782)
@@ -40,8 +40,10 @@

 if sys.platform.startswith('linux'):
     libcpptraj = 'libcpptraj.so'
-else:
+elif sys.platform.startswith('darwin'):
     libcpptraj = 'libcpptraj.dylib'
+else:
+    libcpptraj = 'libcpptraj.dll'

 amber_release = check_flag('--amber_release')
 disable_openmp = check_flag('--disable-openmp')
diff --git a/scripts/build_wheel.py b/scripts/build_wheel.py
--- a/scripts/build_wheel.py	(revision 63d09100066380201ba50aada3da6b72b8e48d24)
+++ b/scripts/build_wheel.py	(date 1690371638383)
@@ -201,7 +201,7 @@
         pytraj_home = os.path.abspath(
             os.path.dirname(__file__).strip('scripts'))
         cpptraj_home = os.environ.get("CPPTRAJHOME", '')
-        ext = 'so' if not sys.platform.startswith('darwin') else 'dylib'
+        ext = 'dll'
         if not cpptraj_home:
             print('CPPTRAJHOME is not set')
             if not self.cpptraj_dir:
