diff --git a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -17,11 +17,6 @@ classifiers=[
         'Topic :: Scientific/Engineering',
 ]
 
-[tool.poetry.scripts]
-c3d2csv = 'c3d.scripts.c3d2csv:main'
-c3d2npz = 'c3d.scripts.c3d2npz:main'
-c3d-metatdata = 'c3d.scripts.c3d_metatdata:main'
-c3d-viewer = 'c3d.scripts.c3d_viewer:main'
-
 [tool.poetry.dependencies]
 python = "^3.9"
 numpy = ">=1.0.0"
