From 4472d074834ae4dedbee3ae8d5e95936bed5460f Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Fri, 20 Aug 2021 06:38:15 -0400
Subject: [PATCH] Conda packaging changes

---
 oss_scripts/configure.sh | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/oss_scripts/configure.sh b/oss_scripts/configure.sh
index 7752de9..2faab46 100755
--- a/oss_scripts/configure.sh
+++ b/oss_scripts/configure.sh
@@ -37,11 +37,11 @@ function is_macos() {
 # Remove .bazelrc if it already exist
 [ -e .bazelrc ] && rm .bazelrc
 
-if [[ $(pip show tensorflow) == *tensorflow* ]] || [[ $(pip show tf-nightly) == *tf-nightly* ]] ; then
+if [[ $(conda list tensorflow) == *tensorflow* ]] || [[ $(pip show tf-nightly) == *tf-nightly* ]] ; then
   echo 'Using installed tensorflow.'
 else
   echo 'Installing tensorflow.'
-  pip install tensorflow==2.6.0
+  conda install -y tensorflow==2.6.0
 fi
 
 if is_windows; then
@@ -49,12 +49,13 @@ if is_windows; then
   sed -i -e 's/":headers",$/":headers", ":windows_static_link_data",/' third_party/icu/BUILD.bzl
 fi
 
-write_to_bazelrc "build:manylinux2010 --crosstool_top=@ubuntu18.04-gcc7_manylinux2010-cuda11.2-cudnn8.1-tensorrt7.2_config_cuda//crosstool:toolchain"
+#write_to_bazelrc "build:manylinux2010 --crosstool_top=@ubuntu18.04-gcc7_manylinux2010-cuda11.2-cudnn8.1-tensorrt7.2_config_cuda//crosstool:toolchain"
 write_to_bazelrc "build --spawn_strategy=standalone"
 write_to_bazelrc "build --strategy=Genrule=standalone"
 write_to_bazelrc "build -c opt"
 write_to_bazelrc "build --define=framework_shared_object=true"
 write_to_bazelrc "build --experimental_repo_remote_exec"
+write_to_bazelrc "build --action_env PYTHON_BIN_PATH=\"$PYTHON\""
 
 # By default, build in C++ 14 mode.
 write_to_bazelrc "build:android --cxxopt=-std=c++14"
-- 
2.23.0

