From c9acd0181263bc30d86c67054e39e98d42c459f4 Mon Sep 17 00:00:00 2001
From: Jason Furmanek <furmanek@us.ibm.com>
Date: Tue, 13 Jul 2021 18:13:15 -0500
Subject: [PATCH] [PATCH] TF Text to use its own protobuf

---
 WORKSPACE | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/WORKSPACE b/WORKSPACE
index f4f1859..713ba8d 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -13,6 +13,16 @@ http_archive(
     patch_args = ["-p1"],
 )
 
+http_archive(
+    name = "com_google_protobuf",
+    sha256 = "bf0e5070b4b99240183b29df78155eee335885e53a8af8683964579c214ad301",
+    strip_prefix = "protobuf-3.14.0",
+    urls = [
+        "https://storage.googleapis.com/mirror.tensorflow.org/github.com/protocolbuffers/protobuf/archive/v3.14.0.zip",
+        "https://github.com/protocolbuffers/protobuf/archive/v3.14.0.zip",
+    ],
+)
+
 http_archive(
     name = "icu",
     strip_prefix = "icu-release-64-2",
-- 
2.23.0

