--- /root/COMPSs/compss/programming_model/bindings/python/setup.py
+++ ./compss/programming_model/bindings/python/setup.py
@@ -50,2 +50,2 @@ 
        os.environ['JAVA_HOME'] + '/include',
-       os.environ['JAVA_HOME'] + '/include/linux/'
+       os.environ['JAVA_HOME'] + '/include/darwin/'

--- /root/COMPSs/compss/programming_model/bindings/python/setup.py
+++ ./compss/programming_model/bindings/python/setup.py
@@ -62,8 +62,8 @@ 
# Thread affinity extension
-thread_affinity = Extension(
+#thread_affinity = Extension(
-    'thread_affinity',
+#    'thread_affinity',
-    include_dirs=['src/ext'],
+#    include_dirs=['src/ext'],
-    extra_compile_args=['-std=c++11'],
+#    extra_compile_args=['-std=c++11'],
    # extra_compile_args=['-fPIC %s' % (' '.join(gcc_debug_flags.split('\n')))],
-    sources=['src/ext/thread_affinity.cc']
+#    sources=['src/ext/thread_affinity.cc']
-)
+#)

--- /root/COMPSs/compss/programming_model/bindings/python/setup.py
+++ ./compss/programming_model/bindings/python/setup.py
@@ -117,2 +117,2 @@ 
    },
-   ext_modules=[compssmodule, thread_affinity]
+   #ext_modules=[compssmodule, thread_affinity]

