--- /root/COMPSs/install	2018-11-09 10:31:38.497241882 +0000
+++ ./install	2019-04-12 07:09:11.000000000 +0000
@@ -100,19 +100,6 @@
   cd "${SCRIPT_DIR}"
   echo "   Success"
 
-  # Install extrae
-  echo "- Install extrae"
-  cd "${SCRIPT_DIR}"/Dependencies/extrae
-  ./install "${targetDir}"/Dependencies/extrae true
-  cd "${SCRIPT_DIR}"
-  echo "   Success"
-
-  # Install PLUTO
-  echo "- Install Pluto"
-  cd "${SCRIPT_DIR}"/Dependencies/pluto
-  ./install_pluto "${targetDir}"/Dependencies/pluto
-  cd "${SCRIPT_DIR}"
-  echo "   Success"
 
   # Set permissions
   echo "- Set COMPSs permissions"
