From 44201bf763309d4acc1c744e0106fc1f3a98efa7 Mon Sep 17 00:00:00 2001
From: Richard Kalling <rck@kallingsoftware.com>
Date: Fri, 13 Mar 2020 19:07:28 -0700
Subject: [PATCH] changing

---
 setup.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/setup.py b/setup.py
index 4206594..8fc1952 100644
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,9 @@
 from numpy.distutils.core import setup, Extension
 import sys
 
+print("I got patched")
 wrapper = Extension('gacode_ext',
+                    extra_link_args=['-lgomp'],
                     sources=['expro/expro.f90',
                              'expro/expro_util.f90',
                              'expro/expro_pycomm.f90',
-- 
2.24.0

