--- lightgbm/CMakeLists.txt	2017-07-18 15:22:37.000000000 -0400
+++ lightgbm/CMakeLists.txt	2017-07-18 15:23:06.000000000 -0400
@@ -1,10 +1,5 @@
 cmake_minimum_required(VERSION 2.8)
 
-if(APPLE)
-    SET(CMAKE_CXX_COMPILER "g++-7")
-    SET(CMAKE_C_COMPILER "gcc-7")
-endif()
-
 PROJECT(lightgbm)
 
 OPTION(USE_MPI "MPI based parallel learning" OFF)
@@ -136,4 +131,4 @@
         LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
         ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 
-install(DIRECTORY ${LightGBM_HEADER_DIR}/LightGBM DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
\ No newline at end of file
+install(DIRECTORY ${LightGBM_HEADER_DIR}/LightGBM DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
