From 916a8331b181d1257bb5df2b5f67d2208ffe7131 Mon Sep 17 00:00:00 2001
Subject: [PATCH] Add project to allow Clang features to be detected

---
 math/minuit2/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/math/minuit2/CMakeLists.txt b/math/minuit2/CMakeLists.txt
index 8ff0b0f8c4..f9f440d873 100644
--- a/math/minuit2/CMakeLists.txt
+++ b/math/minuit2/CMakeLists.txt
@@ -7,4 +7,5 @@
 cmake_minimum_required(VERSION 3.1)
+project(Minuit2)
 
 option(minuit2_mpi "Enable support for MPI in Minuit2")
 option(minuit2_omp "Enable support for OpenMP in Minuit2")
-- 
2.21.0

