diff --git a/setup.py b/setup.py
index 8c04634..3b553fc 100644
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,5 @@
-import setuptools
-with open("mechasvg\\supl\\README.txt", "r") as fh:
+import setuptools, os.path as op
+with open(op.join("mechasvg", "supl", "README.txt"), "r") as fh:
     long_description = fh.read()
 setuptools.setup(
     name="mechaSVG",
@@ -19,4 +19,4 @@ setuptools.setup(
         "Operating System :: OS Independent",
     ],
     python_requires='>=3.6',
-)
\ No newline at end of file
+)
