diff --git a/larch/apps.py b/larch/apps.py
index 6ac85168..e265f359 100644
--- a/larch/apps.py
+++ b/larch/apps.py
@@ -4,8 +4,6 @@ import numpy
 import time
 from argparse import ArgumentParser
 
-from pyshortcuts import make_shortcut, ico_ext
-
 from .site_config import icondir, home_dir, uname
 from .shell import shell
 from .xmlrpc_server import larch_server_cli
@@ -69,37 +67,12 @@ class LarchApp:
         self.name = name
         self.script = script
         self.terminal = terminal
-        self.icon = icon
         bindir = 'Scripts' if uname == 'win' else 'bin'
         self.bindir = os.path.join(sys.prefix, bindir)
 
     def create_shortcut(self):
-        script = os.path.join(self.bindir, self.script)
-        icon = os.path.join(icondir, self.icon)
-        from pyshortcuts import ico_ext
-        if isinstance(ico_ext, (list, tuple)):
-            for ext in ico_ext:
-                ticon = "{:s}.{:s}".format(icon, ext)
-                if os.path.exists(ticon):
-                    icon = ticon
-
-        if HAS_CONDA and uname == 'win':
-            baserunner = os.path.join(self.bindir, 'baserunner.bat')
-            script = "%s %s" % (baserunner, self.script)
-            if not os.path.exists(baserunner) or os.stat(baserunner).st_size < 10:
-                fh = open(baserunner, 'w')
-                fh.write(WIN_BASERUNNER)
-                fh.close()
-                time.sleep(0.5)
-
-        make_shortcut(script, name=self.name, icon=icon,
-                      terminal=self.terminal, folder='Larch')
-
-        if HAS_CONDA and uname == 'darwin':
-            try:
-                fix_darwin_shebang(script)
-            except:
-                print("Warning: could not fix Mac exe for ", script)
+        ...
+
 
 APPS = (LarchApp('Larch CLI', 'larch', terminal=True),
         LarchApp('Larch GUI', 'larch --wxgui'),
@@ -112,12 +85,6 @@ APPS = (LarchApp('Larch CLI', 'larch', terminal=True),
         LarchApp('1D XRD Viewer', 'xrd1d_viewer') )
 
 
-def make_desktop_shortcuts():
-    """make desktop shortcuts for Larch apps"""
-    for app in APPS:
-        app.create_shortcut()
-
-
 def make_cli(description='run larch program', filedesc='data file'):
     usage = "usage: %prog [options] file"
     parser = ArgumentParser(description=description)
@@ -179,16 +146,6 @@ def run_dioptas_larch():
     from dioptas import main
     main()
 
-def run_feff6l():
-    "run feff6l"
-    from larch.xafs.feffrunner import feff6l_cli
-    feff6l_cli()
-
-def run_feff8l():
-    "run feff8l"
-    from larch.xafs.feffrunner import feff8l_cli
-    feff8l_cli()
-
 def run_larch_server():
     "run larch XMLRPC server"
     larch_server_cli()
@@ -238,7 +195,6 @@ def run_larch():
 
     # create desktop icons
     if args.makeicons:
-        make_desktop_shortcuts()
         return
 
     # run in server mode
diff --git a/setup.py b/setup.py
index f9f5f75e..df89402b 100644
--- a/setup.py
+++ b/setup.py
@@ -188,24 +188,6 @@ setup(name = 'xraylarch',
                    'Topic :: Scientific/Engineering'],
       )
 
-# create desktop icons
-if INSTALL or DEVELOP:
-    py_exe = pjoin(sys.exec_prefix, sys.executable)
-    larch_exe = pjoin(sys.exec_prefix, bindir, larchbin)
-    if not pexists(larch_exe):
-        user = os.path.expanduser('~')
-
-        for base in (sys.prefix, sys.exec_prefix,
-                     pjoin(user, 'anaconda3'),
-                     pjoin(user, 'local'),
-                     '/opt/local', '/anaconda3/', '/opt/anaconda3'):
-            if pexists(pjoin(base, larchbin)):
-                larch_exe = pexists(pjoin(base, larchbin))
-            elif pexists(pjoin(base, 'bin', larchbin)):
-                larch_exe = pexists(pjoin(base, larchbin))
-
-    if pexists(larch_exe):
-        subprocess.check_call((py_exe,  larch_exe, '-m'))
 
 if len(missing) > 0:
     dl = "#%s#" % ("="*75)
