diff --git a/setup.py b/setup.py
index 95cc7289..cd7da6f1 100644
--- a/setup.py
+++ b/setup.py
@@ -82,20 +82,18 @@ if uname == 'win':
     larchbin = 'larch-script.py'
 
 # list of top level scripts to add to Python's bin/
-scripts = ['larch', 'larch_server', 'feff6l', 'feff8l', 'xas_viewer',
+scripts = ['larch', 'larch_server', 'xas_viewer',
            'gse_mapviewer', 'gse_dtcorrect', 'xrd1d_viewer','xrd2d_viewer',
            'dioptas_larch', 'xrfdisplay', 'xrfdisplay_epics']
 
 larch_apps = ['{0:s} = larch.apps:run_{0:s}'.format(n) for n in scripts]
 
-packages = ['larch', 'larch.bin']
+packages = ['larch']
 for pname in find_packages('larch'):
     packages.append('larch.%s' % pname)
 
 
-package_data = ['icons/*', 'xray/*.dat', 'xray/*.db', 'xrd/*.db',
-                'bin/darwin64/*', 'bin/linux64/*', 'bin/win64/*',
-                'bin/win32/*']
+package_data = ['icons/*', 'xray/*.dat', 'xray/*.db', 'xrd/*.db']
 
 
 if INSTALL:
