diff --git a/setup.py b/setup.py
index 76a6a7bf..9a1efead 100644
--- a/setup.py
+++ b/setup.py
@@ -50,10 +50,8 @@ def get_hash():
 
 
 def get_version():
-    with open(version_file, 'r', encoding='utf-8') as f:
-        exec(compile(f.read(), version_file, 'exec'))
-    return locals()['__version__']
-
+    from modelscope.version import __version__ as version
+    return version
 
 def parse_requirements(fname='requirements.txt', with_version=True):
     """
