From 7102d2a28e48e05fa491b6bc70b7dfd1eff81e63 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sat, 27 Nov 2021 16:17:13 +0100
Subject: [PATCH] relax_pycollada_req

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index cd8f3b5..09ef543 100644
--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,7 @@
     'networkx==2.2',    # For joint graph
     'numpy',            # Numpy
     'pillow',           # For texture image loading
-    'pycollada==0.6',   # COLLADA (.dae) mesh loading via trimesh
+    'pycollada',   # COLLADA (.dae) mesh loading via trimesh
     'pyrender>=0.1.20', # For visualization
     'scipy',            # For trimesh, annoyingly
     'six',              # Python 2/3 compatability
    
