From 30a4ed9b097f1ea9a2a1e9303e044015d0041d63 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Wed, 27 Jun 2018 12:46:18 -0500
Subject: [PATCH 1/8] add rt to linkopts

---
 tensorflow/BUILD          | 1 +
 tensorflow/tensorflow.bzl | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index 823ad8f..96e12cb 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -429,6 +429,7 @@ tf_cc_shared_object(
         "//tensorflow:darwin": [],
         "//tensorflow:windows": [],
         "//conditions:default": [
+            "-lrt",
             "-Wl,--version-script",  #  This line must be directly followed by the version_script.lds file
             "$(location //tensorflow:tf_framework_version_script.lds)",
         ],
diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index d93e0df..b8438d2 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -377,7 +377,7 @@ def tf_cc_shared_object(
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ['-lrt'],
         framework_so = tf_binary_additional_srcs(),
         kernels = [],
         **kwargs):
@@ -413,7 +413,7 @@ def tf_cc_binary(
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ['-lrt'],
         copts = tf_copts(),
         kernels = [],
         **kwargs):
-- 
2.7.4

