--- setup.py	2021-08-09 14:09:43.158737155 +0200
+++ setup.py	2021-08-09 14:09:24.306864379 +0200
@@ -21,9 +21,9 @@
         "numpy",
     ],
     use_scm_version={
-        "root": "../../",
+        "root": ".",
         "relative_to": __file__,
-        "write_to": os.path.join("packages", "python", "libdlf", "version.py"),
+        "write_to": os.path.join("libdlf", "version.py"),
     },
     setup_requires=["setuptools_scm"],
 )
