diff --git a/configure.ac b/configure.ac
index dd4aded..9919629 100644
--- a/configure.ac
+++ b/configure.ac
@@ -295,10 +295,10 @@ AC_SUBST(PYTHON)
 # udev v190 introduced the btrfs builtin and a udev rule to use it.
 # Our udev rule gives us the friendly dm names but isn't required (or valid)
 # on earlier releases.
-UDEVDIR=
-if ${PKG_CONFIG} udev --atleast-version 190; then
-	UDEVDIR="$(${PKG_CONFIG} udev --variable=udevdir)"
-fi
+# Because we use a CDT for udev, pkg-config (rightfully) outputs a system path
+# (e.g. /usr/lib/udev). We hard-code override this here to put everything into
+# ${prefix}.
+UDEVDIR=${prefix}/lib/udev
 AC_SUBST(UDEVDIR)
 
 dnl lzo library does not provide pkg-config, let use classic way
