--- bindings/python/run.me.orig    2021-03-02 16:00:00.000000000 +0000
+++ bindings/python/run.me         2021-03-02 16:00:00.000000000 +0000
@@ -7,2 +7,4 @@
 import sysconfig
+import os
+import platform
 import hashlib
@@ -37,12 +37,20 @@
     print("compiling _blst.so...") or sys.stdout.flush()
+    cxx = os.environ.get('CXX', 'c++')
+    py_syst = sysconfig.get_platform().split("-")[0]
     if sysconfig.get_config_var('MACHDEP') == 'darwin':
-        cmd = ["c++", "-bundle", "-undefined", "dynamic_lookup"]
+        cmd = [cxx, "-bundle", "-undefined", "dynamic_lookup"]
     else:
-        cmd = ["c++", "-shared", "-Wl,-Bsymbolic"]
+        cmd = [cxx, "-shared", "-Wl,-Bsymbolic"]
-    cmd.extend(["-o", "_blst.so", "-fPIC", "-fvisibility=hidden",
+    print(f"generated: _blst{sysconfig.get_config_var('EXT_SUFFIX')}")
+    extra_cflags = ["-fvisibility=hidden", "-std=c++11"]
+    if py_syst != 'win':
+        extra_cflags.append("-fPIC")
+    else:
+        extra_cflags.extend([])
+    cmd.extend(["-o", f"_blst{sysconfig.get_config_var('EXT_SUFFIX')}", *extra_cflags,
                 "-I"+sysconfig.get_config_var('INCLUDEPY'), "-I..",
                 "-O", "-Wall", "blst_wrap.cpp", "../libblst.a"])
     try:
-        pre = subprocess.check_output(["c++", "-dM", "-E", "-x", "c++", "/dev/null"],
-                                      stderr=subprocess.STDOUT)
+        pre = subprocess.check_output([cxx, "-dM", "-E", "-x", "c++", "/dev/null"],
+                                      stderr=subprocess.STDOUT)
         std = re.search(r'\b__cplusplus\s+([0-9]+)L', pre.decode('ascii'))
@@ -52,3 +53,4 @@
         pass
-    subprocess.check_call(cmd)
+    subprocess.check_call(cmd, stderr=subprocess.STDOUT)
+    print("done.") or sys.stdout.flush()

