--- /dev/null  2021-03-02 16:00:00.000000000 +0000
+++ bindings/python/setup.py        2021-03-02 16:00:00.000000000 +0000
@@ -0,0 +1,21 @@
+from setuptools import setup, Extension
+import platform
+import os
+
+system = platform.system().lower()
+if system == 'darwin':
+    ext = '.dylib'
+elif system == 'windows':
+    ext = '.dll'
+else:
+    ext = '.so'
+
+setup(
+    name='blst',
+    version=os.getenv('PKG_VERSION', '0.1'),
+    packages=['blst'],
+    package_dir={'blst': '.'},
+    package_data={'blst': [f'_blst.*{ext}', '__init__.py']},
+    zip_safe=False,
+)
+
