--- /dev/null  2021-03-02 16:00:00.000000000 +0000
+++ bindings/python/setup.py        2021-03-02 16:00:00.000000000 +0000
@@ -0,0 +1,13 @@
+from setuptools import setup, Extension
+import sysconfig
+import os
+
+setup(
+    name='blst',
+    version=os.getenv('PKG_VERSION', '0.1'),
+    packages=['blst'],
+    package_dir={'blst': '.'},
+    package_data={'blst': [f'_blst{sysconfig.get_config_var("EXT_SUFFIX")}', '__init__.py']},
+    zip_safe=False,
+)
+
