--- build/refresh.sh.orig  2021-03-02 16:00:00.000000000 +0000
+++ build/refresh.sh        2021-03-02 16:00:00.000000000 +0000
@@ -9,3 +9,3 @@
     s=`basename $pl .pl`.asm
-    expr $s : '.*portable' > /dev/null || (set -x; ${PERL} $pl masm > win64/$s)
+    expr $s : '.*portable' > /dev/null || (set -x; ${PERL} $pl masm > win64/$s && ${PERL} $pl nasm > win64-nasm/$s)
     s=`basename $pl .pl`.s
@@ -18,3 +18,3 @@
     s=`basename $pl .pl`.asm
-    (set -x; ${PERL} $pl win64 > win64/$s)
+    (set -x; ${PERL} $pl win64 > win64/$s && ${PERL} $pl nasm > win64-nasm/$s)
     s=`basename $pl .pl`.S
@@ -30,3 +30,3 @@
   echo "EXPORTS"
-  cc -E blst.h | \
+  ${CC} -E blst.h | \
   ${PERL} -ne '{ (/(blst_[\w]+)\s*\(/ || /(BLS12_[\w]+);/) &&  print "\t$1\n" }'
