From ae5e4c3e972aaa03207305d4f2bc021980ed3bb3 Mon Sep 17 00:00:00 2001
From: AntoinePrv <AntoinePrv@users.noreply.github.com>
Date: Thu, 6 Aug 2020 09:46:44 -0400
Subject: [PATCH] Fix boost multiprecision Clang

---
 src/papilo/misc/MultiPrecision.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/papilo/misc/MultiPrecision.hpp b/src/papilo/misc/MultiPrecision.hpp
index 72c43ee..93cd3f8 100644
--- a/src/papilo/misc/MultiPrecision.hpp
+++ b/src/papilo/misc/MultiPrecision.hpp
@@ -53,6 +53,7 @@ using Quad = boost::multiprecision::cpp_bin_float_quad;
 #endif
 
 #ifdef PAPILO_HAVE_GMP
+#include <boost/serialization/nvp.hpp>
 #include <boost/multiprecision/cpp_bin_float.hpp>
 #include <boost/multiprecision/gmp.hpp>
 
@@ -121,6 +122,7 @@ load( Archive& ar,
 } // namespace boost
 
 #else
+#include <boost/serialization/nvp.hpp>
 #include <boost/multiprecision/cpp_bin_float.hpp>
 #include <boost/multiprecision/cpp_int.hpp>
 
-- 
2.28.0

