--- a/cadquery/freecad_impl/__init__.py
+++ b/cadquery/freecad_impl/__init__.py
@@ -63,7 +63,7 @@ def _fc_path():

     # Try to guess if using Anaconda
     if 'env' in sys.prefix:
-        if sys.platform.startswith('linux'):
+        if sys.platform.startswith('linux') or sys.platform.startswith('darwin'):
             _PATH = os.path.join(sys.prefix,'lib')
             # return PATH if FreeCAD.[so,pyd] is present
             if len(glob.glob(os.path.join(_PATH,'FreeCAD.so'))) > 0:
