Fix compilation on macOS

diff --git a/libintervalxt/src/impl/decomposition_state.hpp b/libintervalxt/src/impl/decomposition_state.hpp
index d3928fa..615d00d 100644
--- a/libintervalxt/src/impl/decomposition_state.hpp
+++ b/libintervalxt/src/impl/decomposition_state.hpp
@@ -24,6 +24,7 @@
 #include <boost/logic/tribool.hpp>
 #include <deque>
 #include <list>
+#include <unordered_map>

 #include "../../intervalxt/interval_exchange_transformation.hpp"

diff --git a/libintervalxt/src/dynamical_decomposition.cc b/libintervalxt/src/dynamical_decomposition.cc
index 8a3ef38..7bd40d1 100644
--- a/libintervalxt/src/dynamical_decomposition.cc
+++ b/libintervalxt/src/dynamical_decomposition.cc
@@ -25,6 +25,7 @@
 #include <deque>
 #include <ostream>
 #include <vector>
+#include <unordered_map>

 #include "../intervalxt/component.hpp"
 #include "../intervalxt/connection.hpp"
diff --git a/libintervalxt/src/interval_exchange_transformation.cc b/libintervalxt/src/interval_exchange_transformation.cc
index 4d0bebb..fe1a4ac 100644
--- a/libintervalxt/src/interval_exchange_transformation.cc
+++ b/libintervalxt/src/interval_exchange_transformation.cc
@@ -26,6 +26,7 @@
 #include <algorithm>
 #include <list>
 #include <unordered_set>
+#include <unordered_map>

 #include "../intervalxt/fmt.hpp"
 #include "../intervalxt/induction_step.hpp"

