Fix compilation on macOS

diff --git a/libintervalxt/src/impl/decomposition_state.hpp b/libintervalxt/src/impl/decomposition_state.hpp
index d3928fa..615d00d 100644
--- a/libintervalxt/src/impl/decomposition_state.hpp
+++ b/libintervalxt/src/impl/decomposition_state.hpp
@@ -24,6 +24,7 @@
 #include <boost/logic/tribool.hpp>
 #include <deque>
 #include <list>
+#include <unordered_map>

 #include "../../intervalxt/interval_exchange_transformation.hpp"

