Fix compilation on macOS

diff --git a/libintervalxt/src/impl/decomposition_state.hpp b/libintervalxt/src/impl/decomposition_state.hpp
index d3928fa..615d00d 100644
--- a/libintervalxt/src/impl/decomposition_state.hpp
+++ b/libintervalxt/src/impl/decomposition_state.hpp
@@ -24,6 +24,7 @@
 #include <boost/logic/tribool.hpp>
 #include <deque>
 #include <list>
+#include <unordered_map>

 #include "../../intervalxt/interval_exchange_transformation.hpp"

diff --git a/libintervalxt/src/dynamical_decomposition.cc b/libintervalxt/src/dynamical_decomposition.cc
index 8a3ef38..7bd40d1 100644
--- a/libintervalxt/src/dynamical_decomposition.cc
+++ b/libintervalxt/src/dynamical_decomposition.cc
@@ -25,6 +25,7 @@
 #include <deque>
 #include <ostream>
 #include <vector>
+#include <unordered_map>

 #include "../intervalxt/component.hpp"
 #include "../intervalxt/connection.hpp"
diff --git a/libintervalxt/src/interval_exchange_transformation.cc b/libintervalxt/src/interval_exchange_transformation.cc
index 4d0bebb..fe1a4ac 100644
--- a/libintervalxt/src/interval_exchange_transformation.cc
+++ b/libintervalxt/src/interval_exchange_transformation.cc
@@ -26,6 +26,7 @@
 #include <algorithm>
 #include <list>
 #include <unordered_set>
+#include <unordered_map>

 #include "../intervalxt/fmt.hpp"
 #include "../intervalxt/induction_step.hpp"
diff --git a/libintervalxt/test/connection.test.cc b/libintervalxt/test/connection.test.cc
index b815c8a..20db252 100644
--- a/libintervalxt/test/connection.test.cc
+++ b/libintervalxt/test/connection.test.cc
@@ -37,7 +37,7 @@ TEST_CASE("Connection in a Decomposition") {

   const auto component = decomposition.components()[0];

-  REQUIRE(std::get<Connection>(component.left()[0]) == -std::get<Connection>(component.right()[0]));
+  REQUIRE(*std::get_if<Connection>(&component.left()[0]) == -*std::get_if<Connection>(&component.right()[0]));
 }

 TEST_CASE("Injected Connections in Decompositions") {
diff --git a/libintervalxt/test/separatrix.test.cc b/libintervalxt/test/separatrix.test.cc
index ad6c5af..cb12771 100644
--- a/libintervalxt/test/separatrix.test.cc
+++ b/libintervalxt/test/separatrix.test.cc
@@ -41,8 +41,8 @@ TEST_CASE("Separatrix in a Decomposition") {
   REQUIRE(component.left().size() == 1);
   REQUIRE(component.right().size() == 1);

-  const auto left = std::get<Connection>(component.left()[0]);
-  const auto right = std::get<Connection>(component.right()[0]);
+  const auto left = *std::get_if<Connection>(&component.left()[0]);
+  const auto right = *std::get_if<Connection>(&component.right()[0]);

   REQUIRE(left.source() == right.target());
   REQUIRE(left.source() != right.source());
