diff --git a/setup.py b/setup.py
index 15403a4fe..c5b82a2ff 100644
--- a/setup.py
+++ b/setup.py
@@ -67,14 +67,13 @@ INSTALL_REQUIRES = [
     "sseclient-py>=1.7.2,<2",
     "sse-starlette>=0.10.3,<1",
     "starlette>=0.24.0",
-    "strawberry-graphql==0.138.1",
+    "strawberry-graphql",
     "tabulate",
     "xmltodict",
     "universal-analytics-python3>=1.0.1,<2",
     "pydash==8.0.1",
     # internal packages
     "fiftyone-brain>=0.16.1,<0.17",
-    "fiftyone-db>=0.4,<2.0",
     "voxel51-eta>=0.12.6,<0.13",
 ]
 
