diff --git a/fiftyone/constants.py b/fiftyone/constants.py
index 1c08e3cfe..2096f94d9 100644
--- a/fiftyone/constants.py
+++ b/fiftyone/constants.py
@@ -90,11 +90,7 @@ DEFAULT_COLOR_SCHEME = {
 }
 
 # MongoDB setup
-try:
-    from fiftyone.db import FIFTYONE_DB_BIN_DIR
-except ImportError:
-    # development installation
-    FIFTYONE_DB_BIN_DIR = os.path.join(FIFTYONE_CONFIG_DIR, "bin")
+FIFTYONE_DB_BIN_DIR = str(Path(which("mongod")).parent)
 
 DEFAULT_DB_DIR = os.path.join(FIFTYONE_CONFIG_DIR, "var", "lib", "mongo")
 MIGRATIONS_PATH = os.path.join(FIFTYONE_CONFIG_DIR, "migrations")
