diff --git a/setup.py b/setup.py
index f8ac8e32f..c3e8ab462 100644
--- a/setup.py
+++ b/setup.py
@@ -50,7 +50,7 @@ INSTALL_REQUIRES = [
     # https://plotly.com/python/static-image-export/
     "kaleido!=0.2.1.post1",
     "matplotlib",
-    "mongoengine==0.24.2",
+    "mongoengine",
     "motor>=2.5",
     "numpy",
     "packaging",
@@ -71,13 +71,12 @@ INSTALL_REQUIRES = [
     "sseclient-py>=1.7.2,<2",
     "sse-starlette>=0.10.3,<1",
     "starlette>=0.24.0",
-    "strawberry-graphql==0.138.1",
+    "strawberry-graphql",
     "tabulate",
     "xmltodict",
     "universal-analytics-python3>=1.0.1,<2",
     # internal packages
     "fiftyone-brain>=0.16.1,<0.17",
-    "fiftyone-db>=0.4,<2.0",
     "voxel51-eta>=0.12.6,<0.13",
 ]
 
