diff --git a/modules/domino/src/subset_filters.cpp b/modules/domino/src/subset_filters.cpp
index 03f4f5f951..c224fe7780 100644
--- a/modules/domino/src/subset_filters.cpp
+++ b/modules/domino/src/subset_filters.cpp
@@ -25,7 +25,7 @@ namespace {
 /** A minimum restraint score based SubsetFilter.
     See MinimumRestraintScoreSubsetFilterTable.
  */
-class IMPDOMINOEXPORT MinimumRestraintScoreSubsetFilter : public SubsetFilter {
+class MinimumRestraintScoreSubsetFilter : public SubsetFilter {
   Pointer<RestraintCache> rc_;
   RestraintsTemp rs_;
   Slices slices_;
diff --git a/modules/rmf/src/restraint_io.cpp b/modules/rmf/src/restraint_io.cpp
index 4c7aeb3f5e..6245a5a91f 100644
--- a/modules/rmf/src/restraint_io.cpp
+++ b/modules/rmf/src/restraint_io.cpp
@@ -32,7 +32,7 @@ class RMFRestraint;
 IMP_OBJECTS(RMFRestraint, RMFRestraints);
 /** A dummy restraint object to represent restraints loaded from
     an RMF file.*/
-class IMPRMFEXPORT RMFRestraint : public Restraint {
+class RMFRestraint : public Restraint {
   ParticlesTemp ps_;
   PointerMember<RestraintInfo> info_;
 
