diff --git a/modules/statistics/src/internal/centrality_clustering.cpp b/modules/statistics/src/internal/centrality_clustering.cpp
index 1c6242dfcf..6a290c7042 100644
--- a/modules/statistics/src/internal/centrality_clustering.cpp
+++ b/modules/statistics/src/internal/centrality_clustering.cpp
@@ -34,7 +34,7 @@ using graph::has_no_edges;
 // (Note that this may not be the right #if condition; we have only seen
 // the issue with Boost 1.71 through 1.74 on Macs with Homebrew, but other
 // systems may be affected.)
-#if defined(__clang__) && BOOST_VERSION >= 107100 && BOOST_VERSION <= 107200
+#if defined(__clang__) && BOOST_VERSION >= 106700 && BOOST_VERSION <= 107200
 # include "our_bc_clustering.hpp"
 #elif defined(__clang__) && BOOST_VERSION >= 107300 && BOOST_VERSION <= 107400
 # include "our_bc_clustering_1_73.hpp"
