diff --git a/tools/build/cmake_templates/Dependency.cmake b/tools/build/cmake_templates/Dependency.cmake
index 3df09ad639..dccee68fb1 100644
--- a/tools/build/cmake_templates/Dependency.cmake
+++ b/tools/build/cmake_templates/Dependency.cmake
@@ -22,12 +22,21 @@ find_path("%(PKGNAME)s_INCLUDE_DIR"
 )
 
 # Finally the library itself
+# On Windows CMake does not search the 'lib' prefix, so manually add that;
+# otherwise it will fail to find, e.g. libprotobuf.lib
 foreach(lib %(libraries)s)
-find_library("${lib}_LIBRARY"
-  NAMES ${lib}
-  PATHS ${%(PKGNAME)s_PKGCONF_LIBRARY_DIRS}
-)
-list(APPEND %(PKGNAME)s_LIBRARY "${${lib}_LIBRARY}")
+  if(WIN32)
+    find_library("${lib}_LIBRARY"
+      NAMES ${lib} lib${lib}
+      PATHS ${%(PKGNAME)s_PKGCONF_LIBRARY_DIRS}
+    )
+  else()
+    find_library("${lib}_LIBRARY"
+      NAMES ${lib}
+      PATHS ${%(PKGNAME)s_PKGCONF_LIBRARY_DIRS}
+    )
+  endif()
+  list(APPEND %(PKGNAME)s_LIBRARY "${${lib}_LIBRARY}")
 endforeach(lib)
 
 
