From c5f371adcd47a97a63e7edf3a5c92aee17ed5800 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Wed, 25 Nov 2020 21:29:31 +0100
Subject: [PATCH] Fix time

---
 chronic.py | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/chronic.py b/chronic.py
index 04d8eb2..383e38d 100644
--- a/chronic.py
+++ b/chronic.py
@@ -40,12 +40,7 @@
 
 
 _local = threading.local()
-if sys.platform == "win32":
-    # On Windows, the best timer is time.clock()
-    _clock = systime.clock
-else:
-    # On most other platforms the best timer is time.time()
-    _clock = systime.time
+_clock = systime.time
 post_timing = Signal(name='post timing')
 
 
