--- orocos_kdl/src/velocityprofile_traphalf.cpp
+++ orocos_kdl/src/velocityprofile_traphalf.cpp
@@ -42,6 +42,7 @@
 
 //#include "error.h"
 #include "velocityprofile_traphalf.hpp"
+#include <algorithm>
 
 namespace KDL {
 
--- orocos_kdl/src/CMakeLists.txt
+++ orocos_kdl/src/CMakeLists.txt
@@ -133,6 +133,7 @@
   EXPORT OrocosKDLTargets
   ARCHIVE DESTINATION lib${LIB_SUFFIX}
   LIBRARY DESTINATION lib${LIB_SUFFIX}
+  RUNTIME DESTINATION bin
   PUBLIC_HEADER DESTINATION include/kdl
 )
 
