diff --git a/setup.py b/setup.py
index c5f09f4..e0a2a92 100644
--- a/setup.py
+++ b/setup.py
@@ -28,8 +28,9 @@ for line in read("ecmwflibs/__init__.py").split("\n"):
 
 assert version
 
-libdir = os.path.realpath("install/lib")
-incdir = os.path.realpath("install/include")
+prefix = os.path.join(sys.prefix, "Library") if os.name == "nt" else sys.prefix
+libdir = os.path.join(prefix, "lib")
+incdir = os.path.join(prefix, "include")
 libs = ["eccodes", "MagPlus"]
 
 
