diff --git a/elf/data.hh b/elf/data.hh
index 8085577..d7e9fb4 100644
fixes issue #44 of libelfin - build failure with LLVM 10.
--- a/elf/data.hh
+++ b/elf/data.hh
@@ -560,7 +560,7 @@ struct Sym<Elf64, Order>
                 return (stb)(info >> 4);
         }
 
-        void set_binding(stb v) const
+        void set_binding(stb v)
         {
                 info = (info & 0xF) | ((unsigned char)v << 4);
         }
