From 28e707020cd82a9b756efdc3b527113054169b97 Mon Sep 17 00:00:00 2001
From: Mattias Ellert <mattias.ellert@physics.uu.se>
Date: Tue, 31 Jan 2017 14:18:53 +0100
Subject: [PATCH] Fix for GCC 7

---
 src/server/vomsd.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/server/vomsd.cc b/src/server/vomsd.cc
index 700c5f1..4f4d1f7 100644
--- a/src/server/vomsd.cc
+++ b/src/server/vomsd.cc
@@ -1451,7 +1451,7 @@ static bool determine_group_and_role(std::string command, char *comm, char **gro
     case 'B':
       *role = strchr(string, ':');
       if (*role) {
-        (*role) = '\0';
+        (**role) = '\0';
         (*role)++;
       }
       break;
