From 98ff774ac7f221cfdd657dcc12875469b9342ba8 Mon Sep 17 00:00:00 2001
From: Christopher Burr <christopher.burr@cern.ch>
Date: Wed, 20 Feb 2019 07:37:04 +0000
Subject: [PATCH] HACK: Add NI_MAXHOST and NI_MAXSERV for macOS

---
 src/socklib/ipv6sock.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/socklib/ipv6sock.cc b/src/socklib/ipv6sock.cc
index bac90a5..a05fa44 100644
--- a/src/socklib/ipv6sock.cc
+++ b/src/socklib/ipv6sock.cc
@@ -40,6 +40,13 @@ extern "C" {
 #include "log.h"
 }
 
+#ifndef NI_MAXHOST
+#define NI_MAXHOST  1025
+#endif
+#ifndef NI_MAXSERV
+#define NI_MAXSERV  32
+#endif
+
 static void logconnection(struct sockaddr *client, void *logh)
 {
 
-- 
2.20.0

