From 0953e776f16142915ad3fc05fc6ffe61faee1363 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Tue, 8 Feb 2022 11:28:44 +0000
Subject: [PATCH] Header tests only

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 80b136b..56da08c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@ endif()
 
 ## (5) Options
 option(CODE_COVERAGE "Set ON to add code coverage compile options" OFF)
-option(SDSL_HEADER_TEST "Set ON to run header tests only" OFF)
+option(SDSL_HEADER_TEST "Set ON to run header tests only" ON)
 option(GENERATE_DOC "Set ON to genrate doxygen API reference in build/doc directory" OFF)
 option(USE_LIBCPP "Use the LLVM libc++ instead of GCC libstdc++ on OS X" ON)
 
-- 
2.28.0

