From 726e5594e79ed76a1a815d3ce041b18f1b09d37d Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Tue, 8 Feb 2022 16:41:52 +0000
Subject: [PATCH] Pass CMAKE_ARGS to call

---
 install.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/install.sh b/install.sh
index 3a14b84..f045a08 100755
--- a/install.sh
+++ b/install.sh
@@ -36,7 +36,7 @@ if [ $? != 0 ]; then
 	exit 1
 fi
 
-cmake -DCMAKE_INSTALL_PREFIX="${SDSL_INSTALL_PREFIX}" .. # run cmake 
+cmake ${CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX="${SDSL_INSTALL_PREFIX}" .. # run cmake 
 if [ $? != 0 ]; then
 	echo "ERROR: CMake build failed."
 	exit 1
-- 
2.28.0

