From ede08950746fb85c10b8c5da026bade07c72f5d9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Sun, 14 Feb 2021 21:30:34 +0100
Subject: [PATCH] setup patch

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 30a3252..9393907 100644
--- a/setup.py
+++ b/setup.py
@@ -26,8 +26,7 @@
     'numpy',
     'jax>=0.1.73',
     'jaxlib>=0.1.52',
-    'dm-haiku>=0.0.2',
-    'dataclasses'
+    'dm-haiku>=0.0.2'
 ]
 
 setuptools.setup(
