--- imake.c.orig	2017-02-06 15:34:47.199474238 -0500
+++ imake.c	2017-02-06 15:35:07.663653246 -0500
@@ -160,7 +160,7 @@
 #include <string.h>
 #include <ctype.h>
 #ifdef WIN32
-# include "Xw32defs.h"
+# include <X11/Xw32defs.h>
 #endif
 #include <sys/types.h>
 #include <fcntl.h>
--- imakemdep.h.orig	2017-02-06 15:44:08.735302953 -0500
+++ imakemdep.h	2017-02-06 15:44:11.710327597 -0500
@@ -1422,7 +1422,7 @@
 # endif /* MAKEDEPEND */
 
 # ifndef MAKEDEPEND
-#  if  defined (CROSSCOMPILE_CPP)
+#  if !defined(CROSSCOMPILE) || defined (CROSSCOMPILE_CPP)
 #   ifdef USE_CC_E
 boolean crosscompile_use_cc_e = TRUE;
 #    ifdef DEFAULT_CC
