--- a/src/Infrastructure/src/Emulator/Cores/tlib/tcg/host-utils.h	2025-01-30 17:49:53.632175202 -0600
+++ a/src/Infrastructure/src/Emulator/Cores/tlib/tcg/host-utils.h	2025-01-30 17:49:46.140225383 -0600
@@ -83,2 +83,3 @@
 #if defined(TCG_TARGET_I386) && HOST_LONG_BITS == 64
+#if defined(__x86_64__) || defined(_M_X64)
 #define __HAVE_FAST_MULU64__
@@ -97,2 +98,20 @@
 }
+#elif defined(__aarch64__)
+#define __HAVE_FAST_MULU64__
+static inline void mulu64(uint64_t *plow, uint64_t *phigh, uint64_t a, uint64_t b)
+{
+    __asm__ ("umulh %0, %1, %2"
+             : "=r" (*phigh)
+             : "r" (a), "r" (b));
+    *plow = a * b;
+}
+#define __HAVE_FAST_MULS64__
+static inline void muls64(uint64_t *plow, uint64_t *phigh, int64_t a, int64_t b)
+{
+    __asm__ ("smulh %0, %1, %2"
+             : "=r" (*phigh)
+             : "r" (a), "r" (b));
+    *plow = a * b;
+}
+#endif
 #else
