diff --git a/contrib/Makefile b/contrib/Makefile
index 7ed90d4..fed4d12 100644
--- a/contrib/Makefile
+++ b/contrib/Makefile
@@ -8,16 +8,11 @@ SUBDIRS = \
 		adminpack	\
 		auth_delay	\
 		auto_explain	\
-		btree_gin	\
-		btree_gist	\
 		chkpass		\
-		citext		\
 		cube		\
-		dblink		\
 		dict_int	\
 		dict_xsyn	\
 		earthdistance	\
-		file_fdw	\
 		fuzzystrmatch	\
 		hstore		\
 		intagg		\
@@ -33,22 +28,16 @@ SUBDIRS = \
 		pg_prewarm	\
 		pg_standby	\
 		pg_stat_statements \
-		pg_trgm		\
 		pgcrypto	\
 		pgrowlocks	\
 		pgstattuple	\
 		pgxc_clean	\
 		pgxc_ctl	\
 		pgxc_monitor \
-		postgres_fdw	\
 		seg		\
 		spi		\
-		tablefunc	\
 		tcn		\
-		test_decoding	\
-		tsm_system_rows \
 		tsm_system_time \
-		tsearch2	\
 		unaccent	\
 		vacuumlo	\
 		stormstats
