diff --git a/Cargo.toml b/Cargo.toml
index 28babc9..8bc977e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -8,7 +8,7 @@ name = "polars_strsim"
 crate-type = ["cdylib"]
 
 [dependencies]
-polars = { version = "0.43.1", default-features = false }
+polars = { version = "0.46.0", default-features = false }
 pyo3 = { version = "0.22.3", features = ["extension-module", "abi3-py38"] }
 pyo3-polars = { version = "0.17.0", features = ["derive"] }
 serde = { version = "1", features = ["derive"] }
diff --git a/pyproject.toml b/pyproject.toml
index 56fab40..89c05a0 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,7 +5,7 @@ build-backend = "maturin"
 [project]
 version = "0.2.3"
 name = "polars-strsim"
-dependencies = ["polars>=1.0,<2.0"]
+dependencies = ["polars>=1.2,<2.0"]
 requires-python = ">=3.8"
 classifiers = [
   "Development Status :: 4 - Beta",
