diff -uN scripts/globus-makefile-header.in scripts.new/globus-makefile-header.in
--- scripts/globus-makefile-header.in	2019-02-26 14:54:28.000000000 +0000
+++ scripts/globus-makefile-header.in	2019-09-25 17:26:28.000000000 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl

 #
 # Copyright 1999-2006 University of Chicago
diff -uN scripts/globus-vararg-enums-doxygen-filter.pl scripts.new/globus-vararg-enums-doxygen-filter.pl
--- scripts/globus-vararg-enums-doxygen-filter.pl	2019-02-26 14:54:28.000000000 +0000
+++ scripts/globus-vararg-enums-doxygen-filter.pl	2019-09-25 17:26:34.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl

 #
 # Copyright 1999-2006 University of Chicago
