--- setup.py    2023-12-21 13:00:25.489903713 -0600
+++ setup.py    2023-12-21 12:57:24.286924157 -0600
@@ -222,6 +222,10 @@
             self.define = _build_clib.build_flags['define']

         if os.name == 'nt' or sys.platform == 'win32':
+            print("Building for Windows")
+            print("\tself.libraries: ", self.libraries)
+            print("\t          vars: ", vars(self))
+            print("\t    extensions: ", vars(extensions[0]))
             # Apparently, the linker on Windows interprets -lxxx as xxx.lib, not libxxx.lib
             self.libraries = [lib[1:] for lib in self.libraries if 'secp256k1' in lib]

