--- a/setup_support.py	2023-12-27 17:46:39.000000000 -0600
+++ b/setup_support.py	2023-12-27 17:53:54.607456556 -0600
@@ -62,9 +62,9 @@
     from cffi import FFI
 
     try:
-        subprocess.check_output(['pkg-config', '--exists', 'libsecp256k1'], env=env)  # noqa S603
+        subprocess.check_output(['pkg-config', '--exists', 'libsecp256k1'])  # noqa S603
 
-        includes = subprocess.check_output(['pkg-config', '--cflags-only-I', 'libsecp256k1'], env=env)  # noqa S603
+        includes = subprocess.check_output(['pkg-config', '--cflags-only-I', 'libsecp256k1'])  # noqa S603
         includes = includes.strip().decode('utf-8')
 
         return os.path.exists(os.path.join(includes[2:], 'secp256k1_ecdh.h'))
@@ -75,7 +75,7 @@
                 with suppress(OSError):
                     FFI().dlopen(path)
                     return True
-        # We couldn't locate libsecp256k1 so we'll use the bundled one
+        # We couldn't locate libsecp256k1, so we'll use the bundled one
         return False
 
 
