--- setup.py    2023-12-21 13:00:25.489903713 -0600
+++ setup.py    2023-12-21 12:57:24.286924157 -0600
@@ -223,7 +223,7 @@

         if os.name == 'nt' or sys.platform == 'win32':
             # Apparently, the linker on Windows interprets -lxxx as xxx.lib, not libxxx.lib
-            self.libraries = [lib[1:] for lib in self.libraries if 'secp256k1' in lib]
+            self.extensions[0].libraries = [lib[1:] for lib in self.libraries if 'secp256k1' in lib]

         return _build_ext.run(self)

