diff --git a/setup.py b/setup.py
--- a/setup.py	(revision 3b76f51dd985df76383395c488071d70b2a88e7c)
+++ b/setup.py	(date 1691946069150)
@@ -51,7 +51,7 @@
 def download_library(command):
     if command.dry_run:
         return
-    libdir = absolute('libsecp256k1')
+    libdir = 'libsecp256k1'
     if os.path.exists(os.path.join(libdir, 'autogen.sh')):
         # Library already downloaded
         return
@@ -116,8 +116,8 @@
         download_library(self)
 
         return [
-            absolute(os.path.join(root, filename))
-            for root, _, filenames in os.walk(absolute('libsecp256k1'))
+            os.path.join(root, filename)
+            for root, _, filenames in os.walk('libsecp256k1')
             for filename in filenames
         ]
 
@@ -143,11 +143,11 @@
             if e.errno != errno.EEXIST:
                 raise
 
-        if not os.path.exists(absolute('libsecp256k1')):
+        if not os.path.exists('libsecp256k1'):
             # library needs to be downloaded
             self.get_source_files()
 
-        if not os.path.exists(absolute('libsecp256k1/configure')):
+        if not os.path.exists('libsecp256k1/configure'):
             # configure script hasn't been generated yet
             autogen = absolute('libsecp256k1/autogen.sh')
             os.chmod(absolute(autogen), 0o755)
