diff --git a/setup.py b/setup.py
--- a/setup.py	(revision a3c2fdb643d684090ed33b79de4703b3d49924f8)
+++ b/setup.py	(date 1692282373216)
@@ -38,7 +38,7 @@
 UPSTREAM_REF = os.getenv('COINCURVE_UPSTREAM_REF') or 'ddf2b2910eb19032f8dd657c66735115ae24bfba'
 
 LIB_TARBALL_URL = f'https://github.com/bitcoin-core/secp256k1/archive/{UPSTREAM_REF}.tar.gz'
-
+#LIB_TARBALL_URL = "https://github.com/bitcoin-core/secp256k1/archive/refs/tags/v0.3.2.tar.gz"
 
 # We require setuptools >= 3.3
 if [int(i) for i in setuptools_version.split('.', 2)[:2]] < [3, 3]:
@@ -149,9 +149,22 @@
 
         if not os.path.exists(absolute('libsecp256k1/configure')):
             # configure script hasn't been generated yet
-            autogen = absolute('libsecp256k1/autogen.sh')
-            os.chmod(absolute(autogen), 0o755)
-            subprocess.check_call([autogen], cwd=absolute('libsecp256k1'))
+            # autogenautogen = absolute('libsecp256k1/autogen.sh')
+            # os.chmod(absolute(autogen), 0o755)
+            # subprocess.check_call([autogen], cwd=absolute('libsecp256k1'))
+            # subprocess.check_call(["autoreconf", "-i", "-f", "--warnings=all"],
+            #                       cwd=absolute('libsecp256k1'),
+            #                       shell=False)
+            if os.environ['PREFIX'] is not None and os.path.isfile(
+                autoreconf := os.path.join(os.environ['PREFIX'], 'Library', 'usr', 'bin', 'autoreconf.exe')):
+                subprocess.check_call([autoreconf, "-if --warnings=all"],
+                                      cwd=absolute('libsecp256k1'),
+                                      shell=True)
+            else:
+                subprocess.check_call(["autoreconf", "-if --warnings=all"],
+                                      cwd=absolute('libsecp256k1'),
+                                      shell=True)
+
 
         for filename in [
             'libsecp256k1/configure',
@@ -187,7 +200,6 @@
             '--enable-module-ecdh',
             '--enable-benchmark=no',
             '--enable-tests=no',
-            '--enable-openssl-tests=no',
             '--enable-exhaustive-tests=no',
         ]
         if 'COINCURVE_CROSS_HOST' in os.environ:
@@ -274,7 +286,7 @@
     license='MIT OR Apache-2.0',
 
     python_requires='>=3.7',
-    install_requires=['asn1crypto', 'cffi>=1.3.0'],
+    # install_requires=['asn1crypto', 'cffi>=1.3.0'],
 
     packages=find_packages(exclude=('_cffi_build', '_cffi_build.*', 'libsecp256k1', 'tests')),
     package_data=package_data,
