diff --git a/setup.py b/setup.py
--- a/setup.py	(revision a3c2fdb643d684090ed33b79de4703b3d49924f8)
+++ b/setup.py	(revision 16e2c5d0745348e7b13646a1feaf2e95ed843edd)
@@ -149,9 +149,29 @@
 
         if not os.path.exists(absolute('libsecp256k1/configure')):
             # configure script hasn't been generated yet
-            autogen = absolute('libsecp256k1/autogen.sh')
-            os.chmod(absolute(autogen), 0o755)
-            subprocess.check_call([autogen], cwd=absolute('libsecp256k1'))
+            bash = None
+            for path in ["LIBRARY_BIN", "PREFIX", "BUILD_PREFIX", "SP_DIR"]:
+                if os.environ.get(path, None) is None:
+                    continue
+
+                if bash is not None:
+                    break
+
+                for root, _, filenames in os.walk(os.environ.get(path)):
+                    for filename in filenames:
+                        if filename in ["bash", "bash.exe"]:
+                            bash = os.path.join(root, filename)
+                            break
+
+            if bash is None:
+                bash = "bash"
+
+            subprocess.check_call([f"{bash} -c 'autoreconf -if --warnings=all'"],
+                                  cwd=absolute('libsecp256k1'),
+                                  shell=True)
+            #autogen = absolute('libsecp256k1/autogen.sh')
+            #os.chmod(absolute(autogen), 0o755)
+            #subprocess.check_call([autogen], cwd=absolute('libsecp256k1'))
 
         for filename in [
             'libsecp256k1/configure',
