From 47bda7dd3010b80ab0b9a62ad237ac15e5565b8c Mon Sep 17 00:00:00 2001
From: kiwi0fruit <peter.zagubisalo@gmail.com>
Date: Sun, 10 Feb 2019 13:17:03 +0700
Subject: [PATCH] c

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index ed3d20c..2d1385e 100644
--- a/setup.py
+++ b/setup.py
@@ -113,7 +113,6 @@ def excract_tar_and_move_files(url, hash, move, **kwargs):
 setup(
     name='py-pandoc',
     version=version,
-    cmdclass={'install': PostInstallCommand},
     python_requires='>=3.6',
     description='Pandoc in pip and conda',
     url='https://github.com/kiwi0fruit/py-pandoc',
-- 
2.16.1.windows.1

