diff --git a/autodock/autocomm.h b/autodock/autocomm.h
index 298b6d6..ae641b1 100644
--- a/autodock/autocomm.h
+++ b/autodock/autocomm.h
@@ -77,11 +77,14 @@ const unsigned int MAX_LINES=256;    /* Number of lines in parameter file
 #define PATH_MAX     FILENAME_MAX
 #endif
 
+#ifdef HVD_DEBUG
 #ifdef USE_XCODE
 const /* not unsigned */ int LINE_LEN=140;    /* Line length in characters                      */
 #else
 const /* not unsigned */ int LINE_LEN=256;    /* Line length in characters                      */
 #endif 
+#endif
+const /* not unsigned */ int LINE_LEN=1100;    /* Line length in characters                      */
 
 // Note: MAX_GRID_PTS is a sanity check, and could be enlarged at will   M Pique 2012
 const /* not unsigned */ int MAX_GRID_PTS=1025;	/* Maximum number of grid points in 1 dimension */
@@ -90,7 +93,7 @@ const Real EINTCLAMP=100000.; /* Clamp pairwise internal energies (kcal/mol )  *
 
 #define MAX_MAPS_PAD 0       // Use this to pad MAX_MAPS to a power of 2, for presumably-faster memory access
 #define NUM_NON_VDW_MAPS 2   // Number of electrostatic and desolvation maps
-#define MAX_ATOM_TYPES (16 - NUM_NON_VDW_MAPS)    /* Maximum number of atom types set to keep MAX_MAPS a power of 2 */
+#define MAX_ATOM_TYPES (64 - NUM_NON_VDW_MAPS)    /* Maximum number of atom types set to keep MAX_MAPS a power of 2 */
 #define MAX_MAPS (MAX_ATOM_TYPES + NUM_NON_VDW_MAPS + MAX_MAPS_PAD) /* Maximum number of energy maps        */
                             /* 0,1,2,... are for atomic interactions          */
                             /* last two are for electrostatics and desolvation */
