--- autogrid/configure-universalDarwin	2012-11-01 17:18:04.000000000 -0500
+++ autogrid/configure-universalDarwin.new	2018-08-15 16:15:36.000000000 -0500
@@ -45,9 +45,9 @@
 	sdkver=7
 	sdkver=0 # hack for TSRI since our only 10.7 machine has no SDK dir
 	;;
-  *12)
+  *1[2-9])
 	# Mountain Lion (Intel 32/64-bit)
-	archlist="-arch i386 -arch x86_64"
+	archlist="-arch x86_64"
 	sdkver=0 # hack for TSRI since our only 10.8 machine has no SDK dir
 	;;
   *)
