diff --git a/source/pcraster_aguila/ag_Camera.cc b/source/pcraster_aguila/ag_Camera.cc
index 0c5e06e..7da15d1 100644
--- a/source/pcraster_aguila/ag_Camera.cc
+++ b/source/pcraster_aguila/ag_Camera.cc
@@ -1,6 +1,10 @@
 #include "ag_Camera.h"
 #include <cassert>
-#include <GL/glu.h>
+#if defined(__APPLE__)
+# include <OpenGL/glu.h>
+#elif
+# include <GL/glu.h>
+#endif
 
 
 
diff --git a/source/pcraster_aguila/ag_SceneView.cc b/source/pcraster_aguila/ag_SceneView.cc
index 7bcc145..a6807b7 100644
--- a/source/pcraster_aguila/ag_SceneView.cc
+++ b/source/pcraster_aguila/ag_SceneView.cc
@@ -1,7 +1,11 @@
 #include "ag_SceneView.h"
 #include <cmath>
 #include <vector>
-#include <GL/glu.h>
+#if defined(__APPLE__)
+# include <OpenGL/glu.h>
+#else
+# include <GL/glu.h>
+#endif
 #include <boost/format.hpp>
 #include <QKeyEvent>
 #include <QPoint>
