diff --git a/source/pcraster_python/CMakeLists.txt b/source/pcraster_python/CMakeLists.txt
index 6bb2a9f..4ddd417 100644
--- a/source/pcraster_python/CMakeLists.txt
+++ b/source/pcraster_python/CMakeLists.txt
@@ -152,8 +152,8 @@ add_custom_target(
 #----------------------------------------------------------
 install(
     TARGETS _pcraster
-    LIBRARY DESTINATION python/pcraster
-    RUNTIME DESTINATION python/pcraster
+    LIBRARY DESTINATION lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages/pcraster
+    RUNTIME DESTINATION lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages/pcraster
 )
 
 install(
@@ -170,7 +170,7 @@ install(
         ${PROJECT_SOURCE_DIR}/environment/script/shellscript.py
         ${PROJECT_SOURCE_DIR}/environment/script/utils.py
     DESTINATION
-        python/pcraster
+        lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages/pcraster
 )
 
 #----------------------------------------------------------
