diff --git a/odyl/Makefile b/odyl/Makefile
index 8c8f7307..62031033 100644
--- a/odyl/Makefile
+++ b/odyl/Makefile
@@ -32,7 +32,7 @@ odyl_main.cmx: odyl_main.ml
 odyl_config.ml:
 	echo "let standard_library =" > odyl_config.ml
 	echo "  try Sys.getenv \"CAMLP5LIB\" with" >> odyl_config.ml
-	echo "    Not_found -> \"$(LIBDIR)/$(CAMLP5N)\"" | \
+	echo "    Not_found -> (Sys.getenv \"OCAMLLIB\") ^ \"/camlp5\"" | \
 	sed -e 's|\\|/|g' >> odyl_config.ml
 
 clean::
