From 7723f10811a80258983132bf8b372475cb3401a5 Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Fri, 9 Jan 2026 14:03:36 +0100
Subject: [PATCH] fix SofaGLFW cmake config in with missing gflw

---
 SofaGLFW/SofaGLFWConfig.cmake.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SofaGLFW/SofaGLFWConfig.cmake.in b/SofaGLFW/SofaGLFWConfig.cmake.in
index 009e4de..b84ca58 100644
--- a/SofaGLFW/SofaGLFWConfig.cmake.in
+++ b/SofaGLFW/SofaGLFWConfig.cmake.in
@@ -8,6 +8,9 @@ find_package(Sofa.Simulation.Graph REQUIRED)
 find_package(Sofa.GL REQUIRED)
 find_package(Sofa.Component.Visual REQUIRED)
 find_package(Sofa.GUI.Common QUIET)
+if(NOT TARGET glfw)
+    sofa_find_package(glfw3 QUIET REQUIRED)
+endif()
 
 if(NOT TARGET @PROJECT_NAME@)
     include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
-- 
2.43.0

