From f44785b6c594e80b126e8727fd9c22bc671337c6 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 6 Aug 2022 22:47:41 -0600
Subject: [PATCH] Update requirements.txt

---
 requirements.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index ae91a04..9b57848 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -2,4 +2,4 @@ pillow>=2.2
 reportlab>=2.7
 watchdog>=0.6.0
 pypdf2>=1.23
-evernote
+evernote3
