From 7d663797f331e145b4c2eb4ed6c9af30403d5876 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 6 Aug 2022 23:07:13 -0600
Subject: [PATCH] Fix Python 3

---
 pypdfocr/pypdfocr.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pypdfocr/pypdfocr.py b/pypdfocr/pypdfocr.py
index 7ee7e9e..2341bbb 100644
--- a/pypdfocr/pypdfocr.py
+++ b/pypdfocr/pypdfocr.py
@@ -21,7 +21,7 @@
 import itertools
 from functools import wraps
 
-from version import __version__
+from pypdfocr.version import __version__
 from PIL import Image
 import yaml
 
@@ -467,7 +467,7 @@ def go(self, argv):
                 except KeyboardInterrupt:
                     break
                 except Exception as e:
-                    print traceback.print_exc(e)
+                    print(traceback.print_exc(e))
                     py_watcher.stop()
                     
         else:
