From 251724bc3acc8855809a3e79aae236e006af5e8b Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 6 Aug 2022 22:56:35 -0600
Subject: [PATCH] Fix Python 3

---
 pypdfocr/pypdfocr.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pypdfocr/pypdfocr.py b/pypdfocr/pypdfocr.py
index 7ee7e9e..bb217d2 100644
--- a/pypdfocr/pypdfocr.py
+++ b/pypdfocr/pypdfocr.py
@@ -467,7 +467,7 @@ def go(self, argv):
                 except KeyboardInterrupt:
                     break
                 except Exception as e:
-                    print traceback.print_exc(e)
+                    print(traceback.print_exc(e))
                     py_watcher.stop()
                     
         else:
