From a1bec44b5475be9544f5b3807dbf9c32d9e48621 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 12 Nov 2021 16:42:07 +1100
Subject: [PATCH 3/3] make get_path_to_datafile work outside of source tree

---
 tensorflow_addons/utils/resource_loader.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tensorflow_addons/utils/resource_loader.py b/tensorflow_addons/utils/resource_loader.py
index 507b0d4..6ee10a4 100644
--- a/tensorflow_addons/utils/resource_loader.py
+++ b/tensorflow_addons/utils/resource_loader.py
@@ -46,6 +46,8 @@ def get_path_to_datafile(path, is_so=False):
         bazel_bin_dir = os.path.join(os.path.dirname(root_dir), "bazel-bin")
         if os.path.isdir(bazel_bin_dir):
             root_dir = os.path.join(bazel_bin_dir, "tensorflow_addons")
+    if "SRC_DIR" in os.environ:
+        root_dir = os.path.join(os.environ["SRC_DIR"], "tensorflow_addons")
     return os.path.join(root_dir, path.replace("/", os.sep))
 
 
-- 
2.32.0.windows.2

