--- test/test_server.py	2021-11-20 13:52:33.127670345 -0600
+++ test/test_server.py	2021-11-20 13:53:11.759793896 -0600
@@ -6,11 +6,11 @@
 except:
     from io import StringIO
 root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
-sys.path.insert(0, root_dir)
+
 from fortls.jsonrpc import write_rpc_request, write_rpc_notification, \
     read_rpc_messages, path_to_uri
 
-run_command = os.path.join(root_dir, "fortls.py --incrmental_sync --use_signature_help")
+run_command = "fortls --incrmental_sync --use_signature_help"
 test_dir = os.path.join(root_dir, "test", "test_source")
 
 
@@ -530,4 +530,4 @@
     test_def()
     test_refs()
     test_hover()
-    test_docs()
\ No newline at end of file
+    test_docs()
