diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2392d5..1745fd2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@
 # OpenMM Plumed Plugin
 #----------------------------------------------------
 
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.0)
 
 # We need to know where OpenMM is installed so we can access the headers and libraries.
 SET(OPENMM_DIR "/usr/local/openmm" CACHE PATH "Where OpenMM is installed")
@@ -20,7 +20,7 @@ SET (CMAKE_CXX_STANDARD 11)
 
 # Set flags for linking on mac
 IF(APPLE)
-    SET (CMAKE_INSTALL_NAME_DIR "@rpath")
+    # SET (CMAKE_INSTALL_NAME_DIR "@rpath")
     SET(EXTRA_COMPILE_FLAGS "-msse2 -stdlib=libc++")
 ENDIF(APPLE)
 
