diff --git setup.py setup.py
index 70a66a2..812581a 100644
--- setup.py
+++ setup.py
@@ -28,10 +28,6 @@ try:
         zip_safe = False,
         packages = find_packages(),
         entry_points = {
-        'console_scripts': [
-            'nosetests = nose:run_exit',
-            'nosetests%s = nose:run_exit' % py_vers_tag,
-            ],
         'distutils.commands': [
             ' nosetests = nose.commands:nosetests',
             ],
