diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1407080..e489396 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -49,11 +49,7 @@ target_include_directories(ccd PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 
 if(NOT WIN32)
-  find_library(LIBM_LIBRARY NAMES m)
-  if(NOT LIBM_LIBRARY)
-    message(FATAL_ERROR "Could NOT find required library LibM")
-  endif()
-  target_link_libraries(ccd "${LIBM_LIBRARY}")
+  target_link_libraries(ccd m)
   if(BUILD_SHARED_LIBS)
     set(CCD_PKGCONFIG_EXTRA_LIBS -lm PARENT_SCOPE)
   endif()
