From 2bee6f81208452a805674d5349e54041dfb826e9 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 20 Aug 2021 11:33:56 -0600
Subject: [PATCH] Update setup.py

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 1c45828..fe6de2b 100644
--- a/setup.py
+++ b/setup.py
@@ -49,9 +49,8 @@
                       'pandas',
                       'pymatgen>=2022.0.0',
                       'PyYAML', 
-                      'coverage==4.3.4',
+                      'coverage',
                       'codeclimate-test-reporter',
-                      'fortranformat',
                       'scipy>=1.4.1', 
                       'tqdm',
                       'lxml'],
