diff --git src/graph/inference/blockmodel/graph_blockmodel.hh src/graph/inference/blockmodel/graph_blockmodel.hh
index 3e009edd..e8ae55c7 100644
--- src/graph/inference/blockmodel/graph_blockmodel.hh
+++ src/graph/inference/blockmodel/graph_blockmodel.hh
@@ -22,6 +22,10 @@
 
 #include <vector>
 
+#ifdef __clang__
+#include <boost/algorithm/minmax_element.hpp>
+#endif
+
 #include "../support/graph_state.hh"
 #include "graph_blockmodel_util.hh"
 
@@ -2356,10 +2360,18 @@ public:
                 E += _eweight[e];
             size_t B = num_vertices(_bg);
 
-            auto vi = std::max_element(vertices(_g).first, vertices(_g).second,
-                                       [&](auto u, auto v)
-                                       { return (this->_pclabel[u] <
-                                                 this->_pclabel[v]); });
+// Clang 8.0 fails to correctly recognize these as ForwardIterators,
+// triggering a static_assert in std::max_element(). See #576.
+#ifndef __clang__
+            auto vi = std::max_element(
+#else
+            auto vi = boost::first_max_element(
+#endif
+                vertices(_g).first, vertices(_g).second,
+                [&](auto u, auto v)
+                { return (this->_pclabel[u] <
+                          this->_pclabel[v]); });
+
             size_t C = _pclabel[*vi] + 1;
 
             vector<vector<size_t>> vcs(C);
