diff --git configure.ac configure.ac
index be20296a..58a5235b 100644
--- configure.ac
+++ configure.ac
@@ -145,10 +145,13 @@ AC_SUBST(MOD_DIR)
 
 dnl boost
 AX_BOOST_BASE([[1.55.0]])
-AX_BOOST_PYTHON
+#AX_BOOST_PYTHON
+AC_DEFINE([HAVE_BOOST_PYTHON], [1], "Boost Python is available")
 if test "$BOOST_PYTHON_LIB" = ""; then
    AC_MSG_ERROR([No usable boost::python found])
 fi
+
+
 AX_BOOST_IOSTREAMS
 if test "$BOOST_IOSTREAMS_LIB" = ""; then
    AC_MSG_ERROR([No usable boost::iostreams found])
