From d6519485d18977232efb7d8c960c55b812dc9635 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Mon, 22 Aug 2022 23:21:38 +0530
Subject: [PATCH] Install library & binary

---
 src/CMakeLists.txt  | 2 ++
 tool/CMakeLists.txt | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 346f4c1..9b8e0df 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -120,3 +120,5 @@ target_compile_options(cppast PRIVATE
                            # MSVC warnings
                            $<$<CXX_COMPILER_ID:MSVC>:
                            /W3>)
+
+install(TARGETS cppast)
diff --git a/tool/CMakeLists.txt b/tool/CMakeLists.txt
index f228e83..5d8ea51 100644
--- a/tool/CMakeLists.txt
+++ b/tool/CMakeLists.txt
@@ -5,3 +5,5 @@
 add_executable(cppast_tool main.cpp)
 target_link_libraries(cppast_tool PUBLIC cppast)
 set_target_properties(cppast_tool PROPERTIES CXX_STANDARD 11 OUTPUT_NAME cppast)
+
+install(TARGETS cppast_tool)
-- 
2.37.1

