--- a/libjaylink/meson.build
+++ b/libjaylink/meson.build
@@ -1 +1,10 @@
+if build_system == 'windows'
+  add_project_arguments('-DBUILDING_DLL', language: 'c')
+  win_args = {
+    'vs_module_defs': 'jaylink.def',
+  }
+else
+  win_args = {}
+endif
+
 sources = [
@@ -58,0 +68 @@
+  kwargs: win_args
@@ -60,0 +71,11 @@
+if build_system == 'windows'
+  libjaylink = shared_library(
+    'libjaylink',
+    sources,
+    dependencies: [libusb, ws2_32],
+    version: library_version_string,
+    include_directories: include_dirs,
+    install: true,
+    vs_module_defs: 'jaylink.def'
+  )
+endif
