--- a/libjaylink/meson.build
+++ b/libjaylink/meson.build
@@ -1 +1,12 @@
+if build_system == 'windows'
+  message('Adding BUILDING_DLL define for Windows')
+  add_project_arguments('-DBUILDING_DLL', language: 'c')
+  link_args = ['/DEF:..\libjaylink\jaylink.def']
+  win_args = {
+    'vs_module_defs': '..\libjaylink\jaylink.def',
+  }
+else
+  win_args = {}
+endif
+
 sources = [
@@ -58,0 +70,2 @@
+  kwargs: win_args,
+  link_args: link_args,
@@ -60,0 +74,12 @@
+if build_system == 'windows'
+  libjaylink = shared_library(
+    'libjaylink',
+    sources,
+    dependencies: [libusb, ws2_32],
+    version: library_version_string,
+    include_directories: include_dirs,
+    link_args: link_args,
+    install: true,
+    vs_module_defs: '..\libjaylink\jaylink.def'
+  )
+endif
