diff --git a/libjaylink/meson.build b/libjaylink/meson.build
index xxxxxxx..yyyyyyy 100644
--- a/libjaylink/meson.build
+++ b/libjaylink/meson.build
@@ -1 +1,10 @@
+if build_system == 'windows'
+  win_args = {
+    'vs_module_defs': 'jaylink.def',
+    'name_prefix': 'lib'
+  }
+else
+  win_args = {}
+endif
+
 sources = [
@@ -58,4 +67,17 @@
   install: true,
+  kwargs: win_args
 )

+if build_system == 'windows'
+  jaylink_no_prefix = shared_library(
+    'jaylink',
+    sources,
+    dependencies: [libusb, ws2_32],
+    version: library_version_string,
+    include_directories: include_dirs,
+    install: true,
+    vs_module_defs: 'jaylink.def'
+  )
+endif
+
 pkg = import('pkgconfig')
