--- a/libjaylink/discovery_usb.c
+++ b/libjaylink/discovery_usb.c
@@ -145,1 +145,1 @@
-	enum jaylink_usb_address usb_address;
+	enum jaylink_usb_address usb_address = 0;
--- a/libjaylink/transport_usb.c
+++ b/libjaylink/transport_usb.c
@@ -52,1 +52,1 @@
-	const struct libusb_interface_descriptor *desc;
+	const struct libusb_interface_descriptor *desc = NULL;
@@ -77,1 +77,1 @@
-		desc = &interface->altsetting[0];
+		const struct libusb_interface_descriptor *curr_desc = &interface->altsetting[0];
@@ -89,1 +89,1 @@
-		devh->interface_number = i;
+		devh->interface_number = i; desc = curr_desc;
@@ -93,1 +93,1 @@
-	if (!found_interface) {
+	if (!found_interface || !desc) {
