--- a/libjaylink/socket.c
+++ b/libjaylink/socket.c
@@ -67,1 +67,1 @@
-	ret = connect(sock, address, address_length);
+	ret = connect(sock, address, (int)address_length);
@@ -84,1 +84,1 @@
-	tv.tv_sec = timeout / 1000;
+	tv.tv_sec = (long)(timeout / 1000);
@@ -143,1 +143,1 @@
-	ret = bind(sock, address, length);
+	ret = bind(sock, address, (int)length);
@@ -174,1 +174,1 @@
-	ret = send(sock, buffer, *length, flags);
+	ret = send(sock, buffer, (int)*length, flags);
@@ -206,1 +206,1 @@
-	ret = recv(sock, buffer, *length, flags);
+	ret = recv(sock, buffer, (int)*length, flags);
@@ -243,1 +243,1 @@
-	ret = sendto(sock, buffer, *length, flags, address, address_length);
+	ret = sendto(sock, buffer, (int)*length, flags, address, (int)address_length);
@@ -287,1 +287,1 @@
-	tmp = *address_length;
+	tmp = (int)*address_length;
@@ -288,1 +288,1 @@
-	ret = recvfrom(sock, buffer, *length, flags, address, &tmp);
+	ret = recvfrom(sock, buffer, (int)*length, flags, address, &tmp);
@@ -342,1 +342,1 @@
-	if (!setsockopt(sock, level, option, value, length))
+	if (!setsockopt(sock, level, option, value, (int)length))
