diff --git a/fortran_kernel/tests/test_fortran_kernel.py b/fortran_kernel/tests/test_fortran_kernel.py
index 323ff97..decc959 100644
--- a/fortran_kernel/tests/test_fortran_kernel.py
+++ b/fortran_kernel/tests/test_fortran_kernel.py
@@ -8,7 +8,7 @@ class IRKernelTests(jkt.KernelTests):
 
     file_extension = ".f90"
 
-    code_hello_world = 'print *, "hello, world"'
+    #code_hello_world = 'print *, "hello, world"'
 
     #complete_code_samples = ['1', 'print *, "hello, world"',
     #        "integer :: i"]
