From 7a86dda5ebc2a14aca5b8a2fe76dbfd525dedb43 Mon Sep 17 00:00:00 2001
From: fhoehle <fhoehle@users.noreply.github.com>
Date: Fri, 16 Jul 2021 10:55:09 +0200
Subject: [PATCH 3/7] bug43364.phpt libxml2 update

---
 ext/dom/tests/bug43364.phpt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/dom/tests/bug43364.phpt b/ext/dom/tests/bug43364.phpt
index e4808f19..e540e1c8 100644
--- a/ext/dom/tests/bug43364.phpt
+++ b/ext/dom/tests/bug43364.phpt
@@ -39,4 +39,4 @@ $count = loopElements(array($doc->documentElement));
 var_dump($count);
 ?>
 --EXPECT--
-int(13)
+int(11)
-- 
2.30.2

