From e28744065060ec45d411d98522d86ad8d58a6c18 Mon Sep 17 00:00:00 2001
From: fhoehle <fhoehle@users.noreply.github.com>
Date: Fri, 16 Jul 2021 10:55:31 +0200
Subject: [PATCH 4/7] bug80268.phpt libxml2 update

---
 ext/dom/tests/bug80268.phpt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/ext/dom/tests/bug80268.phpt b/ext/dom/tests/bug80268.phpt
index 0fe50b85..e26a74e3 100644
--- a/ext/dom/tests/bug80268.phpt
+++ b/ext/dom/tests/bug80268.phpt
@@ -19,6 +19,9 @@ var_dump(strpos($html, '<p>foo</p>') !== false);
 <?php
 unlink(__DIR__ . '/80268.html');
 ?>
---EXPECT--
-bool(true)
-bool(true)
+--EXPECTF--
+Warning: DOMDocument::loadHTML(): Char 0x0 out of allowed range in Entity, line: 1 in %s/ext/dom/tests/bug80268.php on line %d
+bool(false)
+
+Warning: DOMDocument::loadHTMLFile(): Char 0x0 out of allowed range in %s/ext/dom/tests/80268.html, line: %d in %s/ext/dom/tests/bug80268.php on line %d
+bool(false)
-- 
2.30.2

