From 56f9dc3e588aa35ab82db451e02485dc4b81ed91 Mon Sep 17 00:00:00 2001
From: fhoehle <fhoehle@users.noreply.github.com>
Date: Fri, 16 Jul 2021 10:56:48 +0200
Subject: [PATCH 6/7] bug61367-read.phpt libxml2 update

---
 ext/libxml/tests/bug61367-read.phpt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/ext/libxml/tests/bug61367-read.phpt b/ext/libxml/tests/bug61367-read.phpt
index b4ecaa03..02df7e2a 100644
--- a/ext/libxml/tests/bug61367-read.phpt
+++ b/ext/libxml/tests/bug61367-read.phpt
@@ -54,7 +54,3 @@ int(4)
 bool(true)
 
 Warning: DOMDocument::loadXML(): I/O warning : failed to load external entity "file:///%s/test_bug_61367-read/bad" in %s on line %d
-
-Warning: DOMDocument::loadXML(): Failure to process entity file in Entity, line: 4 in %s on line %d
-
-Warning: DOMDocument::loadXML(): Entity 'file' not defined in Entity, line: 4 in %s on line %d
-- 
2.30.2

