--- ./ext/libxml/tests/bug61367-read.phpt.bck	2021-06-10 21:13:05.000000000 +0200
+++ ./ext/libxml/tests/bug61367-read.phpt	2021-06-10 21:13:05.000000000 +0200
@@ -54,7 +54,3 @@
 bool(true)
 
 Warning: DOMDocument::loadXML(): I/O warning : failed to load external entity "file:///%s/test_bug_61367-read/bad" in %s on line %d
-
-Warning: DOMDocument::loadXML(): Failure to process entity file in Entity, line: 4 in %s on line %d
-
-Warning: DOMDocument::loadXML(): Entity 'file' not defined in Entity, line: 4 in %s on line %d
