--- ./ext/dom/tests/bug80268.phpt.bck	2021-06-10 21:13:05.000000000 +0200
+++ ./ext/dom/tests/bug80268.phpt	2021-06-10 21:13:05.000000000 +0200
@@ -19,6 +19,9 @@
 <?php
 unlink(__DIR__ . '/80268.html');
 ?>
---EXPECT--
-bool(true)
-bool(true)
+--EXPECTF--
+Warning: DOMDocument::loadHTML(): Char 0x0 out of allowed range in Entity, line: 1 in %s/ext/dom/tests/bug80268.php on line %d
+bool(false)
+
+Warning: DOMDocument::loadHTMLFile(): Char 0x0 out of allowed range in %s/ext/dom/tests/80268.html, line: %d in %s/ext/dom/tests/bug80268.php on line %d
+bool(false)
