--- ./ext/dom/tests/DOMDocument_load_error1.phpt.bck	2021-06-10 21:13:05.000000000 +0200
+++ ./ext/dom/tests/DOMDocument_load_error1.phpt	2021-06-10 21:13:05.000000000 +0200
@@ -21,6 +21,6 @@
 --EXPECTF--
 Warning: DOMDocument::load%r(XML){0,1}%r(): Opening and ending tag mismatch: title line 5 and book %s
 
-Warning: DOMDocument::load%r(XML){0,1}%r(): %rexpected '>'|Opening and ending tag mismatch: book line 5 and books%r %s
+Warning: DOMDocument::load%r(XML){0,1}%r(): %rexpected '>'|Opening and ending tag mismatch: book line 4 and books%r %s
 
 Warning: DOMDocument::load%r(XML){0,1}%r(): %rPremature end of data in tag books|EndTag: '<\/' not found%r %s
