diff --git a/gcc/configure b/gcc/configure
--- a/gcc/configure
+++ b/gcc/configure
@@ -27924,8 +27924,12 @@ fi
     fi
  fi

- { $as_echo "$as_me:${as_lineno-$LINENO}: checking assembler for --gdwarf2 option" >&5
-$as_echo_n "checking assembler for --gdwarf2 option... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: defaulting assembler --gdwarf-2 option to yes" >&5
+$as_echo_n "defaulting assembler --gdwarf-2 option to yes" >&6; }
+gcc_cv_as_gdwarf2_flag=yes
+
+ { $as_echo "$as_me:${as_lineno-$LINENO}: checking assembler for --gdwarf-2 option" >&5
+$as_echo_n "checking assembler for --gdwarf-2 option... " >&6; }
 if test "${gcc_cv_as_gdwarf2_flag+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
@@ -27937,7 +27941,7 @@ else
 fi
   elif test x$gcc_cv_as != x; then
     $as_echo "$insn" > conftest.s
-    if { ac_try='$gcc_cv_as $gcc_cv_as_flags --gdwarf2 -o conftest.o conftest.s >&5'
+    if { ac_try='$gcc_cv_as $gcc_cv_as_flags --gdwarf-2 -o conftest.o conftest.s >&5'
   { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$ac_try\""; } >&5
   (eval $ac_try) 2>&5
   ac_status=$?
@@ -27960,6 +27964,9 @@ $as_echo "#define HAVE_AS_GDWARF2_DEBUG_FLAG 1" >>confdefs.h

 fi

+{ $as_echo "$as_me:${as_lineno-$LINENO}: defaulting assembler --gstabs option to no\n" >&5
+$as_echo_n "defaulting assembler --gstabs option to no\n" >&6; }
+gcc_cv_as_gstabs_flag=no

  { $as_echo "$as_me:${as_lineno-$LINENO}: checking assembler for --gstabs option" >&5
 $as_echo_n "checking assembler for --gstabs option... " >&6; }
