diff --git a/alphacsc/cython_code/tests/__init__.py b/alphacsc/cython_code/tests/__init__.py
new file mode 100644
index 0000000..e69de29
diff --git a/alphacsc/tests/__init__.py b/alphacsc/tests/__init__.py
new file mode 100644
index 0000000..e69de29
diff --git a/alphacsc/utils/arma.py b/alphacsc/utils/arma.py
index 6d1c759..883c276 100644
--- a/alphacsc/utils/arma.py
+++ b/alphacsc/utils/arma.py
@@ -4,7 +4,7 @@
 #
 
 import numpy as np
-from scipy import signal, linalg, fftpack, fft
+from scipy import signal, linalg, fftpack
 
 
 class Arma(object):
@@ -124,7 +124,8 @@ class Arma(object):
             count = 0
             while block[-1] < sig.size:
                 psd[i] += np.abs(
-                    fft(window * sig[block], fft_length, 0))[:n_freq] ** 2
+                    fftpack.fft(window * sig[block], fft_length, 0)
+                )[:n_freq] ** 2
                 count = count + 1
                 block = block + step
             if count == 0:
diff --git a/alphacsc/utils/tests/__init__.py b/alphacsc/utils/tests/__init__.py
new file mode 100644
index 0000000..e69de29
diff --git a/setup.py b/setup.py
index bf05718..6a319e7 100644
--- a/setup.py
+++ b/setup.py
@@ -58,7 +58,7 @@ if __name__ == "__main__":
         ],
         platforms='any',
         ext_modules=ext_modules,
-        packages=find_packages(exclude=["tests"]),
+        packages=find_packages(exclude=["sporco"]),
         setup_requires=['Cython', 'numpy'],
         install_requires=[
             'mne',
