diff --git a/Dictionaries/CMakeLists.txt b/Dictionaries/CMakeLists.txt
index 1234567..abcdefg 100644
--- a/Dictionaries/CMakeLists.txt
+++ b/Dictionaries/CMakeLists.txt
@@ -1,5 +1,4 @@
 find_package(ROOT COMPONENTS Core GenVector CONFIG)
-find_package(GCCXML REQUIRED)

 #-------------------------------------------------------------------------------
 # relax_generate_dictionary(dictionary headerfile selectionfile OPTIONS opt1 opt2 ...)
@@ -21,16 +20,6 @@ macro(relax_generate_dictionary dictionary _headerfile _selectionfile)
   endif()
   set(gensrcdict ${dictionary}Dict.cpp)

-  if(MSVC)
-    set(GCCXML_CXX_COMPILER cl CACHE STRING "Compiler that GCCXML must use.")
-  elseif(APPLE)
-    set(GCCXML_CXX_COMPILER g++ CACHE STRING "Compiler that GCCXML must use.")
-  else()
-    set(GCCXML_CXX_COMPILER ${CMAKE_CXX_COMPILER} CACHE STRING "Compiler that GCCXML must use.")
-  endif()
-  mark_as_advanced(GCCXML_CXX_COMPILER)
-  set(gccxmlopts "--gccxml-compiler ${GCCXML_CXX_COMPILER}")
-
   set(rootmapname ${CMAKE_SHARED_LIBRARY_PREFIX}${dictionary}Rflx.pamtoor)
   set(pcmname     ${dictionary}Dict_rdict.pcm)
   set(rootmapopts --rootmap=${rootmapname} --rootmap-lib=${CMAKE_SHARED_LIBRARY_PREFIX}${dictionary}Rflx)
@@ -45,14 +34,10 @@ macro(relax_generate_dictionary dictionary _headerfile _selectionfile)
    set(definitions ${definitions} -D${d})
   endforeach()

-  if(gccxmlopts)
-    set(gccxmlopts "--gccxmlopt=${gccxmlopts}")
-  endif()
-  get_filename_component(GCCXML_home ${GCCXML} PATH)
   add_custom_command(
     OUTPUT ${gensrcdict} ${rootmapname}
     COMMAND ${ROOT_genreflex_CMD}
-         ${headerfiles} -o ${gensrcdict} ${gccxmlopts} ${rootmapopts} --select=${selectionfile}
-         --gccxmlpath=${GCCXML_home} ${ARG_OPTIONS} ${include_dirs} ${definitions} ${GCCXML_CXX_FLAGS}
+         ${headerfiles} -o ${gensrcdict} ${rootmapopts} --select=${selectionfile}
+         ${ARG_OPTIONS} ${include_dirs} ${definitions}
     DEPENDS ${headerfiles} ${selectionfile})

   # Creating this target at ALL level enables the possibility to generate dictionaries (genreflex step)
