diff --git a/Dictionaries/CMakeLists.txt b/Dictionaries/CMakeLists.txt
--- a/Dictionaries/CMakeLists.txt
+++ b/Dictionaries/CMakeLists.txt
@@ -71,7 +71,7 @@ function(relax_dictionary dictionary headerfile selectionfile)
   separate_arguments(ARG_OPTIONS)

   relax_generate_dictionary(${dictionary} ${headerfile} ${selectionfile} OPTIONS ${ARG_OPTIONS})
-  add_library(${dictionary}Rflx MODULE ${gensrcdict})
+  add_library(${dictionary}Rflx SHARED ${gensrcdict})
   target_link_libraries(${dictionary}Rflx ${ARG_LINK_LIBRARIES} )
   # ensure that *Gen and *Rflx are not built at the same time
   add_dependencies(${dictionary}Rflx ${dictionary}Gen)
