diff --git a/deps/v8/src/inspector/v8-runtime-agent-impl.cc b/deps/v8/src/inspector/v8-runtime-agent-impl.cc
--- a/deps/v8/src/inspector/v8-runtime-agent-impl.cc
+++ b/deps/v8/src/inspector/v8-runtime-agent-impl.cc
@@ -30,7 +30,7 @@
 
 #include "src/inspector/v8-runtime-agent-impl.h"
 
-#include <inttypes.h>
+#include <cinttypes>
 
 #include "src/debug/debug-interface.h"
 #include "src/inspector/injected-script.h"
