diff --git a/scripts/node_build.bash b/scripts/node_build.bash
index fc973c4..369c580 100755
--- a/scripts/node_build.bash
+++ b/scripts/node_build.bash
@@ -13,31 +13,16 @@ function main() {
 
   cp ../../out/patches/thirdPartyMain.js ./lib/_third_party_main.js
 
-  if ! git apply ../../scripts/node.patch ; then
+  if ! patch -p1 < ../../scripts/node.patch ; then
     echo "Failed to patch; assuming already patched"
   fi
 
-  local -i cores=2
-  export CC="ccache gcc"
-  export CXX="ccache g++"
-  if [[ -n ${XDG_CACHE_HOME:-} ]] ; then
-    export CCACHE_DIR="$XDG_CACHE_HOME/ccache"
-  elif [[ $OSTYPE == "darwin"* ]]; then
-    export CCACHE_DIR="$HOME/Library/Caches/ccache"
-  else
-    export CCACHE_DIR="$HOME/.cache/ccache"
-    cores=$(awk '/^processor/{print $3}' /proc/cpuinfo | wc -l)
-  fi
-
-  echo "ccache directory: $CCACHE_DIR"
-  echo "cores: $cores"
-
   ./configure \
     --link-module ./lib/nbin.js \
     --link-module ./lib/_third_party_main.js \
     --openssl-no-asm --openssl-use-def-ca-store
 
-  make "-j$cores"
+  make "-j$CPU_COUNT"
 }
 
 main "$@"
