diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -46,7 +46,7 @@
 # Dependencies
 cc = meson.get_compiler('c')

-libm_dep = cc.find_library('m', required: true)
+libm_dep = cc.find_library('m', required: false)

 glib_req = '>= 2.74.0'
 cairo_req = '>= 1.4'
@@ -59,7 +59,10 @@
 gobject_dep = dependency('gobject-2.0', version: glib_req)
 gio_dep = dependency('gio-2.0', version: glib_req)
 cairo_dep = dependency('cairo', version: cairo_req)
-sqlite_dep = dependency('sqlite3', version: sqlite_req)
+sqlite_dep = dependency('sqlite3', version: sqlite_req, required: false)
+if not sqlite_dep.found()
+  sqlite_dep = cc.find_library('sqlite3')
+endif
 gtk_dep = dependency('gtk4')
 libsoup_dep = dependency('libsoup-3.0', version: '>= 3.0')
