diff --git a/Makefile.vc b/Makefile.vc
index fde3e48..5b1ff33 100644
--- a/Makefile.vc
+++ b/Makefile.vc
@@ -8,12 +8,8 @@
 # place to put the GeoIP.dat database file
 # !!! Please keep the 2 \\ as directory separators !!!
 #
-GEOIPDATADIR="C:\\Windows\\SYSTEM32"
 #
 # System inc, lib, and bin directories
-!ifndef INSTDIR
-INSTDIR="C:\GeoIP"
-!endif

 # Location where GeoIP.lib should be installed my "make install"
 INSTALL_LIB=$(INSTDIR)\Lib
