--- chemfiles/find_chemfiles.py
+++ chemfiles/find_chemfiles.py
@@ -1,7 +1,7 @@
 # -* coding: utf-8 -*
 import os
+import sys
 from ctypes import cdll
-from ctypes.util import find_library

 from chemfiles import ffi

@@ -10,10 +10,15 @@ ROOT = os.path.dirname(__file__)

 def load_clib():
     '''Load chemfiles C++ library'''
-    libpath = find_library("chemfiles")
-    if not libpath:
-        # Rely on the library built by the setup.py function
-        libpath = os.path.join(ROOT, "_chemfiles.so")
+    if os.name == 'nt':
+        libpath = os.path.join(sys.prefix, "Library", "bin", "chemfiles.dll")
+    elif os.name == 'posix':
+        platform = os.uname()[0]
+        if platform == 'Linux':
+            libname = 'libchemfiles.so'
+        else:
+            libname = 'libchemfiles.dylib'
+        libpath = os.path.join(sys.prefix, 'lib', libname)
     try:
         return cdll.LoadLibrary(libpath)
     except OSError:
