Index: tests/test_msgs.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/test_msgs.py b/tests/test_msgs.py
--- a/tests/test_msgs.py	(revision d931fa7f11bb2661bdaf320ee96fc771c7b7dbbb)
+++ b/tests/test_msgs.py	(revision 16586a4de62667d0d21f0c00723a662501c47bd0)
@@ -5,6 +5,7 @@
 import time
 import unittest
 from typing import Optional
+import pydantic
 
 from commlib.msg import Message, MessageHeader, PubSubMessage, RPCMessage
 from commlib.timer import Timer
@@ -46,7 +47,10 @@
 
         _msg = TestPubSubMessage()
         _msg.b = TestObject(c=2, d=3)
-        self.assertEqual(_msg.dict(), _d)
+        if pydantic.__version__ < '2.0.0':
+            self.assertEqual(_msg.dict(), _d)
+        else:
+            self.assertEqual(_msg.model_dump(), _d)
 
     def test_nested_message_from_dict(self):
         _d = {
