diff --git b/pyproject.toml a/pyproject.toml
--- b/pyproject.toml
+++ a/pyproject.toml
@@ -8,11 +8,11 @@ packages = [{include = "commlib"}]

 [tool.poetry.dependencies]
 python = "^3.8"
-pika = "1.3.1"
-paho-mqtt = "1.6.1"
-hiredis = "2.1.1"
-ujson = "5.7.0"
-pydantic = "1.10.4"
+pika = "^1.3.1"
+paho-mqtt = "^1.6.1"
+hiredis = "^2.1.1"
+ujson = "^5.7.0"
+pydantic = ">=1.10.4"

 [tool.poetry.group.dev.dependencies]
 bump2version = "0.5.11"
