diff --git a/m4/cairo.m4 b/m4/cairo.m4
--- a/m4/cairo.m4
+++ b/m4/cairo.m4
@@ -8,10 +8,18 @@ AC_DEFUN([GA_CHECK_LIB_CAIRO],
   cairo_pkgconfig=no
+  cairo_xlib=no
   PKG_CHECK_MODULES([CAIRO],[cairo],[cairo_pkgconfig=yes])
 
+  if test $cairo_pkgconfig = 'yes'; then
+     ga_save_CPPFLAGS=$CPPFLAGS
+     CPPFLAGS="$CPPFLAGS $CAIRO_CFLAGS"
+     AC_CHECK_HEADER([cairo-xlib.h],[cairo_xlib=yes])
+     CPPFLAGS=$ga_save_CPPFLAGS
+  fi
+
-  if test $cairo_pkgconfig = 'yes'; then
+  if test $cairo_pkgconfig = 'yes' -a $cairo_xlib = 'yes'; then
      m4_if([$1], [], [:], [$1])
   else
      m4_if([$2], [], [:], [$2])
   fi
 
 ])
diff --git a/src/gafunc.c b/src/gafunc.c
--- a/src/gafunc.c
+++ b/src/gafunc.c
@@ -5445 +5445 @@ gaint ffoacr (struct gafunc *pfc, struct gastat *pst) {
-gadouble e1,e2,e,wsum,x,y,xmin,xmax,ymin,ymax,d,d2,rad,rad2,w;
+gadouble e1,e2,e,wsum,x,y,xmin,xmax,ymin,ymax,d,d2,rad,radius_squared,weight;
@@ -5643 +5643 @@ gaint ffoacr (struct gafunc *pfc, struct gastat *pst) {
-    rad2 = rad*rad;
+    radius_squared = rad*rad;
@@ -5673,3 +5673,3 @@ gaint ffoacr (struct gafunc *pfc, struct gastat *pst) {
-            w = (rad2-d2)/(rad2+d2);
-            sum += e*w;
-            wsum += w;
+            weight = (radius_squared-d2)/(radius_squared+d2);
+            sum += e*weight;
+            wsum += weight;
