From a28d93befc7771c41f18a7d2e406809cd0da156b Mon Sep 17 00:00:00 2001
From: BjoernBiltzinger <bjoernbiltzinger@gmx.de>
Date: Tue, 25 Jan 2022 13:07:13 +0100
Subject: [PATCH] fix for pandas version 1.4

---
 threeML/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/threeML/__init__.py b/threeML/__init__.py
index 932f0db1..e0acf481 100644
--- a/threeML/__init__.py
+++ b/threeML/__init__.py
@@ -2,7 +2,7 @@
 # Indeed, if no DISPLAY variable is set, matplotlib 2.0 crashes (at the moment, 05/26/2017)
 import pandas as pd
 
-pd.set_option("max_columns", None)
+pd.set_option("display.max_columns", None)
 
 import os
 import traceback
