diff --git a/buildscripts/gen-ldflags.go b/buildscripts/gen-ldflags.go
index a7649056f..644270762 100644
--- a/buildscripts/gen-ldflags.go
+++ b/buildscripts/gen-ldflags.go
@@ -73,18 +73,7 @@ func releaseTag(version string) (string, time.Time) {
 // commitID returns the abbreviated commit-id hash of the last commit.
 func commitID() string {
 	// git log --format="%H" -n1
-	var (
-		commit []byte
-		err    error
-	)
-	cmdName := "git"
-	cmdArgs := []string{"log", "--format=%H", "-n1"}
-	if commit, err = exec.Command(cmdName, cmdArgs...).Output(); err != nil {
-		fmt.Fprintln(os.Stderr, "Error generating git commit-id: ", err)
-		os.Exit(1)
-	}
-
-	return strings.TrimSpace(string(commit))
+	return strings.TrimSpace(string(os.Getenv("GIT_COMMIT")))
 }
 
 func commitTime() time.Time {
