From 9c7883edb68745ec43e6b7df632fa93b9e0393fd Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@lanl.gov>
Date: Wed, 26 Jul 2023 16:45:43 -0600
Subject: [PATCH] patch setup

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 229c17f..dae2bc5 100644
--- a/setup.py
+++ b/setup.py
@@ -20,7 +20,6 @@
     install_requires=[
         "pymatgen>=2023.1.20",
         "numpy",
-        "typing",
         "Sphinx>=4",
         "sphinx-argparse==0.4.0",
         "sphinx-pdj-theme==0.2.1",
