From d21e62e48c32db2e93e8ca1a3d65eb0856bd5b8a Mon Sep 17 00:00:00 2001
From: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 13:38:36 +0100
Subject: [PATCH] Define encoding

---
 quantumflow/config_test.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quantumflow/config_test.py b/quantumflow/config_test.py
index 67fbbef..146f0b1 100644
--- a/quantumflow/config_test.py
+++ b/quantumflow/config_test.py
@@ -35,7 +35,7 @@ def test_copyright() -> None:
             continue
         print("Checking " + fname + " for copyright header...  ", end="")
 
-        with open(fname) as f:
+        with open(fname, encoding='utf-8') as f:
             for line in f.readlines():
                 if not line.strip():
                     continue
-- 
2.33.1

