From d21e62e48c32db2e93e8ca1a3d65eb0856bd5b8a Mon Sep 17 00:00:00 2001
From: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 13:38:36 +0100
Subject: [PATCH 01/16] Define encoding

---
 quantumflow/config_test.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quantumflow/config_test.py b/quantumflow/config_test.py
index 67fbbef..146f0b1 100644
--- a/quantumflow/config_test.py
+++ b/quantumflow/config_test.py
@@ -35,7 +35,7 @@ def test_copyright() -> None:
             continue
         print("Checking " + fname + " for copyright header...  ", end="")
 
-        with open(fname) as f:
+        with open(fname, encoding='utf-8') as f:
             for line in f.readlines():
                 if not line.strip():
                     continue

From 62eaf01feb9beec439d7a621a90ba6768b4dd0c3 Mon Sep 17 00:00:00 2001
From: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 13:45:44 +0100
Subject: [PATCH 02/16] Run tests on all OS

---
 .github/workflows/python-build.yml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.github/workflows/python-build.yml b/.github/workflows/python-build.yml
index 5bbb488..6120662 100644
--- a/.github/workflows/python-build.yml
+++ b/.github/workflows/python-build.yml
@@ -17,6 +17,7 @@ jobs:
     strategy:
       matrix:
         python-version: ['3.7', '3.8', '3.9']
+        os: [ubuntu-latest, macos-latest, windows-latest]
 
     steps:
     - uses: actions/checkout@v2

From addb2b8baf002745a37de0d63730dc3f824abf86 Mon Sep 17 00:00:00 2001
From: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 14:40:03 +0100
Subject: [PATCH 03/16] Use matrix

---
 .github/workflows/python-build.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/python-build.yml b/.github/workflows/python-build.yml
index 6120662..18eb35d 100644
--- a/.github/workflows/python-build.yml
+++ b/.github/workflows/python-build.yml
@@ -13,7 +13,7 @@ on:
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         python-version: ['3.7', '3.8', '3.9']

From 2c037a00a5e614c34ece6bb3bbc664fa3fe38a04 Mon Sep 17 00:00:00 2001
From: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 16:25:38 +0100
Subject: [PATCH 04/16] Disable fail fast

---
 .github/workflows/python-build.yml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.github/workflows/python-build.yml b/.github/workflows/python-build.yml
index 18eb35d..3007f62 100644
--- a/.github/workflows/python-build.yml
+++ b/.github/workflows/python-build.yml
@@ -15,6 +15,7 @@ jobs:
 
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
         python-version: ['3.7', '3.8', '3.9']
         os: [ubuntu-latest, macos-latest, windows-latest]

From 8adf939bdcdd8745fb6d02049f23c1d43fc8ac90 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 20:46:52 +0100
Subject: [PATCH 05/16] Run black

---
 quantumflow/config_test.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quantumflow/config_test.py b/quantumflow/config_test.py
index 146f0b1..7979d11 100644
--- a/quantumflow/config_test.py
+++ b/quantumflow/config_test.py
@@ -35,7 +35,7 @@ def test_copyright() -> None:
             continue
         print("Checking " + fname + " for copyright header...  ", end="")
 
-        with open(fname, encoding='utf-8') as f:
+        with open(fname, encoding="utf-8") as f:
             for line in f.readlines():
                 if not line.strip():
                     continue

From 3e7bf4626897cbedb246819ed5ba783470124ac9 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:05:34 +0100
Subject: [PATCH 06/16] Use shell

---
 examples/examples_test.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/examples/examples_test.py b/examples/examples_test.py
index 21e5206..1f6196f 100644
--- a/examples/examples_test.py
+++ b/examples/examples_test.py
@@ -24,7 +24,7 @@ def test_prepare_w4():
 
 
 def test_prepare_w4_main():
-    rval = subprocess.call(["examples/state_prep_w4.py"])
+    rval = subprocess.call(["examples/state_prep_w4.py"], shell=True)
     assert rval == 0
 
 
@@ -36,12 +36,12 @@ def test_prepare_w16():
 
 
 def test_prepare_w16_main():
-    rval = subprocess.call(["examples/state_prep_w16.py"])
+    rval = subprocess.call(["examples/state_prep_w16.py"], shell=True)
     assert rval == 0
 
 
 def test_prepare_cswap_decomposition():
-    rval = subprocess.call(["examples/cswap_decomposition.py"])
+    rval = subprocess.call(["examples/cswap_decomposition.py"], shell=True)
     assert rval == 0
 
 
@@ -62,17 +62,17 @@ def test_swap_test():
 
 
 def test_swap_test_main():
-    rval = subprocess.call(["examples/swaptest.py"])
+    rval = subprocess.call(["examples/swaptest.py"], shell=True)
     assert rval == 0
 
 
 def test_circuit_identities_main():
-    rval = subprocess.call(["examples/circuit_identities.py"])
+    rval = subprocess.call(["examples/circuit_identities.py"], shell=True)
     assert rval == 0
 
 
 def test_gate_translate_identities_main():
-    rval = subprocess.call(["examples/gate_translations.py"])
+    rval = subprocess.call(["examples/gate_translations.py"], shell=True)
     assert rval == 0
 
 

From 53d2eb78ab61dc252e909ddac6cab192d626437a Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:15:08 +0100
Subject: [PATCH 07/16] Use cwd

---
 examples/examples_test.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/examples/examples_test.py b/examples/examples_test.py
index 1f6196f..00eb413 100644
--- a/examples/examples_test.py
+++ b/examples/examples_test.py
@@ -24,7 +24,7 @@ def test_prepare_w4():
 
 
 def test_prepare_w4_main():
-    rval = subprocess.call(["examples/state_prep_w4.py"], shell=True)
+    rval = subprocess.call(["state_prep_w4.py"], shell=True, cwd="examples")
     assert rval == 0
 
 
@@ -36,12 +36,12 @@ def test_prepare_w16():
 
 
 def test_prepare_w16_main():
-    rval = subprocess.call(["examples/state_prep_w16.py"], shell=True)
+    rval = subprocess.call(["state_prep_w16.py"], shell=True, cwd="examples")
     assert rval == 0
 
 
 def test_prepare_cswap_decomposition():
-    rval = subprocess.call(["examples/cswap_decomposition.py"], shell=True)
+    rval = subprocess.call(["cswap_decomposition.py"], shell=True, cwd="examples")
     assert rval == 0
 
 
@@ -62,17 +62,17 @@ def test_swap_test():
 
 
 def test_swap_test_main():
-    rval = subprocess.call(["examples/swaptest.py"], shell=True)
+    rval = subprocess.call(["swaptest.py"], shell=True, cwd="examples")
     assert rval == 0
 
 
 def test_circuit_identities_main():
-    rval = subprocess.call(["examples/circuit_identities.py"], shell=True)
+    rval = subprocess.call(["circuit_identities.py"], shell=True, cwd="examples")
     assert rval == 0
 
 
 def test_gate_translate_identities_main():
-    rval = subprocess.call(["examples/gate_translations.py"], shell=True)
+    rval = subprocess.call(["gate_translations.py"], shell=True, cwd="examples")
     assert rval == 0
 
 

From 71bf4a48e8dd43cac2fc3205adcd185c10e7edba Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:18:35 +0100
Subject: [PATCH 08/16] Run some tests only on Linux

---
 .github/workflows/python-build.yml | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/.github/workflows/python-build.yml b/.github/workflows/python-build.yml
index 3007f62..6bf3605 100644
--- a/.github/workflows/python-build.yml
+++ b/.github/workflows/python-build.yml
@@ -41,30 +41,39 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-        python -m pip install .[dev]  # install package + test dependencies   
+        python -m pip install .[dev]  # install package + test dependencies
+    - name: Install extra Linux dependencies
+      if: runner.os == 'Linux'
+      run: |
         sudo apt-get install texlive texlive-latex-extra latexmk poppler-utils
         mkdir -p ~/texmf/tex/latex
         wget http://mirrors.ctan.org/graphics/pgf/contrib/quantikz/tikzlibraryquantikz.code.tex -P ~/texmf/tex/latex
     - name: About
+      if: runner.os == 'Linux'
       run: |
         python -m $(python -Wi setup.py --name).about
         echo
         pdflatex --version
         pdftocairo -v
     - name: Test coverage with pytest
+      if: runner.os == 'Linux'
       run: |
         python -m pytest --cov --cov-fail-under 100
     - name: Lint with black and isort (run `make delint` to fix issues)
+      if: runner.os == 'Linux'
       run: |
         black --check .  
         isort --check -m 3 --tc .
     - name: Lint with flake8
+      if: runner.os == 'Linux'
       run: |
         flake8 .            
     - name: Typecheck with mypy
+      if: runner.os == 'Linux'
       run: |
         mypy
     - name: Build documentation with sphinx
+      if: runner.os == 'Linux'
       run: |
         sphinx-build -M html docs docs/_build
     - name: Test notebooks (against master)

From 56567e1953defc462e7516feb433ad90573986c6 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:20:34 +0100
Subject: [PATCH 09/16] Revert "Use cwd"

This reverts commit 53d2eb78ab61dc252e909ddac6cab192d626437a.
---
 examples/examples_test.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/examples/examples_test.py b/examples/examples_test.py
index 00eb413..1f6196f 100644
--- a/examples/examples_test.py
+++ b/examples/examples_test.py
@@ -24,7 +24,7 @@ def test_prepare_w4():
 
 
 def test_prepare_w4_main():
-    rval = subprocess.call(["state_prep_w4.py"], shell=True, cwd="examples")
+    rval = subprocess.call(["examples/state_prep_w4.py"], shell=True)
     assert rval == 0
 
 
@@ -36,12 +36,12 @@ def test_prepare_w16():
 
 
 def test_prepare_w16_main():
-    rval = subprocess.call(["state_prep_w16.py"], shell=True, cwd="examples")
+    rval = subprocess.call(["examples/state_prep_w16.py"], shell=True)
     assert rval == 0
 
 
 def test_prepare_cswap_decomposition():
-    rval = subprocess.call(["cswap_decomposition.py"], shell=True, cwd="examples")
+    rval = subprocess.call(["examples/cswap_decomposition.py"], shell=True)
     assert rval == 0
 
 
@@ -62,17 +62,17 @@ def test_swap_test():
 
 
 def test_swap_test_main():
-    rval = subprocess.call(["swaptest.py"], shell=True, cwd="examples")
+    rval = subprocess.call(["examples/swaptest.py"], shell=True)
     assert rval == 0
 
 
 def test_circuit_identities_main():
-    rval = subprocess.call(["circuit_identities.py"], shell=True, cwd="examples")
+    rval = subprocess.call(["examples/circuit_identities.py"], shell=True)
     assert rval == 0
 
 
 def test_gate_translate_identities_main():
-    rval = subprocess.call(["gate_translations.py"], shell=True, cwd="examples")
+    rval = subprocess.call(["examples/gate_translations.py"], shell=True)
     assert rval == 0
 
 

From 74384268598e32b27ae8b008e28ffade3e34dca7 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:22:30 +0100
Subject: [PATCH 10/16] Use os.path.join

---
 examples/examples_test.py | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/examples/examples_test.py b/examples/examples_test.py
index 1f6196f..7f1c801 100644
--- a/examples/examples_test.py
+++ b/examples/examples_test.py
@@ -9,6 +9,8 @@
 Unit tests for quantumflow examples.
 """
 
+import os
+
 import subprocess
 
 import numpy as np
@@ -24,7 +26,7 @@ def test_prepare_w4():
 
 
 def test_prepare_w4_main():
-    rval = subprocess.call(["examples/state_prep_w4.py"], shell=True)
+    rval = subprocess.call([os.path.join("examples", "state_prep_w4.py")], shell=True)
     assert rval == 0
 
 
@@ -36,12 +38,12 @@ def test_prepare_w16():
 
 
 def test_prepare_w16_main():
-    rval = subprocess.call(["examples/state_prep_w16.py"], shell=True)
+    rval = subprocess.call([os.path.join("examples", "state_prep_w16.py")], shell=True)
     assert rval == 0
 
 
 def test_prepare_cswap_decomposition():
-    rval = subprocess.call(["examples/cswap_decomposition.py"], shell=True)
+    rval = subprocess.call([os.path.join("examples", "cswap_decomposition.py")], shell=True)
     assert rval == 0
 
 
@@ -62,17 +64,17 @@ def test_swap_test():
 
 
 def test_swap_test_main():
-    rval = subprocess.call(["examples/swaptest.py"], shell=True)
+    rval = subprocess.call([os.path.join("examples", "swaptest.py")], shell=True)
     assert rval == 0
 
 
 def test_circuit_identities_main():
-    rval = subprocess.call(["examples/circuit_identities.py"], shell=True)
+    rval = subprocess.call([os.path.join("examples", "circuit_identities.py")], shell=True)
     assert rval == 0
 
 
 def test_gate_translate_identities_main():
-    rval = subprocess.call(["examples/gate_translations.py"], shell=True)
+    rval = subprocess.call([os.path.join("examples", "gate_translations.py")], shell=True)
     assert rval == 0
 
 

From d4d8a415de0f3d9fb530287818059bfc5862235e Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:22:42 +0100
Subject: [PATCH 11/16] Black

---
 examples/examples_test.py | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/examples/examples_test.py b/examples/examples_test.py
index 7f1c801..317ca80 100644
--- a/examples/examples_test.py
+++ b/examples/examples_test.py
@@ -43,7 +43,9 @@ def test_prepare_w16_main():
 
 
 def test_prepare_cswap_decomposition():
-    rval = subprocess.call([os.path.join("examples", "cswap_decomposition.py")], shell=True)
+    rval = subprocess.call(
+        [os.path.join("examples", "cswap_decomposition.py")], shell=True
+    )
     assert rval == 0
 
 
@@ -69,12 +71,16 @@ def test_swap_test_main():
 
 
 def test_circuit_identities_main():
-    rval = subprocess.call([os.path.join("examples", "circuit_identities.py")], shell=True)
+    rval = subprocess.call(
+        [os.path.join("examples", "circuit_identities.py")], shell=True
+    )
     assert rval == 0
 
 
 def test_gate_translate_identities_main():
-    rval = subprocess.call([os.path.join("examples", "gate_translations.py")], shell=True)
+    rval = subprocess.call(
+        [os.path.join("examples", "gate_translations.py")], shell=True
+    )
     assert rval == 0
 
 

From f3271e8014b97623e025989ed62c257946140d11 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:28:29 +0100
Subject: [PATCH 12/16] Remove requirements.txt

Fails on Windows
---
 .github/workflows/python-build.yml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/.github/workflows/python-build.yml b/.github/workflows/python-build.yml
index 6bf3605..d2260c8 100644
--- a/.github/workflows/python-build.yml
+++ b/.github/workflows/python-build.yml
@@ -40,7 +40,6 @@ jobs:
     - name: Install extra dependencies
       run: |
         python -m pip install --upgrade pip
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
         python -m pip install .[dev]  # install package + test dependencies
     - name: Install extra Linux dependencies
       if: runner.os == 'Linux'

From 721716db721d16f96f4beb7d2a4cacb32a7d6ca3 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:28:53 +0100
Subject: [PATCH 13/16] Install wheels

Skips legacy warning
---
 .github/workflows/python-build.yml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.github/workflows/python-build.yml b/.github/workflows/python-build.yml
index d2260c8..d25d6f0 100644
--- a/.github/workflows/python-build.yml
+++ b/.github/workflows/python-build.yml
@@ -40,6 +40,7 @@ jobs:
     - name: Install extra dependencies
       run: |
         python -m pip install --upgrade pip
+        python -m pip install --upgrade wheels
         python -m pip install .[dev]  # install package + test dependencies
     - name: Install extra Linux dependencies
       if: runner.os == 'Linux'

From abe829cd702946016ea859b16ccff20275535d13 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:33:15 +0100
Subject: [PATCH 14/16] isort

---
 examples/examples_test.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/examples/examples_test.py b/examples/examples_test.py
index 317ca80..5c0a8c4 100644
--- a/examples/examples_test.py
+++ b/examples/examples_test.py
@@ -10,7 +10,6 @@
 """
 
 import os
-
 import subprocess
 
 import numpy as np

From 67a3324576ccb098eba48804a3f14209fa0d3740 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:35:39 +0100
Subject: [PATCH 15/16] Fix install

---
 .github/workflows/python-build.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/python-build.yml b/.github/workflows/python-build.yml
index d25d6f0..b7c7efe 100644
--- a/.github/workflows/python-build.yml
+++ b/.github/workflows/python-build.yml
@@ -40,7 +40,7 @@ jobs:
     - name: Install extra dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install --upgrade wheels
+        python -m pip install wheels
         python -m pip install .[dev]  # install package + test dependencies
     - name: Install extra Linux dependencies
       if: runner.os == 'Linux'

From 137ee01d0ad45a6eafc793407975d286f5a13853 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 30 Nov 2021 21:38:28 +0100
Subject: [PATCH 16/16] wheel

---
 .github/workflows/python-build.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/python-build.yml b/.github/workflows/python-build.yml
index b7c7efe..2715459 100644
--- a/.github/workflows/python-build.yml
+++ b/.github/workflows/python-build.yml
@@ -40,7 +40,7 @@ jobs:
     - name: Install extra dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install wheels
+        python -m pip install wheel
         python -m pip install .[dev]  # install package + test dependencies
     - name: Install extra Linux dependencies
       if: runner.os == 'Linux'
