diff --git a/src/engine/engine_collision_convex.c b/src/engine/engine_collision_convex.c
index 0277c63..f12f458 100644
--- a/src/engine/engine_collision_convex.c
+++ b/src/engine/engine_collision_convex.c
@@ -215,7 +215,8 @@ static int mjc_MPRIteration(mjtCCD* obj1, mjtCCD* obj2, const ccd_t* ccd,
   ccd_real_t depth;
   if (ccdMPRPenetration(obj1, obj2, ccd, &depth, &dir, &pos)==0) {
     // contact is found but normal is undefined
-    if (ccdVec3Eq(&dir, ccd_vec3_origin)) {
+    CCD_VEC3(local_ccd_vec3_origin, CCD_ZERO, CCD_ZERO, CCD_ZERO);
+    if (ccdVec3Eq(&dir, &local_ccd_vec3_origin)) {
       return 0;
     }
 
@@ -726,8 +727,9 @@ int mjc_ConvexHField(const mjModel* m, const mjData* d,
           }
 
           // run MPR, save contact
+          CCD_VEC3(local_ccd_vec3_origin, CCD_ZERO, CCD_ZERO, CCD_ZERO);
           if (ccdMPRPenetration(&prism, &obj, &ccd, &depth, &dirccd, &vecccd)==0 &&
-              !ccdVec3Eq(&dirccd, ccd_vec3_origin)) {
+              !ccdVec3Eq(&dirccd, &local_ccd_vec3_origin)) {
             // fill in contact data, transform to global coordinates
             con[cnt].dist = -depth;
             mju_rotVecMat(con[cnt].frame, dirccd.v, mat1);
