diff --git a/python/setup.py b/python/setup.py
index d6291fd..653f3c6 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -146,8 +146,8 @@ def run(self):
       assert ext.name.startswith(EXT_PREFIX)
       assert '.' not in ext.name[len(EXT_PREFIX):]
       self.build_extension(ext)
-    self._copy_external_libraries()
-    self._copy_mujoco_headers()
+    #self._copy_external_libraries()
+    #self._copy_mujoco_headers()
 
   def _find_mujoco(self):
     if MUJOCO_PATH not in os.environ:

diff --git a/python/mujoco/__init__.py b/python/mujoco/__init__.py
index e33a132..1aa6559 100644
--- a/python/mujoco/__init__.py
+++ b/python/mujoco/__init__.py
@@ -22,7 +22,7 @@
 
 _SYSTEM = platform.system()
 if _SYSTEM == 'Windows':
-  ctypes.WinDLL(os.path.join(os.path.dirname(__file__), 'mujoco.dll'))
+  ctypes.WinDLL('mujoco.dll')
 
 from mujoco._callbacks import *
 from mujoco._constants import *
