From 672d44005237acc203a238b76d291bd63ffc8386 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sat, 18 Jun 2022 01:33:24 +0200
Subject: [PATCH] Do not install via pip

---
 python/make_sdist.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/python/make_sdist.sh b/python/make_sdist.sh
index 3308dee..acbd5a0 100755
--- a/python/make_sdist.sh
+++ b/python/make_sdist.sh
@@ -28,8 +28,8 @@ else
   readonly tmp_dir="$(mktemp -d)"
 fi
 
-python -m pip install --upgrade pip setuptools
-python -m pip install absl-py
+# python -m pip install --upgrade pip setuptools
+# python -m pip install absl-py
 pushd ${tmp_dir}
 cp -r "${package_dir}"/* .
