diff --git a/python/mujoco/CMakeLists.txt b/python/mujoco/CMakeLists.txt
index 9e5242f..cbd9b5e 100644
--- a/python/mujoco/CMakeLists.txt
+++ b/python/mujoco/CMakeLists.txt
@@ -265,12 +265,9 @@ target_link_libraries(
             tuple_tools
 )
 
-include(CheckAvxSupport)
-get_avx_compile_options(AVX_COMPILE_OPTIONS)
 
 macro(mujoco_pybind11_module name)
   pybind11_add_module(${name} ${ARGN})
-  target_compile_options(${name} PRIVATE ${AVX_COMPILE_OPTIONS})
   if(NOT MSVC)
     target_compile_options(${name} PRIVATE -Wall -Werror)
     if(CMAKE_C_COMPILER_ID STREQUAL GNU)
