From 41867bea719befa2f9e3e187997acfc803f919b1 Mon Sep 17 00:00:00 2001
From: Rui Chen <rui@chenrui.dev>
Date: Sat, 14 Sep 2024 22:47:23 -0400
Subject: [PATCH] build: dont force CMAKE_OSX_ARCHITECTURES

Signed-off-by: Rui Chen <rui@chenrui.dev>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2d0aa859d..8d7e4410e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.16)
 
 #make a universal binary on macOS
-set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "" FORCE)
+set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "")
 
 project(Clipboard LANGUAGES CXX C VERSION 0.9.0)
 set(CMAKE_CXX_STANDARD 23)
