index 7cd3d9b..0dac9c3 100755
--- a/setup.py
+++ b/setup.py
@@ -37,15 +37,13 @@ def get_output(*args, **kwargs):


 # get the compile and link args
+posix = os.name != 'nt'
 link_args, compile_args = [
-    shlex.split(os.environ[e]) if e in os.environ else None
+    shlex.split(os.environ[e], posix=posix) if e in os.environ else None
     for e in ['GSSAPI_LINKER_ARGS', 'GSSAPI_COMPILER_ARGS']
 ]

 osx_has_gss_framework = False
-if sys.platform == 'darwin':
-    mac_ver = [int(v) for v in platform.mac_ver()[0].split('.')]
-    osx_has_gss_framework = (mac_ver >= [10, 7, 0])

 winkrb_path = None
 if os.name == 'nt':
--
2.27.0.windows.1
