From b261c4da11048dadb3c185e0e803e2b3e7cc2b40 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 2 May 2018 03:32:40 -0700
Subject: [PATCH] Ensure we do not end up wih CRLF line endings on tests

---
 .gitattributes | 3 +++
 1 file changed, 3 insertions(+)
 create mode 100644 .gitattributes

diff --git a/.gitattributes b/.gitattributes
new file mode 100644
index 0000000..f363c35
--- /dev/null
+++ b/.gitattributes
@@ -0,0 +1,3 @@
+# These files are opened as 'rb' and
+# compared against files not written as 'wb
+tests/data/* -text
-- 
2.16.2
