--- a/setup.py
+++ b/setup.py
@@ -295,17 +295,17 @@ if WITH_SETUPTOOLS:
                 if __saltstack_version__.info < (2015, 8):  # pylint: disable=undefined-variable
                     # Install M2Crypto first
                     self.distribution.salt_installing_m2crypto_windows = True
-                    self.run_command('install-m2crypto-windows')
+                    # self.run_command('install-m2crypto-windows')
                     self.distribution.salt_installing_m2crypto_windows = None

                 # Install PyCrypto
                 self.distribution.salt_installing_pycrypto_windows = True
-                self.run_command('install-pycrypto-windows')
+                # self.run_command('install-pycrypto-windows')
                 self.distribution.salt_installing_pycrypto_windows = None

                 # Download the required DLLs
                 self.distribution.salt_download_windows_dlls = True
-                self.run_command('download-windows-dlls')
+                # self.run_command('download-windows-dlls')
                 self.distribution.salt_download_windows_dlls = None

             if self.write_salt_version is True:
@@ -746,15 +746,15 @@ class Install(install):
             if __saltstack_version__.info < (2015, 8):  # pylint: disable=undefined-variable
                 # Install M2Crypto first
                 self.distribution.salt_installing_m2crypto_windows = True
-                self.run_command('install-m2crypto-windows')
+                # self.run_command('install-m2crypto-windows')
                 self.distribution.salt_installing_m2crypto_windows = None
             # Install PyCrypto
             self.distribution.salt_installing_pycrypto_windows = True
-            self.run_command('install-pycrypto-windows')
+            # self.run_command('install-pycrypto-windows')
             self.distribution.salt_installing_pycrypto_windows = None
             # Download the required DLLs
             self.distribution.salt_download_windows_dlls = True
-            self.run_command('download-windows-dlls')
+            # self.run_command('download-windows-dlls')
             self.distribution.salt_download_windows_dlls = None
         # Run install.run
         install.run(self)
