From ab92d8ea1f48397d8a9be1d17c8b63fbfb28468c Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 23 Jul 2024 08:00:24 +0200
Subject: [PATCH] Create MANIFEST.in

---
 MANIFEST.in | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 MANIFEST.in

diff --git a/MANIFEST.in b/MANIFEST.in
new file mode 100644
index 00000000..f051ef07
--- /dev/null
+++ /MANIFEST.in
@@ -0,0 +1 @@
+include src/appfl/comm/grpc/credentials/*
