diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e215b73..6b85b2c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -319,15 +319,15 @@ set(CPPINTEROP_VERSION_MINOR "${CMAKE_MATCH_2}")
 set(CPPINTEROP_VERSION_PATCH "${CMAKE_MATCH_3}")
 
 configure_file(
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CppInterOpConfig.cmake.in
-  ${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/CppInterOpConfig.cmake
+  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CppInterOp/CppInterOpConfig.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/CppInterOp/CppInterOpConfig.cmake
   @ONLY)
 configure_file(
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CppInterOpConfigVersion.cmake.in
-  ${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/CppInterOpConfigVersion.cmake
+  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CppInterOp/CppInterOpConfigVersion.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/CppInterOp/CppInterOpConfigVersion.cmake
   @ONLY)
-install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/
-  DESTINATION lib/cmake
+install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/CppInterOp
+  DESTINATION lib/cmake/CppInterOp
   FILES_MATCHING
   PATTERN "*.cmake"
   )
diff --git a/cmake/CppInterOpConfig.cmake.in b/cmake/CppInterOp/CppInterOpConfig.cmake.in
similarity index 93%
rename from cmake/CppInterOpConfig.cmake.in
rename to cmake/CppInterOp/CppInterOpConfig.cmake.in
index 29e3fb20..f1bc63de 100644
--- a/cmake/CppInterOpConfig.cmake.in
+++ b/cmake/CppInterOp/CppInterOpConfig.cmake.in
@@ -4,6 +4,7 @@
 get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
 get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}" PATH)
 get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}" PATH)
+get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}" PATH)
 
 ### build/install workaround
 
diff --git a/cmake/CppInterOpConfigVersion.cmake.in b/cmake/CppInterOp/CppInterOpConfigVersion.cmake.in
similarity index 100%
rename from cmake/CppInterOpConfigVersion.cmake.in
rename to cmake/CppInterOp/CppInterOpConfigVersion.cmake.in
