diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b85b2c0..d8b340a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -312,7 +312,7 @@ include_directories(BEFORE SYSTEM
   )
 
 
-file(READ "VERSION" CPPINTEROP_VERSION)
+file(STRINGS "VERSION" CPPINTEROP_VERSION)
 string(REGEX MATCH "([0-9]*)\.([0-9]*)\.([0-9]*)" CPPINTEROP_VERSION_ONLY "${CPPINTEROP_VERSION}")
 set(CPPINTEROP_VERSION_MAJOR "${CMAKE_MATCH_1}")
 set(CPPINTEROP_VERSION_MINOR "${CMAKE_MATCH_2}")
diff --git a/cmake/CppInterOp/CppInterOpConfig.cmake.in b/cmake/CppInterOp/CppInterOpConfig.cmake.in
index f1bc63de..51f59ef0 100644
--- a/cmake/CppInterOp/CppInterOpConfig.cmake.in
+++ b/cmake/CppInterOp/CppInterOpConfig.cmake.in
@@ -10,14 +10,16 @@ get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}"
 
 if (IS_DIRECTORY "${CPPINTEROP_INSTALL_PREFIX}/include")
   set(_include "${CPPINTEROP_INSTALL_PREFIX}/include")
+  set(_libs "${CPPINTEROP_INSTALL_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}clangCppInterOp${CMAKE_SHARED_LIBRARY_SUFFIX}")
 else()
   set(_include "@CMAKE_CURRENT_SOURCE_DIR@/include")
+  set(_libs "@CMAKE_CURRENT_SOURCE_DIR@/lib/${CMAKE_SHARED_LIBRARY_PREFIX}clangCppInterOp${CMAKE_SHARED_LIBRARY_SUFFIX}")
 endif()
 
-if (IS_DIRECTORY "${CPPINTEROP_INSTALL_PREFIX}/cmake")
-  set(_cmake "${CPPINTEROP_INSTALL_PREFIX}/cmake")
+if (IS_DIRECTORY "${CPPINTEROP_INSTALL_PREFIX}/lib/cmake/CppInterOp")
+  set(_cmake "${CPPINTEROP_INSTALL_PREFIX}/lib/cmake/CppInterOp")
 else()
-  set(_cmake "@CMAKE_CURRENT_SOURCE_DIR@/cmake")
+  set(_cmake "@CMAKE_CURRENT_SOURCE_DIR@/cmake/CppInterOp")
 endif()
 
 ###
@@ -25,11 +27,13 @@ endif()
 set(CPPINTEROP_EXPORTED_TARGETS "clangCppInterOp")
 set(CPPINTEROP_CMAKE_DIR "${_cmake}")
 set(CPPINTEROP_INCLUDE_DIRS "${_include}")
+set(CPPINTEROP_LIBRARIES "${_libs}")
 
 # Provide all our library targets to users.
 add_library(clangCppInterOp SHARED IMPORTED)
-set_property(TARGET clangCppInterOp PROPERTY IMPORTED_LOCATION "${CPPINTEROP_DIR}/lib/libclangCppInterOp.so")
 set_property(TARGET clangCppInterOp PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${_include}")
+set_property(TARGET clangCppInterOp PROPERTY IMPORTED_LOCATION "${_libs}")
 
-unset(_include)
 unset(_cmake)
+unset(_include)
+unset(_libs)
