--- graphenelib-0.5.9.old/graphenebase/bip38.py	2017-11-15 11:19:11.000000000 +0100
+++ graphenelib-0.5.9.new/graphenebase/bip38.py	2018-02-14 13:41:47.960959200 +0100
@@ -7,9 +7,12 @@
 log = logging.getLogger(__name__)
 
 try:
-    from Crypto.Cipher import AES
+    from Cryptodome.Cipher import AES
 except ImportError:
-    raise ImportError("Missing dependency: pycrypto")
+    try:
+        from Crypto.Cipher import AES
+    except ImportError:
+        raise ImportError("Missing dependency: pyCryptodome")
 
 SCRYPT_MODULE = None
 if not SCRYPT_MODULE:
