--- graphenelib-0.5.9.old/graphenebase/bip38.py	2017-11-15 11:19:11.000000000 +0100
+++ graphenelib-0.5.9.new/graphenebase/bip38.py	2018-02-14 12:21:33.977995900 +0100
@@ -9,7 +9,10 @@
 try:
     from Crypto.Cipher import AES
 except ImportError:
-    raise ImportError("Missing dependency: pycrypto")
+    try:
+        from Cryptodome.Cipher import AES
+    except ImportError:
+        raise ImportError("Missing dependency: pyCryptodome")
 
 SCRYPT_MODULE = None
 if not SCRYPT_MODULE:
