--- graphenelib-0.5.9.old/setup.py	2018-02-05 14:39:12.000000000 +0100
+++ graphenelib-0.5.9.new/setup.py	2018-02-14 11:07:25.982660000 +0100
@@ -38,7 +38,7 @@
                       "requests",
                       "websocket-client",
                       "pylibscrypt",
-                      "pycryptodome",
+                      "pycryptodomex",
                       ],
     classifiers=['License :: OSI Approved :: MIT License',
                  'Operating System :: OS Independent',
