diff --git a/CMakeLists.txt b/CMakeLists.txt
index d85e3dc..3411dd1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,6 +30,9 @@ endif()
 # find_package(CUDA ${CUDA_MIN_VERSION} REQUIRED) #not-required
 find_package(Eigen3 3.3 REQUIRED)
 find_package(COLMAP REQUIRED)
+find_package(pybind11 REQUIRED)
+add_definitions(-DH5_USE_HALF_FLOAT)
+find_package(HighFive REQUIRED)
 
 ################################################################################
 # Select correct C/C++ standard
diff --git a/cmake/CMakeConfig.cmake.in b/cmake/CMakeConfig.cmake.in
index 82cb1e6..b5d909a 100644
--- a/cmake/CMakeConfig.cmake.in
+++ b/cmake/CMakeConfig.cmake.in
@@ -49,16 +49,9 @@ set(PIXSFM_INSTALL_PREFIX "${PIXSFM_INSTALL_PREFIX}/../..")
 set(PIXSFM_FOUND FALSE)
 
 # Find dependency packages.
-message(${PIXSFM_INSTALL_PREFIX}/share/pixsfm/third-party/cmake/)
 set(TEMP_CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH})
 set(CMAKE_MODULE_PATH ${PIXSFM_INSTALL_PREFIX}/share/pixsfm/cmake)
 
-set(pybind11_DIR
-    ${PIXSFM_INSTALL_PREFIX}/share/pixsfm/third-party/cmake/pybind11/)
-
-set(HighFive_DIR
-    ${PIXSFM_INSTALL_PREFIX}/share/pixsfm/third-party/HighFive/CMake/)
-
 if(PIXSFM_FIND_QUIETLY)
     find_package(COLMAP QUIET)
 
diff --git a/third-party/CMakeLists.txt b/third-party/CMakeLists.txt
index c9c513c..281e32d 100644
--- a/third-party/CMakeLists.txt
+++ b/third-party/CMakeLists.txt
@@ -6,10 +6,3 @@ set(CMAKE_INSTALL_DATAROOTDIR ${CMAKE_INSTALL_DATAROOTDIR}/pixsfm/third-party)
 # Add current path to include --> required for HighFive to find half.hpp
 set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH};${CMAKE_CURRENT_SOURCE_DIR})
 set(HIGHFIVE_USE_HALF_FLOAT ON)
-
-# Avoid building docs and examples
-set(HIGHFIVE_BUILD_DOCS OFF)
-set(HIGHFIVE_EXAMPLES OFF)
-
-add_subdirectory(HighFive)
-add_subdirectory(pybind11)
\ No newline at end of file
