diff --git a/src/napari_nd_annotator/_widgets/interpolation_widget.py b/src/napari_nd_annotator/_widgets/interpolation_widget.py
index f9898cb..64612fb 100644
--- a/src/napari_nd_annotator/_widgets/interpolation_widget.py
+++ b/src/napari_nd_annotator/_widgets/interpolation_widget.py
@@ -1,5 +1,5 @@
 import numpy as np
-from cv2 import cv2
+import cv2
 from napari import Viewer
 from napari.layers import Labels
 from scipy.interpolate import interp1d
