diff --git a/setup.py b/setup.py
index 568b97f..c1d3c9e 100755
--- a/setup.py
+++ b/setup.py
@@ -69,9 +69,9 @@ if USE_CYTHON:
     ext = 'pyx'
     write_version(os.path.join(PKG_DIR, 'version.pxi'), {
         '__version__': VERSION,
-        '__git_version__': git_version(),
-        '__croaring_version__': chdir(git_tag, 'CRoaring'),
-        '__croaring_git_version__': chdir(git_version, 'CRoaring'),
+        '__git_version__': '4ca8db8b2cbebd8ec353f71ea414947c609c0cb6',
+        '__croaring_version__': 'v0.2.65',
+        '__croaring_git_version__': '86de9b526c05cbd59deedc101c4f3aee99ddd905',
     })
 else:
     print('Building pyroaring from C sources.')
