diff --git a/wgpu/backends/rs.py b/wgpu/backends/rs.py
index a9b9370..05224b5 100644
--- a/wgpu/backends/rs.py
+++ b/wgpu/backends/rs.py
@@ -59,8 +59,8 @@ logger = logging.getLogger("wgpu")  # noqa
 apidiff = ApiDiff()
 
 # The wgpu-native version that we target/expect
-__version__ = "0.14.2.3"
-__commit_sha__ = "b6a188c6ac386b249d8decaf806e87946058383e"
+__version__ = "0.14.2.4"
+__commit_sha__ = "169b5168aeeced011e311578aa681eb01f33d5d0"
 version_info = tuple(map(int, __version__.split(".")))
 check_expected_version(version_info)  # produces a warning on mismatch
 
