diff --git a/setup.py b/setup.py
index ef993f1..9217431 100755
--- a/setup.py
+++ b/setup.py
@@ -29,7 +29,7 @@ BIND_PATH = path.join(HERE, "bindings")
 # See internal/compile_libs.sh for some detail.  Note that this is *not*
 # simply internal/*.cc
 src_files = [
-    path.join(CLD2_PATH, "internal/", i)
+    path.join("cld2/internal/", i)
     for i in (
         "cld2_generated_cjk_compatible.cc",
         "cld2_generated_deltaocta0122.cc",
@@ -58,7 +58,7 @@ src_files = [
     )
 ]
 src_files.extend(
-    [path.join(BIND_PATH, "pycldmodule.cc"), path.join(BIND_PATH, "encodings.cc")]
+    ["bindings/pycldmodule.cc", "bindings/encodings.cc"]
 )
 for i in src_files:
     if not path.exists(i):
