From 24786ae332bdf0203486b7af28e24d4352c2603d Mon Sep 17 00:00:00 2001
From: Jingbo Chen <cj@yanhuangdata.com>
Date: Mon, 29 Jan 2024 10:50:16 +0800
Subject: [PATCH] fix the pcre versio check error on clang 16

---
 cmake/pcre.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/pcre.cmake b/cmake/pcre.cmake
index e0acda5e7..ebc4a9cca 100644
--- a/cmake/pcre.cmake
+++ b/cmake/pcre.cmake
@@ -30,7 +30,7 @@ if (PCRE_BUILD_SOURCE)
     #if PCRE_MAJOR != ${PCRE_REQUIRED_MAJOR_VERSION} || PCRE_MINOR < ${PCRE_REQUIRED_MINOR_VERSION}
     #error Incorrect pcre version
     #endif
-    main() {}" CORRECT_PCRE_VERSION)
+    int main(void) {return 0;}" CORRECT_PCRE_VERSION)
     set (CMAKE_REQUIRED_INCLUDES "${saved_INCLUDES}")
 
     if (NOT CORRECT_PCRE_VERSION)
