From 0687127e0923819ebcd3a3cbfe2c146397a3f12a Mon Sep 17 00:00:00 2001
From: Klaus Zimmermann <klaus.zimmermann@quansight.com>
Date: Wed, 10 Apr 2024 10:35:31 +0200
Subject: [PATCH] Add find_package(hip)

---
 library/src/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/library/src/CMakeLists.txt b/library/src/CMakeLists.txt
index 24c1058..e6bef3e 100644
--- a/library/src/CMakeLists.txt
+++ b/library/src/CMakeLists.txt
@@ -36,6 +36,10 @@ endfunction( )
 
 add_executable( rocfft_rtc_helper rocfft_rtc_helper.cpp )
 
+set(CMAKE_FIND_DEBUG_MODE TRUE)
+find_package( hip REQUIRED )
+set(CMAKE_FIND_DEBUG_MODE FALSE)
+
 # each backend requires different libraries for host and device code
 if( USE_CUDA )
   set( ROCFFT_HOST_LINK_LIBS -lcuda )
-- 
2.40.1

