From 101e55ebac3a07093393f0dab9aef8380d34b551 Mon Sep 17 00:00:00 2001
From: Klaus Zimmermann <klaus.zimmermann@quansight.com>
Date: Wed, 19 Jun 2024 17:18:36 +0200
Subject: [PATCH] Disable ahead of time kernel compilation

---
 library/src/CMakeLists.txt | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/library/src/CMakeLists.txt b/library/src/CMakeLists.txt
index e6bef3e6..4ed75e98 100644
--- a/library/src/CMakeLists.txt
+++ b/library/src/CMakeLists.txt
@@ -534,7 +534,7 @@ add_custom_command(
   DEPENDS rocfft_aot_helper rocfft_rtc_helper
   COMMENT "Compile default kernels and solution-map kernels into shipped cache file"
 )
-add_custom_target( rocfft_kernel_cache_target ALL
+add_custom_target( rocfft_kernel_cache_target
   DEPENDS rocfft_kernel_cache.db
   VERBATIM
 )
@@ -578,12 +578,12 @@ else()
   set(ROCFFT_KERNEL_CACHE_INSTALL_DIR ${ROCM_INSTALL_LIBDIR}/rocfft)
 endif()
 
-if( NOT ENABLE_ASAN_PACKAGING )
-  rocm_install(FILES ${ROCFFT_KERNEL_CACHE_PATH}
-    DESTINATION "${ROCFFT_KERNEL_CACHE_INSTALL_DIR}"
-    COMPONENT runtime
-  )
-endif()
+# if( NOT ENABLE_ASAN_PACKAGING )
+#   rocm_install(FILES ${ROCFFT_KERNEL_CACHE_PATH}
+#     DESTINATION "${ROCFFT_KERNEL_CACHE_INSTALL_DIR}"
+#     COMPONENT runtime
+#   )
+# endif()
 
 # rtc helper is an internal library executable on Linux, placed in a
 # rocFFT subdirectory of the library directory.  On Windows it goes
-- 
2.40.1

