From 2f9a93ba078aca5443eb86481aa8956dc45cce73 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Mon, 17 Jun 2024 14:38:25 -0500
Subject: [PATCH] Fix mman include for MADV_HUGEPAGE

---
 shared/hostbuf.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/shared/hostbuf.h b/shared/hostbuf.h
index 0a96c7df..c370975f 100644
--- a/shared/hostbuf.h
+++ b/shared/hostbuf.h
@@ -28,6 +28,8 @@
 #ifndef WIN32
 #include <stdlib.h>
 #include <sys/mman.h>
+// GLIBC 2.17 requires linux/mman.h for MADV_HUGEPAGE
+#include <linux/mman.h>
 #endif
 
 // Simple RAII class for host buffers.  T is the type of pointer that
-- 
2.44.0

