From 435aee731df9bc86a5cad93073a177958caf8e8a Mon Sep 17 00:00:00 2001
From: Travis Collins <travis.collins@analog.com>
Date: Fri, 20 Jan 2023 14:18:09 -0700
Subject: [PATCH] Add algorithm headers for min/max on Windows

Signed-off-by: Travis Collins <travis.collins@analog.com>
---
 iiod/context/adalm2000/devices/m2k_dac.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/iiod/context/adalm2000/devices/m2k_dac.cpp b/iiod/context/adalm2000/devices/m2k_dac.cpp
index 5a32409..dfc278d 100644
--- a/iiod/context/adalm2000/devices/m2k_dac.cpp
+++ b/iiod/context/adalm2000/devices/m2k_dac.cpp
@@ -43,6 +43,7 @@
 #include "utils/utility.hpp"
 
 #include <vector>
+#include <algorithm>
 
 using namespace iio_emu;
 
-- 
2.38.1.windows.1

