From 53b57c79a209cd3ce3cd486f0a5864d27f3f6f0e Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Fri, 5 Jan 2024 17:23:52 +0530
Subject: [PATCH] No zig

---
 bindings/python/build.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bindings/python/build.py b/bindings/python/build.py
index 12c3ae73..73bf2840 100644
--- a/bindings/python/build.py
+++ b/bindings/python/build.py
@@ -117,6 +117,9 @@ def compile_target(target_name, target_settings, WHEELS_FOLDER, settings):
             "CFLAGS": "-stdlib=libc++"
         }
 
+    zig = ""
+    env = {}
+
     env["RUSTFLAGS"] = env.get("RUST_FLAGS", "") + " " + rust_flags
 
     environment = os.environ.copy()
-- 
2.42.0

