From b2da7e46e426cd28a2cfe5270802802fd9a1fb1c Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Fri, 22 Dec 2023 14:08:17 +0530
Subject: [PATCH] Fix readme path

---
 bindings/python/pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bindings/python/pyproject.toml b/bindings/python/pyproject.toml
index 5e609ea9f038..0f05de534bef 100644
--- a/bindings/python/pyproject.toml
+++ b/bindings/python/pyproject.toml
@@ -2,7 +2,7 @@
 name="ensmallen"
 version="0.8.96"
 repository= "https://github.com/AnacletoLAB/ensmallen"
-readme = "../../../README.rst"
+readme = "../../README.rst"
 description = "🍇 Ensmallen is the Rust/Python high-performance graph processing submodule of the GRAPE library."
 exclude = ["pytests/*", "Python-*", "./*.py"]
 keywords = ["python", "rust", "graph", "random", "walk"]
