From 033638a06b626fa3119d125fe3474af63eb7ad80 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Wed, 17 Nov 2021 12:53:58 +1000
Subject: [PATCH] Fix missing requirements.txt in pypi release

---
 setup.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 24c6b60..7150ecb 100755
--- a/setup.py
+++ b/setup.py
@@ -9,9 +9,6 @@
 with open('README.rst') as f:
     long_description = f.read()
 
-with open('requirements.txt') as f:
-    requirements = f.read().splitlines()
-
 setup(
     name='snntoolbox',
     version='0.6.0',  # see https://www.python.org/dev/peps/pep-0440/
@@ -47,7 +44,7 @@
 
     keywords='neural networks, deep learning, spiking',
 
-    install_requires=requirements,
+    install_requires=['tensorflow'],
 
     setup_requires=['pytest-runner'],
 
