--- /dev/null	2019-01-06 12:45:33.000000000 -0700
+++ CMakeLists.txt	2019-01-06 12:45:38.000000000 -0700
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.10)
+project(csmapi VERSION 3.0.3 DESCRIPTION "Community Sensor Model API")
+set(CMAKE_CXX_STANDARD 11)
+
+add_library(csmapi SHARED
+             BundleGM.cpp
+             CorrelationModel.cpp
+             csmPointCloud.cpp
+             Ellipsoid.cpp
+             FourParameterCorrelationModel.cpp
+             GeometricModel.cpp
+             Isd.cpp
+             LinearDecayCorrelationModel.cpp
+             ModelIdentifier.cpp
+             Plugin.cpp
+             PointCloudGM.cpp
+             PointCloudIsd.cpp
+             RasterGM.cpp
+             SettableEllipsoid.cpp
+             Version.cpp
+)
+
+file(GLOB HEADER_FILES ${PROJECT_SOURCE_DIR}/*.h)
+set_target_properties(csmapi PROPERTIES
+                    VERSION ${PROJECT_VERSION}
+                    SOVERSION 3
+                    PUBLIC_HEADER "${HEADER_FILES}"
+)
+
+target_include_directories(csmapi PUBLIC ${PROJECT_SOURCE_DIR})
+
+if(WIN32)
+  option(CMAKE_USE_WIN32_THREADS_INIT "using WIN32 threads" ON)
+  option(gtest_disable_pthreads "Disable uses of pthreads in gtest." ON)
+  install(TARGETS csmapi
+          LIBRARY
+            DESTINATION ${CMAKE_INSTALL_PREFIX}/Library/lib
+          RUNTIME
+            DESTINATION ${CMAKE_INSTALL_PREFIX}/Library/lib
+          )
+else()
+        message(STATUS "CMAKE INSTALL LIBDIR: ${CMAKE_INSTALL_PREFIX}")
+        install(TARGETS csmapi LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
+        install(FILES ${HEADER_FILES} DESTINATION ${CMAKE_INSTALL_PREFIX}/include/csm)
+endif()
