diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd3f290..e1de15f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,13 +12,13 @@ if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
 endif()
 
 option(BUILD_SHARED "Build shared library" ON)
-option(BUILD_STATIC "Build static library" ON)
+option(BUILD_STATIC "Build static library" OFF)
 option(BUILD_COVERAGE "Build with code coverage" OFF)
 
-if(${BUILD_CONTAINED_PYTHON}) 
+if(${BUILD_CONTAINED_PYTHON})
   #if make a mostly statically linked python module, use static boost
-   set(BUILD_STATIC ON)
-   set(Boost_USE_STATIC_LIBS ON)
+   set(BUILD_STATIC OFF)
+   set(Boost_USE_STATIC_LIBS OFF)
 endif()
 
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/Modules;${PROJECT_SOURCE_DIR}/cmake")
@@ -58,7 +58,7 @@ if(CMAKE_CXX_COMPILER_ID MATCHES GNU)
 	set(CMAKE_CXX_FLAGS_RELEASE "-O3 -g -fprofile-arcs -ftest-coverage")
     else()
         set(CMAKE_CXX_FLAGS_RELEASE "-O3 -g")
-    endif()	
+    endif()
 endif()
 
 if(CUDA_VERSION_MAJOR LESS 11)
