diff --git a/python/setup.py b/python/setup.py
index 1bd7851..e79c5b9 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -4,19 +4,19 @@ except ImportError:
     from distutils.core import setup
 import sys,os,subprocess
 
-def git_pep440_version(path):
-    def git_command(args):
-        prefix = ['git', '-C', path]
-        return subprocess.check_output(prefix + args).decode().strip()
-    version_full = git_command(['describe', '--tags'])
-    version_tag = git_command(['describe', '--tags', '--abbrev=0'])
-    version_tail = version_full[len(version_tag):]
-    return version_tag + version_tail.replace('-', '.dev', 1).replace('-', '+', 1)
+# def git_pep440_version(path):
+#     def git_command(args):
+#         prefix = ['git', '-C', path]
+#         return subprocess.check_output(prefix + args).decode().strip()
+#     version_full = git_command(['describe', '--tags'])
+#     version_tag = git_command(['describe', '--tags', '--abbrev=0'])
+#     version_tail = version_full[len(version_tag):]
+#     return version_tag + version_tail.replace('-', '.dev', 1).replace('-', '+', 1)
 
-try:
-    version = git_pep440_version(os.path.dirname(os.path.realpath(__file__)))
-except subprocess.CalledProcessError:
-    sys.exit('Cannot obtain version number from git.')
+# try:
+#     version = git_pep440_version(os.path.dirname(os.path.realpath(__file__)))
+# except subprocess.CalledProcessError:
+#     sys.exit('Cannot obtain version number from git.')
 
 if sys.version_info < (2,7):
   sys.exit('Sorry, Python < 2.7 is not supported')
@@ -31,7 +31,7 @@ setup(
   'generation has several options that facilitate obtaining good performance '
   'from machine learning algorithms, including features like data augmentation '
   'and resampling.',
-  version     = version, 
+  version     = "0.5.2",
   author='David Ryan Koes and Jocelyn Sunseri',
   author_email='dkoes@pitt.edu',
   classifiers=[
@@ -43,7 +43,7 @@ setup(
 
       'License :: OSI Approved :: Apache Software License',
       'License :: OSI Approved :: GNU General Public License v2 (GPLv2)',
-  
+
       'Programming Language :: C++',
       'Programming Language :: Python :: 3',
   ],
