--- src\spd\SPDCalcMetrics.cpp.orig	2017-07-06 09:16:20.051694000 +1000
+++ src\spd\SPDCalcMetrics.cpp	2017-07-06 09:19:06.347435200 +1000
@@ -370,7 +370,7 @@
         double lowThreshold = 0;
         double heightUpThreshold = 0;
         double heightLowThreshold = 0;
-
+/*
         try
         {
             if(metricElement->hasAttribute(metricNameXMLStr))
@@ -1759,7 +1759,8 @@
 		catch(SPDProcessingException &e)
 		{
 			throw e;
-		}
+		}*/
+        throw SPDProcessingException("SPD metrics not available on Windows yet");
 
         xercesc::XMLString::release(&metricadd);
         xercesc::XMLString::release(&metricminus);
