--- python/CMakeLists.txt.orig	2017-08-24 10:55:24.970881723 +1000
+++ python/CMakeLists.txt	2017-08-24 11:38:07.634972231 +1000
@@ -126,15 +126,35 @@
 SET(Boost_USE_STATIC_LIBS OFF)
 SET(Boost_USE_MULTITHREADED ON)
 SET(Boost_USE_STATIC_RUNTIME OFF)
+
+FIND_PACKAGE(PythonInterp) # so we can get PYTHON_VERSION_MAJOR for boost below
+IF(NOT PYTHONINTERP_FOUND)
+  MESSAGE(FATAL_ERROR "Unable to find Python")
+ENDIF()
+
 FIND_PACKAGE(Boost)
 IF(Boost_FOUND)
   INCLUDE_DIRECTORIES("${Boost_INCLUDE_DIRS}")
-  FIND_PACKAGE(Boost COMPONENTS python)
-  IF(Boost_PYTHON_FOUND)
-    MESSAGE("FOUND BoostPython: ${Boost_PYTHON_LIBRARY}")
-  ELSEIF(NOT Boost_PYTHON_FOUND)
-    MESSAGE(FATAL_ERROR "Unable to find correct Boost Python version. Did you set BOOST_ROOT?")
-  ENDIF()
+  MESSAGE("Python version ${PYTHON_VERSION_MAJOR}")
+  if (PYTHON_VERSION_MAJOR GREATER 2) # python3
+    MESSAGE("Searching for python3")
+    FIND_PACKAGE(Boost COMPONENTS python3)
+    IF(Boost_PYTHON3_FOUND)
+      MESSAGE("FOUND BoostPython: ${Boost_PYTHON3_LIBRARY}")
+    ELSEIF(NOT Boost_PYTHON3_FOUND)
+      MESSAGE(FATAL_ERROR "Unable to find correct Boost Python3 version. Did you set BOOST_ROOT?")
+    ENDIF()
+
+  else()
+    MESSAGE("Searching for python")
+    FIND_PACKAGE(Boost COMPONENTS python)
+    IF(Boost_PYTHON_FOUND)
+      MESSAGE("FOUND BoostPython: ${Boost_PYTHON_LIBRARY}")
+    ELSEIF(NOT Boost_PYTHON_FOUND)
+      MESSAGE(FATAL_ERROR "Unable to find correct Boost Python version. Did you set BOOST_ROOT?")
+    ENDIF()
+  endif()
+
 ELSEIF(NOT Boost_FOUND)
   MESSAGE(FATAL_ERROR "Unable to find correct Boost version. Did you set BOOST_ROOT?")
 ENDIF()
