diff --git a/bin/meld b/bin/meld
index 576ca374..7e87df6b 100755
--- a/bin/meld
+++ b/bin/meld
@@ -135,9 +135,12 @@ try:
             libintl = cdll['libintl-8']
         else:
             try:
-                libintl = cdll.intl
+                libintl = cdll['intl']
             except FileNotFoundError:
-                libintl = cdll['libintl-8']
+                try:
+                    libintl = cdll['libintl-8']
+                except FileNotFoundError:
+                    libintl = cdll['intl-8']
         libintl.bindtextdomain(locale_domain, str(locale_dir))
         libintl.bind_textdomain_codeset(locale_domain, 'UTF-8')
         libintl.textdomain(locale_domain)
