diff --git a/setup.py b/setup.py
index cba385d..b1e1509 100644
--- a/setup.py
+++ b/setup.py
@@ -194,7 +194,7 @@ def blas_setup(roots, library_names, headers, extra_entries_on_success):
 def openblas_setup(roots):
     return blas_setup(
         roots,
-        ("openblas",),
+        ("cblas", "lapacke"),
         ("cblas.h", "lapacke.h",),
         {"DEFINES": [("HAVE_OPENBLAS", None)]},
     )
