diff --git a/src/matrix/kaldi-blas.h b/src/matrix/kaldi-blas.h
index 477239b..20fb3b8 100644
--- a/src/matrix/kaldi-blas.h
+++ b/src/matrix/kaldi-blas.h
@@ -2,6 +2,8 @@
 
 // Copyright 2009-2011  Ondrej Glembek;  Microsoft Corporation
 
+// Modified 2021 by Sean Robertson, listed.
+
 // See ../../COPYING for clarification regarding multiple authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
@@ -96,6 +98,15 @@
 #elif defined(HAVE_OPENBLAS)
   // getting cblas.h and lapacke.h from <openblas-install-dir>/.
   // putting in "" not <> to search -I before system libraries.
+  
+#ifdef _MSC_VER
+  // sdrobert: following
+  // https://stackoverflow.com/questions/24853450/errors-using-lapack-c-header-in-c-with-visual-studio-2010.
+  // It'll spew a lot of errors, but should be okay.
+  #include <complex>
+  #define lapack_complex_float std::complex<float>
+  #define lapack_complex_double std::complex<double>
+#endif
 
   #include "cblas.h"
   #include "lapacke.h"
