diff --git a/src/matrix/kaldi-blas.h b/src/matrix/kaldi-blas.h
index 477239b..2977d84 100644
--- a/src/matrix/kaldi-blas.h
+++ b/src/matrix/kaldi-blas.h
@@ -96,6 +96,13 @@
 #elif defined(HAVE_OPENBLAS)
   // getting cblas.h and lapacke.h from <openblas-install-dir>/.
   // putting in "" not <> to search -I before system libraries.
+  
+#ifdef _MSC_VER
+  // https://stackoverflow.com/questions/24853450/errors-using-lapack-c-header-in-c-with-visual-studio-2010
+  #include <complex>
+  #define lapack_complex_float std::complex<float>
+  #define lapack_complex_double std::complex<double>
+#endif
 
   #include "cblas.h"
   #include "lapacke.h"
