--- meson.build	2025-09-24 12:12:33.584569358 +0200
+++ meson.build	2025-09-24 12:25:53.554143689 +0200
@@ -37,6 +37,46 @@
   'AdsLib/standalone/NotificationDispatcher.cpp',
 ])
 
+ads_headers = files(
+  'AdsLib/AdsDef.h',
+  'AdsLib/AdsDevice.h',
+  'AdsLib/AdsException.h',
+  'AdsLib/AdsFile.h',
+  'AdsLib/AdsLib.h',
+  'AdsLib/AdsNotification.h',
+  'AdsLib/AdsNotificationOOI.h',
+  'AdsLib/AdsVariable.h',
+  'AdsLib/AmsConnection.h',
+  'AdsLib/AmsHeader.h',
+  'AdsLib/AmsPort.h',
+  'AdsLib/AmsRouter.h',
+  'AdsLib/Frame.h',
+  'AdsLib/LicenseAccess.h',
+  'AdsLib/Log.h',
+  'AdsLib/NotificationDispatcher.h',
+  'AdsLib/RegistryAccess.h',
+  'AdsLib/RingBuffer.h',
+  'AdsLib/RouterAccess.h',
+  'AdsLib/Router.h',
+  'AdsLib/RTimeAccess.h',
+  'AdsLib/Semaphore.h',
+  'AdsLib/Sockets.h',
+  'AdsLib/SymbolAccess.h',
+  'AdsLib/wrap_endian.h',
+  'AdsLib/wrap_registry.h',
+  'AdsLib/wrap_socket.h',
+  'AdsLib/bhf/ParameterList.h',
+  'AdsLib/bhf/StringToInteger.h',
+  'AdsLib/bhf/WindowsQuirks.h',
+  'AdsLib/standalone/AdsDef.h',
+  'AdsLib/standalone/AdsLib.h',
+  'AdsLib/TwinCAT/AdsDef.h',
+  'AdsLib/TwinCAT/AdsLib.h'
+)
+
+
+install_headers(ads_headers, subdir: 'AdsLib')
+
 inc = include_directories([
   'AdsLib',
   'tools',
@@ -62,6 +102,8 @@
   include_directories: inc,
   dependencies: libs,
   link_with: adslib,
+  install: true,
+  install_dir: get_option('bindir'),
 )
 
 adslibooitest = executable('AdsLibOOITest',
@@ -69,6 +111,8 @@
   include_directories: inc,
   dependencies: libs,
   link_with: adslib,
+  install: true,
+  install_dir: get_option('bindir'),
 )
 
 adstool = executable('adstool',
@@ -76,6 +120,8 @@
   include_directories: inc,
   dependencies: libs,
   link_with: adslib,
+  install: true,
+  install_dir: get_option('bindir'),
 )
 
 if get_option('tcadsdll_include') != ''
@@ -86,6 +132,8 @@
       get_option('tcadsdll_include'),
       inc,
     ],
+    install: true,
+    install_dir: get_option('libdir'),
   )
 
   tcadsdll_deps = [
@@ -104,5 +152,7 @@
       tcadsdll_deps
     ],
     link_with: tcadslib,
+    install: true,
+    install_dir: get_option('bindir'),
   )
 endif
