diff --git a/src/LSDJunctionNetwork.cpp b/src/LSDJunctionNetwork.cpp
index 95d49b8..8158977 100644
--- a/src/LSDJunctionNetwork.cpp
+++ b/src/LSDJunctionNetwork.cpp
@@ -3704,7 +3704,10 @@ void LSDJunctionNetwork::write_valley_hilltop_chi_profiles_to_csv(vector<int> so
 		// get the LSDChannel
 		LSDChannel new_channel(hilltop_node, final_node, downslope_chi, m_over_n, A_0, FlowInfo, ElevationRaster);
 		// write to csv
-		string jn_str = static_cast<ostringstream*>( &(ostringstream() << source_junction) )->str();
+		ostringstream tmp_str;
+		tmp_str << source_junction;
+		string jn_str = static_cast<ostringstream*>( &tmp_str )->str();
+		// string jn_str = static_cast<ostringstream*>( &(ostringstream() << source_junction) )->str();
 		string output_csv_filename = DEM_ID+"_chan_profile_"+jn_str;
 		new_channel.write_channel_to_csv(output_path, output_csv_filename, FlowDistance);
 	}
