From 32cadc5c6847c72fb90b83ea694c0025dcb6628b Mon Sep 17 00:00:00 2001
From: Tim Paine <3105306+timkpaine@users.noreply.github.com>
Date: Thu, 7 Mar 2024 11:12:55 -0500
Subject: [PATCH] remove old cmake args

---
 setup.py | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/setup.py b/setup.py
index a968daf..d5abe38 100644
--- a/setup.py
+++ b/setup.py
@@ -44,14 +44,6 @@ if CSP_USE_VCPKG and os.path.exists(vcpkg_toolchain_file):
 else:
     cmake_args.append("-DCSP_USE_VCPKG=OFF")
 
-
-# if "CONDA_PREFIX" in os.environ:
-#     cmake_args.append(f"-DCMAKE_MODULE_PATH={os.environ['CONDA_PREFIX']}/lib/cmake/absl;{os.environ['CONDA_PREFIX']}/lib/cmake/arrow")
-
-if "CMAKE_ARGS" in os.environ:
-    # conda
-    cmake_args.extend(os.environ["CMAKE_ARGS"].split(" "))
-
 if "CXX" in os.environ:
     cmake_args.append(f"-DCMAKE_CXX_COMPILER={os.environ['CXX']}")
 
-- 
2.43.2

