diff --git a/PKG-INFO b/PKG-INFO
index 40355c0..e0f6a39 100644
--- a/PKG-INFO
+++ b/PKG-INFO
@@ -14,7 +14,6 @@ Project-URL: Changelog, https://mesonpy.readthedocs.io/en/latest/changelog.html
 Requires-Python: >=3.7
 Requires-Dist: colorama; os_name == "nt"
 Requires-Dist: meson>=0.60.0
-Requires-Dist: ninja
 Requires-Dist: pep621>=0.3.0
 Requires-Dist: tomli>=1.0.0
 Requires-Dist: typing-extensions>=3.7.4; python_version < "3.8"
diff --git a/pyproject.toml b/pyproject.toml
index d61a210..b66565a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,7 +3,6 @@ build-backend = 'mesonpy'
 backend-path = ['.']
 requires = [
   'meson>=0.62.0',
-  'ninja',
   'pep621>=0.3.0',
   'tomli>=1.0.0',
   'typing-extensions>=3.7.4; python_version<"3.8"',
@@ -28,7 +27,6 @@ classifiers = [
 dependencies = [
   'colorama; os_name == "nt"',
   'meson>=0.60.0',
-  'ninja',
   'pep621>=0.3.0', # not a hard dependency, only needed for projects that use PEP 621 metadata
   'tomli>=1.0.0',
   'typing-extensions>=3.7.4; python_version<"3.8"',
