--- swig/python/setup.py	2015-09-16 07:21:17.000000000 -0300
+++ swig/python/setup.py	2015-12-11 08:55:33.843206824 -0300
@@ -201,6 +201,11 @@
 extra_link_args = []
 extra_compile_args = []
 
+if sys.platform == 'darwin' and [int(x) for x in os.uname()[2].split('.')] >= [11, 0, 0]:
+    # Since MacOS X 10.9, clang no longer accepts -mno-fused-madd.
+    os.environ['ARCHFLAGS'] = '-Wno-error=unused-command-line-argument-hard-error-in-future'
+    extra_compile_args.append('-Qunused-arguments')
+
 gdal_module = Extension('osgeo._gdal',
                         sources=['extensions/gdal_wrap.cpp'],
                         extra_compile_args = extra_compile_args,
