--- swig/python/setup.py	2016-01-25 10:03:50.000000000 -0300
+++ swig/python/setup.py	2016-03-06 15:09:56.361087947 -0300
@@ -242,7 +242,7 @@
     py_modules.append('gdalnumeric')
 
 packages = ["osgeo",]
-
+scripts = glob("./scripts/*.py")
 readme = str(open('README.txt','rb').read())
 
 name = 'GDAL'
@@ -292,6 +292,7 @@
            classifiers = classifiers,
            py_modules = py_modules,
            packages = packages,
+           scripts = scripts,
            url=url,
            data_files = data_files,
            zip_safe = False,
@@ -312,6 +313,7 @@
            classifiers = classifiers,
            py_modules = py_modules,
            packages = packages,
+           scripts = scripts,
            data_files = data_files,
            url=url,
            cmdclass={'build_ext':gdal_ext,
