--- swig/python/osgeo/__init__.py	2015-09-16 07:21:17.000000000 -0300
+++ swig/python/osgeo/__init__.py	2015-12-11 08:50:49.052192641 -0300
@@ -1,6 +1,10 @@
 # __init__ for osgeo package.
 
 # making the osgeo package version the same as the gdal version:
+import os
+import sys
+os.environ['PATH'] = r'%s\Library\bin;%s' % (sys.prefix, os.environ['PATH'])
+
 from sys import version_info
 if version_info >= (2,6,0):
     def swig_import_helper():
