diff --git a/a2x.py b/a2x.py
index 66ae799..78177e8 100755
--- a/a2x.py
+++ b/a2x.py
@@ -649,7 +649,7 @@ class A2X(AttrDict):
         shell('"%s" --backend docbook -a "a2x-format=%s" %s --out-file "%s" "%s"' %
              (self.asciidoc, self.format, self.asciidoc_opts, docbook_file, self.asciidoc_file))
         if not self.no_xmllint and XMLLINT:
-            shell('"%s" --nonet --noout --valid "%s"' % (XMLLINT, docbook_file))
+            shell('"%s" --noout --valid "%s"' % (XMLLINT, docbook_file))
 
     def to_xhtml(self):
         self.to_docbook()
