diff --git a/asciidoc.py b/asciidoc.py
index 09692b0..1a15ae8 100755
--- a/asciidoc.py
+++ b/asciidoc.py
@@ -6209,7 +6209,7 @@ def execute(cmd,opts,args):
     finally:
         sys.stdin,sys.stdout = stdin,stdout
 
-if __name__ == '__main__':
+def main():
     # Process command line options.
     import getopt
     try:
@@ -6263,3 +6263,6 @@ if __name__ == '__main__':
             execute(sys.argv[0],opts,args)
         except KeyboardInterrupt:
             sys.exit(1)
+
+if __name__ == '__main__':
+    main()
