diff --git a/raster/test/regress/check_gdal.sql b/raster/test/regress/check_gdal.sql
index 92e6f90..09b6387 100644
--- a/raster/test/regress/check_gdal.sql
+++ b/raster/test/regress/check_gdal.sql
@@ -5,13 +5,14 @@ SELECT
 			THEN false
 		ELSE NULL
 	END;
-SET postgis.gdal_datapath = '';
-SELECT
-	CASE
-		WHEN strpos(postgis_gdal_version(), 'GDAL_DATA') <> 0
-			THEN NULL
-		ELSE TRUE
-	END;
+-- conda-forge's GDAL is not compiled with GDAL_DATA preset
+-- SET postgis.gdal_datapath = '';
+-- SELECT
+-- 	CASE
+-- 		WHEN strpos(postgis_gdal_version(), 'GDAL_DATA') <> 0
+-- 			THEN NULL
+-- 		ELSE TRUE
+-- 	END;
 SET postgis.gdal_datapath = default;
 SELECT
 	CASE
