diff --git a/src/ImathTest/CMakeLists.txt b/src/ImathTest/CMakeLists.txt
index 19a630a525..709a32385f 100644
--- a/src/ImathTest/CMakeLists.txt
+++ b/src/ImathTest/CMakeLists.txt
@@ -70,11 +70,7 @@ if(NOT "${CMAKE_PROJECT_NAME}" STREQUAL "ImathTest")
 
 endif()
 
-add_executable(ImathHalfPerfTest half_perf_test.cpp)
-set_target_properties(ImathHalfPerfTest PROPERTIES
-RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
-)
-target_link_libraries(ImathHalfPerfTest Imath::Imath)
+# ImathHalfPerfTest not usable on macOS (no CLOCK_MONOTONIC)
 
 function(DEFINE_IMATH_TESTS)
   foreach(curtest IN LISTS ARGN)
