diff --git a/src/github.com/cockroachdb/cockroach/c-deps/jemalloc/include/jemalloc/internal/jemalloc_internal_decls.h b/src/github.com/cockroachdb/cockroach/c-deps/jemalloc/include/jemalloc/internal/jemalloc_internal_decls.h
index c907d910..a4bbac59 100644
--- a/src/github.com/cockroachdb/cockroach/c-deps/jemalloc/include/jemalloc/internal/jemalloc_internal_decls.h
+++ b/src/github.com/cockroachdb/cockroach/c-deps/jemalloc/include/jemalloc/internal/jemalloc_internal_decls.h
@@ -39,7 +39,6 @@
 #include <stdarg.h>
 #include <stdbool.h>
 #include <stdio.h>
-#include <stdlib.h>
 #include <stdint.h>
 #include <stddef.h>
 #ifndef offsetof
