--- configure.ac.old    2017-02-25 11:12:44.106932848 +0530
+++ configure.ac        2017-02-25 11:13:07.990703576 +0530
@@ -962,7 +962,6 @@
     else
         # Normal Sage packages
         echo >&7 "\$(INST)/$PKG_VERSION:$DEPS"
-        echo >&7 "	+\$(A""M_V_at)sage-logger -p '\$(SAGE_SPKG) $PKG_VERSION' '\$(SAGE_LOGS)/$PKG_VERSION.log'"
         echo >&7

         # Add a target with just the bare package name for "sage -i"
