From 8e6b2d3219b19201bb8c02ff91e1367100462593 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Fri, 3 Feb 2017 16:07:08 +0530
Subject: [PATCH] Fix cython bug

---
 src/setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/setup.py b/src/setup.py
index 4d20a0f..25d9d79 100755
--- a/src/setup.py
+++ b/src/setup.py
@@ -409,6 +409,7 @@ class sage_build_ext(build_ext):
                 'fast_getattr': True,
                 'profile': profile,
             },
+            include_path=sys.path,
             # Debugging
             gdb_debug=debug,
             output_dir=SAGE_CYTHONIZED,
-- 
2.7.4

