diff --git a/src/module_list.py b/src/module_list.py
index 96e8081..e670380 100644
--- a/src/module_list.py
+++ b/src/module_list.py
@@ -45,6 +45,7 @@ givaro_cflags = pkgconfig.cflags('givaro').split()
 gsl_pc = pkgconfig.parse('gsl')
 gsl_libs = gsl_pc['libraries']
 gsl_libs.remove('gslcblas')
+gsl_libs += cblas_libs
 gsl_library_dirs = gsl_pc['library_dirs']
 gsl_include_dirs = gsl_pc['include_dirs']
