diff --git a/src/Makefile.in b/src/Makefile.in
index ca6fbe8..51d6993 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -40,7 +40,7 @@ sage:
 		   SAGE_BUILD_DIR=/doesnotexist							\
 		   SAGE_PKGS=$(abs_top_srcdir)/build/pkgs					\
 		   SAGE_CYTHONIZED=$(abs_builddir)/build/cythonized				\
-	 && python -u setup.py build install)
+	 && python -u setup.py build && (python -u setup.py install > /dev/null 2 >&1))

 clean:
 	@echo "Deleting Sage library build artifacts..."
