--- configure.ac.old    2017-02-25 11:12:44.106932848 +0530
+++ configure.ac        2017-02-25 11:13:07.990703576 +0530
@@ -89,7 +89,7 @@ SAGE_EXTCODE="$SAGE_SHARE/sage/ext"
 SAGE_SPKG_INST="$SAGE_LOCAL/var/lib/sage/installed"
 
 #---------------------------------------------------------
-AX_CHECK_ROOT([AC_MSG_ERROR([You cannot build Sage as root, switch to an unpriviledged user])], [])
+#AX_CHECK_ROOT([AC_MSG_ERROR([You cannot build Sage as root, switch to an unpriviledged user])], [])
 
 # Check whether we are on a supported platform
 AC_CANONICAL_BUILD()
