diff --git a/src/bin/sage b/src/bin/sage
index cb3b611..b3ab56c 100755
--- a/src/bin/sage
+++ b/src/bin/sage
@@ -363,7 +363,7 @@ fi
 # Prepare for running Sage, either interactively or non-interactively.
 sage_setup() {
     # Check that we're not in a source tarball which hasn't been built yet (#13561).
-    if [ ! -d "$SAGE_LOCAL/lib/python/site-packages/sage" ]; then
+    if [ ! -d "$SAGE_LOCAL/lib/python*/site-packages/sage" ]; then
         echo >&2 '************************************************************************'
         echo >&2 'It seems that you are attempting to run Sage from an unpacked source'
         echo >&2 'tarball, but you have not compiled it yet (or maybe the build has not'
