diff --git a/cmake/Modules/FindTPLCUBLAS.cmake b/cmake/Modules/FindTPLCUBLAS.cmake
index 4cb0101..8e9e95b 100644
--- a/cmake/Modules/FindTPLCUBLAS.cmake
+++ b/cmake/Modules/FindTPLCUBLAS.cmake
@@ -2,7 +2,9 @@ find_package(CUDAToolkit)
 
 if(CUDAToolkit_FOUND)
   get_target_property(kk_cublas_include_dir_list CUDA::cublas INTERFACE_INCLUDE_DIRECTORIES)
-  list(GET kk_cublas_include_dir_list 0 kk_cublas_include_dir)
+  if(KokkosKernels_CUDA_MATH_INCLUDE_DIR)
+    list(PREPEND kk_cublas_include_dir_list ${KokkosKernels_CUDA_MATH_INCLUDE_DIR})
+  endif()
   get_target_property(kk_cublas_library CUDA::cublas IMPORTED_LOCATION)
   get_filename_component(kk_cublas_library_dir ${kk_cublas_library} DIRECTORY)
   kokkoskernels_find_imported(CUBLAS INTERFACE
@@ -11,7 +13,7 @@ if(CUDAToolkit_FOUND)
     LIBRARIES cublas
     LIBRARY_PATHS ${kk_cublas_library_dir}
     HEADER cublas.h
-    HEADER_PATHS ${kk_cublas_include_dir}
+    HEADER_PATHS ${kk_cublas_include_dir_list}
   )
 elseif(CUBLAS_LIBRARIES AND CUBLAS_LIBRARY_DIRS AND CUBLAS_INCLUDE_DIRS)
   message(WARNING "CUBLAS_LIBRARIES and CUBLAS_LIBRARY_DIRS and CUBLAS_INCLUDE_DIRS are deprecated. Use CUDAToolkit_ROOT to guide CMake's built-in search.")
