diff --git a/src/nlohmann/json.hpp b/src/nlohmann/json.hpp
--- a/src/nlohmann/json.hpp
+++ b/src/nlohmann/json.hpp
@@ -7,9 +7,10 @@
 
 // LAMMPS CUSTOMIZATION: to keep users from including this file directly
 
-#if !defined(LMP_JSON_H)
-#error Please include "json.h" instead of "nlohmann/json.hpp"
-#endif
+#if !defined(LMP_JSON_INCLUDE_PRIVATE)
+#include_next <nlohmann/json.hpp>
+#else
+// LAMMPS uses its private nlohmann_lmp namespace below.
 
 /****************************************************************************\
  * Note on documentation: The source files contain links to the online      *
@@ -25267,3 +25268,5 @@ namespace std
 
 
 #endif  // LMP_NLOHMANN_JSON_HPP_
+
+#endif  // LMP_JSON_INCLUDE_PRIVATE
diff --git a/src/nlohmann/json_fwd.hpp b/src/nlohmann/json_fwd.hpp
--- a/src/nlohmann/json_fwd.hpp
+++ b/src/nlohmann/json_fwd.hpp
@@ -7,9 +7,10 @@
 
 // LAMMPS CUSTOMIZATION: to keep users from including this file directly
 
-#if !defined(LMP_JSON_FWD_H)
-#error Please include "json_fwd.h" instead of "nlohmann/json_fwd.hpp"
-#endif
+#if !defined(LMP_JSON_FWD_INCLUDE_PRIVATE)
+#include_next <nlohmann/json_fwd.hpp>
+#else
+// LAMMPS uses its private nlohmann_lmp namespace below.
 
 #ifndef LMP_NLOHMANN_JSON_FWD_HPP_
 #define LMP_NLOHMANN_JSON_FWD_HPP_
@@ -193,2 +194,4 @@
 
 #endif  // LMP_NLOHMANN_JSON_FWD_HPP_
+
+#endif  // LMP_JSON_FWD_INCLUDE_PRIVATE
diff --git a/src/json.h b/src/json.h
--- a/src/json.h
+++ b/src/json.h
@@ -17,7 +17,9 @@
 // wrapper around including the JSON parsing and writing class
 // Do NOT include in any header file
 
+#define LMP_JSON_INCLUDE_PRIVATE
 #include "nlohmann/json.hpp" // IWYU pragma: export
+#undef LMP_JSON_INCLUDE_PRIVATE
 
 namespace LAMMPS_NS {
 using json = ::nlohmann_lmp::basic_json<nlohmann_lmp::ordered_map>;
diff --git a/src/json_fwd.h b/src/json_fwd.h
--- a/src/json_fwd.h
+++ b/src/json_fwd.h
@@ -17,7 +17,9 @@
 // Forward declarations for header-only JSON class
 // For use in headers
 
+#define LMP_JSON_FWD_INCLUDE_PRIVATE
 #include "nlohmann/json_fwd.hpp" // IWYU pragma: export
+#undef LMP_JSON_FWD_INCLUDE_PRIVATE
 
 namespace LAMMPS_NS {
 using json = ::nlohmann_lmp::basic_json<nlohmann_lmp::ordered_map>;
