diff --git a/source/include/acpixf.h b/source/include/acpixf.h
old mode 100644
new mode 100755
index 64c6b924f..d35e44897
--- a/source/include/acpixf.h
+++ b/source/include/acpixf.h
@@ -154,7 +154,7 @@
 
 /* Current ACPICA subsystem version in YYYYMMDD format */
 
-#define ACPI_CA_VERSION                 0x20250404  
+#define ACPI_CA_VERSION                 0x20250404
 
 #include "acconfig.h"
 #include "actypes.h"
