From 450aedd7d6c0662f0793510556efe720389cf413 Mon Sep 17 00:00:00 2001
From: "Lori A. Burns" <lori.burns@gmail.com>
Date: Wed, 24 May 2023 03:52:39 -0400
Subject: [PATCH 5/5] avoid git for c-f

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d572eb..3991e02 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,8 +57,8 @@ find_package(Python COMPONENTS Interpreter REQUIRED)
 get_filename_component(_compiler_name ${CMAKE_Fortran_COMPILER} NAME)
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/version.f90
-  COMMAND ${Python_EXECUTABLE} version.py ../VERSION ${CMAKE_CURRENT_BINARY_DIR}/version.f90 ${_compiler_name}
-  #COMMAND ${Python_EXECUTABLE} version.py ../VERSION ${CMAKE_CURRENT_BINARY_DIR}/version.f90 ${_compiler_name} --commit c2e0b548  # TODO
+  #COMMAND ${Python_EXECUTABLE} version.py ../VERSION ${CMAKE_CURRENT_BINARY_DIR}/version.f90 ${_compiler_name}
+  COMMAND ${Python_EXECUTABLE} version.py ../VERSION ${CMAKE_CURRENT_BINARY_DIR}/version.f90 ${_compiler_name} --commit c2e0b548  # TODO
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src
   COMMENT "Generating version info"
   )
-- 
2.40.1

