From 3288fee587ba66bf1e40e2b2e070e026d050aa78 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Tue, 1 Apr 2025 08:51:37 -0700
Subject: [PATCH] Fix include path

---
 setup.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/setup.py b/setup.py
index 1017af6..89d8aff 100644
--- a/setup.py
+++ b/setup.py
@@ -112,6 +112,12 @@ if isWindows:
 # this must be after use_pkg_config()!
 include_dirs += [numpy.get_include()]
 
+if useSystemLibraw:
+    prefix = os.environ.get("PREFIX")
+    if not prefix:
+        raise RuntimeError("Environment variable PREFIX is not defined; this package must be built in a conda environment.")
+    include_dirs.append(os.path.join(prefix, "include", "libraw"))
+
 def clone_submodules():
     if not os.path.exists('external/LibRaw/README.md'):
         print('LibRaw git submodule is not cloned yet, will invoke "git submodule update --init" now')
-- 
2.49.0.windows.1

