From 25a8798e38e6e0a576cc78916dbce30a2aa25d1e Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Tue, 1 Apr 2025 08:51:37 -0700
Subject: [PATCH] Fix include path

---
 setup.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/setup.py b/setup.py
index 1017af6..18513b3 100644
--- a/setup.py
+++ b/setup.py
@@ -112,6 +112,13 @@ if isWindows:
 # this must be after use_pkg_config()!
 include_dirs += [numpy.get_include()]
 
+if useSystemLibraw:
+    prefix = os.environ.get("PREFIX")
+    if not prefix:
+        raise RuntimeError("Environment variable PREFIX is not defined; this package must be built in a conda environment.")
+    include_dirs += [os.path.join(prefix, "include", "libraw")]
+    library_dirs += [os.path.join(prefix, "lib")]
+
 def clone_submodules():
     if not os.path.exists('external/LibRaw/README.md'):
         print('LibRaw git submodule is not cloned yet, will invoke "git submodule update --init" now')
-- 
2.49.0.windows.1

