From f17dd5bb5fa245c6ed4e7707dc6ea42a8c1d4265 Mon Sep 17 00:00:00 2001
From: Matus Kosut <matus.kosut@ntnu.no>
Date: Mon, 26 Oct 2020 11:01:35 +0100
Subject: [PATCH] Patch CMakeLists.txt for conda build

---
 CMakeLists.txt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75b8238..64e3863 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,11 +17,14 @@ if(CGET_PREFIX)
     link_directories(${CGET_PREFIX}/lib)
 endif()
 
+include_directories("include" "${SRC_DIR}/include" "${PREFIX}/include" "${PREFIX}/include/shrinkwrap" "${PREFIX}/include/htslib" "${SYS_PREFIX}/include")
+link_directories("lib" "${SRC_DIR}/lib" "${PREFIX}/lib" "${SYS_PREFIX}/lib")
+
 #list(APPEND CMAKE_PREFIX_PATH ${CMAKE_INSTALL_PREFIX})
 #find_package(shrinkwrap CONFIG REQUIRED)
-#if (NOT BUILD_SHARED_LIBS)
+if (NOT BUILD_SHARED_LIBS)
     set(CMAKE_FIND_LIBRARY_SUFFIXES ".a;${CMAKE_FIND_LIBRARY_SUFFIXES}") # Prefer libz.a when both are available
-#endif()
+endif()
 find_library(HTS_LIBRARY hts)
 find_library(ZLIB_LIBRARY z)
 find_library(ZSTD_LIBRARY zstd)
-- 
2.23.0

