From 993be0583e938f12b3ef0508d6cc231f1b345d0c Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 21 Nov 2021 11:20:22 +1100
Subject: [PATCH 2/2] look for zmq.hpp, not zmq.h

---
 cmake/Modules/FindZMQ.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Modules/FindZMQ.cmake b/cmake/Modules/FindZMQ.cmake
index 1131909..13fa1d0 100644
--- a/cmake/Modules/FindZMQ.cmake
+++ b/cmake/Modules/FindZMQ.cmake
@@ -5,7 +5,7 @@
 # ZMQ_LIBRARIES - The libraries needed to use ZMQ
 # ZMQ_DEFINITIONS - Compiler switches required for using ZMQ
 
-find_path ( ZMQ_INCLUDE_DIR zmq.h )
+find_path ( ZMQ_INCLUDE_DIR zmq.hpp )
 find_library ( ZMQ_LIBRARY NAMES zmq )
 
 set ( ZMQ_LIBRARIES ${ZMQ_LIBRARY} )
-- 
2.32.0.windows.2

