From 47119e1ad7daa7b19b27b2ae7b55cb58efbe4f15 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 21 Nov 2021 16:06:56 +1100
Subject: [PATCH 2/3] add patch_level to version_string

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff6356b..a7476ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,7 @@ project(tsnecuda)
 set (CMAKE_PROJECT_VERSION 3)
 set (CMAKE_PROJECT_VERSION_MAJOR 0)
 set (CMAKE_PROJECT_VERSION_MINOR 0)
-set (CMAKE_PROJECT_VERSION_PATH 0)
+set (CMAKE_PROJECT_VERSION_PATCH 0)
 set (CMAKE_SKIP_RULE_DEPENDENCY TRUE)
 set (CMAKE_EXPORT_COMPILE_COMMANDS 1)
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
@@ -16,7 +16,7 @@ option(BUILD_PYTHON "Build python if ON" ON)
 
 # Build Number Generation
 #-------------------------------------------------------------------------------
-SET(VERSION_STRING "${CMAKE_PROJECT_VERSION_MAJOR}.${CMAKE_PROJECT_VERSION_MINOR}")
+SET(VERSION_STRING "${CMAKE_PROJECT_VERSION_MAJOR}.${CMAKE_PROJECT_VERSION_MINOR}.${CMAKE_PROJECT_VERSION_PATCH}")
 ADD_DEFINITIONS(-DVERSION_STRING=${VERSION_STRING})
 #-------------------------------------------------------------------------------
 
-- 
2.32.0.windows.2

