From 801fb0067d4f7880702bd62cee81435ceadeddc5 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 21 Nov 2021 16:07:30 +1100
Subject: [PATCH 3/3] adapt path that depends on invocation in a build
 directiory

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7476ed..b24f4aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -259,7 +259,7 @@ if(BUILD_PYTHON)
     add_dependencies(write_version_string_to_python python_source_files)
     add_custom_command(TARGET write_version_string_to_python POST_BUILD
         COMMAND ${CMAKE_COMMAND} -DVERSION_STRING=${VERSION_STRING} -P
-        ${CMAKE_CURRENT_BINARY_DIR}/../cmake/write_python_version_string.cmake)
+        ${CMAKE_CURRENT_BINARY_DIR}/cmake/write_python_version_string.cmake)
 else()
     message("-- Not building python libraries. To build python libraries use -DBUILD_PYTHON=ON")
 endif()
-- 
2.32.0.windows.2

