--- a/Cargo.toml.orig	2021-09-23 14:00:00.000000000 +0000
+++ b/Cargo.toml	2021-09-23 14:00:00.000000000 +0000
@@ -42,2 +42,3 @@
 sysinfo = { version = "0.30.5", optional = true }
+time = "0.3.35"
 thiserror = "1.0.56"
@@ -54,3 +55,3 @@
-[target.'cfg(target_os = "macos")'.dependencies]
-apple-sys = { version = "0.2.0", features = ["CoreFoundation", "IOKit"] }
-core-foundation = "0.9.3"
+[target.'cfg(target_os = "macos")'.dependencies]
+apple-sys = { version = "0.2.0", features = ["CoreFoundation", "IOKit"] }
+core-foundation = "0.9.3"
