diff --git a/setup.py b/setup.py
index a41e3df..b0ad2dd 100644
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-from setuptools import setup
+from setuptools import setup, find_packages
 
 
 def readme():
@@ -23,7 +23,7 @@ setup(name='cgsn_parsers',
       author='Christopher Wingard',
       author_email='cwingard@coas.oregonstate.edu',
       license='MIT',
-      packages=['cgsn_parsers'],
+      packages=find_packages(),
       install_requires=[
           'nose',
           'numpy',
