From df4d9ebeeeee8dc236531fca5d614cd1b1465a3a Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 11 Jul 2024 14:14:47 +0200
Subject: [PATCH] Update ctypesgentest.py

---
 tests/ctypesgentest.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/ctypesgentest.py b/tests/ctypesgentest.py
index a567e2b4..efd21350 100644
--- a/tests/ctypesgentest.py
+++ b/tests/ctypesgentest.py
@@ -206,11 +206,11 @@ def cleanup_common():
 
 
 def _compile_common(common_lib):
-    subprocess.run(["gcc", "-c", f"{COMMON_DIR}/a.c", "-o", f"{COMMON_DIR}/a.o"])
-    subprocess.run(["gcc", "-c", f"{COMMON_DIR}/b.c", "-o", f"{COMMON_DIR}/b.o"])
+    subprocess.run(["$CC", "-c", f"{COMMON_DIR}/a.c", "-o", f"{COMMON_DIR}/a.o"])
+    subprocess.run(["$CC", "-c", f"{COMMON_DIR}/b.c", "-o", f"{COMMON_DIR}/b.o"])
     subprocess.run(
         [
-            "gcc",
+            "$CC",
             "-shared",
             "-o",
             f"{COMMON_DIR}/{common_lib}",
