--- work.orig/libarchive/archive_private.h	2018-04-22 06:33:58.393242100 -0500
+++ work/libarchive/archive_private.h	2018-04-22 07:13:01.088919200 -0500
@@ -46,6 +46,15 @@
 #define	__LA_DEAD
 #endif
 
+#if defined(_WIN32) && (_MSC_VER<1900)
+// Disable conversion from size_t to int possible loss of data
+#pragma warning(disable:4267)
+#pragma warning(disable:4244)
+// Disable: "The POSIX name for this item is deprecated."
+#pragma warning(disable:4996)
+#endif
+
+
 #define	ARCHIVE_WRITE_MAGIC	(0xb0c5c0deU)
 #define	ARCHIVE_READ_MAGIC	(0xdeb0c5U)
 #define	ARCHIVE_WRITE_DISK_MAGIC (0xc001b0c5U)
