From 39702357f26920b078537762e1877e30fffbf6b9 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 20 Jun 2018 05:43:59 -0700
Subject: [PATCH 2/2] Debugging for this

---
 libarchive/archive_write_disk_windows.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/libarchive/archive_write_disk_windows.c b/libarchive/archive_write_disk_windows.c
index b393a1fe..9fd8ab0b 100644
--- a/libarchive/archive_write_disk_windows.c
+++ b/libarchive/archive_write_disk_windows.c
@@ -283,7 +283,9 @@ file_information(struct archive_write_disk *a, wchar_t *path,
 		        & FILE_ATTRIBUTE_REPARSE_POINT) &&
 		(findData.dwReserved0 == IO_REPARSE_TAG_SYMLINK)))
 		flag |= FILE_FLAG_OPEN_REPARSE_POINT;
-
+	if (flag & FILE_FLAG_OPEN_REPARSE_POINT) {
+		wprintf(L"Hello symlink a->name %s\n",a->name);
+	}
 	h = CreateFileW(a->name, 0, 0, NULL,
 	    OPEN_EXISTING, flag, NULL);
 	if (h == INVALID_HANDLE_VALUE &&
@@ -1275,7 +1277,11 @@ static int
 restore_entry(struct archive_write_disk *a)
 {
 	int ret = ARCHIVE_OK, en;
-
+	if (wcsstr(a->name, L"cluster.tgz"))
+	{
+		// Using e.g.: C:\Users\builder\m64\conda-bld\src_cache\R-3.5.0_fd1725535e.tar.gz
+		printf("Debug this\n");
+	}
 	if (a->flags & ARCHIVE_EXTRACT_UNLINK && !S_ISDIR(a->mode)) {
 		/*
 		 * TODO: Fix this.  Apparently, there are platforms
-- 
2.16.2

