diff --git a/python/setup.py b/python/setup.py
index 98457ae..ed9bf87 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -75,7 +75,7 @@ if compile_with_sources:
     include_dirs += [source_dir, ]
     sources = [os.path.join(source_dir, s) for s in cpp_files]
     sources += ['_alm.c', 'alm_wrapper.cpp']
-    extra_link_args.append(os.path.join(spglib_dir, "libsymspg.a"))
+    extra_link_args.append("-lsymspg")
+    extra_link_args.append("-lgomp")
 else:  # compile with library
     sources = ['_alm.c', 'alm_wrapper.cpp']
     # static link library
