From fa4de41096ceabfe985e720afa6dc0823192ca9f Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Wed, 28 Feb 2024 11:34:32 +0100
Subject: [PATCH] Ignore the vendored eigen3

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 setup.py | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index c6d745b..da67414 100644
--- a/setup.py
+++ b/setup.py
@@ -38,6 +38,8 @@ ext_modules = [
             # Path to pybind11 headers
             get_pybind_include(),
             get_pybind_include(user=True),
+            # Use conda-forge's distribution of Eigen3
+            os.path.join(os.environ["PREFIX"], "include", "eigen3")
         ],
         #        extra_link_args=["-stdlib=libc++"],
         language="c++",
@@ -108,15 +110,6 @@ class BuildExt(build_ext):
         for ext in self.extensions:
             ext.extra_compile_args = opts
             ext.extra_link_args = link_opts
-        # third-party libraries flags
-        localincl = "third-party"
-        if not os.path.exists(os.path.join(localincl, "eigen_3.3.7", "Eigen", "Core")):
-            raise RuntimeError("couldn't find Eigen headers")
-        include_dirs = [
-            os.path.join(localincl, "eigen_3.3.7"),
-        ]
-        for ext in self.extensions:
-            ext.include_dirs = include_dirs + ext.include_dirs
         # run standard build procedure
         build_ext.build_extensions(self)

--
2.43.2

