--- xgboost-0.6a2/xgboost/build-python.sh	2016-08-09 04:32:24.000000000 -0500
+++ xgboost-0.6a2/xgboost/build-python.sh	2017-03-10 15:16:18.000000000 -0600
@@ -15,14 +15,14 @@
 cd ./xgboost/
 #remove the pre-compiled .so and trigger the system's on-the-fly compiling
 make clean
-if make lib/libxgboost.so -j4; then
+if make lib/libxgboost.so -j4 USE_OPENMP=0; then
     echo "Successfully build multi-thread xgboost"
 else
     echo "-----------------------------"
     echo "Building multi-thread xgboost failed"
     echo "Start to build single-thread xgboost"
     make clean
-    make lib/libxgboost.so -j4 no_omp=1
+    make lib/libxgboost.so -j4 USE_OPENMP=0
     echo "Successfully build single-thread xgboost"
     echo "If you want multi-threaded version"
     echo "See additional instructions in doc/build.md"
