diff --git a/modules/cbflib/include/cbf.h b/modules/cbflib/include/cbf.h
index 3d0eb84..1f588cd 100644
--- a/modules/cbflib/include/cbf.h
+++ b/modules/cbflib/include/cbf.h
@@ -321,10 +321,10 @@ _CBF_STR(CBF_VERS_RELEASE) CBF_VERS_SUBRELEASE
     typedef unsigned long long CBF_ull_type;
 #endif
    
-#if defined(CBF_DONT_USE_LONG_LONG) || defined(__cplusplus) || defined(__MINGW32__)
-#undef ULLONG_MAX
-#undef CBF_USE_LONG_LONG
-#endif
+// #if defined(CBF_DONT_USE_LONG_LONG) || defined(__cplusplus) || defined(__MINGW32__)
+// #undef ULLONG_MAX
+// #undef CBF_USE_LONG_LONG
+// #endif
 
 #ifndef SWIG
 #if defined(ULLONG_MAX) && defined(LLONG_MAX)
