diff --git a/modules/cctbx_project/crys3d/hklview/jsview_3d.py b/modules/cctbx_project/crys3d/hklview/jsview_3d.py
index e6fbe2ed54..a30290bea2 100644
--- a/modules/cctbx_project/crys3d/hklview/jsview_3d.py
+++ b/modules/cctbx_project/crys3d/hklview/jsview_3d.py
@@ -1558,7 +1558,7 @@ Distance: %s
   def OpenBrowser(self):
     if self.viewerparams.scene_id is not None and not self.WBmessenger.websockclient \
        and not self.WBmessenger.browserisopen or self.isnewfile:
-      NGLlibpath = libtbx.env.under_root(os.path.join("modules","cctbx_project","crys3d","hklview","ngl.js") )
+      NGLlibpath = libtbx.env.under_dist("crys3d", "hklview/ngl.js") )
       htmlstr = self.hklhtml %(NGLlibpath, os.path.abspath( self.jscriptfname))
       htmlstr += self.htmldiv
       with open(self.hklfname, "w") as f:
