diff --git a/modules/cctbx_project/libtbx/command_line/pytest_launcher.py b/libtbx/command_line/pytest_launcher.py
deleted file mode 100644
index abc1a85845..0000000000
--- a/modules/cctbx_project/libtbx/command_line/pytest_launcher.py
+++ /dev/null
@@ -1,12 +0,0 @@
-# LIBTBX_SET_DISPATCHER_NAME pytest
-# LIBTBX_SET_DISPATCHER_NAME py.test
-from __future__ import absolute_import, division, print_function
-
-import sys
-
-import pytest
-
-# modify sys.argv so the command line help shows the right executable name
-sys.argv[0] = 'pytest'
-
-sys.exit(pytest.main())
