diff --git a/pytest/test_rst.py b/pytest/test_rst.py
index b159f9e..20cc3bf 100644
--- a/pytest/test_rst.py
+++ b/pytest/test_rst.py
@@ -6,6 +6,9 @@ import os
 import re
 import subprocess
 # ----------------------------------------------------------------------------
+pattern_return_newline    = re.compile( r'\r\n' )
+pattern_include_backslash = re.compile( r'\.\. literalinclude:: .*\\' )
+# ----------------------------------------------------------------------------
 def get_index_page_name() :
    file_obj  = open('.readthedocs.yaml', 'r')
    file_data = file_obj.read()
@@ -28,7 +31,7 @@ def run_xrst() :
    # This command should be the same as the one in bin/check_xrst.sh
    # that's used to keep the test_rst directory up to date.
    command = [
-      'python3', '-m', 'xrst',
+      'python', '-m', 'xrst',
       '--local_toc',
       '--rst_only',
       '--config_file',     'xrst.toml',
@@ -86,6 +89,22 @@ def run_test() :
          rst_file_obj.close()
          check_file_obj.close()
          #
+         # rst_data
+         # dos2unix: relpace \r\n by \n
+         replace  = r'\n'
+         rst_data = pattern_return_newline.sub(replace, rst_data)
+         #
+         # rst_data
+         # relpace \ in file names by /
+         m_obj = pattern_include_backslash.search(rst_data)
+         while m_obj != None :
+            start      = m_obj.start()
+            end        = m_obj.end()
+            text       = m_obj.group(0)
+            replace    = text.replace( '\\' , '/' )
+            rst_data   = check_data[:start] + replace + rst_data[end:] 
+            m_obj      = pattern_include_backslash.search(rst_data)
+         #
          if rst_data == check_data :
             print( f'{rst_name}: OK' )
          else :
diff --git a/xrst.toml b/xrst.toml
index 0e52cbf..c26094d 100644
--- a/xrst.toml
+++ b/xrst.toml
@@ -34,7 +34,7 @@ overline  = [ false, false, false, false, false,  ]
 # -----------------------------------------------------------------------------
 # BEGIN_SPELL_PACKAGE
 [spell_package]
-data = 'pyenchant'
+data = 'pyspellchecker'
 # END_SPELL_PACKAGE
 # -----------------------------------------------------------------------------
 # BEGIN_INPUT_FILES
