diff --git a/xrst.toml b/xrst.toml
index 0e52cbf..c26094d 100644
--- a/xrst.toml
+++ b/xrst.toml
@@ -34,7 +34,7 @@ overline  = [ false, false, false, false, false,  ]
 # -----------------------------------------------------------------------------
 # BEGIN_SPELL_PACKAGE
 [spell_package]
-data = 'pyenchant'
+data = 'pyspellchecker'
 # END_SPELL_PACKAGE
 # -----------------------------------------------------------------------------
 # BEGIN_INPUT_FILES
