diff --git a/pyproject.toml b/pyproject.toml
index a5d37a9..81c5145 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ classifiers       = [
 dependencies       = [
    'sphinx', 'toml', 'sphinx-copybutton',             # required
    'pytest',                                          # need for testing xrst
-   'pyenchant', 'pyspellchecker',                     # need the one you use
+   'pyspellchecker',                                  # need the one you use
    'furo', 'sphinx-rtd-theme', 'sphinx-book-theme',   # need the ones you use
 ]
 # END_DEPENDENCIES
diff --git a/pytest/test_rst.py b/pytest/test_rst.py
index b159f9e..39b0647 100644
--- a/pytest/test_rst.py
+++ b/pytest/test_rst.py
@@ -6,6 +6,9 @@ import os
 import re
 import subprocess
 # ----------------------------------------------------------------------------
+pattern_return_newline    = re.compile( r'\r\n' )
+pattern_include_backslash = re.compile( r'\.\. literalinclude:: .*\\' )
+# ----------------------------------------------------------------------------
 def get_index_page_name() :
    file_obj  = open('.readthedocs.yaml', 'r')
    file_data = file_obj.read()
@@ -86,6 +89,22 @@ def run_test() :
          rst_file_obj.close()
          check_file_obj.close()
          #
+         # rst_data
+         # dos2unix: relpace \r\n by \n
+         replace  = r'\n'
+         rst_data = pattern_return_newline.sub(replace, rst_data)
+         #
+         # rst_data
+         # relpace \ in file names by /
+         m_obj = pattern_include_backslash.search(rst_data)
+         while m_obj != None :
+            start      = m_obj.start()
+            end        = m_obj.end()
+            text       = m_obj.group(0)
+            replace    = text.replace( '\\' , '/' )
+            rst_data   = check_data[:start] + replace + rst_data[end:] 
+            m_obj      = pattern_include_backslash.search(rst_data)
+         #
          if rst_data == check_data :
             print( f'{rst_name}: OK' )
          else :
diff --git a/setup.py b/setup.py
index 446c6b0..2a56f82 100644
--- a/setup.py
+++ b/setup.py
@@ -27,7 +27,7 @@ setup(
    install_requires = [
       'sphinx', 'toml', 'sphinx-copybutton',           # required
       'pytest',                                        # need for testing xrst
-      'pyenchant', 'pyspellchecker',                   # need the one you use
+      'pyspellchecker',                                # need the one you use
       'furo', 'sphinx-rtd-theme', 'sphinx-book-theme', # need the ones you use
    ],
    entry_points = {
diff --git a/tox.ini b/tox.ini
index 8a2e158..4b5ffc8 100644
--- a/tox.ini
+++ b/tox.ini
@@ -12,6 +12,5 @@ deps     =
    pytest
    toml
    pyspellchecker
-   pyenchant
 #
 commands = pytest -s pytest
diff --git a/xrst.toml b/xrst.toml
index 0e52cbf..c26094d 100644
--- a/xrst.toml
+++ b/xrst.toml
@@ -34,7 +34,7 @@ overline  = [ false, false, false, false, false,  ]
 # -----------------------------------------------------------------------------
 # BEGIN_SPELL_PACKAGE
 [spell_package]
-data = 'pyenchant'
+data = 'pyspellchecker'
 # END_SPELL_PACKAGE
 # -----------------------------------------------------------------------------
 # BEGIN_INPUT_FILES
