diff --git a/setup.py b/setup.py
index 3c125ff..d0a1e31 100644
--- a/setup.py
+++ b/setup.py
@@ -90,7 +90,7 @@ class BuildFrontEnd(setuptools.command.build_py.build_py):
             rmtree('build/lib/')
 
         try:
-            subprocess.check_call('frontend/gradlew -p frontend clean build', shell=True)
+            subprocess.check_call('frontend/gradlew -x test -p frontend clean build', shell=True)
         except OSError:
             assert 0, "build failed"
         copy2(self.source_server_file, self.dest_file_name)
@@ -128,7 +128,7 @@ class BuildPlugins(Command):
 
         try:
             if self.plugins == "sagemaker":
-                subprocess.check_call('plugins/gradlew -p plugins clean bS', shell=True)
+                subprocess.check_call('plugins/gradlew -x test -p plugins clean bS', shell=True)
             else:
                 raise OSError("No such rule exists")
         except OSError:
