From 41e3dd8231ae1a2f1d8de24d12352df024e26fc9 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Thu, 17 Oct 2024 09:13:22 +0200
Subject: [PATCH] patch requirements

---
 requirements.txt | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 requirements.txt

diff --git a/requirements.txt b/requirements.txt
new file mode 100644
index 0000000..296d654
--- /dev/null
+++ requirements.txt
@@ -0,0 +1 @@
+numpy
\ No newline at end of file
