From e12902472898a8ebda3ed52e43b0b8b31d660a1d Mon Sep 17 00:00:00 2001
From: Ghislain Antony Vaillant <ghisvail@gmail.com>
Date: Thu, 21 Oct 2021 02:16:30 +0200
Subject: [PATCH] No compile test for dependencies

---
 configure | 60 ++++++++++---------------------------------------------
 1 file changed, 10 insertions(+), 50 deletions(-)

diff --git a/configure b/configure
index 06e2f9f8a..343a538a3 100755
--- a/configure
+++ b/configure
@@ -969,20 +969,7 @@ if not openmp:
   eigen_cflags += [ '-DEIGEN_DONT_PARALLELIZE' ]
 
 
-if compile_test ('Eigen3 Unsupported', cpp_flags + eigen_cflags, ld_flags, '''
-#include <iostream>
-#include <Eigen/Core>
-#include <unsupported/Eigen/SpecialFunctions>
-
-using array_type = Eigen::Array<double, 1, 1>;
-
-int main (int argc, char* argv[]) {
-  auto test = Eigen::betainc (array_type::Constant (10.0), array_type::Constant (0.5), array_type::Constant (1.0));
-  std::cout << "Present";
-  return (0);
-}
-''', on_failure='not found; custom functions to be used'):
-  cpp_flags += [ '-DMRTRIX_HAVE_EIGEN_UNSUPPORTED_SPECIAL_FUNCTIONS' ]
+cpp_flags += [ '-DMRTRIX_HAVE_EIGEN_UNSUPPORTED_SPECIAL_FUNCTIONS' ]
 
 
 
@@ -1043,18 +1030,9 @@ int main (int argc, char* argv[])
 tiff_cflags = get_flags ([], 'TIFF_CFLAGS', '--cflags libtiff-4')
 tiff_ldflags = get_flags ([ '-ltiff' ], 'TIFF_LINKFLAGS', '--libs libtiff-4')
 
-if compile_test ('TIFF library', cpp_flags + tiff_cflags, ld_flags + tiff_ldflags, '''
-#include <iostream>
-#include <tiffio.h>
-
-int main() {
-  std::cout << TIFFGetVersion();
-  return (0);
-}
-''', on_failure='not found - TIFF support disabled'):
-  cpp_flags += [ '-DMRTRIX_TIFF_SUPPORT' ] + tiff_cflags
-  ld_flags += tiff_ldflags
-  ld_lib_flags += tiff_ldflags
+cpp_flags += [ '-DMRTRIX_TIFF_SUPPORT' ] + tiff_cflags
+ld_flags += tiff_ldflags
+ld_lib_flags += tiff_ldflags
 
 
 
@@ -1065,18 +1043,9 @@ int main() {
 png_cflags = get_flags ([], 'PNG_CFLAGS', '--cflags libpng')
 png_ldflags = get_flags ([ '-lpng' ], 'PNG_LINKFLAGS', '--libs libpng')
 
-if compile_test ('PNG library', cpp_flags + png_cflags, ld_flags + png_ldflags, '''
-#include <iostream>
-#include <png.h>
-
-int main() {
-  std::cout << "Header: " << PNG_LIBPNG_VER_STRING << "; library: " << png_libpng_ver;
-  return (0);
-}
-''', on_failure='not found - PNG support disabled'):
-  cpp_flags += [ '-DMRTRIX_PNG_SUPPORT' ] + png_cflags
-  ld_flags += png_ldflags
-  ld_lib_flags += png_ldflags
+cpp_flags += [ '-DMRTRIX_PNG_SUPPORT' ] + png_cflags
+ld_flags += png_ldflags
+ld_lib_flags += png_ldflags
 
 
 
@@ -1089,18 +1058,9 @@ int main() {
 fftw_cflags = get_flags ([], 'FFTW_CFLAGS', '--cflags fftw3')
 fftw_ldflags = get_flags ([ '-lfftw3' ], 'FFTW_LINKFLAGS', '--libs fftw3')
 
-if compile_test ('FFTW library', cpp_flags + fftw_cflags, ld_flags + fftw_ldflags, '''
-#include <iostream>
-#include <fftw3.h>
-
-int main() {
-  std::cout << fftw_version << "\\n";
-  return (0);
-}
-''', on_failure='not found - FFTW support disabled'):
-  cpp_flags += [ '-DEIGEN_FFTW_DEFAULT' ] + fftw_cflags
-  ld_flags += fftw_ldflags
-  ld_lib_flags += fftw_ldflags
+cpp_flags += [ '-DEIGEN_FFTW_DEFAULT' ] + fftw_cflags
+ld_flags += fftw_ldflags
+ld_lib_flags += fftw_ldflags
 
 
 
-- 
2.30.2

