--- configure	2019-01-17 02:33:29.000000000 +0000
+++ configure	2019-01-21 14:26:20.000000000 +0000
@@ -14457,7 +14457,7 @@
 $as_echo_n "checking for libraries required to embed python... " >&6; }

 if $PYTHON-config --help 1>/dev/null 2>/dev/null; then
-  PYTHON_LIBS=`$PYTHON-config --ldflags 2>/dev/null`
+  PYTHON_LIBS=`$PYTHON -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_config_var('LDSHARED').split(maxsplit=1)[1], end='')"`
   PYTHON_LIB=`$PYTHON -c "import distutils.sysconfig as s; print(s.get_python_lib(standard_lib=1))"`
   if echo "$host_os" | grep darwin >/dev/null 2>&1; then
                 PYTHON_LIB_LOC=/usr/lib
