From e6da2ccc87806dab1bd4377e28a8d759f088da08 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sat, 16 Sep 2023 11:54:33 +0200
Subject: [PATCH] Add LANGUAGES C option to project CMake call

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb2ecae..8fc6649 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.1)
-project (TRLIB VERSION 0.4)
+project (TRLIB VERSION 0.4 LANGUAGES C)
 
 set(PROJECT_DESCRIPTION "Trust Region Subproblem Solver Library")
 
