From 5447ca3732534e6e455f9038d1c4074685d96e30 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sat, 16 Sep 2023 11:42:15 +0200
Subject: [PATCH] Disable valgrind tests

---
 tests/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 068f559..6896d5a 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -45,6 +45,7 @@ if(CHECK_FOUND)
         target_include_directories("test_${t}" PRIVATE "${PROJECT_BINARY_DIR}/include" "${CMAKE_CURRENT_BINARY_DIR}" ${CHECK_INCLUDE_DIRS})
         add_test("test_${t}" "${CMAKE_CURRENT_BINARY_DIR}/test_${t}")
 
+        #[===[
         if(APPLE)
         else()
             add_custom_command(OUTPUT "test_${t}_valgrind.c" COMMAND sed "'/sr = srunner_create(s);/a    srunner_set_fork_status(sr, CK_NOFORK);'" "${CMAKE_CURRENT_SOURCE_DIR}/test_${t}.c" > "test_${t}_valgrind.c" DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/test_${t}.c")
@@ -54,6 +55,7 @@ if(CHECK_FOUND)
             target_include_directories("test_${t}_valgrind" PRIVATE "${PROJECT_BINARY_DIR}/include" "${CMAKE_CURRENT_BINARY_DIR}" ${CHECK_INCLUDE_DIRS})
             add_test("test_${t}_valgrind" valgrind  --leak-check=full --show-leak-kinds=all --show-reachable=yes --error-exitcode=1 --errors-for-leak-kinds=all "${CMAKE_CURRENT_BINARY_DIR}/test_${t}_valgrind")
         endif()
+        ]===]
     endforeach(t)
 
 endif(CHECK_FOUND)
