From 8a60fce2e9d93ba271297aee397862e63f403d69 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Mon, 24 Feb 2025 02:02:43 -0600
Subject: [PATCH] fix: Add python_requires to setup.py

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 0b36ba1..2676bac 100755
--- a/setup.py
+++ b/setup.py
@@ -17,5 +17,6 @@ setup(name = 'pyslha',
       url = 'http://www.insectnation.org/projects/pyslha',
       description = 'Parsing, manipulating, and visualising SUSY Les Houches Accord data',
       long_description = __doc__,
+      python_requires = ">=3.6",
       keywords = 'supersymmetry susy slha simulation mass decay hep physics particle',
       license = 'GPL')
-- 
2.48.1

