From 8330008f28958ac574ed895ea2f3c61a762968c1 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Mon, 24 Feb 2025 02:00:58 -0600
Subject: [PATCH] fix: Use setuptools over deprecated distutils

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 0b36ba1..5934845 100755
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,6 @@
 #! /usr/bin/env python
 
-# TODO: migrate from distutils for compatibility... to?
-from distutils.core import setup
+from setuptools import setup
 
 ## Setup definition
 import pyslha
-- 
2.48.1

