diff --git a/include/moto/ocp/sym.hpp b/include/moto/ocp/sym.hpp
index a9eec1f..369c112 100644
--- a/include/moto/ocp/sym.hpp
+++ b/include/moto/ocp/sym.hpp
@@ -234,5 +234,11 @@ inline bool operator==(const var &lhs, const var &rhs) noexcept {
     return *lhs == *rhs; // use sym's operator==
 } ///< operator== for var
+inline bool operator==(const var &lhs, const sym &rhs) noexcept {
+    return *lhs == rhs;
+}
+inline bool operator==(const sym &lhs, const var &rhs) noexcept {
+    return lhs == *rhs;
+}
 } // namespace moto
 
 #endif // MOTO_OCP_SYM_HPP
