diff --git a/src/cmd/gz_TEST.cc b/src/cmd/gz_TEST.cc
index 635f48c..d5d64dd 100644
--- a/src/cmd/gz_TEST.cc
+++ b/src/cmd/gz_TEST.cc
@@ -68,7 +68,7 @@ TEST(CmdLine, Ls)
   std::string cmd = "gz launch " + get_config_path("ls.gzlaunch");
   std::string output = customExecStr(cmd);
   EXPECT_TRUE(output.find("CMakeFiles") != std::string::npos) << output;
-  EXPECT_TRUE(output.find("Makefile") != std::string::npos) << output;
+  EXPECT_TRUE(output.find("cmake_install.cmake") != std::string::npos) << output;
 }
 
 /////////////////////////////////////////////////
