diff --git a/CMakeLists.txt b/CMakeLists.txt
index 77cb6c3..9fa7cb4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,7 +91,8 @@ endif()
 
 #--------------------------------------
 # Find prerequistes for backward
-include(src/vendor/BackwardConfig.cmake)
+# include(src/vendor/BackwardConfig.cmake)
+find_package(Backward)
 
 #============================================================================
 # Add plugins

diff --git a/src/Manager.cc b/src/Manager.cc
index 4e17272..f1e990f 100644
--- a/src/Manager.cc
+++ b/src/Manager.cc
@@ -59,7 +59,7 @@
 #include "ignition/launch/config.hh"
 #include "ignition/launch/Plugin.hh"
 
-#include "vendor/backward.hpp"
+#include "backward.hpp"
 
 using namespace ignition::launch;
 using namespace std::chrono_literals;
