From 6cd9afd49bf6c7cc93440e145a7f53518a35118a Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Tue, 8 Jun 2021 11:29:13 +0200
Subject: [PATCH] Try to fix macOS

---
 Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 42d43dd..04f56a7 100644
--- a/Makefile
+++ b/Makefile
@@ -77,13 +77,11 @@ endif
 CC       ?= gcc
 # compiling flags here
 CFLAGS   = -g -L$(PREFIX)/lib -std=c99 -I$(SRCDIR) -I$(LIBDIR)  -Wall -Wextra -fno-common
-ifndef MAC_OS
 ifndef NODPKG
 	CFLAGS   +=$(shell dpkg-buildflags --get CPPFLAGS)
 	CFLAGS   +=$(shell dpkg-buildflags --get CFLAGS)
 endif
-	CFLAGS += $(shell pkg-config --cflags libsecret-1)
-endif
+CFLAGS += $(shell pkg-config --cflags libsecret-1)
 TEST_CFLAGS = $(CFLAGS) -I.
 
 # Linker options
-- 
2.31.1

