From 4d80f824ed853fd8f109e7a066b7f11c97575246 Mon Sep 17 00:00:00 2001
From: Guilherme Leobas <gleobas@quansight.com>
Date: Wed, 20 Apr 2022 18:54:50 +0000
Subject: [PATCH] Build DBEngine

---
 Embedded/DBEngine.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Embedded/DBEngine.cpp b/Embedded/DBEngine.cpp
index d0fc29e658..a84dedffe5 100644
--- a/Embedded/DBEngine.cpp
+++ b/Embedded/DBEngine.cpp
@@ -19,6 +19,7 @@
 #include <stdexcept>
 #include "DataMgr/ForeignStorage/ArrowForeignStorage.h"
 #include "Fragmenter/FragmentDefaultValues.h"
+#include "LockMgr/LockMgr.h"
 #include "Parser/ParserWrapper.h"
 #include "QueryEngine/ArrowResultSet.h"
 #include "QueryEngine/Execute.h"
@@ -210,8 +211,9 @@ class DBEngineImpl : public DBEngine {
                                                        0,
                                                        0),
                            {}};
+    lockmgr::LockedTableDescriptors locks;
     db_handler_->sql_execute(
-        result, session_id, query_str, column_format, first_n, at_most_n);
+        result, session_id, query_str, column_format, first_n, at_most_n, locks);
     auto& targets = result.getTargetsMeta();
     std::vector<std::string> col_names;
     for (const auto target : targets) {
-- 
2.25.1

