diff --git a/Makefile b/Makefile
index 8e49043..8e735b7 100644
--- a/Makefile
+++ b/Makefile
@@ -33,8 +33,8 @@ FMT_LOG=fmt.log
 LINT_LOG=lint.log
 IMPORT_LOG=import.log
 
-GIT_SHA=$(shell git rev-parse HEAD)
-GIT_CLOSEST_TAG=$(shell git describe --abbrev=0 --tags)
+GIT_SHA="8c89e641eacafc5315de3c6486d398d6ce0d441c"
+GIT_CLOSEST_TAG="v${PKG_VERSION}"
 DATE=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 BUILD_INFO_IMPORT_PATH=github.com/jaegertracing/jaeger/pkg/version
 BUILD_INFO=-ldflags "-X $(BUILD_INFO_IMPORT_PATH).commitSHA=$(GIT_SHA) -X $(BUILD_INFO_IMPORT_PATH).latestVersion=$(GIT_CLOSEST_TAG) -X $(BUILD_INFO_IMPORT_PATH).date=$(DATE)"
