From b1d62ce830062f3ae8af7171e63d9e6b9e03a299 Mon Sep 17 00:00:00 2001
From: Howard Butler <hobu.inc@gmail.com>
Date: Sun, 20 Jul 2025 07:23:36 -0500
Subject: [PATCH] Add BUILD_SHARED_LIBS option

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca49824..8a92969 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,8 @@ project(spz
 
 include(GNUInstallDirs)
 
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+
 # zlib is required to build the project
 find_package(ZLIB REQUIRED)
 
