--- a/setup.py
+++ b/setup.py
@@ -65,7 +65,7 @@ def find_fann():
                     return True
         raise Exception("Couldn't find FANN source libs!")
     else:
-        dirs = ['/lib', '/usr/lib', '/usr/lib64', '/usr/local/lib', '/usr/pkg/lib']
+        dirs = [os.environ.get('PREFIX','')+'/lib','/lib', '/usr/lib', '/usr/lib64', '/usr/local/lib', '/usr/pkg/lib']
         for path in dirs:
             if os.path.isdir(path):
                 if find_x(path):
