--- a/setup.py
+++ b/setup.py
@@ -48,6 +48,8 @@ def find_executable(executable, path=None):
 def find_x(path1):
     '''Return true if substring is in string for files
     in specified path'''
+    if not os.path.exists(path1):
+        return False
     libs = os.listdir(path1)
     for lib_dir in libs:
         if "doublefann" in lib_dir:
@@ -65,7 +67,7 @@ def find_fann():
                     return True
         raise Exception("Couldn't find FANN source libs!")
     else:
-        dirs = ['/lib', '/usr/lib', '/usr/lib64', '/usr/local/lib', '/usr/pkg/lib']
+        dirs = [os.environ.get('PREFIX','')+'/lib','/lib', '/usr/lib', '/usr/lib64', '/usr/local/lib', '/usr/pkg/lib']
         for path in dirs:
             if os.path.isdir(path):
                 if find_x(path):
