--- a/lib/maxflow/src/pythongraph.h	2021-04-29 14:28:31.000000000 -0600
+++ b/lib/maxflow/src/pythongraph.h	2021-04-29 14:28:49.000000000 -0600
@@ -17,8 +17,8 @@
 {
 public:
 	Pythongraph(int node_num_max, int edge_num_max) : Graph<captype, tcaptype, flowtype>(node_num_max, edge_num_max, NULL) {};
-	flowtype maxflow() { Graph<captype, tcaptype, flowtype>::maxflow(); };
-	typename Graph<captype, tcaptype, flowtype>::termtype what_segment(int i) { Graph<captype, tcaptype, flowtype>::what_segment(i); };
+	flowtype maxflow() { return Graph<captype, tcaptype, flowtype>::maxflow(); };
+	typename Graph<captype, tcaptype, flowtype>::termtype what_segment(int i) { return Graph<captype, tcaptype, flowtype>::what_segment(i); };
 };
 #endif
 
