diff --git a/CMakeLists.txt b/CMakeLists.txt
index 80c2325d..3c0f5cd7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,14 +34,17 @@ include(pyre_mpi)
 include(pyre_gsl)
 include(pyre_postgres)
 
-# ask git for the pyre version
-pyre_getVersion()
+set(ENV_PKG_VERSION $ENV{PKG_VERSION} CACHE INTERNAL "")
 
 # set up the project
-project(PYRE VERSION ${REPO_MAJOR}.${REPO_MINOR}.${REPO_MICRO} LANGUAGES CXX C)
+project(PYRE VERSION "${ENV_PKG_VERSION}" LANGUAGES CXX C)
 # hmmmm
 include(GNUInstallDirs)
 
+set(MAJOR ${PROJECT_VERSION_MAJOR})
+set(MINOR ${PROJECT_VERSION_MINOR})
+set(MICRO ${PROJECT_VERSION_PATCH})
+
 # programs
 find_program(BASH_PROGRAM bash)
 
