diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a24372e..be57f32d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,8 +33,14 @@ include(pyre_cuda)
 include(pyre_gsl)
 include(pyre_mpi)
 
-# ask git for the pyre version
-pyre_getVersion()
+# set pyre version info
+set(REPO_MAJOR 1)
+set(REPO_MINOR 9)
+set(REPO_MICRO 9)
+set(REVISION "")
+set(MAJOR ${REPO_MAJOR})
+set(MINOR ${REPO_MINOR})
+set(MICRO ${REPO_MICRO})
 
 # set up the project
 project(PYRE VERSION ${REPO_MAJOR}.${REPO_MINOR}.${REPO_MICRO} LANGUAGES CXX)
