diff --git a/CMakeLists.txt b/CMakeLists.txt
index f39b35a..a622695 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,3 +74,5 @@ if(DOXYGEN_FOUND)
     configure_file("${marray_SOURCE_DIR}/doxygen/Doxyfile.in" "${marray_BINARY_DIR}/Doxyfile" @ONLY IMMEDIATE)
     add_custom_target(doc-marray ALL COMMAND ${DOXYGEN} "${marray_BINARY_DIR}/Doxyfile")
 endif()
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/andres" DESTINATION include)
