diff --git a/src/extensions/scatteringcomponentlist_ext.cpp b/src/extensions/scatteringcomponentlist_ext.cpp
index 99f39e8..2663a41 100644
--- a/src/extensions/scatteringcomponentlist_ext.cpp
+++ b/src/extensions/scatteringcomponentlist_ext.cpp
@@ -61,7 +61,7 @@ bool contains(const ScatteringComponentList& scl,
 // Get slices directly from the boost python object
 bp::object getSCSlice(bp::object & scl, bp::slice& s)
 {
-    bp::list l = list(scl);
+    bp::list l(scl);
     return l[s];
 }

