diff --git a/SConstruct b/SConstruct
index 9b14d44..4c6301d 100644
--- a/SConstruct
+++ b/SConstruct
@@ -97,11 +97,6 @@ good_python_flags = lambda n : (
 pyversion = pyoutput('import sys; print("%i.%i" % sys.version_info[:2])')
 pythonconfig = 'python%s-config' % (pyversion if pyversion[0] == '3' else '')
 # Verify python-config comes from the same path as the target python.
-xpython = env.WhereIs(env['python'])
-xpythonconfig = env.WhereIs(pythonconfig)
-if os.path.dirname(xpython) != os.path.dirname(xpythonconfig):
-    print("Inconsistent paths of %r and %r" % (xpython, xpythonconfig))
-    Exit(1)
 # Process the python-config flags here.
 env.ParseConfig(pythonconfig + " --cflags")
 env.Replace(CCFLAGS=filter(good_python_flags, env['CCFLAGS']))

