diff --git a/setup.py b/setup.py
index e60c7f2..02e35bc 100644
--- a/setup.py
+++ b/setup.py
@@ -209,7 +209,7 @@ package_data = {'lensfunpy': []}
 # evil hack, check cmd line for relevant commands
 # custom cmdclasses didn't work out in this case
 cmdline = ''.join(sys.argv[1:])
-needsCompile = any(s in cmdline for s in ['install', 'bdist', 'build_ext', 'wheel', 'nosetests'])
+needsCompile = False
 if isWindows and needsCompile:
     windows_lensfun_compile()
     package_data['lensfunpy'].append('*.dll')
