diff --git a/setup.py b/setup.py
index e60c7f2..b4fa13c 100644
--- a/setup.py
+++ b/setup.py
@@ -222,7 +222,8 @@ package_data = {'lensfunpy': []}
 # evil hack, check cmd line for relevant commands
 # custom cmdclasses didn't work out in this case
 cmdline = ''.join(sys.argv[1:])
-needsCompile = any(s in cmdline for s in ['install', 'bdist', 'build_ext', 'wheel', 'nosetests'])
+#needsCompile = any(s in cmdline for s in ['install', 'bdist', 'build_ext', 'wheel', 'nosetests'])
+needsCompile = False
 if isWindows and needsCompile:
     windows_lensfun_compile()
     package_data['lensfunpy'].append('*.dll')
