diff --git a/setup.py b/setup.py
index ab31a5a..65b18f4 100644
--- a/setup.py
+++ b/setup.py
@@ -64,15 +64,7 @@ def use_pkg_config():
     _ask_pkg_config(extra_link_args,    '--libs-only-other')
     _ask_pkg_config(libraries,          '--libs-only-l', '-l')
 
-if isWindows or isMac:
-    cmake_build = os.path.abspath('external/lensfun/cmake_build')
-    install_dir = os.path.join(cmake_build, 'install')
-    
-    include_dirs += [os.path.join(install_dir, 'include', 'lensfun')]
-    library_dirs += [os.path.join(install_dir, 'lib')]
-else:
-    use_pkg_config()
-
+use_pkg_config()
 if isWindows:
     include_dirs += ['external/stdint']
 
@@ -219,7 +210,7 @@ package_data = {}
 # evil hack, check cmd line for relevant commands
 # custom cmdclasses didn't work out in this case
 cmdline = ''.join(sys.argv[1:])
-needsCompile = any(s in cmdline for s in ['install', 'bdist', 'build_ext', 'nosetests'])
+needsCompile = False
 if isWindows and needsCompile:
     windows_lensfun_compile()
     package_data['lensfunpy'] = ['db_files/*.xml',
