--- safer-4.0.0.orig/setup.py	2020-05-08 14:35:47.000000000 -0300
+++ safer-4.0.0/setup.py	2020-06-08 17:37:33.582870544 -0300
@@ -25,7 +25,7 @@
         tests_require=['pytest'],
         py_modules=['safer'],
         description='A safer file opener',
-        long_description=open('README.rst').read(),
+        # long_description=open('README.rst').read(),
         license='MIT',
         classifiers=_classifiers,
         keywords=['testing', 'modules'],
