diff --git a/hazelbean/ui/auto_ui.py b/hazelbean/ui/auto_ui.py
index b567a62..4f524d6 100644
--- a/hazelbean/ui/auto_ui.py
+++ b/hazelbean/ui/auto_ui.py
@@ -8,7 +8,6 @@ import pandas as pd
 import geopandas as gpd
 from collections import OrderedDict
 import logging
-import fiona
 
 import hazelbean as hb
 
diff --git a/hazelbean/ui/auto_ui_tg.py b/hazelbean/ui/auto_ui_tg.py
index 80a5508..ede866f 100644
--- a/hazelbean/ui/auto_ui_tg.py
+++ b/hazelbean/ui/auto_ui_tg.py
@@ -8,7 +8,6 @@ import pandas as pd
 import geopandas as gpd
 from collections import OrderedDict
 import logging
-import fiona
 
 import hazelbean as hb
 
