diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2da41b2..f06c88a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,7 @@ project(drjit-core
 #  Optional features available to users
 # ----------------------------------------------------------
 
+option(DRJIT_USE_SYSTEM_ROBIN_MAP "Use system-installed robin-map" OFF)
 option(DRJIT_DYNAMIC_LLVM "Resolve LLVM dynamically at run time?" ON)
 option(DRJIT_CORE_ENABLE_TESTS "Build Dr.Jit-Core test suite?" OFF)
 
@@ -170,9 +171,15 @@ add_library(
 
 target_compile_features(drjit-core PUBLIC cxx_std_17)
 
+if (DRJIT_USE_SYSTEM_ROBIN_MAP)
+  find_package(tsl-robin-map REQUIRED CONFIG)
+  target_link_libraries(drjit-core PRIVATE tsl::robin_map)
+else()
+  target_include_directories(drjit-core PRIVATE ext/robin_map/include)
+endif()
+
 target_include_directories(drjit-core PRIVATE
   ext/nanothread/include
-  ext/robin_map/include
   ext/lz4
 )
 
