From 47b34cfe7fd96ba5e50063b8bccd44f24e02025b Mon Sep 17 00:00:00 2001
From: Ilya Shlyakhter <ilya_shl@alum.mit.edu>
Date: Sun, 27 Aug 2017 21:43:09 -0400
Subject: [PATCH 2/2] added setting of DYLD_FALLBACK_LIBRARY_PATH

---
 Makefile.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 0137211..8fa1eb4 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -136,11 +136,11 @@ $(ALIB): $(ALIBOBJS)
 $(OBJS) $(LOBJS): fm.h funcname1.h
 
 tagtable.c: tagtable.tab mktable$(EXT) html.h 
-	./mktable$(EXT) 100 $(srcdir)/tagtable.tab > $@
+	env DYLD_FALLBACK_LIBRARY_PATH=${DYLD_FALLBACK_LIBRARY_PATH}:${PREFIX}/lib ./mktable$(EXT) 100 $(srcdir)/tagtable.tab > $@
 
 entity.h: entity.tab mktable$(EXT)
 	echo '/* $$I''d$$ */' > $@
-	./mktable$(EXT) 100 $(srcdir)/entity.tab >> $@
+	env DYLD_FALLBACK_LIBRARY_PATH=${DYLD_FALLBACK_LIBRARY_PATH}:${PREFIX}/lib ./mktable$(EXT) 100 $(srcdir)/entity.tab >> $@
 
 indep.o: indep.c fm.h funcname1.h entity.h
 func.o: funcname.c functable.c funcname1.h
@@ -173,7 +173,7 @@ funcname2.h: funcname.tab
 
 functable.c: funcname.tab mktable$(EXT)
 	sort funcname.tab | $(AWK) -f $(top_srcdir)/functable.awk > functable.tab
-	./mktable$(EXT) 100 functable.tab > $@
+	env DYLD_FALLBACK_LIBRARY_PATH=${DYLD_FALLBACK_LIBRARY_PATH}:${PREFIX}/lib ./mktable$(EXT) 100 functable.tab > $@
 	-rm -f functable.tab
 
 mktable$(EXT): mktable.o dummy.o Str.o hash.o myctype.o
-- 
2.13.3

