--- lib/Makefile.am	2012-10-12 10:05:48.000000000 -0400
+++ lib/Makefile.am	2017-10-15 00:42:58.919856953 -0400
@@ -68,15 +68,7 @@
 libsasl2_la_DEPENDENCIES = $(LTLIBOBJS)
 libsasl2_la_LIBADD = $(LTLIBOBJS) $(SASL_DL_LIB) $(LIB_SOCKET) $(LIB_DOOR)
 
-if MACOSX
-framedir = /Library/Frameworks/SASL2.framework
 install-exec-hook:
-	$(mkinstalldirs) $(framedir)/Versions/A
-	ln -fs $(libdir)/libsasl2.dylib $(framedir)/Versions/A/SASL2
-	cd $(framedir) ; ln -fs Versions/A/SASL2 .
-else
-install-exec-hook:
-endif
 
 libsasl2.a: libsasl2.la $(SASL_STATIC_OBJS)
 	@echo adding static plugins and dependencies
