diff --git CMakeLists.txt CMakeLists.txt
new file mode 100644
index 0000000..04fc3af
--- /dev/null
+++ CMakeLists.txt
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 2.8)
+project(ann)
+file(GLOB LIB_SOURCES src/*cpp)
+add_library(ann SHARED ${LIB_SOURCES})
+add_definitions(-DDLL_EXPORTS)
+include_directories(${PROJECT_SOURCE_DIR}/include/)
+
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/
+  DESTINATION include)
+
+install(TARGETS ann DESTINATION lib)
