commit 80080f19dc2cc7a849f476612882feb1881c5f9c
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Sat Feb 11 13:22:01 2017 +0530

    Add patch to declare variables "extern"

diff --git a/c/planarity.h b/c/planarity.h
index b0cfa62..79f4687 100644
--- a/c/planarity.h
+++ b/c/planarity.h
@@ -38,7 +38,7 @@ int commandLine(int argc, char *argv[]);
 int legacyCommandLine(int argc, char *argv[]);
 int menu();
 
-char Mode,
+extern char Mode,
      OrigOut,
      EmbeddableOut,
      ObstructedOut,
@@ -49,7 +49,7 @@ void Reconfigure();
 
 /* Low-level Utilities */
 #define MAXLINE 1024
-char Line[MAXLINE];
+extern char Line[MAXLINE];
 
 void Message(char *message);
 void ErrorMessage(char *message);
