From a48246f1ca70912e18815aafcdd6ce7ece58a933 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Wed, 14 Jul 2021 18:21:35 +0200
Subject: [PATCH] Update Makefile

---
 src/Makefile | 52 ----------------------------------------------------
 1 file changed, 52 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 8d4741ee..da6f127d 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -9,57 +9,6 @@ SHELL		= /bin/bash
 # enumlib library paths
 LBD1 = ../symlib/src
 
-FOUND = false
-ifeq (${F90},gfortran)  # gfortran compiler
-  ifeq (${DEBUG},false)
-     FFLAGS = -O3 -ffree-line-length-none -fno-underscoring -I${LBD1}
-     FOUND = true
-  else
-     FFLAGS = -fPIC -g -fbounds-check -Wall -ffree-line-length-none -fno-underscoring -I${LBD1}
-     FOUND = true
-  endif
-endif
-
-ifeq (${F90},ifc)  # Intel compiler
-  FFLAGS = -fPIC -g -error-limit 7 -traceback -check bounds -warn  -I${LBD1}
-  FOUND = true
-endif
-
-ifeq (${F90},ifort)  # Intel compiler
-  ifeq (${DEBUG},false)
-     FFLAGS =  -fPIC -O3 -I${LBD1}
-     FOUND = true
-  else
-#  F90 =  /opt/intel/fc/10.0.016/bin/ifort
-     FFLAGS =  -fPIC -g -debug -error-limit 7 -heap-arrays -traceback -check bounds -warn -I${LBD1}
-     FOUND = true
-#-prof-use -prof-dir .
-   endif
-endif
-
-
-ifeq (${F90},xlf90) # IBM compiler
-  FFLAGS = -g -C -fPIC -qsuffix=f=f90  -I${LBD1}
-  FOUND = true
-endif
-ifeq (${F90},f95) # Absoft PPC compiler
-#  FFLAGS =  -profile -p ${LBDR}
-  FFLAGS = -g -Rb -Rc  -fPIC -nodefaultmod -p ${LBD1} #-ea
-# B80  show entry in subprograms ; Rb bounds; Rc array conformance;
-# z2 warning level
-# -O3 optimization
-# -ea stop after one error
-  FOUND = true
-endif
-ifeq (${F90},)  # If the variable isn't set, make sure compilation bombs
-error:
-	echo Error environmental variable "F90" not set!
-endif
-ifneq (${FOUND},true) # If it's set but no matching flags then bomb
-error:
-	echo Error: makefile doesn\'t have flags for this compiler
-endif
-
 # if type "python" > /dev/null; then
 #     python irevision.py derivative_structure_generator.f90 version.enum
 # fi
@@ -81,7 +30,6 @@ lib: pre_comp libenum.a
 
 pre_comp:
 	+$(MAKE) -C ../symlib/src
-	sed -i.bak "s/\(write(23,'(A)')\)\(.*\)/\1 \"`git describe --tags --dirty --abbrev=4`\"/" derivative_structure_generator.f90
 
 libenum.a: ${OBJS}
 	ar ru $@ $?
