diff --git a/cmake/gauxc-dep-versions.cmake b/cmake/gauxc-dep-versions.cmake
index 8ab0aa11..da69257f 100644
--- a/cmake/gauxc-dep-versions.cmake
+++ b/cmake/gauxc-dep-versions.cmake
@@ -17,7 +17,7 @@ set( GAUXC_GAU2GRID_REPOSITORY https://github.com/dgasmith/gau2grid.git )
 set( GAUXC_GAU2GRID_REVISION   v2.0.6 )
 
 set( GAUXC_INTEGRATORXX_REPOSITORY https://github.com/wavefunction91/IntegratorXX.git )
-set( GAUXC_INTEGRATORXX_REVISION   cf2917c64916583cef1081011beab3085b66e352 )
+set( GAUXC_INTEGRATORXX_REVISION   1369be58d7a3235dac36d75dd964fef058830622 )
 
 set( GAUXC_HIGHFIVE_REPOSITORY https://github.com/BlueBrain/HighFive.git )
 set( GAUXC_HIGHFIVE_REVISION 805f0e13d09b47c4b01d40682621904aa3b31bb8 )
diff --git a/src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_dd_psi.hpp b/src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_dd_psi.hpp
index 4a6b1138..211a4abb 100644
--- a/src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_dd_psi.hpp
+++ b/src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_dd_psi.hpp
@@ -17,7 +17,9 @@
 #include "host/local_host_work_driver.hpp"
 #include <gauxc/molgrid/defaults.hpp>
 #include <stdexcept>
+#ifdef GAUXC_ENABLE_OPENMP
 #include <omp.h>
+#endif
 
 namespace GauXC::detail {
 template <typename ValueType>
@@ -96,12 +98,16 @@ void ReferenceReplicatedXCHostIntegrator<ValueType>::
 
   // Loop over tasks
   const size_t ntasks = tasks.size();
+  #ifdef GAUXC_ENABLE_OPENMP
   #pragma omp parallel
+  #endif
   {
 
   XCHostData<value_type> host_data; // Thread local host data
 
+  #ifdef GAUXC_ENABLE_OPENMP
   #pragma omp for schedule(dynamic) reduction(+:dd_Psi[:natom * ldPsi])
+  #endif
   for( size_t iT = 0; iT < ntasks; ++iT ) {
 
     // Alias current task
diff --git a/src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_dd_psi_potential.hpp b/src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_dd_psi_potential.hpp
index fad35a29..58b9eddf 100644
--- a/src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_dd_psi_potential.hpp
+++ b/src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_dd_psi_potential.hpp
@@ -16,10 +16,13 @@
 #include "integrator_util/spherical_harmonics.hpp"
 #include "host/local_host_work_driver.hpp"
 #include <stdexcept>
-#include <omp.h>
 #include "host/blas.hpp"
 #include "host/util.hpp"
 
+#ifdef GAUXC_ENABLE_OPENMP
+#include <omp.h>
+#endif
+
 namespace GauXC::detail {
 template <typename ValueType>
 void ReferenceReplicatedXCHostIntegrator<ValueType>::
@@ -92,12 +95,16 @@ void ReferenceReplicatedXCHostIntegrator<ValueType>::
   // Loop over tasks
   const size_t ntasks = tasks.size();
 
+  #ifdef GAUXC_ENABLE_OPENMP
   #pragma omp parallel
+  #endif
   {
 
   XCHostData<value_type> host_data; // Thread local host data
 
+  #ifdef GAUXC_ENABLE_OPENMP
   #pragma omp for schedule(dynamic)
+  #endif
   for( size_t iT = 0; iT < ntasks; ++iT ) {
 
     // Alias current task
