diff --git a/cmake/gauxc-dep-versions.cmake b/cmake/gauxc-dep-versions.cmake
index d27e7a0e..7946af25 100644
--- a/cmake/gauxc-dep-versions.cmake
+++ b/cmake/gauxc-dep-versions.cmake
@@ -19,5 +19,5 @@ set( GAUXC_GAU2GRID_REVISION   v2.0.6 )
 set( GAUXC_INTEGRATORXX_REPOSITORY https://github.com/wavefunction91/IntegratorXX.git )
 set( GAUXC_INTEGRATORXX_REVISION   1369be58d7a3235dac36d75dd964fef058830622 )
 
-set( GAUXC_HIGHFIVE_REPOSITORY https://github.com/BlueBrain/HighFive.git )
+set( GAUXC_HIGHFIVE_REPOSITORY https://github.com/highfive-devs/HighFive.git )
 set( GAUXC_HIGHFIVE_REVISION 805f0e13d09b47c4b01d40682621904aa3b31bb8 )
diff --git a/src/external/CMakeLists.txt b/src/external/CMakeLists.txt
index fa1f7f37..46612c81 100644
--- a/src/external/CMakeLists.txt
+++ b/src/external/CMakeLists.txt
@@ -15,20 +15,23 @@ if( GAUXC_ENABLE_HDF5 )
   if(HDF5_FOUND)
     set(GAUXC_HAS_HDF5 TRUE CACHE BOOL "" FORCE)
     message(STATUS "Enabling HDF5 Bindings")
-    message(STATUS "HighFive REPO = ${GAUXC_HIGHFIVE_REPOSITORY}")
-    message(STATUS "HighFive REV  = ${GAUXC_HIGHFIVE_REVISION}  ")
-    FetchContent_Declare( HighFive
-      GIT_REPOSITORY ${GAUXC_HIGHFIVE_REPOSITORY}
-      GIT_TAG        ${GAUXC_HIGHFIVE_REVISION}  
-    )
+    find_package(HighFive QUIET)
+    if(NOT HighFive_FOUND)
+      message(STATUS "HighFive REPO = ${GAUXC_HIGHFIVE_REPOSITORY}")
+      message(STATUS "HighFive REV  = ${GAUXC_HIGHFIVE_REVISION}  ")
+      FetchContent_Declare( HighFive
+        GIT_REPOSITORY ${GAUXC_HIGHFIVE_REPOSITORY}
+        GIT_TAG        ${GAUXC_HIGHFIVE_REVISION}  
+      )
     
-    set(HIGHFIVE_USE_BOOST OFF CACHE BOOL "" )
-    set(HIGHFIVE_UNIT_TESTS OFF CACHE BOOL "" )
-    set(HIGHFIVE_EXAMPLES OFF CACHE BOOL "" )
-    #set(HIGHFIVE_PARALLEL_HDF5 ON CACHE BOOL "" )
-    set(HIGHFIVE_BUILD_DOCS OFF CACHE BOOL "" )
-    FetchContent_MakeAvailable( HighFive )
+      set(HIGHFIVE_USE_BOOST OFF CACHE BOOL "" )
+      set(HIGHFIVE_UNIT_TESTS OFF CACHE BOOL "" )
+      set(HIGHFIVE_EXAMPLES OFF CACHE BOOL "" )
+      #set(HIGHFIVE_PARALLEL_HDF5 ON CACHE BOOL "" )
+      set(HIGHFIVE_BUILD_DOCS OFF CACHE BOOL "" )
+      FetchContent_MakeAvailable( HighFive )
     
+    endif()
     target_sources( gauxc PRIVATE hdf5_write.cxx hdf5_read.cxx )
     target_link_libraries( gauxc PUBLIC HighFive )
   else()
