diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae1db53..5130eb2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,25 +57,19 @@ set(BUILD_SAMPLES OFF CACHE BOOL "")
 set(BUILD_TEST OFF CACHE BOOL "")
 set(BUILD_SHARED_LIBS OFF CACHE BOOL "")
 
-set(BUILD_TESTING OFF CACHE BOOL "")
-set(WITH_GFLAGS OFF CACHE BOOL "")
-add_subdirectory(third_party/glog)
+FIND_PACKAGE(glog REQUIRED)
 target_link_libraries(dynolog_lib PUBLIC glog::glog)
 
-set(GFLAGS_BUILD_TESTING OFF CACHE BOOL "")
-add_subdirectory(third_party/gflags)
+find_package(gflags REQUIRED)
 target_link_libraries(dynolog_lib PUBLIC gflags::gflags)
 
-# https://github.com/nlohmann/json#cmake
-set(JSON_BuildTests OFF CACHE INTERNAL "")
-add_subdirectory(third_party/json)
+find_package(nlohmann_json REQUIRED)
 target_link_libraries(dynolog_lib PUBLIC nlohmann_json::nlohmann_json)
 
-add_subdirectory(third_party/pfs)
-target_include_directories(dynolog_lib PUBLIC third_party/pfs/include)
+find_package(psf REQUIRED)
 target_link_libraries(dynolog_lib PUBLIC pfs)
 
-add_subdirectory(third_party/fmt)
+find_package(fmt REQUIRED)
 target_link_libraries(dynolog_lib PUBLIC fmt::fmt)
 
 if(USE_ODS_GRAPH_API)
