From 04d3b6aee79c1f4b7bf900006102877cd314dd6b Mon Sep 17 00:00:00 2001
From: Michel Zou <xantares09@hotmail.com>
Date: Fri, 11 Jan 2019 20:46:06 +0100
Subject: [PATCH 1/5] Add pybullet install rule

---
 examples/pybullet/CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/examples/pybullet/CMakeLists.txt b/examples/pybullet/CMakeLists.txt
index 2417a3ab2..d8fa12576 100644
--- a/examples/pybullet/CMakeLists.txt
+++ b/examples/pybullet/CMakeLists.txt
@@ -168,4 +168,11 @@ ENDIF(WIN32)
 
 TARGET_LINK_LIBRARIES(pybullet BulletExampleBrowserLib BulletFileLoader BulletWorldImporter BulletSoftBody BulletDynamics BulletCollision BulletInverseDynamicsUtils BulletInverseDynamics LinearMath OpenGLWindow gwen BussIK Bullet3Common ${PYTHON_LIBRARIES})
 
+IF(WIN32)
+  SET(PYTHON_SITE_PACKAGES Lib/site-packages CACHE PATH "Python install path")
+ELSE()
+  SET(PYTHON_SITE_PACKAGES lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages CACHE PATH "Python install path")
+ENDIF()
+
+INSTALL(TARGETS pybullet DESTINATION ${PYTHON_SITE_PACKAGES})
 

From ea515048e934d9c7df8cb157716e5ff336d6b25f Mon Sep 17 00:00:00 2001
From: Michel Zou <xantares09@hotmail.com>
Date: Fri, 11 Jan 2019 20:47:36 +0100
Subject: [PATCH 2/5] OSX Python extension is .so

---
 examples/pybullet/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/examples/pybullet/CMakeLists.txt b/examples/pybullet/CMakeLists.txt
index d8fa12576..caa4c21fb 100644
--- a/examples/pybullet/CMakeLists.txt
+++ b/examples/pybullet/CMakeLists.txt
@@ -163,6 +163,9 @@ IF(WIN32)
 	SET_TARGET_PROPERTIES(pybullet PROPERTIES SUFFIX ".pyd" )
 ENDIF(WIN32)
 
+IF (APPLE)
+  SET_TARGET_PROPERTIES(pybullet PROPERTIES SUFFIX ".so" )
+ENDIF()
 
 
 

From 46e2d280a8b96526d83841b8d375247d15c47a75 Mon Sep 17 00:00:00 2001
From: Michel Zou <xantares09@hotmail.com>
Date: Fri, 11 Jan 2019 20:52:38 +0100
Subject: [PATCH 3/5] Print build time only in debug mode

---
 examples/pybullet/pybullet.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/examples/pybullet/pybullet.c b/examples/pybullet/pybullet.c
index b506fa183..9763d89e7 100644
--- a/examples/pybullet/pybullet.c
+++ b/examples/pybullet/pybullet.c
@@ -10738,7 +10738,9 @@ initpybullet(void)
 	SpamError = PyErr_NewException("pybullet.error", NULL, NULL);
 	Py_INCREF(SpamError);
 	PyModule_AddObject(m, "error", SpamError);
+#ifndef NDEBUG
 	printf("pybullet build time: %s %s\n", __DATE__, __TIME__);
+#endif
 
 	Py_AtExit(b3pybulletExitFunc);
 

From e9112a9f5f94325e69a0901078d08834e56c776e Mon Sep 17 00:00:00 2001
From: Michel Zou <xantares09@hotmail.com>
Date: Sat, 12 Jan 2019 10:39:59 +0100
Subject: [PATCH 4/5] Install examples lib

---
 examples/ExampleBrowser/CMakeLists.txt        | 5 +++++
 examples/OpenGLWindow/CMakeLists.txt          | 5 +++++
 examples/ThirdPartyLibs/BussIK/CMakeLists.txt | 5 +++++
 examples/ThirdPartyLibs/Gwen/CMakeLists.txt   | 6 ++++++
 4 files changed, 21 insertions(+)

diff --git a/examples/ExampleBrowser/CMakeLists.txt b/examples/ExampleBrowser/CMakeLists.txt
index 8298f4137..ef0789e42 100644
--- a/examples/ExampleBrowser/CMakeLists.txt
+++ b/examples/ExampleBrowser/CMakeLists.txt
@@ -71,6 +71,11 @@ IF (BUILD_SHARED_LIBS)
 	ENDIF(WIN32)
 ENDIF(BUILD_SHARED_LIBS)
 
+INSTALL(TARGETS BulletExampleBrowserLib
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX})
+
 ####################
 #
 # Bullet Example Browser main app
diff --git a/examples/OpenGLWindow/CMakeLists.txt b/examples/OpenGLWindow/CMakeLists.txt
index 91a9a014f..6027bda2d 100644
--- a/examples/OpenGLWindow/CMakeLists.txt
+++ b/examples/OpenGLWindow/CMakeLists.txt
@@ -73,3 +73,8 @@ if (BUILD_SHARED_LIBS)
 endif()
 
 #target_link_libraries(OpenGLWindow ${OPENGL_gl_LIBRARY})
+
+INSTALL(TARGETS OpenGLWindow
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX})
diff --git a/examples/ThirdPartyLibs/BussIK/CMakeLists.txt b/examples/ThirdPartyLibs/BussIK/CMakeLists.txt
index 96dc16025..10ec3114c 100644
--- a/examples/ThirdPartyLibs/BussIK/CMakeLists.txt
+++ b/examples/ThirdPartyLibs/BussIK/CMakeLists.txt
@@ -8,3 +8,8 @@ FILE(GLOB BussIK_SRCS "*.cpp" )
 FILE(GLOB BussIK_HDRS "*.h" )
 
 ADD_LIBRARY(BussIK ${BussIK_SRCS} ${BussIK_HDRS})
+
+INSTALL(TARGETS BussIK
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX})
diff --git a/examples/ThirdPartyLibs/Gwen/CMakeLists.txt b/examples/ThirdPartyLibs/Gwen/CMakeLists.txt
index 9e6af37ad..7ca7656a2 100644
--- a/examples/ThirdPartyLibs/Gwen/CMakeLists.txt
+++ b/examples/ThirdPartyLibs/Gwen/CMakeLists.txt
@@ -26,3 +26,9 @@ IF(WIN32 OR APPLE)
   target_link_libraries(gwen ${OPENGL_gl_LIBRARY})
 ENDIF()
 ENDIF()
+
+
+INSTALL(TARGETS gwen
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX})

From fd405446374043d4a98e8e237ea94281d51e7e12 Mon Sep 17 00:00:00 2001
From: Michel Zou <xantares09@hotmail.com>
Date: Sat, 12 Jan 2019 11:36:28 +0100
Subject: [PATCH 5/5] On OSX link with dynamic_lookup

---
 examples/pybullet/CMakeLists.txt | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/examples/pybullet/CMakeLists.txt b/examples/pybullet/CMakeLists.txt
index caa4c21fb..4e7581ee2 100644
--- a/examples/pybullet/CMakeLists.txt
+++ b/examples/pybullet/CMakeLists.txt
@@ -169,7 +169,14 @@ ENDIF()
 
 
 
-TARGET_LINK_LIBRARIES(pybullet BulletExampleBrowserLib BulletFileLoader BulletWorldImporter BulletSoftBody BulletDynamics BulletCollision BulletInverseDynamicsUtils BulletInverseDynamics LinearMath OpenGLWindow gwen BussIK Bullet3Common ${PYTHON_LIBRARIES})
+TARGET_LINK_LIBRARIES(pybullet BulletExampleBrowserLib BulletFileLoader BulletWorldImporter BulletSoftBody BulletDynamics BulletCollision BulletInverseDynamicsUtils BulletInverseDynamics LinearMath OpenGLWindow gwen BussIK Bullet3Common)
+
+IF (WIN32)
+  TARGET_LINK_LIBRARIES(pybullet ${PYTHON_LIBRARIES})
+ELSEIF (APPLE)
+  SET_TARGET_PROPERTIES(pybullet PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+ENDIF ()
+# else Linux: dont link
 
 IF(WIN32)
   SET(PYTHON_SITE_PACKAGES Lib/site-packages CACHE PATH "Python install path")
