diff --git a/setup.py b/setup.py
index 3a64682..87a5c64 100644
--- a/setup.py
+++ b/setup.py
@@ -15,7 +15,7 @@ setup(
     long_description=long_description,
     long_description_content_type="text/markdown",
     license='MIT',
-    install_requires=['numpy', 'opencv-python>=3', 'lxml'],
+    install_requires=['numpy', 'opencv>=3', 'lxml'],
     packages=['imantics'],
     python_requires='>=2.7',
     zip_safe=False,
