From 1059dfe5b363300cf7b76b309ac839400911486a Mon Sep 17 00:00:00 2001
From: Joshua Downer <joshua.downer@gmail.com>
Date: Mon, 17 Aug 2020 11:19:42 -0400
Subject: [PATCH] ul_c_interface: use ctypes.util.find_library

---
 uldaq/ul_c_interface.py | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git uldaq/ul_c_interface.py uldaq/ul_c_interface.py
index e98f933..399c931 100644
--- uldaq/ul_c_interface.py
+++ uldaq/ul_c_interface.py
@@ -12,10 +12,13 @@ from .ul_structs import DaqInChanDescriptor, MemDescriptor, DaqOutChanDescriptor
 from .ul_enums import DaqEventType
 from sys import platform
 
-if platform.startswith('darwin'):
-    lib = CDLL('libuldaq.dylib')
-else:
-    lib = CDLL('libuldaq.so')
+import ctypes.util
+
+libname = ctypes.util.find_library("uldaq")
+if libname is None:
+    raise ImportError("Unable to find 'uldaq' library")
+
+lib = CDLL(libname)
 
 
 #
-- 
2.17.1

