From a18e077766df5cec7d5f7e460d4f88adc356c17e Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sat, 24 Feb 2018 10:35:12 -0600
Subject: [PATCH] Implement mempcpy() via memcpy()

---
 src/kerberosgss.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/kerberosgss.c b/src/kerberosgss.c
index ae440e0..506ad44 100644
--- a/src/kerberosgss.c
+++ b/src/kerberosgss.c
@@ -29,6 +29,13 @@ static void set_gss_error(OM_uint32 err_maj, OM_uint32 err_min);
 extern PyObject *GssException_class;
 extern PyObject *KrbException_class;
 
+#if defined(__APPLE__)
+void *mempcpy(void *dest, const void *src, size_t n) {
+   memcpy(dest, src, n);
+   return dest + n;
+}
+#endif
+
 char* server_principal_details(const char* service, const char* hostname)
 {
     char match[1024];
-- 
2.16.1

