diff --git a/CMakeLists.txt b/CMakeLists.txt
index b63cee1..7cae61e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,6 +30,10 @@ endif()
 
 option(Heffte_SEQUENTIAL_TESTING  "Force all tests to run sequentially to avoid oversubscribing resources."  OFF)
 
+if (Heffte_ENABLE_FFTW)
+    set(Heffte_FFTW_LIBOMP "" CACHE STRING "The OpenMP library used for FFTW if different from the default for the CXX compiler, e.g., if using clang and fftw uses gnu use -DHeffte_FFTW_LIBOMP=-lgomp")
+endif()
+
 # add the file with common CMake helper macros
 include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/heffte_macros.cmake")
 
diff --git a/cmake/FindHeffteFFTW.cmake b/cmake/FindHeffteFFTW.cmake
index 8e862e4..3e96d91 100644
--- a/cmake/FindHeffteFFTW.cmake
+++ b/cmake/FindHeffteFFTW.cmake
@@ -73,14 +73,20 @@ if (NOT FFTW_LIBRARIES)
         PREFIX ${FFTW_ROOT}
         VAR FFTW_LIBRARIES
         REQUIRED "fftw3" "fftw3f"
-        OPTIONAL "fftw3_omp" "fftw3f_omp" "fftw3_threads" "fftw3f_threads"
-                               )
-    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-        find_package(OpenMP REQUIRED)
-        list(APPEND FFTW_LIBRARIES ${OpenMP_CXX_LIBRARIES})
-    else()
-        if ("fftw3_omp" IN_LIST FFTW_LIBRARIES)
-            list(APPEND FFTW_LIBRARIES "-lgomp")
+        OPTIONAL "fftw3_threads" "fftw3f_threads")
+    heffte_find_fftw_libraries(
+        PREFIX ${FFTW_ROOT}
+        VAR FFTW_OMP_LIBRARIES
+        REQUIRED ""
+        OPTIONAL "fftw3_omp" "fftw3f_omp")
+
+    if (FFTW_OMP_LIBRARIES)
+        list(APPEND FFTW_OMP_LIBRARIES "${FFTW_LIBRARIES}")
+        if (Heffte_FFTW_LIBOMP)
+            list(APPEND FFTW_LIBRARIES "${Heffte_FFTW_LIBOMP}")
+        else()
+            find_package(OpenMP REQUIRED)
+            list(APPEND FFTW_LIBRARIES OpenMP::OpenMP_CXX)
         endif()
     endif()
 endif()
diff --git a/cmake/HeffteConfig.cmake b/cmake/HeffteConfig.cmake
index 824dd5d..dc7b432 100644
--- a/cmake/HeffteConfig.cmake
+++ b/cmake/HeffteConfig.cmake
@@ -10,6 +10,9 @@ include("${CMAKE_CURRENT_LIST_DIR}/HeffteTargets.cmake")
 
 if (@Heffte_ENABLE_FFTW@ AND NOT TARGET Heffte::FFTW)
     add_library(Heffte::FFTW INTERFACE IMPORTED GLOBAL)
+    if ("@OpenMP_FOUND@")
+        find_package(OpenMP REQUIRED)
+    endif()
     target_link_libraries(Heffte::FFTW INTERFACE @FFTW_LIBRARIES@)
     set_target_properties(Heffte::FFTW PROPERTIES INTERFACE_INCLUDE_DIRECTORIES @FFTW_INCLUDES@)
 endif()
