--- CMake/macros.cmake.old      2018-02-26 00:54:06.391674683 -0600
+++ CMake/macros.cmake  2018-02-26 00:54:19.327396916 -0600
@@ -38,7 +38,7 @@
 
     set_target_properties(${LIBRARY_NAME}
         PROPERTIES
-        OUTPUT_NAME ${LIBRARY_NAME}
+        OUTPUT_NAME ${LIBNAME}
         DEBUG_OUTPUT_NAME ${LIBRARY_NAME_DEBUG}
         CLEAN_DIRECT_OUTPUT TRUE
         COMPILE_DEFINITIONS "FL_LIBRARY"
