From 9865615cbc0b7938e3cc1e298c0daddb3a731933 Mon Sep 17 00:00:00 2001
From: Daniel Zagaynov <kotopesutility@altlinux.org>
Date: Mon, 24 Jul 2023 12:40:18 +0300
Subject: [PATCH] FIX: Migrate from inspect.getfullargspec to inspect.signature
 (#99)

---
 mdp/configuration.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mdp/configuration.py b/mdp/configuration.py
index 86637827..58c21529 100644
--- a/mdp/configuration.py
+++ b/mdp/configuration.py
@@ -4,7 +4,10 @@
 import mdp
 # python 2/3 compatibility
 try:
-    from inspect import getfullargspec as getargs
+    # Migrate to signature as recommended in
+    # https://docs.python.org/3/library/inspect.html#inspect.getfullargspec
+    from inspect import signature
+    getargs = lambda func: (signature(func)._parameters, None)
 except ImportError:
     from inspect import getargspec as getargs
 import tempfile
