From ff7e42f75ecad616bc2d18f3833bb6a38c8d8d2e Mon Sep 17 00:00:00 2001
From: NiMlr <nils.mueller-p43@rub.de>
Date: Sun, 27 Mar 2022 23:05:21 +0200
Subject: [PATCH] FIX: Deprecation of numpy.typeDict

---
 mdp/utils/routines.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/mdp/utils/routines.py b/mdp/utils/routines.py
index 4e058632..0867e9e9 100644
--- a/mdp/utils/routines.py
+++ b/mdp/utils/routines.py
@@ -292,9 +292,8 @@ def get_dtypes(typecodes_key, _safe=True):
     return types
 
 
-_UNSAFE_DTYPES = [numx.typeDict[d] for d in
-                  ['float16', 'float96', 'float128', 'complex192', 'complex256']
-                  if d in numx.typeDict]
+_UNSAFE_DTYPES = [numx.dtype(d).type for d in
+                  ['float16', 'float128', 'complex256']]
 
 
 def nongeneral_svd(A, range=None, **kwargs):
