--- src/c_client.py.orig	2017-01-19 12:01:35.092979043 -0500
+++ src/c_client.py	2017-01-19 12:02:31.000282059 -0500
@@ -1930,14 +1930,14 @@
                     # from the request size and divide that by the member size
                     return '(((R->length * 4) - sizeof('+ self.c_type + '))/'+'sizeof('+field.type.member.c_wiretype+'))'
                 else:
-		    # use the accessor to get the start of the list, then
-		    # compute the length of it by subtracting it from
+                    # use the accessor to get the start of the list, then
+                    # compute the length of it by subtracting it from
                     # the adress of the first byte after the end of the
                     # request
-		    after_end_of_request = '(((char*)R) + R->length * 4)'
-		    start_of_list = '%s(R)' % (field.c_accessor_name)
+                    after_end_of_request = '(((char*)R) + R->length * 4)'
+                    start_of_list = '%s(R)' % (field.c_accessor_name)
                     bytesize_of_list = '%s - (char*)(%s)' % (after_end_of_request, start_of_list)
-		    return '(%s) / sizeof(%s)' % (bytesize_of_list, field.type.member.c_wiretype)
+                    return '(%s) / sizeof(%s)' % (bytesize_of_list, field.type.member.c_wiretype)
             else:
                 raise Exception(
                     "lengthless lists with varsized members are not supported. Fieldname '%s'"
