--- configure.orig	2017-01-16 22:53:28.550279544 -0500
+++ configure	2017-01-16 22:55:33.037859308 -0500
@@ -20254,7 +20254,7 @@
 have_win32="no"
 lt_enable_auto_import=""
 case $host_os in
-mingw*)
+mingw*|*msys*)
         have_win32="yes"
         lt_enable_auto_import="-Wl,--enable-auto-import"
         ;;
@@ -20429,7 +20429,7 @@
 esac
 
 case $host_os in
-        *mingw*)
+        *mingw*|*msys*)
                 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lws2_32" >&5
 $as_echo_n "checking for main in -lws2_32... " >&6; }
 if ${ac_cv_lib_ws2_32_main+:} false; then :
