From aa58e517457f4d855f67c4f5b084e125b50ee2b9 Mon Sep 17 00:00:00 2001
From: Asaf Kahlon <asafka7@gmail.com>
Date: Fri, 24 Jan 2020 10:28:41 +0200
Subject: [PATCH 1/1] Disable GNU_SOURCE

Signed-off-by: Asaf Kahlon <asafka7@gmail.com>
---
 config.h.in    | 4 ----
 src/internal.h | 3 ---
 2 files changed, 7 deletions(-)

diff --git a/config.h.in b/config.h.in
index d5fb26d..4d1f55c 100644
--- a/config.h.in
+++ b/config.h.in
@@ -81,10 +81,6 @@
 #ifndef _ALL_SOURCE
 # undef _ALL_SOURCE
 #endif
-/* Enable GNU extensions on systems that have them.  */
-#ifndef _GNU_SOURCE
-# undef _GNU_SOURCE
-#endif
 /* Enable threading extensions on Solaris.  */
 #ifndef _POSIX_PTHREAD_SEMANTICS
 # undef _POSIX_PTHREAD_SEMANTICS
diff --git a/src/internal.h b/src/internal.h
index 9ac17e0..e01c15b 100644
--- a/src/internal.h
+++ b/src/internal.h
@@ -15,9 +15,6 @@
 # if !defined(_REENTRANT)
 #  define _REENTRANT
 # endif
-# if !defined(_GNU_SOURCE)
-#  define _GNU_SOURCE
-# endif
 # if !defined(_LARGEFILE_SOURCE)
 #  define _LARGEFILE_SOURCE
 # endif
-- 
2.20.1

