From 539f3545dea182db92a24e46b1fc72ad833cab72 Mon Sep 17 00:00:00 2001
From: Sameeul B Samee <sameeul@gmail.com>
Date: Fri, 21 Apr 2023 15:01:59 -0400
Subject: [PATCH] Install shared library in Windows

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef0da8b..36551af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,6 +95,7 @@ install(TARGETS dcmcjp2k EXPORT fmjpeg2kTargets
 install(TARGETS fmjpeg2k EXPORT fmjpeg2kTargets
   LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin
 )
 
 install(
-- 
2.30.2

