commit 9cc04d4adb4536597ff82e5da7314763b0132641
Author: jay <jlaura@asu.edu>
Date:   Wed Mar 29 07:29:24 2017 -0700

    Added the OSX Makefile

diff --git a/recipes/Makefile.mac b/recipes/Makefile.mac
new file mode 100644
index 0000000..6da024a
--- /dev/null
+++ b/recipes/Makefile.mac
@@ -0,0 +1,6 @@
+include Makefile
+
+CC=clang++
+COPTS=-fPIC -O2 -m64 -Wall -stdlib=libc++ -std=c++11
+LDOPTS=-shared -Wl,-install_name,$(LIBRARY)
+INSTDIR=$(PWD)/mac
