diff --git a/Python/Core.i b/Python/Core.i
index acf79b3..117b800 100644
--- a/Python/Core.i
+++ b/Python/Core.i
@@ -195,7 +195,7 @@ namespace std
 %include "../Core/ShearedVoxelMesh.h"
 %include "../Core/StaggeredVoxelMesh.h"
 %include "../Core/RotatedVoxelMesh.h"
-%include "../core/PrismVoxelMesh.h"
+%include "../Core/PrismVoxelMesh.h"
 %include "../Core/TetgenMesh.h"
 %include "../Core/PeriodicBoundaries.h"
 %include "../Core/ShearedPeriodicBoundaries.h"
@@ -359,9 +359,3 @@ namespace TexGen
 		return *self * dScalar;
 	}
 }
-
-
-
-
-
-
