diff --git a/Core/CMakeLists.txt b/Core/CMakeLists.txt
index 88c3094..b3fc44e 100644
--- a/Core/CMakeLists.txt
+++ b/Core/CMakeLists.txt
@@ -40,8 +40,8 @@ ADD_DEFINITIONS(-DEXPORT)

 IF(WIN32)
 TARGET_LINK_LIBRARIES(TexGenCore triangle triangle-api Octree TinyXML CSparse tetgenlib
-${CMAKE_CURRENT_SOURCE_DIR}/../OctreeRefinement/libsc
-${CMAKE_CURRENT_SOURCE_DIR}/../OctreeRefinement/libp4est)
+${CMAKE_CURRENT_SOURCE_DIR}/../OctreeRefinement/libsc.lib
+${CMAKE_CURRENT_SOURCE_DIR}/../OctreeRefinement/libp4est.lib)
 ELSE(WIN32)
 TARGET_LINK_LIBRARIES(TexGenCore triangle triangle-api Octree TinyXML CSparse tetgenlib)
 ENDIF(WIN32)
