diff --git a/Core/Misc.h b/Core/Misc.h
index b2ca9cf..d02b4f8 100644
--- a/Core/Misc.h
+++ b/Core/Misc.h
@@ -82,16 +82,16 @@ namespace TexGen
 	}
 
 	/// Used to sort double-int pairs
-	struct LessPairDoubleInt : public std::binary_function<std::pair<double, int>, std::pair<double, int>, bool>
-	{
-		bool operator()(std::pair<double, int> x, std::pair<double, int> y) { return x.first < y.first; }
-	};
+	// struct LessPairDoubleInt : public std::binary_function<std::pair<double, int>, std::pair<double, int>, bool>
+	// {
+	// 	bool operator()(std::pair<double, int> x, std::pair<double, int> y) { return x.first < y.first; }
+	// };
 
 	/// Used to sort double-XYZ pairs
-	struct LessPairDoubleXYZ : public std::binary_function<std::pair<double, XYZ>, std::pair<double, XYZ>, bool>
-	{
-		bool operator()(std::pair<double, XYZ> x, std::pair<double, XYZ> y) { return x.first < y.first; }
-	};
+	// struct LessPairDoubleXYZ : public std::binary_function<std::pair<double, XYZ>, std::pair<double, XYZ>, bool>
+	// {
+	// 	bool operator()(std::pair<double, XYZ> x, std::pair<double, XYZ> y) { return x.first < y.first; }
+	// };
 
 	struct COLOR
 	{
@@ -297,4 +297,3 @@ namespace TexGen
 	}
 
 };	// namespace TexGen
-
