diff --git a/Core/ObjectContainer.h b/Core/ObjectContainer.h
index f039a68..a7d0fd1 100644
--- a/Core/ObjectContainer.h
+++ b/Core/ObjectContainer.h
@@ -88,11 +88,11 @@ namespace TexGen
 	};
 
 	/// Used to sort double-objectref pairs
-	template <typename T>
-	struct LessPairDoubleObjectRef : public std::binary_function<std::pair<double, CObjectContainer<T> >, std::pair<double, CObjectContainer<T> >, bool>
-	{
-		bool operator()(std::pair<double, CObjectContainer<T> > x, std::pair<double, CObjectContainer<T> > y) { return x.first < y.first; }
-	};
+	// template <typename T>
+	// struct LessPairDoubleObjectRef : public std::binary_function<std::pair<double, CObjectContainer<T> >, std::pair<double, CObjectContainer<T> >, bool>
+	// {
+	// 	bool operator()(std::pair<double, CObjectContainer<T> > x, std::pair<double, CObjectContainer<T> > y) { return x.first < y.first; }
+	// };
 
 
 
@@ -136,8 +136,3 @@ namespace TexGen
 
 
 };	// namespace TexGen
-
-
-
-
-
