diff --git a/Makefile b/Makefile
index 9aa4360..6d28b62 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,17 @@
 LDFLAGS := $(LDFLAGS) -lcpu_features
+prefix = /usr/local
+exec_prefix = $(prefix)
+bindir = $(exec_prefix)/bin
+
+all: subarch-select
 
 subarch-select: main.cpp
-	g++ $(CXXFLAGS) $(CPPFLAGS) -o $@ $< $(LDFLAGS)
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -o $@ $< $(LDFLAGS)
 
 clean:
-	rm -f subarch-select
+	$(RM) -f subarch-select
+
+install:
+	cp subarch-select ${bindir}/
 
 .PHONY: clean
