--- fast-barnes-py-1.0.0.orig/setup.py	2023-02-17 22:58:11.000000000 +0100
+++ fast-barnes-py-1.0.0/setup.py	2024-06-12 18:02:07.484439377 +0200
@@ -15,8 +15,8 @@
     author_email='bruno.zuercher@meteoswiss.ch',
     url='https://github.com/MeteoSwiss/fast-barnes-py/',
     license='BSD 3-Clause License',
-#    long_description=open('README.md').read(),
-    long_description=open('doc/PyPI_Doc.md').read(),
+   long_description=open('README.md').read(),
+    # long_description=open('doc/PyPI_Doc.md').read(),
     long_description_content_type='text/markdown',
     packages=['fastbarnes', 'fastbarnes.util'],
     install_requires=['numpy', 'scipy', 'numba'],
