--- cmake/GeneratePkgConfigFile.cmake  2023-12-21 14:55:31.228171789 -0600
+++ cmake/GeneratePkgConfigFile.cmake      2023-12-21 14:56:22.620962284 -0600
@@ -1,5 +1,5 @@
 function(generate_pkg_config_file in_file)
-  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(prefix /opt/anaconda1anaconda2anaconda3)
   set(exec_prefix \${prefix})
   set(libdir \${exec_prefix}/${CMAKE_INSTALL_LIBDIR})
   set(includedir \${prefix}/${CMAKE_INSTALL_INCLUDEDIR})
