diff -Nu /dev/null b/libsecp256k1.cmake.pc.in
--- /dev/null
+++ b/libsecp256k1.cmake.pc.in
@@ -0,0 +1,16 @@
+#   @COMMENT_TITLE@
+#   @COMMENT_COPYRIGHT@
+#   @COMMENT_LICENCE@
+
+prefix=@PACKAGE_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: @PACKAGE_NAME@
+Description: @PACKAGE_DESCRIPTION@
+URL: @PACKAGE_URL@
+Version: @PACKAGE_VERSION@
+Cflags: -I${includedir} @PACKAGE_CFLAGS@
+Libs: -L${libdir} -lsecp256k1
+
