diff --git a/setup.py b/setup.py
index 48d35db8..c6e3a2ed 100644
--- a/setup.py
+++ b/setup.py
@@ -46,6 +46,8 @@ class CMakeBuild(build_ext):
 
         if platform.system() == "Windows":
             cmake_args += ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}'.format(cfg.upper(), extdir)]
+            # force cmake to use VS15 2017 with no respect for environment variables
+            cmake_args += ['-G', 'Visual Studio 15 2017']
             if sys.maxsize > 2**32:
                 cmake_args += ['-A', 'x64']
             build_args += ['--', '/m']
