From 024d053d58812871b414e12970d576a870e0a03b Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 11 Jun 2020 08:37:18 +0200
Subject: [PATCH] Use iprpy-data

---
 iprPy/__init__.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/iprPy/__init__.py b/iprPy/__init__.py
index d1703f0..475bbc2 100644
--- a/iprPy/__init__.py
+++ b/iprPy/__init__.py
@@ -8,11 +8,12 @@
     The absolute path to the iprPy package's library directory.
 """
 # Standard Python libraries
+import os
 from pathlib import Path
 
 # Define package-specific directories
 rootdir = Path(__file__).absolute().parent
-libdir = Path(rootdir.parent, 'library')
+libdir = Path(os.environ['CONDA_PREFIX'], 'share', 'iprpy')
 
 # Read version from VERSION file
 with open(Path(rootdir, 'VERSION')) as version_file:
@@ -46,4 +47,4 @@
 
 from . import workflow
 
-from .check_modules import check_modules
\ No newline at end of file
+from .check_modules import check_modules
