diff --git a/makefiles/Makefile.python.mk b/makefiles/Makefile.python.mk
index dea1513c3..1f3b353b0 100755
--- a/makefiles/Makefile.python.mk
+++ b/makefiles/Makefile.python.mk
@@ -27,7 +27,7 @@ python:
 test_python: python
 
 else
-python: install_python_modules pyinit pycp pyalgorithms pygraph pylp
+python: pyinit pycp pyalgorithms pygraph pylp
 test_python: test_python_examples
 BUILT_LANGUAGES +=, python
 endif
@@ -278,11 +278,7 @@ $(PYPI_ARCHIVE_TEMP_DIR) : $(OR_TOOLS_PYTHON_GEN_SCRIPTS)
 	$(COPY) tools$SREADME.pypi $(PYPI_ARCHIVE_TEMP_DIR)$Sortools$SREADME.txt
 	$(COPY) LICENSE-2.0.txt $(PYPI_ARCHIVE_TEMP_DIR)$Sortools
 	$(COPY) tools$Ssetup.py $(PYPI_ARCHIVE_TEMP_DIR)$Sortools
-ifeq ($(PYTHON3),true)
-	$(SED) -i -e 's/ORTOOLS_PYTHON_VERSION/py3-ortools/' $(PYPI_ARCHIVE_TEMP_DIR)$Sortools$Ssetup.py
-else
 	$(SED) -i -e 's/ORTOOLS_PYTHON_VERSION/ortools/' $(PYPI_ARCHIVE_TEMP_DIR)$Sortools$Ssetup.py
-endif
 	$(SED) -i -e 's/VVVV/$(OR_TOOLS_VERSION)/' $(PYPI_ARCHIVE_TEMP_DIR)$Sortools$Ssetup.py
 	$(SED) -i -e 's/PROTOBUF_TAG/$(PROTOBUF_TAG)/' $(PYPI_ARCHIVE_TEMP_DIR)$Sortools$Ssetup.py
 ifeq ($(SYSTEM),win)
@@ -331,4 +327,4 @@ pypi_upload: $(PYPI_UPLOAD)
 
 detect_python:
 	@echo PYTHON3 = $(PYTHON3)
-	@echo SWIG_PYTHON3_FLAG = $(SWIG_PYTHON3_FLAG)
\ No newline at end of file
+	@echo SWIG_PYTHON3_FLAG = $(SWIG_PYTHON3_FLAG)
diff --git a/makefiles/Makefile.third_party.unix b/makefiles/Makefile.third_party.unix
index 63b1ae89d..b8149752d 100755
--- a/makefiles/Makefile.third_party.unix
+++ b/makefiles/Makefile.third_party.unix
@@ -52,7 +52,7 @@ endif
 
 # Detect if patchelf is needed
 ifeq ($(PLATFORM), LINUX)
-    PATCHELF=dependencies/install/bin/patchelf
+    PATCHELF=patchelf
 endif
 
 ACLOCAL_TARGET = \
diff --git a/tools/fix_libraries_on_linux.sh b/tools/fix_libraries_on_linux.sh
index cf9228300..53ecf4eea 100755
--- a/tools/fix_libraries_on_linux.sh
+++ b/tools/fix_libraries_on_linux.sh
@@ -1,4 +1,4 @@
-dependencies/install/bin/patchelf --set-rpath '$ORIGIN' temp/or-tools*/lib/libjniortools.so
-dependencies/install/bin/patchelf --set-rpath '$ORIGIN' temp/or-tools*/lib/libcvrptw_lib.so
-dependencies/install/bin/patchelf --set-rpath '$ORIGIN' temp/or-tools*/lib/libdimacs.so
-dependencies/install/bin/patchelf --set-rpath '$ORIGIN/../../lib' temp/or-tools*/bin/libGoogle.OrTools.so
+patchelf --set-rpath '$ORIGIN' temp/or-tools*/lib/libjniortools.so
+patchelf --set-rpath '$ORIGIN' temp/or-tools*/lib/libcvrptw_lib.so
+patchelf --set-rpath '$ORIGIN' temp/or-tools*/lib/libdimacs.so
+patchelf --set-rpath '$ORIGIN/../../lib' temp/or-tools*/bin/libGoogle.OrTools.so
diff --git a/tools/fix_python_libraries_on_linux.sh b/tools/fix_python_libraries_on_linux.sh
index 99d2893e4..9526ea02a 100755
--- a/tools/fix_python_libraries_on_linux.sh
+++ b/tools/fix_python_libraries_on_linux.sh
@@ -1,4 +1,4 @@
-dependencies/install/bin/patchelf --set-rpath '$ORIGIN/../../ortools' $1/ortools/ortools/constraint_solver/_pywrapcp.so
-dependencies/install/bin/patchelf --set-rpath '$ORIGIN/../../ortools' $1/ortools/ortools/linear_solver/_pywraplp.so
-dependencies/install/bin/patchelf --set-rpath '$ORIGIN/../../ortools' $1/ortools/ortools/algorithms/_pywrapknapsack_solver.so
-dependencies/install/bin/patchelf --set-rpath '$ORIGIN/../../ortools' $1/ortools/ortools/graph/_pywrapgraph.so
+patchelf --set-rpath '$ORIGIN/../../ortools' $1/ortools/ortools/constraint_solver/_pywrapcp.so
+patchelf --set-rpath '$ORIGIN/../../ortools' $1/ortools/ortools/linear_solver/_pywraplp.so
+patchelf --set-rpath '$ORIGIN/../../ortools' $1/ortools/ortools/algorithms/_pywrapknapsack_solver.so
+patchelf --set-rpath '$ORIGIN/../../ortools' $1/ortools/ortools/graph/_pywrapgraph.so
