diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4943c83..372049d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,9 +96,6 @@ set(ASD_binary sd)
 # Find the vernacular compiler name
 string(REGEX MATCH gfortran|g95|ifort|pgf90|f95|f77|nv current_compiler  ${CMAKE_Fortran_COMPILER})
 message(STATUS "Matched compiler: ${current_compiler}.  ${CMAKE_Fortran_COMPILER} ${FCNAME}")
-if(current_compiler)
-   string(APPEND ASD_binary ".${current_compiler}")
-endif()
 # add a `cuda` tag to the binary if used.
 if(USE_CUDA)
    string(APPEND ASD_binary ".cuda")
