diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7620d3..9c2bd38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,6 +47,7 @@ set(LIB ${CMAKE_SOURCE_DIR}/lib)
 message(STATUS "The lib is ${LIB}")
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${LIB})
 message(STATUS "The lib is ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+message(STATUS "The cmake prefix path is ${CMAKE_PREFIX_PATH}")
 file(MAKE_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 ##############################################################################################
 
@@ -137,13 +138,7 @@ ADD_LIBRARY(${UppASD_LIB} ${lib_type})
 set(MKL_LINK static)
 find_package(MKL CONFIG)
 
-include(${CMAKE_MODULE_PATH}/FindFFTW.cmake)
-if (FFTW_FOUND)
-   message(STATUS "FFTW found ${FFTW_FOUND} ${FFTW_INCLUDES} ${FFTW_INCLUDE_DIR}")
-else (FFTW_FOUND)
-   set(FFTW_LIB "")
-   set(FFTW_LIBRARIES "")
-endif (FFTW_FOUND)
+find_package(FFTW CONFIG)
 ##############################################################################################
 
 ##############################################################################################
@@ -387,8 +382,6 @@ elseif(USE_FFTW)
    add_compile_definitions(USE_FFTW)
    message(STATUS "FFT-> ${USE_FFTW} ")
    #include(${CMAKE_MODULE_PATH}/FindFFTW.cmake)
-   include_directories(${FFTW_INCLUDE_DIR})
-   message(STATUS "FFTW: -I${FFTW_INCLUDE_DIR}  -L${FFTW_LIBRARIES} ${FFTW_LIB}")
 endif()
 ##############################################################################################
 
@@ -548,7 +541,7 @@ TARGET_LINK_LIBRARIES(${UppASD_LIB}
    ${LAPACK_LIBRARIES}
    ${CMAKE_THREAD_LIBS_INIT} 
    ${OVF_LIB} 
-   ${FFTW_LIB}  
+   FFTW3::fftw3
    ${CUDA_curand_LIBRARY}
    ${OpenMP_Fortran_LIBRARIES}
    )
@@ -558,9 +551,9 @@ TARGET_LINK_LIBRARIES(${UppASD_EXE}
    ${UppASD_LIB}
    ${BLAS_LIBRARIES}
    ${LAPACK_LIBRARIES}
-   ${CMAKE_THREAD_LIBS_INIT} 
+   ${CMAKE_THREAD_LIBS_INIT}
    ${OVF_LIB} 
-   ${FFTW_LIB}  
+   FFTW3::fftw3
    ${CUDA_curand_LIBRARY}
    ${OpenMP_Fortran_LIBRARIES}
    )
@@ -584,18 +577,6 @@ if (NOT BUILD_PYTHON)
 endif(NOT BUILD_PYTHON)
 ##############################################################################################
 
-##############################################################################################
-# Copy static library to ./lib
-##############################################################################################
-add_custom_command(
-   TARGET ${UppASD_LIB} POST_BUILD
-   COMMAND ${CMAKE_COMMAND} -E copy
-   ${CMAKE_BINARY_DIR}/lib${UppASD_LIB}.a
-   ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
-   #COMMAND ${CMAKE_COMMAND} -E copy
-   #${CMAKE_BINARY_DIR}/_uppasd*.so
-   #${CMAKE_SOURCE_DIR}/python/uppasd/
-   )
 
 
 ###############################################################################################
diff --git a/cmake/SetFortranFlags.cmake b/cmake/SetFortranFlags.cmake
index f7baccf..943768a 100644
--- a/cmake/SetFortranFlags.cmake
+++ b/cmake/SetFortranFlags.cmake
@@ -74,20 +74,6 @@ SET_COMPILE_FLAG(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
                  Fortran "-cpp"
                          "-fpp")
 
-# There is some bug where -march=native doesn't work on Mac
-IF(APPLE)
-    SET(GNUNATIVE "-mtune=native")
-ELSE()
-    SET(GNUNATIVE "-march=native")
-ENDIF()
-# Optimize for the host's architecture
-SET_COMPILE_FLAG(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
-                 Fortran "-xHost"        # Intel
-                         ${GNUNATIVE}    # GNU
-                         "-ta=host"      # Portland Group
-                         "/QxHost"       # Intel Windows
-                )
-
 ###################
 ### DEBUG FLAGS ###
 ###################
