diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7620d3..1f97aa3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -584,18 +584,6 @@ if (NOT BUILD_PYTHON)
 endif(NOT BUILD_PYTHON)
 ##############################################################################################
 
-##############################################################################################
-# Copy static library to ./lib
-##############################################################################################
-add_custom_command(
-   TARGET ${UppASD_LIB} POST_BUILD
-   COMMAND ${CMAKE_COMMAND} -E copy
-   ${CMAKE_BINARY_DIR}/lib${UppASD_LIB}.a
-   ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
-   #COMMAND ${CMAKE_COMMAND} -E copy
-   #${CMAKE_BINARY_DIR}/_uppasd*.so
-   #${CMAKE_SOURCE_DIR}/python/uppasd/
-   )
 
 
 ###############################################################################################
diff --git a/cmake/SetFortranFlags.cmake b/cmake/SetFortranFlags.cmake
index f7baccf..943768a 100644
--- a/cmake/SetFortranFlags.cmake
+++ b/cmake/SetFortranFlags.cmake
@@ -74,20 +74,6 @@ SET_COMPILE_FLAG(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
                  Fortran "-cpp"
                          "-fpp")
 
-# There is some bug where -march=native doesn't work on Mac
-IF(APPLE)
-    SET(GNUNATIVE "-mtune=native")
-ELSE()
-    SET(GNUNATIVE "-march=native")
-ENDIF()
-# Optimize for the host's architecture
-SET_COMPILE_FLAG(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
-                 Fortran "-xHost"        # Intel
-                         ${GNUNATIVE}    # GNU
-                         "-ta=host"      # Portland Group
-                         "/QxHost"       # Intel Windows
-                )
-
 ###################
 ### DEBUG FLAGS ###
 ###################
