diff --git a/source/pt_driver.f90 b/source/pt_driver.f90
index c3f3da5..7a2c07a 100644
--- a/source/pt_driver.f90
+++ b/source/pt_driver.f90
@@ -291,7 +291,7 @@ contains
 
             call calc_mavrg(Natom,Mensemble,pt_emomM(:,:,:,irep),mavg)
 
-            print '(a,i4,a,i4,a,f12.6a,f12.6,a,f12.6)', 'Replica #',irep, " List position: ", templist(irep), &
+            print '(a,i4,a,i4,a,f12.6,a,f12.6,a,f12.6)', 'Replica #',irep, " List position: ", templist(irep), &
                " Average magnetization:", mavg, " Temperature: ", iptemp(irep),"  Energy: " , repene(irep)/Natom
 
       end do
diff --git a/source/sx_driver.f90 b/source/sx_driver.f90
index 7266786..404a368 100644
--- a/source/sx_driver.f90
+++ b/source/sx_driver.f90
@@ -341,7 +341,7 @@ contains
 
          call calc_mavrg(Natom,Mensemble,sx_emomM(:,:,:,irep),mavg)
 
-         print '(a,i4,a,i4,a,f12.6a,f12.6,a,f12.6)', 'Replica #',irep, " List position: ", templist(irep), &
+         print '(a,i4,a,i4,a,f12.6,a,f12.6,a,f12.6)', 'Replica #',irep, " List position: ", templist(irep), &
             " Average magnetization:", mavg, " Temperature: ", iptemp(irep),"  Energy: " , repene_sum(irep)!/Natom
 
       end do
