From e44dbef06c95189ccabf91dbe73cb42f49fca51d Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 16 Feb 2025 01:47:19 +0100
Subject: [PATCH 1/2] Always return library found for conda-forge

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 45ecc06..aae3ed4 100644
--- a/setup.py
+++ b/setup.py
@@ -49,6 +49,7 @@ def find_library(header):
     # if include dir or library dir is None, it means that the library is in
     # standard paths and don't need to be added to compiler / linker search
     # paths
+    return True, None, None
 
     searching_for = f"Searching for {header}"
 
@@ -100,8 +101,6 @@ def make_extension():
     return CppExtension(
         name="torchvision_extra_decoders.extra_decoders_lib",
         sources=sorted(str(s) for s in sources),
-        include_dirs=[heic_include_dir, avif_include_dir],
-        library_dirs=[heic_library_dir, avif_library_dir],
         libraries=["heif", "avif"],
         extra_compile_args={"cxx": ["-g0"]},
     )
-- 
2.43.0

