From 85fd975b85b05b256c13499c8d55205de577cbcd Mon Sep 17 00:00:00 2001
From: Martin Lang <martin.lang@mpsd.mpg.de>
Date: Wed, 19 Nov 2025 11:33:44 +0100
Subject: [PATCH 2/7] Remove copying of the static library to outside the build
 directory

---
 CMakeLists.txt | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b39a319..567fa09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -577,18 +577,6 @@ if (NOT BUILD_PYTHON)
 endif(NOT BUILD_PYTHON)
 ##############################################################################################
 
-##############################################################################################
-# Copy static library to ./lib
-##############################################################################################
-add_custom_command(
-   TARGET ${UppASD_LIB} POST_BUILD
-   COMMAND ${CMAKE_COMMAND} -E copy
-   ${CMAKE_BINARY_DIR}/lib${UppASD_LIB}.a
-   ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
-   #COMMAND ${CMAKE_COMMAND} -E copy
-   #${CMAKE_BINARY_DIR}/_uppasd*.so
-   #${CMAKE_SOURCE_DIR}/python/uppasd/
-   )
 
 
 ###############################################################################################
