From 028812f81d689539ddeafd87aca11d7ab20664d0 Mon Sep 17 00:00:00 2001
From: Martin Lang <martin.lang@mpsd.mpg.de>
Date: Wed, 19 Nov 2025 11:34:52 +0100
Subject: [PATCH 3/7] Remove compiler optimization for host architecture

---
 cmake/SetFortranFlags.cmake | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/cmake/SetFortranFlags.cmake b/cmake/SetFortranFlags.cmake
index f7baccf..943768a 100644
--- a/cmake/SetFortranFlags.cmake
+++ b/cmake/SetFortranFlags.cmake
@@ -74,20 +74,6 @@ SET_COMPILE_FLAG(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
                  Fortran "-cpp"
                          "-fpp")
 
-# There is some bug where -march=native doesn't work on Mac
-IF(APPLE)
-    SET(GNUNATIVE "-mtune=native")
-ELSE()
-    SET(GNUNATIVE "-march=native")
-ENDIF()
-# Optimize for the host's architecture
-SET_COMPILE_FLAG(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
-                 Fortran "-xHost"        # Intel
-                         ${GNUNATIVE}    # GNU
-                         "-ta=host"      # Portland Group
-                         "/QxHost"       # Intel Windows
-                )
-
 ###################
 ### DEBUG FLAGS ###
 ###################
