From 9b4a871f0e535ef462516038512ef355b0c16770 Mon Sep 17 00:00:00 2001
From: Martin Lang <martin.lang@mpsd.mpg.de>
Date: Wed, 19 Nov 2025 11:39:42 +0100
Subject: [PATCH 4/7] Remove -lgfortran, which is not supported by flang

---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 567fa09..9ff40c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -536,7 +536,6 @@ mark_as_advanced( ENABLE_BLACS ENABLE_BLAS95 ENABLE_CDFT ENABLE_CPARDISO ENABLE_
 # Add the needed libraries and special compiler flags
 ##############################################################################################
 TARGET_LINK_LIBRARIES(${UppASD_LIB}
-   -lgfortran
    ${BLAS_LIBRARIES}
    ${LAPACK_LIBRARIES}
    ${CMAKE_THREAD_LIBS_INIT} 
@@ -547,7 +546,6 @@ TARGET_LINK_LIBRARIES(${UppASD_LIB}
    )
 
 TARGET_LINK_LIBRARIES(${UppASD_EXE}
-   -lgfortran
    ${UppASD_LIB}
    ${BLAS_LIBRARIES}
    ${LAPACK_LIBRARIES}
