diff --git a/tasks.py b/tasks.py
index 0e8acb2..12658d6 100644
--- a/tasks.py
+++ b/tasks.py
@@ -80,7 +80,7 @@ def _run_tests(ctx, mark=None, skip=None, junitxml=None):
         cmd.extend(['--junitxml', junitxml])
     cmd.append('-v')
     cmd.append('-x')
-    cmd.extend(['--rerun', '1'])
+    # cmd.extend(['--rerun', '1'])

     marks = []
     if mark is not None:
@@ -94,7 +94,7 @@ def _run_tests(ctx, mark=None, skip=None, junitxml=None):

     if not WINDOWS:
         run(ctx, "ls -a .coverage*")
-        run(ctx, "coverage combine")
+        run(ctx, "coverage combine || true")


 @task
