diff --git a/nbgrader/tests/apps/test_nbgrader.py b/nbgrader/tests/apps/test_nbgrader.py
index 941908b..65964dc 100644
--- a/nbgrader/tests/apps/test_nbgrader.py
+++ b/nbgrader/tests/apps/test_nbgrader.py
@@ -30,6 +30,10 @@ class TestNbGrader(BaseTestApp):

     def test_check_version(self, capfd):
         """Is the version the same regardless of how we run nbgrader?"""
-        out1 = "\n".join(run_command(["nbgrader", "--version"]).splitlines()).strip()
-        out2 = run_nbgrader(["--version"], stdout=True).strip()
+        out1 = '\n'.join(
+            run_command(["nbgrader", "--version"]).splitlines()[-3:]
+        ).strip()
+        out2 = '\n'.join(
+            run_nbgrader(["--version"], stdout=True).splitlines()[-3:]
+        ).strip()
         assert out1 == out2
