From cecaa0a21939dc78657bf1ab08d9770a9ac32ca1 Mon Sep 17 00:00:00 2001
From: Mick Foley <mickf@apple.com>
Date: Mon, 14 May 2018 16:20:37 -0700
Subject: [PATCH] Cleanup - Remove unneeded __packed directive

---
 src/vmm/io/vrtc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vmm/io/vrtc.c b/src/vmm/io/vrtc.c
index 864fe22..1cbebc9 100644
--- a/src/vmm/io/vrtc.c
+++ b/src/vmm/io/vrtc.c
@@ -75,7 +75,7 @@ struct rtcdev {
 	uint8_t nvram[36];
 	uint8_t century;
 	uint8_t nvram2[128 - 51];
-} __packed;
+};
 CTASSERT(sizeof(struct rtcdev) == 128);
 CTASSERT(offsetof(struct rtcdev, century) == RTC_CENTURY);
 
