From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Wed, 25 Oct 2023 12:21:30 -0400
Subject: [PATCH] Let Windows build create a shared library by passing
 "-no-undefined"

---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index a537b1c..068e2fc 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -8,4 +8,4 @@ AM_CFLAGS = -Wall -ffast-math $(FFTW3F_CFLAGS)
 lib_LTLIBRARIES = libosmodsp.la
 
 libosmodsp_la_SOURCES = cfile.c cxvec.c cxvec_math.c iqbal.c
-libosmodsp_la_LDFLAGS = -version-info $(LIBVERSION) $(FFTW3F_LIBS)
+libosmodsp_la_LDFLAGS = -version-info $(LIBVERSION) $(FFTW3F_LIBS) -no-undefined
-- 
2.40.0

