From 8a4edb75e4fa456be53f09fa8b9cb26eaf052768 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Fri, 4 Mar 2022 11:01:49 +0100
Subject: [PATCH] Reorder python-related find_package calls

---
 bindings/CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/bindings/CMakeLists.txt b/bindings/CMakeLists.txt
index d01d045..29cee8b 100644
--- a/bindings/CMakeLists.txt
+++ b/bindings/CMakeLists.txt
@@ -4,9 +4,8 @@
 
 if(ICUB_MODELS_COMPILE_PYTHON_BINDINGS)
 
+    find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
     find_package(pybind11 REQUIRED)
-    find_package(Python3 COMPONENTS Interpreter REQUIRED)
-
 
     # define new line accordingly to the operating system
     if (WIN32)
