diff --git a/cmake/ImportPybind11.cmake b/cmake/ImportPybind11.cmake
index 34d009c..a819e4c 100644
--- a/cmake/ImportPybind11.cmake
+++ b/cmake/ImportPybind11.cmake
@@ -6,7 +6,7 @@
 macro(import_pybind11)
   # If the target already exists, do nothing
   if(NOT TARGET pybind11::pybind11)
-    find_package(pybind11 2.6.2 EXACT QUIET)
+    find_package(pybind11)
     if(TARGET pybind11::pybind11)
       message(STATUS "Found pybind11 at ${pybind11_DIR}")
     else()
