diff --git a/CMakeLists.txt b/CMakeLists.txt
index b3754ef..7e27bc1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,6 +73,7 @@ find_package(Boost
   COMPONENTS regex unit_test_framework program_options filesystem system
   QUIET
 )
+if(NOT SKIP_LIBRARY_BUILD)
 include(ImportNauty)
 import_nauty()
 include(ImportOutcome)
@@ -83,6 +84,10 @@ include(ImportRingDecomposerLib)
 import_rdl()
 
 add_subdirectory(src)
+else()
+  include(ImportMolassembler)
+  import_molassembler()
+endif()
 
 if(SCINE_BUILD_DOCS)
   include(DoxygenDocumentation)
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 379e3898..681fc451 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -11,8 +11,7 @@ set_target_properties(scine_molassembler PROPERTIES
   LIBRARY_OUTPUT_DIRECTORY "scine_molassembler"
 )
 target_include_directories(scine_molassembler SYSTEM PRIVATE ${pybind11_INCLUDE_DIR} ${PYBIND11_INCLUDE_DIR})
-target_compile_options(scine_molassembler PRIVATE ${MOLASSEMBLER_CXX_FLAGS})
-target_link_libraries(scine_molassembler PRIVATE ${MOLASSEMBLER_STATIC_TARGET} Eigen3::Eigen)
+target_link_libraries(scine_molassembler PRIVATE Scine::Molassembler Eigen3::Eigen)
 
 include(FindPythonModule)
 find_python_module(pytest)
