From 2d78ad9c3d8d4a32bfef770f4cd25d2c86d76dc2 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Fri, 24 Jun 2022 17:50:26 +0200
Subject: [PATCH] Disable CMake module installation

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ba7de435c4..12f02a087ec 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -710,7 +710,7 @@ if (OGRE_INSTALL_SAMPLES OR OGRE_INSTALL_SAMPLES_SOURCE)
 endif ()
 
 # Install CMake modules
-add_subdirectory(CMake)
+# add_subdirectory(CMake)
 
 # provide option to install dependencies on Windows
 include(InstallDependencies)
