--- IntegralImage.h	2017-05-04 12:23:02.659428722 +0200
+++ IntegralImage.h	2017-05-04 12:23:29.689430687 +0200
@@ -20,6 +20,7 @@
 #define INTEGRALIMAGE_H
 
 #include <Matrix3D.h>
+#include <cmath>
 #include <vector>
 
 // a box of center (x,y,z), radius (rx,ry,rz)
@@ -73,7 +74,7 @@
     template<typename S>
     static inline S removeNaN( S val )
     {
-    if ( isnan(val) || isinf(val) )
+    if ( std::isnan(val) || std::isinf(val) )
         return (S)0;
     else
         return val;
--- ROIData.h	2017-05-04 12:23:49.352765488 +0200
+++ ROIData.h	2017-05-04 12:24:11.919433878 +0200
@@ -19,6 +19,7 @@
 #ifndef _ROI_DATA_H_
 #define _ROI_DATA_H_
 
+#include <cmath>
 #include <Eigen/Dense>
 #include <vector>
 
@@ -145,7 +146,7 @@
         for (long int i=0; i < numElem; i++)
         {
             auto total = _rotMatrices[i].sum();
-            if ( isnan(total) || isinf(total) )
+            if ( std::isnan(total) || std::isinf(total) )
                 _rotMatrices[i].setIdentity();
         }
     }
