--- python/src/iiboost_python.cpp	2017-05-04 13:40:33.153470980 +0200
+++ python/src/iiboost_python.cpp	2017-05-04 13:41:04.950119078 +0200
@@ -46,7 +46,7 @@
         std::string str;
         ((BoosterModel *)modelPtr)->serializeToString( &str );
 
-        return PyString_FromString( str.c_str() );
+        return PyBytes_FromString( str.c_str() );
     }
 
     // create model from serialized string
--- python/CMakeLists.txt	2017-05-04 13:32:33.316722306 +0200
+++ python/CMakeLists.txt	2017-05-04 13:59:47.086580601 +0200
@@ -6,17 +6,17 @@
 #
 ######################################################################
 
-FIND_PACKAGE(PythonInterp 2 REQUIRED)
+FIND_PACKAGE(PythonInterp REQUIRED)
 
-IF(PYTHONINTERP_FOUND)
-    # check that Python version 2.x is used
-    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
-                         "import sys; print(sys.version[0])"
-                          OUTPUT_VARIABLE PYTHON_MAJOR_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
-    IF(NOT (${PYTHON_MAJOR_VERSION} EQUAL 2))
-        MESSAGE(FATAL_ERROR "Python bindings require Python 2.x.")
-    ENDIF()
-ENDIF()
+#IF(PYTHONINTERP_FOUND)
+#    # check that Python version 2.x is used
+#    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
+#                         "import sys; print(sys.version[0])"
+#                          OUTPUT_VARIABLE PYTHON_MAJOR_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+#    IF(NOT (${PYTHON_MAJOR_VERSION} EQUAL 2))
+#        MESSAGE(FATAL_ERROR "Python bindings require Python 2.x.")
+#    ENDIF()
+#ENDIF()
 
 ######################################################################
 #
@@ -24,21 +24,23 @@
 #
 ######################################################################
 
-execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
-                 "import sys; print sys.exec_prefix"
-                  OUTPUT_VARIABLE PYTHON_PREFIX OUTPUT_STRIP_TRAILING_WHITESPACE)
+find_package(PythonLibs REQUIRED)
 
-IF(APPLE AND ${PYTHON_PREFIX} MATCHES ".*framework.*")
-    SET(PYTHON_LIBRARIES "${PYTHON_PREFIX}/Python"
-        CACHE FILEPATH "Python libraries"
-        FORCE)
-ELSE()
-    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
-                     "import sys; skip = 2 if sys.platform.startswith('win') else 1; print 'python' + sys.version[0:3:skip]"
-                      OUTPUT_VARIABLE PYTHON_LIBRARY_NAME OUTPUT_STRIP_TRAILING_WHITESPACE)
-    FIND_LIBRARY(PYTHON_LIBRARIES ${PYTHON_LIBRARY_NAME} HINTS "${PYTHON_PREFIX}" 
-                 PATH_SUFFIXES lib lib64 libs DOC "Python libraries")
-ENDIF()
+#execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
+#                 "import sys; print(sys.exec_prefix)"
+#                  OUTPUT_VARIABLE PYTHON_PREFIX OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+#IF(APPLE AND ${PYTHON_PREFIX} MATCHES ".*framework.*")
+#    SET(PYTHON_LIBRARIES "${PYTHON_PREFIX}/Python"
+#        CACHE FILEPATH "Python libraries"
+#        FORCE)
+#ELSE()
+#    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
+#                     "import sys; skip = 2 if sys.platform.startswith('win') else 1; print('python' + sys.version[0:3:skip])"
+#                      OUTPUT_VARIABLE PYTHON_LIBRARY_NAME OUTPUT_STRIP_TRAILING_WHITESPACE)
+#    FIND_LIBRARY(PYTHON_LIBRARIES ${PYTHON_LIBRARY_NAME} HINTS "${PYTHON_PREFIX}" 
+#                 PATH_SUFFIXES lib lib64 libs DOC "Python libraries")
+#ENDIF()
 
 ######################################################################
 #
@@ -47,7 +49,7 @@
 ######################################################################
 
 execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
-                "from distutils.sysconfig import *; print get_python_inc()"
+                "from distutils.sysconfig import *; print(get_python_inc())"
                 OUTPUT_VARIABLE PYTHON_INCLUDE OUTPUT_STRIP_TRAILING_WHITESPACE)
 SET(PYTHON_INCLUDE_PATH ${PYTHON_INCLUDE}
     CACHE PATH "Path to Python include files"
@@ -61,7 +63,7 @@
 ######################################################################
 IF(NOT DEFINED IIBOOST_PYTHON_INSTALL_DIR OR IIBOOST_PYTHON_INSTALL_DIR MATCHES "^$")
     execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
-                     "from distutils.sysconfig import *; print get_python_lib(1)"
+                     "from distutils.sysconfig import *; print(get_python_lib(1))"
                       OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE)
     FILE(TO_CMAKE_PATH ${PYTHON_SITE_PACKAGES} IIBOOST_PYTHON_INSTALL_DIR)
 ENDIF()
