From 98cd4381de579271b2467baf2b7d8361b73749e7 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Tue, 22 Aug 2017 13:18:30 -0500
Subject: [PATCH] add petsc_include_as_c flag

---
 proteus/config/default.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/proteus/config/default.py b/proteus/config/default.py
index a31b784c..66ac0144 100644
--- a/proteus/config/default.py
+++ b/proteus/config/default.py
@@ -22,7 +22,7 @@ if sys.platform == 'darwin':
     platform_lapack_integer = '__CLPK_integer'
     platform_blas_h = r'<Accelerate/Accelerate.h>'
     platform_lapack_h = r'<Accelerate/Accelerate.h>'
-    platform_extra_compile_args = ['-flax-vector-conversions','-DPETSC_SKIP_COMPLEX']
+    platform_extra_compile_args = ['-flax-vector-conversions','-DPETSC_SKIP_COMPLEX', '-DPETSC_INCLUDE_AS_C']
 elif sys.platform == 'linux2':
     platform_extra_compile_args = ['-DPETSC_INCLUDE_AS_C', '-DPETSC_SKIP_COMPLEX']
     platform_extra_link_args = ['-L'+PROTEUS_LIB_DIR,'-Wl,-rpath,' + PROTEUS_LIB_DIR]
-- 
2.12.0
