diff --git scripts/clearh5.py scripts/clearh5.py
index 7de825a42..74e481a1e 100755
--- scripts/clearh5.py
+++ scripts/clearh5.py
@@ -40,7 +40,7 @@ def clearh5(filename,dataDir='.',addname="_clean", tCount=None, global_sync=True
         for gridChild in TemporalGridCollection.findall("Grid"):
             counter=int(gridChild.find('Time').attrib['Name'])
             if(counter >= tCount):
-              print counter
+              print(counter)
               Domain[0].remove(gridChild)
 
     import re
