--- a/README.md
+++ b/README.md
@@ -6,6 +6,7 @@

 ccimport 0.2.x support python 3.5.
 ccimport >= 0.3 support python 3.6-3.10.
+ccimport >=0.4.5 support 3.10-3.13

 ## Usage

--- a/ccimport/buildtools/writer.py
+++ b/ccimport/buildtools/writer.py
@@ -7,7 +7,11 @@
 from pathlib import Path
 from typing import Any, Dict, List, Optional, Tuple, Type, Union

-from ninja.ninja_syntax import Writer
+try:
+    from ninja.ninja_syntax import Writer
+except ImportError:
+    from ninja_syntax import Writer  # from system ninja
+
 from ccimport import compat
 from ccimport.constants import get_compiler_map, CXX, CUDACXX, HIPCXX
 from ccimport.buildmeta import BuildMeta
--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@
 # What packages are required for this module to be executed?
 REQUIRED = [
     "pybind11",
-    "ninja",
+    "ninja_syntax",
     "requests",
     "importlib-metadata>=2.0; python_version < \"3.8\"",
     "dataclasses; python_version == \"3.6\"",
--- /dev/null
+++ b/MANIFEST.in
@@ -0,0 +1 @@
+include version.txt
\ No newline at end of file
