diff --git a/src/cpp/skia-builder/build_skia.py b/src/cpp/skia-builder/build_skia.py
index 45ea674..8555bcd 100644
--- a/src/cpp/skia-builder/build_skia.py
+++ b/src/cpp/skia-builder/build_skia.py
@@ -234,11 +234,18 @@
         else [(build_base_dir, args.target_cpu)]
     )

+    build_args = SKIA_BUILD_ARGS
+    # let environment override gn default c/c++ compiler and linker
+    for var in ("CC", "CXX", "AR"):
+        v = os.environ.get(var)
+        if v is not None:
+            build_args.append('{}="{}"'.format(var.lower(), v))
+
     for build_dir, target_cpu in builds:
         build_skia(
             SKIA_SRC_DIR,
             build_dir,
-            SKIA_BUILD_ARGS,
+            build_args,
             target_cpu,
             env,
             args.shared_lib,
