diff --git a/setup.py b/setup.py
index 90c0179..6decb77 100644
--- a/setup.py
+++ b/setup.py
@@ -254,6 +254,11 @@
     if inside_sdist:
         build_cmd.append("--no-sync-deps")

+    if sys.platform.startswith('linux'):
+        gn_exe = os.path.abspath(os.path.join(skia_builder_dir, 'bin', 'linux64', 'gn'))
+        # use gn included in builder
+        build_cmd.append(f"--gn-path={gn_exe}")
+
     env = os.environ
     target_cpu = None
     if sys.platform == "win32":
