diff --git a/setup.py b/setup.py
index 90c0179..6decb77 100644
--- a/setup.py
+++ b/setup.py
@@ -18,7 +18,7 @@
 import re
 import argparse
 import shlex
-
+import shutil

 # export BUILD_SKIA_FROM_SOURCE=0 to not build libskia when building extension
 BUILD_SKIA_FROM_SOURCE = bool(int(os.environ.get("BUILD_SKIA_FROM_SOURCE", "1")))
@@ -276,6 +276,9 @@
     if target_cpu:
         build_cmd.extend(["--target-cpu", target_cpu])

+    # use gn from conda
+    build_cmd.extend([f"--gn-path={shutil.which('gn')}", "--no-fetch-gn"])
+
     subprocess.run(build_cmd, check=True, env=env)
     return build_dir
