--- build.sh.orig     2024-08-20 10:40:17.860801638 -0500
+++ build.sh          2024-08-20 10:40:08.708639616 -0500
@@ -106,4 +106,5 @@
     case $flavour in
-        macosx) (set -x; ${CC} -dynamiclib -o libblst$dll.dylib \
-                               -all_load libblst.a ${CFLAGS}); exit 0;;
+        macosx) (set -x; ${CC} -dynamiclib -o libblst.$PKG_MAJOR_VERSION$dll.dylib \
+                               -all_load libblst.a ${CFLAGS}; \
+                               ln -s libblst.$PKG_MAJOR_VERSION$dll.dylib libblst$dll.dylib); exit 0;;
         mingw*) sharedlib="blst.dll ${TOP}/build/win64/blst.def"
@@ -111,3 +112,3 @@
                 CFLAGS="${CFLAGS} -nostdlib -lgcc";;
-        *)      sharedlib=libblst$dll.so;;
+        *)      sharedlib=libblst$dll.so.$PKG_MAJOR_VERSION;;
     esac
@@ -115,3 +116,3 @@
                    -Wl,--whole-archive,libblst.a,--no-whole-archive ${CFLAGS} \
-                   -Wl,-Bsymbolic)
+                   -Wl,-Bsymbolic; ln -s $sharedlib libblst$dll.so)
 fi
