--- bindings/python/run.me.orig 2024-08-19 15:39:54.069388000 -0500
+++ bindings/python/run.me      2024-08-19 16:01:08.861055413 -0500
@@ -1,2 +1,1 @@
-#!/usr/bin/env python3

@@ -25,6 +25,2 @@

-if newer("../blst.h", "../libblst.a"):
-    print("building libblst.a...") or sys.stdout.flush()
-    subprocess.check_call(["../build.sh", "-fvisibility=hidden"] + sys.argv[1:], cwd="..")
-
 if newer("../blst.swg", "../blst.h", "../blst.hpp", "blst_wrap.cpp"):
@@ -35,3 +31,3 @@

-if newer("blst_wrap.cpp", "../libblst.a", "_blst.so"):
+if newer("blst_wrap.cpp", "../../libblst.so", "../../libblst.dylib", "_blst.so"):
     print("compiling _blst.so...") or sys.stdout.flush()
@@ -44,3 +40,8 @@
                 "-I"+sysconfig.get_config_var('INCLUDEPY'), "-I..",
-                "-O", "-Wall", "blst_wrap.cpp", "../libblst.a"])
+                "-O", "-Wall", "blst_wrap.cpp"])
+    cmd.append(f"../../libblst.so.{os.getenv('PKG_MAJOR_VERSION')}") if py_syst == 'linux' else None
+    cmd.append(f"..\\..\\blst-{os.getenv('PKG_MAJOR_VERSION')}.dll") if py_syst == 'win' else None
+    cmd.append(f"../../libblst.so.{os.getenv('PKG_MAJOR_VERSION')}"") if py_syst == 'darwin' else None
+    cmd.extend(["-L../../", "-lblst") if py_syst not in ['linux','win','darwin'] else None
     try:
