--- bindings/python/run.me.orig 2024-08-19 15:39:54.069388000 -0500
+++ bindings/python/run.me      2024-08-19 16:01:08.861055413 -0500
@@ -1,2 +1,2 @@
-#!/usr/bin/env python3
+

@@ -9,3 +9,3 @@

-here = re.split(r'/(?=[^/]*$)', sys.argv[0])
+here = re.split(r'\\|/', sys.argv[0])
 if len(here) > 1:
@@ -25,9 +25,9 @@

-if newer("../blst.swg", "../blst.h", "../blst.hpp", "blst_wrap.cpp"):
+if newer(os.path.join("..", "blst.swg"), os.path.join("..", "blst.h"), os.path.join("..", "blst.hpp"), "blst_wrap.cpp"):
     print("swig-ing...") or sys.stdout.flush()
     subprocess.check_call(["swig", "-c++", "-python", "-O",
-                                   "-o", "blst_wrap.cpp", "-outdir", ".",
-                                   "../blst.swg"])
+                           "-o", "blst_wrap.cpp", "-outdir", ".",
+                           os.path.join("..", "blst.swg")])

-if newer("blst_wrap.cpp", "../../libblst.so", "_blst.so"):
+if newer("blst_wrap.cpp", os.path.join("..", "..", "libblst.so"), "_blst.so"):
     print("compiling _blst.so...") or sys.stdout.flush()
@@ -39,6 +39,6 @@
     cmd.extend(["-o", "_blst.so", "-fPIC", "-fvisibility=hidden",
-                "-I"+sysconfig.get_config_var('INCLUDEPY'), "-I..",
-                "-O", "-Wall", "blst_wrap.cpp", "../../libblst.so"])
+                "-I" + sysconfig.get_config_var('INCLUDEPY'), "-I..",
+                "-O", "-Wall", "blst_wrap.cpp", os.path.join("..", "..", "libblst.so")])
     try:
-        pre = subprocess.check_output([cxx, "-dM", "-E", "-x", "c++", "/dev/null"],
+        pre = subprocess.check_output([cxx, "-dM", "-E", "-x", "c++", "NUL"],
                                       stderr=subprocess.STDOUT)
