From bb721a9a8639c972499de441bbdf07de8d05491e Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Thu, 27 Apr 2023 13:41:27 +0200
Subject: [PATCH] build: Do not treat compiler warnings as errors for now

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a20a087..f160f3f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,7 +110,10 @@ macro(remotery_create_target target_name linkage output_name)
 
     if(MSVC)
         target_compile_options(
-            ${target_name} PUBLIC /W4 /WX
+            ${target_name} PUBLIC /W4
+            # NOTE(jjerphan): treating warnings as error makes the installation fail.
+            # TODO: remove "/W4"-emitted warnings and use "/WX"
+            # /WX
         )
     else()
         target_compile_options(
-- 
2.40.0

