From d095f9082cb24d41ca941e1aceeff6775fb0bc0d Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Thu, 27 Apr 2023 14:24:50 +0200
Subject: [PATCH] build: Use "Remotery_static" as name for the static library
 on Windows

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f160f3f..a7872de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,7 +140,7 @@ if(REMOTERY_BUILD_STATIC_LIBS)
     # On Windows, a static library should use a different output name
     # to avoid name conflicts with the shared library.
     if(WIN32)
-        remotery_create_target(Remotery-static STATIC Remotery-static)
+        remotery_create_target(Remotery-static STATIC Remotery_static)
     else()
         remotery_create_target(Remotery-static STATIC Remotery)
     endif()
-- 
2.40.0

