From 10662b26dc07b8e2a8f34bb2439318c4c0840625 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Tue, 5 Mar 2024 16:13:11 +0100
Subject: [PATCH 1/2] Add missing include

---
 test/test.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/test.h b/test/test.h
index 45dd23a..014330f 100644
--- a/test/test.h
+++ b/test/test.h
@@ -10,6 +10,8 @@
    systems it'll probably be necessary to run each test individually rather
    than in groups as arranged below */
 
+#include <sys/select.h>
+
 #if 1
 #define TEST_SELFTEST		/* Perform internal self-test */
 #define TEST_LOWLEVEL		/* Test low-level functions */
-- 
2.37.3

