From ef39ded7f2ce0e69075a3d7e50b5dad6430ba317 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Tue, 5 Mar 2024 16:50:31 +0100
Subject: [PATCH 2/2] Use cf compiler

---
 tools/getcompiler.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/getcompiler.sh b/tools/getcompiler.sh
index 1148bbf..ac7f26a 100644
--- a/tools/getcompiler.sh
+++ b/tools/getcompiler.sh
@@ -12,6 +12,9 @@ fi
 
 # Juggle the args around to get them the way that we want them.
 
+echo $CC
+exit
+
 CC=$1
 OSNAME=$2
 shift 2
-- 
2.37.3

