From b72d2bb2a1dcfe38115817e0146ce51244d7bbd3 Mon Sep 17 00:00:00 2001
From: Christian Fobel <christian@fobel.net>
Date: Mon, 29 Apr 2019 15:16:01 -0400
Subject: [PATCH 3/3] fix: use native stdint.h for VS>=2010

---
 src/GKlib/gk_arch.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/GKlib/gk_arch.h b/src/GKlib/gk_arch.h
index e95d376..0b66de0 100644
--- a/src/GKlib/gk_arch.h
+++ b/src/GKlib/gk_arch.h
@@ -32,8 +32,13 @@
 
 
 #ifdef __MSC__
+#if (_MSC_VER < 1600)
   #include "ms_stdint.h"
   #include "ms_inttypes.h"
+#else
+  #include <stdint.h>
+  #include <inttypes.h>
+#endif
   #include "ms_stat.h"
 #else
 #ifndef SUNOS
-- 
2.14.2.windows.1

