diff --git a/xtb/aespot.f90 b/xtb/aespot.f90
index c1e9854..1b52ee9 100644
--- a/xtb/aespot.f90
+++ b/xtb/aespot.f90
@@ -1045,7 +1045,7 @@ subroutine ovlp2(thr,nat,nao,nbf,at,xyz,caoshell,saoshell,nprim,primcount,cont,a
    mm=nbf*(nbf+1)/2
    step=1.0d-4
    step2=0.50_wp/step
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp&         private(iat,ra,iatyp,jat,rb,jatyp,dx,dy,dz,rab2,ish,ishtyp, &
 !$omp&                 icao,naoi,iptyp,jshmax,jsh,jshtyp,jcao,naoj,ss,ip,iprim, &
 !$omp&                 alpi,jp,jprim,ab,est,s00,li,ci,stmp,lj,cc,ii,iao,jj,jao) &
diff --git a/xtb/bfgs.f90 b/xtb/bfgs.f90
index 92defed..828a19d 100644
--- a/xtb/bfgs.f90
+++ b/xtb/bfgs.f90
@@ -64,7 +64,7 @@ subroutine bfgs(nat3,gnorm,grad,grado,dx,hess)
    oodds  = 1.0_wp / dds
 
    if(dds > thrs .and. ddtd > thrs) then
-   !$omp parallel default(none) &
+   !$omp parallel default(shared) &
    !$omp shared(nat3,oodds,ooddtd,svec,tvec) &
    !$omp private(i,j,ii,ij,sdds,tddtd,temp) &
    !$omp shared(hess)
diff --git a/xtb/dftd4.f90 b/xtb/dftd4.f90
index c34b29d..d1db0aa 100644
--- a/xtb/dftd4.f90
+++ b/xtb/dftd4.f90
@@ -1406,7 +1406,7 @@ subroutine dispgrad(nat,ndim,at,q,xyz, &
    enddo
 
 !  print'(" * Entering first OMP section")'
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ii,iii,ia,iz,k,norm,dnorm,twf,tgw,dexpw,expw,gwk,dgwk)  &
 !$omp shared (nat,at,refn,refc,refcovcn,itbl,refq,wf,cn,g_a,g_c,q) &
 !$omp shared (gw,dgw,zvec,dzvec,dzdq)
@@ -1463,7 +1463,7 @@ subroutine dispgrad(nat,ndim,at,q,xyz, &
 !$omp end parallel
 
 !  print'(" * Entering second OMP section")'
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,j,ia,ja,ij,k,l,c6ij,dic6ij,djc6ij,disp,dizij,djzij,  &
 !$omp         rij,r2,r,r4r2ij,r0,oor6,oor8,oor10,door6,door8,door10)  &
 !$omp shared(nat,at,xyz,refn,itbl,zvec,dzvec,c6abns,par,dzdq) &
@@ -1555,7 +1555,7 @@ subroutine dispgrad(nat,ndim,at,q,xyz, &
 !  end select
 
 !  print'(" * Entering third OMP section")'
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,j,ia,ja,ij,rij,r2,r,drdx,den,rcovij,  &
 !$omp         expterm,dcndr,dtmp) reduction(+:g) &
 !$omp shared(nat,at,xyz,dc6dr,dc6dcn)
@@ -1916,7 +1916,7 @@ subroutine dabcappr(nat,ndim,at,xyz,par,  &
 
    eabc = 0.0_wp
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ia,j,ja,ij,r2ij,c6ij,dic6ij,djc6ij,k,l)  &
 !$omp shared (nat,at,r2ab,refn,itbl,c6abns,zvec,dzvec) &
 !$omp shared (c6ab,dc6ab)
@@ -1955,7 +1955,7 @@ subroutine dabcappr(nat,ndim,at,xyz,par,  &
 !$omp enddo
 !$omp end parallel
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,j,ij,ia,ja,k,ka,ik,jk,oorjk,oorik,atm,fdmp,  &
 !$omp         r2ij,cij,oorij,r2ik,r2jk,cik,cjk,r2ijk,rijk,cijk, &
 !$omp         dijatm,djkatm,dikatm,dtmp,dijfdmp,djkfdmp,dikfdmp,  &
@@ -2896,7 +2896,7 @@ subroutine edisp_3d(mol,ndim,q,rep,atm_rep,r_thr,atm_thr,par,g_a,g_c,gw,refc6,mb
    enddo
 
 !  print'(" * Entering first OMP section")'
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ii,iii,ia,iz,k)  &
 !$omp shared (mol,refn,refc,refcovcn,itbl,refq,g_a,g_c,q) &
 !$omp shared (gw,zetavec,zerovec,r_thr)
@@ -2915,7 +2915,7 @@ subroutine edisp_3d(mol,ndim,q,rep,atm_rep,r_thr,atm_thr,par,g_a,g_c,gw,refc6,mb
 !$omp end do
 !$omp end parallel
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,j,ia,ja,ij,k,l,c6ii,c6ij,disp,  &
 !$omp         rij,r2,r,r4r2ij,r0,oor6,oor8,oor10, &
 !$omp         t,tx,ty,tz)  &
@@ -3029,7 +3029,7 @@ subroutine abcappr_3d(mol,ndim,par,zvec,refc6,itbl,rep,r_thr,eabc)
 
    eabc = 0.0_wp
 
-   !$OMP parallel default(none) &
+   !$OMP parallel default(shared) &
    !$omp private(i,ia,j,ja,ij,r2ij,c6ij,k,l,r)  &
    !$omp shared (mol,refn,itbl,refc6,zvec) &
    !$omp shared (c6ab)
@@ -3514,7 +3514,7 @@ subroutine dispgrad_3d(mol,ndim,q,cn,dcndr,dcndL,rep,atm_rep,r_thr,atm_thr,par,
    enddo
 
 !  print'(" * Entering first OMP section")'
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ii,iii,ia,iz,k,norm,dnorm,twf,tgw,dexpw,expw,gwk,dgwk)  &
 !$omp shared (mol,refn,refc,refcovcn,itbl,refq,wf,cn,g_a,g_c,q) &
 !$omp shared (gw,dgw,zvec,dzvec,dzdq,r_thr)
@@ -3570,7 +3570,7 @@ subroutine dispgrad_3d(mol,ndim,q,cn,dcndr,dcndL,rep,atm_rep,r_thr,atm_thr,par,
 !$omp end do
 !$omp end parallel
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,j,ia,ja,ij,k,l,c6ii,c6ij,dic6ii,dic6ij,djc6ij,disp,ddisp,dizii,dizij,djzij,  &
 !$omp         rij,r2,r,r4r2ij,r0,oor6,oor8,oor10,door6,door8,door10, &
 !$omp         t,tx,ty,tz,dtmp,drdx)  &
@@ -3767,7 +3767,7 @@ subroutine dabcappr_3d(mol,ndim,par,zvec,dzvec,refc6,itbl,rep,r_thr, &
 
    eabc = 0.0_wp
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ia,j,ja,ij,r2ij,c6ij,dic6ij,djc6ij,k,l,r)  &
 !$omp shared (mol,refn,itbl,refc6,zvec,dzvec) &
 !$omp shared (c6ab,dc6ab)
@@ -3879,7 +3879,7 @@ subroutine dabcappr_3d_dftd3_like_style(nat,at,xyz,par,thr,rep,dlat,c6ab,dc6ab,
    eabc=0.0_wp
    !        write(*,*)'thr:',sqrt(thr)
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(nat,xyz,c6ab,rep,par,at,dlat,dc6ab,thr) &
 !$omp private(iat,jat,ij,ijvec,c6ij,kat,ik,jk,ikvec,jkvec,c6ik,c6jk,c9, &
 !$omp&        cij,cik,cjk,cijk,tx,ty,tz,repmin,repmax,t,rij2,sx,sy,sz, &
diff --git a/xtb/dispplot.f90 b/xtb/dispplot.f90
index 20de1a1..ed4b0b6 100644
--- a/xtb/dispplot.f90
+++ b/xtb/dispplot.f90
@@ -71,7 +71,7 @@ subroutine dispersion_surfaceplot(nat,at,xyz,q,wf,g_a,g_c,lmbd,dfparam,surface)
    call edisp(nat+1,ndim,at_probe,q_probe,xyz_probe,dfparam,g_a,g_c, &
    &          gw,c6abns,lmbd,save_tmp,emany=save_mbd)
 
-   !$omp parallel default(none) &
+   !$omp parallel default(shared) &
    !$omp shared(at_probe,nat,q_probe,dfparam,wf,g_a,g_c,c6abns,surface,ndim, &
    !$omp        gw,lmbd,save_tmp,save_mbd,xyz) &
    !$omp firstprivate(xyz_probe) &
diff --git a/xtb/eeq_model.f90 b/xtb/eeq_model.f90
index 354e195..3605de5 100644
--- a/xtb/eeq_model.f90
+++ b/xtb/eeq_model.f90
@@ -228,7 +228,7 @@ subroutine goedecker_chrgeq(n,at,xyz,chrg,cn,dcndr,q,dqdr,energy,gradient,&
    if (lverbose) write(istdout,'(72("="),/,1x,a)') &
       "Setup of the A matrix and the RHS X vector"
 !  prepare some arrays
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,at,cn,xi,cnfak,alp,gamm) &
 !$omp private(i,tmp) &
 !$omp shared(Xvec,Xfac,alpha)
@@ -242,7 +242,7 @@ subroutine goedecker_chrgeq(n,at,xyz,chrg,cn,dcndr,q,dqdr,energy,gradient,&
 !$omp enddo
 !$omp endparallel
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,at,xyz,gamm,alpha) &
 !$omp private(i,j,r,gamij) &
 !$omp shared(Xvec,Xfac,Amat)
@@ -326,7 +326,7 @@ do_molecular_gradient: if (lgrad .or. lcpq) then
       "molecular gradient calculation"
    allocate( dAmat(3,n,m), dXvec(3,n,m), Afac(3,n), source = 0.0_wp )
    !allocate( dAmat(3,n,m), Afac(3,n), source = 0.0_wp )
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,dcndr,xyz,alpha,Amat,Xfac,Xtmp) &
 !$omp private(i,j,rij,r2,gamij,arg,dtmp) &
 !$omp shared(dXvec,dAmat) &
@@ -537,7 +537,7 @@ subroutine goedecker_hessian(n,at,xyz,chrg,cn,dcndr,q,dqdr,energy,gradient,hessi
    if (lverbose) write(istdout,'(72("="),/,1x,a)') &
       "Setup of the A matrix and the RHS X vector"
 !  prepare some arrays
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,at,cn,xi,cnfak,alp,gamm) &
 !$omp private(i,tmp) &
 !$omp shared(Xvec,Xfac,alpha)
@@ -551,7 +551,7 @@ subroutine goedecker_hessian(n,at,xyz,chrg,cn,dcndr,q,dqdr,energy,gradient,hessi
 !$omp enddo
 !$omp endparallel
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,at,xyz,gamm,alpha) &
 !$omp private(i,j,r,gamij) &
 !$omp shared(Xvec,Xfac,Amat)
@@ -634,7 +634,7 @@ subroutine goedecker_hessian(n,at,xyz,chrg,cn,dcndr,q,dqdr,energy,gradient,hessi
       "molecular gradient calculation"
    allocate( dAmat(3,n,m), dXvec(3,n,m), Afac(3,n), source = 0.0_wp )
    !allocate( dAmat(3,n,m), Afac(3,n), source = 0.0_wp )
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,dcndr,xyz,alpha,Amat,Xfac,Xtmp) &
 !$omp private(i,j,rij,r2,gamij,arg,dtmp) &
 !$omp shared(dXvec,dAmat) &
@@ -942,7 +942,7 @@ subroutine goedecker_multieq2(n,at,xyz,chrgeq,chrg,cn,dcndr,q,dipm,energy,&
    if (lverbose) write(istdout,'(72("="),/,1x,a)') &
       "Setup of the A matrix and the RHS X vector"
 !  prepare some arrays
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,at,cn,chrgeq) &
 !$omp private(i,ii,tmp) &
 !$omp shared(Xvec,Xfac,alpha,beta)
@@ -958,7 +958,7 @@ subroutine goedecker_multieq2(n,at,xyz,chrgeq,chrg,cn,dcndr,q,dipm,energy,&
 !$omp enddo
 !$omp endparallel
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,m,at,xyz,chrgeq,alpha,beta) &
 !$omp private(i,j,ii,jj,rij,r2,r,rxr,tmp,dtmp) &
 !$omp private(arg,arg2,gamij,gamij2,expterm,erfterm) &
@@ -1182,7 +1182,7 @@ do_molecular_gradient: if (lgrad .or. lcpq) then
       "molecular gradient calculation"
    allocate( dAmat(3,n,m), dXvec(3,n,m), Afac(3,m), source = 0.0_wp )
    !allocate( dAmat(3,n,m), Afac(3,n), source = 0.0_wp )
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,dcndr,xyz,alpha,Amat,Xfac,Xtmp,beta) &
 !$omp private(i,j,ii,jj,rij,r,r2,rxr,r3,gamij,gamij2,arg,arg2,tmp,dtmp) &
 !$omp private(erfterm,expterm) &
@@ -1650,7 +1650,7 @@ subroutine goedecker_multieq(n,at,xyz,chrg,cn,dcndr,q,dipm,energy,&
    if (lverbose) write(istdout,'(72("="),/,1x,a)') &
       "Setup of the A matrix and the RHS X vector"
 !  prepare some arrays
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,at,cn,xi,cnfak,alp,gamm) &
 !$omp private(i,ii,tmp) &
 !$omp shared(Xvec,Xfac,alpha,beta)
@@ -1666,7 +1666,7 @@ subroutine goedecker_multieq(n,at,xyz,chrg,cn,dcndr,q,dipm,energy,&
 !$omp enddo
 !$omp endparallel
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,m,at,xyz,gamm,alpha,beta) &
 !$omp private(i,j,ii,jj,rij,r2,r,rxr,tmp,dtmp) &
 !$omp private(arg,arg2,gamij,gamij2,expterm,erfterm) &
@@ -2039,7 +2039,7 @@ subroutine eeq_chrgeq_core(mol,chrgeq,cn,dcndr,dcndL,q,dqdr,dqdL, &
       "Setup of the A matrix and the RHS X vector"
 
 !  prepare some arrays
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(mol,cn,chrgeq) &
 !$omp private(i,tmp) &
 !$omp shared(Xvec,Xfac,alpha2)
@@ -2056,7 +2056,7 @@ subroutine eeq_chrgeq_core(mol,chrgeq,cn,dcndr,dcndL,q,dqdr,dqdL, &
    if (mol%npbc > 0) then
    cf = sqrtpi/mol%volume**(1.0_wp/3.0_wp)
    ! build Ewald matrix
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp private(i,j,wscAt,riw,gamii,gamij) &
 !$omp shared(mol,chrgeq,alpha2,cf) &
 !$omp reduction(+:Amat)
@@ -2089,7 +2089,7 @@ subroutine eeq_chrgeq_core(mol,chrgeq,cn,dcndr,dcndL,q,dqdr,dqdL, &
 !$omp enddo
 !$omp endparallel
    else
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(mol,chrgeq,alpha2) &
 !$omp private(i,j,r,gamij) &
 !$omp shared(Xvec,Xfac,Amat)
@@ -2174,7 +2174,7 @@ do_molecular_gradient: if (lgrad .or. lcpq) then
    allocate( dAmatdr(3,mol%n,m), dXvecdr(3,mol%n,m), Afac(3,mol%n), source = 0.0_wp )
    if (mol%npbc > 0) then
    allocate( dAmatdL(3,3,m), dXvecdL(3,3,m), source = 0.0_wp )
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(mol,dcndr,dcndL,alpha2,Xfac,Xtmp,cf) &
 !$omp private(i,j,wscAt,riw,gamij,dAtmp,stmp) &
 !$omp shared(dXvecdr,dXvecdL) &
@@ -2228,7 +2228,7 @@ do_molecular_gradient: if (lgrad .or. lcpq) then
 !$omp enddo
 !$omp endparallel
    else
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(mol,dcndr,alpha2,Amat,Xfac,Xtmp) &
 !$omp private(i,j,rij,r2,gamij,arg,dtmp) &
 !$omp shared(dXvecdr,dAmatdr) &
@@ -2615,7 +2615,7 @@ subroutine eeq_chrgeq_gbsa(mol,chrgeq,gbsa,cn,dcndr,q,dqdr, &
       "Setup of the A matrix and the RHS X vector"
 
 !  prepare some arrays
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(mol,cn,chrgeq) &
 !$omp private(i,tmp) &
 !$omp shared(Xvec,Xfac,alpha2)
@@ -2629,7 +2629,7 @@ subroutine eeq_chrgeq_gbsa(mol,chrgeq,gbsa,cn,dcndr,q,dqdr, &
 !$omp enddo
 !$omp endparallel
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(mol,chrgeq,alpha2) &
 !$omp private(i,j,r,gamij) &
 !$omp shared(Xvec,Xfac,Amat)
@@ -2714,7 +2714,7 @@ do_molecular_gradient: if (lgrad .or. lcpq) then
       "molecular gradient calculation"
    allocate( dAmatdr(3,mol%n,m), dXvecdr(3,mol%n,m), Afac(3,mol%n), source = 0.0_wp )
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(mol,dcndr,alpha2,Xfac,Xtmp) &
 !$omp private(i,j,rij,r,r2,gamij,arg,erft,dtmp) &
 !$omp shared(dXvecdr,dAmatdr) &
diff --git a/xtb/gbobc.f90 b/xtb/gbobc.f90
index d77bdb8..3384582 100644
--- a/xtb/gbobc.f90
+++ b/xtb/gbobc.f90
@@ -1654,7 +1654,7 @@ subroutine update_nnlist_gbsa_parallel(this,xyz)
 
    this%nnsas=0
    this%nnlists=0
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp&         shared ( this,xyz,lrcut2,srcut2 ) &
 !$omp&         private( i1,i2,x,y,z,dr2,ip,ip2,thrid,nntmp,nnls ) &
 !$omp&         shared ( plisttr, npid )
diff --git a/xtb/generate_wsc.f90 b/xtb/generate_wsc.f90
index 4fd207b..d18e9a4 100644
--- a/xtb/generate_wsc.f90
+++ b/xtb/generate_wsc.f90
@@ -64,7 +64,7 @@ subroutine generate_wsc(mol,wsc)
 ! ------------------------------------------------------------------------
    wsc%at  = 0
    wsc%itbl= 0
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp private(ii,jj,wc,c,dist,trans,t,lattr,rw) &
 !$omp shared(mol,wsc,rep) &
 !$omp shared(mindist,minpos,nmindist,nminpos)
diff --git a/xtb/grad_core.f90 b/xtb/grad_core.f90
index 0625859..f1fe382 100644
--- a/xtb/grad_core.f90
+++ b/xtb/grad_core.f90
@@ -158,7 +158,7 @@ subroutine hcn_grad_gfn1(g,n,at,ndim,nmat2,matlist2,xyz, &
 
 !  CN dependent part of H0
    hcn=0.0_wp
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(nmat2,matlist2,aoat2,lao2,valao2,P,S,n,at,xyz,hdiag2) &
 !$omp shared(kspd,kmagic,kenscal,kcnao) &
 !$omp private(i,m,j,atj,ati,dum1,dum2,km,dum,ishell,jshell,hji) &
@@ -184,7 +184,7 @@ subroutine hcn_grad_gfn1(g,n,at,ndim,nmat2,matlist2,xyz, &
    enddo
 !$omp end do
 !$omp end parallel
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(ndim,aoat2,P,hdiag2,kcnao) &
 !$omp private(i,ati,dum1) &
 !$omp reduction (+:hcn)
@@ -198,7 +198,7 @@ subroutine hcn_grad_gfn1(g,n,at,ndim,nmat2,matlist2,xyz, &
 !$omp end parallel
 
 !  CN-level shift gradient
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,dcn,hcn) &
 !$omp private(i,gtmp) shared ( g )
 !$omp do
@@ -255,7 +255,7 @@ subroutine hcn_grad_gfn2(g,n,at,ndim,nmat2,matlist2,xyz, &
 
 !  CN dependent part of H0
    hcn=0.0_wp
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(nmat2,matlist2,aoat2,lao2,valao2,P,S,n,at,xyz) &
 !$omp shared(kspd,kmagic,kenscal,aoexp,kcnao) &
 !$omp private(i,m,j,atj,ati,dum1,dum2,km,dum,ishell,jshell,hji,fact) &
@@ -282,7 +282,7 @@ subroutine hcn_grad_gfn2(g,n,at,ndim,nmat2,matlist2,xyz, &
    enddo
 !$omp end do
 !$omp end parallel
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(ndim,aoat2,P,kcnao) &
 !$omp private(i,ati,dum1) &
 !$omp reduction (+:hcn)
@@ -297,7 +297,7 @@ subroutine hcn_grad_gfn2(g,n,at,ndim,nmat2,matlist2,xyz, &
 
 
 !  CN-level shift gradient
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(n,dcn,hcn) &
 !$omp private(i,gtmp) shared ( g )
 !$omp do
diff --git a/xtb/grid_module.f90 b/xtb/grid_module.f90
index a4be32e..b6eb0e2 100644
--- a/xtb/grid_module.f90
+++ b/xtb/grid_module.f90
@@ -112,7 +112,7 @@ subroutine gengrid(grid,nat,at,xyz)
    ! Precompute the grid weights with multiple threads. The job has to be
    ! split in two because the nodes have to be positioned in the array in 
    ! the correct order 
-   !$omp parallel default(none) shared(nat,at,xyz,rr,gridrl,gridx) &
+   !$omp parallel default(shared) shared(nat,at,xyz,rr,gridrl,gridx) &
    !$omp private(nr,nang,rmid,ir,r,il,x,j,k,r1,r2,hypr,cutoff,vp0,vpsum,vpi) &
    !$omp firstprivate(rads,wrads,xang,yang,zang,wang)
    !$omp do
