diff --git a/meson/meson.build b/meson/meson.build
index 363c15c..50e3965 100644
--- a/meson/meson.build
+++ b/meson/meson.build
@@ -71,8 +71,16 @@ elif get_option('la_backend') == 'custom'
     dependencies += fc.find_library(lib)
   endforeach
 else
-  dependencies += fc.find_library('blas', required: true)
-  dependencies += fc.find_library('lapack', required: true)
+  blas_dep = dependency('blas', required: false)
+  if not blas_dep.found()
+    blas_dep = fc.find_library('blas', required: true)
+  endif
+  dependencies += blas_dep
+  lapack_dep = dependency('lapack', required: false)
+  if not lapack_dep.found()
+    lapack_dep = fc.find_library('lapack', required: true)
+  endif
+  dependencies += lapack_dep
 endif
 
 if get_option('openmp')
