diff --git a/meson.build b/meson.build
index 96c1d39..4868e4c 100644
--- a/meson.build
+++ b/meson.build
@@ -35,7 +35,7 @@ config = configuration_data({
   'name': meson.project_name(),
   'description': 'Semiempirical Extended Tight-Binding Program Package',
   'version': meson.project_version(),
-  'commit': run_command(find_program('git'),'show','-s','--format=%h').stdout().strip(),
+  'commit': 'conda-forge',
   'date': run_command(find_program('date'),'-I').stdout().strip(),
   'author': run_command(find_program('whoami')).stdout().strip(),
   'origin': run_command(find_program('hostname')).stdout().strip(),
