diff -Naur basemap-1.0.7rel.orig/lib/mpl_toolkits/basemap/__init__.py basemap-1.0.7rel/lib/mpl_toolkits/basemap/__init__.py
--- lib/mpl_toolkits/basemap/__init__.py	2013-08-17 17:35:32.000000000 -0300
+++ lib/mpl_toolkits/basemap/__init__.py	2016-03-21 13:31:15.369414853 -0300
@@ -45,7 +45,7 @@
     if not os.path.isdir(basemap_datadir):
         raise RuntimeError('Path in environment BASEMAPDATA not a directory')
 else:
-    basemap_datadir = os.sep.join([os.path.dirname(__file__), 'data'])
+    basemap_datadir = os.path.join(sys.prefix, 'share', 'basemap')
 
 __version__ = '1.0.7'
 
diff -Naur basemap-1.0.7rel.orig/lib/mpl_toolkits/basemap/pyproj.py basemap-1.0.7rel/lib/mpl_toolkits/basemap/pyproj.py
--- lib/mpl_toolkits/basemap/pyproj.py	2013-08-17 17:35:32.000000000 -0300
+++ lib/mpl_toolkits/basemap/pyproj.py	2016-03-21 13:33:15.541421826 -0300
@@ -51,7 +51,7 @@
 __version__ =  _proj.__version__
 set_datapath =  _proj.set_datapath
 from array import array
-import os, math
+import os, math, sys
 #import numpy as np
 pj_list={
 'aea': "Albers Equal Area",
@@ -235,7 +235,13 @@
 "sphere":       {'a':6370997.0,'b':6370997.0,'description':"Normal Sphere"},
 }
 
-pyproj_datadir = os.sep.join([os.path.dirname(__file__), 'data'])
+if 'BASEMAPDATA' in os.environ:
+    pyproj_datadir = os.environ['BASEMAPDATA']
+    if not os.path.isdir(pyproj_datadir):
+        raise RuntimeError('Path in environment BASEMAPDATA not a directory')
+else:
+    pyproj_datadir = os.path.join(sys.prefix, 'share', 'basemap')
+
 if not os.path.isdir(pyproj_datadir):
     msg="proj data directory not found. Expecting it at: %s"%pyproj_datadir
     raise IOError(msg)
