--- lib/mpl_toolkits/basemap/__init__.py	2016-03-23 10:25:48.150934435 -0300
+++ lib/mpl_toolkits/basemap/__init__.py	2016-03-23 14:22:02.486302483 -0300
@@ -45,7 +45,10 @@
     if not os.path.isdir(basemap_datadir):
         raise RuntimeError('Path in environment BASEMAPDATA not a directory')
 else:
-    basemap_datadir = os.sep.join([os.path.dirname(__file__), 'data'])
+  if os.name == 'nt':
+    basemap_datadir = os.path.join(sys.prefix, 'Library', 'share', 'basemap')
+  else:
+    basemap_datadir = os.path.join(sys.prefix, 'share', 'basemap')

 __version__ = '1.0.8'

@@ -1205,11 +1208,12 @@
         read boundary data, clip to map projection region.
         """
         msg = dedent("""
-        Unable to open boundary dataset file. Only the 'crude', 'low',
-        'intermediate' and 'high' resolution datasets are installed by default.
-        If you are requesting a 'full' resolution dataset, you may need to
-        download and install those files separately
-        (see the basemap README for details).""")
+        Unable to open boundary dataset file. Only the 'crude' and  'low',
+        resolution datasets are installed by default.
+        If you are requesting an, 'intermediate', 'high' or 'full'
+        resolution dataset, you may need to download and install those
+        files separately with
+        `conda install -c conda-forge basemap-data-hires`.""")
         # only gshhs coastlines can be polygons.
         if name != 'gshhs': as_polygons=False
         try:
@@ -2004,8 +2008,12 @@

         returns a matplotlib.patches.LineCollection object.
         """
-        ax = ax or self._check_ax()
         gis_file = os.path.join(basemap_datadir,'UScounties')
+        if not os.path.isfile(gis_file + '.shp'):
+            msg = ('Cannot find {}.shp\nYou can install it with'
+                   '`conda install -c conda-forge basemap-data-hires`').format
+            raise IOError(msg(gis_file))
+        ax = ax or self._check_ax()
         county_info = self.readshapefile(gis_file,'counties',\
                       default_encoding='latin-1',drawbounds=drawbounds)
         counties = [coords for coords in self.counties]
