--- server_abyss.cpp	2021-06-09 15:16:28.549956388 +1000
+++ server_abyss.cpp	2021-06-09 15:17:02.093957207 +1000
@@ -85,7 +85,7 @@
         sockAddr.sin_port   = htons(portNumber);
         sockAddr.sin_addr.s_addr = 0;
 
-        rc = bind(this->fd, (struct sockaddr *)&sockAddr, sizeof(sockAddr));
+        rc = ::bind(this->fd, (struct sockaddr *)&sockAddr, sizeof(sockAddr));
         
         if (rc != 0) {
             closesock(this->fd);
