diff --git a/python/despyfits/DESImage.py b/python/despyfits/DESImage.py
index 17f4f3d..55d70dd 100644
--- a/python/despyfits/DESImage.py
+++ b/python/despyfits/DESImage.py
@@ -841,7 +841,12 @@ def slice2section(s):
            'Darwin': 'dylib'}
 try:
     libdesimage = ctypes.CDLL(
-        'libdesimage.' + lib_ext[platform.system()])
+        os.path.join(
+            os.getenv('DESPYFITS_DIR'),
+            'lib',
+            'libdesimage.' + lib_ext[platform.system()],
+        )
+    )
 except KeyError:
     raise RuntimeError("Unknown platform: " + platform.system())
