diff --git a/python/despyfits/compressionhdu.py b/python/despyfits/compressionhdu.py
index f173892..b957374 100644
--- a/python/despyfits/compressionhdu.py
+++ b/python/despyfits/compressionhdu.py
@@ -2,12 +2,18 @@

 import ctypes
 import platform
+import os

 lib_ext = {'Linux': 'so',
            'Darwin': 'dylib'}
 try:
     libcompressionhdu = ctypes.CDLL(
-        'libcompressionhdu.' + lib_ext[platform.system()])
+        os.path.join(
+            os.getenv('DESPYFITS_DIR'),
+            'lib',
+            'libcompressionhdu.' + lib_ext[platform.system()],
+        )
+    )
 except KeyError:
     raise RuntimeError("Unknown platform: " + platform.system())
