--- setup.py	2016-02-28 19:09:12.000000000 -0300
+++ setup.py	2016-04-13 09:52:30.507413627 -0300
@@ -28,22 +28,6 @@
 # Custom builder to handler compiler flags. Edit if needed.
 class build_ext_subclass(build_ext):
     def build_extensions(self):
-        comp = self.compiler.compiler_type 
-        if comp in ('unix', 'cygwin', 'mingw32'):
-            # Check if build is with OpenMP 
-            if use_omp:
-                extra_compile_args = ['-std=c99', '-O3', '-fopenmp']
-                extra_link_args=['-lgomp']
-            else:
-                extra_compile_args = ['-std=c99', '-O3']
-                extra_link_args = []
-        else:
-            # Add support for more compilers here
-            raise ValueError(('Compiler flags undefined for %s.', 
-                             'Please modify setup.py and add compiler flags')
-                             % comp)
-        self.extensions[0].extra_compile_args = extra_compile_args
-        self.extensions[0].extra_link_args = extra_link_args
         build_ext.build_extensions(self)
 
     def finalize_options(self):
