diff -ruN inpoly-python-0.2.0/setup.py inpoly-python-0.2.0-patch/setup.py
--- inpoly-python-0.2.0/setup.py	2020-12-20 01:14:36.000000000 +0100
+++ inpoly-python-0.2.0-patch/setup.py	2021-03-28 16:39:12.656196088 +0200
@@ -81,7 +81,7 @@
     python_requires=REQUIRES_PYTHON,
     keywords=KEYWORDS,
     url=URL,
-    packages=find_packages(),
+    packages=find_packages(exclude=['msh',]),
     ext_modules=EXT_MODULES,
     install_requires=REQUIRED,
     classifiers=CLASSIFY
