diff -ruN inpoly-python-0.2.0/example.py inpoly-python-0.2.0-patch/example.py
--- inpoly-python-0.2.0/example.py	2020-12-20 01:14:36.000000000 +0100
+++ inpoly-python-0.2.0-patch/example.py	2021-03-28 19:00:46.344620261 +0200
@@ -40,7 +40,7 @@
     plt.plot(points[ON==1, 0], points[ON==1, 1], "ms")
 
     ax.set_aspect('equal', adjustable='box')
-    plt.show()
+    plt.savefig('ex_1.png')
 
     return
 
@@ -81,7 +81,7 @@
     plt.plot(points[ON==1, 0], points[ON==1, 1], "ms")
 
     ax.set_aspect('equal', adjustable='box')
-    plt.show()
+    plt.savefig('ex_2.png')
 
     return
 
@@ -137,7 +137,7 @@
     plt.plot(points[ON==1, 0], points[ON==1, 1], "ms")
 
     ax.set_aspect('equal', adjustable='box')
-    plt.show()
+    plt.savefig('ex_3.png')
 
     return
 
