From 6d5cbc7a59a2f4fbbd3c40f61c3d0d50d2505c9e Mon Sep 17 00:00:00 2001
From: Axel Huebl <axel.huebl@plasma.ninja>
Date: Fri, 5 Mar 2021 18:33:35 -0800
Subject: [PATCH 3/4] Missing include for std::max

---
 include/blas/batch_common.hh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/blas/batch_common.hh b/include/blas/batch_common.hh
index f15ddff..41bcc62 100644
--- a/include/blas/batch_common.hh
+++ b/include/blas/batch_common.hh
@@ -7,6 +7,7 @@
 #define BLAS_BATCH_COMMON_HH
 
 #include "blas/util.hh"
+#include <algorithm>  // std::min/max
 #include <vector>
 
 namespace blas {
-- 
2.25.1

