--- glib/gutils.c.orig	2016-03-24 00:16:46.764844235 +0000
+++ glib/gutils.c	2016-03-24 00:17:21.136947935 +0000
@@ -1940,7 +1940,7 @@
       gchar *data_dirs = (gchar *) g_getenv ("XDG_DATA_DIRS");
 
       if (!data_dirs || !data_dirs[0])
-          data_dirs = "/usr/local/share/:/usr/share/";
+          data_dirs = CONDA_SYSTEM_DATA_DIRS;
 
       data_dir_vector = g_strsplit (data_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
