diff -Naur DSDP5.8/src/sys/dsdploginfo.c DSDP5.8_patched/src/sys/dsdploginfo.c
--- DSDP5.8/src/sys/dsdploginfo.c	2018-04-20 06:40:53.000000000 +0200
+++ DSDP5.8_patched/src/sys/dsdploginfo.c	2018-04-20 06:41:19.000000000 +0200
@@ -6,7 +6,9 @@
 #include <stdarg.h>
 #include <sys/types.h>
 #include <stdlib.h>
+#ifndef __MACH__
 #include <malloc.h>
+#endif
 #include "dsdpsys.h"
 #include "dsdpbasictypes.h"
 
