diff -urN work.old/configure work/configure
--- work.old/configure	2017-08-22 14:16:00.621910386 +0100
+++ work/configure	2017-08-22 14:17:20.482361609 +0100
@@ -586,7 +586,7 @@
 
 # Identity of this package.
 PACKAGE_NAME='libedit'
-PACKAGE_TARNAME='libedit-20170329'
+PACKAGE_TARNAME='libedit-20170822'
 PACKAGE_VERSION='3.1'
 PACKAGE_STRING='libedit 3.1'
 PACKAGE_BUGREPORT=''
@@ -1374,7 +1374,7 @@
   --localedir=DIR         locale-dependent data [DATAROOTDIR/locale]
   --mandir=DIR            man documentation [DATAROOTDIR/man]
   --docdir=DIR            documentation root
-                          [DATAROOTDIR/doc/libedit-20170329]
+                          [DATAROOTDIR/doc/libedit-20170822]
   --htmldir=DIR           html documentation [DOCDIR]
   --dvidir=DIR            dvi documentation [DOCDIR]
   --pdfdir=DIR            pdf documentation [DOCDIR]
@@ -4223,7 +4223,7 @@
 
 
 # Define the identity of the package.
- PACKAGE='libedit-20170329'
+ PACKAGE='libedit-20170822'
  VERSION='3.1'
 
 
@@ -12580,7 +12580,54 @@
   LIBS="-ltermcap $LIBS"
 
 else
-  as_fn_error $? "libncurses, libcurses, or libtermcap is required!" "$LINENO" 5
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for tgetent in -ltinfo" >&5
+$as_echo_n "checking for tgetent in -ltinfo... " >&6; }
+if ${ac_cv_lib_tinfo_tgetent+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-ltinfo  $LIBS"
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char tgetent ();
+int
+main ()
+{
+return tgetent ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_lib_tinfo_tgetent=yes
+else
+  ac_cv_lib_tinfo_tgetent=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_tinfo_tgetent" >&5
+$as_echo "$ac_cv_lib_tinfo_tgetent" >&6; }
+if test "x$ac_cv_lib_tinfo_tgetent" = xyes; then :
+  cat >>confdefs.h <<_ACEOF
+#define HAVE_LIBTINFO 1
+_ACEOF
+
+  LIBS="-ltinfo $LIBS"
+
+else
+  as_fn_error $? "libncurses, libcurses, libtermcap or libtinfo is required!" "$LINENO" 5
+
+fi
+
 
 fi
 
diff -urN work.old/configure.ac work/configure.ac
--- work.old/configure.ac	2017-08-22 14:16:00.622910417 +0100
+++ work/configure.ac	2017-08-22 14:17:03.964854582 +0100
@@ -49,7 +49,9 @@
 AC_CHECK_LIB(ncurses, tgetent,,
   [AC_CHECK_LIB(curses, tgetent,,
     [AC_CHECK_LIB(termcap, tgetent,,
-      [AC_MSG_ERROR([libncurses, libcurses, or libtermcap is required!])]
+      [AC_CHECK_LIB(tinfo, tgetent,,
+        [AC_MSG_ERROR([libncurses, libcurses, libtermcap or libtinfo is required!])]
+      )]
     )]
   )]
 )
