From 5602fb9f76bb4f981f159d547e2f944b7d2414fa Mon Sep 17 00:00:00 2001
From: Tim Paine <t.paine154@gmail.com>
Date: Sat, 18 Mar 2023 14:27:40 -0400
Subject: [PATCH] .

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 742c344..14ee625 100644
--- a/setup.py
+++ b/setup.py
@@ -47,7 +47,7 @@ setup(
     ],
     extras_require={
         # this version requirement needs to be synchronized with the one in amaranth.back.verilog!
-        "builtin-yosys": ["amaranth-yosys>=0.10.*"],
+        "builtin-yosys": ["amaranth-yosys>=0.10"],
         "remote-build": ["paramiko~=2.7"],
     },
     packages=find_packages(exclude=("tests", "tests.*")),
-- 
2.40.0

