diff --git a/google/cloud/storage/well_known_headers.h b/google/cloud/storage/well_known_headers.h
index d69721aac9..03bfb8bee6 100644
--- a/google/cloud/storage/well_known_headers.h
+++ b/google/cloud/storage/well_known_headers.h
@@ -18,6 +18,7 @@
 #include "google/cloud/internal/random.h"
 #include "google/cloud/optional.h"
 #include "google/cloud/storage/version.h"
+#include <algorithm>
 #include <cstdint>
 #include <iostream>
 #include <limits>
diff --git a/google/cloud/bigtable/benchmarks/setup.cc b/google/cloud/bigtable/benchmarks/setup.cc
index 9e807ab8ad..e2db91c113 100644
--- a/google/cloud/bigtable/benchmarks/setup.cc
+++ b/google/cloud/bigtable/benchmarks/setup.cc
@@ -17,8 +17,8 @@
 #include "google/cloud/internal/build_info.h"
 #include "google/cloud/internal/random.h"
 #include "google/cloud/internal/throw_delegate.h"
+#include <algorithm>
 #include <cctype>
-#include <iomanip>
 #include <iostream>
 #include <sstream>
