diff --git a/gcc/configure.ac b/gcc/configure.ac
--- a/gcc/configure.ac
+++ b/gcc/configure.ac
@@ -4967,11 +4967,11 @@ if test x"$insn" != x; then
     fi
  fi

- gcc_GAS_CHECK_FEATURE([--gdwarf2 option],
+ gcc_GAS_CHECK_FEATURE([-gdwarf-2 option],
   gcc_cv_as_gdwarf2_flag,
-  [elf,2,11,0], [--gdwarf2], [$insn],,
+  [elf,2,11,0], [-gdwarf-2], [$insn],,
   [AC_DEFINE(HAVE_AS_GDWARF2_DEBUG_FLAG, 1,
-[Define if your assembler supports the --gdwarf2 option.])])
+[Define if your assembler supports the -gdwarf-2 option.])])

  gcc_GAS_CHECK_FEATURE([--gstabs option],
   gcc_cv_as_gstabs_flag,
