From 4a306ab7e0f2ba504d8a17a78a6cd8f2b0ed599d Mon Sep 17 00:00:00 2001
From: elbaro <elbaro@users.noreply.github.com>
Date: Mon, 13 Apr 2020 18:29:34 +0900
Subject: [PATCH] patch for conda

---
 src/alloc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/alloc.c b/src/alloc.c
index b1c4cd3..8a24fc3 100644
--- a/src/alloc.c
+++ b/src/alloc.c
@@ -4,6 +4,7 @@ This is free software; you can redistribute it and/or modify it under the
 terms of the MIT license. A copy of the license can be found in the file
 "LICENSE" at the root of this distribution.
 -----------------------------------------------------------------------------*/
+#undef _GNU_SOURCE  // https://github.com/microsoft/mimalloc/issues/219
 #include "mimalloc.h"
 #include "mimalloc-internal.h"
 #include "mimalloc-atomic.h"
-- 
2.26.0

