From 62bd4079507845103dfa5f760addd9b19e5760be Mon Sep 17 00:00:00 2001
From: elbaro <elbaro@users.noreply.github.com>
Date: Thu, 16 Apr 2020 17:16:48 +0900
Subject: [PATCH] patch aligned_malloc

---
 src/alloc-override.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/alloc-override.c b/src/alloc-override.c
index c0e7bc2..cd41147 100644
--- a/src/alloc-override.c
+++ b/src/alloc-override.c
@@ -173,7 +173,7 @@ void* valloc(size_t size)                                   { return mi_valloc(s
 void* pvalloc(size_t size)                                  { return mi_pvalloc(size); }
 void* reallocarray(void* p, size_t count, size_t size)      { return mi_reallocarray(p, count, size); }
 void* memalign(size_t alignment, size_t size)               { return mi_memalign(alignment, size); }
-void* aligned_alloc(size_t alignment, size_t size)          { return mi_aligned_alloc(alignment, size); }
+void* _aligned_malloc(size_t alignment, size_t size) { return mi_aligned_alloc(alignment, size); }
 int posix_memalign(void** p, size_t alignment, size_t size) { return mi_posix_memalign(p, alignment, size); }
 
 #if defined(__GLIBC__) && defined(__linux__)
-- 
2.26.0

