From 6cc628aab31d70549120344d538756e7578f26d5 Mon Sep 17 00:00:00 2001
From: elbaro <elbaro@users.noreply.github.com>
Date: Tue, 14 Apr 2020 20:30:42 +0900
Subject: [PATCH] fix install path

---
 CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e16830a..57cf1af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -146,7 +146,7 @@ endif()
 # Install and output names
 # -----------------------------------------------------------------------------
 
-set(mi_install_dir "${CMAKE_INSTALL_PREFIX}/lib/mimalloc-${mi_version}")
+set(mi_install_dir "${CMAKE_INSTALL_PREFIX}")
 if(MI_SECURE MATCHES "ON")
   set(mi_basename "mimalloc-secure")
 else()
@@ -203,8 +203,8 @@ else()
 endif()
 
 # install static and shared library, and the include files
-install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_dir} LIBRARY)
-install(TARGETS mimalloc-static EXPORT mimalloc DESTINATION ${mi_install_dir})
+install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_dir}/lib LIBRARY)
+install(TARGETS mimalloc-static EXPORT mimalloc DESTINATION ${mi_install_dir}/lib)
 install(FILES include/mimalloc.h DESTINATION ${mi_install_dir}/include)
 install(FILES include/mimalloc-override.h DESTINATION ${mi_install_dir}/include)
 install(FILES include/mimalloc-new-delete.h DESTINATION ${mi_install_dir}/include)
@@ -215,7 +215,7 @@ install(EXPORT mimalloc DESTINATION ${mi_install_dir}/cmake)
 if(NOT WIN32)
   # install a symlink in the /usr/local/lib to the versioned library
   set(mi_symlink "${CMAKE_SHARED_MODULE_PREFIX}${mi_basename}${CMAKE_SHARED_LIBRARY_SUFFIX}")
-  set(mi_soname "mimalloc-${mi_version}/${mi_symlink}.${mi_version}")
+  set(mi_soname "${mi_symlink}.${mi_version}")
   install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${mi_soname} ${mi_symlink} WORKING_DIRECTORY ${CMAKE_INSTALL_PREFIX}/${mi_install_dir}/..)")
   install(CODE "MESSAGE(\"-- Symbolic link: ${CMAKE_INSTALL_PREFIX}/lib/${mi_symlink} -> ${mi_soname}\")")
 endif()
@@ -235,7 +235,7 @@ target_include_directories(mimalloc-obj PUBLIC
 # the FILES expression can also be: $<TARGET_OBJECTS:mimalloc-obj>
 # but that fails cmake versions less than 3.10 so we leave it as is for now
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/mimalloc-obj.dir/src/static.c${CMAKE_C_OUTPUT_EXTENSION}
-        DESTINATION ${mi_install_dir}
+        DESTINATION ${mi_install_dir}/lib
         RENAME ${mi_basename}${CMAKE_C_OUTPUT_EXTENSION} )
 
 # -----------------------------------------------------------------------------
-- 
2.26.0

