diff --git a/dttview/tggtest.cc b/dttview/tggtest.cc
index d4a9237e..387327ee 100644
--- a/dttview/tggtest.cc
+++ b/dttview/tggtest.cc
@@ -10,9 +10,9 @@
 #include <TGraph.h>
 #include <TVirtualX.h>
 #include "PlotSet.hh"
-#include "ligogui/TLGOptions.hh"
-#include "ligogui/TLGPlot.hh"
-#include "ligogui/TLGPad.hh"
+#include "dttview/TLGOptions.hh"
+#include "dttview/TLGPlot.hh"
+#include "dttview/TLGPad.hh"
 
 
    using namespace ligogui;
