diff --git a/setup.py b/setup.py
index c66049e..fab1fa8 100644
--- a/setup.py
+++ b/setup.py
@@ -99,11 +99,11 @@ def setup_package():
         return clean(root)
 
     with chdir(root):
-        with open(os.path.join(root, src_path, 'about.py')) as f:
+        with open(os.path.join(root, src_path, 'about.py'), encoding="utf8") as f:
             about = {}
             exec(f.read(), about)
 
-        with open(os.path.join(root, 'README.rst')) as f:
+        with open(os.path.join(root, 'README.rst'), encoding="utf8") as f:
             readme = f.read()
 
         include_dirs = [
