diff --git a/setup.py b/setup.py
index c66049e..8395695 100644
--- a/setup.py
+++ b/setup.py
@@ -2,6 +2,7 @@
 from __future__ import print_function
 
 import os
+import io
 import shutil
 import subprocess
 import sys
@@ -99,11 +100,11 @@ def setup_package():
         return clean(root)
 
     with chdir(root):
-        with open(os.path.join(root, src_path, 'about.py')) as f:
+        with io.open(os.path.join(root, src_path, 'about.py'), encoding='utf8') as f:
             about = {}
             exec(f.read(), about)
 
-        with open(os.path.join(root, 'README.rst')) as f:
+        with io.open(os.path.join(root, 'README.rst'), encoding='utf8') as f:
             readme = f.read()
 
         include_dirs = [
