From 2c1df4b74fbdf5e7ad2f58485d9459ac8f65eccf Mon Sep 17 00:00:00 2001
From: Lindsey Gray <lindsey.gray@gmail.com>
Date: Tue, 7 Mar 2023 09:49:14 -0600
Subject: [PATCH] apply patch

---
 setup.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index c2e1dc2..461e890 100644
--- a/setup.py
+++ b/setup.py
@@ -53,6 +53,7 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
         if not OUTPUT.exists():
             zip_filename = DIR / pathlib.Path(CGAL_ZIP).parts[-1]
 
+            '''
             with urllib.request.urlopen(CGAL_ZIP) as http_obj:
                 with open(zip_filename, "wb") as file_obj:
                     shutil.copyfileobj(http_obj, file_obj)
@@ -60,7 +61,9 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
             with zipfile.ZipFile(zip_filename) as zip_obj:
                 cgal_dir = DIR / zip_obj.namelist()[0]
                 zip_obj.extractall(DIR)
-
+            '''
+            cgal_dir="${CONDA_PREFIX}/include"
+            
             # Patch for FastJet core version 3.4.0
             # To be removed when https://gitlab.com/fastjet/fastjet/-/merge_requests/1 is merged upstream
             subprocess.run(
@@ -78,7 +81,7 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
             env = os.environ.copy()
             env["PYTHON"] = sys.executable
             env["PYTHON_INCLUDE"] = f'-I{sysconfig.get_path("include")}'
-            env["CXXFLAGS"] = "-O3 -Bstatic -lgmp -Bdynamic -std=c++17"
+            env["CXXFLAGS"] = env.get("CXXFLAGS", "") + " -O3 -Bstatic -lgmp -Bdynamic"
             env["ORIGIN"] = "$ORIGIN"  # if evaluated, it will still be '$ORIGIN'
 
             args = [
@@ -87,10 +90,9 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
                 "--enable-allcxxplugins",
                 "--enable-cgal-header-only",
                 "--enable-cgal",
-                f"--with-cgaldir={cgal_dir}",
                 "--enable-swig",
                 "--enable-pyext",
-                "LDFLAGS=-Wl,-rpath,$$ORIGIN/_fastjet_core/lib:$$ORIGIN",
+                f"LDFLAGS={os.environ.get('LDFLAGS', '')} -Wl,-rpath=$$ORIGIN/_fastjet_core/lib:$$ORIGIN",
             ]
 
             try:
@@ -113,7 +112,7 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
                 [
                     "./configure",
                     f"--fastjet-config={FASTJET}/fastjet-config",
-                    "CXXFLAGS=-O3 -Bstatic -Bdynamic -std=c++17",
+                    "CXX=$CXX CXXFLAGS=-O3 -Bstatic -Bdynamic -std=c++17",
                 ],
                 cwd=FASTJET_CONTRIB,
                 env=env,
-- 
2.37.1 (Apple Git-137.1)

