From 5ef7f73d79a41229f3934ab721812138ca01744a Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Fri, 16 Feb 2024 15:32:46 -0600
Subject: [PATCH] Patch setup.py

---
 setup.py | 35 ++++++++++++++---------------------
 1 file changed, 14 insertions(+), 21 deletions(-)

diff --git a/setup.py b/setup.py
index 72cb832..301fb37 100644
--- a/setup.py
+++ b/setup.py
@@ -51,16 +51,6 @@ def get_version() -> str:
 class FastJetBuild(setuptools.command.build_ext.build_ext):
     def build_extensions(self):
         if not OUTPUT.exists():
-            zip_filename = DIR / pathlib.Path(CGAL_ZIP).parts[-1]
-
-            with urllib.request.urlopen(CGAL_ZIP) as http_obj:
-                with open(zip_filename, "wb") as file_obj:
-                    shutil.copyfileobj(http_obj, file_obj)
-
-            with zipfile.ZipFile(zip_filename) as zip_obj:
-                cgal_dir = DIR / zip_obj.namelist()[0]
-                zip_obj.extractall(DIR)
-
             # Patch for FastJet core version 3.4.0
             # To be removed when https://gitlab.com/fastjet/fastjet/-/merge_requests/1 is merged upstream
             subprocess.run(
@@ -75,10 +65,16 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
                 cwd=FASTJET,
             )
 
+            # RPATH is set for shared libraries in the following locations:
+            # * fastjet/
+            # * fastjet/_fastjet_core/lib/
+            # * fastjet/_fastjet_core/lib/python*/site-packages/
+            _rpath = "'$$ORIGIN/_fastjet_core/lib:$$ORIGIN:$$ORIGIN/../..'"
             env = os.environ.copy()
             env["PYTHON"] = sys.executable
             env["PYTHON_INCLUDE"] = f'-I{sysconfig.get_path("include")}'
-            env["CXXFLAGS"] = "-O3 -Bstatic -lgmp -Bdynamic -std=c++17"
+            env["CXXFLAGS"] = env.get("CXXFLAGS", "") + " -O3 -Bstatic -Bdynamic -std=c++17"
+            env["LDFLAGS"] = env.get("LDFLAGS", "") + f" -Wl,-rpath,{_rpath}"
             env["ORIGIN"] = "$ORIGIN"  # if evaluated, it will still be '$ORIGIN'
 
             args = [
@@ -88,10 +84,9 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
                 "--enable-allcxxplugins",
                 "--enable-cgal-header-only",
                 "--enable-cgal",
-                f"--with-cgaldir={cgal_dir}",
                 "--enable-swig",
                 "--enable-pyext",
-                "LDFLAGS=-Wl,-rpath,$$ORIGIN/_fastjet_core/lib:$$ORIGIN",
+                f'LDFLAGS={env["LDFLAGS"]}',
             ]
 
             try:
@@ -106,22 +101,20 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
                 raise
 
             env = os.environ.copy()
+            env["CXX"] = env.get("CXX", "g++")
+            env["CXXFLAGS"] = env.get("CXXFLAGS", "") + " -O3 -Bstatic -Bdynamic -std=c++17"
+            env["LDFLAGS"] = env.get("LDFLAGS", "") + f" -Wl,-rpath,{_rpath}"
             env["ORIGIN"] = "$ORIGIN"  # if evaluated, it will still be '$ORIGIN'
             subprocess.run(["make", "-j"], cwd=FASTJET, env=env, check=True)
             subprocess.run(["make", "install"], cwd=FASTJET, env=env, check=True)
 
-            subprocess.run(
-                ["patch", "./Makefile.in", DIR / "patch_makefilein.txt"],
-                cwd=FASTJET_CONTRIB,
-                env=env,
-                check=True,
-            )
-
             subprocess.run(
                 [
                     "./configure",
                     f"--fastjet-config={FASTJET}/fastjet-config",
-                    "CXXFLAGS=-O3 -Bstatic -Bdynamic -std=c++17",
+                    f'CXX={env["CXX"]}',
+                    f'CXXFLAGS={env["CXXFLAGS"]}',
+                    f'LDFLAGS={env["LDFLAGS"]}',
                 ],
                 cwd=FASTJET_CONTRIB,
                 env=env,
-- 
2.34.1

