From 89b2d558dc9cf31e9a8007444b7683532a2591dd Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Tue, 20 Feb 2024 17:12:20 -0600
Subject: [PATCH] Patch fastjet-contrib Makefile.in

---
 fastjet-contrib/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fastjet-contrib/Makefile.in b/fastjet-contrib/Makefile.in
index a4645fd..220991d 100644
--- a/fastjet-contrib/Makefile.in
+++ b/fastjet-contrib/Makefile.in
@@ -54,7 +54,7 @@ fragile-shared: libfastjetcontribfragile.@DYNLIBEXT@

 fragile_SHARED_SRC_LIST=@FRAGILE_SHARED_SRC_LIST@
 libfastjetcontribfragile.@DYNLIBEXT@: $(fragile_SHARED_SRC_LIST)
-	$(CXX) @DYNLIBOPT@ -fPIC -DPIC $(CXXFLAGS) `$(FASTJETCONFIG) --cxxflags --libs` $(fragile_SHARED_SRC_LIST) -o libfastjetcontribfragile.@DYNLIBEXT@
+	$(CXX) @DYNLIBOPT@ -fPIC -DPIC $(CXXFLAGS) `$(FASTJETCONFIG) --cxxflags --libs` $(fragile_SHARED_SRC_LIST) $(LDFLAGS) -o libfastjetcontribfragile.@DYNLIBEXT@

 fragile-shared-install: fragile-shared
 	utils/install-sh -c -m 755 libfastjetcontribfragile.@DYNLIBEXT@ $(PREFIX)/lib
--
2.34.1

