From bc37495b45ae99de98c3dfa6b647564149ea44f6 Mon Sep 17 00:00:00 2001
From: Samuel Debionne <samuel.debionne@esrf.fr>
Date: Mon, 16 Jan 2023 22:20:21 +0100
Subject: [PATCH 2/2] Fix static_assert compiler support

---
 src/compilersupport_p.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git src/compilersupport_p.h src/compilersupport_p.h
index 0879801..6cba196 100644
--- src/compilersupport_p.h
+++ src/compilersupport_p.h
@@ -44,7 +44,7 @@
 #  include <stdbool.h>
 #endif
 
-#if __STDC_VERSION__ >= 201112L || (defined(__cplusplus) && __cplusplus >= 201103L) || (defined(__cpp_static_assert) && __cpp_static_assert >= 200410)
+#if __STDC_VERSION__ >= 202311L || (defined(__cplusplus) && __cplusplus >= 201103L) || (defined(__cpp_static_assert) && __cpp_static_assert >= 200410)
 #  define cbor_static_assert(x)         static_assert(x, #x)
 #elif !defined(__cplusplus) && defined(__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__ >= 406) && (__STDC_VERSION__ > 199901L)
 #  define cbor_static_assert(x)         _Static_assert(x, #x)
@@ -202,4 +202,3 @@ static inline bool add_check_overflow(size_t v1, size_t v2, size_t *r)
 }
 
 #endif /* COMPILERSUPPORT_H */
-
-- 
2.34.1
