diff --git a/build.gradle b/build.gradle
index b3a7535..223ace0 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,5 +1,6 @@
 plugins {
     id 'application'
+    id 'com.github.jk1.dependency-license-report' version '3.1.4'
     id 'eclipse'
     id 'maven-publish'
     id 'checkstyle'
@@ -12,9 +13,12 @@ application {
   mainClass = 'com.glencoesoftware.bioformats2raw.Converter'
 }
 java {
-  toolchain {
-    languageVersion = JavaLanguageVersion.of(11)
-  }
+  sourceCompatibility = JavaVersion.VERSION_11
+  targetCompatibility = JavaVersion.VERSION_11
+}
+
+tasks.withType(JavaCompile).configureEach {
+    options.release = 11
 }
 
 repositories {
@@ -71,10 +75,5 @@ jar {
     manifest {
         attributes(
-            "Created-By": "Gradle ${gradle.gradleVersion}",
-            "Build-Jdk": "${System.properties['java.version']} (${System.properties['java.vendor']} ${System.properties['java.vm.version']})",
-            "Built-By": System.properties['user.name'],
-            "Built-On": new java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(new Date()),
-            "Implementation-Build": "git rev-parse --verify HEAD".execute().getText().trim(),
             "Implementation-Title": "bioformats2raw converter",
             "Implementation-Version": archiveVersion,
             "Implementation-Vendor": "Glencoe Software Inc.",
diff --git a/settings.gradle b/settings.gradle
new file mode 100644
index 0000000..cfc6e41
--- /dev/null
+++ b/settings.gradle
@@ -0,0 +1 @@
+rootProject.name = 'bioformats2raw'
