From aba6c015b761ab171a7eb520bae2ee58d03485d6 Mon Sep 17 00:00:00 2001
From: Vasily Litvinov <vasilij.n.litvinov@intel.com>
Date: Thu, 5 Nov 2020 11:59:26 +0300
Subject: [PATCH 1/7] Remove cython from setup requirements - installed by
 conda recipe

Signed-off-by: Vasily Litvinov <vasilij.n.litvinov@intel.com>
---
 python/setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index d64d8402e..8c4546fe0 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -457,7 +457,6 @@ setuptools.setup(
     # The BinaryDistribution argument triggers build_ext.
     distclass=BinaryDistribution,
     install_requires=install_requires,
-    setup_requires=["cython >= 0.29.14", "wheel"],
     extras_require=extras,
     entry_points={
         "console_scripts": [
-- 
2.11.0

