From ef5141611db8a84aaa2788aa3f946d6957d9dda6 Mon Sep 17 00:00:00 2001
From: Vasily Litvinov <vasilij.n.litvinov@intel.com>
Date: Thu, 5 Nov 2020 12:04:25 +0300
Subject: [PATCH] Do not force pickle5 in sys.path

Signed-off-by: Vasily Litvinov <vasilij.n.litvinov@intel.com>
---
 python/ray/__init__.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/python/ray/__init__.py b/python/ray/__init__.py
index 0dea67035..db08a1778 100644
--- a/python/ray/__init__.py
+++ b/python/ray/__init__.py
@@ -24,12 +24,6 @@ if "OMP_NUM_THREADS" not in os.environ:
                  "override this by explicitly setting OMP_NUM_THREADS.")
     os.environ["OMP_NUM_THREADS"] = "1"
 
-# Add the directory containing pickle5 to the Python path so that we find the
-# pickle5 version packaged with ray and not a pre-existing pickle5.
-pickle5_path = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)), "pickle5_files")
-sys.path.insert(0, pickle5_path)
-
 # Importing psutil & setproctitle. Must be before ray._raylet is initialized.
 thirdparty_files = os.path.join(
     os.path.abspath(os.path.dirname(__file__)), "thirdparty_files")
-- 
2.11.0

