From ac6d6bdfa456303703fb35c0b170ab64bef90ccc Mon Sep 17 00:00:00 2001
From: Vasily Litvinov <vasilij.n.litvinov@intel.com>
Date: Thu, 5 Nov 2020 17:09:46 +0300
Subject: [PATCH 03/12] Fix plasma linking for ancient glibc

Signed-off-by: Vasily Litvinov <vasilij.n.litvinov@intel.com>
---
 BUILD.bazel | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/BUILD.bazel b/BUILD.bazel
index 883a31c3b..7fc4fc7e6 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -219,6 +219,7 @@ PLASMA_LINKOPTS = [] + select({
         "-DefaultLib:" + "ws2_32.lib",
     ],
     "//conditions:default": [
+        "-lrt",
     ],
 })
 
@@ -312,6 +313,7 @@ cc_binary(
         "src/ray/plasma/store_exec.cc",
     ],
     copts = PLASMA_COPTS,
+    linkopts = PLASMA_LINKOPTS,
     visibility = ["//visibility:public"],
     deps = [
         ":plasma_store_server_lib",
-- 
2.11.0

