From 2597dfdfec9bbc70f317df9eb489df803c57e63a Mon Sep 17 00:00:00 2001
From: Vasily Litvinov <vasilij.n.litvinov@intel.com>
Date: Tue, 10 Nov 2020 23:26:35 +0300
Subject: [PATCH] Disable making non-core entry scripts

Signed-off-by: Vasily Litvinov <vasilij.n.litvinov@intel.com>
---
 python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index e22f17024..2024ee7fd 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -470,7 +470,7 @@ setuptools.setup(
     entry_points={
         "console_scripts": [
             "ray=ray.scripts.scripts:main",
-            "rllib=ray.rllib.scripts:cli [rllib]", "tune=ray.tune.scripts:cli"
+#            "rllib=ray.rllib.scripts:cli [rllib]", "tune=ray.tune.scripts:cli"
         ]
     },
     include_package_data=True,
-- 
2.11.0

