From 49efc2a9815d1e9b5f3f85ee4991ed9aebf737e4 Mon Sep 17 00:00:00 2001
From: Vasily Litvinov <vasilij.n.litvinov@intel.com>
Date: Tue, 10 Nov 2020 23:26:35 +0300
Subject: [PATCH 7/7] Disable making non-core entry scripts

Signed-off-by: Vasily Litvinov <vasilij.n.litvinov@intel.com>
---
 python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index 8c4546fe0..a2252bec9 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -461,7 +461,7 @@ setuptools.setup(
     entry_points={
         "console_scripts": [
             "ray=ray.scripts.scripts:main",
-            "rllib=ray.rllib.scripts:cli [rllib]", "tune=ray.tune.scripts:cli"
+#            "rllib=ray.rllib.scripts:cli [rllib]", "tune=ray.tune.scripts:cli"
         ]
     },
     include_package_data=True,
-- 
2.11.0

