From 8e5a3c9b85993f5ec00c8ca350ec932ade63d34a Mon Sep 17 00:00:00 2001
From: Vasilij Litvinov <vasilij.n.litvinov@intel.com>
Date: Fri, 20 Nov 2020 14:17:05 +0300
Subject: [PATCH 08/12] Do not crash if BAZEL_SH not set on Windows

Signed-off-by: Vasilij Litvinov <vasilij.n.litvinov@intel.com>
---
 python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index 13c1f4b34..024657039 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -246,7 +246,7 @@ def build(build_python, build_java):
         SHELL = bazel_env.get("SHELL")
         if SHELL:
             bazel_env.setdefault("BAZEL_SH", os.path.normpath(SHELL))
-        BAZEL_SH = bazel_env["BAZEL_SH"]
+        BAZEL_SH = bazel_env.get("BAZEL_SH", "")
         SYSTEMROOT = os.getenv("SystemRoot")
         wsl_bash = os.path.join(SYSTEMROOT, "System32", "bash.exe")
         if (not BAZEL_SH) and SYSTEMROOT and os.path.isfile(wsl_bash):
-- 
2.11.0

