From 479036d8d6e4ab66599f970e2743c4f6e04768c5 Mon Sep 17 00:00:00 2001
From: Vasilij Litvinov <vasilij.n.litvinov@intel.com>
Date: Tue, 24 Nov 2020 00:20:34 +0300
Subject: [PATCH 11/12] Include <process.h> for getpid() explicitly on Windows

Signed-off-by: Vasilij Litvinov <vasilij.n.litvinov@intel.com>
---
 src/ray/util/util.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ray/util/util.h b/src/ray/util/util.h
index 5b678d4f2..85f47e66d 100644
--- a/src/ray/util/util.h
+++ b/src/ray/util/util.h
@@ -23,6 +23,10 @@
 #include <thread>
 #include <unordered_map>
 
+#ifdef _WIN32
+#include <process.h> // to ensure getpid() on Windows
+#endif
+
 // Boost forward-declarations (to avoid forcing slow header inclusions)
 namespace boost {
 
-- 
2.11.0

