From 5f4189f62b7f35e585d62c73198c69cda471248f Mon Sep 17 00:00:00 2001
From: Vasilij Litvinov <vasilij.n.litvinov@intel.com>
Date: Mon, 23 Nov 2020 13:14:24 +0300
Subject: [PATCH 11/11] Add debug prints

Signed-off-by: Vasilij Litvinov <vasilij.n.litvinov@intel.com>
---
 python/setup.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/python/setup.py b/python/setup.py
index 3c1d91e0e..ba387dd3e 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -236,6 +236,12 @@ def replace_symlinks_with_junctions():
         path = os.path.join(root_dir, link)
         if os.path.exists(os.path.join(path, '__init__.py')):
             print('"{}" does not need workarounds, skipping'.format(link))
+            for func in (os.stat, os.lstat, os.path.isdir, os.path.isfile):
+                try:
+                    val = func(path)
+                except BaseException as ex:
+                    val = ex
+                print('[DEBUG] {}("{}") = {}', func.__name__, link, val)
             continue
         try:
             out = subprocess.check_output('DIR /A:L /B', shell=True, cwd=os.path.dirname(path))
-- 
2.11.0

