--- a/build_tools/build_ext.py
+++ b/build_tools/build_ext.py
@@ -56,7 +56,7 @@ class CMakeExtension(setuptools.Extension):
             cmake_path,
             "-B",
             build_dir,
-            f"-DPython_EXECUTABLE={sys.executable}",
+            "-DPython_EXECUTABLE=$PYTHON",
             f"-DPython_INCLUDE_DIR={sysconfig.get_path('include')}",
             f"-DCMAKE_BUILD_TYPE={build_type}",
             f"-DCMAKE_INSTALL_PREFIX={install_dir}",
