--- a/pyproject.toml
+++ b/pyproject.toml
@@ -20,8 +20,8 @@
     "pydantic>=2.0.0",
     "smolagents>=1.21.3",
     "typing-extensions>=4.0.0",
-    "tomli==2.2.1",
-    "aiofiles==24.1.0",
+    "tomli>=2.2.1",
+    "aiofiles>=24.1.0",
     "black>=25.1.0",
     "docstring_parser>=0.16",
     "numpy>=2.2.6",
@@ -33,7 +33,7 @@
     "langchain-text-splitters>=1.0.0",
     "nltk>=3.9.1",
     "scipy>=1.15.3",
-    "onnxruntime==1.23.2 ; python_version == '3.10'", # last version that supports python3.10
+    "onnxruntime>=1.23.2 ; python_version == '3.10'", # last version that supports python3.10
 ]
 
 description = "The Agent Lifecycle Toolkit (ALTK) is a library of components to help agent builders improve their agent with minimal integration effort and setup."
@@ -124,7 +124,7 @@
 build = ["uv~=0.8.22"]
 
 spotlight = [
-    "torch==2.2.2 ; sys_platform == 'darwin' and platform_machine == 'x86_64'",
+    "torch>=2.2.2 ; sys_platform == 'darwin' and platform_machine == 'x86_64'",
     "torch>=2.5.1 ; sys_platform != 'darwin' or platform_machine != 'x86_64'",
     "transformers>=4.53.3",
 ]
