From 170338c9979268673d67e79af4a4905d83611778 Mon Sep 17 00:00:00 2001
From: Francois Cartegnie <fcvlcdev@free.fr>
Date: Fri, 3 May 2019 12:58:31 +0200
Subject: [PATCH 32/33] demux: mp4: remove junk debug

---
 modules/demux/mp4/essetup.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modules/demux/mp4/essetup.c b/modules/demux/mp4/essetup.c
index ec22c1d..6e73572 100644
--- a/modules/demux/mp4/essetup.c
+++ b/modules/demux/mp4/essetup.c
@@ -293,7 +293,6 @@ int SetupVideoES( demux_t *p_demux, mp4_track_t *p_track, MP4_Box_t *p_sample )
 
     /* Flip, unless already flipped */
     if (p_track->i_flip == 1 && (int)p_track->f_rotation != 180) {
-    fprintf(stderr, "p_track->f_rotation %f flip %d\n", p_track->f_rotation, p_track->i_flip);
         video_transform_t transform = (video_transform_t)p_track->fmt.video.orientation;
         /* Flip first then rotate */
         p_track->fmt.video.orientation = ORIENT_HFLIPPED;
-- 
2.48.1

