From 76227a02b0b2515aa038605bbd642df7e93b342c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Denis-Courmont?= <remi@remlab.net>
Date: Sat, 2 Mar 2019 21:29:43 +0200
Subject: [PATCH 33/33] mp4: remove unused variable

---
 modules/demux/mp4/libmp4.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modules/demux/mp4/libmp4.c b/modules/demux/mp4/libmp4.c
index b3a0be0..f18d5fa 100644
--- a/modules/demux/mp4/libmp4.c
+++ b/modules/demux/mp4/libmp4.c
@@ -1227,7 +1227,6 @@ static int MP4_ReadBox_tkhd(  stream_t *p_stream, MP4_Box_t *p_box )
     double scale[2];    // scale factor; sx = scale[0] , sy = scale[1]
     int32_t *matrix = p_box->data.p_tkhd->i_matrix;
 
-    int32_t fmatrix[9];
     int64_t det = (int64_t)matrix[0] * matrix[4] - (int64_t)matrix[1] * matrix[3];
     if (det < 0) {
         /* If determinant is negative copy the matrix and flip it horizontally. */
-- 
2.48.1

