diff --git a/share/vlc.appdata.xml.in.in b/share/vlc.appdata.xml.in.in
index cc9c39a..3c55620 100644
--- a/share/vlc.appdata.xml.in.in
+++ b/share/vlc.appdata.xml.in.in
@@ -18,7 +18,7 @@
     <url type="bugtracker">https://trac.videolan.org/vlc/</url>
     <url type="donation">https://www.videolan.org/contribute.html</url>
     <releases>
-        <release version="@VERSION@" />
+        <release version="@VERSION@" date="@DATE@" />
     </releases>
     <provides>
         <library>libvlc.so.5</library>
@@ -38,4 +38,5 @@
             <image>http://images.videolan.org/vlc/screenshots/2.0.0/vlc-2.0-gnome3-debian.jpg</image>
         </screenshot>
     </screenshots>
+    <content_rating type="oars-1.1" />
 </component>
