diff --git a/src/modules/bank.c b/src/modules/bank.c
index 52037d5b59..c0e5ac3885 100644
--- a/src/modules/bank.c
+++ b/src/modules/bank.c
@@ -282,8 +282,8 @@ static int AllocatePluginFile (module_bank_t *bank, const char *abspath,
          && (plugin->mtime != (int64_t)st->st_mtime
           || plugin->size != (uint64_t)st->st_size))
         {
-            msg_Err(bank->obj, "stale plugins cache: modified %s",
-                    plugin->abspath);
+            // msg_Err(bank->obj, "stale plugins cache: modified %s",
+            //         plugin->abspath);
             vlc_plugin_destroy(plugin);
             plugin = NULL;
         }

