diff --git include/cppad/local/cppad_colpack.hpp include/cppad/local/cppad_colpack.hpp
index ee96ab5fe..5fabe2399 100644
--- include/cppad/local/cppad_colpack.hpp
+++ include/cppad/local/cppad_colpack.hpp
@@ -11,9 +11,14 @@ Secondary License when the conditions for such availability set forth
 in the Eclipse Public License, Version 2.0 are satisfied:
       GNU General Public License, Version 2.0 or later.
 ---------------------------------------------------------------------------- */
-# if CPPAD_HAS_COLPACK
 
 namespace CppAD { namespace local { // BEGIN_CPPAD_LOCAL_NAMESPACE
+
+# if ! CPPAD_HAS_COLPACK
+
+extern void this_routine_should_never_get_called(void);
+
+# else
 /*!
 \file cppad_colpack.hpp
 External interface to Colpack routines used by cppad.
@@ -97,8 +102,8 @@ extern void cppad_colpack_symmetric(
     const CppAD::vector<unsigned int*>&  adolc_pattern
 );
 
-} } // END_CPPAD_LOCAL_NAMESPACE
-
 # endif
+
+} } // END_CPPAD_LOCAL_NAMESPACE
 # endif
 
