From 99c3a9102dea80e8d85cb9ea61481ab92d551a41 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Thu, 7 Jan 2021 17:06:59 +0100
Subject: [PATCH] Add include of cctpye for json_lexer in which std::isdigit is
 used

---
 cppad_lib/json_lexer.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cppad_lib/json_lexer.cpp b/cppad_lib/json_lexer.cpp
index 1e0194099..327248d7b 100644
--- a/cppad_lib/json_lexer.cpp
+++ b/cppad_lib/json_lexer.cpp
@@ -15,6 +15,7 @@ CppAD: C++ Algorithmic Differentiation: Copyright (C) 2003-19 Bradley M. Bell
 # include <cppad/utility/to_string.hpp>
 # include <cppad/utility/thread_alloc.hpp>
 
+# include <cctype>
 
 // BEGIN_CPPAD_LOCAL_GRAPH_NAMESPACE
 namespace CppAD { namespace local { namespace graph {
 
