diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd25337..375343e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,15 +36,11 @@ elseif(NOT TARGET Libxc::xcf90)
     "(maybe Libxc was built without the -DENABLE_FORTRAN=True switch?")
 endif()
 
-find_package(Python3 COMPONENTS Interpreter REQUIRED)
-set(PYTHON_INTERPRETER "${Python3_EXECUTABLE}")
-set(PYTHON_VERSION_MAJOR_MINOR "${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}")
 #################### Add source components ####################
 
 add_subdirectory(common/lib)
 add_subdirectory(slateratom)
 add_subdirectory(sktwocnt)
-add_subdirectory(sktools)
 
 #################### Extra install ####################
 
@@ -52,8 +48,3 @@ configure_file(
   ${CMAKE_CURRENT_SOURCE_DIR}/utils/export/skprogs-activate.sh.in
   ${CMAKE_CURRENT_BINARY_DIR}/skprogs-activate.sh
   @ONLY)
-
-install(
-  PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/skprogs-activate.sh"
-  DESTINATION "${CMAKE_INSTALL_BINDIR}/")
-
