diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6a8b694a..23ef8553 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,9 +32,9 @@ endif()
 ### Eigen
 set(VCG_EIGEN_DIR ${CMAKE_CURRENT_LIST_DIR}/eigenlib)
 
-if(VCG_ALLOW_SYSTEM_EIGEN AND EIGEN3_INCLUDE_DIR)
+if(VCG_ALLOW_SYSTEM_EIGEN)
 	message(STATUS "- Eigen - using system-provided library")
-	set(EIGEN_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
+	find_package(Eigen3 CONFIG REQUIRED)
 elseif(VCG_ALLOW_BUNDLED_EIGEN AND EXISTS "${VCG_EIGEN_DIR}/Eigen/Eigen")
 	message(STATUS "- Eigen - using bundled source")
 	set(EIGEN_INCLUDE_DIRS ${VCG_EIGEN_DIR})
@@ -293,8 +293,11 @@ if (VCG_HEADER_ONLY)
 		add_library(vcglib INTERFACE)
 		target_include_directories(
 				vcglib INTERFACE
-				${CMAKE_CURRENT_LIST_DIR}
-				${EIGEN_INCLUDE_DIRS})
+				${CMAKE_CURRENT_LIST_DIR})
+		target_link_libraries(
+				vcglib INTERFACE
+				Eigen3::Eigen	
+		)
 		if(OPENMP_FOUND)
 			target_link_libraries(vcglib INTERFACE OpenMP::OpenMP_CXX)
 		endif()
