diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6a8b694a..80d4d0d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -309,6 +309,21 @@ else()
 	# if the the required targets (e.g. qt, gl, glew...) exists
 endif()
 
+include(GNUInstallDirs)
+install(
+  DIRECTORY vcg
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  COMPONENT headers
+  FILES_MATCHING PATTERN "*.h"
+  PATTERN "*/examples/*" EXCLUDE
+  PATTERN "*/test/*" EXCLUDE
+  PATTERN "*/website/*" EXCLUDE
+)
+install(
+  FILES wrap/callback.h
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/wrap
+)
+
 if(VCG_BUILD_EXAMPLES)
 	#TODO make the list of samples to build
 	add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/apps)
