From 21e69312bc258fb88e1e80eb20b9bb57e55809c6 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 8 Jun 2024 06:26:27 +0200
Subject: [PATCH] Remove auditwheel

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 2f6801904..5f78797c6 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools==68.2.2", "setuptools-scm==8.0.4", "wheel==0.42.0", "auditwheel==5.4.0"]
+requires = ["setuptools==68.2.2", "setuptools-scm==8.0.4", "wheel==0.42.0"]
 build-backend = "setuptools.build_meta"
 
 [tool.setuptools_scm]
