From 039bfedccde8e7bd66f580139a6c8b69a877c5d3 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 7 Jun 2024 21:56:37 +0200
Subject: [PATCH] relative driver path

---
 playwright/_impl/_driver.py | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/playwright/_impl/_driver.py b/playwright/_impl/_driver.py
index 9e8cdc1e7..9dc41548c 100644
--- a/playwright/_impl/_driver.py
+++ b/playwright/_impl/_driver.py
@@ -23,11 +23,9 @@
 
 
 def compute_driver_executable() -> Tuple[str, str]:
-    driver_path = Path(inspect.getfile(playwright)).parent / "driver"
-    cli_path = str(driver_path / "package" / "cli.js")
-    if sys.platform == "win32":
-        return (str(driver_path / "node.exe"), cli_path)
-    return (os.getenv("PLAYWRIGHT_NODEJS_PATH", str(driver_path / "node")), cli_path)
+    cli_path = Path(inspect.getfile(playwright)).parent / ".." / ".." / ".." / "node_modules" / "playwright" / "node_modules" / "playwright-core" / "cli.js"
+    node_path = Path(inspect.getfile(playwright)).parent / ".." / ".." / ".." / ".." / "bin" / "node"
+    return (str(node_path), str(cli_path))
 
 
 def get_driver_env() -> dict:
