From e467402f29862e24e870bea347164da4c4b1ce1f Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 7 Jun 2024 16:07:55 +0200
Subject: [PATCH] Disable downloads

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 8709e52a2..c09304d74 100644
--- a/setup.py
+++ b/setup.py
@@ -120,7 +120,7 @@ def run(self) -> None:
                 "zip_name": "win32_x64",
             },
         ]
-        self._download_and_extract_local_driver(base_wheel_bundles)
+        # self._download_and_extract_local_driver(base_wheel_bundles)
 
         wheels = base_wheel_bundles
         if not self.all:
@@ -132,7 +132,7 @@ def run(self) -> None:
                     base_wheel_bundles,
                 )
             )[:1]
-        self._build_wheels(wheels)
+        # self._build_wheels(wheels)
 
     def _build_wheels(
         self,
