From d6476831fdfbb31464edf82bfb5bc9670d0fa83c Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 7 Jun 2024 14:08:05 +0200
Subject: [PATCH] Update pyproject.toml

---
 pyproject.toml | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 2f6801904..543ba5b3c 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,8 +2,9 @@
 requires = ["setuptools==68.2.2", "setuptools-scm==8.0.4", "wheel==0.42.0", "auditwheel==5.4.0"]
 build-backend = "setuptools.build_meta"
 
-[tool.setuptools_scm]
-version_file = "playwright/_repo_version.py"
+[project]
+name = "playwright"
+version = "1.44.0"
 
 [tool.pytest.ini_options]
 addopts = "-Wall -rsx -vv -s"
