From 1ff1bd9afa34b673a85be04935ab4ec9c516d5bf Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Fri, 7 Jun 2024 13:43:18 +0200
Subject: [PATCH] patch version

---
 pyproject.toml | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 2f68019..543ba5b 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,8 +2,9 @@
 requires = ["setuptools==68.2.2", "setuptools-scm==8.1.0", "wheel==0.42.0", "auditwheel==5.4.0"]
 build-backend = "setuptools.build_meta"
 
-[tool.setuptools_scm]
-version_file = "playwright/_repo_version.py"
+[project]
+name = "playwright"
+version = "1.44.0"
 
 [tool.pytest.ini_options]
 addopts = "-Wall -rsx -vv -s"
