diff --git a/pkg/rootless/rootless_linux.c b/pkg/rootless/rootless_linux.c
index 2e1fddc..3e47030 100644
--- a/pkg/rootless/rootless_linux.c
+++ b/pkg/rootless/rootless_linux.c
@@ -19,6 +19,16 @@
 #include <sys/select.h>
 #include <stdio.h>
 
+#ifndef HAVE_SETNS
+# include <unistd.h>
+# include <sys/syscall.h>
+/* NB: setns syscall was introduced in Linux 3.0 and is not available on CentOS 6. */
+# if !defined(__NR_setns) && defined(__x86_64__)
+#  define __NR_setns 308
+# endif
+# define setns(fd, nstype) syscall(__NR_setns, fd, nstype)
+#endif
+
 int rename_noreplace (int olddirfd, const char *oldpath, int newdirfd, const char *newpath)
 {
   int ret;
diff --git a/vendor/github.com/containers/storage/pkg/unshare/unshare.c b/vendor/github.com/containers/storage/pkg/unshare/unshare.c
index c0e359b..90b33fe 100644
--- a/vendor/github.com/containers/storage/pkg/unshare/unshare.c
+++ b/vendor/github.com/containers/storage/pkg/unshare/unshare.c
@@ -30,6 +30,18 @@
 #define MFD_CLOEXEC 1U
 #endif
 
+/* NB: O_PATH was introduced in Linux 2.6.39 and is not available on CentOS 6. */
+#if !defined(O_PATH)
+#define O_PATH 010000000
+#endif
+/* NB: memfd_create syscall was introduced in Linux 3.17 and got backported to CentOS 7. */
+#if !defined(__NR_memfd_create) && defined(__x86_64__)
+#define __NR_memfd_create 319
+#endif
+#if !defined(SYS_memfd_create) && defined(__NR_memfd_create)
+#define SYS_memfd_create __NR_memfd_create
+#endif
+
 #ifndef F_LINUX_SPECIFIC_BASE
 #define F_LINUX_SPECIFIC_BASE 1024
 #endif
