diff --git a/gtk/meson.build b/gtk/meson.build
index ac8c1a9..c5479d8 100644
--- a/gtk/meson.build
+++ b/gtk/meson.build
@@ -1046,7 +1046,8 @@ gtk_query_settings = executable(
   'gtk-query-settings.c',
   c_args: gtk_cargs,
   dependencies: libgtk_dep,
-  install: true
+  install: true,
+  install_rpath: gtk_libdir
 )
 
 gtk_builder_tool = executable(
@@ -1054,7 +1055,8 @@ gtk_builder_tool = executable(
   'gtk-builder-tool.c',
   c_args: gtk_cargs,
   dependencies: libgtk_dep,
-  install: true
+  install: true,
+  install_rpath: gtk_libdir
 )
 
 gtk_update_icon_cache = executable(
@@ -1062,7 +1064,8 @@ gtk_update_icon_cache = executable(
   'updateiconcache.c',
   c_args: gtk_cargs,
   dependencies: libgtk_dep,
-  install: true
+  install: true,
+  install_rpath: gtk_libdir
 )
 
  gtk_query_immodules = executable(
@@ -1071,7 +1074,8 @@ gtk_update_icon_cache = executable(
   'gtkutils.c',
   c_args: gtk_cargs,
   dependencies: libgtk_dep,
-  install: true
+  install: true,
+  install_rpath: gtk_libdir
 )
 
 gtk_encode_symbolic_svg = executable(
@@ -1079,7 +1083,8 @@ gtk_encode_symbolic_svg = executable(
   'encodesymbolic.c',
   c_args: gtk_cargs,
   dependencies: libgtk_dep,
-  install: true
+  install: true,
+  install_rpath: gtk_libdir
 )
 
 gtk_launch = executable(
@@ -1087,5 +1092,6 @@ gtk_launch = executable(
   'gtk-launch.c',
   c_args: gtk_cargs,
   dependencies: libgtk_dep,
-  install: true
+  install: true,
+  install_rpath: gtk_libdir
 )
