--- gtk/gtkmodules.c.orig	2016-03-12 16:38:07.046075646 +0000
+++ gtk/gtkmodules.c	2016-03-12 16:39:19.034419061 +0000
@@ -378,8 +378,10 @@
         {
           const gchar *error = g_module_error ();
 
-          g_message ("Failed to load module \"%s\"%s%s",
-                     name, error ? ": " : "", error ? error : "");
+	  if (error) {
+	    g_message ("Failed to load module \"%s\"%s%s",
+		       name, error ? ": " : "", error ? error : "");
+	  }
         }
     }
 
