From 9a423407afe94ed96fedee1ecd87ababb232f0cb Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Wed, 25 Nov 2020 21:03:49 +0100
Subject: [PATCH] Use scikit-learn rather than sklearn

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 1f84b92..3ebb3f6 100644
--- a/setup.py
+++ b/setup.py
@@ -40,7 +40,7 @@
             "scipy",
             "ase",
             "spglib>=1.15.0",
-            "sklearn",
+            "scikit-learn",
             "networkx>=2.4",
             "chronic"
         ],
