From 5eff1ecc8413b0dc93a1ab047d7fed751e6cb40e Mon Sep 17 00:00:00 2001
From: Samuel Debionne <samuel.debionne@esrf.fr>
Date: Fri, 30 Apr 2021 11:55:58 +0200
Subject: [PATCH] Add default value for cxx and cxxflags options for the cxx

---
 bootstrap.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bootstrap.sh b/bootstrap.sh
index 654801e21..1c75a39f4 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -226,7 +226,7 @@ rm -f config.log
 if test "x$BJAM" = x; then
   $ECHO "Building B2 engine.."
   pwd=`pwd`
-  CXX= CXXFLAGS= "$my_dir/tools/build/src/engine/build.sh" ${TOOLSET}
+  "$my_dir/tools/build/src/engine/build.sh" ${TOOLSET} --cxx="$CXX" --cxxflags="$CXXFLAGS"
   if [ $? -ne 0 ]; then
       echo
       echo "Failed to build B2 build engine"
--
2.20.1
