From dcfa761c82a7f64f53a7daf3c11c6cab99263237 Mon Sep 17 00:00:00 2001
From: Ista Zahn <istazahn@gmail.com>
Date: Sun, 13 Mar 2022 11:55:20 -0400
Subject: [PATCH] fix soci

---
 src/cpp/core/DatabaseTests.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/cpp/core/DatabaseTests.cpp b/src/cpp/core/DatabaseTests.cpp
index 6688b2f..d9f3125 100644
--- a/src/cpp/core/DatabaseTests.cpp
+++ b/src/cpp/core/DatabaseTests.cpp
@@ -13,6 +13,10 @@
  *
  */
 
+#define SOCI_USE_BOOST
+#define HAVE_BOOST
+#define SOCI_HAVE_BOOST
+
 #include <tests/TestThat.hpp>
 
 #include <core/Database.hpp>
-- 
2.35.1

