diff --git a/setup.py b/setup.py
index 7385190..b72636a 100755
--- a/setup.py
+++ b/setup.py
@@ -36,7 +36,7 @@ if sys.platform.startswith("linux"):
             Extension(
                 "hid",
                 sources=src,
-                include_dirs=[hidapi_include, "/usr/include/libusb-1.0"],
+                include_dirs=[hidapi_include, os.path.join(os.environ.get("PREFIX", "/usr"), "include", "libusb-1.0")],
                 libraries=libs,
             )
         )
