--- setup.py.orig	2016-09-25 20:32:57.885171517 +0100
+++ setup.py	2016-09-25 20:33:06.368134063 +0100
@@ -52,10 +52,6 @@
 
     # environment variables not set, using defaults instead
     blas_info = get_info('blas_opt')
-    # ugly hack due to scipy bug
-    if 'libraries' in blas_info:
-        if 'mkl_intel_lp64' in blas_info['libraries']:
-            blas_info = get_info('blas')
     if not blas_info:
         blas_info = get_info('blas')
     print(blas_info)
