diff --git a/activemq-cpp/configure.ac b/activemq-cpp/configure.ac
index f7ec827c..afa6c553 100644
--- a/activemq-cpp/configure.ac
+++ b/activemq-cpp/configure.ac
@@ -155,9 +155,7 @@ AC_CHECK_HEADERS([semaphore.h])
 
 AC_CHECK_FUNCS([ioctl select gettimeofday time ftime random srandom])
 
-AMQ_FIND_CPPUNIT( 1.10.2, cppunit=yes, cppunit=no;
-    AC_MSG_RESULT([no. Unit and Integration tests disabled])
-)
+PKG_CHECK_MODULES([CPPUNIT], [cppunit >= 1.10.2], [cppunit=yes], [no_cppunit=yes])
 AM_CONDITIONAL(BUILD_CPPUNIT_TESTS, test x$cppunit = xyes)
 
 # Absolute source/build directory
