--- setuptools/package_index.py
+++ setuptools/package_index.py
@@ -282,6 +282,10 @@ class PackageIndex(Environment):
 
     def process_url(self, url, retrieve=False):
         """Evaluate a URL as a possible download, and maybe retrieve it"""
+        if os.getenv("CONDA_BUILD"):
+            raise RuntimeError("Setuptools downloading is disabled in conda build. "
+                               "Be sure to add all dependencies in the meta.yaml  url=%sr" % url)
+
         if url in self.scanned_urls and not retrieve:
             return
         self.scanned_urls[url] = True
