--- setup.py
+++ setup.py
@@ -145,12 +145,8 @@ setup_params = dict(
         Topic :: Utilities
         """).strip().splitlines(),
     extras_require={
-        "ssl:sys_platform=='win32'": "wincertstore==0.2",
-        "certs": "certifi==2015.11.20",
     },
     dependency_links=[
-        'https://pypi.python.org/packages/source/c/certifi/certifi-2015.11.20.tar.gz#md5=25134646672c695c1ff1593c2dd75d08',
-        'https://pypi.python.org/packages/source/w/wincertstore/wincertstore-0.2.zip#md5=ae728f2f007185648d0c7a8679b361e2',
     ],
     scripts=[],
     tests_require=[
