--- a/bindgen/src/builders.rs
+++ b/bindgen/src/builders.rs
@@ -48,2 +48,5 @@

-        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules"];
+        let sdk_path = self.sdk.path().to_str()
+            .expect("sdk path is not utf-8 representable");
+        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules"
+        "-isysroot", sdk_path, "-I", &format!("{}/usr/include", sdk_path), "--no-system-header-prefix"];
         let target_arg;
@@ -55,9 +58,2 @@

-        clang_args.extend(&[
-            "-isysroot",
-            self.sdk
-                .path()
-                .to_str()
-                .expect("sdk path is not utf-8 representable"),
-        ]);

