--- a/bindgen/src/builder.rs
+++ b/bindgen/src/builder.rs
@@ -1,3 +1,3 @@
 pub use crate::{
-    config::{Config, ConfigMap, FileConfig},
+    config::{Config, ConfigMap},
     sdk::{SdkPath, SdkPathError},
@@ -48,2 +48,6 @@

-        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules"];
+        let sdk_path = self.sdk.path().to_str()
+            .expect("sdk path is not utf-8 representable");
+        let include_path = format!("{}/usr/include", sdk_path);
+        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules",
+        "-isysroot", sdk_path, "-I", &include_path, "--no-system-header-prefix"];
         let target_arg;
@@ -55,9 +59,2 @@

-        clang_args.extend(&[
-            "-isysroot",
-            self.sdk
-                .path()
-                .to_str()
-                .expect("sdk path is not utf-8 representable"),
-        ]);

