--- a/bindgen/src/builder.rs
+++ b/bindgen/src/builder.rs
@@ -31,0 +32,3 @@
+        println!("Building config for framework: {}", framework);
+        let config = ConfigMap::with_builtin_config().framework_config(framework);
+        println!("Config: {:?}", config);
@@ -35 +41 @@
-            ConfigMap::with_builtin_config().framework_config(framework),
+            config,
@@ -47,0 +51 @@
+        println!("Building with SDK path: {:?}", self.sdk.path());
@@ -49 +53,6 @@
-        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules"];
+        let sdk_path = self.sdk.path().to_str()
+            .expect("sdk path is not utf-8 representable");
+        // let include_path = format!("{}/usr/include", sdk_path);
+        let framework_path = format!("{}/System/Library/Frameworks", sdk_path);
+        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules",
+        "-F", &framework_path, "-isysroot", sdk_path, "-nostdinc"];
@@ -58,4 +67 @@
-            self.sdk
-                .path()
-                .to_str()
-                .expect("sdk path is not utf-8 representable"),
+            sdk_path,
@@ -79,0 +86 @@
+        println!("Generated header content:\n{}", format!("@import {};", self.framework));
