--- a/bindgen/src/builder.rs
+++ b/bindgen/src/builder.rs
@@ -31,0 +32,3 @@
+        println!("SDK path: {:?}", sdk);
+        let sdk_path = sdk.try_into()?;
+        println!("Resolved SDK path: {:?}", sdk_path.path());
@@ -49 +52,6 @@
-        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules"];
+        let sdk_path = self.sdk.path().to_str()
+            .expect("sdk path is not utf-8 representable");
+        // let include_path = format!("{}/usr/include", sdk_path);
+        let framework_path = format!("{}/System/Library/Frameworks", sdk_path);
+        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules",
+        "-F", &framework_path];
@@ -58,4 +66 @@
-            self.sdk
-                .path()
-                .to_str()
-                .expect("sdk path is not utf-8 representable"),
+            sdk_path,
