--- a/bindgen/src/builder.rs
+++ b/bindgen/src/builder.rs
@@ -48,2 +48,7 @@

-        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules"];
+        let sdk_path = self.sdk.path().to_str()
+            .expect("sdk path is not utf-8 representable");
+        let include_path = format!("{}/usr/include", sdk_path);
+        let framework_path = format!("{}/System/Library/Frameworks", sdk_path);
+        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules",
+        "-isysroot", sdk_path, "-F", &framework_path, "-I", &include_path, "-nostdinc"];
         let target_arg;
@@ -55,9 +60,2 @@

-        clang_args.extend(&[
-            "-isysroot",
-            self.sdk
-                .path()
-                .to_str()
-                .expect("sdk path is not utf-8 representable"),
-        ]);

