--- a/bindgen/src/builder.rs
+++ b/bindgen/src/builder.rs
@@ -31,2 +31,5 @@
     ) -> Result<Self, SdkPathError> {
+        println!("SDK path: {:?}", sdk);
+        let sdk_path = sdk.try_into()?;
+        println!("Resolved SDK path: {:?}", sdk_path.path());
         Self::new(
@@ -49,3 +52,8 @@
-        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules"];
+        let sdk_path = self.sdk.path().to_str()
+            .expect("sdk path is not utf-8 representable");
+        // let include_path = format!("{}/usr/include", sdk_path);
+        let framework_path = format!("{}/System/Library/Frameworks", sdk_path);
+        let mut clang_args = vec!["-x", "objective-c", "-fblocks", "-fmodules",
+        "-F", &framework_path];
         let target_arg;
@@ -57,6 +65,3 @@
             "-isysroot",
-            self.sdk
-                .path()
-                .to_str()
-                .expect("sdk path is not utf-8 representable"),
+            sdk_path,
         ]);
