diff --git a/source/wrappers/python/win.setup.py.in b/source/wrappers/python/win.setup.py.in
index cb56dd95..8d52b312 100644
--- a/source/wrappers/python/win.setup.py.in
+++ b/source/wrappers/python/win.setup.py.in
@@ -44,7 +44,7 @@ if build_fat:
     extra_macros = [('FATCLIENT', None)]
 else:
     uda_libs = [
-        '@PROJECT_NAME@_client.dll.' + stalib_ext,
+        '@PROJECT_NAME@_client.' + stalib_ext,
     ] + ['@PROJECT_NAME@_' + name + '.' + stalib_ext for name in uda_lib_names]
     extra_libs = '@EXTRA_LIBS@'.split(';') if '@EXTRA_LIBS@' else []
     extra_macros = []
