diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 5ba20c8b..eccc4b6c 100755
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -20,7 +20,7 @@ if( WIN32 AND NOT MINGW )
   add_definitions( /W2 )
   #add_definitions( /W2 /WX )
 else()
-  add_definitions( -Wall -Werror -Wextra -Wno-unused-parameter -Wno-missing-field-initializers )
+  add_definitions( -Wall -Werror -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-cast-function-type )
   if( "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND ${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER 5.0 )
     add_definitions( -Wno-format-truncation -Wno-format-overflow )
     if( NOT MINGW AND NOT ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "IntelLLVM") )
