diff --git a/source/bin/uda_cli.cpp b/source/bin/uda_cli.cpp
index 527dc621..6a7e953d 100644
--- a/source/bin/uda_cli.cpp
+++ b/source/bin/uda_cli.cpp
@@ -410,7 +410,7 @@ int main(int argc, const char** argv)
         if (vm["help"].as<bool>()) {
             std::cout << "Usage: " << argv[0] << " [options] request\n";
             std::cout << desc << "\n";
-            return 1;
+            return 0;
         } else {
             std::cout << "Error: " << err.what() << "\n\n";
             std::cout << "Usage: " << argv[0] << " [options] request\n";
@@ -422,7 +422,7 @@ int main(int argc, const char** argv)
     if (vm["help"].as<bool>()) {
         std::cout << "Usage: " << argv[0] << " [options] request\n";
         std::cout << desc << "\n";
-        return 1;
+        return 0;
     }
 
     if (vm.count("host")) {
