diff --git a/source/client/udaClient.cpp b/source/client/udaClient.cpp
index 61e33170..79aaaac3 100755
--- a/source/client/udaClient.cpp
+++ b/source/client/udaClient.cpp
@@ -121,7 +121,7 @@ int check_file_cache(const REQUEST_DATA* request_data, DATA_BLOCK** p_data_block
     if (client_flags->flags & CLIENTFLAG_FILECACHE && !request_data->put) {
         // Query the cache for the Data
         DATA_BLOCK* data = udaFileCacheRead(request_data, log_malloc_list, user_defined_type_list, proto_version,
-                                            log_struct_list, private_flags, malloc_source)
+                                            log_struct_list, private_flags, malloc_source);
 
         if (data != nullptr) {
             // Success
