diff --git a/fastBPE/fastBPE.hpp b/fastBPE/fastBPE.hpp
index 099ae80..d99b174 100644
--- a/fastBPE/fastBPE.hpp
+++ b/fastBPE/fastBPE.hpp
@@ -12,7 +12,7 @@
 #include <stdio.h>
 #include <string>
 #include <cstring>
-#include <sys/mman.h>
+#include "mman.h"
 #include <sys/stat.h>
 #include <thread>
 #include <unistd.h> // ftruncate
@@ -689,4 +689,4 @@ void applybpe_stream(const char *codesPath, const char *vocabPath) {
   }
 }
 
-};
\ No newline at end of file
+};
