diff --git a/bash_completion b/bash_completion
index 2718c61f..28c6298f 100644
--- a/bash_completion
+++ b/bash_completion
@@ -2042,6 +2042,10 @@ complete -F _minimal ''
 __load_completion()
 {
     local -a dirs=( ${BASH_COMPLETION_USER_DIR:-${XDG_DATA_HOME:-$HOME/.local/share}/bash-completion}/completions )
+
+    # Add completions relative to anaconda's relocatable install path
+    dirs+=( /opt/anaconda1anaconda2anaconda3/share/bash-completion/completions )
+
     local OIFS=$IFS IFS=: dir cmd="${1##*/}" compfile
     [[ -n $cmd ]] || return 1
     for dir in ${XDG_DATA_DIRS:-/usr/local/share:/usr/share}; do
