diff --git a/bash_completion.sh.in b/bash_completion.sh.in
index a28e1a62..4d84c577 100644
--- a/bash_completion.sh.in
+++ b/bash_completion.sh.in
@@ -1,5 +1,5 @@
 # Check for interactive bash and that we haven't already been sourced.
-if [ "x${BASH_VERSION-}" != x -a "x${PS1-}" != x -a "x${BASH_COMPLETION_VERSINFO-}" = x ]; then
+if [ "x${BASH_VERSION-}" != x -a "x${PS1-}" != x -a "x${CONDA_BASH_COMPLETION_LOADED-}" = x ]; then
 
     # Check for recent enough version of bash.
     if [ ${BASH_VERSINFO[0]} -gt 4 ] || \
