From f4197dc56942cfb10ac9ba4b3911f1ed379df009 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 18 Dec 2022 18:43:30 +1100
Subject: [PATCH] do not require clang, resp. specific LLVM version

---
 clif/cmake/modules/CLIFUtils.cmake | 2 --
 1 file changed, 2 deletions(-)

diff --git a/clif/cmake/modules/CLIFUtils.cmake b/clif/cmake/modules/CLIFUtils.cmake
index aa95d37..897e1fd 100644
--- a/clif/cmake/modules/CLIFUtils.cmake
+++ b/clif/cmake/modules/CLIFUtils.cmake
@@ -33,8 +33,6 @@ pkg_check_modules(GOOGLE_PROTOBUF REQUIRED protobuf)
 pkg_check_modules(GOOGLE_GLOG REQUIRED libglog)
 
 find_package(absl REQUIRED)
-find_package(Clang REQUIRED)
-find_package(LLVM 11.1.0 REQUIRED)
 find_package(GTest REQUIRED)
 
 function(add_protobuf_include_directories)
-- 
2.38.1.windows.1

