diff --git a/blosc/CMakeLists.txt b/blosc/CMakeLists.txt
index fdbcb6c..a1229e9 100644
--- a/blosc/CMakeLists.txt
+++ b/blosc/CMakeLists.txt
@@ -238,9 +238,16 @@ if (BUILD_TESTS)
     # TEMP : CMake doesn't automatically add -lpthread here like it does
     # for the blosc2_shared target. Force it for now.
     if (UNIX)
-        set_property(
+        if (APPLE)
+            set_property(
                 TARGET blosc_shared_testing
                 APPEND PROPERTY LINK_FLAGS "-lpthread")
+        else (APPLE)
+            # Pretty old Linuxes like CentOS 6 and 7 need -lrt too
+            set_property(
+                TARGET blosc_shared_testing
+                APPEND PROPERTY LINK_FLAGS "-lpthread -lrt")
+        endif (APPLE)
     endif ()
 endif ()
 
