From 379266c74e0a3fce4e48b4788ad32508b02ec626 Mon Sep 17 00:00:00 2001
From: Carlos O'Ryan <coryan@google.com>
Date: Wed, 20 Dec 2023 16:57:10 +0000
Subject: [PATCH] Backport #12911 from upstream

---
 cmake/GoogleCloudCppFeatures.cmake | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/cmake/GoogleCloudCppFeatures.cmake b/cmake/GoogleCloudCppFeatures.cmake
index a69cd85a..e86b9d03 100644
--- a/cmake/GoogleCloudCppFeatures.cmake
+++ b/cmake/GoogleCloudCppFeatures.cmake
@@ -361,9 +361,6 @@ macro (google_cloud_cpp_enable_cleanup)
         OR (sql IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
         OR (generator IN_LIST GOOGLE_CLOUD_CPP_ENABLE))
         set(GOOGLE_CLOUD_CPP_ENABLE_REST ON)
-        # Backwards compatibility. In the original release of `oauth2` we
-        # automatically compiled the library if REST was enabled
-        list(APPEND GOOGLE_CLOUD_CPP_ENABLE oauth2)
     endif ()
 
     list(REMOVE_DUPLICATES GOOGLE_CLOUD_CPP_ENABLE)
