From 458163c93d96f0b951803c99d77ae15580cbea7e Mon Sep 17 00:00:00 2001
From: Tom Vander Aa <vanderaa@imec.be>
Date: Wed, 4 Aug 2021 11:03:05 +0200
Subject: [PATCH] FIX: for LAPACK on windows, add -DLAPACK_COMPLEX_STRUCTURE=1

---
 cmake/DependenciesConfig.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/DependenciesConfig.cmake b/cmake/DependenciesConfig.cmake
index 89785352..7e851da3 100644
--- a/cmake/DependenciesConfig.cmake
+++ b/cmake/DependenciesConfig.cmake
@@ -45,6 +45,10 @@ macro(configure_lapack)
   find_package(LAPACK REQUIRED)
   find_package(LAPACKE REQUIRED)
   add_definitions(-DEIGEN_USE_BLAS -DEIGEN_USE_LAPACKE)
+  if (MSVC)
+    # needed because MSVC does not have support for c-type _Complex
+    add_definitions(-Dlapack_complex_float=_Fcomplex -Dlapack_complex_double=_Dcomplex)
+  endif()
   message(STATUS LAPACK: ${LAPACK_LIBRARIES})
 endmacro(configure_lapack)
 
-- 
2.30.2

