From 9885dc65807fab677db57fc05f8b6655c9789be1 Mon Sep 17 00:00:00 2001
From: Tom Vander Aa <vanderaa@imec.be>
Date: Tue, 3 Aug 2021 12:23:47 +0200
Subject: [PATCH] WIP: dynamic boost on Windows

---
 cmake/DependenciesConfig.cmake | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/cmake/DependenciesConfig.cmake b/cmake/DependenciesConfig.cmake
index 7b128a41..89785352 100644
--- a/cmake/DependenciesConfig.cmake
+++ b/cmake/DependenciesConfig.cmake
@@ -101,9 +101,6 @@ endmacro(configure_highfive)
 macro(configure_boost)
   message ("Dependency check for boost...")
   if(${ENABLE_BOOST})
-      if(MSVC)
-        set(Boost_USE_STATIC_LIBS ON)
-      endif(MSVC)
       FIND_PACKAGE(Boost COMPONENTS system program_options filesystem REQUIRED)
 
       message("-- Found Boost_VERSION: ${Boost_VERSION}")
-- 
2.25.1

