diff --git a/setup.py b/setup.py
index 6c736c6..904464f 100644
--- a/setup.py
+++ b/setup.py
@@ -26,7 +26,6 @@ setup(
     maintainer='Pavel Raiskup',
     maintainer_email='praiskup@redhat.com',
     packages=find_packages(),
-    scripts=['bin/argparse-manpage'],
     description='Build manual page from python\'s ArgumentParser object.',
     long_description=get_readme(),
     long_description_content_type='text/markdown',
