diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0f12eb..7742f79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,6 +73,7 @@ if(PROJECT_IS_TOP_LEVEL)
 	option(BOX2D_PROFILE "Enable profiling with Tracy" OFF)
 	option(BOX2D_VALIDATE "Enable heavy validation" ON)
 	option(BOX2D_UNIT_TESTS "Build the Box2D unit tests" ON)
+	option(BOX2D_COMPILE_WARNING_AS_ERROR "Compile warnings as errors" OFF)
 
 	if(MSVC AND WIN32)
 		# add_compile_options(/fsanitize=address)
