diff --git a/Bindings/Python/CMakeLists.txt b/Bindings/Python/CMakeLists.txt
index 977133801..4231b770f 100644
--- a/Bindings/Python/CMakeLists.txt
+++ b/Bindings/Python/CMakeLists.txt
@@ -129,10 +129,31 @@ macro(OpenSimAddPythonModule)
 
     add_library(${_libname} SHARED ${_output_cxx_file} ${_output_header_file})
 
-    target_include_directories(${_libname} PRIVATE ${PYTHON_INCLUDE_PATH})
+    target_include_directories(${_libname} PRIVATE
+            "${PYTHON_INCLUDE_PATH}")
     
-    target_link_libraries(${_libname}
-        osimTools osimExampleComponents ${PYTHON_LIBRARIES})
+    # On macOS, the python interpreter might link to the python library 
+    # statically, in which case we do not want to link dynamically to the
+    # python library. This situation occurs with Anaconda's python.
+    # In this situation, we must add a linker flag to mark all undefined
+    # symbols as having to be looked up at runtime, and we ignore
+    # PYTHON_LIBRARIES.
+    # https://github.com/opensim-org/opensim-core/issues/2771
+    # https://stackoverflow.com/questions/25421479/clang-and-undefined-symbols-when-building-a-library
+    # https://github.com/lisitsyn/shogun/commit/961f6109c2a8abab4941bcdde1a19dfe70c440f1
+    # https://github.com/shogun-toolbox/shogun/issues/4068
+    execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c
+        "import sysconfig; print(sysconfig.get_config_var('LDSHARED'))"
+        OUTPUT_VARIABLE PYTHON_LDSHARED
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if("${PYTHON_LDSHARED}" MATCHES "dynamic_lookup")
+        set_target_properties(${_libname} PROPERTIES LINK_FLAGS
+            "-undefined dynamic_lookup")
+    else()
+        target_link_libraries(${_libname} ${PYTHON_LIBRARIES})
+    endif()
+
+    target_link_libraries(${_libname} osimTools osimExampleComponents)
     
     # Set target properties for various platforms.
     # --------------------------------------------
@@ -334,4 +355,4 @@ install(DIRECTORY ../Java/Matlab/OpenSenseExample
         # Exclude Matlab m files
         PATTERN "*.m" EXCLUDE)
 install(DIRECTORY OpenSenseExample
-        DESTINATION "${OPENSIM_INSTALL_PYTHONEXDIR}")
\ No newline at end of file
+        DESTINATION "${OPENSIM_INSTALL_PYTHONEXDIR}")
