diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb71658..3292816 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -168,7 +168,7 @@ target_sources(pinch
 	PUBLIC
 	FILE_SET public_headers TYPE HEADERS
 	BASE_DIRS
-	${CMAKE_CURRENT_SOURCE_DIR}
+	${CMAKE_CURRENT_SOURCE_DIR}/include
 	FILES
 	${PINCH_HEADERS}
 )
@@ -216,4 +216,4 @@ endif()
 
 # if(PINCH_BUILD_EXAMPLES)
 # 	add_subdirectory(examples)
-# endif()
\ No newline at end of file
+# endif()
