From 46077b34ee2d056514b58ca6ee15cbe5ee2d4309 Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Wed, 15 Mar 2017 09:25:34 +0100
Subject: [PATCH 1/2] get_temp_location-keyerror

---
 src/common/core.py | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/common/core.py b/src/common/core.py
index fff82ef..097772b 100644
--- a/src/common/core.py
+++ b/src/common/core.py
@@ -30,12 +30,10 @@ import numpy.ctypeslib as Nct
 
 # location for temporary JModelica files
 def get_temp_location():
-    if sys.platform == 'win32':
-        return os.path.join(tempfile._get_default_tempdir(),'JModelica.org')
-    elif sys.platform == 'darwin':
-        return os.path.join(tempfile._get_default_tempdir(),'JModelica.org')
-    else:
+    if 'USER' in os.environ:
         return os.path.join(tempfile._get_default_tempdir(),os.environ['USER'],'JModelica.org')
+    else:
+        return os.path.join(tempfile._get_default_tempdir(),'JModelica.org')
 
 tmp_location = get_temp_location()
 
-- 
2.7.4

