diff --git a/winfiletime/__init__.py b/winfiletime/__init__.py
index 8ff37a1..06aed48 100644
--- a/winfiletime/__init__.py
+++ b/winfiletime/__init__.py
@@ -1,3 +1,3 @@
-from .filetime import EPOCH_AS_FILETIME, from_datetime, to_datetime, utc
+from .filetime import EPOCH_AS_FILETIME, from_datetime, to_datetime
 
-__all__ = ["EPOCH_AS_FILETIME", "from_datetime", "to_datetime", "utc"]
+__all__ = ["EPOCH_AS_FILETIME", "from_datetime", "to_datetime"]
