diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 6f2a9b4c..54a9fc8f 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -418,7 +418,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang
   else()
     message(STATUS "Enabling Clang-specific build options.")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnull-dereference -Wdangling-else")
-    target_link_libraries(katago "atomic")
+    # target_link_libraries(katago "atomic")
   endif()
 
   if(USE_TCMALLOC)
