diff --git a/config.py b/config.py
index d2b89f111..19148cfa4 100644
--- a/config.py
+++ b/config.py
@@ -470,8 +470,12 @@ def build_interpreter(define_macros, include_dirs, libraries, library_dirs,
     if glob(libpl + '/libpython*mpi*'):
         libs += ' -lpython%s_mpi' % cfgDict['VERSION']
     else:
-        libs += ' ' + cfgDict.get('BLDLIBRARY',
-                                  '-lpython%s' % cfgDict['VERSION'])
+        pylib = os.path.join(*[cfgDict[k] for k in 'LIBDIR INSTSONAME'.split()])
+        if os.path.exists(pylib):
+            libs += ' ' + pylib
+        else:
+            libs += ' ' + cfgDict.get('BLDLIBRARY',
+                                      '-lpython%s' % cfgDict['VERSION'])
     libs = ' '.join([libs, cfgDict['LIBS'], cfgDict['LIBM']])
 
     # Hack taken from distutils to determine option for runtime_libary_dirs
