diff --git a/setup.py b/setup.py
index d1ff183..5a9c0c8 100644
--- a/setup.py
+++ b/setup.py
@@ -5,17 +5,12 @@ import numpy as np
 # Build project with:
 # python setup.py build_ext --inplace
 
-with open("README.rst") as fp:
-    readme = fp.read()
-
-with open("HISTORY.rst") as fp:
-    history = fp.read()
 
 ext_modules = [
     Extension(
         "cyl1tf.interface",
         ["cyl1tf/interface.pyx", "source/l1tf.c"],
-        libraries=["m", "blas", "lapack"],
+        libraries=["msvcr", "blas", "lapack"],
         include_dirs=["cyl1tf", "source", np.get_include()],
         extra_compile_args=["-Ofast"],
     ),
@@ -25,8 +20,6 @@ setup(
     name="cyl1tf",
     version="0.1.3",
     description="Cython implementation of L1 trend fitting.",
-    long_description=readme + "\n\n" + history,
-    long_description_content_type="text/x-rst",
     author="Albert Kottke",
     author_email="albert.kottke@gmail.com",
     url="https://github.com/arkottke/cyl1tf",
