diff --git a/config/cmake/Modules/FindSASL.cmake b/config/cmake/Modules/FindSASL.cmake
index 9ebd37d..20d1974 100644
--- a/config/cmake/Modules/FindSASL.cmake
+++ b/config/cmake/Modules/FindSASL.cmake
@@ -24,7 +24,7 @@ IF( WITH_SASL_NO )
   # Simply do nothing about detecting SASL
   #======================================================================
   MESSAGE(STATUS "SASL: disabled")
-ELSEIF( WITH_SASL_LENGTH EQUAL 0 )
+ELSEIF( NOT WITH_SASL_LENGTH EQUAL 0 )
   MESSAGE(STATUS "SASL: User supplied")
   #======================================================================
   # User is suplying hints of where the SASL library is located.
