diff --git a/bin/mangohud.in b/bin/mangohud.in
index 96468bd..8056785 100755
--- a/bin/mangohud.in
+++ b/bin/mangohud.in
@@ -8,6 +8,16 @@ if [ "$#" -eq 0 ]; then
     exit 1
 fi
 
+# Implicit layers are hard to find by defining other environment variables
+# We don't define the XDG_DATA_DIRS on conda, so we set it here
+# Since this is a subshell, this won't affect the user's main shell
+# Until we decide that it is worthwhile
+# https://github.com/KhronosGroup/Vulkan-Loader/blob/1108bba6c97174d172d45470a7470a3d6a564647/docs/LoaderLayerInterface.md#linux-layer-discovery
+if [ -z "$XDG_DATA_DIRS" ]; then
+    export XDG_DATA_DIRS="${CONDA_PREFIX}/share"
+else
+    export XDG_DATA_DIRS="${CONDA_PREFIX}/share:$XDG_DATA_DIRS"
+fi
 
 # Add exe names newline separated to the string to disable LD_PRELOAD
 DISABLE_LD_PRELOAD="cs2.sh

