diff --git a/bin/mangohud.in b/bin/mangohud.in
index 96468bd..d805d5a 100755
--- a/bin/mangohud.in
+++ b/bin/mangohud.in
@@ -32,6 +32,11 @@ command_line="$*"
 # flag for disable_preload
 disable_preload=false
 
+if [ "$1" = "--disable-preload" ]; then
+    disable_preload=true
+    shift
+fi
+
 # Check if the script name or any of the executables in DISABLE_LD_PRELOAD are in the command line
 for exe in $DISABLE_LD_PRELOAD; do
     if echo "$command_line" | grep -q "$exe"; then

