diff --git a/setup.py b/setup.py
index ee7b146..b38378d 100644
--- a/setup.py
+++ b/setup.py
@@ -57,11 +57,15 @@ if {'build_sphinx'}.intersection(sys.argv):
     cmdclass['build_sphinx'] = BuildDoc

 # define extension
-csegments = Extension(
-    'ligo.__segments',
-    ['src/segments.c', 'src/infinity.c', 'src/segment.c', 'src/segmentlist.c'],
-    include_dirs=['src'],
-)
+if sys.platform.startswith('win'):
+    extensions = []
+else:
+    extensions = [
+        Extension('ligo.__segments',
+                  ['src/segments.c', 'src/infinity.c',
+                   'src/segment.c', 'src/segmentlist.c'],
+                  include_dirs=['src']),
+    ]

 # run setup
 setup(name='ligo-segments',
@@ -75,7 +79,7 @@ setup(name='ligo-segments',
       cmdclass=cmdclass,
       setup_requires=setup_requires,
       install_requires=install_requires,
-      ext_modules=[csegments],
+      ext_modules=extensions,
       classifiers=[
           'Development Status :: 5 - Production/Stable',
           'Programming Language :: Python',
