--- src/XF86DGA.c
+++ src/XF86DGA.c
@@ -321,7 +321,6 @@ Bool XF86DGAViewPortChanged(
 #include <stdlib.h>
 #include <stdio.h>
 #include <fcntl.h>
-#include <sys/mman.h>
 #include <sys/wait.h>
 #include <signal.h>
 #include <unistd.h>
--- src/XF86DGA2.c
+++ src/XF86DGA2.c
@@ -737,7 +737,6 @@ void XDGAKeyEventToXKeyEvent(
 #include <stdlib.h>
 #include <stdio.h>
 #include <fcntl.h>
-#include <sys/mman.h>
 #include <sys/wait.h>
 #include <signal.h>
 #include <unistd.h>
