From cdd2fb45dc3e280d6f3834b54d9f7d612b604155 Mon Sep 17 00:00:00 2001
From: Kasper Peeters <kasper.peeters@phi-sci.com>
Date: Mon, 23 Dec 2019 14:59:17 +0000
Subject: [PATCH] Fix linking to system-supplied jsoncpp.

---
 client_server/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/client_server/CMakeLists.txt b/client_server/CMakeLists.txt
index 8356a96f9b..0453a000de 100644
--- a/client_server/CMakeLists.txt
+++ b/client_server/CMakeLists.txt
@@ -184,6 +184,11 @@ if(NOT ENABLE_JUPYTER)
 	 ${GLIBMM3_LIBRARIES}
 	 Threads::Threads
 	 )
+  if(ENABLE_SYSTEM_JSONCPP)
+    target_link_libraries(cadabra_client
+      ${JSONCPP_LIBRARIES}
+      )
+  endif()
   if(WIN32)
     target_link_libraries(cadabra_client ws2_32 bcrypt)
   endif()
