Index: glfw/library.py
<+>UTF-8
===================================================================
--- glfw/library.py	(date 1558453268000)
+++ glfw/library.py	(date 1562409824000)
@@ -127,12 +127,13 @@
     """
     search_paths = [
         '',
+        os.path.abspath(os.path.dirname(__file__)),
+        sys.prefix + '/lib',
         '/usr/lib64',
         '/usr/local/lib64',
         '/usr/lib', '/usr/local/lib',
         '/run/current-system/sw/lib',
-        '/usr/lib/x86_64-linux-gnu/',
-        os.path.abspath(os.path.dirname(__file__))
+        '/usr/lib/x86_64-linux-gnu/'
     ]

     if sys.platform == 'darwin':
