From 9854889e72e61e03f1b3c7be45b048ac9383489c Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 19 Aug 2021 21:49:51 -0600
Subject: [PATCH] Update setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index c01bb14..fbc466e 100644
--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@
       license='MIT License',
       keywords='Web Framework',
       packages=['pylatte','pylatte.database','pylatte.web','pylatte.web.middleware'],
-      install_requires=['ply','mysql-connector-python','pymongo3'],
+      install_requires=['ply','mysql-connector-python','pymongo'],
       download_url='https://github.com/rucifer1217/pylatte',
       long_description="""
       .. image:: http://pylatte.org/image/pylatte.png
