diff --git a/.gitignore b/.gitignore
index 2722579..9c5e82e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 /makeheaders
 /makeheaders.c~
+build
\ No newline at end of file
diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..492bad2
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.0)
+
+add_executable(makeheaders 
+    makeheaders.c
+)
+
+install(TARGETS makeheaders
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+)
\ No newline at end of file
