From 4cba4c5560b4d8e838f7a56fc437377159671a00 Mon Sep 17 00:00:00 2001
From: "UENO, M." <eunosium-1128@live.jp>
Date: Sat, 25 Apr 2026 08:39:00 +0000
Subject: [PATCH 1/5] Include `fmt/ranges.h` for range formatting support

---
 src/Manager.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Manager.cpp b/src/Manager.cpp
index eccdbae..f729f1c 100644
--- a/src/Manager.cpp
+++ b/src/Manager.cpp
@@ -4,6 +4,7 @@
 #include "fmt/format.h"
 #include "kompute/logger/Logger.hpp"
 #include <fmt/core.h>
+#include <fmt/ranges.h>
 #include <iterator>
 #include <set>
 #include <sstream>
-- 
2.53.0

