From 82c2855b56064bef80766ad0eb97f7d98426abfd Mon Sep 17 00:00:00 2001
From: "UENO, M." <eunosium-1128@live.jp>
Date: Sat, 25 Apr 2026 09:44:24 +0000
Subject: [PATCH 4/5] Include `fmt/format.h` for formatting support in
 Tensor.cpp

---
 src/Tensor.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Tensor.cpp b/src/Tensor.cpp
index ad5cac9..7ad4d6f 100644
--- a/src/Tensor.cpp
+++ b/src/Tensor.cpp
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "kompute/Tensor.hpp"
+#include <fmt/format.h>
 
 namespace kp {
 
-- 
2.53.0

