From 0147d49bf1a7ea10e1e6d483ced1f42efedc5a76 Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:41:21 +0100
Subject: [PATCH 2/9] unvendor dependencies

---
 cmake/targets/BuildBun.cmake | 25 +++++++++++++++----------
 1 file changed, 15 insertions(+), 10 deletions(-)

diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index 64536cc26b..969e779aab 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -50,17 +50,7 @@ set(LLVM_ZIG_CODEGEN_THREADS 0)
 
 set(BUN_DEPENDENCIES
   BoringSSL
-  Brotli
-  Cares
-  Highway
-  LibDeflate
-  LolHtml
-  Lshpack
   Mimalloc
-  Zlib
-  LibArchive # must be loaded after zlib
-  HdrHistogram # must be loaded after zlib
-  Zstd
 )
 
 # TinyCC is optional - disabled on Windows ARM64 where it's not supported
@@ -1304,6 +1294,21 @@ endforeach()
 list(TRANSFORM BUN_DEPENDENCIES TOLOWER OUTPUT_VARIABLE BUN_TARGETS)
 add_custom_target(dependencies DEPENDS ${BUN_TARGETS})
 
+target_link_libraries(${bun} PRIVATE
+  brotlidec
+  brotlienc
+  brotlicommon
+  cares
+  hwy
+  deflate
+  lolhtml
+  ls-hpack
+  z
+  archive
+  hdr_histogram
+  zstd
+)
+
 if(APPLE)
   target_link_libraries(${bun} PRIVATE icucore resolv)
   target_compile_definitions(${bun} PRIVATE U_DISABLE_RENAMING=1)
