From bb18a174376066da25b15d4e63b148878831c2c1 Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:42:01 +0100
Subject: [PATCH 3/9] set required BUN_GIT_SHA variable

---
 cmake/tools/GenerateDependencyVersions.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/tools/GenerateDependencyVersions.cmake b/cmake/tools/GenerateDependencyVersions.cmake
index b7942307be..9658a59cfa 100644
--- a/cmake/tools/GenerateDependencyVersions.cmake
+++ b/cmake/tools/GenerateDependencyVersions.cmake
@@ -126,6 +126,7 @@ function(generate_dependency_versions_header)
   if(NOT BUN_GIT_SHA)
     set(BUN_GIT_SHA "unknown")
   endif()
+  set(BUN_GIT_SHA "not-set-in-conda-build")
   list(APPEND DEPENDENCY_VERSIONS "UWS" "${BUN_GIT_SHA}")
   list(APPEND DEPENDENCY_VERSIONS "USOCKETS" "${BUN_GIT_SHA}")
   
