diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index 64536cc26b..991c6abad3 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -1365,6 +1365,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1465,6 +1467,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1488,6 +1492,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1544,6 +1550,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
