From dd901fb5da260223291aff815b7fb03b1fd50b3d Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:42:34 +0100
Subject: [PATCH 5/9] add targets to register_command

---
 cmake/targets/BuildBun.cmake | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index a62d3b8435..4a84ff822b 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -1347,6 +1347,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1447,6 +1449,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1470,6 +1474,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1526,6 +1532,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
