From 1a2ed06bc47a9fc9b3c8ccf383445d19ed61ae8f Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:43:12 +0100
Subject: [PATCH 6/9] don't use static libgcc, libcxx

---
 cmake/targets/BuildBun.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index 4a84ff822b..b3e654cff6 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -1107,8 +1107,8 @@ if(LINUX)
 
   if(NOT ABI STREQUAL "musl")
     target_link_options(${bun} PUBLIC
-      -static-libstdc++
-      -static-libgcc
+      -lstdc++
+      -lgcc
     )
   else()
     target_link_options(${bun} PUBLIC
