diff --git i/cmake/targets/BuildBun.cmake w/cmake/targets/BuildBun.cmake
index 33d6cbd..ea71c9a 100644
--- i/cmake/targets/BuildBun.cmake
+++ w/cmake/targets/BuildBun.cmake
@@ -1106,8 +1106,8 @@ if(LINUX)
 
   if(NOT ABI STREQUAL "musl")
     target_link_options(${bun} PUBLIC
-      -static-libstdc++
-      -static-libgcc
+      -lstdc++
+      -lgcc
     )
   else()
     target_link_options(${bun} PUBLIC
