From e17f1dab172761f945011284a426c4a71acd6a7b Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:46:55 +0100
Subject: [PATCH 8/9] disable -ld_new

---
 cmake/targets/BuildBun.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index b3e654cff6..ea71c9ad01 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -1060,7 +1060,6 @@ endif()
 
 if(APPLE)
   target_link_options(${bun} PUBLIC
-    -Wl,-ld_new
     -Wl,-no_compact_unwind
     -Wl,-stack_size,0x1200000
     -fno-keep-static-consts
