From 7fb47cec233cfabf05bd6a6c99e388247e0c4a2e Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:47:43 +0100
Subject: [PATCH 9/9] don't use llvm toolchain

---
 cmake/tools/SetupLLVM.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/tools/SetupLLVM.cmake b/cmake/tools/SetupLLVM.cmake
index d8849ed6f0..64ac2604d4 100644
--- a/cmake/tools/SetupLLVM.cmake
+++ b/cmake/tools/SetupLLVM.cmake
@@ -1,3 +1,6 @@
+set(CMAKE_STRIP "$ENV{CMAKE_STRIP}" CACHE FILEPATH "Path to strip" FORCE)
+set(LLD_PROGRAM "$ENV{CMAKE_LLD}" CACHE FILEPATH "Path to lld" FORCE)
+return()
 
 set(DEFAULT_ENABLE_LLVM ON)
 
