From 092b939d987891479089689deb08f744a8ec570d Mon Sep 17 00:00:00 2001
From: Uwe Korn <uwelk@xhochy.com>
Date: Mon, 9 Feb 2026 12:19:07 +0000
Subject: [PATCH 14/14] Use system ICU

---
 cmake/targets/BuildBun.cmake | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index 0a2f9b1..ae7566d 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -1288,11 +1288,6 @@ target_link_libraries(${bun} PRIVATE
   zstd
 )
 
-if(APPLE)
-  target_link_libraries(${bun} PRIVATE icucore resolv)
-  target_compile_definitions(${bun} PRIVATE U_DISABLE_RENAMING=1)
-endif()
-
 if(USE_STATIC_SQLITE)
   target_compile_definitions(${bun} PRIVATE LAZY_LOAD_SQLITE=0)
 else()
@@ -1307,12 +1302,12 @@ if(LINUX)
   else()
     target_link_libraries(${bun} PUBLIC libatomic.so)
   endif()
-
-  target_link_libraries(${bun} PRIVATE ${WEBKIT_LIB_PATH}/libicudata.a)
-  target_link_libraries(${bun} PRIVATE ${WEBKIT_LIB_PATH}/libicui18n.a)
-  target_link_libraries(${bun} PRIVATE ${WEBKIT_LIB_PATH}/libicuuc.a)
 endif()
 
+target_link_libraries(${bun} PRIVATE $ENV{PREFIX}/lib/libicudata$ENV{SHLIB_EXT})
+target_link_libraries(${bun} PRIVATE $ENV{PREFIX}/lib/libicui18n$ENV{SHLIB_EXT})
+target_link_libraries(${bun} PRIVATE $ENV{PREFIX}/lib/libicuuc$ENV{SHLIB_EXT})
+
 if(WIN32)
   target_link_libraries(${bun} PRIVATE
     winmm
-- 
2.43.0

