diff --git src/library/plan.cpp src/library/plan.cpp
index 61bbb64..99cf97a 100644
--- src/library/plan.cpp
+++ src/library/plan.cpp
@@ -165,7 +165,7 @@ static bool split1D_for_inplace(size_t num, vector<vector<size_t> > &splitNums,
 
 	num = num / divide_factor;
 	//now the remaining num should have even number of pow2, pow3 and pow5 and we can do sqrt
-	size_t temp = sqrt(num);
+	size_t temp = (size_t)sqrt((double)num);
 	vector<size_t> splitVec;
 	splitVec.push_back(temp*divide_factor);
 	splitVec.push_back(temp);
