diff --git a/Makefile.in b/Makefile.in
index be203e4..ad7d7d3 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -26,8 +26,8 @@ TESTOBJS = test.o testgrid.o testtree.o showchars.o showkey.o
 NDIALOGOBJS = whiptail.o dialogboxes.o
 WHIPTCLOBJS = shared/whiptcl.o shared/dialogboxes.o
 LIBNEWT = libnewt.a
-LIBNEWTSH = libnewt.$(SOEXT).$(VERSION)
-LIBNEWTSONAME = libnewt.$(SOEXT).$(SONAME)
+LIBNEWTSH = libnewt.$(VERSION).dylib
+LIBNEWTSONAME = libnewt.$(SONAME).dylib
 LIBOBJS = newt.o button.o form.o checkbox.o entry.o label.o listbox.o \
           scrollbar.o textbox.o scale.o grid.o windows.o buttonbar.o \
 	  checkboxtree.o
@@ -95,10 +95,10 @@ _snack.$(SOEXT):   snack.c $(LIBNEWTSH)
 		PIFLAGS=`$$pyconfig --includes`; \
 		PLDFLAGS=`$$pyconfig --ldflags`; \
 		PLFLAGS=`$$pyconfig --libs`; \
-		echo $(CC) $(SHCFLAGS) $(CPPFLAGS) $$PIFLAGS $$PCFLAGS -c -o $$ver/snack.o snack.c; \
-		$(CC) $(SHCFLAGS) $(CPPFLAGS) $$PIFLAGS $$PCFLAGS -c -o $$ver/snack.o snack.c; \
-		echo $(CC) --shared $$PLDFLAGS $$PLFLAGS $(LDFLAGS) -o $$ver/_snack.$(SOEXT) $$ver/snack.o -L.  -lnewt $(LIBS); \
-		$(CC) --shared $$PLDFLAGS $$PLFLAGS $(LDFLAGS) -o $$ver/_snack.$(SOEXT) $$ver/snack.o -L.  -lnewt $(LIBS); \
+		echo $(CC) $(SHCFLAGS) $(CPPFLAGS) $$PIFLAGS -c -o $$ver/snack.o snack.c; \
+		$(CC) $(SHCFLAGS) $(CPPFLAGS) $$PIFLAGS -c -o $$ver/snack.o snack.c; \
+		echo $(CC) --shared $$PLFLAGS $(LDFLAGS) -o $$ver/_snack.$(SOEXT) $$ver/snack.o -L.  -lnewt $(LIBS); \
+		$(CC) --shared $$PLFLAGS $(LDFLAGS) -o $$ver/_snack.$(SOEXT) $$ver/snack.o -L.  -lnewt $(LIBS); \
 	done || :
 	touch $@
 
