diff -uN test/gridmap-test.pl test.2/gridmap-test.pl
--- test/gridmap-test.pl	2019-02-26 14:54:28.000000000 +0000
+++ test.2/gridmap-test.pl	2019-09-26 11:50:10.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -uN test/gridmap-tools-test.pl test.2/gridmap-tools-test.pl
--- test/gridmap-tools-test.pl	2019-02-26 14:54:28.000000000 +0000
+++ test.2/gridmap-tools-test.pl	2019-09-26 11:50:12.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 use File::Basename;
diff -uN test/gss-assist-auth-test.pl test.2/gss-assist-auth-test.pl
--- test/gss-assist-auth-test.pl	2019-02-26 14:54:28.000000000 +0000
+++ test.2/gss-assist-auth-test.pl	2019-09-26 11:50:15.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 use POSIX;
diff -uN test/gss-assist-gridmap-test.pl test.2/gss-assist-gridmap-test.pl
--- test/gss-assist-gridmap-test.pl	2019-02-26 14:54:28.000000000 +0000
+++ test.2/gss-assist-gridmap-test.pl	2019-09-26 11:50:17.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -uN test/read-vhost-cred-dir-test.pl test.2/read-vhost-cred-dir-test.pl
--- test/read-vhost-cred-dir-test.pl	2019-02-26 14:54:28.000000000 +0000
+++ test.2/read-vhost-cred-dir-test.pl	2019-09-26 11:50:22.000000000 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 use strict;
 use warnings;
diff -uN programs/grid-mapfile-add-entry.in programs.2/grid-mapfile-add-entry.in
--- programs/grid-mapfile-add-entry.in	2019-09-26 12:20:24.000000000 +0100
+++ programs/grid-mapfile-add-entry.in	2019-09-26 12:21:34.000000000 +0100
@@ -1,4 +1,4 @@
-#! @PERL@
+#!/usr/bin/env perl
 
 # 
 # Copyright 1999-2016 University of Chicago
diff -uN programs/grid-mapfile-check-consistency.in programs.2/grid-mapfile-check-consistency.in
--- programs/grid-mapfile-check-consistency.in	2019-09-26 12:20:39.000000000 +0100
+++ programs/grid-mapfile-check-consistency.in	2019-09-26 12:21:40.000000000 +0100
@@ -1,4 +1,4 @@
-#! @PERL@
+#!/usr/bin/env perl
                             
 # 
 # Copyright 1999-2006 University of Chicago
diff -uN programs/grid-mapfile-delete-entry.in programs.2/grid-mapfile-delete-entry.in
--- programs/grid-mapfile-delete-entry.in	2019-09-26 12:20:47.000000000 +0100
+++ programs/grid-mapfile-delete-entry.in	2019-09-26 12:21:42.000000000 +0100
@@ -1,4 +1,4 @@
-#! @PERL@
+#!/usr/bin/env perl
 
 # 
 # Copyright 1999-2016 University of Chicago
