From 744e2279aef5cf02aaee9c51ddf811b16d84415c Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Wed, 4 Jun 2025 07:34:17 -0700
Subject: [PATCH] Drop pylibmagic and python-magic-bin to allow pip-check

---
 pyproject.toml | 2 --
 1 file changed, 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index ad25e72..f405e2d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -49,8 +49,6 @@ dependencies = [
     "simple-settings",  # for handling settings more easily
     'importlib_resources; python_version < "3.9"',
     'python-magic; platform_system != "Windows"',  # for getting the filetype, non-Windows
-    'pylibmagic; platform_system != "Windows"',  # for shipping the magic library for python-magic
-    'python-magic-bin; platform_system == "Windows"',  # for getting the filetype, Windows-only
     "typing_extensions >=4.0; python_version<'3.11'",
 ]

--
2.45.2

