diff --git a/Cargo.toml b/Cargo.toml
index 4722616..bd541cc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,7 +13,7 @@ crate-type = ["cdylib"]
 # "extension-module" tells pyo3 we want to build an extension module (skips linking against libpython.so)
 # "abi3-py39" tells pyo3 (and maturin) to build using the stable ABI with minimum Python version 3.9
 # keep in sync with requires-python (pyproject.toml)
-pyo3 = { version = "0.23.4", features = ["extension-module", "abi3-py39"] }
+pyo3 = { version = "0.23.4", features = ["extension-module"] }
 numpy = "0.23.0"
 num-traits = "0.2.19"
 either = "1.14.0"
