diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c5c6e9..7e14503 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,7 +91,7 @@ include(ExternalProject)
 ################################################################################
 # Options
 ################################################################################
-option(CUCIM_SUPPORT_GDS "Support cufile library" ON)
+option(CUCIM_SUPPORT_GDS "Support cufile library" OFF)
 option(CUCIM_STATIC_GDS "Use static cufile library" OFF)
 option(CUCIM_SUPPORT_CUDA "Support CUDA" ON)
 option(CUCIM_SUPPORT_NVTX "Support NVTX" ON)
@@ -227,4 +227,4 @@ export(PACKAGE ${CUCIM_PACKAGE_NAME})
 #include(CPack)
 
 # Unset cached options needed
-unset(CUCIM_STATIC_GDS CACHE)
\ No newline at end of file
+unset(CUCIM_STATIC_GDS CACHE)
