diff --git a/setup.py b/setup.py
index f314ebf..2d9501a 100644
--- a/setup.py
+++ b/setup.py
@@ -15,12 +15,8 @@
 cmake_args = ["-DUNITTESTS=OFF"]
 define_macros = []
 
-# Check if windows linux or mac to pass flag
-if system() == 'Windows':
-    cmake_args += ['-G', 'MinGW Makefiles']
-
-else:  # Linux or Mac
-    cmake_args += ['-G', 'Unix Makefiles']
+# use conda-defined cmake generator
+cmake_args += ['-G', os.environ['CMAKE_GENERATOR']]
 
 # Pass Python option to CMake and Python interface compilation
 cmake_args += ['-DPYTHON=ON']
