Index: CMakeLists.txt
===================================================================
diff --git a/GWOLLUM/trunk/CMakeLists.txt b/GWOLLUM/trunk/CMakeLists.txt
--- a/GWOLLUM/trunk/CMakeLists.txt	(revision 82556)
+++ b/GWOLLUM/trunk/CMakeLists.txt	(working copy)
@@ -30,7 +30,7 @@
 find_package(PkgConfig)

 # specify the C++ standard
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)

 # -- set install paths ------
