diff --git i/src/CMakeLists.txt w/src/CMakeLists.txt
index 31b8985..026e8d2 100644
--- i/src/CMakeLists.txt
+++ w/src/CMakeLists.txt
@@ -168,7 +168,7 @@ endif ()
 
 # =================================================================================================
 # DBCSR LIBRARY
-add_library(dbcsr ${DBCSR_SRCS})
+add_library(dbcsr SHARED ${DBCSR_SRCS})
 
 # -fPIC can also be used in the static case. Addresses are resolved during the
 # linking process
@@ -293,7 +293,7 @@ if (WITH_C_API)
   add_fypp_sources(DBCSR_C_SRCS dbcsr.h dbcsr_api_c.F
                    tensors/dbcsr_tensor_api_c.F tensors/dbcsr_tensor.h)
 
-  add_library(dbcsr_c ${DBCSR_C_SRCS})
+  add_library(dbcsr_c SHARED ${DBCSR_C_SRCS})
   set_target_properties(dbcsr_c PROPERTIES LINKER_LANGUAGE Fortran)
 
   set_target_properties(
