--- a/src/gui.rs.orig 2020-07-29 14:00:00.000000000 +0000
+++ b/src/gui.rs 2020-07-29 14:00:00.000000000 +0000
@@ -159,3 +159,13 @@
 }

+impl Preferences for GuiSettingsContainer {
+    fn load(app_info: &app_dirs2::AppInfo, prefs_key: &str) -> Result<Self, preferences::PreferencesError> {
+        preferences::Preferences::load(app_info, prefs_key)
+    }
+
+    fn save(&self, app_info: &app_dirs2::AppInfo, prefs_key: &str) -> Result<(), preferences::PreferencesError> {
+        preferences::Preferences::save(self, app_info, prefs_key)
+    }
+}
+
 pub fn load_gui_settings() -> GuiSettingsContainer {
