--- a/Cargo.toml.orig	2020-07-29 14:00:00.000000000 +0000
+++ b/Cargo.toml	2020-07-29 14:00:00.000000000 +0000
@@ -15,3 +15,3 @@
 egui_plot = "0.29.0"
-egui-phosphor = { git = "https://github.com/hacknus/egui-phosphor" }
+egui-phosphor = "0.7.3"
 egui-theme-switch = { version = "0.2.0", default-features = true }
@@ -20,4 +20,4 @@
 itertools-num = "0.1"
-keepawake = { git = "https://github.com/segevfiner/keepawake-rs" }
-preferences = { git = "https://github.com/andybarron/preferences-rs" }
+keepawake = "0.5.1"
+preferences = "2.0.0"
 rand = "0.8.5"
@@ -28,3 +28,3 @@
 serde = { version = "1.0", features = ["derive"] }
-serialport = { git = "https://github.com/serialport/serialport-rs", features = ["serde"] }
+serialport = { version = "4.6.1", features = ["serde"] }

