--- CMakeLists.txt.orig      2024-07-05 14:24:49.031916858 -0500
+++ CMakeLists.txt           2024-07-05 14:25:32.952609960 -0500
@@ -73,2 +73,25 @@
        add_library(${PROJECT_NAME} SHARED ${MULTI_SOURCES})
+       set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+       if (WIN32)
+         set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME "${PROJECT_NAME}.${PROJECT_VERSION_MAJOR}")
+         set(MAJOR_DLL_PATH "${CMAKE_CURRENT_BINARY_DIR}/bin/${PROJECT_NAME}.${PROJECT_VERSION_MAJOR}")
+         add_custom_command(
+           TARGET ${PROJECT_NAME} POST_BUILD
+           COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                   "${MAJOR_DLL_PATH}.dll"
+                   "${MAJOR_DLL_PATH}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+           COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                   "${MAJOR_DLL_PATH}.dll"
+                   "${CMAKE_CURRENT_BINARY_DIR}/bin/${PROJECT_NAME}.dll"
+           COMMENT "Copying ${LIB_NAME_C}.${PROJECT_VERSION_MAJOR}.dll to versioned filenames"
+         )
+         add_custom_target(CopyVersionedDLL ALL
+           DEPENDS ${PROJECT_NAME}
+         )
+         install(FILES "${CMAKE_CURRENT_BINARY_DIR}/bin/${PROJECT_NAME}.dll"
+                       "${MAJOR_DLL_PATH}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+                 DESTINATION bin
+                 COMPONENT Runtime
+         )
+       endif ()
 else ()
