--- cmake/GazeboUtils.cmake
+++ cmake/GazeboUtils.cmake
@@ -142,11 +142,9 @@
     # Don't pull in the Windows min/max macros
     add_definitions(-DNOMINMAX)
 
-    #use static libraries for FREEIMAGE
-    add_definitions(-DFREEIMAGE_LIB)
-
     # Use dynamic linking for boost
     add_definitions(-DBOOST_ALL_DYN_LINK)
+    add_definitions(-DBOOST_IOSTREAMS_NO_LIB=1)
 
     # Use dynamic linking for protobuf
     add_definitions(-DPROTOBUF_USE_DLLS)
