--- test/all-in-place.test
+++ test/all-in-place.test (Unsaved)
@@ -339,7 +339,6 @@
   sunicontopnm \
   svgtopam \
   tgatoppm \
-  thinkjettopbm \
   tifftopnm \
   wbmptopbm \
   winicontopam \
@@ -422,7 +422,7 @@
 # These two programs write a usage message on standout in this case.
 # Exit status should be 1.
 
-for i in pamstretch-gen pcdovtoppm
+for i in pamstretch-gen
   do
      $i > /dev/null
      testExitStatus $i 1 $?
--- test/all-in-place.ok
+++ test/all-in-place.ok (Unsaved)
@@ -300,7 +300,6 @@
 sunicontopnm: ok
 svgtopam: ok
 tgatoppm: ok
-thinkjettopbm: ok
 tifftopnm: ok
 wbmptopbm: ok
 winicontopam: ok
@@ -326,4 +326,3 @@
 ppmrainbow: ok
 ppmfade: ok
 pamstretch-gen: ok
-pcdovtoppm: ok

--- test/Test-Order
+++ test/Test-Order (Unsaved)
@@ -133,8 +133,6 @@
 pict-roundtrip.test
 png-roundtrip.test
 png-roundtrip2.test
-ps-roundtrip.test
-ps-alt-roundtrip.test
 sgi-roundtrip.test
 sbig-roundtrip.test
 st4-roundtrip.test

--- test/legacy-names.ok
+++ test/legacy-names.ok (Unsaved)
@@ -25,7 +25,6 @@
 ppmnorm: ok
 ppmtogif: ok
 ppmtojpeg: ok
-ppmtompeg: ok
 ppmtotga: ok
 ppmtouil: ok
 pgmcrater: ok

--- test/legacy-names.test
+++ test/legacy-names.test (Unsaved)
@@ -76,7 +76,6 @@
   ppmnorm \
   ppmtogif \
   ppmtojpeg \
-  ppmtompeg \
   ppmtotga \
   ppmtouil \
 "
