diff --git a/setup.py b/setup.py
index 53ced3c..fc1880f 100755
--- a/setup.py
+++ b/setup.py
@@ -50,13 +50,7 @@ def find_version(*file_paths):
 # platform-dependent helpers
 #
 def is_manylinux():
-    try:
-       for line in open('/etc/redhat-release').readlines():
-           if 'CentOS release 5.11' in line:
-               return True
-    except (OSError, IOError):
-        pass
-    return False
+    return True
 
 
 #
