--- a/python/imas_core/__init__.py
+++ b/python/imas_core/__init__.py
@@ -1,11 +1,5 @@
 """Access Layer Core Python bindings."""

-import os
-import pathlib
-
-if os.name == "nt":  # pragma: no cover
-    os.add_dll_directory(pathlib.Path(__file__).parents[1] / "imas_core.libs")
-
 from . import _al_lowlevel
 from . import imasdef

@@ -13,5 +7,3 @@
 from ._version import version_tuple

 __all__ = ["_al_lowlevel", "imasdef", "__version__", "version_tuple"]
-
-del os, pathlib
