--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,7 +74,7 @@
 # ##############################################################################

 if(WIN32)
-  find_package(PThreads4W REQUIRED)
+  find_package(Threads REQUIRED)
   find_package(dlfcn-win32 CONFIG REQUIRED)
 endif()

@@ -123,7 +123,7 @@
 if(NOT WIN32)
   target_link_libraries(al PRIVATE dl)
 else()
-  target_link_libraries(al PRIVATE dlfcn-win32::dl PThreads4W::PThreads4W ws2_32)
+  target_link_libraries(al PRIVATE dlfcn-win32::dl Threads::Threads ws2_32)
 endif()

 add_subdirectory(src)
