--- a/src/hdf5/CMakeLists.txt
+++ b/src/hdf5/CMakeLists.txt
@@ -20,7 +20,7 @@ target_sources( al PRIVATE
     hdf5_events_handler.cpp
     hdf5_backend_factory.cpp
 )
-target_compile_definitions( al PRIVATE -DHDF5 )
+target_compile_definitions( al PRIVATE -DHDF5 -DH5_BUILT_AS_DYNAMIC_LIB )
 if(WIN32)
 	# Use modern CMake targets if available (vcpkg), otherwise legacy variables
 	if( TARGET hdf5::hdf5-shared )
