--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,15 +26,8 @@
 Repository = "https://github.com/iterorganization/IMAS-Core"

 [build-system]
-requires = [
-    "scikit-build-core>=0.9.3",
-    "cython>=3.0.3",
-    "cython-cmake>=0.2.0",
-    "numpy",
-    "setuptools>=64",
-    "setuptools_scm>=8",
-]
-build-backend = "scikit_build_core.build"
+build-backend = "mesonpy"
+requires = ["meson-python"]

 [tool.cibuildwheel]
 # build-verbosity = 1
