--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,15 +26,8 @@ Homepage = "https://imas.iter.org"
 Repository = "https://github.com/iterorganization/IMAS-Core"

 [build-system]
-requires = [
-    "scikit-build-core>=0.9.3",
-    "cython>=3.0.3",
-    "cython-cmake>=0.2.0",
-    "numpy",
-    "setuptools>=64",
-    "setuptools_scm>=8",
-]
-build-backend = "scikit_build_core.build"
+build-backend = "mesonpy"
+requires = ["meson-python"]

 [tool.cibuildwheel]
 # build-verbosity = 1
@@ -69,7 +62,7 @@ MACOSX_DEPLOYMENT_TARGET = "14.0"
 [tool.cibuildwheel.macos]
 archs = ["arm64"]
 repair-wheel-command = "delocate-wheel -w {dest_dir} {wheel}"
-
+
 [tool.scikit-build.cmake.define]
 BUILD_SHARED_LIBS = "ON"
 DOCS_ONLY = { env = "DOCS_ONLY", default = "OFF" }
