--- a/src/uda/CMakeLists.txt
+++ b/src/uda/CMakeLists.txt
@@ -3,9 +3,9 @@
 # Use pkg config to find uda requirements
 find_package( PkgConfig REQUIRED )
 if( IMAS_UDAFAT )
-    pkg_check_modules( uda REQUIRED IMPORTED_TARGET uda-fat-cpp )
+    pkg_check_modules( uda REQUIRED IMPORTED_TARGET uda-fat-cpp uda-fat-client )
 else()
-    pkg_check_modules( uda REQUIRED IMPORTED_TARGET uda-cpp )
+    pkg_check_modules( uda REQUIRED IMPORTED_TARGET uda-cpp uda-client )
 endif()
 # if( IMAS_UDAFAT )
 #   find_package( PostgreSQL REQUIRED )
