--- a/src/uda/CMakeLists.txt
+++ b/src/uda/CMakeLists.txt
@@ -4,8 +4,10 @@
 find_package( PkgConfig REQUIRED )
 if( IMAS_UDAFAT )
     pkg_check_modules( uda REQUIRED IMPORTED_TARGET uda-fat-cpp )
+    pkg_check_modules( uda_client REQUIRED IMPORTED_TARGET uda-fat-client )
 else()
     pkg_check_modules( uda REQUIRED IMPORTED_TARGET uda-cpp )
+    pkg_check_modules( uda_client REQUIRED IMPORTED_TARGET uda-client )
 endif()
 # if( IMAS_UDAFAT )
 #   find_package( PostgreSQL REQUIRED )
@@ -26,4 +28,4 @@
    target_compile_definitions( al PRIVATE -DUDA )
 endif()

-target_link_libraries( al PRIVATE PkgConfig::uda )
+target_link_libraries( al PRIVATE PkgConfig::uda uda_client )
