From 8af4d1c2a557a22cb81c99b6cd2a30873cb6f4ef Mon Sep 17 00:00:00 2001
From: Julien Malik <julien.malik@unseenlabs.fr>
Date: Mon, 23 May 2022 22:10:58 +0200
Subject: [PATCH 1/2] use pypandoc.convert_file

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 55bd8bd..0438e2c 100644
--- a/setup.py
+++ b/setup.py
@@ -5,7 +5,7 @@ with open('numpy_ringbuffer/__about__.py') as f:
 
 try:
     import pypandoc
-    long_description = pypandoc.convert('README.md', 'rst')
+    long_description = pypandoc.convert_file('README.md', 'rst')
 except (IOError, ImportError):
     print("No")
     long_description = open('README.md').read()
-- 
2.25.1

