From c92c59bb3d9837a352c6c326611e30fd3f7185f3 Mon Sep 17 00:00:00 2001
From: Julien Malik <julien.malik@unseenlabs.fr>
Date: Mon, 23 May 2022 22:12:51 +0200
Subject: [PATCH 2/2] fix from collections import Sequence

---
 numpy_ringbuffer/__init__.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/numpy_ringbuffer/__init__.py b/numpy_ringbuffer/__init__.py
index 719a29b..2ec8284 100644
--- a/numpy_ringbuffer/__init__.py
+++ b/numpy_ringbuffer/__init__.py
@@ -1,5 +1,10 @@
 import numpy as np
-from collections import Sequence
+try:
+    # Python <=3.9
+    from collections import Sequence
+except ImportError:
+    # Python > 3.8
+    from collections.abc import Sequence
 
 from .__about__ import *
 
-- 
2.25.1

