diff --git src/corelib/global/qglobal.cpp src/corelib/global/qglobal.cpp
index 5b63952..471c057 100644
--- src/corelib/global/qglobal.cpp
+++ src/corelib/global/qglobal.cpp
@@ -1850,9 +1850,7 @@ QSysInfo::WinVersion QSysInfo::windowsVersion()
         } else if (osver.dwMajorVersion == 10 && osver.dwMinorVersion == 0) {
             winver = QSysInfo::WV_WINDOWS10;
         } else {
-            qWarning("Qt: Untested Windows version %d.%d detected!",
-                     int(osver.dwMajorVersion), int(osver.dwMinorVersion));
-            winver = QSysInfo::WV_NT_based;
+            winver = QSysInfo::WV_WINDOWS10;
         }
     }
 
