diff --git src/3rdparty/webkit/Source/JavaScriptCore/os-win32/stdint.h src/3rdparty/webkit/Source/JavaScriptCore/os-win32/stdint.h
index b5dff56..500a6e1 100644
--- src/3rdparty/webkit/Source/JavaScriptCore/os-win32/stdint.h
+++ src/3rdparty/webkit/Source/JavaScriptCore/os-win32/stdint.h
@@ -30,6 +30,9 @@
 #error "This stdint.h file should only be compiled with MSVC"
 #endif
 
+// This behavior is fixed in VS 2015.  Now defining these raises errors in compilation.
+#if _MSC_VER < 1900
+
 #include <limits.h>
 
 typedef unsigned char uint8_t;
@@ -65,3 +68,5 @@ CASSERT(sizeof(int64_t) == 8, int64_t_is_eight_bytes)
 CASSERT(sizeof(uint64_t) == 8, uint64_t_is_eight_bytes)
 
 #endif
+
+#endif // VS version < 1900
