--- setup.py	2017-10-12 14:07:19.168340749 +0200
+++ setup.py-fixed	2017-10-12 14:07:07.179340528 +0200
@@ -55,17 +55,17 @@
     name='parcel',
     version='0.1.13',
     packages=["parcel"],
-    cmdclass={
-        'install': ParcelInstall,
-        'develop': ParcelDevelop,
-    },
+    #cmdclass={
+    #    'install': ParcelInstall,
+    #    'develop': ParcelDevelop,
+    #},
     install_requires=[
-        'requests==2.5.1',
-        'progressbar==2.3',
-        'Flask==0.10.1',
-        'intervaltree==2.0.4',
-        'termcolor==1.1.0',
-        'cmd2==0.6.8',
+        'requests',
+        'progressbar',
+        'Flask',
+        'intervaltree',
+        'termcolor',
+        'cmd2',
     ],
     package_data={
         "parcel": [
@@ -78,5 +78,5 @@
         'bin/parcel-tcp2udt',
         'bin/parcel-udt2tcp',
     ],
-    **extra_args
+    #**extra_args
 )
