From d40e571e27de7bc07f5cccc38711ec5786b0a035 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan.janssen@outlook.com>
Date: Tue, 15 Feb 2022 08:27:38 -0700
Subject: [PATCH] Fix collections.abc

---
 basisgen/algebras.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/basisgen/algebras.py b/basisgen/algebras.py
index 917a5e8..6d0fe23 100644
--- a/basisgen/algebras.py
+++ b/basisgen/algebras.py
@@ -1,7 +1,7 @@
 from basisgen.weights import Weight
 
 import abc
-import collections
+import collections.abc
 import enum
 import functools
 import itertools
@@ -335,7 +335,7 @@ def level_vector(self):
             return [10, 6]
 
 
-class SemisimpleAlgebra(collections.Iterable, Algebra):
+class SemisimpleAlgebra(collections.abc.Iterable, Algebra):
     def __init__(self, simple_algebras):
         self.simple_algebras = simple_algebras
         self.rank = sum(
