From 6446a4594456b8b8706a53beb058c8ba90260fca Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 15 Feb 2022 08:13:39 -0700
Subject: [PATCH] Update algebras.py

---
 basisgen/algebras.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/basisgen/algebras.py b/basisgen/algebras.py
index 917a5e8..cd331a6 100644
--- a/basisgen/algebras.py
+++ b/basisgen/algebras.py
@@ -335,7 +335,7 @@ def level_vector(self):
             return [10, 6]
 
 
-class SemisimpleAlgebra(collections.Iterable, Algebra):
+class SemisimpleAlgebra(collections.abc.Iterable, Algebra):
     def __init__(self, simple_algebras):
         self.simple_algebras = simple_algebras
         self.rank = sum(
