diff --git a/test-suite/Makefile.in b/test-suite/Makefile.in
index 45c6f29..c38e15c 100644
--- a/test-suite/Makefile.in
+++ b/test-suite/Makefile.in
@@ -1692,7 +1692,6 @@ SCM_TESTS = tests/00-initial-env.test		\
 	    tests/popen.test			\
 	    tests/popen-child.scm		\
 	    tests/ports.test			\
-	    tests/posix.test			\
 	    tests/q.test			\
 	    tests/r4rs.test			\
 	    tests/r5rs_pitfall.test		\
