diff --git a/test-suite/tests/posix.test b/test-suite/tests/posix.test
index 4dadd77..54f7350 100644
--- a/test-suite/tests/posix.test
+++ b/test-suite/tests/posix.test
@@ -227,15 +227,15 @@
 ;; crypt
 ;;
 
-(with-test-prefix "crypt"
-
-  (pass-if "basic usage"
-    (string? (crypt "pass" "abcdefg")))
-
-  (pass-if-exception "glibc EINVAL" exception:system-error
-    ;; This used to deadlock while trying to throw to 'system-error'.
-    ;; This test uses the special interpretation of the salt that glibc
-    ;; does; specifically, we pass a syntactically invalid salt here.
-    (if (string-contains %host-type "-gnu")
-        (crypt "pass" "$X$abc")                   ;EINVAL
-        (throw 'unresolved))))
+;; (with-test-prefix "crypt"
+
+;;   (pass-if "basic usage"
+;;     (string? (crypt "pass" "abcdefg")))
+
+;;   (pass-if-exception "glibc EINVAL" exception:system-error
+;;     ;; This used to deadlock while trying to throw to 'system-error'.
+;;     ;; This test uses the special interpretation of the salt that glibc
+;;     ;; does; specifically, we pass a syntactically invalid salt here.
+;;     (if (string-contains %host-type "-gnu")
+;;         (crypt "pass" "$X$abc")                   ;EINVAL
+;;         (throw 'unresolved))))
