diff --git a/Makefile b/Makefile
index 4c12c16..0bffa44 100644
--- a/Makefile
+++ b/Makefile
@@ -7,6 +7,8 @@
 LARGE	= $(shell getconf LFS_CFLAGS)
 CFLAGS	= $(RPM_OPT_FLAGS) -pipe -Wall -D_GNU_SOURCE -D_REENTRANT -D_DEFAULT_SOURCE $(LARGE)
 CC	= gcc
+RANLIB  = ranlib
+AR      = ar
 MAJOR	= 1
 MINOR	= 09
 VERSION	= $(MAJOR).$(MINOR)
@@ -89,14 +91,14 @@ obs/unlzw.o: unlzw.c lzw.h
 	$(CC) $(CFLAGS) -fPIC -o $@ -c $<
 
 libzio.a: obj/zio.o obj/unlzw.o
-	ar -rv $@ $^
-	ranlib $@
+	$(AR) -rv $@ $^
+	$(RANLIB) $@
 
 libzio.so.$(VERSION): obs/zio.o obs/unlzw.o
-	gcc -shared -Wl,-soname,$(SONAME),-stats $(LDMAP) -o $@ $^ $(LINK)
+	$(CC) -shared -Wl,-soname,$(SONAME),-stats $(LDMAP) -o $@ $^ $(LINK)
 
-zioP.h: /usr/include/bzlib.h /usr/include/zlib.h
-zio.h:  zio.h.in /usr/include/stdio.h
+zioP.h: $(prefix)/include/bzlib.h $(prefix)/include/zlib.h
+zio.h:  zio.h.in
 	sed 's/@@VERSION@@/$(VERSION)/' < $< > $@
 fzopen.3: fzopen.3.in
 	sed 's/@@VERSION@@/$(VERSION)/' < $< > $@
