From 35ca57833ae96d3e8c240e6d083fe18911dbca55 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Thu, 9 May 2019 17:48:41 +0200
Subject: [PATCH] Fix linking for macOS

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4717cf3..9e13300 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,6 +41,7 @@ ${SMPT_SOURCE_DIR}/src/cropoptimisationengine.cc
 ${SMPT_SOURCE_DIR}/src/cropcutensemble.cc
 ${SMPT_SOURCE_DIR}/src/cropdatastore.cc
 )
+target_link_libraries(cropsimpletools ${ROOT_LIBRARIES})
 
 # List executables and link each of them to the local and ROOT libraries
 set(EXELIST thresholdeffs bwdiv crop stackergen rangefinder sepper mergevars updatedatastore stacker varstocuts eff corr cuttester columnmaker multicolumnmaker tuplesampler tuplescrambler cutapplier)
-- 
2.21.0

