diff --git a/src/tools/impl/tinytex-info.ts b/src/tools/impl/tinytex-info.ts
index 2523a6b40..d31ab03ef 100644
--- a/src/tools/impl/tinytex-info.ts
+++ b/src/tools/impl/tinytex-info.ts
@@ -10,6 +10,15 @@ import { join } from "path/mod.ts";
 import { getenv } from "../../core/env.ts";
 
 export function tinyTexInstallDir(): string | undefined {
+  const condaPrefix = getenv("CONDA_PREFIX");
+  if (condaPrefix) {
+    switch (Deno.build.os) {
+      case "windows":
+        return join(condaPrefix, "Library");
+      default:
+        return condaPrefix;
+    }
+  }
   switch (Deno.build.os) {
     case "windows":
       return expandPath(join(getenv("APPDATA", undefined), "TinyTeX"));
