diff --git a/test_setup.py b/test_setup.py
new file mode 100644
index 0000000..1232460
--- test_setup.py
+++ b/test_setup.py
@@ -0,0 +1,22 @@
+from distutils import ccompiler
+import tempfile
+
+test_src = tempfile.mktemp(suffix='.c') 
+
+content = """
+#include <stdint.h>
+#include <inttypes.h>
+#ifndef  INT64_MIN
+#error "not defined"
+#endif
+void main() {
+    int64_t i;
+}
+"""
+
+with open(test_src, 'w') as f:
+    f.write(content)
+
+compiler = ccompiler.new_compiler()
+compiler.compile([test_src])
+print("compile succeeded")
\ No newline at end of file
