diff --git configure configure
index e3d464b..b1c6025 100755
--- configure
+++ configure
@@ -454,7 +454,7 @@ if [ "$COMMERCIAL_USER" = "ask" ]; then
         echo "Type 'c' if you want to use the Commercial Edition."
         echo "Type 'o' if you want to use the Open Source Edition."
         echo
-        read commercial
+        commercial=o
         echo
         if [ "$commercial" = "c" ]; then
             COMMERCIAL_USER="yes"
@@ -4597,7 +4597,7 @@ elif [ "$Edition" = "OpenSource" ]; then
             echo "Type 'no' to decline this license offer."
             echo
             echo $ECHO_N "Do you accept the terms of $affix license? $ECHO_C"
-            read acceptance
+            acceptance=yes
         fi
         echo
         if [ "$acceptance" = "yes" ] || [ "$acceptance" = "y" ]; then
@@ -4628,7 +4628,7 @@ elif [ "$Edition" = "Preview" ]; then
             echo "Type 'no' to decline this license offer."
             echo
             echo $ECHO_N "Do you accept the terms of the license? $ECHO_C"
-            read acceptance
+            acceptance=yes
         fi
         echo
         if [ "$acceptance" = "yes" ]; then
@@ -4714,7 +4714,7 @@ elif [ "$Edition" != "OpenSource" ]; then
             echo "Type 'no' to decline this license offer."
             echo
             echo $ECHO_N "Do you accept the terms of the $TheLicense? $ECHO_C"
-            read acceptance
+            acceptance=yes
         fi
         echo
         if [ "$acceptance" = "yes" ]; then
