diff --git a/qirlib/build.rs b/qirlib/build.rs
index 732c5b3..fb7dd26 100644
--- a/qirlib/build.rs
+++ b/qirlib/build.rs
@@ -270,7 +270,7 @@ fn compile_llvm_wrapper() -> Result<(), Box<dyn Error>> {
     }
     cfg.cpp(true)
         .cpp_link_stdlib(None)
-        .static_crt(true)
+        .static_crt(false)
         .file("llvm-wrapper/MetadataWrapper.cpp")
         .file("llvm-wrapper/ModuleWrapper.cpp")
         .prefer_clang_cl_over_msvc(true)
