diff --git a/src/libmodplug/sndfile.h b/src/libmodplug/sndfile.h
index a4d0232..bfb8c38 100644
--- a/src/libmodplug/sndfile.h
+++ b/src/libmodplug/sndfile.h
@@ -459,7 +459,7 @@ typedef struct _MODCOMMAND
 class MODPLUG_EXPORT IMixPlugin
 {
 public:
-	virtual ~IMixPlugin();
+	virtual ~IMixPlugin() {};
 	virtual int AddRef() = 0;
 	virtual int Release() = 0;
 	virtual void SaveAllParameters() = 0;
@@ -539,14 +539,25 @@ class MODPLUG_EXPORT CSoundFile
 //==============
 {
 public:	// Static Members
-	static UINT m_nXBassDepth, m_nXBassRange;
-	static UINT m_nReverbDepth, m_nReverbDelay, gnReverbType;
-	static UINT m_nProLogicDepth, m_nProLogicDelay;
+	static UINT m_nXBassDepth;
+	static UINT m_nXBassRange;
+	static UINT m_nReverbDepth;
+	static UINT m_nReverbDelay;
+	static UINT gnReverbType;
+	static UINT m_nProLogicDepth;
+	static UINT m_nProLogicDelay;
 	static UINT m_nStereoSeparation;
 	static UINT m_nMaxMixChannels;
 	static LONG m_nStreamVolume;
-	static DWORD gdwSysInfo, gdwSoundSetup, gdwMixingFreq, gnBitsPerSample, gnChannels;
-	static UINT gnAGC, gnVolumeRampSamples, gnVUMeter, gnCPUUsage;
+	static DWORD gdwSysInfo;
+	static DWORD gdwSoundSetup;
+	static DWORD gdwMixingFreq;
+	static DWORD gnBitsPerSample;
+	static DWORD gnChannels;
+	static UINT gnAGC;
+	static UINT gnVolumeRampSamples;
+	static UINT gnVUMeter;
+	static UINT gnCPUUsage;
 	static LPSNDMIXHOOKPROC gpSndMixHook;
 	static PMIXPLUGINCREATEPROC gpMixPluginCreateProc;
 
