diff --git a/CMakeLists.txt b/CMakeLists.txt
index 92231ae..ecac36f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,12 +5,15 @@ cmake_minimum_required(VERSION 3.12)
 set( CMAKE_EXPORT_COMPILE_COMMANDS 1 )
 
 # Comment out the next line when speed is needed
-set(CMAKE_BUILD_TYPE Debug)
-add_definitions(-O2)
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Debug)
+endif()
 
 # This next compile flag is needed to avoid a compile error on some systems. See
 # https://pybind11.readthedocs.io/en/stable/faq.html#recursive-template-instantiation-exceeded-maximum-depth-of-256
-add_definitions(-ftemplate-depth=1024)
+if(NOT MSVC)
+  add_compile_options(-O2 -ftemplate-depth=1024)
+endif()
 
 #set(CMAKE_MODULE_PATH cmake)
 include(cmake/knownHosts.cmake)
