diff --git a/setup.py b/setup.py
index e61a08b..26f1940 100644
--- a/setup.py
+++ b/setup.py
@@ -6,6 +6,7 @@
 # See also https://www.benjack.io/2018/02/02/python-cpp-revisited.html
 
 import os
+import platform
 import sys
 import subprocess
 from shutil import copyfile, copymode # for doctest unitTests
@@ -109,7 +110,10 @@ class CMakeBuild(build_ext):
         #    os.path.abspath(__file__)), 'tests')
 
         # Copy standalone executable so it can be installed:
-        src_file = os.path.join(self.build_temp, 'xbooz_xform')
+        exe_name = 'xbooz_xform'
+        if platform.system() == 'Windows':
+            exe_name += '.exe'
+        src_file = os.path.join(self.build_temp, exe_name)
         dest_dir = os.path.join(os.path.dirname(
             os.path.abspath(__file__)))
         dest_file = os.path.join(dest_dir, os.path.basename(src_file))
@@ -139,7 +143,10 @@ setup(
     cmdclass={"build_ext": CMakeBuild},
     install_requires=['numpy'],
     # The next line makes the executable available:
-    data_files = [('bin', ['xbooz_xform'])],
+    data_files = [('bin', [
+        'xbooz_xform.exe' if platform.system() == 'Windows'
+        else 'xbooz_xform'
+    ])],
     test_suite='tests',
     zip_safe=False,
 )
