diff --git a/requirements.txt b/requirements.txt
index e69de29..140249a 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -0,0 +1,11 @@
+rastervision_pipeline
+numpy<1.22
+shapely==1.6.*
+pillow==8.3.*
+pyproj==2.6.*
+rasterio==1.2.6
+scikit-learn==0.24.*
+imageio==2.3.*
+pystac==0.5.6
+mask-to-polygons==0.0.2
+opencv-python
\ No newline at end of file
diff --git a/setup.py b/setup.py
index 6bcd880..3b8dd4b 100644
--- a/setup.py
+++ b/setup.py
@@ -31,4 +31,10 @@ setup(
     'raster deep-learning ml computer-vision earth-observation geospatial geospatial-processing',
     packages=find_namespace_packages(exclude=['integration_tests*', 'tests*']),
     install_requires=install_requires,
-    zip_safe=False)
+    zip_safe=False,
+    entry_points = {
+        'console_scripts': [
+            'rastervision=rastervision.core.cli:predict',
+        ],
+    },
+)
