--- qemu/linux-user/elfload.c
+++ qemu/linux-user/elfload.c
@@ -121,2 +121,5 @@

+#ifndef MAP_FIXED_NOREPLACE
+#define MAP_FIXED_NOREPLACE 0x100000       /* MAP_FIXED but do not unmap */
+#endif
 /* this flag is uneffective under linux too, should be deleted */
--- qemu/linux-user/mmap.c
+++ qemu/linux-user/mmap.c
@@ -19,2 +19,3 @@
 #include "qemu/osdep.h"
+#include <linux/mman.h>
 #include "trace.h"
@@ -30,2 +31,5 @@

+#ifndef MAP_FIXED_NOREPLACE
+#define MAP_FIXED_NOREPLACE 0x100000       /* MAP_FIXED but do not unmap */
+#endif
 #ifdef TARGET_ARM
--- qemu/linux-user/strace.c
+++ qemu/linux-user/strace.c
@@ -14,3 +14,6 @@
 #include <linux/in6.h>
 #include <linux/netlink.h>
+#ifndef NETLINK_SMC
+#define NETLINK_SMC 22
+#endif
 #include <sched.h>
--- qemu/linux-user/syscall.c
+++ qemu/linux-user/syscall.c
@@ -86,2 +86,11 @@
 #endif
+#ifndef SOL_ALG
+#define SOL_ALG 279
+#endif
+#ifndef MAP_FIXED_NOREPLACE
+#define MAP_FIXED_NOREPLACE 0x100000       /* MAP_FIXED but do not unmap */
+#endif
+#ifndef MAP_SHARED_VALIDATE
+#define MAP_SHARED_VALIDATE 0x3
+#endif
