--- qemu/Makefile.orig	2024-07-13 15:31:51.826834383 -0500
+++ qemu/Makefile	2024-07-13 16:17:44.996484934 -0500
@@ -131,1 +131,1 @@
-	$(MESON) introspect --targets --tests --benchmarks | $(PYTHON) scripts/mtest2make.py > $@
+	$(MESON) introspect --targets --tests --benchmarks | $(PYTHON_WIN) scripts/mtest2make.py > $@
@@ -131,1 +131,1 @@
-	$(MESON) introspect --buildoptions $(SRC_PATH)/meson.build | $(PYTHON) \
+	$(MESON) introspect --buildoptions $(SRC_PATH)/meson.build | $(PYTHON_WIN) \
