--- scripts/tracetool/__init__.py	2024-02-02 16:07:43.000000000 +0000
+++ scripts/tracetool/__init__.py	2024-02-02 16:07:43.000000000 +0000
@@ -13,6 +13,7 @@

 import re
+import os.path
 import sys
 import weakref

@@ -311,6 +312,7 @@
         self.fmt = fmt
         self.args = args
         self.lineno = int(lineno)
+        filename = os.path.normpath(str(filename)).replace('\\', '/')
         self.filename = str(filename)
         self.event_trans = event_trans
         self.event_exec = event_exec
