From 6772ac31aa5e9be4008a92ecd82c6518d0f745a5 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Thu, 11 Apr 2024 15:13:46 -0500
Subject: [PATCH] add requirements

---
 requirements.txt | 7 +++++++
 1 file changed, 7 insertions(+)
 create mode 100644 requirements.txt

diff --git requirements.txt requirements.txt
new file mode 100644
index 0000000..622767f
--- /dev/null
+++ requirements.txt
@@ -0,0 +1,7 @@
+numpy
+monty
+pymatgen
+vise
+tabulate
+adjustText
+matplotlib-label-lines
