From 10b6e3e79301ce4177c38cb513f647b5fa73e152 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 3 May 2024 13:19:08 +0200
Subject: [PATCH] Remove pytest from requirements

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 37f0376..0d4b4e6 100755
--- a/setup.py
+++ b/setup.py
@@ -33,7 +33,6 @@ PACKAGES = [
 SCRIPTS = []
 
 CORE_REQUIREMENTS = [
-    'pytest',
 ]
 
 # From http://pypi.python.org/pypi?%3Aaction=list_classifiers
