diff -Naur bin/gendesc bin/gendesc
--- bin/gendesc	2018-12-10 18:40:35.387466225 +0000
+++ bin/gendesc	2018-12-10 18:43:01.529759332 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 #
 #   Copyright (c) International Business Machines  Corp., 2002
 #
@@ -35,6 +35,7 @@
 #   2002-09-02: created by Peter Oberparleiter <Peter.Oberparleiter@de.ibm.com>
 #
 
+use warnings;
 use strict;
 use File::Basename; 
 use Getopt::Long;
diff -Naur bin/genhtml bin/genhtml
--- bin/genhtml	2018-12-10 18:40:35.387466225 +0000
+++ bin/genhtml	2018-12-10 18:44:22.854910854 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 #
 #   Copyright (c) International Business Machines  Corp., 2002,2012
 #
@@ -64,6 +64,7 @@
 #                introduced sorting option (enabled per default)
 #
 
+use warnings;
 use strict;
 use File::Basename;
 use File::Temp qw(tempfile);
diff -Naur bin/geninfo bin/geninfo
--- bin/geninfo	2018-12-10 18:40:35.387466225 +0000
+++ bin/geninfo	2018-12-10 18:43:34.358890042 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 #
 #   Copyright (c) International Business Machines  Corp., 2002,2012
 #
@@ -49,6 +49,7 @@
 #                implementation (now enabled per default)
 #
 
+use warnings;
 use strict;
 use File::Basename; 
 use File::Spec::Functions qw /abs2rel catdir file_name_is_absolute splitdir
diff -Naur bin/genpng bin/genpng
--- bin/genpng	2018-12-10 18:40:35.387466225 +0000
+++ bin/genpng	2018-12-10 18:43:59.940753895 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 #
 #   Copyright (c) International Business Machines  Corp., 2002
 #
@@ -29,6 +29,7 @@
 #   2002-08-26: created by Peter Oberparleiter <Peter.Oberparleiter@de.ibm.com>
 #
 
+use warnings;
 use strict;
 use File::Basename; 
 use Getopt::Long;
diff -Naur bin/lcov bin/lcov
--- bin/lcov	2018-12-10 18:40:35.387466225 +0000
+++ bin/lcov	2018-12-10 18:41:58.442909080 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 #
 #   Copyright (c) International Business Machines  Corp., 2002,2012
 #
@@ -59,6 +59,7 @@
 #   2008-08-13 / Peter Oberparleiter: added function coverage support
 #
 
+use warnings;
 use strict;
 use File::Basename;
 use File::Path;
diff -Naur bin/updateversion.pl bin/updateversion.pl
--- bin/updateversion.pl	2018-12-10 18:40:35.387466225 +0000
+++ bin/updateversion.pl	2018-12-10 18:41:24.399139613 +0000
@@ -1,5 +1,6 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 
+use warnings;
 use strict;
 
 use File::Basename;
