diff --git a/cmake/shared.cmake b/cmake/shared.cmake
index cf86991..2189341 100644
--- a/cmake/shared.cmake
+++ b/cmake/shared.cmake
@@ -1,13 +1,3 @@
-execute_process(COMMAND git rev-parse HEAD
-    WORKING_DIRECTORY ${libddwaf_SOURCE_DIR}
-    OUTPUT_VARIABLE BUILD_ID
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-
-message(STATUS "Build id is ${BUILD_ID}")
-string(SUBSTRING "${BUILD_ID}" 0 2 BUILD_ID_PREFIX)
-string(SUBSTRING "${BUILD_ID}" 2 39 BUILD_ID_SUFFIX)
-
 add_library(libddwaf_shared SHARED
     $<TARGET_OBJECTS:libddwaf_shared_objects> $<$<BOOL:${MSVC}>:libddwaf.def>)
 set_target_properties(libddwaf_shared PROPERTIES OUTPUT_NAME ddwaf)
@@ -22,12 +12,11 @@ if(LINUX)
         $<$<BOOL:${LIBDDWAF_ENABLE_LTO}>:-flto>
         -Wl,--no-undefined
         -Wl,-version-script=${libddwaf_SOURCE_DIR}/libddwaf.version
-        -Wl,--build-id=0x${BUILD_ID}
         ${LIBDDWAF_PRIVATE_LIBRARIES}
         -static-libstdc++
         glibc_compat_time64 glibc_compat_math)
 
-    if(NOT (CMAKE_BUILD_TYPE MATCHES Debug))
+    if(FALSE AND NOT (CMAKE_BUILD_TYPE MATCHES Debug))
         set(SYMBOL_FILE $<TARGET_FILE:libddwaf_shared>.debug)
         add_custom_command(TARGET libddwaf_shared POST_BUILD
             COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:libddwaf_shared> ${SYMBOL_FILE}
@@ -41,7 +30,7 @@ if(LINUX)
 elseif (APPLE)
     target_link_libraries(libddwaf_shared PRIVATE -Wl,-undefined,error libddwaf_shared_objects)
 
-    if(NOT (CMAKE_BUILD_TYPE MATCHES Debug))
+    if(FALSE AND NOT (CMAKE_BUILD_TYPE MATCHES Debug))
         # Ensure that dsymutil and strip is present
         find_program(DSYMUTIL dsymutil)
         if (DSYMUTIL STREQUAL "DSYMUTIL-NOTFOUND")
