diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4134fcf..9c48112 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -104,7 +104,18 @@ endif()
 
 add_subdirectory(third_party EXCLUDE_FROM_ALL)
 
-set(LIBDDWAF_PRIVATE_LIBRARIES lib_rapidjson)
+find_package(RapidJSON REQUIRED)
+add_compile_definitions(RAPIDJSON_HAS_STDSTRING=1)
+find_package(re2 REQUIRED)
+find_package(utf8proc REQUIRED)
+add_library(lib_rapidjson ALIAS RapidJSON)
+find_package(GTest REQUIRED)
+add_library(lib_gtest ALIAS GTest::gtest)
+add_library(lib_gmock ALIAS GTest::gmock)
+find_package(yaml-cpp REQUIRED)
+add_library(lib_yamlcpp ALIAS yaml-cpp::yaml-cpp)
+
+set(LIBDDWAF_PRIVATE_LIBRARIES lib_rapidjson re2::re2 utf8proc::utf8proc)
 if(NOT MSVC AND LIBDDWAF_TESTING AND LIBDDWAF_TEST_COVERAGE)
     list(APPEND LIBDDWAF_PRIVATE_LIBRARIES gcov)
 endif()
diff --git a/cmake/objects.cmake b/cmake/objects.cmake
index 19c22af..989eee2 100644
--- a/cmake/objects.cmake
+++ b/cmake/objects.cmake
@@ -101,23 +101,6 @@ set(LIBDDWAF_SOURCE
     ${libddwaf_SOURCE_DIR}/src/vendor/libinjection/src/libinjection_html5.c
     ${libddwaf_SOURCE_DIR}/src/vendor/libinjection/src/libinjection_xss.c
     ${libddwaf_SOURCE_DIR}/src/vendor/libinjection/src/libinjection_sqli.c
-    ${libddwaf_SOURCE_DIR}/src/vendor/utf8proc/utf8proc.c
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/bitstate.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/compile.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/dfa.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/nfa.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/onepass.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/parse.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/perl_groups.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/prog.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/re2.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/regexp.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/simplify.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/tostring.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/unicode_casefold.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/unicode_groups.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/util/rune.cc
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/util/strutil.cc
 )
 
 set(LIBDDWAF_PUBLIC_INCLUDES ${libddwaf_SOURCE_DIR}/include)
@@ -128,8 +111,7 @@ set(LIBDDWAF_PRIVATE_INCLUDES
     ${libddwaf_SOURCE_DIR}/src/vendor/libinjection/src/
     ${libddwaf_SOURCE_DIR}/src/vendor/radixlib/
     ${libddwaf_SOURCE_DIR}/src/vendor/lua-aho-corasick/
-    ${libddwaf_SOURCE_DIR}/src/vendor/utf8proc/
-    ${libddwaf_SOURCE_DIR}/src/vendor/re2/
+    ${CMAKE_INSTALL_PREFIX}/include/re2
 )
 
 function(gen_objects target_name)
