diff --git a/CMakeLists.txt b/CMakeLists.txt
index c3f3b1c..fbaa54d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,12 +48,6 @@ if(NOT MSVC)
         add_compile_options(-ggdb)
     endif()
-else()
+else()  # meaningless change to workaround https://github.com/prefix-dev/rattler-build/issues/2693
-    if (CMAKE_BUILD_TYPE MATCHES Debug)
-        add_compile_options(/MTd)
-    else()
-        add_compile_options(/MT)
-    endif()
-
     add_compile_definitions(-D_CRT_SECURE_NO_WARNINGS=1 -Dstrdup=_strdup -Dputenv=_putenv)
 endif()
 
