diff --git a/cmake/shared.cmake b/cmake/shared.cmake
index 2189341..a07038f 100644
--- a/cmake/shared.cmake
+++ b/cmake/shared.cmake
@@ -3,7 +3,8 @@ add_library(libddwaf_shared SHARED
 set_target_properties(libddwaf_shared PROPERTIES OUTPUT_NAME ddwaf)
 
 install(TARGETS libddwaf_shared EXPORT libddwaf-config
     DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 if(LINUX)
