diff --git a/setup.py b/setup.py
index 571b4a3..d0b4624 100644
--- a/setup.py
+++ b/setup.py
@@ -5,7 +5,7 @@ import io
 from setuptools import (setup, find_namespace_packages)
 
 here = op.abspath(op.dirname(__file__))
-with io.open(op.join(here, 'requirements.txt'), encoding='utf-8') as f:
+with io.open(op.join(here, 'requires.txt'), encoding='utf-8') as f:
     all_reqs = f.read().split('\n')
 install_requires = [x.strip() for x in all_reqs if 'git+' not in x]
 
