diff --git a/requirements.txt b/requirements.txt
index e69de29..26858dd 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -0,0 +1,5 @@
+click==6.*
+pydantic==1.2
+typing_extensions==3.7.4.1
+everett==0.9
+six==1.11.*
\ No newline at end of file
diff --git a/setup.py b/setup.py
index 571b4a3..2eea08e 100644
--- a/setup.py
+++ b/setup.py
@@ -32,8 +32,9 @@ setup(
     packages=find_namespace_packages(exclude=['integration_tests*', 'tests*']),
     install_requires=install_requires,
     zip_safe=False,
-    entry_points='''
-        [console_scripts]
-        rastervision=rastervision.pipeline.cli:main
-    ''',
+    entry_points={
+        'console_scripts': [
+            'rastervision=rastervision.pipeline.cli:main',
+        ],
+    },
 )