diff --git a/setup.py b/setup.py
index 571b4a3..2eea08e 100644
--- a/setup.py
+++ b/setup.py
@@ -32,8 +32,9 @@ setup(
     packages=find_namespace_packages(exclude=['integration_tests*', 'tests*']),
     install_requires=install_requires,
     zip_safe=False,
-    entry_points='''
-        [console_scripts]
-        rastervision=rastervision.pipeline.cli:main
-    ''',
+    entry_points={
+        'console_scripts': [
+            'rastervision=rastervision.pipeline.cli:main',
+        ],
+    },
 )
