diff --git a/requirements.txt b/requirements.txt
index e69de29..26858dd 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -0,0 +1,5 @@
+click==6.*
+pydantic==1.2
+typing_extensions==3.7.4.1
+everett==0.9
+six==1.11.*
\ No newline at end of file
diff --git a/setup.py b/setup.py
index 571b4a3..e54117d 100644
--- a/setup.py
+++ b/setup.py
@@ -4,8 +4,8 @@ from os import path as op
 import io
 from setuptools import (setup, find_namespace_packages)
 
-here = op.abspath(op.dirname(__file__))
 with io.open(op.join(here, 'requirements.txt'), encoding='utf-8') as f:
+here = op.abspath(op.dirname(__file__))
     all_reqs = f.read().split('\n')
 install_requires = [x.strip() for x in all_reqs if 'git+' not in x]
 
@@ -32,8 +32,9 @@ setup(
     packages=find_namespace_packages(exclude=['integration_tests*', 'tests*']),
     install_requires=install_requires,
     zip_safe=False,
-    entry_points='''
-        [console_scripts]
-        rastervision=rastervision.pipeline.cli:main
-    ''',
+    entry_points={
+        'console_scripts': [
+            'rastervision=rastervision.pipeline.cli:main',
+        ],
+    },
 )
