From 61facb86e05f994e21f82eec8ec769441c0e1414 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 28 Mar 2024 21:33:30 -0500
Subject: [PATCH] Update setup.py

---
 setup.py | 22 ++++++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 1fc1f8639..ff5fa974b 100644
--- a/setup.py
+++ b/setup.py
@@ -748,6 +748,10 @@ def get_mupdf():
         libraries = [
                 f'mupdf',
                 f'mupdf-third',
+                f'jbig2dec', 
+                f'openjp2', 
+                f'jpeg', 
+                f'gumbo',
                 ]
     
     elif linux:
@@ -856,16 +860,26 @@ def write_git(name, directory):
 log( f'extra_compile_args={extra_compile_args}')
 log( f'extra_link_args={extra_link_args}')
 
+include_dirs = [
+    os.path.join(os.environ.get("PREFIX"), "include"),
+    os.path.join(os.environ.get("PREFIX"), "include", "mupdf"),
+    os.path.join(os.environ.get("PREFIX"), "include", "freetype2"),
+]
+include_args = [
+    "-I" + incdir
+    for incdir in include_dirs
+]
+
 module = Extension(
     "fitz._fitz",
     ["fitz/fitz.i"],
     language="c++",
     include_dirs=include_dirs,
-    library_dirs=library_dirs,
+    library_dirs=[os.path.join(os.environ.get("PREFIX"), "lib")],
     libraries=libraries,
-    extra_compile_args=extra_compile_args,
-    extra_link_args=extra_link_args,
-    swig_opts=extra_swig_args,
+    # extra_compile_args=extra_compile_args,
+    extra_link_args=["-ljbig2dec"],
+    swig_opts=include_args + ['-w509'],
 )
 
 
