From 55063e637481d7506e3e144007cc253c6553c6c8 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 28 Mar 2024 22:33:30 -0500
Subject: [PATCH] Update setup.py

---
 setup.py | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 1fc1f8639..3ce9f7655 100644
--- a/setup.py
+++ b/setup.py
@@ -856,16 +856,26 @@ def write_git(name, directory):
 log( f'extra_compile_args={extra_compile_args}')
 log( f'extra_link_args={extra_link_args}')
 
+include_dirs = [
+    os.path.join(os.environ.get("PREFIX"), "include"),
+    os.path.join(os.environ.get("PREFIX"), "include", "mupdf"),
+    os.path.join(os.environ.get("PREFIX"), "include", "freetype2"),
+]
+include_args = [
+    "-I" + incdir
+    for incdir in include_dirs
+]
+
 module = Extension(
     "fitz._fitz",
     ["fitz/fitz.i"],
     language="c++",
     include_dirs=include_dirs,
-    library_dirs=library_dirs,
-    libraries=libraries,
-    extra_compile_args=extra_compile_args,
-    extra_link_args=extra_link_args,
-    swig_opts=extra_swig_args,
+    library_dirs=[os.path.join(os.environ.get("PREFIX"), "lib")],
+    libraries=libraries + ["gumbo", "harfbuzz", "jbig2dec", "jpeg", "openjp2"],
+    # extra_compile_args=extra_compile_args,
+    # extra_link_args=extra_link_args,
+    swig_opts=include_args + ['-w509'],
 )
 
 
