From 4a4cbbaad7d752d5b1432f3bab099a7838a9eaae Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 28 Mar 2024 19:32:11 -0500
Subject: [PATCH] Update setup.py

---
 setup.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 1fc1f8639..24f14e14c 100644
--- a/setup.py
+++ b/setup.py
@@ -860,12 +860,12 @@ def write_git(name, directory):
     "fitz._fitz",
     ["fitz/fitz.i"],
     language="c++",
-    include_dirs=include_dirs,
-    library_dirs=library_dirs,
+    # include_dirs=include_dirs,
+    # library_dirs=library_dirs,
     libraries=libraries,
-    extra_compile_args=extra_compile_args,
-    extra_link_args=extra_link_args,
-    swig_opts=extra_swig_args,
+    # extra_compile_args=extra_compile_args,
+    # extra_link_args=extra_link_args,
+    # swig_opts=extra_swig_args,
 )
 
 
