From 07a769535252c9e80df5b5d3478fc333bf408f3d Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 28 Mar 2024 20:09:40 -0500
Subject: [PATCH] Update setup.py

---
 setup.py | 32 +++++++-------------------------
 1 file changed, 7 insertions(+), 25 deletions(-)

diff --git a/setup.py b/setup.py
index 1fc1f8639..00e7ae1ed 100644
--- a/setup.py
+++ b/setup.py
@@ -752,33 +752,18 @@ def get_mupdf():
     
     elif linux:
         # Use system libraries.
-        include_dirs.append( '/usr/include/mupdf')
-        include_dirs.append( '/usr/local/include/mupdf')
-        include_dirs.append( '/usr/include/freetype2')
+        include_dirs.append(os.path.join(os.environ.get("PREFIX"), "include/mupdf"))
+        include_dirs.append(os.path.join(os.environ.get("PREFIX"), "include"))
+        include_dirs.append(os.path.join(os.environ.get("PREFIX"), "freetype"))
         libraries = load_libraries()
-        extra_link_args = []
-        extra_swig_args.append(f'-I/usr/local/include')
-        extra_swig_args.append(f'-I/usr/include')
 
     elif darwin or openbsd or freebsd:
         # Use system libraries.
-        include_dirs.append("/usr/local/include/mupdf")
-        include_dirs.append("/usr/local/include")
-        include_dirs.append("/opt/homebrew/include/mupdf")
-        library_dirs.append("/usr/local/lib")
+        include_dirs.append(os.path.join(os.environ.get("PREFIX"), "include/mupdf"))
+        include_dirs.append(os.path.join(os.environ.get("PREFIX"), "include/freetype"))
+        include_dirs.append(os.path.join(os.environ.get("PREFIX"), "include"))
         libraries = ["mupdf", "mupdf-third"]
-        library_dirs.append("/opt/homebrew/lib")
-
-        include_dirs.append("/usr/include/freetype2")
-        include_dirs.append("/usr/local/include/freetype2")
-        include_dirs.append("/usr/X11R6/include/freetype2")
-        include_dirs.append("/opt/homebrew/include")
-        include_dirs.append("/opt/homebrew/include/freetype2")
-        
-        extra_swig_args.append(f'-I/usr/local/include')
-        extra_swig_args.append(f'-I/opt/homebrew/include')
-
-        library_dirs.append("/opt/homebrew/lib")
+        library_dirs.append(os.path.join(os.environ.get("PREFIX"), "lib"))
 
         if freebsd:
             libraries += [
@@ -863,9 +848,6 @@ def write_git(name, directory):
     include_dirs=include_dirs,
     library_dirs=library_dirs,
     libraries=libraries,
-    extra_compile_args=extra_compile_args,
-    extra_link_args=extra_link_args,
-    swig_opts=extra_swig_args,
 )
 
 
