From ebf3c9d5bc47e9036fb9d67c052e26107aaccab3 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 28 Mar 2024 19:44:21 -0500
Subject: [PATCH] Update setup.py

---
 setup.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 1fc1f8639..a2e8dd29f 100644
--- a/setup.py
+++ b/setup.py
@@ -860,12 +860,12 @@ def write_git(name, directory):
     "fitz._fitz",
     ["fitz/fitz.i"],
     language="c++",
-    include_dirs=include_dirs,
-    library_dirs=library_dirs,
+    include_dirs=[os.path.join(os.environ.get("PREFIX"), 'include'), os.path.join(os.environ.get("PREFIX"), 'include', 'mupdf')],
+    library_dirs=[os.path.join(os.environ.get("PREFIX"), 'lib')],
     libraries=libraries,
-    extra_compile_args=extra_compile_args,
-    extra_link_args=extra_link_args,
-    swig_opts=extra_swig_args,
+    # extra_compile_args=extra_compile_args,
+    # extra_link_args=extra_link_args,
+    # swig_opts=extra_swig_args,
 )
 
 
