From d5eb84df9413b2d950672953238646d184a504e6 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 17 Oct 2024 13:42:56 +0200
Subject: [PATCH] Update setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 72ac91f..7b1ab17 100644
--- a/setup.py
+++ b/setup.py
@@ -14,7 +14,7 @@
                 vars2readme[v] = line.split("=")[1]
 
 deps = []
-with open("/home/zrguo/code/LightRAG/requirements.txt") as f:
+with open("requirements.txt") as f:
     for line in f.readlines():
         if not line.strip():
             continue
