From 689d6b60043ecf9850047e3977ff2b52d4f2dc19 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Mon, 9 Sep 2024 22:16:03 -0400
Subject: [PATCH] don't install volk.c

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d3684b3..e2af272 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,7 +97,7 @@ if(VOLK_INSTALL)
   set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/volk)
 
   # Install files
-  install(FILES volk.h volk.c DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+  install(FILES volk.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
   # Install library target and add it and any dependencies to export set.
   install(TARGETS volk volk_headers
