diff --git a/src/C++/test/Makefile.am b/src/C++/test/Makefile.am
index 8837d178..4afe8c84 100644
--- a/src/C++/test/Makefile.am
+++ b/src/C++/test/Makefile.am
@@ -38,7 +38,6 @@ ut_SOURCES = \
 	SharedArrayTestCase.cpp \
 	SocketAcceptorTestCase.cpp \
 	SocketConnectionTestCase.cpp \
-	SocketConnectorTestCase.cpp \
 	SocketMonitorTestCase.cpp \
 	SocketServerTestCase.cpp \
 	StringUtilitiesTestCase.cpp \
@@ -49,4 +48,4 @@ ut_SOURCES = \
 	UtcTimeStampTestCase.cpp \
 	UtilityTestCase.cpp
 	
-AM_CPPFLAGS = -DCATCH_AMALGAMATED_CUSTOM_MAIN -I..
\ No newline at end of file
+AM_CPPFLAGS = -DCATCH_AMALGAMATED_CUSTOM_MAIN -I..
