diff --git a/test/check.sh b/test/check.sh
index 0fab201c..e72ad15a 100755
--- a/test/check.sh
+++ b/test/check.sh
@@ -15,11 +15,4 @@ then
   exit $?
 fi
 
-./runat.sh $QUICKFIX_TEST_PORT
-RESULT=$?
-if [ $RESULT != 0 ]
-then 
-  exit $RESULT
-fi
-
 sleep 1
diff --git a/test/test.rb b/test/test.rb
index b47fd551..bcfedb50 100644
--- a/test/test.rb
+++ b/test/test.rb
@@ -29,7 +29,7 @@ Object.is_testing = true
 require 'runit/cui/testrunner'
 require 'ReflectorTestCase'
 require 'FixParserTestCase'
-require 'SocketServerTestCase'
+#require 'SocketServerTestCase'
 #require 'ReflectorServerTestCase'
 require 'ReflectorClientTestCase'
 require 'ComparatorTestCase'
@@ -54,7 +54,7 @@ end
 
 RUNIT::CUI::TestRunner.run(ReflectorTestCase.suite)
 RUNIT::CUI::TestRunner.run(FixParserTestCase.suite)
-RUNIT::CUI::TestRunner.run(SocketServerTestCase.suite)
+#RUNIT::CUI::TestRunner.run(SocketServerTestCase.suite)
 #RUNIT::CUI::TestRunner.run(ReflectorClientTestCase.suite)
 RUNIT::CUI::TestRunner.run(ComparatorTestCase.suite)
 #RUNIT::CUI::TestRunner.run(ReflectorServerTestCase.suite)
