diff --git a/setup.py b/setup.py
index 182fe43..374a5e6 100755
--- a/setup.py
+++ b/setup.py
@@ -16,7 +16,7 @@ req_pkgs = [
     'itunespy',
     'requests',
     'colorama',
-    'bs4',
+    'beautifulsoup4',
     'downloader-cli',
     'pyxdg',
     'ffmpeg-python',
@@ -85,5 +85,4 @@ if __name__ == '__main__':
         install_requires=req_pkgs,
         setup_requires=req_pkgs,
         extras_require=extra_features,
-        **params
     )
