From 6e423b51c11ca9730c730b388715aec553d3d2bb Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Sun, 19 Apr 2026 06:40:06 -0400
Subject: [PATCH 2/2] drop ami_atlas_post_install usage

---
 setup.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/setup.py b/setup.py
index e3d1319..d6d1f3d 100644
--- a/setup.py
+++ b/setup.py
@@ -27,12 +27,10 @@ if __name__ == '__main__':
 
 	scripts = [
 		'ami_atlas',
-		'ami_atlas_post_install',
 	]
 
 	if os.name == 'nt':
 		scripts.append('ami_atlas.bat')
-		scripts.append('ami_atlas_post_install.bat')
 
 	#####################################################################
 
-- 
2.45.2

