From 8d55c2b5d573ed51871ea835d770aa245d740bcd Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 31 Mar 2026 22:08:08 -0700
Subject: [PATCH 2/2] add main

---
 pyAMI_atlas/cli.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/pyAMI_atlas/cli.py b/pyAMI_atlas/cli.py
index ef98633..281a950 100755
--- a/pyAMI_atlas/cli.py
+++ b/pyAMI_atlas/cli.py
@@ -227,7 +227,10 @@ def entry_point():
 
 #############################################################################
 
-if __name__ == '__main__':
+def main():
 	sys.exit(entry_point())
 
+if __name__ == '__main__':
+	main()
+
 #############################################################################
-- 
2.45.2

