diff --git a/tools/lib/NQP/Config/Rakudo.pm b/tools/lib/NQP/Config/Rakudo.pm
index 61985e696..d350de21a 100644
--- a/tools/lib/NQP/Config/Rakudo.pm
+++ b/tools/lib/NQP/Config/Rakudo.pm
@@ -560,7 +560,8 @@ sub clean_old_p6_libs {
         my $nqp_config = $self->{impls}{moar}->{config};
         my $lib_dir    = File::Spec->rel2abs(
             File::Spec->catdir(
-                $nqp_config->{'nqp::prefix'},
+                # Strip null character from prefix path
+                ($nqp_config->{'nqp::prefix'} =~ s/\0//gr),
                 'share', 'nqp', 'lib', 'Perl6'
             )
         );
