diff --git a/tools/lib/NQP/Config/Rakudo.pm b/tools/lib/NQP/Config/Rakudo.pm
index 61985e696..3b16a7d04 100644
--- a/tools/lib/NQP/Config/Rakudo.pm
+++ b/tools/lib/NQP/Config/Rakudo.pm
@@ -139,7 +139,8 @@ sub configure_refine_vars {
             );
         }
         elsif ( $self->cfg('relocatable') eq 'reloc'
-            && abs_path($prefix) ne abs_path($nqp_prefix) )
+            && abs_path($prefix) ne abs_path($nqp_prefix)
+            && !$self->opt('ignore-errors') )
         {
             $self->sorry(
                 "Installation directory '$prefix' is different from '",
