diff --git a/tools/templates/Makefile-backend-common.in b/tools/templates/Makefile-backend-common.in
index fc6e16dac..259002737 100644
--- a/tools/templates/Makefile-backend-common.in
+++ b/tools/templates/Makefile-backend-common.in
@@ -4,23 +4,13 @@
 @bpv(GEN_CAT)@ = @shquot(@script(gen-cat.nqp)@)@ @backend@
 
 @bpv(BUILD_DIR)@ = @nfp(gen/@backend@)@
-@make_pp_pfx@ifndef @bpv(BLIB)@
 @bpv(BLIB)@ = blib
-@make_pp_pfx@endif
-@make_pp_pfx@ifndef @bpv(BLIB_RAKUDO)@
 @bpv(BLIB_RAKUDO)@ = @nfp(blib/Perl6)@
 @bpv(BLIB_RAKUDO_R)@ = @nfp(blib/Raku)@
-@make_pp_pfx@endif
 @bpv(NQP_LIBDIR)@ = @nqp::libdir@
-@make_pp_pfx@ifndef @bpv(NQP_RR)@
 @bpv(NQP_RR)@ = @bpm(NQP)@
-@make_pp_pfx@endif
-@make_pp_pfx@ifndef @bpv(NQP_FLAGS)@
 @bpv(NQP_FLAGS)@ = --module-path=blib --ll-exception
-@make_pp_pfx@endif
-@make_pp_pfx@ifndef @bpv(RUNNER_SUFFIX)@
 @bpv(RUNNER_SUFFIX)@ = @runner_suffix@
-@make_pp_pfx@endif
 @bpv(LIBDIR)@	= @nfp($(RAKUDO_HOME)/lib)@
 
 # files we create
@@ -122,9 +112,7 @@
 
 @bpv(HARNESS5)@ = $(PERL5) @nfp(t/harness5)@ --@backend@
 @bpv(HARNESS5_WITH_FUDGE)@ = @bpm(HARNESS5)@ --fudge --keep-exit-code
-@make_pp_pfx@ifndef @bpv(HARNESS_TYPE)@
 @bpv(HARNESS_TYPE)@ = $(HARNESS_TYPE)
-@make_pp_pfx@endif
 
 # --- Utility targets ---
 
