Avoid setenv(3) with zero-length name parameter.
--- tvips/util.c.orig
+++ tvips/util.c
@@ -324,23 +324,22 @@ mktimez(struct tm *tm, const char *zone)
     time_t t;
     int errno_save;
 
-    if (zone != NULL) {
-        zone_save = getenv("TZ");
-        if (setenv("TZ", zone, 1) != 0)
-            return (-1);
-        tzset();
-    }
-    t = mktime(tm);
+    if (zone == NULL || zone[0] == '\0')
+        return (mktime(tm));
 
-    if (zone != NULL) {
-        errno_save = errno;
-        if (zone_save != NULL)
-            setenv("TZ", zone_save, 1);
-        else
-            unsetenv("TZ");
-        tzset();
-        errno = errno_save;
-    }
+    zone_save = getenv("TZ");
+    if (setenv("TZ", zone, 1) != 0)
+        return (-1);
+    tzset();
+
+    t = mktime(tm);
+    errno_save = errno;
+    if (zone_save != NULL)
+        setenv("TZ", zone_save, 1);
+    else
+        unsetenv("TZ");
+    tzset();
+    errno = errno_save;
 
     return (t);
 }
