--- tvips/CMakeLists.txt.orig
+++ tvips/CMakeLists.txt
@@ -391,6 +391,7 @@ add_executable(ser2smv
   "tiff.c"
   "tvips.c"
   "util.c")
+add_dependencies(ser2smv _update_version_h)
 target_compile_features(ser2smv PRIVATE c_restrict cxx_constexpr)
 target_link_libraries(ser2smv
   Iconv::Iconv
@@ -402,6 +403,7 @@ add_executable(smv2pgm
   "frame.c"
   "smv2pgm.c"
   "util.c")
+add_dependencies(smv2pgm _update_version_h)
 target_link_libraries(smv2pgm Iconv::Iconv m)
 
 add_executable(tiff2raw
