diff --git a/pypestutils/helpers.py b/pypestutils/helpers.py
index ea1e9f4..668fd9e 100644
--- a/pypestutils/helpers.py
+++ b/pypestutils/helpers.py
@@ -167,9 +167,9 @@ def mod2obs_mf6(
         print("WARNING: replacing existing 'totim' column in observation dataframe")
     obsdf.loc[:, "totim"] = obsdf.datetime.apply(lambda x: x - start_datetime).dt.days
 
-    usite = obsdf.site.unique()
+    usite = obsdf.site.unique().tolist()
     usite.sort()
-    usite_dict = {s: c for s, c in zip(usite, np.arange(usite.shape[0], dtype=int))}
+    usite_dict = {s: c for s, c in zip(usite, np.arange(len(usite), dtype=int))}
     obsdf.loc[:, "isite"] = obsdf.site.apply(lambda x: usite_dict[x])
     obsdf.sort_values(by=["isite", "totim"], inplace=True)
 
