--- python-stratify-0.1a0.orig/stratify/_vinterp.pyx	2017-02-09 08:20:56.000000000 -0300
+++ python-stratify-0.1a0/stratify/_vinterp.pyx	2017-02-10 14:39:13.952245638 -0300
@@ -8,10 +8,12 @@
 cimport cython
 
 
+cdef extern from "numpy/npy_math.h" nogil:
+    bint isnan "npy_isnan"(long double)
+    float NAN "NPY_NAN"
+    float INFINITY "NPY_INFINITY"
+
 cdef extern from "math.h" nogil:
-    bint isnan(long double)
-    float INFINITY
-    float NAN
     double fabs(double z)
 
 
