diff --git a/unix/Make-arch b/unix/Make-arch
index 1044370..7a18e64 100644
--- a/unix/Make-arch
+++ b/unix/Make-arch
@@ -7,7 +7,6 @@
 
 # Some machines don't have/need ranlib, in their case, use 'touch'
 # this should be overridden by arch specific configuration lines below
-RANLIB= touch
 
 MISCDEFS=$(USEJPEG) $(USEPNG) $(FLT) $(MBOX)
 MISCINC=$(JPEGINC) $(PNGINC) $(SPACEBALLINC)
@@ -1026,12 +1025,17 @@ next:
 macosx:
 	$(MAKE) all \
 	"ARCH = macosx" \
-	"CC = cc" \
 	"CFLAGS = -Os -ffast-math -DBsd $(MISCFLAGS)" \
-	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
-	"RANLIB = ranlib" \
+	"LIBS = -L. -ltachyon $(MISCLIB)"
+
+macosx-64:
+	$(MAKE) all \
+	"ARCH = macosx" \
+	"CFLAGS = -O3 -m64 -ffast-math -DBsd $(MISCFLAGS)" \
+	"ARFLAGS = r" \
+	"STRIP = strip" \
 	"LIBS = -L. -ltachyon $(MISCLIB)"
 
 macosx-thr:
@@ -1216,12 +1220,9 @@ linux-64-debug:
 linux-64-thr:
 	$(MAKE) all \
 	"ARCH = linux-64-thr" \
-	"CC = gcc" \
 	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux  $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS)" \
-	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
-	"RANLIB = ranlib" \
 	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread"
 
 # Linux on AMD64/EM64T, using Intel C 
