--- a/unix/Make-config	2011-03-13 11:01:07.000000000 +0000
+++ b/unix/Make-config	2016-07-04 15:29:57.982923000 +0000
@@ -185,12 +185,12 @@
 # LibPNG can be downlaoded from:
 #   http://www.libpng.org/
 ##########################################################################
 # Uncomment the following lines to disable PNG support
-USEPNG=
-PNGINC=
-PNGLIB=
+USEPNG=-DUSEPNG
+PNGINC=-I$(PREFIX)/include
+PNGLIB=-L$(PREFIX)/lib -lpng -lz
 
 # Uncomment the following lines to enable PNG support
 #USEPNG= -DUSEPNG
 #PNGINC= -I/usr/local/include
 #PNGLIB= -L/usr/local/lib -lpng -lz
