diff --git a/unix/Make-arch b/unix/Make-arch
index 1044370..5ba89d8 100644
--- a/unix/Make-arch
+++ b/unix/Make-arch
@@ -1045,6 +1045,16 @@ macosx-thr:
 	"RANLIB = ranlib" \
 	"LIBS = -L. -ltachyon $(MISCLIB) -lpthread -framework Carbon"
 
+macosx-x86_64-thr:
+	$(MAKE) all \
+	"ARCH = macosx-x86-thr" \
+	"CFLAGS = -O3 -m64 -ffast-math -DBsd $(THREADSFLAGS) $(MISCFLAGS)" \
+	"AR = ar" \
+	"ARFLAGS = r" \
+	"STRIP = strip" \
+	"RANLIB = ranlib" \
+	"LIBS = -L. -ltachyon $(MISCLIB) -lpthread"
+
 macosx-altivec:
 	$(MAKE) all \
 	"ARCH = macosx-altivec" \
@@ -1155,7 +1165,6 @@ win32-mpi:
 linux:
 	$(MAKE) all \
 	"ARCH = linux" \
-	"CC = gcc" \
 	"CFLAGS = -m32 -Wall -O3 -g -ffast-math -DLinux $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
