From c7e96d48ba03ca944f82e92e847409c82439cbfa Mon Sep 17 00:00:00 2001
From: Ghislain Antony Vaillant <ghisvail@gmail.com>
Date: Sun, 30 May 2021 00:14:38 +0200
Subject: [PATCH] Return EXIT_SUCCESS if no command provided

---
 src/PETPVC.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/PETPVC.cxx b/src/PETPVC.cxx
index 84c7309..8b5556c 100644
--- a/src/PETPVC.cxx
+++ b/src/PETPVC.cxx
@@ -154,7 +154,7 @@ int main(int argc, char *argv[])
     //Parse command line.
     if (!command.Parse(argc, argv)) {
 		printPVCMethodList();
-        return EXIT_FAILURE;
+		return EXIT_SUCCESS;
     }
 
     //Get image filenames
-- 
2.30.2

