From 9f768a43fac9ea56d4ea238fec8a7e252ab206c7 Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Fri, 16 Nov 2018 11:19:10 +0000
Subject: [PATCH] Pass through CMAKE_AR when building googletest

---
 cmake/modules/SearchInstalledSoftware.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/modules/SearchInstalledSoftware.cmake b/cmake/modules/SearchInstalledSoftware.cmake
index ca32587d7ae..6abe60366ce 100644
--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -1587,6 +1587,7 @@ if (testing)
                   -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
                   -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
                   -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
+                  -DCMAKE_AR=${CMAKE_AR}
                   -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
                   ${EXTRA_GTEST_OPTS}
     # Disable install step
