diff --git a/setup.py b/setup.py
index 10f8908..30e59b9 100644
--- a/setup.py
+++ b/setup.py
@@ -268,7 +268,7 @@ class build_ext(_build_ext):
             library_dirs=library_dirs,
            libraries=libraries,
             export_symbols=export_symbols,
-            extra_postargs=extra_postargs)
+            extra_postargs=extra_postargs+['-framework', 'CoreFoundation'])
         if needs_manifest:
             temp_dir = os.path.dirname(objects[0])
             manifest_name = lib_name +".manifest"
