diff --git a/setup.py b/setup.py
index 605d486..680b8d2 100644
--- a/setup.py
+++ b/setup.py
@@ -257,6 +257,8 @@ class build_ext(_build_ext):
         needs_manifest = sys.platform == 'win32' and ver.major == 2 and not is_mingw
         extra_postargs = ["/MANIFEST"] if needs_manifest else None
         libraries = ["python{}{}".format(ver.major, ver.minor)] if is_mingw else None
+        if sys.platform == "darwin":
+            extra_postargs = ["-framework", "CoreFoundation"]
         self.compiler.link(
             CCompiler.SHARED_OBJECT,
             objects, lib_name,
