From 559e80bc2c943e295580b63f91cdefe11c5708a1 Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 11 Feb 2025 20:10:03 -0800
Subject: [PATCH] strip debug for osx

---
 CMakeLists.txt | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index aa385466..d44d3d5c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,11 +116,19 @@ if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
   set_property(CACHE CMAKE_INSTALL_PREFIX PROPERTY VALUE "${PROJECT_SOURCE_DIR}")
 endif()
 
+if(APPLE)
+  set(STRIP_DEBUG_OPTION -S)
+  set(STRIP_UNNEEDED_OPTION "")
+else()
+  set(STRIP_DEBUG_OPTION --strip-debug)
+  set(STRIP_UNNEEDED_OPTION --strip-unneeded)
+endif()
+
 function(post_build_debug_library name)
   add_custom_command(TARGET ${name}
     POST_BUILD
     COMMAND ${CMAKE_OBJCOPY} --only-keep-debug $<TARGET_FILE:${name}> ${CMAKE_CURRENT_BINARY_DIR}/lib${name}.so.debug
-    COMMAND ${CMAKE_STRIP} --strip-debug --strip-unneeded $<TARGET_FILE:${name}>
+    COMMAND ${CMAKE_STRIP} ${STRIP_DEBUG_OPTION} ${STRIP_UNNEEDED_OPTION} $<TARGET_FILE:${name}>
     COMMAND ${CMAKE_OBJCOPY} --add-gnu-debuglink=lib${name}.so.debug $<TARGET_FILE:${name}>
   )
 
@@ -133,7 +141,7 @@ function(post_build_debug_executable name)
   add_custom_command(TARGET ${name}
     POST_BUILD
     COMMAND ${CMAKE_OBJCOPY} --only-keep-debug $<TARGET_FILE:${name}> ${CMAKE_CURRENT_BINARY_DIR}/${name}.debug
-    COMMAND ${CMAKE_STRIP} --strip-debug --strip-unneeded $<TARGET_FILE:${name}>
+    COMMAND ${CMAKE_STRIP} ${STRIP_DEBUG_OPTION} ${STRIP_UNNEEDED_OPTION} $<TARGET_FILE:${name}>
     COMMAND ${CMAKE_OBJCOPY} --add-gnu-debuglink=${name}.debug $<TARGET_FILE:${name}>
   )
 
-- 
2.45.2

