From f0ef7c5ac081bc6d007b8efe5820b32733526d81 Mon Sep 17 00:00:00 2001
From: Ruslan Baratov <ruslan_baratov@yahoo.com>
Date: Wed, 7 Nov 2018 22:16:22 +0300
Subject: [PATCH 1/2] CMake: Unify 'install' command

---
 src/CMakeLists.txt | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 83ed499a..376346b8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -103,13 +103,9 @@ set_property(TARGET ${TARGET_NAME} PROPERTY CXX_STANDARD_REQUIRED ON)
 set_property(TARGET ${TARGET_NAME} PROPERTY VERSION "${PROJECT_VERSION}.0")
 set_property(TARGET ${TARGET_NAME} PROPERTY SOVERSION "0")
 
-if(MINGW)
-    # We need to install *.dll into bin/ and *.a into lib/. 
-    install(TARGETS ${TARGET_NAME}
-        RUNTIME DESTINATION bin
-        ARCHIVE DESTINATION lib${LIB_SUFFIX}
-    )
-else()
-    install(TARGETS ${TARGET_NAME} DESTINATION lib${LIB_SUFFIX})
-endif()
+install(TARGETS ${TARGET_NAME}
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
+)
 install(FILES ${HEADERS} DESTINATION include)

From 1212434e177ab15f4a10a7e40251c64843ecdde1 Mon Sep 17 00:00:00 2001
From: "Fomenko, Evarist M" <evarist.m.fomenko@intel.com>
Date: Wed, 28 Nov 2018 09:48:09 +0000
Subject: [PATCH 2/2] put libmklml.dll and libimp5md.dll to bin as well

---
 cmake/MKL.cmake | 21 ++-------------------
 1 file changed, 2 insertions(+), 19 deletions(-)

diff --git a/cmake/MKL.cmake b/cmake/MKL.cmake
index bb020595..ec444761 100644
--- a/cmake/MKL.cmake
+++ b/cmake/MKL.cmake
@@ -168,26 +168,9 @@ function(detect_mkl LIBNAME)
     string(FIND "${MKLLIBPATH}" ${CMAKE_CURRENT_SOURCE_DIR}/external __idx)
     if(${__idx} EQUAL 0)
         if(WIN32)
-            if(MINGW)
-                # We need to install *.dll into bin/ instead of lib/.
-                install(PROGRAMS ${MKLDLL} DESTINATION bin)
-            else()
-                install(PROGRAMS ${MKLDLL} DESTINATION lib)
-            endif()
+            install(PROGRAMS ${MKLDLL} ${MKLIOMP5DLL} DESTINATION bin)
         else()
-            install(PROGRAMS ${MKLLIB} DESTINATION lib)
-        endif()
-        if(MKLIOMP5LIB)
-            if(WIN32)
-                if(MINGW)
-                    # We need to install *.dll into bin/ instead of lib/.
-                    install(PROGRAMS ${MKLIOMP5DLL} DESTINATION bin)
-                else()
-                    install(PROGRAMS ${MKLIOMP5DLL} DESTINATION lib)
-                endif()
-            else()
-                install(PROGRAMS ${MKLIOMP5LIB} DESTINATION lib)
-            endif()
+            install(PROGRAMS ${MKLLIB} ${MKLIOMP5LIB} DESTINATION lib${LIB_SUFFIX})
         endif()
     endif()
 
