--- estnltk/run_tests.py
+++ estnltk/run_tests.py
@@ -2,8 +2,10 @@
 """Shortcut script for running all unit tests in Estnltk."""
 from __future__ import unicode_literals, print_function, absolute_import
 
+import sys
 import unittest
 
 if __name__ == '__main__':
     testsuite = unittest.TestLoader().discover('estnltk.tests')
-    unittest.TextTestRunner(verbosity=1).run(testsuite)
+    result = unittest.TextTestRunner(verbosity=1).run(testsuite)
+    sys.exit(not result.wasSuccessful())
