From 80ae5f9dc548d921416125fc8bbd4435c1624958 Mon Sep 17 00:00:00 2001
From: Mariana Meireles <mariana@psychonautgirl.space>
Date: Mon, 14 Jun 2021 16:31:41 +0200
Subject: [PATCH 4/4] Fix for osx

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 383c76f..5d97dba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -117,12 +117,13 @@ if (GLFW3_OSMESA_BACKEND)
   pkg_check_modules(osmesa REQUIRED IMPORTED_TARGET osmesa)
   target_link_libraries(${EXECUTABLE_NAME} PkgConfig::osmesa)
   set(NOTEBOOK_TOOLKIT_CPU TRUE)
+  target_link_libraries(${EXECUTABLE_NAME} glfw)
 else()
   find_package(OpenGL REQUIRED)
-  target_link_libraries(${EXECUTABLE_NAME} OpenGL::OpenGL)
+  find_package(glfw3)
+  target_link_libraries(${EXECUTABLE_NAME} glfw OpenGL::GL)
 endif()
 
-target_link_libraries(${EXECUTABLE_NAME} glfw)
 target_link_libraries(${EXECUTABLE_NAME} PNG::PNG)
 
 set(NOTEBOOK_TOOLKIT_ENABLED ON)
-- 
2.30.2

