--- a/cmake/ThirdPartyMPI.cmake
+++ b/cmake/ThirdPartyMPI.cmake
@@ -70,6 +70,7 @@ IF( NEKTAR_USE_MPI )
             BINARY_DIR ${TPBUILD}/gsmpi-1.2.1_2
             TMP_DIR ${TPBUILD}/gsmpi-1.2.1_2-tmp
             INSTALL_DIR ${TPDIST}
+            PATCH_COMMAND sed -i "s/VERSION 2.8.7/VERSION 3.5/" ${TPSRC}/gsmpi-1.2.1_2/CMakeLists.txt
             CONFIGURE_COMMAND
                 ${CMAKE_COMMAND}
                 -G ${CMAKE_GENERATOR}
