--- a/cmake/ThirdPartyHDF5.cmake
+++ b/cmake/ThirdPartyHDF5.cmake
@@ -35,27 +35,29 @@
     IF(THIRDPARTY_BUILD_HDF5)
         INCLUDE(ExternalProject)
         EXTERNALPROJECT_ADD(
-            hdf5-1.12.3
+            hdf5-2.0.0
             PREFIX ${TPSRC}
-            URL ${TPURL}/hdf5-1.12.3.tar.bz2
-            URL_MD5 5d609bf2a74f980aa42dbe61de452185
+            URL https://github.com/HDFGroup/hdf5/archive/refs/tags/2.0.0.tar.gz
+            URL_MD5 70ca5f34a5fb2b7c7430ae8814dfb03f
             STAMP_DIR ${TPBUILD}/stamp
             DOWNLOAD_DIR ${TPSRC}
-            SOURCE_DIR ${TPSRC}/hdf5-1.12.3
-            BINARY_DIR ${TPBUILD}/hdf5-1.12.3 
-            TMP_DIR ${TPBUILD}/hdf5-1.12.3-tmp
+            SOURCE_DIR ${TPSRC}/hdf5-2.0.0
+            BINARY_DIR ${TPBUILD}/hdf5-2.0.0
+            TMP_DIR ${TPBUILD}/hdf5-2.0.0-tmp
             INSTALL_DIR ${TPDIST}
             CONFIGURE_COMMAND ${CMAKE_COMMAND}
                 -G ${CMAKE_GENERATOR}
                 -DCMAKE_C_COMPILER:FILEPATH=${CMAKE_C_COMPILER}
                 -DCMAKE_CXX_COMPILER:FILEPATH=${CMAKE_CXX_COMPILER}
                 -DCMAKE_INSTALL_PREFIX:PATH=${TPDIST}
-		-DDEFAULT_API_VERSION=v110
+                -DHDF5_SHARED_LIBS=ON
                 -DHDF5_ENABLE_PARALLEL=ON
+                -DHDF5_ALLOW_UNSUPPORTED=ON
                 -DHDF5_BUILD_CPP_LIB=OFF
+                -DHDF5_BUILD_HL_LIB=ON
                 -DBUILD_TESTING=OFF
                 -DHDF5_BUILD_TOOLS=OFF
-                ${TPSRC}/hdf5-1.12.3
+                ${TPSRC}/hdf5-2.0.0
             )
 
         THIRDPARTY_LIBRARY(HDF5_LIBRARIES SHARED hdf5
@@ -69,7 +71,7 @@
     ELSE()
         MESSAGE(STATUS "Found HDF5: ${HDF5_LIBRARIES}")
         SET(HDF5_CONFIG_INCLUDE_DIR ${HDF5_INCLUDE_DIRS})
-        ADD_CUSTOM_TARGET(hdf5-1.12.3 ALL)
+        ADD_CUSTOM_TARGET(hdf5-2.0.0 ALL)
 
         # Newer HDF5 versions have changed the API
         # We compile deprecated symbols using the old API if newer than 1.10.0
@@ -78,7 +80,7 @@
         ENDIF()
     ENDIF()
 
-    ADD_DEPENDENCIES(thirdparty hdf5-1.12.3)
+    ADD_DEPENDENCIES(thirdparty hdf5-2.0.0)
 
     MARK_AS_ADVANCED(HDF5_LIBRARIES)
     MARK_AS_ADVANCED(HDF5_INCLUDE_DIRS)
