--- a/cmake/ThirdPartyScotch.cmake
+++ b/cmake/ThirdPartyScotch.cmake
@@ -100,11 +100,12 @@
                 ${SCOTCH_SRC}/Makefile.inc
             PATCH_COMMAND ${PATCH} -p0 -f < ${PROJECT_SOURCE_DIR}/cmake/thirdparty-patches/scotch-6_0_4-implicit-function.patch
             BUILD_COMMAND $(MAKE) -C ${SCOTCH_SRC}
-                "CFLAGS=-I${TPDIST}/include ${SCOTCH_CFLAGS}"
-                "LDFLAGS=-L${TPDIST}/lib ${SCOTCH_LDFLAGS}"
-                "CLIBFLAGS=-fPIC"
+                "CCS=${SCOTCH_C_COMPILER}"
                 "CCP=${SCOTCH_C_COMPILER}"
                 "CCD=${SCOTCH_C_COMPILER}"
+                "AR=${CMAKE_AR}"
+                "CFLAGS=-I${TPDIST}/include ${SCOTCH_CFLAGS} -fPIC"
+                "LDFLAGS=-L${TPDIST}/lib ${SCOTCH_LDFLAGS}"
                 "YACC=bison -pscotchyy -y -b y -Wno-yacc"
                 ${SCOTCH_BUILD_TARGET}
             INSTALL_COMMAND $(MAKE) -C ${SCOTCH_SRC}
