# Imported from basf2/externals/patches

diff --git a/circe2/src/Makefile.in b/circe2/src/Makefile.in
index 9e90c92..9f09795 100644
--- a/circe2/src/Makefile.in
+++ b/circe2/src/Makefile.in
@@ -1283,6 +1283,7 @@ circe2_moments.o: $(MODULE_FILES)
 circe2_ls.o: $(MODULE_FILES)
 circe2_generate.o: $(MODULE_FILES)
 
+bigarray_compat.cmx: bigarray_compat.cmi
 $(CIRCE2_NATIVE): $(CIRCE2_CMX)
 $(CIRCE2_BYTECODE): $(CIRCE2_CMO)
 
