From 610aadf1d2761c5d71c26118712da4a1c600375d Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Tue, 5 Oct 2021 12:36:57 +0100
Subject: [PATCH] Update CMakeLists for more portable linking

---
 CMakeLists.txt | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7901df..85ebe55 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,10 +85,11 @@ if(CMAKE_CUDA_COMPILER)
     set_property(TARGET "${TARGET_NAME}" PROPERTY CUDA_RESOLVE_DEVICE_SYMBOLS ON)
 endif()
 
-target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module ZLIB::ZLIB)
-if(DEFINED ENV{CONDA_PREFIX})
-  target_link_libraries("${TARGET_NAME}" PRIVATE gomp)
+target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module)
+if(DEFINED ENV{CONDA_PREFIX} AND (NOT APPLE OR CMAKE_COMPILER_IS_GNUCC))
+  target_link_libraries("${TARGET_NAME}" PRIVATE gomp z)
 else()
+  target_link_libraries("${TARGET_NAME}" PRIVATE ZLIB::ZLIB)
   if(OpenMP_CXX_FOUND)
       target_link_libraries("${TARGET_NAME}" PRIVATE OpenMP::OpenMP_CXX)
   endif()
-- 
2.28.0

