diff --git a/include/lldb/Utility/RegularExpression.h b/include/lldb/Utility/RegularExpression.h
index dba7001..32b18df 100644
--- a/include/lldb/Utility/RegularExpression.h
+++ b/include/lldb/Utility/RegularExpression.h
@@ -11,7 +11,23 @@
 #define liblldb_RegularExpression_h_
 
 #ifdef _WIN32
-#include "../lib/Support/regex_impl.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct llvm_regmatch_t;
+struct llvm_regex_t;
+
+int	llvm_regcomp(llvm_regex_t *, const char *, int);
+size_t	llvm_regerror(int, const llvm_regex_t *, char *, size_t);
+int	llvm_regexec(const llvm_regex_t *, const char *, size_t,
+                     llvm_regmatch_t [], int);
+void	llvm_regfree(llvm_regex_t *);
+
+#ifdef __cplusplus
+}
+#endif
 
 typedef llvm_regmatch_t regmatch_t;
 typedef llvm_regex_t regex_t;
