From https://git.incenp.org/damien/slackbuilds/commit/eaa3d86f2f50368e92f86c2eed5b3078fd9d9f83

diff -Naur libgcab/meson.build libgcab/meson.build
--- libgcab/meson.build    2018-02-09 12:44:30.000000000 +0000
+++ libgcab/meson.build    2018-03-23 11:29:22.330165988 +0000
@@ -93,9 +93,4 @@
     ],
     install : true
   )
-  gnome.generate_vapi('libgcab-1.0',
-    sources: gir[0],
-    packages : ['gio-2.0'],
-    install : true,
-  )
 endif
