From f04daf91097528fe36868a0c68383d35f11b1141 Mon Sep 17 00:00:00 2001
From: Tadeu Manoel <e.tadeu@gmail.com>
Date: Wed, 22 May 2019 14:38:14 -0300
Subject: [PATCH] Fix CMake compilation in Windows

---
 CMakeLists.txt       |  5 ++++-
 tests/CMakeLists.txt | 10 ++++++++--
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 55ed445..a04a11c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,10 @@ cmake_minimum_required(VERSION 2.8)
 set(CMAKE_BUILD_TYPE Release)
 project (sparsehash)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra -Wpedantic -Wno-missing-field-initializers")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall")
+if(NOT MSVC)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wextra -Wpedantic -Wno-missing-field-initializers")
+endif()
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g")
 
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index fe8d44a..749fa3c 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -14,5 +14,11 @@ add_executable(sparsehash_unittests
 add_executable(bench bench.cc)
 
 add_test(sparsehash_unittests sparsehash_unittests)
-target_link_libraries(sparsehash_unittests gtest pthread)
-target_link_libraries(bench pthread)
+
+set(_THREAD_LIB "")
+if(NOT WIN32)
+    set(_THREAD_LIB pthread)
+endif()
+
+target_link_libraries(sparsehash_unittests gtest ${_THREAD_LIB})
+target_link_libraries(bench ${_THREAD_LIB})
-- 
2.20.1.windows.1

