diff --git a/cmake/FindIconv.cmake b/cmake/FindIconv.cmake
index a30f1fd..4a3a76e 100644
--- a/cmake/FindIconv.cmake
+++ b/cmake/FindIconv.cmake
@@ -41,7 +41,11 @@ find_path(ICONV_INCLUDE_DIR iconv.h
 
 if(NOT ICONV_INCLUDE_DIR STREQUAL "ICONV_INCLUDE_DIR-NOTFOUND")
     set(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR})
-    check_function_exists(iconv_open ICONV_IN_GLIBC)
+    file(READ ${ICONV_INCLUDE_DIR}/iconv.h tmptxt)
+    string(FIND "${tmptxt}" "libiconv" matchres)
+    if(${matchres} EQUAL -1)
+        check_function_exists(iconv_open ICONV_IN_GLIBC)
+    endif()
 endif()
 
 if(NOT ICONV_IN_GLIBC)
