diff --git a/cmake/FindIconv.cmake b/cmake/FindIconv.cmake
index a30f1fd..99df4d5 100644
--- a/cmake/FindIconv.cmake
+++ b/cmake/FindIconv.cmake
@@ -39,11 +39,6 @@ find_path(ICONV_INCLUDE_DIR iconv.h
     PATH_SUFFIXES include
 )
 
-if(NOT ICONV_INCLUDE_DIR STREQUAL "ICONV_INCLUDE_DIR-NOTFOUND")
-    set(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR})
-    check_function_exists(iconv_open ICONV_IN_GLIBC)
-endif()
-
 if(NOT ICONV_IN_GLIBC)
     if (CMAKE_CL_64)
         find_library(ICONV_LIBRARY
