--- a/configure
+++ b/configure
@@ -981,10 +981,10 @@ get_pwd() {
     fi
 }

-meson="$(cd pyvenv/bin; get_pwd)/meson"
-if [ -f "$meson$EXESUF" ]; then
-  meson="$meson$EXESUF"
+meson="$PWD/pyvenv/Scripts/meson"
+if [ -f "$meson.exe" ]; then
+  meson="$meson.exe"
 fi

 # On Haiku, meson might show up in a "non-packaged" subfolder instead, see
 # https://github.com/haiku/haiku/blob/r1beta5/docs/user/storage/storageintro.dox
