--- qemu/scripts/tracetool/backend/log.py
+++ qemu/scripts/tracetool/backend/log.py
@@ -29,2 +29,5 @@
 def generate_h(event, group):
+    def normalize_path(path):
+        return os.path.relpath(path).replace('\\', '/')
+
     argnames = ", ".join(event.args.names())
@@ -57,3 +60,3 @@
         event_lineno=event.lineno,
-        event_filename=os.path.relpath(event.filename),
+        event_filename=normalize_path(event.filename),
         name=event.name,
