--- qemu/meson.build
+++ qemu/meson.build
@@ -3998,3 +3998,8 @@
+    # Use 'qemu-system-{arch}' for system emulators, 'qemu-{arch}' for linux-user
+    if target.endswith('-softmmu')
+      exe_name = 'qemu-system-' + target_name
+    else
+      exe_name = 'qemu-' + target_name
+    endif
     execs = [{
-      'name': 'qemu-' + target_name,
+      'name': exe_name,
       'win_subsystem': 'console',
