--- pyproject.toml	2022-12-13 05:00:17.000000000 -0600
+++ pyproject.toml	2023-10-06 16:04:14.668229649 -0500
@@ -1,4 +1,32 @@
 [build-system]
-requires = ["maturin>=0.10,<0.11"]
+requires = ["maturin>=0.14,<0.15"]
 build-backend = "maturin"
 
+[project]
+name = "py-bip39-bindings"
+version = "0.1.11"
+description = "Python bindings for tiny-bip39 RUST crate"
+readme = "README.md"
+requires-python = ">=3.7"
+license = {file = "LICENSE"}
+authors = [
+  {email = "info@polkascan.org"},
+  {name = "Polkascan Foundation"}
+]
+maintainers = [
+  {name = "Polkascan Foundation", email = "info@polkascan.org"}
+]
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "Intended Audience :: Developers",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11"
+]
+
+[project.urls]
+homepage = "https://github.com/polkascan/py-bip39-bindings"
+repository = "https://github.com/polkascan/py-bip39-bindings"
