--- Cargo.toml	2022-12-13 05:00:17.000000000 -0600
+++ Cargo.toml	2023-10-06 16:04:14.668229649 -0500
@@ -20,19 +20,6 @@
 crate-type = ["cdylib"]
 
 [dependencies.pyo3]
-version = "0.11.1"
+version = "0.18.1"
 features = ["extension-module"]
 
-[package.metadata.maturin]
-maintainer = "Stichting Polkascan (Polkascan Foundation)"
-maintainer-email = "legal@polkascan.org"
-classifier = [
-    "License :: OSI Approved :: Apache Software License",
-    "Intended Audience :: Developers",
-    "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.7",
-    "Programming Language :: Python :: 3.8",
-    "Programming Language :: Python :: 3.9",
-    "Programming Language :: Python :: 3.10",
-    "Programming Language :: Python :: 3.11"
-]
