--- setup.py	2024-09-24 10:51:30.000000000 -0400
+++ setup.py	2024-10-26 14:26:01.456152800 -0400
@@ -3,7 +3,8 @@
 
 from setuptools import setup
 from setuptools import find_packages
-from datespan import VERSION
+import os
+VERSION = os.environ["VERSION"]
 
 
 # ...to run the build and deploy process to pypi.org manually:
@@ -67,4 +68,4 @@
         'Documentation': 'https://github.com/Zeutschler/datespan',
         'GitHub': 'https://github.com/Zeutschler/datespan',
     },
-)
\ No newline at end of file
+)
