diff --git a/install/PackageConfig.cmake.in b/install/PackageConfig.cmake.in
index 13fd98f5d..28a77d968 100644
--- a/install/PackageConfig.cmake.in
+++ b/install/PackageConfig.cmake.in
@@ -36,7 +36,7 @@ add_library(zenohc::shared ALIAS __zenohc_shared)
   set_target_properties(__zenohc_shared PROPERTIES
       IMPORTED_NO_SONAME TRUE
       INTERFACE_COMPILE_DEFINITION ZENOHC_DYN_LIB
-      IMPORTED_LOCATION "${_IMPORT_PREFIX}/@CMAKE_INSTALL_LIBDIR@/@DYLIB@"
+      IMPORTED_LOCATION "${_IMPORT_PREFIX}/@ZENOHC_INSTALL_DYLIBDIR@/@DYLIB@"
       INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/@CMAKE_INSTALL_INCLUDEDIR@"
   )
 
