From 04263a44127aff172244b0a3fbbb6e03bfb363a7 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Mon, 24 Jun 2024 14:01:04 +0200
Subject: [PATCH] Update CMakeLists.txt

---
 tests/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 21695a439..12b55c326 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -2,6 +2,8 @@ message(STATUS "zenoh-c tests")
 
 add_custom_target(tests)
 
+find_package(Threads REQUIRED)
+
 file(GLOB files "${CMAKE_CURRENT_SOURCE_DIR}/*.c")
 foreach(file ${files})
     get_filename_component(target ${file} NAME_WE)
@@ -20,7 +22,7 @@ foreach(file ${files})
     add_executable(${target} EXCLUDE_FROM_ALL ${file})
     add_dependencies(tests ${target})
     add_dependencies(${target} zenohc::lib)
-    target_link_libraries(${target} PRIVATE zenohc::lib)
+    target_link_libraries(${target} PRIVATE zenohc::lib Threads::Threads)
     copy_dlls(${target})
     set_property(TARGET ${target} PROPERTY C_STANDARD 11)
     add_test(NAME "${test_type}_${target}" COMMAND ${target})
 
