diff --git a/setup.py b/setup.py
index deb10e0..216e1c2 100644
--- a/setup.py
+++ b/setup.py
@@ -47,14 +47,7 @@
     long_description=readme + '\n\n' + history,
     include_package_data=True,
     keywords='nucleaizer_backend',
-    packages=[
-        'nucleaizer_backend', 
-        'nucleaizer_backend.CycleGAN', 
-        'nucleaizer_backend.mrcnn_interface', 
-        'nucleaizer_backend.Mask_RCNN_tf2',
-        'nucleaizer_backend.Mask_RCNN_tf2.mrcnn',
-        'nucleaizer_backend.pix2pix_interface',
-        ],
+    packages=find_packages(exclude=("tests", "tests.*"),
     setup_requires=setup_requirements,
     test_suite='tests',
     tests_require=test_requirements,
