From c738e1cc925509b591cdda5f207a7ad9391d46d3 Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Thu, 30 Apr 2026 11:00:20 -0700
Subject: [PATCH 1/4] patch to min 3.5

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2206694..07c4b7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@
 # Licensed under the terms of the EUPL version 1.1 or later.
 # See the LICENCE file for more information.
 
-cmake_minimum_required(VERSION 2.9)
+cmake_minimum_required(VERSION 3.5...4.3)
 project(resummino C CXX Fortran)
 
 option(BUILD_LOOPTOOLS "Build LoopTools" ON)
-- 
2.45.2

