From a43a4badcff7ad0024a109433eed68e4dfdb644b Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Thu, 30 Apr 2026 11:57:36 -0700
Subject: [PATCH 2/2] add components for selective install

---
 src/CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e403d52..2c436b0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -43,12 +43,12 @@ endif()
 
 ## changed the looptools configure script that the folder is always called lib
 target_link_libraries(libresummino ${GSL_LIBRARIES}
-	${LHAPDF_LIBRARIES} 
-	${LOOPTOOLS_LIBRARIES} 
+	${LHAPDF_LIBRARIES}
+	${LOOPTOOLS_LIBRARIES}
 	 gfortran)
 
 add_executable(resummino  main.cc)
 target_link_libraries(resummino libresummino)
 
-install(TARGETS resummino RUNTIME DESTINATION bin)
-install(TARGETS libresummino ARCHIVE DESTINATION lib)
+install(TARGETS resummino RUNTIME DESTINATION bin COMPONENT runtime)
+install(TARGETS libresummino ARCHIVE DESTINATION lib COMPONENT library)
-- 
2.25.1

