From 5152ecb4869c20806a32a8c6ea746e27d3f07bc7 Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Fri, 1 May 2026 15:25:41 -0400
Subject: [PATCH 5/5] add link to gfortran in executable

---
 src/CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2c436b0..b644061 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -48,7 +48,13 @@ target_link_libraries(libresummino ${GSL_LIBRARIES}
 	 gfortran)
 
 add_executable(resummino  main.cc)
-target_link_libraries(resummino libresummino)
+target_link_libraries(resummino
+    libresummino
+    ${GSL_LIBRARIES}
+    ${LHAPDF_LIBRARIES}
+    ${LOOPTOOLS_LIBRARIES}
+    gfortran
+)
 
 install(TARGETS resummino RUNTIME DESTINATION bin COMPONENT runtime)
 install(TARGETS libresummino ARCHIVE DESTINATION lib COMPONENT library)
-- 
2.45.2

