 diff -Nu a/setup.py b/setup.py
--- a/setup.py  2023-08-24 10:49:34.111745791 -0500
+++ b/setup.py    2023-08-24 11:08:34.371766808 -0500
@@ -6,8 +6,8 @@
 with io.open("README.md", encoding="utf-8") as f:
     long_description = f.read()

-with io.open("requirements.txt", encoding="utf-8") as f:
-    requirements = [r for r in f.read().split('\n') if len(r)]
+# with io.open("requirements.txt", encoding="utf-8") as f:
+#     requirements = [r for r in f.read().split('\n') if len(r)]

 setup(name="bip32",
       version=bip32.__version__,
@@ -20,4 +20,4 @@
       license="MIT",
       packages=["bip32"],
       keywords=["bitcoin", "bip32", "hdwallet"],
-      install_requires=requirements)
+      install_requires=['coincurve', 'base58'])
