From 6f4b7c8e2bb17119fa95dee1f2751da41e990101 Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Thu, 10 Aug 2023 15:15:23 +0200
Subject: [PATCH] Remove Python2.7 support as deprecated in SOFA.

---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d43e639..1db2b87 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,10 +102,6 @@ if (CollisionOBBCapsule_FOUND)
 endif()
 
 # Config files and install rules for pythons scripts
-sofa_install_pythonscripts(PLUGIN_NAME ${PROJECT_NAME} PYTHONSCRIPTS_SOURCE_DIR "python")
-
-sofa_install_pythonscripts(PLUGIN_NAME ${PROJECT_NAME} PYTHONSCRIPTS_SOURCE_DIR "python3")
-
 find_file(SofaPython3Tools NAMES "SofaPython3/lib/cmake/SofaPython3/SofaPython3Tools.cmake")
 if(SofaPython3Tools)
     message("-- Found SofaPython3Tools.")
-- 
2.34.1

