diff --git a/configure b/configure
index 83b2a53..686218a 100755
--- a/configure
+++ b/configure
@@ -7055,8 +7055,8 @@ else
 fi
 
  if test $enable_test = yes; then
-  ENABLE_TEST_TRUE='#'
-  ENABLE_TEST_FALSE=
+  ENABLE_TEST_TRUE=
+  ENABLE_TEST_FALSE='#'
 else
   ENABLE_TEST_TRUE='#'
   ENABLE_TEST_FALSE=
@@ -7066,8 +7066,8 @@ fi
   ENABLE_CH_IMAGE_TRUE=
   ENABLE_CH_IMAGE_FALSE='#'
 else
-  ENABLE_CH_IMAGE_TRUE=
-  ENABLE_CH_IMAGE_FALSE='#'
+  ENABLE_CH_IMAGE_TRUE='#'
+  ENABLE_CH_IMAGE_FALSE=
 fi
 
  if test $enable_bundled_lark = yes; then
