--- Base/Makefile.in	2021-02-17 22:07:04.000000000 +0000
+++ Base/Makefile.in	2021-02-18 16:14:38.030000000 +0000
@@ -334,7 +334,7 @@
 LAL_CFLAGS = @LAL_CFLAGS@
 LAL_LIBS = @LAL_LIBS@
 LD = @LD@
-LDFLAGS = -pthread
+LDFLAGS = @LDFLAGS@
 LDFLAGS_ENDGROUP = @LDFLAGS_ENDGROUP@
 LDFLAGS_STARTGROUP = @LDFLAGS_STARTGROUP@
 LIBOBJS = @LIBOBJS@
@@ -581,6 +581,7 @@
 AM_CFLAGS = -pthread $(WARNINGCFLAGS) $(opt) $(simd_cflags)
 AM_CXXFLAGS = -pthread $(WARNINGCXXFLAGS) $(opt) $(simd_cflags) \
 			  $(zlibcflags)
+AM_LDFLAGS = $(LDFLAGS) -pthread
 
 all: all-am
 
