--- ./Services/web/Makefile.in	2021-02-17 22:07:05.000000000 +0000
+++ ./Services/web/Makefile.in	2021-02-19 11:54:30.450000000 +0000
@@ -295,7 +295,6 @@
 LAL_CFLAGS = @LAL_CFLAGS@
 LAL_LIBS = @LAL_LIBS@
 LD = @LD@
-LDFLAGS = -static
 LDFLAGS_ENDGROUP = @LDFLAGS_ENDGROUP@
 LDFLAGS_STARTGROUP = @LDFLAGS_STARTGROUP@
 LIBOBJS = @LIBOBJS@
--- ./Services/TrigClient/Makefile.in	2021-02-17 22:07:05.000000000 +0000
+++ ./Services/TrigClient/Makefile.in	2021-02-19 11:55:05.150000000 +0000
@@ -477,7 +477,6 @@
 trigRmNode_SOURCES = \
 	trigRmNode.cc
 
-trigRmNode_LDFLAGS = -static
 AM_CPPFLAGS = \
 	-I$(top_builddir)/config \
 	-I$(top_srcdir)/Base/thread \
--- ./Services/NameClient/Makefile.in	2021-02-17 22:07:05.000000000 +0000
+++ ./Services/NameClient/Makefile.in	2021-02-19 11:55:58.050000000 +0000
@@ -450,7 +450,6 @@
 	-I$(top_srcdir)/IO \
 	-I$(top_srcdir)/Services/AppClient
 
-NameCtrl_LDFLAGS = -static
 AM_CXXFLAGS = $(WARNINGCXXFLAGS) $(opt)
 @online_TRUE@LDADD = \
 @online_TRUE@	libNameClient.la \
--- ./Services/AlarmMgr/Makefile.in	2021-02-17 22:07:05.000000000 +0000
+++ ./Services/AlarmMgr/Makefile.in	2021-02-19 11:56:31.530000000 +0000
@@ -449,7 +449,6 @@
 			$(netlib) $(syslibs)
 
 AM_CXXFLAGS = $(WARNINGCXXFLAGS) $(opt)
-AM_LDFLAGS = -static
 all: all-am
 
 .SUFFIXES:
--- ./IO/daqsocket/Makefile.in	2021-02-17 22:07:04.000000000 +0000
+++ ./IO/daqsocket/Makefile.in	2021-02-19 11:54:00.460000000 +0000
@@ -491,13 +491,9 @@
 			-I$(top_srcdir)/IO/sockutil
 
 prchans_SOURCES = prchans.C
-prchans_LDFLAGS = -static
 prnames_SOURCES = prnames.C
-prnames_LDFLAGS = -static
 prndsvsn_SOURCES = prndsvsn.C
-prndsvsn_LDFLAGS = -static
 prdata_SOURCES = prdata.C
-prdata_LDFLAGS = -static
 LDADD = \
                         libdaqs.la \
                         ../sockutil/libsockutil.la \
--- ./IO/framefast/Makefile.in	2021-02-17 22:07:04.000000000 +0000
+++ ./IO/framefast/Makefile.in	2021-02-19 11:53:21.580000000 +0000
@@ -175,7 +175,7 @@
 fdir_OBJECTS = $(am_fdir_OBJECTS)
 fdir_LDADD = $(LDADD)
 fdir_DEPENDENCIES = libffhelper.la ../frameutils/libfuthelper.la \
-	../../Base/.libs/libgdsbase.a $(am__DEPENDENCIES_1) \
+	../../Base/libgdsbase.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 fdir_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
@@ -184,7 +184,7 @@
 fextract_OBJECTS = $(am_fextract_OBJECTS)
 fextract_LDADD = $(LDADD)
 fextract_DEPENDENCIES = libffhelper.la ../frameutils/libfuthelper.la \
-	../../Base/.libs/libgdsbase.a $(am__DEPENDENCIES_1) \
+	../../Base/libgdsbase.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 fextract_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
@@ -193,7 +193,7 @@
 finfo_OBJECTS = $(am_finfo_OBJECTS)
 finfo_LDADD = $(LDADD)
 finfo_DEPENDENCIES = libffhelper.la ../frameutils/libfuthelper.la \
-	../../Base/.libs/libgdsbase.a $(am__DEPENDENCIES_1) \
+	../../Base/libgdsbase.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 finfo_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
@@ -203,7 +203,7 @@
 framedump_OBJECTS = $(am_framedump_OBJECTS)
 framedump_LDADD = $(LDADD)
 framedump_DEPENDENCIES = libffhelper.la ../frameutils/libfuthelper.la \
-	../../Base/.libs/libgdsbase.a $(am__DEPENDENCIES_1) \
+	../../Base/libgdsbase.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 framedump_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
@@ -212,7 +212,7 @@
 fsettime_OBJECTS = $(am_fsettime_OBJECTS)
 fsettime_LDADD = $(LDADD)
 fsettime_DEPENDENCIES = libffhelper.la ../frameutils/libfuthelper.la \
-	../../Base/.libs/libgdsbase.a $(am__DEPENDENCIES_1) \
+	../../Base/libgdsbase.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 fsettime_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
@@ -541,15 +541,10 @@
 lib_pkgconfigdir = $(libdir)/pkgconfig
 lib_pkgconfig_DATA = gdsframefast.pc
 fdir_SOURCES = fdir.cc
-fdir_LDFLAGS = -static
 finfo_SOURCES = finfo.cc
-finfo_LDFLAGS = -static
 fsettime_SOURCES = fsettime.cc
-fsettime_LDFLAGS = -static
 fextract_SOURCES = fextract.cc
-fextract_LDFLAGS = -static
 @BUILDING_MACPORTS_FALSE@framedump_SOURCES = framedump.cc
-@BUILDING_MACPORTS_FALSE@framedump_LDFLAGS = -static
 AM_CPPFLAGS = \
 			-I$(top_builddir)/config \
 			-I$(top_srcdir)/IO \
@@ -573,7 +568,7 @@
 LDADD = \
 			libffhelper.la \
 			../frameutils/libfuthelper.la \
-			../../Base/.libs/libgdsbase.a \
+			../../Base/libgdsbase.la \
 			$(zliblibs) $(sysmtlibs)
 
 AM_CXXFLAGS = $(WARNINGCXXFLAGS) $(opt)
