--- configure	2020-01-16 15:47:51.000000000 -0800
+++ configure	2020-01-29 03:55:32.863278000 -0800
@@ -19576,9 +19576,9 @@

       readlinelibs=
 case "${host}" in
-*linux*) readlinelibs="-lreadline -ltermcap" ;;
-*solaris*) readlinelibs="-lreadline -ltermcap" ;;
-*darwin*) readlinelibs="-lreadline -ltermcap" ;;
+*linux*) readlinelibs="-lreadline -ltinfow" ;;
+*solaris*) readlinelibs="-lreadline -ltinfow" ;;
+*darwin*) readlinelibs="-lreadline -ltinfow" ;;
 esac

       case "$GDS_CC_FLAVOR" in
