diff --git a/src/napari_ids/_widget.py b/src/napari_ids/_widget.py
index 9e66a6f..3e9ac51 100644
--- a/src/napari_ids/_widget.py
+++ b/src/napari_ids/_widget.py
@@ -12,10 +12,10 @@ from datetime import datetime
 from pathlib import Path
 
 import cv2
-from PyQt5 import QtCore
-from PyQt5.QtCore import QTimer
-from PyQt5.QtGui import QCursor
-from PyQt5.QtWidgets import QMessageBox
+from qtpy import QtCore
+from qtpy.QtCore import QTimer
+from qtpy.QtGui import QCursor
+from qtpy.QtWidgets import QMessageBox
 from ids_peak import ids_peak
 from ids_peak_ipl import ids_peak_ipl
 from magicgui import magic_factory
diff --git a/src/napari_ids/video_ui.py b/src/napari_ids/video_ui.py
index 3badce3..a5de7e4 100644
--- a/src/napari_ids/video_ui.py
+++ b/src/napari_ids/video_ui.py
@@ -1,9 +1,9 @@
 import os
 import pathlib
 
-from PyQt5.QtCore import Qt, QSize
-from PyQt5.QtGui import QPixmap
-from PyQt5.QtWidgets import QGridLayout, QLabel, QPushButton, QComboBox, QSpinBox
+from qtpy.QtCore import Qt, QSize
+from qtpy.QtGui import QPixmap
+from qtpy.QtWidgets import QGridLayout, QLabel, QPushButton, QComboBox, QSpinBox
 
 
 def initui(self):
