From 4e82093ed5a4c1fcc63a97eb8523f6ef8162a4c0 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 9 Apr 2021 02:21:41 +0200
Subject: [PATCH] remove requirements.txt handling from setup.py

---
 setup.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 8eb4d96..4cfd93b 100644
--- a/setup.py
+++ b/setup.py
@@ -3,9 +3,6 @@ import setuptools
 with open("project-description.md", "r") as fh:
     long_description = fh.read()
 
-with open('requirements.txt') as f:
-    requirements = f.read().splitlines()
-
 setuptools.setup(
     name="pyserini",
     version="0.11.0.0",
@@ -18,7 +15,7 @@ setuptools.setup(
         "resources/jars/anserini-0.11.0-fatjar.jar",
      ]},
     url="https://github.com/castorini/pyserini",
-    install_requires=requirements,
+    install_requires=[],
     packages=setuptools.find_packages(),
     classifiers=[
         "Programming Language :: Python :: 3",
-- 
2.29.2.windows.3

