Index: config/FindBoostForALPS.cmake
===================================================================
--- config/FindBoostForALPS.cmake	(revision 7809)
+++ config/FindBoostForALPS.cmake	(working copy)
@@ -60,6 +60,14 @@
   unset(UTF)
   unset(TIMER)
 
+  # Avoid auto link of Boost library
+  add_definitions(-DBOOST_ALL_NO_LIB=1)
+  if(BUILD_SHARED_LIBS)
+    add_definitions(-DBOOST_ALL_DYN_LINK=1)
+  else(BUILD_SHARED_LIBS)
+    add_definitions(-DBOOST_ALL_STATIC_LINK=1)
+  endif(BUILD_SHARED_LIBS)
+
 endif(NOT Boost_ROOT_DIR)
 
 # Boost_FOUND is set only when FindBoost.cmake succeeds.
