Index: alps/tutorials/code-01-python/ising-skeleton.py
===================================================================
--- alps/tutorials/code-01-python/ising-skeleton.py	(revision 7835)
+++ alps/tutorials/code-01-python/ising-skeleton.py	(revision 7836)
@@ -81,13 +81,13 @@
     def step(self):
         for s in range(self.L*self.L):
             # Pick random site k=(i,j)
-            ...
+            #...
             
             # Measure local energy e = -s_k * sum_{l nn k} s_l
-            ...
+            #...
             
             # Flip s_k with probability exp(2 beta e)
-            ...
+            #...
         
     def measure(self):
         E = 0.    # energy
