Index: config/FindHDF5.cmake
===================================================================
--- config/FindHDF5.cmake	(revision 7814)
+++ config/FindHDF5.cmake	(working copy)
@@ -82,7 +82,7 @@
 ENDIF(Hdf5_INCLUDE_DIRS)
 
 IF (WIN32 AND NOT UNIX)
-  SET(HDF5_DEFINITIONS "-D_HDF5USEDLL_" CACHE STRING "HDF5 defintions")
+  SET(HDF5_DEFINITIONS "-D_HDF5USEDLL_ -DH5_BUILT_AS_DYNAMIC_LIB" CACHE STRING "HDF5 defintions")
 ENDIF (WIN32 AND NOT UNIX)
 
 
