Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 7809)
+++ CMakeLists.txt	(working copy)
@@ -408,9 +408,9 @@
   endif(MSVC)
 endif(SZIP_FOUND AND NOT ALPS_USE_VISTRAILS)
 
-if(MSVC AND NOT ALPS_FOR_VISTRAILS)
+if(MSVC AND ALPS_USE_VISTRAILS AND NOT ALPS_FOR_VISTRAILS)
   file(COPY ${SZIP_DLLS} ${ZLIB_DLLS} ${PYTHON_DLLS} DESTINATION ${PROJECT_BINARY_DIR}/bin)
-endif(MSVC AND NOT ALPS_FOR_VISTRAILS)
+endif(MSVC AND ALPS_USE_VISTRAILS AND NOT ALPS_FOR_VISTRAILS)
  
 # HDF5
 find_package(HDF5)
@@ -548,62 +548,6 @@
 endif(APPLE AND (CMAKE_VERSION VERSION_LESS 2.8.12))
 
 ######################################################################
-# install dependent libraries under Windows, from applications
-######################################################################
-
-if(WIN32 AND NOT UNIX AND NOT ALPS_FOR_VISTRAILS AND NOT ALPS_BUILD_SOURCE)
-  # Lapack
-  if(LAPACK_LIBRARIES)
-    foreach(file ${LAPACK_LIBRARIES})
-      get_filename_component(LAPACK_PATH ${file} PATH)
-    endforeach(file ${LAPACK_LIBRARIES})
-    add_custom_target(liblapack ALL
-      COMMAND cmake -E make_directory ${PROJECT_BINARY_DIR}/lib
-      COMMAND cmake -E copy_if_different ${LAPACK_PATH}/lapack.lib ${PROJECT_BINARY_DIR}/lib
-      COMMAND cmake -E copy_if_different ${LAPACK_PATH}/blas.lib ${PROJECT_BINARY_DIR}/lib
-      #COMMAND cmake -E copy_if_different ${LAPACK_PATH}/f2c.lib ${PROJECT_BINARY_DIR}/lib
-    )
-    install(FILES ${PROJECT_BINARY_DIR}/lib/lapack.lib ${PROJECT_BINARY_DIR}/lib/blas.lib
-            DESTINATION lib COMPONENT dependentlibraries
-#            ${PROJECT_BINARY_DIR}/lib/f2c.lib
-    )
-  endif(LAPACK_LIBRARIES)
-  # SQLite
-  if(SQLite_LIBRARIES)
-    get_filename_component(SQLite_PATH ${SQLite_LIBRARIES} PATH)
-    get_filename_component(SQLite_BASE ${SQLite_PATH} PATH)
-    get_filename_component(SQLite_NAME ${SQLite_LIBRARIES} NAME_WE)
-    add_custom_target(libsqlite ALL
-      COMMAND cmake -E make_directory ${PROJECT_BINARY_DIR}/lib
-      COMMAND cmake -E copy_if_different ${SQLite_LIBRARIES} ${PROJECT_BINARY_DIR}/lib
-      COMMAND cmake -E copy_if_different ${SQLite_BASE}/bin/${SQLite_NAME}.dll ${PROJECT_BINARY_DIR}/bin
-    )
-    install(FILES ${PROJECT_BINARY_DIR}/lib/${SQLite_NAME}.lib DESTINATION lib COMPONENT dependentlibraries)
-    install(FILES ${PROJECT_BINARY_DIR}/bin/${SQLite_NAME}.dll DESTINATION bin COMPONENT dependentlibraries)
-  endif(SQLite_LIBRARIES)
-  # alps & boost
-  # add_custom_target(libalps ALL
-    # COMMAND cmake -E make_directory ${PROJECT_BINARY_DIR}/bin
-    # COMMAND cmake -E make_directory ${PROJECT_BINARY_DIR}/lib
-    # COMMAND cmake -E make_directory ${PROJECT_BINARY_DIR}/lib/xml
-    # COMMAND cmake -E copy_if_different ${CMAKE_INSTALL_PREFIX}/bin/archive.exe ${PROJECT_BINARY_DIR}/bin
-    # COMMAND cmake -E copy_if_different ${CMAKE_INSTALL_PREFIX}/bin/parameter2xml.exe ${PROJECT_BINARY_DIR}/bin
-    # COMMAND cmake -E copy_if_different ${CMAKE_INSTALL_PREFIX}/bin/parameter2hdf5.exe ${PROJECT_BINARY_DIR}/bin
-    # COMMAND cmake -E copy_if_different ${CMAKE_INSTALL_PREFIX}/bin/ising.exe ${PROJECT_BINARY_DIR}/bin
-    # COMMAND cmake -E copy_directory ${CMAKE_INSTALL_PREFIX}/lib/xml ${PROJECT_BINARY_DIR}/lib/xml    
-  # )
-  if (ALPS_USE_VISTRAILS)
-    install(FILES ${VISTRAILS_APP_DIR}/${VISTRAILS_APP_NAME}/${VISTRAILS_PYTHON_DIR}/python27.dll DESTINATION bin COMPONENT dependentlibraries)
-    install(FILES ${VISTRAILS_APP_DIR}/${VISTRAILS_APP_NAME}/msvcr71.dll DESTINATION bin COMPONENT dependentlibraries)
-    install(FILES ${VISTRAILS_APP_DIR}/${VISTRAILS_APP_NAME}/msvcr80.dll DESTINATION bin COMPONENT dependentlibraries)
-    install(FILES ${VISTRAILS_APP_DIR}/${VISTRAILS_APP_NAME}/hdf5dll.dll DESTINATION bin COMPONENT dependentlibraries)
-    install(FILES ${VISTRAILS_APP_DIR}/${VISTRAILS_APP_NAME}/szlibdll.dll DESTINATION bin COMPONENT dependentlibraries)
-    install(FILES ${VISTRAILS_APP_DIR}/${VISTRAILS_APP_NAME}/zlib.dll DESTINATION bin COMPONENT dependentlibraries)
-  endif (ALPS_USE_VISTRAILS)
-
-endif(WIN32 AND NOT UNIX AND NOT ALPS_FOR_VISTRAILS AND NOT ALPS_BUILD_SOURCE)
-
-######################################################################
 # setting up ALPSConfig.cmake and include.mk
 ######################################################################
 
