Index: alps/config/FindPython.cmake
===================================================================
--- alps/config/FindPython.cmake	(revision 7834)
+++ alps/config/FindPython.cmake	(working copy)
@@ -179,8 +179,9 @@
       #
       # libraries which must be linked in when embedding
       #
-      EXEC_PYTHON_SCRIPT ("${PYFUNC_NORMALIZE_FLAGS}from distutils.sysconfig import * ;print( normalize_flags( str(get_config_var('LOCALMODLIBS')) + ' ' + str(get_config_var('LIBS')) + ' ' + str(get_config_var('LDFLAGS')) ))"
-                  PYTHON_EXTRA_LIBS)
+      # EXEC_PYTHON_SCRIPT ("${PYFUNC_NORMALIZE_FLAGS}from distutils.sysconfig import * ;print( normalize_flags( str(get_config_var('LOCALMODLIBS')) + ' ' + str(get_config_var('LIBS')) + ' ' + str(get_config_var('LDFLAGS')) ))"
+      #        PYTHON_EXTRA_LIBS)
+      set(PYTHON_EXTRA_LIBS "")
       MESSAGE(STATUS "PYTHON_EXTRA_LIBS =${PYTHON_EXTRA_LIBS}" )
       mark_as_advanced(PYTHON_EXTRA_LIBS)
 
