Index: test/pyalps/mcanalyze.py
===================================================================
--- test/pyalps/mcanalyze.py	(revision 7792)
+++ test/pyalps/mcanalyze.py	(revision 7793)
@@ -32,8 +32,8 @@
 import numpy
 import pyalps
 import mcanalyze
-import matplotlib.pyplot as plt
-import pyalps.plot as alpsplt
+#import matplotlib.pyplot as plt
+#import pyalps.plot as alpsplt
 import pyalps.hdf5 as h5
 
 
