Index: alps/src/alps/python/numpy_array.hpp
===================================================================
--- alps/src/alps/python/numpy_array.hpp	(revision 7838)
+++ alps/src/alps/python/numpy_array.hpp	(revision 7839)
@@ -32,6 +32,7 @@
 #define ALPS_PYTHON_NUMPY_ARRAY
 
 #include <boost/python.hpp>
+#include <alps/config.h>
 #include <alps/python/numpy_import.hpp>
 #include <vector>
 #include <valarray>
@@ -47,17 +48,17 @@
       #endif
 
 
-      alps::python::numpy::array from_pyobject(boost::python::object const & source);
+      ALPS_DECL alps::python::numpy::array from_pyobject(boost::python::object const & source);
 
-      void convert(boost::python::object const & source, std::vector<double> & target);
+      ALPS_DECL void convert(boost::python::object const & source, std::vector<double> & target);
 
-      alps::python::numpy::array convert(double source);
+      ALPS_DECL alps::python::numpy::array convert(double source);
 
-      alps::python::numpy::array convert(std::vector<double> const & source);
+      ALPS_DECL alps::python::numpy::array convert(std::vector<double> const & source);
 
-      alps::python::numpy::array convert(std::vector<std::vector<double> > const & source);
+      ALPS_DECL alps::python::numpy::array convert(std::vector<std::vector<double> > const & source);
 
-      alps::python::numpy::array convert(std::vector<std::vector<std::vector<double> > > const & source);
+      ALPS_DECL alps::python::numpy::array convert(std::vector<std::vector<std::vector<double> > > const & source);
 
 
       // for interchanging purpose between numpy array and std::vector
