Index: src/alps/parapack/util.h
===================================================================
--- src/alps/parapack/util.h	(revision 7819)
+++ src/alps/parapack/util.h	(working copy)
@@ -35,7 +35,7 @@
 
 int hash(int n, int s = 826);
 
-std::string id2string(int id, std::string const& pad = "_");
+ALPS_DECL std::string id2string(int id, std::string const& pad = "_");
 
 ALPS_DECL double parse_percentage(std::string const& str);
 
