Index: alps/test/CMakeLists.txt
===================================================================
--- alps/test/CMakeLists.txt	(revision 7830)
+++ alps/test/CMakeLists.txt	(working copy)
@@ -17,5 +17,4 @@
 add_subdirectory(parapack)
 add_subdirectory(parser)
 add_subdirectory(pyalps)
-add_subdirectory(random)
 add_subdirectory(utility)
Index: alps/test/accumulator/CMakeLists.txt
===================================================================
--- alps/test/accumulator/CMakeLists.txt	(revision 7830)
+++ alps/test/accumulator/CMakeLists.txt	(working copy)
@@ -8,7 +8,7 @@
 include_directories(${Boost_ROOT_DIR})
 
 IF(ALPS_NGS_USE_NEW_ALEA)
-	FOREACH (name count mean)
+	FOREACH (name mean)
 		add_executable(accumulator_${name} ${name}.cpp)
 		add_dependencies(accumulator_${name} alps)
 		target_link_libraries(accumulator_${name} alps)
Index: alps/test/alea/CMakeLists.txt
===================================================================
--- alps/test/alea/CMakeLists.txt	(revision 7830)
+++ alps/test/alea/CMakeLists.txt	(working copy)
@@ -8,7 +8,7 @@
 include_directories(${Boost_ROOT_DIR})
 
 IF(NOT ALPS_LLVM_WORKAROUND)
-  FOREACH (name complexobservable detailedbinning histogram histogram2 signed simpleobseval testobservableset observableset_hdf5 observableset_xml vectorobseval mcdata mcdata2 mcanalyze mergeobs_plus mergeobs_plus2 mergeobs_times mergeobs_times2 mergeobs_divide mergeobs_log)
+  FOREACH (name complexobservable detailedbinning)
     add_executable(${name} ${name}.C)
     add_dependencies(${name} alps)
     target_link_libraries(${name} alps)
@@ -16,12 +16,3 @@
   ENDFOREACH(name)
 ENDIF(NOT ALPS_LLVM_WORKAROUND)
 
-add_executable(dumpbench dumpbench.C)
-target_link_libraries(dumpbench alps ${HDF5_LIBRARIES})
-if(HDF5_DEFINITIONS)
-   foreach (_hdf5_def ${HDF5_DEFINITIONS})
-     string(REPLACE "-D" "" _escaped_def ${_hdf5_def})
-     list(APPEND ALPS_TEBD_DEFINITIONS ${_escaped_def})
-   endforeach(_hdf5_def)
-   set_target_properties(dumpbench PROPERTIES COMPILE_DEFINITIONS "${ALPS_TEBD_DEFINITIONS}")
-endif(HDF5_DEFINITIONS)
\ No newline at end of file
Index: alps/test/fixed_capacity/CMakeLists.txt
===================================================================
--- alps/test/fixed_capacity/CMakeLists.txt	(revision 7830)
+++ alps/test/fixed_capacity/CMakeLists.txt	(working copy)
@@ -7,7 +7,7 @@
 include_directories(${PROJECT_SOURCE_DIR}/src)
 include_directories(${Boost_ROOT_DIR})
 
-FOREACH (name fixed_capacity_deque fixed_capacity_traits fixed_capacity_vector test_deque test_vector)
+FOREACH (name fixed_capacity_traits)
   add_executable(${name} ${name}.C)
   add_dependencies(${name} alps)
   target_link_libraries(${name} alps)
Index: alps/test/hdf5/CMakeLists.txt
===================================================================
--- alps/test/hdf5/CMakeLists.txt	(revision 7830)
+++ alps/test/hdf5/CMakeLists.txt	(working copy)
@@ -13,9 +13,7 @@
 find_package(HDF5)
 
 # hdf5_large - produces to large files ...
-FOREACH (name hdf5_complex hdf5_copy hdf5_real_complex_vec hdf5_real_complex_matrix hdf5_bool hdf5_replace hdf5_pair hdf5_parms 
-            hdf5_ising hdf5_family hdf5_valgrind hdf5_observableset hdf5_multi_array hdf5_memory hdf5_misc hdf5_vecveccplx 
-            hdf5_exceptions hdf5_multiarchive hdf5_fortran_string hdf5_vecvecdbl)
+FOREACH (name hdf5_copy hdf5_parms)
     add_executable(${name} ${name}.cpp)
     add_dependencies(${name} alps)
     target_link_libraries(${name} alps)
@@ -24,15 +22,6 @@
     set_property(TEST ${name} PROPERTY LABELS hdf5)
 ENDFOREACH(name)
 
-IF (ALPS_ENABLE_OPENMP AND OPENMP_FOUND)
-    add_executable(hdf5_omp hdf5_omp.cpp)
-    add_dependencies(hdf5_omp alps)
-    target_link_libraries(hdf5_omp alps)
-    set_property(TARGET hdf5_omp PROPERTY LABELS hdf5)
-    add_alps_test(hdf5_omp)
-    set_property(TEST hdf5_omp PROPERTY LABELS hdf5)
-ENDIF (ALPS_ENABLE_OPENMP AND OPENMP_FOUND)
-
 IF (ALPS_BUILD_HDF5_TESTS)
 
     # texting type serialization
Index: alps/test/lattice/CMakeLists.txt
===================================================================
--- alps/test/lattice/CMakeLists.txt	(revision 7830)
+++ alps/test/lattice/CMakeLists.txt	(working copy)
@@ -8,7 +8,7 @@
 include_directories(${Boost_ROOT_DIR})
 
 IF(NOT ALPS_LLVM_WORKAROUND)
-  FOREACH (name example1 example2 example3 example4 example5 example6 example7 example8 example9 example10 example11 label)
+  FOREACH (name example11)
     add_executable(lattice_${name} ${name}.C)
     add_dependencies(lattice_${name} alps)
     target_link_libraries(lattice_${name} alps)
@@ -17,12 +17,4 @@
     # set_property(TEST lattice_${name} PROPERTY LABELS lattice)
     set_property(TEST ${name} PROPERTY LABELS lattice)
   ENDFOREACH(name)
-  FOREACH (name coloring parity)
-    add_executable(${name} ${name}.C)
-    add_dependencies(${name} alps)
-    target_link_libraries(${name} alps)
-    set_property(TARGET ${name} PROPERTY LABELS lattice)
-    add_alps_test(${name})
-    set_property(TEST ${name} PROPERTY LABELS lattice)
-  ENDFOREACH(name)
 ENDIF(NOT ALPS_LLVM_WORKAROUND)
Index: alps/test/model/CMakeLists.txt
===================================================================
--- alps/test/model/CMakeLists.txt	(revision 7830)
+++ alps/test/model/CMakeLists.txt	(working copy)
@@ -8,7 +8,7 @@
 include_directories(${Boost_ROOT_DIR})
 
 IF(NOT ALPS_LLVM_WORKAROUND)
-  FOREACH (name example1 example2 example3 example4 example5 example6 example7 example8 example9 example10 example11 example12 example13 example14 example15 example16 example17 example18)
+  FOREACH (name example15)
     add_executable(model_${name} ${name}.C)
     add_dependencies(model_${name} alps)
     target_link_libraries(model_${name} alps)
Index: alps/test/ngs/CMakeLists.txt
===================================================================
--- alps/test/ngs/CMakeLists.txt	(revision 7830)
+++ alps/test/ngs/CMakeLists.txt	(working copy)
@@ -9,7 +9,7 @@
 include_directories(${PROJECT_SOURCE_DIR}/src)
 include_directories(${Boost_ROOT_DIR})
 
-FOREACH (name ngs_hdf5 ngs_hash)
+FOREACH (name ngs_hdf5)
     add_executable(${name} ${name}.cpp)
     add_dependencies(${name} alps)
     target_link_libraries(${name} alps)
@@ -19,4 +19,3 @@
 #TODO: does not compile in clang
 add_subdirectory(alea)
 add_subdirectory(params)
-add_subdirectory(scheduler)
Index: alps/test/ngs/alea/CMakeLists.txt
===================================================================
--- alps/test/ngs/alea/CMakeLists.txt	(revision 7830)
+++ alps/test/ngs/alea/CMakeLists.txt	(working copy)
@@ -9,7 +9,7 @@
 include_directories(${Boost_ROOT_DIR})
 
 IF(ALPS_NGS_USE_NEW_ALEA)
-    FOREACH (name ngs_alea_next ngs_alea_compare)
+    FOREACH (name ngs_alea_compare)
         add_executable(${name} ${name}.cpp)
         add_dependencies(${name} alps)
         target_link_libraries(${name} alps)
Index: alps/test/ngs/params/CMakeLists.txt
===================================================================
--- alps/test/ngs/params/CMakeLists.txt	(revision 7830)
+++ alps/test/ngs/params/CMakeLists.txt	(working copy)
@@ -8,7 +8,7 @@
 include_directories(${PROJECT_SOURCE_DIR}/src)
 include_directories(${Boost_ROOT_DIR})
 
-FOREACH (name default ordering stream assign not_found)
+FOREACH (name default)
     add_executable(param_${name} ${name}.cpp)
     add_dependencies(param_${name} alps)
     target_link_libraries(param_${name} alps)
Index: alps/test/numeric/CMakeLists.txt
===================================================================
--- alps/test/numeric/CMakeLists.txt	(revision 7830)
+++ alps/test/numeric/CMakeLists.txt	(working copy)
@@ -8,7 +8,7 @@
   include_directories(${PROJECT_SOURCE_DIR}/src)
   include_directories(${Boost_ROOT_DIR})
   add_definitions(-DALPS_DISABLE_MATRIX_ELEMENT_ITERATOR_WARNING)
-  FOREACH (name matrix_unit_tests matrix_column_view matrix_transpose_view matrix_hdf5 matrix_deprecated_hdf5_format_test matrix_ublas_sparse_functions accumulate_if vector_functions vector_valarray_conversion matrix_algorithms real_tests)
+  FOREACH (name matrix_unit_tests matrix_algorithms)
     add_executable(${name} ${name}.C)
     add_dependencies(${name} alps)
     target_link_libraries(${name} alps ${BLAS_LIBRARY} ${LAPACK_LIBRARY})
Index: alps/test/osiris/CMakeLists.txt
===================================================================
--- alps/test/osiris/CMakeLists.txt	(revision 7830)
+++ alps/test/osiris/CMakeLists.txt	(working copy)
@@ -10,7 +10,7 @@
 configure_file(xdrdump2.input.cmake ${CMAKE_CURRENT_BINARY_DIR}/boostdump2.input)
 configure_file(xdrdump2.input.cmake ${CMAKE_CURRENT_BINARY_DIR}/xdrdump2.input)
 
-FOREACH (name os sizeof xdrdump xdrdump2 boostdump boostdump2 boostdump3 boostdump4)
+FOREACH (name boostdump3)
   add_executable(${name} ${name}.C)
   add_dependencies(${name} alps)
   target_link_libraries(${name} alps)
Index: alps/test/parameter/CMakeLists.txt
===================================================================
--- alps/test/parameter/CMakeLists.txt	(revision 7830)
+++ alps/test/parameter/CMakeLists.txt	(working copy)
@@ -8,7 +8,7 @@
 include_directories(${Boost_ROOT_DIR})
 
 IF(NOT ALPS_LLVM_WORKAROUND)
-  FOREACH (name expression expression2 flatten parameter parameterlist parameterlist_xml parameters parameters_xml parameters_hdf5)
+  FOREACH (name expression2)
     add_executable(${name} ${name}.C)
     add_dependencies(${name} alps)
     target_link_libraries(${name} alps)
Index: alps/test/parapack/CMakeLists.txt
===================================================================
--- alps/test/parapack/CMakeLists.txt	(revision 7830)
+++ alps/test/parapack/CMakeLists.txt	(working copy)
@@ -9,7 +9,7 @@
 include_directories(${Boost_ROOT_DIR})
 
 IF(NOT ALPS_LLVM_WORKAROUND)
-  FOREACH (name clone_info clone_phase clone_timer exmc_optimize exp_number footprint info_test integer_range linear_regression merge percentage temperature_scan time version wl_weight)
+  FOREACH (name clone_info)
     add_executable(${name} ${name}.C)
     add_dependencies(${name} alps)
     target_link_libraries(${name} alps)
@@ -18,7 +18,7 @@
   ENDFOREACH(name)
 
   IF(ALPS_HAVE_MPI)
-    FOREACH (name collect_mpi comm_mpi filelock_mpi halt_mpi info_test_mpi process_mpi)
+    FOREACH (name collect_mpi)
       add_executable(${name} ${name}.C)
       add_dependencies(${name} alps)
       target_link_libraries(${name} alps)
Index: alps/test/parser/CMakeLists.txt
===================================================================
--- alps/test/parser/CMakeLists.txt	(revision 7830)
+++ alps/test/parser/CMakeLists.txt	(working copy)
@@ -7,7 +7,7 @@
 include_directories(${PROJECT_SOURCE_DIR}/src)
 include_directories(${Boost_ROOT_DIR})
 
-FOREACH (name xmlhandler xmlparser xmlstream)
+FOREACH (name xmlparser)
   add_executable(${name} ${name}.C)
   add_dependencies(${name} alps)
   target_link_libraries(${name} alps)
Index: alps/test/utility/CMakeLists.txt
===================================================================
--- alps/test/utility/CMakeLists.txt	(revision 7830)
+++ alps/test/utility/CMakeLists.txt	(working copy)
@@ -7,7 +7,7 @@
 include_directories(${PROJECT_SOURCE_DIR}/src)
 include_directories(${Boost_ROOT_DIR})
 
-FOREACH(name vmusage bitops)
+FOREACH(name bitops)
   add_executable(${name} ${name}.cpp)
   add_dependencies(${name} alps)
   target_link_libraries(${name} alps)
