Index: alps/src/alps/parser/xslt_path.C
===================================================================
--- alps/src/alps/parser/xslt_path.C	(revision 7830)
+++ alps/src/alps/parser/xslt_path.C	(working copy)
@@ -66,7 +66,7 @@
     if (p != 0) 
       path = boost::filesystem::path(p) / "lib" / "xml" / file;
     else {
-      path = boost::filesystem::path(ALPS_XML_DIR) / file;
+      path = boost::filesystem::path(std::string(ALPS_XML_DIR).c_str()) / file;
     #ifdef ALPS_XML_ALTERNATE_DIR
       if (!boost::filesystem::exists(path))
         path = boost::filesystem::path(ALPS_XML_ALTERNATE_DIR) / file;
