Index: src/main/cpp/ndc.cpp
===================================================================
--- src/main/cpp/ndc.cpp	(revision 1414036)
+++ src/main/cpp/ndc.cpp	(revision 1414037)
@@ -309,7 +309,7 @@
         Stack& stack = data->getStack();
         if(!stack.empty())
         {
-                dst = Transcoder::encode(stack.top().message);
+                dst = Transcoder::encode(getMessage(stack.top()));
                 stack.pop();
                 data->recycle();
                 return true;
@@ -326,7 +326,7 @@
         Stack& stack = data->getStack();
         if(!stack.empty())
         {
-                dst = Transcoder::encode(stack.top().message);
+                dst = Transcoder::encode(getMessage(stack.top()));
                 return true;
         }
         data->recycle();
Index: src/main/cpp/transcoder.cpp
===================================================================
--- src/main/cpp/transcoder.cpp	(revision 1414036)
+++ src/main/cpp/transcoder.cpp	(revision 1414037)
@@ -356,7 +356,7 @@
         unsigned char u = (unsigned char) (sv >> 16);
         unsigned char w = (unsigned char) (u - 1);
         unsigned short hs = (0xD800 + ((w & 0xF) << 6) + ((sv & 0xFFFF) >> 10));
-        unsigned short ls = (0xDC00 + (sv && 0x3FF));
+        unsigned short ls = (0xDC00 + (sv & 0x3FF));
         dst.append(1, hs);
         dst.append(1, ls);
     }
Index: src/main/include/log4cxx/helpers/simpledateformat.h
===================================================================
--- src/main/include/log4cxx/helpers/simpledateformat.h	(revision 1414036)
+++ src/main/include/log4cxx/helpers/simpledateformat.h	(revision 1414037)
@@ -29,8 +29,10 @@
 #include <vector>
 #include <time.h>
 
-namespace std { class locale; }
+#include <locale>
 
+using std::locale;
+
 namespace log4cxx
 {
         namespace helpers
Index: src/main/include/log4cxx/helpers/object.h
===================================================================
--- src/main/include/log4cxx/helpers/object.h	(revision 1414036)
+++ src/main/include/log4cxx/helpers/object.h	(revision 1414037)
@@ -61,8 +61,8 @@
 static const log4cxx::helpers::ClassRegistration&  registerClass();
 
 #define IMPLEMENT_LOG4CXX_OBJECT(object)\
-const log4cxx::helpers::Class& object::getClass() const { return getStaticClass(); }\
-const log4cxx::helpers::Class& object::getStaticClass() { \
+const ::log4cxx::helpers::Class& object::getClass() const { return getStaticClass(); }\
+const ::log4cxx::helpers::Class& object::getStaticClass() { \
    static Clazz##object theClass;                         \
    return theClass;                                       \
 }                                                                      \
@@ -71,7 +71,7 @@
     return classReg; \
 }\
 namespace log4cxx { namespace classes { \
-const log4cxx::helpers::ClassRegistration& object##Registration = object::registerClass(); \
+const ::log4cxx::helpers::ClassRegistration& object##Registration = object::registerClass(); \
 } }
 
 
Index: src/main/include/log4cxx/logstring.h
===================================================================
--- src/main/include/log4cxx/logstring.h	(revision 1414036)
+++ src/main/include/log4cxx/logstring.h	(revision 1414037)
@@ -27,7 +27,7 @@
 #include <string>
 #include <log4cxx/log4cxx.h>
 
-#if LOG4CXX_LOGCHAR_IS_WCHAR && LOG4CXX_LOGCHAR_IS_UTF8 && LOG4CXX_LOGCHAR_IS_UNICHAR
+#if (LOG4CXX_LOGCHAR_IS_WCHAR + LOG4CXX_LOGCHAR_IS_UTF8 + LOG4CXX_LOGCHAR_IS_UNICHAR)>1
 #error only one of LOG4CXX_LOGCHAR_IS_WCHAR, LOG4CXX_LOGCHAR_IS_UTF8 or LOG4CXX_LOGCHAR_IS_UNICHAR may be true
 #endif
 
Index: pom.xml
===================================================================
--- pom.xml	(revision 1414036)
+++ pom.xml	(revision 1414037)
@@ -55,6 +55,7 @@
 </organization>
 <build>
     <plugins>
+    <!--
       <plugin>
         <artifactId>maven-antrun-plugin</artifactId>
         <executions>
@@ -166,6 +167,7 @@
           </dependency>
         </dependencies>
       </plugin>  
+      -->
           <plugin>
          <artifactId>maven-assembly-plugin</artifactId>
          <configuration>