From dab4f88222a679e75aa9f509e6d03a8b68b53f11 Mon Sep 17 00:00:00 2001
From: ndana <nathan.dana@radiantsolutions.com>
Date: Tue, 14 Jul 2020 12:35:51 -0400
Subject: [PATCH] fix RoBERTa.py

---
 sentence_transformers/models/RoBERTa.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sentence_transformers/models/RoBERTa.py b/sentence_transformers/models/RoBERTa.py
index b3fbb3d..72fd202 100755
--- a/sentence_transformers/models/RoBERTa.py
+++ b/sentence_transformers/models/RoBERTa.py
@@ -1,6 +1,6 @@
 from torch import Tensor
 from torch import nn
-from pytorch_transformers import RobertaModel, RobertaTokenizer
+from transformers import RobertaModel, RobertaTokenizer
 import json
 from typing import Union, Tuple, List, Dict
 import os
-- 
2.17.0

