From 35b51fb59cdd70a95fa3904aded02302ded63c0f Mon Sep 17 00:00:00 2001
From: ndana <nathan.dana@radiantsolutions.com>
Date: Tue, 14 Jul 2020 12:33:08 -0400
Subject: [PATCH] fix SentenceTransformer.py

---
 sentence_transformers/SentenceTransformer.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sentence_transformers/SentenceTransformer.py b/sentence_transformers/SentenceTransformer.py
index 548b456..0f0eea9 100755
--- a/sentence_transformers/SentenceTransformer.py
+++ b/sentence_transformers/SentenceTransformer.py
@@ -7,7 +7,7 @@ from typing import List, Dict, Tuple, Iterable, Type
 from zipfile import ZipFile
 
 import numpy as np
-import pytorch_transformers
+import transformers as pytorch_transformers
 import torch
 from numpy import ndarray
 from torch import nn, Tensor
-- 
2.17.0

