From 0fc30e05c38d52be8c776974168bdc40a2156bde Mon Sep 17 00:00:00 2001
From: ndana <nathan.dana@radiantsolutions.com>
Date: Tue, 14 Jul 2020 12:36:23 -0400
Subject: [PATCH] fix XLNet.py

---
 sentence_transformers/models/XLNet.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sentence_transformers/models/XLNet.py b/sentence_transformers/models/XLNet.py
index 97bd59b..ddeaeee 100755
--- a/sentence_transformers/models/XLNet.py
+++ b/sentence_transformers/models/XLNet.py
@@ -1,6 +1,6 @@
 from torch import Tensor
 from torch import nn
-from pytorch_transformers import XLNetModel, XLNetTokenizer
+from transformers import XLNetModel, XLNetTokenizer
 import json
 from typing import Union, Tuple, List, Dict
 import os
-- 
2.17.0

