diff --git a/allennlp/run.py b/allennlp/run.py
index ef1ca6c..8b0e44b 100755
--- a/allennlp/run.py
+++ b/allennlp/run.py
@@ -14,5 +14,8 @@ logging.basicConfig(format='%(asctime)s - %(levelname)s - %(name)s - %(message)s
 
 from allennlp.commands import main  # pylint: disable=wrong-import-position
 
-if __name__ == "__main__":
+def run():
     main(prog="allennlp")
+
+if __name__ == "__main__":
+    run()
diff --git a/bin/allennlp b/bin/allennlp
deleted file mode 100755
index c8a2d22..0000000
--- a/bin/allennlp
+++ /dev/null
@@ -1,2 +0,0 @@
-#!/bin/sh
-python -m allennlp.run "$@"
diff --git a/setup.py b/setup.py
index 897b25a..54bf48f 100644
--- a/setup.py
+++ b/setup.py
@@ -135,7 +135,11 @@ setup(name='allennlp',
           'sqlparse==0.2.4',
           'pytorch-pretrained-bert==0.3.0',
       ],
-      scripts=["bin/allennlp"],
+      entry_points={
+          'console_scripts': [
+              "allennlp=allennlp.run:run"
+          ]
+      },
       setup_requires=setup_requirements,
       tests_require=[
           'pytest',
