From cab7f9d429c88ff924bb39788f32862c372ad6be Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Wed, 3 Jul 2024 12:03:26 +0200
Subject: [PATCH] add requirements

---
 requirements | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 requirements

diff --git a/requirements b/requirements
new file mode 100644
index 0000000..561f3ca
--- /dev/null
+++ b/requirements
@@ -0,0 +1,4 @@
+lxml
+tifffile>2020.10.1
+numpy
+imagecodecs
