diff -u a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-12-15 10:44:00.000000000 -0500
+++ b/CMakeLists.txt	2020-06-24 20:16:44.649798357 -0400
@@ -592,7 +592,10 @@
   IF(WIN32)
     # install into place in build-dir
     SET( CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/INSTALL/FILES CACHE FILEPATH "install prefix" FORCE)
-    INCLUDE (win_bundle)
+    # INCLUDE (win_bundle)
+    # install hugin readme, license etc.
+    INSTALL(FILES AUTHORS COPYING.txt 
+            DESTINATION doc/hugin)
 	
 	#Build HuginSetup
 	OPTION (BUILD_HUGINSETUP "Build HuginSetup (NSIS Installer)" OFF)
