From 06d17219e5b7f847dfd5e4e6d2873c2712b7bff2 Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Fri, 18 Jul 2025 17:21:01 +0200
Subject: [PATCH] Skip build of compiler-rt for `buildbot_linux,no_test`

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 utils/build-presets.ini | 1 +
 1 file changed, 1 insertion(+)

diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 81fa54a65e6..0f38487b8fa 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -963,6 +963,7 @@ mixin-preset=
     buildbot_linux
     mixin_buildbot_linux,no_test
 
+skip-build-compiler-rt
 
 [preset: buildbot_linux,no_assertions,no_test]
 mixin-preset=
-- 
2.50.0

