--- a/utils/build-script
+++ b/utils/build-script
@@ -422,6 +422,7 @@ def apply_default_arguments(toolchain, args):
     targets = StdlibDeploymentTarget.get_targets_by_name(
         args.stdlib_deployment_targets)
     args.stdlib_deployment_targets = [
         target.name
         for target in targets
-        if (not target.platform.is_darwin or
+        if target is not None and (not target.platform.is_darwin or
             target.platform.sdk_supports_architecture(
                 target.arch, args.darwin_xcrun_toolchain, args))
     ] 