--- runtime/src/js/pgadmin.js
+++ runtime/src/js/pgadmin.js
@@ -63,5 +63,7 @@
-if (fs.existsSync('dev_config.json')) {
-  try {
-    let dev_config = JSON.parse(fs.readFileSync('dev_config.json'));
-    pythonPath = path.resolve(dev_config['pythonPath']);
-    pgadminFile = path.resolve(dev_config['pgadminFile']);
+const configFilePath = path.join(__dirname, 'dev_config.json');
+if (fs.existsSync(configFilePath)) {
+  try {
+    const configFileContent = fs.readFileSync(configFilePath, 'utf8'); // Specify encoding
+    let dev_config = JSON.parse(configFileContent);
+    pythonPath = path.resolve(__dirname, dev_config['pythonPath']);
+    pgadminFile = path.resolve(__dirname, dev_config['pgadminFile']);
