diff --git a/configure.ac b/configure.ac
index 15854fc..6fc20d9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -153,8 +153,8 @@ AM_CONDITIONAL(LAPACK_SUPPORT, test $acx_lapack_ok = yes)
 AM_CONDITIONAL(COMPLEX_SUPPORT, test $ac_cv_cxx_have_complex = yes)
 
 # Set LIBS
-test "x$BLAS_LIBS" != x && LIBS="$BLAS_LIBS $LIBS"
-test "x$LAPACK_LIBS" != x && LIBS="$LAPACK_LIBS $LIBS"
+test "x$BLAS_LIBS" != x && LIBS="$BLAS_LIBS $LIBS $FLIBS"
+test "x$LAPACK_LIBS" != x && LIBS="$LAPACK_LIBS $BLAS_LIBS $LIBS $FLIBS"
 test "x$UTIL_LIBS" != x && LIBS="$LIBS $UTIL_LIBS"
 
 echo