diff --git a/test/testmasimd.cpp b/test/testmasimd.cpp
index de4e437..d43d4e1 100644
--- a/test/testmasimd.cpp
+++ b/test/testmasimd.cpp
@@ -55,7 +55,7 @@ using std::endl;
          cout << "Vectorizd float arithmetic operation " << relstr << " test failed";                              \
             abort();                                                   \
       }                                                                \
-   cout << "Vectorized float arithmetic operation "relstr" test passed.\n";
+   cout << "Vectorized float arithmetic operation " << relstr << " test passed.\n";
 
 #define TEST_ARITHMETIC_INT( A, rel, B, C, relstr )                    \
    C = (A rel B);                                                      \
@@ -66,7 +66,7 @@ using std::endl;
          cout << "Vectorizd integer arithmetic operation " << relstr << " test failed";                              \
             abort();                                                   \
       }                                                                \
-   cout << "Vectorized integer arithmetic operation "relstr" test passed.\n";
+   cout << "Vectorized integer arithmetic operation " << relstr << " test passed.\n";
 
 void test_float(void);
 void test_double(void);
diff --git a/test/testmasimd2.cpp b/test/testmasimd2.cpp
index d675240..8d404d7 100644
--- a/test/testmasimd2.cpp
+++ b/test/testmasimd2.cpp
@@ -42,14 +42,14 @@ using std::endl;
 #define TEST_RELATION( A, rel, B, C, relstr )  \
    C = (A rel B);                              \
    for( int i=1; i<=N; ++i )                   \
-      LTL_ASSERT_( C(i), "Vectorizd relation "relstr" test failed" ); \
-   cout << "Vectorized relation "relstr" test passed.\n";
+      LTL_ASSERT_( C(i), "Vectorizd relation " << relstr << " test failed" ); \
+   cout << "Vectorized relation " << relstr << " test passed.\n";
 
 #define TEST_LOGICAL( A, rel, B, C, relstr )  \
    C = (A rel B);                              \
    for( int i=1; i<=N; ++i )                   \
-      LTL_ASSERT_( C(i)==(A(i) rel B(i)), "Vectorizd logical operation "relstr" test failed" ); \
-   cout << "Vectorized logical operation "relstr" test passed.\n";
+      LTL_ASSERT_( C(i)==(A(i) rel B(i)), "Vectorizd logical operation " << relstr << " test failed" ); \
+   cout << "Vectorized logical operation " << relstr << " test passed.\n";
 
 void test_double(void);
 void test_float(void);
