diff --git a/ltl/marray/blas.h b/ltl/marray/blas.h
index 9cc5cdb..b8dd227 100644
--- a/ltl/marray/blas.h
+++ b/ltl/marray/blas.h
@@ -47,7 +47,7 @@ extern "C"
     *  We provide our own BLAS prototypes, since finding the right header file
     *  on all supported systems and BLAS versions is a major pain in the ass.
     */
-   double sdot_(const int *n, const float *x, const int *incx, const float *y,
+   float sdot_(const int *n, const float *x, const int *incx, const float *y,
                 const int *incy);
 
    double ddot_(const int *n, const double *x, const int *incx, const double *y,
@@ -183,7 +183,7 @@ template <> struct blas_dot_dispatch<double>
 // FLOAT
 template <> struct blas_dot_dispatch<float>
 {
-   typedef double value_type;
+   typedef float value_type;
    static inline float call(const int *n, const float *x, const int *incx,
                             const float *y, const int *incy)
    {
