--- CMakeLists.txt
+++ CMakeLists.txt
@@ -40,8 +40,8 @@
     set(FOONATHAN_MEMORY_ARCHIVE_INSTALL_DIR   "lib")
     set(FOONATHAN_MEMORY_FRAMEWORK_INSTALL_DIR "bin")
 
-    set(FOONATHAN_MEMORY_CMAKE_CONFIG_INSTALL_DIR "cmake")
-    set(FOONATHAN_MEMORY_ADDITIONAL_FILES_INSTALL_DIR "./")
+    set(FOONATHAN_MEMORY_CMAKE_CONFIG_INSTALL_DIR "share/foonathan_memory/cmake")
+    set(FOONATHAN_MEMORY_ADDITIONAL_FILES_INSTALL_DIR "share/foonathan_memory")
     set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 else()
 	message(FATAL_ERROR "Could not set install folders for this platform!")
