diff --git a/pywildcard.py b/pywildcard.py
index 44b32ec..47d1aa6 100644
--- a/pywildcard.py
+++ b/pywildcard.py
@@ -92,6 +92,7 @@ def translate(pat):
 
     There is no way to quote meta-characters.
     """
+    import os
     i, n = 0, len(pat)
     res = ''
     while i < n:
@@ -102,7 +103,7 @@ def translate(pat):
                 res = res + '.*?'
                 i = i + 1
             else:
-                res = res + r'[^\/]+'
+                res = res + '[^\\' + os.sep + ']+'
         elif c == '?':
             res = res + '.'
         elif c == '[':
diff --git a/test_pywildcard.py b/test_pywildcard.py
index 94a05d1..f451ef6 100644
--- a/test_pywildcard.py
+++ b/test_pywildcard.py
@@ -1,3 +1,4 @@
+import os
 import re
 
 import fnmatch as old
@@ -39,7 +40,7 @@ def test_translate():
     assert regex == '(?ms)example/.*?$'
     assert re.findall(regex, '\n'.join(PATHS2)) == PATHS2
     regex = translate('example/*/*.py')
-    assert regex == '(?ms)example/[^\\/]+/[^\\/]+\\.py$'
+    assert regex == '(?ms)example/[^\\' + os.sep + ']+/[^\\' + os.sep + ']+\\.py$'
     assert re.findall(regex, '\n'.join(PATHS2)) == PATHS2[1:3]
 
 
