diff --git a/setup.py b/setup.py
index 32d78e7..f040705 100644
--- a/setup.py
+++ b/setup.py
@@ -200,7 +200,26 @@ def compile_ne2001(fcomp='gfortran', ar_flags='rc'):
 do_fortran_compile = True
 
 if do_fortran_compile:
-    compile_ne2001()
+    # try and find fortran compiler
+    if os.getenv("F77") is not None:
+        fcomp = os.getenv("F77")
+    else:
+        try:
+            import numpy.distutils.fcompiler as fc
+            
+            for ftype in ["gfortran", "f77"]:
+                fcomp = fc.find_executable(ftype)
+                if fcomp is not None:
+                    break
+        except Exception as e:
+            print("Could not find gfortran compiler: {}".format(e), file=sys.stderr)
+            fcomp = None
+            pass
+
+        if fcomp is None:
+            fcomp = "gfortran"
+
+    compile_ne2001(fcomp=fcomp)
 
 ne2001_shared_objs = list(glob.glob(NE_SRC_PATH + '/*.so'))
 for sobj in ne2001_shared_objs:

