From 94ecd765052da8ff0aa47144bd41ac941c3d3868 Mon Sep 17 00:00:00 2001
From: brendanf <brendan.furneaux@gmail.com>
Date: Thu, 8 Feb 2024 14:20:41 +0200
Subject: [PATCH 2/2] don't try to build dependencies

---
 configure.ucrt | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/configure.ucrt b/configure.ucrt
index d9eb51d0..b59f78e9 100755
--- a/configure.ucrt
+++ b/configure.ucrt
@@ -1,22 +1,2 @@
-# Library versions
-LIB_VER="f338f50"
-TLS_VER="edb8fec"
-
-tar -xf src/mbedtls-$TLS_VER.tar.xz
-cd mbedtls-$TLS_VER
-echo "Compiling 'libmbedtls' from source ..."
-cmake -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX=../install .
-cmake --build . --target install
-cd ..
-rm -rf mbedtls-$TLS_VER
-
-tar -xf src/nng-$LIB_VER.tar.xz
-cd nng-$LIB_VER
-echo "Compiling 'libnng' from source ..."
-cmake -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX=../install .
-cmake --build . --target install
-cd ..
-rm -rf nng-$LIB_VER
-
 # Success
 exit 0
-- 
2.31.1

