From cca0c41adcc8a3050a4fee8a16907d816c2e176b Mon Sep 17 00:00:00 2001
From: brendanf <brendan.furneaux@gmail.com>
Date: Mon, 12 Feb 2024 12:51:27 +0200
Subject: [PATCH 2/2] check win include dir

---
 src/Makevars.win | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/Makevars.win b/src/Makevars.win
index 29130285..c836408f 100644
--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -13,3 +13,11 @@ winlibs:
 	curl -sL https://github.com/shikokuchuo/rwinlib/archive/refs/tags/v1.7.0_3.5.1.zip -o nng.zip
 	unzip -q nng.zip
 	rm -f nng.zip
+	pwd
+	ls
+	echo src-${ARC}
+	ls src-${ARC}
+	echo src-${ARC}/rwinlib-1.7.0_3.5.1
+	ls src-${ARC}/rwinlib-1.7.0_3.5.1
+	echo src-${ARC}/rwinlib-1.7.0_3.5.1/include
+	ls src-${ARC}/rwinlib-1.7.0_3.5.1/include
-- 
2.31.1

