From d3f1c8b4428058089ac0e748be549cac4b5e4111 Mon Sep 17 00:00:00 2001
From: brendanf <brendan.furneaux@gmail.com>
Date: Thu, 8 Feb 2024 17:14:38 +0200
Subject: [PATCH 4/4] windows include path

---
 src/Makevars.win | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makevars.win b/src/Makevars.win
index 4fd01ea3..b50d9d2a 100644
--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -1,2 +1,2 @@
-PKG_CFLAGS=$(C_VISIBILITY)
+PKG_CFLAGS=-I$(LIBRARY_INC) $(C_VISIBILITY)
 PKG_LIBS=-lnng -lmbedtls -lmbedx509 -lmbedcrypto -lbcrypt -lws2_32
-- 
2.31.1

