diff --git a/configure b/configure
index 3d0247e6..9391f32a 100755
--- a/configure
+++ b/configure
@@ -34,20 +34,10 @@ then
   PKG_LIBS="-L$LIB_DIR $PKG_LIBS"
   echo "Found INCLUDE_DIR $INCLUDE_DIR"
   echo "Found LIB_DIR $LIB_DIR"
-elif [ -d "/usr/local/include/mbedtls" ]
+elif [ -d "$PREFIX/include/mbedtls" ]
 then
-  PKG_CFLAGS="-I/usr/local/include $PKG_CFLAGS"
-  PKG_LIBS="-L/usr/local/lib $PKG_LIBS"
-  echo "Found 'libmbedtls' $PKG_CFLAGS"
-elif [ -d "/usr/include/mbedtls" ]
-then
-  PKG_CFLAGS="-I/usr/include $PKG_CFLAGS"
-  PKG_LIBS="-L/usr/lib $PKG_LIBS"
-  echo "Found 'libmbedtls' $PKG_CFLAGS"
-elif [ -d "/usr/local/opt/mbedtls" ]
-then
-  PKG_CFLAGS="-I/usr/local/opt/mbedtls/include $PKG_CFLAGS"
-  PKG_LIBS="-L/usr/local/opt/mbedtls/lib $PKG_LIBS"
+  PKG_CFLAGS="-I$PREFIX/include $PKG_CFLAGS"
+  PKG_LIBS="-L$PREFIX/lib $PKG_LIBS"
   echo "Found 'libmbedtls' $PKG_CFLAGS"
 fi
 
@@ -101,18 +91,10 @@ if [ "$INCLUDE_DIR" ] || [ "$LIB_DIR" ]
 then
   NNG_CFLAGS="-I$INCLUDE_DIR"
   NNG_LIBS="-L$LIB_DIR"
-elif [ -d "/usr/local/include/nng" ]
-then
-  NNG_CFLAGS="-I/usr/local/include"
-  NNG_LIBS="-L/usr/local/lib"
-elif [ -d "/usr/include/nng" ]
-then
-  NNG_CFLAGS="-I/usr/include"
-  NNG_LIBS="-L/usr/lib"
-elif [ -d "/usr/local/opt/nng" ]
+elif [ -d "$PREFIX/include/nng" ]
 then
-  NNG_CFLAGS="-I/usr/local/opt/nng/include"
-  NNG_LIBS="-L/usr/local/opt/nng/lib"
+  NNG_CFLAGS="-I$PREFIX/include"
+  NNG_LIBS="-L$PREFIX/lib"
 fi
 
 echo "#include <nng/nng.h>
