diff -ur a/configure b/configure
--- a/configure	2021-01-20 14:07:31.000000000 +0000
+++ b/configure	2021-02-28 19:59:32.405384898 +0000
@@ -4977,7 +4977,7 @@
 MPD_PUSE=
 CONFIG_UNIVERSAL_AIX=
 case $CC in
-    *gcc*)
+    *gcc*|*gnu-cc*)
         CONFIG_UNIVERSAL_AIX=$CONFIG_UNIVERSAL_AIX_GCC
         MPD_PTHREAD="-pthread"
         MPD_WARN="-Wall -Wextra -Wno-unknown-pragmas -std=c99 -pedantic"
@@ -4985,6 +4985,8 @@
         MPD_OPT="-DNDEBUG -O2"
         MPD_PGEN="-fprofile-generate -fprofile-values"
         MPD_PUSE="-fprofile-use -freorder-blocks"
+        LD="$CC"
+        LDXX="$CXX"
         ;;
     *icc*)
         AR=xiar
diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2021-01-20 14:07:31.000000000 +0000
+++ b/configure.ac	2021-02-28 19:56:59.148988553 +0000
@@ -464,7 +464,7 @@
 MPD_PUSE=
 CONFIG_UNIVERSAL_AIX=
 case $CC in
-    *gcc*)
+    *gcc*|*gnu-cc*)
         CONFIG_UNIVERSAL_AIX=$CONFIG_UNIVERSAL_AIX_GCC
         MPD_PTHREAD="-pthread"
         MPD_WARN="-Wall -Wextra -Wno-unknown-pragmas -std=c99 -pedantic"
@@ -472,6 +472,8 @@
         MPD_OPT="-DNDEBUG -O2"
         MPD_PGEN="-fprofile-generate -fprofile-values"
         MPD_PUSE="-fprofile-use -freorder-blocks"
+        LD="$CC"
+        LDXX="$CXX"
         ;;
     *icc*)
         AR=xiar
