diff -ur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2021-01-20 14:07:31.000000000 +0000
+++ b/Makefile.in	2021-02-28 20:41:07.560318569 +0000
@@ -103,8 +103,6 @@
 	$(INSTALL) -m 644 libmpdec/$(LIBIMPORT) $(DESTDIR)$(libdir)
 
 install_doc:
-	$(INSTALL) -d -m 755 $(DESTDIR)$(docdir)
-	cp -Rp doc/libmpdec $(DESTDIR)$(docdir)
 
 install_libcxx: libcxx install_lib
 	$(INSTALL) -d -m 755 $(DESTDIR)$(includedir)
@@ -117,7 +115,6 @@
 	cd $(DESTDIR)$(libdir) && ln -sf $(LIBSHARED_CXX) $(LIBSONAME_CXX) && ln -sf $(LIBSHARED_CXX) $(LIBNAME_CXX)
 
 install_cxx_doc: install_doc
-	cp -Rp doc/libmpdec++ $(DESTDIR)$(docdir)
 
 
 clean:
