diff -ur a/vcbuild/pgobuild32.bat b/vcbuild/pgobuild32.bat
--- a/vcbuild/pgobuild32.bat	2021-01-20 15:07:31.000000000 +0100
+++ b/vcbuild/pgobuild32.bat	2021-03-02 19:41:42.469312082 +0100
@@ -1,9 +1,9 @@
 @ECHO off
 
-"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -property installationPath > vcpath.txt
-set /p vcpath=<vcpath.txt
-del vcpath.txt
-call "%vcpath%\VC\Auxiliary\Build\vcvarsall.bat" x86
+rem "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -property installationPath > vcpath.txt
+rem set /p vcpath=<vcpath.txt
+rem del vcpath.txt
+rem call "%vcpath%\VC\Auxiliary\Build\vcvarsall.bat" x86
 
 if not exist dist32 mkdir dist32
 if exist dist32\* del /q dist32\*
diff -ur a/vcbuild/pgobuild64.bat b/vcbuild/pgobuild64.bat
--- a/vcbuild/pgobuild64.bat	2021-01-20 15:07:31.000000000 +0100
+++ b/vcbuild/pgobuild64.bat	2021-03-02 19:41:30.525088316 +0100
@@ -1,9 +1,9 @@
 @ECHO off
 
-"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -property installationPath > vcpath.txt
-set /p vcpath=<vcpath.txt
-del vcpath.txt
-call "%vcpath%\VC\Auxiliary\Build\vcvarsall.bat" x64
+rem "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -property installationPath > vcpath.txt
+rem set /p vcpath=<vcpath.txt
+rem del vcpath.txt
+rem call "%vcpath%\VC\Auxiliary\Build\vcvarsall.bat" x64
 
 if not exist dist64 mkdir dist64
 if exist dist64\* del /q dist64\*
diff -ur a/vcbuild/vcbuild32.bat b/vcbuild/vcbuild32.bat
--- a/vcbuild/vcbuild32.bat	2021-01-20 15:07:31.000000000 +0100
+++ b/vcbuild/vcbuild32.bat	2021-03-02 19:40:54.452364957 +0100
@@ -14,10 +14,10 @@
 exit /b 1
 :success
 
-"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -property installationPath > vcpath.txt
-set /p vcpath=<vcpath.txt
-del vcpath.txt
-call "%vcpath%\VC\Auxiliary\Build\vcvarsall.bat" x86
+rem "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -property installationPath > vcpath.txt
+rem set /p vcpath=<vcpath.txt
+rem del vcpath.txt
+rem call "%vcpath%\VC\Auxiliary\Build\vcvarsall.bat" x86
 
 if not exist dist32 mkdir dist32
 if exist dist32\* del /q dist32\*
diff -ur a/vcbuild/vcbuild64.bat b/vcbuild/vcbuild64.bat
--- a/vcbuild/vcbuild64.bat	2021-01-20 15:07:31.000000000 +0100
+++ b/vcbuild/vcbuild64.bat	2021-03-02 19:40:37.560000589 +0100
@@ -13,10 +13,10 @@
 exit /b 1
 :success
 
-"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -property installationPath > vcpath.txt
-set /p vcpath=<vcpath.txt
-del vcpath.txt
-call "%vcpath%\VC\Auxiliary\Build\vcvarsall.bat" x64
+rem "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere" -latest -property installationPath > vcpath.txt
+rem set /p vcpath=<vcpath.txt
+rem del vcpath.txt
+rem call "%vcpath%\VC\Auxiliary\Build\vcvarsall.bat" x64
 
 if not exist dist64 mkdir dist64
 if exist dist64\* del /q dist64\*
