diff --git a/iothub_client/CMakeLists.txt b/iothub_client/CMakeLists.txt
index d04f7bd7d..a9baf9872 100644
--- a/iothub_client/CMakeLists.txt
+++ b/iothub_client/CMakeLists.txt
@@ -332,7 +332,7 @@ if(${use_amqp})
             ${iothub_client_amqp_ws_transport_c_files}
             ${iothub_client_amqp_ws_transport_h_files})
 
-        set(iothub_client_libs ${iothub_client_libs} uamqp)
+        set(iothub_client_libs ${iothub_client_libs})
     endif(build_as_dynamic)
 endif()
 
@@ -431,7 +431,7 @@ if (${build_as_dynamic})
     if (${use_prov_client_core})
         target_link_libraries(iothub_client_dll hsm_security_client prov_auth_client)
     endif()
-    target_link_libraries(iothub_client_dll parson)
+    target_link_libraries(iothub_client_dll parson uamqp)
 
     if (${CMAKE_C_COMPILER_ID} STREQUAL "GNU" OR ${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
         target_link_libraries(iothub_client_dll
@@ -448,7 +448,7 @@ add_library(iothub_client
 )
 setSdkTargetBuildProperties(iothub_client)
 target_link_libraries(iothub_client ${iothub_client_libs})
-target_link_libraries(iothub_client parson)
+target_link_libraries(iothub_client parson uamqp)
 
 if (${use_prov_client_core})
     target_link_libraries(iothub_client hsm_security_client prov_auth_client)
diff --git a/iothub_service_client/CMakeLists.txt b/iothub_service_client/CMakeLists.txt
index e47473e3a..844b8c41c 100644
--- a/iothub_service_client/CMakeLists.txt
+++ b/iothub_service_client/CMakeLists.txt
@@ -109,7 +109,7 @@ if(${use_installed_dependencies})
         set(CMAKE_INSTALL_INCLUDEDIR "include")
     endif()
 
-    install(TARGETS iothub_service_client EXPORT azure_iot_sdksTargets
+    install(TARGETS ${install_libs} EXPORT azure_iot_sdksTargets
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
         RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}/../bin
diff --git a/serializer/CMakeLists.txt b/serializer/CMakeLists.txt
index 419f4d7b6..43f2b5cd3 100644
--- a/serializer/CMakeLists.txt
+++ b/serializer/CMakeLists.txt
@@ -77,7 +77,7 @@ if (${build_as_dynamic})
         serializer_dll SHARED ${serializer_c_files} ${serializer_h_files} ./src/serializer.def
     )
     target_link_libraries(serializer_dll
-        aziotsharedutil_dll
+        aziotsharedutil
         parson
     )
 
@@ -126,7 +126,7 @@ if(${use_installed_dependencies})
         set(CMAKE_INSTALL_LIBDIR "lib")
     endif()
 
-    install(TARGETS serializer EXPORT azure_iot_sdksTargets
+    install(TARGETS ${install_libs} EXPORT azure_iot_sdksTargets
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
         RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}/../bin
