diff -rupN gimagereader-3.3.1/CMakeLists.txt gimagereader-3.3.1-new/CMakeLists.txt
--- gimagereader-3.3.1/CMakeLists.txt	2019-07-28 22:59:40.000000000 +0200
+++ gimagereader-3.3.1-new/CMakeLists.txt	2020-07-29 10:33:51.491682621 +0200
@@ -1,4 +1,8 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.11)
+IF(${CMAKE_VERSION} VERSION_GREATER "3.17.0")
+CMAKE_POLICY(SET CMP0100 NEW)
+ENDIF()
+
 LIST(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 INCLUDE(GSettings)
 SET(ENV{LC_ALL} "C")
