diff --git a/CMakeLists.txt b/CMakeLists.txt
index 397e5dc..93a7693 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -178,8 +178,8 @@ ELSEIF("${INTERFACE_TYPE}" STREQUAL "qt5")
     FIND_PACKAGE(Qt5PrintSupport REQUIRED)
     FIND_PACKAGE(QuaZip5)
     IF(NOT QUAZIP_FOUND)
-        FIND_PATH(QUAZIP_INCLUDE_DIR NAME quazipfile.h PATH_SUFFIXES quazip5)
-        FIND_LIBRARY(QUAZIP_LIBRARIES quazip5)
+        FIND_PATH(QUAZIP_INCLUDE_DIR NAME quazipfile.h PATH_SUFFIXES quazip5 QuaZip-Qt5-1.0/quazip QuaZip-Qt5-1.1/quazip)
+        FIND_LIBRARY(QUAZIP_LIBRARIES NAMES quazip5 quazip1-qt5)
         IF(NOT QUAZIP_INCLUDE_DIR OR NOT QUAZIP_LIBRARIES)
             MESSAGE(FATAL_ERROR "QuaZip5 is required but was not found")
         ENDIF()
