From 853bf2a20ec2be551ead786c1e12f5ce7f571f1b Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 6 Feb 2021 17:20:34 +0100
Subject: [PATCH] Update Makefile

---
 make/Makefile | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/make/Makefile b/make/Makefile
index 56da73b..ddd0f29 100644
--- a/make/Makefile
+++ b/make/Makefile
@@ -19,9 +19,9 @@ SRCLMP 		= $(SRCPATH)external/MLIP4LAMMPS/lammps_interface.cpp
 HEADERSMLP 	= $(wildcard $(SRCPATH)*.h) $(wildcard $(SRCPATH)common/*.h) $(wildcard $(SRCPATH)drivers/*.h)
 HEADERSUTL 	= $(wildcard $(SRCPATH)utils/*.h)
 
-INCDIR		= ../blas/include
+INCDIR		= $(PREFIX)/include
 INCLIB		= ../blas/lib/libopenblas.a
-BLAS_LIBPATH 	= ../blas/lib
+BLAS_LIBPATH 	= $(PREFIX)/lib
 BLAS_SLIB 	= -lopenblas
 BLAS_OBJPATH 	= ../lib/blas_obj/
 
@@ -43,7 +43,7 @@ CC_SRL	= g++
 CFLAGS 		= -O2 -std=c++11 
 LINK 		= mpicxx 
 LINK_SRL	= g++
-LINKFLAGS 	= -o 
+LINKFLAGS 	= $(LDFLAGS) -o
 ARCHIVE 	= ar
 ARCHFLAG1 	= -x
 ARCHFLAG2 	= -rvs
@@ -61,7 +61,6 @@ lib: 	$(OBJCMN_SRL) $(OBJLMP)
 	rm -f $(OBJCMN_SRL) $(OBJLMP)
 
 mlp:	$(OBJCMN) $(OBJBIN)
-	(cd ../blas/OpenBLAS/ && make && make install)
 	$(LINK) $^  $(LINKFLAGS)  $(BIN) -L$(BLAS_LIBPATH) -lopenblas
 
 mlp_srl:	$(OBJCMN_SRL) $(OBJBIN_SRL)
