From 4d12a649a158d4948890de9b72b969c65ce1c690 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 6 Feb 2021 19:45:15 +0100
Subject: [PATCH] Update Makefile

---
 make/Makefile | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/make/Makefile b/make/Makefile
index 56da73b..dfe26f5 100644
--- a/make/Makefile
+++ b/make/Makefile
@@ -19,9 +19,9 @@ SRCLMP 		= $(SRCPATH)external/MLIP4LAMMPS/lammps_interface.cpp
 HEADERSMLP 	= $(wildcard $(SRCPATH)*.h) $(wildcard $(SRCPATH)common/*.h) $(wildcard $(SRCPATH)drivers/*.h)
 HEADERSUTL 	= $(wildcard $(SRCPATH)utils/*.h)
 
-INCDIR		= ../blas/include
+INCDIR		= $(PREFIX)/include
 INCLIB		= ../blas/lib/libopenblas.a
-BLAS_LIBPATH 	= ../blas/lib
+BLAS_LIBPATH 	= $(PREFIX)/lib
 BLAS_SLIB 	= -lopenblas
 BLAS_OBJPATH 	= ../lib/blas_obj/
 
@@ -40,10 +40,10 @@ BIN 		= $(BINPATH)mlp
 
 CC 		= mpicxx
 CC_SRL	= g++
-CFLAGS 		= -O2 -std=c++11 
+# CFLAGS 		= -O2 -std=c++11 
 LINK 		= mpicxx 
 LINK_SRL	= g++
-LINKFLAGS 	= -o 
+LINKFLAGS 	= $(LDFLAGS) -o
 ARCHIVE 	= ar
 ARCHFLAG1 	= -x
 ARCHFLAG2 	= -rvs
@@ -61,8 +61,7 @@ lib: 	$(OBJCMN_SRL) $(OBJLMP)
 	rm -f $(OBJCMN_SRL) $(OBJLMP)
 
 mlp:	$(OBJCMN) $(OBJBIN)
-	(cd ../blas/OpenBLAS/ && make && make install)
-	$(LINK) $^  $(LINKFLAGS)  $(BIN) -L$(BLAS_LIBPATH) -lopenblas
+	$(LINK) $^  $(LINKFLAGS)  $(BIN) -L$(BLAS_LIBPATH) -lcblas
 
 mlp_srl:	$(OBJCMN_SRL) $(OBJBIN_SRL)
 	(cd ../blas/OpenBLAS/ && make && make install)
@@ -87,4 +86,3 @@ $(OBJBIN_SRL): ADDCFLAG := -I$(INCDIR) # -DMLIP_DEBUG
 clean:
 	rm -f $(OBJCMN) $(OBJCMN_SRL) $(OBJBIN) $(OBJBIN_SRL) $(BIN) $(OBJLMP) $(SRCPATH)external/MLIP4USPEX/relax_mpi.o  \
 	$(SRCPATH)utils/*.o $(LIB) $(SRCPATH)*~ 
-
