diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e5b780e..78fd4ec 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -17,7 +17,8 @@ if(NOT WIN32)
   # Link the libraries that we don't expect to find on Windows
   target_link_libraries(${PROJECT_LIBRARY_TARGET_NAME}
     PUBLIC UUID::UUID
-    pthread)
+    pthread
+    rt)
 else()
   target_link_libraries(${PROJECT_LIBRARY_TARGET_NAME}
     PRIVATE shlwapi)
