From 8d90eee1109a79135b5858d0d7a5fcb0f6cab641 Mon Sep 17 00:00:00 2001
From: Diego <diego.ferigo@iit.it>
Date: Tue, 6 Jul 2021 15:47:04 +0200
Subject: [PATCH] Fix path in conda due to its binary relocation hacks

---
 src/SystemPaths.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/SystemPaths.cc b/src/SystemPaths.cc
index f809686..16e6e66 100644
--- a/src/SystemPaths.cc
+++ b/src/SystemPaths.cc
@@ -235,6 +235,8 @@ void SystemPaths::AddFilePaths(const std::string &_path)
 std::string SystemPaths::NormalizeDirectoryPath(const std::string &_path)
 {
   std::string path = _path;
+  path.erase(std::find(path.begin(), path.end(), '\0'), path.end());
+
   // Use '/' because it works on Linux, OSX, and Windows
   std::replace(path.begin(), path.end(), '\\', '/');
   // Make last character '/'
