From f4b62ca947dcfb1aca65ed80a3459e753579acfc Mon Sep 17 00:00:00 2001
From: Daniel Tipping <daniel@oldreliable.tech>
Date: Fri, 12 Apr 2019 16:30:13 +0100
Subject: [PATCH] Find proj_6_0.lib if proj.lib isn't found

---
 cmake/FindProj4.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/FindProj4.cmake b/cmake/FindProj4.cmake
index a41abe8..8fd9379 100644
--- a/cmake/FindProj4.cmake
+++ b/cmake/FindProj4.cmake
@@ -44,13 +44,13 @@ endif()
 
 if( PROJ4_PATH )
 
-    find_path(PROJ4_INCLUDE_DIR NAMES proj_api.h PATHS ${PROJ4_PATH} ${PROJ4_PATH}/include PATH_SUFFIXES proj4 NO_DEFAULT_PATH )
-    find_library(PROJ4_LIBRARY  NAMES proj       PATHS ${PROJ4_PATH} ${PROJ4_PATH}/lib     PATH_SUFFIXES proj4 NO_DEFAULT_PATH )
+    find_path(PROJ4_INCLUDE_DIR NAMES proj_api.h    PATHS ${PROJ4_PATH} ${PROJ4_PATH}/include PATH_SUFFIXES proj4 NO_DEFAULT_PATH )
+    find_library(PROJ4_LIBRARY  NAMES proj proj_6_1 PATHS ${PROJ4_PATH} ${PROJ4_PATH}/lib     PATH_SUFFIXES proj4 NO_DEFAULT_PATH )
 
 endif()
 
-find_path(PROJ4_INCLUDE_DIR NAMES proj_api.h PATHS PATH_SUFFIXES proj4 )
-find_library( PROJ4_LIBRARY NAMES proj       PATHS PATH_SUFFIXES proj4 )
+find_path(PROJ4_INCLUDE_DIR NAMES proj_api.h    PATHS PATH_SUFFIXES proj4 )
+find_library( PROJ4_LIBRARY NAMES proj proj_6_1 PATHS PATH_SUFFIXES proj4 )
 
 
 # handle the QUIETLY and REQUIRED arguments and set GRIBAPI_FOUND
-- 
2.21.0

