--- a/src/cbf_hdf5.c
+++ b/src/cbf_hdf5.c
@@ -283,6 +283,10 @@
 #include <assert.h>
 #include <errno.h>
 
+#ifndef HAVE_REALPATH
+char *realpath(const char *path, char *resolved_path);
+#endif
+
 #if !defined(CBF_NO_REGEX)
 #ifdef CBF_REGEXLIB_REGEX
 #include <regex.h>
--- a/src/realpath.c
+++ b/src/realpath.c
@@ -1,6 +1,19 @@
-#include <stdlib.h>
+#ifdef _WIN32
+#    include <windows.h>
+#endif
+
 char *realpath(const char *path, char *resolved_path)
 {
-    return 0;
+#ifdef _WIN32
+    if (path == NULL) {
+        errno = EINVAL;
+        return (NULL);
+    }
+
+    return (_fullpath(resolved_path, path, MAX_PATH));
+#else
+    errno = ENOSYS;
+    return (NULL);
+#endif
 }
 
