--- CMakeLists.txt
+++ CMakeLists.txt
@@ -35,16 +35,16 @@
 
 # CPACK
 if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-  install(DIRECTORY "bin/Debug/" DESTINATION "./")
+  install(DIRECTORY "bin/Debug/" DESTINATION "bin")
 else()
-  install(DIRECTORY "bin/Release/" DESTINATION "./")
+  install(DIRECTORY "bin/Release/" DESTINATION "bin")
 endif()
 
-install(DIRECTORY "custom_build_rules/" DESTINATION "./custom_build_rules/")
-install(DIRECTORY "bison/data/" DESTINATION "./data/")
-install(FILES "flex/src/FlexLexer.h" DESTINATION "./")
-install(FILES "changelog.md" DESTINATION "./")
-install(FILES "README.md" DESTINATION "./")
+install(DIRECTORY "custom_build_rules/" DESTINATION "share/winflexbison/custom_build_rules/")
+install(DIRECTORY "bison/data/" DESTINATION "share/winflexbison/data/")
+install(FILES "flex/src/FlexLexer.h" DESTINATION "include")
+install(FILES "changelog.md" DESTINATION "share/winflexbison/")
+install(FILES "README.md" DESTINATION "share/winflexbison/")
 
 set(PACKAGE_GENERATORS_DEFAULT ZIP)
 
