--- /dev/null	2022-07-14 12:55:30.000000000 +0200
+++ third_party/com_github_nlohmann_json/system.BUILD.bazel	2022-07-14 12:52:30.000000000 +0200
@@ -0,0 +1,12 @@
+load("@com_google_tensorstore//:utils.bzl", "cc_library_with_strip_include_prefix")
+
+licenses(["notice"])
+
+exports_files(["LICENSE.MIT"])
+
+cc_library_with_strip_include_prefix(
+    name = "nlohmann_json",
+    hdrs = glob(["$(CB_PREFIX)/include/**/*.hpp"]),
+    strip_include_prefix = "$(CB_PREFIX)/include",
+    visibility = ["//visibility:public"],
+)
