--- third_party/com_github_nlohmann_json/workspace.bzl.orig	2022-07-14 12:53:49.000000000 +0200
+++ third_party/com_github_nlohmann_json/workspace.bzl	2022-07-14 12:53:56.000000000 +0200
@@ -28,6 +28,7 @@
         ],
         sha256 = "b94997df68856753b72f0d7a3703b7d484d4745c567f3584ef97c96c25a5798e",
         build_file = Label("//third_party:com_github_nlohmann_json/bundled.BUILD.bazel"),
+        system_build_file = Label("//third_party:com_github_nlohmann_json/bundled.BUILD.bazel"),
     )
 
 cmake_add_dep_mapping(target_mapping = {
