--- python/tensorstore/BUILD	2022-07-18 11:20:35.000000000 +0200
+++ python/tensorstore/BUILD	2022-07-18 11:23:52.000000000 +0200
@@ -27,6 +27,10 @@
     name = "_tensorstore",
     srcs = ["tensorstore.cc"],
     imports = [".."],
+    linkopts = select({
+        "@platforms//os:macos": ["-Wl,-undefined,dynamic_lookup"],
+        "//conditions:default": [],
+    }),
     deps = [
         ":chunk_layout",
         ":context",
