--- third_party/com_github_pybind_pybind11/workspace.bzl	2022-07-14 12:01:29.000000000 +0200
+++ third_party/com_github_pybind_pybind11/workspace.bzl	2022-07-14 12:01:37.000000000 +0200
@@ -28,6 +28,7 @@
         ],
         sha256 = "6bd528c4dbe2276635dc787b6b1f2e5316cf6b49ee3e150264e455a0d68d19c1",
         build_file = Label("//third_party:com_github_pybind_pybind11/bundled.BUILD.bazel"),
+        system_build_file = Label("//third_party:com_github_pybind_pybind11/system.BUILD.bazel"),
     )

 cmake_fetch_content_package(
