From c60936644de33a2bc7e5b3e1e7e6a9a15f25fd30 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sun, 24 May 2020 21:14:47 +0000
Subject: [PATCH 02/12] Relax constraint on minimum version of GCC

---
 SConstruct |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/SConstruct b/SConstruct
index abf905d..a625431 100644
--- a/SConstruct
+++ b/SConstruct
@@ -2029,14 +2029,14 @@ def doConfigure(myenv):
         }
         """ % compiler_minimum_string)
     elif myenv.ToolchainIs('gcc'):
-        compiler_minimum_string = "GCC 8.2"
+        compiler_minimum_string = "GCC 7.5"
         compiler_test_body = textwrap.dedent(
         """
         #if !defined(__GNUC__) || defined(__clang__)
         #error
         #endif
 
-        #if (__GNUC__ < 8) || (__GNUC__ == 8 && __GNUC_MINOR__ < 2)
+        #if (__GNUC__ < 7) || (__GNUC__ == 7 && __GNUC_MINOR__ < 5)
         #error %s or newer is required to build MongoDB
         #endif
 
-- 
1.7.1

