From c010f6cccbbd77d51b07a574571079f9c371fc0e Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 25 May 2020 15:22:50 +0000
Subject: [PATCH 06/12] Make SCons inherit environment vars

---
 SConstruct |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/SConstruct b/SConstruct
index d7553d8..e2b8650 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1027,6 +1027,7 @@ envDict = dict(BUILD_ROOT=buildDir,
                )
 
 env = Environment(variables=env_vars, **envDict)
+env.Append(ENV = os.environ)
 del envDict
 
 for var in ['CC', 'CXX']:
-- 
1.7.1

