From a71efe15b0e723f63795cfd3ac04719305933a46 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 25 May 2020 16:48:04 +0000
Subject: [PATCH 08/12] Relax minimum Darwin version check to 10.9

---
 SConstruct |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/SConstruct b/SConstruct
index 2e042dd..8ab02fe 100644
--- a/SConstruct
+++ b/SConstruct
@@ -2112,7 +2112,7 @@ def doConfigure(myenv):
 
     conf.Finish()
 
-    # We require macOS 10.12 or newer
+    # We require macOS 10.9 or newer
     if env.TargetOSIs('darwin'):
 
         # TODO: Better error messages, mention the various -mX-version-min-flags in the error, and
@@ -2123,7 +2123,7 @@ def doConfigure(myenv):
             #include <AvailabilityMacros.h>
             #include <TargetConditionals.h>
 
-            #if TARGET_OS_OSX && (__MAC_OS_X_VERSION_MIN_REQUIRED < __MAC_10_12)
+            #if TARGET_OS_OSX && (__MAC_OS_X_VERSION_MIN_REQUIRED < __MAC_10_9)
             #error 1
             #endif
             """
@@ -2138,7 +2138,7 @@ def doConfigure(myenv):
         })
 
         if not conf.CheckDarwinMinima():
-            conf.env.ConfError("Required target minimum of macOS 10.12 not found")
+            conf.env.ConfError("Required target minimum of macOS 10.9 not found")
 
         conf.Finish()
 
-- 
1.7.1

