From 0c7c3e9e4471777f6dd17b912e7a4da4fa0fdf7d Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 25 May 2020 19:50:43 +0000
Subject: [PATCH 10/12] Fix openssl 1.1.x library names on Windows

---
 SConstruct |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/SConstruct b/SConstruct
index 8ab02fe..cf469eb 100644
--- a/SConstruct
+++ b/SConstruct
@@ -3006,9 +3006,9 @@ def doConfigure(myenv):
             sslLinkDependencies = ["crypto"]
 
         if conf.env.TargetOSIs('windows'):
-            sslLibName = "ssleay32"
-            cryptoLibName = "libeay32"
-            sslLinkDependencies = ["libeay32"]
+            sslLibName = "libssl"
+            cryptoLibName = "libcrypto"
+            sslLinkDependencies = ["libcrypto"]
 
         # Used to import system certificate keychains
         if conf.env.TargetOSIs('darwin'):
-- 
1.7.1

