From dba8ebbd11cffdeea425c0141420e81b7d0624fa Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 25 May 2020 20:34:47 +0000
Subject: [PATCH 11/12] Fix icu library names for Windows

---
 SConstruct |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/SConstruct b/SConstruct
index cf469eb..169d183 100644
--- a/SConstruct
+++ b/SConstruct
@@ -3268,14 +3268,14 @@ def doConfigure(myenv):
         conf.FindSysLibDep("intel_decimal128", ["bid"])
 
     if use_system_version_of_library("icu"):
-        conf.FindSysLibDep("icudata", ["icudata"])
+        conf.FindSysLibDep("icudata", ["icudt" if conf.env.TargetOSIs('windows') else "icudata"])
         # We can't use FindSysLibDep() for icui18n and icuuc below, since SConf.CheckLib() (which
         # FindSysLibDep() relies on) doesn't expose an 'extra_libs' parameter to indicate that the
         # library being tested has additional dependencies (icuuc depends on icudata, and icui18n
         # depends on both). As a workaround, we skip the configure check for these two libraries and
         # manually assign the library name. We hope that if the user has icudata installed on their
         # system, then they also have icu18n and icuuc installed.
-        conf.env['LIBDEPS_ICUI18N_SYSLIBDEP'] = 'icui18n'
+        conf.env['LIBDEPS_ICUI18N_SYSLIBDEP'] = 'icuin' if conf.env.TargetOSIs('windows') else 'icui18n'
         conf.env['LIBDEPS_ICUUC_SYSLIBDEP'] = 'icuuc'
 
     if use_system_version_of_library("abseil-cpp"):
-- 
1.7.1

