From 4803bec3624536c04b135f7e90534afa4d5a0854 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Tue, 26 May 2020 14:11:37 +0000
Subject: [PATCH 12/12] Don't force disable auto linking if --use-system-boost

---
 SConstruct |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/SConstruct b/SConstruct
index 169d183..b0f759d 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1626,11 +1626,12 @@ elif env.TargetOSIs('windows'):
         print("NOTE: Tool configuration did not find 'cl' compiler, falling back to os environment")
         env['ENV'] = dict(os.environ)
 
-    env.Append(CPPDEFINES=[
-    # This tells the Windows compiler not to link against the .lib files
-    # and to use boost as a bunch of header-only libraries
-        "BOOST_ALL_NO_LIB",
-    ])
+    if not use_system_version_of_library("boost"):
+        env.Append(CPPDEFINES=[
+        # This tells the Windows compiler not to link against the .lib files
+        # and to use boost as a bunch of header-only libraries
+            "BOOST_ALL_NO_LIB",
+        ])
 
     env.Append( CPPDEFINES=[ "_UNICODE" ] )
     env.Append( CPPDEFINES=[ "UNICODE" ] )
-- 
1.7.1

