From 0a93c182bba10adbc2b7f8617eba7b2f62d7802d Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Fri, 11 Aug 2023 14:47:08 +0200
Subject: [PATCH] Remove Python2.7 support as deprecated in SOFA.

---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d21b38..2a676e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,8 +102,6 @@ if(SoftRobots_FOUND)
     message("-- Found dependency: 'SoftRobots' plugin.")
 endif()
 
-sofa_install_pythonscripts(PLUGIN_NAME ${PROJECT_NAME} PYTHONSCRIPTS_SOURCE_DIR "python")
-
 find_file(SofaPython3Tools NAMES "SofaPython3/lib/cmake/SofaPython3/SofaPython3Tools.cmake")
 if(SofaPython3Tools)
     message("-- Found SofaPython3Tools.")
-- 
2.34.1

