From 1eb2650f1ad69be586ed904a044b9c11f21d1991 Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Fri, 11 Aug 2023 14:47:08 +0200
Subject: [PATCH] Remove Python2.7 support as deprecated in SOFA.

---
 CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d21b38..3a598e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,8 +102,6 @@ if(SoftRobots_FOUND)
     message("-- Found dependency: 'SoftRobots' plugin.")
 endif()
 
-sofa_install_pythonscripts(PLUGIN_NAME ${PROJECT_NAME} PYTHONSCRIPTS_SOURCE_DIR "python")
-
 find_file(SofaPython3Tools NAMES "SofaPython3/lib/cmake/SofaPython3/SofaPython3Tools.cmake")
 if(SofaPython3Tools)
     message("-- Found SofaPython3Tools.")
@@ -145,6 +143,6 @@ if(COSSERATPLUGIN_BUILD_TESTS)
 endif()
 
 # Config files and install rules for pythons scripts
-sofa_install_pythonscripts(PLUGIN_NAME ${PROJECT_NAME} PYTHONSCRIPTS_SOURCE_DIR "examples/python3/")
+#sofa_install_pythonscripts(PLUGIN_NAME ${PROJECT_NAME} PYTHONSCRIPTS_SOURCE_DIR "examples/python3/")
 
 include(cmake/packaging.cmake)
\ No newline at end of file
-- 
2.34.1

