From 0aed2206328ce18f91d770579b855ebb27ac7d77 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 14 Feb 2021 20:07:03 +0100
Subject: [PATCH] patch dependencies

---
 setup.py | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index 11ded37..ef66292 100755
--- a/setup.py
+++ b/setup.py
@@ -19,15 +19,15 @@
 DEPENDENCIES = (
     # Note: these dependency versions should be kept in-sync with the versions
     # specified in the docker container requirements files.
-    'google-auth~=1.4.0',
-    'ipykernel~=4.6.0',
-    'ipython~=5.5.0',
-    'notebook~=5.2.0',
-    'six~=1.12.0',
-    'pandas~=0.24.0',
-    'portpicker~=1.2.0',
-    'requests~=2.21.0',
-    'tornado~=4.5.0',
+    'google-auth>=1.4.0',
+    'ipykernel>=4.6.0',
+    'ipython>=5.5.0',
+    'notebook>=5.2.0',
+    'six>=1.12.0',
+    'pandas>=0.24.0',
+    'portpicker>=1.2.0',
+    'requests>=2.21.0',
+    'tornado>=4.5.0',
 )
 
 setup(
