From e14dc89c30125c143c4a83b9e2c101ed4ac57275 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Wed, 7 Apr 2021 11:38:27 +0200
Subject: [PATCH] add LDFLAGS to LIBS

---
 browndye2/source/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/browndye2/source/Makefile b/browndye2/source/Makefile
index ba550d0..e14eeda 100644
--- a/browndye2/source/Makefile
+++ b/browndye2/source/Makefile
@@ -37,7 +37,7 @@ ifdef BOOST
   FILELIB := $(FILELIB_BST)
 endif
 
-LIB         := -lm -lpthread -lbrowndye -lexpat -llapack $(FILELIB) 
+LIB         := $(LDFLAGS) -lm -lpthread -lbrowndye -lexpat -llapack $(FILELIB) 
 #INC         := -I$(INCDIR) 
 #INCDEP      := -I$(INCDIR)
 
-- 
2.30.2
