diff --git a/setup.py b/setup.py
index d32c116..4ec82df 100644
--- a/setup.py
+++ b/setup.py
@@ -27,6 +27,6 @@ setup(name='codecov',
       packages=['codecov'],
       include_package_data=True,
       zip_safe=True,
-      install_requires=["requests>=2.0.0", "coverage", "argparse"],
+      install_requires=["requests>=2.0.0", "coverage"],
       tests_require=["unittest2"],
       entry_points={'console_scripts': ['codecov=codecov:main']})