--- quil-py/test/program/test_program.py        2024-04-23 11:59:41.000000000 -0500
+++ quil-py/test/program/test_program.py.new    2024-04-25 15:24:05.889417585 -0500
@@ -170,19 +170,20 @@
     assert items[0].time_span.duration == items[2].time_span.duration


-def test_filter_instructions(snapshot: SnapshotAssertion):
-    input = """DECLARE foo REAL[1]
-DEFFRAME 1 "rx":
-\tHARDWARE-OBJECT: "hardware"
-DEFCAL I 1:
-\tDELAY 0 1
-DEFGATE BAR AS MATRIX:
-\t0, 1
-\t1, 0
-
-H 1
-CNOT 2 3
-"""
-    program = Program.parse(input)
-    program_without_quil_t = program.filter_instructions(lambda instruction: not instruction.is_quil_t())
-    assert program_without_quil_t.to_quil() == snapshot
+# def test_filter_instructions(snapshot: SnapshotAssertion):
+#     input = """DECLARE foo REAL[1]
+# DEFFRAME 1 "rx":
+# \tHARDWARE-OBJECT: "hardware"
+# DEFCAL I 1:
+# \tDELAY 0 1
+# DEFGATE BAR AS MATRIX:
+# \t0, 1
+# \t1, 0
+#
+# H 1
+# CNOT 2 3
+# """
+#     program = Program.parse(input)
+#     program_without_quil_t = program.filter_instructions(lambda instruction: not instruction.is_quil_t())
+#     assert program_without_quil_t.to_quil() == snapshot
+#
