--- a/setup.py
+++ b/setup.py
@@ -26,2 +26,8 @@
 from setuptools.command.build_ext import build_ext
+from build_mgclient_with_conda_lib_ext import build_mgclient_with_conda_lib_ext
+
+import distutils.util
+if platform.system() == "Windows":
+    import distutils.ccompiler
+    distutils.ccompiler.get_default_compiler = lambda: 'mingw32'

@@ -316,3 +322,3 @@
     ext_modules=[
-        Extension(EXTENSION_NAME, sources=sources, depends=headers, extra_link_args=extra_link_args)
+        build_mgclient_with_conda_lib_ext()
     ],
@@ -322,3 +333,2 @@
     },
-    cmdclass={"build_ext": BuildMgclientExt},
     install_requires=[
