--- a/setup.py
+++ b/setup.py
@@ -26,2 +26,22 @@
 from setuptools.command.build_ext import build_ext
+from build_mgclient_with_conda_lib_ext import build_mgclient_with_conda_lib_ext
+
+class GccBuildExt(build_ext):
+    def build_extensions(self):
+        if sys.platform == "win32":
+            build_prefix = os.environ.get('BUILD_PREFIX', '')
+            cc_path = os.path.join(build_prefix, 'Library', 'bin', 'x86_64-w64-mingw32-gcc.exe')
+            print(f"DEBUG: Looking for GCC at: {cc_path}")
+            if cc_path and os.path.exists(cc_path):
+                print(f"DEBUG: Looking for GCC at: {cc_path}")
+                from distutils.cygwinccompiler import Mingw32CCompiler
+                self.compiler = Mingw32CCompiler()
+                self.compiler.set_executables(
+                    compiler=cc_path,
+                    compiler_so=cc_path,
+                    linker_exe=cc_path,
+                    linker_so=cc_path
+                )
+                self.compiler.initialize()
+        super().build_extensions()

@@ -316,3 +336,3 @@
     ext_modules=[
-        Extension(EXTENSION_NAME, sources=sources, depends=headers, extra_link_args=extra_link_args)
+        build_mgclient_with_conda_lib_ext()
     ],
@@ -322,3 +342,3 @@
     },
-    cmdclass={"build_ext": BuildMgclientExt},
+    cmdclass={"build_ext": GccBuildExt},
     install_requires=[
