--- a/setup.py
+++ b/setup.py
@@ -26,2 +26,3 @@
 from setuptools.command.build_ext import build_ext
+from build_mgclient_with_conda_lib_ext import build_mgclient_with_conda_lib_ext

@@ -316,3 +317,3 @@
     ext_modules=[
-        Extension(EXTENSION_NAME, sources=sources, depends=headers, extra_link_args=extra_link_args)
+        build_mgclient_with_conda_lib_ext()
     ],
@@ -322,3 +323,1 @@
     },
-    cmdclass={"build_ext": BuildMgclientExt},
     install_requires=[
