diff --git a/src/context/simulation_context.cpp b/src/context/simulation_context.cpp
index 392a3958..ebb80e8b 100644
--- a/src/context/simulation_context.cpp
+++ b/src/context/simulation_context.cpp
@@ -629,8 +629,6 @@ void Simulation_context::print_info() const
     std::printf("maximum number of OMP threads : %i\n", omp_get_max_threads());
     std::printf("number of MPI ranks per node  : %i\n", num_ranks_per_node());
     std::printf("page size (Kb)                : %li\n", utils::get_page_size() >> 10);
-    std::printf("number of pages               : %li\n", utils::get_num_pages());
-    std::printf("available memory (GB)         : %li\n", utils::get_total_memory() >> 30);
 
     std::string headers[]         = {"FFT context for density and potential", "FFT context for coarse grid"};
     double cutoffs[]              = {pw_cutoff(), 2 * gk_cutoff()};
diff --git a/src/utils/utils.hpp b/src/utils/utils.hpp
index 611511e5..5600d143 100644
--- a/src/utils/utils.hpp
+++ b/src/utils/utils.hpp
@@ -350,16 +350,6 @@ inline long get_page_size()
     return sysconf(_SC_PAGESIZE);
 }
 
-inline long get_num_pages()
-{
-    return sysconf(_SC_PHYS_PAGES);
-}
-
-inline long get_total_memory()
-{
-    return get_page_size() * get_num_pages();
-}
-
 ///// Check if lambda F(Args) is of type T.
 //template <typename T, typename F, typename ...Args>
 //constexpr bool check_lambda_type()
