diff --git a/pyproject.toml b/pyproject.toml
index 14acd0506..701408d38 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -17,11 +17,6 @@ classifiers = [
     "Programming Language :: Python :: 3.13",
 ]
 requires-python = ">=3.8"
-dependencies = [
-    # required by the basedpyright cli & langserver wrapper scripts. only binaries are required (no cli)
-    # since the user shouldn't have a node/npm binary unknowingly added to their PATH
-    "nodejs-wheel-binaries>=20.13.1",
-]
 [dependency-groups]
 dev = [
     "pylint>=3.0.0a7",
