--- src/EditresCom.c.orig	2017-01-28 17:22:38.909762304 -0500
+++ src/EditresCom.c	2017-01-28 17:23:10.819859656 -0500
@@ -2193,11 +2193,11 @@
 	    case sizeof(short):
 		XmuSnprintf(buffer, sizeof(buffer), "%d", (int)(value & 0xffff));
 		break;
-	    case sizeof(int):
+	    case 4:
 		XmuSnprintf(buffer, sizeof(buffer), "0x%08x", (int)value);
 		break;
 #ifdef LONG64
-	    case sizeof(long):
+	    case 8:
 		XmuSnprintf(buffer, sizeof(buffer), "0x%016lx", value);
 		break;
 #endif
