--- configure.ac.orig	2017-02-06 14:59:51.284072932 -0500
+++ configure.ac	2017-02-06 15:03:39.715929946 -0500
@@ -53,7 +53,7 @@
 platform_darwin=no
 LIBEXT=so
 case $host_os in
-    cygwin*|mingw*)
+    cygwin*|mingw*|*msys*)
 	LIBEXT=dll.a
 	platform_win32=yes
 	;;
@@ -99,7 +99,7 @@
 
 # Link with winsock if mingw target
 case $host_os in
-	*mingw*)
+	*mingw*|*msys*)
 		AC_CHECK_LIB([ws2_32],[main])
 	;;
 	*)
