diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7bbdd6c1..65dcd35d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -781,10 +781,6 @@ install(FILES mapd.thrift DESTINATION ".")
 install(FILES common.thrift DESTINATION ".")
 install(FILES QueryEngine/serialized_result_set.thrift DESTINATION "QueryEngine/")
 
-if(NOT PREFER_STATIC_LIBS)
-  install(FILES ${Boost_LIBRARIES} DESTINATION ThirdParty/lib)
-endif()
-
 if("${MAPD_EDITION_LOWER}" STREQUAL "ee")
   set(EULA_FILE "${CMAKE_SOURCE_DIR}/EULA-EE.txt")
 else()
