--- lib/Makefile.in	2019-12-13 01:26:06.000000000 +0000
+++ lib/Makefile.in	2020-02-11 15:56:09.000000000 +0000
@@ -461,7 +461,7 @@
 lib_LTLIBRARIES = libgstlalinspiral.la
 libgstlalinspiral_la_SOURCES = gstlal_snglinspiral.h gstlal_snglinspiral.c snglinspiralrowtype.h snglinspiralrowtype.c
 libgstlalinspiral_la_CFLAGS = $(AM_CFLAGS) $(LAL_CFLAGS) $(GSTLAL_CFLAGS) $(gstreamer_CFLAGS)
-libgstlalinspiral_la_LDFLAGS = -version-info $(LIBVERSION) $(AM_LDFLAGS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS)
+libgstlalinspiral_la_LDFLAGS = -version-info $(LIBVERSION) $(AM_LDFLAGS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(GSL_LIBS)
 all: all-recursive

 .SUFFIXES:
