diff --git a/src/Makevars.win b/src/Makevars.win
index 22bc85c..d15cba0 100644
--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -1,5 +1,8 @@
 ## This is a C++14 package
 CXX_STD = CXX14
+CXX14FLAGS=-O3 -mtune=native -march=native -Wno-unused-variable -Wno-unused-function  -Wno-macro-redefined
+CXX14 = $(BINPREF)g++ -m$(WIN) -std=c++1y
+CXX11FLAGS=-O3 -Wno-unused-variable -Wno-unused-function
 PKG_CFLAGS = -Ilib/eclat
 PKG_CXXFLAGS = -Iinclude -Ilib -Ilib/spectra -Ilib/eigen3 -Ilib/eclat -Ilib/infomap
 PKG_CPPFLAGS = -DNS_INFOMAP -DONLY_C_LOCALE=1
