diff --git a/Makefile b/Makefile
index 05f7ca5..8682011 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ else ifeq ($(OS), WINDOWS)
 CC ?= i586-mingw32msvc-gcc
 CFLAGS ?= -O2 -Wall
 teensy_loader_cli.exe: teensy_loader_cli.c
-       $(CC) $(CFLAGS) -s -DUSE_WIN32 -o teensy_loader_cli.exe teensy_loader_cli.c -lhid -lsetupapi -lwinmm
+       $(CC) $(CFLAGS) -s -DUSE_WIN32 -o teensy_loader_cli.exe teensy_loader_cli.c -lhid -lsetupapi -lwinmm $(LDFLAGS)


 else ifeq ($(OS), MACOSX)
@@ -27,7 +27,7 @@ ifeq ($(USE_LIBUSB), YES)
 CC ?= gcc
 CFLAGS ?= -O2 -Wall
 teensy_loader_cli: teensy_loader_cli.c
-       $(CC) $(CFLAGS) -s -DUSE_LIBUSB -DMACOSX -o teensy_loader_cli teensy_loader_cli.c -lusb -I /usr/local/include -L/usr/local/lib
+       $(CC) $(CFLAGS) -s -DUSE_LIBUSB -DMACOSX -o teensy_loader_cli teensy_loader_cli.c -lusb $(LDFLAGS)

 else
 CC ?= gcc
