From ef411d531d52ffe9a4101b79ef87f1d796760f9f Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 11 Jul 2024 21:53:14 +0200
Subject: [PATCH] Update setup.py

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 0a33e6a..9b8b34f 100755
--- a/setup.py
+++ b/setup.py
@@ -51,8 +51,7 @@
     ],
     data_files=[
         ('lib', ['lib/xenon-grpc-{}.jar'.format(xenon_grpc_version)]),  # noqa
-        ('bin', ['bin/xenon-grpc',
-                 'bin/xenon-grpc.bat'])],
+        ('bin', ['bin/xenon-grpc.bat'])],
     install_requires=['grpcio', 'grpcio-tools', 'pyxdg', 'pyopenssl'],
     extras_require={
         'test': ['pytest', 'flake8', 'coverage', 'pep8', 'tox'],
