--- a/01-devices.post
+++ b/01-devices.post
@@ -17,7 +17,8 @@ maybe_create_devs ()
   fi
 
   # Create it if necessary
-  mkdir -m 755 "${DEVDIR}" 2> /dev/null
+  mkdir -p "${DEVDIR}" 2>/dev/null || true
+  chmod 755 "${DEVDIR}"
   if [ ! -e "${DEVDIR}" ]
   then
     echo
@@ -38,22 +39,20 @@ maybe_create_devs ()
       echo "${DEVDIR}/shm is existant but not a directory."
       echo "POSIX semaphores and POSIX shared memory will not work"
       echo
+      exit 1
     fi
   fi
 
   # Create it if necessary
+  mkdir -p "${DEVDIR}/shm" 2>/dev/null || true
+  chmod 1777 "${DEVDIR}/shm"
   if [ ! -e "${DEVDIR}/shm" ]
   then
-    mkdir -m 1777 "${DEVDIR}/shm"
-    if [ ! -e "${DEVDIR}/shm" ]
-    then
-      echo
-      echo "Creating ${DEVDIR}/shm directory failed."
-      echo "POSIX semaphores and POSIX shared memory will not work"
-      echo
-    fi
-  else
-    chmod 1777 "${DEVDIR}/shm"
+    echo
+    echo "Creating ${DEVDIR}/shm directory failed."
+    echo "POSIX semaphores and POSIX shared memory will not work"
+    echo
+    exit 1
   fi
 
   # Check for ${DEVDIR}/mqueue directory (for POSIX message queues)
@@ -67,27 +66,25 @@ maybe_create_devs ()
       echo "${DEVDIR}/mqueue is existant but not a directory."
       echo "POSIX message queues will not work"
       echo
+      exit 1
     fi
   fi
 
   # Create it if necessary
+  mkdir -p "${DEVDIR}/mqueue" 2>/dev/null || true
+  chmod 1777 "${DEVDIR}/mqueue"
   if [ ! -e "${DEVDIR}/mqueue" ]
   then
-    mkdir -m 1777 "${DEVDIR}/mqueue"
-    if [ ! -e "${DEVDIR}/mqueue" ]
-    then
-      echo
-      echo "Creating ${DEVDIR}/mqueue directory failed."
-      echo "POSIX message queues will not work"
-      echo
-    fi
-  else
-    chmod 1777 "${DEVDIR}/mqueue"
+    echo
+    echo "Creating ${DEVDIR}/mqueue directory failed."
+    echo "POSIX message queues will not work"
+    echo
+    exit 1
   fi
-  
+
   # Install /dev/fd, /dev/std{in,out,err}.  The bash builtin test was compiled
   # to assume these exist, so use /bin/test to really check.
-  /usr/bin/test -h /dev/stdin  || ln -sf /proc/self/fd/0 /dev/stdin
+  /usr/bin/test -h /dev/stdin  || ln -sf /proc/self/fd/0 /dev/stdin 2>/dev/null || true
   /usr/bin/test -h /dev/stdout || ln -sf /proc/self/fd/1 /dev/stdout
   /usr/bin/test -h /dev/stderr || ln -sf /proc/self/fd/2 /dev/stderr
   /usr/bin/test -h /dev/fd     || ln -sf /proc/self/fd   /dev/fd
