--- a/01-devices.post
+++ b/01-devices.post
@@ -44,7 +44,7 @@ maybe_create_devs ()
   # Create it if necessary
   if [ ! -e "${DEVDIR}/shm" ]
   then
-    mkdir -m 1777 "${DEVDIR}/shm"
+    mkdir -m 1777 "${DEVDIR}/shm" 2> /dev/null
     if [ ! -e "${DEVDIR}/shm" ]
     then
       echo
@@ -53,7 +53,7 @@ maybe_create_devs ()
       echo
     fi
   else
-    chmod 1777 "${DEVDIR}/shm"
+    chmod 1777 "${DEVDIR}/shm" 2> /dev/null
   fi
 
   # Check for ${DEVDIR}/mqueue directory (for POSIX message queues)
@@ -73,7 +73,7 @@ maybe_create_devs ()
   # Create it if necessary
   if [ ! -e "${DEVDIR}/mqueue" ]
   then
-    mkdir -m 1777 "${DEVDIR}/mqueue"
+    mkdir -m 1777 "${DEVDIR}/mqueue" 2> /dev/null
     if [ ! -e "${DEVDIR}/mqueue" ]
     then
       echo
@@ -82,7 +82,7 @@ maybe_create_devs ()
       echo
     fi
   else
-    chmod 1777 "${DEVDIR}/mqueue"
+    chmod 1777 "${DEVDIR}/mqueue" 2> /dev/null
   fi
   
   # Install /dev/fd, /dev/std{in,out,err}.  The bash builtin test was compiled
