diff --git i/meson_scripts/init.py w/meson_scripts/init.py
index 4ed6f05..9fda51f 100755
--- i/meson_scripts/init.py
+++ w/meson_scripts/init.py
@@ -179,7 +179,7 @@ def is_git_directory(path="."):
             "Directory was not cloned using git. Using fall-back method to init submodules"
         )
         return False
-    return p == 0
+    return False
 
 
 def submodule_status(path, sha_commit):
