From 2fb284b2d1c2ded5ef586715b84eeb1f8fd0e974 Mon Sep 17 00:00:00 2001
From: David Wilson <dw@botanicus.net>
Date: Tue, 25 Oct 2016 21:44:36 +0100
Subject: [PATCH] issue #138: disable tests that have no meaning on Windows.

---
 tests/env_test.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/env_test.py b/tests/env_test.py
index 9ba6a7e..b56aa55 100644
--- a/tests/env_test.py
+++ b/tests/env_test.py
@@ -164,6 +164,8 @@ def test_map_async(self):
             path, env = testlib.temp_env(map_async=flag)
             assert env.flags()['map_async'] == flag
 
+    @unittest.skipIf(sys.platform == 'win32',
+        'Mode argument is ignored on Windows; see lmdb.h')
     def test_mode_subdir_create(self):
         oldmask = os.umask(0)
         try:
@@ -177,6 +179,8 @@ def test_mode_subdir_create(self):
         finally:
             os.umask(oldmask)
 
+    @unittest.skipIf(sys.platform == 'win32',
+        'Mode argument is ignored on Windows; see lmdb.h')
     def test_mode_subdir_nocreate(self):
         oldmask = os.umask(0)
         try:
@@ -489,6 +493,8 @@ def _test_reader_check_child(path):
         txn = env.begin()
         os._exit(0)
 
+    @unittest.skipIf(sys.platform == 'win32',
+        'Stale writers are cleared automatically on Windows, see lmdb.h')
     def test_reader_check(self):
         path, env = testlib.temp_env(max_spare_txns=0)
         rc = env.reader_check()
