--- ./utils/cxi_context.c
+++ ./utils/cxi_context.c
@@ -19,7 +19,15 @@
 #include <stdlib.h>
 #include <stdbool.h>
 #include <err.h>
+#if __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 25)
 #include <sys/random.h>
+#else
+#include <sys/syscall.h>
+#ifndef SYS_getrandom
+#define SYS_getrandom 318 /* For x86_64 */
+#endif
+#define getrandom(buf, buflen, flags) syscall(SYS_getrandom, (buf), (buflen), (flags))
+#endif
 #include <sys/mman.h>
 #include <linux/mman.h>
 
