--- ./utils/cxi_udp_gen.c
+++ ./utils/cxi_udp_gen.c
@@ -165,10 +165,10 @@
 	struct udphdr *udp = &skb->udphdr;
 	uint32_t sum;
 
-	udp->uh_sport = htons(src_port);
-	udp->uh_dport = htons(55555);
-	udp->uh_ulen = htons(udp_payload_len);
-	udp->uh_sum = 0;
+	udp->source = htons(src_port);
+	udp->dest = htons(55555);
+	udp->len = htons(udp_payload_len);
+	udp->check = 0;
 
 	sum = header_checksum(skb, udp_payload_len);
 
@@ -197,7 +197,7 @@
 		sum = sum ?: 0xffff;
 	}
 
-	udp->uh_sum = sum;
+	udp->check = sum;
 }
 
 #define MIN(a, b) (((a) < (b)) ? (a) : (b))
