diff --git a/configure b/configure
index ead47c5..a9b366e 100755
--- a/configure
+++ b/configure
@@ -3198,7 +3198,7 @@ toolchain "clang"
     set_toolset "mxx" "clang" "clang++"
     set_toolset "ld" "clang++" "clang"
     set_toolset "sh" "clang++" "clang"
-    set_toolset "ar" "ar"
+    set_toolset "ar" "llvm-ar" "ar"
 toolchain_end
 
 # gcc toolchain
diff --git a/core/src/cli/xmake.sh b/core/src/cli/xmake.sh
index 4d7bf42..af5d168 100755
--- a/core/src/cli/xmake.sh
+++ b/core/src/cli/xmake.sh
@@ -19,7 +19,7 @@ target "cli"
     if is_plat "macosx" && is_config "runtime" "luajit"; then
         add_ldflags "-all_load" "-pagezero_size 10000" "-image_base 100000000"
     elif is_plat "mingw"; then
-        add_ldflags "-static-libgcc" "-lwintrust" "-lcrypt32"
+        add_ldflags "-lwintrust" "-lcrypt32"
     fi
 
     # add install files
@@ -52,7 +52,7 @@ target "cli"
     # add syslinks
     add_options "atomic"
     if is_plat "mingw" "msys" "cygwin"; then
-        add_syslinks "ws2_32" "pthread" "m"
+        add_syslinks "ws2_32" "user32"
     elif is_plat "bsd" "solaris"; then
         add_syslinks "pthread" "m"
     elif is_plat "haiku"; then
