diff --git a/core/xmake.sh b/core/xmake.sh
index 850c37a..f159480 100755
--- a/core/xmake.sh
+++ b/core/xmake.sh
@@ -100,21 +100,26 @@ option_find_lua() {
     local ldflags=""
     local cflags=""
     option "lua"
-        # detect lua5.4 on debian
-        cflags=`pkg-config --cflags lua5.4 2>/dev/null`
-        ldflags=`pkg-config --libs lua5.4 2>/dev/null`
-        # detect it on fedora
-        if test_z "${cflags}"; then
-            cflags=`pkg-config --cflags lua 2>/dev/null`
-        fi
-        if test_z "${ldflags}"; then
-            ldflags=`pkg-config --libs lua 2>/dev/null`
-        fi
-        if test_z "${cflags}"; then
-            cflags="-I/usr/include/lua5.4"
-        fi
-        if test_z "${ldflags}"; then
-            ldflags="-llua5.4"
+        if is_plat "mingw"; then
+            cflags="-I${PREFIX}/include"
+            ldflags="-L${PREFIX}/lib -llua"
+        else
+            # detect lua5.4 on debian
+            cflags=`pkg-config --cflags lua5.4 2>/dev/null`
+            ldflags=`pkg-config --libs lua5.4 2>/dev/null`
+            # detect it on fedora
+            if test_z "${cflags}"; then
+                cflags=`pkg-config --cflags lua 2>/dev/null`
+            fi
+            if test_z "${ldflags}"; then
+                ldflags=`pkg-config --libs lua 2>/dev/null`
+            fi
+            if test_z "${cflags}"; then
+                cflags="-I/usr/include/lua5.4"
+            fi
+            if test_z "${ldflags}"; then
+                ldflags="-llua5.4"
+            fi
         fi
         add_cflags "${cflags}"
         add_ldflags "${ldflags}"
@@ -207,13 +212,18 @@ option_find_tbox() {
     local ldflags=""
     local cflags=""
     option "tbox"
-        cflags=`pkg-config --cflags libtbox 2>/dev/null`
-        ldflags=`pkg-config --libs libtbox 2>/dev/null`
-        if test_z "${cflags}"; then
-            cflags="-I/usr/include"
-        fi
-        if test_z "${ldflags}"; then
-            ldflags="-ltbox"
+        if is_plat "mingw"; then
+            cflags="-I${PREFIX}/include"
+            ldflags="-L${PREFIX}/lib -ltbox"
+        else
+            cflags=`pkg-config --cflags libtbox 2>/dev/null`
+            ldflags=`pkg-config --libs libtbox 2>/dev/null`
+            if test_z "${cflags}"; then
+                cflags="-I/usr/include"
+            fi
+            if test_z "${ldflags}"; then
+                ldflags="-ltbox"
+            fi
         fi
         add_cflags "${cflags}"
         add_ldflags "${ldflags}"
