diff --git a/core/src/xmake/hash/xxhash.c b/core/src/xmake/hash/xxhash.c
index 77bf32d..55d68b0 100644
--- a/core/src/xmake/hash/xxhash.c
+++ b/core/src/xmake/hash/xxhash.c
@@ -29,10 +29,7 @@
  * includes
  */
 #include "prefix.h"
-#define XXH_NAMESPACE XM_
-#define XXH_STATIC_LINKING_ONLY
-#define XXH_IMPLEMENTATION
-#include "xxhash/xxhash.h"
+#include <xxhash.h>
 
 /* //////////////////////////////////////////////////////////////////////////////////////
  * implementation
@@ -65,13 +62,13 @@ tb_int_t xm_hash_xxhash(lua_State *lua) {
         XXH64_hash_t value64;
         XXH128_hash_t value128;
         if (mode == 128) {
-            value128 = XM_XXH3_128bits(data, size);
+            value128 = XXH3_128bits(data, size);
             buffer = (tb_byte_t const *)&value128;
         } else if (mode == 64) {
-            value64 = XM_XXH3_64bits(data, size);
+            value64 = XXH3_64bits(data, size);
             buffer = (tb_byte_t const *)&value64;
         } else if (mode == 32) {
-            value64 = XM_XXH3_64bits(data, size);
+            value64 = XXH3_64bits(data, size);
             value32 = (value64 >> 32) ^ (value64 & 0xffffffff);
             buffer = (tb_byte_t const *)&value32;
         }
@@ -100,13 +97,13 @@ tb_int_t xm_hash_xxhash(lua_State *lua) {
     tb_stream_ref_t stream = tb_stream_init_from_file(filename, TB_FILE_MODE_RO);
     if (stream) {
         // open stream
-        XXH3_state_t *state = XM_XXH3_createState();
+        XXH3_state_t *state = XXH3_createState();
         if (tb_stream_open(stream) && state) {
             // reset xxhash
             if (mode == 32 || mode == 64) {
-                XM_XXH3_64bits_reset(state);
+                XXH3_64bits_reset(state);
             } else {
-                XM_XXH3_128bits_reset(state);
+                XXH3_128bits_reset(state);
             }
 
             // read data and update xxhash
@@ -117,9 +114,9 @@ tb_int_t xm_hash_xxhash(lua_State *lua) {
 
                 if (real > 0) {
                     if (mode == 32 || mode == 64) {
-                        XM_XXH3_64bits_update(state, data, real);
+                        XXH3_64bits_update(state, data, real);
                     } else {
-                        XM_XXH3_128bits_update(state, data, real);
+                        XXH3_128bits_update(state, data, real);
                     }
                 }
                 // no data? continue it
@@ -143,13 +140,13 @@ tb_int_t xm_hash_xxhash(lua_State *lua) {
             XXH64_hash_t value64;
             XXH128_hash_t value128;
             if (mode == 128) {
-                value128 = XM_XXH3_128bits_digest(state);
+                value128 = XXH3_128bits_digest(state);
                 buffer = (tb_byte_t const *)&value128;
             } else if (mode == 64) {
-                value64 = XM_XXH3_64bits_digest(state);
+                value64 = XXH3_64bits_digest(state);
                 buffer = (tb_byte_t const *)&value64;
             } else if (mode == 32) {
-                value64 = XM_XXH3_64bits_digest(state);
+                value64 = XXH3_64bits_digest(state);
                 value32 = (value64 >> 32) ^ (value64 & 0xffffffff);
                 buffer = (tb_byte_t const *)&value32;
             }
@@ -169,7 +166,7 @@ tb_int_t xm_hash_xxhash(lua_State *lua) {
 
         // exit xxhash
         if (state) {
-            XM_XXH3_freeState(state);
+            XXH3_freeState(state);
         }
     }
     if (!ok) {
diff --git a/core/src/xmake/xmake.sh b/core/src/xmake/xmake.sh
index 2bb44fc..7634ddb 100755
--- a/core/src/xmake/xmake.sh
+++ b/core/src/xmake/xmake.sh
@@ -10,6 +10,7 @@ target "xmake"
     # add deps
     if has_config "external"; then
         add_deps "sv"
+        add_links "xxhash" "{public}"
         local libs="lz4 tbox"
         for lib in $libs; do
             if has_config "$lib"; then
@@ -53,7 +54,6 @@ target "xmake"
     # add includes directory
     add_includedirs ".." "{public}"
     add_includedirs "${builddir}/${plat}/${arch}/${mode}" "{public}"
-    add_includedirs "../xxhash"
     add_includedirs "${projectdir}/xmake/scripts/module"
 
     # add the common source files
