diff --git a/core/src/xmake/io/iscygpty.c b/core/src/xmake/io/iscygpty.c
index e46d8c4..c1f7d5a 100644
--- a/core/src/xmake/io/iscygpty.c
+++ b/core/src/xmake/io/iscygpty.c
@@ -97,7 +97,7 @@ static void setup_fileid_api(void) {
 #define is_wprefix(s, prefix) (wcsncmp((s), (prefix), sizeof(prefix) / sizeof(WCHAR) - 1) == 0)
 
 /* Check if the fd handle is a cygwin/msys's pty. */
-int is_cygpty(HANDLE h) {
+static int is_cygpty(HANDLE h) {
 #if defined(STUB_IMPL)
     return 0;
 #elif _WIN32_WINNT >= 0x0600
