diff --git a/configure b/configure
index 8dd6e3d..54e59d8 100755
--- a/configure
+++ b/configure
@@ -1796,13 +1796,16 @@ _get_macosx_sysroot_flags() {
             else
                 flags="${flags} -isysroot ${sdkdir}"
             fi
-            path_basename "${sdkdir}"; local basename="${_ret}"
-            if test_nz "${basename}" && string_startswith "${basename}" "MacOSX"; then
-                string_replace "${basename}" "MacOSX" ""; local target_minver="${_ret}"
-                if test_nz "${target_minver}"; then
-                    flags="${flags} -target ${_target_arch}-apple-macos${target_minver}"
+            local target_minver="${MACOSX_DEPLOYMENT_TARGET:-}"
+            if test_z "${target_minver}"; then
+                path_basename "${sdkdir}"; local basename="${_ret}"
+                if test_nz "${basename}" && string_startswith "${basename}" "MacOSX"; then
+                    string_replace "${basename}" "MacOSX" ""; target_minver="${_ret}"
                 fi
             fi
+            if test_nz "${target_minver}"; then
+                flags="${flags} -target ${_target_arch}-apple-macos${target_minver}"
+            fi
         fi
     fi
     _ret="${flags}"
