From e2fac421eb6524d95d83f6b418899517a3615736 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sun, 4 Aug 2019 19:38:35 -0500
Subject: [PATCH] Fix for new NTL versions

---
 code/latte/rational.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/code/latte/rational.cpp b/code/latte/rational.cpp
index e4c3dcb..678561a 100644
--- a/code/latte/rational.cpp
+++ b/code/latte/rational.cpp
@@ -200,6 +200,11 @@ void RationalNTL::canonicalize()
 		denominator *= -1;
 		numerator *= -1;
 	}//make denominator positive.
+	else if (denominator == 0) 
+	{
+		denominator = 1;
+		return;
+	}
 
 	ZZ gcd = GCD(numerator, denominator); //GCD(numerator, denominator);
 
