From b530b275f1af85d2ae8da140e58796aa506186f8 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 1 Dec 2021 21:07:42 +1100
Subject: [PATCH 1/2] do not build third_party libs

---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed39e644..46312680 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,5 +63,8 @@ endif()
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_COMPILED_WITH_CXX_ABI} -Wall ${TORCH_CXX_FLAGS}")
 
-add_subdirectory(third_party)
+find_package(re2 REQUIRED)
+find_package(double-conversion REQUIRED)
+find_package(sentencepiece REQUIRED)
+
 add_subdirectory(torchtext/csrc)
-- 
2.38.1.windows.1

