From 28c4b93702571e5d1d128c4887b2e99d77ebff39 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 1 Dec 2021 21:07:42 +1100
Subject: [PATCH] do not build third_party libs in setup.py

---
 build_tools/setup_helpers/extension.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/build_tools/setup_helpers/extension.py b/build_tools/setup_helpers/extension.py
index f3a2097..38501e7 100644
--- a/build_tools/setup_helpers/extension.py
+++ b/build_tools/setup_helpers/extension.py
@@ -182,6 +182,4 @@ def get_ext_modules(debug=False):
 
 class BuildExtension(TorchBuildExtension):
     def build_extension(self, ext):
-        if ext.name == _EXT_NAME:
-            _configure_third_party(self.debug)
         super().build_extension(ext)
-- 
2.32.0.windows.2

