From 55627e85bccfccc9ad9344556e52f6b9a178034c Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 18 Jan 2023 22:06:33 +1100
Subject: [PATCH 3/5] remove unnecessary installation requirements

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 080415f7..d3ca73d2 100644
--- a/setup.py
+++ b/setup.py
@@ -86,7 +86,7 @@ setup_info = dict(
     description="Text utilities and datasets for PyTorch",
     long_description=read("README.rst"),
     license="BSD",
-    install_requires=["tqdm", "requests", pytorch_package_dep, "numpy"],
+    install_requires=[pytorch_package_dep, "numpy"],
     python_requires=">=3.7",
     classifiers=[
         "Programming Language :: Python :: 3.7",
-- 
2.38.1.windows.1

