From 12ec329580410d3bfd02ed2a950c2d7de812bbbf Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 18 Jan 2023 22:06:33 +1100
Subject: [PATCH 3/6] remove unnecessary installation requirements

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index d8afa7a5..8501f17b 100644
--- a/setup.py
+++ b/setup.py
@@ -100,7 +100,7 @@ setup_info = dict(
     description="Text utilities and datasets for PyTorch",
     long_description=read("README.rst"),
     license="BSD",
-    install_requires=["tqdm", "requests", pytorch_package_dep, "numpy"],
+    install_requires=[pytorch_package_dep, "numpy"],
     python_requires=">=3.7",
     classifiers=[
         "Programming Language :: Python :: 3.7",
-- 
2.38.1.windows.1

