From a2e6933a9ae206f216e4b20d14010912f5ff9758 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 18 Jan 2023 22:58:13 +1100
Subject: [PATCH 5/6] don't try to init submodules

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 8501f17b..f49b00fb 100644
--- a/setup.py
+++ b/setup.py
@@ -89,7 +89,6 @@ class clean(distutils.command.clean.clean):
                 shutil.rmtree(str(path), ignore_errors=True)
 
 
-_init_submodule()
 setup_info = dict(
     # Metadata
     name="torchtext",
-- 
2.38.1.windows.1

