From 37c5c5e4e1c729bf28ac53fa6eedc0a55361bd4f Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 20 Jan 2023 17:16:20 +1100
Subject: [PATCH 6/6] must use C++17 to match abseil

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 46312680..cc136ad5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,7 @@ if(env_cxx_standard GREATER -1)
       "PyTorch requires -std=c++14. Please remove -std=c++ settings in your environment.")
 endif()
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_C_STANDARD 11)
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-- 
2.38.1.windows.1

