Index: work/third_party/pthreadpool/src/gcd.c
===================================================================
--- work.orig/third_party/pthreadpool/src/gcd.c
+++ work/third_party/pthreadpool/src/gcd.c
@@ -118,7 +118,9 @@ PTHREADPOOL_INTERNAL void pthreadpool_pa
 		/* The next subrange starts where the previous ended */
 		range_start = range_end;
 	}
-
+        #ifndef DISPATCH_APPLY_AUTO
+        #define DISPATCH_APPLY_AUTO ((dispatch_queue_t _Nonnull)0)
+        #endif
 	dispatch_apply_f(threads_count.value, DISPATCH_APPLY_AUTO, threadpool, thread_main);
 
 	/* Unprotect the global threadpool structures */
