diff --git a/third_party/XNNPACK/src/jit/memory.c b/third_party/XNNPACK/src/jit/memory.c
index 257c2a50b..af3a3679c 100644
--- a/third_party/XNNPACK/src/jit/memory.c
+++ b/third_party/XNNPACK/src/jit/memory.c
@@ -32,7 +32,7 @@ enum xnn_status xnn_allocate_code_memory(struct xnn_code_buffer* buf, size_t siz
     return xnn_status_out_of_memory;
   }
 #else
-  void* p = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+  void* p = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
   if (p == MAP_FAILED) {
     xnn_log_error("failed to allocate %zu bytes for JIT code buffer, error code: %d", size, errno);
     return xnn_status_out_of_memory;

