From dedd790a90a904853f92f5b6d26588e0697bd3c7 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@lanl.gov>
Date: Thu, 14 Jul 2022 17:40:40 -0600
Subject: [PATCH] Fix collections.abc

---
 fn/iters.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/fn/iters.py b/fn/iters.py
index 9da133a..46e011c 100644
--- a/fn/iters.py
+++ b/fn/iters.py
@@ -1,5 +1,5 @@
 from sys import version_info
-from collections import deque, Iterable
+from collections import deque
 from operator import add, itemgetter, attrgetter, not_
 from functools import partial
 from itertools import (islice, 
@@ -17,6 +17,12 @@
 from .underscore import shortcut as _
 from .uniform import *
 
+try:
+    from collections import Iterable
+except ImportError:
+    from collections.abc import Iterable
+
+
 def take(limit, base): 
     return islice(base, limit)
 
