From a0dab56e36172ceb2264c5d5a902cafa932aa519 Mon Sep 17 00:00:00 2001
From: Tom de Geus <tom@geus.me>
Date: Tue, 26 Nov 2019 17:10:39 +0100
Subject: [PATCH] Updating setup.py

---
 setup.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index f614095..6c818a6 100644
--- a/setup.py
+++ b/setup.py
@@ -2,13 +2,13 @@ from setuptools import setup
 
 setup(
     name = 'pdfcombine',
-    description = 'Script of combine PDF files (wrapper around GhostScript)',
-    long_description = 'Script of combine PDF files (wrapper around GhostScript)',
-    keywords = 'PDF, GhostScript',
-    version = '0.4.0',
+    version = '0.4.1',
     license = 'MIT',
     author = 'Tom de Geus',
     author_email = 'tom@geus.me',
+    description = 'Script of combine PDF files (wrapper around GhostScript)',
+    long_description = 'Script of combine PDF files (wrapper around GhostScript)',
+    keywords = 'PDF, GhostScript',
     url = 'https://github.com/tdegeus/pdfcombine',
     entry_points = {
         'console_scripts': ['pdfcombine = pdfcombine:main']}
-- 
2.21.0 (Apple Git-122.2)

