diff --git a/glomap/estimators/bundle_adjustment.cc b/glomap/estimators/bundle_adjustment.cc
index a03bea5..a993590 100644
--- a/glomap/estimators/bundle_adjustment.cc
+++ b/glomap/estimators/bundle_adjustment.cc
@@ -1,7 +1,6 @@
 #include "bundle_adjustment.h"
 
 #include <colmap/estimators/cost_functions.h>
-#include <colmap/estimators/manifold.h>
 #include <colmap/sensor/models.h>
 
 namespace glomap {
diff --git a/glomap/estimators/gravity_refinement.cc b/glomap/estimators/gravity_refinement.cc
index 6015d66..2b7f6e1 100644
--- a/glomap/estimators/gravity_refinement.cc
+++ b/glomap/estimators/gravity_refinement.cc
@@ -3,7 +3,7 @@
 #include "glomap/estimators/cost_function.h"
 #include "glomap/math/gravity.h"
 
-#include <colmap/estimators/manifold.h>
+#include <colmap/estimators/cost_functions.h>
 
 namespace glomap {
 void GravityRefiner::RefineGravity(const ViewGraph& view_graph,
