diff --git a/freecad/__init__.py.template b/freecad/__init__.py.template
index daded10..f9204d0 100644
--- a/freecad/__init__.py.template
+++ b/freecad/__init__.py.template
@@ -3,7 +3,7 @@ sys.path.append("{{ PATH_TO_FREECAD }}")
 import FreeCAD as app
 import Part as part
 import PartDesign as partdesign
-import Draft as draft
+# import Draft as draft # somehow not working in no gui mode
 import Fem as fem
 import Mesh as mesh
 import Path as path
diff --git a/setup.py b/setup.py
index 79099fa..677a145 100644
--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,12 @@ proc = sub.Popen(['FreeCAD', '-c', "import os; import FreeCADGui; print(os.path.
 out, err = proc.communicate()
 if err:
     raise(RuntimeError(err.decode("utf8")))
-freecad_lib_path = out.decode("utf8").replace("\n", "")
+freecad_lib_path = out.decode("utf8").split(" ")[0].replace("\n", "")
+
+print("\n\n##############################")
+print("path to the freecad library is: \n")
+print(freecad_lib_path)
+print("##############################\n\n")
 
 template_dir = os.path.join(os.path.dirname(__file__), "freecad")
 template_fn = os.path.join(template_dir, "__init__.py.template")
