commit 7cb99b70f424cbc8e8110b1738cd6e485b2a514f
Author: looooo <sppedflyer@gmail.com>
Date:   Thu Nov 7 11:45:31 2019 +0100

    add debug prints

diff --git a/setup.py b/setup.py
index 9062310..54f5649 100644
--- a/setup.py
+++ b/setup.py
@@ -10,6 +10,11 @@ if err:
     raise(RuntimeError(err.decode("utf8")))
 freecad_lib_path = out.decode("utf8").replace("\n", "")
 
+print("\n\n##############################")
+print("path to the freecad library is: \n")
+print(freecad_lib_path)
+print("##############################\n\n")
+
 template_dir = os.path.join(os.path.dirname(__file__), "freecad")
 template_fn = os.path.join(template_dir, "__init__.py.template")
 render_fn = os.path.join(template_dir, "__init__.py")
