diff --git a/libxsd-frontend/xsd-frontend/semantic-graph/elements.cxx b/libxsd-frontend/xsd-frontend/semantic-graph/elements.cxx
index fa48a9a..3159906 100644
--- a/libxsd-frontend/xsd-frontend/semantic-graph/elements.cxx
+++ b/libxsd-frontend/xsd-frontend/semantic-graph/elements.cxx
@@ -3,6 +3,7 @@
 // license   : GNU GPL v2 + exceptions; see accompanying LICENSE file
 
 #include <algorithm>
+#include <iostream>
 
 #include <cutl/compiler/type-info.hxx>
 
