diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf39c27..41a3366 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -829,6 +829,11 @@ if(PASTIX_WITH_PAPI)
   )
 endif()
 
+message(STATUS "###################################")
+message(STATUS "PASTIX_LIB_SRCS: ${PASTIX_LIB_SRCS}")
+
+message(STATUS "#####################################################")
+message(STATUS "generated_common_sources: ${generated_common_sources}")
 
 add_library( pastix
   ${PASTIX_LIB_SRCS}
