diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf39c27..bf5d9be 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -829,6 +829,14 @@ if(PASTIX_WITH_PAPI)
   )
 endif()
 
+message(STATUS "###################################")
+message(STATUS "PASTIX_LIB_SRCS: ${PASTIX_LIB_SRCS}")
+
+message(STATUS "#####################################################")
+message(STATUS "generated_common_sources: ${generated_common_sources}")
+
+file(TO_CMAKE_PATH "${PASTIX_LIB_SRCS}" PASTIX_LIB_SRCS)
+message(STATUS "PASTIX_LIB_SRCS: ${PASTIX_LIB_SRCS}")
 
 add_library( pastix
   ${PASTIX_LIB_SRCS}
