From 1b1ac91b9362ff7beb28eb01bcce975d671c4473 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 26 Nov 2021 14:14:52 +1100
Subject: [PATCH] do not try to build sox/kaldi

---
 third_party/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index ef984d5..f499005 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -6,7 +6,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
 # sox
 ################################################################################
 add_library(libsox INTERFACE)
-if (BUILD_SOX)
+if (FALSE)
   add_subdirectory(sox)
   target_include_directories(libsox INTERFACE ${SOX_INCLUDE_DIR})
   target_link_libraries(libsox INTERFACE ${SOX_LIBRARIES})
@@ -16,7 +16,7 @@ endif()
 ################################################################################
 # kaldi
 ################################################################################
-if (BUILD_KALDI)
+if (FALSE)
   add_subdirectory(kaldi)
   list(APPEND TORCHAUDIO_THIRD_PARTIES kaldi)
 endif()
-- 
2.32.0.windows.2

