From 2b2c9099288b9e2db6951aef0055e96369192906 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 10 Jun 2023 00:36:55 +1100
Subject: [PATCH 1/2] don't rebuild libs we have already

---
 third_party/CMakeLists.txt | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index f2b82966..687a5b69 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -21,11 +21,9 @@ endif()
 # Decoder
 ################################################################################
 if (BUILD_CTC_DECODER)
-  add_subdirectory(zlib)
-  add_subdirectory(bzip2)
-  if (NOT MSVC)
-    add_subdirectory(lzma)
-  endif()
+  find_package(zlib)
+  find_package(bzip2)
+  find_package(lzma)
   add_subdirectory(kenlm)
   add_subdirectory(flashlight-text)
 endif()
