From 54bd2248d75b16a347651182d3879e6b60e74c6d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 10 Jun 2023 01:24:08 +1100
Subject: [PATCH 2/2] point to correct prefix

---
 tools/setup_helpers/extension.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tools/setup_helpers/extension.py b/tools/setup_helpers/extension.py
index 093f1367..b6314a44 100644
--- a/tools/setup_helpers/extension.py
+++ b/tools/setup_helpers/extension.py
@@ -111,8 +111,9 @@ class CMakeBuild(build_ext):
 
         cmake_args = [
             f"-DCMAKE_BUILD_TYPE={cfg}",
-            f"-DCMAKE_PREFIX_PATH={torch.utils.cmake_prefix_path}",
-            f"-DCMAKE_INSTALL_PREFIX={extdir}",
+            f"-DCMAKE_INSTALL_PREFIX={os.environ['SP_DIR']}/torch/lib",
+            f"-DCMAKE_LIBRARY_PREFIX={os.environ['PREFIX']}",
+            f"-DCMAKE_PREFIX_PATH={os.environ['PREFIX']}",
             "-DCMAKE_VERBOSE_MAKEFILE=ON",
             f"-DPython_INCLUDE_DIR={distutils.sysconfig.get_python_inc()}",
             f"-DBUILD_SOX:BOOL={'ON' if _BUILD_SOX else 'OFF'}",
