diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7738d19..4904ae4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,6 +57,9 @@ add_library(QGLViewer SHARED ${QGLViewer_SRC})
 target_include_directories(QGLViewer INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 target_link_libraries(QGLViewer PRIVATE ${QtLibs} OpenGL::GL OpenGL::GLU)
 
+option(LIBQGLVIEWER_BUILD_EXAMPLES "Build libQGLViewer examples" ON)
+if(LIBQGLVIEWER_BUILD_EXAMPLES)
+
 # Example: animation.
 set(animation_SRC
     "${PROJECT_SOURCE_DIR}/examples/animation/animation.cpp"
@@ -259,6 +262,8 @@ set(stereoViewer_SRC
 target_include_directories(stereoViewer PRIVATE QGLViewer ${QtLibs} OpenGL::GL OpenGL::GLU)
 target_link_libraries(stereoViewer QGLViewer ${QtLibs} OpenGL::GL OpenGL::GLU)
 
+endif()
+
 # Install.
 include(GNUInstallDirs)
 install(TARGETS QGLViewer
