diff --git a/src/cmdLineParser.cpp b/src/cmdLineParser.cpp
index 29702cf..5b1c10c 100644
--- a/src/cmdLineParser.cpp
+++ b/src/cmdLineParser.cpp
@@ -58,6 +58,14 @@ void parseCmdLine(
             else if( strcmp(args[i],"-o") == 0 ) {
                 currentSwitch.assign( args[i] ) ;
             }
+            else if( strcmp(args[i],"-h") == 0 ) {
+                cout << "  Usage:\n";
+                cout << "      mmligner <pdb1>(:<chainIDs>) <pdb2>(:<chainIDs)\n";
+                cout << "               (--ivalue <fasta>)\n";
+                cout << "               ( -o  <output_prefix_string>) \n";
+                cout << "               ( --superpose) \n\n";
+                exit(0);
+            }
             else  {
                 cerr << "::CmdLineError:: Unsupported option: " 
                     << args[i] << endl ;
