From 52ed1b376eeb1f6aba896d382b5421864d50a175 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Wed, 10 Mar 2021 17:25:00 +0100
Subject: [PATCH] Update setup.cfg.gnu

---
 setup.cfg.gnu | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/setup.cfg.gnu b/setup.cfg.gnu
index ebd3181..c3ce687 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -2,13 +2,13 @@
 # Rename to setup.cfg.
 
 [config_fc]
-fcompiler=gfortran
-f90flags=-cpp -fPIC -ffree-form -ffree-line-length-none -x f95-cpp-input
-f77flags=-cpp -fPIC -x f77-cpp-input
+fcompiler=${FC}
+f90flags=${FFLAGS}
+f77flags=${FFLAGS}
 # For gcc-10, add this to f77flags: -fallow-argument-mismatch
 
 [build_ext]
-libraries=gfortran
+libraries=${FC}
 
 # See the docstring in versioneer.py for instructions. Note that you must
 # re-run 'versioneer.py setup' after changing this section, and commit the
