From 04ccc4831858e48aa366f2c91859535101ee3b14 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Mon, 12 Aug 2024 22:18:07 +0200
Subject: [PATCH] Install fatrop.hpp

---
 fatrop/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fatrop/CMakeLists.txt b/fatrop/CMakeLists.txt
index 52ee289..d46d3cd 100644
--- a/fatrop/CMakeLists.txt
+++ b/fatrop/CMakeLists.txt
@@ -147,6 +147,8 @@ install(DIRECTORY "${CMAKE_SOURCE_DIR}/fatrop/${incl}" # source directory
         PATTERN "*.h" # select C header files
 )
 endforeach()
+install(FILES fatrop.hpp
+        DESTINATION "include/fatrop/")
 
 set(INCLUDE_INSTALL_DIR "include")
 set(LIBRARY_INSTALL_DIR "lib")
