diff --git a/wrapping/src/CMakeLists.txt b/wrapping/src/CMakeLists.txt
index d636303c..3d281f47 100644
--- a/wrapping/src/CMakeLists.txt
+++ b/wrapping/src/CMakeLists.txt
@@ -61,7 +61,7 @@ execute_process(COMMAND ${PYTHON_EXECUTABLE} "-c" "from distutils.sysconfig impo
                 OUTPUT_VARIABLE PYTHON_SITE_ARCH
                 RESULT_VARIABLE PYTHON_SITEARCH_NOT_FOUND)
 #extract correctly python location: e.g lib64/python2.7/site-packages
-string(REGEX MATCH "[Ll]ib.*packages" PYTHON_REL_SITE_ARCH ${PYTHON_SITE_ARCH})
+set(PYTHON_REL_SITE_ARCH "${CMAKE_INSTALL_PREFIX}/../Lib/site-packages/")
 
 install(TARGETS  "_openmeeg"
         LIBRARY DESTINATION ${PYTHON_REL_SITE_ARCH}
