diff --git a/wrapping/src/CMakeLists.txt b/wrapping/src/CMakeLists.txt
index a7440569..745be104 100644
--- a/wrapping/src/CMakeLists.txt
+++ b/wrapping/src/CMakeLists.txt
@@ -58,18 +58,20 @@ swig_add_library(openmeeg LANGUAGE python SOURCES openmeeg.i)
 swig_link_libraries(openmeeg ${PYTHON_LIBRARIES} OpenMEEG::OpenMEEG)
 
 execute_process(COMMAND ${PYTHON_EXECUTABLE} "-c" "from distutils.sysconfig import get_python_lib; print(get_python_lib(1))"
-                OUTPUT_VARIABLE PYTHON_SITE_ARCH
+                OUTPUT_VARIABLE PYTHON_SITE_ARCH_fail
                 RESULT_VARIABLE PYTHON_SITEARCH_NOT_FOUND)
-#extract correctly python location: e.g lib64/python2.7/site-packages
-string(REGEX MATCH "[Ll]ib.*packages" PYTHON_REL_SITE_ARCH ${PYTHON_SITE_ARCH})
+set(PYTHON_SITE_ARCH_work "${CMAKE_INSTALL_PREFIX}/../Lib/site-packages/")
+file(TO_CMAKE_PATH "${PYTHON_SITE_ARCH_fail}" PYTHON_SITE_ARCH_fail)
+message(STATUS "PYTHON_SITE_ARCH_work ${PYTHON_SITE_ARCH_work}")
+message(STATUS "PYTHON_SITE_ARCH_fail ${PYTHON_SITE_ARCH_fail}")
 
 install(TARGETS  "_openmeeg"
-        LIBRARY DESTINATION ${PYTHON_REL_SITE_ARCH}
-        RUNTIME DESTINATION ${PYTHON_REL_SITE_ARCH}
+        LIBRARY DESTINATION ${PYTHON_SITE_ARCH_work}
+        RUNTIME DESTINATION ${PYTHON_SITE_ARCH_work}
         PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ)
 
 install(FILES  ${OpenMEEG_BINARY_DIR}/wrapping/src/openmeeg.py
-        DESTINATION ${PYTHON_REL_SITE_ARCH}
+        DESTINATION ${PYTHON_SITE_ARCH_fail}
         PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
 
 # Test Python wrapping
