diff --git a/octomap/src/binvox2bt.cpp b/octomap/src/binvox2bt.cpp
index 9b6abec..c2d8048 100644
--- a/octomap/src/binvox2bt.cpp
+++ b/octomap/src/binvox2bt.cpp
@@ -56,7 +56,7 @@
 using namespace std;
 using namespace octomap;
 
-typedef unsigned char byte;
+typedef unsigned char my_byte;
 
 int main(int argc, char **argv)
 {
@@ -227,8 +227,8 @@ int main(int argc, char **argv)
         cout.flush();
             
         // read voxel data
-        byte value;
-        byte count;
+        my_byte value;
+        my_byte count;
         int index = 0;
         int end_index = 0;
         unsigned nr_voxels = 0;
diff --git a/octomap/src/edit_octree.cpp b/octomap/src/edit_octree.cpp
index 75d3fa3..489cdf2 100644
--- a/octomap/src/edit_octree.cpp
+++ b/octomap/src/edit_octree.cpp
@@ -41,8 +41,6 @@
 using namespace std;
 using namespace octomap;
 
-typedef unsigned char byte;
-
 int main(int argc, char **argv)
 {
 
