From bae4e86520800e7e1b634f642764bca76da930cf Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Fri, 4 Aug 2023 09:04:37 +0200
Subject: [PATCH] Force usage of correct python version from environment.

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cffb130..0f70663 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,11 +79,11 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 # Enable use of folder for IDE like VS
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
-# Set the minimum python version to 3.7
-set(PYBIND11_PYTHON_VERSION 3.7)
+# XXX
+set(PYBIND11_PYTHON_VERSION $ENV{PYTHON_VERSION})
 
 # Find Python3
-find_package(Python ${PYBIND11_PYTHON_VERSION} COMPONENTS Interpreter Development REQUIRED)
+find_package(Python ${PYBIND11_PYTHON_VERSION} EXACT COMPONENTS Interpreter Development REQUIRED)
 
 # Save PYTHON_* vars
 set(PYTHON_VERSION_RESET "${PYTHON_VERSION}")
-- 
2.34.1

