diff -ruN squadgen-1.2.1/src/SQuadGen.cpp squadgen-1.2.1-patch/src/SQuadGen.cpp
--- squadgen-1.2.1/src/SQuadGen.cpp	2024-05-11 15:31:02.000000000 -0600
+++ squadgen-1.2.1-patch/src/SQuadGen.cpp	2024-05-12 11:20:44.022644614 -0600
@@ -14,7 +14,6 @@
 ///		or implied warranty.
 ///	</remarks>
 
-#include <png.h>
 #include "netcdfcpp.h"
 
 #include <string>
@@ -195,7 +194,7 @@
 	if ((nRefinementLevel > 0) &&
 		((strRefineRect != "") || (strRefineFile != "") || (fLoadCSRefinementMap))
 	) {
-		int nCommandLineCount = 
+		int nCommandLineCount =
 			  ((strRefineRect != "")?(1):(0))
 			+ ((strRefineFile != "")?(1):(0))
 			+ (fLoadCSRefinementMap?(1):(0));
