diff --git a/cmake/KDAB/modules/KDPySide2ModuleBuild.cmake b/cmake/KDAB/modules/KDPySide2ModuleBuild.cmake
index 29be1168..b79faf43 100644
--- a/cmake/KDAB/modules/KDPySide2ModuleBuild.cmake
+++ b/cmake/KDAB/modules/KDPySide2ModuleBuild.cmake
@@ -134,6 +134,7 @@ macro(create_python_bindings
     set(TARGET_NAME "Py${libraryName}")
     set(MODULE_NAME "${libraryName}")
     add_library(${TARGET_NAME} MODULE ${outputSource})
+    target_link_directories(${TARGET_NAME} PUBLIC "${CMAKE_PREFIX_PATH}/../libs")
 
     set_target_properties(
         ${TARGET_NAME} PROPERTIES
diff --git a/cmake/KDAB/modules/KDPySide6ModuleBuild.cmake b/cmake/KDAB/modules/KDPySide6ModuleBuild.cmake
index 058aac03..5494e727 100644
--- a/cmake/KDAB/modules/KDPySide6ModuleBuild.cmake
+++ b/cmake/KDAB/modules/KDPySide6ModuleBuild.cmake
@@ -134,6 +134,7 @@ macro(create_python_bindings
     set(TARGET_NAME "Py${libraryName}")
     set(MODULE_NAME "${libraryName}")
     add_library(${TARGET_NAME} MODULE ${outputSource})
+    target_link_directories(${TARGET_NAME} PUBLIC "${CMAKE_PREFIX_PATH}/../libs")
 
     set_target_properties(
         ${TARGET_NAME} PROPERTIES
