diff --git a/src/Python/CMakeLists.txt b/src/Python/CMakeLists.txt
index e622ab8..22fb44a 100644
--- src/Python/CMakeLists.txt
+++ src/Python/CMakeLists.txt
@@ -89,8 +89,12 @@ IF(OPENMESH_BUILD_PYTHON_BINDINGS)
 
 						FILE(GLOB SOURCES *.cc *hh)
 						ADD_LIBRARY(openmesh SHARED ${SOURCES})
-
-						install(TARGETS openmesh DESTINATION ${ACG_PROJECT_LIBDIR}/python )
+                        if (WIN32)
+                            set (PYTHON_INSTALL_DIR Lib/site-packages)
+                        else ()
+                            set (PYTHON_INSTALL_DIR ${ACG_PROJECT_LIBDIR}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages)
+                        endif ()
+						install (TARGETS openmesh DESTINATION ${PYTHON_INSTALL_DIR})
 
 						INCLUDE_DIRECTORIES(${PYTHON_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} ../)
 
