diff --git a/CMakeLists.txt b/CMakeLists.txt.edit
index 5650947..6cdc869 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt.edit
@@ -1916,6 +1916,8 @@ if(NOT DISABLE_GMSH_TESTS)
   file(GLOB_RECURSE TESTFILES
        tutorial/*.geo demos/*.geo benchmarks/?d/*.geo benchmarks/extrude/*.geo
        benchmarks/boolean/*.geo)
+  list(REMOVE_ITEM TESTFILES "${CMAKE_SOURCE_DIR}/demos/post_processing/lowmem_anim.geo")
+  list(REMOVE_ITEM TESTFILES "${CMAKE_SOURCE_DIR}/tutorial/t16.geo")
   foreach(TESTFILE ${TESTFILES})
     # use relative path for cygwin/mingw (the pure win exe built with the mingw
     # compilers does not understand a full cygwin-style path)
