diff --git a/Makefile b/Makefile
index 1509e77..e30c7ae 100644
--- a/Makefile
+++ b/Makefile
@@ -183,7 +183,7 @@ JOVETOOLM = $(DMANDIR)/jovetool.$(MANEXT)
 # Add -DDEBUGCRASH if you want to not trap on SEGV, BUS, etc, useful
 # running jove under a debugger (gdb, dbx)
 
-SYSDEFS = -DLinux
+SYSDEFS = -DDarwin
 
 # Select optimization level (flags passed to compiling and linking steps).
 # On most systems: -g for debugging, -O for optimization.
