From 05980594e812bc19f2e835de941be643fbfcfbf8 Mon Sep 17 00:00:00 2001
From: Mattias Ellert <mattias.ellert@physics.uu.se>
Date: Wed, 28 Feb 2018 19:23:53 +0100
Subject: [PATCH] Missing declaration

---
 src/dcap.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/dcap.c b/src/dcap.c
index ad9c241..ce4d311 100644
--- a/src/dcap.c
+++ b/src/dcap.c
@@ -119,6 +119,7 @@ static int get_data_socket();
 static int ascii_open_conversation(struct vsp_node *);
 static int getDataMessage(struct vsp_node *);
 static void getRevision( revision * );
+static void freeRevision( revision * );
 static int init_hostname();
 static void getPortRange();
 static int isActive();
