--- SConstruct	2017-12-12 08:41:10.469790395 -0600
+++ SConstruct	2017-12-12 14:18:59.199747714 -0600
@@ -55,7 +55,7 @@
 # To be used to ensure a PREFIX directory is only created when installing.
 def createPathIsDirCreateWithTarget(target):
   def my_validator(key, val, env):
-    build_targets = (map(str, BUILD_TARGETS))
+    build_targets = (list(map(str, BUILD_TARGETS)))
     if target in build_targets:
       return PathVariable.PathIsDirCreate(key, val, env)
     else:
@@ -157,6 +157,8 @@
                   CPPPATH=['.', ],
                   )
 
+env.Append( ENV = {'PATH': os.environ['PATH'] } )
+
 env.Append(BUILDERS = {
     'GenDef' : 
       Builder(action = sys.executable + ' build/gen_def.py $SOURCES > $TARGET',
@@ -166,7 +168,7 @@
 match = re.search('SERF_MAJOR_VERSION ([0-9]+).*'
                   'SERF_MINOR_VERSION ([0-9]+).*'
                   'SERF_PATCH_VERSION ([0-9]+)',
-                  env.File('serf.h').get_contents(),
+                  env.File('serf.h').get_contents().decode('utf-8'),
                   re.DOTALL)
 MAJOR, MINOR, PATCH = [int(x) for x in match.groups()]
 env.Append(MAJOR=str(MAJOR))
@@ -183,7 +185,7 @@
 
 unknown = opts.UnknownVariables()
 if unknown:
-  print 'Warning: Used unknown variables:', ', '.join(unknown.keys())
+  print('Warning: Used unknown variables:', ', '.join(list(unknown.keys())))
 
 apr = str(env['APR'])
 apu = str(env['APU'])
