--- pyproject.toml       2024-02-11 18:53:54.000000000 -0600
+++ pyproject.toml       2024-02-24 15:45:31.700265695 -0600
@@ -15,7 +15,7 @@
 pybip39 = "0.1.0"
 based58 = "0.1.1"
 bumpversion = "0.6.0"
-"jsonalias" = "0.1.1"
+"jsonalias" = "0.1.2"
 ruff = "0.1.6"
 mnemonic = "0.20"
 mypy = "0.991"
@@ -40,7 +40,7 @@
     "Programming Language :: Python :: Implementation :: CPython",
     "Programming Language :: Python :: Implementation :: PyPy",
 ]
-dependencies = ["typing-extensions >=4.2.0", "jsonalias ==0.1.1"]
+dependencies = ["typing-extensions >=4.2.0", "jsonalias ==0.1.2"]

 [project.urls]
 homepage = "https://github.com/kevinheavey/solders"
