--- lib/gstlal-burst/Makefile.in	2019-10-21 19:14:38.000000000 +0100
+++ lib/gstlal-burst/Makefile.in	2020-02-12 10:16:01.000000000 +0000
@@ -415,7 +415,7 @@
 	sngltriggerrowtype.h sngltriggerrowtype.c

 libgstlalburst_la_CFLAGS = $(AM_CFLAGS) $(LAL_CFLAGS) $(GSTLAL_CFLAGS) $(gstreamer_CFLAGS)
-libgstlalburst_la_LDFLAGS = -version-info $(LIBVERSION) $(AM_LDFLAGS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS)
+libgstlalburst_la_LDFLAGS = -version-info $(LIBVERSION) $(AM_LDFLAGS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(GSL_LIBS)
 all: all-am

 .SUFFIXES:
