diff --git a/libcuckoo/CMakeLists.txt b/libcuckoo/CMakeLists.txt
index 5ec8902..8569889 100644
--- a/libcuckoo/CMakeLists.txt
+++ b/libcuckoo/CMakeLists.txt
@@ -28,7 +28,7 @@ target_include_directories(libcuckoo INTERFACE
 target_link_libraries(libcuckoo INTERFACE Threads::Threads)
 
 # cmake packaging
-set (libcuckoo_pkgloc "share/cmake/libcuckoo")
+set (libcuckoo_pkgloc "lib/cmake/libcuckoo")
 
 write_basic_package_version_file(
     "libcuckoo-config-version.cmake" VERSION ${libcuckoo_VERSION}
