From 3b50bc2b7464ed527d1709b5c7ba86688e6aa3ec Mon Sep 17 00:00:00 2001
From: Jussi <jussi.hietanen@iki.fi>
Date: Mon, 3 Jul 2023 15:59:07 +0300
Subject: [PATCH] Add missing cstdint header for uint8_t type

--- a/src/libserial/SerialPortConstants.h
+++ b/src/libserial/SerialPortConstants.h
@@ -33,6 +33,7 @@
 #pragma once
 
+#include <cstdint>
 #include <limits>
 #include <stdexcept>
 #include <string>
