--- a/src/SerialPort.cpp	2018-12-03 06:35:00.000000000 +0100
+++ b/src/SerialPort.cpp	2026-03-28 20:44:32.056518672 +0100
@@ -37,11 +37,14 @@
 #include <cstdlib>
 #include <cstring>
 #include <fcntl.h>
-#include <linux/serial.h>
 #include <sys/ioctl.h>
 #include <type_traits>
 #include <unistd.h>
 
+#ifdef __linux__
+#include <linux/serial.h>
+#endif
+
 namespace LibSerial 
 {
     /**
--- a/src/SerialStreamBuf.cpp	2018-12-03 06:35:00.000000000 +0100
+++ b/src/SerialStreamBuf.cpp	2026-03-28 20:44:32.056518672 +0100
@@ -37,7 +37,7 @@
 #include <cassert>
 #include <cstring>
 #include <fcntl.h>
-#include <linux/serial.h>
+
 #include <sys/ioctl.h>
 #include <unistd.h>
 
--- a/src/SerialStream.cpp	2018-12-03 06:35:00.000000000 +0100
+++ b/src/SerialStream.cpp	2026-03-28 20:44:32.056518672 +0100
@@ -895,6 +895,7 @@
         throw ;
     }
 
+#ifdef __linux__
     std::vector<std::string>
     SerialStream::GetAvailableSerialPorts()
     try
@@ -916,6 +917,7 @@
         setstate(badbit) ;
         return {} ;
     }
+#endif
     catch (const std::exception&)
     {
         setstate(std::ios_base::failbit) ;
