--- a/src/SerialPort.cpp	2026-03-28 20:47:49.744643430 +0100
+++ b/src/SerialPort.cpp	2026-03-28 23:41:30.468602689 +0100
@@ -1198,6 +1198,7 @@
             baud_rate_as_int = 230400 ;
             break ;
 
+#ifdef __linux__
         case BaudRate::BAUD_460800:
             baud_rate_as_int = 460800 ;
             break ;
@@ -1245,6 +1246,7 @@
         case BaudRate::BAUD_4000000:
             baud_rate_as_int = 4000000 ;
             break ;
+#endif // __linux__
         default:
             // If an incorrect baud rate was specified, throw an exception.
             throw std::runtime_error(ERR_MSG_INVALID_BAUD_RATE) ;
