From 2d0fbf8a5af8e5b1997a672698bfb766dec08b24 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 23 Feb 2021 16:43:33 +0100
Subject: [PATCH] Update Makefile

---
 src/Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 315db85..4fe10c2 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -35,12 +35,12 @@
 
 ### gfortran/MacOS settings:
 ###-------------------------
-FC     = gfortran
+# FC     = gfortran
 USE_C_FILES = yes
-CC = gcc
-CFLAGS =
-LIB           = -L/Users/jbehler/Daten-Goettingen/Codes/LAPACK/lapack-3.8.0 -llapack -lblas
-FFLAGS        = -O3
+# CC = gcc
+# CFLAGS =
+LIB           = ${LDFLAGS} -llapack -lblas
+# FFLAGS        = -O3
 
 ### GWDG
 ###-----
