From 691a1fe67c5cc9a8e5fd1706a20d3ea473a567eb Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Sat, 30 Apr 2022 00:13:08 +0200
Subject: [PATCH] Not all import are required during setup

---
 geomstats/__init__.py | 59 ++++++++++++++++++++++++++++---------------
 setup.py              |  6 +++++
 2 files changed, 45 insertions(+), 20 deletions(-)

diff --git a/geomstats/__init__.py b/geomstats/__init__.py
index d79474d0..66be03c1 100644
--- a/geomstats/__init__.py
+++ b/geomstats/__init__.py
@@ -2,23 +2,42 @@
 
 __version__ = "2.5.0"
 
-import geomstats._backend
-import geomstats._logging
-import geomstats.geometry.discrete_curves
-import geomstats.geometry.euclidean
-import geomstats.geometry.hyperbolic
-import geomstats.geometry.hypersphere
-import geomstats.geometry.invariant_metric
-import geomstats.geometry.landmarks
-import geomstats.geometry.lie_algebra
-import geomstats.geometry.lie_group
-import geomstats.geometry.manifold
-import geomstats.geometry.minkowski
-import geomstats.geometry.poincare_polydisk
-import geomstats.geometry.product_manifold
-import geomstats.geometry.product_riemannian_metric
-import geomstats.geometry.riemannian_metric
-import geomstats.geometry.skew_symmetric_matrices
-import geomstats.geometry.spd_matrices
-import geomstats.geometry.special_euclidean
-import geomstats.geometry.special_orthogonal  # NOQA
+try:
+    # This variable is injected in the __builtins__ by the build
+    # process. It is used to enable importing subpackages of sklearn when
+    # the binaries are not built
+    # mypy error: Cannot determine type of '__GEOMSTATS_SETUP__'
+    __GEOMSTATS_SETUP__  # type: ignore
+except NameError:
+    __GEOMSTATS_SETUP__ = False
+
+if __GEOMSTATS_SETUP__:
+    import sys
+
+    sys.stderr.write(
+        "Partial import of geomstats during build "
+        "process as some run dependencie are missing.\n"
+    )
+    # We are not importing the rest of geomstats during the build
+    # process, as it may not be compiled yet
+else:
+    import geomstats._backend
+    import geomstats._logging
+    import geomstats.geometry.discrete_curves
+    import geomstats.geometry.euclidean
+    import geomstats.geometry.hyperbolic
+    import geomstats.geometry.hypersphere
+    import geomstats.geometry.invariant_metric
+    import geomstats.geometry.landmarks
+    import geomstats.geometry.lie_algebra
+    import geomstats.geometry.lie_group
+    import geomstats.geometry.manifold
+    import geomstats.geometry.minkowski
+    import geomstats.geometry.poincare_polydisk
+    import geomstats.geometry.product_manifold
+    import geomstats.geometry.product_riemannian_metric
+    import geomstats.geometry.riemannian_metric
+    import geomstats.geometry.skew_symmetric_matrices
+    import geomstats.geometry.spd_matrices
+    import geomstats.geometry.special_euclidean
+    import geomstats.geometry.special_orthogonal  # NOQA
diff --git a/setup.py b/setup.py
index b908cbe5..b3a563de 100644
--- a/setup.py
+++ b/setup.py
@@ -1,3 +1,9 @@
+"""Setup geomstats."""
+
+import builtins
+
 import setuptools
 
+builtins.__GEOMSTATS_SETUP__ = True
+
 setuptools.setup()
-- 
2.17.1

