diff --git a/CMakeLists.txt b/CMakeLists.txt
index d2eb3c51..0e38d6dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -282,11 +282,8 @@ if(BUILD_EXTERNAL AND IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/External)
     add_subdirectory(External)
 endif()
 
-if(NOT TARGET SPIRV-Tools-opt)
-    set(ENABLE_OPT OFF)
-endif()
-
 if(ENABLE_OPT)
+    find_package(SPIRV-Tools-opt REQUIRED)
     message(STATUS "optimizer enabled")
     add_definitions(-DENABLE_OPT=1)
 else()
@@ -301,7 +298,6 @@ add_subdirectory(OGLCompilersDLL)
 if(ENABLE_GLSLANG_BINARIES)
     add_subdirectory(StandAlone)
 endif()
-add_subdirectory(SPIRV)
 if(ENABLE_HLSL)
     add_subdirectory(hlsl)
 endif()
