--- a/SConscript	2021-01-09 12:35:30.000000000 +1000
+++ b/SConscript	2021-08-02 06:40:00.535074593 +1000
@@ -660,7 +660,11 @@
           "SHLINKFLAGS",
           ]:
     if i in os.environ:
-        env.MergeFlags(Split(os.getenv(i)))
+        if i == 'CPPFLAGS' or i == 'CFLAGS' or i == 'CXXFLAGS':
+            p = env.ParseFlags(os.getenv(i))
+            env.MergeFlags(p)
+        else:
+            env.MergeFlags(Split(os.getenv(i)))
 
 
 # Keep scan-build options in the environment
@@ -1155,6 +1159,7 @@
             confdefs.append("#define HAVE_LIBUSB 1\n")
             try:
                 usbflags = pkg_config('libusb-1.0')
+                usbflags.append(["-ludev"])
             except OSError:
                 announce("pkg_config is confused about the state "
                          "of libusb-1.0.")
@@ -1529,7 +1534,7 @@
 
 target_python_path = ''
 py_config_text = str(eval(PYTHON_CONFIG_CALL))
-python_libdir = str(eval(PYTHON_LIBDIR_CALL))
+python_libdir = os.environ.get('SP_DIR')
 config.env['xgps_deps'] = False
 
 if not cleaning and not helping and config.env['python']:
@@ -2526,6 +2531,7 @@
      not env['nostrip'] and not sys.platform.startswith('darwin'))):
     env.AddPostAction(binaryinstall, '$STRIP $TARGET')
 
+python_libdir = os.environ.get('SP_DIR')
 python_module_dir = str(python_libdir) + os.sep + 'gps'
 
 python_modules_install = env.Install(DESTDIR + python_module_dir,

