--- pypy3/rpython/translator/platform/darwin.py	2019-03-01 10:57:51.000000000 +0100
+++ pypy3/rpython/translator/platform/darwin.py	2019-03-01 10:58:07.000000000 +0100
@@ -9,7 +9,7 @@
 # since 10.5, so we use that as minimum requirement. Bumped to 10.7
 # to allow the use of thread-local in __thread in C.
 #
-DARWIN_VERSION_MIN = '-mmacosx-version-min=10.7'
+DARWIN_VERSION_MIN = '-mmacosx-version-min=10.9'
 
 class Darwin(posix.BasePosix):
     name = "darwin"
