--- pypy3/pypy/module/_minimal_curses/fficurses.py	2018-04-24 08:05:26.000000000 +0200
+++ pypy3/pypy/module/_minimal_curses/fficurses.py	2018-06-19 23:44:26.000000000 +0200
@@ -35,12 +35,7 @@
         pass
 
 def try_eci():
-    for eci in try_tools():
-        yield eci.merge(ExternalCompilationInfo(includes=['curses.h',
-                                                          'term.h']))
-    for eci1 in try_cflags():
-        for eci2 in try_ldflags():
-            yield eci1.merge(eci2)
+    yield ExternalCompilationInfo(includes=['curses.h', 'term.h']).merge(ExternalCompilationInfo.from_pkg_config("ncursesw"))
 
 def guess_eci():
     for eci in try_eci():
