--- pypy3/lib_pypy/pyrepl/_minimal_curses.py	2018-06-19 16:04:27.000000000 +0200
+++ pypy3/lib_pypy/pyrepl/_minimal_curses.py	2018-06-19 16:07:03.000000000 +0200
@@ -16,11 +16,12 @@
 
 
 def _find_clib():
-    trylibs = ['ncurses', 'curses']
+    trylibs = ['ncursesw']
 
     for lib in trylibs:
         path = ctypes.util.find_library(lib)
         if path:
+
             return path
     raise ImportError("curses library not found")
 
