diff --git a/src/Pegasus/cli/pegasus-globus-online-init.py b/src/Pegasus/cli/pegasus-globus-online-init.py
index 7becaf59b..45da0be53 100755
--- a/src/Pegasus/cli/pegasus-globus-online-init.py
+++ b/src/Pegasus/cli/pegasus-globus-online-init.py
@@ -5,7 +5,7 @@ import os
 from optparse import OptionParser

 import globus_sdk
-from ConfigParser import ConfigParser
+from configparser import ConfigParser

 client_id = "d7382f5a-4ea3-4b69-b094-99c392fc820d"
 config_file = os.path.expanduser("~/.pegasus/globus.conf")
