From 9e8b4c862ef6556da1865885e3897b5bec5456ff Mon Sep 17 00:00:00 2001
From: AntoinePrv <AntoinePrv@users.noreply.github.com>
Date: Mon, 25 Jan 2021 15:22:08 -0500
Subject: [PATCH] Leave default library type

---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7f22692..426c13b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -119,7 +119,7 @@ file(GLOB headers
 )
 
     # create soplex library with pic
-    add_library(libsoplex-pic STATIC ${sources})
+    add_library(libsoplex-pic ${sources})
     set_target_properties(libsoplex-pic PROPERTIES
     POSITION_INDEPENDENT_CODE on
     VERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR}.${SOPLEX_VERSION_PATCH}.${SOPLEX_VERSION_SUB}
@@ -128,7 +128,7 @@ add_dependencies(libsoplex-pic soplex_update_githash)
 target_link_libraries(libsoplex-pic ${libs})
 
 # create soplex library without pic
-add_library(libsoplex STATIC ${sources})
+add_library(libsoplex ${sources})
 set_target_properties(libsoplex PROPERTIES
     VERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR}.${SOPLEX_VERSION_PATCH}.${SOPLEX_VERSION_SUB}
     SOVERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR})
-- 
2.29.2

