diff --git a/client/python/opensysml/binary.py b/client/python/opensysml/binary.py
index 7c6b416..adc7ab4 100644
--- a/client/python/opensysml/binary.py
+++ b/client/python/opensysml/binary.py
@@ -384,8 +384,12 @@ def get_binary_path():
     Returns:
         str: Absolute path to binary (e.g. ~/.opensysml/bin/sysml-grpc)
     """
-    base_dir = os.path.expanduser('~/.opensysml/bin')
-    return os.path.join(base_dir, service_binary_name())
+
+    return os.path.join(
+        os.environ['CONDA_PREFIX'],
+        ('Library/bin' if os.name == 'nt' else 'bin'),
+        service_binary_name()
+    )
 
 
 def is_executable(path):
