diff --git a/init.py b/init.py
index b2b1dde..c36e381 100644
--- a/init.py
+++ b/init.py
@@ -53,14 +53,8 @@ CDockFocusController = _ads.CDockFocusController
 CDockManager = _ads.CDockManager
 CDockSplitter = _ads.CDockSplitter
 CDockOverlay = _ads.CDockOverlay
-CDockOverlayCross = _ads.CDockOverlayCross
 CDockWidget = _ads.CDockWidget
 CDockWidgetTab = _ads.CDockWidgetTab
-CDockingStateReader = _ads.CDockingStateReader
 CElidingLabel = _ads.CElidingLabel
 CFloatingDockContainer = _ads.CFloatingDockContainer
-CFloatingDragPreview = _ads.CFloatingDragPreview
-IFloatingWidget = _ads.IFloatingWidget
 CIconProvider = _ads.CIconProvider
-CSpacerWidget = _ads.CSpacerWidget
-CTitleBarButton = _ads.CTitleBarButton
