From c014f66bee5e41355843614337120df669cef429 Mon Sep 17 00:00:00 2001
From: Henrik Finsberg <henriknf@simula.no>
Date: Thu, 19 Feb 2026 09:16:36 +0100
Subject: [PATCH] Lower absolute tolerance in vtkhdf tests

---
 tests/test_vtkhdf.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/test_vtkhdf.py b/tests/test_vtkhdf.py
index 566d5bb..5a79f07 100644
--- a/tests/test_vtkhdf.py
+++ b/tests/test_vtkhdf.py
@@ -105,7 +105,7 @@ def test_write_point_data(dtype, tmp_path, cell_type):
             filename=filename, name="u", mesh=grid, time=tj, backend="vtkhdf"
         )
         v_ref = Function(u.function_space, dtype=u.x.array.dtype)
-        atol = 10 * np.finfo(u.x.array.dtype).eps
+        atol = 8 * np.finfo(u.x.array.dtype).eps
         v_ref.interpolate(lambda x: f(x, tj))
         np.testing.assert_allclose(u.x.array, v_ref.x.array, atol=atol)
 
@@ -124,7 +124,7 @@ def test_write_point_data(dtype, tmp_path, cell_type):
             filename=blocked_file, name="u", mesh=grid, time=tk, backend="vtkhdf"
         )
         v_ref = Function(u.function_space, dtype=u.x.array.dtype)
-        atol = 10 * np.finfo(u.x.array.dtype).eps
+        atol = 8 * np.finfo(u.x.array.dtype).eps
         v_ref.interpolate(lambda x: g(x, tk))
         np.testing.assert_allclose(u.x.array, v_ref.x.array, atol=atol)
 
@@ -160,7 +160,7 @@ def test_write_cell_data(dtype, tmp_path, cell_type):
             filename=filename, name="u", mesh=grid, time=tj, backend="vtkhdf"
         )
         v_ref = Function(u.function_space, dtype=u.x.array.dtype)
-        atol = 10 * np.finfo(u.x.array.dtype).eps
+        atol = 8 * np.finfo(u.x.array.dtype).eps
         v_ref.interpolate(lambda x: f(x, tj))
         np.testing.assert_allclose(u.x.array, v_ref.x.array, atol=atol)
 
@@ -179,7 +179,7 @@ def test_write_cell_data(dtype, tmp_path, cell_type):
             filename=blocked_file, name="u", mesh=grid, time=tk, backend="vtkhdf"
         )
         v_ref = Function(u.function_space, dtype=u.x.array.dtype)
-        atol = 10 * np.finfo(u.x.array.dtype).eps
+        atol = 8 * np.finfo(u.x.array.dtype).eps
         v_ref.interpolate(lambda x: g(x, tk))
         np.testing.assert_allclose(u.x.array, v_ref.x.array, atol=atol)
 
-- 
2.50.1 (Apple Git-155)

