From 848fae2cae64350c0f87dd7bf8fe208aa7c93154 Mon Sep 17 00:00:00 2001
From: Henrik Finsberg <henriknf@simula.no>
Date: Thu, 19 Feb 2026 13:28:59 +0100
Subject: [PATCH 3/3] More fixes for tolerance in vtkhdf

---
 tests/test_vtkhdf.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/test_vtkhdf.py b/tests/test_vtkhdf.py
index ac7d4f3..968288c 100644
--- a/tests/test_vtkhdf.py
+++ b/tests/test_vtkhdf.py
@@ -254,7 +254,7 @@ def test_write_meshtags(dtype, tmp_path, generate_reference_map):
         )
         org_maps[dim] = generate_reference_map(mesh, et, comm, root)
 
-    tol = 10 * np.finfo(dtype).eps
+    tol = 15 * np.finfo(dtype).eps
     # Read in hex grid from second time step
     hex_mesh = io4dolfinx.read_mesh(
         filename, comm, time=1.0, backend_args={"name": "hex"}, backend="vtkhdf"
@@ -288,7 +288,7 @@ def test_write_meshtags(dtype, tmp_path, generate_reference_map):
 
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
 def test_read_write_pointdata(dtype, tmp_path):
-    tol = 10 * np.finfo(dtype).eps
+    tol = 15 * np.finfo(dtype).eps
 
     comm = MPI.COMM_WORLD
     tmp_path = comm.bcast(tmp_path, root=0)
@@ -360,7 +360,7 @@ def test_read_write_pointdata(dtype, tmp_path):
 
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
 def test_read_write_celldata(dtype, tmp_path):
-    tol = 10 * np.finfo(dtype).eps
+    tol = 15 * np.finfo(dtype).eps
 
     comm = MPI.COMM_WORLD
     tmp_path = comm.bcast(tmp_path, root=0)
@@ -436,7 +436,7 @@ def test_read_write_celldata(dtype, tmp_path):
 
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
 def test_read_write_mix_data(dtype, tmp_path):
-    tol = 10 * np.finfo(dtype).eps
+    tol = 15 * np.finfo(dtype).eps
 
     mesh = create_unit_square(MPI.COMM_WORLD, 5, 7, dtype=dtype)
 
-- 
2.50.1 (Apple Git-155)

