From a1621e778c75b709dc280589284cf8a94f477f80 Mon Sep 17 00:00:00 2001
From: Henrik Finsberg <henriknf@simula.no>
Date: Thu, 26 Feb 2026 09:53:52 +0100
Subject: [PATCH 4/4] Skip tests requiring petsc if it is not available

---
 tests/test_legacy_readers.py | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/tests/test_legacy_readers.py b/tests/test_legacy_readers.py
index 37f083f..4de5fec 100644
--- a/tests/test_legacy_readers.py
+++ b/tests/test_legacy_readers.py
@@ -14,7 +14,6 @@ import dolfinx
 import numpy as np
 import pytest
 import ufl
-from dolfinx.fem.petsc import LinearProblem
 
 from io4dolfinx import (
     read_cell_data,
@@ -110,6 +109,11 @@ def test_legacy_function(backend):
     f = ufl.conditional(ufl.gt(x[0], 0.5), x[1], 2 * x[0])
     L = ufl.inner(f, v) * ufl.dx
 
+    try:
+        from dolfinx.fem.petsc import LinearProblem
+    except ImportError:
+        pytest.skip("dolfinx.fem.petsc.LinearProblem not available")
+
     uh = dolfinx.fem.Function(V)
     if "petsc_options_prefix" in inspect.signature(LinearProblem.__init__).parameters.keys():
         extra_options = {"petsc_options_prefix": "legacy_test"}
@@ -149,6 +153,11 @@ def test_read_legacy_function_from_checkpoint(backend):
     f = ufl.conditional(ufl.gt(x[0], 0.5), x[1], 2 * x[0])
     L = ufl.inner(f, v) * ufl.dx
 
+    try:
+        from dolfinx.fem.petsc import LinearProblem
+    except ImportError:
+        pytest.skip("dolfinx.fem.petsc.LinearProblem not available")
+
     uh = dolfinx.fem.Function(V)
     if "petsc_options_prefix" in inspect.signature(LinearProblem.__init__).parameters.keys():
         extra_options = {"petsc_options_prefix": "legacy_checkpoint_test"}
-- 
2.50.1 (Apple Git-155)

