From 0212dab1ce140314c5f6e07358fe94256d09ad29 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 13 Feb 2021 14:35:17 +0100
Subject: [PATCH] Update makefile

---
 src/application/makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/application/makefile b/src/application/makefile
index c8a2b6529..3cc63d646 100644
--- a/src/application/makefile
+++ b/src/application/makefile
@@ -27,9 +27,9 @@ include $(PROJECT_DIR)/src/makefile.$(COMP)
 # PROJECT COMPILER AND FLAGS
 ###############################################################################
 # Include directories and linking flags are target specific (see below).
-CC=$(PROJECT_CC)
+# CC=$(PROJECT_CC)
 MPICC=$(PROJECT_MPICC)
-CFLAGS=$(PROJECT_CFLAGS)
+# CFLAGS=$(PROJECT_CFLAGS)
 OPTIONS+=$(PROJECT_OPTIONS)
 DEBUG=$(PROJECT_DEBUG)
 ifeq ($(MODE), test)
@@ -72,9 +72,9 @@ CLEAN_APP=$(patsubst %, clean-%, $(APP))
 all: $(APP_CORE) $(APP_TRAIN) nnp-train
 
 # Applications which require only libnnp:
-$(APP_CORE): INCLUDES=-I./ -I$(PROJECT_INCLUDE)/
+$(APP_CORE): INCLUDES=-I./ -I$(PREFIX)/include/
 ifeq ($(MODE), shared)
-$(APP_CORE): LDFLAGS=-L$(PROJECT_LIB) -lnnp
+$(APP_CORE): LDFLAGS=-L$(PREFIX)/lib -lnnp
 else
 $(APP_CORE): LDFLAGS=$(PROJECT_LIB)/libnnp.a
 endif
@@ -85,9 +85,9 @@ $(APP_CORE):
 
 # Applications which require libnnp and libnnptrain:
 # (nnp-train itself is actually a special case).
-$(APP_TRAIN): INCLUDES=-I./ -I$(PROJECT_INCLUDE)/ -I$(PROJECT_GSL)
+$(APP_TRAIN): INCLUDES=-I./ -I$(PREFIX)/include/
 ifeq ($(MODE), shared)
-$(APP_TRAIN): LDFLAGS=-L$(PROJECT_LIB) -lnnptrain -lnnp -lgsl $(PROJECT_LDFLAGS_BLAS)
+$(APP_TRAIN): LDFLAGS=-L$(PREFIX)/lib -lnnptrain -lnnp -lgsl -lblas
 else
 $(APP_TRAIN): LDFLAGS=$(PROJECT_LIB)/libnnptrain.a $(PROJECT_LIB)/libnnp.a -lgsl $(PROJECT_LDFLAGS_BLAS)
 endif
@@ -97,9 +97,9 @@ $(APP_TRAIN):
 	cp $@ $(PROJECT_BIN)/
 
 # Separate rule for nnp-train, requires Eigen.
-nnp-train: INCLUDES=-I./ -I$(PROJECT_INCLUDE)/ -I$(PROJECT_GSL) -I$(PROJECT_EIGEN)
+nnp-train: INCLUDES=-I./ -I$(PREFIX)/include/ -I$(PREFIX)/include/eigen3/
 ifeq ($(MODE), shared)
-nnp-train: LDFLAGS=-L$(PROJECT_LIB) -lnnptrain -lnnp -lgsl $(PROJECT_LDFLAGS_BLAS)
+nnp-train: LDFLAGS=-L$(PREFIX)/lib -lnnptrain -lnnp -lgsl -lblas
 else
 nnp-train: LDFLAGS=$(PROJECT_LIB)/libnnptrain.a $(PROJECT_LIB)/libnnp.a -lgsl $(PROJECT_LDFLAGS_BLAS)
 endif
