From 3375363d79c88be7a76b64d5772b93162bf9f987 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 13 Feb 2021 09:52:44 +0100
Subject: [PATCH] Update makefile.gnu

---
 src/makefile.gnu | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/makefile.gnu b/src/makefile.gnu
index 856e4cf6a..4712a0c24 100644
--- a/src/makefile.gnu
+++ b/src/makefile.gnu
@@ -11,14 +11,14 @@ PROJECT_EIGEN=/usr/include/eigen3/
 ###############################################################################
 # COMPILERS AND FLAGS
 ###############################################################################
-PROJECT_CC=g++
+PROJECT_CC=$(CC)
 PROJECT_MPICC=mpic++
 # OpenMP parallelization is disabled by default, add flag "-fopenmp" to enable.
-PROJECT_CFLAGS=-O3 -march=native -std=c++11
-PROJECT_CFLAGS_MPI=-Wno-long-long
+PROJECT_CFLAGS=$(CFLAGS)
+PROJECT_CFLAGS_MPI=$(CFLAGS)
 PROJECT_DEBUG=-g -pedantic-errors -Wall -Wextra
 PROJECT_TEST=--coverage -fno-default-inline -fno-inline -fno-inline-small-functions -fno-elide-constructors
-PROJECT_AR=ar
+PROJECT_AR=$(AR)
 PROJECT_ARFLAGS=-rcsv
 PROJECT_CFLAGS_BLAS=
 PROJECT_LDFLAGS_BLAS=-lblas -lgsl -lgslcblas
