From 1163456f9f0419d7b8732f874c3b58a6d709bbe8 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 13 Feb 2021 10:49:24 +0100
Subject: [PATCH] shared patch

---
 src/makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/makefile b/src/makefile
index 580c91d4e..ff15e4378 100644
--- a/src/makefile
+++ b/src/makefile
@@ -19,13 +19,13 @@ COMP=gnu
 # Possible modes are "static", "shared" and "test".
 # Note: For target "pynnp" there is no "static" mode, "shared" is automatically
 # used instead.
-MODE=static
+MODE=shared
 
 # Installation directory for binaries.
-INSTALL_BIN=$(HOME)/local/bin
+INSTALL_BIN=$(PREFIX)/bin
 
 # Installation directory for libraries.
-INSTALL_LIB=$(HOME)/local/lib
+INSTALL_LIB=$(PREFIX)/lib
 
 
 ###############################################################################
@@ -36,7 +36,7 @@ INSTALL_LIB=$(HOME)/local/lib
 # space-separated list of the following available options:
 #    * LAMMPS
 #    * CabanaMD
-INTERFACES=LAMMPS CabanaMD
+INTERFACES=LAMMPS
 
 
 ###############################################################################
