diff -ruN '--exclude=*.pyc' '--exclude=__pycache__' '--exclude=test' pyflowy-0.0.1.orig/subprojects/flowy/flowy/include/definitions.hpp pyflowy-0.0.1/subprojects/flowy/flowy/include/definitions.hpp
--- pyflowy-0.0.1.orig/subprojects/flowy/flowy/include/definitions.hpp	2026-02-16 23:14:04.522955556 +0100
+++ pyflowy-0.0.1/subprojects/flowy/flowy/include/definitions.hpp	2026-02-16 23:22:33.738554302 +0100
@@ -3,14 +3,14 @@
 // Copyright 2023--present Flowy developers
 
 #include "xtensor-blas/xlinalg.hpp"
-#include "xtensor/xbuilder.hpp"
-#include "xtensor/xmanipulation.hpp"
-#include "xtensor/xmath.hpp"
-#include "xtensor/xtensor_forward.hpp"
-#include <xtensor/xcsv.hpp>
-#include <xtensor/xfixed.hpp>
-#include <xtensor/xtensor.hpp>
-#include <xtensor/xview.hpp>
+#include "xtensor/generators/xbuilder.hpp"
+#include "xtensor/misc/xmanipulation.hpp"
+#include "xtensor/core/xmath.hpp"
+#include "xtensor/core/xtensor_forward.hpp"
+#include <xtensor/io/xcsv.hpp>
+#include <xtensor/containers/xfixed.hpp>
+#include <xtensor/containers/xtensor.hpp>
+#include <xtensor/views/xview.hpp>
 
 namespace Flowy
 {
diff -ruN '--exclude=*.pyc' '--exclude=__pycache__' '--exclude=test' pyflowy-0.0.1.orig/subprojects/flowy/flowy/include/dump_csv.hpp pyflowy-0.0.1/subprojects/flowy/flowy/include/dump_csv.hpp
--- pyflowy-0.0.1.orig/subprojects/flowy/flowy/include/dump_csv.hpp	2026-02-16 23:14:04.522961187 +0100
+++ pyflowy-0.0.1/subprojects/flowy/flowy/include/dump_csv.hpp	2026-02-16 23:22:33.739704735 +0100
@@ -8,8 +8,8 @@
 #include <string>
 #include <utility>
 
-#include <xtensor/xtensor.hpp>
-#include <xtensor/xtensor_config.hpp>
+#include <xtensor/containers/xtensor.hpp>
+#include <xtensor/core/xtensor_config.hpp>
 
 namespace Flowy::Utility
 {
diff -ruN '--exclude=*.pyc' '--exclude=__pycache__' '--exclude=test' pyflowy-0.0.1.orig/subprojects/flowy/flowy/include/topography.hpp pyflowy-0.0.1/subprojects/flowy/flowy/include/topography.hpp
--- pyflowy-0.0.1.orig/subprojects/flowy/flowy/include/topography.hpp	2026-02-16 23:14:04.522994373 +0100
+++ pyflowy-0.0.1/subprojects/flowy/flowy/include/topography.hpp	2026-02-16 23:22:33.732490259 +0100
@@ -3,9 +3,9 @@
 // Copyright 2023--present Flowy developers
 #include "flowy/include/definitions.hpp"
 #include "flowy/include/lobe.hpp"
-#include "xtensor/xbuilder.hpp"
-#include "xtensor/xmath.hpp"
-#include "xtensor/xsort.hpp"
+#include "xtensor/generators/xbuilder.hpp"
+#include "xtensor/core/xmath.hpp"
+#include "xtensor/misc/xsort.hpp"
 #include <iterator>
 #include <memory>
 #include <optional>
diff -ruN '--exclude=*.pyc' '--exclude=__pycache__' '--exclude=test' pyflowy-0.0.1.orig/subprojects/flowy/src/netcdf_file.cpp pyflowy-0.0.1/subprojects/flowy/src/netcdf_file.cpp
--- pyflowy-0.0.1.orig/subprojects/flowy/src/netcdf_file.cpp	2026-02-16 23:14:04.523589054 +0100
+++ pyflowy-0.0.1/subprojects/flowy/src/netcdf_file.cpp	2026-02-16 23:22:33.731212927 +0100
@@ -2,8 +2,8 @@
 // Copyright 2023--present Flowy developers
 #include "flowy/include/netcdf_file.hpp"
 #include "flowy/include/definitions.hpp"
-#include "xtensor/xmanipulation.hpp"
-#include "xtensor/xmath.hpp"
+#include "xtensor/misc/xmanipulation.hpp"
+#include "xtensor/core/xmath.hpp"
 #include <fmt/format.h>
 #include <fmt/ostream.h>
 #include <netcdf.h>
diff -ruN '--exclude=*.pyc' '--exclude=__pycache__' '--exclude=test' pyflowy-0.0.1.orig/subprojects/flowy/src/simulation.cpp pyflowy-0.0.1/subprojects/flowy/src/simulation.cpp
--- pyflowy-0.0.1.orig/subprojects/flowy/src/simulation.cpp	2026-02-16 23:14:04.523598295 +0100
+++ pyflowy-0.0.1/subprojects/flowy/src/simulation.cpp	2026-02-16 23:22:33.732592343 +0100
@@ -11,9 +11,9 @@
 #include "fmt/core.h"
 #include "pdf_cpplib/include/probability_dist.hpp"
 #include "pdf_cpplib/include/reservoir_sampling.hpp"
-#include "xtensor/xbuilder.hpp"
-#include "xtensor/xmath.hpp"
-#include "xtensor/xsort.hpp"
+#include "xtensor/generators/xbuilder.hpp"
+#include "xtensor/core/xmath.hpp"
+#include "xtensor/misc/xsort.hpp"
 #include <fmt/chrono.h>
 #include <fmt/format.h>
 #include <fmt/ranges.h>
diff -ruN '--exclude=*.pyc' '--exclude=__pycache__' '--exclude=test' pyflowy-0.0.1.orig/subprojects/flowy/src/topography.cpp pyflowy-0.0.1/subprojects/flowy/src/topography.cpp
--- pyflowy-0.0.1.orig/subprojects/flowy/src/topography.cpp	2026-02-16 23:14:04.523612836 +0100
+++ pyflowy-0.0.1/subprojects/flowy/src/topography.cpp	2026-02-16 23:22:33.734050696 +0100
@@ -5,8 +5,8 @@
 #include "flowy/include/definitions.hpp"
 #include "flowy/include/lobe.hpp"
 #include "thirdparty/tsl/robin_set.h"
-#include "xtensor/xbuilder.hpp"
-#include "xtensor/xtensor_forward.hpp"
+#include "xtensor/generators/xbuilder.hpp"
+#include "xtensor/core/xtensor_forward.hpp"
 #include <fmt/ranges.h>
 #include <fmt/std.h>
 #include <algorithm>
