From 5a95e7dca114bc7894f038ffecfadef70435a663 Mon Sep 17 00:00:00 2001
From: Uwe Korn <uwelk@xhochy.com>
Date: Wed, 21 Aug 2024 18:55:20 +0000
Subject: [PATCH] Collect built targets

---
 src/Makevars | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Makevars b/src/Makevars
index 34bac1b..8e8821e 100644
--- a/src/Makevars
+++ b/src/Makevars
@@ -26,6 +26,7 @@ $(STATLIB):
 	fi && \
 		export PATH="$(PATH):$(HOME)/.cargo/bin" && \
 			cargo build $(CRAN_FLAGS) --lib --release --manifest-path=./rust/Cargo.toml --target-dir $(TARGET_DIR) && \
+			find $(TARGET_DIR) -name 'libb64.*' -exec cp '{}' $(LIBDIR) ';'
 			echo `cargo --version` && echo `rustc --version`;
 		rm -Rf $(CARGOTMP) $(VENDOR_DIR) $(LIBDIR)/build; \
 
