# Imported from basf2/externals/patches

--- Makefile
+++ Makefile
@@ -3,7 +3,7 @@
 #
 # Doug Wright
 
-default: setup lib test
+default: setup lib
 
 lib:
 	$(MAKE) -C src
--- src/Makefile
+++ src/Makefile
@@ -6,10 +6,11 @@
 OBJ = $(SRC:%.cc=%.o)
 DEP = $(SRC:%.cc=.%.d)  
 LIB = ../lib/libCRY.a
+SLIB = ../lib/libCRY.so
 
-CPPFLAGS += -Wall
+CPPFLAGS += -Wall -fPIC
 
-default: depend $(LIB)
+default: depend $(LIB) $(SLIB)
 
 depend: $(DEP)
 
@@ -23,8 +24,15 @@
 	@ar -crs $(LIB) $(OBJ)
 	@echo "Done making $(LIB)"
 
+$(SLIB): $(OBJ)
+	@if (test -f $(SLIB) ) then (rm $(SLIB)) fi
+	$(CXX) -shared \
+	-o $(SLIB) \
+	$(OBJ)
+	@echo "Done making $(SLIB)"
+
 clean:
-	rm -f *.o .*.d ../lib/libCRY.a 
+	rm -f *.o .*.d ../lib/libCRY.a ../lib/libCRY.so
 
 .PHONY: depend clean
