diff --git a/guiqwt/curve.py b/guiqwt/curve.py
index fa86834..6746947 100644
--- a/guiqwt/curve.py
+++ b/guiqwt/curve.py
@@ -1399,9 +1399,12 @@ class CurvePlot(BasePlot):
         # Sometimes, an obscure exception happens when we quit an application
         # because if we don't remove the eventFilter it can still be called
         # after the filter object has been destroyed by Python.
-        canvas = self.canvas()
-        if canvas:
-            canvas.removeEventFilter(self.filter)
+        try:
+            canvas = self.canvas()
+            if canvas:
+                canvas.removeEventFilter(self.filter)
+        except RuntimeError:
+            pass
 
     # generic helper methods        
     def canvas2plotitem(self, plot_item, x_canvas, y_canvas):
