From 80f2995dc85335719602ec3ec3d7b6b32c5445a0 Mon Sep 17 00:00:00 2001
From: Yannik Tausch <yannik.tausch@quantco.com>
Date: Thu, 31 Oct 2024 18:45:50 +0100
Subject: [PATCH] remove tests from distribution

---
 MANIFEST.in | 1 +
 setup.py    | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/MANIFEST.in b/MANIFEST.in
index 5dd2aa4..ef437b0 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1 +1,2 @@
 recursive-include streamlit_searchbox/frontend/build *
+recursive-exclude tests *
diff --git a/setup.py b/setup.py
index 7480aee..8ebf3c3 100644
--- a/setup.py
+++ b/setup.py
@@ -9,7 +9,7 @@ setuptools.setup(
     + "and provides a list of suggestions based on a provided function",
     long_description_content_type="text/plain",
     url="https://github.com/m-wrzr/streamlit-searchbox",
-    packages=setuptools.find_packages(),
+    packages=setuptools.find_packages(exclude=("tests",)),
     include_package_data=True,
     classifiers=[],
     python_requires=">=3.9, !=3.9.7",
-- 
2.46.1

