diff --git a/build.rs b/build.rs
index 1140455..4127ef3 100644
--- a/build.rs
+++ b/build.rs
@@ -3,7 +3,6 @@ use std::path::PathBuf;
 use std::process::Command;
 
 fn main() {
-    println!("cargo:rerun-if-changed=ffi/webgpu-headers/webgpu.h");
     println!("cargo:rerun-if-changed=ffi/wgpu.h");
 
     #[rustfmt::skip]
@@ -33,7 +32,6 @@ fn main() {
         ("WGPUTextureView", "WGPUTextureViewImpl"),
     ];
     let mut builder = bindgen::Builder::default()
-        .header("ffi/webgpu-headers/webgpu.h")
         .header("ffi/wgpu.h")
         .parse_callbacks(Box::new(bindgen::CargoCallbacks))
         .blocklist_function("wgpuGetProcAddress")
diff --git a/ffi/wgpu.h b/ffi/wgpu.h
index 6e576ae..dd36031 100644
--- a/ffi/wgpu.h
+++ b/ffi/wgpu.h
@@ -1,7 +1,7 @@
 #ifndef WGPU_H_
 #define WGPU_H_
 
-#include "webgpu-headers/webgpu.h"
+#include "webgpu.h"
 
 typedef enum WGPUNativeSType {
     // Start at 6 to prevent collisions with webgpu STypes

