diff --git a/libkineto/libkineto_defs.bzl b/libkineto/libkineto_defs.bzl
index 57e230d..51c7141 100644
--- a/libkineto/libkineto_defs.bzl
+++ b/libkineto/libkineto_defs.bzl
@@ -73,6 +73,7 @@ def get_libkineto_public_headers():
         "include/ThreadUtil.h",
         "include/libkineto.h",
         "include/time_since_epoch.h",
+        "include/LoggingAPI.h",
     ]
 
 # kineto code should be updated to not have to
