diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-03-30 08:52:11.085826500 -0400
+++ b/CMakeLists.txt	2023-03-30 08:52:54.062112700 -0400
@@ -7,18 +7,18 @@
 if(SKBUILD)
     set(Python_EXECUTABLE "${PYTHON_EXECUTABLE}")
 
-    if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
-        message(STATUS "Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
-        file(DOWNLOAD "https://github.com/conan-io/cmake-conan/raw/v0.16.1/conan.cmake" "${CMAKE_BINARY_DIR}/conan.cmake")
-    endif()
-    include(${CMAKE_BINARY_DIR}/conan.cmake)
-    conan_cmake_configure()
-    conan_cmake_autodetect(settings)
-    conan_cmake_install(
-            PATH_OR_REFERENCE ${CMAKE_SOURCE_DIR}
-            BUILD missing
-            SETTINGS ${settings}
-    )
+    # if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
+    #     message(STATUS "Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
+    #     file(DOWNLOAD "https://github.com/conan-io/cmake-conan/raw/v0.16.1/conan.cmake" "${CMAKE_BINARY_DIR}/conan.cmake")
+    # endif()
+    # include(${CMAKE_BINARY_DIR}/conan.cmake)
+    # conan_cmake_configure()
+    # conan_cmake_autodetect(settings)
+    # conan_cmake_install(
+    #         PATH_OR_REFERENCE ${CMAKE_SOURCE_DIR}
+    #         BUILD missing
+    #         SETTINGS ${settings}
+    # )
 else()
     include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
     conan_basic_setup()
