diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2023-03-30 08:52:11.085826500 -0400
+++ b/src/CMakeLists.txt	2023-03-30 08:52:54.062112700 -0400
@@ -17,9 +17,9 @@
 target_link_libraries( _pycharls
     PRIVATE
         pybind11::headers
-        charls::charls
-        Eigen3::Eigen3
-        fmt::fmt
+        charls
+        Eigen3
+        fmt
         )
 
 install(TARGETS _pycharls
