From ca076edfd04597674316eee3be10bff1d8cb2186 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Tue, 17 Sep 2024 14:52:51 +0200
Subject: [PATCH] remove uuid

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 5e4ac53..fdfc512 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -9,7 +9,6 @@ readme = "README.md"
 python = "^3.11"
 numpy = "^1.26.2"
 pathlib = "^1.0.1"
-uuid = "^1.30"
 jupyterlab = "^4.0.8"
 yachalk = "^0.1.5"
 groq = "^0.6.0"
