diff --git a/Cargo.toml b/Cargo.toml
index 998ef4e..bc0d67e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,15 +28,12 @@ tracing-subscriber = { version = "0.3", features = ["env-filter"] }
 
 # For macos, use vendored gssapi
 [target.'cfg(target_os = "macos")'.dependencies]
-rdkafka = { version = "0.28.0", features = [ "cmake-build", "gssapi-vendored", "ssl-vendored" ] }
+rdkafka = { version = "0.28.0", features = [ "cmake-build", "gssapi", "ssl" ] }
 
 # For everything else, use system sasl-lib
 [target.'cfg(not(target_os = "macos"))'.dependencies]
 rdkafka = { version = "0.28.0", features = [ "cmake-build", "gssapi", "ssl" ] }
 
-[target.'cfg(target_os = "windows")'.dependencies]
-sasl2-sys = {version="0.1.20",features=["vendored","openssl-vendored"]}
-
 [build-dependencies]
 openssl-src = { version = "111", default-features = false, features = ["force-engine"] }
 
