From 77f2419ce8cd75e6552276ddedfe0bc145cf2d04 Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Thu, 18 Oct 2018 11:07:27 -0500
Subject: [PATCH] lower numpy req

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index c21d005..c85d3ec 100644
--- a/setup.py
+++ b/setup.py
@@ -59,7 +59,7 @@ class build_ext(_build_ext):
         _build_ext.build_extensions(self)
 
 
-SETUP_REQUIREMENTS = {'numpy': '1.12'}
+SETUP_REQUIREMENTS = {'numpy': '1.11'}
 INSTALL_REQUIREMENTS = SETUP_REQUIREMENTS.copy()
 INSTALL_REQUIREMENTS.update({'scipy': '0.19',
                              'pandas': '0.20',
-- 
2.18.0

