From cd1d184f62ee45fa0030a85ede698b7aed865aa3 Mon Sep 17 00:00:00 2001
From: Daeho Ro <40587651+daeho-ro@users.noreply.github.com>
Date: Wed, 22 Oct 2025 10:19:48 +0900
Subject: [PATCH] feature: update CMakeFiles for corrosion fetch

---
 src/taskchampion-cpp/CMakeLists.txt | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/src/taskchampion-cpp/CMakeLists.txt b/src/taskchampion-cpp/CMakeLists.txt
index 95846700122d6a0d36e06b273cd6e1c97362b3a8..e5b8d37bae265884a439abbb5ef30911d169ca5b 100644
--- a/src/taskchampion-cpp/CMakeLists.txt
+++ b/src/taskchampion-cpp/CMakeLists.txt
@@ -36,16 +36,15 @@ if(SYSTEM_CORROSION)
   set(CMAKE_REQUIRE_FIND_PACKAGE_Corrosion ON)
 endif()
 
-FetchContent_Declare(Corrosion
-    GIT_REPOSITORY https://github.com/corrosion-rs/corrosion.git
-    GIT_TAG        ${TASK_CORROSION_VERSION}
-    FIND_PACKAGE_ARGS CONFIG
-)
-FetchContent_MakeAvailable(Corrosion)
-
 if(SYSTEM_CORROSION)
   find_package(Corrosion REQUIRED)
 else()
+  FetchContent_Declare(Corrosion
+      GIT_REPOSITORY https://github.com/corrosion-rs/corrosion.git
+      GIT_TAG        ${TASK_CORROSION_VERSION}
+      FIND_PACKAGE_ARGS CONFIG
+  )
+  FetchContent_MakeAvailable(Corrosion)
   add_subdirectory(${CMAKE_SOURCE_DIR}/src/taskchampion-cpp/corrosion)
 endif()
 
