--- ./iceoryx_posh/include/iceoryx_posh/internal/gateway/channel.inl	2025-11-05 12:07:33.451205398 -0800
+++ ./iceoryx_posh/include/iceoryx_posh/internal/gateway/channel.inl	2025-11-05 12:07:46.230647677 -0800
@@ -52,7 +52,7 @@
 constexpr inline bool Channel<IceoryxTerminal, ExternalTerminal>::operator==(
     const Channel<IceoryxTerminal, ExternalTerminal>& rhs) const noexcept
 {
-    return m_service == rhs.getService();
+    return m_service == rhs.getServiceDescription();
 }
 
 template <typename IceoryxTerminal, typename ExternalTerminal>
