--- ./iceoryx_hoofs/test/moduletests/test_cxx_function_ref.cpp	2025-11-05 11:57:19.441791521 -0800
+++ ./iceoryx_hoofs/test/moduletests/test_cxx_function_ref.cpp	2025-11-05 11:57:27.903955514 -0800
@@ -15,6 +15,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include <cstdint>
 #include "iceoryx_hoofs/cxx/attributes.hpp"
 #include "iceoryx_hoofs/cxx/function_ref.hpp"
 #include "test.hpp"

--- ./iceoryx_hoofs/test/moduletests/test_cxx_reference_counter.cpp	2025-11-05 11:58:22.514084370 -0800
+++ ./iceoryx_hoofs/test/moduletests/test_cxx_reference_counter.cpp	2025-11-05 11:58:32.083294162 -0800
@@ -15,6 +15,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include <cstdint>
 #include "iceoryx_hoofs/internal/cxx/reference_counter.hpp"
 #include "test.hpp"
 
--- ./iceoryx_hoofs/test/moduletests/test_cxx_type_traits.cpp	2025-11-05 11:58:48.393659571 -0800
+++ ./iceoryx_hoofs/test/moduletests/test_cxx_type_traits.cpp	2025-11-05 11:59:00.676941094 -0800
@@ -15,6 +15,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include <cstdint>
 #include "iceoryx_hoofs/cxx/type_traits.hpp"
 #include "test.hpp"

--- iceoryx_hoofs/test/moduletests/test_atomic_relocatable_ptr.cpp	2025-11-05 17:53:49.522416010 -0800
+++ iceoryx_hoofs/test/moduletests/test_atomic_relocatable_ptr.cpp	2025-11-05 17:53:59.351352913 -0800
@@ -20,6 +20,7 @@
 #include "test.hpp"
 
 #include <cstring>
+#include <cstdint>
 
 namespace
 { 
