--- ./iceoryx_posh/cmake/iceoryxversions.cmake	2025-11-05 12:48:15.570087058 -0800
+++ ./iceoryx_posh/cmake/iceoryxversions.cmake	2025-11-05 12:48:37.545210004 -0800
@@ -15,11 +15,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Find GIT revisions
-execute_process(COMMAND
-  git describe --match=None --always --abbrev=40 --dirty
-  WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-  OUTPUT_VARIABLE ICEORYX_SHA1
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
+set (ICEORYX_SHA1 "000000000000")
 
 string(TIMESTAMP ICEORYX_BUILDDATE UTC)
 

