cmake patch

diff -r 035e53e53730 -r bc079a6aebea CMakeLists.txt
--- a/CMakeLists.txt	Tue Jul 05 18:33:42 2016 +0100
+++ b/CMakeLists.txt	Thu Oct 20 12:43:02 2016 +0200
@@ -24,6 +24,8 @@
 option(HAVE_SOUND "Sound support." ON)
 option(HAVE_3DS_IMPORT_CAPABILITIES "3ds import capabilities." ON)
 
+include_directories(${CMAKE_INSTALL_PREFIX}/include)
+
 # it is assumed that simage library has been installed to same directory as Coin
 set(COIN_LINK_LIBSIMAGE "no" CACHE STRING "Use and linkage of simage library.")
 set_property(CACHE COIN_LINK_LIBSIMAGE PROPERTY STRINGS no runtime dynamic static)
