diff -r 035e53e53730 include/Inventor/system/inttypes.h.cmake.in
--- a/include/Inventor/system/inttypes.h.cmake.in	Tue Jul 05 18:33:42 2016 +0100
+++ b/include/Inventor/system/inttypes.h.cmake.in	Thu Feb 16 10:15:31 2017 -0800
@@ -76,7 +76,7 @@
    configure?  20010711 mortene. */
 
 /* The <inttypes.h> header file. */
-#cmakedefine HAVE_INTTYPES_H
+#cmakedefine HAVE_INTTYPES_H 1
 /* The <stdint.h> header file. */
 #cmakedefine HAVE_STDINT_H 1
 /* The <sys/types.h> header file. */
diff -r 035e53e53730 src/config.h.cmake.in
--- a/src/config.h.cmake.in	Tue Jul 05 18:33:42 2016 +0100
+++ b/src/config.h.cmake.in	Thu Feb 16 10:15:31 2017 -0800
@@ -208,7 +208,7 @@
 #define HAVE_INTPTR_T 1
 
 /* Define to 1 if you have the <inttypes.h> header file. */
-#cmakedefine HAVE_INTTYPES_H
+#cmakedefine HAVE_INTTYPES_H 1
 
 /* Define to 1 if you have the <io.h> header file. */
 #cmakedefine HAVE_IO_H
