--- work/CMakeLists.txt.orig	2018-06-10 09:22:12.000000000 -0700
+++ work/CMakeLists.txt	2018-06-10 16:59:25.000000000 -0700
@@ -10,11 +10,17 @@
 include(CppcheckTargets)
 include(ExternalProject)
 
-find_package(Git REQUIRED)
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git" AND IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/.git")
+  find_package(Git REQUIRED)
 
-# Information from git
-# ====================
-include(LIEFGit)
+  # Information from git
+  # ====================
+  include(LIEFGit)
+else()
+  set(LIEF_VERSION_MAJOR "0")
+  set(LIEF_VERSION_MINOR "9")
+  set(LIEF_VERSION_PATCH "0")
+endif()
 
 # LIEF Project
 # ============
