From f3709db789ace6222c80333a165f54b6902012ef Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 15 Jan 2019 00:23:02 -0600
Subject: [PATCH] MachO: Fix has_dynamic_symbol_command()

---
 src/MachO/Binary.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/MachO/Binary.cpp b/src/MachO/Binary.cpp
index d9035d2..dc7e348 100644
--- a/src/MachO/Binary.cpp
+++ b/src/MachO/Binary.cpp
@@ -1709,7 +1709,7 @@ const SymbolCommand& Binary::symbol_command(void) const {
 // DynamicSymbolCommand command
 // ++++++++++++++++++++++++++++
 bool Binary::has_dynamic_symbol_command(void) const {
-  return this->has_command<SymbolCommand>();
+  return this->has_command<DynamicSymbolCommand>();
 }
 
 DynamicSymbolCommand& Binary::dynamic_symbol_command(void) {
-- 
2.17.2 (Apple Git-113)

