--- a/setup.py
+++ b/setup.py
@@ -61,7 +61,7 @@
     long_description_content_type="text/markdown",
     url="https://github.com/google/differential-privacy/",
     packages=setuptools.find_packages(),
-    install_requires=_parse_requirements("requirements.txt"),
+    # install_requires=_parse_requirements("requirements.txt"),
     classifiers=[
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
