--- a/setup.py
+++ b/setup.py
@@ -21,15 +21,8 @@
 
 
 def _get_version():
-  """Parses the version number from VERSION file."""
-  with open(os.path.join(here, "VERSION")) as f:
-    try:
-      version_line = next(
-          line for line in f if not line.startswith("\"\"\""))
-    except StopIteration:
-      raise ValueError("Version not defined in VERSION")
-    else:
-      return version_line.strip('\n \'"')
+  """Return the version number from conda build."""
+  return "0.4.3"
 
 
 def _parse_requirements(path):
