From f428ce32c4ff4083cef284946005d4ff63ed680a Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sun, 31 May 2020 13:59:03 -0400
Subject: [PATCH 2/2] Make CMAKE_INSTALL_PREFIX override conditional

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e67fc09..1b3f62c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -477,7 +477,9 @@ endif()
 # prepare files here...
 IF(WIN32)
   # install into place in build-dir
-  SET( CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/INSTALL/FILES CACHE FILEPATH "install prefix" FORCE)
+  if (NOT DEFINED CMAKE_INSTALL_PREFIX)
+    SET( CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/INSTALL/FILES CACHE FILEPATH "install prefix" FORCE)
+  endif ()
   SET( INSTALL_WIN_FILES AUTHORS ChangeLog COPYING NEWS README )
   INSTALL(FILES ${INSTALL_WIN_FILES} DESTINATION ${CMAKE_INSTALL_PREFIX})
   # install Eriks droplets
-- 
2.24.1

