diff -r b236375ad1a8 src/filenameparse.cc
--- a/src/filenameparse.cc	Tue Nov 26 14:00:20 2019 +0100
+++ b/src/filenameparse.cc	Sun Apr 26 22:04:24 2020 -0700
@@ -22,7 +22,7 @@
 // Life is tough and then you die.  -- Jack Dempsey
 
 
-#include <filesystem>
+#include <experimental/filesystem>
 #include <numeric>
 #include <string>
 #include <vector>
@@ -39,7 +39,7 @@
 #define PARENT_DIRECTORY ".."
 
 
-typedef std::filesystem::path basic_path;
+typedef std::experimental::filesystem::path basic_path;
 
 
 namespace enblend {
