diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49f5080c..cb7f64ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -761,6 +761,7 @@ add_custom_target( sparse-testing DEPENDS ${sparse-testing} )
 # ----------------------------------------
 # what to install
 install( TARGETS magma magma_sparse ${blas_fix}
+         OPTIONAL
          RUNTIME DESTINATION bin
          LIBRARY DESTINATION lib
          ARCHIVE DESTINATION lib )
