diff --git a/setup.py b/setup.py
index 3cd8c99..1a6429d 100755
--- a/setup.py
+++ b/setup.py
@@ -27,21 +27,6 @@ setup(
     url="https://github.com/rigetticomputing/pyquil.git",
     packages=find_packages(exclude=["*.tests", "*.tests.*", "tests.*", "tests"]),
     license="LICENSE",
-    install_requires=[
-        'requests >= 2.4.2',
-        'numpy >= 1.10',
-        'matplotlib >= 1.5',
-        'typing >= 3.6',
-        'urllib3 >= 1.21.1',
-        "antlr4-python2-runtime>=4.7; python_version < '3'",
-        "antlr4-python3-runtime>=4.7; python_version >= '3'",
-    ],
-    setup_requires=['pytest-runner'],
-    tests_require=[
-        'pytest >= 3.0.0',
-        'mock',
-    ],
-    test_suite='pyquil.tests',
     entry_points={
         'console_scripts': ['pyquil-config-setup=pyquil.setup.pyquil_config_setup:main']
     },
