From 1bac6f78cfb77b5c4ab4814e8693a0b7ab6762f9 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Tue, 13 Aug 2019 07:43:35 -0400
Subject: [PATCH 1/2] Rename and replace entrypoint stub exe

---
 conda/core/path_actions.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/conda/core/path_actions.py b/conda/core/path_actions.py
index 01c6938f7..70ca2166b 100644
--- a/conda/core/path_actions.py
+++ b/conda/core/path_actions.py
@@ -258,8 +258,12 @@ class LinkPathAction(CreateInPrefixPathAction):
     def create_python_entry_point_windows_exe_action(cls, transaction_context, package_info,
                                                      target_prefix, requested_link_type,
                                                      entry_point_def):
-        source_directory = context.conda_prefix
-        source_short_path = 'Scripts/conda.exe'
+        if lexists(join(sys.prefix, 'entry_point_base.exe')):
+            source_directory = sys.prefix
+            source_short_path = 'entry_point_base.exe'
+        else:
+            source_directory = context.conda_prefix
+            source_short_path = 'Scripts/conda.exe'
         command, _, _ = parse_entry_point_def(entry_point_def)
         target_short_path = "Scripts/%s.exe" % command
         source_path_data = PathDataV1(
--
2.22.0
