From a3edb498065b15298ebb298492c83d1124b36138 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Tue, 13 Aug 2019 07:46:10 -0400
Subject: [PATCH 2/2] Manipulate PATH directly instead of _call_ing conda.bat

---
 conda/utils.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/conda/utils.py b/conda/utils.py
index 4b689f888..bd940491c 100644
--- a/conda/utils.py
+++ b/conda/utils.py
@@ -362,7 +362,8 @@ def wrap_subprocess_call(on_win, root_prefix, prefix, dev_mode, debug_wrapper_sc
             # after all!
             # fh.write("@FOR /F \"tokens=100\" %%F IN ('chcp') DO @SET CONDA_OLD_CHCP=%%F\n")
             # fh.write('@chcp 65001>NUL\n')
-            fh.write('{0}CALL \"{1}\" activate \"{2}\"\n'.format(silencer, conda_bat, prefix))
+            # fh.write('{0}CALL \"{1}\" activate \"{2}\"\n'.format(silencer, conda_bat, prefix))
+            fh.write('SET PATH={p};{p}\\Library\\mingw-w64\\bin;{p}\Library\\usr\\bin;{p}\\Library\\bin;{p}\\Scripts;{p}\\bin;%PATH%'.format(p=prefix))
             fh.write("{}IF %ERRORLEVEL% NEQ 0 EXIT /b %ERRORLEVEL%\n".format(silencer))
             if debug_wrapper_scripts:
                 fh.write('echo *** environment after *** 1>&2\n')
-- 
2.22.0

