--- sdl2/__init__.py
+++ sdl2/__init__.py
@@ -1,4 +1,16 @@
 """SDL2 wrapper package"""
+# Direct this module to Ana/Miniconda's lib folder, where the SDL2 libraries
+# can be found. Only do this if the variable hasn't already been set (maybe
+# the user has set a custom path already).
+import os
+
+if os.name == "posix":
+	os.environ["PYSDL2_DLL_PATH"] = os.path.abspath(os.path.join(
+		os.path.dirname(__file__), os.pardir, os.pardir, os.pardir))
+elif os.name == "nt":
+	os.environ["PYSDL2_DLL_PATH"] = os.path.abspath(os.path.join(
+		os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, "Library", "bin"))
+
 from .dll import get_dll_file, _bind
 from ctypes import c_int as _cint
 
