diff --git a/setup.py b/setup.py
index 13d32cf..d81b645 100644
--- a/setup.py
+++ b/setup.py
@@ -5,7 +5,7 @@ from setuptools import setup
 HERE = pathlib.Path(__file__).parent
 
 # The text of the README file
-README = (HERE / "README.md").read_text()
+README = (HERE / "README.md").read_text(encoding="utf-8")
 
 # This call to setup() does all the work
 setup(
@@ -20,4 +20,4 @@ setup(
     packages=["bertviz"],
     include_package_data=True,
     install_requires=["transformers>=2.0", "torch>=1.0", "tqdm", "boto3", "requests", "regex", "sentencepiece", "IPython>=7.14"],
-)
\ No newline at end of file
+)
