--- CMakeLists.txt.orig	2015-05-12 02:35:22.286561235 +0000
+++ CMakeLists.txt	2015-05-12 02:35:57.101519807 +0000
@@ -36,7 +36,7 @@
        "set ON to use libc++ for standard C++ library (only effective for clang)"
        ON)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -DDISABLE_ALIGNAS")
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     if(USE_LIBCXX)
     	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
@@ -57,5 +57,3 @@
 #set(CMAKE_BUILD_TYPE Release)
 
 add_subdirectory(src bin)
-add_subdirectory(gtest bingtest)
-add_subdirectory(test bintest)
--- src/libsakura/localdef.h.orig	2015-05-12 02:31:17.774852194 +0000
+++ src/libsakura/localdef.h	2015-05-12 02:33:32.852691457 +0000
@@ -166,7 +166,7 @@
   #define __has_builtin(x) 0  // Compatibility with non-clang compilers.
 #endif
 
-#if defined(__GNUG__) && (!defined(__clang__) || __has_builtin(__builtin_assume_aligned))
+#if defined(__GNUG__) && !defined(__clang__) && __has_builtin(__builtin_assume_aligned)
 /**
  * @~japanese
  * @brief @a ptr がsakuraのアライメント要件を満たしていると見なし、そのアドレスを返す。
--- src/convolution.cc.orig	2016-01-08 09:08:10.000000000 +0000
+++ src/convolution.cc	2016-04-11 22:28:29.603543227 +0000
@@ -38,6 +38,7 @@
 #include <climits>
 #include <fstream>
 #include <sstream>
+#include <stdexcept>
 
 #include <libsakura/sakura.h>
 #include <libsakura/localdef.h>
--- src/gridding.cc.orig	2015-05-12 02:38:13.036358050 +0000
+++ src/gridding.cc	2015-05-12 02:38:24.548344351 +0000
@@ -27,6 +27,7 @@
 #include <cassert>
 #include <cmath>
 #include <climits>
+#include <cstdint>
 
 #include "libsakura/sakura.h"
 #include "libsakura/localdef.h"
