From 9b53861672de42d284473b88c2d5862531bd9ee8 Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Wed, 20 Feb 2019 00:16:17 +0100
Subject: [PATCH] Disable tcl/tk version check

---
 scilab/configure | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/scilab/configure b/scilab/configure
index e8a17ec1c2f..b47abf035fa 100755
--- a/scilab/configure
+++ b/scilab/configure
@@ -32942,14 +32942,6 @@ done
 	fi
   done
 
-  #perform tk tests if tcl test passed
-  if test $TCL_LIB_OK = 0; then
-	if test $TCL_INCLUDE_OK = 0; then
-        	as_fn_error $? "no header file tcl.h found for 8.4* or 8.5*. Please install the dev library of tcl 8.4 or 8.5" "$LINENO" 5
-        else
-			as_fn_error $? "header file tcl.h has been found for 8.4* or 8.5* but no corresponding tcl library (ie libtcl8.4.so or libtcl8.5.so)" "$LINENO" 5
-        fi
-  fi
 # Check if X11/Xlib.h is available or not (tk.h needs it ...)
    for ac_header in X11/Xlib.h
 do :
@@ -33058,18 +33050,6 @@ case $host_os in
 esac
 
 TK_LIB=tk$TK_VERSION
-if test $TK_VERSION_OK = 1; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ($TK_VERSION) yes" >&5
-$as_echo "($TK_VERSION) yes" >&6; }
-else
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ($TK_VERSION) no" >&5
-$as_echo "($TK_VERSION) no" >&6; }
-	if $TK_VERSION = "can't happen"; then
-		as_fn_error $? "can(t happen" "$LINENO" 5
-        else
-	   as_fn_error $? "you need at least version 8.4 of tk" "$LINENO" 5
-	fi
-fi
 CFLAGS=$saved_cflags
 CPPFLAGS=$saved_cppflags
 
@@ -33257,18 +33237,8 @@ done
 	fi
   done
 
-  if test $TK_LIB_OK = 0; then
-	if test $TK_INCLUDE_OK = 0; then
-        	as_fn_error $? "no header file tk.h found for 8.4* or 8.5*. Please install the dev library of tk 8.4 or 8.5" "$LINENO" 5
-        else
-		as_fn_error $? "header file tk.h has been found for 8.4* or 8.5* but no corresponding tk library (libtk8.4.so or libtk8.5.so)" "$LINENO" 5
-        fi
-
-  else
         TCLTK_LIBS=$TK_LIB" $TCLTK_LIBS"
         WITH_TKSCI=yes
-  fi
-
 
     # set variables
     if test "$WITH_TKSCI" = yes; then
-- 
2.20.1

