diff --git a/tests/conftest.py b/tests/conftest.py
index 8673c90..ff933b9 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,16 +1,18 @@
 """Pytest conftest module containing common test configuration and fixtures."""
+
+import os
 import shutil
 
 import pytest
-from sphinx.testing.path import path
+from pathlib import Path
 
 pytest_plugins = "sphinx.testing.fixtures"
 
 
 def copy_srcdir_to_tmpdir(srcdir, tmp):
-    srcdir = path(__file__).parent.abspath() / srcdir
-    tmproot = tmp / path(srcdir).basename()
-    shutil.copytree(srcdir, tmproot)
+    srcdir = Path(__file__).parent.resolve / srcdir
+    tmproot = Path(tmp) / srcdir.name
+    shutil.copytree(srcdir.as_posix(), tmproot.as_posix())
     return tmproot
 
 
