diff --git a/etc/config.sh/functions b/etc/config.sh/functions

--- a/etc/config.sh/functions
+++ b/etc/config.sh/functions
@@ -104,3 +104,3 @@ if [ -z "$foamVar_prefix
        then
            unset foamVar_prefix foamVar_end
-            return 1
+            return 0
@@ -135,3 +135,3 @@ fi

        unset foamVar_prefix foamVar_end
-        return 1    # Nothing set
+        return 0    # Nothing set
