From ee3983628f5b12134793c96639fc778b145ddd89 Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Fri, 31 Jul 2020 22:54:44 +0200
Subject: [PATCH] Patch get-version for conda-forge

---
 scripts/get-version | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/get-version b/scripts/get-version
index 234006ebfb..55b1a1eb63 100755
--- a/scripts/get-version
+++ b/scripts/get-version
@@ -1,4 +1,7 @@
 #!/bin/sh
+set -eu
+echo ${PKG_VERSION}
+exit 0
 
 # optionally take commit-ish as argument for debugging purposes
 head=$1
