From b1d700639cb3ec0806982c6fac5a7771e9aca2f6 Mon Sep 17 00:00:00 2001
From: Sebastian Ehlert <sehlert@microsoft.com>
Date: Fri, 25 Jul 2025 11:40:48 +0200
Subject: [PATCH] Include package data when installing mattergen

---
 pyproject.toml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pyproject.toml b/pyproject.toml
index 6df3406..faa3a81 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -93,6 +93,9 @@ csv-to-dataset = "mattergen.scripts.csv_to_dataset:main"
 [tool.setuptools.packages.find]
 include = ["mattergen*"]
 
+[tool.setuptools.package-data]
+"*" = ["**.yaml", "**.csv", "**.pt", "**.json"]
+
 [tool.uv.sources]
 torch = { index = "pytorch_linux",  marker = "sys_platform == 'linux'" }
 torchvision = { index = "pytorch_linux",  marker = "sys_platform == 'linux'" }
