From 918ed72257c3658ab32584441fcb37a1faef22db Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 24 Feb 2022 22:52:36 +1100
Subject: [PATCH 1/5] point to shared libcutlass.so

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8fb73e67..77f798b3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,7 +98,7 @@ if(BUILD_PYTHON_MODULE)
     if(NOT("${CUTLASS_INCLUDE_DIR}" STREQUAL "") AND NOT("${CUTLASS_LIBRARY_DIR}" STREQUAL ""))
         set(CUTLASS_SRC ${PYTHON_SRC_PATH}/cutlass.cc)
         add_definitions(-DWITH_CUTLASS_BINDINGS)
-        set(CUTLASS_LIBRARIES "cutlass.a")
+        set(CUTLASS_LIBRARIES "libcutlass.so")
     endif()
     include_directories("." ${PYTHON_SRC_PATH} ${PYTHON_INCLUDE_DIRS} ${CUTLASS_INCLUDE_DIR})
     link_directories(${PYTHON_LINK_DIRS} ${CUTLASS_LIBRARY_DIR})
-- 
2.35.1.windows.2

