From 6d7d78f83c037106f409dc4efc23244a08b53fc7 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 24 Feb 2022 23:20:34 +1100
Subject: [PATCH 2/4] patch include paths in setup.py

---
 python/setup.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/python/setup.py b/python/setup.py
index f77b9266..2f6417c8 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -77,7 +77,8 @@ class CMakeBuild(build_ext):
             self.build_extension(ext)
 
     def build_extension(self, ext):
-        llvm_include_dir, llvm_library_dir = get_llvm()
+        llvm_include_dir = os.environ["LLVM_INCLUDE_DIRS"]
+        llvm_library_dir = os.environ["LLVM_LIBRARY_DIR"]
         # self.debug = True
         extdir = os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.path)))
         # create build directories
@@ -88,7 +89,9 @@ class CMakeBuild(build_ext):
         if not os.path.exists(llvm_build_dir):
             os.makedirs(llvm_build_dir)
         # python directories
-        python_include_dirs = [distutils.sysconfig.get_python_inc()] + ['/usr/local/cuda/include']
+        python_include_dirs = [
+            distutils.sysconfig.get_python_inc()
+        ] + ("CUDA_HOME" in os.environ) * [os.environ.get("CUDA_HOME", "") + "/include"]
         cmake_args = [
             "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=" + extdir,
             "-DBUILD_TUTORIALS=OFF",
-- 
2.35.1.windows.2

