From 3fb9b65ca108143fff3bc1fbac2c9a6f2f14164f Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 25 Feb 2022 10:01:52 +1100
Subject: [PATCH 3/4] add install step

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 77f798b3..f3c6dfb7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,3 +121,5 @@ if(BUILD_PYTHON_MODULE)
     endif()
     target_link_libraries(triton ${CUTLASS_LIBRARIES} ${PYTHON_LDFLAGS})
 endif()
+
+install(TARGETS triton LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-- 
2.32.0.windows.2

