From 033bb80737e3bcdf133729a0bbdc9bb54a0eb281 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 4 Mar 2022 12:02:00 +1100
Subject: [PATCH 4/4] point to system CUDA instead of vendored header

---
 include/triton/driver/dispatch.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/triton/driver/dispatch.h b/include/triton/driver/dispatch.h
index 5503baca..27ead520 100755
--- a/include/triton/driver/dispatch.h
+++ b/include/triton/driver/dispatch.h
@@ -7,8 +7,8 @@
 #include <dlfcn.h>
 
 //CUDA Backend
-#include "triton/external/CUDA/cuda.h"
-#include "triton/external/CUDA/nvml.h"
+#include "cuda.h"
+#include "nvml.h"
 
 //// HIP backend
 //#define __HIP_PLATFORM_AMD__
-- 
2.35.1.windows.2

