diff -Naur /home/laurent/Downloads/SWASHES-1.03.00/CMakeLists.txt SWASHES-1.03.00/CMakeLists.txt
--- /home/laurent/Downloads/SWASHES-1.03.00/CMakeLists.txt	1969-12-31 18:00:00.000000000 -0600
+++ SWASHES-1.03.00/CMakeLists.txt	2017-12-20 08:45:35.317797639 -0600
@@ -0,0 +1,33 @@
+cmake_minimum_required (VERSION 3.0)
+
+project(SWASHES)
+
+set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+if(UNIX)
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -Wall -pedantic -O3")
+endif(UNIX)
+
+if(APPLE)
+    set(CMAKE_INSTALL_NAME_DIR @executable_path)
+    set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
+endif(APPLE)
+
+# includes and source files
+include_directories(Headers)
+include_directories(Sources)
+file(GLOB SWASHES_SOURCES Sources/*.cpp)
+
+# executable
+add_executable(swashes ${SWASHES_SOURCES})
+target_compile_definitions(swashes PRIVATE CLE=TRUE)
+
+if(WIN32)
+    target_link_libraries(swashes)
+else(WIN32)
+    target_link_libraries(swashes m pthread)
+endif(WIN32)
+
+# install
+install(TARGETS swashes DESTINATION bin)
diff -Naur /home/laurent/Downloads/SWASHES-1.03.00/Makefile SWASHES-1.03.00/Makefile
--- /home/laurent/Downloads/SWASHES-1.03.00/Makefile	2016-01-29 09:19:31.000000000 -0600
+++ SWASHES-1.03.00/Makefile	1969-12-31 18:00:00.000000000 -0600
@@ -1,79 +0,0 @@
-########################## Makefile ##############################################
-#
-# author Carine Lucas <carine.lucas@univ-orleans.fr> (2010-2012)
-# version 1.01.01
-# date 2012-03-06
-#
-# copyright License Cecill-V2 <http://www.cecill.info/licences/Licence_CeCILL_V2-en.html>
-#
-# (c) CNRS - Universite d'Orleans - INRA (France)
-#
-# This file is part of SWASHES software. 
-# <https://sourcesup.renater.fr/projects/swashes/> 
-#
-# SWASHES = Shallow-Water Analytic Solutions for Hydraulic and 
-# Environmental Studies.
-# This software is a computer program whose purpose is to compute analytic
-# solutions for Shallow-Water equations.
-#
-# LICENSE
-#
-# This software is governed by the CeCILL license under French law and
-# abiding by the rules of distribution of free software.  You can  use, 
-# modify and/ or redistribute the software under the terms of the CeCILL
-# license as circulated by CEA, CNRS and INRIA at the following URL
-# <http://www.cecill.info>. 
-#
-# As a counterpart to the access to the source code and  rights to copy,
-# modify and redistribute granted by the license, users are provided only
-# with a limited warranty  and the software's author,  the holder of the
-# economic rights,  and the successive licensors  have only  limited
-# liability. 
-#
-# In this respect, the user's attention is drawn to the risks associated
-# with loading,  using,  modifying and/or developing or reproducing the
-# software by the user in light of its specific status of free software,
-# that may mean  that it is complicated to manipulate,  and  that  also
-# therefore means  that it is reserved for developers  and  experienced
-# professionals having in-depth computer knowledge. Users are therefore
-# encouraged to load and test the software's suitability as regards their
-# requirements in conditions enabling the security of their systems and/or 
-# data to be ensured and,  more generally, to use and operate it in the 
-# same conditions as regards security. 
-#
-# The fact that you are presently reading this means that you have had
-# knowledge of the CeCILL license and that you accept its terms.
-#
-##################################################################################
-
-CC=@g++
-
-SRCDIR=Sources
-HEADDIR=Headers
-OBJDIR=obj
-BINDIR=bin
-
-CFLAGS= -I$(HEADDIR) -g -Wall -pedantic -O3
-
-EXEC=swashes
-
-SRC=$(wildcard $(SRCDIR)/*.cpp)
-OBJ=$(SRC:$(SRCDIR)/%.cpp=$(OBJDIR)/%.o)
-
-all: $(EXEC)
-
-$(EXEC): $(OBJ) $(OBJDIR)/swashes.o
-	$(CC) -o $(BINDIR)/$@ $^ $(CFLAGS)
-	@ echo " swashes => $(BINDIR)"
-
-$(OBJDIR)/%.o: $(SRCDIR)/%.cpp $(HEADDIR)/%.hpp
-	$(CC) -o $@ -c $< $(CFLAGS) 
-
-$(OBJDIR)/swashes.o: $(SRCDIR)/swashes.cpp
-	$(CC) -o $@ -c $< $(CFLAGS)
-
-clean:
-	@rm -f $(OBJDIR)/*
-
-cleanall: clean
-	@rm -f $(BINDIR)/swashes
diff -Naur /home/laurent/Downloads/SWASHES-1.03.00/Sources/bump.cpp SWASHES-1.03.00/Sources/bump.cpp
--- /home/laurent/Downloads/SWASHES-1.03.00/Sources/bump.cpp	2016-01-29 09:19:31.000000000 -0600
+++ SWASHES-1.03.00/Sources/bump.cpp	2017-12-20 09:18:29.399587622 -0600
@@ -53,6 +53,9 @@
  * knowledge of the CeCILL license and that you accept its terms.
  *
  ******************************************************************************/
+#if defined(WIN32)
+#include <minmax.h>
+#endif
 
 #include "bump.hpp"
 
diff -Naur /home/laurent/Downloads/SWASHES-1.03.00/Sources/dressler_dam.cpp SWASHES-1.03.00/Sources/dressler_dam.cpp
--- /home/laurent/Downloads/SWASHES-1.03.00/Sources/dressler_dam.cpp	2016-01-29 09:19:31.000000000 -0600
+++ SWASHES-1.03.00/Sources/dressler_dam.cpp	2017-12-20 09:18:20.943207675 -0600
@@ -52,6 +52,9 @@
  * knowledge of the CeCILL license and that you accept its terms.
  *
  ******************************************************************************/
+#if defined(WIN32)
+#include <minmax.h>
+#endif
 
 #include "dressler_dam.hpp"
 
diff -Naur /home/laurent/Downloads/SWASHES-1.03.00/Sources/inclined_plane.cpp SWASHES-1.03.00/Sources/inclined_plane.cpp
--- /home/laurent/Downloads/SWASHES-1.03.00/Sources/inclined_plane.cpp	2016-01-29 09:19:31.000000000 -0600
+++ SWASHES-1.03.00/Sources/inclined_plane.cpp	2017-12-20 09:18:25.611417595 -0600
@@ -53,6 +53,9 @@
  * knowledge of the CeCILL license and that you accept its terms.
  *
  ******************************************************************************/
+#if defined(WIN32)
+#include <minmax.h>
+#endif
 
 #include "inclined_plane.hpp"
 
diff -Naur /home/laurent/Downloads/SWASHES-1.03.00/Sources/swash.cpp SWASHES-1.03.00/Sources/swash.cpp
--- /home/laurent/Downloads/SWASHES-1.03.00/Sources/swash.cpp	2016-01-29 09:19:31.000000000 -0600
+++ SWASHES-1.03.00/Sources/swash.cpp	2017-12-20 09:18:16.158992072 -0600
@@ -52,6 +52,9 @@
  * knowledge of the CeCILL license and that you accept its terms.
  *
  ******************************************************************************/
+#if defined(WIN32)
+#include <minmax.h>
+#endif
 
 #include "swash.hpp"
 
