diff -urN pixell-0.26.0_orig/cython/distances_core.c pixell-0.26.0/cython/distances_core.c
--- pixell-0.26.0_orig/cython/distances_core.c	2024-07-15 16:04:54.000000000 -0700
+++ pixell-0.26.0/cython/distances_core.c	2024-08-28 11:16:21.157693504 -0700
@@ -15,7 +15,7 @@
 double wall_time() { struct timeval tv; gettimeofday(&tv,0); return tv.tv_sec + 1e-6*tv.tv_usec; }
 int max(int a, int b) { return a > b ? a : b; }
 int min(int a, int b) { return a < b ? a : b; }
-int compar_int(int * a, int * b) { return *a-*b; }
+int compar_int(void const * a, void const * b) { return (*((int*)a)-*((int*)b)); }
 int wrap1(int a, int n) { return a < 0 ? a+n : a >= n ? a-n : a; }
 
 // The simple functions are too slow to serve as the basis for a distance transform.
