--- configure.orig	2015-12-10 00:36:41.059789685 -0500
+++ configure	2015-12-10 00:37:33.949897903 -0500
@@ -11998,6 +11998,7 @@
     ;;
 esac
 
+if false ; then
 INTLTOOL_REQUIRED_VERSION_AS_INT=`echo 0.40.0 | awk -F. '{ print $ 1 * 1000 + $ 2 * 100 + $ 3; }'`
 INTLTOOL_APPLIED_VERSION=`intltool-update --version | head -1 | cut -d" " -f3`
 INTLTOOL_APPLIED_VERSION_AS_INT=`echo $INTLTOOL_APPLIED_VERSION | awk -F. '{ print $ 1 * 1000 + $ 2 * 100 + $ 3; }'`
@@ -12009,6 +12010,10 @@
     test "$INTLTOOL_APPLIED_VERSION_AS_INT" -ge "$INTLTOOL_REQUIRED_VERSION_AS_INT" ||
 	as_fn_error $? "Your intltool is too old.  You need intltool 0.40.0 or later." "$LINENO" 5
 fi
+else
+INTLTOOL_APPLIED_VERSION=0.40.0
+INTLTOOL_APPLIED_VERSION_AS_INT=4000
+fi
 
 # Extract the first word of "intltool-update", so it can be a program name with args.
 set dummy intltool-update; ac_word=$2
@@ -12130,9 +12135,11 @@
 fi
 
 
+if false ; then
 if test -z "$INTLTOOL_UPDATE" -o -z "$INTLTOOL_MERGE" -o -z "$INTLTOOL_EXTRACT"; then
     as_fn_error $? "The intltool scripts were not found. Please install intltool." "$LINENO" 5
 fi
+fi
 
 if test -z "$AM_DEFAULT_VERBOSITY"; then
   AM_DEFAULT_VERBOSITY=1
