--- setup.py	2018-12-20 11:38:26.000000000 +0000
+++ setup.py	2019-09-18 10:03:59.074162200 +0100
@@ -6,6 +6,9 @@
 from codecs import open
 import os
 import re
+import platform
+
+WINDOWS = platform.system().lower() == "windows"
 
 # see https://stackoverflow.com/a/21621689/1862861 for why this is here
 class build_ext(_build_ext):
@@ -24,11 +27,16 @@
 else:
     have_cython = True
 
+if WINDOWS:
+    libraries = []
+else:
+    libraries = ["m"]
+
 # set extension
 if have_cython: # convert the pyx file to a .c file if cython is available
-    ext_modules = [ Extension("cpnest.parameter", sources =[ "cpnest/parameter.pyx"], include_dirs=['cpnest/'], libraries=['m']) ]
+    ext_modules = [ Extension("cpnest.parameter", sources =[ "cpnest/parameter.pyx"], include_dirs=['cpnest/'], libraries=libraries) ]
 else: # just compile the included parameter.c (already converted from parameter.pyx) file
-    ext_modules = [ Extension("cpnest.parameter", sources =[ "cpnest/parameter.c"], include_dirs=['cpnest/'], libraries=['m']) ]
+    ext_modules = [ Extension("cpnest.parameter", sources =[ "cpnest/parameter.c"], include_dirs=['cpnest/'], libraries=libraries) ]
 
 here = os.path.abspath(os.path.dirname(__file__))
 
