--- src/cfchecker/cfchecks.py	2013-11-11 10:43:16.000000000 -0300
+++ src/cfchecker/cfchecks.py	2016-05-04 12:53:14.153974154 -0300
@@ -48,8 +48,18 @@
 
 # Use ctypes to interface to the UDUNITS-2 shared library
 # The udunits2 library needs to be in a standard path o/w export LD_LIBRARY_PATH
-from ctypes import *
-udunits=CDLL("libudunits2.so")
+from ctypes import CDLL
+import os.path
+import sys
+
+if 'linux' in sys.platform:
+  ext = 'so'
+elif 'darwin' in sys.platform:
+  ext = 'dylib'
+else:
+  raise ValueError('Unrecognized platform %s' % sys.platform)
+  
+udunits = CDLL(os.path.join(sys.prefix, 'lib', 'libudunits2.%s' % ext))
  
 STANDARDNAME = 'http://cf-pcmdi.llnl.gov/documents/cf-standard-names/standard-name-table/current/cf-standard-name-table.xml'
 AREATYPES = 'http://cf-pcmdi.llnl.gov/documents/cf-standard-names/area-type-table/current/area-type-table.xml'
