diff --git a/meson.build b/meson.build
index aabfe0f..2790d1d 100644
--- a/meson.build
+++ b/meson.build
@@ -352,7 +352,19 @@ else
         endif
     endif
 
-    bases = ['/usr/local', '/usr', homebrew_prefix, '/opt/local']
+    conda_prefix = run_command(
+        [
+            find_program('python3', 'python'),
+            '-c',
+            'import os; print(os.environ.get("PREFIX", ""))',
+        ],
+        check: true,
+    ).stdout().strip()
+
+    bases = ['/usr/local', '/usr', homebrew_prefix, '/opt/local']
+    if conda_prefix != '' and fs.is_dir(conda_prefix)
+        bases = [conda_prefix] + bases
+    endif
+
     foreach inc_dir : bases
         foreach sub_inc : [
             '',
