From 557f87232046e8e6135e553444a736ff341bef8e Mon Sep 17 00:00:00 2001
From: "Korn, Uwe" <Uwe.Korn@blue-yonder.com>
Date: Tue, 19 Mar 2019 11:16:02 +0100
Subject: [PATCH] Deactivate aggressive failures on warnings

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4484e373..e17405bb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,6 +98,8 @@ elseif (MSVC)
   set (WARN_FLAGS "${WARN_FLAGS} -wd4146") # unary minus operator applied to unsigned type, result still unsigned
 endif ()
 
+set(WARN_FLAGS "")
+
 enable_testing()
 
 INCLUDE(CheckSourceCompiles)
-- 
2.20.1

