diff --git a/libs/lensfun/cpuid.cpp b/libs/lensfun/cpuid.cpp
index c435f8ac..5732050d 100644
--- a/libs/lensfun/cpuid.cpp
+++ b/libs/lensfun/cpuid.cpp
@@ -92,9 +92,9 @@ guint _lf_detect_cpu_features ()
 {
 #define cpuid(cmd) \
     __asm volatile ( \
-        "push %%"R_BX"\n" \
+        "push %%" R_BX "\n" \
         "cpuid\n" \
-        "pop %%"R_BX"\n" \
+        "pop %%" R_BX "\n" \
        : "=a" (ax), "=c" (cx),  "=d" (dx) \
        : "0" (cmd))
 
@@ -110,7 +110,7 @@ guint _lf_detect_cpu_features ()
 
     g_mutex_lock (&lock);
 #else
-    static GStaticMutex lock = G_STATIC_MUTEX_INIT; 
+    static GStaticMutex lock = G_STATIC_MUTEX_INIT;
 
     g_static_mutex_lock (&lock);
 #endif
