From ecf71a03c3aa9772227ad080c759e799cedc5a29 Mon Sep 17 00:00:00 2001
From: Marco Bonifacio <bonifacio.marco@gmail.com>
Date: Mon, 21 Feb 2022 18:11:48 +0100
Subject: [PATCH] Update setup.py

---
 setup.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 7f076a5..198861d 100644
--- a/setup.py
+++ b/setup.py
@@ -17,7 +17,10 @@ setup(
     name=name,
     version=version,
     py_modules=[name],
-    scripts=["jsonlint"],
+    entry_points={
+        'console_scripts': [
+            'jsonlint = demjson3.jsonlint:main'
+         ]},
     author="Deron Meranda",
     author_email="deron.meranda@gmail.com",
     description="encoder, decoder, and lint/validator for JSON (JavaScript Object Notation) compliant with RFC 7159",
-- 
2.30.0.windows.2

