diff --git a/setup.py b/setup.py
index e0ff69b..a283a36 100644
--- a/setup.py
+++ b/setup.py
@@ -64,6 +64,7 @@ setup(
         'License :: OSI Approved :: BSD License',
         'Operating System :: OS Independent',
         'Programming Language :: Python :: 3 :: Only',
+        'Programming Language :: Python :: 3.9',
         'Programming Language :: Python :: 3.10',
         'Programming Language :: Python :: 3.11',
         'Programming Language :: Python :: 3.12',
@@ -72,7 +73,7 @@ setup(
         'Topic :: Software Development :: Libraries :: Python Modules',
     ],
     packages=find_packages(exclude=["tests*"]),
-    python_requires='>=3.10',
+    python_requires='>=3.9',
     install_requires=[
         'numpy>=1.24.0',
     ],

