diff --git a/wrappers/python/CMakeLists.txt b/wrappers/python/CMakeLists.txt
index c0bb792..4da5fb2 100644
--- a/wrappers/python/CMakeLists.txt
+++ b/wrappers/python/CMakeLists.txt
@@ -10,6 +10,8 @@ include_directories(
     ${JsonCpp_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIRS})
 link_directories(${Boost_LIBRARY_DIRS})
 
+add_definitions(-D HAVE_SNPRINTF)
+
 file(GLOB_RECURSE Header_Files "*.h")
 file(GLOB_RECURSE Source_Files "*.cpp")
 set(Python_Files "__init__.py")
