diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fd10591..3088a5f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,4 +1,4 @@
-find_package(Boost REQUIRED COMPONENTS date_time filesystem log system)
+find_package(Boost REQUIRED COMPONENTS date_time exception filesystem log system)
 find_package(ICU REQUIRED)
 find_package(JsonCpp REQUIRED)
 if(WITH_DCMTK)
@@ -31,6 +31,7 @@ add_definitions(
     ${DCMTK_DEFINITIONS}
     -D BOOST_ASIO_SEPARATE_COMPILATION
     -D ODIL_MAJOR_VERSION=${odil_MAJOR_VERSION}
+    -D BOOST_UUID_FORCE_AUTO_LINK
 )
 if(BUILD_SHARED_LIBS)
     add_definitions(-D BOOST_ALL_DYN_LINK)
