--- configure    2004-02-05 04:34:07.000000000 -0500
+++ configure    2019-07-31 23:21:16.000000000 -0400
@@ -19099,18 +19099,18 @@
     case "$optimize" in
     -O|"-O "*)
         optimize="-O"
-        optimize="$optimize -fforce-mem"
+        : #optimize="$optimize -fforce-mem"
         optimize="$optimize -fforce-addr"
         : #x optimize="$optimize -finline-functions"
         : #- optimize="$optimize -fstrength-reduce"
-        optimize="$optimize -fthread-jumps"
-        optimize="$optimize -fcse-follow-jumps"
-        optimize="$optimize -fcse-skip-blocks"
+        : #optimize="$optimize -fthread-jumps"
+        : #optimize="$optimize -fcse-follow-jumps"
+        : #optimize="$optimize -fcse-skip-blocks"
         : #x optimize="$optimize -frerun-cse-after-loop"
         : #x optimize="$optimize -frerun-loop-opt"
         : #x optimize="$optimize -fgcse"
         optimize="$optimize -fexpensive-optimizations"
-        optimize="$optimize -fregmove"
+        : #optimize="$optimize -fregmove"
         : #* optimize="$optimize -fdelayed-branch"
         : #x optimize="$optimize -fschedule-insns"
         optimize="$optimize -fschedule-insns2"
@@ -21657,7 +21657,7 @@
 echo "${ECHO_T}${enable_experimental-no}" >&6


-test -n "$arch"     && CFLAGS="$CFLAGS $arch"
+# test -n "$arch"     && CFLAGS="$CFLAGS $arch"
 test -n "$debug"    && CFLAGS="$CFLAGS $debug"
 test -n "$optimize" && CFLAGS="$CFLAGS $optimize"
 test -n "$profile"  && CFLAGS="$CFLAGS $profile" LDFLAGS="$LDFLAGS $profile"
@@ -23103,4 +23103,3 @@
   # would make configure fail if this is the last instruction.
   $ac_cs_success || { (exit 1); exit 1; }
 fi
-
