diff --git a/pyproject.toml b/pyproject.toml
index 5831fe6..6ec79cd 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -18,7 +18,7 @@ classifiers = [
 dependencies = [
     "torch",
     "triton",
-    "ninja",
+    
     "einops",
     "transformers",
     "packaging",
@@ -42,6 +42,6 @@ requires = [
     "wheel",
     "torch",
     "packaging",
-    "ninja",
+    
 ]
 build-backend = "setuptools.build_meta"
