diff --git a/client/Makefile b/client/Makefile
index 474833e..01c0112 100644
--- a/client/Makefile
+++ b/client/Makefile
@@ -2,10 +2,10 @@ TARGET_NAME=eonclient
 
 #------------------------------------
 #C compiler settings
-CC=gcc
-CXX=g++
-LD=ld
-AR=ar cru
+# CC=gcc
+# CXX=g++
+# LD=ld
+AR=$(AR) cru
 RANLIB=ranlib
 DEPFLAGS=-MMD
 
@@ -14,8 +14,8 @@ DEPFLAGS=-MMD
 #if building lammps with meam
 #LDFLAGS += potentials/LAMMPS/libmeam.a
 #LDFLAGS += potentials/LAMMPS/libreax.a
-LDFLAGS += -lcurl -ldl
-CXXFLAGS += -Wall -Wno-unused-local-typedefs -Wno-unused-function -Wfatal-errors  $(OPT) #  -I flags depend on location of Python.h library
+# LDFLAGS += -lcurl -ldl
+# CXXFLAGS += -Wall -Wno-unused-local-typedefs -Wno-unused-function -Wfatal-errors  $(OPT) #  -I flags depend on location of Python.h library
 
 #CXXFLAGS += -Wall -Wno-unused-local-typedefs -Wno-unused-function -Wfatal-errors -lpython3 -I/export/tarpit/Python-3.7.4/ -I/export/tarpit/Python-3.7.4/Include/ $(OPT) #-I flags depend on location of Python.h library
 
