--- src/CMakeLists.txt	2019-11-07 21:56:50.884002939 -0600
+++ src/CMakeLists.txt	2019-11-07 21:57:04.904276526 -0600
@@ -120,7 +120,7 @@
 
 set ( LNKSCR "${CMAKE_CURRENT_SOURCE_DIR}/hsacore.so.link" )
 
-set ( CMAKE_SHARED_LINKER_FLAGS "-Wl,-Bdynamic -Wl,-z,noexecstack -Wl,${LNKSCR} -Wl,--version-script=${DRVDEF}" )
+set ( CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-Bdynamic -Wl,-z,noexecstack -Wl,${LNKSCR} -Wl,--version-script=${DRVDEF}" )
 
 set ( CMAKE_SKIP_BUILD_RPATH TRUE )
 
