From c799a3947f1b43b7d1ed902bdaeff2a0b31bab0d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 11 Apr 2020 22:12:25 +0200
Subject: [PATCH 1/4] comment out requires in setup.py

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index b4efb00..539a424 100644
--- a/setup.py
+++ b/setup.py
@@ -119,8 +119,8 @@ setup(name='qdldl',
       long_description=readme(),
       package_dir={'qdldl': 'module'},
       include_package_data=True,  # Include package data from MANIFEST.in
-      setup_requires=["setuptools>=18.0", "pybind11"],
-      install_requires=["numpy >= 1.7", "scipy >= 0.13.2"],
+      # setup_requires=["setuptools>=18.0", "pybind11"],
+      # install_requires=["numpy >= 1.7", "scipy >= 0.13.2"],
       license='Apache 2.0',
       url="https://github.com/oxfordcontrol/qdldlpy/",
       cmdclass={'build_ext': build_ext_qdldl},
-- 
2.26.0.windows.1

