From e5d44c3516fdea0940e9897b87132ca6331c4fcf Mon Sep 17 00:00:00 2001
From: Bartolomeo Stellato <bartolomeo.stellato@gmail.com>
Date: Wed, 3 Jun 2020 11:40:50 -0400
Subject: [PATCH] Change cmake dir

---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index e0e9288..35ec9c8 100644
--- a/setup.py
+++ b/setup.py
@@ -58,7 +58,8 @@ else:
 
 # Compile QDLDL using CMake
 current_dir = os.getcwd()
-qdldl_dir = os.path.join(current_dir, 'c',)
+root_dir = os.path.dirname(os.path.abspath(__file__))
+qdldl_dir = os.path.join(root_dir, 'c')
 qdldl_build_dir = os.path.join(qdldl_dir, 'build')
 qdldl_lib = [qdldl_build_dir, 'out'] + lib_subdir + [lib_name]
 qdldl_lib = os.path.join(*qdldl_lib)
-- 
2.24.3 (Apple Git-128)

