From e240260c0340b8f57f7caf3444eac6e61683853b Mon Sep 17 00:00:00 2001
From: Bartolomeo Stellato <bartolomeo.stellato@gmail.com>
Date: Tue, 12 May 2020 17:50:35 -0400
Subject: [PATCH 1/2] Include cpp folder in setup.py

---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index e0e9288..4e8cb0a 100644
--- a/setup.py
+++ b/setup.py
@@ -97,9 +97,10 @@ if sys.platform == 'darwin':
             os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
 
 qdldl = Extension('qdldl',
-                   sources= glob(os.path.join('cpp', '*.cpp')),
+                   sources=glob(os.path.join('cpp', '*.cpp')),
                    include_dirs=[os.path.join('c'),
                                  os.path.join('c', 'qdldl', 'include'),
+                                 os.path.join('cpp'),
                                  get_pybind_include(),
                                  get_pybind_include(user=False)],
                    language='c++',
-- 
2.17.2 (Apple Git-113)

