From b910df80cb756339b7272471378916cf314762af Mon Sep 17 00:00:00 2001
From: Bartolomeo Stellato <bartolomeo.stellato@gmail.com>
Date: Tue, 12 May 2020 19:43:31 -0400
Subject: [PATCH 2/2] Consider all sources

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 4e8cb0a..180b734 100644
--- a/setup.py
+++ b/setup.py
@@ -97,7 +97,7 @@ if sys.platform == 'darwin':
             os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
 
 qdldl = Extension('qdldl',
-                   sources=glob(os.path.join('cpp', '*.cpp')),
+                   sources=glob(os.path.join('cpp', '*')),
                    include_dirs=[os.path.join('c'),
                                  os.path.join('c', 'qdldl', 'include'),
                                  os.path.join('cpp'),
-- 
2.17.2 (Apple Git-113)

