From 503016a437aef7bdcdf5c2f88a0483a83f28712d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 11 Apr 2020 22:51:33 +0200
Subject: [PATCH 3/4] setup: use nmake on windows

---
 setup.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 3d1bae5..f0bd2b7 100644
--- a/setup.py
+++ b/setup.py
@@ -38,7 +38,9 @@ lib_subdir = []
 
 # Check if windows linux or mac to pass flag
 if system() == 'Windows':
-    cmake_args += ['-G', 'Visual Studio 14 2015']
+    # cmake_args += ['-G', 'Visual Studio 14 2015']
+    cmake_args += ['-G', 'NMake Makefiles']
+    cmake_args += ['-DCMAKE_BUILD_TYPE=Release']
     # Differentiate between 32-bit and 64-bit
     if sys.maxsize // 2 ** 32 > 0:
         cmake_args[-1] += ' Win64'
-- 
2.26.0.windows.1

