From 04ba879fdd075e808dd2e5faa7ed7cc019c174be Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 12 Apr 2020 18:09:05 +0200
Subject: [PATCH 4/4] include all of cpp-folder as source

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index f0bd2b7..3121001 100644
--- a/setup.py
+++ b/setup.py
@@ -99,7 +99,7 @@ if sys.platform == 'darwin':
             os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
 
 qdldl = Extension('qdldl',
-                   sources= glob(os.path.join('cpp', '*.cpp')),
+                   sources=[os.path.join('cpp')],
                    include_dirs=[os.path.join('c'),
                                  os.path.join('c', 'qdldl', 'include'),
                                  get_pybind_include(),
-- 
2.24.0.windows.2

