diff --git a/pyproject.toml b/pyproject.toml
index 107d7e4..1640d70 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,7 +42,7 @@ dependencies = [
   'pandas>=2.1.0, <3.0.0',
   'rdflib>=6.1.1, <8.0.0',
   'pyoxigraph>=0.3.0, <0.4.0',
-  'ruamel.yaml==0.18.0, <0.19.0',
+  'ruamel.yaml>=0.18.0, <0.19.0',
   'jsonpath-python>=1.0.6, <2.0.0',
   'elementpath>=4.0.0, <5.0.0',
   'duckdb>=0.10.0, <2.0.0',
