From c1e7bd1d2850c64cf63b6d169229c68f8cb8365d Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Tue, 26 Mar 2024 14:20:33 +0100
Subject: [PATCH] fix patch in version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2940cb95fa..219edc22fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ include(CMakeDependentOption)
 # Manually define VERSION
 set(Sofa_VERSION_MAJOR 23)
 set(Sofa_VERSION_MINOR 12)
-set(Sofa_VERSION_PATCH 00)
+set(Sofa_VERSION_PATCH 01)
 set(Sofa_VERSION ${Sofa_VERSION_MAJOR}.${Sofa_VERSION_MINOR}.${Sofa_VERSION_PATCH})
 
 set(SOFA_VERSION_STR "\"${Sofa_VERSION}\"")
-- 
2.34.1

