From 2b23235e5c48b5b65126f4302fc54511fee36c0e Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Mon, 18 Mar 2024 11:08:17 +0100
Subject: [PATCH] force Sofa.GUI.Qt to use external qglviewer

---
 Sofa/GUI/Qt/CMakeLists.txt | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/Sofa/GUI/Qt/CMakeLists.txt b/Sofa/GUI/Qt/CMakeLists.txt
index 6076ed5619..6ec758eeec 100644
--- a/Sofa/GUI/Qt/CMakeLists.txt
+++ b/Sofa/GUI/Qt/CMakeLists.txt
@@ -74,10 +74,7 @@ if(Sofa.GL_FOUND)
     option(SOFA_GUI_QT_ENABLE_QGLVIEWER "Compile the QGLViewer for the Qt GUI" ON)
 
     if(SOFA_GUI_QT_ENABLE_QGLVIEWER)
-        find_package(QGLViewer)
-        if(NOT QGLViewer_FOUND)
-            add_subdirectory(libQGLViewer/QGLViewer)
-        endif()
+        find_package(QGLViewer REQUIRED)
     endif()
 else()
     message(STATUS "Sofa.GUI.Qt: Sofa.GL has not been activated, QtViewer/QGLViewer will be disabled.")
-- 
2.34.1

