From f763f968e60e16dac5a216dad891acb61995f0d5 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Thu, 12 Jan 2023 15:47:20 +0100
Subject: [PATCH] install python component

---
 CMakeLists.txt | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a9a7e4..91b763c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -357,8 +357,18 @@ endif()
 # pybind11 module
 option(BUILDPYTHONMODULE "Build Python binding." OFF)
 if (BUILDPYTHONMODULE)
-  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/misc/pymodule/pybind11)
+  find_package(
+    Python3
+    COMPONENTS Interpreter Development.Module
+    REQUIRED
+  )
+  find_package(pybind11 REQUIRED)
   pybind11_add_module(kahip_python_binding ${CMAKE_CURRENT_SOURCE_DIR}/misc/pymodule/kahip.cpp)
   target_link_libraries(kahip_python_binding PUBLIC kahip_static)
   set_target_properties(kahip_python_binding PROPERTIES OUTPUT_NAME "kahip")
+  install(TARGETS kahip_python_binding
+    LIBRARY DESTINATION "${Python3_SITELIB}"
+    COMPONENT python
+    EXCLUDE_FROM_ALL
+  )
 endif ()
--
2.39.0
