From 2a21fcf4599d4c63bbccf2b967b7e7ba6d81dcfb Mon Sep 17 00:00:00 2001
From: Dean Moldovan <dean0x7d@gmail.com>
Date: Thu, 26 Jan 2017 12:05:41 +0100
Subject: [PATCH] Workaround for conda-build glob pattern expansion


diff --git a/tests/baseline_data/kpm/ldos[graphene-const_potential].pbz b/tests/baseline_data/kpm/ldos_graphene-const_potential_.pbz
similarity index 100%
rename from tests/baseline_data/kpm/ldos[graphene-const_potential].pbz
rename to tests/baseline_data/kpm/ldos_graphene-const_potential_.pbz
diff --git a/tests/baseline_data/kpm/ldos[graphene-magnetic_field].pbz b/tests/baseline_data/kpm/ldos_graphene-magnetic_field_.pbz
similarity index 100%
rename from tests/baseline_data/kpm/ldos[graphene-magnetic_field].pbz
rename to tests/baseline_data/kpm/ldos_graphene-magnetic_field_.pbz
diff --git a/tests/baseline_data/kpm/ldos[graphene-pristine-oversized].pbz b/tests/baseline_data/kpm/ldos_graphene-pristine-oversized_.pbz
similarity index 100%
rename from tests/baseline_data/kpm/ldos[graphene-pristine-oversized].pbz
rename to tests/baseline_data/kpm/ldos_graphene-pristine-oversized_.pbz
diff --git a/tests/baseline_data/kpm/ldos[graphene-pristine].pbz b/tests/baseline_data/kpm/ldos_graphene-pristine_.pbz
similarity index 100%
rename from tests/baseline_data/kpm/ldos[graphene-pristine].pbz
rename to tests/baseline_data/kpm/ldos_graphene-pristine_.pbz
diff --git a/tests/baseline_data/lattice/expected[graphene-bilayer].pbz b/tests/baseline_data/lattice/expected_graphene-bilayer_.pbz
similarity index 100%
rename from tests/baseline_data/lattice/expected[graphene-bilayer].pbz
rename to tests/baseline_data/lattice/expected_graphene-bilayer_.pbz
diff --git a/tests/baseline_data/lattice/expected[graphene-monolayer-4atom].pbz b/tests/baseline_data/lattice/expected_graphene-monolayer-4atom_.pbz
similarity index 100%
rename from tests/baseline_data/lattice/expected[graphene-monolayer-4atom].pbz
rename to tests/baseline_data/lattice/expected_graphene-monolayer-4atom_.pbz
diff --git a/tests/baseline_data/lattice/expected[graphene-monolayer-alt].pbz b/tests/baseline_data/lattice/expected_graphene-monolayer-alt_.pbz
similarity index 100%
rename from tests/baseline_data/lattice/expected[graphene-monolayer-alt].pbz
rename to tests/baseline_data/lattice/expected_graphene-monolayer-alt_.pbz
diff --git a/tests/baseline_data/lattice/expected[graphene-monolayer-nn].pbz b/tests/baseline_data/lattice/expected_graphene-monolayer-nn_.pbz
similarity index 100%
rename from tests/baseline_data/lattice/expected[graphene-monolayer-nn].pbz
rename to tests/baseline_data/lattice/expected_graphene-monolayer-nn_.pbz
diff --git a/tests/baseline_data/lattice/expected[graphene-monolayer].pbz b/tests/baseline_data/lattice/expected_graphene-monolayer_.pbz
similarity index 100%
rename from tests/baseline_data/lattice/expected[graphene-monolayer].pbz
rename to tests/baseline_data/lattice/expected_graphene-monolayer_.pbz
diff --git a/tests/baseline_data/shape/polygon[diamond].pbz b/tests/baseline_data/shape/polygon_diamond_.pbz
similarity index 100%
rename from tests/baseline_data/shape/polygon[diamond].pbz
rename to tests/baseline_data/shape/polygon_diamond_.pbz
diff --git a/tests/baseline_data/shape/polygon[pentagon].pbz b/tests/baseline_data/shape/polygon_pentagon_.pbz
similarity index 100%
rename from tests/baseline_data/shape/polygon[pentagon].pbz
rename to tests/baseline_data/shape/polygon_pentagon_.pbz
diff --git a/tests/baseline_data/shape/polygon[square].pbz b/tests/baseline_data/shape/polygon_square_.pbz
similarity index 100%
rename from tests/baseline_data/shape/polygon[square].pbz
rename to tests/baseline_data/shape/polygon_square_.pbz
diff --git a/tests/baseline_data/shape/polygon[triangle90].pbz b/tests/baseline_data/shape/polygon_triangle90_.pbz
similarity index 100%
rename from tests/baseline_data/shape/polygon[triangle90].pbz
rename to tests/baseline_data/shape/polygon_triangle90_.pbz
diff --git a/tests/baseline_data/shape/polygon[triangle].pbz b/tests/baseline_data/shape/polygon_triangle_.pbz
similarity index 100%
rename from tests/baseline_data/shape/polygon[triangle].pbz
rename to tests/baseline_data/shape/polygon_triangle_.pbz
diff --git a/tests/baseline_data/solver/dos[arpack-graphene-magnetic_field].pbz b/tests/baseline_data/solver/dos_arpack-graphene-magnetic_field_.pbz
similarity index 100%
rename from tests/baseline_data/solver/dos[arpack-graphene-magnetic_field].pbz
rename to tests/baseline_data/solver/dos_arpack-graphene-magnetic_field_.pbz
diff --git a/tests/baseline_data/solver/dos[feast-graphene-magnetic_field].pbz b/tests/baseline_data/solver/dos_feast-graphene-magnetic_field_.pbz
similarity index 100%
rename from tests/baseline_data/solver/dos[feast-graphene-magnetic_field].pbz
rename to tests/baseline_data/solver/dos_feast-graphene-magnetic_field_.pbz
diff --git a/tests/baseline_data/solver/eigenvalues[arpack-graphene-magnetic_field].pbz b/tests/baseline_data/solver/eigenvalues_arpack-graphene-magnetic_field_.pbz
similarity index 100%
rename from tests/baseline_data/solver/eigenvalues[arpack-graphene-magnetic_field].pbz
rename to tests/baseline_data/solver/eigenvalues_arpack-graphene-magnetic_field_.pbz
diff --git a/tests/baseline_data/solver/eigenvalues[feast-graphene-magnetic_field].pbz b/tests/baseline_data/solver/eigenvalues_feast-graphene-magnetic_field_.pbz
similarity index 100%
rename from tests/baseline_data/solver/eigenvalues[feast-graphene-magnetic_field].pbz
rename to tests/baseline_data/solver/eigenvalues_feast-graphene-magnetic_field_.pbz
diff --git a/tests/baseline_data/solver/spatial_ldos[arpack-graphene-magnetic_field].pbz b/tests/baseline_data/solver/spatial_ldos_arpack-graphene-magnetic_field_.pbz
similarity index 100%
rename from tests/baseline_data/solver/spatial_ldos[arpack-graphene-magnetic_field].pbz
rename to tests/baseline_data/solver/spatial_ldos_arpack-graphene-magnetic_field_.pbz
diff --git a/tests/baseline_data/solver/spatial_ldos[feast-graphene-magnetic_field].pbz b/tests/baseline_data/solver/spatial_ldos_feast-graphene-magnetic_field_.pbz
similarity index 100%
rename from tests/baseline_data/solver/spatial_ldos[feast-graphene-magnetic_field].pbz
rename to tests/baseline_data/solver/spatial_ldos_feast-graphene-magnetic_field_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-bilayer].pbz b/tests/baseline_data/system/expected_graphene-bilayer_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-bilayer].pbz
rename to tests/baseline_data/system/expected_graphene-bilayer_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-monolayer-4atom-periodic-2d].pbz b/tests/baseline_data/system/expected_graphene-monolayer-4atom-periodic-2d_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-monolayer-4atom-periodic-2d].pbz
rename to tests/baseline_data/system/expected_graphene-monolayer-4atom-periodic-2d_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-monolayer-4atom].pbz b/tests/baseline_data/system/expected_graphene-monolayer-4atom_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-monolayer-4atom].pbz
rename to tests/baseline_data/system/expected_graphene-monolayer-4atom_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-monolayer-alt].pbz b/tests/baseline_data/system/expected_graphene-monolayer-alt_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-monolayer-alt].pbz
rename to tests/baseline_data/system/expected_graphene-monolayer-alt_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-monolayer-nn].pbz b/tests/baseline_data/system/expected_graphene-monolayer-nn_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-monolayer-nn].pbz
rename to tests/baseline_data/system/expected_graphene-monolayer-nn_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-monolayer-periodic-1d-alt].pbz b/tests/baseline_data/system/expected_graphene-monolayer-periodic-1d-alt_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-monolayer-periodic-1d-alt].pbz
rename to tests/baseline_data/system/expected_graphene-monolayer-periodic-1d-alt_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-monolayer-periodic-1d].pbz b/tests/baseline_data/system/expected_graphene-monolayer-periodic-1d_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-monolayer-periodic-1d].pbz
rename to tests/baseline_data/system/expected_graphene-monolayer-periodic-1d_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-monolayer-periodic-2d].pbz b/tests/baseline_data/system/expected_graphene-monolayer-periodic-2d_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-monolayer-periodic-2d].pbz
rename to tests/baseline_data/system/expected_graphene-monolayer-periodic-2d_.pbz
diff --git a/tests/baseline_data/system/expected[graphene-monolayer].pbz b/tests/baseline_data/system/expected_graphene-monolayer_.pbz
similarity index 100%
rename from tests/baseline_data/system/expected[graphene-monolayer].pbz
rename to tests/baseline_data/system/expected_graphene-monolayer_.pbz
diff --git a/tests/utils/path.py b/tests/utils/path.py
index 1c352e1..3a74269 100644
--- a/tests/utils/path.py
+++ b/tests/utils/path.py
@@ -33,4 +33,5 @@ def path_from_fixture(request, prefix, variant='', ext='', override_group=''):
         part = name.partition('[')
         name = override_group + part[1] + part[2]
 
+    name = name.replace("[", "_").replace("]", "_")
     return (test_dir / prefix / module_name / name).with_suffix(ext)
-- 
2.11.0

