--- make-target-2.sh    2024-03-29 12:02:12.000000000 -0500
+++ make-target-2.sh    2024-04-28 19:10:54.914542271 -0500
@@ -52,6 +52,11 @@
 fi
 if [ "$warm_compile" = yes ]; then
     echo //doing warm init - compilation phase
+    patchelf \
+        --set-interpreter $LIBC_INTERPRETER \
+        --set-rpath $LIBC_RPATH \
+        ./src/runtime/sbcl
+
     ./src/runtime/sbcl --core output/cold-sbcl.core \
      --lose-on-corruption $SBCL_MAKE_TARGET_2_OPTIONS --no-sysinit --no-userinit \
      --eval '(sb-fasl::!warm-load "src/cold/warm.lisp")' --quit
