--- tests/relocation.test.sh    2024-03-29 12:02:13.000000000 -0500
+++ tests/relocation.test.sh    2024-04-29 09:31:58.979437524 -0500
@@ -29,6 +29,10 @@
 while [ $i -le 6 ]
 do
   export SBCL_FAKE_MMAP_INSTRUCTION_LINE=$i
+  patchelf \
+      --set-interpreter $LIBC_INTERPRETER \
+      --set-rpath $LIBC_RPATH \
+      $test_sbcl
   $test_sbcl --lose-on-corruption --disable-ldb --noinform --core ../output/sbcl.core \
               --no-sysinit --no-userinit --noprint --disable-debugger \
               --eval '(gc :full t)' \
