diff --git a/configure.py b/configure.py
index a263efb..d1bce53 100644
--- a/configure.py
+++ b/configure.py
@@ -663,23 +663,23 @@ class TargetConfiguration:
 
             config_args = ducfg.get('CONFIG_ARGS', '')
 
-            dynamic_pylib = '--enable-shared' in config_args
-            if not dynamic_pylib:
-                dynamic_pylib = '--enable-framework' in config_args
-
-            if dynamic_pylib:
-                pyshlib = ducfg.get('LDLIBRARY', '')
-
-                exec_prefix = ducfg['exec_prefix']
-                multiarch = ducfg.get('MULTIARCH', '')
-                libdir = ducfg['LIBDIR']
-
-                if glob.glob('%s/lib/libpython%d.%d*' % (exec_prefix, py_major, py_minor)):
-                    pylib_dir = exec_prefix + '/lib'
-                elif multiarch != '' and glob.glob('%s/lib/%s/libpython%d.%d*' % (exec_prefix, multiarch, py_major, py_minor)):
-                    pylib_dir = exec_prefix + '/lib/' + multiarch
-                elif glob.glob('%s/libpython%d.%d*' % (libdir, py_major, py_minor)):
-                    pylib_dir = libdir
+            pyshlib = ducfg.get('LDLIBRARY', '')
+            if pyshlib.endswith('.a'):
+                sh_suffix = ".so"
+                if sys.platform == 'darwin':
+                    sh_suffix = ".dylib"
+                pyshlib = pyshlib.replace('.a', sh_suffix)
+
+            exec_prefix = ducfg['exec_prefix']
+            multiarch = ducfg.get('MULTIARCH', '')
+            libdir = ducfg['LIBDIR']
+
+            if glob.glob('%s/lib/libpython%d.%d*' % (exec_prefix, py_major, py_minor)):
+                pylib_dir = exec_prefix + '/lib'
+            elif multiarch != '' and glob.glob('%s/lib/%s/libpython%d.%d*' % (exec_prefix, multiarch, py_major, py_minor)):
+                pylib_dir = exec_prefix + '/lib/' + multiarch
+            elif glob.glob('%s/libpython%d.%d*' % (libdir, py_major, py_minor)):
+                pylib_dir = libdir
 
         self.py_pylib_dir = pylib_dir
         self.py_pylib_lib = pylib_lib
