--- configure.py        2019-10-07 04:00:39.515797720 +0000
+++ configure.py        2019-10-07 04:01:23.260211526 +0000
@@ -1551,12 +1551,12 @@
     path = os.environ['PATH']
 
     dll_dir = os.path.dirname(__file__) + '\\\\Qt\\\\bin'
-    if os.path.isfile(dll_dir + '\\\\Qt5Core.dll'):
+    if os.path.isfile(dll_dir + '\\\\Qt5Core_conda.dll'):
         path = dll_dir + ';' + path
         os.environ['PATH'] = path
     else:
         for dll_dir in path.split(';'):
-            if os.path.isfile(dll_dir + '\\\\Qt5Core.dll'):
+            if os.path.isfile(dll_dir + '\\\\Qt5Core_conda.dll'):
                 break
         else:
             raise ImportError("unable to find Qt5Core.dll on PATH")

