diff --git a/setup.py b/setup.py
index 66603c1..ed9d772 100755
--- a/setup.py
+++ b/setup.py
@@ -26,15 +26,13 @@ DEPENDENCIES = [
 
 EXTRAS = {}
 
-with open("DESCRIPTION.md", "r", encoding="utf-8") as f:
-    README = f.read()
 inline_license = ""
 
 setup(
     name="opendp-smartnoise",
     version=SELFVERSION,
     description="",
-    long_description=README,
+    long_description="",
     long_description_content_type="text/markdown",
     author="opendp-smartnoise",
     license=inline_license,
