diff --git a/graalpython/com.oracle.graal.python.cext/lzma/config.h b/graalpython/com.oracle.graal.python.cext/lzma/config.h
index 0d1d0704f7..f96f33f045 100644
--- a/graalpython/com.oracle.graal.python.cext/lzma/config.h
+++ b/graalpython/com.oracle.graal.python.cext/lzma/config.h
@@ -108,7 +108,7 @@
 #define STDC_HEADERS 1
 #define TUKLIB_CPUCORES_SCHED_GETAFFINITY 1
 #define TUKLIB_FAST_UNALIGNED_ACCESS 1
-#define TUKLIB_PHYSMEM_SYSCONF 1
+#define TUKLIB_PHYSMEM_SYSCTL 1
 #define VERSION "5.2.6"
 #ifndef _DARWIN_USE_64_BIT_INODE
 # define _DARWIN_USE_64_BIT_INODE 1
