diff --git a/graalpython/com.oracle.graal.python/src/com/oracle/graal/python/PythonLanguage.java b/graalpython/com.oracle.graal.python/src/com/oracle/graal/python/PythonLanguage.java
index 2ead49b03c..acca52ba00 100644
--- a/graalpython/com.oracle.graal.python/src/com/oracle/graal/python/PythonLanguage.java
+++ b/graalpython/com.oracle.graal.python/src/com/oracle/graal/python/PythonLanguage.java
@@ -39,6 +39,7 @@ import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Semaphore;
 import java.util.logging.Level;

+import org.graalvm.home.Version;
 import org.graalvm.options.OptionDescriptors;
 import org.graalvm.options.OptionKey;
 import org.graalvm.options.OptionValues;
@@ -174,7 +175,17 @@ public final class PythonLanguage extends TruffleLanguage<PythonContext> {
     public static final String FROZEN_FILENAME_PREFIX = "<frozen ";
     public static final String FROZEN_FILENAME_SUFFIX = ">";

+    /**
+     * GraalVM version, as used by {@link Version}. Unfortunately, we cannot just use {@link
+     * Version#getCurrent} as it relies on a GraalVM build, but we may run from Jar files directly
+     * during development. So we hardcode the version here and have an assert below that it is
+     * equal to the version in the GraalVM we are built into.
+     */
+    public static final int[] GRAALVM_VERSION = new int[]{22, 3, 0};
+
     static {
+        assert Version.getCurrent().compareTo(GRAALVM_VERSION) == 0 || Version.getCurrent().toString().equals("snapshot");
+
         switch (RELEASE_LEVEL) {
             case RELEASE_LEVEL_ALPHA:
                 RELEASE_LEVEL_STRING = tsLiteral("alpha");
diff --git a/graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/SysModuleBuiltins.java b/graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/SysModuleBuiltins.java
index 1668968a32..6154f6d579 100644
--- a/graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/SysModuleBuiltins.java
+++ b/graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/SysModuleBuiltins.java
@@ -414,7 +414,7 @@ public class SysModuleBuiltins extends PythonBuiltins {
     protected static PSimpleNamespace makeImplementation(PythonObjectFactory factory, PTuple versionInfo, TruffleString gmultiarch) {
         final PSimpleNamespace ns = factory.createSimpleNamespace();
         ns.setAttribute(tsLiteral("name"), T_GRAALPYTHON_ID);
-        ns.setAttribute(T_CACHE_TAG, toTruffleStringUncached(J_GRAALPYTHON_ID + "-" + PythonLanguage.MAJOR + PythonLanguage.MINOR));
+        ns.setAttribute(T_CACHE_TAG, toTruffleStringUncached(J_GRAALPYTHON_ID + PythonLanguage.GRAALVM_VERSION[0] + PythonLanguage.GRAALVM_VERSION[1] + "-" + PythonLanguage.MAJOR + PythonLanguage.MINOR));
         ns.setAttribute(T_VERSION, versionInfo);
         ns.setAttribute(T__MULTIARCH, gmultiarch);
         ns.setAttribute(tsLiteral("hexversion"), PythonLanguage.VERSION_HEX);
-- 
2.25.1

