diff --git a/mx.py b/mx.py
index 11a83ca8..8f592ad1 100755
--- a/mx.py
+++ b/mx.py
@@ -1829,6 +1829,9 @@ class Suite(object):
             elif isinstance(value, str):
                 value = expandvars(value)
                 if '$' in value or '%' in value:
+                    if is_linux() and value == r"\$$ORIGIN":
+                        # this is the escaping that CMake uses, see https://cmake.org/cmake/help/latest/policy/CMP0095.html#policy:CMP0095
+                        return "$ORIGIN"
                     abort('value of ' + '.'.join(context) + ' contains an undefined environment variable: ' + value)
             elif isinstance(value, bool):
                 pass
