diff --git a/mx.py b/mx.py
index 6cff6da1..b3371144 100755
--- a/mx.py
+++ b/mx.py
@@ -7470,7 +7470,7 @@ class JavaBuildTask(ProjectBuildTask):
         ProjectBuildTask.initSharedMemoryState(self)
         try:
             self._newestBox = multiprocessing.Array('c', 2048)
-        except TypeError:
+        except (TypeError, ValueError):
             self._newestBox = multiprocessing.Value('c', '')
 
     def pushSharedMemoryState(self):
