diff --git a/sulong/mx.sulong/suite.py b/sulong/mx.sulong/suite.py
index 64aada01e2a..38b00797483 100644
--- a/sulong/mx.sulong/suite.py
+++ b/sulong/mx.sulong/suite.py
@@ -929,6 +929,32 @@ suite = {
             "ninja_install_targets" : ["install-cxxabi", "install-cxx"],
             "results" : ["native"],
             "cmakeConfig" : {
+              # the only difference to darwin is CMAKE_INSTALL_RPATH
+              "CMAKE_INSTALL_RPATH" : "\\$$ORIGIN",
+              "LLVM_ENABLE_RUNTIMES" : "libcxx;libcxxabi",
+              "LIBCXXABI_INCLUDE_TESTS": "NO",
+              "LIBCXXABI_ENABLE_STATIC" : "NO",
+              "LIBCXX_INCLUDE_BENCHMARKS": "NO",
+              "LIBCXX_INCLUDE_TESTS": "NO",
+              "LIBCXX_ENABLE_STATIC" : "NO",
+              "LIBCXX_ENABLE_EXPERIMENTAL_LIBRARY" : "NO",
+              "CMAKE_C_COMPILER" : "<path:SULONG_BOOTSTRAP_TOOLCHAIN_NO_HOME>/bin/<cmd:clang>",
+              "CMAKE_CXX_COMPILER" : "<path:SULONG_BOOTSTRAP_TOOLCHAIN_NO_HOME>/bin/<cmd:clang++>",
+              "CMAKE_INSTALL_PREFIX" : "native",
+              # workaround for build problem with cmake >=3.22
+              # see https://lists.llvm.org/pipermail/llvm-dev/2021-December/154144.html
+              "CMAKE_BUILD_WITH_INSTALL_RPATH" : "YES",
+            },
+          },
+        },
+        "darwin" : {
+          "<others>" : {
+            "ninja_targets" : ["cxxabi", "cxx"],
+            "ninja_install_targets" : ["install-cxxabi", "install-cxx"],
+            "results" : ["native"],
+            "cmakeConfig" : {
+              # the only difference to the other unixes is CMAKE_INSTALL_RPATH
+              "CMAKE_INSTALL_RPATH" : "@loader_path/",
               "LLVM_ENABLE_RUNTIMES" : "libcxx;libcxxabi",
               "LIBCXXABI_INCLUDE_TESTS": "NO",
               "LIBCXXABI_ENABLE_STATIC" : "NO",
