diff --git a/sulong/mx.sulong/suite.py b/sulong/mx.sulong/suite.py
index 30dbb391c07..64aada01e2a 100644
--- a/sulong/mx.sulong/suite.py
+++ b/sulong/mx.sulong/suite.py
@@ -305,6 +305,7 @@ suite = {
       "subDir" : "tests",
       "native" : True,
       "vpath" : True,
+      "defaultBuild" : False,
       "results" : [
         "bin/"
       ],
@@ -1850,6 +1851,7 @@ suite = {
 
     "SULONG_TEST" : {
       "subDir" : "tests",
+      "defaultBuild" : False,
       "dependencies" : [
         "com.oracle.truffle.llvm.tests",
         "com.oracle.truffle.llvm.tests.types",
@@ -2093,6 +2095,7 @@ suite = {
       "native" : True,
       "relpath" : True,
       "platformDependent" : True,
+      "defaultBuild" : False,
       "layout" : {
         "./" : [
           "dependency:com.oracle.truffle.llvm.tests.tck.native/*",
