diff --git a/setup.py b/setup.py
index 24d97e1..5d0383a 100644
--- a/setup.py
+++ b/setup.py
@@ -176,13 +176,6 @@ class cmake_build_ext(build_ext):
             # custom LDFLAGS - depends on platform
             'LDFLAGS': ldflags,
         }
-        self.build_cmake_dependency(JPEG_DIR, [
-            *flags,
-            '-DWITH_CRT_DLL=1',  # fixes https://bugs.python.org/issue24872
-            '-DENABLE_SHARED=0',
-            '-DREQUIRE_SIMD=1',
-            '-DCMAKE_POSITION_INDEPENDENT_CODE=ON',
-        ], env=env)
 
     def build_cmake_dependency(self, path, options, env=None):
         cur_dir = pt.abspath(os.curdir)
@@ -222,7 +215,6 @@ def _staticlib():
 def make_jpeg_module():
     include_dirs = [
         NumpyImport(),
-        pt.join(JPEG_DIR, 'src'),
         pt.join(PACKAGE_DIR, 'simplejpeg'),
     ]
     static_libs = [pt.join(_libdir(), _staticlib())]
