diff --git a/setup.py b/setup.py
index 24d97e1..85bedbb 100644
--- a/setup.py
+++ b/setup.py
@@ -158,11 +158,8 @@ class cmake_build_ext(build_ext):
         cflags = os.getenv('CFLAGS', '')
         ldflags = os.getenv('LDFLAGS', '')
         if OS == 'linux':
-            # enable LTO
-            cflags = '-flto ' + cflags
             # same as extension
             ldflags = (
-                '-flto '
                 '-Wl,'  # following are linker options
                 '--strip-all,'  # Remove all symbols
                 '--exclude-libs,ALL,'  # Do not export symbols
@@ -247,9 +244,6 @@ def make_jpeg_module():
             '--exclude-libs,ALL,'  # Do not export symbols
             '--gc-sections'  # Remove unused sections
         ])
-        extra_compile_args.extend([
-            '-flto',  # enable LTO
-        ])
     return Extension(
         'simplejpeg._jpeg',
         sources,
