diff --git a/pyproject.toml b/pyproject.toml
index baf8fad..5b97bdc 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -21,7 +21,7 @@ Source = "https://gitlab.com/jfolz/simplejpeg"
 Tracker = "https://gitlab.com/jfolz/simplejpeg/issues"
 
 [build-system]
-requires = ["setuptools>=50.0.3", "wheel", "cmake>=3.6.3", "cython~=3.0.0", "numpy>=2.0.0"]
+requires = ["setuptools>=50.0.3", "wheel", "cython~=3.0.0", "numpy>=2.0.0"]
 build-backend = "setuptools.build_meta"
 
 [tool.cibuildwheel]
diff --git a/setup.py b/setup.py
index 549f09d..00ea9c3 100644
--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,6 @@ import tarfile
 import sysconfig
 import subprocess
 
-from cmake import CMAKE_BIN_DIR
 from setuptools import setup
 from setuptools import find_packages
 from setuptools import Extension
@@ -163,9 +162,8 @@ class cmake_build_ext(build_ext):
         os.chdir(build_dir)
         config = 'Debug' if self.debug else 'Release'
         env = dict(os.environ, **(env or {}))
-        cmake = os.path.join(CMAKE_BIN_DIR, 'cmake')
         subprocess.check_call([
-            cmake,
+            'cmake',
             '-G' + make_type(), '-Wno-dev',
             '-DCMAKE_BUILD_TYPE=' + config,
             *options,
