diff --git a/whisper/transcribe.py b/whisper/transcribe.py
index 0a4cc36..6006274 100644
--- a/whisper/transcribe.py
+++ b/whisper/transcribe.py
@@ -1,5 +1,6 @@
 import argparse
 import os
+import sys
 import traceback
 import warnings
 from typing import TYPE_CHECKING, List, Optional, Tuple, Union
@@ -517,6 +518,9 @@ def transcribe(
 def cli():
     from . import available_models

+    if hasattr(sys.stdout, "reconfigure"):
+        sys.stdout.reconfigure(encoding="utf-8")
+
     def valid_model_name(name):
         if name in available_models() or os.path.exists(name):
             return name
