From b70320d1c5d612672d9f65f0d27932041f3adea5 Mon Sep 17 00:00:00 2001
From: Mariana Meireles <mariana@psychonautgirl.space>
Date: Sun, 6 Feb 2022 18:14:47 -0800
Subject: [PATCH] add conda_prefix

---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index bdae9b1..115ac9f 100644
--- a/Makefile
+++ b/Makefile
@@ -52,3 +52,6 @@ pgo: exprtk_benchmark.cpp exprtk.hpp
 
 clean:
 	rm -f core.* *~ *.o *.bak *stackdump gmon.out *.gcda *.gcno *.gcnor *.gch
+
+install: exprtk.hpp
+	install exprtk.hpp $(CONDA_PREFIX)/include
-- 
2.33.1

