From 17447733831edbd8f04b61fb25d4f8304954fa07 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 28 May 2020 11:35:48 +0200
Subject: [PATCH] Update configure.ac

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 8c5f64e..356cc75 100644
--- a/configure.ac
+++ b/configure.ac
@@ -173,7 +173,7 @@ ATP_FC_FEATURES
 # Initialize Libtool
 #   Disable this shared-library feature because it is not convenient
 #   to have a script and not an executable for atompaw
-LT_INIT([disable-shared])
+LT_INIT([enable-shared])
 LT_PREREQ([2.2.2])
 LTOBJEXT="lo"
 AC_SUBST(LTOBJEXT)
