From 0b4e5acd8fc65b316d2ef17a1e1f47a4b8dc9102 Mon Sep 17 00:00:00 2001
From: Joris Vaillant <joris.vaillant@inria.fr>
Date: Tue, 31 Dec 2024 16:54:01 +0100
Subject: [PATCH] cmake: Don't link with all pinocchio

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8c39dd3..a53e324 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,7 +144,7 @@ target_include_directories(
                             $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 target_link_libraries(${PROJECT_NAME} INTERFACE Boost::serialization)
 if(CURVES_WITH_PINOCCHIO_SUPPORT)
-  target_link_libraries(${PROJECT_NAME} INTERFACE pinocchio::pinocchio)
+  target_link_libraries(${PROJECT_NAME} INTERFACE pinocchio::pinocchio_headers)
   target_compile_definitions(${PROJECT_NAME}
                              INTERFACE -DCURVES_WITH_PINOCCHIO_SUPPORT)
 endif(CURVES_WITH_PINOCCHIO_SUPPORT)
-- 
2.34.1

