diff --git a/src/dtt/CMakeLists.txt b/src/dtt/CMakeLists.txt
index 3f0b13c..f839d88 100644
--- a/src/dtt/CMakeLists.txt
+++ b/src/dtt/CMakeLists.txt
@@ -30,7 +30,7 @@ add_compile_definitions(
     u_short=unsigned\ short
 )
 elseif(UNIX)
-SET( RPCFLAGS -C -N -M -DPORTMAP -DLIGO_GDS -DGDS_NO_EPICS ${max_chnname_size})
+SET( RPCFLAGS -C -N -M -DPORTMAP -DLIGO_GDS -DGDS_NO_EPICS ${max_chnname_size} -Y $ENV{BUILD_PREFIX}/bin)
 FIND_PROGRAM(RPCGEN_PROG rpcgen)
 endif()
 
