--- pylibczirw-6.0.0.orig/setup.py	2026-03-10 12:31:50.000000000 -0300
+++ pylibczirw-6.0.0/setup.py	2026-03-11 08:28:16.000209730 -0300
@@ -24,7 +24,7 @@
     changelog = changelog_file.read()
 
 # List any runtime requirements here
-requirements = ["numpy", "cmake", "xmltodict", "validators", "packaging"]
+requirements = ["numpy", "xmltodict", "validators", "packaging"]
 
 
 class CMakeExtension(Extension):
