From 8562f542a978567e646e6d903724b57869462a95 Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Thu, 21 May 2020 13:21:23 -0400
Subject: [PATCH] Make Framework creation optional through OSX_FRAMEWORK
 option.

---
 CMakeLists.txt | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 34cb7f9..2fc0473 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,6 +27,16 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 	set(SKIP_INSTALL_ALL ${OSX_PACKAGE})
 endif()
 
+# build a framework bundle by default if CMAKE_FRAMEWORK is unset and OSX_PACKAGE is ON
+if(CMAKE_FRAMEWORK)
+	set(OSX_FRAMEWORK_DEFAULT ${CMAKE_FRAMEWORK})
+elseif(OSX_PACKAGE)
+	set(OSX_FRAMEWORK_DEFAULT ON)
+else()
+	set(OSX_FRAMEWORK_DEFAULT OFF)
+endif()
+option(OSX_FRAMEWORK "Build a Framework Bundle on OSX" ${OSX_FRAMEWORK_DEFAULT})
+
 include(FindGit OPTIONAL)
 if (GIT_FOUND)
 	execute_process(
@@ -82,7 +92,7 @@ add_subdirectory(test)
 set_target_properties(ad9361 PROPERTIES
 	VERSION ${VERSION}
 	SOVERSION ${LIBAD9361_VERSION_MAJOR}
-	FRAMEWORK TRUE
+	FRAMEWORK ${OSX_FRAMEWORK}
 	PUBLIC_HEADER ${LIBAD9361_HEADERS}
 	C_STANDARD 99
 	C_STANDARD_REQUIRED ON
-- 
2.20.1

