diff --git a/plugins/cpd/CMakeLists.txt b/plugins/cpd/CMakeLists.txt
index 50d0ea58b..ba85e27a7 100644
--- a/plugins/cpd/CMakeLists.txt
+++ b/plugins/cpd/CMakeLists.txt
@@ -2,12 +2,14 @@ cmake_minimum_required(VERSION 3.13)
 
 if (STANDALONE)
     project(pdal-cpd)
-    include(../../cmake/standalone.cmake NO_POLICY_SCOPE)
+    include("${CMAKE_CURRENT_LIST_DIR}/../../cmake/standalone.cmake" )
+    find_package(PDAL REQUIRED)
 endif(STANDALONE)
 
 set(Cpd_VERSION 0.5)
 
-find_package(Cpd ${Cpd_VERSION} REQUIRED)
+find_package(Cpd ${Cpd_VERSION} REQUIRED CONFIG)
+find_package(Fgt REQUIRED CONFIG)
 option(BUILD_PLUGIN_CPD "Build Coherent Point Drift support" ${Cpd_FOUND})
 
 set(files filters/CpdFilter.cpp)
@@ -18,8 +20,10 @@ PDAL_ADD_PLUGIN(filter_libname filter cpd
         filters/CpdFilter.cpp
     LINK_WITH
         Cpd::Library-C++
+        ${PDAL_LIBRARIES}
     INCLUDES
-        "${include_dirs}"
+        ${PDAL_VENDOR_DIR}/eigen
+        ${CMAKE_CURRENT_LIST_DIR}
     )
 
 if(${WITH_TESTS})
@@ -27,7 +31,7 @@ if(${WITH_TESTS})
         FILES
             test/CpdFilterTest.cpp
         LINK_WITH
-            ${filter_libname}
+            Cpd::Library-C++
         INCLUDES
             "${include_dirs}"
         )
