--- plugins/e57/CMakeLists.txt  2024-06-10 22:08:55.125471649 -0500
+++ plugins/e57/CMakeLists.txt  2024-06-10 22:09:08.013771307 -0500
@@ -3,6 +3,7 @@
 if (STANDALONE)
     find_package(PDAL REQUIRED)
     include("${PROJECT_SOURCE_DIR}/../../cmake/standalone.cmake" )
+    include("${PROJECT_SOURCE_DIR}/../../cmake/nlohmann.cmake" )
 endif(STANDALONE)
 
 add_subdirectory(libE57Format)
@@ -17,6 +18,7 @@
         ${PDAL_VENDOR_DIR}
     LINK_WITH
         E57Format
+        ${PDAL_LIBRARIES}
     INCLUDES
         libE57Format/include
 )
@@ -33,6 +35,7 @@
         io/Uuid.cpp
     LINK_WITH
         E57Format
+        ${PDAL_LIBRARIES}
     INCLUDES
         libE57Format/include
 )
