diff --git Source/LibTIFF4/tif_config.h Source/LibTIFF4/tif_config.h
index 64391b8..252b94d 100644
--- Source/LibTIFF4/tif_config.h
+++ Source/LibTIFF4/tif_config.h
@@ -81,7 +81,9 @@ If your big endian system isn't being detected, add an OS specific check
 #endif // BYTE_ORDER
 
 #ifdef _WIN32
+#if defined(_MSC_VER) && _MSC_VER < 1900
 #define snprintf _snprintf
+#endif // _MSC_VER check
 #define lfind _lfind
 #endif // _WIN32
 
