diff -ru unit_tests/aub_tests/CMakeLists.txt unit_tests/aub_tests/CMakeLists.txt
--- unit_tests/aub_tests/CMakeLists.txt     2019-09-18 16:12:11.000000000 -0500
+++ unit_tests/aub_tests/CMakeLists.txt    2019-10-02 01:59:54.676099184 -0500
@@ -64,7 +64,7 @@
 target_link_libraries(igdrcl_aub_tests igdrcl_mocks)
 target_link_libraries(igdrcl_aub_tests ${NEO_MOCKABLE_LIB_NAME})
 target_link_libraries(igdrcl_aub_tests igdrcl_mocks)
-target_link_libraries(igdrcl_aub_tests gmock-gtest ${IGDRCL_EXTRA_LIBS})
+target_link_libraries(igdrcl_aub_tests gmock-gtest ${IGDRCL_EXTRA_LIBS} -lrt)
 
 target_include_directories(igdrcl_aub_tests BEFORE PRIVATE ${IGDRCL_SOURCE_DIR}/unit_tests/gen_common${BRANCH_DIR_SUFFIX})
 
diff -ru unit_tests/CMakeLists.txt unit_tests/CMakeLists.txt
--- unit_tests/CMakeLists.txt       2019-09-18 16:12:11.000000000 -0500
+++ unit_tests/CMakeLists.txt      2019-10-02 01:56:40.216745561 -0500
@@ -131,7 +131,7 @@
 endif()
 
 target_link_libraries(igdrcl_tests ${NEO_MOCKABLE_LIB_NAME})
-target_link_libraries(igdrcl_tests igdrcl_mocks)
+target_link_libraries(igdrcl_tests igdrcl_mocks -lrt)
 
 option(SHOW_VERBOSE_UTESTS_RESULTS "Use the default/verbose test output" OFF)
 
diff -ru unit_tests/linux/CMakeLists.txt unit_tests/linux/CMakeLists.txt
--- unit_tests/linux/CMakeLists.txt 2019-09-18 16:12:11.000000000 -0500
+++ unit_tests/linux/CMakeLists.txt        2019-10-02 01:57:58.370088895 -0500
@@ -50,7 +50,7 @@
     $<TARGET_OBJECTS:igdrcl_libult>
     $<TARGET_OBJECTS:igdrcl_libult_cs>
   )
-  target_link_libraries(igdrcl_${target_name} ${NEO_MOCKABLE_LIB_NAME} igdrcl_mocks ${IGDRCL_EXTRA_LIBS} gmock-gtest)
+  target_link_libraries(igdrcl_${target_name} ${NEO_MOCKABLE_LIB_NAME} igdrcl_mocks ${IGDRCL_EXTRA_LIBS} gmock-gtest -lrt)
 
   set_property(TARGET igdrcl_${target_name} APPEND_STRING PROPERTY COMPILE_FLAGS ${ASAN_FLAGS})
   target_include_directories(igdrcl_${target_name} PRIVATE
diff -ru unit_tests/tbx/CMakeLists.txt unit_tests/tbx/CMakeLists.txt
--- unit_tests/tbx/CMakeLists.txt   2019-09-18 16:12:11.000000000 -0500
+++ unit_tests/tbx/CMakeLists.txt  2019-10-02 02:01:15.829504810 -0500
@@ -15,7 +15,7 @@
   $<TARGET_OBJECTS:igdrcl_libult>
 )
 
-target_link_libraries(igdrcl_tbx_tests ${NEO_MOCKABLE_LIB_NAME} igdrcl_mocks gmock-gtest ${IGDRCL_EXTRA_LIBS})
+target_link_libraries(igdrcl_tbx_tests ${NEO_MOCKABLE_LIB_NAME} igdrcl_mocks gmock-gtest ${IGDRCL_EXTRA_LIBS} -lrt)
 
 if(WIN32)
   target_include_directories(igdrcl_tbx_tests PRIVATE

