--- package.cmake   2019-09-18 16:12:11.000000000 -0500
+++ package.cmake  2019-10-02 02:05:32.445942562 -0500
@@ -22,12 +22,7 @@
 
   get_os_release_info(os_name os_version os_codename)
 
-  if("${os_name}" STREQUAL "clear-linux-os")
-    # clear-linux-os distribution avoids /etc for distribution defaults.
-    set(_dir_etc "/usr/share/defaults/etc")
-  else()
-    set(_dir_etc "/etc")
-  endif()
+    set(_dir_etc "${CMAKE_INSTALL_PREFIX}/etc")
 
   set(OCL_ICD_RUNTIME_NAME ${CMAKE_SHARED_LIBRARY_PREFIX}${NEO_DLL_NAME_BASE}${CMAKE_SHARED_LIBRARY_SUFFIX})
   install(
