diff --git a/src/match.c b/src/match.c
index 79ea1f4..4377ea3 100644
--- a/src/match.c
+++ b/src/match.c
@@ -106,7 +106,7 @@ VERSION	29/09/2012
 void	match_field(fieldstruct *field, fieldstruct *reffield)
   {
    setstruct	*refset, *refset2, *set, *fieldset;
-   char		str[128];
+   char		str[MAXCHAR];
    double	wcspos[NAXIS],
 		angle,scale,dlng,dlat,sig, dlng2,dlat2,sig2, asig,
 		dangle,dscale, ddlng,ddlat, shear,sangle, sheartot, sangletot,
@@ -114,7 +114,7 @@ void	match_field(fieldstruct *field, fieldstruct *reffield)
    int		i,j,k, lng,lat,
 		naxis, nmax;
 
-  sprintf(str, "Matching field %s...", field->rfilename);
+  sprintf(str, "Matching field %.*s ...", MAXCHAR - 20, field->rfilename);
   NFPRINTF(OUTPUT, str);
 
 /* We are going to use FFT routines */
