--- setup.py	2016-10-09 09:51:46.000000000 +0200
+++ setup.py	2018-05-03 18:58:48.767656453 +0200
@@ -146,9 +146,6 @@
       package_data={LIBNAME:
                     get_package_data(LIBNAME, ('.png', '.svg', '.mo'))},
       data_files=[(r'Doc', [CHM_DOC])] if CHM_DOC else [],
-      entry_points={'gui_scripts':
-                    ['guidata-tests-py%d = guidata.tests:run'\
-                     % sys.version_info.major,]},
       extras_require = {
                         'Doc':  ["Sphinx>=1.1"],
                         },
