diff --git a/setup.py b/setup.py
index 7843679..bd5fc85 100644
--- a/setup.py
+++ b/setup.py
@@ -146,9 +146,6 @@ def build_chm_doc(libname):
       package_data={LIBNAME:
                     get_package_data(LIBNAME, ('.png', '.svg', '.mo'))},
       data_files=[(r'Doc', [CHM_DOC])] if CHM_DOC else [],
-      entry_points={'gui_scripts':
-                    ['guidata-tests-py%d = guidata.tests:run'\
-                     % sys.version_info.major,]},
       extras_require = {
                         'Doc':  ["Sphinx>=1.1"],
                         },
