diff --git a/include/dcgp/problems/symbolic_regression.hpp b/include/dcgp/problems/symbolic_regression.hpp
index 8a84381..46ebf7a 100644
--- a/include/dcgp/problems/symbolic_regression.hpp
+++ b/include/dcgp/problems/symbolic_regression.hpp
@@ -7,7 +7,13 @@
 #include <pagmo/io.hpp>
 #include <pagmo/population.hpp>
 #include <pagmo/types.hpp>
+#define and &&
+#define or ||
+#define not !
 #include <symengine/expression.h>
+#undef and
+#undef or
+#undef not
 #include <vector>
 
 #include <dcgp/expression.hpp>
