diff --git a/dcgpy/expose_symbolic_regression.cpp b/dcgpy/expose_symbolic_regression.cpp
index 9ca39e1..8e20833 100644
--- a/dcgpy/expose_symbolic_regression.cpp
+++ b/dcgpy/expose_symbolic_regression.cpp
@@ -1,3 +1,5 @@
+#include <ciso646>
+
 // See: https://docs.scipy.org/doc/numpy/reference/c-api.array.html#importing-the-api
 // In every cpp file We need to make sure this is included before everything else,
 // with the correct #defines.
@@ -107,4 +109,4 @@ void expose_symbolic_regression()
     expose_data_from_the_gym<&gym::generate_lanczos2>("generate_lanczos2", generate_lanczos2_doc());
     expose_data_from_the_gym<&gym::generate_misra1b>("generate_misra1b", generate_misra1b_doc());
 }
-} // namespace dcgpy
\ No newline at end of file
+} // namespace dcgpy
