diff --git a/include/dcgp/problems/symbolic_regression.hpp b/include/dcgp/problems/symbolic_regression.hpp
index 8a84381..ef5b257 100644
--- a/include/dcgp/problems/symbolic_regression.hpp
+++ b/include/dcgp/problems/symbolic_regression.hpp
@@ -7,7 +7,9 @@
 #include <pagmo/io.hpp>
 #include <pagmo/population.hpp>
 #include <pagmo/types.hpp>
+#define and &&
 #include <symengine/expression.h>
+#undef and
 #include <vector>
 
 #include <dcgp/expression.hpp>
