--- a/casacore/python3/CMakeLists.txt	2017-06-02 14:20:42.513447648 -0400
+++ b/casacore/python3/CMakeLists.txt	2017-06-02 14:21:36.515767923 -0400
@@ -23,13 +23,7 @@
 find_package(Python REQUIRED)
 
 if (PYTHONINTERP_FOUND)
-    if (APPLE)
-        find_package(Boost REQUIRED COMPONENTS python3)
-    else ()
-        # NOTE: the name of the python3 version of boost is probably Debian/Ubuntu specific
-        find_package(Boost REQUIRED COMPONENTS python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
-    endif (APPLE)
-
+    find_package(Boost REQUIRED COMPONENTS python)
     find_package (NUMPY REQUIRED)
 
     # copy the variables to their final destination
