--- casacore/CMakeLists.txt.orig	2015-05-13 20:23:33.020772229 +0000
+++ casacore/CMakeLists.txt	2015-05-13 20:27:00.367197771 +0000
@@ -314,6 +314,10 @@
     add_definitions(-DUSE_STACKTRACE)
 endif(USE_STACKTRACE)
 
+# conda-build hack
+add_definitions(-DCONDA_CASA_ROOT="${CONDA_CASA_ROOT}")
+# end conda-build hack
+
 # Add the modules to be built.
 add_subdirectory (build-tools)
 foreach (module ${_modules})
--- casacore/casa/System/Aipsrc.cc.orig	2015-05-13 20:23:25.424756643 +0000
+++ casacore/casa/System/Aipsrc.cc	2015-05-13 20:29:57.773561857 +0000
@@ -215,9 +215,9 @@
     } else { 
       aipsPath = extAipsPath;
     }
-    // Set the path to home if not defined in any way.
+    // conda-build hack: default is known; hack to fix zero-pad issues on OSX
     if (aipsPath.empty()) {
-      setAipsPath(uhome);
+      setAipsPath(String(String(CONDA_CASA_ROOT).chars()));
       aipsPath = extAipsPath;
     }
     Int n = aipsPath.freq(' ') + aipsPath.freq('	') + 4;
--- casacore/casa/System/Aipsrc.cc.orig	2016-02-02 22:06:41.314849499 +0000
+++ casacore/casa/System/Aipsrc.cc	2016-02-02 22:08:02.513055617 +0000
@@ -422,7 +422,7 @@
     if (find(x, String("user.aipsdir"), String("/aips++"))) {
       home = x;
     } else {
-      home = uhome + x;
+      home = String(String(CONDA_CASA_ROOT).chars());
     }
   }
 }
--- casacore/measures/Measures/MeasIERS.cc.orig	2016-04-11 21:49:32.215695000 +0000
+++ casacore/measures/Measures/MeasIERS.cc	2016-04-11 21:56:01.060660664 +0000
@@ -422,7 +422,8 @@
           }
         }
         if (!found) {
-          String casadata=String(CASADATA);
+	  // Conda NUL-padding workaround:                                                                        
+          String casadata=String(String(CASADATA).chars());
           casadata.gsub("%CASAROOT%", Aipsrc::aipsRoot());
           casadata.gsub("%CASAHOME%", Aipsrc::aipsHome());
           Path cdatapath(casadata);
--- casacore/measures/Measures/MeasTable.cc.orig	2016-04-11 21:49:32.231695000 +0000
+++ casacore/measures/Measures/MeasTable.cc	2016-04-11 21:54:36.895451727 +0000
@@ -2988,7 +2988,8 @@
 	isValid = Table::isReadable(absPathName);
       }
       if(!isValid){
-        String casadata=String(CASADATA);
+	// Conda NUL-padding workaround:                                                                        
+        String casadata=String(String(CASADATA).chars());
         casadata.gsub("%CASAROOT%", Aipsrc::aipsRoot());
         casadata.gsub("%CASAHOME%", Aipsrc::aipsHome());
         Path lPath(casadata + "/" + antRespPath);
