--- casacore/cmake/FindReadline.cmake.orig	2016-06-27 00:25:35.953287749 +0000
+++ casacore/cmake/FindReadline.cmake	2016-06-27 00:25:44.792279239 +0000
@@ -36,7 +36,7 @@
     HINTS ${READLINE_ROOT_DIR} PATH_SUFFIXES include)
   find_library(READLINE_LIBRARY readline
     HINTS ${READLINE_ROOT_DIR} PATH_SUFFIXES lib)
-  find_library(NCURSES_LIBRARY ncurses)   # readline depends on libncurses
+  find_library(NCURSES_LIBRARY ncursesw)   # readline depends on libncurses
   mark_as_advanced(READLINE_INCLUDE_DIR READLINE_LIBRARY NCURSES_LIBRARY)
 
   include(FindPackageHandleStandardArgs)
