diff --git a/SConscript b/SConscript
index 94e691b..21fd9ce 100644
--- a/ndarray_ext/SConscript
+++ b/ndarray_ext/SConscript
@@ -22,6 +22,6 @@ if len(Glob('#/ndarray'))==1:
     standardExternalPackage('ndarray', PREFIX=node.abspath, 
                             LOCAL=True, INCDIR='include')
 else:
-    standardCondaPackage('ndarray', INCDIR='include/ndarray', LINKLIBS=None)
+    standardCondaPackage('ndarray-psana', INCDIR='include/ndarray', LINKLIBS=None)
 
 
