diff --git PyInstaller/compat.py PyInstaller/compat.py
index dd8c7ac..f3ed99c 100644
--- PyInstaller/compat.py
+++ PyInstaller/compat.py
@@ -56,7 +56,10 @@
 elif is_cygwin:
     PYDYLIB_NAMES = {'libpython%d%d.dll' % _pyver}
 elif is_darwin:
-    PYDYLIB_NAMES = {'Python', '.Python', 'libpython%d.%d.dylib' % _pyver}
+    # libpython%d.%dm.dylib for Conda virtual environment installations
+    PYDYLIB_NAMES = {'Python', '.Python', 
+                     'libpython%d.%d.dylib' % _pyver, 
+                     'libpython%d.%dm.dylib' % _pyver}
 elif is_aix:
     # Shared libs on AIX are archives with shared object members, thus the ".a" suffix.
     PYDYLIB_NAMES = {'libpython%d.%d.a' % _pyver}
