diff --git a/CMakeLists.txt b/CMakeLists.txt
index 658b71d..2ecf754 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,9 +48,7 @@ if (LLAMA_BUILD)
     add_subdirectory(vendor/llama.cpp/examples/llava)
     set_target_properties(llava_shared PROPERTIES OUTPUT_NAME "llava")
     # Set CUDA_ARCHITECTURES to OFF on windows
-    if (WIN32)
-        set_target_properties(llava_shared PROPERTIES CUDA_ARCHITECTURES OFF)
-    endif()
+    set_target_properties(llava_shared PROPERTIES CUDA_ARCHITECTURES OFF)
     install(
         TARGETS llava_shared
         LIBRARY DESTINATION ${SKBUILD_PLATLIB_DIR}/llama_cpp