diff --git a/CMakeLists.txt b/CMakeLists.txt
index a705d4b..77e14ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.1)
+cmake_minimum_required (VERSION 3.16)
 
 if (EXISTS ${CMAKE_BINARY_DIR}/conan_toolchain.cmake)
     cmake_policy(SET CMP0091 NEW)
diff --git a/examples/Demo/CMakeLists.txt b/examples/Demo/CMakeLists.txt
index a8cba26..c141498 100644
--- a/examples/Demo/CMakeLists.txt
+++ b/examples/Demo/CMakeLists.txt
@@ -5,7 +5,7 @@
 # Second, it allows us to scan for 3rd party library dependencies and install
 # them with tixi (cmake doesn't allow scanning dlls only).
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.16)
 
 project(Tixi-Demos)
 
