From c07afa0bfcb8e887828cd5099858513e70e1eddc Mon Sep 17 00:00:00 2001
From: Martin Siggel <martin.siggel@dlr.de>
Date: Tue, 29 Jun 2021 21:53:42 +0200
Subject: [PATCH] Disabled fortran example

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6f34e0c..0ed290b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,7 +111,7 @@ endif(TIXI_BUILD_TESTS)
 #demos
 add_subdirectory(examples/Demo)
 if (TIXI_ENABLE_FORTRAN)
-  add_subdirectory(examples/fortran77)
+  #add_subdirectory(examples/fortran77)
 endif(TIXI_ENABLE_FORTRAN)
 
 # create the doc
-- 
2.31.1.windows.1

