diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ad7f650..ffc8c69 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -100,8 +100,6 @@ set(MARVEL_INCLUDE_DIR
         "../thirdparty/stb"
         "../thirdparty/imnodes"
         "../thirdparty/ImGuiFileDialog"
-        "../thirdparty/glfw/include/"
-        "../thirdparty/glfw/deps/"
         "../thirdparty/imgui/"
         "../thirdparty/gl3w/"
         "../thirdparty/imgui/misc/freetype/"
diff --git a/src/distribution.cmake b/src/distribution.cmake
index b858384..aabd837 100644
--- a/src/distribution.cmake
+++ b/src/distribution.cmake
@@ -40,7 +40,9 @@ if(WIN32)

 	set_target_properties(_dearpygui PROPERTIES SUFFIX ".pyd")
 	set_target_properties(_dearpygui PROPERTIES PREFIX "")
-	target_link_libraries(_dearpygui PUBLIC d3d11 dxgi dwmapi ${Python_LIBRARIES} freetype)
+	find_package(Freetype REQUIRED)
+	target_link_libraries(_dearpygui PUBLIC d3d11 dxgi dwmapi ${Python_LIBRARIES} Freetype::Freetype)
 
 elseif(APPLE)
 
diff --git a/thirdparty/CMakeLists.txt b/thirdparty/CMakeLists.txt
index 4e4f721..ddb1ac2 100644
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -1,13 +1,7 @@
-﻿
-set(GLFW_BUILD_DOCS OFF)
-set(GLFW_BUILD_EXAMPLES OFF)
-set(GLFW_BUILD_TESTS OFF)
-set(GLFW_INSTALL OFF)
-
 if(WIN32)
+    # freetype is found via find_package in src/distribution.cmake
 elseif(APPLE)
-    add_subdirectory ("glfw")
+    find_package(glfw3 REQUIRED)
 else() #Linux
-	add_subdirectory ("glfw")
+    find_package(glfw3 REQUIRED)
 endif()
