Fix compiler selection
author: Jan Janssen (MPIE)

---
 setup.py | 7 +-
 1 file changed, 3 insertion(+), 4 deletion(-)

diff --git a/setup.py b/etup.py
index 0849e03d..a366828d 100644
--- setup.py	(date 1568712669000)
+++ setup.py	(date 1568722198133)
@@ -13,11 +13,10 @@
 
 
 def check_compiler():
-    import subprocess
-    output = subprocess.Popen(['gcc'], stderr=subprocess.PIPE).communicate()[1]
-    if b'clang' in output:
+    import platform
+    if 'Darwin' in platform.system():
         return 'clang'
-    if b'gcc' in output:
+    elif 'Linux' in platform.system():
         return 'gcc'
 
 
