diff --git a/host/CMakeLists.txt b/host/CMakeLists.txt
index 916138ee9..0bce6c726 100644
--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -260,6 +260,7 @@ if(MSVC)
         -D_CRT_NONSTDC_NO_WARNINGS
         -D_CRT_NONSTDC_NO_DEPRECATE
         -D_WINSOCK_DEPRECATED_NO_WARNINGS
+        -DHAVE_SNPRINTF
     )
     # multi-threaded build and increases the number of addressable sections in an .obj file.
     add_compile_options(/MP /bigobj)
