From 91e5a52878c90e8eb4e5ed8893a6b7f42b2361b2 Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Fri, 10 Aug 2018 18:07:17 -0400
Subject: [PATCH 4/4] Fix build RPATH remaining on installed libpyuhd by not
 setting it.

---
 host/python/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/host/python/CMakeLists.txt b/host/python/CMakeLists.txt
index 830c93cc1..6847501a1 100644
--- a/host/python/CMakeLists.txt
+++ b/host/python/CMakeLists.txt
@@ -51,6 +51,9 @@ else()
         target_link_options(pyuhd PRIVATE "LINKER:-undefined,dynamic_lookup")
     endif(APPLE)
 endif(WIN32)
+# skip setting pyuhd's RPATH because it is installed as a python data file so
+# the build RPATH would never be overwritten by an install RPATH
+set_target_properties(pyuhd PROPERTIES SKIP_BUILD_RPATH ON BUILD_WITH_INSTALL_RPATH OFF)
 # Copy pyuhd library to the staging directory
 add_custom_command(TARGET pyuhd
     POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:pyuhd> ${CMAKE_CURRENT_BINARY_DIR}/uhd/$<TARGET_FILE_NAME:pyuhd>)
-- 
2.20.1

