From e38d6b088bce4c43ebcc6015a0ff860b734535e8 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 4 Dec 2020 15:08:16 +0100
Subject: [PATCH] Fix Import

---
 gb_code/gb_generator.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gb_code/gb_generator.py b/gb_code/gb_generator.py
index 74cc195..c458f51 100644
--- a/gb_code/gb_generator.py
+++ b/gb_code/gb_generator.py
@@ -18,7 +18,7 @@
 import numpy as np
 from numpy import dot, cross
 from numpy.linalg import det, norm
-import csl_generator as cslgen
+import gb_code.csl_generator as cslgen
 import warnings
 
 
