From 1e731482d14b97840a4bc148e2735136ee0aed39 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Fri, 11 Feb 2022 14:54:36 +0100
Subject: [PATCH] Bump patch version to 3.2.1

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ef6de7..06bbc79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,7 +33,7 @@ cmake_minimum_required(VERSION 2.6)
 
 PROJECT(qpOASES CXX)
 SET(PACKAGE_NAME "qpOASES")
-SET(PACKAGE_VERSION "3.2.0")
+SET(PACKAGE_VERSION "3.2.1")
 SET(PACKAGE_SO_VERSION "3.2")
 SET(PACKAGE_DESCRIPTION "An implementation of the online active set strategy")
 SET(PACKAGE_AUTHOR "Hans Joachim Ferreau, Andreas Potschka, Christian Kirches et al.")
