--- tests/Makefile	2017-02-11 20:49:48.000000000 -0500
+++ tests/Makefile_new	2017-02-11 21:06:53.000000000 -0500
@@ -1,9 +1,9 @@
 
-LIB_FLAGS = -larmadillo
+LIB_FLAGS = -larmadillo -L${PREFIX}/lib
 #LIB_FLAGS = -lblas -llapack 
 #LIB_FLAGS = -lopenblas -llapack 
 
-CXX_FLAGS = -std=c++11 -Wshadow -Wall -pedantic -O0
+CXX_FLAGS = -std=c++11 -Wshadow -Wall -pedantic -O0 -I${PREFIX}/include
 #CXX_FLAGS = -std=c++11 -Wshadow -Wall -pedantic -O0 -DARMA_DONT_USE_WRAPPER
 #CXX_FLAGS = -std=c++11 -Wshadow -Wall -pedantic -O2
 
