From e738c7384002d206c48daefbfdcc46c541fa5363 Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Tue, 4 Feb 2025 17:33:47 -0600
Subject: [PATCH 2/2] BLD: Move CMake config install directory

---
 src/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5f7c3ec..0ed8ffe 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -121,7 +121,7 @@ endif()
 #   * <prefix>/cmake/
 #   * <prefix>/lib*/
 #   * <prefix>/include/
-set(config_install_dir "${CMAKE_INSTALL_PREFIX}/cmake")
+set(config_install_dir "${CMAKE_INSTALL_PREFIX}/lib/cmake")
 
 set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
 
@@ -173,7 +173,7 @@ install(
 #   * <prefix>/cmake/nvimgcodec/nvimgcodecConfigVersion.cmake
 install(
     FILES "${project_config}" "${version_config}"
-    DESTINATION "cmake/nvimgcodec"
+    DESTINATION "${config_install_dir}/nvimgcodec"
     COMPONENT lib
 )
 
@@ -182,7 +182,7 @@ install(
 install(
     EXPORT "${TARGETS_EXPORT_NAME}"
     NAMESPACE "${namespace}"
-    DESTINATION "cmake/nvimgcodec"
+    DESTINATION "${config_install_dir}/nvimgcodec"
     COMPONENT lib
 )
 
-- 
2.47.1

