From a3327de7c8b08ac2224efa6193f2918f411152f8 Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Tue, 17 Sep 2024 11:09:47 -0500
Subject: [PATCH 3/9] DOC: Add Fatal Error if extensions are built without
 library

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7e19bdf..5e4162b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,7 +87,13 @@ option(BUILD_CVCUDA_SAMPLES "Build CVCUDA samples" OFF)
 option(BUILD_DOCS "Build documentation" OFF)
 option(BUILD_EXTENSIONS "Build extensions modules" ON)
 option(BUILD_NVJPEG_EXT "Build nvjpeg extensions module" ON)
+if(BUILD_NVJPEG_EXT AND NOT BUILD_LIBRARY)
+    message(FATAL_ERROR "BUILD_LIBRARY must be enabled if BUILD_NVJPEG_EXT is enabled. The NVJPEG extension links to libimgcodec_static.")
+endif()
 option(BUILD_NVJPEG2K_EXT "Build nvjpeg2k extensions module" ON)
+if(BUILD_NVJPEG2K_EXT AND NOT BUILD_LIBRARY)
+    message(FATAL_ERROR "BUILD_LIBRARY must be enabled if BUILD_NVJPEG2K_EXT is enabled. The NVJPEG2K extension links to libimgcodec_static.")
+endif()
 option(BUILD_NVBMP_EXT "Build nvbmp extensions module" ON)
 option(BUILD_NVPNM_EXT "Build nvpnm extensions module" ON)
 option(BUILD_LIBJPEG_TURBO_EXT "Build libjpeg-turbo extensions module" ON)
-- 
2.34.1

