--- setup_old.py	2020-09-09 18:30:23.984852234 -0500
+++ setup.py	2020-09-09 19:38:06.969621657 -0500
@@ -8,11 +8,9 @@
 
 os.chdir(os.path.dirname(sys.argv[0]) or ".")
 
-import grass_session  # isort:skip
-
 setup(
     name="grass-session",
-    version=grass_session.version,
+    version="0.5",
     description="GRASS GIS session utilities",
     long_description=open("README.rst", "rt").read(),
     url="https://github.com/zarch/grass-session",
