Index: src/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/CMakeLists.txt  (revision 9740b708b5dd8dc0b3c017eafe4dd2f716d86822)
+++ src/CMakeLists.txt  (date 1590162983115)
@@ -8,10 +8,6 @@
    option(BUILD_TESTS "Build unit tests." ON)
    option(USE_OPENMP "Parallelization using OpenMP." ON)
    option (BUILD_SHARED_LIBS "Build diffusioncma as a shared library" OFF)
-   if(NOT CMAKE_COMPILER_IS_GNUCC)
-       message("Not using OpenMP")
-       set(USE_OPENMP OFF)
-   endif(NOT CMAKE_COMPILER_IS_GNUCC)
 
 
 # ----- HANDLE OPTIONS -----

