--- a/cpp/open3d/utility/ExtractZIP.cpp	2024-01-05 11:44:03
+++ b/cpp/open3d/utility/ExtractZIP.cpp	2024-01-05 11:44:13
@@ -12,7 +12,7 @@
 
 #include <errno.h>
 #include <stdio.h>
-#include <unzip.h>
+#include <minizip/unzip.h>
 
 #include <iostream>
 #include <string>
@@ -43,7 +43,7 @@
     int err = UNZ_OK;
     FILE *fout = nullptr;
     void *buf;
-    uInt size_buf;
+    unsigned int size_buf;
 
     unz_file_info64 file_info;
     err = unzGetCurrentFileInfo64(uf, &file_info, filename_inzip,
@@ -179,7 +179,7 @@
     // ExtractFromZIP supports password. Can be exposed if required in future.
     const std::string password = "";
 
-    for (uLong i = 0; i < gi.number_entry; ++i) {
+    for (unsigned long i = 0; i < gi.number_entry; ++i) {
         err = ExtractCurrentFile(uf, extract_dir, password);
         if (err != UNZ_OK) {
             // Close file, before throwing exception.

