diff --git a/3rdparty/find_dependencies.cmake b/3rdparty/find_dependencies.cmake
index 169bfbe1a54..7b7aedd9d21 100644
--- a/3rdparty/find_dependencies.cmake
+++ b/3rdparty/find_dependencies.cmake
@@ -1174,8 +1174,9 @@ endif()
 if(BUILD_GUI)
     if(USE_SYSTEM_IMGUI)
         open3d_find_package_3rdparty_library(3rdparty_imgui
-            PACKAGE ImGui
-            TARGETS ImGui::ImGui
+            REQUIRED
+            PACKAGE imgui
+            TARGETS imgui::imgui
         )
         if(NOT 3rdparty_imgui_FOUND)
             set(USE_SYSTEM_IMGUI OFF)
