diff --git a/3rdparty/find_dependencies.cmake b/3rdparty/find_dependencies.cmake
index 169bfbe1a54..7b7aedd9d21 100644
--- a/3rdparty/find_dependencies.cmake
+++ b/3rdparty/find_dependencies.cmake
@@ -1174,8 +1174,9 @@ endif()
 if(BUILD_GUI)
     if(USE_SYSTEM_IMGUI)
         open3d_find_package_3rdparty_library(3rdparty_imgui
-            PACKAGE ImGui
-            TARGETS ImGui::ImGui
+            REQUIRED
+            PACKAGE imgui
+            TARGETS imgui::imgui
         )
         if(NOT 3rdparty_imgui_FOUND)
             set(USE_SYSTEM_IMGUI OFF)
diff --git a/3rdparty/cmake/FindImGui.cmake b/3rdparty/cmake/FindImGui.cmake
deleted file mode 100644
index a7033c59885..00000000000
--- a/3rdparty/cmake/FindImGui.cmake
+++ /dev/null
@@ -1,19 +0,0 @@
-find_path(ImGui_INCLUDE_DIR NAMES imgui.h PATH_SUFFIXES imgui)
-find_library(ImGui_LIBRARY NAMES imgui)
-find_library(stb_LIBRARY NAMES stb)
-if(ImGui_INCLUDE_DIR AND ImGui_LIBRARY)
-    add_library(ImGui::ImGui UNKNOWN IMPORTED)
-    set_target_properties(ImGui::ImGui PROPERTIES
-        INTERFACE_INCLUDE_DIRECTORIES "${ImGui_INCLUDE_DIR}"
-        IMPORTED_LOCATION "${ImGui_LIBRARY}"
-    )
-    if(stb_LIBRARY)
-        set_target_properties(ImGui::ImGui PROPERTIES
-            INTERFACE_LINK_LIBRARIES "${stb_LIBRARY}"
-        )
-    endif()
-endif()
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(ImGui  DEFAULT_MSG  ImGui_LIBRARY ImGui_INCLUDE_DIR)
-
