--- 3rdparty/find_dependencies.cmake 2024-01-05 12:17:53
+++ 3rdparty/find_dependencies.cmake 2024-01-05 12:24:45
@@ -874,6 +874,10 @@
             LIB_DIR      ${BORINGSSL_LIB_DIR}
             LIBRARIES    ${BORINGSSL_LIBRARIES}
             DEPENDS      ext_zlib ext_boringssl
+        )
+        open3d_pkg_config_3rdparty_library(3rdparty_libidn2
+            SEARCH_ARGS libidn2
+            
         )
     endif()
 
@@ -895,9 +899,10 @@
         #     _Curl_resolv in libcurl.a(hostip.c.o)
         # ```
         # The "Foundation" framework is already linked by GLFW.
-        target_link_libraries(3rdparty_curl INTERFACE "-framework SystemConfiguration")
+        target_link_libraries(3rdparty_curl INTERFACE "-framework SystemConfiguration -framework CoreFoundation")
     endif()
     target_link_libraries(3rdparty_curl INTERFACE 3rdparty_openssl)
+    target_link_libraries(3rdparty_curl INTERFACE 3rdparty_libidn2)
 endif()
 list(APPEND Open3D_3RDPARTY_PRIVATE_TARGETS_FROM_CUSTOM Open3D::3rdparty_curl)
 
