From 08897b66cf89c405ba8d9edbe18580e5126ef844 Mon Sep 17 00:00:00 2001
From: kiwi0fruit <peter.zagubisalo@gmail.com>
Date: Tue, 22 Jan 2019 00:48:43 +0700
Subject: [PATCH] patch

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 48b7b37..2f81db3 100644
--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,7 @@ import sys
 basedir = path.dirname(path.abspath(__file__))
 extra_args = {}
 
-if (3, 6) > sys.version_info >= (3, 5):
+if False:
     in_dir = path.join(basedir, 'pyppeteer')
     out_dir = path.join(basedir, '.pyppeteer')
     packages = ['pyppeteer']
@@ -88,7 +88,7 @@ setup(
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
     ],
-    python_requires='>=3.5',
+    python_requires='>=3.6',
     test_suite='tests',
     tests_require=test_requirements,
     **extra_args
-- 
2.16.1.windows.1

