diff --git a/build/gen.py b/build/gen.py
index 99de57e..da2564f 100755
--- a/build/gen.py
+++ b/build/gen.py
@@ -330,6 +330,7 @@ def WriteGNNinja(path, platform, host, options):
       ])
       # This is needed by libc++.
       libs.append('-ldl')
+      libs.append('-lrt')
     elif platform.is_darwin():
       min_mac_version_flag = '-mmacosx-version-min=10.9'
       cflags.append(min_mac_version_flag)
