--- a/setup.py	2019-02-26 11:05:15.000000000 +0000
+++ b/setup.py	2021-06-03 10:28:01.819901366 +0100
@@ -2,16 +2,11 @@
 from setuptools import setup, find_packages
 from os import path
 
-here = path.abspath(path.dirname(__file__))
-
-with io.open(path.join(here, 'DESCRIPTION.rst'), encoding='utf-8') as f:
-   long_description = f.read()
 
 setup(
     name='signalr-client-threads',
     version='0.0.12',
     description='Fork of SignalR client for Python based on threads instead of gevent',
-    long_description=long_description,
     url='https://github.com/PawelTroka/signalr-client-threads',
     author='Taucraft Limited, Michael Herrmann, Pawel Troka',
     author_email='info@taucraft.com, pawel.troka@outlook.com',
