From 64732e76c1cb3d2f7ee62d05d0fa2bd8601b9243 Mon Sep 17 00:00:00 2001
From: Gabriel Reis <gabrielcnr@gmail.com>
Date: Thu, 3 Jun 2021 09:11:28 +0100
Subject: [PATCH] FIX: setup.py description file missing

---
 setup.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/setup.py b/setup.py
index 5f6c40f..e9ce535 100644
--- a/setup.py
+++ b/setup.py
@@ -2,16 +2,11 @@ import io
 from setuptools import setup, find_packages
 from os import path
 
-here = path.abspath(path.dirname(__file__))
-
-with io.open(path.join(here, 'DESCRIPTION.rst'), encoding='utf-8') as f:
-   long_description = f.read()
 
 setup(
     name='signalr-client-threads',
     version='0.0.12',
     description='Fork of SignalR client for Python based on threads instead of gevent',
-    long_description=long_description,
     url='https://github.com/PawelTroka/signalr-client-threads',
     author='Taucraft Limited, Michael Herrmann, Pawel Troka',
     author_email='info@taucraft.com, pawel.troka@outlook.com',
-- 
2.24.3 (Apple Git-128)
