--- v4-client-cpp/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -9,2 +9,8 @@
-set(CMAKE_CXX_STANDARD 20)
-set(warnings "-Wall -Wextra")
+set(CMAKE_CXX_STANDARD 17)
+if(MSVC)
+  set(warnings "/W3 /wd4996 /wd4251")
+  set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+  add_compile_options(/D_WIN32_WINNT=0x0601 /bigobj)
+else()
+  set(warnings "-Wall -Wextra")
+endif()
@@ -19,2 +19,2 @@

-find_package(Catch2 REQUIRED)
+find_package(Catch2 3 REQUIRED)
@@ -24,3 +24,2 @@

-add_subdirectory(lib)
-add_subdirectory(examples)
\ No newline at end of file
+add_subdirectory(lib/proto)
