--- v4-client-cpp/lib/proto/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -3,6 +3,39 @@
 file(GLOB_RECURSE proto_files RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.proto)
-protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS ${proto_files})

-add_library(dydx_v4_proto ${BALOO_BIN_TYPE} ${PROTO_SRCS} ${PROTO_HDRS})
-target_include_directories(dydx_v4_proto PUBLIC ${Protobuf_INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(dydx_v4_proto PUBLIC ${Protobuf_LIBRARIES})
\ No newline at end of file
+find_package(Protobuf REQUIRED)
+include_directories(${Protobuf_INCLUDE_DIRS})
+include(GNUInstallDirs)
+
+set(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include/dydx_v4_proto")
+set(PROTO_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/_generated")
+file(MAKE_DIRECTORY ${PROTO_BINARY_DIR})
+
+add_library(dydx_v4_proto_obj OBJECT ${proto_files})
+set_property(TARGET dydx_v4_proto_obj PROPERTY POSITION_INDEPENDENT_CODE 1)
+target_sources(dydx_v4_proto_obj PRIVATE ${proto_files})
+target_include_directories(dydx_v4_proto_obj PUBLIC ${PROTO_BINARY_DIR} $ENV{PREFIX}/include $ENV{PREFIX}/Library/include)
+
+if(WIN32)
+    add_definitions(-DBUILDING_DLL)
+    protobuf_generate(
+        TARGET dydx_v4_proto_obj
+        PROTO_OUT_DIR ${PROTO_BINARY_DIR}
+        OUT_VAR PROTO_GENERATED_FILES
+        EXPORT_MACRO DLL_EXPORT_API)
+else()
+    protobuf_generate(
+        TARGET dydx_v4_proto_obj
+        PROTO_OUT_DIR ${PROTO_BINARY_DIR}
+        OUT_VAR PROTO_GENERATED_FILES)
+endif()
+
+add_library(dydx_v4_proto SHARED $<TARGET_OBJECTS:dydx_v4_proto_obj>)
+if(WIN32)
+    target_link_libraries(dydx_v4_proto PUBLIC "${Protobuf_LIBRARIES}" abseil_dll)
+else()
+    target_link_libraries(dydx_v4_proto PUBLIC protobuf::libprotobuf)
+endif()
+
+include($ENV{RECIPE_DIR}/cmake/InstallDYDXProtobufHeaders.txt)
+include($ENV{RECIPE_DIR}/cmake/InstallLibraryVersion.txt)
+include($ENV{RECIPE_DIR}/cmake/InstallCmakeConfigFiles.txt)
+include($ENV{RECIPE_DIR}/cmake/InstallPkgConfigFiles.txt)

