--- v4-client-cpp/lib/proto/cosmos/base/v1beta1/coin.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/cosmos/base/v1beta1/coin.proto	2021-10-14 15:00:00.000000000 +0000
@@ -6,2 +6,10 @@
 import "amino/amino.proto";
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50000;
+}
+extend google.protobuf.FieldOptions {
+  string cpp_field_export = 50002;
+}

@@ -18,3 +26,3 @@

-  string denom  = 1;
+  string denom  = 1 [(cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];
   string amount = 2 [
@@ -23,3 +31,4 @@
     (gogoproto.nullable)   = false,
-    (amino.dont_omitempty) = true
+    (amino.dont_omitempty) = true,
+    (cpp_field_export) = "DYDX_V4_PROTO_EXPORT"
   ];
@@ -34,5 +43,5 @@

-  string denom  = 1;
+  string denom  = 1 [(cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];
   string amount = 2
-      [(cosmos_proto.scalar) = "cosmos.Dec", (gogoproto.customtype) = "Dec", (gogoproto.nullable) = false];
+      [(cosmos_proto.scalar) = "cosmos.Dec", (gogoproto.customtype) = "Dec", (gogoproto.nullable) = false, (cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];
 }
@@ -41,3 +50,3 @@
 message IntProto {
-  string int = 1 [(cosmos_proto.scalar) = "cosmos.Int", (gogoproto.customtype) = "Int", (gogoproto.nullable) = false];
+  string int = 1 [(cosmos_proto.scalar) = "cosmos.Int", (gogoproto.customtype) = "Int", (gogoproto.nullable) = false, (cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];
 }
@@ -46,3 +55,5 @@
 message DecProto {
-  string dec = 1 [(cosmos_proto.scalar) = "cosmos.Dec", (gogoproto.customtype) = "Dec", (gogoproto.nullable) = false];
+  string dec = 1 [(cosmos_proto.scalar) = "cosmos.Dec", (gogoproto.customtype) = "Dec", (gogoproto.nullable) = false, (cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];
 }
+
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
