--- v4-client-cpp/lib/proto/cosmos/base/v1beta1/coin.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/cosmos/base/v1beta1/coin.proto	2021-10-14 15:00:00.000000000 +0000
@@ -2,3 +2,2 @@
 package cosmos.base.v1beta1;
-
 import "gogoproto/gogo.proto";
@@ -6,3 +5,10 @@
 import "amino/amino.proto";
+import "google/protobuf/descriptor.proto";

+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50000;
+}
+extend google.protobuf.MessageOptions {
+  string cpp_class_export = 50001;
+}
 option go_package                       = "github.com/cosmos/cosmos-sdk/types";
@@ -10,3 +16,2 @@
 option (gogoproto.stringer_all)         = false;
-
 // Coin defines a token with a denomination and an amount.
@@ -17,3 +22,3 @@
   option (gogoproto.equal) = true;
-
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   string denom  = 1;
@@ -26,3 +31,2 @@
 }
-
 // DecCoin defines a token with a denomination and a decimal amount.
@@ -33,3 +37,3 @@
   option (gogoproto.equal) = true;
-
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   string denom  = 1;
@@ -38,11 +42,13 @@
 }
-
 // IntProto defines a Protobuf wrapper around an Int object.
 message IntProto {
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   string int = 1 [(cosmos_proto.scalar) = "cosmos.Int", (gogoproto.customtype) = "Int", (gogoproto.nullable) = false];
 }
-
 // DecProto defines a Protobuf wrapper around a Dec object.
 message DecProto {
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   string dec = 1 [(cosmos_proto.scalar) = "cosmos.Dec", (gogoproto.customtype) = "Dec", (gogoproto.nullable) = false];
 }
+
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
