--- v4-client-cpp/lib/proto/cosmos/tx/v1beta1/tx.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/cosmos/tx/v1beta1/tx.proto	2021-10-14 15:00:00.000000000 +0000
@@ -9,2 +9,10 @@
 import "cosmos_proto/cosmos.proto";
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50002;
+}
+extend google.protobuf.FieldOptions {
+  string cpp_field_export = 50003;
+}

@@ -51,3 +59,3 @@
   // representation in TxRaw.
-  bytes body_bytes = 1;
+  bytes body_bytes = 1 [(cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];

@@ -55,3 +63,3 @@
   // representation in TxRaw.
-  bytes auth_info_bytes = 2;
+  bytes auth_info_bytes = 2 [(cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];

@@ -60,3 +68,3 @@
   // attacker
-  string chain_id = 3;
+  string chain_id = 3 [(cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];

@@ -256 +264,3 @@
 }
+
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
