--- v4-client-cpp/lib/proto/cosmos/tx/v1beta1/tx.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/cosmos/tx/v1beta1/tx.proto	2021-10-14 15:00:00.000000000 +0000
@@ -9,2 +9,10 @@
 import "cosmos_proto/cosmos.proto";
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50000;
+}
+extend google.protobuf.MessageOptions {
+  string cpp_class_export = 50001;
+}

@@ -49,2 +57,3 @@
 message SignDoc {
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   // body_bytes is protobuf serialization of a TxBody that matches the
@@ -52,2 +61,3 @@
   bytes body_bytes = 1;
+  option (gogoproto.moretags) = "DYDX_V4_PROTO_EXPORT:mutable_body_bytes";

@@ -56,2 +66,3 @@
   bytes auth_info_bytes = 2;
+  option (gogoproto.moretags) = "DYDX_V4_PROTO_EXPORT:mutable_auth_info_bytes";

@@ -61,2 +72,3 @@
   string chain_id = 3;
+  option (gogoproto.moretags) = "DYDX_V4_PROTO_EXPORT:mutable_chain_id";

@@ -256 +268,3 @@
 }
+
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
