--- v4-client-cpp/lib/proto/cosmos/crypto/secp256k1/keys.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/cosmos/crypto/secp256k1/keys.proto	2021-10-14 15:00:00.000000000 +0000
@@ -2,7 +2,15 @@
 package cosmos.crypto.secp256k1;
 import "amino/amino.proto";
 import "gogoproto/gogo.proto";
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50000;
+}
+extend google.protobuf.MessageOptions {
+  string cpp_class_export = 50001;
+}
 option go_package = "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1";
+
 // PubKey defines a secp256k1 public key
 // Key is the compressed form of the pubkey. The first byte depends is a 0x02 byte
 // if the y-coordinate is the lexicographically largest of the two associated with
@@ -10,6 +18,7 @@
 // This prefix is followed with the x-coordinate.
 message PubKey {
   option (amino.name) = "tendermint/PubKeySecp256k1";
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   // The Amino encoding is simply the inner bytes field, and not the Amino
   // encoding of the whole PubKey struct.
   //
@@ -22,11 +31,13 @@
   option (amino.message_encoding)     = "key_field";
   option (gogoproto.goproto_stringer) = false;
   bytes key = 1;
+  option (gogoproto.moretags) = "DYDX_V4_PROTO_EXPORT:mutable_key";
 }
 // PrivKey defines a secp256k1 private key.
 message PrivKey {
   option (amino.name)             = "tendermint/PrivKeySecp256k1";
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   option (amino.message_encoding) = "key_field";
   bytes key = 1;
+  option (gogoproto.moretags) = "DYDX_V4_PROTO_EXPORT:mutable_key";
 }
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
