--- v4-client-cpp/lib/proto/cosmos/crypto/secp256k1/keys.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/cosmos/crypto/secp256k1/keys.proto	2021-10-14 15:00:00.000000000 +0000
@@ -5,2 +5,10 @@
 import "gogoproto/gogo.proto";
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50004;
+}
+extend google.protobuf.FieldOptions {
+  string cpp_field_export = 50005;
+}

@@ -28,3 +36,3 @@

-  bytes key = 1;
+  bytes key = 1 [(cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];
 }
@@ -36,3 +44,5 @@

-  bytes key = 1;
+  bytes key = 1 [(cpp_field_export) = "DYDX_V4_PROTO_EXPORT"];
 }
+
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
