--- v4-client-cpp/lib/proto/dydxprotocol/subaccounts/subaccounts.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/dydxprotocol/subaccounts/subaccounts.proto	2021-10-14 15:00:00.000000000 +0000
@@ -3,12 +3,24 @@
 import "cosmos_proto/cosmos.proto";
 import "dydxprotocol/subaccounts/asset_position.proto";
 import "dydxprotocol/subaccounts/perpetual_position.proto";
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50000;
+}
+extend google.protobuf.MessageOptions {
+  string cpp_class_export = 50001;
+}
+
 option go_package = "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types";
+
 // SubaccountId defines a unique identifier for a Subaccount.
 message SubaccountId {
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   // The address of the wallet that owns this subaccount.
   string owner = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  option (gogoproto.moretags) = "DYDX_V4_PROTO_EXPORT:mutable_owner";
   // < 128 Since 128 should be enough to start and it fits within
   // 1 Byte (1 Bit needed to indicate that the first byte is the last).
   uint32 number = 2;
@@ -16,6 +28,7 @@
 // Subaccount defines a single sub-account for a given address.
 // Subaccounts are uniquely indexed by a subaccountNumber/owner pair.
 message Subaccount {
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   // The Id of the Subaccount
   SubaccountId id = 1;
   // All `AssetPosition`s associated with this subaccount.
@@ -28,3 +41,5 @@
   // subaccount.
   bool margin_enabled = 4;
 }
+
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
