--- v4-client-cpp/lib/proto/dydxprotocol/subaccounts/subaccounts.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/dydxprotocol/subaccounts/subaccounts.proto	2021-10-14 15:00:00.000000000 +0000
@@ -2,3 +2,2 @@
 package dydxprotocol.subaccounts;
-
 import "cosmos_proto/cosmos.proto";
@@ -6,2 +5,10 @@
 import "dydxprotocol/subaccounts/perpetual_position.proto";
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50000;
+}
+extend google.protobuf.MessageOptions {
+  string cpp_class_export = 50001;
+}

@@ -11,4 +18,6 @@
 message SubaccountId {
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   // The address of the wallet that owns this subaccount.
   string owner = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  option (gogoproto.moretags) = "DYDX_V4_PROTO_EXPORT:mutable_owner";
   // < 128 Since 128 should be enough to start and it fits within
@@ -17,3 +26,2 @@
 }
-
 // Subaccount defines a single sub-account for a given address.
@@ -21,2 +29,3 @@
 message Subaccount {
+  option (cpp_class_export) = "DYDX_V4_PROTO_EXPORT";
   // The Id of the Subaccount
@@ -33 +42,3 @@
 }
+
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
