--- v4-client-cpp/lib/proto/dydxprotocol/subaccounts/subaccount.proto.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/dydxprotocol/subaccounts/subaccount.proto	2021-10-14 15:00:00.000000000 +0000
@@ -2,3 +2,2 @@
 package dydxprotocol.subaccounts;
-
 import "cosmos_proto/cosmos.proto";
@@ -6,5 +5,12 @@
 import "dydxprotocol/subaccounts/perpetual_position.proto";
+import "google/protobuf/descriptor.proto";

-option go_package = "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types";
+extend google.protobuf.FileOptions {
+  string cpp_file_export = 50000;
+}
+extend google.protobuf.FieldOptions {
+  string cpp_field_export = 50002;
+}

+option go_package = "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types";
 // SubaccountId defines a unique identifier for a Subaccount.
@@ -12,3 +18,4 @@
   // The address of the wallet that owns this subaccount.
-  string owner = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  string owner = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString",
+                     (cpp_field_export) = "DYDX_V4_PROTO_EXPORT" ];
   // < 128 Since 128 should be enough to start and it fits within
@@ -17,3 +24,2 @@
 }
-
 // Subaccount defines a single sub-account for a given address.
@@ -33 +39,3 @@
 }
+
+option (cpp_file_export) = "DYDX_V4_PROTO_EXPORT";
