From 9cb4011fa2ccf85aa8fb2d49b009319f9616eea0 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 1 Dec 2019 16:55:53 +0100
Subject: [PATCH] Disable shared library on Windows

---
 CMakeLists.txt | 26 +++++++++++++++-----------
 1 file changed, 15 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6336dfa..ae92635 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -117,17 +117,21 @@ if(OPENGL_FOUND)
   add_library(lib STATIC gl2ps.c gl2ps.h)
   set_target_properties(lib PROPERTIES OUTPUT_NAME gl2ps)
 
-  add_library(shared SHARED gl2ps.c gl2ps.h)
-  target_link_libraries(shared ${EXTERNAL_LIBRARIES})
-  set_target_properties(shared PROPERTIES OUTPUT_NAME gl2ps)
-  set_target_properties(shared PROPERTIES
-    VERSION ${GL2PS_MAJOR_VERSION}.${GL2PS_MINOR_VERSION}.${GL2PS_PATCH_VERSION}
-    SOVERSION ${GL2PS_MAJOR_VERSION})
-  if(MSVC)
-    set_target_properties(shared PROPERTIES COMPILE_FLAGS "-DGL2PSDLL -DGL2PSDLL_EXPORTS")
-  endif(MSVC)
-
-  install(TARGETS lib shared DESTINATION lib${LIB_SUFFIX})
+  if(${GL2PS_OS} MATCHES "Windows")
+    install(TARGETS lib DESTINATION lib${LIB_SUFFIX})
+  else()
+    add_library(shared SHARED gl2ps.c gl2ps.h)
+    target_link_libraries(shared ${EXTERNAL_LIBRARIES})
+    set_target_properties(shared PROPERTIES OUTPUT_NAME gl2ps)
+    set_target_properties(shared PROPERTIES
+      VERSION ${GL2PS_MAJOR_VERSION}.${GL2PS_MINOR_VERSION}.${GL2PS_PATCH_VERSION}
+      SOVERSION ${GL2PS_MAJOR_VERSION})
+    if(MSVC)
+      set_target_properties(shared PROPERTIES COMPILE_FLAGS "-DGL2PSDLL -DGL2PSDLL_EXPORTS")
+    endif(MSVC)
+
+    install(TARGETS lib shared DESTINATION lib${LIB_SUFFIX})
+  endif()
 endif(OPENGL_FOUND)
 
 if(WIN32)
-- 
2.21.0

