From e7397626d03ac29b6215decae50589d9055b321d Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 1 Dec 2019 17:33:01 +0100
Subject: [PATCH] Disable static library on Windows

---
 CMakeLists.txt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6336dfa..7c74d4a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,8 +114,11 @@ endif(MATH_LIBRARY)
 include_directories(${EXTERNAL_INCLUDES})
 
 if(OPENGL_FOUND)
-  add_library(lib STATIC gl2ps.c gl2ps.h)
-  set_target_properties(lib PROPERTIES OUTPUT_NAME gl2ps)
+  if(NOT ${GL2PS_OS} MATCHES "Windows")
+    add_library(lib STATIC gl2ps.c gl2ps.h)
+    set_target_properties(lib PROPERTIES OUTPUT_NAME gl2ps)
+    install(TARGETS lib DESTINATION lib${LIB_SUFFIX})
+  endif()
 
   add_library(shared SHARED gl2ps.c gl2ps.h)
   target_link_libraries(shared ${EXTERNAL_LIBRARIES})
@@ -127,7 +130,7 @@ if(OPENGL_FOUND)
     set_target_properties(shared PROPERTIES COMPILE_FLAGS "-DGL2PSDLL -DGL2PSDLL_EXPORTS")
   endif(MSVC)
 
-  install(TARGETS lib shared DESTINATION lib${LIB_SUFFIX})
+  install(TARGETS shared DESTINATION lib${LIB_SUFFIX})
 endif(OPENGL_FOUND)
 
 if(WIN32)
-- 
2.21.0

