From 8a84c8d5848d198b0d914e7f55acf3d919ccab5f Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 1 Dec 2019 16:34:17 +0100
Subject: [PATCH] Disable static library on Windows

---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6336dfa..7204a63 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,8 +114,10 @@ endif(MATH_LIBRARY)
 include_directories(${EXTERNAL_INCLUDES})
 
 if(OPENGL_FOUND)
-  add_library(lib STATIC gl2ps.c gl2ps.h)
-  set_target_properties(lib PROPERTIES OUTPUT_NAME gl2ps)
+  if(NOT ${GL2PS_OS} MATCHES "Windows")
+    add_library(lib STATIC gl2ps.c gl2ps.h)
+    set_target_properties(lib PROPERTIES OUTPUT_NAME gl2ps)
+  endif()
 
   add_library(shared SHARED gl2ps.c gl2ps.h)
   target_link_libraries(shared ${EXTERNAL_LIBRARIES})
-- 
2.21.0

