From 3777ac46631f14f132e8f4e4172c21ff4969e1d4 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 1 Dec 2019 17:14:40 +0100
Subject: [PATCH] Remove test applications

---
 CMakeLists.txt | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7204a63..6bff1ea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,12 +82,6 @@ if(OPENGL_FOUND)
   list(APPEND EXTERNAL_LIBRARIES ${OPENGL_LIBRARIES})
 endif(OPENGL_FOUND)
 
-find_package(GLUT)
-if(GLUT_FOUND)
-  list(APPEND EXTERNAL_INCLUDES ${GLUT_INCLUDE_DIR})
-  list(APPEND EXTERNAL_LIBRARIES ${GLUT_LIBRARIES})
-endif(GLUT_FOUND)
-
 if(ENABLE_ZLIB)
   find_package(ZLIB)
   if(ZLIB_FOUND)
@@ -145,13 +139,6 @@ install(FILES ${CMAKE_SOURCE_DIR}/COPYING.GL2PS DESTINATION ${GL2PS_DOC})
 install(FILES ${CMAKE_SOURCE_DIR}/gl2psTest.c DESTINATION ${GL2PS_DOC})
 install(FILES ${CMAKE_SOURCE_DIR}/gl2psTestSimple.c DESTINATION ${GL2PS_DOC})
 
-if(GLUT_FOUND)
-  add_executable(gl2psTest WIN32 gl2psTest.c)
-  target_link_libraries(gl2psTest lib ${EXTERNAL_LIBRARIES})
-  add_executable(gl2psTestSimple WIN32 gl2psTestSimple.c)
-  target_link_libraries(gl2psTestSimple lib ${EXTERNAL_LIBRARIES})
-endif(GLUT_FOUND)
-
 find_package(LATEX)
 if(PDFLATEX_COMPILER)
   add_custom_command(OUTPUT gl2ps.pdf DEPENDS gl2ps.tex 
-- 
2.21.0

