From 6732446c6b8bdd8d631d96b4f6a7d1763e284097 Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Thu, 31 Oct 2024 14:50:32 -0500
Subject: [PATCH 06/10] BLD: Use consistent install directory for cmake configs

---
 src/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 940e2a6..83fdfe6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -113,7 +113,7 @@ endif()
 #   * <prefix>/cmake/
 #   * <prefix>/lib*/
 #   * <prefix>/include/
-set(config_install_dir "${CMAKE_INSTALL_PREFIX}/cmake")
+set(config_install_dir "${CMAKE_INSTALL_PREFIX}/lib/cmake/nvimgcodec")
 
 set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
 
@@ -165,7 +165,7 @@ install(
 #   * <prefix>/cmake/nvimgcodec/nvimgcodecConfigVersion.cmake
 install(
     FILES "${project_config}" "${version_config}"
-    DESTINATION "cmake/nvimgcodec"
+    DESTINATION "${config_install_dir}"
     COMPONENT lib
 )
 
@@ -174,7 +174,7 @@ install(
 install(
     EXPORT "${TARGETS_EXPORT_NAME}"
     NAMESPACE "${namespace}"
-    DESTINATION "cmake/nvimgcodec"
+    DESTINATION "${config_install_dir}"
     COMPONENT lib
 )
 
-- 
2.47.0

