From 1914106b92eb9c71dcc0d7b7e2e52cb4abc12cfb Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Thu, 31 Oct 2024 23:22:31 -0500
Subject: [PATCH 08/10] BLD: Decouple nvimgcodec and dynlink_* targets

---
 CMakeLists.txt     | 17 ++++-------------
 src/CMakeLists.txt |  1 -
 2 files changed, 4 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f15b7c1..dae85f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,13 +87,7 @@ option(BUILD_CVCUDA_SAMPLES "Build CVCUDA samples" OFF)
 option(BUILD_DOCS "Build documentation" OFF)
 option(BUILD_EXTENSIONS "Build extensions modules" ON)
 option(BUILD_NVJPEG_EXT "Build nvjpeg extensions module" ON)
-if(BUILD_NVJPEG_EXT AND NOT BUILD_LIBRARY)
-    message(FATAL_ERROR "BUILD_LIBRARY must be enabled if BUILD_NVJPEG_EXT is enabled. The NVJPEG extension links to libimgcodec_static.")
-endif()
 option(BUILD_NVJPEG2K_EXT "Build nvjpeg2k extensions module" ON)
-if(BUILD_NVJPEG2K_EXT AND NOT BUILD_LIBRARY)
-    message(FATAL_ERROR "BUILD_LIBRARY must be enabled if BUILD_NVJPEG2K_EXT is enabled. The NVJPEG2K extension links to libimgcodec_static.")
-endif()
 option(BUILD_NVBMP_EXT "Build nvbmp extensions module" ON)
 option(BUILD_NVPNM_EXT "Build nvpnm extensions module" ON)
 option(BUILD_LIBJPEG_TURBO_EXT "Build libjpeg-turbo extensions module" ON)
@@ -105,13 +99,6 @@ option(NVIMG_CODEC_USE_SYSTEM_DLPACK "Use system installed dlpack" OFF)
 option(NVIMG_CODEC_USE_SYSTEM_PYBIND "Use system installed pybind11" OFF)
 option(NVIMG_CODEC_COPY_LIBS_TO_PYTHON_DIR "Move nvimgcodec from install prefix to the python directory" ON)
 
-cmake_dependent_option(WITH_DYNAMIC_NVJPEG "Dynamically loads nvjpeg at runtime" ON
-                      "BUILD_NVJPEG_EXT" OFF)
-propagate_option(WITH_DYNAMIC_NVJPEG)
-cmake_dependent_option(WITH_DYNAMIC_NVJPEG2K "Dynamically loads nvjpeg2k at runtime" ON
-                      "BUILD_NVJPEG2K_EXT" OFF)
-propagate_option(WITH_DYNAMIC_NVJPEG2K)
-
 string(TOLOWER ${CMAKE_SYSTEM_NAME} SYS_NAME)
 
 # Introduce variables:
@@ -256,6 +243,10 @@ include(Dependencies)
 
 add_subdirectory(external)
 
+if(BUILD_LIBRARY OR BUILD_PYTHON OR BUILD_EXTENSIONS OR BUILD_TEST)
+    add_subdirectory(src/dynlink)
+endif()
+
 if(BUILD_LIBRARY)
     add_subdirectory(src)
 endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 83fdfe6..306ad46 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -69,7 +69,6 @@ add_library(${NVIMGCODEC_LIBRARY_NAME}_static STATIC ${NVIMGCODEC_SRCS} ${Produc
 set(LIBTOPACK ${NVIMGCODEC_LIBRARY_NAME} ${NVIMGCODEC_LIBRARY_NAME}_static)
 
 find_package(CUDAToolkit REQUIRED)
-add_subdirectory(dynlink)
 
 target_link_libraries(${NVIMGCODEC_LIBRARY_NAME} PRIVATE CUDA::cudart_static dynlink_cuda)
 
-- 
2.47.0

