From 7fd9348235602b98ca7738053df1af0a8b0e843a Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Thu, 31 Oct 2024 23:23:20 -0500
Subject: [PATCH 9/9] BLD: Link system nvimgcodec to python module

---
 python/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 39f54c7..10c2f7e 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -16,6 +16,10 @@
 include(Utils)
 include(GNUInstallDirs)
 
+if(NOT TARGET nvimgcodec)
+    find_package(nvimgcodec REQUIRED)
+endif()
+
 if(UNIX)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -fPIC -fvisibility=hidden")
 endif()
-- 
2.34.1

