--- RadTraQ-0.0.7.orig/versioneer.py	2024-01-03 12:17:33.570486650 -0600
+++ RadTraQ-0.0.7/versioneer.py	2024-01-03 12:17:44.647513217 -0600
@@ -339,7 +339,7 @@
     # configparser.NoOptionError (if it lacks "VCS="). See the docstring at
     # the top of versioneer.py for instructions on writing your setup.cfg .
     setup_cfg = os.path.join(root, "setup.cfg")
-    parser = configparser.SafeConfigParser()
+    parser = configparser.ConfigParser()
     with open(setup_cfg, "r") as f:
         parser.readfp(f)
     VCS = parser.get("versioneer", "VCS")  # mandatory
