diff --git a/src/cmake/ie_parallel.cmake b/src/cmake/ie_parallel.cmake
index 1e6b942f88..30f3ec8ec7 100644
--- a/src/cmake/ie_parallel.cmake
+++ b/src/cmake/ie_parallel.cmake
@@ -142,7 +142,7 @@ macro(ov_find_package_tbb)
                 endif()
             endforeach()
 
-            if(WIN32 AND TARGET TBB::tbbbind_2_5)
+            if(WIN32 AND TARGET TBB::tbbbind_2_5 AND NOT CPACK_GENERATOR STREQUAL "CONDA-FORGE")
                 # Add HWLOC::hwloc_2_5 target to check via Apivalidator
                 get_target_property(TBB_location TBB::tbb IMPORTED_LOCATION_RELEASE)
                 get_filename_component(TBB_dir "${TBB_location}" DIRECTORY)
