diff --git a/samples/cpp/CMakeLists.txt b/samples/cpp/CMakeLists.txt
index aa3d4bc1c8..4c9869f2fa 100644
--- a/samples/cpp/CMakeLists.txt
+++ b/samples/cpp/CMakeLists.txt
@@ -100,7 +100,12 @@ endif()
 if(NOT TARGET gflags)
     if(APPLE OR DEFINED ENV{CONDA_PREFIX} OR DEFINED ENV{HOMEBREW_PREFIX})
         # conda-forge and brew contains only shared version of gflags
-        find_package(gflags QUIET COMPONENTS nothreads_shared)
+        if(ENV{CONDA_PREFIX} AND WIN32)
+            # conda-forge windows compiled as gflags
+            find_package(gflags QUIET COMPONENTS shared)
+        else()
+            find_package(gflags QUIET COMPONENTS nothreads_shared)
+        endif()
     else()
         find_package(gflags QUIET COMPONENTS nothreads_static)
     endif()
