diff --git a/samples/cpp/CMakeLists.txt b/samples/cpp/CMakeLists.txt
index 3a10819c47..e399abdd84 100644
--- a/samples/cpp/CMakeLists.txt
+++ b/samples/cpp/CMakeLists.txt
@@ -105,10 +105,10 @@ endif()
 if(TARGET gflags)
     set(GFLAGS_TARGET gflags)
 elseif(gflags_required)
-    if(EXISTS /etc/debian_version)
-        set(gflags_component nothreads_static)
-    else()
+    if(WIN32 OR DEFINED ENV{CONDA_PREFIX})
         set(gflags_component shared)
+    else()
+        set(gflags_component nothreads_static)
     endif()
     find_package(gflags QUIET OPTIONAL_COMPONENTS ${gflags_component})
     if(gflags_FOUND)
