diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ba62e48..51a51ca 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -107,7 +107,7 @@ write_basic_package_version_file(
     COMPATIBILITY SameMajorVersion
 )
 install(FILES ${ads_HEADERS}
-    DESTINATION include
+    DESTINATION include/qtadvanceddocking
     COMPONENT headers
 )
 install(FILES
@@ -121,7 +121,7 @@ install(TARGETS qtadvanceddocking
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
-    INCLUDES DESTINATION include
+    INCLUDES DESTINATION include/qtadvanceddocking
 )
 
 install(EXPORT adsTargets
