From 9d29369a4b351a4d46eec7c45d42aed957ba1f0e Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sat, 24 Jun 2023 18:11:39 -0400
Subject: [PATCH 2/5] Update the location of SPRV headers

---
 libshaderc_util/src/compiler.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libshaderc_util/src/compiler.cc b/libshaderc_util/src/compiler.cc
index 525cbc5..f91ab09 100644
--- a/libshaderc_util/src/compiler.cc
+++ b/libshaderc_util/src/compiler.cc
@@ -20,7 +20,7 @@
 #include <thread>
 #include <tuple>
 
-#include "SPIRV/GlslangToSpv.h"
+#include "glslang/SPIRV/GlslangToSpv.h"
 #include "libshaderc_util/format.h"
 #include "libshaderc_util/io_shaderc.h"
 #include "libshaderc_util/message.h"
-- 
2.39.2

