--- a/setup.py
+++ b/setup.py
@@ -261,6 +261,10 @@
 # shared libaries since pip does not integrate well with the system package
 # management.
 for library_name in project_information.library_names:
+  # libpff itself comes from the shared library the conda package provides.
+  if library_name == project_information.library_name:
+    continue
+
   for source_file in glob.glob(os.path.join(library_name, "*.[ly]")):
     generated_source_file = "{0:s}.c".format(source_file[:-2])
     if not os.path.exists(generated_source_file):
@@ -283,7 +287,7 @@
             project_information.module_name,
             define_macros=DEFINE_MACROS,
             include_dirs=project_information.include_directories,
-            libraries=[],
+            libraries=["pff"],
             library_dirs=[],
             sources=SOURCES
         )
