--- PyAudio-0.2.11.orig/setup.py	2017-03-18 20:29:01.000000000 -0300
+++ PyAudio-0.2.11/setup.py	2017-07-15 17:04:58.411856212 -0300
@@ -79,7 +79,7 @@
 else:
     include_dirs = [os.path.join(portaudio_path, 'include/')]
     extra_link_args = [
-        os.path.join(portaudio_path, 'lib/.libs/libportaudio.a')
+        os.path.join(portaudio_path, 'lib/libportaudio.a')
         ]

     # platform specific configuration
