From 99b15b3ddede054a14e210da65ee7037bd52140d Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Tue, 1 Apr 2025 08:23:27 -0700
Subject: [PATCH] Use os specific path seperator

---
 setup/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup/setup.py b/setup/setup.py
index a7acc89..c44f1d8 100644
--- a/setup/setup.py
+++ b/setup/setup.py
@@ -32,7 +32,7 @@ def get_dirlist(_rootdir):
 # Get subfolders recursively
 os.chdir('..')
 rootdir = 'python-sdk'
-packages = [d.replace('/', '.').replace('{}.'.format(rootdir), '') for d in get_dirlist(rootdir)]
+packages = [os.path.relpath(d, rootdir).replace(os.sep, '.') for d in get_dirlist(rootdir)]
 
 # Filter out Python cache folders
 packages = [p for p in packages if not p.endswith('__pycache__')]
-- 
2.49.0.windows.1

