From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: conda-forge staged-recipes <conda-forge@users.noreply.github.com>
Date: Mon, 11 May 2026 01:55:00 -0700
Subject: [PATCH] Guard newer Metal command buffer errors

---
 filament/backend/src/metal/MetalContext.mm | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/filament/backend/src/metal/MetalContext.mm b/filament/backend/src/metal/MetalContext.mm
index a2c298d6c..ce70ca0cb 100644
--- a/filament/backend/src/metal/MetalContext.mm
+++ b/filament/backend/src/metal/MetalContext.mm
@@ -109,16 +109,20 @@ void logMTLCommandBufferError(MTLCommandBufferError error) {
     MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorInternal)
     MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorTimeout)
     MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorPageFault)
+#if defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120000
     MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorAccessRevoked)
+#endif
     MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorNotPermitted)
     MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorOutOfMemory)
     MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorInvalidResource)
 
     MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorMemoryless)
 
+#if defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120000
     if (@available(iOS 15.0, macOS 12.0, *)) {
         MTL_COMMAND_ERROR_CASE(MTLCommandBufferErrorStackOverflow)
     }
+#endif
 
     LOG(ERROR) << "Filament Metal unknown error.";
 
-- 
2.51.0
