From c318f2b8e6d327ea117baeed7455d767acb19f84 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 2 May 2023 17:09:29 -0600
Subject: [PATCH] remove wheel and versioneer from requirements

---
 requirements.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index 5ad9cd6..401cff8 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,3 @@
-wheel
 numpy
 scipy == 1.9.1
 matplotlib
