From d74d804efee0bb1ae77d3e28816fec3ea0bbaccd Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 2 May 2023 15:12:50 -0600
Subject: [PATCH] Replace zmq with pyzmq

---
 requirements.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index 5ad9cd6..d4a785d 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -7,7 +7,7 @@ pandas
 ophyd
 dask >= 2021.6.2
 distributed >= 2021.6.2
-zmq
+pyzmq
 fvgp == 3.3.7
 plotly
 notebook
