From 752676f2256101b7e0b5bcc4a1014d2afefa28c8 Mon Sep 17 00:00:00 2001
From: Andrew Annex <ama6fy@virginia.edu>
Date: Sat, 6 Jun 2020 15:44:34 -0400
Subject: [PATCH] fixing version import

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index ed76c0c7..855b1ce7 100644
--- a/setup.py
+++ b/setup.py
@@ -1,12 +1,12 @@
 from setuptools import setup, find_packages
-from gempy import __version__
+#from gempy import __version__
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
 setup(
     name='gempy',
-    version=__version__,
+    version='2.2.1',
     packages=find_packages(exclude=('test', 'docs')),
     include_package_data=True,
     install_requires=[
-- 
2.24.1

