diff --git a/configure.cmake b/configure.cmake
index 9c17711..a5c8d28 100644
--- a/qtwebengine/configure.cmake
+++ b/qtwebengine/configure.cmake
@@ -547,6 +547,7 @@ add_check_for_support(
 add_check_for_support(
    MODULES QtWebEngine QtPdf
    CONDITION
+       TRUE OR
        (LINUX AND CMAKE_CXX_COMPILER_ID STREQUAL GNU) OR
        (LINUX AND CMAKE_CXX_COMPILER_ID STREQUAL Clang) OR
        (WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL MSVC) OR
