--- setup.py	2022-01-14 18:59:30.000000000 -0500
+++ setup.py	2022-03-08 15:13:33.096795400 -0500
@@ -10,7 +10,7 @@
 here = path.abspath(path.dirname(__file__))
 
 # long description from README file
-with open(path.join(here, 'README.md'), encoding='utf-8') as f:
+with open(path.join(here, 'README.rst'), encoding='utf-8') as f:
     long_description = f.read()
 
 # version number and all other params from package.json
