diff --git a/CMakeLists.txt b/CMakeLists.txt
index f55568e..e189b2f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,7 +96,8 @@ if(ENABLE_NO_ALIGNED_ALLOC)
     add_definitions(-DNO_ALIGNED_ALLOC)
 else()
     # this is available in C11 N.B. std::aligned_alloc is available in C++17
-    CHECK_FUNCTION_EXISTS(aligned_alloc HAVE_ALIGNED_ALLOC)
+    include(CheckFunctionExists)
+    check_function_exists(aligned_alloc HAVE_ALIGNED_ALLOC)
     if(NOT HAVE_ALIGNED_ALLOC)
         add_definitions(-DNO_ALIGNED_ALLOC)
     endif()
