diff --git a/source/dftd4.f90 b/source/dftd4.f90
index e6a1dae..726c1e7 100644
--- a/source/dftd4.f90
+++ b/source/dftd4.f90
@@ -163,7 +163,7 @@ subroutine initialize_d4_model(self,atoms,mode,g_a,g_c)
    enddo
 
    ! integrate C6 coefficients
-   !$omp parallel default(none) private(i,j,ii,jj,alpha,c6) shared(self)
+   !$omp parallel default(shared) private(i,j,ii,jj,alpha,c6) shared(self)
    !$omp do schedule(runtime)
    do i = 1, max_elem
       do j = 1, i
@@ -232,7 +232,7 @@ subroutine initialize_d3_model(self,atoms)
    enddo
 
    ! integrate C6 coefficients
-   !$omp parallel default(none) private(i,j,ii,jj,alpha,c6) shared(self)
+   !$omp parallel default(shared) private(i,j,ii,jj,alpha,c6) shared(self)
    !$omp do schedule(runtime)
    do i = 1, max_elem
       do j = 1, i
@@ -1195,7 +1195,7 @@ subroutine edisp_3d(mol,dispm,ndim,q,r_thr,mbd_thr,par, &
    enddo
 
 !  print'(" * Entering first OMP section")'
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ii,iii,ia,iz,k)  &
 !$omp shared (dispm,mol,itbl,q) &
 !$omp shared (gw,zetavec,zerovec,r_thr)
@@ -1214,7 +1214,7 @@ subroutine edisp_3d(mol,dispm,ndim,q,r_thr,mbd_thr,par, &
 !$omp end do
 !$omp end parallel
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,j,ia,ja,ij,k,l,c6ii,c6ij,disp,  &
 !$omp         rij,r2,r,r4r2ij,r0,oor6,oor8,oor10, &
 !$omp         t,tx,ty,tz)  &
@@ -1329,7 +1329,7 @@ subroutine abcappr_3d(mol,dispm,ndim,par,zvec,refc6,itbl,rep,r_thr,eabc)
 
    eabc = 0.0_wp
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ia,j,ja,ij,r2ij,c6ij,k,l,r)  &
 !$omp shared (mol,dispm,itbl,refc6,zvec) &
 !$omp shared (c6ab)
@@ -1427,7 +1427,7 @@ subroutine abcappr_3d_dftd3_like_style(nat,at,xyz,par,thr,rep,dlat,c6ab,eabc)
    !        write(*,*)'thr:',sqrt(thr)
 
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(nat,xyz,c6ab,rep,par,at,dlat,thr) &
 !$omp private(iat,jat,ij,ijvec,c6ij,kat,ik,jk,ikvec,jkvec,c6ik,c6jk,c9, &
 !$omp&        cij,cik,cjk,cijk,tx,ty,tz,repmin,repmax,t,rij2,sx,sy,sz, &
@@ -1841,7 +1841,7 @@ subroutine dispgrad_3d(mol,dispm,ndim,q,cn,dcndr,dcndL,r_thr,mbd_thr,par,wf, &
    enddo
 
 !  print'(" * Entering first OMP section")'
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ii,iii,ia,iz,k,norm,dnorm,twf,tgw,dexpw,expw,gwk,dgwk)  &
 !$omp shared (mol,dispm,itbl,wf,cn,q) &
 !$omp shared (gw,dgw,zvec,dzvec,dzdq,r_thr)
@@ -1897,7 +1897,7 @@ subroutine dispgrad_3d(mol,dispm,ndim,q,cn,dcndr,dcndL,r_thr,mbd_thr,par,wf, &
 !$omp end do
 !$omp end parallel
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,j,ia,ja,ij,k,l,c6ii,c6ij,dic6ii,dic6ij,djc6ij,disp,ddisp, &
 !$omp         rij,r2,r,r4r2ij,r0,oor6,oor8,oor10,door6,door8,door10, &
 !$omp         t,tx,ty,tz,dtmp,drdx,dizii,dizij,djzij)  &
@@ -2094,7 +2094,7 @@ subroutine dabcappr_3d(mol,dispm,ndim,par,zvec,dzvec,refc6,itbl,rep,r_thr,g,sigm
 
    eabc = 0.0_wp
 
-!$OMP parallel default(none) &
+!$OMP parallel default(shared) &
 !$omp private(i,ia,j,ja,ij,r2ij,c6ij,dic6ij,djc6ij,k,l,r)  &
 !$omp shared (mol,dispm,itbl,refc6,zvec,dzvec) &
 !$omp shared (c6ab,dc6ab)
@@ -2206,7 +2206,7 @@ subroutine dabcappr_3d_dftd3_like_style(nat,at,xyz,par,thr,rep,dlat,c6ab,dc6ab,
    eabc=0.0_wp
    !        write(*,*)'thr:',sqrt(thr)
 
-!$omp parallel default(none) &
+!$omp parallel default(shared) &
 !$omp shared(nat,xyz,c6ab,rep,par,at,dlat,dc6ab,thr) &
 !$omp private(iat,jat,ij,ijvec,c6ij,kat,ik,jk,ikvec,jkvec,c6ik,c6jk,c9, &
 !$omp&        cij,cik,cjk,cijk,tx,ty,tz,repmin,repmax,t,rij2,sx,sy,sz, &
