diff --git a/setup.py b/setup.py
index 67c36a9..7b1b709 100644
--- a/setup.py
+++ b/setup.py
@@ -193,7 +193,7 @@ if 'darwin' in sys.platform and PY3:
   mac_target = sysconfig.get_config_var('MACOSX_DEPLOYMENT_TARGET')
   if mac_target and (pkg_resources.parse_version(mac_target) <
                      pkg_resources.parse_version('10.7.0')):
-    os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.7'
+    os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
     os.environ['_PYTHON_HOST_PLATFORM'] = re.sub(
         r'macosx-[0-9]+\.[0-9]+-(.+)',
         r'macosx-10.7-\1',
