diff --git a/setup.py b/setup.py
index 5eb54b4..24d1566 100644
--- a/setup.py
+++ b/setup.py
@@ -192,7 +192,10 @@ if not "win32" in sys.platform:
 if "win32" in sys.platform:
   EXTENSION_LIBRARIES += ('advapi32', 'ws2_32',)
 if BUILD_WITH_SYSTEM_OPENSSL:
-  EXTENSION_LIBRARIES += ('ssl', 'crypto',)
+  if not "win32" in sys.platform:
+    EXTENSION_LIBRARIES += ('ssl', 'crypto',)
+  else:
+    EXTENSION_LIBRARIES += ('ssleay32', 'libeay32',)
 if BUILD_WITH_SYSTEM_ZLIB:
   EXTENSION_LIBRARIES += ('z',)
 if BUILD_WITH_SYSTEM_CARES:
