diff -ur BAT-0.9.4.1-orig/models/base/Makefile.in BAT-0.9.4.1/models/base/Makefile.in
--- BAT-0.9.4.1-orig/models/base/Makefile.in	2026-02-13 04:42:14
+++ BAT-0.9.4.1/models/base/Makefile.in	2026-02-13 04:43:58
@@ -779,7 +779,7 @@
 	done; \
 	rm -rf base-tmp;
 @WITH_CLING_TRUE@libBATmodels_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_TRUE@	$(ROOTCLING) -f $@.tmp -s libBATmodels@SHLIBEXT@ -rml libBATmodels@SHLIBEXT@ -rmf libBATmodels.rootmap -c $(CPPFLAGS) $(CXXFLAGS) -I$(includedir) $+
+@WITH_CLING_TRUE@	$(ROOTCLING) -f $@.tmp -s libBATmodels@SHLIBEXT@ -rml libBATmodels@SHLIBEXT@ -rmf libBATmodels.rootmap -c $(ROOTCFLAGS) -I$(includedir) $+
 @WITH_CLING_TRUE@	@# Some magic to prefix header names with "$(PACKAGE)/", and only that, in dictionary and rootmap
 @WITH_CLING_TRUE@	@# avoid sed -i: not portable between Mac OS X and linux => use back-up files
 @WITH_CLING_TRUE@	$(GREP) -F -v '"'"`pwd`"'/",' $@.tmp | $(SED) 's|"\([^"]*/\)\?\([^/"]*[.]h\)",|"'$(PACKAGE)/'\2",| ; s|\\"\([^"]*/\)\?\([^/"]*[.]h\)\\"\\n"|\\"'$(PACKAGE)/'\2\\"\\n"|' > $@.tmp2
@@ -794,7 +794,7 @@
 ### ROOT-5: ###
 
 @WITH_CLING_FALSE@libBATmodels_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_FALSE@	$(ROOTCINT) -f $@ -c -p $(CPPFLAGS) $(CXXFLAGS) $+
+@WITH_CLING_FALSE@	$(ROOTCINT) -f $@ -c -p $(ROOTCFLAGS) $+
 
 @WITH_CLING_FALSE@.libs/libBATmodels.rootmap: libBATmodels.la | $(library_include_HEADERS) LinkDef.h
 @WITH_CLING_FALSE@	$(RLIBMAP) -o $@ -l .libs/libBATmodels@SHLIBEXT@ -c $|
diff -ur BAT-0.9.4.1-orig/models/mtf/Makefile.in BAT-0.9.4.1/models/mtf/Makefile.in
--- BAT-0.9.4.1-orig/models/mtf/Makefile.in	2026-02-13 04:42:14
+++ BAT-0.9.4.1/models/mtf/Makefile.in	2026-02-13 04:43:58
@@ -774,7 +774,7 @@
 	done; \
 	rm -rf mtf-tmp;
 @WITH_CLING_TRUE@libBATmtf_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_TRUE@	$(ROOTCLING) -f $@.tmp -s libBATmtf@SHLIBEXT@ -rml libBATmtf@SHLIBEXT@ -rmf libBATmtf.rootmap -c $(CPPFLAGS) $(CXXFLAGS) -I$(includedir) $+
+@WITH_CLING_TRUE@	$(ROOTCLING) -f $@.tmp -s libBATmtf@SHLIBEXT@ -rml libBATmtf@SHLIBEXT@ -rmf libBATmtf.rootmap -c $(ROOTCFLAGS) -I$(includedir) $+
 @WITH_CLING_TRUE@	@# Some magic to prefix header names with "$(PACKAGE)/", and only that, in dictionary and rootmap:
 @WITH_CLING_TRUE@	@# avoid sed -i: not portable between Mac OS X and linux => use back-up files
 @WITH_CLING_TRUE@	$(GREP) -F -v '"'"`pwd`"'/",' $@.tmp | $(SED) 's|"\([^"]*/\)\?\([^/"]*[.]h\)",|"'$(PACKAGE)/'\2",| ; s|\\"\([^"]*/\)\?\([^/"]*[.]h\)\\"\\n"|\\"'$(PACKAGE)/'\2\\"\\n"|' > $@.tmp2
@@ -789,7 +789,7 @@
 ### ROOT-5: ###
 
 @WITH_CLING_FALSE@libBATmtf_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_FALSE@	$(ROOTCINT) -f $@ -c -p $(CPPFLAGS) $(CXXFLAGS) $+
+@WITH_CLING_FALSE@	$(ROOTCINT) -f $@ -c -p $(ROOTCFLAGS) $+
 
 @WITH_CLING_FALSE@.libs/libBATmtf.rootmap: libBATmtf.la | $(library_include_HEADERS) LinkDef.h
 @WITH_CLING_FALSE@	$(RLIBMAP) -o $@ -l .libs/libBATmtf@SHLIBEXT@ -c $|
diff -ur BAT-0.9.4.1-orig/models/mvc/Makefile.in BAT-0.9.4.1/models/mvc/Makefile.in
--- BAT-0.9.4.1-orig/models/mvc/Makefile.in	2026-02-13 04:42:14
+++ BAT-0.9.4.1/models/mvc/Makefile.in	2026-02-13 04:43:58
@@ -769,7 +769,7 @@
 	done; \
 	rm -rf mvc-tmp;
 @WITH_CLING_TRUE@libBATmvc_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_TRUE@	$(ROOTCLING) -f $@.tmp -s libBATmvc@SHLIBEXT@ -rml libBATmvc@SHLIBEXT@ -rmf libBATmvc.rootmap -c $(CPPFLAGS) $(CXXFLAGS) -I$(includedir) $+
+@WITH_CLING_TRUE@	$(ROOTCLING) -f $@.tmp -s libBATmvc@SHLIBEXT@ -rml libBATmvc@SHLIBEXT@ -rmf libBATmvc.rootmap -c $(ROOTCFLAGS) -I$(includedir) $+
 @WITH_CLING_TRUE@	@# Some magic to prefix header names with "$(PACKAGE)/", and only that, in dictionary and rootmap:
 @WITH_CLING_TRUE@	@# avoid sed -i: not portable between Mac OS X and linux => use back-up files
 @WITH_CLING_TRUE@	$(GREP) -F -v '"'"`pwd`"'/",' $@.tmp | $(SED) 's|"\([^"]*/\)\?\([^/"]*[.]h\)",|"'$(PACKAGE)/'\2",| ; s|\\"\([^"]*/\)\?\([^/"]*[.]h\)\\"\\n"|\\"'$(PACKAGE)/'\2\\"\\n"|' > $@.tmp2
@@ -784,7 +784,7 @@
 ### ROOT-5: ###
 
 @WITH_CLING_FALSE@libBATmvc_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_FALSE@	$(ROOTCINT) -f $@ -c -p $(CPPFLAGS) $(CXXFLAGS) $+
+@WITH_CLING_FALSE@	$(ROOTCINT) -f $@ -c -p $(ROOTCFLAGS) $+
 
 @WITH_CLING_FALSE@.libs/libBATmvc.rootmap: libBATmvc.la | $(library_include_HEADERS) LinkDef.h
 @WITH_CLING_FALSE@	$(RLIBMAP) -o $@ -l .libs/libBATmvc@SHLIBEXT@ -c $|
diff -ur BAT-0.9.4.1-orig/src/Makefile.in BAT-0.9.4.1/src/Makefile.in
--- BAT-0.9.4.1-orig/src/Makefile.in	2026-02-13 04:42:14
+++ BAT-0.9.4.1/src/Makefile.in	2026-02-13 04:43:58
@@ -801,7 +801,7 @@
 	done; \
 	rm -rf src-tmp;
 @WITH_CLING_TRUE@libBAT_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_TRUE@	$(ROOTCLING) -f $@ -s libBAT@SHLIBEXT@ -rml libBAT@SHLIBEXT@ -rmf libBAT.rootmap -c $(CPPFLAGS) $(CXXFLAGS) -I$(top_srcdir) $(subst $(top_srcdir)/,,$(library_include_HEADERS)) $(srcdir)/LinkDef.h
+@WITH_CLING_TRUE@	$(ROOTCLING) -f $@ -s libBAT@SHLIBEXT@ -rml libBAT@SHLIBEXT@ -rmf libBAT.rootmap -c $(ROOTCFLAGS) -I$(top_srcdir) $(subst $(top_srcdir)/,,$(library_include_HEADERS)) $(srcdir)/LinkDef.h
 
 @WITH_CLING_TRUE@.libs/libBAT.rootmap: libBAT.la
 @WITH_CLING_TRUE@	cp libBAT.rootmap libBAT_rdict.pcm .libs/
@@ -811,7 +811,7 @@
 ### ROOT-5: ###
 
 @WITH_CLING_FALSE@libBAT_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_FALSE@	$(ROOTCINT) -f $@ -c -p $(CPPFLAGS) $(CXXFLAGS) $+
+@WITH_CLING_FALSE@	$(ROOTCINT) -f $@ -c -p $(ROOTCFLAGS) $+
 
 @WITH_CLING_FALSE@.libs/libBAT.rootmap: libBAT.la | $(library_include_HEADERS) LinkDef.h
 @WITH_CLING_FALSE@	$(RLIBMAP) -o $@ -l .libs/libBAT@SHLIBEXT@ -c $|
