diff -ur BAT-0.9.4.1-orig/src/Makefile.in BAT-0.9.4.1-fixed/src/Makefile.in
--- BAT-0.9.4.1-orig/src/Makefile.in	2015-01-19 14:30:55.000000000 +0100
+++ BAT-0.9.4.1-fixed/src/Makefile.in	2018-05-02 10:12:25.850626081 +0200
@@ -356,7 +356,7 @@
 libBAT_la_LDFLAGS = -version-info 6:0:0 -L$(ROOTLIBDIR)
 libBAT_la_LIBADD = $(ROOTLIBS) $(am__append_1)
 AM_CXXFLAGS = $(ROOTAUXCFLAGS) -Wall -fPIC -I$(top_srcdir)/BAT \
-	$(ROOTCFLAGS) $(am__append_2) $(am__append_3)
+	-I$(top_srcdir) $(ROOTCFLAGS) $(am__append_2) $(am__append_3)
 AM_LDFLAGS = -O
 AM_SOFLAGS = -shared
 @IS_DARWIN_TRUE@CREATE_SO_LINK = cd $(DESTDIR)$(libdir); rm -f libBAT.so; ln -s libBAT.dylib libBAT.so
@@ -801,12 +801,7 @@
 	done; \
 	rm -rf src-tmp;
 @WITH_CLING_TRUE@libBAT_rdict.cxx: $(library_include_HEADERS) LinkDef.h
-@WITH_CLING_TRUE@	$(ROOTCLING) -f $@.tmp -s libBAT@SHLIBEXT@ -rml libBAT@SHLIBEXT@ -rmf libBAT.rootmap -c $(CPPFLAGS) $(CXXFLAGS) -I$(includedir) $+
-@WITH_CLING_TRUE@	@# Some magic to prefix header names with "$(PACKAGE)/", and only that, in dictionary and rootmap:
-@WITH_CLING_TRUE@	@# avoid sed -i: not portable between Mac OS X and linux => use back-up files
-@WITH_CLING_TRUE@	$(GREP) -F -v '"'"`pwd`"'/",' $@.tmp | $(SED) 's|"\([^"]*/\)\?\([^/"]*[.]h\)",|"'$(PACKAGE)/'\2",| ; s|\\"\([^"]*/\)\?\([^/"]*[.]h\)\\"\\n"|\\"'$(PACKAGE)/'\2\\"\\n"|' > $@.tmp2
-@WITH_CLING_TRUE@	$(SED) 's|\$$clingAutoload\$$\([^/""]\+/\)*|$$clingAutoload$$'$(PACKAGE)'/|; /.*DICTPAYLOAD(.*/,/.*)DICTPAYLOAD.*/ s|#include "\([^/"]\+/\)*\(.*\)"|#include <'$(PACKAGE)'/\2>|' $@.tmp2 > $@ && $(RM) $@.tmp $@.tmp2
-@WITH_CLING_TRUE@	$(SED) 's|\(header \+\)\([^ ].*/\)\?\([^ ].*[.]h\)|\1'$(PACKAGE)/'\3|' $(rtmap) > $(rtmap).tmp && mv $(rtmap).tmp $(rtmap)
+@WITH_CLING_TRUE@	$(ROOTCLING) -f $@ -s libBAT@SHLIBEXT@ -rml libBAT@SHLIBEXT@ -rmf libBAT.rootmap -c $(CPPFLAGS) $(CXXFLAGS) -I$(top_srcdir) $(subst $(top_srcdir)/,,$(library_include_HEADERS)) $(srcdir)/LinkDef.h
 
 @WITH_CLING_TRUE@.libs/libBAT.rootmap: libBAT.la
 @WITH_CLING_TRUE@	cp libBAT.rootmap libBAT_rdict.pcm .libs/
