diff -ur BAT-0.9.4.1-orig/models/mvc/BCMVCombination.cxx BAT-0.9.4.1-patched/models/mvc/BCMVCombination.cxx
--- BAT-0.9.4.1-orig/models/mvc/BCMVCombination.cxx	2014-11-20 12:24:57.000000000 +0100
+++ BAT-0.9.4.1-patched/models/mvc/BCMVCombination.cxx	2021-05-10 13:34:53.991100033 +0200
@@ -18,6 +18,8 @@
 #include "../../BAT/BCMath.h"
 #include "../../BAT/BCParameter.h"
 
+#include <TString.h>
+
 #include <fstream>
 #include <iomanip>
 #include <iostream>
