diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 67881875..aa7726df 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -9,9 +9,6 @@
 if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../lib/UserInterface/textinput
    OR CLING_INCLUDE_TESTS)
   add_subdirectory(driver)
-  add_subdirectory(Jupyter)
-  add_subdirectory(libcling)
-  add_subdirectory(demo)
 endif()
 
 add_subdirectory(plugins)
