--- bin/nvcc.profile.bak	2023-05-12 17:14:18.957422747 -0500
+++ bin/nvcc.profile	2023-05-12 18:08:02.746114493 -0500
@@ -1,14 +1,14 @@
 
-TOP              = $(_HERE_)/..
+TOP              = $(_HERE_)/../$(_TARGET_DIR_)
 
 NVVMIR_LIBRARY_DIR = $(TOP)/$(_NVVM_BRANCH_)/libdevice
 
 LD_LIBRARY_PATH += $(TOP)/lib:
-PATH            += $(TOP)/$(_NVVM_BRANCH_)/bin:$(_HERE_):
+PATH            += $(TOP)/bin:$(TOP)/$(_NVVM_BRANCH_)/bin:$(TOP)/../../bin:$(TOP)/../../$(_NVVM_BRANCH_)/bin:
 
-INCLUDES        +=  "-I$(TOP)/$(_TARGET_DIR_)/include" $(_SPACE_)
+INCLUDES        +=  "-I$(TOP)/include" $(_SPACE_)
 
-LIBRARIES        =+ $(_SPACE_) "-L$(TOP)/$(_TARGET_DIR_)/lib$(_TARGET_SIZE_)/stubs" "-L$(TOP)/$(_TARGET_DIR_)/lib$(_TARGET_SIZE_)"
+LIBRARIES        =+ $(_SPACE_) "-L$(TOP)/lib/stubs" "-L$(TOP)/lib"
 
 CUDAFE_FLAGS    +=
 PTXAS_FLAGS     +=
