diff --git a/Sources/Infomap-0.15.7/Makefile b/Sources/Infomap-0.15.7/Makefile
index ee00447..9922f63 100644
--- a/Sources/Infomap-0.15.7/Makefile
+++ b/Sources/Infomap-0.15.7/Makefile
@@ -1,8 +1,7 @@
 #Generated Makefile
-CXX  = g++
-CXXFLAGS = -Wall -O4 -pipe -fopenmp
+CXXFLAGS = -std=c++03 -Wall -O4 -pipe -fopenmp -I$(CONDA_PREFIX)/include
 LINKER = $(CXX)
-LFLAGS = -fopenmp
+LFLAGS = -fopenmp -L$(CONDA_PREFIX)/lib
 
 HEADERS = \
 src/infomap/Edge.h \
@@ -61,7 +60,7 @@ src/io/version.cpp \
 src/utils/FileURI.cpp \
 src/utils/Logger.cpp \
 
-TARGET = Infomap
+TARGET = $(PREFIX)/bin/Infomap
 
 OBJECTS = $(SOURCES:src/%.cpp=build/%.o)
 
