diff --git a/Sources/Infomap-0.15.7/Makefile b/Sources/Infomap-0.15.7/Makefile
index ee00447..859e824 100644
--- a/Sources/Infomap-0.15.7/Makefile
+++ b/Sources/Infomap-0.15.7/Makefile
@@ -1,6 +1,5 @@
 #Generated Makefile
-CXX  = g++
-CXXFLAGS = -Wall -O4 -pipe -fopenmp
+CXXFLAGS = -std=c++03 -Wall -O4 -pipe -fopenmp -I$(CONDA_PREFIX)/include
 LINKER = $(CXX)
 LFLAGS = -fopenmp
 
@@ -61,7 +60,7 @@ src/io/version.cpp \
 src/utils/FileURI.cpp \
 src/utils/Logger.cpp \
 
-TARGET = Infomap
+TARGET = $(PREFIX)/bin/Infomap
 
 OBJECTS = $(SOURCES:src/%.cpp=build/%.o)
 
