diff --git a/Sources/TopicMapping/docmap.cpp b/Sources/TopicMapping/docmap.cpp
index 6e8d6dd..e8f2e99 100644
--- a/Sources/TopicMapping/docmap.cpp
+++ b/Sources/TopicMapping/docmap.cpp
@@ -3,7 +3,7 @@
 
 // global variables (really minor things)
 bool global_verbose=false;
-const string INFOMAP_PATH="___INSERT_INFOMAP_PATH_HERE___";
+const string INFOMAP_PATH="Infomap";
 
 #include "../SingleSliceInfomap/infomap_single_function.cpp"
 #include "word_corpus.h"
