diff --git a/src/multimedia/platform/darwin/qdarwinintegration.mm b/src/multimedia/platform/darwin/qdarwinintegration.mm
index a1d1cfe..f0ba102 100644
--- a/src/multimedia/platform/darwin/qdarwinintegration.mm
+++ b/src/multimedia/platform/darwin/qdarwinintegration.mm
@@ -54,7 +54,7 @@ QT_BEGIN_NAMESPACE
 
 QDarwinIntegration::QDarwinIntegration()
 {
-#ifdef Q_OS_MACOS
+#if 0
     if (__builtin_available(macOS 11.0, *))
         VTRegisterSupplementalVideoDecoderIfAvailable(kCMVideoCodecType_VP9);
 #endif
