diff --git a/clayer/pysertype.c b/clayer/pysertype.c
index 6d04560..6a8ac2b 100644
--- a/clayer/pysertype.c
+++ b/clayer/pysertype.c
@@ -606,7 +606,14 @@ static void sertype_free(struct ddsi_sertype* tpcmn)
 #endif
 
     // dds_free the python type if python isn't already shutting down (deadlock).
-#if PY_MINOR_VERSION > 6
+#if PY_MINOR_VERSION >= 13
+  if (!Py_IsFinalizing ())
+  {
+    PyGILState_STATE state = PyGILState_Ensure ();
+    Py_DECREF (this->my_py_type);
+    PyGILState_Release (state);
+  }
+#elif PY_MINOR_VERSION > 6
     if (!_Py_IsFinalizing()) {
         PyGILState_STATE state = PyGILState_Ensure();
         Py_DECREF(this->my_py_type);
