From 48355ed9c72dae11aaaacb31bcd2605703dc4ed3 Mon Sep 17 00:00:00 2001
From: martinRenou <martin.renou@gmail.com>
Date: Tue, 26 Jan 2021 18:15:16 +0100
Subject: [PATCH] Fix missing import in setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 7fc1e66..7728b38 100644
--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ from setuptools import find_packages
 from jupyter_packaging import (
     create_cmdclass, install_npm, ensure_targets,
     combine_commands, ensure_python,
-    get_version
+    get_version, skip_if_exists
 )
 
 from setuptools import setup
-- 
2.29.2

