From 7078100416d6f3d55f5c4c1e6935c4f64af2a580 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Fri, 7 Feb 2020 14:11:11 +0100
Subject: [PATCH 2/3] Fix configuration for conda-forge

---
 mlocal/checks/project-post.chk | 38 +++++++++++++++++-----------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/mlocal/checks/project-post.chk b/mlocal/checks/project-post.chk
index b164e152d..5ab22f02f 100644
--- a/mlocal/checks/project-post.chk
+++ b/mlocal/checks/project-post.chk
@@ -17,26 +17,26 @@ config_add_def BUILDDIR \"$builddir\"
 config_add_def SOURCEDIR \"$sourcedir\"
 config_add_def PREFIX \"$prefix\"
 config_add_def EXECPREFIX \"$exec_prefix\"
-config_add_def BINDIR \"$bindir\"
-config_add_def SBINDIR \"$sbindir\"
-config_add_def LIBEXECDIR \"$libexecdir\"
-config_add_def DATAROOTDIR \"$datarootdir\"
-config_add_def DATADIR \"$datadir\"
-config_add_def SYSCONFDIR \"$sysconfdir\"
-config_add_def SHAREDSTATEDIR \"$sharedstatedir\"
-config_add_def LOCALSTATEDIR \"$localstatedir\"
-config_add_def RUNSTATEDIR \"$runstatedir\"
-config_add_def INCLUDEDIR \"$includedir\"
+config_add_def BINDIR EXECPREFIX \"/bin\"
+config_add_def SBINDIR EXECPREFIX \"/sbin\"
+config_add_def LIBEXECDIR EXECPREFIX \"/libexec\"
+config_add_def DATAROOTDIR PREFIX \"/share\"
+config_add_def DATADIR DATAROOTDIR
+config_add_def SYSCONFDIR PREFIX \"/etc\"
+config_add_def SHAREDSTATEDIR PREFIX \"/com\"
+config_add_def LOCALSTATEDIR PREFIX \"/var\"
+config_add_def RUNSTATEDIR LOCALSTATEDIR \"/run\"
+config_add_def INCLUDEDIR PREFIX \"/include\"
 config_add_def OLDINCLUDEDIR \"$oldincludedir\"
-config_add_def DOCDIR  \"$docdir\"
-config_add_def INFODIR \"$infodir\"
-config_add_def HTMLDIR \"$htmldir\"
-config_add_def DVIDIR \"$dvidir\"
-config_add_def PDFDIR \"$pdfdir\"
-config_add_def PSDIR \"$psdir\"
-config_add_def LIBDIR \"$libdir\"
-config_add_def LOCALEDIR \"$localedir\"
-config_add_def MANDIR \"$mandir\"
+config_add_def DOCDIR  DATADIR \"/doc/singularity\"
+config_add_def INFODIR DATADIR \"/info\"
+config_add_def HTMLDIR DOCDIR
+config_add_def DVIDIR DOCDIR
+config_add_def PDFDIR DOCDIR
+config_add_def PSDIR DOCDIR
+config_add_def LIBDIR PREFIX \"/lib\"
+config_add_def LOCALEDIR DATAROOTDIR \"/locale\"
+config_add_def MANDIR DATAROOTDIR \"/man\"
 config_add_def SINGULARITY_CONFDIR SYSCONFDIR \"/singularity\"
 config_add_def SINGULARITY_CONF_FILE SINGULARITY_CONFDIR \"/singularity.conf\"
 config_add_def CAPABILITY_FILE SINGULARITY_CONFDIR \"/capability.json\"
-- 
2.27.0

