From 8cc3543a6890a20b95edefd88177aff93654cfae Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Wed, 12 Oct 2022 21:53:40 +0200
Subject: [PATCH 2/3] Remove singularity

---
 mlocal/frags/build_cli.mk | 18 +++---------------
 1 file changed, 3 insertions(+), 15 deletions(-)

diff --git a/mlocal/frags/build_cli.mk b/mlocal/frags/build_cli.mk
index e2e348e11..7b298e157 100644
--- a/mlocal/frags/build_cli.mk
+++ b/mlocal/frags/build_cli.mk
@@ -31,20 +31,14 @@ $(apptainer_INSTALL): $(apptainer)
 	$(V)umask 0022 && mkdir -p $(@D)
 	$(V)install -m 0755 $(apptainer) $(apptainer_INSTALL) # set cp to install
 
-singularity_INSTALL := $(DESTDIR)$(BINDIR)/singularity
-$(singularity_INSTALL):
-	@echo " INSTALL" $@
-	$(V)ln -sf apptainer $(singularity_INSTALL)
-
 CLEANFILES += $(apptainer)
-INSTALLFILES += $(apptainer_INSTALL) $(singularity_INSTALL)
+INSTALLFILES += $(apptainer_INSTALL)
 ALL += $(apptainer)
 
 
 # bash_completion files
 bash_completion1 :=  $(BUILDDIR)/etc/bash_completion.d/apptainer
-bash_completion2 :=  $(BUILDDIR)/etc/bash_completion.d/singularity
-bash_completions := $(bash_completion1) $(bash_completion2)
+bash_completions := $(bash_completion1)
 $(bash_completions): $(apptainer_build_config)
 	@echo " GEN" $@
 	$(V)rm -f $@
@@ -53,8 +47,7 @@ $(bash_completions): $(apptainer_build_config)
 		$(SOURCEDIR)/cmd/bash_completion/bash_completion.go $@
 
 bash_completion1_INSTALL := $(DESTDIR)$(DATADIR)/bash-completion/completions/apptainer
-bash_completion2_INSTALL := $(DESTDIR)$(DATADIR)/bash-completion/completions/singularity
-bash_completions_INSTALL := $(bash_completion1_INSTALL) $(bash_completion2_INSTALL)
+bash_completions_INSTALL := $(bash_completion1_INSTALL)
 $(bash_completion1_INSTALL): $(bash_completion1)
 	@echo " INSTALL" $@
 	$(V)umask 0022 && mkdir -p $(@D)
@@ -106,11 +99,6 @@ $(man_pages): apptainer
 	mkdir -p $@
 	$(V)$(GO) run $(GO_MODFLAGS) -tags "$(GO_TAGS)" \
 		$(SOURCEDIR)/cmd/docs/docs.go man --dir $@
-	$(V)cd $@;							\
-		for M in apptainer*; do					\
-			S="`echo $$M|sed 's/apptainer/singularity/'`";	\
-			ln -fs $$M $$S;					\
-		done
 
 man_pages_INSTALL := $(DESTDIR)$(MANDIR)/man1
 $(man_pages_INSTALL): $(man_pages)
-- 
2.37.3

