diff --git a/pkg/image/unpacker/squashfs.go b/pkg/image/unpacker/squashfs.go
index 25ef856f8..1acac6fcf 100644
--- a/pkg/image/unpacker/squashfs.go
+++ b/pkg/image/unpacker/squashfs.go
@@ -62,7 +62,10 @@ type Squashfs struct {
 // NewSquashfs initializes and returns a Squahfs unpacker instance
 func NewSquashfs() *Squashfs {
 	s := &Squashfs{}
+	defaultPath := os.Getenv("PATH")
+	os.Setenv("PATH", os.Getenv("USER_PATH"))
 	s.UnsquashfsPath, _ = exec.LookPath("unsquashfs")
+	os.Setenv("PATH", defaultPath)
 	return s
 }
 
