--- xorg-cf-files-1.0.6/linux.cf.orig	2016-01-16 15:23:08.121009598 +0000
+++ xorg-cf-files-1.0.6/linux.cf	2016-01-16 15:23:50.711116569 +0000
@@ -578,11 +578,7 @@
 #define MkdirHierCmd		mkdir -p
 
 #ifndef HaveLib64
-# if defined (AMD64Architecture) || defined (s390xArchitecture) || defined (Ppc64Architecture) || defined (AArch64Architecture)
-#  define HaveLib64	YES
-# else
-#  define HaveLib64	NO
-# endif
+# define HaveLib64	NO
 #endif
 
 #if UseElfFormat
@@ -1072,7 +1068,7 @@
 
 #if HaveLib64
 # ifndef LibDirName
-#  define LibDirName		lib64
+#  define LibDirName		lib
 # endif
 # ifndef SystemUsrLibDir
 #  define SystemUsrLibDir	/usr/lib64
