From 527139c0aa29e4c97fd8fd39c906e940e4d08cd6 Mon Sep 17 00:00:00 2001
From: Gustavo Sampaio <gbritosampaio@gmail.com>
Date: Mon, 2 Jul 2018 21:02:39 -0300
Subject: [PATCH] No git deps

---
 configure.ac | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6f19a2f0..5213862e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@
 
 # Get the hash of the last commit, to be used if it is not an
 # official release.
-AC_DEFUN([MSC_GIT_HASH], m4_esyscmd_s(git log -1 --format="%h" --abbrev-commit))
+AC_DEFUN([MSC_GIT_HASH], 8d0f51be)
 AC_DEFUN([MSC_MAJOR], m4_esyscmd_s(cat headers/modsecurity/modsecurity.h | grep "define MODSECURITY_MAJOR " | awk {'print $3'} | sed 's/\"//g'))
 AC_DEFUN([MSC_MINOR], m4_esyscmd_s(cat headers/modsecurity/modsecurity.h | grep "define MODSECURITY_MINOR " | awk {'print $3'} | sed 's/\"//g'))
 AC_DEFUN([MSC_PATCHLEVEL], m4_esyscmd_s(cat headers/modsecurity/modsecurity.h | grep "define MODSECURITY_PATCHLEVEL " | awk {'print $3'} | sed 's/\"//g'))
@@ -22,8 +22,7 @@ m4_define([msc_version_c_plus_a], [m4_eval(msc_version_major + msc_version_minor
 m4_define([msc_version],
     [msc_version_major.msc_version_minor])
 
-m4_define([msc_version_git],
-    [m4_esyscmd_s(git describe)])
+m4_define([msc_version_git], v3.0.2)
 
 m4_define([msc_version_info],
     [msc_version_c_plus_a:msc_version_patchlevel:msc_version_minor])
@@ -69,10 +68,10 @@ test "x$HAS_LIBINJECTION" = "x" && AC_MSG_ERROR([\
 
    ])
 # Libinjection version
-AC_DEFUN([LIBINJECTION_VERSION], m4_esyscmd_s(PWD="others/libinjection" git describe))
+AC_DEFUN([LIBINJECTION_VERSION], v3.9.2-30-gbf234eb)
 
 # SecLang test version
-AC_DEFUN([SECLANG_TEST_VERSION], m4_esyscmd_s(PWD="test/test-cases/secrules-language-tests" git log -1 --format="%h" --abbrev-commit))
+AC_DEFUN([SECLANG_TEST_VERSION], e6b03e4)
 
 
 # Check for yajl
-- 
2.17.1

