--- a/Signal/General/Makefile.in
+++ b/Signal/General/Makefile.in
@@ -102,7 +102,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-bin_PROGRAMS = dmsmear$(EXEEXT) digitxt$(EXEEXT) digimon$(EXEEXT) \
+bin_PROGRAMS = digitxt$(EXEEXT) digimon$(EXEEXT) \
 	digihist$(EXEEXT) filterbank_speed$(EXEEXT) $(am__EXEEXT_1) \
 	$(am__EXEEXT_2) $(am__EXEEXT_3) $(am__EXEEXT_4) \
 	$(am__EXEEXT_5) $(am__EXEEXT_6)
