diff --git a/Signal/General/Makefile.am b/Signal/General/Makefile.am
index 67e7d823..eb49d2d4 100644
--- a/Signal/General/Makefile.am
+++ b/Signal/General/Makefile.am
@@ -53,7 +53,7 @@ libdspdsp_la_SOURCES = optimize_fft.c cross_detect.c cross_detect.h  \
 	InverseFilterbankResponse.C
 
 
-bin_PROGRAMS = dmsmear digitxt digimon digihist filterbank_speed
+bin_PROGRAMS = digitxt digimon digihist filterbank_speed
 # bin_PROGRAMS += test_InverseFilterbankResponse test_FIRFilter
 
 if HAVE_CUFFT
