diff --git a/Signal/Pulsar/dspsr.C b/Signal/Pulsar/dspsr.C
index f38c25a8..023b2f1e 100644
--- a/Signal/Pulsar/dspsr.C
+++ b/Signal/Pulsar/dspsr.C
@@ -733,9 +733,14 @@ void parse_options (int argc, char** argv) try
       string ascii_predictor = lines.str();
       char* line_buffer = (char*) (ascii_predictor.c_str());
 
-      FILE* virtual_ptr = fmemopen( line_buffer, strlen(line_buffer) ,"r" );
+      FILE* virtual_ptr = tmpfile();
       if (virtual_ptr)
+      {
+        fputs( line_buffer, virtual_ptr );
+        rewind( virtual_ptr );
         config->predictors.push_back ( factory<Pulsar::Predictor> ( virtual_ptr ));
+        fclose( virtual_ptr );
+      }
     }
   }
 
