--- a/python/Makefile.in
+++ b/python/Makefile.in
@@ -450,7 +450,7 @@
 		   $(top_builddir)/Signal/General/libdspdsp.la \
 		   $(top_builddir)/Signal/Statistics/libdspstats.la \
 		   $(top_builddir)/Kernel/libdspbase.la \
-		   @CUFFT_LIBS@ @CUDA_LIBS@
+		   @CUFFT_LIBS@ @CUDA_LIBS@ @PYTHON_LDFLAGS@
 
 SWIG_FLAGS = -Wall -python -I$(top_builddir)/local_include @PSRCHIVE_INCLUDE@
 CLEANFILES = dspsr.py dspsr.pyc dspsr_wrap.cxx
