diff --git a/python/Makefile.am b/python/Makefile.am
index beae614e..c901751b 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -14,7 +14,7 @@ _dspsr_la_LIBADD = \
 		   $(top_builddir)/Signal/General/libdspdsp.la \
 		   $(top_builddir)/Signal/Statistics/libdspstats.la \
 		   $(top_builddir)/Kernel/libdspbase.la \
-		   @CUFFT_LIBS@ @CUDA_LIBS@
+		   @CUFFT_LIBS@ @CUDA_LIBS@ @PYTHON_LDFLAGS@
 
 SWIG_FLAGS = -Wall -python -I$(top_builddir)/local_include @PSRCHIVE_INCLUDE@
 
