diff --git a/Management/dspsr_cflags.in b/Management/dspsr_cflags.in
--- a/Management/dspsr_cflags.in
+++ b/Management/dspsr_cflags.in
@@ -1,4 +1,4 @@
-#! /bin/csh -f
+#! /bin/sh
 
 # @configure_input@
 
diff --git a/Management/dspsr_ldflags.in b/Management/dspsr_ldflags.in
--- a/Management/dspsr_ldflags.in
+++ b/Management/dspsr_ldflags.in
@@ -1,4 +1,4 @@
-#! /bin/csh -f
+#! /bin/sh
 
 # @configure_input@
 
