--- a/Signal/Statistics/Makefile.in
+++ b/Signal/Statistics/Makefile.in
@@ -96,7 +96,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-bin_PROGRAMS = test_Romberg$(EXEEXT) sklimit$(EXEEXT) $(am__EXEEXT_1)
+bin_PROGRAMS = sklimit$(EXEEXT) $(am__EXEEXT_1)
 @HAVE_PGPLOT_TRUE@am__append_1 = sk_distrib_plot sk_limit_plot
 subdir = Signal/Statistics
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
