From 04aade130e9280b867f56e8f30d1990c0432ed4a Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 31 Mar 2026 22:31:07 -0700
Subject: [PATCH] install with python sitearch

---
 CMakeLists.txt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5251869..264a09d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,9 @@ set(root_components Core;Net;Hist;Graf;Graf3d;Gpad;Tree;Rint;
         Postscript;Matrix;Physics;Gui;RooFitCore;RooFit;RooStats;
         HistFactory;XMLParser;Foam;Minuit)

+include(GNUInstallDirs)
+find_package(Python3 COMPONENTS Interpreter REQUIRED)
+
 #Require root
 find_package(ROOT 6 REQUIRED ${root_components})
 list(TRANSFORM root_components PREPEND ROOT::)
@@ -60,8 +63,8 @@ install(DIRECTORY ./macros DESTINATION "${CMAKE_INSTALL_PREFIX}/share/histfitter
 install(DIRECTORY ./test DESTINATION "${CMAKE_INSTALL_PREFIX}/share/histfitter")

 #Copy python files to install location
-install(DIRECTORY ./python/ DESTINATION "${CMAKE_INSTALL_PREFIX}/python/histfitter")
-install(DIRECTORY ./scripts/ DESTINATION "${CMAKE_INSTALL_PREFIX}/bin/histfitter"
+install(DIRECTORY ./python/ DESTINATION "${Python3_SITEARCH}/HistFitter")
+install(DIRECTORY ./scripts/ DESTINATION "${CMAKE_INSTALL_PREFIX}/bin/histfitter"
             USE_SOURCE_PERMISSIONS)

 #Copy C++ headers before generating root dictionary
--
2.45.2

