From 4592321e6c1dec862af7df7ff663ae2b8eedbada Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 31 Mar 2026 22:58:08 -0700
Subject: [PATCH 4/4] clean up installs

---
 CMakeLists.txt | 47 +++++++++++++++++++++++++++++++++++++++--------
 1 file changed, 39 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 264a09d..f98e227 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,6 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
   set(CMAKE_INSTALL_PREFIX "${PROJECT_SOURCE_DIR}/install" CACHE PATH "..." FORCE)
 endif()
-set(CMAKE_INSTALL_LIBDIR "${CMAKE_INSTALL_PREFIX}/lib")
 
 #Define equired root components
 set(root_components Core;Net;Hist;Graf;Graf3d;Gpad;Tree;Rint;
@@ -55,20 +54,52 @@ install(PROGRAMS setup_histfitter.sh DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
 install(FILES ./histfitter_env_setup.sh DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
 
 #Copy files needed for working directories
-install(FILES ./config/HistFactorySchema.dtd DESTINATION "${CMAKE_INSTALL_PREFIX}/share/histfitter/config")
-install(DIRECTORY ./analysis DESTINATION "${CMAKE_INSTALL_PREFIX}/share/histfitter")
-install(DIRECTORY ./macros DESTINATION "${CMAKE_INSTALL_PREFIX}/share/histfitter")
+install(FILES ./config/HistFactorySchema.dtd DESTINATION "${CMAKE_INSTALL_DATADIR}/histfitter/config")
+install(DIRECTORY ./analysis DESTINATION ${CMAKE_INSTALL_DATADIR}/histfitter
+    PATTERN "__pycache__" EXCLUDE
+    PATTERN "*.pyc" EXCLUDE
+)
+install(DIRECTORY ./macros DESTINATION ${CMAKE_INSTALL_DATADIR}/histfitter
+    PATTERN "__pycache__" EXCLUDE
+    PATTERN "*.pyc" EXCLUDE
+)
 
 #Copy test files for testing build
-install(DIRECTORY ./test DESTINATION "${CMAKE_INSTALL_PREFIX}/share/histfitter")
+# install(DIRECTORY ./test DESTINATION "${CMAKE_INSTALL_PREFIX}/share/histfitter")
 
 #Copy python files to install location
 install(DIRECTORY ./python/ DESTINATION "${Python3_SITEARCH}/HistFitter")
-install(DIRECTORY ./scripts/ DESTINATION "${CMAKE_INSTALL_PREFIX}/bin/histfitter"
-            USE_SOURCE_PERMISSIONS)
+install(PROGRAMS scripts/HistFitter.py
+        DESTINATION ${CMAKE_INSTALL_BINDIR}
+        RENAME histfitter)
+install(PROGRAMS
+  scripts/GenerateJSONOutput.py
+  scripts/GenerateTreeDescriptionFromJSON.py
+  scripts/HypoTest.py
+  scripts/PrintFitResult.py
+  scripts/PrintFitResultTex.py
+  scripts/SysTable.py
+  scripts/SysTableTex.py
+  scripts/SystRankingPlot.py
+  scripts/SystRankingPlotMP.py
+  scripts/UpperLimitTable.py
+  scripts/UpperLimitTableTex.py
+  scripts/YieldsTable.py
+  scripts/YieldsTableTex.py
+  scripts/contourPlotter.py
+  scripts/getFinalFullJER.py
+  scripts/harvestHepdataToContours.py
+  scripts/harvestToContours.py
+  scripts/mergeSysTableBreakdown.py
+  scripts/multiplexContours.py
+  scripts/multiplexJSON.py
+  scripts/plotUpDown.py
+  scripts/pull_maker.py
+  DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
 
 #Copy C++ headers before generating root dictionary
-install(FILES ${HistFitter_headers} DESTINATION "${CMAKE_INSTALL_PREFIX}/include/histfitter")
+install(FILES ${HistFitter_headers} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/histfitter")
 
 #Add libraries
 add_library(HistFitter SHARED ${HistFitter_sources})
-- 
2.45.2

