From 68b71c90668e8f3e5b047e1175b98fadba365678 Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 31 Mar 2026 23:04:12 -0700
Subject: [PATCH 5/5] add init.py

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f98e227..abe658f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,6 +69,8 @@ install(DIRECTORY ./macros DESTINATION ${CMAKE_INSTALL_DATADIR}/histfitter
 
 #Copy python files to install location
 install(DIRECTORY ./python/ DESTINATION "${Python3_SITEARCH}/HistFitter")
+file(WRITE "${CMAKE_BINARY_DIR}/HistFitter/__init__.py" "")
+install(FILES "${CMAKE_BINARY_DIR}/HistFitter/__init__.py" DESTINATION "${Python3_SITEARCH}/HistFitter")
 install(PROGRAMS scripts/HistFitter.py
         DESTINATION ${CMAKE_INSTALL_BINDIR}
         RENAME histfitter)
-- 
2.45.2

