From 82623683ac6959a6489ed2a81fd870bd618b70cf Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 31 Mar 2026 23:23:32 -0700
Subject: [PATCH 6/6] remove extraneous caches

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index abe658f..7e3800f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -157,3 +157,10 @@ if(MSVC)
 else()
     target_compile_options(HistFitter PRIVATE -Wall -Wextra -Wpedantic)
 endif()
+
+install(CODE "execute_process(COMMAND rm -f
+    \"\${CMAKE_INSTALL_PREFIX}/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache\"
+    \"\${CMAKE_INSTALL_PREFIX}/lib/graphviz/config8\"
+    \"\${CMAKE_INSTALL_PREFIX}/lib/gtk-3.0/3.0.0/immodules.cache\"
+    \"\${CMAKE_INSTALL_PREFIX}/share/glib-2.0/schemas/gschemas.compiled\"
+    \"\${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/icon-theme.cache\")")
-- 
2.45.2

