--- a/src/HypoTestTool.h
+++ b/src/HypoTestTool.h
@@ -113,10 +113,8 @@
         bool mOptimize;
         bool mUseVectorStore;
         bool mGenerateBinned;
-        bool mUseProof;
         bool mEnableDetailedOutput;
         bool mRebuild;
-        int     mNWorkers;
         int     mNToyToRebuild;
         int     mPrintLevel;
         int     mInitialFit;
--- a/src/HypoTestTool.cxx
+++ b/src/HypoTestTool.cxx
@@ -16,7 +16,7 @@
  * Copyright (c):                                                                 *
  *      CERN, Switzerland                                                         *
  *                                                                                *
- * http://root.cern.ch/root/html534/tutorials/roostats/StandardHypoTestInvDemo.C.html
+ * http://root.cern.ch/root/html534/tutorials/roostats/StandardHypoTestInvDemo.C.html
  *                                                                                *
  * (http://root.cern.ch/drupal/content/license)                                   *
  **********************************************************************************/
@@ -68,26 +68,23 @@

 /* Options:
  *  bool plotHypoTestResult = true;          // plot test statistic result at each point
- *  bool writeResult = true;                 // write HypoTestInverterResult in a file
+ *  bool writeResult = true;                 // write HypoTestInverterResult in a file
  *  TString resultFileName;                  // file with results (by default is built automatically using teh ws input file name)
- *  bool optimize = true;                    // optmize evaluation of test statistic
- *  bool useVectorStore = true;              // convert data to use new roofit data store
- *  bool generateBinned = false;             // generate binned data sets
+ *  bool optimize = true;                    // optmize evaluation of test statistic
+ *  bool useVectorStore = true;              // convert data to use new roofit data store
+ *  bool generateBinned = false;             // generate binned data sets
  *  bool noSystematics = false;              // force all systematics to be off (i.e. set all nuisance parameters as constat to their nominal values)
  *  double nToysRatio = 2;                   // ratio Ntoys S+b/ntoysB
- *  double maxPOI = -1;                      // max value used of POI (in case of auto scan)
- *  bool useProof = false;                    // use Proof Light when using toys (for freq or hybrid)
+ *  double maxPOI = -1;                      // max value used of POI (in case of auto scan)
  *  bool enableDetailedOutput = false;       // enable detailed output with all fit information for each toys (output will be written in result file
- *  int nworkers = 4;                        // number of worker for Proof
  *  bool rebuild = false;                    // re-do extra toys for computing expected limits and rebuild test stat
  *  // distributions (N.B this requires much more CPU (factor is equivalent to nToyToRebuild)
- *  int nToyToRebuild = 100;                 // number of toys used to rebuild
- *  int initialFit = -1;                     // do a first  fit to the model (-1 : default, 0 skip fit, 1 do always fit)
+ *  int nToyToRebuild = 100;                 // number of toys used to rebuild
+ *  int initialFit = -1;                     // do a first  fit to the model (-1 : default, 0 skip fit, 1 do always fit)
  *  int randomSeed = -1;                     // random seed (if = -1: use default value, if = 0 always random )
- *  // NOTE: Proof uses automatically a random seed
- *  std::string massValue = "";              // extra string to tag output file of result
+ *  std::string massValue = "";              // extra string to tag output file of result
  *  std::string  minimizerType = "";         // minimizer type (default is what is in ROOT::Math::MinimizerOptions::DefaultMinimizerType()
- *  int   printLevel = 0;                    // print level for debugging PL test statistics and calculators
+ *  int   printLevel = 0;                    // print level for debugging PL test statistics and calculators
  */

 //_______________________________________________________________________________________
@@ -97,10 +94,8 @@
     mOptimize(true),
     mUseVectorStore(true),
     mGenerateBinned(true),
-    mUseProof(false),
     mEnableDetailedOutput(false),
     mRebuild(false),
-    mNWorkers(4),
     mNToyToRebuild(100),
     mPrintLevel(0),
     mInitialFit(1),
@@ -134,7 +129,6 @@
     if (s_name.find("Optimize") != std::string::npos) mOptimize = value;
     if (s_name.find("UseVectorStore") != std::string::npos) mUseVectorStore = value;
     if (s_name.find("GenerateBinned") != std::string::npos) mGenerateBinned = value;
-    if (s_name.find("UseProof") != std::string::npos) mUseProof = value;
     if (s_name.find("EnableDetailedOutput") != std::string::npos) mEnableDetailedOutput = value;
     if (s_name.find("Rebuild") != std::string::npos) mRebuild = value;
     if (s_name.find("GenerateAsimovDataForObserved") != std::string::npos) m_generateAsimovDataForObserved = value;
@@ -152,7 +146,6 @@

     std::string s_name(name);

-    if (s_name.find("NWorkers") != std::string::npos) mNWorkers = value;
     if (s_name.find("NToyToRebuild") != std::string::npos) mNToyToRebuild = value;
     if (s_name.find("PrintLevel") != std::string::npos) mPrintLevel = value;
     if (s_name.find("InitialFit") != std::string::npos) mInitialFit = value;
@@ -202,19 +195,19 @@
 void
 RooStats::HypoTestTool::AnalyzeResult( RooStats::HypoTestInverterResult * r,
         int calculatorType,
-        int testStatType,
-        bool useCLs,
+        int testStatType,
+        bool useCLs,
         int npoints,
         const char* outfilePrefix,
         const char* outfiletype
-        )
-{
+        )
+{
     if (r==0) {
         m_logger << kERROR << "Input HypoTestInverterResult is empty. Exit." << GEndl;
         return;
     }

-    // analyize result produced by the inverter, optionally save it in a file
+    // analyize result produced by the inverter, optionally save it in a file

     double upperLimit = r->UpperLimit();
     double ulError = r->UpperLimitEstimatedError();
@@ -228,13 +221,13 @@
     m_logger << kINFO << " expected limit (-2 sig) " << r->GetExpectedUpperLimit(-2) << GEndl;
     m_logger << kINFO << " expected limit (+2 sig) " << r->GetExpectedUpperLimit(2) << GEndl;

-    // write result in a file
+    // write result in a file
     // write to a file the results
     const char *  calcType = (calculatorType == 0) ? "Freq" : (calculatorType == 1) ? "Hybr" : "Asym";
     const char *  limitType = (useCLs) ? "CLs" : "Cls+b";
     const char * scanType = (npoints < 0) ? "auto" : "grid";
     if (mResultFileName.IsNull())
-        mResultFileName = TString::Format("%s_%s_%s_%s_ts%d.root",outfilePrefix,calcType,limitType,scanType,testStatType);
+        mResultFileName = TString::Format("%s_%s_%s_%s_ts%d.root",outfilePrefix,calcType,limitType,scanType,testStatType);
     //strip the / from the filename
     if (mMassValue.size()>0) {
         mResultFileName += mMassValue.c_str();
@@ -244,18 +237,18 @@
     if (mWriteResult) {
         TFile * fileOut = new TFile(mResultFileName,"RECREATE");
         r->Write();
-        fileOut->Close();
+        fileOut->Close();
     }

-    // plot the result ( p values vs scan points)
+    // plot the result ( p values vs scan points)
     std::string typeName = "";
     if (calculatorType == 0 )
         typeName = "Frequentist";
     if (calculatorType == 1 )
-        typeName = "Hybrid";
-    else if (calculatorType == 2 || calculatorType == 3) {
+        typeName = "Hybrid";
+    else if (calculatorType == 2 || calculatorType == 3) {
         typeName = "Asymptotic";
-        mPlotHypoTestResult = false;
+        mPlotHypoTestResult = false;
     }

     const char * resultName = r->GetName();
@@ -264,7 +257,7 @@

     // plot in a new canvas with style
     TString c1Name = TString::Format("%s_Scan",typeName.c_str());
-    TCanvas *c_sig = new TCanvas(c1Name);
+    TCanvas *c_sig = new TCanvas(c1Name);
     c_sig->SetLogy(false);

     plot->Draw("CLb 2CL");  // plot all and Clb
@@ -278,15 +271,15 @@
     //split mResultFileName - it might contain a directory
     TString outputDir = TString(gSystem->DirName(mResultFileName))+"/";
     TString ULfilename = TString("upperlimit_cls_poi_") + gSystem->BaseName(mResultFileName) + outfiletype;
-    m_logger << kINFO << " writing result plot to " << TString(outputDir+ULfilename).Data() << GEndl;
+    m_logger << kINFO << " writing result plot to " << TString(outputDir+ULfilename).Data() << GEndl;
     c_sig->SaveAs( TString(outputDir+ULfilename).Data() );

     const int nEntries = r->ArraySize();

-    // plot test statistics distributions for the two hypothesis
-    if (mPlotHypoTestResult) {
+    // plot test statistics distributions for the two hypothesis
+    if (mPlotHypoTestResult) {
         TCanvas * c2 = new TCanvas();
-        if (nEntries > 1) {
+        if (nEntries > 1) {
             int ny = TMath::CeilNint( sqrt(nEntries) );
             int nx = TMath::CeilNint(double(nEntries)/ny);
             c2->Divide( nx,ny);
@@ -308,8 +301,8 @@
  */
 RooStats::HypoTestInverterResult *
 RooStats::HypoTestTool::RunHypoTestInverter(RooWorkspace * w,
-        const char * modelSBName, const char * modelBName,
-        const char * dataName, int type,  int testStatType,
+        const char * modelSBName, const char * modelBName,
+        const char * dataName, int type,  int testStatType,
         bool useCLs, int npoints, double poimin, double poimax, int ntoys,
         bool useNumberCounting,
         const char * nuisPriorName )
@@ -316,7 +309,7 @@
 {
     m_logger << kINFO << ">>> Running HypoTestInverter on the workspace " << w->GetName() << GEndl;

-    bool ok = this->SetupHypoTestInverter(w, modelSBName, modelBName, dataName, type, testStatType,
+    bool ok = this->SetupHypoTestInverter(w, modelSBName, modelBName, dataName, type, testStatType,
             useCLs, npoints, poimin, poimax, ntoys, useNumberCounting, nuisPriorName );
     if (!ok) {
         return 0;
@@ -323,7 +316,7 @@
     }

     /// by now m_calc has been setup okay ...
-    TStopwatch tw;
+    TStopwatch tw;
     tw.Start();
     RooStats::HypoTestInverterResult * r = m_calc->GetInterval();

@@ -331,16 +324,15 @@
     tw.Print();

     if (mRebuild) {
-        m_calc->SetCloseProof(1);
         tw.Start();
         SamplingDistribution * limDist = m_calc->GetUpperLimitDistribution(true,mNToyToRebuild);
         m_logger << kINFO << "Time to rebuild distributions " << GEndl;
         tw.Print();

-        if (limDist) {
-            m_logger << kINFO << "expected up limit " << limDist->InverseCDF(0.5) << " +/- "
-                << limDist->InverseCDF(0.16) << "  "
-                << limDist->InverseCDF(0.84) << "\n";
+        if (limDist) {
+            m_logger << kINFO << "expected up limit " << limDist->InverseCDF(0.5) << " +/- "
+                << limDist->InverseCDF(0.16) << "  "
+                << limDist->InverseCDF(0.84) << "\n";

             // update r to a new updated result object containing the rebuilt expected p-values distributions
             // (it will not recompute the expected limit)
@@ -348,8 +340,8 @@
             r = m_calc->GetInterval();

         }
-        else
-            m_logger << kINFO << "ERROR : failed to re-build distributions " << GEndl;
+        else
+            m_logger << kINFO << "ERROR : failed to re-build distributions " << GEndl;
     }

     m_logger << kINFO << ">>> Done running HypoTestInverter on the workspace " << w->GetName() << GEndl;
@@ -362,13 +354,13 @@
 /// internal routine to run the hypothesis test
 RooStats::HypoTestResult*
 RooStats::HypoTestTool::RunHypoTest(RooWorkspace * w, bool doUL,
-        const char * modelSBName, const char * modelBName,
-        const char * dataName, int type, int testStatType,
+        const char * modelSBName, const char * modelBName,
+        const char * dataName, int type, int testStatType,
         int ntoys,
         bool useNumberCounting,
         const char * nuisPriorName )
 {
-    bool ok = this->SetupHypoTestCalculator(w, doUL, modelSBName, modelBName, dataName, type, testStatType,
+    bool ok = this->SetupHypoTestCalculator(w, doUL, modelSBName, modelBName, dataName, type, testStatType,
             ntoys, useNumberCounting, nuisPriorName );
     if (!ok) {
         return 0;
@@ -378,7 +370,7 @@

     m_logger << kINFO << ">>> Running HypoTestCalculator on the workspace " << w->GetName() << GEndl;

-    TStopwatch tw;
+    TStopwatch tw;
     tw.Start();
     RooStats::HypoTestResult * r = m_hc->GetHypoTest();
     tw.Print();
@@ -394,21 +386,21 @@
 //_______________________________________________________________________________________
 bool
 RooStats::HypoTestTool::SetupHypoTestCalculator(RooWorkspace * w, bool doUL,
-        const char * modelSBName, const char * modelBName,
-        const char * dataName, int type,  int testStatType,
+        const char * modelSBName, const char * modelBName,
+        const char * dataName, int type,  int testStatType,
         int ntoys,
         bool useNumberCounting,
-        const char * nuisPriorName )
+        const char * nuisPriorName )
 {
     m_logger << kINFO << ">>> Setting up HypoTestCalculator on the workspace <" << w->GetName() << ">" << GEndl;
     m_logger << kINFO << ">>> Setting up HypoTest for : " << (doUL ? "exclusion" : "discovery") << GEndl;

-    RooAbsData * data = w->data(dataName);
-    if (!data) {
+    RooAbsData * data = w->data(dataName);
+    if (!data) {
         Error("HypoTestTool","Not existing data %s",dataName);
         return false;
     }
-    else
+    else
         m_logger << kINFO << "Using data set " << dataName << GEndl;

     ModelConfig* bModel = (ModelConfig*) w->obj(modelBName);
@@ -418,8 +410,8 @@
         Error("HypoTestTool","Not existing ModelConfig %s",modelSBName);
         return false;
     }
-    // check the model
-    if (!sbModel->GetPdf()) {
+    // check the model
+    if (!sbModel->GetPdf()) {
         Error("HypoTestTool","Model %s has no pdf ",modelSBName);
         return false;
     }
@@ -431,7 +423,7 @@
         Error("HypoTestTool","Model %s has no observables ",modelSBName);
         return false;
     }
-    if (!sbModel->GetSnapshot() ) {
+    if (!sbModel->GetSnapshot() ) {
         Info("HypoTestTool","Model %s has no snapshot  - make one using model poi",modelSBName);
         RooRealVar * var = dynamic_cast<RooRealVar*>(sbModel->GetParametersOfInterest()->first());
         if (var) {
@@ -443,14 +435,14 @@

     // case of no systematics
     // remove nuisance parameters from model
-    if (mNoSystematics) {
+    if (mNoSystematics) {
         const RooArgSet * nuisPar = sbModel->GetNuisanceParameters();
-        if (nuisPar && nuisPar->getSize() > 0) {
+        if (nuisPar && nuisPar->getSize() > 0) {
             m_logger << kINFO << "HypoTestTool" << "  -  Switch off all systematics by setting them constant to their initial values" << GEndl;
             RooStats::SetAllConstant(*nuisPar);
         }
         const RooArgSet * bnuisPar = bModel->GetNuisanceParameters();
-        if (bnuisPar)
+        if (bnuisPar)
             RooStats::SetAllConstant(*bnuisPar);
     }

@@ -463,7 +455,7 @@
             m_logger << kERROR << "HypoTestTool: sbModel has no POI!" << GEndl;
             return false;
         }
-
+
         RooRealVar * var = dynamic_cast<RooRealVar*>(bModel->GetParametersOfInterest()->first());
         if (!var){
           Info("HypoTestTool","Cast failed");
@@ -470,7 +462,7 @@
           m_logger << kINFO << "HypoTestTool: Cast failed part 2" << GEndl;
           return false;
         }
-
+
         double oldval = var->getVal();
         var->setVal(0);
         bModel->SetSnapshot( RooArgSet(*var)  );
@@ -477,10 +469,10 @@
         var->setVal(oldval);

         //LoadSnapshotInWorkspace(w, "snapshot_paramsVals_RooExpandedFitResult_afterFit")
-    } else if (!bModel->GetSnapshot() ) { // MB : note, this resets all parameters!
+    } else if (!bModel->GetSnapshot() ) { // MB : note, this resets all parameters!
         Info("HypoTestTool", "Background model %s has no snapshot - make one using model poi and 0 values ",modelBName);
         RooRealVar * var = dynamic_cast<RooRealVar*>(bModel->GetParametersOfInterest()->first());
-        if (var) {
+        if (var) {
             double oldval = var->getVal();
             var->setVal(0);
             bModel->SetSnapshot( RooArgSet(*var)  );
@@ -488,7 +480,7 @@

             Error("HypoTestTool","Model %s has no valid poi",modelBName);
             return false;
-        }
+        }
     } else { // reset
         sbModel->GetSnapshot();
     }
@@ -495,13 +487,13 @@

     // check model  has global observables when there are nuisance pdf
     // for the hybrid case the globobs are not needed
-    if (type != 1 ) {
+    if (type != 1 ) {
         bool hasNuisParam = (sbModel->GetNuisanceParameters() && sbModel->GetNuisanceParameters()->getSize() > 0);
         bool hasGlobalObs = (sbModel->GetGlobalObservables() && sbModel->GetGlobalObservables()->getSize() > 0);
-        if (hasNuisParam && !hasGlobalObs ) {
-            // try to see if model has nuisance parameters first
+        if (hasNuisParam && !hasGlobalObs ) {
+            // try to see if model has nuisance parameters first
             RooAbsPdf * constrPdf = RooStats::MakeNuisancePdf(*sbModel,"nuisanceConstraintPdf_sbmodel");
-            if (constrPdf) {
+            if (constrPdf) {
                 Warning("StandardHypoTestInvDemo","Model %s has nuisance parameters but no global observables associated",sbModel->GetName());
                 Warning("StandardHypoTestInvDemo","\tThe effect of the nuisance parameters will not be treated correctly ");
             }
@@ -508,11 +500,11 @@
         }
     }

-    // run first a data fit
+    // run first a data fit
     const RooArgSet * poiSet = sbModel->GetParametersOfInterest();
     RooRealVar *poi = (RooRealVar*)poiSet->first();

-    m_logger << kINFO << "HypoTestTool : POI initial value:   " << poi->GetName() << " = " << poi->getVal()   << GEndl;
+    m_logger << kINFO << "HypoTestTool : POI initial value:   " << poi->GetName() << " = " << poi->getVal()   << GEndl;

     //poi->setRange(0,100);

@@ -525,12 +517,12 @@

     if(m_generateAsimovDataForObserved) {
         m_logger << kINFO << "HypoTestTool: generateAsimovDataForObserved is set to true; enabling doFit" << GEndl;
-        doFit = true;
+        doFit = true;
     }

-    if (doFit)  {
+    if (doFit)  {
         // do the fit : By doing a fit the POI snapshot (for S+B) is set to the fit value
-        // and the nuisance parameters nominal values will be set to the fit value.
+        // and the nuisance parameters nominal values will be set to the fit value.
         // This is relevant when using LEP test statistics
         const RooArgSet* prevSnapSet = sbModel->GetSnapshot();
         const RooArgSet* tPoiSet = sbModel->GetParametersOfInterest();
@@ -538,7 +530,7 @@
         Info( "HypoTestTool", "Doing a first fit to the observed data ");
         if (mMinimizerType.size()==0) {
             mMinimizerType = ROOT::Math::MinimizerOptions::DefaultMinimizerType();
-        } else {
+        } else {
             ROOT::Math::MinimizerOptions::SetDefaultMinimizer(mMinimizerType.c_str());
         }
         Info("HypoTestTool", "Using %s as minimizer for computing the test statistic",
@@ -549,7 +541,7 @@
         TStopwatch tw;
         tw.Start();
         Bool_t verbose = (m_logger.GetMinLevel() <= kDEBUG) ? kTRUE : kFALSE;
-
+
         RooLinkedList fitArgs;
         RooCmdArg _InitialHesse(InitialHesse(false));
         RooCmdArg _Hesse(Hesse(false));
@@ -572,25 +564,25 @@
         fitArgs.Add(dynamic_cast<TObject*>(&_PrintLevel));
         fitArgs.Add(dynamic_cast<TObject*>(&_Constrain));
         fitArgs.Add(dynamic_cast<TObject*>(&_Save));
-
+
         RooLinkedList fitArgs_speed(fitArgs);
-
+
         fitArgs.Add(dynamic_cast<TObject*>(&_Strategy_default));
         fitArgs_speed.Add(dynamic_cast<TObject*>(&_Strategy_speed));
-
+
         RooFitResult *fitres = sbModel->GetPdf()->fitTo(*data, fitArgs_speed);
-
+
         if (fitres->status() != 0) {
             Warning("HypoTestTool", "Fit to the model failed - try with strategy 1");
             fitres = sbModel->GetPdf()->fitTo(*data, fitArgs);
         }
-        if (fitres->status() != 0)
+        if (fitres->status() != 0)
             Warning("HypoTestTool", "Fit still failed - continue anyway.....");

         poihat  = poi->getVal();
-        m_logger << kINFO << "HypoTestTool - Best Fit value : " << poi->GetName() << " = "
+        m_logger << kINFO << "HypoTestTool - Best Fit value : " << poi->GetName() << " = "
             << poihat << " +/- " << poi->getError() << GEndl;
-        m_logger << kINFO << "Time for fitting : "; tw.Print();
+        m_logger << kINFO << "Time for fitting : "; tw.Print();

         RooArgSet newSnapSet;
         if (tPoiSet!=0) newSnapSet.add(*tPoiSet); // make sure this is the full poi set.
@@ -602,27 +594,27 @@
                 if (vr==0) continue;
                 TString vrName = vr->GetName();
                 RooRealVar* par = (RooRealVar*)newSnapSet.find(vrName.Data());
-                if (par==0) { newSnapSet.add(*vr); } // add if not yet present
+                if (par==0) { newSnapSet.add(*vr); } // add if not yet present
             }
         }

-        //save best fit value in the poi snapshot
+        //save best fit value in the poi snapshot
         //sbModel->SetSnapshot(*sbModel->GetParametersOfInterest());
         sbModel->SetSnapshot(newSnapSet);
-        m_logger << kINFO << "HypoTestTool: snapshot of S+B Model " << sbModel->GetName()
-            << " is set to the best fit value" << GEndl;
+        m_logger << kINFO << "HypoTestTool: snapshot of S+B Model " << sbModel->GetName()
+            << " is set to the best fit value" << GEndl;
     }

     // and reset poi
     poi->setVal(poihat);

-    // build test statistics and hypotest calculators for running the inverter
+    // build test statistics and hypotest calculators for running the inverter

-    // print a message in case of LEP test statistics because it affects result by doing or not doing a fit
+    // print a message in case of LEP test statistics because it affects result by doing or not doing a fit
     if (testStatType == 0) {
-        if (!doFit)
+        if (!doFit)
             Info("HypoTestTool", "Using LEP test statistic - an initial fit is not done and the TS will use the nuisances at the model value");
-        else
+        else
             Info("HypoTestTool", "Using LEP test statistic - an initial fit has been done and the TS will use the nuisances at the best fit value");
     }

@@ -630,12 +622,12 @@
     ModelConfig* nullModel = ( doUL ? sbModel : bModel );
     ModelConfig* altModel  = ( doUL ? bModel : sbModel );

-    // build test statistics and hypotest calculators for running the inverter
+    // build test statistics and hypotest calculators for running the inverter
     SimpleLikelihoodRatioTestStat* slrts(0);
     if (testStatType == 0) {
         slrts = new SimpleLikelihoodRatioTestStat(*nullModel->GetPdf(),*altModel->GetPdf());

-        // null parameters must includes snapshot of poi plus the nuisance values
+        // null parameters must includes snapshot of poi plus the nuisance values
         RooArgSet nullParams(*nullModel->GetSnapshot());
         if (nullModel->GetNuisanceParameters()) nullParams.add(*nullModel->GetNuisanceParameters());
         if (nullModel->GetSnapshot()) slrts->SetNullParameters(nullParams);
@@ -645,7 +637,7 @@

         if (mEnableDetailedOutput) slrts->EnableDetailedOutput();
         slrts->SetReuseNLL(mOptimize);
-    }
+    }

     // ratio of profile likelihood - need to pass snapshot for the alt
     RatioOfProfiledLikelihoodsTestStat* ropl(0);
@@ -657,12 +649,12 @@
         ropl->SetReuseNLL(mOptimize);
         if (mEnableDetailedOutput) ropl->EnableDetailedOutput();
         if (mOptimize) ropl->SetStrategy(0);
-    }
+    }

     ProfileLikelihoodTestStat* profll(0);
     if (testStatType == 2 || testStatType==3) {
         profll = new ProfileLikelihoodTestStat(*nullModel->GetPdf());
-        if (testStatType == 3) {
+        if (testStatType == 3) {
             profll->SetOneSided(true);
             if (!doUL) profll->SetOneSidedDiscovery(true);
         }
@@ -680,10 +672,10 @@

     MaxLikelihoodEstimateTestStat* maxll(0);
     if (testStatType == 4) {
-        maxll = new MaxLikelihoodEstimateTestStat(*nullModel->GetPdf(),*poi);
-    }
+        maxll = new MaxLikelihoodEstimateTestStat(*nullModel->GetPdf(),*poi);
+    }

-    // create the HypoTest calculator class
+    // create the HypoTest calculator class
     this->ResetHypoTestCalculator(); // first clear

     // build some asimov data here
@@ -692,7 +684,7 @@
         fAsimovGlobObs.removeAll();
         const RooArgSet *altSnapshot = altModel->GetSnapshot();
         RooArgSet poiAlt(*altSnapshot);
-        RooArgSet *tmp = (RooArgSet*) poiAlt.snapshot();
+        RooArgSet *tmp = (RooArgSet*) poiAlt.snapshot();
         auto fAsimovData = AsymptoticCalculator::MakeAsimovData(*data, *nullModel, poiAlt, fAsimovGlobObs, tmp);

         data = dynamic_cast<RooAbsData*>(fAsimovData->Clone());
@@ -703,13 +695,13 @@
     if (type == 0) m_hc = new FrequentistCalculator(*data, *altModel, *nullModel);
     else if (type == 1) m_hc = new HybridCalculator(*data, *altModel, *nullModel);
     else if (type == 2) m_hc = new AsymptoticCalculator(*data, *altModel, *nullModel);
-    else if (type == 3) m_hc = new AsymptoticCalculator(*data, *altModel, *nullModel, true);  // for using Asimov data generated with nominal values
+    else if (type == 3) m_hc = new AsymptoticCalculator(*data, *altModel, *nullModel, true);  // for using Asimov data generated with nominal values
     else {
         Error("HypoTestTool","Invalid - calculator type = %d supported values are only :\n\t\t\t 0 (Frequentist) , 1 (Hybrid) , 2 (Asymptotic) ",type);
         return false;
     }

-    // set the test statistic
+    // set the test statistic
     TestStatistic * testStat = 0;
     if (testStatType == 0) {
         testStat = slrts;
@@ -717,23 +709,23 @@
     if (testStatType == 1) {
         testStat = ropl;
     }
-    if (testStatType == 2 || testStatType == 3) {
+    if (testStatType == 2 || testStatType == 3) {
         testStat = profll;
     }
     if (testStatType == 4) {
         testStat = maxll;
     }
-    if (testStat == 0) {
+    if (testStat == 0) {
         Error("HypoTestTool","Invalid - test statistic type = %d supported values are only :\n\t\t\t 0 (SLR) , 1 (Tevatron) , 2 (PLR), 3 (PLR1), 4(MLE)",testStatType);
         return false;
-    }
+    }

     ToyMCSampler *toymcs = dynamic_cast<ToyMCSampler*>(m_hc->GetTestStatSampler());
-    if (toymcs) {
+    if (toymcs) {
         if (useNumberCounting) toymcs->SetNEventsPerToy(1);
         toymcs->SetTestStatistic(testStat);

-        if (data->isWeighted() && !mGenerateBinned) {
+        if (data->isWeighted() && !mGenerateBinned) {
             Info("HypoTestTool","Data set is weighted, nentries = %d and sum of weights = %8.1f but toy generation is unbinned - it would be faster to set mGenerateBinned to true\n",
                     data->numEntries(), data->sumEntries());
         }
@@ -741,26 +733,26 @@

         toymcs->SetUseMultiGen(mOptimize);

-        if (mGenerateBinned &&  nullModel->GetObservables()->getSize() > 2) {
+        if (mGenerateBinned &&  nullModel->GetObservables()->getSize() > 2) {
             Warning("HypoTestTool","generate binned is activated but the number of ovservable is %d. Too much memory could be needed for allocating all the bins",
                     nullModel->GetObservables()->getSize() );

             // set the random seed if needed
-            if (mRandomSeed >= 0) RooRandom::randomGenerator()->SetSeed(mRandomSeed);
-        }
+            if (mRandomSeed >= 0) RooRandom::randomGenerator()->SetSeed(mRandomSeed);
+        }
     }

-    if (type == 1) {
+    if (type == 1) {
         HybridCalculator *hhc = dynamic_cast<HybridCalculator*> (m_hc);
         assert(hhc);

-        hhc->SetToys( static_cast<Int_t>(ntoys), static_cast<Int_t>(ntoys/mNToysRatio) ); // can use less ntoys for b hypothesis
+        hhc->SetToys( static_cast<Int_t>(ntoys), static_cast<Int_t>(ntoys/mNToysRatio) ); // can use less ntoys for b hypothesis

         // remove global observables from ModelConfig (this is probably not needed anymore in 5.32)
         altModel->SetGlobalObservables(RooArgSet() );
         nullModel->SetGlobalObservables(RooArgSet() );

-        // check for nuisance prior pdf in case of nuisance parameters
+        // check for nuisance prior pdf in case of nuisance parameters
         if (altModel->GetNuisanceParameters() || nullModel->GetNuisanceParameters() ) {

             // fix for using multigen (does not work in this case)
@@ -767,22 +759,22 @@
             toymcs->SetUseMultiGen(false);
             ToyMCSampler::SetAlwaysUseMultiGen(false);

-            RooAbsPdf * nuisPdf = 0;
+            RooAbsPdf * nuisPdf = 0;
             if (nuisPriorName) nuisPdf = w->pdf(nuisPriorName);
             // use prior defined first in altModel (then in nullModel)
-            if (!nuisPdf)  {
+            if (!nuisPdf)  {
                 Info("HypoTestTool","No nuisance pdf given for the HybridCalculator - try to deduce  pdf from the model");
-                if (altModel->GetPdf() && altModel->GetObservables() )
+                if (altModel->GetPdf() && altModel->GetObservables() )
                     nuisPdf = RooStats::MakeNuisancePdf(*altModel,"nuisancePdf_bmodel");
-                else
+                else
                     nuisPdf = RooStats::MakeNuisancePdf(*nullModel,"nuisancePdf_sbmodel");
-            }
+            }
             if (!nuisPdf ) {
-                if (altModel->GetPriorPdf())  {
+                if (altModel->GetPriorPdf())  {
                     nuisPdf = altModel->GetPriorPdf();
-                    Info("HypoTestTool","No nuisance pdf given - try to use %s that is defined as a prior pdf in the B model",nuisPdf->GetName());
+                    Info("HypoTestTool","No nuisance pdf given - try to use %s that is defined as a prior pdf in the B model",nuisPdf->GetName());
                 }
-                else {
+                else {
                     Error("HypoTestTool","Cannnot run Hybrid calculator because no prior on the nuisance parameter is specified or can be derived");
                     return false;
                 }
@@ -793,23 +785,23 @@

             const RooArgSet* nuisParams = (altModel->GetNuisanceParameters() ) ? altModel->GetNuisanceParameters() : nullModel->GetNuisanceParameters();
             RooArgSet* np = nuisPdf->getObservables(*nuisParams);
-            if (np->getSize() == 0) {
+            if (np->getSize() == 0) {
                 Warning("HypoTestTool","Prior nuisance does not depend on nuisance parameters. They will be smeared in their full range");
             }
             delete np;

             hhc->ForcePriorNuisanceAlt(*nuisPdf);
-            hhc->ForcePriorNuisanceNull(*nuisPdf);
+            hhc->ForcePriorNuisanceNull(*nuisPdf);
         }
-    } else if (type == 2 || type == 3) {
-        if (testStatType == 3) dynamic_cast<AsymptoticCalculator*>(m_hc)->SetOneSided(true);
-        if (testStatType != 2 && testStatType != 3)
+    } else if (type == 2 || type == 3) {
+        if (testStatType == 3) dynamic_cast<AsymptoticCalculator*>(m_hc)->SetOneSided(true);
+        if (testStatType != 2 && testStatType != 3)
             Warning("StandardHypoTestInvDemo","Only the PL test statistic can be used with AsymptoticCalculator - use by default a two-sided PL");
-
+
         dynamic_cast<AsymptoticCalculator*>(m_hc)->SetQTilde(true); // bug in roostats. Need to turn on explicitly.
-        dynamic_cast<AsymptoticCalculator*>(m_hc)->SetPrintLevel(mPrintLevel+1);
+        dynamic_cast<AsymptoticCalculator*>(m_hc)->SetPrintLevel(mPrintLevel+1);
     } else if (type == 0 || type == 1)  {
-        ((FrequentistCalculator*) m_hc)->SetToys( static_cast<Int_t>(ntoys), static_cast<Int_t>(ntoys/mNToysRatio) );
+        ((FrequentistCalculator*) m_hc)->SetToys( static_cast<Int_t>(ntoys), static_cast<Int_t>(ntoys/mNToysRatio) );
     }

     // Get the result
@@ -824,9 +816,9 @@
 //_______________________________________________________________________________________
 bool
 RooStats::HypoTestTool::SetupHypoTestInverter(RooWorkspace * w,
-        const char * modelSBName, const char * modelBName,
-        const char * dataName, int type,  int testStatType,
-        bool useCLs, int npoints, double poimin, double poimax,
+        const char * modelSBName, const char * modelBName,
+        const char * dataName, int type,  int testStatType,
+        bool useCLs, int npoints, double poimin, double poimax,
         int ntoys,
         bool useNumberCounting,
         const char * nuisPriorName ){
@@ -833,7 +825,7 @@

     m_logger << kINFO << ">>> Setting up HypoTestInverter on the workspace " << w->GetName() << GEndl;

-    bool ok = this->SetupHypoTestCalculator(w, true, modelSBName, modelBName, dataName, type, testStatType,
+    bool ok = this->SetupHypoTestCalculator(w, true, modelSBName, modelBName, dataName, type, testStatType,
             ntoys, useNumberCounting, nuisPriorName ); /// MB : Note the true : always assuming exclusion here

     if (!ok) {
@@ -850,13 +842,6 @@
     m_calc->UseCLs(useCLs);
     m_calc->SetVerbose(true);

-    // can speed up using proof-lite
-    if (mUseProof && mNWorkers > 1) {
-        ToyMCSampler *toymcs = (ToyMCSampler*)m_hc->GetTestStatSampler();
-        ProofConfig *pc = new ProofConfig(*w, mNWorkers, "", kFALSE);
-        toymcs->SetProofConfig(pc);    // enable proof
-    }
-
     // get models from WS
     // get the modelConfig out of the file
     ModelConfig* sbModel = (ModelConfig*) w->obj(modelSBName);
@@ -865,8 +850,8 @@
         Error("HypoTestTool","Not existing ModelConfig %s",modelSBName);
         return false;
     }
-    // check the model
-    if (!sbModel->GetPdf()) {
+    // check the model
+    if (!sbModel->GetPdf()) {
         Error("HypoTestTool","Model %s has no pdf ",modelSBName);
         return false;
     }
@@ -878,7 +863,7 @@
         Error("HypoTestTool","Model %s has no observables ",modelSBName);
         return false;
     }
-    if (!sbModel->GetSnapshot() ) {
+    if (!sbModel->GetSnapshot() ) {
         Info("HypoTestTool","Model %s has no snapshot  - make one using model poi",modelSBName);
         sbModel->SetSnapshot( *sbModel->GetParametersOfInterest() );
     }
@@ -889,8 +874,8 @@
     double poihat  = poi->getVal();

     if (npoints > 0) {
-        if (poimin > poimax) {
-            // if no min/max given scan between MLE and +4 sigma
+        if (poimin > poimax) {
+            // if no min/max given scan between MLE and +4 sigma
             if (poimin > poihat) { poimin = poihat - 4 * poi->getError(); }
             if (poimin < 0) { poimin = 0; }
             poimax = ( poihat +  20 * poi->getErrorHi() );
@@ -899,7 +884,7 @@
         if ( poimax > poi->getMax() ) { poi->setMax( poimax ); }
         m_calc->SetFixedScan(npoints,poimin,poimax);
     }
-    else {
+    else {
         m_logger << kINFO << "Doing an automatic scan in interval: " << poi->getMin() << " , " << poi->getMax() << GEndl;
     }

