diff --git a/repoze.who-friendlyform-1.0.8/repoze/who/plugins/friendlyform.py b/repoze.who-friendlyform-1.0.8/repoze/who/plugins/friendlyform.py
index cf1efa6..83b682b 100644
--- a/repoze.who-friendlyform-1.0.8/repoze/who/plugins/friendlyform.py
+++ b/repoze.who-friendlyform-1.0.8/repoze/who/plugins/friendlyform.py
@@ -14,9 +14,16 @@
 ##############################################################################
 
 """Collection of :mod:`repoze.who` friendly forms"""
+try:
+    from urlparse import urlparse, urlunparse #Python 2.x
+except ImportError:
+    from urllib.parse import urlparse, urlunparse #Python 3+
+
+try:
+    from urllib import urlencode #Python 2.x
+except ImportError:
+    from urllib.parse import urlencode #Python 3+
 
-from urlparse import urlparse, urlunparse
-from urllib import urlencode
 try:
     from urlparse import parse_qs
 except ImportError:#pragma: no cover
@@ -26,8 +33,8 @@ from webob import Request
 # TODO: Stop using Paste; we already started using WebOb
 from webob.exc import HTTPFound, HTTPUnauthorized
 from paste.request import construct_url, parse_dict_querystring, parse_formvars
-from zope.interface import implements
-
+#from zope.interface import implements
+from zope.interface import implementer
 from repoze.who.interfaces import IChallenger, IIdentifier
 
 __all__ = ['FriendlyFormPlugin']
@@ -58,7 +65,8 @@ class FriendlyFormPlugin(object):
     supported) and ISO-8859-1 (aka "Latin-1") is the default one.
 
     """
-    implements(IChallenger, IIdentifier)
+#    implements(IChallenger, IIdentifier)
+    implementer(IChallenger, IIdentifier)
 
     classifications = {
         IIdentifier: ["browser"],
