diff --git a/setup.py b/setup.py
index 9fc0efc..d864ee9 100644
--- a/setup.py
+++ b/setup.py
@@ -1,10 +1,11 @@
-from setuptools import setup
-
+from setuptools import setup, find_packages
 
 def readme():
     with open('README.md') as f:
         return f.read()
 
+package_data = {'': ['process/templates/*.xml']}
+
 setup(
     name='cgsn_processing',
     version='0.1.0',
@@ -24,7 +25,7 @@ setup(
     author='Christopher Wingard',
     author_email='cwingard@coas.oregonstate.edu',
     license='MIT',
-    packages=['cgsn_processing'],
+    packages=find_packages(),
     install_requires=[
         'numpy >= 1.9.2',
         'scipy >= 0.15.1',
@@ -37,5 +38,5 @@ setup(
         'pyaxiom',
         'requests'
     ],
-    include_package_data=True,
+    package_data=package_data,
     zip_safe=False)
