--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -124,10 +124,6 @@ if(KOMPUTE_OPT_INSTALL)
         install(TARGETS ${vk_header_target}
             EXPORT komputeTargets)
     endif ()
-
-    if(KOMPUTE_OPT_BUILD_PYTHON)
-        # We can't export pybind11::headers because it's alias target, so we unwrap the alias and install it.
-        get_target_property(pybind11_aliased_target pybind11::headers ALIASED_TARGET)
-        install(TARGETS ${pybind11_aliased_target}
-            EXPORT komputeTargets)
-    endif ()
+    # When using external pybind11, we don't need to install pybind11 targets
 endif()
