--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -232,10 +232,6 @@ if(KOMPUTE_OPT_ANDROID_BUILD)
     add_compile_definitions(VK_USE_PLATFORM_ANDROID_KHR=1)
 endif()

-if(KOMPUTE_OPT_BUILD_PYTHON)
-    add_compile_definitions(KOMPUTE_BUILD_PYTHON=1)
-endif()
-
 if(KOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS)
     add_compile_definitions(KOMPUTE_DISABLE_VK_DEBUG_LAYERS=1)
 endif()
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -6,3 +6,6 @@ include_directories(
 target_link_libraries(
         kp PRIVATE
         kompute::kompute)
+
+# Only the Python module needs KOMPUTE_BUILD_PYTHON
+target_compile_definitions(kp PRIVATE KOMPUTE_BUILD_PYTHON=1)
