diff --git a/fortran/common/debug.F90 b/fortran/common/debug.F90
index 22ae025..ffb4403 100644
--- a/fortran/common/debug.F90
+++ b/fortran/common/debug.F90
@@ -111,7 +111,7 @@ if (present(code)) then
     ! N.B.: In Fortran 2008, stop code must be a scalar default character or integer CONSTANT
     ! expression, but Fortran 2018 lifts the requirement on constancy. gfortran is strict in this
     ! aspect. Consequently, for gfortran, compile with either `-std=f2018` or no `-std` at all.
-    error stop code
+    error stop
 else
     error stop
 end if
