From b9fd5ddebf3f576b21ca3e5b3405b8372f4a9834 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 18 Jan 2024 16:30:03 +0100
Subject: [PATCH] Update system.make

---
 src/system.make | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/system.make b/src/system.make
index d0d1d93b..68293ee9 100644
--- a/src/system.make
+++ b/src/system.make
@@ -5,12 +5,12 @@ FC=mpif90
 F77=mpif77
 
 # Linking flags
-LINKFLAGS= -L/usr/local/lib
+LINKFLAGS= $(LDFLAGS)
 ARFLAGS=
 
 # Compilation flags
 # NB for gcc10 you need to add -fallow-argument-mismatch
-COMPFLAGS= -O3 $(XC_COMPFLAGS)
+COMPFLAGS= $(FFLAGS)
 COMPFLAGS_F77= $(COMPFLAGS)
 
 # Set BLAS and LAPACK libraries
@@ -18,7 +18,7 @@ COMPFLAGS_F77= $(COMPFLAGS)
 # BLAS= -lvecLibFort
 # Intel MKL use the Intel tool
 # Generic
-# BLAS= -llapack -lblas
+BLAS= -llapack -lblas
 
 # Full library call; remove scalapack if using dummy diag module
 LIBS= $(FFT_LIB) $(XC_LIB) -lscalapack $(BLAS)
@@ -35,7 +35,7 @@ LIBS= $(FFT_LIB) $(XC_LIB) -lscalapack $(BLAS)
 # XC_LIBRARY = LibXC_v4
 XC_LIBRARY = LibXC_v5
 XC_LIB = -lxcf90 -lxc
-XC_COMPFLAGS = -I/usr/local/include
+XC_COMPFLAGS = -I/$(PREFIX)/include
 
 # Set FFT library
 FFT_LIB=-lfftw3
