diff --git a/src/system/system.make b/src/system/system.make
new file mode 100644
index 00000000..0e03b445
--- /dev/null
+++ src/system/system.make
@@ -0,0 +1,17 @@
+# This is an example system-specific makefile. You will need to adjust
+# it for the actual system you are running on.
+
+# Set compilers
+FC=mpif90
+LINKFLAGS=$(LDFLAGS) -fopenmp
+BLAS= -llapack -lblas
+SCALAPACK = -lscalapack
+XC_LIBRARY = LibXC_v5
+XC_LIB = -lxcf90 -lxc
+XC_COMPFLAGS = -I$(PREFIX)/include
+FFT_LIB=-lfftw3
+FFT_OBJ=fft_fftw3.o
+MULT_KERN = default
+DIAG_DUMMY =
+LIBS= $(XC_LIB) -lscalapack-openmpi $(BLAS) $(FFT_LIB)
+COMPFLAGS=$(FFLAGS) -O3 -fopenmp $(XC_COMPFLAGS) -fallow-argument-mismatch
