diff --git a/src/system/system.make b/src/system/system.make
new file mode 100644
index 00000000..0e03b445
--- /dev/null
+++ src/system/system.make
@@ -0,0 +1,17 @@
+# This is an example system-specific makefile. You will need to adjust
+# it for the actual system you are running on.
+
+# Set compilers
+FC=mpif90
+LINKFLAGS=$(LDFLAGS) -fopenmp
+BLAS= -llapack -lblas
+SCALAPACK = -lscalapack
+XC_LIBRARY = LibXC_v5
+XC_LIB = -lxcf90 -lxc
+XC_COMPFLAGS = -I$(PREFIX)/include
+FFT_LIB=-lfftw3
+FFT_OBJ=fft_fftw3.o
+MULT_KERN = default
+DIAG_DUMMY =
+LIBS= $(XC_LIB) -lscalapack $(BLAS) $(FFT_LIB)
+COMPFLAGS=$(FFLAGS) -O3 -fopenmp $(XC_COMPFLAGS) -fallow-argument-mismatch
diff --git a/src/Makefile b/src/Makefile
index 6f3509b2..b1ccafe2 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -75,7 +75,7 @@ deps.obj.inc: $(SRCS_NODS) $(SYSTEM_PATH)
 	$(ECHOSTR) '  character(len=*), parameter :: datestr="'`date "+%Y/%m/%d at %H:%M %z"`'"' >> datestamp.f90
 	sed "s/BBB/"`git symbolic-ref HEAD | cut -b 12-`"/" $(COMMENT) | sed "s/RRR/"`git describe --tags`"/" >> datestamp.f90
 	$(ECHOSTR) "end module datestamp" >> datestamp.f90
-	./makedeps makedeps.txt $^
+	$(BUILD_PREFIX)/bin/perl makedeps makedeps.txt $^
 	sed /"^mpi.o"/D makedeps.txt > deps.obj.inc
 
 #Target
