diff --git a/setup.py b/setup.py
index 2e3f282..176b17e 100644
--- a/setup.py
+++ b/setup.py
@@ -69,8 +69,7 @@ class NPM(Command):

     targets = [
         os.path.join(here, 'itkwidgets', 'static', 'extension.js'),
-        os.path.join(here, 'itkwidgets', 'static', 'index.js'),
-        os.path.join(here, 'js', 'dist', 'labextension.js')
+        os.path.join(here, 'itkwidgets', 'static', 'index.js')
     ]

     def initialize_options(self):
