From 2e944159f7dc6c1a791bee1facb3a24cdd8d7c3e Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 14 Sep 2022 19:32:13 +0200
Subject: [PATCH] Remove flags

---
 CMakeLists.txt | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ada8340..cba3014 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,25 +2,6 @@ cmake_minimum_required (VERSION 2.8)
 project (tinyrenderer)
 
 add_definitions(-D_USE_MATH_DEFINES)
-include(CheckCXXCompilerFlag)
-
-function(enable_cxx_compiler_flag_if_supported flag)
-    string(FIND "${CMAKE_CXX_FLAGS}" "${flag}" flag_already_set)
-    if(flag_already_set EQUAL -1)
-        check_cxx_compiler_flag("${flag}" flag_supported)
-        if(flag_supported)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flag}" PARENT_SCOPE)
-        endif()
-        unset(flag_supported CACHE)
-    endif()
-endfunction()
-
-enable_cxx_compiler_flag_if_supported("-Wall")
-enable_cxx_compiler_flag_if_supported("-Wextra")
-enable_cxx_compiler_flag_if_supported("-pedantic")
-enable_cxx_compiler_flag_if_supported("-std=c++14")
-enable_cxx_compiler_flag_if_supported("-O3")
-enable_cxx_compiler_flag_if_supported("-fopenmp")
 
 file(GLOB SOURCES *.h *.cpp)
 
