From c52c43c19245cc857bfdae7cca605fbc6bf5f9b8 Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Fri, 23 Oct 2020 20:40:27 +0200
Subject: [PATCH] rt_fix

---
 tests/CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git tests/CMakeLists.txt tests/CMakeLists.txt
index e9ece4a..42e0efb 100644
--- tests/CMakeLists.txt
+++ tests/CMakeLists.txt
@@ -22,6 +22,7 @@
 #
 # This macro will create a binary from `NAME.cc' and add it to the test suite.
 #
+find_library(RT_LIB rt)
 MACRO(ADD_TESTCASE NAME GENERATED)
   IF (${GENERATED} STREQUAL TRUE)
    ADD_UNIT_TEST(${NAME} ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.cc)
@@ -30,7 +31,11 @@ MACRO(ADD_TESTCASE NAME GENERATED)
   ENDIF()
 
   PKG_CONFIG_USE_DEPENDENCY(${NAME} tinyxml)
-  TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME})
+  if(RT_LIB)
+    TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME} ${RT_LIB})
+  else()
+    TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME} )
+  endif()
 ENDMACRO(ADD_TESTCASE)
 
 CONFIG_FILES(srdfparser.cc)
-- 
2.17.1

