From 03924c56f12c82b5820cb903dd7e782c4a975457 Mon Sep 17 00:00:00 2001
From: Santiago Castro <bryant1410@gmail.com>
Date: Tue, 19 Jul 2022 08:15:26 +0000
Subject: [PATCH 4/7] Fix typo in device docstring (#171)

---
 python/decord/ndarray.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/decord/ndarray.py b/python/decord/ndarray.py
index 56e7324..1f6878e 100644
--- a/python/decord/ndarray.py
+++ b/python/decord/ndarray.py
@@ -36,7 +36,7 @@ def cpu(dev_id=0):
     return DECORDContext(1, dev_id)
 
 def gpu(dev_id=0):
-    """Construct a CPU device
+    """Construct a GPU device
 
     Parameters
     ----------
-- 
2.43.0

