From b63f61a267d9bf0b3ef1c034e5bd775c911b5be6 Mon Sep 17 00:00:00 2001
From: Santiago Castro <bryant1410@gmail.com>
Date: Tue, 19 Jul 2022 08:16:48 +0000
Subject: [PATCH 6/7] Avoid creating extra mem in string join (#173)

---
 python/decord/video_loader.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/decord/video_loader.py b/python/decord/video_loader.py
index ace40c5..ce80771 100644
--- a/python/decord/video_loader.py
+++ b/python/decord/video_loader.py
@@ -42,7 +42,7 @@ class VideoLoader(object):
         self._handle = None
         assert isinstance(uris, (list, tuple))
         assert (len(uris) > 0)
-        uri = ','.join([x.strip() for x in uris])
+        uri = ','.join(x.strip() for x in uris)
         if isinstance(ctx, DECORDContext):
             ctx = [ctx]
         for _ctx in ctx:
-- 
2.43.0

