diff --git a/nengo_gui/compat.py b/nengo_gui/compat.py
index 2f1dbfee..1d49e08c 100644
--- a/nengo_gui/compat.py
+++ b/nengo_gui/compat.py
@@ -5,6 +5,11 @@
 
 import numpy as np
 
+try:
+    collectionsAbc = collections.abc
+except AttributeError:
+    collectionsAbc = collections
+
 # Only test for Python 2 so that we have less changes for Python 4
 PY2 = sys.version_info[0] == 2
 
@@ -32,4 +37,4 @@ def is_iterable(obj):
     if isinstance(obj, np.ndarray):
         return obj.ndim > 0  # 0-d arrays give error if iterated over
     else:
-        return isinstance(obj, collections.Iterable)
+        return isinstance(obj, collectionsAbc.Iterable)
diff --git a/nengo_gui/completion.py b/nengo_gui/completion.py
index 989afda2..a3c3692f 100644
--- a/nengo_gui/completion.py
+++ b/nengo_gui/completion.py
@@ -10,7 +10,7 @@ class Script(object):
         def __init__(self, *args, **kwargs):
             pass
 
-        def completions(self):
+        def complete(self, line, column):
             return []
 
 
@@ -19,5 +19,5 @@ def completions(self):
 
 def get_completions(code, line, column, path=None):
     with _jedi_lock:
-        script = Script(code, line, column, path=path)
-        return script.completions()
+        script = Script(code, path=path)
+        return script.complete(line, column)
