diff --git ah_bootstrap.py ah_bootstrap.py
index 786b8b1..e18ce7e 100644
--- ah_bootstrap.py
+++ ah_bootstrap.py
@@ -150,9 +150,9 @@ PACKAGE_NAME = 'astropy_helpers'
 # Defaults for other options
 DOWNLOAD_IF_NEEDED = True
 INDEX_URL = 'https://pypi.python.org/simple'
-USE_GIT = True
-OFFLINE = False
-AUTO_UPGRADE = True
+USE_GIT = False
+OFFLINE = True
+AUTO_UPGRADE = False
 
 # A list of all the configuration options and their required types
 CFG_OPTIONS = [
