diff -uraN -x .git -x conda_build.sh -x build_env_setup.sh old/tools/install.py new/tools/install.py
--- old/tools/install.py	2020-10-24 20:32:59.000000000 +0200
+++ new/tools/install.py	2020-11-09 20:25:54.058733184 +0100
@@ -97,7 +97,7 @@
   logging.basicConfig(level=logging.DEBUG if args.verbose else logging.INFO)
   os.makedirs(target)
   copy_emscripten(target)
-  add_revision_file(target)
+  # add_revision_file(target)
   return 0
 
 
