diff --git a/src/cvcuda/priv/CMakeLists.txt b/src/cvcuda/priv/CMakeLists.txt
index deac31c..26233db 100644
--- a/src/cvcuda/priv/CMakeLists.txt
+++ b/src/cvcuda/priv/CMakeLists.txt
@@ -102,9 +102,9 @@ target_link_libraries(cvcuda_priv
         cvcuda_headers
         nvcv_util_sanitizer
         cvcuda_legacy
-        CUDA::cudart_static
-	    CUDA::cusolver_static
-        CUDA::cublas_static
-        CUDA::cublasLt_static
+        CUDA::cudart_static
+	    CUDA::cusolver
+        CUDA::cublas
+        CUDA::cublasLt
         -lrt
 )
