From 0fbb106b94b90455d30a9bbaff1a333f22d957b3 Mon Sep 17 00:00:00 2001
From: Iain Russell <Iain.Russell@ecmwf.int>
Date: Thu, 25 Jul 2019 09:25:46 +0100
Subject: [PATCH] Pass-CPP-to-rpcgen

---
 src/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git mars-client/src/CMakeLists.txt b/src/CMakeLists.txt
index f64825f..21dcbc3 100644
--- mars-client/src/CMakeLists.txt
+++ mars-client/src/CMakeLists.txt
@@ -180,6 +180,7 @@ else()
 endif()
 
 
+
 if( DEFINED RPCGEN_PATH )
     find_program( RPCGEN_EXECUTABLE NAMES rpcgen PATHS ${RPCGEN_PATH} PATH_SUFFIXES bin NO_DEFAULT_PATH )
 endif()
@@ -189,6 +190,14 @@ if( NOT RPCGEN_EXECUTABLE )
 endif()
 
 
+if(RPCGEN_USE_CPP_ENV AND DEFINED ENV{CPP})
+    get_filename_component(_rpcgen_cpp $ENV{CPP} DIRECTORY)
+    list(APPEND RPCGEN_EXECUTABLE -Y ${_rpcgen_cpp})
+elseif(RPCGEN_USE_CPP_ENV)
+    ecbuild_critical("$CPP not found in environment")
+endif()
+
+
 # targets
 
 configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/rpcmars.x ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x COPYONLY )
-- 
2.13.7

