From 29206ce5c8b9d99b86c15eb8b7bb279714069beb Mon Sep 17 00:00:00 2001
From: Iain Russell <Iain.Russell@ecmwf.int>
Date: Fri, 26 Jul 2019 13:18:06 +0100
Subject: [PATCH] Allow to build without Fortran compiler [METV-2470]

---
 CMakeLists.txt | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index eb26108..687c815 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ set( CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake;${CMAKE_CURRENT_SOURCE
 
 include( ecbuild_bundle )
 
-project( MetviewMiniBundle C CXX Fortran )
+project( MetviewMiniBundle C CXX )
 set(CPACK_PACKAGE_NAME Metview)  # so that the source tarball will have the name 'Metview'
 
 
@@ -30,11 +30,22 @@ endforeach()
 #general
 set(ENABLE_ODB           OFF CACHE BOOL "ODB support")
 set(ENABLE_EXPERIMENTAL  OFF CACHE BOOL "Experimental features")
+set(ENABLE_FORTRAN       NOT_SET CACHE BOOL "Fortran features")
 
 #mars
 set(ECMWF             OFF CACHE BOOL "build the client as for ECMWF")
 set(CONFIG_FDB        OFF CACHE BOOL "database configuration to access FDB")
 set(ENABLE_UDP_STATS  OFF CACHE BOOL  "")
+
+
+# if the user did bot supply a value for ENABLE_FORTRAN, then unset it
+# (we want to then use the default behaviour, per package)
+if(ENABLE_FORTRAN STREQUAL "NOT_SET")
+    unset(ENABLE_FORTRAN)
+    unset(ENABLE_FORTRAN CACHE)
+endif()
+
+
 # -------------------------------------------------------------------------------------
 
 macro(my_bundle_pkg)
-- 
2.13.7

