diff --git a/Makefile b/Makefile
index 5fada13..e450191 100755
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,9 @@
 # LayNii makefile
 
-CC		= c++
-CFLAGS	= -std=c++11 -DHAVE_ZLIB
-LFLAGS	= -lm -lz
+CC      = $(CXX)
+CFLAGS  = $(CPPFLAGS) -std=c++11 -DHAVE_ZLIB
+LFLAGS  = $(LDFLAGS) -lm -lz
+
 # CFLAGS	= -std=c++11 -pedantic -DHAVE_ZLIB -lm -lz
 
 # =============================================================================
