diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0358646..35bd6fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,10 +85,6 @@ if (${CMAKE_Fortran_COMPILER_ID} STREQUAL "GNU")
     set(CMAKE_Fortran_FLAGS_DEBUG
         # Only gtkf-sketcher.f90 needs Fortran 2008 for execute_command_line():
         "-g -pthread -Wall -Wextra -pedantic -std=f2008 -Wtabs -fcheck=all -fbacktrace -Wno-unused-dummy-argument")
-    set(CMAKE_Fortran_FLAGS_RELEASE "-pthread -O3 -mtune=native -march=native")
-    if (UNIX)
-        set(CMAKE_EXE_LINKER_FLAGS "-rdynamic")
-    endif()
 elseif(${CMAKE_Fortran_COMPILER_ID} STREQUAL "Intel")
     # ifort compiler:    set(GTKname "gtk+-${GTKv}.0")
     set(CMAKE_Fortran_FLAGS_DEBUG "-g -warn all -warn nounused")
