--- setup.py    2021-09-20 09:58:42.568513470 +0200
+++ setup.conda.py      2021-09-20 13:48:13.798211550 +0200
@@ -31,7 +31,7 @@
 
 CONF_DIR = '/udocker/etc'
 REQUIREMENTS = []
-SETUP_REQUIREMENTS = ['pytest-runner', ]
+SETUP_REQUIREMENTS = []
 TEST_REQUIREMENTS = ['pytest', ]
 
 setup(
@@ -74,6 +74,5 @@
     test_suite='tests',
     tests_require=TEST_REQUIREMENTS,
     url='https://github.com/indigo-dc/udocker',
-    use_2to3=True,
     zip_safe=False,
 )
