diff -Naur awesome-slugify-1.6.5.orig/setup.py awesome-slugify-1.6.5/setup.py
--- setup.py	2015-06-05 03:29:19.000000000 -0300
+++ setup.py	2015-08-27 21:47:51.072227404 -0300
@@ -1,4 +1,4 @@
-# coding=utf8
+# -*- coding: utf-8 -*-
 
 from setuptools import setup, find_packages
 
diff -Naur awesome-slugify-1.6.5.orig/slugify/alt_translates.py awesome-slugify-1.6.5/slugify/alt_translates.py
--- slugify/alt_translates.py	2014-04-07 00:15:48.000000000 -0300
+++ slugify/alt_translates.py	2015-08-27 21:59:30.308262228 -0300
@@ -1,4 +1,4 @@
-# coding=utf8
+# -*- coding: utf-8 -*-
 
 CYRILLIC = {       # instead of:
     u'ё': u'e',    # io / yo
diff -Naur awesome-slugify-1.6.5.orig/slugify/main.py awesome-slugify-1.6.5/slugify/main.py
--- slugify/main.py	2015-06-05 03:29:19.000000000 -0300
+++ slugify/main.py	2015-08-27 22:01:22.407267811 -0300
@@ -1,4 +1,4 @@
-# coding=utf8
+# -*- coding: utf-8 -*-
 
 import sys
 
diff -Naur awesome-slugify-1.6.5.orig/slugify/tests.py awesome-slugify-1.6.5/slugify/tests.py
--- slugify/tests.py	2015-06-05 03:29:19.000000000 -0300
+++ slugify/tests.py	2015-08-27 22:01:29.084268144 -0300
@@ -1,4 +1,4 @@
-# coding=utf8
+# -*- coding: utf-8 -*-
 
 import unittest
 
