--- htrmopo-0.5.orig/htrmopo/util.py	2026-06-17 11:34:35.250441730 +0200
+++ htrmopo-0.5/htrmopo/util.py	2026-06-17 11:37:04.315245301 +0200
@@ -22,17 +22,17 @@
 ###############################################################################
 
 
-with open(resources.files('htrmopo').joinpath('v1.metadata.schema.json')) as fp:
+with open(resources.files('htrmopo').joinpath('v1.metadata.schema.json'), encoding="utf-8") as fp:
     _v1_schema = json.load(fp)
 
-with open(resources.files('htrmopo').joinpath('v0.metadata.schema.json')) as fp:
+with open(resources.files('htrmopo').joinpath('v0.metadata.schema.json'), encoding="utf-8") as fp:
     _v0_schema = json.load(fp)
 
 ###############################################################################
 # Controlled vocabularies
 ###############################################################################
 
-with open(resources.files('htrmopo').joinpath('iso15924.txt')) as fp:
+with open(resources.files('htrmopo').joinpath('iso15924.txt'), encoding="utf-8") as fp:
     _iso15924 = {}
     for line in fp.readlines():
         try:
@@ -41,7 +41,7 @@
         except Exception:
             continue
 
-with open(resources.files('htrmopo').joinpath('iso639-3.txt')) as fp:
+with open(resources.files('htrmopo').joinpath('iso639-3.txt'), encoding="utf-8") as fp:
     _iso639_3 = {}
     for line in fp.readlines()[1:]:
         try:
@@ -50,7 +50,7 @@
         except Exception:
             continue
 
-with open(resources.files('htrmopo').joinpath('licenses.json')) as fp:
+with open(resources.files('htrmopo').joinpath('licenses.json'), encoding="utf-8") as fp:
     _licenses = json.load(fp)
 
 
