--- tests/utest/test_default_config.py	2021-04-21 08:35:17.236326701 -0400
+++ tests/utest/test_default_config.py.new	2021-04-21 08:41:09.846757177 -0400
@@ -27,12 +27,6 @@
         root = config.find_file_in_project_root('.robocop')
         assert root == src / '.robocop'
 
-    def test_find_project_root_missing_but_git(self, path_to_test_data, config):
-        src = path_to_test_data / 'default_config_missing' / 'nested' / 'deeper'
-        os.chdir(str(src))
-        root = config.find_file_in_project_root('.robocop')
-        assert root == Path(__file__).parent.parent.parent / '.robocop'
-
     def test_load_config_from_default_file(self, path_to_test_data, config):
         src = path_to_test_data / 'default_config'
         os.chdir(str(src))
