diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/ b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/
diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/__init__.py b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/__init__.py
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/__init__.py
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/__init__.py
diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/__init__.py b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/__init__.py
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/__init__.py
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/__init__.py
diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/lr/__init__.py b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/lr/__init__.py
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/lr/__init__.py
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/lr/__init__.py
diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/lr/pr.py b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/lr/pr.py
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/lr/pr.py
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/lr/pr.py
diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/util/__init__.py b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/util/__init__.py
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/util/__init__.py
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/util/__init__.py
diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/util/_base.py b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/util/_base.py
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/fadm/util/_base.py
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/fadm/util/_base.py
diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/predict_lr.py b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/predict_lr.py
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/predict_lr.py
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/predict_lr.py
diff --git a/aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/train_pr.py b/aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/train_pr.py
similarity index 100%
rename from aif360/algorithms/inprocessing/kamfadm-2012ecmlpkdd/train_pr.py
rename to aif360/algorithms/inprocessing/kamfadm_2012ecmlpkdd/train_pr.py
diff --git a/aif360/algorithms/inprocessing/prejudice_remover.py b/aif360/algorithms/inprocessing/prejudice_remover.py
index e70c9ae..59290e2 100644
--- a/aif360/algorithms/inprocessing/prejudice_remover.py
+++ b/aif360/algorithms/inprocessing/prejudice_remover.py
@@ -17,15 +17,15 @@ See: https://github.com/algofairness/fairness-comparison/tree/master/fairness/al
 
 Changes made to fairness-comparison code:
     * removed all files not used by PrejudiceRemover algorithm:
-        - kamfadm-2012ecmlpkdd/data/*
-        - kamfadm-2012ecmlpkdd/fadm/eval/*
-        - kamfadm-2012ecmlpkdd/fadm/nb/*
-        - kamfadm-2012ecmlpkdd/fai_bin_bin.py
-        - kamfadm-2012ecmlpkdd/predict_nb.py
-        - kamfadm-2012ecmlpkdd/train_cv2nb.py
-        - kamfadm-2012ecmlpkdd/train_lr.py
-        - kamfadm-2012ecmlpkdd/train_nb.py
-    * fixed typo in kamfadm-2012ecmlpkdd/fadm/lr/pr.py:244 (typeError -> TypeError)
+        - kamfadm_2012ecmlpkdd/data/*
+        - kamfadm_2012ecmlpkdd/fadm/eval/*
+        - kamfadm_2012ecmlpkdd/fadm/nb/*
+        - kamfadm_2012ecmlpkdd/fai_bin_bin.py
+        - kamfadm_2012ecmlpkdd/predict_nb.py
+        - kamfadm_2012ecmlpkdd/train_cv2nb.py
+        - kamfadm_2012ecmlpkdd/train_lr.py
+        - kamfadm_2012ecmlpkdd/train_nb.py
+    * fixed typo in kamfadm_2012ecmlpkdd/fadm/lr/pr.py:244 (typeError -> TypeError)
     * removed commands.py and instead use subprocess.getoutput
 
 Notes from fairness-comparison's KamishimaAlgorithm.py on changes made to
@@ -61,7 +61,7 @@ original Kamishima code.
     second-to-last column as the sensitive attribute, and pr.py:265-268
     will take the remaining columns as non-sensitive.
 
-The code in kamfadm-2012ecmlpkdd/ is the (fairness-comparison) modified version
+The code in kamfadm_2012ecmlpkdd/ is the (fairness-comparison) modified version
 of the original ECML paper code.
 
 See: changes-to-downloaded-code.diff and KamishimaAlgorithm.py for more details.
@@ -152,7 +152,7 @@ class PrejudiceRemover(Transformer):
         # ADDED FOLLOWING LINE to get absolute path of this file, i.e.
         # prejudice_remover.py
         k_path = os.path.dirname(os.path.abspath(__file__))
-        train_pr = os.path.join(k_path, 'kamfadm-2012ecmlpkdd', 'train_pr.py')
+        train_pr = os.path.join(k_path, 'kamfadm_2012ecmlpkdd', 'train_pr.py')
         # changed paths in the calls below to (a) specify path of train_pr,
         # predict_lr RELATIVE to this file, and (b) compute & use absolute path
         #  and (c) replace python3 with python
@@ -194,7 +194,7 @@ class PrejudiceRemover(Transformer):
         # ADDED FOLLOWING LINE to get absolute path of this file, i.e.
         # prejudice_remover.py
         k_path = os.path.dirname(os.path.abspath(__file__))
-        predict_lr = os.path.join(k_path, 'kamfadm-2012ecmlpkdd', 'predict_lr.py')
+        predict_lr = os.path.join(k_path, 'kamfadm_2012ecmlpkdd', 'predict_lr.py')
         # changed paths in the calls below to (a) specify path of train_pr,
         # predict_lr RELATIVE to this file, and (b) compute & use absolute path,
         # and (c) replace python3 with python
