From f6f10b09edb511f6404d9b42b99c7a87514bc824 Mon Sep 17 00:00:00 2001
From: gmatteo <gmatteo@gmail.com>
Date: Sat, 19 Oct 2019 00:47:54 +0200
Subject: [PATCH] Update readme

---
 abipy/gui/mixins.py   |  2 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/abipy/gui/mixins.py b/abipy/gui/mixins.py
index 334b797d8..79c94605d 100644
--- a/abipy/gui/mixins.py
+++ b/abipy/gui/mixins.py
@@ -342,7 +342,7 @@ def OnNetcdf_WxNcView(self, event):
         NcViewerFrame(self, filepaths=self.nc_filepaths).Show()
 
 
-class Has_Phbands(metaclass=abc.ABCMeta)
+class Has_Phbands(metaclass=abc.ABCMeta):
     """
     Mixin class that provides a menu and callbacks for analyzing phonon bands.
     """
