diff --git a/src/datatrove/tools/check_dataset.py b/src/datatrove/tools/check_dataset.py
index 763e5c3..507dc6f 100644
--- a/src/datatrove/tools/check_dataset.py
+++ b/src/datatrove/tools/check_dataset.py
@@ -94,10 +94,13 @@ def check_dataset(input_folder: DataFolder, tokenizer: str = "gpt2", eos_token:
             assert last_token == eos_token, f"no EOS at doc end of doc {doci}"
 
 
-if __name__ == "__main__":
+def main():
     args = parser.parse_args()
 
     input_folder: DataFolder = get_datafolder(args.data)
 
     check_dataset(input_folder, args.tokenizer, args.eos)
     print("All checks ok")
+
+if __name__ == "__main__":
+    main()
