diff --git a/sdk/core/azure-core/vcpkg/Config.cmake.in b/sdk/core/azure-core/vcpkg/Config.cmake.in
index 412cce6d..e1f1d037 100644
--- a/sdk/core/azure-core/vcpkg/Config.cmake.in
+++ b/sdk/core/azure-core/vcpkg/Config.cmake.in
@@ -10,10 +10,6 @@ if(@BUILD_TRANSPORT_CURL@)
   find_dependency(CURL @CURL_MIN_REQUIRED_VERSION@)
 endif()
 
-if(@BUILD_TRANSPORT_WINHTTP@)
-  find_dependency(wil)
-endif()
-
 if (NOT WIN32)
   find_dependency(OpenSSL)
 endif()
