diff --git a/CMakeLists.txt b/CMakeLists.txt
index d28c114..723c343 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,7 @@ find_package(GTSAM 4.2 REQUIRED)
 find_package(gtsam_points 1.2.0 REQUIRED)
 find_package(OpenMP REQUIRED)
 find_package(spdlog REQUIRED)
+find_package(nlohmann_json REQUIRED)
 
 if(BUILD_WITH_VIEWER)
   find_package(Iridescence REQUIRED)
@@ -107,7 +108,6 @@ add_library(glim SHARED
 )
 target_include_directories(glim PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/json/include>
   $<INSTALL_INTERFACE:include/glim>
   ${OpenCV_INCLUDE_DIRS}
 )
@@ -118,6 +118,7 @@ target_link_libraries(glim
   gtsam
   gtsam_points::gtsam_points
   OpenMP::OpenMP_CXX
   spdlog::spdlog
+  nlohmann_json::nlohmann_json
   $<TARGET_NAME_IF_EXISTS:Iridescence::Iridescence>
   ${OpenCV_LIBRARIES}
 )
diff --git a/cmake/glim-config.cmake.in b/cmake/glim-config.cmake.in
index 64f51e4..514c62c 100644
--- a/cmake/glim-config.cmake.in
+++ b/cmake/glim-config.cmake.in
@@ -23,6 +23,7 @@ find_dependency(gtsam_points REQUIRED)
 find_dependency(GTSAM REQUIRED)
 find_dependency(OpenMP REQUIRED)
 find_dependency(spdlog REQUIRED)
+find_dependency(nlohmann_json REQUIRED)
 
 if(BUILD_WITH_CUDA)
   find_dependency(CUDAToolkit REQUIRED)
