--- summa-2.0.0.orig/build/Makefile	2019-09-27 08:57:44.801473676 -0300
+++ summa-2.0.0/build/Makefile	2019-09-27 09:07:36.468901838 -0300
@@ -117,7 +117,7 @@
 MOD_PATH = $(F_MASTER)/build
 
 # Define the directory for the executables
-EXE_PATH = $(F_MASTER)/bin
+# EXE_PATH = $(F_MASTER)/bin
 
 #========================================================================
 # PART 2: Assemble all of the SUMMA sub-routines
@@ -347,7 +347,7 @@
 
 # link routines
 link:
-	$(FC_EXE) -pg -g *.o $(LIBNETCDF) $(LIBLAPACK) -o $(DRIVER__EX)
+	$(FC_EXE) -g *.o $(LIBNETCDF) $(LIBLAPACK) -o $(DRIVER__EX)
 
 # Remove object files
 clean:
