--- a/src/backports/zoneinfo/_tzpath.py
+++ b/src/backports/zoneinfo/_tzpath.py
@@ -22,17 +22,8 @@ def reset_tzpath(to=None):
         env_var = os.environ.get("PYTHONTZPATH", None)
         if env_var is not None:
             base_tzpath = _parse_python_tzpath(env_var)
-        elif sys.platform != "win32":
-            base_tzpath = [
-                "/usr/share/zoneinfo",
-                "/usr/lib/zoneinfo",
-                "/usr/share/lib/zoneinfo",
-                "/etc/zoneinfo",
-            ]
-
-            base_tzpath.sort(key=lambda x: not os.path.exists(x))
         else:
-            base_tzpath = ()
+            base_tzpath = [os.path.join(sys.base_prefix, "share", "zoneinfo")]

     TZPATH = tuple(base_tzpath)

--
