From 9dc68de62170695e7cb653efd2a7ff7721bbcb36 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 25 Apr 2025 11:34:39 +0200
Subject: [PATCH] patch license

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 3f12642..3e4e300 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -14,7 +14,7 @@ authors = [
   { name = "Marnik Bercx", email = "marnik.bercx@psi.ch" },
   { name = "Christina Ertural", email = "christina.ertural@bam.de" },
 ]
-license = { file = "../LICENSE" }
+license = { file = "LICENSE" }
 dependencies = [
   "aiida-workgraph>=0.5.1,<=0.5.2",
   "numpy>=1.21,<2",
