diff --git a/CMakeLists.txt b/CMakeLists.txt
index 25adacc..4fc1ed1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -263,6 +263,8 @@ else()
 endif()
 
 FIND_PACKAGE(ZLIB REQUIRED QUIET)
+FIND_PACKAGE(PkgConfig REQUIRED)
+PKG_CHECK_MODULES(LZ4 REQUIRED IMPORTED_TARGET liblz4)
 
-FIND_PACKAGE(SQLite3 QUIET)
+FIND_PACKAGE(SQLite3 REQUIRED)
 IF(SQLite3_FOUND)
diff --git a/corelib/src/CMakeLists.txt b/corelib/src/CMakeLists.txt
index 6969c5c..b706bb5 100644
--- a/corelib/src/CMakeLists.txt
+++ b/corelib/src/CMakeLists.txt
@@ -128,8 +128,6 @@ SET(SRC_FILES
     
     GainCompensator.cpp
 
-    rtflann/ext/lz4.c
-    rtflann/ext/lz4hc.c
     FlannIndex.cpp
     
     #clams stuff
@@ -176,6 +174,7 @@ SET(PUBLIC_INCLUDE_DIRS
 
 SET(LIBRARIES
 	${ZLIB_LIBRARIES}
+	PkgConfig::LZ4
 )
 
 # Issue that qhull dependency uses optimized and debug keywords,
diff --git a/corelib/src/rtflann/util/serialization.h b/corelib/src/rtflann/util/serialization.h
index 8c25166..8099339 100644
--- a/corelib/src/rtflann/util/serialization.h
+++ b/corelib/src/rtflann/util/serialization.h
@@ -6,8 +6,8 @@
 #include <cstdlib>
 #include <cstring>
 #include <stdio.h>
-#include "rtflann/ext/lz4.h"
-#include "rtflann/ext/lz4hc.h"
+#include <lz4.h>
+#include <lz4hc.h>
 
 
 namespace rtflann
