diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e0a184..f35b45b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -322,7 +322,11 @@ ENDIF(ZLIB_FOUND)
 
 SET(ADD_VTK_GUI_SUPPORT_QT_TO_CONF FALSE)
 IF(WITH_QT)
+  IF(PCL_VTK_COMPONENTS)
+    FIND_PACKAGE(VTK COMPONENTS ${PCL_VTK_COMPONENTS})
+  ELSE()
     FIND_PACKAGE(VTK)
+  ENDIF()
     IF(NOT VTK_FOUND)
       MESSAGE(FATAL_ERROR "VTK is required when using Qt. Set -DWITH_QT=OFF if you don't want gui tools.")
     ENDIF(NOT VTK_FOUND)
diff --git a/utilite/CMakeLists.txt b/utilite/CMakeLists.txt
index b152c0f..7bc18f3 100644
--- a/utilite/CMakeLists.txt
+++ b/utilite/CMakeLists.txt
@@ -2,7 +2,8 @@ SET(UTILITE_VERSION "0.3.0")
 ADD_DEFINITIONS(-DUTILITE_VERSION="${UTILITE_VERSION}")
 
 if(UNIX AND NOT ANDROID)
-   FIND_PACKAGE(Pthreads REQUIRED)
+   SET(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../cmake_modules;${CMAKE_MODULE_PATH}")
+   FIND_PACKAGE(Pthreads REQUIRED MODULE)
 ENDIF(UNIX AND NOT ANDROID)
 
 ADD_SUBDIRECTORY( src )
diff --git a/utilite/src/CMakeLists.txt b/utilite/src/CMakeLists.txt
--- a/utilite/src/CMakeLists.txt
+++ b/utilite/src/CMakeLists.txt
@@ -20,8 +20,8 @@ generate_export_header(rtabmap_utilite
   BASE_NAME utilite)
 
 target_include_directories(rtabmap_utilite PUBLIC 
-  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include;${CMAKE_CURRENT_BINARY_DIR}/include;${PTHREADS_INCLUDE_DIR}>"
-  "$<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR};${PTHREADS_INCLUDE_DIR}>")
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include;${CMAKE_CURRENT_BINARY_DIR}/include;${PTHREADS_INCLUDE_DIR}>"
+  "$<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>")
 
 IF(MINGW)
     TARGET_LINK_LIBRARIES(rtabmap_utilite PRIVATE ${PTHREADS_LIBRARY} "-lpsapi")
diff --git a/corelib/src/CMakeLists.txt b/corelib/src/CMakeLists.txt
--- a/corelib/src/CMakeLists.txt
+++ b/corelib/src/CMakeLists.txt
@@ -883,8 +883,7 @@ target_include_directories(rtabmap_core PUBLIC
   "$<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>")
 
 target_include_directories(rtabmap_core SYSTEM PUBLIC 
-  "$<BUILD_INTERFACE:${PUBLIC_INCLUDE_DIRS};${INCLUDE_DIRS}>"
-  "$<INSTALL_INTERFACE:${PUBLIC_INCLUDE_DIRS}>")
+  "$<BUILD_INTERFACE:${PUBLIC_INCLUDE_DIRS};${INCLUDE_DIRS}>")
   
 TARGET_LINK_LIBRARIES(rtabmap_core 
   PUBLIC 
diff --git a/guilib/src/CMakeLists.txt b/guilib/src/CMakeLists.txt
--- a/guilib/src/CMakeLists.txt
+++ b/guilib/src/CMakeLists.txt
@@ -202,8 +202,7 @@ target_include_directories(rtabmap_gui PUBLIC
   "$<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>")
 
 target_include_directories(rtabmap_gui SYSTEM PUBLIC 
-  "$<BUILD_INTERFACE:${PUBLIC_INCLUDE_DIRS};${INCLUDE_DIRS}>"
-  "$<INSTALL_INTERFACE:${PUBLIC_INCLUDE_DIRS}>")
+  "$<BUILD_INTERFACE:${PUBLIC_INCLUDE_DIRS};${INCLUDE_DIRS}>")
 
 TARGET_LINK_LIBRARIES(rtabmap_gui 
   PUBLIC
