From be69d3c157557ae01e6b9e78beec4fe271632d8c Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Thu, 28 Oct 2021 11:28:13 +0200
Subject: [PATCH 3/8] Parametrize LIBPROTOBUF_VERSION

---
 WORKSPACE | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/WORKSPACE b/WORKSPACE
index 65a5f81..486b7b0 100755
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -124,8 +124,8 @@ load("@rules_jvm_external//:defs.bzl", "maven_install")
 # TODO: Use installed protobuf version
 maven_install(
     artifacts = [
-        "com.google.protobuf:protobuf-java:3.15.6",
-        "com.google.protobuf:protobuf-java-util:3.15.6",
+        "com.google.protobuf:protobuf-java:LIBPROTOBUF_VERSION",
+        "com.google.protobuf:protobuf-java-util:LIBPROTOBUF_VERSION",
     ],
     repositories = [
         "https://repo1.maven.org/maven2",
-- 
2.35.0

