diff --git a/jupyterhub_traefik_proxy/proxy.py b/jupyterhub_traefik_proxy/proxy.py
index ddbefbe..28c218d 100644
--- a/jupyterhub_traefik_proxy/proxy.py
+++ b/jupyterhub_traefik_proxy/proxy.py
@@ -153,7 +153,7 @@ class TraefikProxy(Proxy):
     )
 
     traefik_api_url = Unicode(
-        "http://localhost:8099",
+        "http://127.0.0.1:8099",
         config=True,
         help="""traefik authenticated api endpoint url""",
     )
diff --git a/tests/conftest.py b/tests/conftest.py
index e35c0da..c8dfed5 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -224,13 +224,13 @@ def _make_etcd_proxy(*, auth=False, client_ca=None, **extra_kwargs):
         client_ca = str(client_ca)
         kwargs.update(
             dict(
-                etcd_url="https://localhost:2379",
+                etcd_url="https://127.0.0.1:2379",
                 etcd_username=Config.etcd_user,
                 etcd_password=Config.etcd_password,
                 etcd_client_kwargs=dict(
                     grpc_options=[
-                        ("grpc.ssl_target_name_override", "localhost"),
-                        ("grpc.default_authority", "localhost"),
+                        ("grpc.ssl_target_name_override", "127.0.0.1"),
+                        ("grpc.default_authority", "127.0.0.1"),
                     ],
                     ca_cert=client_ca,
                 ),
@@ -550,16 +550,16 @@ async def launch_etcd_auth(etcd_ssl_key_cert, etcd_client_ca):
             f"--peer-trusted-ca-file={etcd_client_ca}",
             f"--cert-file={cert}",
             f"--key-file={key}",
-            "--initial-cluster=default=https://localhost:2380",
-            "--initial-advertise-peer-urls=https://localhost:2380",
-            "--listen-peer-urls=https://localhost:2380",
-            "--listen-client-urls=https://localhost:2379",
-            "--advertise-client-urls=https://localhost:2379",
+            "--initial-cluster=default=https://127.0.0.1:2380",
+            "--initial-advertise-peer-urls=https://127.0.0.1:2380",
+            "--listen-peer-urls=https://127.0.0.1:2380",
+            "--listen-client-urls=https://127.0.0.1:2379",
+            "--advertise-client-urls=https://127.0.0.1:2379",
             "--log-level=debug",
         ],
     )
     etcdctl_args = [
-        "--endpoints=localhost:2379",
+        "--endpoints=127.0.0.1:2379",
         "--user",
         f"{Config.etcd_user}:{Config.etcd_password}",
         f"--cacert={etcd_client_ca}",
@@ -575,11 +575,11 @@ async def launch_etcd_auth(etcd_ssl_key_cert, etcd_client_ca):
         c = etcd3.client(
             user=Config.etcd_user,
             password=Config.etcd_password,
-            host="localhost",
+            host="127.0.0.1",
             port=2379,
             grpc_options=[
-                ("grpc.ssl_target_name_override", "localhost"),
-                ("grpc.default_authority", "localhost"),
+                ("grpc.ssl_target_name_override", "127.0.0.1"),
+                ("grpc.default_authority", "127.0.0.1"),
             ],
             ca_cert=etcd_client_ca,
         )
