--- v4-client-cpp/examples/composite_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
+++ v4-client-cpp/examples/composite_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
@@ -1,2 +1,3 @@
-find_package(json REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(absl REQUIRED)
 
@@ -5,2 +6,2 @@
 target_include_directories(dydx_v4_composite_client_example PRIVATE ${INCLUDE_DIR})
-target_link_libraries(dydx_v4_composite_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json)
\ No newline at end of file
+target_link_libraries(dydx_v4_composite_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json absl_synchronization absl_log_internal_message absl_base)
--- v4-client-cpp/examples/faucet_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
+++ v4-client-cpp/examples/faucet_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
@@ -1,2 +1,3 @@
-find_package(json REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(absl REQUIRED)
 
@@ -5,2 +6,2 @@
 target_include_directories(dydx_v4_faucet_client_example PRIVATE ${INCLUDE_DIR})
-target_link_libraries(dydx_v4_faucet_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json)
\ No newline at end of file
+target_link_libraries(dydx_v4_faucet_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json absl_synchronization absl_log_internal_message absl_base)
--- v4-client-cpp/examples/indexer_rest_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
+++ v4-client-cpp/examples/indexer_rest_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
@@ -1,2 +1,3 @@
-find_package(json REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(absl REQUIRED)
 
@@ -5,2 +6,2 @@
 target_include_directories(dydx_v4_indexer_rest_client_example PRIVATE ${INCLUDE_DIR})
-target_link_libraries(dydx_v4_indexer_rest_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json)
\ No newline at end of file
+target_link_libraries(dydx_v4_indexer_rest_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json absl_synchronization absl_log_internal_message absl_base)
--- v4-client-cpp/examples/indexer_ws_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
+++ v4-client-cpp/examples/indexer_ws_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
@@ -1,2 +1,3 @@
-find_package(json REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(absl REQUIRED)
 
@@ -5,2 +6,2 @@
 target_include_directories(dydx_v4_indexer_ws_client_example PRIVATE ${INCLUDE_DIR})
-target_link_libraries(dydx_v4_indexer_ws_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json)
\ No newline at end of file
+target_link_libraries(dydx_v4_indexer_ws_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json absl_synchronization absl_log_internal_message absl_base)
--- v4-client-cpp/examples/node_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
+++ v4-client-cpp/examples/node_client/CMakeLists.txt    2024-03-26 18:58:46.466212418 -0500
@@ -1,2 +1,3 @@
-find_package(json REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(absl REQUIRED)
 
@@ -5,2 +6,2 @@
 target_include_directories(dydx_v4_node_client_example PRIVATE ${INCLUDE_DIR})
-target_link_libraries(dydx_v4_node_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json)
+target_link_libraries(dydx_v4_node_client_example dydx_v4_client_lib nlohmann_json::nlohmann_json absl_synchronization absl_log_internal_message absl_base)
