--- v4-client-cpp/lib/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -3,3 +3,3 @@
 # Proto files
-add_subdirectory(proto)
+find_package(dydx_v4_proto)

@@ -12,5 +13,59 @@
 file(GLOB_RECURSE SOURCES ${SOURCE_DIR}/*.cpp)
-add_library(dydx_v4_client_lib ${SOURCES})
-target_include_directories(dydx_v4_client_lib PUBLIC ${INCLUDE_DIR})
-target_link_libraries(dydx_v4_client_lib PUBLIC dydx_v4_proto OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
+
+# Create an object library
+add_library(dydx_v4_client_lib_obj OBJECT ${SOURCES})
+set_property(TARGET dydx_v4_client_lib_obj PROPERTY POSITION_INDEPENDENT_CODE 1)
+target_include_directories(dydx_v4_client_lib_obj PUBLIC ${INCLUDE_DIR} ${dydx_v4_proto_INCLUDE_DIRS})
+
+# Create shared and static libraries from the object library
+add_library(dydx_v4_client_lib SHARED $<TARGET_OBJECTS:dydx_v4_client_lib_obj>)
+set_target_properties(dydx_v4_client_lib PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+target_link_libraries(dydx_v4_client_lib PUBLIC dydx_v4_proto Boost::boost OpenSSL::SSL crypto bip3x fmt nlohmann_json::nlohmann_json)
+
+# Install the libraries
+include(GNUInstallDirs)
+install(TARGETS dydx_v4_client_lib EXPORT dydx_v4_client_lib-config DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT client)
+install(DIRECTORY include/common DESTINATION include/dydx_v4_client COMPONENT client)
+install(DIRECTORY include/dydx_v4_futures DESTINATION include/dydx_v4_client COMPONENT client)
+#install(EXPORT dydx_v4_client_lib-config DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/dydx_v4_client_lib COMPONENT client)
+include(CMakePackageConfigHelpers)
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/dydx_v4_client_lib-config.cmake.in
+    "
+include(CMakeFindDependencyMacro)
+find_dependency(dydx_v4_proto)
+find_dependency(Boost)
+find_dependency(OpenSSL)
+find_dependency(bip3x)
+find_dependency(fmt)
+find_dependency(nlohmann_json)
+@PACKAGE_INIT@
+set_and_check(dydx_v4_client_lib_INCLUDE_DIRS \"\${PACKAGE_PREFIX_DIR}/include/dydx_v4_client/common\" \"\${PACKAGE_PREFIX_DIR}/include/dydx_v4_client/dydx_v4_futures\")
+set_and_check(dydx_v4_client_lib_LIBRARIES \"\${PACKAGE_PREFIX_DIR}/lib/libdydx_v4_client_lib${CMAKE_SHARED_LIBRARY_SUFFIX}\")
+"
+)
+configure_package_config_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/dydx_v4_client_lib-config.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/dydx_v4_client_lib-config.cmake
+  INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/dydx_v4_client_lib)
+write_basic_package_version_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/dydx_v4_client_lib-config-version.cmake
+  VERSION ${PROJECT_VERSION}
+  COMPATIBILITY SameMajorVersion )
+install(
+  FILES ${CMAKE_CURRENT_BINARY_DIR}/dydx_v4_client_lib-config.cmake
+        ${CMAKE_CURRENT_BINARY_DIR}/dydx_v4_client_lib-config-version.cmake
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/dydx_v4_client_lib
+  COMPONENT client)
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/dydx_v4_client_lib.pc
+    "
+Name: dydx_v4_client_lib
+Description: DYDX C++ client library
+Version: ${PROJECT_VERSION}
+Requires: bip3x >= 3.0, fmt >= 8.0, nlohmann_json >= 3.11.3, OpenSSL >= 3, Boost >= 1.82.0
+Libs: -L${CMAKE_INSTALL_PREFIX}/lib -ldydx_v4_client_lib
+Cflags: -I${CMAKE_INSTALL_PREFIX}/include/dydx_v4_client
+    "
+)
+
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/dydx_v4_client_lib.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig COMPONENT client)

@@ -20,3 +60,9 @@
 add_executable(dydx_v4_client_lib_tests ${TEST_SOURCES})
-target_link_libraries(dydx_v4_client_lib_tests PRIVATE Catch2::Catch2WithMain dydx_v4_client_lib)
+#set(Boost_USE_STATIC_LIBS ON)
+#unset(Boost_DIR_CACHE)
+#find_package(Boost 1.82.0 REQUIRED)
+#add_library(dydx_v4_client_lib_static STATIC $<TARGET_OBJECTS:dydx_v4_client_lib_obj>)
+#target_link_libraries(dydx_v4_client_lib_static PUBLIC dydx_v4_proto Boost::boost OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
+#target_link_libraries(dydx_v4_client_lib_tests PRIVATE Catch2::Catch2WithMain dydx_v4_client_lib_static Boost::boost crypto OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
+#target_include_directories(dydx_v4_client_lib_tests PUBLIC ${INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR}/proto)
 catch_discover_tests(dydx_v4_client_lib_tests)
--- v4-client-cpp/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -24,2 +24,2 @@

-add_subdirectory(lib/proto)
+add_subdirectory(lib)
