--- v4-client-cpp/lib/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -75,9 +75,9 @@
 add_executable(dydx_v4_client_lib_tests ${TEST_SOURCES})
-#set(Boost_USE_STATIC_LIBS ON)
-#unset(Boost_DIR_CACHE)
-#find_package(Boost 1.82.0 REQUIRED)
-#add_library(dydx_v4_client_lib_static STATIC $<TARGET_OBJECTS:dydx_v4_client_lib_obj>)
-#target_link_libraries(dydx_v4_client_lib_static PUBLIC dydx_v4_proto Boost::boost OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
-#target_link_libraries(dydx_v4_client_lib_tests PRIVATE Catch2::Catch2WithMain dydx_v4_client_lib_static Boost::boost crypto OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
-#target_include_directories(dydx_v4_client_lib_tests PUBLIC ${INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR}/proto)
+set(Boost_USE_STATIC_LIBS ON)
+unset(Boost_DIR_CACHE)
+find_package(Boost 1.82.0 REQUIRED)
+add_library(dydx_v4_client_lib_static STATIC $<TARGET_OBJECTS:dydx_v4_client_lib_obj>)
+target_link_libraries(dydx_v4_client_lib_static PUBLIC dydx_v4_proto Boost::boost OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
+target_link_libraries(dydx_v4_client_lib_tests PRIVATE Catch2::Catch2WithMain dydx_v4_client_lib_static Boost::boost crypto OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
+target_include_directories(dydx_v4_client_lib_tests PUBLIC ${INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR}/proto)
 catch_discover_tests(dydx_v4_client_lib_tests)
