diff --git a/src/Mod/Drawing/Gui/DrawingView.cpp b/src/Mod/Drawing/Gui/DrawingView.cpp
index bbfb05465..90d3677f4 100644
--- a/src/Mod/Drawing/Gui/DrawingView.cpp
+++ b/src/Mod/Drawing/Gui/DrawingView.cpp
@@ -306,7 +306,7 @@ void DrawingView::findPrinterSettings(const QString& fileName)
     pageSizes[QPrinter::A7] = QString::fromLatin1("A7");
     pageSizes[QPrinter::A8] = QString::fromLatin1("A8");
     pageSizes[QPrinter::A9] = QString::fromLatin1("A9");
-    pageSizes[QPrinter::B0] = QString::fromLatin1("B0");
+    // pageSizes[QPrinter::B0] = QString::fromLatin1("B0");
     pageSizes[QPrinter::B1] = QString::fromLatin1("B1");
     pageSizes[QPrinter::B2] = QString::fromLatin1("B2");
     pageSizes[QPrinter::B3] = QString::fromLatin1("B3");
@@ -627,7 +627,7 @@ QPrinter::PageSize DrawingView::getPageSize(int w, int h) const
         {74, 105}, // A7
         {52, 74}, // A8
         {37, 52}, // A8
-        {1000, 1414}, // B0
+        // {1000, 1414}, // B0
         {707, 1000}, // B1
         {31, 44}, // B10
         {500, 707}, // B2
@@ -646,7 +646,7 @@ QPrinter::PageSize DrawingView::getPageSize(int w, int h) const
     };
 
     QPrinter::PageSize ps = QPrinter::Custom;
-    for (int i=0; i<30; i++) {
+    for (int i=0; i<29; i++) {
         if (std::abs(paperSizes[i][0]-w) <= 1 &&
             std::abs(paperSizes[i][1]-h) <= 1) {
             ps = static_cast<QPrinter::PageSize>(i);
