diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1cfe238..43a9fb5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,7 +109,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
     INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
 
     # get linker errors as soon as possible and not at runtime e.g. for modules
-    if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT BUILD_WITH_CONDA)
         set(CMAKE_SHARED_LINKER_FLAGS "-Wl,-undefined,error")
     elseif(UNIX)
         set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined")
@@ -199,6 +199,7 @@ OPTION(FREECAD_USE_EXTERNAL_SMESH "Use system installed smesh instead of the bun
 OPTION(FREECAD_USE_EXTERNAL_KDL "Use system installed orocos-kdl instead of the bundled." OFF)
 OPTION(FREECAD_USE_FREETYPE "Builds the features using FreeType libs" ON)
 OPTION(FREECAD_BUILD_DEBIAN "Prepare for a build of a Debian package" OFF)
+OPTION(BUILD_WITH_CONDA "Set ON if you build freecad with conda" OFF)
 if (WIN32 OR APPLE)
     OPTION(FREECAD_USE_QT_FILEDIALOG "Use Qt's file dialog instead of the native one." OFF)
 else()
@@ -449,7 +450,7 @@ if(NOT FREECAD_LIBPACK_USE OR FREECAD_LIBPACK_CHECKFILE_CLBUNDLER)
 set(Python_ADDITIONAL_VERSIONS "2.7")
 
 # For building on OS X
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT BUILD_WITH_CONDA)
 
     # If the user doesn't tell us which package manager they're using
     if(NOT DEFINED MACPORTS_PREFIX AND NOT DEFINED HOMEBREW_PREFIX)
@@ -559,7 +560,7 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     endif(NOT DEFINED PYTHON_LIBRARY)
 
 
-endif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+endif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT BUILD_WITH_CONDA)
 
     find_package(PythonInterp REQUIRED)
     set(Python_ADDITIONAL_VERSIONS ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR})
diff --git a/cMake/FindOpenCasCade.cmake b/cMake/FindOpenCasCade.cmake
index 9944b8a..296b76c 100644
--- a/cMake/FindOpenCasCade.cmake
+++ b/cMake/FindOpenCasCade.cmake
@@ -56,7 +56,10 @@ else(OCE_FOUND) #look for OpenCASCADE
       )
     endif(CYGWIN OR MINGW)
   else(WIN32)
+    MESSAGE(STATUS "################################")
+    MESSAGE(STATUS ${CMAKE_PREFIX_PATH}/include/opencascade)
     FIND_PATH(OCC_INCLUDE_DIR Standard_Version.hxx
+      ${CMAKE_PREFIX_PATH}/include/opencascade
       /usr/include/opencascade
       /usr/local/include/opencascade
       /opt/opencascade/include
diff --git a/src/Mod/Part/App/CMakeLists.txt b/src/Mod/Part/App/CMakeLists.txt
index fc33a8f..1fa6c4d 100644
--- a/src/Mod/Part/App/CMakeLists.txt
+++ b/src/Mod/Part/App/CMakeLists.txt
@@ -22,6 +22,10 @@ include_directories(
     ${FREETYPE_INCLUDE_DIRS}
 )
 
+message(STATS "#####################")
+message(STATUS ${OCC_LIBRARY_DIR})
+message(STATUS "####################")
+
 link_directories(${OCC_LIBRARY_DIR})
 
 set(Part_LIBS 
