diff --git a/CMakeLists.txt b/CMakeLists.txt
index 353940dcc..ca055cfb4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -748,7 +748,13 @@ endif()
         else(NOT FREECAD_USE_EXTERNAL_SMESH)
             find_package(SMESH CONFIG)
             set (SMESH_INCLUDE_DIR ${SMESH_INCLUDE_PATH})
-            SET(EXTERNAL_SMESH_LIBS ${SMESH_LIBRARIES})
+            foreach(LIB ${SMESH_LIBRARIES})
+                set(FOUND_LIB "FOUND_LIB-NOTFOUND")
+                find_library(FOUND_LIB ${LIB} PATHS ${SMESH_LIB_PATH})
+                list(APPEND SMESH_ABS_LIBS ${FOUND_LIB})
+            endforeach(LIB)
+            unset(FOUND_LIB CACHE)
+            set(EXTERNAL_SMESH_LIBS ${SMESH_ABS_LIBS})
             if(NOT SMESH_FOUND)
                 message(ERROR "================\n"
                               "SMESH not found.\n"
