commit 0a1d9e582e36e7e1e8cdbc0fad59733b9358c08e
Author: looooo <sppedflyer@gmail.com>
Date:   Wed May 30 10:31:29 2018 +0200

    fix windows problem:
    https://forum.freecadweb.org/viewtopic.php?f=4&t=29014

diff --git a/src/Main/MainPy.cpp b/src/Main/MainPy.cpp
index e8e6965..7ea3a26 100644
--- a/src/Main/MainPy.cpp
+++ b/src/Main/MainPy.cpp
@@ -60,7 +60,7 @@ BOOL APIENTRY DllMain(HANDLE hModule, DWORD  ul_reason_for_call, LPVOID lpReserv
     case DLL_PROCESS_ATTACH: {
         // This name is preliminary, we pass it to Application::init() in initFreeCAD()
         // which does the rest.
-        char  szFileName [MAX_PATH];
+        TCHAR  szFileName [MAX_PATH];
         GetModuleFileName((HMODULE)hModule, szFileName, MAX_PATH-1);
         App::Application::Config()["AppHomePath"] = szFileName;
     }
@@ -97,7 +97,7 @@ PyMOD_INIT_FUNC(FreeCAD)
     argv[0][MAX_PATH-1] = '\0'; // ensure null termination
 #elif defined(FC_OS_CYGWIN)
     HMODULE hModule = GetModuleHandle("FreeCAD.dll");
-    char szFileName [MAX_PATH];
+    TCHAR szFileName [MAX_PATH];
     GetModuleFileName(hModule, szFileName, MAX_PATH-1);
     argv[0] = (char*)malloc(MAX_PATH);
     strncpy(argv[0],szFileName,MAX_PATH);
