diff --git a/src/DockContainerWidget.h b/src/DockContainerWidget.h
index 06ea710..5f4e668 100644
--- a/src/DockContainerWidget.h
+++ b/src/DockContainerWidget.h
@@ -107,7 +107,7 @@ protected:
 	/**
 	 * Adds the given dock area to this container widget
 	 */
-	void addDockArea(CDockAreaWidget* DockAreaWidget, DockWidgetArea area = CenterDockWidgetArea);
+	void addDockArea(CDockAreaWidget* DockAreaWidget, DockWidgetArea area = ads::CenterDockWidgetArea);
 
 	/**
 	 * Removes the given dock area from this container
