diff --git a/setup.py b/setup.py
index 9f50f5e..30fd57b 100644
--- a/setup.py
+++ b/setup.py
@@ -28,7 +28,7 @@ setup(name='gym',
                 if package.startswith('gym')],
       zip_safe=False,
       install_requires=[
-          'scipy', 'numpy>=1.10.4', 'six', 'pyglet>=1.4.0,<=1.5.0', 'cloudpickle>=1.2.0,<1.4.0',
+          'scipy', 'numpy>=1.10.4', 'mujoco_py', 'six', 'atari_py', 'pyglet>=1.4.0,<=1.5.0', 'cloudpickle>=1.2.0,<1.4.0',
           'enum34~=1.1.6;python_version<"3.4"',
       ],
       extras_require=extras,
