diff --git a/webhelpers/html/builder.py b/webhelpers/html/builder.py
--- a/webhelpers/html/builder.py
+++ b/webhelpers/html/builder.py
@@ -156,8 +156,23 @@
 
 """
 import re
-from urllib import quote as url_escape
-from UserDict import DictMixin
+try:
+    from urllib import quote as url_escape # Python 2.X
+except ImportError:
+    from urllib.parse import quote as url_escape # Python 3+
+#from urllib import quote as url_escape
+
+try:
+    from UserDict import DictMixin # Python 2.X
+except ImportError:
+    from collections import MutableMapping as DictMixin # Python 3+
+
+    def __len__(self):
+        return len(self.mylist)
+
+    def __iter__(self):
+        for i in self.mylist:
+            yield i
 
 import markupsafe
 try:
@@ -293,7 +308,8 @@
 
 
 def make_tag(tag, *args, **kw):
-    if kw.has_key("c"):
+#    if kw.has_key("c"):
+    if "c" in kw:
         assert not args, "The special 'c' keyword argument cannot be used "\
 "in conjunction with non-keyword arguments"
         args = kw.pop("c")
@@ -332,7 +348,8 @@
     literal(u'')
     """
     strings = [u' %s="%s"' % (_attr_decode(attr), escape(value))
-        for attr, value in sorted(attrs.iteritems())
+#        for attr, value in sorted(attrs.iteritems())
+        for attr, value in sorted(attrs.items())       
         if value is not None]
     return literal("".join(strings))
 
