--- CMakeLists.txt	Mon Nov 25 12:07:31 2013
+++ CMakeLists.txt	Wed Dec 04 18:11:10 2013
@@ -144,6 +144,8 @@
 if (SNAPPY_FOUND AND ZLIB_FOUND)  # Snappy borrows crc32 from zlib
     set(SNAPPY_PKG libsnappy)
     add_definitions(-DSNAPPY_CODEC)
+    include_directories(${SNAPPY_INCLUDE_DIR})
+    link_directories(${SNAPPY_LIBRARIES})
     message("Enabled snappy codec")
 else (SNAPPY_FOUND AND ZLIB_FOUND)
     set(SNAPPY_PKG "")
