diff --git a/build/cmake/CondorPackageConfig.cmake b/build/cmake/CondorPackageConfig.cmake
index 88ae8154cd..cc774f0e58 100644
--- a/build/cmake/CondorPackageConfig.cmake
+++ b/build/cmake/CondorPackageConfig.cmake
@@ -125,7 +125,7 @@ set( C_BIN                  bin)
 set( C_LIB                     lib)
 set( C_LIB_PUBLIC              lib)
 set( C_LIB32           lib)
-set( C_LIBEXEC         libexec )
+set( C_LIBEXEC         libexec/condor )
 set( C_SBIN                    sbin)

 set( C_PYTHONARCHLIB lib/python)
@@ -138,14 +138,14 @@ set( C_SQL                        sql)
 set( C_SERVICES     services)

 set( C_INIT                    etc/init.d )
-set( C_ETC                     etc/examples )
+set( C_ETC                     etc/examples/condor )
 set( C_CONFIGD         etc/condor/config.d )
 set( C_GANGLIAD                etc/condor/ganglia.d )
 set( C_SYSCONFIG       etc/sysconfig )

-set( C_ETC_EXAMPLES etc/examples )
-set( C_SHARE_EXAMPLES .)
-set( C_DOC                     . )
+set( C_ETC_EXAMPLES etc/examples/condor )
+set( C_SHARE_EXAMPLES share/examples/condor )
+set( C_DOC share/doc/condor )

 set( C_LOCAL_DIR       var/lib/condor )
 set( C_LOG_DIR         var/log/condor )
