From 82c9334deb3cfa4279bda80a1d4e86b7d78aca15 Mon Sep 17 00:00:00 2001
From: Jason Patton <jpatton@cs.wisc.edu>
Date: Thu, 3 Oct 2019 13:54:14 -0500
Subject: [PATCH] Build bindings if either Python 2 or 3 libs are found #7298

---
 src/python-bindings/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/python-bindings/CMakeLists.txt b/src/python-bindings/CMakeLists.txt
index c2e8673092..2da966ca1a 100644
--- a/src/python-bindings/CMakeLists.txt
+++ b/src/python-bindings/CMakeLists.txt
@@ -204,7 +204,7 @@ if(WINDOWS)
     endif(NOT (MSVC_VERSION LESS 1700))
   endif()
 else()
-  if ( WITH_PYTHON_BINDINGS AND PYTHONLIBS_FOUND AND Boost_PYTHON_LIBRARY AND NOT SOLARIS )
+  if ( WITH_PYTHON_BINDINGS AND ( PYTHONLIBS_FOUND OR PYTHON3LIBS_FOUND ) AND Boost_PYTHON_LIBRARY AND NOT SOLARIS )
     configure_file (
       "${PROJECT_SOURCE_DIR}/src/python-bindings/test_driver.in"
       "${CMAKE_CURRENT_BINARY_DIR}/test_driver"
@@ -490,5 +490,5 @@ else()
         endif()
       endif ( LINUX OR DARWIN )
     endif (BUILD_TESTING)
-  endif ( WITH_PYTHON_BINDINGS AND PYTHONLIBS_FOUND AND Boost_PYTHON_LIBRARY AND NOT SOLARIS )
+  endif ( WITH_PYTHON_BINDINGS AND ( PYTHONLIBS_FOUND OR PYTHON3LIBS_FOUND ) AND Boost_PYTHON_LIBRARY AND NOT SOLARIS )
 endif(WINDOWS)
-- 
2.22.1

