--- src/python-bindings/CMakeLists.txt	2019-10-15 13:04:50.458272716 +0100
+++ src/python-bindings/CMakeLists.txt	2019-10-15 13:49:49.456517268 +0100
@@ -332,7 +332,7 @@
       add_library( py3classad_module SHARED classad_module.cpp classad_module_impl.cpp classad_parsers.cpp )
       target_include_directories( py3classad_module PUBLIC ${PYTHON3_INCLUDE_DIRS} )
       target_link_libraries( py3classad_module py3classad -l${PYTHON3_BOOST_LIB} ${PYTHON3_LIBRARIES} )
-      set_target_properties(py3classad_module PROPERTIES PREFIX "" OUTPUT_NAME py3classad )
+      set_target_properties(py3classad_module PROPERTIES PREFIX "" OUTPUT_NAME classad )
       set_target_properties(py3classad_module PROPERTIES SUFFIX ".so" )
       if ( NOT CMAKE_SKIP_RPATH )
         set_target_properties(py3classad_module PROPERTIES INSTALL_RPATH "${PYTHON_RPATH}")
@@ -367,7 +367,7 @@
       add_library( py3htcondor SHARED htcondor.cpp ${HTCONDOR_BINDINGS_SOURCES} log_reader.cpp )
       target_include_directories( py3htcondor PUBLIC ${PYTHON3_INCLUDE_DIRS} )
       target_link_libraries( py3htcondor py3classad condor_utils -l${PYTHON3_BOOST_LIB} ${PYTHON3_LIBRARIES} )
-      set_target_properties( py3htcondor PROPERTIES PREFIX "" )
+      set_target_properties( py3htcondor PROPERTIES PREFIX "" OUTPUT_NAME htcondor )
       set_target_properties( py3htcondor PROPERTIES SUFFIX ".so" )
       if ( NOT CMAKE_SKIP_RPATH )
         set_target_properties(py3htcondor PROPERTIES INSTALL_RPATH "${PYTHON_RPATH}")
