diff --git a/build/cmake/macros/CondorSetLinkLibs.cmake b/build/cmake/macros/CondorSetLinkLibs.cmake
index b21f93c0a4..101b02bbd1 100644
--- a/build/cmake/macros/CondorSetLinkLibs.cmake
+++ b/build/cmake/macros/CondorSetLinkLibs.cmake
@@ -32,7 +32,7 @@ if (${_CNDR_TARGET}LinkLibs)
                if (DARWIN OR AIX OR SOLARIS)
                        target_link_libraries( ${_CNDR_TARGET} ${${_CNDR_TARGET}LinkLibs} ${${_CNDR_TARGET}LinkLibs}  )
                else()
-                       target_link_libraries( ${_CNDR_TARGET} -Wl,--start-group ${${_CNDR_TARGET}LinkLibs} -Wl,--end-group -Wl,--enable-new-dtags )
+                       target_link_libraries( ${_CNDR_TARGET} -Wl,--start-group ${${_CNDR_TARGET}LinkLibs} -Wl,--end-group )
                endif()
         else()
                target_link_libraries( ${_CNDR_TARGET} ${${_CNDR_TARGET}LinkLibs};${CONDOR_WIN_LIBS} )
