diff --git a/Algebraic_kernel_d/include/CGAL/RS/dyadic.h b/Algebraic_kernel_d/include/CGAL/RS/dyadic.h
index 9f35ea8..ef9f514 100644
--- Algebraic_kernel_d/include/CGAL/RS/dyadic.h
+++ Algebraic_kernel_d/include/CGAL/RS/dyadic.h
@@ -21,7 +21,7 @@
 
 #include <stdio.h>
 #include <math.h>
-#include <gmp.h>
+#include <mpir.h>
 #include <mpfr.h>
 #include <CGAL/assertions.h>
 
diff --git a/Algebraic_kernel_d/include/CGAL/RS/signat_1.h b/Algebraic_kernel_d/include/CGAL/RS/signat_1.h
index af30329..3da53ea 100644
--- Algebraic_kernel_d/include/CGAL/RS/signat_1.h
+++ Algebraic_kernel_d/include/CGAL/RS/signat_1.h
@@ -25,7 +25,7 @@
 #include <CGAL/Polynomial_traits_d.h>
 #include "exact_signat_1.h"
 //#include <boost/mpl/assert.hpp>
-#include <gmp.h>
+#include <mpir.h>
 
 namespace CGAL{
 namespace RS_AK1{
diff --git a/CGAL_Core/include/CGAL/CORE/CoreAux_impl.h b/CGAL_Core/include/CGAL/CORE/CoreAux_impl.h
index cfb72ec..5a5b2c6 100644
--- CGAL_Core/include/CGAL/CORE/CoreAux_impl.h
+++ CGAL_Core/include/CGAL/CORE/CoreAux_impl.h
@@ -40,7 +40,7 @@
 #endif
 
 #include <CGAL/CORE/CoreAux.h>
-#include <gmp.h>
+#include <mpir.h>
 
 namespace CORE { 
 
diff --git a/CGAL_Core/include/CGAL/CORE/Gmp.h b/CGAL_Core/include/CGAL/CORE/Gmp.h
index cfa7574..d4373cd 100644
--- CGAL_Core/include/CGAL/CORE/Gmp.h
+++ CGAL_Core/include/CGAL/CORE/Gmp.h
@@ -25,7 +25,7 @@
 #define _CORE_GMP_H_
 
 #include <CGAL/CORE/Impl.h>
-#include <gmp.h>
+#include <mpir.h>
 
 namespace CORE { 
 
diff --git a/Installation/cmake/modules/FindGMP.cmake b/Installation/cmake/modules/FindGMP.cmake
index 11f1add..7e0824d 100644
--- Installation/cmake/modules/FindGMP.cmake
+++ Installation/cmake/modules/FindGMP.cmake
@@ -28,7 +28,7 @@ if (GMP_in_cache)
 else()
 
   find_path(GMP_INCLUDE_DIR
-            NAMES gmp.h
+            NAMES mpir.h
             HINTS ENV GMP_INC_DIR
                   ENV GMP_DIR
                   ${CGAL_INSTALLATION_PACKAGE_DIR}/auxiliary/gmp/include
@@ -40,7 +40,7 @@ else()
     cache_set( GMP_IN_CGAL_AUXILIARY TRUE )
   endif()
 
-  find_library(GMP_LIBRARIES NAMES gmp libgmp-10
+  find_library(GMP_LIBRARIES NAMES mpir
     HINTS ENV GMP_LIB_DIR
           ENV GMP_DIR
           ${CGAL_INSTALLATION_PACKAGE_DIR}/auxiliary/gmp/lib
diff --git a/Number_types/include/CGAL/GMP/Gmpzf_type.h b/Number_types/include/CGAL/GMP/Gmpzf_type.h
index b5e943e..660d3c1 100644
--- Number_types/include/CGAL/GMP/Gmpzf_type.h
+++ Number_types/include/CGAL/GMP/Gmpzf_type.h
@@ -82,7 +82,7 @@ public:
 
   // exponent type
   // --------------------
-  typedef long Exponent; // may overflow, but if it does, the mantissa is
+  typedef mpir_si Exponent; // may overflow, but if it does, the mantissa is
                          // potentially too large to be useful, anyway;
                          // still, repeated squaring of a power of two
                          // quickly brings this type to its limits...
diff --git a/Number_types/include/CGAL/gmp.h b/Number_types/include/CGAL/gmp.h
index bac2fcb..0bb3837 100644
--- Number_types/include/CGAL/gmp.h
+++ Number_types/include/CGAL/gmp.h
@@ -27,7 +27,7 @@
                                      // warning on - applied on unsigned number
 #endif
 
-#include <gmp.h>
+#include <mpir.h>
 
 
 #if defined(BOOST_MSVC)
