From 31bf2bbc79522e26fbc4b12466a5c0253a0e3c15 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 9 Aug 2021 15:48:04 -0600
Subject: [PATCH] Pymatgen Fix

---
 pmg_lammps/inputs.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pmg_lammps/inputs.py b/pmg_lammps/inputs.py
index f024128..e6c42e5 100644
--- a/pmg_lammps/inputs.py
+++ b/pmg_lammps/inputs.py
@@ -2,8 +2,8 @@
 from collections import OrderedDict
 
 import pymatgen as pmg
-from pymatgen.core import Specie, Element, Structure
-from pymatgen.core.periodic_table import _pt_data as periodic_table
+from pymatgen.core.structure import Element, Structure
+from pymatgen.core.periodic_table import Specie, _pt_data as periodic_table
 import numpy as np
 
 from .core import LammpsBox, LammpsPotentials
