From 6224adcd14c21cd76702c6d6f7c6166058bc6e38 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 9 Aug 2021 15:27:06 -0600
Subject: [PATCH] Fix pymatgen compatibility

---
 pmg_lammps/inputs.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pmg_lammps/inputs.py b/pmg_lammps/inputs.py
index f024128..7e0cf7a 100644
--- a/pmg_lammps/inputs.py
+++ b/pmg_lammps/inputs.py
@@ -2,7 +2,7 @@
 from collections import OrderedDict
 
 import pymatgen as pmg
-from pymatgen.core import Specie, Element, Structure
+from pymatgen.core.structure import Specie, Element, Structure
 from pymatgen.core.periodic_table import _pt_data as periodic_table
 import numpy as np
 
