From f028860f41174600245fc60245fe1f184d0625ea Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 29 Mar 2020 13:44:08 +0200
Subject: [PATCH 2/4] relax pandas version-check to minor level

---
 modin/pandas/__init__.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/modin/pandas/__init__.py b/modin/pandas/__init__.py
index 2065f07..b2abd0a 100644
--- a/modin/pandas/__init__.py
+++ b/modin/pandas/__init__.py
@@ -1,13 +1,14 @@
 import pandas
 
-__pandas_version__ = "1.0.1"
+# pandas follows SemVer since 1.0; no need to specify down to patch level
+__pandas_maj_min_version__ = "1.0"
 
-if pandas.__version__ != __pandas_version__:
+if pandas.__version__[:3] != __pandas_maj_min_version__:
     import warnings
 
     warnings.warn(
         "The pandas version installed does not match the required pandas version in "
-        "Modin. This may cause undesired side effects!".format(__pandas_version__)
+        "Modin ({}.x). This may cause undesired side effects!".format(__pandas_maj_min_version__)
     )
 
 from pandas import (
-- 
2.24.0.windows.2

