From 00ffbdf46050a6733ce3f4a6627f76751a27197f Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 29 Mar 2020 13:49:44 +0200
Subject: [PATCH 3/4] remove unused (wrong) version string

---
 modin/__init__.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modin/__init__.py b/modin/__init__.py
index fb651bb..7bdde9f 100644
--- a/modin/__init__.py
+++ b/modin/__init__.py
@@ -76,7 +76,6 @@ def get_partition_format():
     return os.environ.get("MODIN_BACKEND", "Pandas").title()
 
 
-__version__ = "0.6.3"
 __execution_engine__ = get_execution_engine()
 __partition_format__ = get_partition_format()
 
-- 
2.24.0.windows.2

