--- CMakeLists.txt
+++ CMakeLists.txt
@@ -39,6 +39,8 @@
 
 message (STATUS "${PROJECT_NAME} version ${SDF_VERSION_FULL}")
 set (CMAKE_INCLUDE_DIRECTORIES_PROJECT_BEFORE ON)
+set (CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES "")
+set (CMAKE_CXX_IMPLICIT_LINK_DIRECTORIES "")
 
 #============================================================================
 # We turn off extensions because (1) we do not ever want to use non-standard
@@ -102,17 +104,6 @@
 include (${sdf_cmake_dir}/SDFUtils.cmake)
 
 message (STATUS "\n\n====== Finding 3rd Party Packages ======")
- # Use of tinyxml. System installation on UNIX. Internal copy on WIN
-if (UNIX)
-  message (STATUS "Using system tinyxml")
-  set (USE_EXTERNAL_TINYXML True)
-elseif(WIN32)
-  message (STATUS "Using internal tinyxml code")
-  set (USE_EXTERNAL_TINYXML False)
-else()
-  message (STATUS "Unknown platform")
-  BUILD_ERROR("Unknown platform")
-endif()
 include (${sdf_cmake_dir}/SearchForStuff.cmake)
 message (STATUS "----------------------------------------\n")
 
--- cmake/SDFUtils.cmake
+++ cmake/SDFUtils.cmake
@@ -128,10 +128,6 @@
     elseif(WIN32)
       add_executable(${BINARY_NAME}
         ${GTEST_SOURCE_file}
-        ${PROJECT_SOURCE_DIR}/src/win/tinyxml/tinystr.cpp
-        ${PROJECT_SOURCE_DIR}/src/win/tinyxml/tinyxmlerror.cpp
-        ${PROJECT_SOURCE_DIR}/src/win/tinyxml/tinyxml.cpp
-        ${PROJECT_SOURCE_DIR}/src/win/tinyxml/tinyxmlparser.cpp
       )
     else()
       message(FATAL_ERROR "Unsupported platform")
@@ -157,6 +153,7 @@
         gtest.lib
         gtest_main.lib
         sdformat.dll
+        ${tinyxml_LIBRARIES}
         ${IGNITION-MATH_LIBRARIES}
       )
 
--- cmake/SearchForStuff.cmake
+++ cmake/SearchForStuff.cmake
@@ -70,7 +70,7 @@
 
 if (NOT DEFINED USE_INTERNAL_URDF OR NOT USE_INTERNAL_URDF)
   # check for urdfdom with pkg-config
-  pkg_check_modules(URDF urdfdom>=1.0)
+  pkg_check_modules(URDF urdfdom>=1.0 IMPORTED_TARGET)
 
   if (NOT URDF_FOUND)
     if (NOT DEFINED USE_INTERNAL_URDF)
@@ -79,10 +79,6 @@
     else()
       BUILD_ERROR("Couldn't find the urdfdom >= 1.0 system installation")
     endif()
-  else()
-    # what am I doing here? pkg-config and cmake
-    set(URDF_INCLUDE_DIRS ${URDF_INCLUDEDIR})
-    set(URDF_LIBRARY_DIRS ${URDF_LIBDIR})
   endif()
 endif()
 
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -105,7 +105,11 @@
 message (STATUS "URDF_LIBRARIES=${URDF_LIBRARIES}")
 
 if (NOT USE_INTERNAL_URDF)
-  target_link_libraries(sdformat ${URDF_LIBRARIES})
+  if (TARGET PkgConfig::URDF)
+    target_link_libraries(sdformat PkgConfig::URDF)
+  else()
+    target_link_libraries(sdformat ${URDF_LIBRARIES})
+  endif()
 endif()
 
 sdf_install_library(sdformat)
