diff -ur FMILibrary-2.0.2.orig/ThirdParty/c99_snprintf/CMakeLists.txt FMILibrary-2.0.2/ThirdParty/c99_snprintf/CMakeLists.txt
--- ThirdParty/c99_snprintf/CMakeLists.txt	2016-10-17 19:12:50.795570338 +0200
+++ ThirdParty/c99_snprintf/CMakeLists.txt	2016-10-17 19:18:05.862250814 +0200
@@ -87,8 +87,6 @@
     endif()
 endforeach()
 add_definitions(
-    -Dvsnprintf=rpl_vsnprintf
-    -Dsnprintf=rpl_snprintf
     -Dvasprintf=rpl_vasprintf
     -Dasprintf=rpl_asprintf
     ${DEFINES})
