diff --git a/SConstruct b/SConstruct
--- a/SConstruct
+++ b/SConstruct
@@ -353,6 +353,8 @@
 opts.Add("CXX", "C++ compiler binary")
 opts.Add("CC", "C compiler binary")
 opts.Add("LINK", "Linker binary")
+opts.Add("AR", "Archiver binary")
+opts.Add("RANLIB", "Archive indexer binary")
 opts.Add("cppdefines", "Custom defines for the pre-processor")
 opts.Add("ccflags", "Custom flags for both the C and C++ compilers")
 opts.Add("cxxflags", "Custom flags for the C++ compiler")
