diff --git a/platform/linuxbsd/wayland/wayland_thread.cpp b/platform/linuxbsd/wayland/wayland_thread.cpp
--- a/platform/linuxbsd/wayland/wayland_thread.cpp
+++ b/platform/linuxbsd/wayland/wayland_thread.cpp
@@ -38,9 +38,11 @@
 
 #ifdef __FreeBSD__
 #include <dev/evdev/input-event-codes.h>
-#else
-// Assume Linux.
+#elif __has_include(<linux/input-event-codes.h>)
 #include <linux/input-event-codes.h>
+#else
+// Older Linux kernel headers define the event codes in input.h.
+#include <linux/input.h>
 #endif
 
 // For the actual polling thread.
diff --git a/platform/linuxbsd/wayland/wayland_embedder.cpp b/platform/linuxbsd/wayland/wayland_embedder.cpp
--- a/platform/linuxbsd/wayland/wayland_embedder.cpp
+++ b/platform/linuxbsd/wayland/wayland_embedder.cpp
@@ -36,11 +36,13 @@
 
 #include <sys/stat.h>
 
 #ifdef __FreeBSD__
 #include <dev/evdev/input-event-codes.h>
-#else
-// Assume Linux.
+#elif __has_include(<linux/input-event-codes.h>)
 #include <linux/input-event-codes.h>
+#else
+// Older Linux kernel headers define the event codes in input.h.
+#include <linux/input.h>
 #endif
 
 #include "core/os/os.h"
