diff --git a/platform/linuxbsd/detect.py b/platform/linuxbsd/detect.py
--- a/platform/linuxbsd/detect.py
+++ b/platform/linuxbsd/detect.py
@@ -550,7 +550,7 @@ def configure(env: "Environment"):
     env.Append(LIBS=["pthread"])
 
     if platform.system() == "Linux":
-        env.Append(LIBS=["dl"])
+        env.Append(LIBS=["dl", "iconv"])
 
     if platform.libc_ver()[0] != "glibc":
         if env["execinfo"]:
