diff --git a/platform/windows/detect.py b/platform/windows/detect.py
--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -505,8 +505,55 @@ def configure_msvc(env: "SConsEnvironment"):
         if not env["use_volk"]:
-            LIBS += ["vulkan"]
+            LIBS += ["vulkan-1"]
 
+    env.Prepend(CPPPATH=[os.environ["LIBRARY_INC"]])
+    env.Prepend(LIBPATH=[os.environ["LIBRARY_LIB"]])
+
+    if env["brotli"] and not env["builtin_brotli"]:
+        LIBS += ["brotlicommon", "brotlidec"]
+    if not env["builtin_freetype"]:
+        env.Prepend(CPPPATH=[os.path.join(os.environ["LIBRARY_INC"], "freetype2")])
+        LIBS += ["freetype"]
+    if not env["builtin_graphite"]:
+        LIBS += ["graphite2"]
+    if not env["builtin_libjpeg_turbo"]:
+        LIBS += ["turbojpeg"]
+    if not env["builtin_libogg"]:
+        LIBS += ["ogg"]
+    if not env["builtin_libpng"]:
+        LIBS += ["libpng16"]
+    if not env["builtin_libtheora"]:
+        env["builtin_libogg"] = False
+        env["builtin_libvorbis"] = False
+        LIBS += ["theora", "theoradec", "theoraenc"]
+    if not env["builtin_libvorbis"]:
+        env["builtin_libogg"] = False
+        LIBS += ["vorbis", "vorbisfile", "vorbisenc"]
+    if not env["builtin_libwebp"]:
+        LIBS += ["libwebp", "libsharpyuv"]
+    if not env["builtin_pcre2"]:
+        LIBS += ["pcre2-32"]
+    if not env["builtin_zlib"]:
+        LIBS += ["zlib"]
+    if not env["builtin_zstd"]:
+        LIBS += ["zstd"]
+    if not env["builtin_embree"]:
+        LIBS += ["embree4"]
+    if not env["builtin_glslang"]:
+        LIBS += [
+            "glslang",
+            "SPIRV",
+            "glslang-default-resource-limits",
+            "SPIRV-Tools-opt",
+            "SPIRV-Tools",
+        ]
+    if not env["builtin_msdfgen"]:
+        env.Prepend(CPPPATH=[os.path.join(os.environ["LIBRARY_INC"], "msdfgen")])
+        LIBS += ["msdfgen-core"]
+
     if env["sdl"]:
         env.Append(CPPDEFINES=["SDL_ENABLED"])
+        if not env["builtin_sdl"]:
+            LIBS += ["SDL3"]
 
     if env["d3d12"]:
         check_d3d12_installed(env, env["arch"] + "-msvc")
