Fix make file
author: Jan Janssen (MPIE)

---
 src/Makefile        | 8 +-
 1 file changed, 4 insertion(+), 4 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 2e6ad9a..25556d0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -2,7 +2,7 @@
 #
 # See doc/ for more information on building the extension modules.
 
-F2PY=f2py3
+F2PY=f2py
 F2PY_FLAGS=--opt='-O3' \
 			--f90exec=$(F90) \
 			--f77exec=$(F90) \
@@ -24,7 +24,7 @@
 ifort-omp: OMP_F90_FLAGS=-openmp -D__OPENMP
 ifort-omp: F2PY_OMP_F90_FLAGS=-liomp5
 
-gfortran: F90=gfortran
+gfortran: F90=$(GFORTRAN)
 gfortran: F90FLAGS=-x f95-cpp-input -Wall
 gfortran: ARCH=-march=native
 gfortran-omp: OMP_F90_FLAGS=-fopenmp -D__OPENMP
@@ -86,8 +86,8 @@
 #
 # Here we use a make feature called target pattern. Fancy, eh?
 $(OBJ_LOCAL): _%.so: %.f90
-	CC=gcc CXX=g++ $(F2PY) -h $*.pyf $? -m _$* --overwrite-signature; \
-	CC=gcc CXX=g++ $(F2PY) -c $*.pyf $? $(F2PY_FLAGS); \
+	CC=$(GCC) CXX=$(GXX) $(F2PY) -h $*.pyf $? -m _$* --overwrite-signature; \
+	CC=$(GCC) CXX=$(GXX) $(F2PY) -c $*.pyf $? $(F2PY_FLAGS); \
 	rm -f $@; mv _$**.so $@
 
 
