Fix make file
author: Jan Janssen (MPIE)

---
 src/Makefile        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 2e6ad9a..25556d0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -2,7 +2,7 @@
 #
 # See doc/ for more information on building the extension modules.
 
-F2PY=f2py3
+F2PY=f2py
 F2PY_FLAGS=--opt='-O3' \
 			--f90exec=$(F90) \
 			--f77exec=$(F90) \
