--- a/cmake/ga-linalg.cmake
+++ b/cmake/ga-linalg.cmake
@@ -119,7 +119,7 @@
       set(use_openmp OFF)
     endif()
 
-    if(_blis_essl_set OR ${LINALG_VENDOR} MATCHES "OpenBLAS")
+    if(_blis_essl_set OR ${LINALG_VENDOR} MATCHES "OpenBLAS" OR ${LINALG_VENDOR} MATCHES "ReferenceBLAS")
       set(use_openmp OFF)
       # if(_blis_essl_set) #Assume openblas does not have lapack
         set(LAPACK_PREFERENCE_LIST ReferenceLAPACK)
