From f858f9909832f45ab96dc20e75ea3efd29734647 Mon Sep 17 00:00:00 2001
From: Alex Ford <alex.ford@abcellera.com>
Date: Fri, 4 Nov 2022 18:23:18 -0700
Subject: [PATCH] fix_cpp_build

---
 setup.py | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index 34b1663..e6d4763 100644
--- a/setup.py
+++ b/setup.py
@@ -58,13 +58,13 @@ def limited_api_args():
 #    return ['-DPy_LIMITED_API=0x030a0000']
     return []
 
-class CppExtension(build_ext):
-    def build_extensions(self):
-        if sys.platform == "linux":
-            self.compiler.compiler_so[0] = "g++"
-            self.compiler.compiler_cxx[0] = "g++"
-            self.compiler.linker_so[0] = "g++"
-        build_ext.build_extensions(self)
+# class CppExtension(build_ext):
+#     def build_extensions(self):
+#         if sys.platform == "linux":
+#             self.compiler.compiler_so[0] = "g++"
+#             self.compiler.compiler_cxx[0] = "g++"
+#             self.compiler.linker_so[0] = "g++"
+#         build_ext.build_extensions(self)
 
 tracker = setuptools.extension.Extension(
             'slipcover.tracker',
@@ -72,7 +72,7 @@ tracker = setuptools.extension.Extension(
             extra_compile_args=cxx_version('c++17') + platform_compile_args() + limited_api_args(),
             extra_link_args=platform_link_args(),
             py_limited_api=bool(limited_api_args()),
-            language='C++'
+            language='c++'
 )
 
 
@@ -93,7 +93,7 @@ setuptools.setup(
     install_requires=[
         "tabulate"
     ],
-    cmdclass={"build_ext": CppExtension},
+    # cmdclass={"build_ext": CppExtension},
     classifiers=[
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
-- 
2.35.3

