From 2dae408a58701616fa1aec2c9aeab49b70b03bf5 Mon Sep 17 00:00:00 2001
From: Alex Ford <alex.ford@abcellera.com>
Date: Fri, 4 Nov 2022 18:23:18 -0700
Subject: [PATCH] use conda build flags

---
 setup.py | 30 +++++++++++++++++++++---------
 1 file changed, 21 insertions(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index 34b1663..652b865 100644
--- a/setup.py
+++ b/setup.py
@@ -1,6 +1,7 @@
 import setuptools
 from setuptools.command.build_ext import build_ext
 import sys
+import os
 from pathlib import Path
 
 def get_version():
@@ -35,6 +36,11 @@ def cxx_version(v):
     return [f"-std={v}" if sys.platform != "win32" else f"/std:{v}"]
 
 def platform_compile_args():
+    # If flags are specified as a global env var use then,
+    # this happens during conda build.
+    if flags := os.environ.get("CXXFLAGS", "").split():
+        return flags
+
     if sys.platform == 'darwin':
         return "-arch x86_64 -arch arm64 -arch arm64e".split()
     if sys.platform == 'win32':
@@ -58,13 +64,19 @@ def limited_api_args():
 #    return ['-DPy_LIMITED_API=0x030a0000']
     return []
 
-class CppExtension(build_ext):
-    def build_extensions(self):
-        if sys.platform == "linux":
-            self.compiler.compiler_so[0] = "g++"
-            self.compiler.compiler_cxx[0] = "g++"
-            self.compiler.linker_so[0] = "g++"
-        build_ext.build_extensions(self)
+# class CppExtension(build_ext):
+#     def build_extensions(self):
+#         if sys.platform == "linux":
+#             self.compiler.compiler_so[0] = "g++"
+#             self.compiler.compiler_cxx[0] = "g++"
+#             self.compiler.linker_so[0] = "g++"
+
+#         if "CXX" in os.environ:
+#             self.compiler.compiler_so[0] = os.environ["CXX"]
+#             self.compiler.compiler_cxx[0] = os.environ["CXX"]
+#             self.compiler.linker_so[0] = os.environ["CXX"]
+
+#         build_ext.build_extensions(self)
 
 tracker = setuptools.extension.Extension(
             'slipcover.tracker',
@@ -72,7 +84,7 @@ tracker = setuptools.extension.Extension(
             extra_compile_args=cxx_version('c++17') + platform_compile_args() + limited_api_args(),
             extra_link_args=platform_link_args(),
             py_limited_api=bool(limited_api_args()),
-            language='C++'
+            language='c++'
 )
 
 
@@ -88,7 +100,7 @@ setuptools.setup(
     author_email="juan@altmayer.com, emery@cs.umass.edu",
     license="Apache License 2.0",
     packages=['slipcover'],
-    ext_modules=([tracker]),
+    # ext_modules=([tracker]),
     python_requires=">=3.8,<3.12",
     install_requires=[
         "tabulate"
-- 
2.35.3

