diff -urN libvpx.old/third_party/libwebm/mkvparser/mkvreader.cc libvpx/third_party/libwebm/mkvparser/mkvreader.cc
--- libvpx.old/third_party/libwebm/mkvparser/mkvreader.cc	2017-04-25 05:31:28.000000000 -0500
+++ libvpx/third_party/libwebm/mkvparser/mkvreader.cc	2017-04-25 05:35:39.000000000 -0500
@@ -8,6 +8,9 @@
 #include "mkvparser/mkvreader.h"

 #include <cassert>
+#ifdef linux
+#include <unistd.h>
+#endif

 namespace mkvparser {

diff -urN libvpx.old/third_party/libwebm/mkvmuxer/mkvwriter.cc libvpx/third_party/libwebm/mkvmuxer/mkvwriter.cc
--- libvpx.old/third_party/libwebm/mkvmuxer/mkvwriter.cc	2017-04-25 05:53:42.000000000 -0500
+++ libvpx/third_party/libwebm/mkvmuxer/mkvwriter.cc	2017-04-25 05:56:02.000000000 -0500
@@ -11,6 +11,9 @@
 #ifdef _MSC_VER
 #include <share.h>  // for _SH_DENYWR
 #endif
+#ifdef linux
+#include <unistd.h>
+#endif

 namespace mkvmuxer {
