--- a/src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp
+++ b/src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp
@@ -79,8 +79,7 @@ Containers::Array<char> BasisImageConverter::doExportToData(const ImageView2D& i
     PARAM_CONFIG(max_selector_clusters, int);
     PARAM_CONFIG(y_flip, bool);
     PARAM_CONFIG(check_for_alpha, bool);
     PARAM_CONFIG(force_alpha, bool);
-    PARAM_CONFIG_FIX_NAME(seperate_rg_to_color_alpha, bool, "separate_rg_to_color_alpha");
 
     UnsignedInt threadCount = configuration().value<Int>("threads");
     if(threadCount == 0) threadCount = std::thread::hardware_concurrency();
@@ -141,7 +140,8 @@ Containers::Array<char> BasisImageConverter::doExportToData(const ImageView2D& i
        basis image from existing data as it is based on a std::vector, moreover
        we need to tightly pack it and flip Y. The `dst` is an Y-flipped view
        already to make the following loops simpler. */
-    params.m_source_images.emplace_back(image.size().x(), image.size().y());
+    params.m_source_images.resize(1);
+    params.m_source_images.back().resize(image.size().x(), image.size().y());
     auto dst = Containers::arrayCast<Color4ub>(Containers::StridedArrayView2D<basisu::color_rgba>({params.m_source_images.back().get_ptr(), params.m_source_images.back().get_total_pixels()}, {std::size_t(image.size().y()), std::size_t(image.size().x())})).flipped<0>();
 
     /* basis image is always RGBA, fill in alpha if necessary */
