From db1a5ed30ca1042b1f4c9db51752f16beceb3e3b Mon Sep 17 00:00:00 2001
From: Tadeu Manoel <e.tadeu@gmail.com>
Date: Fri, 15 Feb 2019 11:14:26 -0200
Subject: [PATCH] Remove tlmdif test example

---
 examples/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index d5cacd8..b85d428 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -8,7 +8,7 @@ if (BUILD_EXAMPLES)
   # Make sure the linker can find the cminpack library once it is built.
   link_directories (${CMINPACK_BINARY_DIR})
 
-  set (FPGM tchkder thybrd thybrd1 thybrj thybrj1 tlmder tlmder1 tlmdif tlmdif1 tlmstr tlmstr1)
+  set (FPGM tchkder thybrd thybrd1 thybrj thybrj1 tlmder tlmder1 tlmdif1 tlmstr tlmstr1)
   set (CPGM ${FPGM} tfdjac2)
 
   # cmpfiles could be used by runtest.cmake... for now it's unused
-- 
2.7.4

