From 4c7404ca0cd5cee921993419793e290d111e9218 Mon Sep 17 00:00:00 2001
From: Tadeu Manoel <e.tadeu@gmail.com>
Date: Wed, 13 Feb 2019 15:48:05 -0200
Subject: [PATCH] Fix compilation of `examples/tlmdif_` on Windows

---
 minpack.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/minpack.h b/minpack.h
index a9d31ce..43405b1 100644
--- a/minpack.h
+++ b/minpack.h
@@ -296,7 +296,7 @@ void __minpack_func__(rwupdt)(const int *n, __minpack_real__ *r, const int *ldr,
              const __minpack_real__ *w, __minpack_real__ *b, __minpack_real__ *alpha, __minpack_real__ *cos, 
              __minpack_real__ *sin);
 __minpack_attr__
-void __minpack_func__(covar)(const int *n, __minpack_real__ *r, const int *ldr, 
+void MINPACK_EXPORT __minpack_func__(covar)(const int *n, __minpack_real__ *r, const int *ldr,
            const int *ipvt, const __minpack_real__ *tol, __minpack_real__ *wa);
 #ifdef __cplusplus
 }
-- 
2.20.1.windows.1

