diff --git a/libcap/cap_file.c b/libcap/cap_file.c
index 40756ea..e27ca80 100644
--- a/libcap/cap_file.c
+++ b/libcap/cap_file.c
@@ -25,7 +25,7 @@ extern int fremovexattr(int, const char *);

 #include "libcap.h"

-#ifdef VFS_CAP_U32
+#if defined (VFS_CAP_U32) && defined (XATTR_NAME_CAPS)

 #if VFS_CAP_U32 != __CAP_BLKS
 # error VFS representation of capabilities is not the same size as kernel
--
2.8.1