--- shyaml-orig 2018-08-24 10:11:02.361128184 -0400
+++ shyaml  2018-08-24 10:12:11.900480130 -0400
@@ -73,7 +73,7 @@
     DEFAULT   if not provided and given KEY do not match any value in
               the provided YAML, then DEFAULT will be returned. If no
               default is provided and the KEY do not match any value
-              in the provided YAML, %(exname)s will fail with an error
+              in the provided YAML, %(exname)s will fail with an error
               message.

 Examples:
@@ -374,9 +374,9 @@
         type(value).__name__


-def main(args):  ## pylint: disable=too-many-branches
+def main(args=None):  ## pylint: disable=too-many-branches
     """Entrypoint of the whole application"""
-
+    args = sys.argv[1:] if args is None else args
     dump = magic_dump
     for arg in ["-y", "--yaml"]:
         if arg in args:
