diff --git a/setup.py b/setup.py
index 388dca21e..38e36d313 100755
--- a/setup.py
+++ b/setup.py
@@ -226,7 +226,7 @@ setup (
     extras_require = extras_require,
     install_requires = install_requires,
     scripts  = find_files('bin', relpath='./') + ['tools/einsteinathome/pycbc_build_eah.sh'],
-    packages = find_packages(),
+    packages = find_packages(exclude=('gwave*',)),
     package_data = {'pycbc.workflow': find_files('pycbc/workflow'),
                     'pycbc.results': find_files('pycbc/results'),
                     'pycbc.tmpltbank': find_files('pycbc/tmpltbank')},
