diff --git a/modules/FindMagnumBindings.cmake b/modules/FindMagnumBindings.cmake
index 3a73b61..f2d16f3 100644
--- a/modules/FindMagnumBindings.cmake
+++ b/modules/FindMagnumBindings.cmake
@@ -130,6 +130,9 @@ endif()
 # and include name derived from the library name in the loop below. (So far no
 # special cases.)
 
+set(_MAGNUMBINDINGS_PYTHON_INCLUDE_PATH_NAMES
+    Magnum/PythonBindings.h)
+
 # Find all components
 foreach(_component ${MagnumBindings_FIND_COMPONENTS})
     string(TOUPPER ${_component} _COMPONENT)
