--- pyface/ipython_widget.py.orig  2015-10-27 14:57:28.000000000 -0400
+++ pyface/ipython_widget.py    2016-04-18 23:05:18.000000000 -0400
@@ -13,18 +13,17 @@
 #------------------------------------------------------------------------------
 """ The implementation of an IPython shell. """

+from __future__ import absolute_import

 # Import the toolkit specific version.
 try:
     import IPython.frontend
 except ImportError:
-    raise ImportError, '''
+    raise ImportError('''
 ________________________________________________________________________________
 Could not load the Wx frontend for ipython.
-You need to have ipython >= 0.9 installed to use the ipython widget.'''
-
+You need to have ipython >= 0.9 installed to use the ipython widget.''')

-from __future__ import absolute_import

 from .toolkit import toolkit_object
 IPythonWidget= toolkit_object('ipython_widget:IPythonWidget')
