From 3944e8dbf2b8d92ea0e22312c287b3bfd540a13a Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 31 Mar 2026 14:43:33 -0700
Subject: [PATCH] add main function for entrypoint

---
 pyAMI/app.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pyAMI/app.py b/pyAMI/app.py
index 2cbdb23..d0ca008 100644
--- a/pyAMI/app.py
+++ b/pyAMI/app.py
@@ -186,3 +186,6 @@ class App(object):
 			return 1
 
 #############################################################################
+
+def main():
+	sys.exit(App('local', 'text').run())
-- 
2.45.2

