From d5d6665dfe179448dda296d8f32c94d84e34a812 Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 31 Mar 2026 14:46:15 -0700
Subject: [PATCH 4/4] add lxml dep

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 849d528..d006815 100755
--- a/setup.py
+++ b/setup.py
@@ -40,7 +40,7 @@ if __name__ == '__main__':
 		packages = ['pyAMI'],
 		package_data = {'': ['README', 'CHANGELOG', '*.txt'], 'pyAMI': ['*.txt']},
 		scripts = scripts,
-		install_requires = [],
+		install_requires = ['lxml'],
 		platforms = 'any',
 		zip_safe = False
 	)
-- 
2.45.2

