From 3acaff7ef2a08824d9355356cf68e0d3e53941be Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 31 Mar 2026 14:48:55 -0700
Subject: [PATCH 5/5] raw strings for \W

---
 pyAMI/utils.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyAMI/utils.py b/pyAMI/utils.py
index 543bdbf..7af798d 100644
--- a/pyAMI/utils.py
+++ b/pyAMI/utils.py
@@ -174,7 +174,7 @@ def get_entity(table):
 def get_primary_fields(table):
 
 	try:
-		return re.split('\W+', tables[table]['@primary'])
+		return re.split(r'\W+', tables[table]['@primary'])
 
 	except KeyError:
 		raise pyAMI.exception.Error('no primary field for table `%s`' % table)
@@ -184,7 +184,7 @@ def get_primary_fields(table):
 def get_foreign_tables(table):
 
 	try:
-		return re.split('\W+', tables[table]['@foreign'])
+		return re.split(r'\W+', tables[table]['@foreign'])
 
 	except KeyError:
 		return (((((((((((((((((((((((((((((([ ]))))))))))))))))))))))))))))))
-- 
2.45.2

