From 9718ffeefaac0f8d58f5c460abe19e8360549b16 Mon Sep 17 00:00:00 2001
From: Valentin Dovidaytis <horis.vdovidaytis@gmail.com>
Date: Wed, 5 Mar 2025 11:42:32 +0100
Subject: [PATCH] Remove static links for Linux

---
 python-package/setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/python-package/setup.py b/python-package/setup.py
index fa3e20c6b..c2e7b7017 100644
--- a/python-package/setup.py
+++ b/python-package/setup.py
@@ -66,8 +66,7 @@ elif this_system == 'Windows':
     distutils.cygwinccompiler.get_msvcr = lambda: []
 
 elif this_system == 'Linux':
-    static_link_libraries_list += ['stdc++']
-    extra_link = ['-lz']
+    extra_link = []
 
 else:
     raise ValueError("Unsupported platform.")
-- 
