From b0abb516dcb3fe93f79ae6ebfc8d2165e49f315f Mon Sep 17 00:00:00 2001
From: Valentin Dovidaytis <horis.vdovidaytis@gmail.com>
Date: Wed, 5 Mar 2025 12:24:36 +0100
Subject: [PATCH] Remove static link of zlib for Linux

---
 python-package/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python-package/setup.py b/python-package/setup.py
index fa3e20c6b..cab9455c8 100644
--- a/python-package/setup.py
+++ b/python-package/setup.py
@@ -67,7 +67,7 @@ elif this_system == 'Windows':
 
 elif this_system == 'Linux':
     static_link_libraries_list += ['stdc++']
-    extra_link = ['-lz']
+    extra_link = []
 
 else:
     raise ValueError("Unsupported platform.")
-- 
