From fc29e3c7ca5fb2fdde0d311691b5c953e1695b4b Mon Sep 17 00:00:00 2001
From: Hadrien Mary <hadrien.mary@gmail.com>
Date: Mon, 31 Aug 2020 17:53:15 -0400
Subject: [PATCH 1/4] Use bind instead of bind2nd

---
 tmap/tmap/analyse.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tmap/tmap/analyse.cc b/tmap/tmap/analyse.cc
index b282b74..7cfe52e 100644
--- a/tmap/tmap/analyse.cc
+++ b/tmap/tmap/analyse.cc
@@ -30,7 +30,7 @@ namespace {

     std::vector<float> diff(weights.size());
     std::transform(weights.begin(), weights.end(), diff.begin(),
-                  std::bind2nd(std::minus<float>(), mean));
+                  std::bind(std::minus<float>(), std::placeholders::_1, mean));
     float sq_sum = std::inner_product(diff.begin(), diff.end(), diff.begin(), 0.0);
     float stdev = std::sqrt(sq_sum / weights.size());

@@ -64,7 +64,7 @@ namespace {

       std::vector<float> diff(weights.size());
       std::transform(weights.begin(), weights.end(), diff.begin(),
-                    std::bind2nd(std::minus<float>(), mean));
+                    std::bind(std::minus<float>(), std::placeholders::_1, mean));
       float sq_sum = std::inner_product(diff.begin(), diff.end(), diff.begin(), 0.0);
       float stdev = std::sqrt(sq_sum / weights.size());


From ec7a69b9939c62d147b479c2bc1d7f8dc506909e Mon Sep 17 00:00:00 2001
From: Hadrien Mary <hadrien.mary@gmail.com>
Date: Mon, 31 Aug 2020 17:53:52 -0400
Subject: [PATCH 2/4] Fix x64 flag in setup.py

---
 tmap/setup.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tmap/setup.py b/tmap/setup.py
index 28c0d06..4d7e240 100644
--- a/tmap/setup.py
+++ b/tmap/setup.py
@@ -54,8 +54,6 @@ def build_extension(self, ext):
             cmake_args += [
                 "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}".format(cfg.upper(), extdir)
             ]
-            if sys.maxsize > 2 ** 32:
-                cmake_args += ["-A", "x64"]
             build_args += ["--", "/m"]
         elif platform.system() == 'Darwin':
             cmake_args += ['-DOpenMP_C_FLAG=-fopenmp']

From b566512d3cb0d45202d334a33a5b4e79b5d342e6 Mon Sep 17 00:00:00 2001
From: Hadrien Mary <hadrien.mary@gmail.com>
Date: Mon, 31 Aug 2020 18:09:32 -0400
Subject: [PATCH 3/4] remove conda logic in cmake

---
 tmap/CMakeLists.txt | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/tmap/CMakeLists.txt b/tmap/CMakeLists.txt
index 723a7ae..4621a7f 100644
--- a/tmap/CMakeLists.txt
+++ b/tmap/CMakeLists.txt
@@ -11,15 +11,6 @@ else()
     set(CMAKE_CXX_FLAGS_RELEASE "-O3 -funroll-loops -fopenmp")
 endif(MSVC)

-
-if (DEFINED ENV{CONDA_PREFIX})
-    SET(OGDF_INCLUDE "$ENV{CONDA_PREFIX}/include")
-    SET(OGDF_LIB "$ENV{CONDA_PREFIX}/lib")
-
-    include_directories(${OGDF_INCLUDE})
-    link_directories(${OGDF_LIB})
-endif()
-
 # find_package(OpenMP)
 # if (OPENMP_FOUND)
 #    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")

From 508b7958d8dd26568ec8256b05c5dc2a335a1a6a Mon Sep 17 00:00:00 2001
From: Matthew R Becker <becker.mr@gmail.com>
Date: Tue, 1 Sep 2020 08:53:32 -0500
Subject: [PATCH 4/4] only comment; do not delete

---
 tmap/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tmap/CMakeLists.txt b/tmap/CMakeLists.txt
index 4621a7f..337cd0c 100644
--- a/tmap/CMakeLists.txt
+++ b/tmap/CMakeLists.txt
@@ -29,7 +29,7 @@ include_directories(${SOURCE_DIR})
 SET(SOURCES "${SOURCE_DIR}/layout.cc" "${SOURCE_DIR}/lshforest.cc" "${SOURCE_DIR}/minhash.cc" "${SOURCE_DIR}/analyse.cc")

 # Generate python module
-add_subdirectory(lib/pybind11)
+# add_subdirectory(lib/pybind11)

 pybind11_add_module(tmap ${SOURCES} "${SOURCE_DIR}/bindings.cc")
 if(MSVC)
