diff --git setup.py setup.py
index 40921b8..3f0f35f 100644
--- setup.py
+++ setup.py
@@ -5,10 +5,9 @@ import errno
 
 sources = ['c/_cffi_backend.c']
 libraries = ['ffi']
-include_dirs = ['/usr/include/ffi',
-                '/usr/include/libffi']    # may be changed by pkg-config
+include_dirs = [os.path.join(sys.prefix, 'include')]
 define_macros = []
-library_dirs = []
+library_dirs = [os.path.join(sys.prefix, 'lib')]
 extra_compile_args = []
 extra_link_args = []
 
