From 18480a41cfb23cdabf1241cb20ea9e793b4349b2 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Fri, 4 Nov 2022 16:49:35 +0100
Subject: [PATCH] Fix LC_ID_DYLIB on macOS

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index baf73e2..e8043f7 100644
--- a/Makefile
+++ b/Makefile
@@ -105,7 +105,7 @@ lib: lib/libProfessor2.so
 
 lib/libProfessor2.so: $(LIBOBJECTS)
 	mkdir -p lib
-	$(CXX) -shared -o $@ $(LIBOBJECTS)
+	cd lib && $(CXX) -shared -o $(notdir $@) $(addprefix ../,$(LIBOBJECTS))
 
 obj/%.o: src/%.cc $(LIBHEADERS)
 	mkdir -p obj
-- 
2.37.1 (Apple Git-137.1)

