--- a/AutoHotkeyx.vcxproj
+++ b/AutoHotkeyx.vcxproj
@@ -400,7 +400,7 @@
       </ClCompile>
     </ItemGroup>
   </Target>
-  <Target Name="SetAhkVersion" Condition="Exists('@(AhkVersionInput)')" BeforeTargets="PrepareForBuild">
+  <Target Name="SetAhkVersion" Condition="Exists('@(AhkVersionInput)') or '$(RawAhkVersion)'!=''" BeforeTargets="PrepareForBuild">
     <Exec Condition="'$(RawAhkVersion)'==''" Command="git describe --dirty --match v*" ConsoleToMSBuild="true" IgnoreExitCode="true" EchoOff="true">
       <Output PropertyName="GitDescription" TaskParameter="ConsoleOutput" />
     </Exec>
@@ -408,6 +408,10 @@
       <RawAhkVersion>$([System.Text.RegularExpressions.Regex]::Replace($(GitDescription.Substring(1)), "-(?=\d+-)|-g|-dirty$", "+"))</RawAhkVersion>
       <AhkVersionN4>$([System.Text.RegularExpressions.Regex]::Replace($(GitDescription), ".*-(\d+)-g.*", "$1"))</AhkVersionN4>
       <AhkVersionN4 Condition="'$(AhkVersionN4)'=='$(GitDescription)'">0</AhkVersionN4>
+      <AhkVersionN>$([System.Text.RegularExpressions.Regex]::Replace($(RawAhkVersion), "\.(\d+)([^\.\d].*)?", ",$1")),$(AhkVersionN4)</AhkVersionN>
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(AhkVersionN)'=='' and '$(RawAhkVersion)'!=''">
+      <AhkVersionN4 Condition="'$(AhkVersionN4)'==''">0</AhkVersionN4>
       <AhkVersionN>$([System.Text.RegularExpressions.Regex]::Replace($(RawAhkVersion), "\.(\d+)([^\.\d].*)?", ",$1")),$(AhkVersionN4)</AhkVersionN>
     </PropertyGroup>
     <Message Text="RAW_AHK_VERSION sourced from git: '$(RawAhkVersion)'" Importance="high" />
