From 62b249fbab21fc9423c8f01174eab9b8bd9ecefa Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 21 Feb 2024 15:03:46 +0100
Subject: [PATCH 2/6] Be compatible with current libprotobuf pin

---
 mysqlx-connector-python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mysqlx-connector-python/setup.py b/mysqlx-connector-python/setup.py
index f9bfb5c..a3fadaf 100644
--- a/mysqlx-connector-python/setup.py
+++ b/mysqlx-connector-python/setup.py
@@ -141,7 +141,7 @@ def main() -> None:
         ext_modules=EXTENSIONS,
         cmdclass=COMMAND_CLASSES,
         python_requires=">=3.8",
-        install_requires=["protobuf>=4.21.1,<=4.21.12"],
+        install_requires=[f"protobuf~={os.environ['libprotobuf']}.0"],
         extras_require={
             "dns-srv": ["dnspython>=1.16.0,<=2.3.0"],
             "compression": ["lz4>=2.1.6,<=4.3.2", "zstandard>=0.12.0,<=0.19.0"],
-- 
2.45.2

