From 3793b95935c5e027fd378204b37feae79ac4be2f Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 21 Feb 2024 20:27:00 +0100
Subject: [PATCH 5/6] Link to abseil_dll on Windows

---
 mysqlx-connector-python/cpydist/__init__.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mysqlx-connector-python/cpydist/__init__.py b/mysqlx-connector-python/cpydist/__init__.py
index 1a47152..efbfa2b 100644
--- a/mysqlx-connector-python/cpydist/__init__.py
+++ b/mysqlx-connector-python/cpydist/__init__.py
@@ -317,6 +317,7 @@ class BuildExt(build_ext, BaseCommand):
                 ext.libraries.append("libprotobuf" if os.name == "nt" else "protobuf")
                 # Add required standard needed for libprotobuf >=3.6.1
                 if os.name == "nt":
+                    ext.libraries.append("abseil_dll")
                     ext.extra_compile_args.append("/std:c++17")
                 else:
                     ext.extra_compile_args.append("-std=c++17")
-- 
2.45.2

