From 4759f9f9347cafbb502839938378ac5bfa83b24b Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Thu, 8 Aug 2024 22:05:18 +0100
Subject: [PATCH 6/6] Don't ignore user provided xtra compile & link args

---
 mysqlx-connector-python/cpydist/__init__.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mysqlx-connector-python/cpydist/__init__.py b/mysqlx-connector-python/cpydist/__init__.py
index efbfa2b..893eed4 100644
--- a/mysqlx-connector-python/cpydist/__init__.py
+++ b/mysqlx-connector-python/cpydist/__init__.py
@@ -315,6 +315,12 @@ class BuildExt(build_ext, BaseCommand):
                 ext.include_dirs.append(self.with_protobuf_include_dir)
                 ext.library_dirs.append(self._build_protobuf_lib_dir)
                 ext.libraries.append("libprotobuf" if os.name == "nt" else "protobuf")
+                # Add extra compile args
+                if self.extra_compile_args:
+                    ext.extra_compile_args.extend(self.extra_compile_args.split())
+                # Add extra link args
+                if self.extra_link_args:
+                    ext.extra_link_args.extend(self.extra_link_args.split())
                 # Add required standard needed for libprotobuf >=3.6.1
                 if os.name == "nt":
                     ext.libraries.append("abseil_dll")
-- 
2.45.2

