diff --git a/setup.py b/setup.py
index e933dc8..5b37080 100644
--- a/setup.py
+++ b/setup.py
@@ -13,6 +13,7 @@ def read(fname):
 
 setup(
     name='pytest-chalice',
+    version='0.0.4',
     license='MIT',
 
     author='Satoshi SUZUKI',
@@ -30,9 +31,6 @@ setup(
         'pytest>=3.5.0',
     ],
 
-    setup_requires=['setuptools_scm'],
-    use_scm_version=True,
-
     classifiers=[
         'Development Status :: 4 - Beta',
         'Framework :: Pytest',
