--- sat-search-0.2.1.orig/setup.py	2019-02-14 19:38:00.000000000 -0200
+++ sat-search-0.2.1/setup.py	2019-06-21 13:12:18.988181777 -0300
@@ -42,7 +42,7 @@
     packages=find_packages(exclude=['docs', 'tests*']),
     include_package_data=True,
     install_requires=install_requires,
-    dependency_links=dependency_links,
+    # dependency_links=dependency_links,
     setup_requires=['pytest-runner'],
     tests_require=['pytest'],
 )
