--- a/interface_cuda/interface.cpp	Fri Nov 15 18:07:15 2019 -0800
+++ b/interface_cuda/interface.cpp	Fri Nov 15 18:08:09 2019 -0800
@@ -447,7 +447,7 @@
             err = cudaPointerGetAttributes( &attr, const_cast<void*>( A ));
             if ( ! err ) {
                 // definitely know type
-                return (attr.memoryType == cudaMemoryTypeDevice);
+                return (attr.type == cudaMemoryTypeDevice);
             }
             else if ( err == cudaErrorInvalidValue ) {
                 // clear error; see http://icl.cs.utk.edu/magma/forum/viewtopic.php?f=2&t=529
