diff --git a/common/Container.i b/common/Container.i
index 86a0de311..9c83fc165 100644
--- a/common/Container.i
+++ b/common/Container.i
@@ -30,7 +30,7 @@
 #ifndef CONTAINER_I
 #define CONTAINER_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 
diff --git a/common/Detectors.i b/common/Detectors.i
index 3b9c62d48..bda2e0260 100644
--- a/common/Detectors.i
+++ b/common/Detectors.i
@@ -30,7 +30,7 @@
 #ifndef SWIG__COMMON__FR_DECTOR_I
 #define SWIG__COMMON__FR_DETECTOR_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %import "framecpp/Detectors.hh"
 
diff --git a/common/Dimension.i b/common/Dimension.i
index 882c576ab..c43437648 100644
--- a/common/Dimension.i
+++ b/common/Dimension.i
@@ -30,7 +30,7 @@
 #ifndef DIMENSION_I
 #define DIMENSION_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "ldastoolsal/types.hh"
diff --git a/common/FrAdcData.i b/common/FrAdcData.i
index 3978b3e22..33b4ffb44 100644
--- a/common/FrAdcData.i
+++ b/common/FrAdcData.i
@@ -31,7 +31,7 @@
 #define FR_ADC_DATA_I
 
 /** @cond IGNORE_BY_DOXYGEN */
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "ldastoolsal/types.hh"
diff --git a/common/FrDetector.i b/common/FrDetector.i
index 9c0fd1e9f..79a4e0ed6 100644
--- a/common/FrDetector.i
+++ b/common/FrDetector.i
@@ -30,7 +30,7 @@
 #ifndef SWIG__COMMON__FR_DETECTOR_I
 #define SWIG__COMMON__FR_DETECTOR_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "ldastoolsal/types.hh"
diff --git a/common/FrEvent.i b/common/FrEvent.i
index 8c942ab0f..e6d782478 100644
--- a/common/FrEvent.i
+++ b/common/FrEvent.i
@@ -30,7 +30,7 @@
 #ifndef FR_EVENT_I
 #define FR_EVENT_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %include "std_pair.i"
 %include "std_string.i"
diff --git a/common/FrHistory.i b/common/FrHistory.i
index 512bb5b61..e84ca2640 100644
--- a/common/FrHistory.i
+++ b/common/FrHistory.i
@@ -30,7 +30,7 @@
 #ifndef FR_HISTORY_I
 #define FR_HISTORY_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "ldastoolsal/types.hh"
diff --git a/common/FrProcData.i b/common/FrProcData.i
index a4c9be616..e48db4689 100644
--- a/common/FrProcData.i
+++ b/common/FrProcData.i
@@ -30,7 +30,7 @@
 #ifndef FR_PROC_DATA_I
 #define FR_PROC_DATA_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "ldastoolsal/types.hh"
diff --git a/common/FrSimData.i b/common/FrSimData.i
index 02b9fb70f..767a6b877 100644
--- a/common/FrSimData.i
+++ b/common/FrSimData.i
@@ -30,7 +30,7 @@
 #ifndef FR_SIM_DATA_I
 #define FR_SIM_DATA_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %include <attribute.i>
 
diff --git a/common/FrSimEvent.i b/common/FrSimEvent.i
index 9dfb8f421..09c349a1b 100644
--- a/common/FrSimEvent.i
+++ b/common/FrSimEvent.i
@@ -30,7 +30,7 @@
 #ifndef FR_SIM_EVENT_I
 #define FR_SIM_EVENT_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %include <attribute.i>
 
diff --git a/common/FrTOC.i b/common/FrTOC.i
index 10f8bb201..919fe1947 100644
--- a/common/FrTOC.i
+++ b/common/FrTOC.i
@@ -27,7 +27,7 @@
 
 /* -*- mode: c++ -*- */
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %include <attribute.i>
 
diff --git a/common/FrTOCAdc.i b/common/FrTOCAdc.i
index a078dfc02..1dbec33ee 100644
--- a/common/FrTOCAdc.i
+++ b/common/FrTOCAdc.i
@@ -27,7 +27,7 @@
 
 /* -*- mode: c++ -*- */
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 typedef FrameCPP::FrTOC::MapADC_type	MapADC_type;
diff --git a/common/FrTOCProc.i b/common/FrTOCProc.i
index 9e6c19778..8fde55d33 100644
--- a/common/FrTOCProc.i
+++ b/common/FrTOCProc.i
@@ -30,7 +30,7 @@
 #ifndef FR_TOC_PROC_I
 #define FR_TOC_PROC_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 typedef FrameCPP::FrTOC::MapProc_type	MapProc_type;
diff --git a/common/FrTOCSim.i b/common/FrTOCSim.i
index 46b36c526..d1c2e4895 100644
--- a/common/FrTOCSim.i
+++ b/common/FrTOCSim.i
@@ -27,7 +27,7 @@
 
 /* -*- mode: c++ -*- */
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 typedef FrameCPP::FrTOC::MapSim_type	MapSim_type;
diff --git a/common/FrTable.i b/common/FrTable.i
index eb49f5292..787a7f5ec 100644
--- a/common/FrTable.i
+++ b/common/FrTable.i
@@ -30,7 +30,7 @@
 #ifndef SWIG__COMMON__FR_TABLE_I
 #define SWIG__COMMON__FR_TABLE_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "ldastoolsal/types.hh"
diff --git a/common/FrVect.i b/common/FrVect.i
index 2283975b3..3ccbcdef4 100644
--- a/common/FrVect.i
+++ b/common/FrVect.i
@@ -36,7 +36,7 @@
  * @{
  */
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %feature("autodoc","1");
 
diff --git a/common/GPSTime.i b/common/GPSTime.i
index 3c6b5c4d3..fe30a3796 100644
--- a/common/GPSTime.i
+++ b/common/GPSTime.i
@@ -27,7 +27,7 @@
 
 /* -*- mode: c++ -*- */
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "framecpp/GPSTime.hh"
diff --git a/common/IFrameFStream.i b/common/IFrameFStream.i
index 2ad31e8f9..8584681b8 100644
--- a/common/IFrameFStream.i
+++ b/common/IFrameFStream.i
@@ -30,7 +30,7 @@
 #ifndef IFRAME_FSTREAM_I
 #define IFRAME_FSTREAM_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "framecpp/IFrameStream.hh"
diff --git a/common/OFrameFStream.i b/common/OFrameFStream.i
index 6737df6d0..15a9015ef 100644
--- a/common/OFrameFStream.i
+++ b/common/OFrameFStream.i
@@ -30,7 +30,7 @@
 #ifndef OFRAME_FSTREAM_I
 #define OFRAME_FSTREAM_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %import "framecpp/OFrameStream.hh"
 
diff --git a/common/STRING.i b/common/STRING.i
index de640e50d..566b5f834 100644
--- a/common/STRING.i
+++ b/common/STRING.i
@@ -27,7 +27,7 @@
 
 /* -*- mode: c++ -*- */
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 
diff --git a/common/SearchContainer.i b/common/SearchContainer.i
index 5f04b6236..713ae5e96 100644
--- a/common/SearchContainer.i
+++ b/common/SearchContainer.i
@@ -30,7 +30,7 @@
 #ifndef SEARCH_CONTAINER_I
 #define SEARCH_CONTAINER_I
 
-%module frameCPP
+%module(package="LDAStools") frameCPP
 
 %{
 #include "ldastoolsal/types.hh"
diff --git a/common/frameCPP.i b/common/frameCPP.i
index 628af6c9c..03d6ce7fa 100644
--- a/common/frameCPP.i
+++ b/common/frameCPP.i
@@ -37,7 +37,7 @@ developers to interface with the frameCPP library."
 %enddef
 #endif /* SWIGIMPORTED */
 
-%module(docstring=DOCSTRING) frameCPP
+%module(docstring=DOCSTRING, package="LDAStools") frameCPP
 %feature("autodoc","1");
 
 %include <boost_shared_ptr.i>
