diff --git a/setup.py b/setup.py
index 7429a8e..9a1e2ae 100644
--- a/setup.py
+++ b/setup.py
@@ -108,9 +108,13 @@ if not os.getenv("READTHEDOCS"):
                         "-DCMAKE_SYSTEM_PROCESSOR=ARM64",
                     ]
                 else:  # X64 arch
-                    llvmpath = (
-                        subprocess.check_output(["brew", "--prefix", "llvm"]).decode().strip()
-                    )
+                    if shutil.which("brew"):
+                        llvmpath = (
+                            subprocess.check_output(["brew", "--prefix", "llvm"]).decode().strip()
+                        )
+                    else:
+                        llvmpath = shutil.which("clang++")
+                        llvmpath = Path(llvmpath).parent.parent
                     configure_args += [
                         f"-DCMAKE_CXX_COMPILER={llvmpath}/bin/clang++",
                         f"-DCMAKE_LINKER={llvmpath}/bin/lld",
