diff --git a/pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp b/pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp
index ca033a6..2c007aa 100644
--- a/pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp
+++ b/pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp
@@ -18,7 +18,6 @@
 
 #pragma once
 #include <cstdlib>
-#include <memory>
 #include <unordered_map>
 #include <unordered_set>
 #include <vector>
diff --git a/pennylane_lightning_kokkos/src/util/Util.hpp b/pennylane_lightning_kokkos/src/util/Util.hpp
index d261d63..af0088a 100644
--- a/pennylane_lightning_kokkos/src/util/Util.hpp
+++ b/pennylane_lightning_kokkos/src/util/Util.hpp
@@ -24,7 +24,6 @@
 #include <cstddef>
 #include <iostream>
 #include <limits>
-#include <memory>
 #include <numeric>
 #include <set>
 #include <stdexcept>
