From 4c376bf2281e7d2f3f4cb90295464ca28c4ca723 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 21 Feb 2021 11:41:36 +0100
Subject: [PATCH] Update Makefile

---
 Makefile | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/Makefile b/Makefile
index 85baa2eee..d6c26723c 100644
--- a/Makefile
+++ b/Makefile
@@ -111,8 +111,8 @@ $(info Using QUIP_ARCH=${QUIP_ARCH}, MODULES=${MODULES}, QUIP_ROOT=${QUIP_ROOT})
 all: ${MODULES} ${GAPFIT} ${GAP_PROGRAMS} ${PROGRAMS}
 
 FOX = fox
-export FOX_LIBDIR=${QUIP_ROOT}/src/${FOX}/objs.${QUIP_ARCH}/lib
-export FOX_INCDIR=${QUIP_ROOT}/src/${FOX}/objs.${QUIP_ARCH}/finclude
+export FOX_LIBDIR=${PREFIX}/lib
+export FOX_INCDIR=${PREFIX}/finclude
 
 # now we can include the config makefile, it needs to come after the default target
 include Makefile.config
@@ -126,19 +126,6 @@ ${BUILDDIR}/Makefile.inc:
 		exit 1 ;\
 		fi
 
-# Automatically pull the submodules if the user didn't.
-# Remove the Makefile.QUIP if it has failed previously.
-src/${FOX}/configure:
-	@echo "Attempting to automatically clone fox submodule"
-	rm -f src/${FOX}/Makefile.QUIP
-	git submodule update --init src/${FOX} || \
-	    { echo "fox clone failed. Download it manually" ; exit 1 ; }
-
-${FOX}: src/${FOX}/configure src/${FOX}/objs.${QUIP_ARCH}/lib/libFoX_common.a
-src/${FOX}/objs.${QUIP_ARCH}/lib/libFoX_common.a:
-	cp Makefile.fox src/${FOX}/Makefile.QUIP
-	make -C src/${FOX} -I${PWD} -I${PWD}/arch -I${BUILDDIR} -f Makefile.QUIP
-
 FOX_STATIC_LIBFILES = $(patsubst -l%,${FOX_LIBDIR}/lib%.a,${FOX_LIBS})
 FOX_STATIC_LIBFILE_OBJS = $(shell for i in ${FOX_STATIC_LIBFILES}; do ar -t $$i; done | grep \.o)
 
