From 6ae52010e71293f5714defe80caf05182abc1300 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 21 Feb 2021 11:28:31 +0100
Subject: [PATCH] Update Makefile

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 85baa2eee..12a6a83f1 100644
--- a/Makefile
+++ b/Makefile
@@ -111,8 +111,8 @@ $(info Using QUIP_ARCH=${QUIP_ARCH}, MODULES=${MODULES}, QUIP_ROOT=${QUIP_ROOT})
 all: ${MODULES} ${GAPFIT} ${GAP_PROGRAMS} ${PROGRAMS}
 
 FOX = fox
-export FOX_LIBDIR=${QUIP_ROOT}/src/${FOX}/objs.${QUIP_ARCH}/lib
-export FOX_INCDIR=${QUIP_ROOT}/src/${FOX}/objs.${QUIP_ARCH}/finclude
+export FOX_LIBDIR=${PREFIX}/lib
+export FOX_INCDIR=${PREFIX}/finclude
 
 # now we can include the config makefile, it needs to come after the default target
 include Makefile.config
