From eef713711029b14f365ef2d8d009cc06c45e1f26 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Wed, 5 Aug 2020 22:17:48 +0200
Subject: [PATCH] Update Makefile.linux_x86_64_gfortran

---
 arch/Makefile.linux_x86_64_gfortran | 43 +++++++++++++++++++++++------
 1 file changed, 35 insertions(+), 8 deletions(-)

diff --git a/arch/Makefile.linux_x86_64_gfortran b/arch/Makefile.linux_x86_64_gfortran
index f0921732b..f5b335a77 100644
--- a/arch/Makefile.linux_x86_64_gfortran
+++ b/arch/Makefile.linux_x86_64_gfortran
@@ -30,13 +30,13 @@
 
 # declarations
 
-F77 = gfortran
-F90 = gfortran
-F95 = gfortran
-CC = gcc
-CPLUSPLUS = g++
-LINKER = gfortran
-FPP = gfortran -E -x f95-cpp-input
+F77 = ${GFORTRAN}
+F90 = ${GFORTRAN}
+F95 = ${GFORTRAN}
+CC = ${GCC}
+CPLUSPLUS = ${GCC}
+LINKER = ${GFORTRAN}
+FPP = ${GFORTRAN} -E -x f95-cpp-input
 
 SAMPLE_DEBUG = -O0 -g -DDUMP_CORE_ON_ABORT -DDEBUG -fbounds-check
 DEBUG= 
@@ -58,9 +58,36 @@ ATOMEYE_LIBS = -L/usr/lib -Wl,-Bstatic  -lm \
 QUIPPY_FCOMPILER = gnu95
 QUIPPY_F90FLAGS = -ffree-line-length-none
 QUIPPY_DEFINES =  -DGETARG_F2003 -DGETENV_F2003 -DGFORTRAN -DFORTRAN_UNDERSCORE
-QUIPPY_CPP = gfortran -E -x f95-cpp-input
+QUIPPY_CPP = ${FC} -E -x f95-cpp-input
 QUIPPY_DEBUG = 0
 
 export DEFAULT_MATH_LINKOPTS = -llapack -lblas
 
 FPP_PRE_TARGET_STRING = -o
+
+USE_MAKEDEP=1
+HAVE_CP2K=0
+HAVE_VASP=0
+HAVE_TB=0
+HAVE_PRECON=1
+HAVE_LOTF=0
+HAVE_ONIOM=0
+HAVE_LOCAL_E_MIX=0
+HAVE_QC=0
+HAVE_GAP=0
+HAVE_DESCRIPTORS_EXTERNAL=0
+HAVE_TURBOGAP=0
+HAVE_QR=1
+HAVE_THIRDPARTY=0
+HAVE_FX=0
+HAVE_SCME=0
+HAVE_MTP=0
+HAVE_MBD=0
+HAVE_TTM_NF=0
+HAVE_CH4=0
+HAVE_NETCDF4=0
+HAVE_MDCORE=0
+HAVE_ASAP=0
+HAVE_CGAL=0
+HAVE_METIS=0
+HAVE_LMTO_TBE=0
