From 879dd1b1db8a8d8efe5432040a8b219ca6f355e9 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Wed, 5 Aug 2020 21:53:15 +0200
Subject: [PATCH] Add files via upload

---
 arch/Makefile.linux_x86_64_gfortran | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/Makefile.linux_x86_64_gfortran b/arch/Makefile.linux_x86_64_gfortran
index f0921732b..341096c2b 100644
--- a/arch/Makefile.linux_x86_64_gfortran
+++ b/arch/Makefile.linux_x86_64_gfortran
@@ -64,3 +64,30 @@ QUIPPY_DEBUG = 0
 export DEFAULT_MATH_LINKOPTS = -llapack -lblas
 
 FPP_PRE_TARGET_STRING = -o
+
+USE_MAKEDEP=1
+HAVE_CP2K=0
+HAVE_VASP=0
+HAVE_TB=0
+HAVE_PRECON=1
+HAVE_LOTF=0
+HAVE_ONIOM=0
+HAVE_LOCAL_E_MIX=0
+HAVE_QC=0
+HAVE_GAP=0
+HAVE_DESCRIPTORS_EXTERNAL=0
+HAVE_TURBOGAP=0
+HAVE_QR=1
+HAVE_THIRDPARTY=0
+HAVE_FX=0
+HAVE_SCME=0
+HAVE_MTP=0
+HAVE_MBD=0
+HAVE_TTM_NF=0
+HAVE_CH4=0
+HAVE_NETCDF4=0
+HAVE_MDCORE=0
+HAVE_ASAP=0
+HAVE_CGAL=0
+HAVE_METIS=0
+HAVE_LMTO_TBE=0
