From e08611602afb2946174a652223476391accd45e5 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 21 Feb 2021 16:20:18 +0100
Subject: [PATCH] Update Makefile.linux_x86_64_gfortran

---
 arch/Makefile.linux_x86_64_gfortran | 20 ++++++++------------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/arch/Makefile.linux_x86_64_gfortran b/arch/Makefile.linux_x86_64_gfortran
index f0921732b..c8c81a2b1 100644
--- a/arch/Makefile.linux_x86_64_gfortran
+++ b/arch/Makefile.linux_x86_64_gfortran
@@ -30,23 +30,19 @@
 
 # declarations
 
-F77 = gfortran
-F90 = gfortran
-F95 = gfortran
-CC = gcc
-CPLUSPLUS = g++
-LINKER = gfortran
-FPP = gfortran -E -x f95-cpp-input
+CPLUSPLUS = $(CPP)
+LINKER = $(FC)
+FPP = $(FC) -E -x f95-cpp-input
 
 SAMPLE_DEBUG = -O0 -g -DDUMP_CORE_ON_ABORT -DDEBUG -fbounds-check
 DEBUG= 
 OPTIM = -O3
 
 DEFINES += -DGETARG_F2003 -DGETENV_F2003 -DGFORTRAN -DFORTRAN_UNDERSCORE
-F95FLAGS += -x f95-cpp-input -ffree-line-length-none -ffree-form -fno-second-underscore -fPIC
-F77FLAGS += -x f77-cpp-input -fno-second-underscore -fPIC
+F95FLAGS += $(FFLAGS) -x f95-cpp-input -ffree-line-length-none -ffree-form -fno-second-underscore -fPIC
+F77FLAGS += $(FFLAGS) -x f77-cpp-input -fno-second-underscore -fPIC
 CFLAGS += -fPIC 
-CPLUSPLUSFLAGS += -fPIC 
+CPLUSPLUSFLAGS += $(CPPFLAGS) -fPIC 
 
 AR_ADD = src
 
@@ -56,9 +52,9 @@ ATOMEYE_LIBS = -L/usr/lib -Wl,-Bstatic  -lm \
    -llapack -lblas -Wl,-Bdynamic
 
 QUIPPY_FCOMPILER = gnu95
-QUIPPY_F90FLAGS = -ffree-line-length-none
+QUIPPY_F90FLAGS = $(FFLAGS) -ffree-line-length-none
 QUIPPY_DEFINES =  -DGETARG_F2003 -DGETENV_F2003 -DGFORTRAN -DFORTRAN_UNDERSCORE
-QUIPPY_CPP = gfortran -E -x f95-cpp-input
+QUIPPY_CPP = $(FC) -E -x f95-cpp-input
 QUIPPY_DEBUG = 0
 
 export DEFAULT_MATH_LINKOPTS = -llapack -lblas
