From 0f33c94835ef25995f218d2747cd239b6db9f116 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 21 Feb 2021 15:18:53 +0100
Subject: [PATCH] Update Makefile

---
 quippy/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quippy/Makefile b/quippy/Makefile
index db63c5fd8..f5c374fa5 100644
--- a/quippy/Makefile
+++ b/quippy/Makefile
@@ -64,7 +64,7 @@ F2PY_LINK_ARGS = $(shell ${PYTHON} -c 'import sys; print(" ".join([arg for arg i
 
 F90WRAP_PATH = $(shell ${PIP} show f90wrap | grep Location | sed -e 's/Location: //' -E -e 's|/lib(64)?/.*|/bin|')
 
-F90WRAP = ${F90WRAP_PATH}/f90wrap
+F90WRAP = ${PREFIX}/bin/f90wrap
 F2PY_F90WRAP = ${F90WRAP_PATH}/f2py-f90wrap
 
 #################################
