diff --git a/meson.build b/meson.build
index b82cf72e..a22f6386 100644
--- a/meson.build
+++ b/meson.build
@@ -17,7 +17,6 @@ endif
 
 # These requirements are kept in both `bin/meld` and `meson.build`. If you
 # update one, update the other.
-dependency('python3', version : '>= 3.6')
 dependency('gtk+-3.0', version: '>= 3.22')
 dependency('glib-2.0', version: '>= 2.48')
 dependency('gtksourceview-4', version: '>= 4.0.0')
