diff --git a/python/uv/_find_uv.py b/python/uv/_find_uv.py
index 00b7d88..8112137 100644
--- a/python/uv/_find_uv.py
+++ b/python/uv/_find_uv.py
@@ -7,7 +7,10 @@ import sysconfig
 
 def find_uv_bin() -> str:
     """Return the uv binary path."""
-
+    return os.path.join(
+        sys.prefix,
+        *(("Library", "bin", "uv.exe") if os.name == "nt" else ("bin", "uv"))
+    )
     uv_exe = "uv" + sysconfig.get_config_var("EXE")
 
     path = os.path.join(sysconfig.get_path("scripts"), uv_exe)
