--- a/prov/cxi/src/cxip_msg_hpc.c
+++ b/prov/cxi/src/cxip_msg_hpc.c
@@ -2097,8 +2097,12 @@
 
 	do {
 		cur_ule_count = pte_status.ule_count;
-		*ule_offsets = reallocarray(*ule_offsets, cur_ule_count,
-					    sizeof(*ule_offsets));
+                size_t new_size = cur_ule_count * sizeof(uint64_t);
+                if (cur_ule_count > SIZE_MAX / sizeof(uint64_t)) {
+                        ret = -FI_ENOMEM;  /* Overflow */
+                        goto err;
+                }
+                *ule_offsets = realloc(*ule_offsets, new_size);
 		if (*ule_offsets == NULL) {
 			RXC_WARN(rxc, "Failed allocate ule offset memory\n");
 			ret = -FI_ENOMEM;
