From 7e7dc84f731a841fa67594de10b0cfc8da46632a Mon Sep 17 00:00:00 2001
From: Michalis Agathos <magathos@damtp.cam.ac.uk>
Date: Wed, 7 Jun 2023 14:14:25 +0100
Subject: [PATCH] minimise dependencies

---
 pyproject.toml   | 1 -
 requirements.txt | 1 -
 2 files changed, 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index f496a16..ac5f475 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,7 +32,6 @@ license = { text = "MIT" }
 version = "0.1a1"
 dependencies = [
     "bilby",
-    "lalsuite",
     "numpy>=1.17"
 ]
 classifiers = [
diff --git a/requirements.txt b/requirements.txt
index f32e630..fb6da21 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,3 +1,2 @@
 bilby
-lalsuite
 numpy>=1.17
-- 
2.37.1 (Apple Git-137.1)

