diff --git a/squashfs-tools/Makefile b/squashfs-tools/Makefile
index 52d2582..e20d9b5 100644
--- a/squashfs-tools/Makefile
+++ b/squashfs-tools/Makefile
@@ -26,7 +26,7 @@ GZIP_SUPPORT = 1
 # To build using XZ Utils liblzma - install the library and uncomment
 # the XZ_SUPPORT line below.
 #
-#XZ_SUPPORT = 1
+XZ_SUPPORT = 1
 
 
 ############ Building LZO support ##############
@@ -37,7 +37,7 @@ GZIP_SUPPORT = 1
 # LZO_SUPPORT line below. If needed, uncomment and set LZO_DIR to the
 # installation prefix.
 #
-#LZO_SUPPORT = 1
+LZO_SUPPORT = 1
 #LZO_DIR = /usr/local
 
 
@@ -51,7 +51,7 @@ GZIP_SUPPORT = 1
 # install and uncomment
 # the LZ4_SUPPORT line below.
 #
-#LZ4_SUPPORT = 1
+LZ4_SUPPORT = 1
 
 
 ########### Building LZMA support #############
@@ -108,7 +108,7 @@ XATTR_DEFAULT = 1
 ###############################################
 
 INCLUDEDIR = -I.
-INSTALL_DIR = /usr/local/bin
+INSTALL_DIR = $(PREFIX)/bin
 
 MKSQUASHFS_OBJS = mksquashfs.o read_fs.o action.o swap.o pseudo.o compressor.o \
 	sort.o progressbar.o read_file.o info.o restore.o process_fragments.o \
