From aa608aa07c34dbb85a02c18310aa2f1418df5a38 Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Fri, 13 Nov 2020 12:28:36 +0100
Subject: [PATCH] [OSX] Fix python dep for osx

---
 makepanda/makewheel.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git makepanda/makewheel.py makepanda/makewheel.py
index 4caebcc9e3..d995cb7e52 100644
--- makepanda/makewheel.py
+++ makepanda/makewheel.py
@@ -609,7 +609,7 @@ def makewheel(version, output_dir, platform=None):
         pylib_name = 'python{0}{1}.dll'.format(*sys.version_info)
         pylib_path = os.path.join(get_config_var('BINDIR'), pylib_name)
     elif sys.platform == 'darwin':
-        pylib_name = 'libpython{0}.{1}.dylib'.format(*sys.version_info)
+        pylib_name = get_config_var('LDLIBRARY')
         pylib_path = os.path.join(get_config_var('LIBDIR'), pylib_name)
     else:
         pylib_name = get_config_var('LDLIBRARY')
-- 
2.17.1

