diff --git a/sources/pyside6/cmake/Macros/PySideModules.cmake b/sources/pyside6/cmake/Macros/PySideModules.cmake
index 462c47441..fc3a63b07 100644
--- a/sources/pyside6/cmake/Macros/PySideModules.cmake
+++ b/sources/pyside6/cmake/Macros/PySideModules.cmake
@@ -207,7 +207,7 @@ macro(create_pyside_module)
     endif()
 
     # Add target to generate pyi file, which depends on the module target.
-    add_custom_target("${module_NAME}_pyi" ALL
+    add_custom_target("${module_NAME}_pyi"
                       COMMAND ${CMAKE_COMMAND} -E env ${ld_prefix}
                       "${SHIBOKEN_PYTHON_INTERPRETER}"
                       "${CMAKE_CURRENT_SOURCE_DIR}/../support/generate_pyi.py" ${generate_pyi_options})
@@ -216,9 +216,9 @@ macro(create_pyside_module)
     # install
     install(TARGETS ${module_NAME} LIBRARY DESTINATION "${PYTHON_SITE_PACKAGES}/PySide6")
 
-    file(GLOB hinting_stub_files RELATIVE "${CMAKE_CURRENT_BINARY_DIR}/PySide6" "${CMAKE_CURRENT_BINARY_DIR}/PySide6/*.pyi")
-    install(FILES ${hinting_stub_files}
-            DESTINATION "${PYTHON_SITE_PACKAGES}/PySide6")
+    #file(GLOB hinting_stub_files RELATIVE "${CMAKE_CURRENT_BINARY_DIR}/PySide6" "${CMAKE_CURRENT_BINARY_DIR}/PySide6/*.pyi")
+    #install(FILES ${hinting_stub_files}
+    #        DESTINATION "${PYTHON_SITE_PACKAGES}/PySide6")
 
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/PySide6/${module_NAME}/pyside6_${lower_module_name}_python.h
             DESTINATION include/PySide6${pyside6_SUFFIX}/${module_NAME}/)
