diff --git a/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp b/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp
index fb42b78..9596e8d 100644
--- a/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp
+++ b/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp
@@ -323,6 +323,14 @@ QByteArrayList emulatedCompilerOptions()
     case Compiler::Gpp:
         if (needsClangBuiltinIncludes())
             appendClangBuiltinIncludes(&headerPaths);
+
+        // Append the c++ include paths since Clang is unable to find <type_traits> etc
+        const HeaderPaths gppPaths = gppInternalIncludePaths(compilerFromCMake(QStringLiteral("g++")));
+        for (const HeaderPath & h : gppPaths) {
+            if (h.path.contains("c++") || h.path.contains("sysroot")) {
+                headerPaths.append(h);
+            }
+        }
         break;
     }
 
