diff --git a/tests/conftest.py b/tests/conftest.py
index e6f8b4f..dbf0c84 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -37,7 +37,7 @@ def new_project(plugin_dir, compiled_extension, tmp_path) -> Generator[Path, Non
     project_file.write_text(
         f"""\
 [build-system]
-requires = ["hatchling", "hatch-mypyc @ {plugin_dir.as_uri()}"]
+requires = ["hatchling", "hatch-mypyc"]
 build-backend = "hatchling.build"

 [project]
