--- ../../_h_env/Library/include/xtensor/xmath.hpp
+++ ../../_h_env/Library/include/xtensor/xmath.hpp	
@@ -279,7 +279,9 @@ XTENSOR_INT_SPECIALIZATION_IMPL(FUNC_NAME, RETURN_VAL, unsigned long long);
         using std::arg;

         using std::atan2;
-        using std::copysign;
+        #if !defined(_MSC_VER)
+            using std::copysign;
+        #endif
         using std::fdim;
         using std::fmax;
         using std::fmin;
