diff --git a/warp/build_dll.py b/warp/build_dll.py
--- a/warp/build_dll.py
+++ b/warp/build_dll.py
@@ -101,8 +101,10 @@ def find_host_compiler():
     else:
         # try and find g++
         try:
-            return run_cmd("which g++").decode()
+            return run_cmd("echo $CXX").decode()
         except:
+            return run_cmd("which g++").decode()
+        finally:
             return ""
