diff --git a/build_llvm.py b/build_llvm.py
--- a/build_llvm.py
+++ b/build_llvm.py
@@ -43,10 +43,6 @@ for arch in packages:
 
 
 def build_from_source_for_arch(args, arch):
-    # Install dependencies
-    subprocess.check_call([sys.executable, "-m", "pip", "install", "gitpython"])
-    subprocess.check_call([sys.executable, "-m", "pip", "install", "cmake"])
-    subprocess.check_call([sys.executable, "-m", "pip", "install", "ninja"])
 
     from git import Repo
 
