From 9fe2004de4630cdf2cb4e8c21d250808e75f9ee5 Mon Sep 17 00:00:00 2001
From: Rolando Espinoza <rndmax84@gmail.com>
Date: Mon, 13 Mar 2017 17:32:31 -0300
Subject: [PATCH 1/2] Fix VS2008 solution format version

---
 build/VS2008/zstd.sln | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/VS2008/zstd.sln b/build/VS2008/zstd.sln
index 97b88c8..96c8441 100644
--- a/build/VS2008/zstd.sln
+++ b/build/VS2008/zstd.sln
@@ -1,5 +1,5 @@
 
-Microsoft Visual Studio Solution File, Format Version 10.00
+Microsoft Visual Studio Solution File, Format Version 9.00
 # Visual Studio 2008
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "zstd", "zstd\zstd.vcproj", "{1A2AB08E-5CE7-4C5B-BE55-458157C14051}"
 EndProject
-- 
2.6.2

