 diff -ur gap.shi gap.shi
--- gap.shi 2012-02-20 20:04:01.000000000 +0000
+++ gap.shi 2012-09-02 10:44:27.488429081 +0100
@@ -20,7 +20,7 @@
 ##  You won't have to change this unless you move the installation.
 ##
 if [ "x$GAP_DIR" = "x" ];  then
-GAP_DIR="@gapdir@"
+    GAP_DIR=`dirname $0`/../gap/latest
 fi
 
 
@@ -35,9 +35,9 @@
 ##  in parallel with other programs you may want to set this value close
 ##  to the amount of memory your computer has.
 ##
-#if [ "x$GAP_MEM" = "x" ];  then
-#GAP_MEM="-m 256m"
-#fi
+if [ "x$GAP_MEM" = "x" ];  then
+    GAP_MEM="-m 24m"
+fi

