diff --git a/rembg/cli.py b/rembg/cli.py
index 6935dc9..2f1e9f9 100644
--- a/rembg/cli.py
+++ b/rembg/cli.py
@@ -8,6 +8,9 @@ def main() -> None:
         if extra == "cli":
             requirements = package_distribution.requires(extras=(extra,))
             for requirement in requirements:
+                if requirement.project_name == 'opencv-python-headless':
+                    continue
+
                 try:
                     pkg_resources.require(requirement.project_name)
                 except pkg_resources.DistributionNotFound:
