diff --git a/scripts/build.ts b/scripts/build.ts
index 57f0fe3..0a46fe2 100644
--- a/scripts/build.ts
+++ b/scripts/build.ts
@@ -371,6 +371,7 @@ class Builder {
 	 */
 	private async binary(targetPath: string, binariesPath: string, binaryName: string): Promise<void> {
 		const bin = new Binary({
+			nodePath: process.env.NODE_BINARY_PATH,
 			mainFile: path.join(targetPath, "out/vs/server/main.js"),
 			target: await this.target(),
 		});
