diff --git a/CMakeLists.txt b/CMakeLists.txt
index a75f15d8..6a7a0c91 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ project(vsg
 )
 set(VSG_SOVERSION 14)
 SET(VSG_RELEASE_CANDIDATE 0)
-set(Vulkan_MIN_VERSION 1.1.70.0)
+set(Vulkan_MIN_VERSION )

 set(VSG_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}" CACHE INTERNAL "Root source directory of VulkanSceneGraph.")
 set(VSG_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}" CACHE INTERNAL "Root binary directory of VulkanSceneGraph.")
