diff --git ebooklib/setup.py ebooklib/setup.py
index b3a9a6d..37a25e8 100644
--- ebooklib/setup.py
+++ ebooklib/setup.py
@@ -10,7 +10,7 @@
     url = 'https://github.com/aerkalov/ebooklib',
     license = 'GNU Affero General Public License',
     description = 'Ebook library which can handle EPUB2/EPUB3 and Kindle format',
-    long_description = open('README.md').read(),
+    long_description = 'Windows and py3k had trouble with ingesting README.md here. See README.md directly',
     keywords = ['ebook', 'epub', 'kindle'],
     classifiers = [
         "Development Status :: 4 - Beta",
