diff --git a/test/src/reader01.cpp b/test/src/reader01.cpp
index f6848ef..eb01ccc 100644
--- a/test/src/reader01.cpp
+++ b/test/src/reader01.cpp
@@ -8,6 +8,11 @@
 
 #include <hffix.hpp>
 
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
 const size_t chunksize = 4096; // Choose a preferred I/O chunk size.
 
 char buffer[1 << 20]; // Must be larger than the largest FIX message size.
diff --git a/util/src/fixprint.cpp b/util/src/fixprint.cpp
index 306dd75..41ada19 100644
--- a/util/src/fixprint.cpp
+++ b/util/src/fixprint.cpp
@@ -6,6 +6,11 @@
 #include <cstring>
 #include <map>
 
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
 const char color_field[]   = "\x1b" "[33m"; // Yellow
 const char color_value[]   = "\x1b" "[37m"; // White
 const char color_msgtype[] = "\x1b" "[32m"; // Green
