From c28d49b015b2c47b283401878d8f45cdfaf4371d Mon Sep 17 00:00:00 2001
From: Constantin Pape <c.pape@gmx.net>
Date: Sun, 7 Nov 2021 17:58:22 +0100
Subject: [PATCH 1/1] Skip tests that require nifty with hdf5 if it is not
 available

---
 test/transformation/test_affine.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/transformation/test_affine.py b/test/transformation/test_affine.py
index 589c2f7..e1ab83b 100644
--- a/test/transformation/test_affine.py
+++ b/test/transformation/test_affine.py
@@ -2,6 +2,7 @@ import os
 import unittest
 from shutil import rmtree
 
+import nifty
 import numpy as np
 from scipy.ndimage import affine_transform
 from elf.io import open_file
@@ -123,6 +124,7 @@ class TestAffine(unittest.TestCase):
     def test_affine_subvolume_3d_z5(self):
         self._test_affine_subvolume_3d_chunked('tmp.n5')
 
+    @unittest.skipUnless(nifty.Configuration.WITH_HDF5, "Needs nifty built with hdf5")
     def test_affine_subvolume_3d_h5(self):
         self._test_affine_subvolume_3d_chunked('tmp.h5')
 
-- 
2.23.0

