diff --git a/meson.build b/meson.build
index efeac23..8ae3237 100644
--- a/meson.build
+++ b/meson.build
@@ -38,7 +38,7 @@ conf.set('includedir', includedir)
 pkg = configure_file(configuration: conf,
                      input: 'gsettings-desktop-schemas.pc.in',
                      install: true,
-                     install_dir: join_paths(datadir, 'pkgconfig'),
+                     install_dir: join_paths(libdir, 'pkgconfig'),
                      output: 'gsettings-desktop-schemas.pc')
 
 glib_mkenums = find_program('glib-mkenums')
