--- msvc_project/VapourSynth.sln
+++ msvc_project/VapourSynth.sln
@@ -4,9 +4,6 @@
-VisualStudioVersion = 17.3.32901.215
+VisualStudioVersion = 17.1.32414.318
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "VapourSynth", "Core\Core.vcxproj", "{8C2696F2-47FC-425A-989E-9C02AE4E76D2}"
-	ProjectSection(ProjectDependencies) = postProject
-		{C8629E57-4D6F-40FA-82FC-B39824195932} = {C8629E57-4D6F-40FA-82FC-B39824195932}
-	EndProjectSection
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "VSVFW", "VSVFW\VSVFW.vcxproj", "{2B4C263D-1EF9-4656-8E15-FBF7C1B91BE0}"
 	ProjectSection(ProjectDependencies) = postProject
