diff --git a/dask-gateway-server/setup.py b/dask-gateway-server/setup.py
index 5df1357..8c84ca8 100644
--- a/dask-gateway-server/setup.py
+++ b/dask-gateway-server/setup.py
@@ -127,11 +127,7 @@ setup(
         open("README.rst").read() if os.path.exists("README.rst") else ""
     ),
     url="http://github.com/jcrist/dask-gateway/",
-    packages=[
-        "dask_gateway_server",
-        "dask_gateway_server.proxy",
-        "dask_gateway_server.managers",
-    ],
+    packages=setuptools.find_packages(),
     package_data={"dask_gateway_server": ["proxy/dask-gateway-proxy"]},
     install_requires=install_requires,
     extras_require=extras_require,

