From 4c3f4988da20d553e0feda08cc95dfa860923761 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sat, 30 Nov 2024 13:57:25 +0530
Subject: [PATCH 2/3] Do not force static freetype

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cbd57077..c010d2eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,7 +166,7 @@ endif()
 ibd_check_freetype_availability(freetype_default)
 # Note: to reduce wasm size, you may want to disable freetype on emscripten.
 option(HELLOIMGUI_USE_FREETYPE "Use freetype for text rendering" ${freetype_default})
-ibd_force_freetype_static_for_python()
+# ibd_force_freetype_static_for_python()
 
 # IMGUI_BUNDLE_BUILD_DEMOS: Build demos
 # -----------------------------------------------------------------------------
-- 
2.45.2

