diff --git a/external/hello_imgui/hello_imgui/hello_imgui_cmake/hello_imgui_build_lib.cmake b/external/hello_imgui/hello_imgui/hello_imgui_cmake/hello_imgui_build_lib.cmake
index 923da1b..26b0873 100644
--- a/external/hello_imgui/hello_imgui/hello_imgui_cmake/hello_imgui_build_lib.cmake
+++ b/external/hello_imgui/hello_imgui/hello_imgui_cmake/hello_imgui_build_lib.cmake
@@ -280,10 +280,11 @@ function(him_build_imgui)
             _him_do_build_imgui()
         endif()
         if (HELLOIMGUI_USE_FREETYPE)
-            _him_add_freetype_to_imgui()
-            if (HELLOIMGUI_USE_FREETYPE_PLUTOSVG)
-                _him_add_freetype_plutosvg_to_imgui()
-            endif()
+            find_package(Freetype REQUIRED)
+            target_link_libraries(imgui PRIVATE ${FREETYPE_LIBRARIES})
+            target_include_directories(imgui PRIVATE ${FREETYPE_INCLUDE_DIRS})
+            find_library(PLUTOVG_LIBRARIES plutovg REQUIRED)
+            target_link_libraries(imgui PRIVATE ${PLUTOVG_LIBRARIES})
         endif()
     endif()
 endfunction()
