From 49b9de2136c6d45c1fef514a7e5112b97ad00d10 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Wed, 4 Dec 2024 23:12:06 -0500
Subject: [PATCH] Find glfw correctly for conda-forge

---
 bindings/imgui_bundle/__init__.py | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/bindings/imgui_bundle/__init__.py b/bindings/imgui_bundle/__init__.py
index 950ac670..79f3b2a7 100644
--- a/bindings/imgui_bundle/__init__.py
+++ b/bindings/imgui_bundle/__init__.py
@@ -124,17 +124,6 @@ if has_submodule("immapp_cpp"):  # immapp is a Python wrapper around immapp_cpp
 from imgui_bundle import imgui_fig as imgui_fig
 __all__.extend(["imgui_fig"])
 
-
-# Glfw setup:
-# By importing imgui_bundle.glfw_utils, we make sure that glfw provided by pip will use our glfw dynamic library.
-# (imgui_bundle.glfw_utils will call _set_glfw_pip_search_path automatically)
-if has_submodule("with_glfw"):
-    from imgui_bundle._glfw_set_search_path import _glfw_set_search_path
-
-    _glfw_set_search_path()
-    from imgui_bundle import glfw_utils as glfw_utils  # noqa: E402
-
-
 #
 
 # Override assets folder
-- 
2.45.2

