diff --git a/bindings/imgui_bundle/demos_cpp/demos_immapp/CMakeLists.txt b/bindings/imgui_bundle/demos_cpp/demos_immapp/CMakeLists.txt
index 16358f94..a17f4c99 100644
--- a/bindings/imgui_bundle/demos_cpp/demos_immapp/CMakeLists.txt
+++ b/bindings/imgui_bundle/demos_cpp/demos_immapp/CMakeLists.txt
@@ -4,7 +4,7 @@ if(EMSCRIPTEN)
 else()
     if (UNIX OR (APPLE AND NOT IOS))
         find_package(OpenGL REQUIRED)
-        target_link_libraries(imgui_example_glfw_opengl3 PUBLIC OpenGL::GL)
+        target_link_libraries(imgui_example_glfw_opengl3 PRIVATE OpenGL::GL)
     endif()
 endif()
 
Submodule external/imgui-node-editor/imgui-node-editor contains modified content
diff --git a/external/imgui-node-editor/imgui-node-editor/examples/cmake/Findimgui.cmake b/external/imgui-node-editor/imgui-node-editor/examples/cmake/Findimgui.cmake
index f764f88..71bb95c 100644
--- a/external/imgui-node-editor/imgui-node-editor/examples/cmake/Findimgui.cmake
+++ b/external/imgui-node-editor/imgui-node-editor/examples/cmake/Findimgui.cmake
@@ -55,13 +55,13 @@ foreach(component ${imgui_FIND_COMPONENTS})
     target_link_libraries(imgui_${component} PUBLIC imgui)
 
     if (component MATCHES "dx11")
-        target_link_libraries(imgui_${component} PUBLIC d3d11)
+        target_link_libraries(imgui_${component} PRIVATE d3d11)
     elseif (component MATCHES "opengl*")
         find_package(OpenGL REQUIRED COMPONENTS OpenGL)
-        target_link_libraries(imgui_${component} PUBLIC OpenGL::GL)
+        target_link_libraries(imgui_${component} PRIVATE OpenGL::GL)
     elseif(component STREQUAL "glfw")
         find_package(glfw REQUIRED)
-        target_link_libraries(imgui_${component} PUBLIC glfw)
+        target_link_libraries(imgui_${component} PRIVATE glfw)
     endif()
 
     add_library(imgui::${component} ALIAS imgui_${component})
Submodule external/immvision/immvision contains modified content
diff --git a/external/immvision/immvision/src/immvision_gl_loader/CMakeLists.txt b/external/immvision/immvision/src/immvision_gl_loader/CMakeLists.txt
index 14df249..b60a364 100644
--- a/external/immvision/immvision/src/immvision_gl_loader/CMakeLists.txt
+++ b/external/immvision/immvision/src/immvision_gl_loader/CMakeLists.txt
@@ -28,7 +28,7 @@ target_include_directories(immvision_gl_loader PUBLIC $<BUILD_INTERFACE:${CMAKE_
 
 if (NOT EMSCRIPTEN)
     find_package(OpenGL REQUIRED)
-    target_link_libraries(immvision_gl_loader PUBLIC OpenGL::GL)
+    target_link_libraries(immvision_gl_loader PRIVATE OpenGL::GL)
 endif()
 
 if (IMMVISION_USE_GLAD)
diff --git a/imgui_bundle_cmake/internal/add_imgui_bundle_bindings.cmake b/imgui_bundle_cmake/internal/add_imgui_bundle_bindings.cmake
index 312e2bf4..24e605cf 100644
--- a/imgui_bundle_cmake/internal/add_imgui_bundle_bindings.cmake
+++ b/imgui_bundle_cmake/internal/add_imgui_bundle_bindings.cmake
@@ -85,7 +85,7 @@ function(add_imgui_bundle_bindings)
     # Link with OpenGL (necessary for nanobind)
     if (NOT  EMSCRIPTEN)
         find_package(OpenGL REQUIRED)
-        target_link_libraries(${python_native_module_name} PUBLIC OpenGL::GL)
+        target_link_libraries(${python_native_module_name} PRIVATE OpenGL::GL)
     endif()
 
     # Disable optimizations on release build for msvc

