From aec96b4661f86a11ed8e01e81578382fcf62cc92 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Wed, 4 Dec 2024 23:15:09 -0500
Subject: [PATCH 6/7] Enable shared libraries for conda-forge

---
 CMakeLists.txt | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bfc37dd8..fe4017d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -253,11 +253,6 @@ if(TARGET imgui_bundle)
     return()
 endif()
 
-# Disable system libraries for python bindings (only if not using vcpkg or conan)
-if (IMGUI_BUNDLE_BUILD_PYTHON)
-    ibd_disable_system_libraries_for_python()
-endif()
-
 
 # Reset installable dependencies at startup
 ibd_reset_installable_dependencies()
@@ -293,7 +288,6 @@ set(CMAKE_CXX_STANDARD 17)
 if (IMGUI_BUNDLE_BUILD_PYTHON)
     add_compile_definitions(IMGUI_BUNDLE_BUILD_PYTHON)
     add_compile_definitions(IMGUI_BUNDLE_PYTHON_API)
-    ibd_disable_system_libraries_for_python()
 endif()
 
 # IMGUI_BUNDLE_WITH_TEST_ENGINE is propagated to HELLOIMGUI_WITH_TEST_ENGINE
-- 
2.45.2

