From 57202bb80db74201bd6ecb9303dad18d6709fd40 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Wed, 4 Dec 2024 23:15:21 -0500
Subject: [PATCH 7/7] Remove optional dependencies to slim package

---
 pyproject.toml | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index e414d039..745074f3 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -29,10 +29,10 @@ classifiers = [
 
 dependencies = [
     "numpy",
-    "munch",
-    "glfw",
-    "PyOpenGL",
-    "pillow",
+    # "munch",
+    # "glfw",
+    # "PyOpenGL",
+    # "pillow",
     "pydantic",
 ]
 
-- 
2.45.2

