diff --git test/Makefile.in test/Makefile.in
index 02ab570..f2b9889 100644
--- test/Makefile.in
+++ test/Makefile.in
@@ -1071,7 +1071,7 @@ check_SCRIPTS = $(TEST_SCRIPT)
 # These tests (fheap, btree2) are under development and are not used by
 # the library yet. Move them to the end so that their failure do not block
 # other current library code tests.
-TEST_PROG = testhdf5 lheap ohdr stab gheap cache cache_api \
+TEST_PROG = testhdf5 lheap ohdr stab gheap cache_api \
            pool accum hyperslab istore bittests dt_arith \
            dtypes dsets cmpd_dset filter_fail extend external efc objcopy links unlink \
            big mtime fillval mount flush1 flush2 app_ref enum \
