commit 49366194cf4c7e4764b97f37f451392a72346466
Author: Tom Schoonjans <Tom.Schoonjans@diamond.ac.uk>
Date:   Fri Sep 20 13:21:39 2019 +0530

    Fix detection of fftw on Windows

diff --git a/cmake/generic/modules/FindFFTW.cmake b/cmake/generic/modules/FindFFTW.cmake
index c80164cb2..a0e868467 100644
--- a/cmake/generic/modules/FindFFTW.cmake
+++ b/cmake/generic/modules/FindFFTW.cmake
@@ -27,7 +27,15 @@ set(CMAKE_FIND_LIBRARY_SUFFIXES ${ba_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES})
 if (WIN32)
     string( REPLACE ".lib" ".dll" FFTW_LIBRARY_DLL "${FFTW_LIBRARY}" )
     if (NOT EXISTS ${FFTW_LIBRARY_DLL})
-        message(FATAL_ERROR "File ${FFTW_LIBRARY_DLL} does not exist")
+        string(FIND "${FFTW_LIBRARY_DLL}" "lib" LIB_POSITION)
+        if ("${LIB_POSITION}" GREATER "-1")
+                string( REPLACE "lib" "bin" FFTW_LIBRARY_DLL "${FFTW_LIBRARY_DLL}" )
+                if (NOT EXISTS ${FFTW_LIBRARY_DLL})
+                        message(FATAL_ERROR "File ${FFTW_LIBRARY_DLL} does not exist")
+                endif (NOT EXISTS ${FFTW_LIBRARY_DLL})
+	else ()
+                message(FATAL_ERROR "File ${FFTW_LIBRARY_DLL} does not exist")
+        endif ("${LIB_POSITION}" GREATER "-1")
     endif(NOT EXISTS ${FFTW_LIBRARY_DLL})
 endif()
 
