commit 6c21469d5602827699d9a16939077fbacdbb3557
Author: Tom Schoonjans <Tom.Schoonjans@diamond.ac.uk>
Date:   Fri Sep 20 09:47:48 2019 +0530

    Use external CMAKE_SHARED_LINKER_FLAGS in Linux builds

diff --git a/cmake/bornagain/modules/SetUpLinux.cmake b/cmake/bornagain/modules/SetUpLinux.cmake
index a92a17cc5..c83db1037 100644
--- a/cmake/bornagain/modules/SetUpLinux.cmake
+++ b/cmake/bornagain/modules/SetUpLinux.cmake
@@ -22,4 +22,4 @@ set(CMAKE_C_FLAGS_DEBUG            "-g")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -I${CMAKE_SOURCE_DIR}/Wrap")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pipe ${BIT_ENVIRONMENT} -Wall -W -Woverloaded-virtual -fPIC")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe ${BIT_ENVIRONMENT} -Wall -W -fPIC")
-set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined")
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
