commit e2d4325899cd035def3ca695b41d579ac46f5fac
Author: Tom Schoonjans <Tom.Schoonjans@diamond.ac.uk>
Date:   Sun Sep 22 17:13:37 2019 +0530

    Windows: do not copy Qt5 dlls into install prefix

diff --git a/GUI/coregui/CMakeLists.txt b/GUI/coregui/CMakeLists.txt
index 62899ff3d..fbd9e52bf 100644
--- a/GUI/coregui/CMakeLists.txt
+++ b/GUI/coregui/CMakeLists.txt
@@ -117,46 +117,3 @@ target_link_libraries(${library_name}
 # --- Installation ---------
 install (TARGETS ${library_name} DESTINATION ${destination_lib} COMPONENT Libraries)
 
-if(WIN32)
-    # --- Qt ---
-    string(REGEX REPLACE "\\\\" "/" QTDIR $ENV{QTDIR} ) # Convert C:\Qt\ to C:/Qt/
-    install(FILES
-    ${QTDIR}/bin/Qt5Widgets.DLL
-    ${QTDIR}/bin/Qt5Core.dll
-    ${QTDIR}/bin/Qt5Gui.dll
-    ${QTDIR}/bin/Qt5Designer.dll
-    ${QTDIR}/bin/Qt5Xml.dll
-    ${QTDIR}/bin/Qt5Svg.dll
-    ${QTDIR}/bin/Qt5Network.dll
-    ${QTDIR}/bin/Qt5PrintSupport.dll
-    DESTINATION ${destination_lib} COMPONENT Libraries)
-
-    install(FILES
-    ${QTDIR}/plugins/platforms/qwindows.dll
-    DESTINATION bin/platforms COMPONENT Libraries)
-
-    install(FILES
-    ${QTDIR}/plugins/iconengines/qsvgicon.dll
-    DESTINATION bin/iconengines COMPONENT Libraries)
-
-    install(FILES
-    ${QTDIR}/plugins/imageformats/qjpeg.dll
-    ${QTDIR}/plugins/imageformats/qsvg.dll
-    DESTINATION bin/imageformats COMPONENT Libraries)
-
-    install(FILES
-    ${QTDIR}/bin/libEGL.dll
-    ${QTDIR}/bin/libGLESv2.dll
-    DESTINATION ${destination_lib} COMPONENT Libraries)
-
-    set(QT_VISTA_STYLE "${QTDIR}/plugins/styles/qwindowsvistastyle.dll")
-    if(EXISTS "${QT_VISTA_STYLE}")
-        install(FILES
-		${QT_VISTA_STYLE}
-                DESTINATION bin/styles COMPONENT Libraries)
-    else(EXISTS "${QT_VISTA_STYLE}")
-        MESSAGE(AUTHOR_WARNING "File ${QT_VISTA_STYLE} not found.")
-    endif(EXISTS "${QT_VISTA_STYLE}")
-    unset(QT_VISTA_STYLE)
-
-endif()
