commit 699e59409cfd5da2e75f544c0d30f38e59953e2d
Author: Tom Schoonjans <Tom.Schoonjans@diamond.ac.uk>
Date:   Sun Sep 22 16:30:59 2019 +0530

    Fix detection of libtiff on Windows

diff --git a/cmake/generic/modules/FindTIFF.cmake b/cmake/generic/modules/FindTIFF.cmake
index ecd840a63..ddff5ef28 100644
--- a/cmake/generic/modules/FindTIFF.cmake
+++ b/cmake/generic/modules/FindTIFF.cmake
@@ -72,7 +72,15 @@ if(TIFF_FOUND)
     else()
         string( REPLACE ".lib" ".dll" TIFF_LIBRARY_DLL "${TIFF_LIBRARY}" )
         if (NOT EXISTS ${TIFF_LIBRARY_DLL})
-            message(FATAL_ERROR "File ${TIFF_LIBRARY_DLL} does not exist")
+            string(FIND "${TIFF_LIBRARY_DLL}" "lib" LIB_POSITION)
+            if ("${LIB_POSITION}" GREATER "-1")
+                string( REPLACE "lib" "bin" TIFF_LIBRARY_DLL "${TIFF_LIBRARY_DLL}" )
+                if (NOT EXISTS ${TIFF_LIBRARY_DLL})
+                    message(FATAL_ERROR "File ${TIFF_LIBRARY_DLL} does not exist")
+                endif (NOT EXISTS ${TIFF_LIBRARY_DLL})
+            else ()
+                message(FATAL_ERROR "File ${TIFF_LIBRARY_DLL} does not exist")
+            endif ("${LIB_POSITION}" GREATER "-1")
         endif (NOT EXISTS ${TIFF_LIBRARY_DLL})
     endif(NOT WIN32)
 endif()
