commit 7d03213ce15c50a5a5ea70ad458dcf3cb9b298f8
Author: Tom Schoonjans <Tom.Schoonjans@diamond.ac.uk>
Date:   Sun Sep 22 15:59:46 2019 +0530

    Remove windows dependency on boost_zlib and boost_bzip2

diff --git a/cmake/bornagain/modules/SearchInstalledSoftware.cmake b/cmake/bornagain/modules/SearchInstalledSoftware.cmake
index 218eadf20..e42252f1d 100644
--- a/cmake/bornagain/modules/SearchInstalledSoftware.cmake
+++ b/cmake/bornagain/modules/SearchInstalledSoftware.cmake
@@ -19,9 +19,6 @@ set(Boost_USE_STATIC_RUNTIME OFF)
 add_definitions(-DBOOST_ALL_DYN_LINK) # line is needed for MSVC
 #add_definitions(-DBOOST_LIB_DIAGNOSTIC) # shows during compilation auto-linked libraries
 set(boost_libraries_required program_options iostreams regex system filesystem)
-if(WIN32)
-    set(boost_libraries_required ${boost_libraries_required} zlib bzip2)
-endif()
 
 find_package(Boost 1.48.0 COMPONENTS ${boost_libraries_required} REQUIRED)
 # In spite of the "REQUIRED" flag, FindBoost will not terminate if some components are missing.
