diff --git a/catkin_tools/common.py b/catkin_tools/common.py
index b04b04e..d018d0e 100644
--- a/catkin_tools/common.py
+++ b/catkin_tools/common.py
@@ -20,7 +20,10 @@ import os
 import re
 import sys
 
-import trollius as asyncio
+try:
+    import asyncio
+except ImportError:
+    import trollius as asyncio
 
 from shlex import split as _cmd_split
 try:
diff --git a/catkin_tools/execution/executor.py b/catkin_tools/execution/executor.py
index db04773..76b5568 100644
--- a/catkin_tools/execution/executor.py
+++ b/catkin_tools/execution/executor.py
@@ -18,7 +18,10 @@ import traceback
 
 from itertools import tee
 
-import trollius as asyncio
+try:
+    import asyncio
+except ImportError:
+    import trollius as asyncio
 
 from concurrent.futures import ThreadPoolExecutor
 from concurrent.futures import FIRST_COMPLETED
diff --git a/catkin_tools/verbs/catkin_build/build.py b/catkin_tools/verbs/catkin_build/build.py
index 5ed56ee..5dca706 100644
--- a/catkin_tools/verbs/catkin_build/build.py
+++ b/catkin_tools/verbs/catkin_build/build.py
@@ -22,7 +22,10 @@ import time
 import traceback
 import yaml
 
-import trollius as asyncio
+try:
+    import asyncio
+except ImportError:
+    import trollius as asyncio
 
 try:
     # Python3
diff --git a/setup.py b/setup.py
index 03b728d..c566471 100644
--- a/setup.py
+++ b/setup.py
@@ -14,9 +14,10 @@ install_requires = [
     'catkin-pkg > 0.2.9',
     'setuptools',
     'PyYAML',
-    'osrf-pycommon > 0.1.1',
-    'trollius'
+    'osrf-pycommon > 0.1.1'
 ]
+if sys.version_info[0] == 3 and sys.version_info[1] <= 4:
+    install_requires.append('trollius')
 if sys.version_info[0] == 2 and sys.version_info[1] <= 6:
     install_requires.append('argparse')
 
