diff --git a/sources/pyside6/cmake/PySideSetup.cmake b/sources/pyside6/cmake/PySideSetup.cmake
index a167f4ddd..37c8d5a17 100644
--- a/sources/pyside6/cmake/PySideSetup.cmake
+++ b/sources/pyside6/cmake/PySideSetup.cmake
@@ -211,7 +211,9 @@ if(NOT TARGET uninstall)
                       -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
 endif()
 
-shiboken_internal_set_python_site_packages()
+if (NOT DEFINED PYTHON_SITE_PACKAGES)
+  shiboken_internal_set_python_site_packages()
+endif()
 
 set(GENERATOR_EXTRA_FLAGS --generator-set=shiboken
                           --enable-parent-ctor-heuristic
