diff --git a/sources/pyside6/CMakeLists.txt b/sources/pyside6/CMakeLists.txt
index c24777f36..926f5aff8 100644
--- a/sources/pyside6/CMakeLists.txt
+++ b/sources/pyside6/CMakeLists.txt
@@ -14,11 +14,11 @@ if(Qt${QT_MAJOR_VERSION}Qml_FOUND)
     add_subdirectory(libpysideqml)
 endif()
 
-find_package(Qt${QT_MAJOR_VERSION}Designer)
-if(Qt${QT_MAJOR_VERSION}UiTools_FOUND AND Qt${QT_MAJOR_VERSION}Designer_FOUND)
+if(Qt${QT_MAJOR_VERSION}UiTools_FOUND)
     add_subdirectory(plugins/uitools)
+    find_package(Qt${QT_MAJOR_VERSION}Designer)
     # PYSIDE-535: Enable when PyPy supports embedding
-    if (NOT PYPY_VERSION)
+    if (Qt${QT_MAJOR_VERSION}Designer_FOUND AND NOT PYPY_VERSION)
         add_subdirectory(plugins/designer)
     endif()
 endif()
