diff --git a/Makefile b/Makefile
index 71dcc5f..f8dcd15 100644
--- a/Makefile
+++ b/Makefile
@@ -6,8 +6,8 @@
 # LICENSE file in the root directory of this source tree.
 #
 
-CXX = c++
-CXXFLAGS = -pthread -std=c++11
+CXX = ${CXX}
+CXXFLAGS = ${CXXFLAGS}
 OBJS = args.o autotune.o matrix.o dictionary.o MurmurHash3.o loss.o productquantizer.o densematrix.o quantmatrix.o vector.o model.o utils.o meter.o fasttext.o
 INCLUDES = -I.
 
