From 0f40913a43cb41533535519c9b4e4fdc62b60914 Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Thu, 18 Sep 2025 15:39:49 +0200
Subject: [PATCH] only requires Qt5 if interactive control cmake option is ON

---
 SoftRobotsConfig.cmake.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SoftRobotsConfig.cmake.in b/SoftRobotsConfig.cmake.in
index 9000ee8..d665d04 100644
--- a/SoftRobotsConfig.cmake.in
+++ b/SoftRobotsConfig.cmake.in
@@ -13,7 +13,10 @@ find_package(Sofa.Component.Controller QUIET REQUIRED)
 find_package(Sofa.Component.SolidMechanics.Spring QUIET REQUIRED)
 find_package(Sofa.Component.Mapping QUIET REQUIRED)
 find_package(Sofa.Component.StateContainer QUIET REQUIRED)
-find_package(Qt5 QUIET REQUIRED COMPONENTS Network)
+
+if(SOFTROBOTS_INTERACTIVECONTROL)
+  find_package(Qt5 QUIET REQUIRED COMPONENTS Network)
+endif()
 
 if(SOFTROBOTS_HAVE_SOFAPYTHON)
     find_package(SofaPython QUIET REQUIRED)
-- 
2.34.1

