From 45748f650dbd8ba05428924a82d72339fb67b562 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 6 Aug 2022 22:07:23 -0600
Subject: [PATCH] Fix Python 3

---
 scanpdf/scanpdf.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scanpdf/scanpdf.py b/scanpdf/scanpdf.py
index ceb2079..43a8b7d 100644
--- a/scanpdf/scanpdf.py
+++ b/scanpdf/scanpdf.py
@@ -68,7 +68,7 @@ def cmd(self, cmd_list):
             logging.debug(out)
             return out
         except subprocess.CalledProcessError as e:
-            print (e.output)
+            print(e.output)
             self._error("Could not run command %s" % cmd_list)
             
 
