From 8c73c5e8b00d4dca3169d12480a5139a53dee64c Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 6 Aug 2022 22:18:23 -0600
Subject: [PATCH] Python 3 support

---
 scanpdf/scanpdf.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scanpdf/scanpdf.py b/scanpdf/scanpdf.py
index b211987..b7de74a 100644
--- a/scanpdf/scanpdf.py
+++ b/scanpdf/scanpdf.py
@@ -67,7 +67,7 @@ def cmd(self, cmd_list):
             logging.debug(out)
             return out
         except subprocess.CalledProcessError as e:
-            print e.output
+            print(e.output)
             self._error("Could not run command %s" % cmd_list)
             
 
