diff --git a/fibsem/imaging/_tile.py b/fibsem/imaging/_tile.py
index 257914d..3612c31 100644
--- a/fibsem/imaging/_tile.py
+++ b/fibsem/imaging/_tile.py
@@ -332,7 +332,7 @@ def _minimap(minimap_image: FibsemImage, positions: list[FibsemStagePosition]):
         import matplotlib.pyplot as plt
         from matplotlib.backends.backend_agg import FigureCanvasAgg
         from PIL import Image
-        from PyQt5.QtGui import QImage, QPixmap
+        from qtpy.QtGui import QImage, QPixmap
         pil_image = None
 
         import matplotlib.pyplot as plt
diff --git a/fibsem/ui/FibsemAlignmentWidget.py b/fibsem/ui/FibsemAlignmentWidget.py
index 3d07e4d..d6b7356 100644
--- a/fibsem/ui/FibsemAlignmentWidget.py
+++ b/fibsem/ui/FibsemAlignmentWidget.py
@@ -3,7 +3,7 @@ import logging
 from copy import deepcopy
 import napari
 import napari.utils.notifications
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 
 import numpy as np
 from fibsem import alignment
diff --git a/fibsem/ui/FibsemCryoSputterWidget.py b/fibsem/ui/FibsemCryoSputterWidget.py
index 9bda194..46daf69 100644
--- a/fibsem/ui/FibsemCryoSputterWidget.py
+++ b/fibsem/ui/FibsemCryoSputterWidget.py
@@ -2,7 +2,7 @@ import logging
 import napari
 import napari.utils.notifications
 import numpy as np
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 from fibsem import config as cfg
 from fibsem import constants, conversions, utils
 from fibsem.microscope import FibsemMicroscope, TescanMicroscope, ThermoMicroscope, DemoMicroscope
diff --git a/fibsem/ui/FibsemDetectionUI.py b/fibsem/ui/FibsemDetectionUI.py
index 07b0ec0..4e6a144 100644
--- a/fibsem/ui/FibsemDetectionUI.py
+++ b/fibsem/ui/FibsemDetectionUI.py
@@ -14,7 +14,7 @@ from fibsem.detection.detection import DetectedFeatures
                                      
 
 from fibsem.ui.qtdesigner_files import detection_dialog as detection_gui
-from PyQt5 import QtCore, QtWidgets
+from qtpy import QtCore, QtWidgets
 
 # TODO: convert to napari
 class FibsemDetectionUI(detection_gui.Ui_Dialog, QtWidgets.QDialog):
diff --git a/fibsem/ui/FibsemDetectionWidget.py b/fibsem/ui/FibsemDetectionWidget.py
index adffb49..bfbe636 100644
--- a/fibsem/ui/FibsemDetectionWidget.py
+++ b/fibsem/ui/FibsemDetectionWidget.py
@@ -6,7 +6,7 @@ import napari
 import napari.utils.notifications
 import numpy as np
 import tifffile as tff
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 
 from fibsem.detection import detection
 from fibsem.detection import utils as det_utils
@@ -20,7 +20,7 @@ from fibsem.structures import (
     Point,
 )
 from fibsem.ui import _stylesheets
-from PyQt5.QtCore import pyqtSignal
+from qtpy.QtCore import pyqtSignal
 from fibsem.ui.qtdesigner_files import FibsemDetectionWidget
 import logging
 
diff --git a/fibsem/ui/FibsemEmbeddedDetectionWidget.py b/fibsem/ui/FibsemEmbeddedDetectionWidget.py
index 05f0240..d6ef714 100644
--- a/fibsem/ui/FibsemEmbeddedDetectionWidget.py
+++ b/fibsem/ui/FibsemEmbeddedDetectionWidget.py
@@ -6,7 +6,7 @@ import napari
 import napari.utils.notifications
 import numpy as np
 import tifffile as tff
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 
 from fibsem.detection import detection
 from fibsem.detection import utils as det_utils
@@ -21,7 +21,7 @@ from fibsem.structures import (
 )
 from fibsem import utils
 from fibsem.ui import _stylesheets
-from PyQt5.QtCore import pyqtSignal
+from qtpy.QtCore import pyqtSignal
 from fibsem.ui.qtdesigner_files import FibsemEmbeddedDetectionWidget
 import logging
 
diff --git a/fibsem/ui/FibsemGISWidget.py b/fibsem/ui/FibsemGISWidget.py
index 87ece7e..053c0e5 100644
--- a/fibsem/ui/FibsemGISWidget.py
+++ b/fibsem/ui/FibsemGISWidget.py
@@ -2,7 +2,7 @@ import logging
 import napari
 import napari.utils.notifications
 import numpy as np
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 from fibsem import config as cfg
 from fibsem import constants, conversions, utils
 from fibsem.microscope import FibsemMicroscope, TescanMicroscope, ThermoMicroscope, DemoMicroscope
diff --git a/fibsem/ui/FibsemImageSettingsWidget.py b/fibsem/ui/FibsemImageSettingsWidget.py
index 893ec63..bede1b1 100644
--- a/fibsem/ui/FibsemImageSettingsWidget.py
+++ b/fibsem/ui/FibsemImageSettingsWidget.py
@@ -25,8 +25,8 @@ from fibsem.ui.qtdesigner_files import ImageSettingsWidget
 from fibsem.ui import _stylesheets
 # from napari.qt.threading import thread_worker
    
-from PyQt5 import QtWidgets
-from PyQt5.QtCore import pyqtSignal
+from qtpy import QtWidgets
+from qtpy.QtCore import pyqtSignal
 import threading
 from queue import Queue
 
diff --git a/fibsem/ui/FibsemImageViewer.py b/fibsem/ui/FibsemImageViewer.py
index b29f537..5c60b81 100644
--- a/fibsem/ui/FibsemImageViewer.py
+++ b/fibsem/ui/FibsemImageViewer.py
@@ -1,5 +1,5 @@
 import napari
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 from datetime import datetime
 from pathlib import Path
 from fibsem import constants
diff --git a/fibsem/ui/FibsemLabellingUI.py b/fibsem/ui/FibsemLabellingUI.py
index 4c395ef..a5e7227 100644
--- a/fibsem/ui/FibsemLabellingUI.py
+++ b/fibsem/ui/FibsemLabellingUI.py
@@ -11,7 +11,7 @@ import tifffile as tff
 import zarr
 from PIL import Image
 from fibsem.ui.qtdesigner_files import FibsemLabellingUI
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 import dask.array as da
 from fibsem.segmentation.config import CLASS_COLORS
 from fibsem.segmentation import utils as seg_utils
diff --git a/fibsem/ui/FibsemManipulatorWidget.py b/fibsem/ui/FibsemManipulatorWidget.py
index 2ebdca9..f2a9b0b 100644
--- a/fibsem/ui/FibsemManipulatorWidget.py
+++ b/fibsem/ui/FibsemManipulatorWidget.py
@@ -4,8 +4,8 @@ import os
 import napari
 import napari.utils.notifications
 import numpy as np
-from PyQt5 import QtWidgets
-from PyQt5.QtWidgets import QMessageBox
+from qtpy import QtWidgets
+from qtpy.QtWidgets import QMessageBox
 
 from fibsem import constants, conversions
 from fibsem.microscope import (DemoMicroscope, FibsemMicroscope,
diff --git a/fibsem/ui/FibsemMillingWidget.py b/fibsem/ui/FibsemMillingWidget.py
index b83e7ab..fec60f5 100644
--- a/fibsem/ui/FibsemMillingWidget.py
+++ b/fibsem/ui/FibsemMillingWidget.py
@@ -6,7 +6,7 @@ import numpy as np
 from copy import deepcopy
 import napari
 import napari.utils.notifications
-from PyQt5 import QtWidgets, QtCore
+from qtpy import QtWidgets, QtCore
 
 from fibsem import config as cfg
 from fibsem import constants, conversions, milling, patterning, utils
diff --git a/fibsem/ui/FibsemMinimapWidget.py b/fibsem/ui/FibsemMinimapWidget.py
index 5931e40..cb92ce5 100644
--- a/fibsem/ui/FibsemMinimapWidget.py
+++ b/fibsem/ui/FibsemMinimapWidget.py
@@ -2,7 +2,7 @@ import logging
 import napari
 import napari.utils.notifications
 import numpy as np
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 from fibsem import config as cfg
 from fibsem import constants, conversions, utils, acquire
 from fibsem.microscope import FibsemMicroscope
@@ -21,7 +21,7 @@ PATH = os.path.join(cfg.DATA_PATH, "tile")
 os.makedirs(PATH, exist_ok=True)
 from fibsem.ui import utils as ui_utils 
 
-from PyQt5.QtCore import pyqtSignal, pyqtSlot
+from qtpy.QtCore import pyqtSignal, pyqtSlot
 
 
 class FibsemMinimapWidget(FibsemMinimapWidget.Ui_MainWindow, QtWidgets.QMainWindow):
diff --git a/fibsem/ui/FibsemModelTrainingWidget.py b/fibsem/ui/FibsemModelTrainingWidget.py
index e334c0a..df78d7b 100644
--- a/fibsem/ui/FibsemModelTrainingWidget.py
+++ b/fibsem/ui/FibsemModelTrainingWidget.py
@@ -7,7 +7,7 @@ import glob
 import napari
 import napari.utils.notifications
 from PIL import Image
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 import dask.array as da
 from fibsem.segmentation import config as cfg
 from fibsem.segmentation import utils as seg_utils
@@ -19,7 +19,7 @@ from fibsem.segmentation.config import CLASS_COLORS
 
 
 # BASE_PATH = os.path.join(os.path.dirname(fibsem.__file__), "config")
-from PyQt5.QtCore import pyqtSignal
+from qtpy.QtCore import pyqtSignal
 
 from napari.qt.threading import thread_worker
 
diff --git a/fibsem/ui/FibsemMovementWidget.py b/fibsem/ui/FibsemMovementWidget.py
index 70f7283..a6aaf26 100644
--- a/fibsem/ui/FibsemMovementWidget.py
+++ b/fibsem/ui/FibsemMovementWidget.py
@@ -12,8 +12,8 @@ from napari.qt.threading import thread_worker
 import numpy as np
 import yaml
 from PIL import Image
-from PyQt5 import QtCore, QtWidgets
-from PyQt5.QtGui import QImage, QPixmap
+from qtpy import QtCore, QtWidgets
+from qtpy.QtGui import QImage, QPixmap
 from fibsem import config as cfg
 from fibsem import constants, conversions
 from fibsem.microscope import FibsemMicroscope
diff --git a/fibsem/ui/FibsemMultiChemWidget.py b/fibsem/ui/FibsemMultiChemWidget.py
index 1444b5c..5a1bfa0 100644
--- a/fibsem/ui/FibsemMultiChemWidget.py
+++ b/fibsem/ui/FibsemMultiChemWidget.py
@@ -7,7 +7,7 @@ from autoscript_sdb_microscope_client import SdbMicroscopeClient
 from fibsem import calibration, constants, utils
 from fibsem.structures import BeamType, GammaSettings, ImageSettings, MultiChemSettings
 from fibsem.ui.qtdesigner_files import FibsemMultiChemWidget
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 
 
 class FibsemMultiChemWidget(FibsemMultiChemWidget.Ui_Form, QtWidgets.QWidget):
diff --git a/fibsem/ui/FibsemPositionsWidget.py b/fibsem/ui/FibsemPositionsWidget.py
index 8c3b598..4aca480 100644
--- a/fibsem/ui/FibsemPositionsWidget.py
+++ b/fibsem/ui/FibsemPositionsWidget.py
@@ -5,8 +5,8 @@ from pathlib import Path
 import napari
 import napari.utils.notifications
 import numpy as np
-from PyQt5 import QtWidgets
-from PyQt5.QtGui import QPixmap, QImage
+from qtpy import QtWidgets
+from qtpy.QtGui import QPixmap, QImage
 from copy import deepcopy
 from fibsem import constants, conversions
 from fibsem.microscope import FibsemMicroscope
diff --git a/fibsem/ui/FibsemSegmentationModelWidget.py b/fibsem/ui/FibsemSegmentationModelWidget.py
index f6a1346..cc0c31b 100644
--- a/fibsem/ui/FibsemSegmentationModelWidget.py
+++ b/fibsem/ui/FibsemSegmentationModelWidget.py
@@ -6,7 +6,7 @@ import napari
 import napari.utils.notifications
 import numpy as np
 import tifffile as tff
-from PyQt5 import QtWidgets
+from qtpy import QtWidgets
 
 from fibsem.detection import detection
 from fibsem.detection import utils as det_utils
@@ -18,7 +18,7 @@ from fibsem.structures import (
     FibsemImage,
     Point,
 )
-from PyQt5.QtCore import pyqtSignal
+from qtpy.QtCore import pyqtSignal
 from fibsem.ui.qtdesigner_files import FibsemSegmentationModelWidget
 from fibsem.segmentation.model import SegmentationModel
 import logging
diff --git a/fibsem/ui/FibsemSystemSetupWidget.py b/fibsem/ui/FibsemSystemSetupWidget.py
index 461877c..a48c4de 100644
--- a/fibsem/ui/FibsemSystemSetupWidget.py
+++ b/fibsem/ui/FibsemSystemSetupWidget.py
@@ -4,8 +4,8 @@ import yaml
 import os 
 import napari
 import napari.utils.notifications
-from PyQt5 import QtWidgets
-from PyQt5.QtCore import pyqtSignal
+from qtpy import QtWidgets
+from qtpy.QtCore import pyqtSignal
 
 import fibsem
 from fibsem import config as cfg
@@ -212,7 +212,7 @@ class FibsemSystemSetupWidget(FibsemSystemSetupWidget.Ui_Form, QtWidgets.QWidget
         system_dict["system"]["name"] = self.microscope.model
         system_dict["system"]["manufacturer"] = self.comboBox_manufacturer.currentText()
         system_dict["system"]["version"] = fibsem.__version__
-        from PyQt5.QtWidgets import QInputDialog
+        from qtpy.QtWidgets import QInputDialog
         system_dict["system"]["id"] = QInputDialog.getText(self, "Microscope ID", "Please enter ID of microscope")[0]
         system_dict["system"]["description"] = QInputDialog.getText(self, "Description", "Please enter system description")[0]
     
diff --git a/fibsem/ui/FibsemUI.py b/fibsem/ui/FibsemUI.py
index 82d62cc..4d717a0 100644
--- a/fibsem/ui/FibsemUI.py
+++ b/fibsem/ui/FibsemUI.py
@@ -19,9 +19,9 @@ from fibsem.ui.FibsemSystemSetupWidget import FibsemSystemSetupWidget
 from fibsem.ui.qtdesigner_files import FibsemUI
 from fibsem.ui.utils import message_box_ui
 
-from PyQt5 import QtWidgets
-from PyQt5.QtWidgets import QMessageBox
-from PyQt5.QtCore import pyqtSignal
+from qtpy import QtWidgets
+from qtpy.QtWidgets import QMessageBox
+from qtpy.QtCore import pyqtSignal
 
 
 class FibsemUI(FibsemUI.Ui_MainWindow, QtWidgets.QMainWindow):
diff --git a/fibsem/ui/qtdesigner_files/CurrentAlignmentWidget.py b/fibsem/ui/qtdesigner_files/CurrentAlignmentWidget.py
index 89e5a69..622241a 100644
--- a/fibsem/ui/qtdesigner_files/CurrentAlignmentWidget.py
+++ b/fibsem/ui/qtdesigner_files/CurrentAlignmentWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'CurrentAlignmentWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_BeamAlignment(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemCryoSputterWidget.py b/fibsem/ui/qtdesigner_files/FibsemCryoSputterWidget.py
index 4bcd816..43017d9 100644
--- a/fibsem/ui/qtdesigner_files/FibsemCryoSputterWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemCryoSputterWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemCryoSputterWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Dialog(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemDetectionWidget.py b/fibsem/ui/qtdesigner_files/FibsemDetectionWidget.py
index e4a32c2..e4c9fc1 100644
--- a/fibsem/ui/qtdesigner_files/FibsemDetectionWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemDetectionWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemDetectionWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemEmbeddedDetectionWidget.py b/fibsem/ui/qtdesigner_files/FibsemEmbeddedDetectionWidget.py
index 9957ffd..6c20fc1 100644
--- a/fibsem/ui/qtdesigner_files/FibsemEmbeddedDetectionWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemEmbeddedDetectionWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemEmbeddedDetectionWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemGISWidget.py b/fibsem/ui/qtdesigner_files/FibsemGISWidget.py
index cb7b1a9..617641e 100644
--- a/fibsem/ui/qtdesigner_files/FibsemGISWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemGISWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemGISWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemLabellingUI.py b/fibsem/ui/qtdesigner_files/FibsemLabellingUI.py
index f407a7c..feaba0f 100644
--- a/fibsem/ui/qtdesigner_files/FibsemLabellingUI.py
+++ b/fibsem/ui/qtdesigner_files/FibsemLabellingUI.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemLabellingUI.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Dialog(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemManipulatorWidget.py b/fibsem/ui/qtdesigner_files/FibsemManipulatorWidget.py
index e0ee0ef..cd6506b 100644
--- a/fibsem/ui/qtdesigner_files/FibsemManipulatorWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemManipulatorWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemManipulatorWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemMillingWidget.py b/fibsem/ui/qtdesigner_files/FibsemMillingWidget.py
index 77134b7..fcb73d2 100644
--- a/fibsem/ui/qtdesigner_files/FibsemMillingWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemMillingWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemMillingWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemMinimapWidget.py b/fibsem/ui/qtdesigner_files/FibsemMinimapWidget.py
index 01df52d..921bb11 100644
--- a/fibsem/ui/qtdesigner_files/FibsemMinimapWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemMinimapWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemMinimapWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_MainWindow(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemModelTrainingWidget.py b/fibsem/ui/qtdesigner_files/FibsemModelTrainingWidget.py
index 8d65498..10670d6 100644
--- a/fibsem/ui/qtdesigner_files/FibsemModelTrainingWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemModelTrainingWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemModelTrainingWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemMovementWidget.py b/fibsem/ui/qtdesigner_files/FibsemMovementWidget.py
index 38db158..0cab2db 100644
--- a/fibsem/ui/qtdesigner_files/FibsemMovementWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemMovementWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemMovementWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemMultiChemWidget.py b/fibsem/ui/qtdesigner_files/FibsemMultiChemWidget.py
index e3bc593..27dd32a 100644
--- a/fibsem/ui/qtdesigner_files/FibsemMultiChemWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemMultiChemWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemMultiChemWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.7
+# Created by: qtpy UI code generator 5.15.7
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemPositionsWidget.py b/fibsem/ui/qtdesigner_files/FibsemPositionsWidget.py
index 2ed39b6..90d961b 100644
--- a/fibsem/ui/qtdesigner_files/FibsemPositionsWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemPositionsWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemPositionsWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemSegmentationModelWidget.py b/fibsem/ui/qtdesigner_files/FibsemSegmentationModelWidget.py
index 529c8ab..ce163b1 100644
--- a/fibsem/ui/qtdesigner_files/FibsemSegmentationModelWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemSegmentationModelWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemSegmentationModelWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.7
+# Created by: qtpy UI code generator 5.15.7
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemSettingUI.py b/fibsem/ui/qtdesigner_files/FibsemSettingUI.py
index 85fa27c..99cc099 100644
--- a/fibsem/ui/qtdesigner_files/FibsemSettingUI.py
+++ b/fibsem/ui/qtdesigner_files/FibsemSettingUI.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemSettingsUI.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Dialog(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemSettings.py b/fibsem/ui/qtdesigner_files/FibsemSettings.py
index f022669..172974a 100644
--- a/fibsem/ui/qtdesigner_files/FibsemSettings.py
+++ b/fibsem/ui/qtdesigner_files/FibsemSettings.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemSettingsUI.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.7
+# Created by: qtpy UI code generator 5.15.7
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Dialog(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemSystemSetupWidget.py b/fibsem/ui/qtdesigner_files/FibsemSystemSetupWidget.py
index 814b1b9..7687583 100644
--- a/fibsem/ui/qtdesigner_files/FibsemSystemSetupWidget.py
+++ b/fibsem/ui/qtdesigner_files/FibsemSystemSetupWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemSystemSetupWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/FibsemUI.py b/fibsem/ui/qtdesigner_files/FibsemUI.py
index c788ee1..60b764c 100644
--- a/fibsem/ui/qtdesigner_files/FibsemUI.py
+++ b/fibsem/ui/qtdesigner_files/FibsemUI.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'FibsemUI.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_MainWindow(object):
diff --git a/fibsem/ui/qtdesigner_files/ImageSettingsWidget.py b/fibsem/ui/qtdesigner_files/ImageSettingsWidget.py
index 034fc1a..597d280 100644
--- a/fibsem/ui/qtdesigner_files/ImageSettingsWidget.py
+++ b/fibsem/ui/qtdesigner_files/ImageSettingsWidget.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'ImageSettingsWidget.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/detection_dialog.py b/fibsem/ui/qtdesigner_files/detection_dialog.py
index a867e74..b80d3dc 100644
--- a/fibsem/ui/qtdesigner_files/detection_dialog.py
+++ b/fibsem/ui/qtdesigner_files/detection_dialog.py
@@ -2,11 +2,11 @@
 
 # Form implementation generated from reading ui file 'detection_dialog.ui'
 #
-# Created by: PyQt5 UI code generator 5.9.2
+# Created by: qtpy UI code generator 5.9.2
 #
 # WARNING! All changes made in this file will be lost!
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 class Ui_Dialog(object):
     def setupUi(self, Dialog):
diff --git a/fibsem/ui/qtdesigner_files/image_viewer.py b/fibsem/ui/qtdesigner_files/image_viewer.py
index adcaa75..1d49bf2 100644
--- a/fibsem/ui/qtdesigner_files/image_viewer.py
+++ b/fibsem/ui/qtdesigner_files/image_viewer.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'image_viewer.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.9
+# Created by: qtpy UI code generator 5.15.9
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_MainWindow(object):
diff --git a/fibsem/ui/qtdesigner_files/needle_popup.py b/fibsem/ui/qtdesigner_files/needle_popup.py
index f338e3a..59be27c 100644
--- a/fibsem/ui/qtdesigner_files/needle_popup.py
+++ b/fibsem/ui/qtdesigner_files/needle_popup.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'needle_popup.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.7
+# Created by: qtpy UI code generator 5.15.7
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Form(object):
diff --git a/fibsem/ui/qtdesigner_files/user_dialog.py b/fibsem/ui/qtdesigner_files/user_dialog.py
index 3a6860c..0ba9660 100644
--- a/fibsem/ui/qtdesigner_files/user_dialog.py
+++ b/fibsem/ui/qtdesigner_files/user_dialog.py
@@ -2,13 +2,13 @@
 
 # Form implementation generated from reading ui file 'user_dialog.ui'
 #
-# Created by: PyQt5 UI code generator 5.15.7
+# Created by: qtpy UI code generator 5.15.7
 #
 # WARNING: Any manual changes made to this file will be lost when pyuic5 is
 # run again.  Do not edit this file unless you know what you are doing.
 
 
-from PyQt5 import QtCore, QtGui, QtWidgets
+from qtpy import QtCore, QtGui, QtWidgets
 
 
 class Ui_Dialog(object):
diff --git a/fibsem/ui/utils.py b/fibsem/ui/utils.py
index c4096e4..3778790 100644
--- a/fibsem/ui/utils.py
+++ b/fibsem/ui/utils.py
@@ -12,7 +12,7 @@ from fibsem.structures import Point, FibsemImage, FibsemPatternSettings, FibsemP
 from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
 from matplotlib.figure import Figure
 from matplotlib.patches import Rectangle
-from PyQt5.QtWidgets import QMessageBox, QSizePolicy, QVBoxLayout, QWidget
+from qtpy.QtWidgets import QMessageBox, QSizePolicy, QVBoxLayout, QWidget
 from fibsem.patterning import FibsemMillingStage
 import napari
 from fibsem.utils import load_yaml, save_yaml
@@ -140,10 +140,10 @@ import fibsem.patterning as patterning
 #     return response
 
 
-from PyQt5 import QtWidgets
-from PyQt5.QtCore import Qt
-from PyQt5.QtGui import QImage, QPixmap
-from PyQt5.QtWidgets import QGridLayout, QLabel
+from qtpy import QtWidgets
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QImage, QPixmap
+from qtpy.QtWidgets import QGridLayout, QLabel
 
 COLOURS = ["yellow", "cyan", "magenta", "lime", "orange", "hotpink", "green", "blue", "red", "purple"]
 
@@ -494,7 +494,7 @@ def message_box_ui(title: str, text: str, buttons=QMessageBox.Yes | QMessageBox.
 
     return response
 
-from PyQt5 import QtGui
+from qtpy import QtGui
 def _display_logo(path, label, shape=[50, 50]):
     label.setScaledContents(True)
     label.setFixedSize(*shape)
diff --git a/fibsem/ui/windows.py b/fibsem/ui/windows.py
index a58e378..8f2212a 100644
--- a/fibsem/ui/windows.py
+++ b/fibsem/ui/windows.py
@@ -4,7 +4,7 @@ from pathlib import Path
 
 import napari
 import numpy as np
-from PyQt5.QtWidgets import QMessageBox
+from qtpy.QtWidgets import QMessageBox
 
 from fibsem import acquire, conversions, validation
 from fibsem.acquire import take_reference_images
