diff --git a/pyproject.toml b/pyproject.toml
index 929880a..e594bbc 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,6 +23,8 @@ include-package-data = true
 include = ["fibsem*"]
 exclude = ["fibsem/**/*.ui", "fibsem/segmentation/docs.*", "fibsem/segmentation/docs", "fibsem/segmentation/docs*"]  # exclude packages matching these glob patterns (empty by default)
 
+[tool.setuptools.package-data]
+mypkg = ["*.yaml"]
 
 [tool.pytest.ini_options]
 minversion = "6.0"
