diff --git a/CMakeLists.txt b/CMakeLists.txt
index f2e9f1a8..8a8d3e48 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -245,7 +245,7 @@ if(NRGLJUBLJANA_CONDA_BUILD)
       add_library(gtest_main ALIAS GTest::gtest_main)
     endif()
   endif()
-  find_package(Eigen3 3.4 CONFIG REQUIRED)
+  find_package(Eigen3 5.0 CONFIG REQUIRED)
   set(HIGHFIVE_FIND_HDF5 Off)
   find_package(HighFive CONFIG REQUIRED)
   find_package(fmt 11 CONFIG REQUIRED)
