diff --git a/CMakeLists.txt b/CMakeLists.txt
index b54a9d4..9ed2609 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,7 +51,19 @@ else()
 endif()
 
 install(TARGETS ccache-storage-http RUNTIME DESTINATION bin)
-install(CODE "execute_process(COMMAND \${CMAKE_COMMAND} -E create_symlink ccache-storage-http \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin/ccache-storage-https)")
+if(WIN32)
+  install(
+    CODE
+      "execute_process(COMMAND \${CMAKE_COMMAND} -E copy_if_different \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin/ccache-storage-http.exe \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin/ccache-storage-https.exe)"
+  )
+else()
+  install(
+    CODE
+      "execute_process(COMMAND \${CMAKE_COMMAND} -E create_symlink ccache-storage-http \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin/ccache-storage-https)"
+  )
+endif()
 
 find_program(CLANG_FORMAT clang-format)
 if(CLANG_FORMAT)