diff --git a/setup.py b/setup.py
index 8e1c220..5c67ad4 100644
--- a/setup.py
+++ b/setup.py
@@ -51,10 +51,7 @@ setup(
     author_email='Niel Hu <hu.niel92@gmail.com>',
     url='https://github.com/uber/h3-py.git',
     packages=find_packages(exclude=['tests', 'tests.*']),
-    install_requires=[],
-    cmdclass={
-        'build_ext': CustomBuildExtCommand,
-    },
+    install_requires=[h3],
     package_data={
         'h-py':
         ['out/*.dylib' if platform.system() == 'Darwin' else (
