diff --git a/vispy/ext/fontconfig.py b/vispy/ext/fontconfig.py
index ff24662..29cdea5 100644
--- a/vispy/ext/fontconfig.py
+++ b/vispy/ext/fontconfig.py
@@ -1,5 +1,7 @@
 # -*- coding: utf-8 -*-
 
+import os
+import sys
 import warnings
 from ctypes import (util, cdll, c_void_p, c_char_p, c_double, c_int, c_bool,
                     Union, Structure, byref, POINTER)
@@ -7,7 +9,13 @@ from ..util.wrappers import run_subprocess
 
 # Some code adapted from Pyglet
 
-fc = util.find_library('fontconfig')
+from sys import platform as _platform
+
+if _platform == "linux" or _platform == "linux2":
+   fc = os.path.join(sys.prefix, 'lib', 'libfontconfig.so')
+elif _platform == "darwin" or _platform == "win32":
+   fc = util.find_library('fontconfig')
+
 if fc is None:
     raise ImportError('fontconfig not found')
 fontconfig = cdll.LoadLibrary(fc)
