diff --git a/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/TypeOrmAndNodeFsStorageFilesRepository.ts b/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/TypeOrmAndNodeFsStorageFilesRepository.ts
index 0000000..1111111 100644
--- a/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/TypeOrmAndNodeFsStorageFilesRepository.ts
+++ b/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/TypeOrmAndNodeFsStorageFilesRepository.ts
@@ -2,7 +2,7 @@ import { Service } from 'diod';
 import { Readable } from 'form-data';
 import { readFile, unlink } from 'fs/promises';
 import path from 'path';
-import { DataSource, Repository } from '../../../../../../../apps/node_modules/typeorm';
+import { DataSource, Repository } from 'typeorm';
 import { ensureFolderExists } from '../../../../../../../apps/shared/fs/ensure-folder-exists';
 import { WriteReadableToFile } from '../../../../../../../apps/shared/fs/write-readable-to-file';
 import { StorageFile } from '../../../../domain/StorageFile';
diff --git a/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/TypeOrmStorageFilesDataSourceFactory.ts b/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/TypeOrmStorageFilesDataSourceFactory.ts
index 0000000..1111111 100644
--- a/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/TypeOrmStorageFilesDataSourceFactory.ts
+++ b/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/TypeOrmStorageFilesDataSourceFactory.ts
@@ -1,4 +1,4 @@
-import { DataSource } from '../../../../../../../apps/node_modules/typeorm';
+import { DataSource } from 'typeorm';
 import { TypeOrmStorageFile } from './entities/TypeOrmStorageFile';
 import { app } from 'electron';
 
diff --git a/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/entities/TypeOrmStorageFile.ts b/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/entities/TypeOrmStorageFile.ts
index 0000000..1111111 100644
--- a/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/entities/TypeOrmStorageFile.ts
+++ b/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/entities/TypeOrmStorageFile.ts
@@ -1,4 +1,4 @@
-import { Column, Entity, PrimaryColumn } from '../../../../../../../../apps/node_modules/typeorm';
+import { Column, Entity, PrimaryColumn } from 'typeorm';
 import { StorageFileAttributes } from '../../../../../domain/StorageFile';
 
 @Entity('storage_file')
diff --git a/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/__test-helpers__/sqlDataSource.ts b/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/__test-helpers__/sqlDataSource.ts
index 0000000..1111111 100644
--- a/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/__test-helpers__/sqlDataSource.ts
+++ b/src/context/storage/StorageFiles/infrastructure/persistance/repository/typeorm/__test-helpers__/sqlDataSource.ts
@@ -1,5 +1,5 @@
 import path from 'node:path';
-import { DataSource } from '../../../../../../../../apps/node_modules/typeorm';
+import { DataSource } from 'typeorm';
 import { TypeOrmStorageFile } from '../entities/TypeOrmStorageFile';
 
 export function obtainSqliteDataSource(directory: string): Promise<DataSource> {
