From f91df7d1c9954a5b002e38abf7275e84640d147f Mon Sep 17 00:00:00 2001
From: Setu Shah <setu4993@yahoo.co.in>
Date: Sun, 15 Mar 2020 15:37:41 -0700
Subject: [PATCH] Fix for parse_requirements

---
 setup.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 8ead33e..44ff31d 100644
--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,10 @@
 from operator import attrgetter
 from os import path
 
-from pip.req import parse_requirements
+try:  # for pip >= 10
+    from pip._internal.req import parse_requirements
+except ImportError:  # for pip <= 9.0.3
+    from pip.req import parse_requirements
 from setuptools import setup
 
 def read(fname):
-- 
2.23.0

