diff --git a/test/corrosion_install/install_lib/CMakeLists.txt b/test/corrosion_install/install_lib/CMakeLists.txt
index 4587ec8..dd764ce 100644
--- a/test/corrosion_install/install_lib/CMakeLists.txt
+++ b/test/corrosion_install/install_lib/CMakeLists.txt
@@ -2,13 +2,14 @@ cmake_minimum_required(VERSION 3.15)
 project(test_project VERSION 0.1.0)
 include(ExternalProject)
 
+find_package(Threads REQUIRED)
+
 add_library(static_lib STATIC IMPORTED)
 add_library(shared_lib SHARED IMPORTED)
 set(install_prefix "${CMAKE_CURRENT_BINARY_DIR}/rust_lib")
 set(static_lib_install_path "${install_prefix}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}rust_lib${CMAKE_STATIC_LIBRARY_SUFFIX}")
 set(shared_lib_install_path "${install_prefix}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}rust_lib${CMAKE_SHARED_LIBRARY_SUFFIX}")
 
-
 set_target_properties(static_lib PROPERTIES
                       IMPORTED_LOCATION
                       "${static_lib_install_path}")
@@ -18,7 +19,7 @@ set_target_properties(shared_lib PROPERTIES
                       "${shared_lib_install_path}")
 
 add_executable(main-static main.cpp)
-target_link_libraries(main-static PRIVATE static_lib)
+target_link_libraries(main-static PRIVATE static_lib Threads::Threads)
 
 ExternalProject_Add(
         rust_lib
diff --git a/test/custom_target/custom_target/CMakeLists.txt b/test/custom_target/custom_target/CMakeLists.txt
index d6396bc..7445a55 100644
--- a/test/custom_target/custom_target/CMakeLists.txt
+++ b/test/custom_target/custom_target/CMakeLists.txt
@@ -11,6 +11,8 @@ if(NOT "${Rust_CARGO_TARGET}" MATCHES ".json")
     )
 endif()
 
+find_package(Threads REQUIRED)
+
 corrosion_import_crate(MANIFEST_PATH rust/Cargo.toml)
 corrosion_set_cargo_flags(rust_lib "-Zbuild-std")
 # custom target triples require nightly features
@@ -20,4 +22,4 @@ corrosion_set_cargo_flags(rust-bin "-Zbuild-std")
 corrosion_set_env_vars(rust-bin RUSTC_BOOTSTRAP=1)
 
 add_executable(test-exe main.cpp)
-target_link_libraries(test-exe PUBLIC rust_lib)
+target_link_libraries(test-exe PUBLIC rust_lib Threads::Threads)
