From 4b569a7dcf6cf1dba40cdba39380b9a93d8f7f2d Mon Sep 17 00:00:00 2001
From: Anand Balakrishnan <anandbala1597@gmail.com>
Date: Tue, 15 Oct 2024 18:01:43 -0700
Subject: [PATCH 1/2] don't hardcode gcc as the compiler

---
 download.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/download.py b/download.py
index 88f0987..d71e78c 100644
--- a/download.py
+++ b/download.py
@@ -97,7 +97,7 @@ def extensions(
         cudd_cflags.append('-DCYTHON_TRACE=1')
         sylvan_cflags.append('-DCYTHON_TRACE=1')
         # directives['binding'] = True
-    os.environ['CC'] = CC
+    # os.environ['CC'] = CC
     path = args.cudd if args.cudd else CUDD_PATH
     cudd_include = [(path, s) for s in CUDD_INCLUDE]
     cudd_link = [(path, s) for s in CUDD_LINK]
-- 
2.46.2

