diff --git a/setup.py b/setup.py
index e8bd98b..0fad0e5 100644
--- a/setup.py
+++ b/setup.py
@@ -51,7 +51,11 @@ setup(
     packages=find_packages(),
     include_package_data=True,
     zip_safe=False,
-    scripts=['superset/bin/superset'],
+    entry_points={
+        'console_scripts': [
+            "superset = superset.bin.superset:cli"
+        ]
+    },
     install_requires=[
         'bleach',
         'boto3==1.4.7',
