diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index ed9c3b9..a4356a0 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -13,6 +13,7 @@ FetchContent_Declare(
 
 # For Windows, Prevent overriding the parent project's compiler/linker settings
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+set(INSTALL_GTEST OFF CACHE BOOL "" FORCE)
 FetchContent_MakeAvailable(GoogleTest)
 include(GoogleTest)
 
