diff --git a/src/conmon.c b/src/conmon.c
index e43dc68..d3c42b6 100644
--- a/src/conmon.c
+++ b/src/conmon.c
@@ -22,6 +22,11 @@
 #include <sys/prctl.h>
 #include <sys/stat.h>
 
+/* NB: PR_{SET,GET}_CHILD_SUBREAPER were introduced in Linux 3.4 and are not available on CentOS 6. */
+#ifndef PR_SET_CHILD_SUBREAPER
+#define PR_SET_CHILD_SUBREAPER 36
+#endif
+
 int main(int argc, char *argv[])
 {
        _cleanup_gerror_ GError *err = NULL;
