From d754f5df0a50ba72835c0b4555e443b77240b2b8 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Tue, 14 Dec 2021 16:55:38 -0500
Subject: [PATCH] Allow windows to resize to be smaller than the default size

---
 deeplabcut/gui/widgets.py | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/deeplabcut/gui/widgets.py b/deeplabcut/gui/widgets.py
index 4d3d0cf89..d34cc53e9 100644
--- a/deeplabcut/gui/widgets.py
+++ b/deeplabcut/gui/widgets.py
@@ -51,7 +51,7 @@ class BaseFrame(wx.Frame):
     """Contains the main GUI and button boxes"""
 
     def __init__(self, frame_title="", parent=None, imtypes=None):
-        # Settting the GUI size and panels design
+        # Setting the GUI size and panels design
         displays = (
             wx.Display(i) for i in range(wx.Display.GetCount())
         )  # Gets the number of displays
@@ -77,9 +77,8 @@ def __init__(self, frame_title="", parent=None, imtypes=None):
         self.statusbar = self.CreateStatusBar()
         self.statusbar.SetStatusText("")
 
-        self.SetSizeHints(
-            wx.Size(self.gui_size)
-        )  #  This sets the minimum size of the GUI. It can scale now!
+        # This sets the minimum size of the GUI. It can scale now!
+        self.SetSizeHints(wx.Size(800, 600))
 
     @staticmethod
     def calc_distance(x1, y1, x2, y2):
