--- setup.py	2019-05-04 14:11:20.409186851 -0400
+++ setup.py	2019-05-04 21:53:34.400244170 -0400
@@ -81,7 +81,7 @@

     extras = {
         'excel': ['openpyxl', 'networkx'],
-        'plot': ['graphviz', 'regex', 'flask', 'Pygments', 'lxml', 'bs4',
+        'plot': ['graphviz', 'regex', 'flask', 'Pygments', 'lxml', 'beautifulsoup4',
                  'jinja2', 'docutils']  # ['schedula[plot]>=0.2.0']
     }
     extras['all'] = sorted(functools.reduce(set.union, extras.values(), set()))
