diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d5fa32..e76a71b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,3 +21,5 @@ add_library(libgama OBJECT ${SRC_GAMA} ${SRC_EXPAT})
 
 add_executable(gama-local bin/gama-local.cpp $<TARGET_OBJECTS:libgama>)
 add_executable(gama-g3 bin/gama-g3.cpp $<TARGET_OBJECTS:libgama>)
+
+install(TARGETS gama-local gama-g3 DESTINATION bin)
\ No newline at end of file
