diff --git a/setup.py b/setup.py
index 02db597e..6418cea3 100644
--- a/setup.py
+++ b/setup.py
@@ -22,11 +22,6 @@ import _updateversion as up
 from distutils.command.clean import clean as Clean
 
 
-# == Checking platform ========================================================
-is_platform_windows = False
-if platform.system() == "Windows":
-    is_platform_windows = True
-
 
 # === Setting clean command ===================================================
 logging.basicConfig(level=logging.INFO)
@@ -99,47 +94,6 @@ class CleanCommand(Clean):
 # =============================================================================
 
 
-# =============================================================================
-# Check if openmp available
-# see http://openmp.org/wp/openmp-compilers/
-omp_test = r"""
-#include <omp.h>
-#include <stdio.h>
-int main() {
-#pragma omp parallel
-printf("Hello from thread %d, nthreads %d\n", omp_get_thread_num(),
-       omp_get_num_threads());
-}
-"""
-
-
-def check_for_openmp(cc_var):
-    import tempfile
-
-    tmpdir = tempfile.mkdtemp()
-    curdir = os.getcwd()
-    os.chdir(tmpdir)
-
-    filename = r"test.c"
-    with open(filename, "w") as file:
-        file.write(omp_test)
-    with open(os.devnull, "w") as fnull:
-        result = subprocess.call(
-            [cc_var, "-fopenmp", filename], stdout=fnull, stderr=fnull
-        )
-
-    os.chdir(curdir)
-    # clean up
-    shutil.rmtree(tmpdir)
-    return result
-
-
-# ....... Using function
-if is_platform_windows:
-    openmp_installed = False
-else:
-    openmp_installed = not check_for_openmp("cc")
-# =============================================================================
 
 
 # == Getting tofu version =====================================================
@@ -212,7 +166,7 @@ else:
 
 # =============================================================================
 #  Compiling files
-if openmp_installed:
+if platform.system() == 'Linux':
     extra_compile_args = ["-O3", "-Wall", "-fopenmp", "-fno-wrapv"]
     extra_link_args = ["-fopenmp"]
 else:
