diff --git a/MANIFEST.in b/MANIFEST.in
index abda6c90..263ea18b 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -3,4 +3,6 @@
 include MANIFEST.in
 include pyproject.toml
 include _updateversion.py
-recursive-include tofu/geom *.pxd
\ No newline at end of file
+recursive-include tofu/geom *.pxd
+recursive-include tofu/geom/inputs *.txt
+include LICENSE.txt
\ No newline at end of file
diff --git a/tofu/version.py b/tofu/version.py
index 3ee8339e..d428c894 100644
--- a/tofu/version.py
+++ b/tofu/version.py
@@ -1,2 +1,2 @@
 # Do not edit, pipeline versioning governed by git tags!
-__version__ = '1.4.3b4-34-g4c767105'
+__version__ = '1.4.3b4-28-g91c648a7'
