From 2412d98f139ed329a3f1dfe118386a96bfef5c79 Mon Sep 17 00:00:00 2001
From: Tim Snyder <timothy.snyder@sifive.com>
Date: Wed, 29 Jun 2022 14:50:29 +0000
Subject: [PATCH 3/3] rm buildJobCache for sifive/wake#925

---
 build.wake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.wake b/build.wake
index 31087b11..9bd805ca 100644
--- a/build.wake
+++ b/build.wake
@@ -38,7 +38,7 @@ export def install: List String => Result String Error = match _
 
 # Build all wake targets
 def targets =
-    buildWake, buildWakeBox, buildFuseDaemon, buildShim, buildBSP, buildLSP, buildJobCache, Nil
+    buildWake, buildWakeBox, buildFuseDaemon, buildShim, buildBSP, buildLSP, Nil
 
 def all variant =
     require Pass x =
-- 
2.35.3

