--- a/tests.py
+++ b/tests.py
@@ -404,6 +404,19 @@
         fi;
         """
         testfiles = " ".join([pipes.quote(p) for p in pythonTestFiles])
+
+        ####################################
+        # debug stuff added by stackvana
+        print(' ', flush=True)
+        print(' ', flush=True)
+        print('environment:', os.environ, flush=True)
+        print('target:', target, flush=True)
+        print('testing command:', cmd.format(interpreter, testfiles, libpathstr), flush=True)
+        print(' ', flush=True)
+        print(' ', flush=True)
+        # end of stuff added by stackvana
+        ####################################
+
         result = self._env.Command(target, None, cmd.format(interpreter, testfiles, libpathstr))

         self._env.Alias(os.path.basename(target), target)
