diff --git a/CMakeLists.txt b/CMakeLists.txt
index 996beef..b043ab9 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -228,3 +228,6 @@ if((COMPILER_SUPPORTS_CXX0X OR COMPILER_SUPPORTS_CXX11) AND HDF5_FOUND)
         target_include_directories(solve-lmp-grid-graph PRIVATE ${HDF5_INCLUDE_DIRS})
     endif()
 endif()
+
+# Install the headers.
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/andres" DESTINATION include)
