From c7b452f67f04ad70b77062b6cd5199e03a02b3ad Mon Sep 17 00:00:00 2001
From: Tim Paine <t.paine154@gmail.com>
Date: Sat, 18 Mar 2023 21:52:15 -0400
Subject: [PATCH] apply patch

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index fe8ba6bc..bdeeaf25 100755
--- a/setup.py
+++ b/setup.py
@@ -6,6 +6,7 @@ from setuptools import find_packages
 
 setup(
     name="litex",
+    version="2022.12",
     description="Python SoC/Core builder for building FPGA based systems.",
     author="Florent Kermarrec",
     author_email="florent@enjoy-digital.fr",
-- 
2.40.0

