diff --git a/src/arkode/CMakeLists.txt b/src/arkode/CMakeLists.txt
index cfed8e6..523a967 100644
--- a/src/arkode/CMakeLists.txt
+++ b/src/arkode/CMakeLists.txt
@@ -138,6 +138,19 @@ IF(BUILD_SHARED_LIBS)
   ADD_LIBRARY(sundials_arkode_shared SHARED 
     ${arkode_SOURCES}  ${arkode_BL_SOURCES}  ${shared_SOURCES})
 
+
+  IF(UNIX)
+    TARGET_LINK_LIBRARIES(sundials_arkode_shared m)
+  ENDIF()
+
+  IF(LAPACK_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_arkode_shared ${LAPACK_LIBRARIES})
+  ENDIF()
+
+  IF(KLU_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_arkode_shared ${KLU_LIBRARIES})
+  ENDIF ()
+
   # Set the library name and make sure it is not deleted
   SET_TARGET_PROPERTIES(sundials_arkode_shared
     PROPERTIES OUTPUT_NAME sundials_arkode CLEAN_DIRECT_OUTPUT 1)
diff --git a/src/cvode/CMakeLists.txt b/src/cvode/CMakeLists.txt
index d47cccc..0265046 100644
--- a/src/cvode/CMakeLists.txt
+++ b/src/cvode/CMakeLists.txt
@@ -145,6 +145,18 @@ IF(BUILD_SHARED_LIBS)
   ADD_LIBRARY(sundials_cvode_shared SHARED 
     ${cvode_SOURCES}  ${cvode_BL_SOURCES}  ${shared_SOURCES})
 
+  IF(UNIX)
+    TARGET_LINK_LIBRARIES(sundials_cvode_shared m)
+  ENDIF()
+
+  IF(LAPACK_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_cvode_shared ${LAPACK_LIBRARIES})
+  ENDIF()
+
+  IF(KLU_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_cvode_shared ${KLU_LIBRARIES})
+  ENDIF ()
+
   # Set the library name and make sure it is not deleted
   SET_TARGET_PROPERTIES(sundials_cvode_shared
     PROPERTIES OUTPUT_NAME sundials_cvode CLEAN_DIRECT_OUTPUT 1)
diff --git a/src/cvodes/CMakeLists.txt b/src/cvodes/CMakeLists.txt
index 38229df..c7b4f60 100644
--- a/src/cvodes/CMakeLists.txt
+++ b/src/cvodes/CMakeLists.txt
@@ -132,6 +132,18 @@ IF(BUILD_SHARED_LIBS)
   ADD_LIBRARY(sundials_cvodes_shared SHARED 
     ${cvodes_SOURCES}  ${cvodes_BL_SOURCES}  ${shared_SOURCES})
 
+  IF(UNIX)
+    TARGET_LINK_LIBRARIES(sundials_cvodes_shared m)
+  ENDIF()
+
+  IF(LAPACK_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_cvodes_shared ${LAPACK_LIBRARIES})
+  ENDIF()
+
+  IF(KLU_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_cvodes_shared ${KLU_LIBRARIES})
+  ENDIF()
+ 
   # Set the library name and make sure it is not deleted
   SET_TARGET_PROPERTIES(sundials_cvodes_shared
     PROPERTIES OUTPUT_NAME sundials_cvodes CLEAN_DIRECT_OUTPUT 1)
diff --git a/src/ida/CMakeLists.txt b/src/ida/CMakeLists.txt
index 7aa8c84..cea11dc 100644
--- a/src/ida/CMakeLists.txt
+++ b/src/ida/CMakeLists.txt
@@ -135,6 +135,18 @@ IF(BUILD_SHARED_LIBS)
   ADD_LIBRARY(sundials_ida_shared SHARED
     ${ida_SOURCES}  ${ida_BL_SOURCES} ${shared_SOURCES})
 
+  IF(UNIX)
+    TARGET_LINK_LIBRARIES(sundials_ida_shared m)
+  ENDIF()
+
+  IF(LAPACK_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_ida_shared ${LAPACK_LIBRARIES})
+  ENDIF()
+
+  IF(KLU_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_ida_shared ${KLU_LIBRARIES})
+  ENDIF()
+
   # Set the library name and make sure it is not deleted
   SET_TARGET_PROPERTIES(sundials_ida_shared
     PROPERTIES OUTPUT_NAME sundials_ida CLEAN_DIRECT_OUTPUT 1)
diff --git a/src/idas/CMakeLists.txt b/src/idas/CMakeLists.txt
index 400a228..708a7f0 100644
--- a/src/idas/CMakeLists.txt
+++ b/src/idas/CMakeLists.txt
@@ -137,6 +137,18 @@ IF(BUILD_SHARED_LIBS)
   ADD_LIBRARY(sundials_idas_shared SHARED 
     ${idas_SOURCES}  ${idas_BL_SOURCES} ${idas_SLU_SOURCES} ${shared_SOURCES})
 
+  IF(UNIX)
+    TARGET_LINK_LIBRARIES(sundials_idas_shared m)
+  ENDIF()
+
+  IF(LAPACK_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_idas_shared ${LAPACK_LIBRARIES})
+  ENDIF()
+
+  IF(KLU_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_idas_shared ${KLU_LIBRARIES})
+  ENDIF()
+
   # Set the library name and make sure it is not deleted
   SET_TARGET_PROPERTIES(sundials_idas_shared
     PROPERTIES OUTPUT_NAME sundials_idas CLEAN_DIRECT_OUTPUT 1)
diff --git a/src/kinsol/CMakeLists.txt b/src/kinsol/CMakeLists.txt
index d22ed62..86da83a 100644
--- a/src/kinsol/CMakeLists.txt
+++ b/src/kinsol/CMakeLists.txt
@@ -145,6 +145,18 @@ IF(BUILD_SHARED_LIBS)
   ADD_LIBRARY(sundials_kinsol_shared SHARED
     ${kinsol_SOURCES} ${kinsol_BL_SOURCES} ${kinsol_SLU_SOURCES} ${kinsol_KLU_SOURCES} ${shared_SOURCES})
 
+  IF(UNIX)
+    TARGET_LINK_LIBRARIES(sundials_kinsol_shared m)
+  ENDIF()
+
+  IF(LAPACK_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_kinsol_shared ${LAPACK_LIBRARIES})
+  ENDIF()
+
+  IF(KLU_FOUND)
+    TARGET_LINK_LIBRARIES(sundials_kinsol_shared ${KLU_LIBRARIES})
+  ENDIF()
+
   # Set the library name and make sure it is not deleted
   SET_TARGET_PROPERTIES(sundials_kinsol_shared
     PROPERTIES OUTPUT_NAME sundials_kinsol CLEAN_DIRECT_OUTPUT 1)
diff --git a/src/nvec_ser/CMakeLists.txt b/src/nvec_ser/CMakeLists.txt
index de9fb12..4df70ba 100644
--- a/src/nvec_ser/CMakeLists.txt
+++ b/src/nvec_ser/CMakeLists.txt
@@ -54,6 +54,9 @@ ENDIF(BUILD_STATIC_LIBS)
 #  - Install the NVECSERIAL library
 IF(BUILD_SHARED_LIBS)
   ADD_LIBRARY(sundials_nvecserial_shared SHARED ${nvecserial_SOURCES} ${shared_SOURCES})
+  IF(UNIX)
+    TARGET_LINK_LIBRARIES(sundials_nvecserial_shared m)
+  ENDIF()
   SET_TARGET_PROPERTIES(sundials_nvecserial_shared
     PROPERTIES OUTPUT_NAME sundials_nvecserial CLEAN_DIRECT_OUTPUT 1)
   SET_TARGET_PROPERTIES(sundials_nvecserial_shared
