--- a/solar_utils/src/spectrl2_2.c
+++ b/solar_utils/src/spectrl2_2.c
@@ -187,5 +187,5 @@
     static int   nr;       /* indicates the wavelength range */
     static int   i;              /* Loop counter */
-    static int   false  = 0;     /* 0 is false */
-    static int   true   = 1;     /* non-0 is true */
+    static int   spectrl_false = 0;     /* 0 is false */
+    static int   spectrl_true  = 1;     /* non-0 is true */
     static int   retval;    /* solpos return code */
@@ -239,4 +239,4 @@
     /* Angles of incidence and tilt angles must be preset */
-    track  = false;
+    track  = spectrl_false;
     if ( soldat->tilt < 0 ) // MM 2012-01-20, change tilt < 0, same as solpos
-        track    = true;
+        track    = spectrl_true;