--- a/setup.py
+++ b/setup.py
@@ -83,14 +83,19 @@
     def link_dylib_lib(self, objects, output_libname, output_dir=None,
                        libraries=None, library_dirs=None,
                        runtime_library_dirs=None, export_symbols=None,
                        debug=0, extra_preargs=None, extra_postargs=None,
                        build_temp=None, target_lang=None):
         """implementation of link_shared_lib"""
-        self.link("shared_library", objects,
+        self.link("shared_lib", objects,
                   self.library_filename(output_libname, lib_type='dylib'),
                   output_dir,
                   libraries, library_dirs, runtime_library_dirs,
                   export_symbols, debug,
                   extra_preargs, extra_postargs, build_temp, target_lang)