--- a/setup.py
+++ b/setup.py
@@ -41,6 +41,7 @@
 elif PLATFORM == 'darwin':
     LIB_FILE = 'lib%s.dylib'
     RPATH = "-Wl,-rpath,@loader_path/"
     INSTALL_NAME = "@rpath/" + LIB_FILE
-    CCFLAGS = LDFLAGS = ['-fPIC']
+    CCFLAGS = ['-fPIC']
+    LDFLAGS = ['-fPIC', '-dynamiclib']
 elif PLATFORM in ['linux', 'linux2']:
@@ -151,3 +152,8 @@
     from distutils.sysconfig import customize_compiler
     customize_compiler(CC)
+    if PLATFORM == 'darwin':
+        CC.set_executables(
+            linker_so=[
+                flag for flag in CC.linker_so
+                if flag not in ('-bundle', '-shared')
+            ])
     CC.add_include_dir(SRC_DIR)  # set includes directory