--- test/test_create_extrinsics.py      2023-12-29 04:09:41.000000000 -0600
+++ test/test_create_extrinsics.py      2024-02-18 14:10:18.294410998 -0600
@@ -53,7 +53,7 @@
         # Create balance transfer call
         call = self.kusama_substrate.compose_call(
             call_module='Balances',
-            call_function='transfer',
+            call_function='transfer_allow_death',
             call_params={
                 'dest': 'EaG2CRhJWPb7qmdcJvy3LiWdh26Jreu9Dx6R1rXxPmYXoDk',
                 'value': 3 * 10 ** 3
@@ -66,7 +66,7 @@
         decoded_extrinsic.decode(extrinsic.data)

         self.assertEqual(decoded_extrinsic['call']['call_module'].name, 'Balances')
-        self.assertEqual(decoded_extrinsic['call']['call_function'].name, 'transfer')
+        self.assertEqual(decoded_extrinsic['call']['call_function'].name, 'transfer_allow_death')
         self.assertEqual(extrinsic['nonce'], 0)
         self.assertEqual(extrinsic['tip'], 1)

@@ -77,7 +77,7 @@
             # Create balance transfer call
             call = substrate.compose_call(
                 call_module='Balances',
-                call_function='transfer',
+                call_function='transfer_allow_death',
                 call_params={
                     'dest': 'EaG2CRhJWPb7qmdcJvy3LiWdh26Jreu9Dx6R1rXxPmYXoDk',
                     'value': 3 * 10 ** 3
@@ -99,7 +99,7 @@

         balance_call = self.polkadot_substrate.compose_call(
             call_module='Balances',
-            call_function='transfer',
+            call_function='transfer_allow_death',
             call_params={
                 'dest': 'EaG2CRhJWPb7qmdcJvy3LiWdh26Jreu9Dx6R1rXxPmYXoDk',
                 'value': 3 * 10 ** 3
@@ -126,7 +126,7 @@

         call = self.kusama_substrate.compose_call(
             call_module='Balances',
-            call_function='transfer',
+            call_function='transfer_allow_death',
             call_params={
                 'dest': 'EaG2CRhJWPb7qmdcJvy3LiWdh26Jreu9Dx6R1rXxPmYXoDk',
                 'value': 3 * 10 ** 3
@@ -172,7 +172,7 @@

         call = self.kusama_substrate.compose_call(
             call_module='Balances',
-            call_function='transfer',
+            call_function='transfer_allow_death',
             call_params={
                 'dest': 'EaG2CRhJWPb7qmdcJvy3LiWdh26Jreu9Dx6R1rXxPmYXoDk',
                 'value': 2000
@@ -218,7 +218,7 @@
         # Create balance transfer call
         call = self.kusama_substrate.compose_call(
             call_module='Balances',
-            call_function='transfer',
+            call_function='transfer_allow_death',
             call_params={
                 'dest': 'EaG2CRhJWPb7qmdcJvy3LiWdh26Jreu9Dx6R1rXxPmYXoDk',
                 'value': 3 * 10 ** 3
--- test/test_type_registry.py      2024-02-18 16:08:54.936349966 -0600
+++ test/test_type_registry.py  2024-02-18 16:09:33.116719083 -0600
@@ -196,7 +196,7 @@

         call = substrate.compose_call(
             call_module='Balances',
-            call_function='transfer',
+            call_function='transfer_allow_death',
             call_params={
                 'dest': keypair_alice.ss58_address,
                 'value': 2000
@@ -216,7 +216,7 @@

         call = substrate.compose_call(
             call_module='Balances',
-            call_function='transfer',
+            call_function='transfer_allow_death',
             call_params={
                 'dest': keypair_alice.ss58_address,
                 'value': 2000
--- test/test_runtime_call.py       2024-02-18 16:29:15.883438375 -0600
+++ test/test_runtime_call.py   2024-02-18 16:29:39.479564108 -0600
@@ -51,7 +51,7 @@
     def test_transaction_payment(self):
         call = self.substrate.compose_call(
             call_module='Balances',
-            call_function='transfer',
+            call_function='transfer_allow_death',
             call_params={
                 'dest': 'EaG2CRhJWPb7qmdcJvy3LiWdh26Jreu9Dx6R1rXxPmYXoDk',
                 'value': 3 * 10 ** 3
--- test/test_subscriptions.py      2024-02-18 17:08:13.826631783 -0600
+++ test/test_subscriptions.py  2024-02-18 17:08:23.554717925 -0600
@@ -28,17 +28,6 @@
             url=settings.POLKADOT_NODE_URL
         )

-    def test_query_subscription(self):
-
-        def subscription_handler(obj, update_nr, subscription_id):
-
-            return {'update_nr': update_nr, 'subscription_id': subscription_id}
-
-        result = self.substrate.query("System", "Events", [], subscription_handler=subscription_handler)
-
-        self.assertEqual(result['update_nr'], 0)
-        self.assertIsNotNone(result['subscription_id'])
-
     def test_subscribe_storage_multi(self):

         def subscription_handler(storage_key, updated_obj, update_nr, subscription_id):
