diff --git a/graphframes/examples/belief_propagation.py b/graphframes/examples/belief_propagation.py
index 1d609ce..5b71ef4 100644
--- a/graphframes/examples/belief_propagation.py
+++ b/graphframes/examples/belief_propagation.py
@@ -19,7 +19,6 @@ import math
 
 from pyspark import SparkConf, SparkContext
 from pyspark.sql import SQLContext, functions as sqlfunctions, types
-from pyspark.tests import QuietTest as SuppressSparkLogs
 
 from graphframes import GraphFrame
 from graphframes.lib import AggregateMessages as AM
