diff --git a/src/Makefile b/src/Makefile
--- a/src/Makefile	(revision db09f6eb2e70ae0661a0cd9ad9b58b9b566311a9)
+++ b/src/Makefile	(date 1631795654521)
@@ -35,14 +35,7 @@
 OPT=$(OPTIMIZATION)

 # Detect if the compiler supports C11 _Atomic
-C11_ATOMIC := $(shell sh -c 'echo "\#include <stdatomic.h>" > foo.c; \
-	$(CC) -std=c11 -c foo.c -o foo.o > /dev/null 2>&1; \
-	if [ -f foo.o ]; then echo "yes"; rm foo.o; fi; rm foo.c')
-ifeq ($(C11_ATOMIC),yes)
-	STD+=-std=c11
-else
-	STD+=-std=c99
-endif
+STD+=-std=c11

 PREFIX?=/usr/local
 INSTALL_BIN=$(PREFIX)/bin
