diff --git a/src/sta/CMakeLists.txt b/src/sta/CMakeLists.txt
index 159ee19..badd706 100644
--- a/src/sta/CMakeLists.txt
+++ b/src/sta/CMakeLists.txt
@@ -30,6 +30,7 @@ project(STA VERSION 2.3.1
 
 option(USE_CUDD "Use CUDD BDD package")
 option(CUDD_DIR "CUDD BDD package directory")
+option(INSTALL_LIBOPENSTA "Install OpenSTA library and headers")
 
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
@@ -511,11 +512,13 @@ message(STATUS "STA executable: ${STA_HOME}/app/sta")
 # executable
 install(TARGETS sta DESTINATION bin)
 
+if (INSTALL_LIBOPENSTA)
 # library
 install(TARGETS OpenSTA DESTINATION lib)
 
 # include
 install(DIRECTORY include/sta  DESTINATION include)
+endif()
 
 ################################################################
 
