diff --git a/go.mod b/go.mod
index dac57141..fff8a77c 100644
--- a/go.mod
+++ b/go.mod
@@ -8,3 +8,5 @@ require (
 	gitlab.com/ZaberTech/zaber-device-db-service v0.0.0-20191130221157-4b4564b0b58e
 	gotest.tools v2.2.0+incompatible
 )
+
+go 1.13
diff --git a/py/zaber_motion/zaber_motion/bindings.py b/py/zaber_motion/zaber_motion/bindings.py
index e3ef5c91..6c785757 100644
--- a/py/zaber_motion/zaber_motion/bindings.py
+++ b/py/zaber_motion/zaber_motion/bindings.py
@@ -1,8 +1,12 @@
 from ctypes import c_void_p, c_int, c_int64, c_uint8, CFUNCTYPE
+from ctypes import cdll
 from typing import Any
 import platform
 import sys
-import importlib
+
+
+def load_library(name):
+    return cdll.LoadLibrary(name)
 
 
 def _load_library() -> Any:
@@ -18,8 +22,7 @@ def _load_library() -> Any:
     if os_system == "darwin":
         ext = ".dylib"
     lib_name = "zaber-motion-lib-{}-{}{}".format(os_system, arch, ext)
-    module_name = "zaber_motion_bindings_{}".format(os_system)
-    return importlib.import_module(module_name).load_library(lib_name)  # type: ignore
+    return load_library(lib_name)
 
 
 lib = _load_library()
