From cd9c30ab1d9abc7cba28aeb2fd313382898d004a Mon Sep 17 00:00:00 2001
From: Nicholas Devenish <ndevenish@gmail.com>
Date: Wed, 4 Sep 2024 14:15:36 +0100
Subject: [PATCH 1/4] Use lib3mf-python

---
 src/build123d/mesher.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/build123d/mesher.py b/src/build123d/mesher.py
index 884ff53..6aea957 100644
--- a/src/build123d/mesher.py
+++ b/src/build123d/mesher.py
@@ -109,7 +109,7 @@ from OCP.TopExp import TopExp_Explorer
 from OCP.TopoDS import TopoDS_Compound
 from OCP.TopLoc import TopLoc_Location
 
-from py_lib3mf import Lib3MF
+from lib3mf import Lib3MF
 from build123d.build_enums import MeshType, Unit
 from build123d.geometry import Color, TOLERANCE, Vector
 from build123d.topology import Compound, Face, Shape, Shell, Solid, downcast
-- 
2.39.3 (Apple Git-146)

