From ac2a590e0578901868fd296e1d2227c1231b8849 Mon Sep 17 00:00:00 2001
From: Nicholas Devenish <ndevenish@gmail.com>
Date: Thu, 5 Sep 2024 08:42:54 +0100
Subject: [PATCH 2/4] Don't try to help lib3mf find it's library

---
 src/build123d/mesher.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/build123d/mesher.py b/src/build123d/mesher.py
index 6aea957..462f41e 100644
--- a/src/build123d/mesher.py
+++ b/src/build123d/mesher.py
@@ -148,8 +148,7 @@ class Mesher:
 
     def __init__(self, unit: Unit = Unit.MM):
         self.unit = unit
-        libpath = os.path.dirname(Lib3MF.__file__)
-        self.wrapper = Lib3MF.Wrapper(os.path.join(libpath, "lib3mf"))
+        self.wrapper = Lib3MF.Wrapper()
         self.model = self.wrapper.CreateModel()
         self.model.SetUnit(Mesher._map_b3d_to_3mf_unit[unit])
         self.meshes: list[Lib3MF.MeshObject] = []
-- 
2.39.3 (Apple Git-146)

