From 31b299547439909c4580ca4e448e501e9747dc43 Mon Sep 17 00:00:00 2001
From: Nicholas Devenish <ndevenish@gmail.com>
Date: Thu, 5 Sep 2024 08:55:47 +0100
Subject: [PATCH 3/4] Change explicit dependency lock version

---
 tests/test_mesher.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test_mesher.py b/tests/test_mesher.py
index b1119a4..4b8b7e2 100644
--- a/tests/test_mesher.py
+++ b/tests/test_mesher.py
@@ -36,7 +36,7 @@ class DirectApiTestCase(unittest.TestCase):
 class TestProperties(unittest.TestCase):
     def test_version(self):
         exporter = Mesher()
-        self.assertEqual(exporter.library_version, "2.3.1")
+        self.assertEqual(exporter.library_version, "2.3.2")
 
     def test_units(self):
         for unit in Unit:
-- 
2.39.3 (Apple Git-146)

