From bbb4528a308f5310490f0c07ba81e8ea3dd3f266 Mon Sep 17 00:00:00 2001
From: Nicholas Devenish <ndevenish@gmail.com>
Date: Thu, 5 Sep 2024 16:58:27 +0100
Subject: [PATCH 4/4] Fix raw regex

---
 tests/test_exporters3d.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test_exporters3d.py b/tests/test_exporters3d.py
index 6af0d53..e3743ed 100644
--- a/tests/test_exporters3d.py
+++ b/tests/test_exporters3d.py
@@ -90,7 +90,7 @@ class TestExportStep(DirectApiTestCase):
         os.remove("assembly.step")
         self.assertNotEqual(step_data.find("DRAUGHTING_PRE_DEFINED_COLOUR('red')"), -1)
         self.assertNotEqual(step_data.find("DRAUGHTING_PRE_DEFINED_COLOUR('blue')"), -1)
-        self.assertEqual(len(re.findall("[\(\,]25.4[\,\)]", step_data)), 45)
+        self.assertEqual(len(re.findall(r"[\(\,]25.4[\,\)]", step_data)), 45)
         self.assertNotEqual(step_data.find("PRODUCT('sphere',"), -1)
         self.assertNotEqual(step_data.find("PRODUCT('box',"), -1)
         self.assertNotEqual(step_data.find("PRODUCT('assembly',"), -1)
-- 
2.39.3 (Apple Git-146)

