--- setup.py.orig	2017-04-11 07:22:57.569193300 -0700
+++ setup.py	2017-04-11 07:50:38.729951100 -0700
@@ -125,7 +125,6 @@
 Name: Pycairo
 Description: Python %(py_version)d bindings for cairo
 Version: %(version)s
-Requires: cairo
 Cflags: -I${prefix}/include/pycairo
 Libs:
 """ % {
@@ -161,7 +160,7 @@
         )
 
     def run(self):
-        pkg_config_version_check('cairo', CAIRO_VERSION_REQUIRED)
+        #pkg_config_version_check('cairo', CAIRO_VERSION_REQUIRED)
 
         if self.enable_xpyb:
             if sys.version_info[0] != 2:
@@ -206,6 +205,11 @@
 
 
 def main():
+    import os
+    #pfx = subprocess.check_output(['cygpath.exe', '-u', os.environ['LIBRARY_PREFIX']]).decode().strip()
+    #if pfx == '/':
+    #    pfx = ''
+    pfx = os.environ['LIBRARY_PREFIX']
 
     cairo_ext = Extension(
         name='cairo._cairo',
@@ -219,9 +223,9 @@
             'cairo/region.c',
             'cairo/surface.c',
         ],
-        include_dirs=pkg_config_parse('--cflags-only-I', 'cairo'),
-        library_dirs=pkg_config_parse('--libs-only-L', 'cairo'),
-        libraries=pkg_config_parse('--libs-only-l', 'cairo'),
+        include_dirs=['%s\include\cairo' % pfx],
+        library_dirs=['%s\lib' % pfx],
+        extra_link_args=['cairo.lib'],
     )
 
     setup(
