diff --git Makefile.am Makefile.am
index 1e7ae54..eb4d8e3 100644
--- Makefile.am
+++ Makefile.am
@@ -277,7 +277,6 @@ curve_keygen_SOURCES = tools/curve_keygen.cpp
 # tests
 #
 test_apps = \
-	test_system \
 	test_pair_inproc \
 	test_pair_tcp \
 	test_reqrep_inproc \
@@ -334,9 +333,6 @@ test_apps = \
 	test_capabilities \
 	test_xpub_nodrop
 
-test_system_SOURCES = tests/test_system.cpp
-test_system_LDADD = libzmq.la
-
 test_pair_inproc_SOURCES = \
 	tests/test_pair_inproc.cpp \
 	tests/testutil.hpp
