diff --git Makefile.am Makefile.am
index 1e7ae54..cdd8304 100644
--- Makefile.am
+++ Makefile.am
@@ -515,8 +515,7 @@ test_apps += \
 	test_shutdown_stress \
 	test_pair_ipc \
 	test_reqrep_ipc \
-	test_timeo \
-	test_filter_ipc
+	test_timeo
 
 test_shutdown_stress_SOURCES = tests/test_shutdown_stress.cpp
 test_shutdown_stress_LDADD = libzmq.la
@@ -534,9 +533,6 @@ test_reqrep_ipc_LDADD = libzmq.la
 test_timeo_SOURCES = tests/test_timeo.cpp
 test_timeo_LDADD = libzmq.la
 
-test_filter_ipc_SOURCES = tests/test_filter_ipc.cpp
-test_filter_ipc_LDADD = libzmq.la
-
 if HAVE_FORK
 test_apps += test_fork
 
