From 2c9292a05f43a1a7ebf3de36c128729f4bcff5d9 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 22 Aug 2022 09:54:11 -0600
Subject: [PATCH] Update setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index a8be0a4..219d9b6 100644
--- a/setup.py
+++ b/setup.py
@@ -93,7 +93,7 @@ def find_package_data(packages):
     url="http://github.com/chopralab/paddy",
     packages=packages,
     zip_safe=False,
-    install_requires=['sklearn', 'future', 'numpy', 'matplotlib'],
+    install_requires=['scikit-learn', 'future', 'numpy', 'matplotlib'],
     tests_require=['pandas'],
     classifiers=[
         "Development Status :: 4 - Beta",
