From da61816c3a41ffbf8381c04b6df5e4aada262e01 Mon Sep 17 00:00:00 2001
From: Jinzhe Zeng <jinzhe.zeng@rutgers.edu>
Date: Thu, 28 Nov 2024 18:44:21 -0500
Subject: [PATCH] hack ROOT_DIR

---
 nodejs_wheel/executable.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nodejs_wheel/executable.py b/nodejs_wheel/executable.py
index 42f216e..dede038 100644
--- a/nodejs_wheel/executable.py
+++ b/nodejs_wheel/executable.py
@@ -10,7 +10,7 @@ try:
 except ImportError:
     from typing_extensions import Literal  # type: ignore
 
-ROOT_DIR = os.path.dirname(__file__)
+ROOT_DIR = os.path.join(os.path.dirname(__file__), "..", "..", "..", "..")
 
 
 @overload
-- 
2.34.1

