From 3c744fcf21d755f393a533df49afdb0c52ce9be3 Mon Sep 17 00:00:00 2001
From: Jinzhe Zeng <jinzhe.zeng@rutgers.edu>
Date: Thu, 28 Nov 2024 18:44:21 -0500
Subject: [PATCH] hack ROOT_DIR

---
 nodejs_wheel/executable.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/nodejs_wheel/executable.py b/nodejs_wheel/executable.py
index 42f216e..e435399 100644
--- a/nodejs_wheel/executable.py
+++ b/nodejs_wheel/executable.py
@@ -10,7 +10,12 @@ try:
 except ImportError:
     from typing_extensions import Literal  # type: ignore
 
-ROOT_DIR = os.path.dirname(__file__)
+if os.name == "nt":
+    ROOT_DIR = os.path.join(os.path.dirname(__file__), "..", "..", "..")
+    NODE_MODULES_DIR = os.path.join(ROOT_DIR, "node_modules")
+else:
+    ROOT_DIR = os.path.join(os.path.dirname(__file__), "..", "..", "..", "..")
+    NODE_MODULES_DIR = os.path.join(ROOT_DIR, "lib", "node_modules")
 
 
 @overload
@@ -179,7 +184,7 @@ def npm(
     if args is None:
         args = sys.argv[1:]
     return call_node(
-        os.path.join(ROOT_DIR, "lib", "node_modules", "npm", "bin", "npm-cli.js"),
+        os.path.join(NODE_MODULES_DIR, "npm", "bin", "npm-cli.js"),
         *args,
         return_completed_process=return_completed_process,
         **kwargs,
@@ -233,7 +238,7 @@ def npx(
     if args is None:
         args = sys.argv[1:]
     return call_node(
-        os.path.join(ROOT_DIR, "lib", "node_modules", "npm", "bin", "npx-cli.js"),
+        os.path.join(NODE_MODULES_DIR, "npm", "bin", "npx-cli.js"),
         *args,
         return_completed_process=return_completed_process,
         **kwargs,
-- 
2.34.1

