From 3d745d3ea26f938aec5decbf290298184a46a541 Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Wed, 28 Feb 2024 08:27:02 +0100
Subject: [PATCH 2/5] Fix build on MinGW

---
 src/common/Fork.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/common/Fork.c b/src/common/Fork.c
index e7ad293..add5368 100644
--- a/src/common/Fork.c
+++ b/src/common/Fork.c
@@ -155,10 +155,7 @@ Extern void SUFFIX(cubawait)(Spin **pspin)
 
 Extern void SUFFIX(cubafork)(Spin **pspin) {}
 
-Extern void SUFFIX(cubawait)(Spin **pspin)
-{
-  MasterExit();
-}
+Extern void SUFFIX(cubawait)(Spin **pspin) {}
 
 #endif
 
-- 
2.44.0

