diff --git a/shiboken/CMakeLists.txt b/shiboken/CMakeLists.txt
index 191a58c..fc89439 100644
--- a/shiboken/CMakeLists.txt
+++ b/shiboken/CMakeLists.txt
@@ -78,6 +78,7 @@ set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
     --avoid-protected-hack --enable-pyside-extensions
     ${qt_includes} ${qmn_includes}
     -I${CMAKE_SOURCE_DIR}/src
+    -I$ENV{BUILD_PREFIX}/include/c++/v1/
     -T${PYSIDE_TYPESYSTEMS}
     --output-directory=${CMAKE_CURRENT_BINARY_DIR}
     )
