diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 1f78168..6414a95 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -8,6 +8,10 @@
 # (4) Addition of the test executable via add_test.
 
 ### Core tests, which rely only on boost-test and ndarray.
+set(Boost_NO_BOOST_CMAKE OFF)
+set(Boost_USE_STATIC_LIBS OFF)
+set(BUILD_SHARED_LIBS ON)
+
 find_package(Boost COMPONENTS unit_test_framework REQUIRED)
 
 include_directories(
@@ -17,12 +21,12 @@ include_directories(
 
 add_executable(ndarray ndarray.cc)
 add_dependencies(ndarray preprocess_includes2)
-target_link_libraries(ndarray ${Boost_LIBRARIES})
+target_link_libraries(ndarray Boost::unit_test_framework)
 add_test(test_ndarray ndarray)
 
 add_executable(views views.cc)
 add_dependencies(views preprocess_includes2)
-target_link_libraries(views ${Boost_LIBRARIES})
+target_link_libraries(views Boost::unit_test_framework)
 add_test(test_views views)
 
 
@@ -34,7 +38,7 @@ if(NDARRAY_EIGEN)
 
     add_executable(ndarray-eigen ndarray-eigen.cc)
     add_dependencies(ndarray-eigen preprocess_includes2)
-    target_link_libraries(ndarray-eigen ${Boost_LIBRARIES})
+    target_link_libraries(ndarray-eigen Boost::unit_test_framework)
     add_test(test_ndarray_eigen ndarray-eigen)
 endif(NDARRAY_EIGEN)
 
@@ -45,7 +49,7 @@ if(NDARRAY_FFTW)
 
     add_executable(ndarray-fft ndarray-fft.cc)
     add_dependencies(ndarray-fft preprocess_includes2)
-    target_link_libraries(ndarray-fft ${Boost_LIBRARIES} ${FFTW_LIBRARIES})
+    target_link_libraries(ndarray-fft Boost::unit_test_framework ${FFTW_LIBRARIES})
     add_test(test_ndarray-fft ndarray-fft)
 endif(NDARRAY_FFTW)
 
