--- a/test/h5cpp_tests/node_tests/dataset_io_test.py
+++ b/test/h5cpp_tests/node_tests/dataset_io_test.py
@@ -131,6 +131,20 @@
         else:
             self.assertEqual(read, bdata)

+    def testWriteVariableLengthEmptycalar(self):
+        data = u""
+        bdata = b""
+        dtype = h5cpp.datatype.String.variable()
+        dataset = Dataset(
+            self.root, h5cpp.Path("VariableLengthStringEmptyScalar"),
+            dtype, Scalar())
+        # dataset.write(data)
+        read = dataset.read()
+        if sys.version_info > (3,):
+            self.assertEqual(read, data)
+        else:
+            self.assertEqual(read, bdata)
+
     def testWriteIntegerArray(self):

         data = numpy.array([[1, 2, 3, 4], [5, 6, 7, 8]])
