--- a/setup.py
+++ b/setup.py
@@ -69,6 +69,7 @@

     nexus_config.add_link_library('pninexus')
     nexus_config.add_link_library('h5cpp')
+    nexus_config.add_link_library('hdf5_hl')
     nexus_config.add_link_library(
         "boost_python{major}{minor}".format(major=sys.version_info.major,
                                             minor=sys.version_info.minor))
@@ -77,7 +78,6 @@
     hdf5_hl_path = os.environ.get('HDF5_HL_LOCAL_PATH')
     if hdf5_hl_path:
         # use when h5cpp compiled with --as-needed
-        nexus_config.add_link_library('hdf5_hl')
         nexus_config.add_library_directory(
             # '/usr/lib/x86_64-linux-gnu/hdf5/serial/'
             hdf5_hl_path
@@ -149,11 +149,18 @@
 # ----------------------------------------------------------------------------
 # set compiler options
 # ----------------------------------------------------------------------------
-if sys.platform == "win32":
-    pass
+if sys.platform in ["win32", "win64"]:
+    arguments = ['/std:c++17']
 else:
-    arguments = ['-std=c++17', '-Wall', '-Wextra', '-fdiagnostics-show-option']
-    nexus_config.add_compiler_arguments(arguments)
+    # old version                                                                                                                                        # arguments = ['-std=c++17', '-Wall', '-Wextra',
+    #              '-fdiagnostics-show-option']
+    # lto does not work
+    # arguments = ['-std=c++17', '-Wall', '-Wextra',
+    #       '-fdiagnostics-show-option', '-flto=auto', '-ffat-lto-objects']
+    # no-lto to be on the save side
+    arguments = ['-std=c++17', '-Wall', '-Wextra', '-fdiagnostics-show-option',
+                 '-fno-lto']
+nexus_config.add_compiler_arguments(arguments)

 # ----------------------------------------------------------------------------
 # creating the extension factories
