--- a/setup.py
+++ b/setup.py
@@ -78,11 +78,22 @@
     if hdf5_hl_path:
         # use when h5cpp compiled with --as-needed
         nexus_config.add_link_library('hdf5_hl')
-        nexus_config.add_library_directory(
-            # '/usr/lib/x86_64-linux-gnu/hdf5/serial/'
-            hdf5_hl_path
-        )
+        if hdf5_hl_path != '__SYS__':
+            nexus_config.add_library_directory(
+                # '/usr/lib/x86_64-linux-gnu/hdf5/serial/'
+                hdf5_hl_path
+            )

+    hdf5_path = os.environ.get('HDF5_LOCAL_PATH')
+    if hdf5_path:
+        # use when h5cpp compiled with --as-needed
+        nexus_config.add_link_library('hdf5')
+        if hdf5_path != '__SYS__':
+            nexus_config.add_library_directory(
+                # '/usr/lib/x86_64-linux-gnu/hdf5/serial/'
+                hdf5_path
+            )
+
     h5cpp_path = os.environ.get('H5CPP_LOCAL_PATH')
     if h5cpp_path:
         # use when h5cpp locally installed
@@ -149,11 +160,19 @@
 # ----------------------------------------------------------------------------
 # set compiler options
 # ----------------------------------------------------------------------------
-if sys.platform == "win32":
-    pass
+if sys.platform in ["win32", "win64"]:
+    arguments = ['/std:c++17']
 else:
-    arguments = ['-std=c++17', '-Wall', '-Wextra', '-fdiagnostics-show-option']
-    nexus_config.add_compiler_arguments(arguments)
+    # old version
+    # arguments = ['-std=c++17', '-Wall', '-Wextra',
+    #              '-fdiagnostics-show-option']
+    # lto does not work
+    # arguments = ['-std=c++17', '-Wall', '-Wextra',
+    #       '-fdiagnostics-show-option', '-flto=auto', '-ffat-lto-objects']
+    # no-lto to be on the save side
+    arguments = ['-std=c++17', '-Wall', '-Wextra', '-fdiagnostics-show-option',
+                 '-fno-lto']
+nexus_config.add_compiler_arguments(arguments)

 # ----------------------------------------------------------------------------
 # creating the extension factories
