--- a/src/cpp/h5cpp/numpy/hdf5_variable_strings_support.hpp
+++ b/src/cpp/h5cpp/numpy/hdf5_variable_strings_support.hpp
@@ -116,16 +116,25 @@
     PyObject ***dataptr = (PyObject***)NpyIter_GetDataPtrArray(iter);
     for(auto string: buffer)
     {
+      PyObject *ptr;
       if(string == NULL)
 	{
-	  char empty[] = {'\0'};
-	  string = empty;
+#if PY_VERSION_HEX >= 0x030d0000
+	  ptr = Py_GetConstant(Py_CONSTANT_EMPTY_STR);
+#elif PY_MAJOR_VERSION >= 3
+	  ptr = PyUnicode_FromString("");
+#else
+	  ptr = PyString_FromString("");
+#endif
 	}
+      else
+	{
 #if PY_MAJOR_VERSION >= 3
-      PyObject *ptr = PyUnicode_FromString(string);
+	  ptr = PyUnicode_FromString(string);
 #else
-      PyObject *ptr = PyString_FromString(string);
+	  ptr = PyString_FromString(string);
 #endif
+	}
       if(ptr==NULL)
       {
         std::cerr<<"could not create python string!"<<std::endl;
