--- cykdtree-0.2.4.orig/cykdtree/__init__.py	2017-02-10 00:52:30.000000000 -0200
+++ cykdtree-0.2.4/cykdtree/__init__.py	2017-08-10 18:19:55.205095007 -0300
@@ -1,6 +1,4 @@
 from cykdtree.kdtree import PyKDTree, PyNode
-from cykdtree import tests
-import nose
 import sys
 import os
 
@@ -34,4 +32,4 @@
 
 
 
-__all__ = ["PyKDTree", "PyNode", "tests", "run_nose"]
+__all__ = ["PyKDTree", "PyNode", "run_nose"]
