--- setup.py	2024-09-23 08:38:56.000000000 -0400
+++ setup.py	2024-10-26 14:10:02.602186200 -0400
@@ -3,7 +3,8 @@
 
 from setuptools import setup
 from setuptools import find_packages
-from cubedpandas import VERSION as CUBEDPANDAS_VERSION
+import os
+CUBEDPANDAS_VERSION = os.environ["VERSION"]
 
 
 # ...to run the build and deploy process to pypi.org manually:
@@ -78,4 +79,4 @@
         'Documentation': 'https://zeutschler.github.io/cubedpandas/',
         'GitHub': 'https://github.com/Zeutschler/cubedpandas',
     },
-)
\ No newline at end of file
+)
