diff --git a/openemacs.c b/openemacs.c
index 534a364..cd47244 100644
--- a/openemacs.c
+++ b/openemacs.c
@@ -133,10 +133,10 @@ char *PYTHON_SYNTAX_HIGHLIGHT_KEYWORDS[] = {
 };
 
 struct editor_syntax SYNTAX_HIGHLIGHT_DATABASE[] = {
-    { .file_match = C_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = C_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = "//", .multi_line_comment_start = "/*", .multi_line_comment_end = "*/" },
+    { .file_match = C_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = C_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = {'/', '/'}, .multi_line_comment_start = {'/', '*'}, .multi_line_comment_end = {'*', '/'} },
     { .file_match = PYTHON_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = PYTHON_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = "#\0", .multi_line_comment_start = "", .multi_line_comment_end = "" },
-    { .file_match = GO_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = GO_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = "//", .multi_line_comment_start = "", .multi_line_comment_end = "" },
-    { .file_match = V_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = V_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = "//", .multi_line_comment_start = "", .multi_line_comment_end = "" }
+    { .file_match = GO_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = GO_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = {'/', '/'}, .multi_line_comment_start = "", .multi_line_comment_end = "" },
+    { .file_match = V_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = V_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = {'/', '/'}, .multi_line_comment_start = "", .multi_line_comment_end = "" }
 };
 
 __attribute__((format(printf, 1, 2)))
