diff --git a/openemacs.c b/openemacs.c
index 534a364..b994b91 100644
--- a/openemacs.c
+++ b/openemacs.c
@@ -35,9 +35,10 @@ enum SYNTAX_HIGHLIGHT_MODE {
 struct editor_syntax {
     char **file_match;
     char **keywords;
-    char single_line_comment_start[2];
-    char multi_line_comment_start[3];
-    char multi_line_comment_end[3];
+    // unterminated-string-initialization warning is a false positive here:
+    char __attribute__((__nonstring__)) single_line_comment_start[2];
+    char __attribute__((__nonstring__)) multi_line_comment_start[3];
+    char __attribute__((__nonstring__)) multi_line_comment_end[3];
 };
 
 // A single line of the file we are editing.
@@ -133,10 +134,10 @@ char *PYTHON_SYNTAX_HIGHLIGHT_KEYWORDS[] = {
 };
 
 struct editor_syntax SYNTAX_HIGHLIGHT_DATABASE[] = {
-    { .file_match = C_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = C_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = "//", .multi_line_comment_start = "/*", .multi_line_comment_end = "*/" },
+    { .file_match = C_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = C_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = {'/', '/'}, .multi_line_comment_start = {'/', '*'}, .multi_line_comment_end = {'*', '/'} },
     { .file_match = PYTHON_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = PYTHON_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = "#\0", .multi_line_comment_start = "", .multi_line_comment_end = "" },
-    { .file_match = GO_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = GO_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = "//", .multi_line_comment_start = "", .multi_line_comment_end = "" },
-    { .file_match = V_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = V_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = "//", .multi_line_comment_start = "", .multi_line_comment_end = "" }
+    { .file_match = GO_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = GO_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = {'/', '/'}, .multi_line_comment_start = "", .multi_line_comment_end = "" },
+    { .file_match = V_SYNTAX_HIGHLIGHT_FILE_EXTENSIONS, .keywords = V_SYNTAX_HIGHLIGHT_KEYWORDS, .single_line_comment_start = {'/', '/'}, .multi_line_comment_start = "", .multi_line_comment_end = "" }
 };
 
 __attribute__((format(printf, 1, 2)))
