---
 setup.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 7a37a6d..b4eaaed 100644
--- a/setup.py
+++ b/setup.py
@@ -25,6 +25,7 @@
 import os
 import platform
 import sys
+import sysconfig
 from setuptools import setup, Extension

 # base source directory
@@ -61,11 +62,10 @@ thin_depends = [os.path.join(impl_dir, n) \
 thin_depends.append(base_pxd)

 # if the platform is macOS, add arguments required for cross-compilation for
-# both x86_64 and arm64 architectures.
-# Use a Universal 2 Python binary to build, or set an archtecture
-# before building, e.g. export ARCHFLAGS="-arch x86_64"
+# both x86_64 and arm64 architectures if the python interpreter is a
+# universal2 version.
 extra_compile_args = []
-if sys.platform == "darwin":
+if sys.platform == "darwin" and "universal2" in sysconfig.get_platform():
     if platform.machine() == "x86_64":
         target = "arm64-apple-macos"
     else:
--
2.36.1