diff --git a/pexsi_v1.2.0/src/CMakeLists.txt b/pexsi_v1.2.0/src/CMakeLists.txt
index 8f475cb..d2d9b29 100644
--- a/pexsi_v1.2.0/src/CMakeLists.txt
+++ b/pexsi_v1.2.0/src/CMakeLists.txt
@@ -56,7 +56,7 @@ if( PEXSI_ENABLE_FORTRAN )
   set( PEXSI_SRC_F90 f_interface.f90 )
 endif()
 
-add_library( pexsi STATIC ${PEXSI_SRC_C} ${PEXSI_SRC_CXX} ${PEXSI_SRC_F90})
+add_library( pexsi ${PEXSI_SRC_C} ${PEXSI_SRC_CXX} ${PEXSI_SRC_F90})
 add_library( PEXSI::pexsi ALIAS pexsi )
 
 # Set up Fortran module directory
