--- a/src/superlu_dist_internal_complex.cpp	2020-02-25 16:12:53.779494396 +0000
+++ b/src/superlu_dist_internal_complex.cpp	2020-02-25 16:13:22.075702493 +0000
@@ -51,22 +51,22 @@ such enhancements or derivative works th
 #include <superlu_zdefs.h>
 
 #include <numeric>
-#include <Cnames.h>
+#include <superlu_FCnames.h>
 extern "C"{ void
 #ifdef SUPERLU_DIST_MAJOR_VERSION
 #if SUPERLU_DIST_MAJOR_VERSION < 5
-pzsymbfact(superlu_options_t *options, SuperMatrix *A, 
+pzsymbfact(superlu_options_t *options, SuperMatrix *A,
     ScalePermstruct_t *ScalePermstruct, gridinfo_t *grid,
     LUstruct_t *LUstruct, SuperLUStat_t *stat, int *numProcSymbFact,
     int *info, double *totalMemory, double *maxMemory );
 #else
-pzsymbfact(superlu_dist_options_t *options, SuperMatrix *A, 
+pzsymbfact(superlu_dist_options_t *options, SuperMatrix *A,
     ScalePermstruct_t *ScalePermstruct, gridinfo_t *grid,
     LUstruct_t *LUstruct, SuperLUStat_t *stat, int *numProcSymbFact,
     int *info, double *totalMemory, double *maxMemory );
 #endif
 #else
-pzsymbfact(superlu_dist_options_t *options, SuperMatrix *A, 
+pzsymbfact(superlu_dist_options_t *options, SuperMatrix *A,
     ScalePermstruct_t *ScalePermstruct, gridinfo_t *grid,
     LUstruct_t *LUstruct, SuperLUStat_t *stat, int *numProcSymbFact,
     int *info, double *totalMemory, double *maxMemory );
