diff --git a/src/superlu_dist_internal_complex.cpp b/src/superlu_dist_internal_complex.cpp
index 03c085a..1b40e4f 100644
--- a/src/superlu_dist_internal_complex.cpp
+++ b/src/superlu_dist_internal_complex.cpp
@@ -51,7 +51,7 @@ such enhancements or derivative works thereof, in binary and source code form.
 #include <superlu_zdefs.h>
 
 #include <numeric>
-#include <Cnames.h>
+#include <superlu_FCnames.h>
 extern "C"{ void
 #ifdef SUPERLU_DIST_MAJOR_VERSION
 #if SUPERLU_DIST_MAJOR_VERSION < 5
diff --git a/src/superlu_dist_internal_real.cpp b/src/superlu_dist_internal_real.cpp
index 5a6f69e..ee312e3 100644
--- a/src/superlu_dist_internal_real.cpp
+++ b/src/superlu_dist_internal_real.cpp
@@ -51,7 +51,7 @@ such enhancements or derivative works thereof, in binary and source code form.
 #include <superlu_ddefs.h>
 #include <numeric>
 
-#include <Cnames.h>
+#include <superlu_FCnames.h>
 extern "C"{ void
 #ifdef SUPERLU_DIST_MAJOR_VERSION
 #if SUPERLU_DIST_MAJOR_VERSION < 5
