From d3b625da189409455610a3450705a7b5bad4457a Mon Sep 17 00:00:00 2001
From: Rolando Espinoza <rndmax84@gmail.com>
Date: Sun, 5 Feb 2017 15:17:32 -0400
Subject: [PATCH 1/9] Remove wrongly placed break on cmake file.

---
 cmake/FindLZO.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/FindLZO.cmake b/cmake/FindLZO.cmake
index d501100..aace042 100644
--- a/cmake/FindLZO.cmake
+++ b/cmake/FindLZO.cmake
@@ -13,6 +13,5 @@ if (EXISTS ${LZO_INCLUDE_DIR})
   if (EXISTS ${LZO_LIBRARY})
     set (LZO_LIBRARIES ${LZO_LIBRARY})
     set (LZO_FOUND "LZO_FOUND")
-    break ()
   endif ()
 endif ()
-- 
2.6.2

