From 4522079095149c641cd1cd8c46144095bf1b517a Mon Sep 17 00:00:00 2001
From: Rolando Espinoza <rndmax84@gmail.com>
Date: Sun, 5 Feb 2017 15:18:05 -0400
Subject: [PATCH 2/9] Add directive to have _SC_PAGE_SIZE defined.

---
 squash/buffer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/squash/buffer.c b/squash/buffer.c
index f6843cc..999e2f4 100644
--- a/squash/buffer.c
+++ b/squash/buffer.c
@@ -23,6 +23,7 @@
  * Authors:
  *   Evan Nemerson <evan@nemerson.com>
  */
+#define _POSIX_C_SOURCE 200809L
 
 #include <squash/config.h>
 
-- 
2.6.2

