From 894231198548f3c02c330b80ab5b7a39b482eb5d Mon Sep 17 00:00:00 2001
From: Rolando Espinoza <rndmax84@gmail.com>
Date: Sun, 5 Feb 2017 15:18:42 -0400
Subject: [PATCH 3/9] Fix squash missing endian declarations in OSX.

---
 plugins/snappy/squash-snappy.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/plugins/snappy/squash-snappy.c b/plugins/snappy/squash-snappy.c
index 967658c..4d711b8 100644
--- a/plugins/snappy/squash-snappy.c
+++ b/plugins/snappy/squash-snappy.c
@@ -31,7 +31,14 @@
 #include <stdlib.h>
 #include <string.h>
 #include <strings.h>
-#include <endian.h>
+
+#if defined(__APPLE__)
+# include <machine/endian.h>
+# include <libkern/OSByteOrder.h>
+# define be32toh OSSwapBigToHostInt32
+#else
+# include <endian.h>
+#endif
 
 #include <squash/squash.h>
 #include <snappy-c.h>
-- 
2.6.2

