diff --git a/pyproject.toml b/pyproject.toml
index 03103d7..046d5a2 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,7 +32,7 @@ dependencies= [
     # Make sure this version does not fall behind the builder's tiledbsoma version.
     "tiledbsoma>=1.12.3,!=1.14.1,<1.15",
     "anndata",
-    "numpy>=1.23,<2.0",
+    "numpy>=1.23",
     "requests",
     "typing_extensions",
     "s3fs>=2021.06.1",
