diff --git a/config/defaults/config.LINUX_GFORTRAN.mk b/config/defaults/config.LINUX_GFORTRAN.mk
index 5601b72..26dfa78 100644
--- a/config/defaults/config.LINUX_GFORTRAN.mk
+++ b/config/defaults/config.LINUX_GFORTRAN.mk
@@ -2,8 +2,8 @@
 AR       = ar
 AR_FLAGS = -rvs
 
-FF90       = gfortran
+FF90       = ${FC}
 FF90_FLAGS = -O2 -fdefault-real-8 -fPIC
 
 F2PY = f2py
-F2PY_FF90 = gnu95
+F2PY_FF90 = ${FC}
