--- qwtpolarconfig.pri	2016-07-06 23:29:07.551745854 -0400
+++ _qwtpolarconfig.pri	2016-07-07 02:10:42.354257941 -0400
@@ -15,24 +15,22 @@
 QWT_POLAR_VER_PAT      = 1
 QWT_POLAR_VERSION      = $${QWT_POLAR_VER_MAJ}.$${QWT_POLAR_VER_MIN}.$${QWT_POLAR_VER_PAT}
 
-unix {
-    QWT_POLAR_INSTALL_PREFIX    = /usr/local/qwtpolar-$$QWT_POLAR_VERSION
-}
-
-win32 {
-    QWT_POLAR_INSTALL_PREFIX    = C:/QwtPolar-$$QWT_POLAR_VERSION
-}
+QWT_POLAR_INSTALL_PREFIX = $$(PREFIX)
 
 QWT_POLAR_INSTALL_DOCS      = $${QWT_POLAR_INSTALL_PREFIX}/doc
 QWT_POLAR_INSTALL_HEADERS   = $${QWT_POLAR_INSTALL_PREFIX}/include
 QWT_POLAR_INSTALL_LIBS      = $${QWT_POLAR_INSTALL_PREFIX}/lib
 
+# Allow QMake to find Qwt lib/include
+INCLUDEPATH += $$(PREFIX)/include
+LIBS += -lqwt
+
 ######################################################################
 # Designer plugin
 ######################################################################
 
-QWT_POLAR_INSTALL_PLUGINS   = $${QWT_POLAR_INSTALL_PREFIX}/plugins/designer
-# QWT_POLAR_INSTALL_PLUGINS   = $${QT_INSTALL_PREFIX}/plugins/designer
+# QWT_POLAR_INSTALL_PLUGINS   = $${QWT_POLAR_INSTALL_PREFIX}/plugins/designer
+QWT_POLAR_INSTALL_PLUGINS   = $$[QT_INSTALL_PREFIX]/plugins/designer
 
 ######################################################################
 # Build the static/shared libraries.
@@ -61,8 +59,8 @@
 # with every Qt upgrade. 
 ######################################################################
 
-QWT_POLAR_INSTALL_FEATURES  = $${QWT_POLAR_INSTALL_PREFIX}/features
-# QWT_INSTALL_FEATURES  = $${QT_INSTALL_PREFIX}/features
+# QWT_POLAR_INSTALL_FEATURES  = $${QWT_POLAR_INSTALL_PREFIX}/features
+QWT_POLAR_INSTALL_FEATURES  = $$[QT_INSTALL_PREFIX]/mkspecs/features
 
 ######################################################################
 # If you want to build the QwtPolar designer plugin,
@@ -77,7 +75,7 @@
 # Otherwise you have to build them from the examples directory.
 ######################################################################
 
-QWT_POLAR_CONFIG     += QwtPolarExamples
+# QWT_POLAR_CONFIG     += QwtPolarExamples
 
 ######################################################################
 # When Qt has been built as framework qmake wants 
