From ff46d31bdcbab54e1ac01816187f0537e1d79583 Mon Sep 17 00:00:00 2001
From: Asaf Kahlon <asafka7@gmail.com>
Date: Sat, 18 Jan 2020 10:09:06 +0200
Subject: [PATCH 1/1] install to /bin

Signed-off-by: Asaf Kahlon <asafka7@gmail.com>
---
 makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/makefile b/makefile
index 4241df2..cd59901 100644
--- a/makefile
+++ b/makefile
@@ -192,11 +192,11 @@ uftp_keymgt: $(UFTP_KEYMGT_OBJS)
 	$(CC) $(CPPFLAGS) $(OPTIONS) $(INCLUDE) $(CFLAGS) $(MTFLAGS) -c $<
 
 install: all
-	install -m 755 -d $(DESTDIR)/usr/bin
-	install -m 755 -d $(DESTDIR)/usr/share/man/man1
-	install -m 755  uftp $(DESTDIR)/usr/bin
-	install -m 755  uftpd $(DESTDIR)/usr/bin
-	install -m 755  uftpproxyd $(DESTDIR)/usr/bin
-	install -m 755  uftp_keymgt $(DESTDIR)/usr/bin
-	install -m 644  uftp.1 uftpd.1 uftpproxyd.1 uftp_keymgt.1 $(DESTDIR)/usr/share/man/man1
+	install -m 755 -d $(DESTDIR)/bin
+	install -m 755 -d $(DESTDIR)/share/man/man1
+	install -m 755  uftp $(DESTDIR)/bin
+	install -m 755  uftpd $(DESTDIR)/bin
+	install -m 755  uftpproxyd $(DESTDIR)/bin
+	install -m 755  uftp_keymgt $(DESTDIR)/bin
+	install -m 644  uftp.1 uftpd.1 uftpproxyd.1 uftp_keymgt.1 $(DESTDIR)/share/man/man1
 
-- 
2.20.1

