--- a/pycompilation/util.py
+++ b/pycompilation/util.py
@@ -309,7 +309,10 @@ def import_module_from_file(filename, only_if_newer_than=None):
     ImportError if any of the files specified in only_if_newer_than are newer
     than the file given by filename.
     """
-    import imp
+    try:
+        import imp
+    except ImportError:
+        import importlib as imp
     path, name = os.path.split(filename)
     name, ext = os.path.splitext(name)
     name = name.split('.')[0]
