diff -Naur orig/src/runtime_src/core/pcie/tools/xbflash.qspi/xqspips.cpp work/src/runtime_src/core/pcie/tools/xbflash.qspi/xqspips.cpp
--- orig/src/runtime_src/core/pcie/tools/xbflash.qspi/xqspips.cpp	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/pcie/tools/xbflash.qspi/xqspips.cpp	2022-07-01 17:52:52.712810766 -0600
@@ -15,6 +15,7 @@
  * under the License.
  */
 
+#include <array>
 #include <fstream>
 #include <iostream>
 #include <cassert>
diff -Naur orig/src/runtime_src/core/pcie/tools/xbflash.qspi/xspi.cpp work/src/runtime_src/core/pcie/tools/xbflash.qspi/xspi.cpp
--- orig/src/runtime_src/core/pcie/tools/xbflash.qspi/xspi.cpp	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/pcie/tools/xbflash.qspi/xspi.cpp	2022-07-01 17:52:06.202827103 -0600
@@ -14,6 +14,7 @@
  * under the License.
  */
 
+#include <array>
 #include <iostream>
 #include <string>
 #include <cassert>
diff -Naur orig/src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp work/src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp
--- orig/src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp	2022-07-02 14:18:31.324205367 -0600
+++ work/src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp	2022-07-02 13:51:47.089339393 -0600
@@ -15,6 +15,7 @@
  * under the License.
  */
 
+#include <array>
 #include <fstream>
 #include <iostream>
 #include <cassert>
