diff -Naur orig/src/CMake/icd.cmake work/src/CMake/icd.cmake
--- orig/src/CMake/icd.cmake	2022-04-14 13:29:02.000000000 -0600
+++ work/src/CMake/icd.cmake	2022-07-01 21:54:27.367830329 -0600
@@ -10,6 +10,6 @@
   ${ICD_FILE_NAME}
   )
 
-set(OCL_ICD_INSTALL_PREFIX "/etc/OpenCL/vendors")
+set(OCL_ICD_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/etc/OpenCL/vendors")
 
 install (FILES ${CMAKE_CURRENT_BINARY_DIR}/${ICD_FILE_NAME} DESTINATION ${OCL_ICD_INSTALL_PREFIX})
