diff -Naur orig/src/CMake/nativeLnx.cmake work/src/CMake/nativeLnx.cmake
--- orig/src/CMake/nativeLnx.cmake	2022-04-14 13:29:02.000000000 -0600
+++ work/src/CMake/nativeLnx.cmake	2022-07-02 00:27:33.775082257 -0600
@@ -59,19 +59,6 @@
   OUTPUT_STRIP_TRAILING_WHITESPACE
   )
 
-# Static linking creates and installs static tools and libraries. The
-# static libraries have system boost dependencies which must be
-# resolved in final target.  The tools (currently xbutil2 and xbmgmt2)
-# will be statically linked.  Enabled only for ubuntu.
-option(XRT_STATIC_BUILD "Enable static building of XRT" OFF)
-if ( (${CMAKE_VERSION} VERSION_GREATER "3.16.0")
-    AND (${XRT_NATIVE_BUILD} STREQUAL "yes")
-    AND (${LINUX_FLAVOR} MATCHES "^(Ubuntu)")
-    )
-  message("-- Enabling static artifacts of XRT")
-  set(XRT_STATIC_BUILD ON)
-endif()
-
 # --- Boost ---
 #set(Boost_DEBUG 1)
 
