diff -Naur orig/src/CMake/nativeLnx.cmake work/src/CMake/nativeLnx.cmake
--- orig/src/CMake/nativeLnx.cmake	2022-07-02 00:27:33.775082257 -0600
+++ work/src/CMake/nativeLnx.cmake	2022-07-02 12:06:01.861184713 -0600
@@ -101,13 +101,23 @@
 INCLUDE (FindCurses)
 find_package(Curses REQUIRED)
 
+find_program(PYTHON_EXECUTABLE python3)
+execute_process (COMMAND ${PYTHON_EXECUTABLE} -c "import site; sp=site.getsitepackages(); assert len(sp) == 1, f'site packages has len {len(sp)} != 1'; print(sp[0])"
+                 RESULT_VARIABLE _result
+                 OUTPUT_VARIABLE PYTHON_SITE_PACKAGES
+                 ERROR_QUIET
+                 OUTPUT_STRIP_TRAILING_WHITESPACE)
+if (_result)
+    message(FATAL_ERROR "non-zero exit when looking for site-packages. got ${PYTHON_SITE_PACKAGES}")
+endif()
+
 # --- XRT Variables ---
-set (XRT_INSTALL_DIR           "xrt")
+set (XRT_INSTALL_DIR           "${CMAKE_INSTALL_PREFIX}")
 set (XRT_INSTALL_BIN_DIR       "${XRT_INSTALL_DIR}/bin")
 set (XRT_INSTALL_UNWRAPPED_DIR "${XRT_INSTALL_BIN_DIR}/unwrapped")
 set (XRT_INSTALL_INCLUDE_DIR   "${XRT_INSTALL_DIR}/include")
 set (XRT_INSTALL_LIB_DIR       "${XRT_INSTALL_DIR}/lib${LIB_SUFFIX}")
-set (XRT_INSTALL_PYTHON_DIR    "${XRT_INSTALL_DIR}/python")
+set (XRT_INSTALL_PYTHON_DIR    "${PYTHON_SITE_PACKAGES}/pyxrt")
 set (XRT_VALIDATE_DIR          "${XRT_INSTALL_DIR}/test")
 set (XRT_NAMELINK_ONLY NAMELINK_ONLY)
 set (XRT_NAMELINK_SKIP NAMELINK_SKIP)
@@ -119,7 +129,7 @@
 # Note, that in order to disable RPATH insertion for a specific
 # target (say a static executable), use
 #  set_target_properties(<target> PROPERTIES INSTALL_RPATH "")
-SET(CMAKE_INSTALL_RPATH "$ORIGIN/../lib${LIB_SUFFFIX}:$ORIGIN/../..:$ORIGIN/../../lib${LIB_SUFFIX}")
+#SET(CMAKE_INSTALL_RPATH "$ORIGIN/../lib${LIB_SUFFFIX}:$ORIGIN/../..:$ORIGIN/../../lib${LIB_SUFFIX}")
 
 # --- Release: eula ---
 file(GLOB XRT_EULA
@@ -153,7 +163,7 @@
 #XMA settings END
 
 # --- Python bindings ---
-set(PY_INSTALL_DIR "${XRT_INSTALL_DIR}/python")
+set(PY_INSTALL_DIR "${XRT_INSTALL_PYTHON_DIR}")
 add_subdirectory(python)
 
 # --- Python tests ---
@@ -167,7 +177,6 @@
   PERMISSIONS OWNER_READ OWNER_EXECUTE OWNER_WRITE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
   DESTINATION ${XRT_INSTALL_DIR}/test)
 
-add_subdirectory("../tests/validate" "${CMAKE_CURRENT_BINARY_DIR}/validate_build")
 message("-- XRT version: ${XRT_VERSION_STRING}")
 
 # -- CPack
@@ -175,16 +184,11 @@
 
 set (XRT_DKMS_DRIVER_SRC_BASE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/runtime_src/core")
 
-include (CMake/dkms.cmake)
-include (CMake/dkms-aws.cmake)
-include (CMake/dkms-azure.cmake)
-include (CMake/dkms-container.cmake)
 
 # --- ICD ---
 include (CMake/icd.cmake)
 
 # --- Change Log ---
-include (CMake/changelog.cmake)
 
 # --- Package Config ---
 include (CMake/pkgconfig.cmake)
diff -Naur orig/src/runtime_src/core/tools/xbtop/CMakeLists.txt work/src/runtime_src/core/tools/xbtop/CMakeLists.txt
--- orig/src/runtime_src/core/tools/xbtop/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/tools/xbtop/CMakeLists.txt	2022-07-02 12:01:09.714365924 -0600
@@ -2,13 +2,13 @@
 # Copyright (C) 2022 Xilinx, Inc. All rights reserved.
 #
 
-message(${LINUX_FLAVOR} ${LINUX_VERSION})
+message("${LINUX_FLAVOR} ${LINUX_VERSION}")
 string(SUBSTRING "${LINUX_VERSION}" 0 1 major_version)
 set (PYTHON_3 TRUE)
 
-if (${LINUX_FLAVOR} STREQUAL "Amazon")
+if ("${LINUX_FLAVOR}" STREQUAL "Amazon")
   set (PYTHON_3 FALSE)
-elseif ((${LINUX_FLAVOR} STREQUAL "CentOS") AND (${major_version} STREQUAL "7"))
+elseif (("${LINUX_FLAVOR}" STREQUAL "CentOS") AND ("${major_version}" STREQUAL "7"))
   set (PYTHON_3 FALSE)
 endif ()
 
