diff -Naur orig/src/CMake/pkgconfig.cmake work/src/CMake/pkgconfig.cmake
--- orig/src/CMake/pkgconfig.cmake	2022-04-14 13:29:02.000000000 -0600
+++ work/src/CMake/pkgconfig.cmake	2022-07-02 12:10:34.139318480 -0600
@@ -3,13 +3,7 @@
 #
 message("-- Preparing XRT pkg-config")
 
-if (${LINUX_FLAVOR} MATCHES "^(Ubuntu)")
-  set(XRT_PKG_CONFIG_DIR "/usr/lib/pkgconfig")
-elseif (${LINUX_FLAVOR} MATCHES "^(RedHat|CentOS|Amazon|Fedora|SUSE)")
-  set(XRT_PKG_CONFIG_DIR "/usr/lib64/pkgconfig")
-else ()
-  set(XRT_PKG_CONFIG_DIR "/usr/share/pkgconfig")
-endif ()
+set(XRT_PKG_CONFIG_DIR "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
 
 configure_file (
   ${CMAKE_SOURCE_DIR}/CMake/config/xrt.pc.in
