diff -Naur orig/src/CMake/version.cmake work/src/CMake/version.cmake
--- orig/src/CMake/version.cmake	2022-04-14 13:29:02.000000000 -0600
+++ work/src/CMake/version.cmake	2022-07-02 12:16:16.107363071 -0600
@@ -63,8 +63,3 @@
 endif()
 
 # This is not required on MPSoC platform. To avoid yocto error, do NOT intall
-if ((${XRT_NATIVE_BUILD} STREQUAL "yes") AND (NOT WIN32))
-  # Copied over from dkms.cmake. TODO: cleanup
-  set (XRT_DKMS_INSTALL_DIR "/usr/src/xrt-${XRT_VERSION_STRING}")
-  install(FILES ${CMAKE_BINARY_DIR}/gen/version.h DESTINATION ${XRT_DKMS_INSTALL_DIR}/driver/include)
-endif()
