diff -Naur orig/src/CMakeLists.txt work/src/CMakeLists.txt
--- orig/src/CMakeLists.txt	2022-07-01 18:09:59.626973884 -0600
+++ work/src/CMakeLists.txt	2022-07-02 12:19:55.332982905 -0600
@@ -67,10 +67,6 @@
   set (CMAKE_BUILD_TYPE RelWithDebInfo)
 endif (NOT CMAKE_BUILD_TYPE)
 
-set (CMAKE_INSTALL_PREFIX "/opt/xilinx")
-if (${XRT_NATIVE_BUILD} STREQUAL "no")
-  set (CMAKE_INSTALL_PREFIX "/usr")
-endif()
 
 # Enable AIE even on x86.  This is POC, once
 # complete we will remove the need for these
