diff -Naur orig/src/include/1_2/CL/cl_ext.h work/src/include/1_2/CL/cl_ext.h
--- orig/src/include/1_2/CL/cl_ext.h	2022-04-14 13:29:02.000000000 -0600
+++ work/src/include/1_2/CL/cl_ext.h	1969-12-31 17:00:00.000000000 -0700
@@ -1,41 +0,0 @@
-/*******************************************************************************
- * Copyright (c) 2008-2015 The Khronos Group Inc.
- *
- * Permission is hereby granted, free of charge, to any person obtaining a
- * copy of this software and/or associated documentation files (the
- * "Materials"), to deal in the Materials without restriction, including
- * without limitation the rights to use, copy, modify, merge, publish,
- * distribute, sublicense, and/or sell copies of the Materials, and to
- * permit persons to whom the Materials are furnished to do so, subject to
- * the following conditions:
- *
- * The above copyright notice and this permission notice shall be included
- * in all copies or substantial portions of the Materials.
- *
- * MODIFICATIONS TO THIS FILE MAY MEAN IT NO LONGER ACCURATELY REFLECTS
- * KHRONOS STANDARDS. THE UNMODIFIED, NORMATIVE VERSIONS OF KHRONOS
- * SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
- *    https://www.khronos.org/registry/
- *
- * THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
- * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
- * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
- * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
- * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
- * MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
- ******************************************************************************/
-
-/*
- * Copyright (C) 2018, Xilinx Inc - All rights reserved
- */
-
-#ifndef __XOCL_CL_EXT_H
-#define __XOCL_CL_EXT_H
-
-// This isn't the coolest.  If an application truly needs cl_ext.h
-// and finds this one first, then the warning is bogus.
-//#warning Deprecated Xilinx cl_ext.h included, please replace with cl_ext_xilinx.h
-#include <CL/cl_ext_xilinx.h>
-
-#endif
diff -Naur orig/src/include/1_2/CL/cl_ext_xilinx.h work/src/include/1_2/CL/cl_ext_xilinx.h
--- orig/src/include/1_2/CL/cl_ext_xilinx.h	2022-04-14 13:29:02.000000000 -0600
+++ work/src/include/1_2/CL/cl_ext_xilinx.h	2022-07-02 12:26:25.556131211 -0600
@@ -42,7 +42,7 @@
 
 // Do *not* include cl_ext.h from this directory
 #ifndef _WIN32
-# include_next <CL/cl_ext.h>
+# include <CL/cl_ext.h>
 #else
 # pragma warning( push )
 # pragma warning( disable : 4201 )
diff -Naur orig/src/runtime_src/xocl/CMakeLists.txt work/src/runtime_src/xocl/CMakeLists.txt
--- orig/src/runtime_src/xocl/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/xocl/CMakeLists.txt	2022-07-02 12:27:51.046394143 -0600
@@ -115,7 +115,6 @@
 # Release OpenCL extension headers
 install (FILES
   ${XRT_SRC_DIR}/include/1_2/CL/cl_ext_xilinx.h
-  ${XRT_SRC_DIR}/include/1_2/CL/cl_ext.h
   ${XRT_SRC_DIR}/include/1_2/CL/cl2xrt.hpp
   DESTINATION ${XRT_INSTALL_INCLUDE_DIR}/CL
   COMPONENT ${XRT_DEV_COMPONENT}
diff -Naur orig/src/runtime_src/xocl/core/device.h work/src/runtime_src/xocl/core/device.h
--- orig/src/runtime_src/xocl/core/device.h	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/xocl/core/device.h	2022-07-02 12:29:50.516544331 -0600
@@ -26,6 +26,7 @@
 #include "core/common/unistd.h"
 #include "core/common/scope_guard.h"
 
+#include <CL/cl_ext_xilinx.h>
 #include <cassert>
 
 namespace xocl {
diff -Naur orig/src/runtime_src/xocl/core/memory.h work/src/runtime_src/xocl/core/memory.h
--- orig/src/runtime_src/xocl/core/memory.h	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/xocl/core/memory.h	2022-07-02 12:30:47.957424693 -0600
@@ -28,6 +28,7 @@
 #include "core/common/api/bo.h"
 #include "core/include/experimental/xrt_bo.h"
 
+#include <CL/cl_ext_xilinx.h>
 #include <map>
 
 #ifdef _WIN32
diff -Naur orig/tests/unit_test/002_bitonic_sort/utils.cpp work/tests/unit_test/002_bitonic_sort/utils.cpp
--- orig/tests/unit_test/002_bitonic_sort/utils.cpp	2022-04-14 13:29:02.000000000 -0600
+++ work/tests/unit_test/002_bitonic_sort/utils.cpp	2022-07-02 12:31:33.097260000 -0600
@@ -20,7 +20,7 @@
 #include <memory.h>
 //#include <windows.h>
 #include "CL/cl.h"
-#include "CL/cl_ext.h"
+#include "CL/cl_ext_xilinx.h"
 #include "utils.h"
 #include <assert.h>
 
diff -Naur orig/tests/unit_test/2kernelglobal_002_rw_4ddr_512/main.cpp work/tests/unit_test/2kernelglobal_002_rw_4ddr_512/main.cpp
--- orig/tests/unit_test/2kernelglobal_002_rw_4ddr_512/main.cpp	2022-04-14 13:29:02.000000000 -0600
+++ work/tests/unit_test/2kernelglobal_002_rw_4ddr_512/main.cpp	2022-07-02 12:32:09.234330439 -0600
@@ -22,7 +22,7 @@
 #include <cmath>
 #include <chrono>
 #include <CL/opencl.h>
-#include <CL/cl_ext.h>
+#include <CL/cl_ext_xilinx.h>
 
 //pointer access pipeline loop
 #define PIPELINELOOP                        2
diff -Naur orig/tests/unit_test/icdext/Stream.h work/tests/unit_test/icdext/Stream.h
--- orig/tests/unit_test/icdext/Stream.h	2022-04-14 13:29:02.000000000 -0600
+++ work/tests/unit_test/icdext/Stream.h	2022-07-02 12:32:41.940109373 -0600
@@ -18,7 +18,7 @@
  */
 
 #include <CL/cl2.hpp>
-#include <CL/cl_ext.h>
+#include <CL/cl_ext_xilinx.h>
 
 #include <iostream>
 #include <string>
