diff -Naur orig/src/runtime_src/core/pcie/tools/CMakeLists.txt work/src/runtime_src/core/pcie/tools/CMakeLists.txt
--- orig/src/runtime_src/core/pcie/tools/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/pcie/tools/CMakeLists.txt	2022-07-02 12:39:55.684967028 -0600
@@ -4,7 +4,6 @@
 if(NOT WIN32)
   add_subdirectory(xbutil)
   add_subdirectory(xbmgmt)
-  add_subdirectory(cloud-daemon)
   add_subdirectory(xbflash.qspi)
 endif(NOT WIN32)
 
