diff -Naur orig/src/runtime_src/core/pcie/tools/xbflash.qspi/CMakeLists.txt work/src/runtime_src/core/pcie/tools/xbflash.qspi/CMakeLists.txt
--- orig/src/runtime_src/core/pcie/tools/xbflash.qspi/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/pcie/tools/xbflash.qspi/CMakeLists.txt	2022-07-02 13:00:05.514557575 -0600
@@ -15,7 +15,7 @@
 set(XBFLASH_QSPI_SRC ${XBFLASH_QSPI_FILES})
 set(XBFLASH_NAME "xbflash.qspi")
 set(XBFLASH_NAME_NEW "xbflash")
-set(XBFLASH_INSTALL_DEST "/usr/local/bin")
+set(XBFLASH_INSTALL_DEST "${XRT_INSTALL_BIN_DIR}")
 set(XBFLASH_COMPONENT "xbflash")
 
 add_executable(${XBFLASH_NAME} ${XBFLASH_QSPI_SRC})
diff -Naur orig/src/runtime_src/core/tools/xbflash2/CMakeLists.txt work/src/runtime_src/core/tools/xbflash2/CMakeLists.txt
--- orig/src/runtime_src/core/tools/xbflash2/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/tools/xbflash2/CMakeLists.txt	2022-07-02 13:02:49.564455522 -0600
@@ -28,7 +28,7 @@
 
 set(XBFLASH_QSPI_SRC ${XBFLASH_BASE_FILES} ${XBFLASH_QSPI_FILES})
 set(XBFLASH_NAME "xbflash2")
-set(XBFLASH_INSTALL_DEST "/usr/local/bin")
+set(XBFLASH_INSTALL_DEST "${XRT_INSTALL_BIN_DIR}")
 set(XBFLASH_COMPONENT "xbflash2")
 
 add_executable(${XBFLASH_NAME} ${XBFLASH_QSPI_SRC})
