diff -Naur orig/src/runtime_src/tools/scripts/CMakeLists.txt work/src/runtime_src/tools/scripts/CMakeLists.txt
--- orig/src/runtime_src/tools/scripts/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/tools/scripts/CMakeLists.txt	2022-07-02 13:07:20.704429259 -0600
@@ -25,4 +25,3 @@
 
 install (PROGRAMS ${XRT_SCRIPTS} DESTINATION ${XRT_INSTALL_BIN_DIR})
 install (PROGRAMS ${XRT_LOADER_SCRIPTS} DESTINATION ${XRT_INSTALL_UNWRAPPED_DIR})
-install (FILES ${XRT_SETUP_SCRIPTS} DESTINATION ${XRT_INSTALL_DIR})
diff -Naur orig/src/runtime_src/tools/scripts/loader work/src/runtime_src/tools/scripts/loader
--- orig/src/runtime_src/tools/scripts/loader	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/tools/scripts/loader	2022-07-02 13:10:48.465035537 -0600
@@ -41,22 +41,6 @@
   exit 1
 fi
 
-# -- Find the setup script and configure environment
-XRT_SETUP_SCRIPT="`dirname \"$0\"`/../../setup.sh"
-
-if [ ! -f "$XRT_SETUP_SCRIPT" ]; then
-  echo "ERROR: Could not find XRT setup script."
-  echo "ERROR: ${XRT_SETUP_SCRIPT} does not exist"
-  exit 1
-fi
-
-. "${XRT_SETUP_SCRIPT}" > /dev/null
-
-# XRT uses system librairies whereas Vitis and Vivado ship there own.
-# when XRT tools are invoked from Vitis/Vivado the LD_LIBRARY_PATH will be set to use libraries shipped with Vitis/vivado
-# so we reset the LD_LIBRARY_PATH to make XRT tool select the correct libraries.
-LD_LIBRARY_PATH=$XILINX_XRT/lib:$(echo $LD_LIBRARY_PATH | tr ':' '\n' | grep -E -v "/(Vitis|Vivado).*/(lnx|lin)" | tr '\n' ':')
-
 # -- Execute the wrapped program
 "${XRT_PROG_UNWRAPPED}" "${XRT_LOADER_ARGS[@]}"
 
