diff -Naur orig/src/runtime_src/xdp/CMakeLists.txt work/src/runtime_src/xdp/CMakeLists.txt
--- orig/src/runtime_src/xdp/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/xdp/CMakeLists.txt	2022-07-02 13:14:43.148926023 -0600
@@ -449,7 +449,7 @@
 
 # Only install these files for PCIe device for now, which is .
 if (${XRT_NATIVE_BUILD} STREQUAL "yes")
-set (APPDEBUG_INSTALL_PREFIX "/opt/xilinx/xrt/share/appdebug")
+set (APPDEBUG_INSTALL_PREFIX "${XRT_INSTALL_DIR}/share/appdebug")
 install (FILES
   ${XRT_XDP_APPDEBUG_DIR}/appdebug.py
   ${XRT_XDP_APPDEBUG_DIR}/appdebugint.py
