diff -Naur orig/src/CMake/cpackLin.cmake work/src/CMake/cpackLin.cmake
--- orig/src/CMake/cpackLin.cmake	2022-04-14 13:29:02.000000000 -0600
+++ work/src/CMake/cpackLin.cmake	2022-07-02 19:15:48.082641756 -0600
@@ -73,6 +73,12 @@
       uuid-dev (>= 2.27.1)")
   endif()
 
+  if (NOT ${XRT_SHARED_COMPONENT} STREQUAL "xrt")
+     SET(CPACK_DEBIAN_XRT-SHARED_PACKAGE_NAME ${XRT_SHARED_COMPONENT})
+     SET(CPACK_DEBIAN_XRT-SHARED_PACKAGE_DEPENDS
+       "${XRT_DEV_COMPONENT} (>= ${XRT_VERSION_MAJOR}.${XRT_VERSION_MINOR}.${XRT_VERSION_PATCH})" )
+   endif()
+
   if (DEFINED CROSS_COMPILE)
     if (${aarch} STREQUAL "aarch64")
       SET(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "arm64")
@@ -128,6 +134,12 @@
     SET(CPACK_RPM_XRT-DEV_PACKAGE_REQUIRES "xrt >= ${XRT_VERSION_MAJOR}.${XRT_VERSION_MINOR}.${XRT_VERSION_PATCH}, ocl-icd-devel >= 2.2, libuuid-devel >= 2.23.2")
   endif()
 
+  if (NOT ${XRT_SHARED_COMPONENT} STREQUAL "xrt")
+     SET(CPACK_RPM_XRT-SHARED_PACKAGE_NAME ${XRT_SHARED_COMPONENT})
+     SET(CPACK_RPM_XRT-SHARED_PACKAGE_DEPENDS
+       "${XRT_DEV_COMPONENT} >= ${XRT_VERSION_MAJOR}.${XRT_VERSION_MINOR}.${XRT_VERSION_PATCH}" )
+   endif()
+
   if (DEFINED CROSS_COMPILE)
     if (${aarch} STREQUAL "aarch64")
       SET(CPACK_RPM_PACKAGE_ARCHITECTURE "aarch64")
diff -Naur orig/src/CMakeLists.txt work/src/CMakeLists.txt
--- orig/src/CMakeLists.txt	2022-07-02 13:32:00.650794623 -0600
+++ work/src/CMakeLists.txt	2022-07-02 19:17:53.226297984 -0600
@@ -87,7 +87,15 @@
 # If XRT_DEV_COMPONENT is same DEFAULT_COMPONENT then only that package
 # is created with both development and run-time content.
 #set (XRT_DEV_COMPONENT "xrt-dev")
-set (XRT_DEV_COMPONENT "xrt")
+set (XRT_DEV_COMPONENT "xrt" CACHE STRING "Set to 'xrt-dev' to split dev artifacts into separate component&package")
+
+# Enable static package by setting XRT_STATIC_COMPONENT to "xrt-static"
+# If XRT_STATIC_COMPONENT is same DEFAULT_COMPONENT then only that package
+# is created with both dynamic and static content
+# It should also be possible to set it to "xrt-dev" if you have XRT_DEV_COMPONENT
+# set to "xrt-dev" and you want to bundle the static librares into the development package.
+#set (XRT_STATIC_COMPONENT "xrt-static"
+set (XRT_STATIC_COMPONENT "xrt" CACHE STRING "Set to 'xrt-static' to split dev artifacts into separate component&package")
 
 set (XRT_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 
diff -Naur orig/src/runtime_src/core/common/CMakeLists.txt work/src/runtime_src/core/common/CMakeLists.txt
--- orig/src/runtime_src/core/common/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/common/CMakeLists.txt	2022-07-02 19:22:46.521270864 -0600
@@ -86,12 +86,18 @@
   RUNTIME DESTINATION ${XRT_INSTALL_BIN_DIR}
 )
 
-install(TARGETS xrt_coreutil xrt_coreutil_static
+install(TARGETS xrt_coreutil
   EXPORT xrt-dev-targets
   ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT}
   LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT} ${XRT_NAMELINK_ONLY}
 )
 
+install(TARGETS xrt_coreutil_static
+  EXPORT xrt-static-targets
+  ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT}
+  LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT} ${XRT_NAMELINK_ONLY}
+)
+
 # The scheduler object files are for auto config of scheduler. These
 # files reference xrt_core symbols, hence are excluded from
 # xrt_corecommon shared library and instead linked explicitly into
diff -Naur orig/src/runtime_src/core/pcie/emulation/cpu_em/CMakeLists.txt work/src/runtime_src/core/pcie/emulation/cpu_em/CMakeLists.txt
--- orig/src/runtime_src/core/pcie/emulation/cpu_em/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/pcie/emulation/cpu_em/CMakeLists.txt	2022-07-02 19:24:36.670649025 -0600
@@ -67,8 +67,14 @@
   RUNTIME DESTINATION ${XRT_INSTALL_BIN_DIR}
 )
 
-install (TARGETS xrt_swemu xrt_swemu_static
+install (TARGETS xrt_swemu
   EXPORT xrt-dev-targets
   ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT}
   LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT} ${XRT_NAMELINK_ONLY}
 )
+
+install (TARGETS xrt_swemu_static
+  EXPORT xrt-static-targets
+  ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT}
+  LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT} ${XRT_NAMELINK_ONLY}
+)
diff -Naur orig/src/runtime_src/core/pcie/emulation/hw_em/CMakeLists.txt work/src/runtime_src/core/pcie/emulation/hw_em/CMakeLists.txt
--- orig/src/runtime_src/core/pcie/emulation/hw_em/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/pcie/emulation/hw_em/CMakeLists.txt	2022-07-02 19:26:04.918162222 -0600
@@ -71,9 +71,14 @@
   RUNTIME DESTINATION ${XRT_INSTALL_BIN_DIR}
 )
 
-install (TARGETS xrt_hwemu xrt_hwemu_static
+install (TARGETS xrt_hwemu
   EXPORT xrt-dev-targets
   ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT}
   LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT} ${XRT_NAMELINK_ONLY}
 )
 
+install (TARGETS xrt_hwemu_static
+  EXPORT xrt-static-targets
+  ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT}
+  LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT} ${XRT_NAMELINK_ONLY}
+)
diff -Naur orig/src/runtime_src/core/pcie/linux/CMakeLists.txt work/src/runtime_src/core/pcie/linux/CMakeLists.txt
--- orig/src/runtime_src/core/pcie/linux/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/pcie/linux/CMakeLists.txt	2022-07-02 19:27:11.036791282 -0600
@@ -79,8 +79,14 @@
   RUNTIME DESTINATION ${XRT_INSTALL_BIN_DIR}
 )
 
-install(TARGETS xrt_core xrt_core_static
+install(TARGETS xrt_core
   EXPORT xrt-dev-targets
   ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT}
   LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT} ${XRT_NAMELINK_ONLY}
 )
+
+install(TARGETS xrt_core_static
+  EXPORT xrt-static-targets
+  ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT}
+  LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT} ${XRT_NAMELINK_ONLY}
+)
diff -Naur orig/src/runtime_src/core/pcie/windows/mcdm/CMakeLists.txt work/src/runtime_src/core/pcie/windows/mcdm/CMakeLists.txt
--- orig/src/runtime_src/core/pcie/windows/mcdm/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/core/pcie/windows/mcdm/CMakeLists.txt	2022-07-02 19:29:03.210341011 -0600
@@ -52,8 +52,14 @@
   RUNTIME DESTINATION ${XRT_INSTALL_BIN_DIR}
 )
 
-install(TARGETS xrt_core xrt_core_static
+install(TARGETS xrt_core
   EXPORT xrt-dev-targets
   ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT}
   LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT} ${XRT_NAMELINK_ONLY}
 )
+
+install(TARGETS xrt_core_static
+  EXPORT xrt-static-targets
+  ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT}
+  LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT} ${XRT_NAMELINK_ONLY}
+)
diff -Naur orig/src/runtime_src/xocl/CMakeLists.txt work/src/runtime_src/xocl/CMakeLists.txt
--- orig/src/runtime_src/xocl/CMakeLists.txt	2022-07-02 13:32:01.535869544 -0600
+++ work/src/runtime_src/xocl/CMakeLists.txt	2022-07-02 19:30:23.436170726 -0600
@@ -106,12 +106,18 @@
   RUNTIME DESTINATION ${XRT_INSTALL_BIN_DIR}
 )
 
-install(TARGETS xilinxopencl xilinxopencl_static
+install(TARGETS xilinxopencl
   EXPORT xrt-dev-targets
   ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT}
   LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT} ${XRT_NAMELINK_ONLY}
 )
 
+install(TARGETS xilinxopencl_static
+  EXPORT xrt-static-targets
+  ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT}
+  LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT} ${XRT_NAMELINK_ONLY}
+)
+
 # Release OpenCL extension headers
 install (FILES
   ${XRT_SRC_DIR}/include/1_2/CL/cl_ext_xilinx.h
diff -Naur orig/src/runtime_src/xrt/CMakeLists.txt work/src/runtime_src/xrt/CMakeLists.txt
--- orig/src/runtime_src/xrt/CMakeLists.txt	2022-04-14 13:29:02.000000000 -0600
+++ work/src/runtime_src/xrt/CMakeLists.txt	2022-07-02 19:33:45.721390809 -0600
@@ -68,8 +68,14 @@
   RUNTIME DESTINATION ${XRT_INSTALL_BIN_DIR}
 )
 
-install(TARGETS xrt++ xrt++_static
+install(TARGETS xrt++
   EXPORT xrt-dev-targets
   ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT}
   LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_DEV_COMPONENT} ${XRT_NAMELINK_ONLY}
 )
+
+install(TARGETS xrt++_static
+  EXPORT xrt-static-targets
+  ARCHIVE DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT}
+  LIBRARY DESTINATION ${XRT_INSTALL_LIB_DIR} COMPONENT ${XRT_STATIC_COMPONENT} ${XRT_NAMELINK_ONLY}
+)
