--- grass-7.4.1/configure	2017-08-11 03:02:18.000000000 -0600
+++ grass-7.4.1.mine/configure	2017-12-20 12:19:03.000000000 -0700
@@ -1522,8 +1522,8 @@
 	*-apple-darwin*)
 	    SHLIB_CFLAGS="-fno-common"
 	    SHLIB_SUFFIX=".dylib"
-	    SHLIB_LD="cc -dynamiclib -compatibility_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -current_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -install_name \${INST_DIR}/lib/lib\${LIB_NAME}\${SHLIB_SUFFIX}"
-	    LD_LIBRARY_PATH_VAR="DYLD_LIBRARY_PATH"
+	    SHLIB_LD="clang -dynamiclib -compatibility_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -current_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -install_name @rpath/lib\${LIB_NAME}\${SHLIB_SUFFIX}"
+	    LD_LIBRARY_PATH_VAR="LD_RUN_PATH"
 	    ;;
 	*-sun-solaris*)
 	    # Note: If _REENTRANT isn't defined, then Solaris
