--- grass-7.4.1/include/Make/Install.make	2017-08-24 13:58:38.000000000 -0600
+++ grass-7.4.1.mine/include/Make/Install.make	2017-08-24 13:59:17.000000000 -0600
@@ -118,7 +118,8 @@
 
 ifneq ($(findstring darwin,$(ARCH)),)
 	@# enable OSX Help Viewer
-	@/bin/ln -sfh "$(INST_DIR)/docs/html" /Library/Documentation/Help/GRASS-$(GRASS_VERSION_MAJOR).$(GRASS_VERSION_MINOR)
+	@echo "=== NOT INSTALLING OSX Help Viewer DOCS ==="
+	@# /bin/ln -sfh "$(INST_DIR)/docs/html" /Library/Documentation/Help/GRASS-$(GRASS_VERSION_MAJOR).$(GRASS_VERSION_MINOR)
 endif
 
 $(INST_DIR) $(UNIX_BIN):
