--- grass-7.4.1/lib/python/ctypes/loader.py	2016-05-04 07:23:07.000000000 -0600
+++ grass-7.4.1.mine/lib/python/ctypes/loader.py	2017-08-25 11:17:52.000000000 -0600
@@ -124,7 +124,9 @@
         dyld_fallback_library_path = _environ_path("DYLD_FALLBACK_LIBRARY_PATH")
         if not dyld_fallback_library_path:
             dyld_fallback_library_path = [os.path.expanduser('~/lib'),
-                                          '/usr/local/lib', '/usr/lib']
+                                          '/usr/local/lib', '/usr/lib',
+                                          os.path.join(sys.prefix, 'lib')]
+        dyld_fallback_library_path.extend(_environ_path('LD_RUN_PATH'))
 
         dirs = []
 
