--- include/Make/Platform.make.in.orig
+++ include/Make/Platform.make.in
@@ -37,7 +37,7 @@
 exec_prefix         = @exec_prefix@
 ARCH                = @host@
 UNIX_BIN            = @BINDIR@
-INST_DIR            = @INSTDIR@
+INST_DIR            = @exec_prefix@
 
 GRASS_HOME          = @GRASS_HOME@
 RUN_GISBASE         = @GISBASE@
@@ -56,7 +56,7 @@
 GRASS_LIBRARY_TYPE  = @GRASS_LIBRARY_TYPE@
 
 #static libs:
-STLIB_LD            = @STLIB_LD@
+STLIB_LD            = @SHLIB_LD@ -liconv ${LINK_FLAGS}
 STLIB_PREFIX        = lib
 STLIB_SUFFIX        = @STLIB_SUFFIX@
 
@@ -75,7 +75,7 @@
 CFLAGS              = @CFLAGS@
 CXXFLAGS            = @CXXFLAGS@
 INCLUDE_DIRS        = @INCLUDE_DIRS@
-LINK_FLAGS          = @LDFLAGS@
+LINK_FLAGS          = @LDFLAGS@ -Xlinker -rpath -Xlinker ${prefix}/lib -Xlinker -rpath -Xlinker ${RUN_GISBASE}
 
 DLLIB               = @DLLIB@
 XCFLAGS             = @X_CFLAGS@
@@ -212,7 +212,7 @@
 CAIRO_HAS_XRENDER_SURFACE = @CAIRO_HAS_XRENDER_SURFACE@
 
 #Python
-PYTHON              = python3
+PYTHON              = ${prefix}/bin/python3
 
 #regex
 REGEXINCPATH        = @REGEXINCPATH@
