--- grass-7.4.1/include/Make/Platform.make.in	2015-12-22 12:54:38.000000000 -0700
+++ grass-7.4.1.mine/include/Make/Platform.make.in	2017-08-28 11:31:04.632332530 -0600
@@ -37,7 +37,7 @@
 exec_prefix         = @exec_prefix@
 ARCH                = @host@
 UNIX_BIN            = @BINDIR@
-INST_DIR            = @INSTDIR@
+INST_DIR            = @exec_prefix@
 
 GRASS_HOME          = @GRASS_HOME@
 RUN_GISBASE         = @GISBASE@
@@ -62,7 +62,7 @@
 
 #shared libs
 SHLIB_PREFIX        = lib
-SHLIB_LD            = @SHLIB_LD@
+SHLIB_LD            = @SHLIB_LD@ -liconv ${LINK_FLAGS}
 SHLIB_LDFLAGS       = @SHLIB_LD_FLAGS@
 SHLIB_CFLAGS        = @SHLIB_CFLAGS@
 SHLIB_SUFFIX        = @SHLIB_SUFFIX@
@@ -75,7 +75,7 @@
 CFLAGS              = @CFLAGS@
 CXXFLAGS            = @CXXFLAGS@
 INCLUDE_DIRS        = @INCLUDE_DIRS@
-LINK_FLAGS          = @LDFLAGS@
+LINK_FLAGS          = @LDFLAGS@ -Xlinker -rpath -Xlinker ${prefix}/lib -Xlinker -rpath -Xlinker ${RUN_GISBASE}
 
 DLLIB               = @DLLIB@
 XCFLAGS             = @X_CFLAGS@
@@ -208,7 +208,7 @@
 CAIRO_HAS_XRENDER_SURFACE = @CAIRO_HAS_XRENDER_SURFACE@
 
 #Python
-PYTHON              = python
+PYTHON              = ${prefix}/bin/python
 
 #regex
 REGEXINCPATH        = @REGEXINCPATH@
