--- include/Make/Platform.make.in.orig
+++ include/Make/Platform.make.in
@@ -37,7 +37,7 @@
 exec_prefix         = @exec_prefix@
 ARCH                = @host@
 UNIX_BIN            = @BINDIR@
-INST_DIR            = @INSTDIR@
+INST_DIR            = @exec_prefix@
 
 GRASS_HOME          = @GRASS_HOME@
 RUN_GISBASE         = @GISBASE@
@@ -62,8 +62,8 @@
 
 #shared libs
 SHLIB_PREFIX        = lib
-SHLIB_LD            = @SHLIB_LD@
-SHLIB_LDX           = @SHLIB_LDX@
+SHLIB_LD            = @SHLIB_LD@ ${ICONVLIB}
+SHLIB_LDX           = @SHLIB_LDX@ ${ICONVLIB}
 SHLIB_LDFLAGS       = @SHLIB_LD_FLAGS@
 SHLIB_CFLAGS        = @SHLIB_CFLAGS@
 SHLIB_SUFFIX        = @SHLIB_SUFFIX@
@@ -212,7 +212,7 @@
 CAIRO_HAS_XRENDER_SURFACE = @CAIRO_HAS_XRENDER_SURFACE@
 
 #Python
-PYTHON              = python3
+PYTHON              = ${prefix}/bin/pythonw
 
 #regex
 REGEXINCPATH        = @REGEXINCPATH@
