--- include/Make/Rules.make.orig
+++ include/Make/Rules.make
@@ -46,8 +46,8 @@
 	GISRC=$(RUN_GISRC) \
 	GISBASE=$(RUN_GISBASE) \
 	PATH="$(ARCH_DISTDIR)/bin:$(GISBASE)/bin:$(GISBASE)/scripts:$$PATH" \
+	LD_LIBRARY_PATH="$(BIN):$(GISBASE)/bin:$(GISBASE)/scripts:$(ARCH_LIBDIR):$(BASE_LIBDIR)" \
 	PYTHONPATH="$(GRASS_PYTHONPATH)" \
-	$(LD_LIBRARY_PATH_VAR)="$(BIN):$(GISBASE)/bin:$(GISBASE)/scripts:$(ARCH_LIBDIR):$(BASE_LIBDIR):$($(LD_LIBRARY_PATH_VAR))" \
 	LC_ALL=C LANG=C LANGUAGE=C \
 	$(1)
 
