--- grass-7.4.1/include/Make/Shlib.make	2011-05-21 03:35:08.000000000 -0600
+++ grass-7.4.1.mine/include/Make/Shlib.make	2017-12-19 14:03:53.000000000 -0700
@@ -11,5 +11,6 @@
 ifndef MINGW
 	(cd $(ARCH_LIBDIR); ln -f -s $(notdir $@) $(patsubst %.$(GRASS_VERSION_NUMBER)$(SHLIB_SUFFIX),%$(SHLIB_SUFFIX),$@))
 endif
+	install_name_tool -change libgdal.20.dylib @rpath/libgdal.20.dylib $@
 
 shlib: $(SHLIB)
