Index: mfront/src/MohrCoulombStressCriterion.cxx
===================================================================
--- mfront/src/MohrCoulombStressCriterion.cxx	(révision 2460)
+++ mfront/src/MohrCoulombStressCriterion.cxx	(copie de travail)
@@ -159,7 +159,7 @@
       if (r == FLOWCRITERION) {
 #if __cplusplus >= 201703L
         code += "const auto [seqf" + id + ",dseq" + id + "_ds" + id + "] = ";
-        code += "computeMohrCoulombStressCriterionNormal(this->" + cn + ", s" +
+        code += "computeMohrCoulombStressCriterionNormal(this->" + pn + ", s" +
                 id + ");\n";
 #else  /* __cplusplus >= 201703L */
         code += "stress seqf" + id + ";\n";
@@ -227,4 +227,4 @@
 
   }  // end of namespace bbrick
 
-}  // end of namespace mfront
\ No newline at end of file
+}  // end of namespace mfront
