diff -ur tfel-TFEL-3.3.0/bindings/python/mfront/CMakeLists.txt TFEL-3.3.0/bindings/python/mfront/CMakeLists.txt
--- tfel-TFEL-3.3.0/bindings/python/mfront/CMakeLists.txt	2019-12-16 15:42:07.000000000 +0100
+++ TFEL-3.3.0/bindings/python/mfront/CMakeLists.txt	2020-05-01 11:03:16.541182572 +0200
@@ -1,22 +1,6 @@
 add_subdirectory(umatxx)
 add_subdirectory(mtest)
 
-include_directories("${PROJECT_SOURCE_DIR}/bindings/python/include")
-include_directories("${PROJECT_SOURCE_DIR}/mfront/include")
-include_directories("${PROJECT_SOURCE_DIR}/include")
-include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
-include_directories(SYSTEM "${PYTHON_INCLUDE_DIRS}")
-
-link_directories("${Boost_LIBRARY_DIRS}")
-link_directories("${PYTHON_LIBRARY_PATH}")
-link_directories("${PROJECT_BINARY_DIR}/src/Exception")
-link_directories("${PROJECT_BINARY_DIR}/src/Utilities")
-link_directories("${PROJECT_BINARY_DIR}/src/Math")
-link_directories("${PROJECT_BINARY_DIR}/src/System")
-link_directories("${PROJECT_BINARY_DIR}/src/Glossary")
-link_directories("${PROJECT_BINARY_DIR}/src/Tests")
-link_directories("${PROJECT_BINARY_DIR}/mfront/src")
-
 mfront_python_module(_mfront mfront.cxx
   AbstractDSL.cxx
   FileDescription.cxx
@@ -35,10 +19,13 @@
   GeneratorOptions.cxx
   MakefileGenerator.cxx
   CMakeGenerator.cxx)
-target_link_libraries(py_mfront__mfront TFELMFront
+target_link_libraries(py_mfront__mfront
+  PRIVATE
+  TFELMFront
   MFrontLogStream TFELMaterial
   TFELMathParser  TFELSystem
   TFELUtilities   TFELGlossary
-  TFELException   TFELConfig)
+  TFELException   TFELConfig
+  ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
 tfel_python_script(mfront __init__.py)
diff -ur tfel-TFEL-3.3.0/bindings/python/mtest/CMakeLists.txt TFEL-3.3.0/bindings/python/mtest/CMakeLists.txt
--- tfel-TFEL-3.3.0/bindings/python/mtest/CMakeLists.txt	2019-12-16 15:42:07.000000000 +0100
+++ TFEL-3.3.0/bindings/python/mtest/CMakeLists.txt	2020-05-01 11:03:48.841198868 +0200
@@ -1,20 +1,3 @@
-include_directories("${PROJECT_SOURCE_DIR}/bindings/python/include")
-include_directories("${PROJECT_SOURCE_DIR}/include")
-include_directories("${PROJECT_SOURCE_DIR}/mfront/include")
-include_directories("${PROJECT_SOURCE_DIR}/mtest/include")
-include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
-include_directories(SYSTEM "${PYTHON_INCLUDE_DIRS}")
-
-link_directories("${PROJECT_BINARY_DIR}/mtest/src")
-link_directories("${PROJECT_BINARY_DIR}/src/Exception")
-link_directories("${PROJECT_BINARY_DIR}/src/Utilities")
-link_directories("${PROJECT_BINARY_DIR}/src/Material")
-link_directories("${PROJECT_BINARY_DIR}/src/Math")
-link_directories("${PROJECT_BINARY_DIR}/src/System")
-link_directories("${PROJECT_BINARY_DIR}/src/Tests")
-link_directories("${Boost_LIBRARY_DIRS}")
-link_directories("${PYTHON_LIBRARY_PATH}")
-
 mtest_python_module(_mtest
   pymtest.cxx
   Behaviour.cxx
@@ -32,10 +15,10 @@
   MFrontLogStream.cxx)
 
 target_link_libraries(py_mtest__mtest
-  TFELMTest
-  TFELMaterial
-  TFELMath
-  TFELUtilities
-  TFELException)
+  PRIVATE
+  TFELMTest TFELMaterial
+  TFELMath  TFELUtilities
+  TFELException
+  ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
 tfel_python_script(mtest __init__.py)
diff -ur tfel-TFEL-3.3.0/bindings/python/std/CMakeLists.txt TFEL-3.3.0/bindings/python/std/CMakeLists.txt
--- tfel-TFEL-3.3.0/bindings/python/std/CMakeLists.txt	2019-12-16 15:42:07.000000000 +0100
+++ TFEL-3.3.0/bindings/python/std/CMakeLists.txt	2020-05-01 11:01:10.705119054 +0200
@@ -1,14 +1,14 @@
-include_directories("${PROJECT_SOURCE_DIR}/bindings/python/include")
-include_directories("${PROJECT_SOURCE_DIR}/include")
-include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
-include_directories(SYSTEM "${PYTHON_INCLUDE_DIRS}")
-
-link_directories("${Boost_LIBRARY_DIRS}")
-link_directories("${PYTHON_LIBRARY_PATH}")
-
 std_python_module(_stl_vector vector.cxx)
+target_link_libraries(py_std__stl_vector
+  PRIVATE ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 std_python_module(_stl_map    map.cxx)
+target_link_libraries(py_std__stl_map
+  PRIVATE ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 std_python_module(_stl_pair   pair.cxx)
+target_link_libraries(py_std__stl_pair
+  PRIVATE ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 std_python_module(_stl_tuple  tuple.cxx)
+target_link_libraries(py_std__stl_tuple
+  PRIVATE ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
 tfel_python_script(std __init__.py)
diff -ur tfel-TFEL-3.3.0/bindings/python/tfel/CMakeLists.txt TFEL-3.3.0/bindings/python/tfel/CMakeLists.txt
--- tfel-TFEL-3.3.0/bindings/python/tfel/CMakeLists.txt	2019-12-16 15:42:07.000000000 +0100
+++ TFEL-3.3.0/bindings/python/tfel/CMakeLists.txt	2020-05-01 14:25:21.806921750 +0200
@@ -14,32 +14,41 @@
 
 tfel_python_module(_tfel tfel.cxx
   PhysicalConstants.cxx)
+target_link_libraries(py_tfel__tfel
+  PRIVATE ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
 tfel_python_module(tests tests.cxx
   TestResult.cxx
   XMLTestOutput.cxx)
-target_link_libraries(py_tfel_tests TFELTests)
+target_link_libraries(py_tfel_tests
+  PRIVATE TFELTests ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
 if(NOT WIN32)
 tfel_python_module(utilities utilities.cxx
   Data.cxx
   TextData.cxx)
-target_link_libraries(py_tfel_utilities TFELUtilities)
+target_link_libraries(py_tfel_utilities
+  PRIVATE TFELUtilities ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 endif(NOT WIN32)
 
 tfel_python_module(glossary
   pyglossary.cxx
   Glossary.cxx
   GlossaryEntry.cxx)
-target_link_libraries(py_tfel_glossary TFELGlossary)
+target_link_libraries(py_tfel_glossary
+  PRIVATE TFELGlossary ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
-tfel_python_module(math math.cxx
+tfel_python_module(math
+  math.cxx
   vector.cxx
   tvector.cxx
   stensor.cxx
   st2tost2.cxx
   Evaluator.cxx)
-target_link_libraries(py_tfel_math TFELMath TFELMathParser TFELException)
+target_link_libraries(py_tfel_math
+   PRIVATE
+   TFELMath TFELMathParser TFELException
+   ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
 tfel_python_module(material
   material.cxx
@@ -49,7 +58,10 @@
   PiPlane.cxx
   Barlat.cxx
   Hosford.cxx)
-target_link_libraries(py_tfel_material TFELMaterial TFELMath TFELException)
+target_link_libraries(py_tfel_material
+  PRIVATE
+  TFELMaterial TFELMath TFELException
+  ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
 if(NOT WIN32)
   tfel_python_module(system system.cxx
@@ -65,6 +77,8 @@
     ExternalBehaviourDescription.cxx)
 endif(NOT WIN32)
 target_link_libraries(py_tfel_system
-  TFELSystem)
+  PRIVATE
+  TFELSystem
+  ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
 
 tfel_python_script(tfel __init__.py)
diff -ur tfel-TFEL-3.3.0/cmake/modules/tfel.cmake TFEL-3.3.0/cmake/modules/tfel.cmake
--- tfel-TFEL-3.3.0/cmake/modules/tfel.cmake	2019-12-16 15:42:07.000000000 +0100
+++ TFEL-3.3.0/cmake/modules/tfel.cmake	2020-05-01 11:00:10.113088471 +0200
@@ -405,13 +405,19 @@
   endif(WIN32)
   set_target_properties(py_${fullname} PROPERTIES PREFIX "")
   set_target_properties(py_${fullname} PROPERTIES OUTPUT_NAME ${name})
-  if(TFEL_USES_CONAN)
-    target_link_libraries(py_${fullname}
-      ${CONAN_LIBS} ${PYTHON_LIBRARIES})
-  else(TFEL_USES_CONAN)
-    target_link_libraries(py_${fullname}
-      ${Boost_PYTHON_LIBRARY} ${PYTHON_LIBRARIES})
-  endif(TFEL_USES_CONAN)
+  target_include_directories(py_${fullname}
+    PRIVATE "${PROJECT_SOURCE_DIR}/bindings/python/include"
+    PRIVATE "${PROJECT_SOURCE_DIR}/include")
+  target_include_directories(py_${fullname}
+    SYSTEM
+    PRIVATE "${Boost_INCLUDE_DIRS}"
+    PRIVATE "${PYTHON_INCLUDE_DIRS}")
+  if(python-static-interpreter-workaround)
+    if(APPLE)
+      target_link_options(py_${fullname}
+        PRIVATE "-undefined" "dynamic_lookup")
+    endif(APPLE)
+  endif(python-static-interpreter-workaround)
 endmacro(python_module_base)
 
 # macro(python_module name)
@@ -427,7 +433,7 @@
 #     endif(TFEL_APPEND_SUFFIX)
 # endmacro(python_module)
 
-macro(python_lib_module name package)
+function(python_lib_module name package)
   python_module_base(${package}_${name} ${name} ${ARGN})
   if(TFEL_APPEND_SUFFIX)
     if(WIN32)
@@ -450,25 +456,32 @@
         COMPONENT python_bindings)
     endif(WIN32)
   endif(TFEL_APPEND_SUFFIX)
-endmacro(python_lib_module)
+endfunction(python_lib_module)
 
-macro(std_python_module name)
+function(std_python_module name)
   python_lib_module(${name} std ${ARGN})
-endmacro(std_python_module)
+endfunction(std_python_module)
 
-macro(tfel_python_module name)
+function(tfel_python_module name)
   python_lib_module(${name} tfel ${ARGN})
-endmacro(tfel_python_module)
+endfunction(tfel_python_module)
 
-macro(mfront_python_module name)
+function(mfront_python_module name)
   python_lib_module(${name} mfront ${ARGN})
-endmacro(mfront_python_module)
+  set(fullname "mfront_${name}")
+  target_include_directories(py_${fullname}
+    PRIVATE "${PROJECT_SOURCE_DIR}/mfront/include")
+endfunction(mfront_python_module)
 
-macro(mtest_python_module name)
+function(mtest_python_module name)
   python_lib_module(${name} mtest ${ARGN})
-endmacro(mtest_python_module)
+  set(fullname "mtest_${name}")
+  target_include_directories(py_${fullname}
+    PRIVATE "${PROJECT_SOURCE_DIR}/mtest/include"
+    PRIVATE "${PROJECT_SOURCE_DIR}/mfront/include")
+endfunction(mtest_python_module)
 
-macro(tfel_python_script_base dir)
+function(tfel_python_script_base dir)
   if(${ARGC} LESS 1)
     message(FATAL_ERROR "tfel_python_script_base : no script specified")
   endif(${ARGC} LESS 1)
@@ -500,12 +513,12 @@
         COMPONENT python_bindings)
     endif(WIN32)
   endforeach(pyscript ${ARGN})
-endmacro(tfel_python_script_base)
+endfunction(tfel_python_script_base)
 
-macro(tfel_python_script dir)
+function(tfel_python_script dir)
   if(TFEL_APPEND_SUFFIX)
     tfel_python_script_base(${dir}_${TFEL_SUFFIX_FOR_PYTHON_MODULES} ${ARGN})
   else(TFEL_APPEND_SUFFIX)
     tfel_python_script_base(${dir} ${ARGN})
   endif(TFEL_APPEND_SUFFIX)
-endmacro(tfel_python_script)
+endfunction(tfel_python_script)
diff -ur tfel-TFEL-3.3.0/CMakeLists.txt TFEL-3.3.0/CMakeLists.txt
--- tfel-TFEL-3.3.0/CMakeLists.txt	2019-12-16 15:42:07.000000000 +0100
+++ TFEL-3.3.0/CMakeLists.txt	2020-05-01 22:13:52.834219820 +0200
@@ -268,6 +268,14 @@
 option(enable-broken-boost-python-module-visibility-handling
   "workaround a bug in boost python which occurs on FreeBSD+clang. See https://github.com/boostorg/python/issues/173 for details" OFF)
 
+if( (DEFINED ENV{"CONDA_PREFIX"}) AND APPLE)
+  message("Enabling workaround for Issue 235")
+  set(python-static-interpreter-workaround ON)
+else()
+ option(python-static-interpreter-workaround
+   "a workaround to handle the fact the conda' python interpreter is statically linked, which causes issues on Mac Os. See for details https://github.com/ContinuumIO/anaconda-issues/issues/9078. A proper solution may only appear with CMake 3.15 with the `Python::module` target, see https://gitlab.kitware.com/cmake/cmake/issues/18100. Until this version is widely available, this flag proposes a hack. Use with care." OFF)
+endif()
+ 
 if(enable-python OR enable-python-bindings)
   find_package(PythonLibs REQUIRED)
   set(HAVE_PYTHON ON)
@@ -388,6 +396,33 @@
     message(FATAL_ERROR "Boost python libraries not found.\n"
       "Required packages are :" ${BOOST_MANDATORY_COMPONENTS})
   endif(Boost_PYTHON_FOUND)
+if(TFEL_USES_CONAN)
+    if(NOT python-static-interpreter-workaround)
+      set(TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES
+        ${CONAN_LIBS} ${PYTHON_LIBRARIES})
+    else(python-static-interpreter-workaround)
+      if(APPLE)
+        set(TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES
+          ${CONAN_LIBS})
+      else(APPLE)
+        set(TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES
+          ${CONAN_LIBS} ${PYTHON_LIBRARIES})
+      endif(APPLE)
+    endif()
+  else(TFEL_USES_CONAN)
+    if(NOT python-static-interpreter-workaround)
+      set(TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES
+        ${Boost_PYTHON_LIBRARY} ${PYTHON_LIBRARIES})
+    else()
+      if(APPLE)
+        set(TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES
+          ${Boost_PYTHON_LIBRARY})
+      else(APPLE)
+        set(TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES
+          ${Boost_PYTHON_LIBRARY} ${PYTHON_LIBRARIES})
+      endif(APPLE)
+    endif()
+  endif(TFEL_USES_CONAN)
   if(DEFINED TFEL_TMP_BUILD_SHARED_LIBS)
     unset(TFEL_TMP_BUILD_SHARED_LIBS)
     unset(BUILD_SHARED_LIBS)
Seulement dans TFEL-3.3.0: .git
