diff --git a/mfront/src/BehaviourProfiler.cxx b/mfront/src/BehaviourProfiler.cxx
index cb2b5633..f5e832ad 100644
--- a/mfront/src/BehaviourProfiler.cxx
+++ b/mfront/src/BehaviourProfiler.cxx
@@ -172,13 +172,13 @@ namespace mfront
 
   BehaviourProfiler::Timer::Timer(BehaviourProfiler& t, const unsigned short cn)
       : gtimer(t), c(cn) {
-#if !(defined _WIN32 || defined _WIN64)
+#if !(defined _WIN32 || defined _WIN64 || defined __APPLE__)
     ::clock_gettime(CLOCK_THREAD_CPUTIME_ID,&(this->start));
 #endif
   }  // end of BehaviourProfiler::Timer
 
   BehaviourProfiler::Timer::~Timer() {
-#if !(defined _WIN32 || defined _WIN64)
+#if !(defined _WIN32 || defined _WIN64 || defined __APPLE__)
     ::clock_gettime(CLOCK_THREAD_CPUTIME_ID,&(this->end));
     add_measure(this->gtimer.measures[this->c],this->start,this->end);
 #endif
