diff --git a/requirements.txt b/requirements.txt
new file mode 100644
index 0000000..81ea9bb
--- /dev/null
+++ b/requirements.txt
@@ -0,0 +1,2 @@
+rastervision_pipeline==0.13.1
+gdal==3.0.4
\ No newline at end of file
diff --git a/setup.py b/setup.py
index bc17402..0ac7f10 100644
--- a/setup.py
+++ b/setup.py
@@ -31,4 +31,10 @@ setup(
     'raster deep-learning ml computer-vision earth-observation geospatial geospatial-processing',
     packages=find_namespace_packages(exclude=['integration_tests*', 'tests*']),
     install_requires=install_requires,
-    zip_safe=False)
+    zip_safe=False,
+    entry_points={
+        'console_scripts': [
+            'rastervision=rastervision.gdal_vsi.vsi_file_system:main',
+        ],
+    }
+)
\ No newline at end of file
