diff --git config/darwin config/darwin
index 2c78cb2..0e45fa5 100644
--- config/darwin
+++ config/darwin
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /usr/bin/mpicxx 
+CXX      = ${MPICXX} 
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC} 
@@ -17,7 +17,7 @@ FC = ${F77}
 
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /usr/bin/mpicxx 
+LD =  ${MPICXX} 
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin32 config/darwin32
index 71a3676..14643f6 100644
--- config/darwin32
+++ config/darwin32
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /usr/bin/mpicxx -arch i386 -arch x86_64
+CXX      = ${MPICXX} -arch i386 -arch x86_64
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = /sw/bin/gfortran -march=i386 -march=x86-64
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /usr/bin/mpicc -arch i386 -arch x86_64
+LD =  ${MPICC} -arch i386 -arch x86_64
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin32-petsc-dev config/darwin32-petsc-dev
index c55c48f..dbfa349 100644
--- config/darwin32-petsc-dev
+++ config/darwin32-petsc-dev
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /opt/local/bin/openmpicxx
+CXX      = ${MPICXX}
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = /opt/local/bin/openmpif90
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /opt/local/bin/openmpicxx
+LD =  ${MPICXX}
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = -lgcc_s.10.5 -lSystem ${PETSC_LIB}
 
diff --git config/darwin32_openmpi config/darwin32_openmpi
index 8021e10..a77823b 100644
--- config/darwin32_openmpi
+++ config/darwin32_openmpi
@@ -2,10 +2,10 @@
 include ${PETSC_DIR}/bmake/common/variables
 ARCHIVE_SUFFIX = .a
 PETSC_FLAGS =  ${CFLAGS} ${CCPPFLAGS}
-CXX = /usr/local/bin/mpiCC
-CC  = /usr/local/bin/mpicc
-F   = /usr/local/bin/mpif77
-LD  = /usr/local/bin/mpiCC
+CXX = ${MPICXX}
+CC  = ${MPICC}
+F   = ${MPIF77}
+LD  = ${MPICC}
 BLAS= -framework Accelerate
 MPI_INCLUDE = -I/usr/local/mpich-1.2.5/ch_p4/include
 
diff --git config/darwin64 config/darwin64
index f011c01..dd76037 100644
--- config/darwin64
+++ config/darwin64
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /usr/bin/mpicxx -arch x86_64
+CXX      = ${MPICXX} -arch x86_64
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC} -arch x86_64
@@ -17,7 +17,7 @@ FC = ${F77} -m64
 
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /usr/bin/mpicxx -arch x86_64
+LD =  ${MPICXX} -arch x86_64
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin64-petsc-dev config/darwin64-petsc-dev
index e9e772b..44b2a87 100644
--- config/darwin64-petsc-dev
+++ config/darwin64-petsc-dev
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = mpicxx  -arch x86_64 
+CXX      = ${MPICXX}  -arch x86_64 
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS} 
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = /sw/bin/gfortran -march=x86-64
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  mpicxx -arch x86_64 
+LD =  ${MPICXX} -arch x86_64 
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin_i386 config/darwin_i386
index 682b528..7124313 100644
--- config/darwin_i386
+++ config/darwin_i386
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /usr/bin/mpicxx 
+CXX      = ${MPICXX} 
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = gfortran 
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /usr/bin/mpicc
+LD =  ${MPICC}
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin_i386_macports config/darwin_i386_macports
index 96116a6..144d626 100644
--- config/darwin_i386_macports
+++ config/darwin_i386_macports
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /opt/local/bin/openmpicxx
+CXX      = ${MPICXX}
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = /opt/local/bin/openmpif90
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /opt/local/bin/openmpicxx
+LD =  ${MPICXX}
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin_ppc config/darwin_ppc
index 96116a6..144d626 100644
--- config/darwin_ppc
+++ config/darwin_ppc
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /opt/local/bin/openmpicxx
+CXX      = ${MPICXX}
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = /opt/local/bin/openmpif90
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /opt/local/bin/openmpicxx
+LD =  ${MPICXX}
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin_ppc_macports config/darwin_ppc_macports
index 96116a6..144d626 100644
--- config/darwin_ppc_macports
+++ config/darwin_ppc_macports
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /opt/local/bin/openmpicxx
+CXX      = ${MPICXX}
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = /opt/local/bin/openmpif90
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /opt/local/bin/openmpicxx
+LD =  ${MPICXX}
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin_universal config/darwin_universal
index 21ade26..29b23db 100644
--- config/darwin_universal
+++ config/darwin_universal
@@ -7,7 +7,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = mpicxx -arch ppc -arch ppc64 -arch i386 -arch x86_64
+CXX      = ${MPICXX} -arch ppc -arch ppc64 -arch i386 -arch x86_64
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
diff --git config/darwin_x86 config/darwin_x86
index 71a3676..14643f6 100644
--- config/darwin_x86
+++ config/darwin_x86
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /usr/bin/mpicxx -arch i386 -arch x86_64
+CXX      = ${MPICXX} -arch i386 -arch x86_64
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = /sw/bin/gfortran -march=i386 -march=x86-64
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /usr/bin/mpicc -arch i386 -arch x86_64
+LD =  ${MPICC} -arch i386 -arch x86_64
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/darwin_x86_64 config/darwin_x86_64
index 6d70378..b1c9bcb 100644
--- config/darwin_x86_64
+++ config/darwin_x86_64
@@ -6,7 +6,7 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = /usr/bin/mpicxx -arch x86_64
+CXX      = ${MPICXX} -arch x86_64
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
@@ -15,7 +15,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = ${F77}
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  /usr/bin/mpicc -arch x86_64
+LD =  ${MPICC} -arch x86_64
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/linux32 config/linux32
index 0290201..44d22a9 100644
--- config/linux32
+++ config/linux32
@@ -6,16 +6,16 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = mpicxx -fPIC
+CXX      = ${MPICXX} -fPIC
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC} -fPIC
 CCFLAGS = ${PETSC_CCPPFLAGS}
 
-FC = mpif77
+FC = ${MPIF77}
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  mpicxx -fPIC
+LD =  ${MPICXX} -fPIC
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/linux64 config/linux64
index f363388..8ea6dfd 100644
--- config/linux64
+++ config/linux64
@@ -16,7 +16,7 @@ CCFLAGS = ${PETSC_CCPPFLAGS}
 FC = gfortran 
 FCFLAGS = ${PETSC_FCPPFLAGS} -m64 -g
 
-LD =  mpicxx -m64 -g
+LD =  ${MPICXX} -m64 -g
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/lonestar config/lonestar
index e452cd0..5753238 100644
--- config/lonestar
+++ config/lonestar
@@ -6,16 +6,16 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = mpicxx -cxx=g++ -fPIC
+CXX      = ${MPICXX} -cxx=g++ -fPIC
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS} 
 
 CC      = ${PCC}
 CCFLAGS = ${PETSC_CCPPFLAGS}
 
-FC = mpif77 -fPIC
+FC = ${MPIF77} -fPIC
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  mpicxx -fPIC
+LD =  ${MPICXX} -fPIC
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/petsc config/petsc
index 2eca772..292967b 100644
--- config/petsc
+++ config/petsc
@@ -6,16 +6,16 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = mpicxx
+CXX      = ${MPICXX}
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
 CCFLAGS = ${PETSC_CCPPFLAGS}
 
-FC = mpif77
+FC = ${MPIF77}
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  mpicxx
+LD =  ${MPICXX}
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git config/ranger config/ranger
index 69d6037..83efd81 100644
--- config/ranger
+++ config/ranger
@@ -6,16 +6,16 @@ ARCHIVE_SUFFIX = .a
 
 INCLUDES = -I${DAETK_DIR} -I${DAETK_DIR}/pete/pete-2.1.0/src -I${DAETK_DIR}/pdetk ${PETSC_INCLUDE}
 
-CXX      = mpicxx -fPIC
+CXX      = ${MPICXX} -fPIC
 CXXFLAGS = -DF77_POST_UNDERSCORE -DUSE_BLAS -DCMRVEC_BOUNDS_CHECK -DMV_VECTOR_BOUNDS_CHECK -DPETSCVEC_BOUNDS_CHECK ${PETSC_CCPPFLAGS}
 
 CC      = ${PCC}
 CCFLAGS = ${PETSC_CCPPFLAGS}
 
-FC = mpif77 -fPIC
+FC = ${MPIF77} -fPIC
 FCFLAGS = ${PETSC_FCPPFLAGS}
 
-LD =  mpicxx -fPIC
+LD =  ${MPICXX} -fPIC
 LDFLAGS = ${PCC_LINKER_FLAGS} 
 LDLIBS  = ${PETSC_LIB}
 
diff --git newDaspk/examples/MPI/Makefile newDaspk/examples/MPI/Makefile
index f36a930..552f8a1 100644
--- newDaspk/examples/MPI/Makefile
+++ newDaspk/examples/MPI/Makefile
@@ -20,7 +20,7 @@ OBJS = $(SOLVR)/ddaspk.o $(SOLVR)/daux.o $(SOLVR)/dlinpk.o \
 	$(SOLVR)/adf_dummy.o
 
 # MPI fortran compile comands
-MPIF = mpif77
+MPIF = ${MPIF77}
 
 #MPI compile options
 MPIFLAGS = -lmpi
