diff --git a/tests/test_ify.py b/tests/test_ify.py
index 8463596..e040b09 100644
--- a/tests/test_ify.py
+++ b/tests/test_ify.py
@@ -144,8 +144,8 @@ class HBReaderTestCase(unittest.TestCase):
             metadata = FileInfo()
             with hbopen(self.data_file, metadata) as f:
                 ...
-            self.assertEqual('hbreader/tests/data',  str(metadata.base_path))
-            self.assertEqual('hbreader/tests/data/test data 1.txt', str(metadata.source_file))
+            self.assertEqual('test_tmp/tests/data',  str(metadata.base_path))
+            self.assertEqual('test_tmp/tests/data/test data 1.txt', str(metadata.source_file))
         finally:
             FileInfo.rel_offset = None
         # Use a rel_offset that *doesn't* resolve to the current situation
@@ -158,10 +158,10 @@ class HBReaderTestCase(unittest.TestCase):
             self.assertEqual('tests/data/test data 1.txt', str(metadata.source_file))
         finally:
             FileInfo.rel_offset = None
-        self.assertNotEqual('hbreader/tests/data', str(metadata.base_path))
-        self.assertTrue(str(metadata.base_path).endswith('hbreader/tests/data'))
-        self.assertNotEqual('hbreader/tests/data/test data 1.txt', str(metadata.source_file))
-        self.assertTrue(str(metadata.source_file), 'hbreader/tests/data/test data 1.txt')
+        self.assertNotEqual('test_tmp/tests/data', str(metadata.base_path))
+        self.assertTrue(str(metadata.base_path).endswith('test_tmp/tests/data'))
+        self.assertNotEqual('test_tmp/tests/data/test data 1.txt', str(metadata.source_file))
+        self.assertTrue(str(metadata.source_file), 'test_tmp/tests/data/test data 1.txt')

     # TODO: Make sure the FileInfo is filled out the way we want it
     @unittest.skip("Test needs to be completed")
