--- setup.py	2022-07-06 15:24:04.000000000 -0400
+++ setup.py	2022-07-19 11:53:00.985587000 -0400
@@ -123,13 +123,15 @@
         'Z3_BUILD_DOCUMENTATION' : False,
         'Z3_BUILD_TEST_EXECUTABLES' : False,
         'Z3_BUILD_DOTNET_BINDINGS' : False,
-        'Z3_BUILD_JAVA_BINDINGS' : False
+        'Z3_BUILD_JAVA_BINDINGS' : False,
+        'CMAKE_CXX_COMPILER': 'vs2019',
     }
     # Convert cmake options to CLI arguments
     for key, val in cmake_options.items():
         if type(val) is bool:
             cmake_options[key] = str(val).upper()
     cmake_args = [ '-D' + key + '=' + value for key,value in cmake_options.items() ]
+    cmake_args = cmake_args + [ '-G' + ' ' + '"Visual Studio 16 2019"']
     args = [ 'cmake', *cmake_args, SRC_DIR ]
     if subprocess.call(args, env=build_env, cwd=BUILD_DIR) != 0:
         raise LibError("Unable to configure Z3.")
