--- setup.py	2022-07-06 15:24:04.000000000 -0400
+++ setup.py.new	2022-07-19 17:34:42.793030500 -0400
@@ -123,7 +123,8 @@
         'Z3_BUILD_DOCUMENTATION' : False,
         'Z3_BUILD_TEST_EXECUTABLES' : False,
         'Z3_BUILD_DOTNET_BINDINGS' : False,
-        'Z3_BUILD_JAVA_BINDINGS' : False
+        'Z3_BUILD_JAVA_BINDINGS' : False,
+        'CMAKE_CXX_COMPILER': 'vs2019',
     }
     # Convert cmake options to CLI arguments
     for key, val in cmake_options.items():
