--- .deps/Makefile	2016-12-29 18:04:09.411121864 +0000
+++ .deps/Makefile	2016-12-29 18:11:50.121471888 +0000
@@ -140,14 +140,14 @@
 # Target rules for targets named jemalloc
 
 # Build rule for target.
-jemalloc: cmake_check_build_system
-	$(MAKE) -f CMakeFiles/Makefile2 jemalloc
-.PHONY : jemalloc
-
-# fast build rule for target.
-jemalloc/fast:
-	$(MAKE) -f CMakeFiles/jemalloc.dir/build.make CMakeFiles/jemalloc.dir/build
-.PHONY : jemalloc/fast
+#jemalloc: cmake_check_build_system
+#	$(MAKE) -f CMakeFiles/Makefile2 jemalloc
+#.PHONY : jemalloc
+#
+# fast build rule for target.
+#jemalloc/fast:
+#	$(MAKE) -f CMakeFiles/jemalloc.dir/build.make CMakeFiles/jemalloc.dir/build
+#.PHONY : jemalloc/fast
 
 #=============================================================================
 # Target rules for targets named luv-static
@@ -309,14 +309,14 @@
 # Target rules for targets named libuv
 
 # Build rule for target.
-libuv: cmake_check_build_system
-	$(MAKE) -f CMakeFiles/Makefile2 libuv
-.PHONY : libuv
+#libuv: cmake_check_build_system
+#	$(MAKE) -f CMakeFiles/Makefile2 libuv
+#.PHONY : libuv
 
 # fast build rule for target.
-libuv/fast:
-	$(MAKE) -f CMakeFiles/libuv.dir/build.make CMakeFiles/libuv.dir/build
-.PHONY : libuv/fast
+#libuv/fast:
+#	$(MAKE) -f CMakeFiles/libuv.dir/build.make CMakeFiles/libuv.dir/build
+#.PHONY : libuv/fast
 
 #=============================================================================
 # Target rules for targets named msgpack
