--- .deps/build/src/libuv/autogen.sh	2015-12-14 19:54:30.000000000 +0000
+++ .deps/build/src/libuv/autogen.sh	2016-11-04 08:28:13.118519169 +0000
@@ -16,9 +16,9 @@
 
 cd `dirname "$0"`
 
-if [ "$LIBTOOLIZE" = "" ] && [ "`uname`" = "Darwin" ]; then
-  LIBTOOLIZE=glibtoolize
-fi
+#if [ "$LIBTOOLIZE" = "" ] && [ "`uname`" = "Darwin" ]; then
+#  LIBTOOLIZE=glibtoolize
+#fi
 
 ACLOCAL=${ACLOCAL:-aclocal}
 AUTOCONF=${AUTOCONF:-autoconf}
