diff --git a/install/build.py b/install/build.py
index 76606dc89..426f6aab6 100644
--- a/install/build.py
+++ b/install/build.py
@@ -164,7 +164,8 @@ def _get_compiler_base_options():
         proc = subprocess.Popen(
             nvcc_path + ['-o', test_out_path, test_cu_path],
             stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE)
+            stderr=subprocess.PIPE,
+            env=os.environ)
         stdoutdata, stderrdata = proc.communicate()
         stderrlines = stderrdata.split(b'\n')
         if proc.returncode != 0:
