--- setup.cfg	2019-03-07 20:27:28.000000000 -0600
+++ setup.cfg	2019-03-08 09:40:18.000000000 -0600
@@ -20,6 +20,7 @@

 [ah_bootstrap]
 auto_use = True
+auto_upgrade = False

 [flake8]
 ignore=
@@ -55,6 +56,3 @@
 url = https://baseband.readthedocs.io
 # Note: you will also need to change this in your package's __init__.py
 minimum_python_version = 2.7
-
-[entry_points]
-astropy-package-template-example = packagename.example_mod:main
