diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff1ff21..575593d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -180,11 +180,6 @@ endif()
 # we can include cmake files from this directory.
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
 
-find_package(LibXml2 2.5.3 QUIET)
-if (LIBXML2_FOUND)
-  set(CLANG_HAVE_LIBXML 1)
-endif()
-
 include(CheckIncludeFile)
 check_include_file(sys/resource.h CLANG_HAVE_RLIMITS)
 

