diff --git a/setup.py b/setup.py
index 800da34..014bef6 100644
--- a/setup.py
+++ b/setup.py
@@ -17,8 +17,7 @@
 setuptools.setup(
     name="dpdata",
     version_format='{tag}.dev{commitcount}+{gitsha}',
-    setup_requires=['setuptools-git-version'],
-    # version="0.0.2",
+    version="0.1.0",
     author="Han Wang",
     author_email="wang_han@iapcm.ac.cn",
     description="Manipulating DeePMD-kit, VASP and LAMMPS data formats",
