From 10326aa9f43928d0dc4132043bfb6a0659197486 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 12 May 2019 01:17:44 +0200
Subject: [PATCH 2/3] Fix gettext directory on Darwin

---
 cmake/modules/FindGLIB2.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/modules/FindGLIB2.cmake b/cmake/modules/FindGLIB2.cmake
index 4ddaddf..32c81d9 100644
--- a/cmake/modules/FindGLIB2.cmake
+++ b/cmake/modules/FindGLIB2.cmake
@@ -46,7 +46,7 @@ if (GLIB2_INCLUDE_DIRS)
 endif (GLIB2_INCLUDE_DIRS)
 
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    link_directories ("${CMAKE_INSTALL_PREFIX}/opt/gettext/lib")
+    link_directories ("${CMAKE_INSTALL_PREFIX}/lib/gettext")
 endif ()
 
 # -----------------------------------------------------
-- 
2.21.0

