From 1fc54662657df72cb6e82e66954cf3d815c57b8f Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 12 May 2019 01:18:16 +0200
Subject: [PATCH 3/3] Use statically linked Python

---
 src/CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 82f1563..a09d89e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -18,7 +18,7 @@ link_directories (
 add_library(gfal-py MODULE ${src_python})
 target_link_libraries(gfal-py
     ${Boost_LIBRARIES} ${GLIB2_LIBRARIES}  ${GTHREAD2_LIBRARIES}
-    ${GFAL2_LIBRARIES} ${PYTHON_LIBRARIES}
+    ${GFAL2_LIBRARIES}
 )
 set_target_properties(gfal-py
     PROPERTIES
@@ -30,6 +30,7 @@ set_target_properties(gfal-py
 set_property(TARGET gfal-py
     APPEND PROPERTY INCLUDE_DIRECTORIES ${PYTHON_INCLUDE_PATH}
 ) 
+set_property(TARGET gfal-py APPEND PROPERTY LINK_FLAGS "-undefined dynamic_lookup")
 
 install(TARGETS gfal-py 
 	    RUNTIME		DESTINATION ${CMAKE_INSTALL_PREFIX}
@@ -41,7 +42,7 @@ if (Boost_PYTHON_3_LIBRARY)
     add_library(gfal-py3 MODULE ${src_python})
     target_link_libraries(gfal-py3
         ${Boost_PYTHON_3_LIBRARY} ${GLIB2_LIBRARIES} ${GTHREAD_LIBRARIES}
-        ${GFAL2_LIBRARIES} ${PYTHON_LIBRARIES_3}
+        ${GFAL2_LIBRARIES}
     )
     set_target_properties(gfal-py3
         PROPERTIES
@@ -53,6 +54,7 @@ if (Boost_PYTHON_3_LIBRARY)
     set_property(TARGET gfal-py3
         APPEND PROPERTY INCLUDE_DIRECTORIES ${PYTHON_INCLUDE_PATH_3}
     )
+    set_property(TARGET gfal-py3 APPEND PROPERTY LINK_FLAGS "-undefined dynamic_lookup")
 
     install(TARGETS gfal-py3
         RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}
-- 
2.21.0

