From 0ae42ec7f4048de127a1524229c0e1c7365e0ffc Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 12 May 2019 02:27:05 +0200
Subject: [PATCH] Fix finding Boost for Python 2

---
 CMakeLists.txt     | 15 +++++++++++++++
 src/CMakeLists.txt |  2 +-
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c43d86a..2a6f8c5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,6 +21,21 @@ find_package(PythonEasy REQUIRED)
 
 message("Python: ${PYTHON_EXECUTABLE}")
 
+string(REPLACE "." "" Boost_PYTHON_SUFFIX "${PYTHON_CURRENT_VERSION}")
+
+find_library(Boost_PYTHON_LIBRARY
+    NAMES boost_python boost_python${Boost_PYTHON_SUFFIX} boost_python-py${Boost_PYTHON_SUFFIX}
+    HINTS
+        ${Boost_LIBRARYDIR}
+        "/usr/lib"
+        "/usr/lib64"
+)
+if (Boost_PYTHON_LIBRARY)
+    message("Boost Python library found! ${Boost_PYTHON_LIBRARY}")
+else()
+    message("Boost Python library not found")
+endif ()
+
 if (PYTHON_EXECUTABLE_3)
     message("Python 3 installed! ${PYTHON3_CURRENT_VERSION}")
     message("Python 3: ${PYTHON_EXECUTABLE_3}")
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 82f1563..34364d2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -17,7 +17,7 @@ link_directories (
 # Python 2
 add_library(gfal-py MODULE ${src_python})
 target_link_libraries(gfal-py
-    ${Boost_LIBRARIES} ${GLIB2_LIBRARIES}  ${GTHREAD2_LIBRARIES}
+    ${Boost_PYTHON_LIBRARY} ${GLIB2_LIBRARIES}  ${GTHREAD2_LIBRARIES}
     ${GFAL2_LIBRARIES} ${PYTHON_LIBRARIES}
 )
 set_target_properties(gfal-py
-- 
2.21.0

