diff --git a/setup.py b/setup.py
index 36ad4c7..04ece62 100644
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,7 @@
 import io
 from setuptools import setup
 
-GITHUB_URL = "https://github.com/cheshirekow/cmakelang"
+GITHUB_URL = "https://github.com/cheshirekow/cmake-format"
 
 VERSION = None
 with io.open("cmakelang/__init__.py", encoding="utf-8") as infile:
@@ -13,8 +13,8 @@ with io.open("cmakelang/__init__.py", encoding="utf-8") as infile:
 assert VERSION is not None
 
 
-with io.open("cmakelang/doc/README.rst", encoding="utf-8") as infile:
-  long_description = infile.read()
+VERSION = "v" + VERSION
+long_description = ""
 
 setup(
     name="cmakelang",
