From 9fa0d1e38227874ed53ede54e38827ac7d0cf433 Mon Sep 17 00:00:00 2001
From: Dylan Grafmyre <dylang@conversica.com>
Date: Thu, 5 Jan 2017 14:47:03 -0800
Subject: [PATCH] Update setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index c6c3fa5..c7abd4d 100644
--- a/setup.py
+++ b/setup.py
@@ -6,7 +6,7 @@
     author       = 'Wijnand Modderman-Lenstra',
     author_email = 'maze@pyth0n.org',
     description  = 'Python interface to the POSIX getent family of commands',
-    long_description = file('README.rst').read(),
+    long_description = open('README.rst').read(),
     license      = 'MIT',
     keywords     = 'getent group passwd shadow network alias host',
     packages     = ['getent'],
