From af5803f205cc87268b471987ef7dbf15bd72ac91 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 7 Jun 2026 19:39:36 +1100
Subject: [PATCH] don't install nanoarrow_testing

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 5c3032d..4f008f8 100644
--- a/meson.build
+++ b/meson.build
@@ -223,7 +223,7 @@ if needs_testing
         sources: ['src/nanoarrow/testing/testing.cc'],
         dependencies: [nanoarrow_dep, nlohmann_json_dep],
         include_directories: incdir,
-        install: true,
+        install: false,
         gnu_symbol_visibility: 'hidden',
     )
     pkg.generate(nanoarrow_testing_lib, filebase: 'nanoarrow-testing')
