diff --git a/setup.py b/setup.py
index 67778d4..873af24 100644
--- a/setup.py
+++ b/setup.py
@@ -25,7 +25,7 @@ setup(name='mapbox-vector-tile',
       author_email='hello@mapzen.com',
       url='https://github.com/tilezen/mapbox-vector-tile',
       license='MIT',
-      packages=find_packages(),
+      packages=find_packages(exclude=['tests']),
       include_package_data=True,
       zip_safe=False,
       test_suite="setup.test_suite",
