diff --git a/Directory.Build.props b/Directory.Build.props
index 3177de8..c1ef78f 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -109,12 +109,12 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <GFORTRAN_BIN>"C:\Program Files\mingw-w64\x86_64-8.1.0-posix-seh-rt_v6-rev0\mingw64\bin"</GFORTRAN_BIN>
+    <GFORTRAN_BIN>$(LIBRARY_BIN)</GFORTRAN_BIN>
   </PropertyGroup>
 
   <PropertyGroup>
     <OutputPath>$(StagingOutputRootPath)$(MSBuildProjectName)\</OutputPath>
-    <VCToolsVersion>14.15.26726</VCToolsVersion>
+    <VCToolsVersion>14.16.27023</VCToolsVersion>
     <WindowsTargetPlatformVersion>10.0.16299.0</WindowsTargetPlatformVersion>
     <PlatformToolset>v141</PlatformToolset>
     <OutDir>$(OutputPath)</OutDir>
