diff --git a/Directory.Build.props b/Directory.Build.props
index 3177de8..b653061 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -114,7 +114,7 @@
 
   <PropertyGroup>
     <OutputPath>$(StagingOutputRootPath)$(MSBuildProjectName)\</OutputPath>
-    <VCToolsVersion>14.15.26726</VCToolsVersion>
+    <VCToolsVersion>14.16.27023</VCToolsVersion>
     <WindowsTargetPlatformVersion>10.0.16299.0</WindowsTargetPlatformVersion>
     <PlatformToolset>v141</PlatformToolset>
     <OutDir>$(OutputPath)</OutDir>
