diff --git a/cmake/project_common.cmake b/cmake/project_common.cmake
index 15939a5..113145e 100644
--- a/cmake/project_common.cmake
+++ b/cmake/project_common.cmake
@@ -1,7 +1,7 @@
 message(STATUS "**** ${PROJECT_NAME} ****")
 
 set(PDCURSES_SRCDIR ${CMAKE_SOURCE_DIR})
-set(PDCURSES_DIST ${CMAKE_INSTALL_PREFIX}/${CMAKE_BUILD_TYPE})
+set(PDCURSES_DIST ${CMAKE_INSTALL_PREFIX})
 
 set(osdir ${PDCURSES_SRCDIR}/${PROJECT_NAME})
 set(demodir ${PDCURSES_SRCDIR}/demos)
