diff --git a/python_bindings/Makefile b/python_bindings/Makefile
index 71dddc09b..12cfa85bc 100644
--- a/python_bindings/Makefile
+++ b/python_bindings/Makefile
@@ -55,7 +55,7 @@ CCFLAGS := $(filter-out -Wstrict-prototypes,$(CCFLAGS))
 
 # DON'T link libpython* - leave those symbols to lazily resolve at load time
 # Cf. https://github.com/pybind/pybind11/blob/master/docs/compiling.rst#building-manually
-LDFLAGS=-lz
+LDFLAGS += -lz
 ifeq ($(UNAME), Darwin)
     # Keep OS X builds from complaining about missing libpython symbols
     LDFLAGS += -undefined dynamic_lookup
-- 
