--- scons\gallium.py	Thu Feb 02 14:30:08 2017
+++ scons\gallium.py_new	Thu Feb 02 14:30:34 2017
@@ -497,10 +497,10 @@
         # Choose the appropriate MSVC CRT
         # http://msdn.microsoft.com/en-us/library/2kzt1wy3.aspx
         if env['build'] in ('debug', 'checked'):
-            env.Append(CCFLAGS = ['/MTd'])
+            env.Append(CCFLAGS = ['/MDd'])
             env.Append(SHCCFLAGS = ['/LDd'])
         else:
-            env.Append(CCFLAGS = ['/MT'])
+            env.Append(CCFLAGS = ['/MD'])
             env.Append(SHCCFLAGS = ['/LD'])
     
     # Static code analysis
