--- src/mpeg1audio/__init__.py.orig	2021-04-19 23:05:01.712898311 -0500
+++ src/mpeg1audio/__init__.py	2021-04-19 23:07:15.211934511 -0500
@@ -72,7 +72,7 @@
 from datetime import timedelta
 from mpeg1audio import headers
 from mpeg1audio import utils
-from headers import MPEGAudioHeaderEOFException, MPEGAudioHeaderException
+from mpeg1audio.headers import MPEGAudioHeaderEOFException, MPEGAudioHeaderException
 import math
 import struct
 
