From baa06327b4b6126d2b5a07c63ed51f22f66e4838 Mon Sep 17 00:00:00 2001
From: Henrik Finsberg <henriknf@simula.no>
Date: Sat, 30 May 2026 11:39:12 +0200
Subject: [PATCH] Make pydicom and dcm2niix optional

---
 pyproject.toml | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index d6dfa23..3574c97 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -24,8 +24,6 @@ dependencies = [
   "pandas",
   "scipy",
   "scikit-image",
-  "pydicom",
-  "dcm2niix",
 ]
 
 [project.optional-dependencies]
@@ -40,8 +38,13 @@ napari = [
 test = [
     "pytest",
     "pytest-cov",
+    "mritk[dicom]",
 
 ]
+dicom = [
+    "pydicom", # Used for handling DICOM files
+    "dcm2niix", # Used for converting DICOM to NIfTI
+]
 pypi = [
     "build" # Used for building wheels and uploading to pypi
 ]
-- 
2.50.1 (Apple Git-155)

