diff --git a/clingwrapper/setup.py b/clingwrapper/setup.py
index ece5922..ce24b3c 100755
--- a/clingwrapper/setup.py
+++ b/clingwrapper/setup.py
@@ -29,13 +29,7 @@ with codecs.open(os.path.join(here, 'README.rst'), encoding='utf-8') as f:
 # platform-dependent helpers
 #
 def is_manylinux():
-    try:
-        for line in open('/etc/redhat-release').readlines():
-            if 'CentOS release 5.11' in line:
-                return True
-    except (OSError, IOError):
-        pass
-    return False
+    return True
 
 def _get_linker_options():
     if 'win32' in sys.platform:
