diff -Nur flux-sched-0.27.0/t/Makefile.in flux-sched-0.27.0.patched/t/Makefile.in
--- flux-sched-0.27.0/t/Makefile.in	2023-04-01 00:50:41.000000000 +0000
+++ flux-sched-0.27.0.patched/t/Makefile.in	2023-04-23 23:06:09.693246186 +0000
@@ -559,7 +559,6 @@
     t1007-recovery-full.t \
     t1009-recovery-multiqueue.t \
     t1010-sync-modules.t \
-    t1011-dynstate-change.t \
     t1012-find-status.t \
     t1013-exclusion.t \
     t1013-qmanager-priority.t \
