diff --git a/ccx_2.14/src/Makefile_MT b/ccx_2.14/src/Makefile_MT
index 9b8c363..2ccd1d2 100755
--- a/ccx_2.14/src/Makefile_MT
+++ b/ccx_2.14/src/Makefile_MT
@@ -1,5 +1,5 @@
 
-CFLAGS = -Wall -O3 -fopenmp -I ../../../SPOOLES.2.2 -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE -DUSE_MT=1 
+CFLAGS = -Wall -O3 -fopenmp -I $(PREFIX)/include/spooles -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE -DUSE_MT=1 
 FFLAGS = -Wall -O3 -fopenmp 
 
 CC=cc
@@ -18,12 +18,12 @@ OCCXF = $(SCCXF:.f=.o)
 OCCXC = $(SCCXC:.c=.o)
 OCCXMAIN = $(SCCXMAIN:.c=.o)
 
-DIR=../../../SPOOLES.2.2
 
 LIBS = \
-       $(DIR)/MT/src/spoolesMT.a \
-       $(DIR)/spooles.a \
-       ../../../ARPACK/libarpack_INTEL.a \
+       $(PREFIX)/lib/spoolesMT.a \
+       $(PREFIX)/lib/spooles.a \
+       $(PREFIX)/lib/libarpack.a \
+       $(PREFIX)/lib/liblapack.a \
        -lpthread -lm
 
 ccx_2.14_MT: $(OCCXMAIN) ccx_2.14_MT.a  $(LIBS)
diff --git a/ccx_2.14/src/date.pl b/ccx_2.14/src/date.pl
index 934d7f7..855cf87 100755
--- a/ccx_2.14/src/date.pl
+++ b/ccx_2.14/src/date.pl
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 chomp($date=`date`);
 
