diff --git a/ccx_2.14/src/ccx_2.14.c b/ccx_2.14/src/ccx_2.14.c
index e53d6f4..fd7303b 100644
--- a/ccx_2.14/src/ccx_2.14.c
+++ b/ccx_2.14/src/ccx_2.14.c
@@ -14,11 +14,11 @@
 /*     You should have received a copy of the GNU General Public License */
 /*     along with this program; if not, write to the Free Software       */
 /*     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.         */
-
+/*
 #ifdef __WIN32
   _set_output_format(_TWO_DIGIT_EXPONENT);
 #endif
-
+*/
 #ifdef CALCULIX_MPI
 #include <spoolesMPI.h>
 #endif
