diff --git a/CalculiX/ccx_2.14/src/u_free.c b/CalculiX/ccx_2.14/src/u_free.c
index 7aacc52..7a12fe0 100644
--- a/CalculiX/ccx_2.14/src/u_free.c
+++ b/CalculiX/ccx_2.14/src/u_free.c
@@ -41,5 +41,5 @@ void *u_free(void* ptr,const char *file,const int line, const char* ptr_name){
   if(log_realloc==1) {
       printf("FREEING of variable %s, file %s, line=%d: oldaddress= %ld\n",ptr_name,file,line,(long int)ptr);
   }      
-  return;
+  return NULL;
 }
