--- setup.py	2014-01-27 19:07:37.000000000 -0600
+++ setup.py.new	2018-04-02 14:08:59.359088004 -0500
@@ -15,10 +15,10 @@
 import sys
 import os
 from setuptools import setup
+from importlib import reload
 
 reload(sys)
-sys.setdefaultencoding('utf-8')
-
+#sys.setdefaultencoding('utf-8')
 
 files = ["pyscrap/*"]
 
--- pyscrap/item.py	2014-01-27 19:06:36.000000000 -0600
+++ pyscrap/item.py.new	2018-04-02 14:14:10.814624755 -0500
@@ -16,7 +16,7 @@
 #from types import MethodType
 
 reload(sys)
-sys.setdefaultencoding('utf-8')
+# sys.setdefaultencoding('utf-8')
 
 
 class FieldNotDefined(Exception):
--- pyscrap/pipeline.py	2014-01-27 19:06:36.000000000 -0600
+++ pyscrap/pipeline.py.new	2018-04-02 14:14:20.817706236 -0500
@@ -15,7 +15,7 @@
 import sys
 
 reload(sys)
-sys.setdefaultencoding("utf-8")
+# sys.setdefaultencoding("utf-8")
 
 
 def spiderDefaultSave(item):
--- pyscrap/settings.py	2014-01-27 19:06:36.000000000 -0600
+++ pyscrap/settings.py.new	2018-04-02 14:14:29.824779597 -0500
@@ -15,7 +15,7 @@
 import sys
 
 reload(sys)
-sys.setdefaultencoding('utf-8')
+# sys.setdefaultencoding('utf-8')
 
 
 def getHeaders():
--- pyscrap/spiders.py	2014-01-27 19:07:13.000000000 -0600
+++ pyscrap/spiders.py.new	2018-04-02 14:14:39.564858935 -0500
@@ -27,7 +27,7 @@
 
 
 reload(sys)
-sys.setdefaultencoding("utf-8")
+# sys.setdefaultencoding("utf-8")
 
 
 def getHtml(url):
--- pyscrap/templates.py	2014-01-27 19:06:36.000000000 -0600
+++ pyscrap/templates.py.new	2018-04-02 14:14:49.163937120 -0500
@@ -15,7 +15,7 @@
 import sys
 
 reload(sys)
-sys.setdefaultencoding('utf-8')
+# sys.setdefaultencoding('utf-8')
 
 
 def createSetup(name):
