From ecf914768b824db5d6b09aa8bfbb592237eca3c9 Mon Sep 17 00:00:00 2001
From: martinRenou <martin.renou@gmail.com>
Date: Mon, 12 Feb 2024 11:47:20 +0100
Subject: [PATCH 1/1] Missing header

---
 src/exprparse/exprparse.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/exprparse/exprparse.cpp b/src/exprparse/exprparse.cpp
index c78ebac..3a1d59b 100644
--- a/src/exprparse/exprparse.cpp
+++ b/src/exprparse/exprparse.cpp
@@ -25,6 +25,7 @@
 // SOFTWARE.
 
 #include "exprparse.h"
+#include <cstdint>
 #include <cstddef>
 #include <cstring>
 #include <list>
-- 
2.43.0

