From d35a258f077621d2ff55576476e9d6a8a676d5b8 Mon Sep 17 00:00:00 2001
From: gmatteo <gmatteo@gmail.com>
Date: Thu, 17 Oct 2019 19:58:40 +0200
Subject: [PATCH] drop support for py2, remove six, migrate to flowtk API

---
 pseudo_dojo/core/__init__.py                     |  4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pseudo_dojo/core/__init__.py b/pseudo_dojo/core/__init__.py
index 4cb1ed86..89ae5d26 100644
--- a/pseudo_dojo/core/__init__.py
+++ b/pseudo_dojo/core/__init__.py
@@ -1,5 +1,5 @@
 from .testing import *
 from .atom import *
 
-from pymatgen.io.abinit.tasks import TaskManager
-from pymatgen.io.abinit.pseudos import PseudoParser, PseudoTable
+from abipy.flowtk.tasks import TaskManager
+from abipy.flowtk.pseudos import PseudoParser, PseudoTable
