diff --git a/runtime/libpgmath/lib/CMakeLists.txt b/runtime/libpgmath/lib/CMakeLists.txt
index ea65005b..39fd7c48 100644
--- a/runtime/libpgmath/lib/CMakeLists.txt
+++ b/runtime/libpgmath/lib/CMakeLists.txt
@@ -91,13 +91,13 @@ if(${LIBPGMATH_SYSTEM_PROCESSOR} MATCHES "x86_64" AND NOT ${LIBPGMATH_WITH_GENER
     set(DEFINITIONS_L1
       TARGET_WIN_X8664 TARGET_WIN _PGI_NOBUILTINS MAXCPUS=256
       MAXCPUSL=8 MAXCPUSR=8 WINNT WIN64 LONG_IS_32
-      HOST_WIN TARGET_X8664 _NO_CRT_STDIO_INLINE)
+      HOST_WIN TARGET_X8664)
 #djp      HOST_WIN _DLL TARGET_X8664 _NO_CRT_STDIO_INLINE)
 
     set(DEFINITIONS_L2
       TARGET_WIN_X8664 TARGET_WIN _PGI_NOBUILTINS MAXCPUS=256
       MAXCPUSL=8 MAXCPUSR=8 WINNT WIN64 LONG_IS_32
-      HOST_WIN TARGET_X8664 _NO_CRT_STDIO_INLINE)
+      HOST_WIN TARGET_X8664)
 #djp      HOST_WIN _DLL TARGET_X8664 _NO_CRT_STDIO_INLINE)
 
     # common
diff --git a/runtime/libpgmath/lib/common/pgstdinit.h b/runtime/libpgmath/lib/common/pgstdinit.h
index d12f65b6..8b96ad50 100644
--- a/runtime/libpgmath/lib/common/pgstdinit.h
+++ b/runtime/libpgmath/lib/common/pgstdinit.h
@@ -227,7 +227,9 @@ char *fgets(char *, int, FILE *);
 FILE *fopen64(const char *, const char *);
 #endif
 FILE *fopen(const char *, const char *);
+#if !defined(TARGET_WIN)
 int fprintf(FILE *, const char *, ...);
+#endif
 #ifdef TARGET_WIN_X8664
 int __io_fputc(int, FILE *);
 #endif
