diff --git a/runtime/libpgmath/lib/common/pgstdinit.h b/runtime/libpgmath/lib/common/pgstdinit.h
index d12f65b6..977d0cca 100644
--- a/runtime/libpgmath/lib/common/pgstdinit.h
+++ b/runtime/libpgmath/lib/common/pgstdinit.h
@@ -227,7 +227,6 @@ char *fgets(char *, int, FILE *);
 FILE *fopen64(const char *, const char *);
 #endif
 FILE *fopen(const char *, const char *);
-int fprintf(FILE *, const char *, ...);
 #ifdef TARGET_WIN_X8664
 int __io_fputc(int, FILE *);
 #endif
@@ -319,7 +318,6 @@ void rewind();
 int scanf();
 void setbuf();
 int setvbuf();
-int sprintf();
 int sscanf();
 FILE *tmpfile();
 char *tmpnam();
