diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 17b7b41..31565e4 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -64,7 +64,7 @@ foreach(file ${files})
 		target_include_directories(${testcase} PRIVATE ${THIRDPARTY_INCLUDE_DIR})
 	endif()
 
-    if(CMAKE_CXX_COMPILER_ID STREQUAL GNU OR CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL GNU OR (CMAKE_CXX_COMPILER_ID STREQUAL Clang AND NOT APPLE))
         target_link_libraries(${testcase} PRIVATE stdc++fs)
     endif()
 
