diff --git a/poetry_pyproject.toml b/poetry_pyproject.toml
index 7830d91..92c249e 100644
--- a/poetry_pyproject.toml
+++ b/poetry_pyproject.toml
@@ -8,7 +8,7 @@ packages = [{include = "skelly_tracker"}]
 
 [tool.poetry.dependencies]
 python = ">=3.9,<3.13"
-opencv-contrib-python = "4.8.*"
+opencv-python = "4.8.*"
 PySide6 = "^6.6.0"
 pydantic = "^1.10.7"
 mediapipe = "^0.10.0"
diff --git a/pyproject.toml b/pyproject.toml
index b3a9024..c24d2fa 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,7 +55,7 @@ keywords = [
 
 #dynamic = ["dependencies"]
 dependencies = [
-    "opencv-contrib-python==4.8.*",
+    "opencv-python==4.8.*",
     "pydantic==2.*",
     "numpy<2",
     "tqdm==4.*",
