--- gst/lal/Makefile.in	2019-12-13 01:15:03.000000000 +0000
+++ gst/lal/Makefile.in	2020-02-06 15:29:02.000000000 +0000
@@ -435,7 +435,7 @@

 lib@GSTPLUGINPREFIX@gstlalugly_la_CFLAGS = $(AM_CFLAGS) $(GSL_CFLAGS) $(FFTW_CFLAGS) $(LAL_CFLAGS) $(GSTLAL_CFLAGS) $(gstreamer_CFLAGS)
 #lib@GSTPLUGINPREFIX@gstlalugly_la_LIBADD = $(top_builddir)/lib/libgstlalugly.la
-lib@GSTPLUGINPREFIX@gstlalugly_la_LDFLAGS = $(AM_LDFLAGS) $(GSL_LIBS) $(FFTW_LIBS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(GSTLAL_PLUGIN_LDFLAGS)
+lib@GSTPLUGINPREFIX@gstlalugly_la_LDFLAGS = $(AM_LDFLAGS) $(GSL_LIBS) $(FFTW_LIBS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(GSTLAL_PLUGIN_LDFLAGS) -lgstaudio-1.0
 all: all-am

 .SUFFIXES:
