--- Global.h.orig	2017-03-04 10:35:19.948562037 +0530
+++ Global.h	2017-03-04 10:35:36.904887354 +0530
@@ -19,7 +19,9 @@
 necessary to set even Long to 64 bits.
 */
 
+#ifndef POLY_Dmax
 #define   		POLY_Dmax	6	/* max dim of polytope	    */
+#endif
 /*
 POLY_Dmax should be set to the dimension of the polytopes that are analysed.
 While the programs still work if POLY_Dmax is set to a higher value, they may
