diff --git a/CMake/ConfigureBuild.cmake b/CMake/ConfigureBuild.cmake
index ffdcca170..e4d8133fe 100644
--- a/CMake/ConfigureBuild.cmake
+++ b/CMake/ConfigureBuild.cmake
@@ -162,7 +162,7 @@ if (BUILD_TYPE_LOWER STREQUAL "debug" AND WIN32)
 endif ()
 
 # Create the pkg-config package files on Unix systems
-if (UNIX)
+# if (UNIX)
   set(OGRE_PLUGIN_PREFIX "")
   set(OGRE_PLUGIN_EXT ".so")
   set(OGRE_PAGING_ADDITIONAL_PACKAGES "")
@@ -171,14 +171,19 @@ if (UNIX)
     set(OGRE_PLUGIN_EXT ".a")
   endif ()
 
+  if (WIN32)
+    set(OGRE_PLUGIN_PREFIX "")
+    set(OGRE_PLUGIN_EXT ".lib")
+  endif()
+
   set(OGRE_ADDITIONAL_LIBS "")
   set(OGRE_ADDITIONAL_INCLUDE_DIRS "")
   if (APPLE AND NOT APPLE_IOS)
     set(OGRE_ADDITIONAL_INCLUDE_DIRS "-I\${includedir}/OGRE/OSX")
   endif ()
   set(OGRE_CFLAGS "")
-  set(OGRE_PREFIX_PATH ${CMAKE_INSTALL_PREFIX})
-  if (OGRE_CONFIG_THREADS GREATER 0)
+  set(OGRE_PREFIX_PATH "")
+  if (UNIX AND OGRE_CONFIG_THREADS GREATER 0)
     set(OGRE_CFLAGS "-pthread")
     set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lpthread")
   endif ()
@@ -256,7 +261,7 @@ if (UNIX)
     set(CMAKE_COMPILER_IS_CLANGXX 1)
   endif ()
 
-endif ()
+# endif ()
 
 if(OGRE_CONFIG_STATIC_LINK_CRT)
 #We statically link to reduce dependencies
diff --git a/CMake/Templates/OGRE.pc.in b/CMake/Templates/OGRE.pc.in
index 5d752f299..a4baaafa6 100644
--- a/CMake/Templates/OGRE.pc.in
+++ b/CMake/Templates/OGRE.pc.in
@@ -2,7 +2,7 @@ prefix=@OGRE_PREFIX_PATH@
 exec_prefix=${prefix}
 libdir=${prefix}/@OGRE_LIB_DIRECTORY@
 includedir=${prefix}/include
-plugindir=${libdir}/OGRE
+plugindir=${prefix}/@OGRE_PLUGIN_DIRECTORY@
 
 Name: OGRE
 Description: Object-Oriented Graphics Rendering Engine
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac277f3c9..866d39ddd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -443,6 +443,11 @@ cmake_dependent_option(OGRE_FULL_RPATH "Build executables with the full required
 option(OGRE_PROFILING "Enable internal profiling support." FALSE)
 cmake_dependent_option(OGRE_CONFIG_STATIC_LINK_CRT "Statically link the MS CRT dlls (msvcrt)" FALSE "MSVC" FALSE)
 set(OGRE_LIB_DIRECTORY "lib${LIB_SUFFIX}" CACHE STRING "Install path for libraries, e.g. 'lib64' on some 64-bit Linux distros.")
+if (WIN32)
+  set(OGRE_PLUGIN_DIRECTORY "bin" CACHE STRING "Install path for plugins.")
+else()
+  set(OGRE_PLUGIN_DIRECTORY "lib${LIB_SUFFIX}/OGRE" CACHE STRING "Install path for plugins.")
+endif()
 if (WIN32)
 	option(OGRE_INSTALL_VSPROPS "Install Visual Studio Property Page." FALSE)
 	if (OGRE_INSTALL_VSPROPS)
