--- setup.py	2017-12-03 20:14:18.000000000 -0500
+++ setup.py.new	2017-12-03 23:29:35.000000000 -0500
@@ -10,7 +10,14 @@
 from setuptools.command.install import install
 
 
-# Build anltr files on installation
+ANTLR = 'antlr4'
+
+
+if sys.platform == 'win32':
+    ANTLR += '.cmd'
+
+
+# Build antlr files on installation
 #   this is such a mess... it looks like there are
 #   no common steps to develop and install
 
@@ -19,34 +26,39 @@
         compile_grammar()
         develop.run(self)
 
+
 class AntlrInstallCommand(install):
     def run(self):
         compile_grammar()
         install.run(self)
 
+
 def compile_grammar():
     here = os.path.dirname(__file__) or '.'
     package_dir = os.path.join(here, 'latex2sympy')
-    subprocess.check_output(['antlr4',  'PS.g4', '-o', 'gen'], cwd=package_dir)
+    subprocess.check_output([ANTLR,  'PS.g4', '-o', 'gen'], cwd=package_dir)
+
 
 if sys.version_info.major == 3:
     REQUIRES = ['antlr4-python3-runtime',  'sympy']
 else:
     REQUIRES = ['antlr4-python2-runtime',  'sympy']
 
+
 setuptools.setup(
     name='latex2sympy',
-    version=0.1,
+    version='0.1.0',
     author='august.codes',
     author_email='augustt198@gmail.com',
     description='Parse latex markup into sympy: suitable for programmatic modifcation',
-    license='GPLv3',
-    keywords='MIT',
+    license='MIT',
+    keywords='sympy latex',
     url='',
-    packages=['latex2sympy'],
+    packages=setuptools.find_packages(),
     classifiers=[
     ],
     install_requires=REQUIRES,
+    include_package_data=True,
     cmdclass=dict(
         install=AntlrInstallCommand,
         develop=AntlrDevelopCommand),
