From d7f3721604c24d04a8ca2896dbbc651488534db8 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Wed, 7 Aug 2024 11:27:14 +0200
Subject: [PATCH] fix protobuf

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 974e4f9..707a844 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,7 +37,7 @@ dependencies = [
     "ngboost>=0.3.12,<1",
     "numpy>=1.24.1,<2",
     "pandas>=1.4.2",
-    "protobuf<=3.20.3,<4",
+    "protobuf",
     "scikit-learn>=1.1.1,<2",
     "scikit-learn-extra>=0.3.0,<1",
     "scipy>=1.10.1,<2",
