From 01dfc743be3bef48f79aa615dbbfcb5ad90235c1 Mon Sep 17 00:00:00 2001
From: Mervin Fansler <mmfansler@gmail.com>
Date: Wed, 15 Dec 2021 19:30:50 -0500
Subject: [PATCH] force conftest.so

---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 8a45474..7cb5fa8 100755
--- a/configure
+++ b/configure
@@ -30,14 +30,14 @@ int main ()
 EOF
 
 _R_SHLIB_BUILD_OBJECTS_SYMBOL_TABLES_=false \
-  "${R}" CMD SHLIB conftest.cc ${GLPK_LIBS} >/dev/null 2>&1 \
+  "${R}" CMD SHLIB conftest.cc ${GLPK_LIBS} -o conftest.so >/dev/null 2>&1 \
   && "$R" --slave --vanilla -e 'dyn.load("conftest.so")'
 status=${?}
 if test ${status} -ne 0; then
   rm -f conftest.*o
   GLPK_LIBS="-lglpk -lgmp -lm"
   _R_SHLIB_BUILD_OBJECTS_SYMBOL_TABLES_=false \
-    "${R}" CMD SHLIB conftest.cc ${GLPK_LIBS} >/dev/null 2>&1 \
+    "${R}" CMD SHLIB conftest.cc ${GLPK_LIBS} -o conftest.so >/dev/null 2>&1 \
     && "$R" --slave --vanilla -e 'dyn.load("conftest.so")'
   status=${?}
 fi
-- 
2.33.0

