From 8050ee3e9f8df400641e0cbe6a2b2ea2e68438bd Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sat, 18 Jan 2020 23:02:05 -0800
Subject: [PATCH 1/2] Override custom install layout

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8bc5a106..183d84a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -476,9 +476,9 @@ set_target_properties(connector PROPERTIES
 
 install(TARGETS connector
   CONFIGURATIONS Release RelWithDebInfo
-  ARCHIVE DESTINATION "${INSTALL_LIB_DIR_STATIC}" COMPONENT XDevAPIDev
-  RUNTIME DESTINATION "${INSTALL_LIB_DIR}" COMPONENT XDevAPIDll
-  LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT XDevAPIDll
+  ARCHIVE COMPONENT XDevAPIDev
+  RUNTIME COMPONENT XDevAPIDll
+  LIBRARY COMPONENT XDevAPIDll
 )
 
 install(TARGETS connector
-- 
2.24.1

