diff --git a/tests/test_cli.py b/tests/test_cli.py
index 00fc50c..6803012 100644
--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -28,9 +28,7 @@ import num2words
 class CliCaller(object):

     def __init__(self):
-        self.cmd = os.path.realpath(os.path.join(os.path.dirname(__file__),
-                                    "..", "bin", "num2words"))
-        self.cmd_list = ["python", self.cmd]
+        self.cmd_list = ["num2words"]

     def run_cmd(self, *args):
         cmd_list = self.cmd_list + [str(arg) for arg in args]
