diff --git a/hdwallets/_bip32.py b/hdwallets/_bip32.py
--- a/hdwallets/_bip32.py
+++ b/hdwallets/_bip32.py
@@ -91,6 +94,7 @@
         if isinstance(path, str):
             path = _deriv_path_str_to_list(path)
         chaincode, key = self.master_chaincode, self.master_privkey
+        pubkey = self.master_pubkey
         # We'll need the private key at some point anyway, so let's derive
         # everything from private keys.
         if _hardened_index_in_path(path):
