diff --git a/aksetup_helper.py b/aksetup_helper.py
index f90d085..3e96a4a 100644
--- a/aksetup_helper.py
+++ b/aksetup_helper.py
@@ -4,6 +4,7 @@ import sys
 from setuptools.command.build_ext import (  # noqa: N812
         build_ext as BaseBuildExtCommand)
 
+CXXFLAGS = ['-DBOOST_PYTHON_NO_PY_SIGNATURES','-DBOOST_NO_INCLASS_MEMBER_INITIALIZATION']
 
 def count_down_delay(delay):
     from time import sleep
