diff --git a/setup.py b/setup.py
index b07a88f..f4232ca 100644
--- a/setup.py
+++ b/setup.py
@@ -5,22 +5,6 @@
 
 import os
 
-if hasattr(cpp_extension, "get_compiler_abi_compatibility_and_version"):
-    # - New function since torch 1.12
-    if cpp_extension.get_compiler_abi_compatibility_and_version("g++")[0]:
-        os.environ["CC"] = "g++"
-        os.environ["CXX"] = "g++"
-    else:
-        os.environ["CC"] = "c++"
-        os.environ["CXX"] = "c++"
-else:
-    # - This works up to torch 1.11
-    if cpp_extension.check_compiler_abi_compatibility("g++"):
-        os.environ["CC"] = "g++"
-        os.environ["CXX"] = "g++"
-    else:
-        os.environ["CC"] = "c++"
-        os.environ["CXX"] = "c++"
 
 # Class for clean command
 class Cleaner(Command):
