diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index babe02138..c29ef1960 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -321,7 +321,7 @@ else()
         target_link_libraries(vulkan PRIVATE Threads::Threads)
     endif()
 
-    if(APPLE)
+    if(FALSE)
         find_library(COREFOUNDATION_LIBRARY NAMES CoreFoundation)
         target_link_libraries(vulkan PRIVATE "-framework CoreFoundation")
 
