diff --git a/setup.py b/setup.py
index 3e7bbd8..c613e7b 100644
--- a/setup.py
+++ b/setup.py
@@ -24,7 +24,7 @@ except:
     input()
     exit()
 
-from numpy.distutils.core import setup
+# from numpy.distutils.core import setup
 
 DOC_FILES = []
 
@@ -63,7 +63,7 @@ if __name__ == "__main__":
 
     #data_files = ['test_data/' + i for i in TEST_DATA_FILES]
     #data_files.extend(['docs/' + i for i in doc_files])
-    setup(configuration = configuration,
+    setuptools.setup(configuration = configuration,
         install_requires=['numpy', 'scipy'], # can also add version specifiers   #namespace_packages=['kernel'],
         #py_modules = ['kernel', 'tests', 'examples', 'solvers'],
         packages=setuptools.find_packages(),
