diff --git a/data/meson.build b/data/meson.build
index 1234567..abcdefg 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -23,16 +23,17 @@ metainfo_with_relinfo = custom_target('gen-output',
     command : [ascli_exe, 'news-to-metainfo', '--limit=6', '@INPUT0@', '@INPUT1@', '@OUTPUT@']
 )

-metainfo_i18n = i18n.itstool_join(
-    input:  metainfo_with_relinfo,
-    output: 'org.freedesktop.appstream.cli.metainfo.xml',
-    mo_targets: i18n_result[0],
-    its_files: [join_paths(meson.current_source_dir(), 'its', 'metainfo.its')],
+metainfo_i18n = custom_target('metainfo-no-i18n',
+    input: metainfo_with_relinfo,
+    output: 'org.freedesktop.appstream.cli.metainfo.xml',
+    command: ['cp', '@INPUT@', '@OUTPUT@'],
     install: true,
     install_dir: metainfo_dir,
 )

 test('as-validate_metainfo.cli',
     ascli_exe,
     args: ['validate',
            '--pedantic',
