From 5bfd572df7650ef8c316950dbe8dc9558de7153f Mon Sep 17 00:00:00 2001
From: Joseph Hunkeler <jhunkeler@gmail.com>
Date: Tue, 9 Dec 2025 11:28:17 -0500
Subject: [PATCH 2/2] Handle missing UFFD define

---
 src/compression/compression.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/compression/compression.c b/src/compression/compression.c
index b9f513e..a28f0a7 100644
--- a/src/compression/compression.c
+++ b/src/compression/compression.c
@@ -25,6 +25,11 @@
 #include <sys/ioctl.h>
 #include <sys/poll.h>
 #include <sys/syscall.h>
+
+#ifndef UFFD_USER_MODE_ONLY
+#pragma message "warning: UFFD_USER_MODE_ONLY missing"
+#define UFFD_USER_MODE_ONLY 0
+#endif
 #endif
 
 
-- 
2.51.2

