From febffadb33ea2a48365998dd7900187ee6c665d4 Mon Sep 17 00:00:00 2001
From: Joseph Hunkeler <jhunkeler@gmail.com>
Date: Mon, 8 Dec 2025 15:42:01 -0500
Subject: [PATCH 2/3] Check userfaultfd user mode exists

---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index 1102b4b..2236fdd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -80,10 +80,12 @@ AC_COMPILE_IFELSE(
   ]], [[
     struct uffdio_api api;
     api.api = UFFD_API;
+    int mode = UFFD_USER_MODE_ONLY;
   ]])],
   [have_userfaultfd=yes],
   [have_userfaultfd=no]
 )
+
 AM_CONDITIONAL([HAVE_USERFAULTFD], [test "x$have_userfaultfd" = "xyes"])
 AS_IF([test "x$have_userfaultfd" = "xyes"], [
   AC_DEFINE([HAVE_USERFAULTFD], [1],
@@ -91,6 +93,7 @@ AS_IF([test "x$have_userfaultfd" = "xyes"], [
 ])
 AC_CHECK_DECLS([SYS_userfaultfd], [], [], [[#include <sys/syscall.h>]])
 
+
 # http://www.gnu.org/software/autoconf-archive/ax_valgrind_check.html
 # - make check-valgrind
 AX_VALGRIND_CHECK
-- 
2.51.2

