--- tests/Makefile.am	2025-12-08 12:15:42.204663551 -0500
+++ tests/Makefile.am	2025-12-08 12:29:21.650297233 -0500
@@ -7,7 +7,10 @@ AM_CPPFLAGS = \
 
 asdf_standard_dir = $(abs_top_srcdir)/asdf-standard
 asdf_standard_submodule = $(asdf_standard_dir)/.git
-munit_submodule = $(srcdir)/tests/munit.git
+munit_dir = $(srcdir)/tests/munit
+munit_cflags = -I$(munit_dir)
+munit_ldflags = $(builddir)/libmunit.a
+munit_submodule = $(munit_dir).git
 submodules = $(asdf_standard_submodule) $(munit_submodule)
 
 $(asdf_standard_submodule) $(munit_submodule):
@@ -43,14 +46,14 @@ unit_test_cppflags = \
     -DREFERENCE_FILES_DIR=\"$(asdf_standard_dir)/reference_files\" \
     -DFIXTURES_DIR=\"$(abs_srcdir)/fixtures\"
 unit_test_cflags = \
-    $(FYAML_CFLAGS) $(STATGRAB_CFLAGS) $(LZ4_CFLAGS) $(CODE_COVERAGE_CFLAGS) $(ASDF_CFLAGS)
+     $(FYAML_CFLAGS) $(STATGRAB_CFLAGS) $(LZ4_CFLAGS) $(CODE_COVERAGE_CFLAGS) $(ASDF_CFLAGS)
 
 if DEBUG
 unit_test_cflags += -DMUNIT_NO_FORK
 endif
 
 unit_test_ldflags = $(CODE_COVERAGE_LDFLAGS)
-unit_test_ldadd = $(top_builddir)/libasdf.la libmunit.a $(LZ4_LIBS) $(STATGRAB_LIBS)
+unit_test_ldadd = $(top_builddir)/libasdf.la $(munit_ldflags) $(FYAML_LIBS) $(LZ4_LIBS) $(STATGRAB_LIBS)
 
 # Unit tests
 check_LIBRARIES = libmunit.a
