diff -r -u -p a/libasdf-0.1.0a2/configure.ac b/libasdf-0.1.0a2/configure.ac
--- configure.ac	2025-12-05 03:58:39.000000000 -0500
+++ configure.ac	2025-12-08 13:31:04.799161987 -0500
@@ -80,10 +80,12 @@ AC_COMPILE_IFELSE(
   ]], [[
     struct uffdio_api api;
     api.api = UFFD_API;
+    int mode = UFFD_USER_MODE_ONLY;
   ]])],
   [have_userfaultfd=yes],
   [have_userfaultfd=no]
 )
+
 AM_CONDITIONAL([HAVE_USERFAULTFD], [test "x$have_userfaultfd" = "xyes"])
 AS_IF([test "x$have_userfaultfd" = "xyes"], [
   AC_DEFINE([HAVE_USERFAULTFD], [1],
@@ -91,6 +93,7 @@ AS_IF([test "x$have_userfaultfd" = "xyes
 ])
 AC_CHECK_DECLS([SYS_userfaultfd], [], [], [[#include <sys/syscall.h>]])
 
+
 # http://www.gnu.org/software/autoconf-archive/ax_valgrind_check.html
 # - make check-valgrind
 AX_VALGRIND_CHECK
