From 2787684e7d0aaa6a128b8f36d3551fdb89567be7 Mon Sep 17 00:00:00 2001
From: Joseph Hunkeler <jhunkeler@gmail.com>
Date: Mon, 8 Dec 2025 15:44:28 -0500
Subject: [PATCH 3/3] Fix test linkage

---
 tests/Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 7f3424b..b48bfd8 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -52,7 +52,7 @@ if DEBUG
 unit_test_cflags += -DMUNIT_NO_FORK
 endif
 
-unit_test_ldflags = $(CODE_COVERAGE_LDFLAGS)
+unit_test_ldflags = $(LDFLAGS) $(CODE_COVERAGE_LDFLAGS)
 unit_test_ldadd = $(top_builddir)/libasdf.la $(munit_ldflags) $(FYAML_LIBS) $(LZ4_LIBS) $(STATGRAB_LIBS)
 
 # Unit tests
@@ -194,7 +194,7 @@ $(cpp_header_test): $(asdf_public_headers) $(srcdir)/scripts/generate_test_cpp_h
 
 test-cpp-headers$(EXEEXT): $(cpp_header_test) $(top_builddir)/libasdf.la
 	$(AM_V_at)$(LIBTOOL) --mode=link $(CXX) -std=c++17 -I$(top_srcdir)/include \
-	    $(ASDF_LDFLAGS) $(cpp_header_test) $(top_builddir)/libasdf.la \
+	    $(LDFLAGS) $(ASDF_LDFLAGS) $(cpp_header_test) $(top_builddir)/libasdf.la \
 	    -o $@ $(QUIET_OUT)
 
 TESTS += test-cpp-headers
@@ -214,7 +214,7 @@ $(DOC_EXAMPLES_DIR)/test-doc-examples.sh: $(top_srcdir)/README.rst $(srcdir)/scr
 	$(AM_V_at)for f in $(DOC_EXAMPLES_DIR)/*.c; do \
 	    exe=$${f%.c}; \
 	    $(LIBTOOL) --mode=link $(CC) $(ASDF_CFLAGS) -I$(top_srcdir)/include \
-	      $(ASDF_LDFLAGS) $$f $(top_builddir)/libasdf.la -o $$exe $(QUIET_OUT); \
+	      $(LDFLAGS) $(ASDF_LDFLAGS) $$f $(top_builddir)/libasdf.la -o $$exe $(QUIET_OUT); \
 	done
 
 endif
-- 
2.51.2

