diff --git a/setup.py b/setup.py
index ff680c2..5796cbd 100644
--- a/setup.py
+++ b/setup.py
@@ -40,6 +40,7 @@ def build_extension(self, ext):
             '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
             '-DPYTHON_EXECUTABLE=' + sys.executable,
             '-DVERSION_INFO={}'.format(self.distribution.get_version()),
+            '-DCMAKE_BUILD_TYPE=Release',
         ]
         eigen_dir = os.environ.get('EIGEN3_INCLUDE_DIRS')
         if eigen_dir is not None:
@@ -52,19 +53,7 @@ def build_extension(self, ext):
         build_args = ['--config', cfg]
 
         if platform.system() == "Windows":
-            if os.environ.get('CMAKE_TOOLCHAIN_FILE') is not None:
-                cmake_toolchain_file = os.environ.get('CMAKE_TOOLCHAIN_FILE')
-                # print(f'-DCMAKE_TOOLCHAIN_FILE={cmake_toolchain_file}')
-                cmake_args += [f'-DCMAKE_TOOLCHAIN_FILE={cmake_toolchain_file}']
-            cmake_args += ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}'.format(cfg.upper(), extdir)]
-            if sys.maxsize > 2**32:
-                if os.environ.get('CMAKE_TOOLCHAIN_FILE') is not None:
-                    cmake_args += ['-DVCPKG_TARGET_TRIPLET=x64-windows']
-                cmake_args += ['-A', 'x64']
-            build_args += ['--', '/m']
-        else:
-            cmake_args += ['-DCMAKE_BUILD_TYPE=' + cfg]
-            build_args += ['--', '-j2']
+            cmake_args += ['-G Ninja', '-DCMAKE_CXX_FLAGS=-DNOMINMAX']
 
         env = os.environ.copy()
         env['CXXFLAGS'] = '{} -DVERSION_INFO=\\"{}\\"'.format(
