--- setup.py	2024-10-26 13:10:54.582141000 -0400
+++ setup.py	2024-10-26 13:11:46.167990600 -0400
@@ -3,7 +3,8 @@
 
 from setuptools import setup
 from setuptools import find_packages
-from nanocube import VERSION as NANOCUBE_VERSION
+import os
+NANOCUBE_VERSION = os.environ["VERSION"]
 
 
 # ...to run the build and deploy process to pypi.org manually:
@@ -81,4 +82,4 @@
         'Documentation': 'https://zeutschler.github.io/nanocube/',
         'GitHub': 'https://github.com/Zeutschler/nanocube',
     },
-)
\ No newline at end of file
+)
