diff --git a/CMakeLists.txt b/CMakeLists.txt
index d643b0f..549ef7a 100644
--- a/extern/datetime-fortran/CMakeLists.txt
+++ b/extern/datetime-fortran/CMakeLists.txt
@@ -25,7 +25,7 @@ endif()
 if(MINGW)
   # MinGW / MSYS2 GCC needs this for strftime to work
   # https://sourceforge.net/p/mingw-w64/bugs/793/
-  target_link_libraries(datetime PRIVATE "ucrtbase")
+  target_link_libraries(datetime PRIVATE stdc++)
 endif(MINGW)
 
 # tests
