diff --git a/src/pelagic_base.F90 b/src/pelagic_base.F90
index 446a109..a3dc656 100644
--- a/extern/fabm/extern/ersem/src/pelagic_base.F90
+++ b/extern/fabm/extern/ersem/src/pelagic_base.F90
@@ -34,7 +34,7 @@ module ersem_pelagic_base
    contains
       procedure :: initialize
       procedure :: do_bottom
-      procedure :: do
+      procedure :: do => pelagic_base_do
 
       procedure :: initialize_ersem_base
       procedure :: get_sinking_rate
@@ -269,7 +269,7 @@ contains
       rm = self%rm
    end function
    
-   subroutine DO (self, _ARGUMENTS_DO_)
+   subroutine pelagic_base_do(self, _ARGUMENTS_DO_)
       class (type_ersem_pelagic_base), intent(in) :: self
       
       real (rk) :: k1_rate,state, et, ETW
