diff --git a/um_ppibm/setup.py b/um_ppibm/setup.py
index 73122dc..8346170 100644
--- a/um_ppibm/setup.py
+++ b/um_ppibm/setup.py
@@ -66,7 +66,7 @@ setuptools.setup(
             "um_ppibm.um_ieee2ibm32",
             ["lib/um_ppibm/um_ieee2ibm32.c"],
             include_dirs=[np.get_include()],
-            libraries=["shum_string_conv", "shum_byteswap", "shum_data_conv"],
+            libraries=["shum_string_conv", "shum_byteswap", "shum_data_conv", "shum_string_conv"],
         )
     ],
 )
