From bb8cd4f95b16ad508e7401c77bb231d12d05b260 Mon Sep 17 00:00:00 2001
From: Daniel Tipping <daniel@oldreliable.tech>
Date: Mon, 25 Mar 2019 20:50:15 +0000
Subject: [PATCH] Use find_package instead of find_library for threads

---
 CMakeLists.txt | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/metview/CMakeLists.txt b/metview/CMakeLists.txt
index 4a9efdda..64659b80 100644
--- a/metview/CMakeLists.txt
+++ b/metview/CMakeLists.txt
@@ -446,8 +446,8 @@ ecbuild_debug("VERSION: ${METVIEW_VERSION_STR}")
 ###############################################################################
 # Find external dependencies
 
-#set( CMAKE_THREAD_PREFER_PTHREAD 1 )
-#find_package( Threads )
+set( CMAKE_THREAD_PREFER_PTHREAD 1 )
+find_package( Threads REQUIRED )
 
 #ecbuild_add_extra_search_paths( boost )
 #find_package( Boost 1.49.0 )
@@ -471,16 +471,6 @@ if(CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
     endif()
 endif()
 
-
-find_library(PTHREAD_LIB pthread)
-if(PTHREAD_LIB)
-    ecbuild_debug("pthread found: ${PTHREAD_LIB}")
-else()
-    ecbuild_critical("library pthread not found")
-endif()
-
-
-
 # FLEXTRA
 
 if(DEFINED FLEXTRA_PATH)
-- 
2.20.1

