From 56b348f4eaef372e51e24a464ae9ec865ca4ed14 Mon Sep 17 00:00:00 2001
From: Daniel Tipping <daniel@oldreliable.tech>
Date: Fri, 29 Mar 2019 16:46:56 +0000
Subject: [PATCH] Pass $CPP to rcpgen

---
 mars-client/src/CMakeLists.txt | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/mars-client/src/CMakeLists.txt b/mars-client/src/CMakeLists.txt
index cbc7cb7..4c0f274 100644
--- a/mars-client/src/CMakeLists.txt
+++ b/mars-client/src/CMakeLists.txt
@@ -171,6 +171,14 @@ else()
     ecbuild_critical("Found neither flex nor lex")
 endif()
 
+set(_rpcgen_cmd rpcgen)
+if(RPCGEN_USE_CPP_ENV AND DEFINED ENV{CPP})
+    get_filename_component(_rpcgen_cpp $ENV{CPP} DIRECTORY)
+    list(APPEND _rpcgen_cmd -Y ${_rpcgen_cpp})
+elseif(RPCGEN_USE_CPP_ENV)
+    ecbuild_critical("$CPP not found in environment")
+endif()
+
 # targets
 
 configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/rpcmars.x ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x COPYONLY )
@@ -178,13 +186,13 @@ configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/rpcmars.x ${CMAKE_CURRENT_BINARY_DIR
 add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.h
 	COMMAND ${CMAKE_COMMAND} -E remove ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.h
-	COMMAND rpcgen -h -o ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.h ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x
+    COMMAND ${_rpcgen_cmd} -h -o ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.h ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x
     DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x )
 
 add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/marsxdr.c
 	COMMAND ${CMAKE_COMMAND} -E remove ${CMAKE_CURRENT_BINARY_DIR}/marsxdr.c
-	COMMAND rpcgen -c -o ${CMAKE_CURRENT_BINARY_DIR}/marsxdr.c ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x
+	COMMAND ${_rpcgen_cmd} -c -o ${CMAKE_CURRENT_BINARY_DIR}/marsxdr.c ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x
     DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x )
 
 add_custom_command(
-- 
2.19.1

