From cb03f9fd059f818f3f15da948aae9845398c2312 Mon Sep 17 00:00:00 2001
From: Daniel Tipping <daniel@oldreliable.tech>
Date: Fri, 12 Apr 2019 12:08:40 +0100
Subject: [PATCH] Pass $CPP to rpcgen

---
 metview/src/libMars/CMakeLists.txt | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/metview/src/libMars/CMakeLists.txt b/metview/src/libMars/CMakeLists.txt
index ee4e528..b166b28 100644
--- a/metview/src/libMars/CMakeLists.txt
+++ b/metview/src/libMars/CMakeLists.txt
@@ -120,18 +120,26 @@ endif()
 # and work on it there. Note: will this cause a problem with in-source builds?
 # If so, we may need to first check whether the file already exists.
 
+set(_rpcgen_cmd ${RPCGEN_EXECUTABLE})
+if(RPCGEN_USE_CPP_ENV AND DEFINED ENV{CPP})
+    get_filename_component(_rpcgen_cpp $ENV{CPP} DIRECTORY)
+    list(APPEND _rpcgen_cmd -Y ${_rpcgen_cpp})
+elseif(RPCGEN_USE_CPP_ENV)
+    ecbuild_critical("$CPP not found in environment")
+endif()
+
 if(NOT CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR)
     configure_file(${MARS_SOURCE}/rpcmars.x ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x)
 endif()
 
 ADD_CUSTOM_COMMAND(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.h
-    COMMAND ${RPCGEN_EXECUTABLE} -h -o ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.h ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x
+	COMMAND ${_rpcgen_cmd} -h -o ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.h ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x
     DEPENDS ${MARS_SOURCE}/rpcmars.x)
 
 ADD_CUSTOM_COMMAND(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.c
-    COMMAND ${RPCGEN_EXECUTABLE} -c -o ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.c ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x
+	COMMAND ${_rpcgen_cmd} -c -o ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.c ${CMAKE_CURRENT_BINARY_DIR}/rpcmars.x
     DEPENDS ${MARS_SOURCE}/rpcmars.x)
 
 ADD_CUSTOM_COMMAND(
-- 
2.19.1

