From 6502e9172c674c84e07da1f8c4a39eff33d797de Mon Sep 17 00:00:00 2001
From: Daniel Tipping <daniel@oldreliable.tech>
Date: Fri, 12 Apr 2019 13:35:48 +0100
Subject: [PATCH] Set CMAKE_CXX_STANDARD_REQUIRED ON for Metview

---
 metview/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/metview/CMakeLists.txt b/metview/CMakeLists.txt
index 4a9efdd..bc842c6 100644
--- a/metview/CMakeLists.txt
+++ b/metview/CMakeLists.txt
@@ -52,6 +52,7 @@ ecbuild_requires_macro_version( 1.9 )
 
 #ecbuild_add_cxx11_flags()  # Magics 3 requires C++11/14
 set (CMAKE_CXX_STANDARD 14)
+set (CMAKE_CXX_STANDARD_REQUIRED ON)
 set (CMAKE_C_STANDARD 99)
 
 ###############################################################################
-- 
2.20.1

