From 8aeb8fd1ff2686f89913155742757dba109a1171 Mon Sep 17 00:00:00 2001
From: Daniel Tipping <daniel@oldreliable.tech>
Date: Mon, 15 Apr 2019 15:21:57 +0100
Subject: [PATCH] Remove cxx11 flags

---
 metview/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/metview/CMakeLists.txt b/metview/CMakeLists.txt
index bc842c6..42d5005 100644
--- a/metview/CMakeLists.txt
+++ b/metview/CMakeLists.txt
@@ -83,7 +83,6 @@ ecbuild_add_option( FEATURE PPROC_MIR
     REQUIRED_PACKAGES "PROJECT mir" )
 
 if( HAVE_PPROC_MIR )
-    ecbuild_add_cxx11_flags()
     list( APPEND MARS_FEATURE_LIBRARIES mir )
 endif()
 
-- 
2.19.1

