diff --git a/src/CSGCGALDomain3D.cpp b/src/CSGCGALDomain3D.cpp
index 3ef7742..087e7d1 100644
--- a/src/CSGCGALDomain3D.cpp
+++ b/src/CSGCGALDomain3D.cpp
@@ -1506,7 +1506,7 @@ namespace
       parameters = default_parameters();

       // Create the polyhedron
-      BuildFromFacetList<Exact_HalfedgeDS> builder(vertices, facets, {});
+      BuildFromFacetList<Exact_HalfedgeDS> builder(vertices, facets, std::set<std::size_t> {});
       impl->p.delegate(builder);
     }
     //-----------------------------------------------------------------------------
