diff --git a/swig/CMakeLists.txt b/swig/CMakeLists.txt
index 832318a..2e4753e 100644
--- a/swig/CMakeLists.txt
+++ b/swig/CMakeLists.txt
@@ -10,9 +10,12 @@ endif()
 find_package(SWIG REQUIRED)
 include(${SWIG_USE_FILE})

-# FIXME: Instead use DOLFIN_PYTHON_EXECUTABLE when
-# this has been added to DOLFINConfig.cmake
-find_package(PythonInterp 2)
+if (NOT DEFINED DOLFIN_PYTHON_EXECUTABLE )
+  message(WARNING "DOLFIN_PYTHON_EXECUTABLE not defined. This should be in DOLFINConfig.cmake")
+  find_package(PythonInterp)
+else()
+  set(PYTHON_EXECUTABLE ${DOLFIN_PYTHON_EXECUTABLE})
+endif()
