diff --git a/swig/CMakeLists.txt b/swig/CMakeLists.txt
index 832318a..fd3bec7 100644
--- a/swig/CMakeLists.txt
+++ b/swig/CMakeLists.txt
@@ -12,7 +12,7 @@ include(${SWIG_USE_FILE})
 
 # FIXME: Instead use DOLFIN_PYTHON_EXECUTABLE when
 # this has been added to DOLFINConfig.cmake
-find_package(PythonInterp 2)
+find_package(PythonInterp 3)
 
 # Don't use these. Instad get the values from Dolfin
 # find_package(PythonLibs)
