diff --git a/setup.py b/setup.py
index cb10518..6559634 100644
--- a/setup.py
+++ b/setup.py
@@ -27,6 +27,9 @@ os.chdir(SCRIPT_DIR)
 RUST_DIR = str(SCRIPT_DIR / "rust")
 
 def build_native(spec):
+    cargo_build_target = os.getenv('CARGO_BUILD_TARGET', '')
+    target_path=str(os.path.join('target', cargo_build_target, 'release'))
+
     arch_flags = os.getenv("ARCHFLAGS")
     if arch_flags is not None and ("x86_64" in arch_flags and "arm64" in arch_flags):
         spec.add_external_build(
@@ -47,7 +50,7 @@ def build_native(spec):
         spec.add_cffi_module(
             module_path='cmsis_pack_manager._native',
             dylib=lambda: build.find_dylib('cmsis_cffi',
-                                           in_path='target/release'),
+                                           in_path=target_path),
             header_filename=lambda: build.find_header('cmsis.h', in_path='cmsis-cffi')
         )
     else:
@@ -59,7 +62,7 @@ def build_native(spec):
         spec.add_cffi_module(
             module_path='cmsis_pack_manager._native',
             dylib=lambda: build.find_dylib('cmsis_cffi',
-                                           in_path='target/release/deps'),
+                                           in_path='{}/deps'.format(target_path)),
             header_filename=lambda: build.find_header('cmsis.h', in_path='cmsis-cffi')
         )
 
