diff --git a/SRC/CMakeLists.txt b/SRC/CMakeLists.txt
index 36b55d1..a71c80b 100644
--- a/SRC/CMakeLists.txt
+++ b/SRC/CMakeLists.txt
@@ -121,16 +121,22 @@ if(enable_complex16)
 endif()

 add_library(superlu_dist ${sources} ${HEADERS})
+# build both shared and static libs
+add_library(superlu_dist-static STATIC ${sources} ${HEADERS})
+
+foreach(target superlu_dist superlu_dist-static)
-target_link_libraries(superlu_dist
+target_link_libraries(${target}
                       ${MPI_C_LIBRARIES} ${BLAS_LIB} ${PARMETIS_LIB} m)
-set_target_properties(superlu_dist PROPERTIES
+set_target_properties(${target} PROPERTIES
+                      OUTPUT_NAME superlu_dist
                       VERSION ${PROJECT_VERSION} SOVERSION ${VERSION_MAJOR}
 )
+endforeach(target)

 # Define GNU standard installation directories
 include(GNUInstallDirs)

-install(TARGETS superlu_dist
+install(TARGETS superlu_dist superlu_dist-static
 # DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
      DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
--
2.16.1

