Index: setup.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup.py	(revision )
+++ setup.py	(revision )
@@ -329,6 +329,8 @@
             pass
         elif platform == 'sunos5':
             lflags += ['-Wl,-s']
+        elif platform == 'darwin':
+            pass
         else:
             lflags += ['-Wl,-S']
 
