Index: jcc3/sources/jcc.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- jcc3/sources/jcc.cpp	(revision )
+++ jcc3/sources/jcc.cpp	(revision )
@@ -210,7 +210,7 @@
     uintmax_t hash = (uintmax_t) PyObject_Hash(arg);
     char buffer[hexdig + 1];
 
-    sprintf(buffer, "%0*"PRIxMAX, (int) hexdig, hash);
+    sprintf(buffer, "%0*" PRIxMAX, (int) hexdig, hash);
     return PyUnicode_FromStringAndSize(buffer, hexdig);
 }
 #endif
