From 83da04eb709fec0b0ebc35694ab7a22ce0a0bc76 Mon Sep 17 00:00:00 2001
From: Anand Balakrishnan <anandbala1597@gmail.com>
Date: Tue, 15 Oct 2024 18:17:20 -0700
Subject: [PATCH] don't upperbound ply version

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index e8cde5c..5e0335a 100644
--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,7 @@ def run_setup():
         url=PACKAGE_URL,
         license='BSD',
         python_requires=PYTHON_REQUIRES,
-        install_requires=['ply >= 3.4, <= 3.10'],
+        install_requires=["ply >= 3.4"],
         tests_require=TESTS_REQUIRE,
         packages=[PACKAGE_NAME],
         package_dir={PACKAGE_NAME: PACKAGE_NAME},
-- 
2.46.2

