From b82e92622c3c9b8bac7f924fc6fb9cf386f8940f Mon Sep 17 00:00:00 2001
From: Jimmy Shen <14003693+jmmshn@users.noreply.github.com>
Date: Mon, 20 Mar 2023 16:36:42 -0700
Subject: [PATCH] Update pyproject.toml

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 893af5c..c9261e0 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,7 +25,7 @@ keywords = ["machine-learning", "dft", "vasp", "volumetric", "pymatgen"]
 license = {text = "modified BSD"}
 name = "mp-pyrho"
 readme = "README.md"
-requires-python = '>="3.8"'
+requires-python = ">=3.8"
 
 [project.optional-dependencies]
 dev = ["pre-commit==2.17.0"]
