From 03952355bcc3343e3f5474faf8ef525d529e6669 Mon Sep 17 00:00:00 2001
From: Bas Zalmstra <bas@prefix.dev>
Date: Wed, 30 Apr 2025 14:04:54 +0200
Subject: [PATCH 2/5] allow using cmake on windows

---
 CMakeLists.txt | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf37bde7..2f12de8c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -177,12 +177,12 @@ if(MEGA)
 		# automatically.
 	endif()
 else()
-	if(MSVC)
-		message(FATAL_ERROR "
-It is currently only possible to build with megasource or (experimental) mingw on Windows.
-Please see https://github.com/love2d/megasource
-")
-	endif()
+# 	if(MSVC)
+# 		message(FATAL_ERROR "
+# It is currently only possible to build with megasource or (experimental) mingw on Windows.
+# Please see https://github.com/love2d/megasource
+# ")
+# 	endif()
 
 	find_package(Freetype REQUIRED)
 	find_package(ModPlug REQUIRED)
