From 0b86e0c5d221094e37028bdfbbffe64a966f6273 Mon Sep 17 00:00:00 2001
From: Bas Zalmstra <bas@prefix.dev>
Date: Wed, 30 Apr 2025 14:09:49 +0200
Subject: [PATCH 4/5] find correct openal library on windows

---
 extra/cmake/FindOpenAL.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extra/cmake/FindOpenAL.cmake b/extra/cmake/FindOpenAL.cmake
index 112beca9..7bdce30b 100644
--- a/extra/cmake/FindOpenAL.cmake
+++ b/extra/cmake/FindOpenAL.cmake
@@ -18,7 +18,7 @@ find_path(OPENAL_INCLUDE_DIR
 	PATHS ${OPENAL_SEARCH_PATHS})
 
 find_library(OPENAL_LIBRARY
-	NAMES openal
+	NAMES openal openal32
 	PATH_SUFFIXES lib
 	PATHS ${OPENAL_SEARCH_PATHS})
