From 154c72553948df3804a903e7d592be449312874d Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Mon, 8 Apr 2024 09:55:15 -0500
Subject: [PATCH] add requirements.txt

---
 requirements.txt | 3 +++
 1 file changed, 3 insertions(+)
 create mode 100644 requirements.txt

diff --git a/requirements.txt b/requirements.txt
new file mode 100644
index 0000000..c60a4fc
--- /dev/null
+++ requirements.txt
@@ -0,0 +1,3 @@
+torch
+transformers
+bitsandbytes
\ No newline at end of file
