From 0f43c3bebdc2d7ca1f68308d9d29287a27086dbf Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 4 Mar 2026 09:46:06 +0100
Subject: [PATCH] Update pyproject.toml

Signed-off-by: Silvio Traversaro <silvio@traversaro.it>
---
 pyproject.toml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index f4fb7d24967..1342ca30d33 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -66,7 +66,8 @@ dependencies = [
 
     # Core dependencies
     "setuptools>=71.0.0,<81.0.0",
-    "cmake>=3.29.0.1,<4.2.0",
+    # In conda-forge cmake does not install pypi metadata
+    # "cmake>=3.29.0.1,<4.2.0",
     "einops>=0.8.0,<0.9.0",
     "opencv-python-headless>=4.9.0,<4.13.0",
     "av>=15.0.0,<16.0.0",
