From 0f94b36b9bd76ad7cc4a9b0246aaabced562a85d Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Fri, 10 Nov 2023 13:42:18 -0600
Subject: [PATCH] Add imported target with pkg-config to support windows

---
 blosc2/CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/blosc2/CMakeLists.txt b/blosc2/CMakeLists.txt
index 597be1a..9e19431 100644
--- a/blosc2/CMakeLists.txt
+++ b/blosc2/CMakeLists.txt
@@ -4,9 +4,8 @@ add_library(blosc2_ext MODULE ${blosc2_ext})
 if(USE_SYSTEM_BLOSC2)
     set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
     find_package(PkgConfig REQUIRED)
-    pkg_check_modules(Blosc2 REQUIRED blosc2)
-    target_include_directories(blosc2_ext PRIVATE ${Blosc2_INCLUDE_DIRS})
-    target_link_libraries(blosc2_ext ${Blosc2_LDFLAGS})
+    pkg_check_modules(Blosc2 REQUIRED IMPORTED_TARGET blosc2)
+    target_link_libraries(blosc2_ext PkgConfig::Blosc2)
 else()
     set(STATIC_LIB ON CACHE BOOL "Build a static version of the blosc library.")
     set(SHARED_LIB ON CACHE BOOL "Build a shared library version of the blosc library.")
-- 
2.42.0

