diff --git a/.gitignore b/.gitignore
index b887d00..9c9e095 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,2 @@
-inc/CL/
 inc/EGL/
 inc/KHR/
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e08865..f95df8d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,3 +29,9 @@ target_link_libraries (OpenCL ${CMAKE_DL_LIBS})
 
 enable_testing()
 add_subdirectory (test)
+
+install(TARGETS OpenCL
+        INCLUDES DESTINATION include
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
