diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e08865..5cf2116 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,3 +29,11 @@ target_link_libraries (OpenCL ${CMAKE_DL_LIBS})
 
 enable_testing()
 add_subdirectory (test)
+
+install(TARGETS OpenCL
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
+
+install(DIRECTORY inc/CL DESTINATION include
+        FILES_MATCHING PATTERN "*.h")
