From 3d1b6551e20d35d66af5e24f3b905086fe342674 Mon Sep 17 00:00:00 2001
From: Xylar Asay-Davis <xylarstorm@gmail.com>
Date: Thu, 26 Mar 2020 09:04:38 +0100
Subject: [PATCH] skip installing "if" directory

---
 CMakeLists.txt | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 354ca75..babbefb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -209,15 +209,6 @@ set(
 # install header files
 install(FILES ${haru_HDRS} DESTINATION include)

-# install various files
-install(FILES README CHANGES INSTALL DESTINATION .)
-if(NOT DEVPAK)
-  install(DIRECTORY if DESTINATION .)
-endif(NOT DEVPAK)
-if(DEVPAK)
-  install(FILES ${CMAKE_BINARY_DIR}/libharu.DevPackage DESTINATION .)
-endif(DEVPAK)
-
 # =======================================================================
 # print out some information
 # =======================================================================
--
2.23.0

