--- setup.cfg   2020-10-19 20:12:54.846108000 +0100
+++ setup.cfg   2021-01-07 12:44:31.940000000 +0000
@@ -23,11 +23,14 @@
 license = MIT

 [options]
-scripts = git-filter-repo
 py_modules = git_filter_repo
 python_requires = >= 3.5
 setup_requires = setuptools_scm

+[options.entry_points]
+console_scripts =
+       git-filter-repo = git_filter_repo:main
+
 [bdist_wheel]
 universal = 1

--- git_filter_repo.py  2020-10-18 01:07:18.000000000 +0100
+++ git_filter_repo.py  2021-01-07 12:45:25.310000000 +0000
@@ -3940,11 +3940,14 @@
     print(_("Completely finished after {:.2f} seconds.")
           .format(time.time()-start))

-if __name__ == '__main__':
+def main(args=sys.argv[1:]):
   setup_gettext()
-  args = FilteringOptions.parse_args(sys.argv[1:])
+  args = FilteringOptions.parse_args(args)
   if args.analyze:
     RepoAnalyze.run(args)
   else:
     filter = RepoFilter(args)
     filter.run()
+
+if __name__ == '__main__':
+  main()
