From d9d9382cc9669959f99d44c1c5c330f0b351bf5c Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan.janssen@outlook.com>
Date: Sat, 2 Oct 2021 17:59:15 -0600
Subject: [PATCH] fix setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 83a2b06..bf874c3 100755
--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@
 here = path.abspath(path.dirname(__file__))
 root = path.dirname(here)
 
-readme = path.join(here, 'README.md')
+readme = path.join(here, 'README.rst')
 package_json = path.join(here, 'package.json')
 
 # a workaround when installing locally from git repository with pip install -e .
