diff --git a/bench/perl/AsmGen.pl b/bench/perl/AsmGen.pl
index efec7c32..258de897 100755
--- a/bench/perl/AsmGen.pl
+++ b/bench/perl/AsmGen.pl
@@ -48,6 +48,7 @@ my %OPT;
 my $INPUTFILE;
 my $OUTPUTFILE;
 my $CPP_ARGS='';
+my $CC=$ENV{'CC'};
 
 # Enable warnings within the Parse::RecDescent module.
 $::RD_ERRORS = 1; # Make sure the parser dies when it encounters an error
@@ -269,7 +270,7 @@ _EOGRAMMAR_
 #=======================================
 init();
 print "INFO: Calling cpp with arguments $CPP_ARGS.\n" if ($VERBOSE);
-my $text = `cpp -x assembler-with-cpp $CPP_ARGS $INPUTFILE.pas`;
+my $text = `$CC -E -x assembler-with-cpp $CPP_ARGS $INPUTFILE.pas`;
 
 if ($OPT{p}) {
     open FILE,">$INPUTFILE.Pas";
