--- a/thirdparty/lz4/CMakeLists.txt
+++ b/thirdparty/lz4/CMakeLists.txt
@@ -134,7 +134,12 @@
       LOG_CONFIGURE ON
       LOG_BUILD ON
       LOG_INSTALL ON
-      BUILD_BYPRODUCTS ${lz4_LIBRARY_DIR}/liblz4.a
+      BUILD_BYPRODUCTS
+        ${lz4_LIBRARY_DIR}/liblz4.a
+        ${lz4_INCLUDE_DIR}/lz4.h
+        ${lz4_INCLUDE_DIR}/lz4hc.h
+        ${lz4_INCLUDE_DIR}/lz4frame.h
+        ${lz4_INCLUDE_DIR}/lz4frame_static.h
     )
 endif()
 add_library(lz4 STATIC IMPORTED GLOBAL)
