--- a/thirdparty/lz4/CMakeLists.txt
+++ b/thirdparty/lz4/CMakeLists.txt
@@ -128,7 +128,7 @@
       URL "${CMAKE_CURRENT_SOURCE_DIR}/lz4-1.9.4"
       CONFIGURE_COMMAND ""
         BUILD_COMMAND env ${_lz4_env} BUILD_SHARED=no make -j
-      INSTALL_COMMAND make DESTDIR=${EXTERNAL_BINARY_DIR} BUILD_SHARED=no install
+      INSTALL_COMMAND make DESTDIR=${EXTERNAL_BINARY_DIR} PREFIX=/usr/local BUILD_SHARED=no install
       BUILD_IN_SOURCE ON
       LOG_DOWNLOAD ON
       LOG_CONFIGURE ON
