--- a/thirdparty/rocksdb/rocksdb-8.1.1/db/blob/blob_file_meta.h
+++ b/thirdparty/rocksdb/rocksdb-8.1.1/db/blob/blob_file_meta.h
@@ -6,6 +6,7 @@
 #pragma once

 #include <cassert>
+#include <cstdint>
 #include <iosfwd>
 #include <memory>
 #include <string>
--- a/thirdparty/rocksdb/rocksdb-8.1.1/include/rocksdb/trace_record.h
+++ b/thirdparty/rocksdb/rocksdb-8.1.1/include/rocksdb/trace_record.h
@@ -5,6 +5,7 @@

 #pragma once

+#include <cstdint>
 #include <memory>
 #include <string>
 #include <vector>
