--- a/db/blob/blob_file_meta.h
+++ b/db/blob/blob_file_meta.h
@@ -6,6 +6,7 @@
 #pragma once

 #include <cassert>
+#include <cstdint>
 #include <iosfwd>
 #include <memory>
 #include <string>
--- a/include/rocksdb/trace_record.h
+++ b/include/rocksdb/trace_record.h
@@ -5,6 +5,7 @@

 #pragma once

+#include <cstdint>
 #include <memory>
 #include <string>
 #include <vector>
--- a/include/rocksdb/trace_record_result.h
+++ b/include/rocksdb/trace_record_result.h
@@ -5,6 +5,7 @@

 #pragma once

+#include <cstdint>
 #include <string>
 #include <vector>

--- a/include/rocksdb/utilities/checkpoint.h
+++ b/include/rocksdb/utilities/checkpoint.h
@@ -7,6 +7,7 @@

 #pragma once

+#include <cstdint>
 #include <string>
 #include <vector>

--- a/include/rocksdb/utilities/replayer.h
+++ b/include/rocksdb/utilities/replayer.h
@@ -5,6 +5,7 @@

 #pragma once

+#include <cstdint>
 #include <functional>
 #include <memory>

--- a/port/port_example.h
+++ b/port/port_example.h
@@ -14,6 +14,8 @@

 #pragma once

+#include <cstdint>
+
 namespace ROCKSDB_NAMESPACE {
 namespace port {

--- a/util/aligned_buffer.h
+++ b/util/aligned_buffer.h
@@ -10,6 +10,7 @@

 #include <algorithm>
 #include <cassert>
+#include <cstdint>

 #include "port/port.h"

--- a/utilities/transactions/lock/range/range_tree/lib/util/partitioned_counter.h
+++ b/utilities/transactions/lock/range/range_tree/lib/util/partitioned_counter.h
@@ -53,6 +53,8 @@

 #pragma once

+#include <cstdint>
+
 // Overview: A partitioned_counter provides a counter that can be incremented
 // and the running sum can be read at any time.
 //  We assume that increments are frequent, whereas reading is infrequent.
