From 8df707113be195c61c170b3991df22f1feb3c03d Mon Sep 17 00:00:00 2001
From: Zac Bowling <zac@zacbowling.com>
Date: Fri, 24 Jan 2025 16:23:56 -0800
Subject: [PATCH] Don't vendor openssl

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 29f9d8a..c5df2ef 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,7 +10,7 @@ crate-type = ["cdylib"]
 
 [dependencies]
 futures = "0.3"
-openssl = { version = "0.10", features = ["vendored"] }
+openssl = { version = "0.10" }
 pyo3 = { version = "0.23", features = ["extension-module", "abi3-py38"] }
 rand = "0.8.5"
 reqwest = { version = "0.12", features = ["stream"] }