From 834cb5262f2b69f421638d5ac1e7213e149cc378 Mon Sep 17 00:00:00 2001
From: Adam Ehlers Nyholm Thomsen <aen@atp.dk>
Date: Mon, 5 Aug 2024 14:59:43 +0200
Subject: [PATCH] Got error about missing std::tolower on MSVC

---
 src/util/stringutil.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/util/stringutil.cpp b/src/util/stringutil.cpp
index 5b0de58e7..bdcd4d9de 100644
--- a/src/util/stringutil.cpp
+++ b/src/util/stringutil.cpp
@@ -12,6 +12,7 @@
 
 #include <algorithm>  // for std::transform
 #include <cassert>
+#include <cctype>
 
 /*
 void strRemoveWhitespace(char* str) {
