--- a/src/logger.h
+++ b/src/logger.h
@@ -36,7 +36,7 @@
         inline void info(std::string_view fmt, const Args &...args)
         {
             #ifndef DISABLE_SPDLOG
-            get()->info(fmt, args...);
+            spdlog::info(fmt::runtime(fmt), args...);
             #else
             doPrint(std::cout, fmt, args...);
             #endif
@@ -46,7 +46,7 @@
         inline void warn(std::string_view fmt, const Args &...args)
         {
             #ifndef DISABLE_SPDLOG
-            get()->warn(fmt, args...);
+            spdlog::warn(fmt::runtime(fmt), args...);
             #else
             doPrint(std::cout, fmt, args...);
             #endif
