--- a/src/io.h
+++ b/src/io.h
@@ -29,10 +29,10 @@
 {

     //////////////// IO ////////////////
-    void SaveMesh(const string &filename, Model &mesh);
+    void SaveMesh(const std::string &filename, Model &mesh);
     void SaveConfig(Params params);
-    void SaveOBJ(const string &filename, vector<Model> parts, Params &params);
-    void SaveOBJs(const string &foldername, const string &filename, vector<Model> parts, Params &params);
+    void SaveOBJ(const std::string &filename, vector<Model> parts, Params &params);
+    void SaveOBJs(const std::string &foldername, const std::string &filename, vector<Model> parts, Params &params);
     bool WriteVRML(ofstream &fout, Model mesh);
-    void SaveVRML(const string &fileName, vector<Model>& meshes, Params &params);
+    void SaveVRML(const std::string &fileName, vector<Model>& meshes, Params &params);
 }
--- a/src/process.h
+++ b/src/process.h
@@ -25,7 +25,7 @@
 {
   extern thread_local std::mt19937 random_engine;

-  void DecimateCH(Model &ch, int tgt_pts, string apx_mode);
+  void DecimateCH(Model &ch, int tgt_pts, std::string apx_mode);
   void DecimateConvexHulls(vector<Model> &cvxs, Params &params);
   void MergeCH(Model &ch1, Model &ch2, Model &ch, Params &params);
   double MergeConvexHulls(Model &m, vector<Model> &meshs, vector<Model> &cvxs, Params &params, double epsilon = 0.02, double threshold = 0.01);
--- a/main.cpp
+++ b/main.cpp
@@ -18,7 +18,7 @@
   Params params;

   // Model files
-  string input_model;
+  std::string input_model;
   params.seed = (unsigned)time(NULL);

   // args
@@ -113,7 +113,7 @@
     }
   }

-  string ext;
+  std::string ext;
   if (params.input_model.length() > 4)
   {
     ext = params.input_model.substr(params.input_model.length() - 4);
@@ -187,8 +187,8 @@

   RecoverParts(parts, bbox, rot, params);

-  string objName = regex_replace(params.output_name, regex("wrl"), "obj");
-  string wrlName = regex_replace(params.output_name, regex("obj"), "wrl");
+  std::string objName = regex_replace(params.output_name, regex("wrl"), "obj");
+  std::string wrlName = regex_replace(params.output_name, regex("obj"), "wrl");

   SaveVRML(wrlName, parts, params);
   SaveOBJ(objName, parts, params);
