--- a/python/package/__init__.py
+++ b/python/package/__init__.py
@@ -13,13 +13,23 @@
     c_uint,
     c_char_p,
 )
+import platform
+from ctypes.util import find_library

+sys = platform.system()
+prefix = os.environ['CONDA_PREFIX']
+if sys == 'Windows':
+    lib_dir = os.path.join(prefix, 'Library', 'bin')
+    os.add_dll_directory(lib_dir)
+    lib_name = '_coacd.dll'
+else:
+    lib_dir = os.path.join(prefix, 'lib')
+    lib_name = '_coacd'

-_lib_files = os.listdir(os.path.dirname(os.path.abspath(__file__)))
-for _file in _lib_files:
-    if _file.startswith("lib_coacd"):
-        _lib = ctypes.CDLL(os.path.join(os.path.dirname(os.path.abspath(__file__)), _file))
-
+_lib_path = find_library(lib_name) or os.path.join(lib_dir, lib_name if sys == 'Windows' else f'lib{lib_name}.so')
+if not os.path.exists(_lib_path):
+    raise ImportError(f"Cannot find '{lib_name}' in conda prefix {prefix}")
+_lib = ctypes.CDLL(_lib_path)

 class CoACD_Mesh(ctypes.Structure):
     _fields_ = [
