--- a/python/package/bin/coacd
+++ b/python/package/bin/coacd
@@ -13,7 +13,7 @@
 import argparse
 import coacd

-if __name__ == "__main__":
+def main():
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "-i",
--- a/setup.py
+++ b/setup.py
@@ -102,9 +102,7 @@
     packages=["coacd"],
     python_requires=">=3.9",
     install_requires=["numpy"],
-    ext_modules=[CMakeExtension("coacd")],
-    cmdclass={"build_ext": CMakeBuild, "bdist_wheel": bdist_wheel_abi3},
     zip_safe=False,
     package_dir={"coacd": os.path.join("python/package")},
-    scripts=["python/package/bin/coacd"]
 )
