diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8b3193..a386f99 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,6 @@ if(MSVC)
     # For Microsoft Visual Studio (Windows)
     add_compile_options("/bigobj")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W0 /D_USE_MATH_DEFINES")
-    set(CMAKE_SHARED_LIBRARY_PREFIX "lib")
 else()
     # For non-MSVC (Linux, macOS, etc.)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
