diff --git a/include/libcaer/portable_endian.h b/include/libcaer/portable_endian.h
index ba1a4b4..fe5ff8a 100644
--- a/include/libcaer/portable_endian.h
+++ b/include/libcaer/portable_endian.h
@@ -56,7 +56,7 @@
 #      define be64toh(x) betoh64(x)
 #      define le64toh(x) letoh64(x)
 #elif defined(__WINDOWS__)
-#      include <sys/param.h>
+#      include <windows.h>
 #      include <winsock2.h>
 
 #      if BYTE_ORDER == LITTLE_ENDIAN

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 629035d..d3ab644 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -16,7 +16,11 @@ SET(LIBCAER_SOURCES
 
 # This is often the case due to how we try to have multiple compatible
 # definitions of functions in C.
-SET(LIBCAER_COMPILE_OPTIONS "-Wno-unused-function")
+IF (WIN32)
+       SET(LIBCAER_COMPILE_OPTIONS "")
+ELSE()
+       SET(LIBCAER_COMPILE_OPTIONS "-Wno-unused-function")
+ENDIF()
 
 SET(LIBCAER_LINK_LIBRARIES_PRIVATE PkgConfig::libusb ${BASE_LIBS})
