--- src/libsystemd-network/arp-util.c.orig	2019-10-15 11:47:39.117534233 -0400
+++ src/libsystemd-network/arp-util.c	2019-10-15 11:47:49.785614116 -0400
@@ -12,6 +12,8 @@
 #include "unaligned.h"
 #include "util.h"
 
+#include "linux/bpf_common.h"
+
 int arp_network_bind_raw_socket(int ifindex, be32_t address, const struct ether_addr *eth_mac) {
         struct sock_filter filter[] = {
                 BPF_STMT(BPF_LD + BPF_W + BPF_LEN, 0),                                         /* A <- packet length */
