--- src/coredump/coredump-vacuum.c.orig	2019-10-16 13:07:59.214734707 -0400
+++ src/coredump/coredump-vacuum.c	2019-10-16 13:12:30.328745417 -0400
@@ -17,6 +17,8 @@
 #include "time-util.h"
 #include "user-util.h"
 
+#include "missing_fcntl.h"
+
 #define DEFAULT_MAX_USE_LOWER (uint64_t) (1ULL*1024ULL*1024ULL)           /* 1 MiB */
 #define DEFAULT_MAX_USE_UPPER (uint64_t) (4ULL*1024ULL*1024ULL*1024ULL)   /* 4 GiB */
 #define DEFAULT_KEEP_FREE_UPPER (uint64_t) (4ULL*1024ULL*1024ULL*1024ULL) /* 4 GiB */
