--- src/libsystemd-network/dhcp-network.c.orig	2019-10-15 11:13:20.549996641 -0400
+++ src/libsystemd-network/dhcp-network.c	2019-10-15 11:12:31.313614545 -0400
@@ -19,6 +19,8 @@
 #include "socket-util.h"
 #include "unaligned.h"
 
+#include "linux/bpf_common.h"
+
 static int _bind_raw_socket(int ifindex, union sockaddr_union *link,
                             uint32_t xid, const uint8_t *mac_addr,
                             size_t mac_addr_len,
