--- src/shared/dissect-image.c.orig	2019-10-15 13:17:42.686739421 -0400
+++ src/shared/dissect-image.c	2019-10-15 13:20:19.995498515 -0400
@@ -47,6 +47,8 @@
 #include "user-util.h"
 #include "xattr-util.h"
 
+#include "missing_loop.h"
+
 int probe_filesystem(const char *node, char **ret_fstype) {
         /* Try to find device content type and return it in *ret_fstype. If nothing is found,
          * 0/NULL will be returned. -EUCLEAN will be returned for ambiguous results, and an
