--- src/basic/khash.c.orig	2019-10-14 13:13:18.631817258 -0400
+++ src/basic/khash.c	2019-10-14 13:17:00.977799575 -0400
@@ -30,7 +30,7 @@
                 struct sockaddr sa;
                 struct sockaddr_alg alg;
         } sa = {
-                .alg.salg_family = AF_ALG,
+                .alg.salg_family = PF_ALG,
                 .alg.salg_type = "hash",
                 .alg.salg_name = "sha256", /* a very common algorithm */
         };
@@ -41,7 +41,7 @@
                 _cleanup_close_ int fd1 = -1, fd2 = -1;
                 uint8_t buf[LONGEST_DIGEST+1];
 
-                fd1 = socket(AF_ALG, SOCK_SEQPACKET|SOCK_CLOEXEC, 0);
+                fd1 = socket(PF_ALG, SOCK_SEQPACKET|SOCK_CLOEXEC, 0);
                 if (fd1 < 0) {
                         /* The kernel returns EAFNOSUPPORT if AF_ALG is not supported at all */
                         if (IN_SET(errno, EAFNOSUPPORT, EOPNOTSUPP))
@@ -90,7 +90,7 @@
                 struct sockaddr sa;
                 struct sockaddr_alg alg;
         } sa = {
-                .alg.salg_family = AF_ALG,
+                .alg.salg_family = PF_ALG,
                 .alg.salg_type = "hash",
         };
 
@@ -116,7 +116,7 @@
         if (supported == 0)
                 return -EOPNOTSUPP;
 
-        fd = socket(AF_ALG, SOCK_SEQPACKET|SOCK_CLOEXEC, 0);
+        fd = socket(PF_ALG, SOCK_SEQPACKET|SOCK_CLOEXEC, 0);
         if (fd < 0)
                 return -errno;
 
