--- src/shared/loop-util.c.orig	2019-10-15 13:27:00.825749250 -0400
+++ src/shared/loop-util.c	2019-10-15 13:27:16.089816936 -0400
@@ -11,6 +11,8 @@
 #include "loop-util.h"
 #include "stat-util.h"
 
+#include "missing_loop.h"
+
 int loop_device_make(int fd, int open_flags, LoopDevice **ret) {
         const struct loop_info64 info = {
                 .lo_flags = LO_FLAGS_AUTOCLEAR|LO_FLAGS_PARTSCAN|(open_flags == O_RDONLY ? LO_FLAGS_READ_ONLY : 0),
