--- src/machine/machine-dbus.c.orig	2019-10-15 16:34:44.418350793 -0400
+++ src/machine/machine-dbus.c	2019-10-15 16:35:03.158407429 -0400
@@ -39,6 +39,8 @@
 #include "tmpfile-util.h"
 #include "user-util.h"
 
+#include "missing_syscall.h"
+
 static BUS_DEFINE_PROPERTY_GET_ENUM(property_get_class, machine_class, MachineClass);
 static BUS_DEFINE_PROPERTY_GET2(property_get_state, "s", Machine, machine_get_state, machine_state_to_string);
 
