--- minstall.py.orig	2019-10-17 15:19:03.695228570 -0400
+++ minstall.py	2019-10-17 15:19:42.506929835 -0400
@@ -349,6 +349,7 @@
                 if not self.did_install_something:
                     print('Nothing to install.')
         except PermissionError:
+            return
             if shutil.which('pkexec') is not None and 'PKEXEC_UID' not in os.environ:
                 print('Installation failed due to insufficient permissions.')
                 print('Attempting to use polkit to gain elevated privileges...')
