--- src/basic/missing_fs.h.orig	2019-09-26 11:56:16.447099354 -0400
+++ src/basic/missing_fs.h	2019-10-14 11:44:38.089808560 -0400
@@ -65,3 +65,15 @@
 #ifndef FS_PROJINHERIT_FL
 #define FS_PROJINHERIT_FL 0x20000000
 #endif
+
+#ifndef FS_NOCOW_FL
+#define FS_NOCOW_FL     0x00800000 /* Do not cow file */
+#endif
+
+#ifndef AT_EMPTY_PATH
+#define AT_EMPTY_PATH   0x1000  /* Allow empty relative pathname.  */
+#endif
+
+#ifndef FALLOC_FL_PUNCH_HOLE
+#define FALLOC_FL_PUNCH_HOLE  0x02 /* de-allocates range */
+#endif
