--- src/basic/missing_prctl.h.orig	2019-10-14 13:24:23.357737475 -0400
+++ src/basic/missing_prctl.h	2019-10-15 13:39:33.764317363 -0400
@@ -12,3 +12,37 @@
 #define PR_CAP_AMBIENT_LOWER     3
 #define PR_CAP_AMBIENT_CLEAR_ALL 4
 #endif
+
+/*
+ * Tune up process memory map specifics.
+ */
+#ifndef PR_SET_MM
+#define PR_SET_MM   35
+# define PR_SET_MM_START_CODE   1
+# define PR_SET_MM_END_CODE   2
+# define PR_SET_MM_START_DATA   3
+# define PR_SET_MM_END_DATA   4
+# define PR_SET_MM_START_STACK    5
+# define PR_SET_MM_START_BRK    6
+# define PR_SET_MM_BRK      7
+# define PR_SET_MM_ARG_START    8
+# define PR_SET_MM_ARG_END    9
+# define PR_SET_MM_ENV_START    10
+# define PR_SET_MM_ENV_END    11
+# define PR_SET_MM_AUXV     12
+# define PR_SET_MM_EXE_FILE   13
+# define PR_SET_MM_MAP      14
+# define PR_SET_MM_MAP_SIZE   15
+#endif
+
+#ifndef OOM_SCORE_ADJ_MIN
+#define OOM_SCORE_ADJ_MIN (-1000)
+#endif
+
+#ifndef OOM_SCORE_ADJ_MAX
+#define OOM_SCORE_ADJ_MAX 1000
+#endif
+
+#ifndef PR_SET_NO_NEW_PRIVS
+#define PR_SET_NO_NEW_PRIVS 38
+#endif
