--- src/basic/missing_socket.h.orig	2019-10-14 13:00:17.637171354 -0400
+++ src/basic/missing_socket.h	2019-10-16 12:33:33.457848278 -0400
@@ -62,3 +62,13 @@
 #ifndef IP_TRANSPARENT
 #define IP_TRANSPARENT 19
 #endif
+
+#ifndef PF_ALG
+#define PF_ALG    38  /* Algorithm sockets.  */
+#endif
+
+typedef unsigned short __kernel_sa_family_t;
+
+#ifndef MSG_FASTOPEN
+#define MSG_FASTOPEN 0x20000000
+#endif
