--- src/basic/missing_capability.h.orig	2019-10-16 11:48:35.862081863 -0400
+++ src/basic/missing_capability.h	2019-10-16 11:50:25.409289837 -0400
@@ -10,3 +10,15 @@
 #undef  CAP_LAST_CAP
 #define CAP_LAST_CAP   CAP_AUDIT_READ
 #endif
+
+#ifndef CAP_SYSLOG
+#define CAP_SYSLOG           34
+#endif
+
+#ifndef CAP_WAKE_ALARM
+#define CAP_WAKE_ALARM            35
+#endif
+
+#ifndef CAP_BLOCK_SUSPEND
+#define CAP_BLOCK_SUSPEND    36
+#endif
