--- src/basic/linux/netlink.h.orig	2019-09-24 14:15:49.172054121 -0400
+++ src/basic/linux/netlink.h	2019-10-15 15:33:19.937940329 -0400
@@ -6,6 +6,8 @@
 #include <linux/socket.h> /* for __kernel_sa_family_t */
 #include <linux/types.h>
 
+#include "missing_socket.h"
+
 #define NETLINK_ROUTE		0	/* Routing/device hook				*/
 #define NETLINK_UNUSED		1	/* Unused number				*/
 #define NETLINK_USERSOCK	2	/* Reserved for user mode socket protocols 	*/
