--- src/nspawn/nspawn.c.orig	2019-09-24 12:01:19.696185772 -0400
+++ src/nspawn/nspawn.c	2019-10-16 15:18:27.423164506 -0400
@@ -4655,8 +4655,10 @@
                         if (IN_SET(rl, RLIMIT_NPROC, RLIMIT_SIGPENDING)) {
                                 /* For these two let's read the limits off PID 1. See above for an explanation. */
 
+                                /* no prlimit available
                                 if (prlimit(1, rl, NULL, &buffer) < 0)
                                         return log_error_errno(errno, "Failed to read resource limit RLIMIT_%s of PID 1: %m", rlimit_to_string(rl));
+                                */
 
                                 v = &buffer;
                         } else
