--- src/nspawn/nspawn-patch-uid.c.orig	2019-10-15 16:26:29.240903731 -0400
+++ src/nspawn/nspawn-patch-uid.c	2019-10-15 16:26:45.136948144 -0400
@@ -320,8 +320,7 @@
         }
 
         /* Also, if we hit a read-only file system, then don't bother, skip the whole subtree */
-        if ((sfs.f_flags & ST_RDONLY) ||
-            access_fd(fd, W_OK) == -EROFS)
+        if (access_fd(fd, W_OK) == -EROFS)
                 goto read_only;
 
         if (S_ISDIR(st->st_mode)) {
