--- src/libsystemd/sd-event/sd-event.c.orig	2019-10-15 10:58:51.319137635 -0400
+++ src/libsystemd/sd-event/sd-event.c	2019-10-15 11:44:49.880266831 -0400
@@ -26,6 +26,9 @@
 #include "strxcpyx.h"
 #include "time-util.h"
 
+#include "missing_time.h"
+#include "missing_inotify.h"
+
 #define DEFAULT_ACCURACY_USEC (250 * USEC_PER_MSEC)
 
 static const char* const event_source_type_table[_SOURCE_EVENT_SOURCE_TYPE_MAX] = {
