--- src/timesync/timesyncd-manager.c.orig	2019-10-16 13:41:43.078709363 -0400
+++ src/timesync/timesyncd-manager.c	2019-10-16 13:44:53.505261336 -0400
@@ -297,9 +297,11 @@
                 break;
         }
 
+        /* clock_adjtime not available on older versions of linux
         r = clock_adjtime(CLOCK_REALTIME, &tmx);
         if (r < 0)
                 return -errno;
+        */
 
         /* If touch fails, there isn't much we can do. Maybe it'll work next time. */
         (void) touch("/var/lib/systemd/timesync/clock");
