--- src/udev/v4l_id/v4l_id.c.orig	2019-10-15 14:06:07.995148380 -0400
+++ src/udev/v4l_id/v4l_id.c	2019-10-15 14:25:27.523347237 -0400
@@ -29,6 +29,8 @@
 #include "fd-util.h"
 #include "util.h"
 
+#include "missing_videodev2.h"
+
 int main(int argc, char *argv[]) {
         static const struct option options[] = {
                 { "help", no_argument, NULL, 'h' },
@@ -68,10 +70,7 @@
                 printf("ID_V4L_VERSION=2\n");
                 printf("ID_V4L_PRODUCT=%s\n", v2cap.card);
                 printf("ID_V4L_CAPABILITIES=:");
-                if (v2cap.capabilities & V4L2_CAP_DEVICE_CAPS)
-                        capabilities = v2cap.device_caps;
-                else
-                        capabilities = v2cap.capabilities;
+                capabilities = v2cap.capabilities;
                 if ((capabilities & V4L2_CAP_VIDEO_CAPTURE) > 0 ||
                     (capabilities & V4L2_CAP_VIDEO_CAPTURE_MPLANE) > 0)
                         printf("capture:");
