diff --git a/setup.py b/setup.py
index 283f03d..3fd0155 100755
--- a/setup.py
+++ b/setup.py
@@ -18,7 +18,7 @@ install_requires=['numpy>=1.14.3', 'dpdata>=0.1.3', 'pymatgen>=2017.9.1', 'ase',
 setuptools.setup(
     name=NAME,
     version_format='{tag}.dev{commitcount}+{gitsha}',
-    setup_requires=['setuptools-git-version'],
+    version='0.3.0',
     author="Han Wang",
     author_email="wang_han@iapcm.ac.cn",
     description="DPGen: The deep potential generator",
