From 3b288e1bbbddcff5457e84f1eaa7b2d4c1e11407 Mon Sep 17 00:00:00 2001
From: Casey Clements <casey.clements@continuum.io>
Date: Wed, 10 Oct 2018 21:44:56 -0400
Subject: [PATCH] fix imports

---
 pennies/calculators/__init__.py | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/pennies/calculators/__init__.py b/pennies/calculators/__init__.py
index 482d915..46f4a41 100644
--- a/pennies/calculators/__init__.py
+++ b/pennies/calculators/__init__.py
@@ -1,5 +1,6 @@
 # import modules that define dispatched methods, eg presentvalue, par_rate
-import assets
-import payments
-import swaps
-import trades
\ No newline at end of file
+from . import assets
+from . import payments
+from . import swaps
+from . import trades
+
-- 
2.14.3 (Apple Git-98)

