diff --git a/src/libusockets.h b/src/libusockets.h
index f5a6c12..b04e0b3 100644
--- a/src/libusockets.h
+++ b/src/libusockets.h
@@ -322,14 +322,6 @@ void us_socket_remote_address(int ssl, struct us_socket_t *s, char *buf, int *le
 #endif
 
 /* Decide what eventing system to use by default */
-#if !defined(LIBUS_USE_EPOLL) && !defined(LIBUS_USE_LIBUV) && !defined(LIBUS_USE_GCD) && !defined(LIBUS_USE_KQUEUE) && !defined(LIBUS_USE_ASIO)
-#if defined(_WIN32)
 #define LIBUS_USE_LIBUV
-#elif defined(__APPLE__) || defined(__FreeBSD__)
-#define LIBUS_USE_KQUEUE
-#else
-#define LIBUS_USE_EPOLL
-#endif
-#endif
 
 #endif // LIBUSOCKETS_H
