diff --git a/setup.py b/setup.py
index 95bf706..ca6f2e1 100644
--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@ import subprocess
 from distutils.ccompiler import new_compiler
 from setuptools import setup, Extension
 
-import numpy
+import os
 try:
     from Cython.Distutils import build_ext
     setup_kwargs = {'cmdclass': {'build_ext': build_ext}}
@@ -51,7 +51,7 @@ setup(name='openmoltools',
       author_email='kyleabeauchamp@gmail.com',
       description=DOCLINES[0],
       long_description="\n".join(DOCLINES[2:]),
-      version=__version__,
+      version=os.environ["PKG_VERSION"],
       license='MIT',
       url='http://github.com/choderalab/openmoltools',
       platforms=['Linux', 'Mac OS-X', 'Unix'],
