diff --git a/pymorton/pymorton.py b/pymorton/pymorton.py
index 895d9c7..d787307 100644
--- a/pymorton/pymorton.py
+++ b/pymorton/pymorton.py
@@ -89,7 +89,7 @@ def __unpart1by2_64(n):
     return n
 
 
-if getattr(sys, 'maxint', 0) and sys.maxint <= 2 ** 31 - 1:
+if sys.version_info[0] == 2 and getattr(sys, 'maxint', 0) and sys.maxint <= 2 ** 31 - 1:
     __part1by1 = __part1by1_32
     __part1by2 = __part1by2_32
     __unpart1by1 = __unpart1by1_32

