From 8eaaab7e567542d41973a35661b8f6080a2d71e7 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 27 Mar 2020 11:44:47 +0000
Subject: [PATCH] Patch for C macros

---
 CMakeLists.txt   | 1 +
 src/robin_hood.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 61296d3..fa72e66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,7 @@ if(UNIX AND NOT APPLE)
     if(CMAKE_CXX_COMPILER STREQUAL "icpc")  
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fast -xCASCADELAKE -DMKL_ILP64 -m64 -static-intel")
     else()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS")
         set(CMAKE_LD_FLAGS "${CMAKE_LDFLAGS} -Wl,--as-needed")
     endif()
 endif()
diff --git a/src/robin_hood.h b/src/robin_hood.h
index 7ad9dae..ef1c73b 100644
--- a/src/robin_hood.h
+++ b/src/robin_hood.h
@@ -41,6 +41,7 @@
 
 #include <algorithm>
 #include <cstdlib>
+#include <cstdint>
 #include <cstring>
 #include <functional>
 #include <stdexcept>
-- 
2.25.0

