From f2321968ab5d64edbcc9d76e48aeaa4b2108b173 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Mon, 30 Mar 2020 14:50:06 +0100
Subject: [PATCH] Replace cmake with make

---
 setup.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 879fb0e..9aa387c 100644
--- a/setup.py
+++ b/setup.py
@@ -79,8 +79,11 @@ class CMakeBuild(build_ext):
                                                               self.distribution.get_version())
         if not os.path.exists(self.build_temp):
             os.makedirs(self.build_temp)
-        subprocess.check_call(['cmake', ext.sourcedir] + cmake_args, cwd=self.build_temp, env=env)
-        subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
+        
+        subprocess.check_call(['make', 'python'], cwd=ext.sourcedir + '/src', env=env)
+        subprocess.check_call(['make', 'install_python', 'PYTHON_LIB_PATH=' + extdir], cwd=ext.sourcedir + '/src', env=env)
+        #subprocess.check_call(['cmake', ext.sourcedir] + cmake_args, cwd=self.build_temp, env=env)
+        #subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
 
 
 here = path.abspath(path.dirname(__file__))
-- 
2.25.0

