From 1ebdcbd439010aeae3a4d280ff2bdd8ab4b20daf Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 27 Mar 2020 11:04:49 +0000
Subject: [PATCH] Add cint types to robin hood hash includes

---
 src/robin_hood.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/robin_hood.h b/src/robin_hood.h
index 7ad9dae..ef1c73b 100644
--- a/src/robin_hood.h
+++ b/src/robin_hood.h
@@ -41,6 +41,7 @@
 
 #include <algorithm>
 #include <cstdlib>
+#include <cstdint>
 #include <cstring>
 #include <functional>
 #include <stdexcept>
-- 
2.25.0

