From 0d565562974a04d9dc995ee98432566a93222029 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 27 Mar 2020 12:57:16 +0000
Subject: [PATCH] Update CMakeLists with improved CUDA compile directives

---
 CMakeLists.txt  | 4 +++-
 environment.yml | 5 ++---
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 61296d3..9950fa1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,7 @@ if(UNIX AND NOT APPLE)
     if(CMAKE_CXX_COMPILER STREQUAL "icpc")  
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fast -xCASCADELAKE -DMKL_ILP64 -m64 -static-intel")
     else()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS")
         set(CMAKE_LD_FLAGS "${CMAKE_LDFLAGS} -Wl,--as-needed")
     endif()
 endif()
@@ -56,6 +57,7 @@ if(CMAKE_CUDA_COMPILER)
     set_property(TARGET "${TARGET_NAME}_CUDA" 
                  PROPERTY POSITION_INDEPENDENT_CODE ON
                  CUDA_SEPARABLE_COMPILATION ON
+                 CUDA_RESOLVE_DEVICE_SYMBOLS ON   # device link with nvcc (ar alone insufficient)
                  CUDA_VISIBILITY_PRESET "hidden")
 else()
     message(STATUS "CUDA not found, compiling CPU code only")
@@ -84,4 +86,4 @@ if(CMAKE_CUDA_COMPILER)
 endif()
 target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module Eigen3::Eigen 
                                                ${ARMADILLO_LIBRARIES} 
-                                               z bz2 hdf5_cpp hdf5 openblas m dl)
\ No newline at end of file
+                                               z bz2 hdf5_cpp hdf5 openblas gfortran m dl)
diff --git a/environment.yml b/environment.yml
index 258c51d..9e32bd9 100644
--- a/environment.yml
+++ b/environment.yml
@@ -11,10 +11,9 @@ dependencies:
   - cmake >= 3.12
   - pybind11
   - zlib
-  - bzip2
   - eigen
   - hdf5
   - h5py
   - openblas
-  - lapack
-  - armadillo
\ No newline at end of file
+  - armadillo
+  - libgfortan-ng
-- 
2.25.0

