From a4473f733ebe92f986987a10bbe7feae8e779425 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 27 Mar 2020 15:36:28 +0000
Subject: [PATCH] Force use of make not gmake

---
 CMakeLists.txt | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e371cd..8c3b196 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,6 +21,8 @@ if(UNIX AND NOT APPLE)
     endif()
 endif()
 
+set(CMAKE_MAKE_PROGRAM make CACHE FILEPATH "" FORCE)
+
 # Add libraries
 find_package(pybind11 REQUIRED)
 find_package (Eigen3 3.3 REQUIRED NO_MODULE)
@@ -84,6 +86,15 @@ set_target_properties("${TARGET_NAME}" PROPERTIES
 if(CMAKE_CUDA_COMPILER)
     target_link_libraries("${TARGET_NAME}" PRIVATE "${TARGET_NAME}_CUDA")
 endif()
-target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module Eigen3::Eigen 
+
+# use gfortran explicitly on Linux
+if(UNIX AND NOT APPLE)
+    target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module Eigen3::Eigen 
                                                ${ARMADILLO_LIBRARIES} 
-                                               z bz2 hdf5_cpp hdf5 openblas gfortran m dl)
+                                               z hdf5_cpp hdf5 openblas gfortran m dl)
+else()
+    target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module Eigen3::Eigen 
+                                               ${ARMADILLO_LIBRARIES} 
+                                               z hdf5_cpp hdf5 openblas m dl)
+endif()
+
-- 
2.25.0

