From 60e0a38ff49b2c640f2d8a796bd7467cc1202210 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 27 Mar 2020 18:18:02 +0000
Subject: [PATCH] Use Makefile to build extension

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 879fb0e..9f865c9 100644
--- a/setup.py
+++ b/setup.py
@@ -116,7 +116,6 @@ setup(
     },
     install_requires=['numpy',
                       'h5py'],
-    ext_modules=[CMakeExtension('pp_sketchlib')],
-    cmdclass=dict(build_ext=CMakeBuild),
+    package_data={'pp_sketchlib': ['*.so', 'pp_sketchlib/*.so']},
     zip_safe=False
 )
-- 
2.25.0

