From 016755c48d3d00c59c9e58e88c9ab6321285b304 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 27 Mar 2020 11:58:14 +0000
Subject: [PATCH 2/2] Add gfortran to linker

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fa72e66..c1528ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,4 +85,4 @@ if(CMAKE_CUDA_COMPILER)
 endif()
 target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module Eigen3::Eigen 
                                                ${ARMADILLO_LIBRARIES} 
-                                               z bz2 hdf5_cpp hdf5 openblas m dl)
\ No newline at end of file
+                                               z bz2 hdf5_cpp hdf5 openblas gfortran m dl)
\ No newline at end of file
-- 
2.25.0

