From 738a9b72973a0279517eae92786828a6a6241361 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 27 Mar 2020 18:36:17 +0000
Subject: [PATCH 1/2] Add C macros to Makefile

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 57ceded..ec4d53d 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,5 +1,5 @@
 
-CXXFLAGS+=-Wall -Wextra -std=c++14 -flto -fno-fat-lto-objects -fPIC -fvisibility=hidden
+CXXFLAGS+=-Wall -Wextra -std=c++14 -flto -fno-fat-lto-objects -fPIC -fvisibility=hidden -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS
 ifdef DEBUG
   CXXFLAGS+= -O0 -g
 else ifdef PROFILE
-- 
2.25.0


From 9e447fd3bd6891fc84d06e3d7c92ac38213ded0b Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Sun, 29 Mar 2020 19:42:34 +0100
Subject: [PATCH 2/2] Single optimisation option to nvcc

---
 src/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index ec4d53d..a510c56 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -27,7 +27,7 @@ LDFLAGS+= -L$(LIBLOC)/lib
 LDLIBS+=-lz -lhdf5_cpp -lhdf5 -lopenblas -llapack -lcudadevrt -lcudart_static $(LALIBS)
 
 CUDA_LDFLAGS =-L$(LIBLOC)/lib -L${CUDA_HOME}/targets/x86_64-linux/lib/stubs -L${CUDA_HOME}/targets/x86_64-linux/lib
-CUDAFLAGS =-Xcompiler -fPIC -Xptxas -dlcm=ca --cudart static --relocatable-device-code=true -gencode arch=compute_35,code=compute_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_75,code=sm_75 -O3
+CUDAFLAGS =-Xcompiler -fPIC -Xptxas -dlcm=ca --cudart static --relocatable-device-code=true -gencode arch=compute_35,code=compute_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_75,code=sm_75
 
 # python specific options
 python: CPPFLAGS += -DGPU_AVAILABLE -DPYTHON_EXT -DNDEBUG -Dpp_sketchlib_EXPORTS -DVERSION_INFO=\"1.2.0\" $(shell python3 -m pybind11 --includes)
@@ -53,7 +53,7 @@ read_test: $(SKETCH_OBJS) read_test.o
 	$(LINK.cpp) $^ -o $@
 
 python: $(SKETCH_OBJS) $(CUDA_OBJS) sketchlib_bindings.o
-	nvcc $(CUDAFLAGS) $(CUDA_LDFLAGS) -Wno-deprecated-gpu-targets -shared -dlink $^ -o device_link.o -Xnvlink $(LDLIBS)
+	nvcc $(CUDAFLAGS) $(CUDA_LDFLAGS) -O2 -Wno-deprecated-gpu-targets -shared -dlink $^ -o device_link.o -Xnvlink $(LDLIBS)
 	$(LINK.cpp) $(CUDA_LDFLAGS) $(LDFLAGS) -shared $^ device_link.o -o pp_sketchlib$(shell python3-config --extension-suffix) $(LDLIBS)
 
 dist.cu.o:
-- 
2.25.0

