From 644871c8351667919999f616916cfe9249529ec5 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 27 Mar 2020 12:40:51 +0000
Subject: [PATCH 3/3] Try making CUDA link dynamic

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c1528ca..f703bd9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,7 +52,7 @@ if(CMAKE_CUDA_COMPILER)
     set(CMAKE_CUDA_FLAGS "${CUDA_OPTS}")
     
     add_compile_definitions(GPU_AVAILABLE)
-    add_library("${TARGET_NAME}_CUDA" STATIC src/dist.cu)
+    add_library("${TARGET_NAME}_CUDA" SHARED src/dist.cu)
     # target_include_directories("${TARGET_NAME}_CUDA" PRIVATE "${EIGEN3_INCLUDE_DIR}")
     set_property(TARGET "${TARGET_NAME}_CUDA" 
                  PROPERTY POSITION_INDEPENDENT_CODE ON
@@ -85,4 +85,4 @@ if(CMAKE_CUDA_COMPILER)
 endif()
 target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module Eigen3::Eigen 
                                                ${ARMADILLO_LIBRARIES} 
-                                               z bz2 hdf5_cpp hdf5 openblas gfortran m dl)
\ No newline at end of file
+                                               z bz2 hdf5_cpp hdf5 openblas gfortran m dl)
-- 
2.25.0

