diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 756df376..ec5475f7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -282,7 +282,7 @@ if (BUILD_LIBRARY)
     SOVERSION ${Ledger_VERSION_MAJOR})
 
   add_executable(ledger main.cc global.cc)
-  target_link_libraries(ledger libledger)
+  target_link_libraries(ledger libledger rt)
   if (CMAKE_SYSTEM_NAME STREQUAL Darwin AND HAVE_BOOST_PYTHON)
     target_link_libraries(ledger ${PYTHON_LIBRARIES})
   endif()
diff --git a/test/unit/CMakeLists.txt b/test/unit/CMakeLists.txt
index 3611b00a..b36b6ba3 100644
--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -1,5 +1,5 @@
 macro(add_ledger_test _name)
-  target_link_libraries(${_name} libledger)
+  target_link_libraries(${_name} libledger rt)
   add_test(Ledger${_name} ${PROJECT_BINARY_DIR}/${_name})
   set_tests_properties(Ledger${_name}
     PROPERTIES ENVIRONMENT "TZ=${Ledger_TEST_TIMEZONE}")
diff --git a/test/unit/t_amount.cc b/test/unit/t_amount.cc
index b82de510..6c45250e 100644
--- a/test/unit/t_amount.cc
+++ b/test/unit/t_amount.cc
@@ -1,5 +1,6 @@
 #define BOOST_TEST_DYN_LINK
 #define BOOST_TEST_MODULE math
+#include <boost/test/included/unit_test.hpp>
 #include <boost/test/unit_test.hpp>
 
 #include <system.hh>
diff --git a/test/unit/t_times.cc b/test/unit/t_times.cc
index c6251664..bb050c28 100644
--- a/test/unit/t_times.cc
+++ b/test/unit/t_times.cc
@@ -1,5 +1,6 @@
 #define BOOST_TEST_DYN_LINK
 #define BOOST_TEST_MODULE util
+#include <boost/test/included/unit_test.hpp>
 #include <boost/test/unit_test.hpp>
 
 #include <system.hh>
