diff --git a/setup.py b/setup.py
index c8debdc..fc21def 100644
--- a/setup.py
+++ b/setup.py
@@ -41,7 +41,6 @@ setup(
     url="https://github.com/abourget/gevent-socketio",
     download_url="https://github.com/abourget/gevent-socketio",
     install_requires=get_reqs('pip-requirements.txt'),
-    setup_requires=('versiontools >= 1.7'),
     cmdclass = {'test': PyTest},
     tests_require=get_reqs('pip-requirements-test.txt'),
     packages=find_packages(exclude=["examples", "tests"]),
