diff --git a/CMake/C17Config.cmake b/CMake/C17Config.cmake
index 3a24399cf..812142f55 100644
--- a/CMake/C17Config.cmake
+++ b/CMake/C17Config.cmake
@@ -24,7 +24,9 @@ check_cxx_compiler_flag("-fPIC" COMPILER_SUPPORTS_FPIC)
 if(COMPILER_SUPPORTS_FPIC)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
 endif()
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
+if(NOT WIN32)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
+endif()
 # ==============================================================================
 include(CheckSymbolExists)
 check_symbol_exists("fseek64" "stdio.h" HAVE_FSEEK64)
diff --git a/modules/api_nelson/CMakeLists.txt b/modules/api_nelson/CMakeLists.txt
index d2f0269ed..b7d5804b8 100644
--- a/modules/api_nelson/CMakeLists.txt
+++ b/modules/api_nelson/CMakeLists.txt
@@ -14,6 +14,9 @@ file(GLOB API_NELSON_SRC "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/*.cpp")
 list(FILTER API_NELSON_SRC EXCLUDE REGEX "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp") 
 # ==============================================================================
 add_library(${module_library_name} ${API_NELSON_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSAPI_NELSON_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/characters_encoding/CMakeLists.txt b/modules/characters_encoding/CMakeLists.txt
index 0db31d32b..8e125275b 100644
--- a/modules/characters_encoding/CMakeLists.txt
+++ b/modules/characters_encoding/CMakeLists.txt
@@ -14,6 +14,9 @@ list(FILTER CHARACTERS_ENCODING_SRC EXCLUDE REGEX
 "${CMAKE_CURRENT_SOURCE_DIR}/no-src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${CHARACTERS_ENCODING_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSCHARACTERS_ENCODING_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/commons/CMakeLists.txt b/modules/commons/CMakeLists.txt
index 0e59910cc..f9f1b5943 100644
--- a/modules/commons/CMakeLists.txt
+++ b/modules/commons/CMakeLists.txt
@@ -16,6 +16,9 @@ list(FILTER COMMONS_SRC EXCLUDE REGEX
 "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${COMMONS_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSCOMMONS_EXPORTS")
+endif()
 target_compile_definitions(${module_library_name} PRIVATE FMT_EXPORT)
 set(COMMONS_INCLUDE_DIRS
     ${CMAKE_CURRENT_SOURCE_DIR}/src/include
diff --git a/modules/core/CMakeLists.txt b/modules/core/CMakeLists.txt
index fe2b5a9dd..b25ddb44e 100644
--- a/modules/core/CMakeLists.txt
+++ b/modules/core/CMakeLists.txt
@@ -14,6 +14,9 @@ list(FILTER CORE_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${CORE_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSCORE_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/display_format/CMakeLists.txt b/modules/display_format/CMakeLists.txt
index 458c8da7c..5714cc810 100644
--- a/modules/display_format/CMakeLists.txt
+++ b/modules/display_format/CMakeLists.txt
@@ -14,6 +14,9 @@ file(GLOB DISPLAY_FORMAT_SRC "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/*.cpp")
 list(FILTER DISPLAY_FORMAT_SRC EXCLUDE REGEX "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp") 
 # ==============================================================================
 add_library(${module_library_name} ${DISPLAY_FORMAT_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSDISPLAY_FORMAT_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/elementary_mathematics/CMakeLists.txt b/modules/elementary_mathematics/CMakeLists.txt
index 7b38614d9..15e6c62fb 100644
--- a/modules/elementary_mathematics/CMakeLists.txt
+++ b/modules/elementary_mathematics/CMakeLists.txt
@@ -18,6 +18,9 @@ file(GLOB ELEMENTARY_MATHEMATICS_SRC
 list(FILTER ELEMENTARY_MATHEMATICS_SRC EXCLUDE REGEX "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp") 
 # ==============================================================================
 add_library(${module_library_name} ${ELEMENTARY_MATHEMATICS_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSELEMENTARY_MATHEMATICS_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/engine/CMakeLists.txt b/modules/engine/CMakeLists.txt
index 8430c9ecc..750d7d74e 100644
--- a/modules/engine/CMakeLists.txt
+++ b/modules/engine/CMakeLists.txt
@@ -14,6 +14,9 @@ list(FILTER ENGINE_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${ENGINE_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSENGINE_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/error_manager/CMakeLists.txt b/modules/error_manager/CMakeLists.txt
index da6c6d358..b31b4c6be 100644
--- a/modules/error_manager/CMakeLists.txt
+++ b/modules/error_manager/CMakeLists.txt
@@ -14,6 +14,9 @@ file(GLOB ERROR_MANAGER_SRC "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/*.cpp")
 list(FILTER ERROR_MANAGER_SRC EXCLUDE REGEX "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp") 
 # ==============================================================================
 add_library(${module_library_name} ${ERROR_MANAGER_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSERROR_MANAGER_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/files_folders_functions/CMakeLists.txt b/modules/files_folders_functions/CMakeLists.txt
index 1586d39e7..705842552 100644
--- a/modules/files_folders_functions/CMakeLists.txt
+++ b/modules/files_folders_functions/CMakeLists.txt
@@ -15,6 +15,9 @@ list(FILTER FILES_FOLDERS_FUNCTIONS_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${FILES_FOLDERS_FUNCTIONS_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSFILES_FOLDERS_FUNCTIONS_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/functions_manager/CMakeLists.txt b/modules/functions_manager/CMakeLists.txt
index 889ed4527..8b29ba4a0 100644
--- a/modules/functions_manager/CMakeLists.txt
+++ b/modules/functions_manager/CMakeLists.txt
@@ -16,6 +16,9 @@ list(FILTER FUNCTIONS_MANAGER EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${FUNCTIONS_MANAGER})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSFUNCTIONS_MANAGER_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/i18n/CMakeLists.txt b/modules/i18n/CMakeLists.txt
index 097f79542..babec4d2d 100644
--- a/modules/i18n/CMakeLists.txt
+++ b/modules/i18n/CMakeLists.txt
@@ -14,6 +14,9 @@ list(FILTER I18N_SRC EXCLUDE REGEX
 "${CMAKE_CURRENT_SOURCE_DIR}/no-src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${I18N_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSI18N_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/interpreter/CMakeLists.txt b/modules/interpreter/CMakeLists.txt
index e4178a928..4c0dd6172 100644
--- a/modules/interpreter/CMakeLists.txt
+++ b/modules/interpreter/CMakeLists.txt
@@ -15,6 +15,9 @@ file(GLOB INTERPRETER_SRC "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/*.cpp"
 list(FILTER INTERPRETER_SRC EXCLUDE REGEX "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp") 
 # ==============================================================================
 add_library(${module_library_name} ${INTERPRETER_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSINTERPRETER_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/localization/CMakeLists.txt b/modules/localization/CMakeLists.txt
index 6e0c5104c..a0cc194a4 100644
--- a/modules/localization/CMakeLists.txt
+++ b/modules/localization/CMakeLists.txt
@@ -14,6 +14,9 @@ list(FILTER LOCALIZATION_SRC EXCLUDE REGEX
 "${CMAKE_CURRENT_SOURCE_DIR}/no-src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${LOCALIZATION_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSLOCALIZATION_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/memory_manager/CMakeLists.txt b/modules/memory_manager/CMakeLists.txt
index 02526c43d..c555623aa 100644
--- a/modules/memory_manager/CMakeLists.txt
+++ b/modules/memory_manager/CMakeLists.txt
@@ -14,6 +14,9 @@ list(FILTER MEMORY_MANAGER_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${MEMORY_MANAGER_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSLOCALIZATION_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/nelson_manager/CMakeLists.txt b/modules/nelson_manager/CMakeLists.txt
index bebc2f5a1..4f2d1f403 100644
--- a/modules/nelson_manager/CMakeLists.txt
+++ b/modules/nelson_manager/CMakeLists.txt
@@ -17,6 +17,9 @@ list(FILTER NELSON_MANAGER_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${NELSON_MANAGER_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSNELSON_MANAGER_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/operators/CMakeLists.txt b/modules/operators/CMakeLists.txt
index 9b51c1dd2..a98214529 100644
--- a/modules/operators/CMakeLists.txt
+++ b/modules/operators/CMakeLists.txt
@@ -15,6 +15,9 @@ list(FILTER OPERATORS_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${OPERATORS_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSOPERATORS_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/os_functions/CMakeLists.txt b/modules/os_functions/CMakeLists.txt
index 46f875dd0..d36b36674 100644
--- a/modules/os_functions/CMakeLists.txt
+++ b/modules/os_functions/CMakeLists.txt
@@ -14,6 +14,9 @@ file(GLOB OS_FUNCTIONS_SRC "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/*.cpp")
 list(FILTER OS_FUNCTIONS_SRC EXCLUDE REGEX "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp") 
 # ==============================================================================
 add_library(${module_library_name} ${OS_FUNCTIONS_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSOS_FUNCTIONS_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/profiler/CMakeLists.txt b/modules/profiler/CMakeLists.txt
index f2bb3684e..8057a1d50 100644
--- a/modules/profiler/CMakeLists.txt
+++ b/modules/profiler/CMakeLists.txt
@@ -14,6 +14,9 @@ list(FILTER PROFILER_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${PROFILER_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSPROFILER_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/stream_manager/CMakeLists.txt b/modules/stream_manager/CMakeLists.txt
index 184bbc1a2..f8bb32674 100644
--- a/modules/stream_manager/CMakeLists.txt
+++ b/modules/stream_manager/CMakeLists.txt
@@ -15,6 +15,9 @@ list(FILTER STREAM_MANAGER_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${STREAM_MANAGER_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSSTREAM_MANAGER_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/terminal/CMakeLists.txt b/modules/terminal/CMakeLists.txt
index 5ab6eaae5..d5b50bafe 100644
--- a/modules/terminal/CMakeLists.txt
+++ b/modules/terminal/CMakeLists.txt
@@ -16,6 +16,9 @@ list(FILTER TERMINAL_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${TERMINAL_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSTERMINAL_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/text_completion/CMakeLists.txt b/modules/text_completion/CMakeLists.txt
index 85a87018a..8892c7c84 100644
--- a/modules/text_completion/CMakeLists.txt
+++ b/modules/text_completion/CMakeLists.txt
@@ -15,6 +15,9 @@ list(FILTER TEXT_COMPLETION_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${TEXT_COMPLETION_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSTEXT_COMPLETION_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
diff --git a/modules/types/CMakeLists.txt b/modules/types/CMakeLists.txt
index 161a09417..a33feb0d6 100644
--- a/modules/types/CMakeLists.txt
+++ b/modules/types/CMakeLists.txt
@@ -14,6 +14,9 @@ list(FILTER TYPES_SRC EXCLUDE REGEX
      "${CMAKE_CURRENT_SOURCE_DIR}/src/cpp/dllMain.cpp")
 # ==============================================================================
 add_library(${module_library_name} ${TYPES_SRC})
+if(WIN32)
+    target_compile_definitions(${module_library_name} PUBLIC "NLSTYPES_EXPORTS")
+endif()
 target_include_directories(
     ${module_library_name}
     PUBLIC
