diff --git a/modules/interpreter/src/include/LocalFunctionsTable.hpp b/modules/interpreter/src/include/LocalFunctionsTable.hpp
index 9a1f5f48b..46e6c667a 100644
--- a/modules/interpreter/src/include/LocalFunctionsTable.hpp
+++ b/modules/interpreter/src/include/LocalFunctionsTable.hpp
@@ -11,11 +11,12 @@
 //=============================================================================
 #include <utility>
 #include <string>
+#include "nlsInterpreter_exports.h"
 #include "FunctionDef.hpp"
 //=============================================================================
 namespace Nelson {
 //=============================================================================
-class LocalFunctionsTable
+class NLSINTERPRETER_IMPEXP LocalFunctionsTable
 {
 private:
     void* cachedLocalFunctionsTable;
diff --git a/modules/interpreter/src/include/VariablesTable.hpp b/modules/interpreter/src/include/VariablesTable.hpp
index 96c103405..53ce4743a 100644
--- a/modules/interpreter/src/include/VariablesTable.hpp
+++ b/modules/interpreter/src/include/VariablesTable.hpp
@@ -10,11 +10,12 @@
 #pragma once
 //=============================================================================
 #include <string>
+#include "nlsInterpreter_exports.h"
 #include "ArrayOf.hpp"
 #include "Types.hpp"
 //=============================================================================
 namespace Nelson {
-class VariablesTable
+class NLSINTERPRETER_IMPEXP VariablesTable
 {
     //=============================================================================
 private:
