--- CMakeLists.txt	2019-11-07 01:39:44.868380641 -0600
+++ CMakeLists.txt	2019-11-07 01:40:04.560737080 -0600
@@ -72,9 +72,10 @@
 endif()
 
 if(UNIX)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fno-rtti -Wall")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
   add_definitions(-D__STDC_LIMIT_MACROS)
   add_definitions(-D__STDC_CONSTANT_MACROS)
+  add_definitions(-D__STDC_FORMAT_MACROS)
 endif()
 
 enable_testing()
