--- src/driver/CMakeLists.txt	2019-11-07 01:33:22.905463368 -0600
+++ src/driver/CMakeLists.txt	2019-11-07 01:33:10.605240454 -0600
@@ -77,17 +77,22 @@
 endif (WIN32)
 
 target_link_libraries(opencl_driver
-  clangDriver
-  clangFrontend
-  clangEdit
-  clangLex
-  clangBasic
-  clangCodeGen
+  clangCodeGen clangFrontendTool clangFrontend clangDriver clangSerialization
+  clangParse clangSema clangRewrite clangRewriteFrontend
+  clangStaticAnalyzerFrontend clangStaticAnalyzerCheckers
+  clangStaticAnalyzerCore clangAnalysis clangEdit clangAST clangASTMatchers clangLex clangBasic
   lldELF
   lldCore
-  LLVMDebugInfoDWARF
+  lldCommon
+  ${LLVM_PTHREAD_LIB}
 )
-target_link_libraries(opencl_driver ${llvm_libs})
+
+if (LLVM_LINK_LLVM_DYLIB)
+  target_link_libraries(opencl_driver LLVM)
+else()
+  target_link_libraries(opencl_driver LLVMDebugInfoDWARF ${llvm_libs})
+endif()
+
 target_include_directories(opencl_driver PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
 install(TARGETS opencl_driver DESTINATION lib)
