From f27075f0d04fd6c759a7c16b271d9af24a87dc96 Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@continuum.io>
Date: Wed, 13 Jun 2018 10:24:10 -0500
Subject: [PATCH 2/2] fix setting complex vars on win

---
 src/C/fftw.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/C/fftw.c b/src/C/fftw.c
index 124774a..e39adef 100644
--- a/src/C/fftw.c
+++ b/src/C/fftw.c
@@ -207,7 +207,7 @@ static PyObject *idft(PyObject *self, PyObject *args, PyObject *kwrds)
   Py_END_ALLOW_THREADS
 
   number a;
-  a.z = 1.0/m;
+  a.z = _DCOMPLEX_(1.0/m, 0);
   int mn = m*n, ix = 1;
   zscal_(&mn, &a.z, MAT_BUFZ(X), &ix);
 
@@ -313,7 +313,7 @@ static PyObject *idftn(PyObject *self, PyObject *args, PyObject *kwrds)
   }
 
   number a;
-  a.z = 1.0/proddim;
+  a.z = _DCOMPLEX_(1.0/proddim, 0);
 
   int ix = 1;
   zscal_(&proddim, &a.z, MAT_BUFZ(X), &ix);
-- 
2.17.0.windows.1

