diff --git a/pytoniq_core/crypto/ciphers.py b/pytoniq_core/crypto/ciphers.py
--- a/pytoniq_core/crypto/ciphers.py
+++ b/pytoniq_core/crypto/ciphers.py
@@ -3,3 +3,3 @@
 import hashlib
-import x25519
+from nacl import bindings as c
 
@@ -120,3 +120,3 @@
    """
-    return x25519.scalar_mult(private_key, public_key)
+    return c.crypto_scalarmult(private_key, public_key)
 
diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -28,4 +28,2 @@
         "requests>=2.31.0",
-        "x25519>=0.0.2",
-        "setuptools>=65.5.1",
         "PyNaCl>=1.5.0"
