# Imported from basf2/externals/patches

--- src/RaveBase/Converters/src/CmsToRaveObjects.cc	2024-06-13 06:17:58.578696293 +0900
+++ src/RaveBase/Converters/src/CmsToRaveObjects.cc	2024-06-13 06:17:45.968270000 +0900
@@ -41,7 +41,7 @@
   struct ltVertex
   {
     bool operator() ( const pair < float, rave::Vertex > & r1,
-                      const pair < float, rave::Vertex > & r2 )
+                      const pair < float, rave::Vertex > & r2 ) const
     {
       return ( r1.first > r2.first );
     }
