# Imported from basf2/externals/patches

--- src/DataFormats/GeometrySurface/interface/ReferenceCounted.h	2023-06-14 01:58:15.231823000 +0900
+++ src/DataFormats/GeometrySurface/interface/ReferenceCounted.h	2023-06-14 01:59:15.562354405 +0900
@@ -94,7 +94,7 @@
 #else
 template<typename T>
 struct LocalCache {
-  std::auto_ptr<T> ptr;
+  std::unique_ptr<T> ptr;
 };
 
 #endif
--- src/DataFormats/GeometrySurface/interface/BoundSurface.h	2023-06-14 01:58:27.331541000 +0900
+++ src/DataFormats/GeometrySurface/interface/BoundSurface.h	2023-06-14 01:59:28.874708000 +0900
@@ -31,7 +31,7 @@
       theBounds( iToCopy.theBounds->clone() ) {}
 
     const BoundSurface& operator=(const BoundSurface& iRHS ) {
-      theBounds = std::auto_ptr<Bounds>( iRHS.theBounds->clone() );
+      theBounds = std::unique_ptr<Bounds>( iRHS.theBounds->clone() );
       return *this;
     }
 
@@ -39,7 +39,7 @@
 
 private:
   //own_ptr<Bounds,OwnerPolicy::Clone> theBounds;
-  std::auto_ptr<Bounds> theBounds;
+  std::unique_ptr<Bounds> theBounds;
 };
 
 
--- src/DataFormats/GeometrySurface/interface/BlockWipedAllocator.h	2023-06-14 01:58:38.282943000 +0900
+++ src/DataFormats/GeometrySurface/interface/BlockWipedAllocator.h	2023-06-14 01:59:36.322346203 +0900
@@ -137,7 +137,7 @@
 
 template<typename T>
 struct LocalCache : public BlockWipedAllocator::LocalCache {
-  std::auto_ptr<T> ptr;
+  std::unique_ptr<T> ptr;
   LocalCache(){ 
     blockWipedAllocator<sizeof(T)>().registerCache(this);
   }
--- src/RecoVertex/KinematicFitPrimitives/src/KinematicRefittedTrackState.cc	2023-06-14 01:58:46.723503000 +0900
+++ src/RecoVertex/KinematicFitPrimitives/src/KinematicRefittedTrackState.cc	2023-06-14 01:59:41.645724000 +0900
@@ -73,7 +73,7 @@
 TrajectoryStateOnSurface KinematicRefittedTrackState::trajectoryStateOnSurface(const Surface & surface, 
                                                    const Propagator & propagator) const
 {
- std::auto_ptr<Propagator> thePropagator( propagator.clone());
+ std::unique_ptr<Propagator> thePropagator( propagator.clone());
  thePropagator->setPropagationDirection(anyDirection);
  return thePropagator->propagate(freeTrajectoryState(), surface);
 }
--- src/RecoVertex/VertexTools/src/PerigeeRefittedTrackState.cc	2023-06-14 01:58:52.936844000 +0900
+++ src/RecoVertex/VertexTools/src/PerigeeRefittedTrackState.cc	2023-06-14 01:59:49.943971000 +0900
@@ -46,7 +46,7 @@
 PerigeeRefittedTrackState::trajectoryStateOnSurface(const Surface & surface,
 				const Propagator & propagator) const
 {
-  std::auto_ptr<Propagator> thePropagator( propagator.clone());
+  std::unique_ptr<Propagator> thePropagator( propagator.clone());
   thePropagator->setPropagationDirection(anyDirection);
 
   TrajectoryStateOnSurface tsos = thePropagator->propagate(freeTrajectoryState(), surface);
--- src/RecoVertex/ZvresReco/src/ZvresVertexReconstructor.cc	2023-06-14 01:59:00.791072000 +0900
+++ src/RecoVertex/ZvresReco/src/ZvresVertexReconstructor.cc	2023-06-14 01:59:54.445770870 +0900
@@ -422,7 +422,7 @@
   clog << "----------------------------------------------------------------------------" << endl;
 #endif
 
-  std::auto_ptr<TransientVertexVector > ret(from_lcfi(zres));
+  std::unique_ptr<TransientVertexVector > ret(from_lcfi(zres));
 
   // free up any dynamic lcfi stuff...
   vertex_lcfi::MetaMemoryManager::Event()->delAllObjects();
