--- a/CMakeLists.txt	2026-03-27 23:23:22
+++ b/CMakeLists.txt	2026-03-27 23:26:01
@@ -136,21 +136,21 @@
   ENDIF()
 ENDMACRO()
 
-MACRO(GET_MOD_ENABLE_RANK macro_modname macro_mod_enable_val macro_output_rank)
+FUNCTION(GET_MOD_ENABLE_RANK macro_modname macro_mod_enable_val macro_output_rank)
   IF(${macro_mod_enable_val} STREQUAL "O")
-    SET(${macro_output_rank} 0)
+    SET(${macro_output_rank} 0 PARENT_SCOPE)
   ELSEIF(${macro_mod_enable_val} STREQUAL "i")
-    SET(${macro_output_rank} 1)
+    SET(${macro_output_rank} 1 PARENT_SCOPE)
   ELSEIF(${macro_mod_enable_val} STREQUAL "I")
-    SET(${macro_output_rank} 2)
+    SET(${macro_output_rank} 2 PARENT_SCOPE)
   ELSEIF(${macro_mod_enable_val} STREQUAL "a")
-    SET(${macro_output_rank} 3)
+    SET(${macro_output_rank} 3 PARENT_SCOPE)
   ELSEIF(${macro_mod_enable_val} STREQUAL "A")
-    SET(${macro_output_rank} 4)
+    SET(${macro_output_rank} 4 PARENT_SCOPE)
   ELSE()
     MESSAGE(FATAL_ERROR "Unexpected enablement value \"${macro_mod_enable_val}\" for ${macro_modname}")
   ENDIF()
-ENDMACRO()
+ENDFUNCTION()
 
 GET_MOD_ENABLE_RANK("ENABLE_MODULES setting" ${ENABLE_MODULES} enable_modules_rank)
 
