diff --git a/src/interface.cpp b/src/interface.cpp
index b95fe35..f2f4889 100644
--- a/src/interface.cpp
+++ b/src/interface.cpp
@@ -88,7 +88,7 @@ static void CopyWhatTextToErrorMessage(const system_error& e, char* errorMessage
 }
 
 
-CHARLS_IMEXPORT(ApiResult) JpegLsEncodeStream(ByteStreamInfo compressedStreamInfo, size_t& pcbyteWritten,
+ApiResult JpegLsEncodeStream(ByteStreamInfo compressedStreamInfo, size_t& pcbyteWritten,
     ByteStreamInfo rawStreamInfo, const struct JlsParameters& params, char* errorMessage)
 {
     try
@@ -152,7 +152,7 @@ CHARLS_IMEXPORT(ApiResult) JpegLsEncodeStream(ByteStreamInfo compressedStreamInf
 }
 
 
-CHARLS_IMEXPORT(ApiResult) JpegLsDecodeStream(ByteStreamInfo rawStream, ByteStreamInfo compressedStream, const JlsParameters* info, char* errorMessage)
+ApiResult JpegLsDecodeStream(ByteStreamInfo rawStream, ByteStreamInfo compressedStream, const JlsParameters* info, char* errorMessage)
 {
     try
     {
@@ -181,7 +181,7 @@ CHARLS_IMEXPORT(ApiResult) JpegLsDecodeStream(ByteStreamInfo rawStream, ByteStre
 }
 
 
-CHARLS_IMEXPORT(ApiResult) JpegLsReadHeaderStream(ByteStreamInfo rawStreamInfo, JlsParameters* params, char* errorMessage)
+ApiResult JpegLsReadHeaderStream(ByteStreamInfo rawStreamInfo, JlsParameters* params, char* errorMessage)
 {
     try
     {
@@ -207,7 +207,7 @@ CHARLS_IMEXPORT(ApiResult) JpegLsReadHeaderStream(ByteStreamInfo rawStreamInfo,
 
 extern "C"
 {
-    CHARLS_IMEXPORT(ApiResult) JpegLsEncode(void* destination, size_t destinationLength, size_t* bytesWritten, const void* source, size_t sourceLength, const struct JlsParameters* params, char* errorMessage)
+    ApiResult JpegLsEncode(void* destination, size_t destinationLength, size_t* bytesWritten, const void* source, size_t sourceLength, const struct JlsParameters* params, char* errorMessage)
     {
         if (!destination || !bytesWritten || !source || !params)
             return ApiResult::InvalidJlsParameters;
@@ -219,13 +219,13 @@ extern "C"
     }
 
 
-    CHARLS_IMEXPORT(ApiResult) JpegLsReadHeader(const void* compressedData, size_t compressedLength, JlsParameters* params, char* errorMessage)
+    ApiResult JpegLsReadHeader(const void* compressedData, size_t compressedLength, JlsParameters* params, char* errorMessage)
     {
         return JpegLsReadHeaderStream(FromByteArray(compressedData, compressedLength), params, errorMessage);
     }
 
 
-    CHARLS_IMEXPORT(ApiResult) JpegLsDecode(void* destination, size_t destinationLength, const void* source, size_t sourceLength, const struct JlsParameters* params, char* errorMessage)
+    ApiResult JpegLsDecode(void* destination, size_t destinationLength, const void* source, size_t sourceLength, const struct JlsParameters* params, char* errorMessage)
     {
         ByteStreamInfo compressedStream = FromByteArray(source, sourceLength);
         ByteStreamInfo rawStreamInfo = FromByteArray(destination, destinationLength);
@@ -234,7 +234,7 @@ extern "C"
     }
 
 
-    CHARLS_IMEXPORT(ApiResult) JpegLsVerifyEncode(const void* uncompressedData, size_t uncompressedLength, const void* compressedData, size_t compressedLength, char* errorMessage)
+    ApiResult JpegLsVerifyEncode(const void* uncompressedData, size_t uncompressedLength, const void* compressedData, size_t compressedLength, char* errorMessage)
     {
         try
         {
@@ -292,7 +292,7 @@ extern "C"
     }
 
 
-    CHARLS_IMEXPORT(ApiResult) JpegLsDecodeRect(void* uncompressedData, size_t uncompressedLength, const void* compressedData, size_t compressedLength,
+    ApiResult JpegLsDecodeRect(void* uncompressedData, size_t uncompressedLength, const void* compressedData, size_t compressedLength,
         JlsRect roi, JlsParameters* info, char* errorMessage)
     {
         try
