From 7aa66b3d0c5812f41e98ac152c1994a19ede4976 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Tue, 21 Apr 2026 16:23:14 +0200
Subject: [PATCH] fix

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 73014f2..ae05c6a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -68,7 +68,6 @@ cmake.version = ">=3.15"
 build.verbose = true
 wheel.packages = ["src/ase2sprkkr"]
 sdist.exclude = [
-	"**/*",
 	"**/*.sw?",
 	"**/.pytest_cache",
 	"**/.pytest_cache/**",
