diff --git a/CMakeLists.txt b/CMakeLists.txt
index 80c5118..1a407f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,14 +71,7 @@ else()
   set(PROTOBUF_LIB "protobuf::libprotobuf")
 endif()
 
-if(NOT TARGET onnx_proto)
-  # Note: This avoids libprotobuf.so complaining about name collisions at runtime
-  if(NOT ONNX_NAMESPACE)
-    set(ONNX_NAMESPACE "onnx2trt_onnx")
-  endif()
-  add_definitions("-DONNX_NAMESPACE=${ONNX_NAMESPACE}")
-  add_subdirectory(third_party/onnx EXCLUDE_FROM_ALL)
-endif()
+FIND_PACKAGE(ONNX REQUIRED)
 
 # CUDA
 if (NOT CUDA_TOOLKIT_ROOT_DIR)
