diff --git a/m4/acx_openblas.m4 b/m4/acx_openblas.m4
index 5421eec..fe43f65 100644
--- a/m4/acx_openblas.m4
+++ b/m4/acx_openblas.m4
@@ -100,7 +100,7 @@ dnl ----------------------------
 		[Define if you have the OpenBLAS parallel library and header files.]),
       unset ac_cv_search_LAPACKE_dpotrf
       [AC_SEARCH_LIBS(
-        [LAPACKE_dpotrf], ["openblas"$acx_openblas_suffix],
+        [LAPACKE_dpotrf], ["openblas"$acx_openblas_suffix "lapacke"$acx_openblas_suffix],
 	[OPENBLAS_WARN="parallel OpenBLAS"$acx_openblas_suffix" not found, reverting to scalar OpenBLAS"$acx_openblas_suffix"!"],
         [OPENBLAS_ERROR="OpenBLAS"$acx_openblas_suffix" library file not found!"],
         $acx_openblas_libopt
@@ -109,7 +109,7 @@ dnl ----------------------------
     )
   else
     AC_SEARCH_LIBS(
-      [LAPACKE_dpotrf], ["openblas"$acx_openblas_suffix],,
+      [LAPACKE_dpotrf], ["openblas"$acx_openblas_suffix "lapacke"$acx_openblas_suffix],,
       [OPENBLAS_ERROR="OpenBLAS"$acx_openblas_suffix" library file not found!"],
       $acx_openblas_libopt
     )
