diff --git a/src/Makefile.am b/src/Makefile.am
index 3017766..4f75357 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -34,7 +34,7 @@ if USE_MODEL
   LEVDIR		= levmar
 endif
 SUBDIRS			= fits $(LEVDIR) wcs
-bin_PROGRAMS		= sex ldactoasc
+bin_PROGRAMS		= sex
 check_PROGRAMS		= sex
 sex_SOURCES		= analyse.c assoc.c astrom.c back.c bpro.c catout.c \
 			  check.c clean.c dgeo.c extract.c $(FFTSOURCE) \
