diff --git a/configure.ac b/configure.ac
index 7ff6cce..57290bd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -335,7 +335,7 @@ debian/changelog
 doc/Makefile
 doc/src/conf.py
 man/Makefile
-man/sex.1
+man/s-extractor.1
 src/Makefile
 src/fits/Makefile
 src/levmar/Makefile
diff --git a/man/Makefile.am b/man/Makefile.am
index 8e1f78c..40cfdfc 100644
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -26,5 +26,5 @@
 #
 #%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

-dist_man_MANS = sex.1 sex.x
+dist_man_MANS = s_extractor.1 s_extractor.x

diff --git a/man/sex.1.in b/man/sex.1.in
index d0f7208..1fe7e80 100644
--- a/man/sex.1.in
+++ b/man/sex.1.in
@@ -1,18 +1,18 @@
 .TH SEXTRACTOR "1" "@DATE3@" "SExtractor @PACKAGE_VERSION@" "User Commands"
 .SH NAME
-sex \- extract a source catalogue from an astronomical FITS image
+s-extractor \- extract a source catalogue from an astronomical FITS image
 .SH SYNOPSIS
-.B sex \fIimage\fR [\fI-c configuration-file\fR]
+.B s-extractor \fIimage\fR [\fI-c configuration-file\fR]
 .RS
 [\fI-parameter1 value1 -parameter2 value2 ...\fR]
 .RE
 .TP
-.B sex \fIimage\_detect\fR \fIimage_measure\fR [\fI-c configuration-file\fR]
+.B s-extractor \fIimage\_detect\fR \fIimage_measure\fR [\fI-c configuration-file\fR]
 .RS
 [\fI-parameter1 value1 -parameter2 value2 ...\fR]
 .RE
 .TP
-.B sex \fI-d\fR
+.B s-extractor \fI-d\fR
 .SH DESCRIPTION
 SExtractor is a program that builds a catalogue of objects from an astronomical
 image. Although it is particularly oriented towards reduction of large scale
diff --git a/man/sex.x b/man/sex.x
index c6f97ad..95311ae 100644
--- a/man/sex.x
+++ b/man/sex.x
@@ -1,2 +1,2 @@
 [name]
-sex \- extract a source catalog from an astronomical FITS image
+s_extractor \- extract a source catalog from an astronomical FITS image
diff --git a/src/Makefile.am b/src/Makefile.am
index 3017766..e5c075f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -34,9 +34,9 @@ if USE_MODEL
   LEVDIR		= levmar
 endif
 SUBDIRS			= fits $(LEVDIR) wcs
-bin_PROGRAMS		= sex
-check_PROGRAMS		= sex
-sex_SOURCES		= analyse.c assoc.c astrom.c back.c bpro.c catout.c \
+bin_PROGRAMS		= s-extractor
+check_PROGRAMS		= s-extractor
+s_extractor_SOURCES		= analyse.c assoc.c astrom.c back.c bpro.c catout.c \
 			  check.c clean.c dgeo.c extract.c $(FFTSOURCE) \
 			  field.c filter.c fitswcs.c flag.c graph.c growth.c \
 			  header.c image.c interpolate.c main.c makeit.c \
@@ -53,9 +53,9 @@ sex_SOURCES		= analyse.c assoc.c astrom.c back.c bpro.c catout.c \
 			  sexheadsc.h som.h threads.h types.h wcscelsys.h \
 			  weight.h winpos.h xml.h
 ldactoasc_SOURCES 	= ldactoasc.c ldactoasc.h
-sex_LDADD		= $(srcdir)/fits/libfits.a \
+s_extractor_LDADD		= $(srcdir)/fits/libfits.a \
 			  $(srcdir)/wcs/libwcs_c.a \
-			  $(LEVLIB)
+			  $(LEVLIB)
 ldactoasc_LDADD		= $(srcdir)/fits/libfits.a
 DATE=`date +"%Y-%m-%d"`

