From 60ca0463ed28d00998c77decb5fd67ff3f887392 Mon Sep 17 00:00:00 2001
From: Samee <sameeul@gmail.com>
Date: Wed, 21 Sep 2022 20:17:47 +0000
Subject: [PATCH] fix filesystem issue

---
 tests/test_initialization.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/tests/test_initialization.h b/tests/test_initialization.h
index 8bdd5b6..f602d5f 100644
--- a/tests/test_initialization.h
+++ b/tests/test_initialization.h
@@ -5,7 +5,15 @@
 #include <unordered_map>
 #include <unordered_set>
 #include <iostream>
-#include <filesystem>
+#if __has_include(<filesystem>)
+  #include <filesystem>
+  namespace fs = std::filesystem;
+#elif __has_include(<experimental/filesystem>)
+  #include <experimental/filesystem> 
+  namespace fs = std::experimental::filesystem;
+#else
+  error "Missing the <filesystem> header."
+#endif
 
 #include "../src/nyx/dirs_and_files.h"
 #include "../src/nyx/environment.h"
-- 
2.32.0

