From 1c9662b77a0c8c07c02a8ab2ea36f6077578b706 Mon Sep 17 00:00:00 2001
From: Uwe Korn <uwelk@xhochy.com>
Date: Sun, 21 Jan 2024 09:38:54 +0000
Subject: [PATCH] Add missing stdexcept include

---
 src/range_map.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/range_map.h b/src/range_map.h
index 9faa9e8..9aed8ed 100644
--- a/src/range_map.h
+++ b/src/range_map.h
@@ -34,6 +34,7 @@
 
 #include <exception>
 #include <map>
+#include <stdexcept>
 #include <vector>
 
 #include "absl/strings/str_cat.h"
-- 
2.25.1

