From f91f6cd0739c7b060e8cf7e1cd8749f9798f1f8f Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwelk@xhochy.com>
Date: Sat, 16 Dec 2023 13:52:17 +0100
Subject: [PATCH] Don't force C++11

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08965ac..c3aaf05 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,6 @@ cmake_minimum_required(VERSION 3.5)
 cmake_policy(SET CMP0048 NEW)
 project (Bloaty VERSION 1.0)
 project (Bloaty VERSION 1.1)
-set(CMAKE_CXX_STANDARD 11)
 
 # Options we define for users.
 option(BLOATY_ENABLE_ASAN "Enable address sanitizer." OFF)
-- 
2.39.3 (Apple Git-145)

