From 0687488446d48feedf6e5a6c3472b4c2ec45b846 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sat, 27 Jun 2020 11:13:35 +0200
Subject: [PATCH] Add cmath include to EvtMatrix.hh

---
 R02-00-00/EvtGenBase/EvtMatrix.hh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/R02-00-00/EvtGenBase/EvtMatrix.hh b/R02-00-00/EvtGenBase/EvtMatrix.hh
index dec2312..93e07c1 100644
--- a/R02-00-00/EvtGenBase/EvtMatrix.hh
+++ b/R02-00-00/EvtGenBase/EvtMatrix.hh
@@ -21,6 +21,7 @@
 #ifndef __EVT_MATRIX_HH__
 #define __EVT_MATRIX_HH__
 
+#include <cmath>
 #include <sstream>
 #include <vector>
 
-- 
2.26.0

