From 396c12741e1487c55c2070610e66a5ae4f94f692 Mon Sep 17 00:00:00 2001
From: David Laehnemann <1379875+dlaehnemann@users.noreply.github.com>
Date: Mon, 1 Dec 2025 14:50:45 +0100
Subject: [PATCH] remove HISTORY.md from setup.py

---
 setup.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index e22dc08..2fdae18 100644
--- a/setup.py
+++ b/setup.py
@@ -3,15 +3,12 @@ from setuptools import setup, find_packages
 with open('README.md') as readme_file:
     README = readme_file.read()
 
-with open('HISTORY.md') as history_file:
-    HISTORY = history_file.read()
-
 setup_args = dict(
     name='multiprocesspandas',
     version='1.1.5',
     description='Extends Pandas to run apply methods for  dataframe, series and groups on multiple cores at same time.',
     long_description_content_type="text/markdown",
-    long_description=README + '\n\n' + HISTORY,
+    long_description=README,
     license='MIT',
     packages=find_packages(),
     author='Shahnawaz Akhtar',
-- 
2.51.1

