From bfe6d8eb9141c33f68c3b5c791dc55d75a311255 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 17 Jul 2022 20:17:04 +0200
Subject: [PATCH] use DLLs from conda-forge

---
 buildconfig/config_win.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/buildconfig/config_win.py b/buildconfig/config_win.py
index 1ea07457..415c395a 100644
--- a/buildconfig/config_win.py
+++ b/buildconfig/config_win.py
@@ -473,10 +473,10 @@ def setup_prebuilt_sdl2(prebuilt_dir):
 
 def main():
     machine_type = get_machine_type()
-    prebuilt_dir = 'prebuilt-%s' % machine_type
+    prebuilt_dir = os.environ["LIBRARY_LIB"]
     use_prebuilt = '-prebuilt' in sys.argv
 
-    auto_download = 'PYGAME_DOWNLOAD_PREBUILT' in os.environ
+    auto_download = False
     if auto_download:
         auto_download = os.environ['PYGAME_DOWNLOAD_PREBUILT'] == '1'
 
-- 
2.37.0.windows.1

