From 21c99189c993172cda5e5171829e8091f68537ff Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 17 Jul 2022 20:17:04 +0200
Subject: [PATCH 1/9] use DLLs from conda-forge

---
 buildconfig/config_win.py | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/buildconfig/config_win.py b/buildconfig/config_win.py
index 856e2587..6309a1c9 100644
--- a/buildconfig/config_win.py
+++ b/buildconfig/config_win.py
@@ -30,9 +30,9 @@ def get_machine_type():
     return as_machine_type(get_ptr_size())
 
 class Dependency:
-    huntpaths = ['..', '..\\..', '..\\*', '..\\..\\*']
-    inc_hunt = ['include']
-    lib_hunt = ['VisualC\\SDL\\Release', 'VisualC\\Release', 'Release', 'lib']
+    huntpaths = [os.environ['LIBRARY_PREFIX'], os.environ['LIBRARY_LIB'], os.environ['LIBRARY_INC']]
+    inc_hunt = ['include', 'include\\SDL2']
+    lib_hunt = ['lib', 'bin']
     check_hunt_roots = True
     def __init__(self, name, wildcards, libs=None, required=0, find_header='', find_lib=''):
         if libs is None:
@@ -455,10 +455,11 @@ def setup_prebuilt_sdl2(prebuilt_dir):
 
 def main(auto_config=False):
     machine_type = get_machine_type()
-    prebuilt_dir = f'prebuilt-{machine_type}'
-    use_prebuilt = '-prebuilt' in sys.argv
+    # avoid triggering unwanted branches below
+    prebuilt_dir = ''
+    use_prebuilt = False
 
-    auto_download = 'PYGAME_DOWNLOAD_PREBUILT' in os.environ
+    auto_download = False
     if auto_download:
         auto_download = os.environ['PYGAME_DOWNLOAD_PREBUILT'] == '1'
 
-- 
2.38.1.windows.1

