From fccb6608d6605240fe38ed1db0fea961c1e8ae91 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 17 Jul 2022 23:41:27 +0200
Subject: [PATCH 02/10] fix PREFIX-handling on OSX

---
 buildconfig/config_darwin.py | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/buildconfig/config_darwin.py b/buildconfig/config_darwin.py
index c785e183..e7fce48e 100644
--- a/buildconfig/config_darwin.py
+++ b/buildconfig/config_darwin.py
@@ -146,16 +146,11 @@ def main():
     ])
 
     print ('Hunting dependencies...')
-    incdirs = ['/usr/local/include', '/opt/homebrew/include']
-    incdirs.extend(['/usr/local/include/SDL2', '/opt/homebrew/include/SDL2', '/opt/local/include/SDL2'])
+    incdirs = [os.environ['PREFIX'] + '/include',
+               os.environ['PREFIX'] + '/include/SDL2',
+               os.environ['PREFIX'] + '/include/freetype2/freetype']
 
-    incdirs.extend([
-       #'/usr/X11/include',
-       '/opt/local/include',
-       '/opt/local/include/freetype2/freetype']
-    )
-    #libdirs = ['/usr/local/lib', '/usr/X11/lib', '/opt/local/lib']
-    libdirs = ['/usr/local/lib', '/opt/local/lib', '/opt/homebrew/lib']
+    libdirs = [os.environ['PREFIX'] + '/lib']
 
     for d in DEPS:
         if isinstance(d, (list, tuple)):
-- 
2.37.0.windows.1

