From 5af582d7805360df919b20246d115d8649fcaae3 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 19 Jul 2022 19:32:47 +0200
Subject: [PATCH 03/10] try to fix whacky search patterns for windows

---
 buildconfig/config_win.py | 28 +++++++++++++++-------------
 1 file changed, 15 insertions(+), 13 deletions(-)

diff --git a/buildconfig/config_win.py b/buildconfig/config_win.py
index ffa0e56a..eb925e22 100644
--- a/buildconfig/config_win.py
+++ b/buildconfig/config_win.py
@@ -371,27 +371,29 @@ def _add_sdl2_dll_deps(DEPS):
     DEPS.add_dll(r'(lib)?opusfile[-0-9]*\.dll$', 'opusfile', ['*opusfile-[0-9]*'])
     # IMAGE
     DEPS.add_dll(r'(lib){0,1}tiff[-0-9]*\.dll$', 'tiff', ['tiff-[0-9]*'], ['jpeg', 'z'])
-    DEPS.add_dll(r'(z|zlib1)\.dll$', 'z', ['zlib-[1-9].*'])
+    DEPS.add_dll(r'z(lib|dll)?\.dll$', 'z', ['zlib-[1-9].*'])
     DEPS.add_dll(r'(lib)?webp[-0-9]*\.dll$', 'webp', ['*webp-[0-9]*'])
 
 def setup():
     DEPS = DependencyGroup()
 
-    DEPS.add('SDL', 'SDL2', ['SDL2-[1-9].*'], r'(lib){0,1}SDL2\.dll$', required=1)
+    # arguments: name, lib, wildcards, dll_regex, libs=None, required=0, find_header='', find_lib=''
+    DEPS.add('SDL', 'SDL2', ['SDL2-[1-9].*'], r'(lib)?SDL2\.dll$', find_header=r'SDL\.h', required=1)
     DEPS.add('PORTMIDI', 'portmidi', ['portmidi'], r'portmidi\.dll$', find_header=r'portmidi\.h')
     #DEPS.add('PORTTIME', 'porttime', ['porttime'], r'porttime\.dll$')
     DEPS.add_dummy('PORTTIME')
-    DEPS.add('MIXER', 'SDL2_mixer', ['SDL2_mixer-[1-9].*'], r'(lib){0,1}SDL2_mixer\.dll$',
-             ['SDL', 'vorbisfile'])
-    DEPS.add('PNG', 'png', ['SDL2_image-[2-9].*', 'libpng-[1-9].*'], r'(png|libpng)[-0-9]*\.dll$', ['z'],
-             find_header=r'png\.h', find_lib=r'(lib)?png1[-0-9]*\.lib')
-    DEPS.add('JPEG', 'jpeg', ['SDL2_image-[2-9].*', 'jpeg-9*'], r'(lib){0,1}jpeg-9\.dll$',
-             find_header=r'jpeglib\.h', find_lib=r'(lib)?jpeg-9\.lib')
-    DEPS.add('IMAGE', 'SDL2_image', ['SDL2_image-[1-9].*'], r'(lib){0,1}SDL2_image\.dll$',
-             ['SDL', 'jpeg', 'png', 'tiff'], 0)
-    DEPS.add('FONT', 'SDL2_ttf', ['SDL2_ttf-[2-9].*'], r'(lib){0,1}SDL2_ttf\.dll$', ['SDL', 'z', 'freetype'])
-    DEPS.add('FREETYPE', 'freetype', ['freetype-[1-9].*'], r'(lib){0,1}freetype[-0-9]*\.dll$',
-             find_header=r'ft2build\.h', find_lib=r'(lib)?freetype[-0-9]*\.lib')
+    DEPS.add('MIXER', 'SDL2_mixer', ['SDL2_mixer-[1-9].*'], r'(lib)?SDL2_mixer\.dll$',
+             ['SDL', 'vorbisfile'], find_header=r'SDL_mixer\.h')
+    DEPS.add('PNG', 'png', ['libpng-[1-9].*'], r'(lib)?png\d*\.dll$', ['z'],
+             find_header=r'png\.h', find_lib=r'(lib)?png\d*\.lib')
+    DEPS.add('JPEG', 'jpeg', ['jpeg-9*'], r'(lib)?jpeg\.dll$',
+             find_header=r'jpeglib\.h', find_lib=r'(lib)?jpeg\.lib')
+    DEPS.add('IMAGE', 'SDL2_image', ['SDL2_image-[1-9].*'], r'(lib)?SDL2_image\.dll$',
+             ['SDL', 'jpeg', 'png', 'tiff'], required=0, find_header=r'SDL_image\.h')
+    DEPS.add('FONT', 'SDL2_ttf', ['SDL2_ttf-[2-9].*'], r'(lib)?SDL2_ttf\.dll$',
+             ['SDL', 'z', 'freetype'], find_header=r'SDL_ttf\.h')
+    DEPS.add('FREETYPE', 'freetype', ['freetype-[1-9].*'], r'(lib)?freetype\.dll$',
+             find_header=r'ft2build\.h', find_lib=r'(lib)?freetype\.lib')
     DEPS.configure()
     _add_sdl2_dll_deps(DEPS)
     for d in get_definitions():
-- 
2.37.0.windows.1

