From 58e0a62d48b94f5c5f99b68edcf916216e421d25 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 18 Jul 2022 00:08:38 +0200
Subject: [PATCH 3/3] also look for SDL2 on windows

---
 buildconfig/config_win.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/buildconfig/config_win.py b/buildconfig/config_win.py
index 62d7cd64..b21a0980 100644
--- a/buildconfig/config_win.py
+++ b/buildconfig/config_win.py
@@ -31,7 +31,7 @@ def get_machine_type():
 
 class Dependency(object):
     huntpaths = ['..', '..\\..', '..\\*', '..\\..\\*']
-    inc_hunt = ['include']
+    inc_hunt = ['include', 'include\\SDL2']
     lib_hunt = ['VisualC\\SDL\\Release', 'VisualC\\Release', 'Release', 'lib']
     check_hunt_roots = True
     def __init__(self, name, wildcards, libs=None, required=0, find_header='', find_lib=''):
-- 
2.37.0.windows.1

