From 829147b02c48fcade1cfd0118780379ccaeac7ab Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 19 Jul 2022 21:44:52 +0200
Subject: [PATCH 04/10] do not copy DLLs

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index f8c309a0..ea012bb2 100644
--- a/setup.py
+++ b/setup.py
@@ -659,8 +659,8 @@
         f = the_dlls[lib].replace('/', os.sep)
         if f == '_':
             print(f"WARNING, DLL for {lib} library not found.")
-        else:
-            pygame_data_files.append(f)
+        # else:
+        #     pygame_data_files.append(f)
 
     if '-enable-msvc-analyze' in sys.argv:
         # calculate the MSVC compiler version as an int
