From c2c07d88ff664127b35fe19fa0499676c3beae07 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 19 Jul 2022 19:55:46 +0200
Subject: [PATCH 4/4] remove spurious error that's in a logging branch

---
 buildconfig/config_win.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/buildconfig/config_win.py b/buildconfig/config_win.py
index c762c3ba..45527c60 100644
--- a/buildconfig/config_win.py
+++ b/buildconfig/config_win.py
@@ -85,8 +85,6 @@ class Dependency(object):
                 print ("Path for %s not found." % self.name)
                 for info in self.prune_info:
                     print(info)
-                if self.required:
-                    print ('Too bad that is a requirement! Hand-fix the "Setup"')
             return False
         elif len(self.paths) == 1:
             self.path = self.paths[0]
-- 
2.37.0.windows.1

