From 829147b02c48fcade1cfd0118780379ccaeac7ab Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 19 Jul 2022 21:44:52 +0200
Subject: [PATCH 04/10] do not copy DLLs

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index f8c309a0..ea012bb2 100644
--- a/setup.py
+++ b/setup.py
@@ -612,8 +612,8 @@ if sys.platform == 'win32' and not 'WIN32_DO_NOT_INCLUDE_DEPS' in os.environ:
         f = the_dlls[lib].replace('/', os.sep)
         if f == '_':
             print ("WARNING, DLL for %s library not found." % lib)
-        else:
-            pygame_data_files.append(f)
+        # else:
+        #     pygame_data_files.append(f)
 
 
     if '-enable-msvc-analyze' in sys.argv:
-- 
2.37.0.windows.1

