From 4a6bbf415c5a40a672b9a46b9cce6b88796cdbce Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 20 Jul 2022 08:07:27 +0200
Subject: [PATCH 05/10] avoid downloading stuff despite auto_download=False

---
 buildconfig/config_win.py | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/buildconfig/config_win.py b/buildconfig/config_win.py
index eb925e22..2b007c50 100644
--- a/buildconfig/config_win.py
+++ b/buildconfig/config_win.py
@@ -495,12 +495,7 @@ def main():
     }
     download_kwargs[machine_type] = True
 
-    if not auto_download:
-        if (not download_win_prebuilt.cached(**download_kwargs) or\
-            not os.path.isdir(prebuilt_dir))\
-            and download_win_prebuilt.ask(**download_kwargs):
-            use_prebuilt = True
-    else:
+    if auto_download:
         download_win_prebuilt.update(**download_kwargs)
 
     if os.path.isdir(prebuilt_dir):
-- 
2.37.0.windows.1

