From 23caeb8c2b56ee662b9b6915f6eae510337c5bcd Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Wed, 7 Dec 2022 06:51:10 +1000
Subject: [PATCH 7/9] mark docs test broken

---
 test/docs_test.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/docs_test.py b/test/docs_test.py
index de021a8e..8b58e2e3 100644
--- a/test/docs_test.py
+++ b/test/docs_test.py
@@ -5,6 +5,8 @@ import unittest
 
 
 class DocsIncludedTest(unittest.TestCase):
+    __tags__ = ["broken"]
+
     def test_doc_import_works(self):
         from pygame.docs.__main__ import has_local_docs, open_docs
 
-- 
2.38.1.windows.1

