From be34b7c86bd065394b3cd5231089581ad6afd6c3 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Wed, 7 Dec 2022 07:01:04 +1000
Subject: [PATCH 8/9] fix strict prototype

---
 src_c/simd_blitters.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src_c/simd_blitters.h b/src_c/simd_blitters.h
index 36cb8c2a..2b196d66 100644
--- a/src_c/simd_blitters.h
+++ b/src_c/simd_blitters.h
@@ -40,7 +40,7 @@ blit_blend_premultiplied_sse2(SDL_BlitInfo *info);
 #endif /* (defined(__SSE2__) || defined(PG_ENABLE_ARM_NEON)) */
 
 int
-pg_has_avx2();
+pg_has_avx2(void);
 void
 blit_blend_rgba_mul_avx2(SDL_BlitInfo *info);
 void
-- 
2.38.1.windows.1

