diff --git a/setup.py b/setup.py
index 888d7a455..5e4a3d8e5 100644
--- a/setup.py
+++ b/setup.py
@@ -435,7 +435,7 @@ def run_install_headers(self):
         # some warnings are skipped here
         ("/W3", "/wd4142", "/wd4996")
         if sys.platform == "win32"
-        else ("-Wall", "-Wno-error=unknown-pragmas")
+        else ("-Wall", "-Wno-error=unknown-pragmas", "-Wno-error=unused-command-line-argument")
     )
 
     if "surface" in e.name and sys.platform == "darwin":
