commit 083dc0a5ad1e72309fba68b3cd50ecbe0ea30c63
Author: Tom Schoonjans <Tom.Schoonjans@diamond.ac.uk>
Date:   Wed Nov 6 13:55:24 2019 +0000

    build: decode strings before split

diff --git a/commands/java.py b/commands/java.py
index ccbda7e..f6b600f 100644
--- a/commands/java.py
+++ b/commands/java.py
@@ -208,7 +208,9 @@ class setup_java(Command):
                     'INFO: the MACOSX_DEPLOYMENT_TARGET environment variable is set:', target)
 
                 result = shell('sw_vers -productVersion')
-                if target.split('.')[:2] != result.stdout.split('.')[:2]:
+                if target.split('.')[:2] != result.stdout.decode("utf-8").split('.')[:2]:
                     warning(
                         'This target appears to be incorrect for the system version:', result.stdout)
 
