commit 3c20b173549a2419e699a2b4233fd6e3bd8796fd
Author: Tom Schoonjans <Tom.Schoonjans@diamond.ac.uk>
Date:   Thu Nov 14 15:25:22 2019 +0000

    Fix overlinking against JVM and Python shared libs

diff --git a/commands/java.py b/commands/java.py
index a55d6b3..0386b07 100644
--- a/commands/java.py
+++ b/commands/java.py
@@ -120,7 +120,7 @@ def get_java_lib():
 
 
 def get_java_libraries():
-    if not is_osx():
+    if is_windows():
         return ['jvm']
     return []
 
diff --git a/commands/python.py b/commands/python.py
index 26acf38..6886735 100644
--- a/commands/python.py
+++ b/commands/python.py
@@ -15,7 +15,9 @@ def get_python_libs():
     ldv = sysconfig.get_config_var('LDVERSION')
     if ldv:
         v = ldv
-    libs = ['python' + v]
+    libs = []
+    if is_windows():
+        libs.append('python' + v)
     if not is_windows() and not is_bsd():
         libs.append('dl')
     return libs
