commit b92c4ad8a8f2461a2c19ebadadc861837903a825
Author: Tom Schoonjans <Tom.Schoonjans@diamond.ac.uk>
Date:   Thu Nov 14 15:25:22 2019 +0000

    Fix overlinking against JVM shared lib

diff --git a/commands/java.py b/commands/java.py
index a55d6b3..0386b07 100644
--- a/commands/java.py
+++ b/commands/java.py
@@ -120,7 +120,7 @@ def get_java_lib():
 
 
 def get_java_libraries():
-    if not is_osx():
+    if is_windows():
         return ['jvm']
     return []
 
