diff --git a/docs/command.rst b/docs/command.rst
index c61993a..2d7cc0a 100644
--- a/docs/command.rst
+++ b/docs/command.rst
@@ -182,7 +182,7 @@ variable "folder" for rendering. For example,
 
     doctr-versions-menu --label '<releases>'  "{{ folder | replace('v', '', 1) }}" --label master '{{ folder }} (latest dev branch)'
 
-drops the initial "v" from the folder name of released versions (``v1.0.0`` →
+drops the initial "v" from the folder name of released versions (``v1.0.0`` ->
 ``1.0.0``) and appends a label " (latest dev branch)" to the label for the
 ``master`` folder.
 
diff --git a/src/doctr_versions_menu/cli.py b/src/doctr_versions_menu/cli.py
index 75c3fec..cb0d3e7 100644
--- a/src/doctr_versions_menu/cli.py
+++ b/src/doctr_versions_menu/cli.py
@@ -222,7 +222,7 @@ def main(
     file (``doctr-versions-menu.conf`` in the ``gh-pages`` root)
     instead of via command line flags. Every long-form-flag has a corresponding
     config file variable, obtained by replacing hyphens with underscores
-    (``--write-index-html`` → ``write_index_html``).
+    (``--write-index-html`` -> ``write_index_html``).
     """
     logging.basicConfig(level=logging.WARNING)
     logger = logging.getLogger(__name__)
