From 6231a28e013ac931f760ce1a860ce4d31991cba7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Sun, 18 Feb 2018 14:49:01 +0100
Subject: [PATCH] -shared breaks build on OSX

---
 python/setup.py.in | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/python/setup.py.in b/python/setup.py.in
index 4913e993..3ff70fb4 100644
--- a/python/setup.py.in
+++ b/python/setup.py.in
@@ -74,11 +74,6 @@ def _init_posix(init):
             config_vars['CC'] = 'gcc'
             config_vars['CXX'] = 'g++'
 
-        # FIXME raises hardening-no-fortify-functions lintian warning.
-        else:
-            # Non-Sun needs linkage with g++
-            config_vars['LDSHARED'] = 'g++ -shared -g -W -Wall -Wno-deprecated'
-
         config_vars['CFLAGS'] = '-g -W -Wall -Wno-deprecated -std=c++11'
         config_vars['OPT'] = '-g -W -Wall -Wno-deprecated -std=c++11'
 
-- 
2.16.1

