From f6e0083f6c59506e88764615bcd5aab9bb091005 Mon Sep 17 00:00:00 2001
From: Mate Soos <soos.mate@gmail.com>
Date: Sun, 7 Jan 2018 21:50:06 +0000
Subject: [PATCH] Trying to fix the python module build for OSX

---
 python/setup.py.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/python/setup.py.in b/python/setup.py.in
index 4913e993..37ad295b 100644
--- a/python/setup.py.in
+++ b/python/setup.py.in
@@ -46,10 +46,10 @@ def cleanup(dat):
         if elem != "":
             ret.append(elem)
 
-    if is_apple != "":
-        for x in ret:
-            if x == "-lpython" or x == "-lframework":
-                x = "-undefined dynamic_lookup"
+#    if is_apple != "":
+#        for x in ret:
+#            if x == "-lpython" or x == "-lframework":
+#                x = "-undefined dynamic_lookup"
 
     return ret
     # return []
-- 
2.16.1

