diff --git a/CMakeLists.txt b/CMakeLists.txt
index bfc1fc38..d800e017 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -544,28 +544,6 @@ endif()
 if (ENABLE_PYTHON_INTERFACE)
     # check python-config
     if (PYTHON_EXECUTABLE AND PYTHON_LIBRARIES AND PYTHON_INCLUDE_DIRS AND NOT COVERAGE)
-        SET(PY_CONFIG_EXE "${PYTHON_EXECUTABLE}.${PYTHON_VERSION_MINOR}-config")
-        message(STATUS "Will execute python config ${PY_CONFIG_EXE}")
-        execute_process(
-            COMMAND ${PY_CONFIG_EXE} --help
-            OUTPUT_VARIABLE PY_CONF_OPTIONS
-            RESULT_VARIABLE RES_VAR)
-
-        if(NOT ${RES_VAR} MATCHES "0")
-            message(WARNING "That python config did not work, trying another one...")
-            SET(PY_CONFIG_EXE "${PYTHON_EXECUTABLE}-config")
-            message(STATUS "Will execute python config ${PY_CONFIG_EXE}")
-            execute_process(
-                COMMAND ${PY_CONFIG_EXE} --help
-                OUTPUT_VARIABLE PY_CONF_OPTIONS
-                RESULT_VARIABLE RES_VAR)
-            if(NOT ${RES_VAR} MATCHES "0")
-                UNSET(PY_CONFIG_EXE)
-            endif()
-        endif()
-    endif()
-
-    if (PYTHON_EXECUTABLE AND PYTHON_LIBRARIES AND PYTHON_INCLUDE_DIRS AND PY_CONFIG_EXE AND NOT COVERAGE)
         message(STATUS "Found python interpreter, libs and header files")
         message(STATUS "Building python interface")
         add_subdirectory(python pycryptosat)
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index cdea949f..594cc622 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -20,10 +20,10 @@
 # THE SOFTWARE.
 
 execute_process(
-    COMMAND ${PY_CONFIG_EXE} --cflags
+    COMMAND ${PYTHON_EXECUTABLE} -c 'import sysconfig; print(sysconfig.get_config_vars("CFLAGS")[0])'
     OUTPUT_VARIABLE PY_C_CONFIG)
 execute_process(
-    COMMAND ${PY_CONFIG_EXE} --ldflags
+    COMMAND ${PYTHON_EXECUTABLE} -c 'import sysconfig; print(sysconfig.get_config_vars("LDFLAGS")[0])'
     OUTPUT_VARIABLE PY_LD_CONFIG)
 
 string(REPLACE "\n" " " PY_C_CONFIG ${PY_C_CONFIG})
