From 67f94da79943c051ad25d6b9d78c206b0c046fb7 Mon Sep 17 00:00:00 2001
From: Mate Soos <soos.mate@gmail.com>
Date: Wed, 10 Jan 2018 00:09:00 +0000
Subject: [PATCH] Fixing python-config check

---
 CMakeLists.txt        | 23 +++++++++++++++++++++++
 python/CMakeLists.txt | 13 -------------
 2 files changed, 23 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86163d87..e655d4be 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -541,7 +541,30 @@ else()
 endif()
 
 if (ENABLE_PYTHON_INTERFACE)
+    # check python-config
     if (PYTHON_EXECUTABLE AND PYTHON_LIBRARIES AND PYTHON_INCLUDE_DIRS AND NOT COVERAGE)
+        SET(PY_CONFIG_EXE "${PYTHON_EXECUTABLE}.${PYTHON_VERSION_MINOR}-config")
+        message(STATUS "Will execute python config ${PY_CONFIG_EXE}")
+        execute_process(
+            COMMAND ${PY_CONFIG_EXE} --help
+            OUTPUT_VARIABLE PY_CONF_OPTIONS
+            RESULT_VARIABLE RES_VAR)
+
+        if(NOT ${RES_VAR} MATCHES "0")
+            message(WARNING "That python config did not work, trying another one...")
+            SET(PY_CONFIG_EXE "${PYTHON_EXECUTABLE}-config")
+            message(STATUS "Will execute python config ${PY_CONFIG_EXE}")
+            execute_process(
+                COMMAND ${PY_CONFIG_EXE} --help
+                OUTPUT_VARIABLE PY_CONF_OPTIONS
+                RESULT_VARIABLE RES_VAR)
+            if(NOT ${RES_VAR} MATCHES "0")
+                UNSET(PY_CONFIG_EXE)
+            endif()
+        endif()
+    endif()
+
+    if (PYTHON_EXECUTABLE AND PYTHON_LIBRARIES AND PYTHON_INCLUDE_DIRS AND PY_CONFIG_EXE AND NOT COVERAGE)
         message(STATUS "Found python interpreter, libs and header files")
         message(STATUS "Building python interface")
         add_subdirectory(python pycryptosat)
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 0dd05396..cdea949f 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -19,19 +19,6 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
-SET(PY_CONFIG_EXE "${PYTHON_EXECUTABLE}.${PYTHON_VERSION_MINOR}-config")
-message(STATUS "Will execute python config ${PY_CONFIG_EXE}")
-execute_process(
-    COMMAND ${PY_CONFIG_EXE} --help
-    OUTPUT_VARIABLE PY_CONF_OPTIONS
-    RESULT_VARIABLE RES_VAR)
-
-if(NOT ${RES_VAR} MATCHES "0")
-    message(WARNING "That python config did not work, trying another one...")
-    SET(PY_CONFIG_EXE "${PYTHON_EXECUTABLE}-config")
-    message(STATUS "Will execute python config ${PY_CONFIG_EXE}")
-endif()
-
 execute_process(
     COMMAND ${PY_CONFIG_EXE} --cflags
     OUTPUT_VARIABLE PY_C_CONFIG)
-- 
2.16.1

