diff --git a/pyproject.toml b/pyproject.toml
index 7619914..636c3dc 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [build-system]
-requires      = ["maturin>=1.4"]
-build-backend = "maturin"
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
 
 [project]
 name = "taplo"
@@ -22,7 +22,6 @@ classifiers = [
   "Programming Language :: Rust",
   "Topic :: Software Development :: Quality Assurance",
 ]
-dynamic = ["version"]
 
 [tool.maturin]
 bindings      = "bin"
