From daf9121eaaf9e833c70d69002fc747d6568cf5eb Mon Sep 17 00:00:00 2001
From: Iain Russell <Iain.Russell@ecmwf.int>
Date: Mon, 7 Oct 2019 15:33:57 +0100
Subject: [PATCH] Fix for Windows build

---
 src/web/MagConfig.cc | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/web/MagConfig.cc b/src/web/MagConfig.cc
index 1c8e2683..44de60c9 100644
--- a/src/web/MagConfig.cc
+++ b/src/web/MagConfig.cc
@@ -230,10 +230,11 @@ void StyleLibrary::init() {
                     else
                         MagConfigHandler(path + "/" + current_, *this);
                 }
-                while (!_findnext(handle, &fileinfo))
-                    ;
-                _findclose(handle);
             }
+            while (!_findnext(handle, &fileinfo));
+
+            _findclose(handle);
+        }
 #endif
     }
 }
-- 
2.13.7

