--- CMakeLists.txt	2019-01-28 18:06:29.524677281 +0100
+++ src/CMakeLists.txt	2019-01-28 18:10:48.029933286 +0100
@@ -1,38 +1,23 @@
 cmake_minimum_required(VERSION 2.8.10)
 project(blosc_hdf5)
-include(ExternalProject)
 
-# options
-option(BUILD_TESTS
-    "Build test programs form the blosc filter" ON)
-
-option(BUILD_PLUGIN
-    "Build dynamically loadable plugin for HDF5 version > 1.8.11" ON)
-if(BUILD_PLUGIN)
-    set(PLUGIN_INSTALL_PATH "/usr/local/hdf5/lib/plugin" CACHE PATH 
-      "Where to install the dynamic HDF5-plugin")
-endif(BUILD_PLUGIN)
-
-set(BLOSC_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/blosc")
-set(BLOSC_INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/blosc")
-set(BLOSC_CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${BLOSC_INSTALL_DIR})
-
-message("BLOSC_PREFIX='${BLOSC_PREFIX}'")
-message("BLOSC_INSTALL_DIR='${BLOSC_INSTALL_DIR}'")
-message("BLOSC_CMAKE_ARGS='${BLOSC_CMAKE_ARGS}'")
-message("GIT_EXECUTABLE='${GIT_EXECUTABLE}'")
-
-ExternalProject_Add(project_blosc
-  PREFIX ${BLOSC_PREFIX}
-  GIT_REPOSITORY https://github.com/Blosc/c-blosc.git
-  INSTALL_DIR ${BLOSC_INSTALL_DIR}
-  CMAKE_ARGS ${BLOSC_CMAKE_ARGS}
-)
+# find local blosc install
+#SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+#find_package(blosc REQUIRED)
 
+find_path(BLOSC_INCLUDE_DIR "blosc.h")
+find_library(BLOSC_LIBRARY NAMES blosc)
+
+
+message("BLOSC lib: ${BLOSC_LIBRARY}")
+message("BLOSC include: ${BLOSC_INCLUDE_DIRS}")
+
+# TODO: set these paths for OSX and Windoze
+set(PLUGIN_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}/lib/hdf5/plugin" CACHE PATH
+            "Where to install the dynamic HDF5-plugin")
 
 # sources
-set(SOURCES src/blosc_filter.c)
-set(PLUGIN_SOURCES src/blosc_filter.c src/blosc_plugin.c )
+set(SOURCES src/blosc_filter.c src/blosc_plugin.c)
 
 # dependencies
 if(MSVC)
@@ -49,40 +34,19 @@
 endif(MSVC)
 include_directories(${HDF5_INCLUDE_DIRS})
 
-
+# TODO: below block only needed for internal blosc build.
 # add blosc libraries
 add_library(blosc_shared SHARED IMPORTED)
-set_property(TARGET blosc_shared PROPERTY IMPORTED_LOCATION ${BLOSC_INSTALL_DIR}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}blosc${CMAKE_SHARED_LIBRARY_SUFFIX})
-add_dependencies(blosc_shared project_blosc)
-include_directories(${BLOSC_INSTALL_DIR}/include)
+#set_property(TARGET blosc_shared PROPERTY IMPORTED_LOCATION ${BLOSC_INSTALL_DIR}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}blosc${CMAKE_SHARED_LIBRARY_SUFFIX})
+#add_dependencies(blosc_shared project_blosc)
+#include_directories(${BLOSC_INSTALL_DIR}/include)
+###
 
 add_library(blosc_filter_shared SHARED ${SOURCES})
-set_target_properties(
-  blosc_filter_shared PROPERTIES OUTPUT_NAME blosc_filter)
-target_link_libraries(blosc_filter_shared blosc_shared ${HDF5_LIBRARIES})
-
-if(BUILD_PLUGIN)
-    add_library(blosc_plugin_shared SHARED ${PLUGIN_SOURCES})
-    set_target_properties(
-      blosc_plugin_shared PROPERTIES OUTPUT_NAME H5Zblosc)
-    target_link_libraries(blosc_plugin_shared blosc_shared ${HDF5_LIBRARIES})
-
-    install(TARGETS blosc_plugin_shared DESTINATION ${PLUGIN_INSTALL_PATH} COMPONENT HDF5_FILTER_DEV)
-endif(BUILD_PLUGIN)
+set_target_properties(blosc_filter_shared PROPERTIES OUTPUT_NAME blosc_filter)
+target_link_libraries(blosc_filter_shared
+        ${BLOSC_LIBRARY}
+        ${HDF5_LIBRARIES})
 
 # install
-install(FILES src/blosc_filter.h DESTINATION include COMPONENT HDF5_FILTER_DEV)
-install(TARGETS blosc_filter_shared DESTINATION lib COMPONENT HDF5_FILTER_DEV)
-
-
-# test
-message("LINK LIBRARIES='blosc_filter_shared ${HDF5_LIBRARIES}'")
-if(BUILD_TESTS)
-    enable_testing()
-    set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
-    find_package(Threads REQUIRED)
-    set(LIBS ${LIBS} ${CMAKE_THREAD_LIBS_INIT})
-    add_executable(example src/example.c)
-    target_link_libraries(example blosc_filter_shared ${HDF5_LIBRARIES} ${LIBS})
-    add_test(test_hdf5_filter example)
-endif(BUILD_TESTS)
+install(TARGETS blosc_filter_shared DESTINATION ${PLUGIN_INSTALL_PATH} COMPONENT HDF5_FILTER_DEV)
