diff --git a/src/Utils/CMakeLists.txt b/src/Utils/CMakeLists.txt
index f435593..cfc8fb7 100644
--- a/src/Utils/CMakeLists.txt
+++ b/src/Utils/CMakeLists.txt
@@ -29,6 +29,7 @@ if(SCINE_PARALLELIZE)
   find_package(OpenMP REQUIRED)
 endif()
 
+if(NOT SCINE_SKIP_LIBRARY)
 # Obey standard CMake behavior regarding shared/static libraries
 add_library(UtilsOS ${UTILS_HEADERS} ${UTILS_CPPS})
 if(NOT BUILD_SHARED_LIBS)
@@ -119,6 +120,10 @@ if(SCINE_BUILD_TESTS)
     DESTINATION ${CMAKE_CURRENT_BINARY_DIR}
   )
 endif()
+else()
+include(ImportUtilsOS)
+import_utils_os()
+endif()
 
 # Python bindings
 if(SCINE_BUILD_PYTHON_BINDINGS)
@@ -164,11 +169,13 @@ if(SCINE_BUILD_PYTHON_BINDINGS)
     ${CMAKE_CURRENT_BINARY_DIR}/scine_utilities/__init__.py
   )
   file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/Python/README.rst DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+  if(NOT SCINE_SKIP_LIBRARY)
   add_custom_command(TARGET UtilsOSModule POST_BUILD
     COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:UtilsOSModule> ${CMAKE_CURRENT_BINARY_DIR}/scine_utilities
     COMMENT "Copying UtilsOS module into python package directory"
   )
   add_dependencies(scine_utilities UtilsOSModule)
+  endif()
 
   # Figure out which targets need to be copied along
   set(_py_targets_to_copy Scine::Core)  # Core is always shared
@@ -187,11 +194,13 @@ if(SCINE_BUILD_PYTHON_BINDINGS)
     string(APPEND utils_PY_DEPS ", \"${_target_filename}\"")
   endforeach()
 
+  if(NOT SCINE_SKIP_LIBRARY)
   add_custom_command(TARGET UtilsOS POST_BUILD
     COMMAND ${CMAKE_COMMAND} -E copy ${_py_target_gen_exprs} ${CMAKE_CURRENT_BINARY_DIR}/scine_utilities
     COMMENT "Copying dependent shared libraries into python package directory"
   )
   message(STATUS "Targets to copy for python bindings: ${_py_targets_to_copy}")
+  endif()
   unset(_py_targets_to_copy)
 
   # Typing stubs
@@ -227,10 +236,12 @@ if(SCINE_BUILD_PYTHON_BINDINGS)
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/doc-py
     DOCTEST
   )
+  if(NOT SCINE_SKIP_LIBRARY)
   # The UtilsOSModule is technically also a dependency of the documentation
   if(TARGET scine_utilitiesDocumentation)
     add_dependencies(scine_utilitiesDocumentation UtilsOSModule)
   endif()
+  endif()
 endif()
 
 if(SCINE_BUILD_PYTHON_BINDINGS)
