diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4038142..0db41d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,8 +97,7 @@ else()
                 MODULES ${modules}
                 PYTHON_PACKAGE "vtkbool"
                 BUILD_STATIC OFF
-                INSTALL_HEADERS OFF
-                MODULE_DESTINATION "${VTK_PYTHON_SITE_PACKAGES_SUFFIX}")
+                INSTALL_HEADERS OFF)
 
             if (VTKBOOL_TESTING)
                 include(CTest)
