diff --git a/setup.cfg b/setup.cfg
index 1065384..6a49338 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -21,11 +21,12 @@ remote_data_strict = True
 
 [ah_bootstrap]
 auto_use = True
+offline = True
 
 [metadata]
 package_name = sbpy
 description = A Python Module for Small-Body Planetary Astronomy
-long_description = 
+long_description =
 author = sbpy team
 author_email = msk@astro.umd.edu
 license = BSD 3-Clause
@@ -62,8 +63,3 @@ ignore = E402,E741,E226,E501
 # Excluding files that are directly copied from the package template or
 # generated
 exclude = _astropy_init.py,version.py,ah_bootstrap.py
-
-
-[entry_points]
-
-astropy-package-template-example = packagename.example_mod:main
