From 6c2a0cda046d972c6e2ab25553e01f42e6be7667 Mon Sep 17 00:00:00 2001
From: Daniel Bast <2790401+dbast@users.noreply.github.com>
Date: Fri, 11 Oct 2024 10:40:10 +0200
Subject: [PATCH 1/1] Patch libncursesw naming on darwin

---
 configure | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/configure b/configure
index dd663c3..fb1a5d3 100755
--- a/configure
+++ b/configure
@@ -7620,11 +7620,6 @@ fi
 
 if test "$utf8" = "yes"; then
   libncursesw=ncursesw
-  # Simply called libncurses on OS X
-  case "$host_os" in
-    *darwin*) libncursesw=ncurses
-    ;;
-  esac
 
   as_ac_Lib=`printf "%s\n" "ac_cv_lib_$libncursesw""_mvaddwstr" | sed "$as_sed_sh"`
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for mvaddwstr in -l$libncursesw" >&5
-- 
2.47.0

