--- a/src/flash/nor/rsl10.c
+++ b/src/flash/nor/rsl10.c
@@ -158,2 +158,2 @@
-	struct rsl10_bank *nbank = bank->driver_priv;
-	struct rsl10_info *chip  = nbank->chip;
+	/* struct rsl10_bank *nbank = bank->driver_priv; */
+	/* struct rsl10_info *chip  = nbank->chip; */
--- a/src/target/adi_v5_swd.c
+++ b/src/target/adi_v5_swd.c
@@ -483,1 +383,1 @@
-	const struct swd_driver *swd = adiv5_dap_swd_driver(dap);
+	/* const struct swd_driver *swd = adiv5_dap_swd_driver(dap); */
