--- a/configure.ac 2019-01-01 14:00:00.000000000 +0100
+++ b/configure.ac 2019-01-01 14:00:00.000000000 +0100
@@ -667,0 +667,17 @@
+AS_IF([test -d "$PREFIX/lib64"], [
+  LDFLAGS="$LDFLAGS -Wl,-rpath -Wl,$PREFIX/lib64 -Wl,-rpath-link -Wl,$PREFIX/lib64"
+])
+if test "x$use_libjaylink" = "xyes"; then
+  LIBJAYLINK_LIBS="-lusb-1.0 -ljaylink"
+  AC_SUBST(LIBJAYLINK_LIBS)
+fi
+PKG_CHECK_MODULES([JIMTCL], [jimtcl], [
+  # This will set JIMTCL_LIBS which should include -lssl -lcrypto if needed
+  use_external_jimtcl=yes
+], [
+  use_external_jimtcl=no
+])
+
+AS_IF([test "x$use_external_jimtcl" = "xyes"], [
+  LIBS="$JIMTCL_LIBS $LIBS"
+])
