--- python-ghostscript-v0.7.orig/setup.cfg	2023-12-20 13:18:30.452497083 -0600
+++ python-ghostscript-v0.7/setup.cfg	2023-12-20 13:18:40.747546844 -0600
@@ -22,7 +22,7 @@
 
 [metadata]
 name = ghostscript
-version = attr:ghostscript.__version__
+version = 0.7
 author = Hartmut Goebel
 author_email = h.goebel@crazy-compilers.com
 description = Interface to the Ghostscript C-API, both high- and low-level, based on ctypes
