--- python-ghostscript-v0.7.orig/ghostscript/_gsprint.py	2023-12-20 13:18:30.453497088 -0600
+++ python-ghostscript-v0.7/ghostscript/_gsprint.py	2023-12-20 13:45:31.513348016 -0600
@@ -487,6 +487,8 @@
     from distutils.version import LooseVersion
     import os
 
+    return os.environ.get("GS_DLL", os.environ["CONDA_PREFIX"] + "\\library\\bin\\gsdll64.dll")
+
     dlls = []
     # Look up different variants of Ghostscript and take the highest
     # version for which the DLL is to be found in the filesystem.
