diff --git a/cmake/LocatePythonLibs.cmake b/cmake/LocatePythonLibs.cmake
index bf8c2b8..0fed3a4 100644
--- a/cmake/LocatePythonLibs.cmake
+++ b/cmake/LocatePythonLibs.cmake
@@ -17,10 +17,10 @@ message(STATUS "Python include path: ${PYTHON_INCLUDE_DIR}")
 if (NOT DEFINED ${PYTHON_LIB_PATH})
   # Find the Python lib dir
   execute_process(COMMAND ${PYTHON_BIN}
-                  -c "from distutils.sysconfig import get_config_var; print(get_config_var('LIBDIR'))"
-                  OUTPUT_VARIABLE PYTHON_LIB_DIR
+                  -c "import sys; import os; print(os.path.join(sys.prefix, 'libs'))"
+                  OUTPUT_VARIABLE PYTHON_LIB_PATH
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
-  message(STATUS "Python library path: ${PYTHON_LIB_DIR}")
+  message(STATUS "Python library path: ${PYTHON_LIB_PATH}")
 
   # Get the Python version
   execute_process(COMMAND ${PYTHON_BIN} -c "from distutils.sysconfig import get_python_version; print(get_python_version())"
@@ -36,7 +36,7 @@ if (NOT DEFINED ${PYTHON_LIB_PATH})
   set (PYTHON_LIB_PREFIX python${PYTHON_VERSION})
   find_library(PYTHON_LIB_PATH
                NAMES ${PYTHON_LIB_PREFIX} ${PYTHON_LIB_PREFIX}m ${PYTHON_LIB_PREFIX}dm
-               PATHS ${PYTHON_LIB_DIR}
+               PATHS ${PYTHON_LIB_PATH}
                NO_DEFAULT_PATH)
 
   # If the targeted search fails, look in cmake default locations
@@ -44,7 +44,7 @@ if (NOT DEFINED ${PYTHON_LIB_PATH})
     message(STATUS "Expanding search for libpython")
     find_library(PYTHON_LIB_PATH
                  NAMES ${PYTHON_LIB_PREFIX} ${PYTHON_LIB_PREFIX}m ${PYTHON_LIB_PREFIX}dm
-                 PATHS ${PYTHON_LIB_DIR})
+                 PATHS ${PYTHON_LIB_PATH})
   endif()
 endif()
 message(STATUS "Python library path: ${PYTHON_LIB_PATH}")
