diff --git a/cmake/LocatePythonLibs.cmake b/cmake/LocatePythonLibs.cmake
index bf8c2b8..f4889e5 100644
--- a/cmake/LocatePythonLibs.cmake
+++ b/cmake/LocatePythonLibs.cmake
@@ -7,7 +7,7 @@ message(STATUS "Python binary: ${PYTHON_BIN}")
 # Python include path
 if (NOT DEFINED ${PYTHON_INCLUDE_DIR})
   execute_process(COMMAND ${PYTHON_BIN}
-                  -c "from distutils.sysconfig import get_python_inc; print(get_python_inc())"
+                  -c "import sys; import os; print(os.path.join(sys.prefix, 'libs'))"
                   OUTPUT_VARIABLE PYTHON_INCLUDE_DIR
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
