diff --git a/cmake/LocatePythonLibs.cmake b/cmake/LocatePythonLibs.cmake
index bf8c2b8..9f8bc74 100644
--- a/cmake/LocatePythonLibs.cmake
+++ b/cmake/LocatePythonLibs.cmake
@@ -17,13 +17,13 @@ message(STATUS "Python include path: ${PYTHON_INCLUDE_DIR}")
 if (NOT DEFINED ${PYTHON_LIB_PATH})
   # Find the Python lib dir
   execute_process(COMMAND ${PYTHON_BIN}
-                  -c "from distutils.sysconfig import get_config_var; print(get_config_var('LIBDIR'))"
+                  -c "import sys; import os; print(os.path.join(sys.prefix, 'libs'))"
                   OUTPUT_VARIABLE PYTHON_LIB_DIR
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
   message(STATUS "Python library path: ${PYTHON_LIB_DIR}")
 
   # Get the Python version
-  execute_process(COMMAND ${PYTHON_BIN} -c "from distutils.sysconfig import get_python_version; print(get_python_version())"
+  execute_process(COMMAND ${PYTHON_BIN} -c "from distutils.sysconfig import get_python_version; print(get_python_version().replace('.',''))"
                   OUTPUT_VARIABLE PYTHON_VERSION
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
   message(STATUS "Python version: ${PYTHON_VERSION}")
