diff --git a/cmake/LocatePythonLibs.cmake b/cmake/LocatePythonLibs.cmake
index bf8c2b8..3087b99 100644
--- a/cmake/LocatePythonLibs.cmake
+++ b/cmake/LocatePythonLibs.cmake
@@ -11,7 +11,6 @@ if (NOT DEFINED ${PYTHON_INCLUDE_DIR})
                   OUTPUT_VARIABLE PYTHON_INCLUDE_DIR
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
-message(STATUS "Python include path: ${PYTHON_INCLUDE_DIR}")
 
 # Python library path
 if (NOT DEFINED ${PYTHON_LIB_PATH})
@@ -20,7 +19,6 @@ if (NOT DEFINED ${PYTHON_LIB_PATH})
                   -c "from distutils.sysconfig import get_config_var; print(get_config_var('LIBDIR'))"
                   OUTPUT_VARIABLE PYTHON_LIB_DIR
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
-  message(STATUS "Python library path: ${PYTHON_LIB_DIR}")
 
   # Get the Python version
   execute_process(COMMAND ${PYTHON_BIN} -c "from distutils.sysconfig import get_python_version; print(get_python_version())"
@@ -46,6 +44,16 @@ if (NOT DEFINED ${PYTHON_LIB_PATH})
                  NAMES ${PYTHON_LIB_PREFIX} ${PYTHON_LIB_PREFIX}m ${PYTHON_LIB_PREFIX}dm
                  PATHS ${PYTHON_LIB_DIR})
   endif()
+
+  # If the targeted search fails, look in cmake default locations
+  if (NOT PYTHON_LIB_PATH)
+    message(STATUS "Expanding search for Windows")
+    execute_process(COMMAND ${PYTHON_BIN}
+                    -c "import sys; import os; print(os.path.join(sys.prefix, 'libs'))"
+                    OUTPUT_VARIABLE PYTHON_INCLUDE_DIR
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  endif()
+  message(STATUS "Python library path: ${PYTHON_LIB_DIR}")
 endif()
 message(STATUS "Python library path: ${PYTHON_LIB_PATH}")
 
