From 8830f7a9ea4bc4426ed07727081b2ddbb5dbb24e Mon Sep 17 00:00:00 2001
From: tnigon <nigo0024@umn.edu>
Date: Wed, 21 Apr 2021 07:48:55 -0500
Subject: [PATCH] exclude tests from setup.py for conda forge compatibility

---
 setup.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index 8d6398b..675d851 100644
--- a/setup.py
+++ b/setup.py
@@ -18,8 +18,8 @@ setup(
     version=version,
     description='A Python package for finite difference derivatives in any number of dimensions.',
     long_description="""A Python package for finite difference derivatives in any number of dimensions.
-    
-    Features: 
+
+    Features:
 
         * Differentiate arrays of any number of dimensions along any axis
         * Partial derivatives of any desired order
@@ -52,7 +52,7 @@ setup(
         'Programming Language :: Python :: 3.9',
     ],
     keywords=['finite-differences',  'numerical-derivatives', 'scientific-computing'],
-    packages=find_packages(),
+    packages=find_packages(exclude=("tests",)),
     package_dir={name: name},
     include_package_data=True,
     install_requires=['numpy', 'scipy', 'sympy'],
-- 
2.8.3.windows.1

