diff --git a/python/common/CMakeLists.txt b/python/common/CMakeLists.txt
index 31db045..943552f 100644
--- a/python/common/CMakeLists.txt
+++ b/python/common/CMakeLists.txt
@@ -33,6 +33,6 @@ target_link_libraries(cvcuda_python_common
         nvcv_types
         nvcv_util
         cvcuda
-        CUDA::cudart_static
+        CUDA::cudart
         pybind11::pybind11
 )
diff --git a/python/mod_cvcuda/CMakeLists.txt b/python/mod_cvcuda/CMakeLists.txt
index 5c15aa9..56113b5 100644
--- a/python/mod_cvcuda/CMakeLists.txt
+++ b/python/mod_cvcuda/CMakeLists.txt
@@ -149,7 +149,7 @@ target_include_directories(cvcuda_module_python

 target_link_libraries(cvcuda_module_python
     PRIVATE
-        CUDA::cudart_static
+        CUDA::cudart
         nvcv_types
         nvcv_util_compat
         cvcuda
diff --git a/src/cvcuda/CMakeLists.txt b/src/cvcuda/CMakeLists.txt
index 64c6c3d..dcc382e 100644
--- a/src/cvcuda/CMakeLists.txt
+++ b/src/cvcuda/CMakeLists.txt
@@ -99,7 +99,7 @@ add_library(cvcuda SHARED

 target_link_libraries(cvcuda
     PUBLIC
-        CUDA::cudart_static
+        CUDA::cudart
         nvcv_types
         -lrt
     PRIVATE
diff --git a/src/cvcuda/priv/CMakeLists.txt b/src/cvcuda/priv/CMakeLists.txt
index deac31c..26233db 100644
--- a/src/cvcuda/priv/CMakeLists.txt
+++ b/src/cvcuda/priv/CMakeLists.txt
@@ -102,9 +102,9 @@ target_link_libraries(cvcuda_priv
         cvcuda_headers
         nvcv_util_sanitizer
         cvcuda_legacy
-        CUDA::cudart_static
-	    CUDA::cusolver_static
-        CUDA::cublas_static
-        CUDA::cublasLt_static
+        CUDA::cudart
+	    CUDA::cusolver
+        CUDA::cublas
+        CUDA::cublasLt
         -lrt
 )
diff --git a/src/cvcuda/priv/legacy/CMakeLists.txt b/src/cvcuda/priv/legacy/CMakeLists.txt
index 4a265c8..1dccd25 100644
--- a/src/cvcuda/priv/legacy/CMakeLists.txt
+++ b/src/cvcuda/priv/legacy/CMakeLists.txt
@@ -110,7 +110,7 @@ add_library(cvcuda_legacy STATIC

 target_link_libraries(cvcuda_legacy
     PUBLIC
-        CUDA::cudart_static
+        CUDA::cudart
         nvcv_types
         nvcv_util
         cvcuda_headers
diff --git a/src/cvcuda/util/CMakeLists.txt b/src/cvcuda/util/CMakeLists.txt
index 346ab73..dad2b17 100644
--- a/src/cvcuda/util/CMakeLists.txt
+++ b/src/cvcuda/util/CMakeLists.txt
@@ -24,6 +24,6 @@ add_library(cvcuda_util STATIC
 target_link_libraries(cvcuda_util
     PUBLIC
         nvcv_util
-        CUDA::cudart_static
+        CUDA::cudart
         -lrt
 )
diff --git a/src/nvcv/src/priv/CMakeLists.txt b/src/nvcv/src/priv/CMakeLists.txt
index 7e298d9..98c6ce0 100644
--- a/src/nvcv/src/priv/CMakeLists.txt
+++ b/src/nvcv/src/priv/CMakeLists.txt
@@ -52,7 +52,7 @@ target_link_libraries(nvcv_types_priv
     PUBLIC
         nvcv_types_headers
         nvcv_util_sanitizer
-        CUDA::cudart_static
+        CUDA::cudart
 )

 if(NOT PLATFORM_IS_QNX)
diff --git a/src/nvcv/util/CMakeLists.txt b/src/nvcv/util/CMakeLists.txt
index aa72802..81d4af5 100644
--- a/src/nvcv/util/CMakeLists.txt
+++ b/src/nvcv/util/CMakeLists.txt
@@ -192,7 +192,7 @@ target_link_libraries(nvcv_util
     PUBLIC
         nvcv_util_sanitizer
         nvcv_types_headers
-        CUDA::cudart_static
+        CUDA::cudart
         nvcv_util_compat
 )

diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index ea0e91d..53eebd3 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -36,7 +36,7 @@ target_link_libraries(nvcv_test_main
         nvcv_util_compat
     PRIVATE
         nvcv_types
-        CUDA::cudart_static
+        CUDA::cudart
         -lrt
 )

