From 726caa6ff11321b208ab0ae9fb10c615435b6e72 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Thu, 8 May 2025 09:25:31 -0700
Subject: [PATCH] Adjust dependencies

---
 pyproject.toml | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 6d284ae1..26ad737c 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -20,7 +20,6 @@ dependencies = [
     "cryptography>=38",
     "tyro>=0.6.6",
     "gdown>=4.6.0",
-    "ninja>=1.10",
     "h5py>=2.9.0",
     "imageio>=2.21.1",
     'importlib-metadata>=6.0.0; python_version < "3.10"',
@@ -33,10 +32,10 @@ dependencies = [
     "msgpack_numpy>=0.4.8",
     "nerfacc==0.5.2",
     "open3d>=0.16.0",
-    "opencv-python-headless==4.10.0.84",
+    "opencv==4.10",
     "Pillow>=10.3.0",
     "plotly>=5.7.0",
-    "protobuf<=3.20.3,!=3.20.0",
+    "protobuf<6,!=3.20.0",
     # TODO(1480) enable when pycolmap windows wheels are available
     # "pycolmap==0.3.0",
     # TODO(3461) and external issue cnr-isti-vclab/PyMeshLab/issues/398: Unrestrict Windows version when it isn't broken anymore.
-- 
2.49.0.windows.1

