From 44aa5fb9e44d729af6fb205226dea1afb10c3b58 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Wed, 2 Apr 2025 18:00:35 -0700
Subject: [PATCH] Remove ninja from install dependency

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 6d284ae1..dcd153c7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -20,7 +20,6 @@ dependencies = [
     "cryptography>=38",
     "tyro>=0.6.6",
     "gdown>=4.6.0",
-    "ninja>=1.10",
     "h5py>=2.9.0",
     "imageio>=2.21.1",
     'importlib-metadata>=6.0.0; python_version < "3.10"',
-- 
2.20.1.windows.1

