diff --git a/napari_file_watcher/main_module.py b/napari_file_watcher/main_module.py
index b0f5f80..601064b 100644
--- a/napari_file_watcher/main_module.py
+++ b/napari_file_watcher/main_module.py
@@ -1,6 +1,6 @@
 import os
 from qtpy.QtWidgets import QWidget, QTreeWidget, QTreeWidgetItem
-from PyQt5 import Qsci
+from qtpy import Qsci
 from qtpy import QtGui, QtWidgets, QtCore
 
 from datetime import datetime
