Avoid using uv for running tests

--- a/Makefile
+++ b/Makefile
@@ -180,8 +180,8 @@
 	./scripts/publish-release.sh

 # -k test_vec0_update
-test-loadable: loadable
-	uv run --managed-python --project tests pytest -vv -s -x . tests/test-*.py
+test-loadable:
+	python -m pytest -vvv -s -x tests/test-*.py

 test-loadable-snapshot-update: loadable
 	uv run --managed-python --project tests pytest -vv tests/test-loadable.py --snapshot-update
