diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d85488..63ab834 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,9 +24,9 @@ set(SMESH_VERSION_TWEAK 0)
 set(BUILD_SHARED_LIBS ON)
 
 # Compiler flags
-if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
+if(CMAKE_COMPILER_IS_GNUCXX)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-endif(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
+endif(CMAKE_COMPILER_IS_GNUCXX)
 
 # --------------------------------------------------------------------------- #
 # OpenCASCADE
