--- pygeos-0.3.orig/setup.py	2019-07-08 12:43:17.000000000 -0300
+++ pygeos-0.3/setup.py	2019-09-09 20:12:49.288652657 -0300
@@ -2,11 +2,17 @@
 from setuptools import setup, Extension
 import geosconfig
 
+import numpy
+
+include_dirs = [os.environ.get("GEOS_INCLUDE_PATH")]
+library_dirs = [os.environ.get("GEOS_LIBRARY_PATH")]
+include_dirs.append(numpy.get_include())
+
 module_ufuncs = Extension(
     'pygeos.ufuncs',
     sources=['src/ufuncs.c'],
-    include_dirs=geosconfig.include_dirs,
-    library_dirs=geosconfig.library_dirs,
+    include_dirs=include_dirs,
+    library_dirs=library_dirs,
     libraries=geosconfig.libraries,
     extra_link_args=geosconfig.extra_link_args
 )
