diff --git a/src/ufuncs.c b/src/ufuncs.c
index a0e5c03..68a12c5 100644
--- a/src/ufuncs.c
+++ b/src/ufuncs.c
@@ -5,9 +5,10 @@
 #include <math.h>
 #include <geos_c.h>
 #include <structmember.h>
-#include "numpy/ndarraytypes.h"
-#include "numpy/ufuncobject.h"
-#include "numpy/npy_3kcompat.h"
+
+#include <numpy/ndarraytypes.h>
+#include <numpy/ufuncobject.h>
+#include <numpy/npy_3kcompat.h>
 
 #include "fast_loop_macros.h"
