diff --git a/setup.py b/setup.py
index ae5a502..6012669 100644
--- a/setup.py
+++ b/setup.py
@@ -65,7 +65,7 @@ args = dict(
     description="Tephigram plotting in Python",
     long_description=long_description(),
     long_description_content_type="text/markdown",
-    setup_requires=["setuptools>=40.8.0", "pytest-runner"],
+    setup_requires=["setuptools>=40.8.0"],
     install_requires=load("requirements.txt"),
     tests_require=load("requirements-dev.txt"),
     test_suite=f"{NAME}.tests",

