diff --git setup.py setup.py
index 9cb0667..71b3f25 100644
--- setup.py
+++ setup.py
@@ -299,7 +299,7 @@ class ExtensionConfiguration(object):
         if str.find(sys.version, "MSC") >= 0:
             self.extra_compile_args.append("-O2")
             self.extra_compile_args.append("-GF")        # enable read-only string pooling
-            self.extra_compile_args.append("-WX")        # treat warnings as errors
+            self.extra_compile_args.append("-W2")        # treat warnings as errors
             p = subprocess.Popen(['cl.exe'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             out, err = p.communicate()
             match = re.search(r'Version (\d+)', err.decode().split("\n")[0])
