diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5d8aeb0..0b084af 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -114,11 +114,6 @@ endif()
 # * CMAKE_INSTALL_BINDIR
 # * CMAKE_INSTALL_INCLUDEDIR
 include(GNUInstallDirs)
-if(UNIX)
-    set(CMAKE_INSTALL_LIBDIR "lib64")  # use lib64 instead of lib
-else()
-    set(CMAKE_INSTALL_LIBDIR "lib")
-endif()
 set(CMAKE_INSTALL_FULL_LIBDIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")  # adjust accordingly
 
 # Layout. This works for all platforms:
@@ -190,4 +185,4 @@ install(
     COMPONENT lib
 )
 
-# }
\ No newline at end of file
+# }
