From 0cd205438388a4cd8d1a597c45b01bd3fadeaf92 Mon Sep 17 00:00:00 2001
From: kiwi0fruit <peter.zagubisalo@gmail.com>
Date: Sun, 10 Feb 2019 13:13:01 +0700
Subject: [PATCH] com

---
 pymathjax/conda.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pymathjax/conda.py b/pymathjax/conda.py
index 12c26d2..985df44 100644
--- a/pymathjax/conda.py
+++ b/pymathjax/conda.py
@@ -1 +1 @@
-conda = False
+conda = True
-- 
2.16.1.windows.1

