From 0068e3f0ef1c0858d501f5185a716a75b0122d6d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 13 Oct 2023 16:16:50 +0200
Subject: [PATCH] fix makefile

---
 v1.3/Makefile | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/v1.3/Makefile b/v1.3/Makefile
index 4cc0bcd..fddfc47 100755
--- a/v1.3/Makefile
+++ b/v1.3/Makefile
@@ -25,22 +25,22 @@
 #***************************************************************************************************************************
 
 nanosculpt:  mk_config_param.o mk_config_calc.o mk_config.o inhedron.o simplTri.o
-	gcc  -O4 -g -o nanoSCULPT -lm mk_config.o mk_config_param.o mk_config_calc.o inhedron.o simplTri.o
+	${LD} ${LDFLAGS_LD} -o nanoSCULPT -ldl -lm mk_config.o mk_config_param.o mk_config_calc.o inhedron.o simplTri.o
 
 mk_config_param.o:  mk_config_param.c mk_config_globals.h
-	gcc  -O4 -g -static -Wall -c  mk_config_param.c 
+	${CC} ${CFLAGS} -c mk_config_param.c 
 
 mk_config_calc.o:  mk_config_calc.c  mk_config_globals.h 
-	gcc  -O4 -g -static -Wall -c mk_config_calc.c 
+	${CC} ${CFLAGS} -c mk_config_calc.c 
 
 mk_config.o: mk_config.c  mk_config_globals.h
-	gcc  -O4 -g -static -Wall -c mk_config.c
+	${CC} ${CFLAGS} -c mk_config.c
 
 inhedron.o: mk_config_globals.h inhedron.c
-	gcc  -O4 -g -static -Wall -c inhedron.c
+	${CC} ${CFLAGS} -c inhedron.c -lm
 
 simplTri.o: mk_config_globals.h simplTri.c
-	gcc -O4 -g -static -Wall -c simplTri.c
+	${CC} ${CFLAGS} -c simplTri.c
 
 
 clean:
