--- linecache2/tests/inspect_fodder2.py
+++ linecache2/tests/inspect_fodder2.py
@@ -99,16 +99,16 @@
 method_in_dynamic_class = f().g

 #line 101
-def keyworded(*arg1, arg2=1):
-    pass
+# def keyworded(*arg1, arg2=1):
+#     pass

 #line 105
-def annotated(arg1: list):
-    pass
+# def annotated(arg1: list):
+#     pass

 #line 109
-def keyword_only_arg(*, arg):
-    pass
+# def keyword_only_arg(*, arg):
+#     pass

 from functools import wraps
