diff --git a/pyproject.toml b/pyproject.toml
index 2111a21..b1d6026 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,10 +1,10 @@
 [build-system]
-requires = ["hatchling>=1.1.0", "typing-extensions", "hatch-ci", "jinja2"]
+requires = ["hatchling>=1.1.0", "typing-extensions", "jinja2"]
 build-backend = "hatchling.build"
 
 [project]
 name = "hatch-ci"
-dynamic = ["version"]
+version = "0.1.0"
 description = "Hatch plugin for ci system versioning"
 readme = "README.md"
 license = { text = "MIT" }
@@ -43,27 +43,6 @@ hatch-ci = "hatch_ci:script.main"
 Issues = "https://github.com/cav71/hatch-ci/issues"
 Source = "https://github.com/cav71/hatch-ci"
 
-[project.entry-points.hatch]
-ci = "hatch_ci.hooks"
-
-[tool.hatch.version]
-source = "ci"
-version-file = "src/hatch_ci/__init__.py"
-paths = [ "README.md" ]
-fixers = [
-    # for the github actions
-    [ 're:(https://github.com/.+/actions/workflows/)(master)(.yml/badge.svg)',
-        '\1{{ ctx.workflow }}\3' ],
-    [ 're:(https://github.com/.+/actions)/(workflows/)(master.yml)(?!/)',
-        '\1/runs/{{ ctx.runid }}' ],
-    
-    # for the codecov part
-    [ 're:(https://codecov.io/gh/.+/branch)/master(/graph/badge.svg[?]token=.+)',
-        '\1/{{ ctx.branch|urlquote }}\2' ],
-    [ 're:(https://codecov.io/gh/.+/branch)/master(?!/)',
-        '\1/{{ ctx.branch|urlquote }}' ]
-]
-
 [tool.ruff]
 target-version = "py38"
 line-length = 88
