diff -urN cctools.orig/otool/main.c cctools/otool/main.c
--- cctools.orig/otool/main.c	2017-08-01 13:46:44.000000000 +0100
+++ cctools/otool/main.c	2017-08-01 13:48:11.000000000 +0100
@@ -4087,38 +4087,4 @@
 	}
 }
 
-#ifndef __DYNAMIC__
-/*
- * To avoid linking in libm.  These variables are defined as they are used in
- * pthread_init() to put in place a fast sqrt().
- */
-size_t hw_sqrt_len = 0;
-
-double
-sqrt(double x)
-{
-	return(0.0);
-}
-double
-hw_sqrt(double x)
-{
-	return(0.0);
-}
-
-/*
- * More stubs to avoid linking in libm.  This works as along as we don't use
- * long doubles.
- */
-int32_t
-__fpclassifyd(double x)
-{
-	return(0);
-}
-int32_t
-__fpclassify(long double x)
-{
-	return(0);
-}
-#endif /* !defined(__DYNAMIC__) */
-
 #endif /* !defined(LLVM_OTOOL) */
