--- include/mach/i386/thread_status.h       2018-10-04 17:13:10.000000000 -0500
+++ include/mach/i386/thread_status.h     2019-09-16 01:18:31.483854331 -0500
@@ -67,12 +67,13 @@
 #ifndef        _MACH_I386_THREAD_STATUS_H_
 #define _MACH_I386_THREAD_STATUS_H_
 
-#include <mach/i386/_structs.h>
+#include <mach/machine/_structs.h>
 #include <mach/message.h>
 #include <mach/i386/fp_reg.h>
 #include <mach/i386/thread_state.h>
 #include <i386/eflags.h>
 
+
 /*
  * the i386_xxxx form is kept for legacy purposes since these types
  * are externally known... eventually they should be deprecated.

