diff -urN cctools.orig/as/app.h cctools/as/app.h
--- cctools.orig/as/app.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/app.h	2017-08-01 13:54:10.000000000 +0100
@@ -1,4 +1,4 @@
-#import <stdio.h>
+#include <stdio.h>
 
 extern FILE *scrub_file;
 extern char *scrub_string;
diff -urN cctools.orig/as/as.h cctools/as/as.h
--- cctools.orig/as/as.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/as.h	2017-08-01 13:54:10.000000000 +0100
@@ -58,9 +58,9 @@
 
 /* These #imports are for type definitions etc. */
 #include <stdint.h>
-#import <stdio.h>
-#import <assert.h>
-#import <mach/machine.h>
+#include <stdio.h>
+#include <assert.h>
+#include <mach/machine.h>
 
 /* These defines are potentially useful */
 #undef FALSE
diff -urN cctools.orig/as/atof-ieee.h cctools/as/atof-ieee.h
--- cctools.orig/as/atof-ieee.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/atof-ieee.h	2017-08-01 13:54:10.000000000 +0100
@@ -1,4 +1,4 @@
-#import "flonum.h"
+#include "flonum.h"
 
 extern char *atof_ieee(
     char *str,
diff -urN cctools.orig/as/expr.h cctools/as/expr.h
--- cctools.orig/as/expr.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/expr.h	2017-08-01 13:54:10.000000000 +0100
@@ -20,9 +20,9 @@
 the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.  */
 
 #include "as.h"
-#import "struc-symbol.h"
-#import "bignum.h"
-#import "flonum.h"
+#include "struc-symbol.h"
+#include "bignum.h"
+#include "flonum.h"
 
 enum {
   /* FROM expr.h line 46 */
diff -urN cctools.orig/as/flonum.h cctools/as/flonum.h
--- cctools.orig/as/flonum.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/flonum.h	2017-08-01 13:54:10.000000000 +0100
@@ -34,7 +34,7 @@
 *									*
 \***********************************************************************/
 
-#import "bignum.h"
+#include "bignum.h"
 
 /***********************************************************************\
 *									*
diff -urN cctools.orig/as/frags.h cctools/as/frags.h
--- cctools.orig/as/frags.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/frags.h	2017-08-01 13:54:10.000000000 +0100
@@ -19,9 +19,9 @@
 along with GAS; see the file COPYING.  If not, write to
 the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.  */
 
-#import "as.h"
-#import "relax.h"
-#import "struc-symbol.h"
+#include "as.h"
+#include "relax.h"
+#include "struc-symbol.h"
 
 /*
  * A code fragment (frag) is some known number of chars, followed by some
diff -urN cctools.orig/as/md.h cctools/as/md.h
--- cctools.orig/as/md.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/md.h	2017-08-01 13:54:10.000000000 +0100
@@ -19,13 +19,13 @@
 in a file named COPYING.  Among other things, the copyright
 notice and this notice must be preserved on all copies.  */
 
-#import <mach/machine.h>
-#import "stuff/bytesex.h"
-#import "frags.h"
-#import "relax.h"
-#import "struc-symbol.h"
-#import "fixes.h"
-#import "read.h"
+#include <mach/machine.h>
+#include "stuff/bytesex.h"
+#include "frags.h"
+#include "relax.h"
+#include "struc-symbol.h"
+#include "fixes.h"
+#include "read.h"
 
 /* These are the default cputype and cpusubtype for this target MACHINE */
 extern const cpu_type_t md_cputype;
diff -urN cctools.orig/as/read.h cctools/as/read.h
--- cctools.orig/as/read.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/read.h	2017-08-01 13:54:10.000000000 +0100
@@ -19,7 +19,7 @@
 along with GAS; see the file COPYING.  If not, write to
 the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.  */
 
-#import "expr.h"
+#include "expr.h"
 
 /* Define to make whitespace be allowed in many syntactically
    unnecessary places.  Normally undefined.  For compatibility with
diff -urN cctools.orig/as/struc-symbol.h cctools/as/struc-symbol.h
--- cctools.orig/as/struc-symbol.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/struc-symbol.h	2017-08-01 13:54:10.000000000 +0100
@@ -21,7 +21,7 @@
 
 #ifdef NeXT_MOD
 #include "arch64_32.h"
-#import <mach-o/nlist.h>
+#include <mach-o/nlist.h>
 #else /* !defined(NeXT_MOD) */
 #ifndef		VMS
 #include "a.out.h"		/* Needed to define struct nlist. Sigh. */
diff -urN cctools.orig/as/symbols.h cctools/as/symbols.h
--- cctools.orig/as/symbols.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/as/symbols.h	2017-08-01 13:54:10.000000000 +0100
@@ -17,8 +17,8 @@
 along with GAS; see the file COPYING.  If not, write to
 the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.  */
 
-#import "struc-symbol.h"
-#import "hash.h"
+#include "struc-symbol.h"
+#include "hash.h"
 
 extern struct hash_control *sy_hash;
 extern struct obstack notes;
diff -urN cctools.orig/gprof/gprof.h cctools/gprof/gprof.h
--- cctools.orig/gprof/gprof.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/gprof/gprof.h	2017-08-01 13:54:10.000000000 +0100
@@ -57,17 +57,17 @@
  *
  *	@(#)gprof.h	8.1 (Berkeley) 6/6/93
  */
-#import <stdio.h>
-#import <sys/types.h>
-#import <sys/stat.h>
-#import <gnu/a.out.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <gnu/a.out.h>
 #ifdef __OPENSTEP__
-#import <mach-o/rld_state.h>
-#import <mach-o/gmon.h>
+#include <mach-o/rld_state.h>
+#include <mach-o/gmon.h>
 #else
-#import <sys/gmon.h>
+#include <sys/gmon.h>
 #endif
-#import "stuff/bool.h"
+#include "stuff/bool.h"
 
 /*
  * Used for comparison routine return values.
diff -urN cctools.orig/include/architecture/m88k/fp_regs.h cctools/include/architecture/m88k/fp_regs.h
--- cctools.orig/include/architecture/m88k/fp_regs.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/architecture/m88k/fp_regs.h	2017-08-01 13:54:10.000000000 +0100
@@ -41,7 +41,7 @@
 #ifndef	_ARCH_M88K_FP_REGS_H_
 #define	_ARCH_M88K_FP_REGS_H_
 
-#import <architecture/m88k/reg_help.h>
+#include <architecture/m88k/reg_help.h>
 
 /*
  * m88k_xrf_t -- data types that MAY be in extended register file
diff -urN cctools.orig/include/architecture/m88k/reg_help.h cctools/include/architecture/m88k/reg_help.h
--- cctools.orig/include/architecture/m88k/reg_help.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/architecture/m88k/reg_help.h	2017-08-01 13:54:10.000000000 +0100
@@ -41,7 +41,7 @@
 #ifndef _ARCH_M88K_REG_HELP_H_
 #define _ARCH_M88K_REG_HELP_H_
 
-#import <architecture/nrw/reg_help.h>
+#include <architecture/nrw/reg_help.h>
 
 /* Stack pointer must always be a multiple of 16 */
 #define	STACK_INCR	16
diff -urN cctools.orig/include/mach/m88k/thread_status.h cctools/include/mach/m88k/thread_status.h
--- cctools.orig/include/mach/m88k/thread_status.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/mach/m88k/thread_status.h	2017-08-01 13:54:10.000000000 +0100
@@ -49,8 +49,8 @@
 #ifndef	_MACH_M88K_THREAD_STATE_
 #define	_MACH_M88K_THREAD_STATE_
 
-#import <architecture/m88k/fp_regs.h>
-#import <architecture/m88k/reg_help.h>
+#include <architecture/m88k/fp_regs.h>
+#include <architecture/m88k/reg_help.h>
 
 /**************************************************************************
  * Data Typedefs used by thread_getstatus() and thread_setstatus()        *
diff -urN cctools.orig/include/mach/machine.h cctools/include/mach/machine.h
--- cctools.orig/include/mach/machine.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/mach/machine.h	2017-08-01 13:54:10.000000000 +0100
@@ -80,8 +80,8 @@
 #ifndef	_MACH_MACHINE_H_
 #define _MACH_MACHINE_H_
 
-#import <mach/machine/vm_types.h>
-#import <mach/boolean.h>
+#include <mach/machine/vm_types.h>
+#include <mach/boolean.h>
 
 /*
  *	For each host, there is a maximum possible number of
diff -urN cctools.orig/include/mach-o/hppa/swap.h cctools/include/mach-o/hppa/swap.h
--- cctools.orig/include/mach-o/hppa/swap.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/mach-o/hppa/swap.h	2017-08-01 13:54:10.000000000 +0100
@@ -20,8 +20,8 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <architecture/byte_order.h>
-#import <mach/hppa/thread_status.h>
+#include <architecture/byte_order.h>
+#include <mach/hppa/thread_status.h>
 
 extern void swap_hppa_integer_thread_state(
     struct hp_pa_integer_thread_state *regs,
diff -urN cctools.orig/include/mach-o/i860/swap.h cctools/include/mach-o/i860/swap.h
--- cctools.orig/include/mach-o/i860/swap.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/mach-o/i860/swap.h	2017-08-01 13:54:10.000000000 +0100
@@ -20,8 +20,8 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <architecture/byte_order.h>
-#import <mach/i860/thread_status.h>
+#include <architecture/byte_order.h>
+#include <mach/i860/thread_status.h>
 
 extern void swap_i860_thread_state_regs(
     struct i860_thread_state_regs *cpu,
diff -urN cctools.orig/include/mach-o/m68k/swap.h cctools/include/mach-o/m68k/swap.h
--- cctools.orig/include/mach-o/m68k/swap.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/mach-o/m68k/swap.h	2017-08-01 13:54:10.000000000 +0100
@@ -20,8 +20,8 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <architecture/byte_order.h>
-#import <mach/m68k/thread_status.h>
+#include <architecture/byte_order.h>
+#include <mach/m68k/thread_status.h>
 
 extern void swap_m68k_thread_state_regs(
     struct m68k_thread_state_regs *cpu,
diff -urN cctools.orig/include/mach-o/m88k/swap.h cctools/include/mach-o/m88k/swap.h
--- cctools.orig/include/mach-o/m88k/swap.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/mach-o/m88k/swap.h	2017-08-01 13:54:10.000000000 +0100
@@ -20,8 +20,8 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <architecture/byte_order.h>
-#import <mach/m88k/thread_status.h>
+#include <architecture/byte_order.h>
+#include <mach/m88k/thread_status.h>
 
 extern void swap_m88k_thread_state_grf_t(
     m88k_thread_state_grf_t *cpu,
diff -urN cctools.orig/include/mach-o/sparc/swap.h cctools/include/mach-o/sparc/swap.h
--- cctools.orig/include/mach-o/sparc/swap.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/mach-o/sparc/swap.h	2017-08-01 13:54:10.000000000 +0100
@@ -20,8 +20,8 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <architecture/byte_order.h>
-#import <mach/sparc/thread_status.h>
+#include <architecture/byte_order.h>
+#include <mach/sparc/thread_status.h>
 
 void swap_sparc_thread_state_regs(
     struct sparc_thread_state_regs *cpu,
diff -urN cctools.orig/include/standalone/libsa.h cctools/include/standalone/libsa.h
--- cctools.orig/include/standalone/libsa.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/standalone/libsa.h	2017-08-01 13:54:10.000000000 +0100
@@ -22,13 +22,13 @@
  */
 /* Exported API for standalone library */
 #if !(defined(KLD) && defined(__STATIC__))
-#import <mach/mach.h>
+#include <mach/mach.h>
 #else /* defined(KLD) && defined(__STATIC__) */
 #include <mach/kern_return.h>
 #endif /* !(defined(KLD) && defined(__STATIC__)) */
-#import <mach-o/loader.h>
-#import <stdarg.h>
-#import <stddef.h>
+#include <mach-o/loader.h>
+#include <stdarg.h>
+#include <stddef.h>
 
 #ifndef bcopy
 #ifdef __OPENSTEP__
diff -urN cctools.orig/include/stuff/breakout.h cctools/include/stuff/breakout.h
--- cctools.orig/include/stuff/breakout.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/stuff/breakout.h	2017-08-01 13:54:10.000000000 +0100
@@ -24,7 +24,7 @@
 #define __private_extern__ __declspec(private_extern)
 #endif
 
-#import "stuff/ofile.h"
+#include "stuff/ofile.h"
 
 /*
  * This is used to build the table of contents of an archive.  Each toc_entry
diff -urN cctools.orig/include/stuff/errors.h cctools/include/stuff/errors.h
--- cctools.orig/include/stuff/errors.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/stuff/errors.h	2017-08-01 13:54:40.000000000 +0100
@@ -25,7 +25,7 @@
  * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#import "mach/mach.h"
+#include "mach/mach.h"
 
 /* user defined (imported) */
 extern char *progname __attribute__((visibility("hidden")));
diff -urN cctools.orig/include/stuff/ofile.h cctools/include/stuff/ofile.h
--- cctools.orig/include/stuff/ofile.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/stuff/ofile.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,17 +28,17 @@
 #define __private_extern__ __declspec(private_extern)
 #endif
 
-#import <ar.h>
+#include <ar.h>
 #ifndef AR_EFMT1
 #define	AR_EFMT1	"#1/"		/* extended format #1 */
 #endif
-#import <mach-o/loader.h>
+#include <mach-o/loader.h>
 #ifdef OFI
-#import <mach-o/dyld.h>
+#include <mach-o/dyld.h>
 #endif
-#import "stuff/bytesex.h"
-#import "stuff/bool.h"
-#import "stuff/arch.h"
+#include "stuff/bytesex.h"
+#include "stuff/bool.h"
+#include "stuff/arch.h"
 
 enum ofile_type {
     OFILE_UNKNOWN,
diff -urN cctools.orig/include/stuff/print.h cctools/include/stuff/print.h
--- cctools.orig/include/stuff/print.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/stuff/print.h	2017-08-01 13:54:10.000000000 +0100
@@ -24,7 +24,7 @@
 #define __private_extern__ __declspec(private_extern)
 #endif
 
-#import <stdarg.h>
+#include <stdarg.h>
 
 __private_extern__ void print(
     const char *format, ...)
diff -urN cctools.orig/include/stuff/reloc.h cctools/include/stuff/reloc.h
--- cctools.orig/include/stuff/reloc.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/stuff/reloc.h	2017-08-01 13:54:10.000000000 +0100
@@ -24,8 +24,8 @@
 #define __private_extern__ __declspec(private_extern)
 #endif
 
-#import <mach/machine.h>
-#import "stuff/bool.h"
+#include <mach/machine.h>
+#include "stuff/bool.h"
 
 __private_extern__ uint32_t reloc_pair_r_type(
     cpu_type_t cputype);
diff -urN cctools.orig/include/stuff/vm_flush_cache.h cctools/include/stuff/vm_flush_cache.h
--- cctools.orig/include/stuff/vm_flush_cache.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/include/stuff/vm_flush_cache.h	2017-08-01 13:54:10.000000000 +0100
@@ -24,7 +24,7 @@
 #define __private_extern__ __declspec(private_extern)
 #endif
 
-#import <mach/mach.h>
+#include <mach/mach.h>
 __private_extern__ kern_return_t vm_flush_cache(
     mach_port_t target_task,
     vm_address_t address,
diff -urN cctools.orig/ld/layout.c cctools/ld/layout.c
--- cctools.orig/ld/layout.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/ld/layout.c	2017-08-01 13:54:10.000000000 +0100
@@ -51,21 +51,21 @@
 #include "stuff/openstep_mach.h"
 #include <mach-o/fat.h>
 #include <mach-o/loader.h>
-#import <mach/m68k/thread_status.h>
+#include <mach/m68k/thread_status.h>
 #undef MACHINE_THREAD_STATE	/* need to undef these to avoid warnings */
 #undef MACHINE_THREAD_STATE_COUNT
 #undef THREAD_STATE_NONE
 #undef VALID_THREAD_STATE_FLAVOR
-#import <mach/ppc/thread_status.h>
+#include <mach/ppc/thread_status.h>
 #undef MACHINE_THREAD_STATE	/* need to undef these to avoid warnings */
 #undef MACHINE_THREAD_STATE_COUNT
 #undef THREAD_STATE_NONE
 #undef VALID_THREAD_STATE_FLAVOR
-#import <mach/m88k/thread_status.h>
-#import <mach/i860/thread_status.h>
-#import <mach/i386/thread_status.h>
-#import <mach/hppa/thread_status.h>
-#import <mach/sparc/thread_status.h>
+#include <mach/m88k/thread_status.h>
+#include <mach/i860/thread_status.h>
+#include <mach/i386/thread_status.h>
+#include <mach/hppa/thread_status.h>
+#include <mach/sparc/thread_status.h>
 #include <mach-o/nlist.h>
 #include <mach-o/reloc.h>
 #if defined(RLD) && !defined(SA_RLD) && !(defined(KLD) && defined(__STATIC__))
diff -urN cctools.orig/libmacho/hppa_swap.c cctools/libmacho/hppa_swap.c
--- cctools.orig/libmacho/hppa_swap.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libmacho/hppa_swap.c	2017-08-01 13:54:10.000000000 +0100
@@ -21,7 +21,7 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 #ifndef RLD
-#import <mach-o/hppa/swap.h>
+#include <mach-o/hppa/swap.h>
 
 void
 swap_hppa_integer_thread_state(
diff -urN cctools.orig/libmacho/i386_swap.c cctools/libmacho/i386_swap.c
--- cctools.orig/libmacho/i386_swap.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libmacho/i386_swap.c	2017-08-01 13:54:10.000000000 +0100
@@ -133,7 +133,7 @@
 
 #include <string.h>
 #include <libkern/OSByteOrder.h>
-#import <mach-o/i386/swap.h>
+#include <mach-o/i386/swap.h>
 
 void
 swap_i386_thread_state(
diff -urN cctools.orig/libmacho/i860_swap.c cctools/libmacho/i860_swap.c
--- cctools.orig/libmacho/i860_swap.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libmacho/i860_swap.c	2017-08-01 13:54:10.000000000 +0100
@@ -21,7 +21,7 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 #ifndef RLD
-#import <mach-o/i860/swap.h>
+#include <mach-o/i860/swap.h>
 
 void
 swap_i860_thread_state_regs(
diff -urN cctools.orig/libmacho/m68k_swap.c cctools/libmacho/m68k_swap.c
--- cctools.orig/libmacho/m68k_swap.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libmacho/m68k_swap.c	2017-08-01 13:54:10.000000000 +0100
@@ -21,7 +21,7 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 #ifndef RLD
-#import <mach-o/m68k/swap.h>
+#include <mach-o/m68k/swap.h>
 
 void
 swap_m68k_thread_state_regs(
diff -urN cctools.orig/libmacho/m88k_swap.c cctools/libmacho/m88k_swap.c
--- cctools.orig/libmacho/m88k_swap.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libmacho/m88k_swap.c	2017-08-01 13:54:10.000000000 +0100
@@ -22,8 +22,8 @@
  */
 #ifndef RLD
 #include <string.h>
-#import <mach-o/m88k/swap.h>
-#import <architecture/nrw/reg_help.h>
+#include <mach-o/m88k/swap.h>
+#include <architecture/nrw/reg_help.h>
 
 void
 swap_m88k_thread_state_grf_t(
diff -urN cctools.orig/libmacho/ppc_swap.c cctools/libmacho/ppc_swap.c
--- cctools.orig/libmacho/ppc_swap.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libmacho/ppc_swap.c	2017-08-01 13:54:10.000000000 +0100
@@ -71,8 +71,8 @@
 #define __vrsave vrsave
 #define __xer xer
 
-#import <mach-o/ppc/swap.h>
-#import <architecture/nrw/reg_help.h>
+#include <mach-o/ppc/swap.h>
+#include <architecture/nrw/reg_help.h>
 
 void
 swap_ppc_thread_state_t(
diff -urN cctools.orig/libmacho/sparc_swap.c cctools/libmacho/sparc_swap.c
--- cctools.orig/libmacho/sparc_swap.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libmacho/sparc_swap.c	2017-08-01 13:54:10.000000000 +0100
@@ -22,8 +22,8 @@
  */
 #ifndef RLD
 #include <string.h>
-#import <mach-o/sparc/swap.h>
-#import <architecture/nrw/reg_help.h>
+#include <mach-o/sparc/swap.h>
+#include <architecture/nrw/reg_help.h>
 
 void
 swap_sparc_thread_state_regs(
diff -urN cctools.orig/libstuff/SymLoc.c cctools/libstuff/SymLoc.c
--- cctools.orig/libstuff/SymLoc.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libstuff/SymLoc.c	2017-08-01 13:54:10.000000000 +0100
@@ -21,18 +21,18 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 #ifndef RLD
-#import <libc.h>
-#import <ctype.h>
-#import <sys/types.h>
+#include <libc.h>
+#include <ctype.h>
+#include <sys/types.h>
 #ifdef __OPENSTEP__
 #define _POSIX_SOURCE
 #endif
-#import <dirent.h>
-#import <pwd.h>
-#import "stuff/bool.h"
-#import "stuff/errors.h"
-#import "stuff/allocate.h"
-#import "stuff/SymLoc.h"
+#include <dirent.h>
+#include <pwd.h>
+#include "stuff/bool.h"
+#include "stuff/errors.h"
+#include "stuff/allocate.h"
+#include "stuff/SymLoc.h"
 
 const char *
 symLocForDylib(const char *installName, const char *releaseName,
diff -urN cctools.orig/libstuff/ofile.c cctools/libstuff/ofile.c
--- cctools.orig/libstuff/ofile.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/libstuff/ofile.c	2017-08-01 13:54:10.000000000 +0100
@@ -50,17 +50,17 @@
 #include <sys/mman.h>
 #include <mach-o/fat.h>
 #include <mach-o/loader.h>
-#import <mach/m68k/thread_status.h>
-#import <mach/ppc/thread_status.h>
+#include <mach/m68k/thread_status.h>
+#include <mach/ppc/thread_status.h>
 #undef MACHINE_THREAD_STATE     /* need to undef these to avoid warnings */
 #undef MACHINE_THREAD_STATE_COUNT
 #undef THREAD_STATE_NONE
 #undef VALID_THREAD_STATE_FLAVOR
-#import <mach/m88k/thread_status.h>
-#import <mach/i860/thread_status.h>
-#import <mach/i386/thread_status.h>
-#import <mach/sparc/thread_status.h>
-#import <mach/arm/thread_status.h>
+#include <mach/m88k/thread_status.h>
+#include <mach/i860/thread_status.h>
+#include <mach/i386/thread_status.h>
+#include <mach/sparc/thread_status.h>
+#include <mach/arm/thread_status.h>
 #include <mach-o/nlist.h>
 #include <mach-o/reloc.h>
 #include "stuff/bool.h"
diff -urN cctools.orig/misc/cmpdylib.c cctools/misc/cmpdylib.c
--- cctools.orig/misc/cmpdylib.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/misc/cmpdylib.c	2017-08-01 13:54:10.000000000 +0100
@@ -20,13 +20,13 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stdio.h>
-#import <stdlib.h>
+#include <stdio.h>
+#include <stdlib.h>
 
-#import "stuff/ofile.h"
-#import "stuff/errors.h"
-#import "stuff/bytesex.h"
-#import "stuff/allocate.h"
+#include "stuff/ofile.h"
+#include "stuff/errors.h"
+#include "stuff/bytesex.h"
+#include "stuff/allocate.h"
 
 /* name of the program for error messages (argv[0]) */
 __private_extern__ char *progname = NULL;
diff -urN cctools.orig/misc/pagestuff.c cctools/misc/pagestuff.c
--- cctools.orig/misc/pagestuff.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/misc/pagestuff.c	2017-08-01 13:54:10.000000000 +0100
@@ -21,12 +21,12 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 #include <stdint.h>
-#import <stdio.h>
-#import <stdlib.h>
-#import <string.h>
-#import "stuff/ofile.h"
-#import "stuff/errors.h"
-#import "stuff/allocate.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "stuff/ofile.h"
+#include "stuff/errors.h"
+#include "stuff/allocate.h"
 
 enum file_part_type {
     FP_FAT_HEADERS,
diff -urN cctools.orig/misc/redo_prebinding.c cctools/misc/redo_prebinding.c
--- cctools.orig/misc/redo_prebinding.c	2017-08-01 13:52:43.000000000 +0100
+++ cctools/misc/redo_prebinding.c	2017-08-01 13:54:10.000000000 +0100
@@ -83,31 +83,31 @@
 #include <mach-o/redo_prebinding.h>
 #endif /* defined(LIBRARY_API) */
 
-#import <stdio.h>
-#import <stdlib.h>
-#import <string.h>
-#import <limits.h>
-#import <libc.h>
-#import <malloc/malloc.h>
-#import <sys/types.h>
-#import <sys/stat.h>
-#import <mach-o/stab.h>
-#import <mach-o/loader.h>
-#import <mach-o/reloc.h>
-#import <mach-o/hppa/reloc.h>
-#import <mach-o/sparc/reloc.h>
-#import <mach-o/ppc/reloc.h>
-#import <mach-o/arm/reloc.h>
-#import <stuff/breakout.h>
-#import <stuff/best_arch.h>
-#import <stuff/allocate.h>
-#import <stuff/errors.h>
-#import <stuff/rnd.h>
-#import <stuff/hppa.h>
-#import <stuff/execute.h>
-#import <stuff/guess_short_name.h>
-#import <stuff/seg_addr_table.h>
-#import <stuff/macosx_deployment_target.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <libc.h>
+#include <malloc/malloc.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <mach-o/stab.h>
+#include <mach-o/loader.h>
+#include <mach-o/reloc.h>
+#include <mach-o/hppa/reloc.h>
+#include <mach-o/sparc/reloc.h>
+#include <mach-o/ppc/reloc.h>
+#include <mach-o/arm/reloc.h>
+#include <stuff/breakout.h>
+#include <stuff/best_arch.h>
+#include <stuff/allocate.h>
+#include <stuff/errors.h>
+#include <stuff/rnd.h>
+#include <stuff/hppa.h>
+#include <stuff/execute.h>
+#include <stuff/guess_short_name.h>
+#include <stuff/seg_addr_table.h>
+#include <stuff/macosx_deployment_target.h>
 
 #include <mach-o/dyld.h>
 
diff -urN cctools.orig/otool/arm_disasm.h cctools/otool/arm_disasm.h
--- cctools.orig/otool/arm_disasm.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/arm_disasm.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,10 +28,10 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stuff/bytesex.h>
-#import <mach-o/reloc.h>
-#import <mach-o/nlist.h>
-#import <stuff/bool.h>
+#include <stuff/bytesex.h>
+#include <mach-o/reloc.h>
+#include <mach-o/nlist.h>
+#include <stuff/bool.h>
 #include "stuff/symbol.h"
 #include "llvm-c/Disassembler.h"
 
diff -urN cctools.orig/otool/hppa_disasm.h cctools/otool/hppa_disasm.h
--- cctools.orig/otool/hppa_disasm.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/hppa_disasm.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,10 +28,10 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stuff/bytesex.h>
-#import <mach-o/reloc.h>
-#import <mach-o/nlist.h>
-#import <stuff/bool.h>
+#include <stuff/bytesex.h>
+#include <mach-o/reloc.h>
+#include <mach-o/nlist.h>
+#include <stuff/bool.h>
 #include "stuff/symbol.h"
 
 extern uint32_t hppa_disassemble(
diff -urN cctools.orig/otool/i386_disasm.h cctools/otool/i386_disasm.h
--- cctools.orig/otool/i386_disasm.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/i386_disasm.h	2017-08-01 13:55:16.000000000 +0100
@@ -28,10 +28,10 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stuff/bytesex.h>
-#import <mach-o/reloc.h>
-#import <mach-o/nlist.h>
-#import <stuff/bool.h>
+#include <stuff/bytesex.h>
+#include <mach-o/reloc.h>
+#include <mach-o/nlist.h>
+#include <stuff/bool.h>
 #include "otool.h"
 #include "stuff/symbol.h"
 #include "llvm-c/Disassembler.h"
diff -urN cctools.orig/otool/i860_disasm.h cctools/otool/i860_disasm.h
--- cctools.orig/otool/i860_disasm.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/i860_disasm.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,10 +28,10 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stuff/bytesex.h>
-#import <mach-o/reloc.h>
-#import <mach-o/nlist.h>
-#import <stuff/bool.h>
+#include <stuff/bytesex.h>
+#include <mach-o/reloc.h>
+#include <mach-o/nlist.h>
+#include <stuff/bool.h>
 #include "stuff/symbol.h"
 
 extern uint32_t i860_disassemble(
diff -urN cctools.orig/otool/m68k_disasm.h cctools/otool/m68k_disasm.h
--- cctools.orig/otool/m68k_disasm.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/m68k_disasm.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,10 +28,10 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stuff/bytesex.h>
-#import <mach-o/reloc.h>
-#import <mach-o/nlist.h>
-#import <stuff/bool.h>
+#include <stuff/bytesex.h>
+#include <mach-o/reloc.h>
+#include <mach-o/nlist.h>
+#include <stuff/bool.h>
 #include "stuff/symbol.h"
 
 extern uint32_t m68k_disassemble(
diff -urN cctools.orig/otool/m88k_disasm.h cctools/otool/m88k_disasm.h
--- cctools.orig/otool/m88k_disasm.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/m88k_disasm.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,10 +28,10 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stuff/bytesex.h>
-#import <mach-o/reloc.h>
-#import <mach-o/nlist.h>
-#import <stuff/bool.h>
+#include <stuff/bytesex.h>
+#include <mach-o/reloc.h>
+#include <mach-o/nlist.h>
+#include <stuff/bool.h>
 #include "stuff/symbol.h"
 
 extern uint32_t m88k_disassemble(
diff -urN cctools.orig/otool/ofile_print.h cctools/otool/ofile_print.h
--- cctools.orig/otool/ofile_print.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/ofile_print.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,13 +28,13 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <ar.h>
-#import <mach-o/fat.h>
-#import <mach-o/loader.h>
-#import <mach-o/nlist.h>
-#import <mach-o/reloc.h>
-#import <stuff/bytesex.h>
-#import <stuff/bool.h>
+#include <ar.h>
+#include <mach-o/fat.h>
+#include <mach-o/loader.h>
+#include <mach-o/nlist.h>
+#include <mach-o/reloc.h>
+#include <stuff/bytesex.h>
+#include <stuff/bool.h>
 #include "stuff/symbol.h"
 
 extern void print_fat_headers(
diff -urN cctools.orig/otool/ppc_disasm.h cctools/otool/ppc_disasm.h
--- cctools.orig/otool/ppc_disasm.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/ppc_disasm.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,10 +28,10 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stuff/bytesex.h>
-#import <mach-o/reloc.h>
-#import <mach-o/nlist.h>
-#import <stuff/bool.h>
+#include <stuff/bytesex.h>
+#include <mach-o/reloc.h>
+#include <mach-o/nlist.h>
+#include <stuff/bool.h>
 #include "stuff/symbol.h"
 
 extern uint32_t ppc_disassemble(
diff -urN cctools.orig/otool/sparc_disasm.h cctools/otool/sparc_disasm.h
--- cctools.orig/otool/sparc_disasm.h	2017-08-01 13:52:43.000000000 +0100
+++ cctools/otool/sparc_disasm.h	2017-08-01 13:54:10.000000000 +0100
@@ -28,10 +28,10 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
-#import <stuff/bytesex.h>
-#import <mach-o/reloc.h>
-#import <mach-o/nlist.h>
-#import <stuff/bool.h>
+#include <stuff/bytesex.h>
+#include <mach-o/reloc.h>
+#include <mach-o/nlist.h>
+#include <stuff/bool.h>
 #include "stuff/symbol.h"
 
 extern uint32_t sparc_disassemble(
