diff -urN a/libstuff/macosx_deployment_target.c b/libstuff/macosx_deployment_target.c
--- a/libstuff/macosx_deployment_target.c	2013-10-07 17:09:08.475770020 +0100
+++ b/libstuff/macosx_deployment_target.c	2013-10-07 17:09:08.632439115 +0100
@@ -115,8 +115,8 @@
 	osversion_name[0] = CTL_KERN;
 	osversion_name[1] = KERN_OSRELEASE;
 	osversion_len = sizeof(osversion) - 1;
-	if(sysctl(osversion_name, 2, osversion, &osversion_len, NULL, 0) == -1)
-	    system_error("sysctl for kern.osversion failed");
+	strcpy(osversion,"12.0");
+	    
 
 	/*
 	 * Now parse this out.  It is expected to be of the form "x.y.z" where
@@ -146,10 +146,10 @@
 	 * As a last resort we set the default to the highest known shipping
 	 * system to date.
 	 */
-	value->major = 6;
+	value->major = 5;
 	value->minor = 0;
-	value->name = allocate(strlen("10.6") + 1);
-	strcpy(value->name, "10.6");
+	value->name = allocate(strlen("10.5") + 1);
+	strcpy(value->name, "10.5");
 	warning("unknown value returned by sysctl() for kern.osrelease: %s "
 		"ignored (using %s)", osversion, value->name);
 	/* fall through to also warn about a possble bad user value */
