diff -urN a/ar/archive.h b/ar/archive.h
--- a/ar/archive.h	2013-10-07 17:09:09.955792881 +0100
+++ b/ar/archive.h	2013-10-07 17:09:10.119128752 +0100
@@ -112,11 +112,11 @@
 } CHDR;
 
 /* Header format strings. */
-#define	HDR1	"%s%-13d%-12ld%-6u%-6u%-8o%-10qd%2s"
-#define	HDR2	"%-16.16s%-12ld%-6u%-6u%-8o%-10qd%2s"
+#define	HDR1	"%s%-13d%-12ld%-6u%-6u%-8o%-10lld%2s"
+#define	HDR2	"%-16.16s%-12ld%-6u%-6u%-8o%-10lld%2s"
 
 #define	OLDARMAXNAME	15
-#define	HDR3	"%-16.15s%-12ld%-6u%-6u%-8o%-10qd%2s"
+#define	HDR3	"%-16.15s%-12ld%-6u%-6u%-8o%-10lld%2s"
 
 
 #include <sys/cdefs.h>
diff -urN a/ar/contents.c b/ar/contents.c
--- a/ar/contents.c	2013-10-07 17:09:09.955792881 +0100
+++ b/ar/contents.c	2013-10-07 17:09:10.119128752 +0100
@@ -104,8 +104,8 @@
 			goto next;
 		if (options & AR_V) {
 			(void)strmode(chdr.mode, buf);
-			(void)printf("%s %6d/%-6d %8qd ",
-			    buf + 1, chdr.uid, chdr.gid, chdr.size);
+			(void)printf("%s %6d/%-6d %8lld ",
+			    buf + 1, chdr.uid, chdr.gid, (int64_t)chdr.size);
 			tp = localtime(&chdr.date);
 			(void)strftime(buf, sizeof(buf), "%b %e %H:%M %Y", tp);
 			(void)printf("%s %s\n", buf, file);
diff -urN a/as/messages.c b/as/messages.c
--- a/as/messages.c	2013-10-07 17:09:09.999126896 +0100
+++ b/as/messages.c	2013-10-07 17:09:10.169129524 +0100
@@ -236,5 +236,5 @@
 void
 sprint_value (char *buf, signed_expr_t val)
 {
-    sprintf (buf, "%qd", val);
+    sprintf (buf, "%lld", val);
 }
