diff -urN a/ld/ld.c b/ld/ld.c
--- a/ld/ld.c	2013-10-07 17:09:10.449133848 +0100
+++ b/ld/ld.c	2013-10-07 17:09:10.632470050 +0100
@@ -1908,6 +1908,11 @@
 	else{
 	    next_root = p;
 	}
+       if(next_root == NULL) {
+#ifdef CROSS_SYSROOT
+         next_root = CROSS_SYSROOT;
+#endif
+       }
 	if(next_root != NULL){
 	    for(i = 0; standard_dirs[i] != NULL; i++){
 		p = allocate(strlen(next_root) +
--- a/misc/libtool.c	2013-10-07 17:09:10.429133546 +0100
+++ b/misc/libtool.c	2013-10-07 17:09:10.612469708 +0100
@@ -1110,6 +1110,11 @@
 	else{
 	    next_root = getenv("NEXT_ROOT");
 	}
+	if(next_root == NULL) {
+#ifdef CROSS_SYSROOT
+          next_root = CROSS_SYSROOT;
+#endif
+	}
 	if(next_root != NULL){
 	    for(i = 0; standard_dirs[i] != NULL; i++){
 		p = allocate(strlen(next_root) +
