diff -urN a/libstuff/get_arch_from_host.c b/libstuff/get_arch_from_host.c
--- a/libstuff/get_arch_from_host.c	2013-10-07 17:09:10.939141407 +0100
+++ b/libstuff/get_arch_from_host.c	2013-10-07 17:09:11.102477312 +0100
@@ -62,6 +62,8 @@
 	    return(0);
 	}
 	mach_port_deallocate(mach_task_self(), my_mach_host_self);
+	host_basic_info.cpu_type = EMULATED_HOST_CPU_TYPE;
+	host_basic_info.cpu_subtype = EMULATED_HOST_CPU_SUBTYPE;
 
 	if(family_arch_flag != NULL){
 	    family_arch_flag->cputype = host_basic_info.cpu_type;
