diff -urN cctools.orig/otool/print_objc.c cctools/otool/print_objc.c
--- cctools.orig/otool/print_objc.c	2017-08-01 13:57:33.000000000 +0100
+++ cctools/otool/print_objc.c	2017-08-01 14:00:56.000000000 +0100
@@ -145,7 +145,13 @@
     uint32_t characters;	/* char * (32-bit pointer) */
     uint32_t _length;
 };
+/*
+ * On Ubuntu GCC 4.5.32, this fails. It seems NXConstantString is baked into the
+ * compiler (i.e. changing it to NXConstantString2 allows it to compile)
+ */
+/*
 typedef struct objc_string_object_t NXConstantString;
+*/
 
 #define SIZEHASHTABLE 821
 struct _hashEntry_t {
