diff -urN a/as/input-scrub.c b/as/input-scrub.c
--- a/as/input-scrub.c	2013-10-07 17:09:13.529181513 +0100
+++ b/as/input-scrub.c	2013-10-07 17:09:13.682517196 +0100
@@ -384,14 +384,7 @@
   fprintf (stderr,"as:file(%s) %s! ",
 	   filename, gripe
 	   );
-  if (errno > sys_nerr)
-    {
-      fprintf (stderr, "Unknown error #%d.", errno);
-    }
-  else
-    {
-      fprintf (stderr, "%s.", sys_errlist [errno]);
-    }
+      fprintf (stderr, "%s.", strerror(errno));
   (void)putc('\n', stderr);
   errno = 0;			/* After reporting, clear it. */
   if (input_file_is_open())	/* RMS says don't mention line # if not needed. */
