diff -urN cctools.orig/misc/redo_prebinding.c cctools/misc/redo_prebinding.c
--- cctools.orig/misc/redo_prebinding.c	2017-08-01 13:57:33.000000000 +0100
+++ cctools/misc/redo_prebinding.c	2017-08-01 14:24:10.000000000 +0100
@@ -9897,7 +9897,7 @@
 	errors++;
 }
 #endif /* defined(LIBRARY_API) */
-
+#if HAVE_GETATTRLIST
 #include <sys/attr.h>
 /*
  * Structure defining what's returned from getattrlist.  It returns all the
@@ -9982,3 +9982,13 @@
 
 	return(FALSE);
 }
+#else
+static
+enum bool
+has_resource_fork(
+                char *filename)
+{
+  return FALSE;
+}
+#endif
+
