diff -urN a/include/mach/ppc/_structs.h b/include/mach/ppc/_structs.h
--- a/include/mach/ppc/_structs.h	2013-10-07 17:09:14.432528785 +0100
+++ b/include/mach/ppc/_structs.h	2013-10-07 17:09:14.575864368 +0100
@@ -136,7 +136,7 @@
 };
 #endif /* __DARWIN_UNIX03 */
 
-#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+//#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
 
 #pragma pack(4)			/* Make sure the structure stays as we defined it */
 
@@ -236,7 +236,7 @@
 
 #pragma pack()
 
-#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) */
+//#endif /* _POSIX_C_SOURCE #endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE)#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) !_DARWIN_C_SOURCE */
 
 /* This structure should be double-word aligned for performance */
 
@@ -353,7 +353,7 @@
 };
 #endif /* __DARWIN_UNIX03 */
 
-#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+//#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
 #if __DARWIN_UNIX03
 #define _STRUCT_PPC_EXCEPTION_STATE64	struct __darwin_ppc_exception_state64
 _STRUCT_PPC_EXCEPTION_STATE64
@@ -385,7 +385,7 @@
 #endif
 };
 #endif /* __DARWIN_UNIX03 */
-#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) */
+//#endif /* _POSIX_C_SOURCE #endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE)#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) !_DARWIN_C_SOURCE */
 
 #pragma pack()
 
diff -urN a/include/mach/ppc/thread_status.h b/include/mach/ppc/thread_status.h
--- a/include/mach/ppc/thread_status.h	2013-10-07 17:09:14.432528785 +0100
+++ b/include/mach/ppc/thread_status.h	2013-10-07 17:09:14.575864368 +0100
@@ -70,9 +70,9 @@
 */
 
 typedef _STRUCT_PPC_THREAD_STATE	ppc_thread_state_t;
-#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+//#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
 typedef _STRUCT_PPC_THREAD_STATE64	ppc_thread_state64_t;
-#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) */
+//#endif /* _POSIX_C_SOURCE #endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE)#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) !_DARWIN_C_SOURCE */
 typedef _STRUCT_PPC_FLOAT_STATE		ppc_float_state_t;
 typedef _STRUCT_PPC_VECTOR_STATE	ppc_vector_state_t;
 
@@ -113,9 +113,9 @@
 /* of covering the full potential state (hi half available). */
 
 typedef _STRUCT_PPC_EXCEPTION_STATE	ppc_exception_state_t;
-#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+//#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
 typedef _STRUCT_PPC_EXCEPTION_STATE64	ppc_exception_state64_t;
-#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) */
+//#endif /* _POSIX_C_SOURCE #endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE)#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) !_DARWIN_C_SOURCE */
 
 /*
  * Save State Flags
