diff -urN cctools.orig/ld/pass1.c cctools/ld/pass1.c
--- cctools.orig/ld/pass1.c	2017-08-01 14:05:15.000000000 +0100
+++ cctools/ld/pass1.c	2017-08-01 14:25:09.000000000 +0100
@@ -508,7 +508,7 @@
 	    pass1_object(file_name, file_addr, file_size, base_name, FALSE,
 			 FALSE, bundle_loader, force_weak);
 	}
-#ifdef VM_SYNC_DEACTIVATE
+#if defined(VM_SYNC_DEACTIVATE) && (HAVE_DECL_VM_MSYNC)
 	vm_msync(mach_task_self(), (vm_address_t)file_addr,
 		 (vm_size_t)file_size, VM_SYNC_DEACTIVATE);
 #endif /* VM_SYNC_DEACTIVATE */
diff -urN cctools.orig/ld/pass2.c cctools/ld/pass2.c
--- cctools.orig/ld/pass2.c	2017-08-01 13:57:33.000000000 +0100
+++ cctools/ld/pass2.c	2017-08-01 14:25:09.000000000 +0100
@@ -269,7 +269,7 @@
 		    }
 		}
 		output_local_symbols();
-#ifdef VM_SYNC_DEACTIVATE
+#if defined(VM_SYNC_DEACTIVATE) && (HAVE_DECL_VM_MSYNC)
 		vm_msync(mach_task_self(), (vm_address_t)cur_obj->obj_addr,
 			 (vm_size_t)cur_obj->obj_size, VM_SYNC_DEACTIVATE);
 #endif /* VM_SYNC_DEACTIVATE */
diff -urN cctools.orig/misc/libtool.c cctools/misc/libtool.c
--- cctools.orig/misc/libtool.c	2017-08-01 14:23:25.000000000 +0100
+++ cctools/misc/libtool.c	2017-08-01 14:25:09.000000000 +0100
@@ -2795,7 +2795,7 @@
 		}
 		memcpy(p, arch->members[j].object_addr,
 		       arch->members[j].object_size);
-#ifdef VM_SYNC_DEACTIVATE
+#if defined(VM_SYNC_DEACTIVATE) && (HAVE_DECL_VM_MSYNC)
 		vm_msync(mach_task_self(),
 			 (vm_address_t)arch->members[j].object_addr,
 			 (vm_size_t)arch->members[j].object_size,
