diff -urN a/ar/archive.c b/ar/archive.c
--- a/ar/archive.c	2013-10-07 17:09:15.399210407 +0100
+++ b/ar/archive.c	2013-10-07 17:09:15.549212722 +0100
@@ -68,6 +68,7 @@
 
 #include <sys/param.h>
 #include <sys/stat.h>
+#include <unistd.h>
 
 #include <ar.h>
 #include <dirent.h>
diff -urN a/ar/contents.c b/ar/contents.c
--- a/ar/contents.c	2013-10-07 17:09:15.399210407 +0100
+++ b/ar/contents.c	2013-10-07 17:09:15.549212722 +0100
@@ -78,6 +78,8 @@
 #include <string.h>
 #include <tzfile.h>
 #include <unistd.h>
+#include <stdint.h>
+
 
 #include "archive.h"
 #include "extern.h"
diff -urN a/as/obstack.c b/as/obstack.c
--- a/as/obstack.c	2013-10-07 17:09:15.455877937 +0100
+++ b/as/obstack.c	2013-10-07 17:09:15.602546888 +0100
@@ -16,6 +16,8 @@
 Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.  */
 
 #include <stdlib.h>
+#include <stdint.h>
+
 #include "obstack.h"
 
 #ifdef __STDC__
diff -urN a/as/sections.c b/as/sections.c
--- a/as/sections.c	2013-10-07 17:09:15.455877937 +0100
+++ b/as/sections.c	2013-10-07 17:09:15.602546888 +0100
@@ -25,6 +25,8 @@
  */
 #include <stdlib.h>
 #include <strings.h>
+#include <string.h>
+
 #include "sections.h"
 #include "obstack.h"
 #include "xmalloc.h"
