diff -urN a/libstuff/macosx_deployment_target.c b/libstuff/macosx_deployment_target.c
--- a/libstuff/macosx_deployment_target.c	2013-10-07 17:09:16.299224328 +0100
+++ b/libstuff/macosx_deployment_target.c	2013-10-07 17:09:16.419226158 +0100
@@ -26,6 +26,9 @@
 #include <string.h>
 #include <mach/mach.h>
 #include <sys/types.h>
+#if defined(__unused) && defined(__linux__)
+#undef __unused
+#endif
 #include <sys/sysctl.h>
 #include "stuff/errors.h"
 #include "stuff/allocate.h"
