diff -urN ld64.orig/src/ld/ld.cpp ld64/src/ld/ld.cpp
--- ld64.orig/src/ld/ld.cpp	2017-08-07 16:38:43.144118949 +0100
+++ ld64/src/ld/ld.cpp	2017-08-07 16:38:50.376340747 +0100
@@ -1406,6 +1406,7 @@
 
 #ifndef NDEBUG
 // implement assert() function to print out a backtrace before aborting
+extern "C" void __assert_rtn(const char* func, const char* file, int line, const char* failedexpr);
 void __assert_rtn(const char* func, const char* file, int line, const char* failedexpr)
 {
     Snapshot *snapshot = Snapshot::globalSnapshot;
