diff -urN a/ar/misc.c b/ar/misc.c
--- a/ar/misc.c	2013-10-07 17:09:19.469273356 +0100
+++ b/ar/misc.c	2013-10-07 17:09:19.615942300 +0100
@@ -83,6 +83,14 @@
 
 char *tname = "temporary file";		/* temporary file "name" */
 
+#ifdef __MINGW32__
+ #define TMPDIR "TEMP"
+ #define SEP "\\"
+#else
+ #define TMPDIR "TMPDIR"
+ #define SEP "/"
+#endif
+
 int
 tmp()
 {
@@ -93,12 +101,12 @@
 	char path[MAXPATHLEN];
 
 	if (!first && !envtmp) {
-		envtmp = getenv("TMPDIR");
+		envtmp = getenv(TMPDIR);
 		first = 1;
 	}
 
 	if (envtmp)
-		(void)sprintf(path, "%s/%s", envtmp, _NAME_ARTMP);
+		(void)sprintf(path, "%s%s%s", envtmp, SEP, _NAME_ARTMP);
 	else
 		strcpy(path, _PATH_ARTMP);
 	
