diff -urN cctools.orig/libstuff/execute.c cctools/libstuff/execute.c
--- cctools.orig/libstuff/execute.c	2017-08-01 14:23:25.000000000 +0100
+++ cctools/libstuff/execute.c	2017-08-01 14:42:03.000000000 +0100
@@ -25,7 +25,11 @@
 #include <mach/mach.h> /* first to get rid of pre-comp warning */
 #include "stdio.h"
 #include <signal.h>
+#if defined(__MINGW32__)
+#include <process.h>
+#else
 #include <sys/wait.h>
+#endif
 #include <sys/file.h>
 #include <errno.h>
 #include "stuff/errors.h"
@@ -61,7 +65,10 @@
 		    fprintf(stderr, "%s ", *p++);
 	    fprintf(stderr, "\n");
 	}
-
+#if defined __MINGW32__
+    termsig = _spawnvp(_P_WAIT, name, (const char * const*)argv);
+    return(termsig==0);
+#else
 	forkpid = fork();
 	if(forkpid == -1)
 	    system_fatal("can't fork a new process to execute: %s", name);
@@ -93,6 +100,7 @@
 #endif
 		termsig == 0);
 	}
+#endif
 }
 
 /*
