diff -urN ld64.orig/src/ld/parsers/libunwind/AddressSpace.hpp ld64/src/ld/parsers/libunwind/AddressSpace.hpp
--- ld64.orig/src/ld/parsers/libunwind/AddressSpace.hpp	2013-03-07 23:10:19.000000000 +0000
+++ ld64/src/ld/parsers/libunwind/AddressSpace.hpp	2017-08-07 16:48:58.681022116 +0100
@@ -86,7 +86,7 @@
 {
 public:
 	
-	#if __LP64__
+	#if defined(__LP64__) || defined(_WIN64) || defined(__x86_64__)
 		typedef uint64_t	pint_t;
 		typedef  int64_t	sint_t;
 	#else
@@ -113,7 +113,7 @@
 
 inline uintptr_t LocalAddressSpace::getP(pint_t addr)
 {
-#if __LP64__
+#if defined(__LP64__) || defined(_WIN64) || defined(__x86_64__)
 	return get64(addr);
 #else
 	return get32(addr);
diff -urN ld64.orig/src/ld/parsers/macho_relocatable_file.cpp ld64/src/ld/parsers/macho_relocatable_file.cpp
--- ld64.orig/src/ld/parsers/macho_relocatable_file.cpp	2017-08-07 16:37:49.260466487 +0100
+++ ld64/src/ld/parsers/macho_relocatable_file.cpp	2017-08-07 16:48:58.683022177 +0100
@@ -5502,7 +5502,7 @@
 template <typename A>
 unsigned long Literal8Section<A>::contentHash(const class Atom<A>* atom, const ld::IndirectBindingTable& ind) const
 {
-#if __LP64__
+#if defined(__LP64__) || defined(_WIN64)
 	const uint64_t* literalContent = (uint64_t*)atom->contentPointer();
 	return *literalContent;
 #else
