diff -urN cctools.orig/as/driver.c cctools/as/driver.c
--- cctools.orig/as/driver.c	2017-08-01 14:23:25.000000000 +0100
+++ cctools/as/driver.c	2017-08-01 14:44:09.000000000 +0100
@@ -209,14 +209,14 @@
 	 */
 	if(arch_name == NULL){
 	    if(get_arch_from_host(&arch_flag, NULL)){
-#if __LP64__
+#if defined(__LP64__) || defined(_WIN64)
 		/*
 		 * If runing as a 64-bit binary and on an Intel x86 host
 		 * default to the 64-bit assember.
 		 */
 		if(arch_flag.cputype == CPU_TYPE_I386)
 		    arch_flag = *get_arch_family_from_cputype(CPU_TYPE_X86_64);
-#endif /* __LP64__ */
+#endif /* defined(__LP64__) || defined(_WIN64) */
 		arch_name = arch_flag.name;
 	    }
 	    else
diff -urN cctools.orig/include/mach-o/loader.h cctools/include/mach-o/loader.h
--- cctools.orig/include/mach-o/loader.h	2017-08-01 13:57:33.000000000 +0100
+++ cctools/include/mach-o/loader.h	2017-08-01 14:44:22.000000000 +0100
@@ -313,7 +313,7 @@
  */
 union lc_str {
 	uint32_t	offset;	/* offset to the string */
-#ifndef __LP64__
+#if !defined(__LP64__) && !defined(_WIN64)
 	char		*ptr;	/* pointer to the string */
 #endif 
 };
diff -urN cctools.orig/include/mach-o/nlist.h cctools/include/mach-o/nlist.h
--- cctools.orig/include/mach-o/nlist.h	2017-08-01 13:57:33.000000000 +0100
+++ cctools/include/mach-o/nlist.h	2017-08-01 14:44:22.000000000 +0100
@@ -75,7 +75,7 @@
  */
 struct nlist {
 	union {
-#ifndef __LP64__
+#if !defined(__LP64__) && !defined(_WIN64)
 		char *n_name;	/* for use when in-core */
 #endif
 		uint32_t n_strx;	/* index into the string table */
diff -urN cctools.orig/include/mach-o/ranlib.h cctools/include/mach-o/ranlib.h
--- cctools.orig/include/mach-o/ranlib.h	2017-08-01 13:57:33.000000000 +0100
+++ cctools/include/mach-o/ranlib.h	2017-08-01 14:44:22.000000000 +0100
@@ -58,7 +58,7 @@
 struct	ranlib {
     union {
 	uint32_t	ran_strx;	/* string table index of */
-#ifndef __LP64__
+#if !defined(__LP64__) && !defined(_WIN64)
 	char		*ran_name;	/* symbol defined by */
 #endif
     } ran_un;
diff -urN cctools.orig/libstuff/ofile.c cctools/libstuff/ofile.c
--- cctools.orig/libstuff/ofile.c	2017-08-01 14:43:46.000000000 +0100
+++ cctools/libstuff/ofile.c	2017-08-01 14:46:40.000000000 +0100
@@ -398,7 +398,7 @@
 	     */
 	    if(all_archs == FALSE){
 		(void)get_arch_from_host(&host_arch_flag, &specific_arch_flag);
-#if __LP64__
+#if defined(__LP64__) || defined(_WIN64)
 		/*
 		 * If runing as a 64-bit binary and on an Intel x86 host
 		 * default to 64-bit.
