diff -urN a/include/architecture/sparc/reg.h b/include/architecture/sparc/reg.h
--- a/include/architecture/sparc/reg.h	2013-11-18 23:05:37.725490376 +0000
+++ b/include/architecture/sparc/reg.h	2013-11-18 23:05:38.013486756 +0000
@@ -40,7 +40,7 @@
  * Usage is u.u_ar0[XX].
  */
 #define	PSR	(0)
-#define	PC	(1)
+/*#define	PC	(1) .. defining PC breaks llvm-3.4 */
 #define	nPC	(2)
 #define	SPARC_Y	(3)
 #define	G1	(4)
