diff -urN ld64.orig/src/ld/Options.cpp ld64/src/ld/Options.cpp
--- ld64.orig/src/ld/Options.cpp	2017-08-10 11:03:08.000000000 -0700
+++ ld64/src/ld/Options.cpp	2017-08-10 11:08:32.000000000 -0700
@@ -3993,8 +3993,10 @@
 		fSDKPaths.push_back(CROSS_SYSROOT);
 #endif
 	if ( addStandardLibraryDirectories ) {
-		libraryPaths.push_back("/usr/lib");
-		libraryPaths.push_back("/usr/local/lib");
+		// Not that conda will care, but fix:
+		// https://langui.sh/2015/07/24/osx-clang-include-lib-search-paths/
+		libraryPaths.push_back("/usr/local/lib");
+		libraryPaths.push_back("/usr/lib");
 
 		frameworkPaths.push_back("/Library/Frameworks/");
 		frameworkPaths.push_back("/System/Library/Frameworks/");
