Binary files ld64.orig/a.out and ld64/a.out differ
diff -urN ld64.orig/src/ld/Options.cpp ld64/src/ld/Options.cpp
--- ld64.orig/src/ld/Options.cpp	2017-08-10 11:13:45.000000000 -0700
+++ ld64/src/ld/Options.cpp	2017-08-10 13:25:52.000000000 -0700
@@ -3907,11 +3907,27 @@
 
 void Options::buildSearchPaths(int argc, const char* argv[])
 {
-	bool addStandardLibraryDirectories = true;
+	int addStandardLibraryDirectories = 1;
 	std::vector<const char*> libraryPaths;
 	std::vector<const char*> frameworkPaths;
+	char rawPath[PATH_MAX];
+	char path[PATH_MAX];
 	libraryPaths.reserve(10);
 	frameworkPaths.reserve(10);
+
+	// Conda-specific modification:
+	// If CONDA_BUILD_SYSROOT is set then add that as fSDKPaths, in the same way and
+	// place that CROSS_SYSROOT is handled (though it is one or the other).
+	const char* condaBuildDisableLD64Hacks = getenv("CONDA_BUILD_DISABLE_LD64_HACKS");
+	const char* envCondaBuildSysroot = (condaBuildDisableLD64Hacks == NULL) ? getenv("CONDA_BUILD_SYSROOT") : NULL;
+	if (envCondaBuildSysroot != NULL) {
+		addStandardLibraryDirectories = 2;
+		realpath(envCondaBuildSysroot, path);
+		if (strlen(path)) {
+			envCondaBuildSysroot = strdup(path);
+		}
+	}
+
 	// scan through argv looking for -L, -F, -Z, and -syslibroot options
 	for(int i=0; i < argc; ++i) {
 		if ( (argv[i][0] == '-') && (argv[i][1] == 'L') ) {
@@ -3943,7 +3959,7 @@
 			frameworkPaths.push_back(frameworkSearchDir);
 		}
 		else if ( strcmp(argv[i], "-Z") == 0 )
-			addStandardLibraryDirectories = false;
+			addStandardLibraryDirectories = 0;
 		else if ( strcmp(argv[i], "-v") == 0 ) {
 			fVerbose = true;
 			extern const char ldVersionString[];
@@ -3991,12 +4007,19 @@
 #ifdef CROSS_SYSROOT
 	if(fSDKPaths.size() == 0)
 		fSDKPaths.push_back(CROSS_SYSROOT);
+#else
+	if((envCondaBuildSysroot != NULL) && (fSDKPaths.size() == 0))
+		fSDKPaths.push_back(envCondaBuildSysroot);
 #endif
 	if ( addStandardLibraryDirectories ) {
 		// Not that conda will care, but fix:
 		// https://langui.sh/2015/07/24/osx-clang-include-lib-search-paths/
-		libraryPaths.push_back("/usr/local/lib");
-		libraryPaths.push_back("/usr/lib");
+		if ( addStandardLibraryDirectories == 1 ) {
+			libraryPaths.push_back("/usr/local/lib");
+			libraryPaths.push_back("/usr/lib");
+		} else {
+			libraryPaths.push_back("/usr/lib");
+		}
 
 		frameworkPaths.push_back("/Library/Frameworks/");
 		frameworkPaths.push_back("/System/Library/Frameworks/");
@@ -4012,7 +4035,8 @@
 	}
 
 	// now merge sdk and library paths to make real search paths
-	fLibrarySearchPaths.reserve(libraryPaths.size()*(fSDKPaths.size()+1));
+	fLibrarySearchPaths.reserve(libraryPaths.size()*(fSDKPaths.size()+2));
+
 	int libIndex = 0;
 	for (std::vector<const char*>::iterator it = libraryPaths.begin(); it != libraryPaths.end(); ++it, ++libIndex) {
 		const char* libDir = *it;
@@ -4057,7 +4081,7 @@
 					else
 						fLibrarySearchPaths.push_back(libDir);
 				}
-				else if ( !addStandardLibraryDirectories || (strcmp(libDir, "/usr/local/lib") != 0) ) {
+				else if ( (addStandardLibraryDirectories == 0) || (strcmp(libDir, "/usr/local/lib") != 0) ) {
 					warning("directory not found for option '-L%s'", libDir);
 				}
 			}
@@ -4118,6 +4142,27 @@
 		}
 	}
 
+	// Conda-specific modification:
+	// Add $(dirname of this executable)/../lib as the first library path so that our libc++ gets found first.
+	// No -L or -F concatenations are performed on this path.
+	if (condaBuildDisableLD64Hacks == NULL) {
+		uint32_t bufSize = PATH_MAX;	
+		if ( _NSGetExecutablePath(rawPath, &bufSize) != -1 ) {
+			if ( realpath(rawPath, path) != NULL ) {
+				char* lastSlash = strrchr(path, '/');
+				if ( lastSlash != NULL ) {
+					*lastSlash = '\0';
+					char* lastSlash = strrchr(path, '/');
+					lastSlash = strrchr(path, '/');
+					if ( lastSlash != NULL ) {
+						strcpy(lastSlash+1, "lib");
+						fLibrarySearchPaths.insert(fLibrarySearchPaths.begin(), strdup(path));
+					}
+				}
+			}
+		}
+	}
+	
 	if ( fVerbose ) {
 		fprintf(stderr,"Library search paths:\n");
                 for (std::vector<const char*>::iterator it = fLibrarySearchPaths.begin();
