--- ld64-409.12/src/ld/parsers/textstub_dylib_file.cpp  2018-09-18 22:03:37.000000000 +0000
+++ ld64-409.12/src/ld/parsers/textstub_dylib_file.cpp     2019-09-16 22:48:53.425416705 +0000
@@ -119,15 +119,6 @@
                platforms.add({ld::kPlatform_bridgeOS, 0});
                break;
        #endif
-       #if ((TAPI_API_VERSION_MAJOR == 1 &&  TAPI_API_VERSION_MINOR >= 4) || (TAPI_API_VERSION_MAJOR > 1))
-       case tapi::Platform::iOSMac:
-               platforms.add({ld::kPlatform_iOSMac, 0});
-               break;
-       case tapi::Platform::zippered:
-               platforms.add({ld::kPlatform_macOS, 0});
-               platforms.add({ld::kPlatform_iOSMac, 0});
-               break;
-       #endif
        }
 
        return platforms;
