diff --git a/setup.py b/setup.py
index fed64e8..7338a17 100644
--- a/setup.py
+++ b/setup.py
@@ -58,6 +58,7 @@ class CMakeBuild(build_ext):
             cmake_args += [
                 "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}".format(cfg.upper(), extdir)
             ]
+            self.build_temp = "C:/"+str(hex(hash(self.build_temp)))[1:5]
         else:
             build_args += ["--", "-j4"]
 
