diff --git a/setup.py b/setup.py
index 14427dc..3ee2cd2 100644
--- a/setup.py
+++ b/setup.py
@@ -55,7 +55,8 @@ class CMakeBuild(build_ext):
 
         if platform.system() == "Windows":
             print("-----------------Windows...")
-            build_args += ["--parallel 1"]
+            path_hash = hash(self.build_temp)
+            self.build_temp = "C:/"+str(path_hash)[1:5]
         else:
             build_args += ["--parallel 4"]
         cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
