diff --git a/CMakeLists.txt b/CMakeLists.txt
index 52f7856..16471fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,10 @@ pybind11_add_module(libbfiocpp
   ${SOURCE}
 )
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zm1000")
+endif()
+
 # VERSION_INFO is defined by setup.py and passed into the C++ code as a define (VERSION_INFO) here.
 target_compile_definitions(libbfiocpp PRIVATE VERSION_INFO=${VERSION_INFO})
 
diff --git a/setup.py b/setup.py
index 2996b62..14a4f3d 100644
--- a/setup.py
+++ b/setup.py
@@ -59,7 +59,7 @@ class CMakeBuild(build_ext):
             ]
             if sys.maxsize > 2**32:
                 cmake_args += ["-A", "x64"]
-            build_args += ["--", "/m"]
+            build_args += ["--parallel 1" ]
         else:
             cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
             build_args += ["--", "-j4"]
