diff --git a/setup.py b/setup.py
index 2996b62..ad3e4d0 100644
--- a/setup.py
+++ b/setup.py
@@ -54,12 +54,12 @@ class CMakeBuild(build_ext):
 
         if platform.system() == "Windows":
             print("-----------------Windows...")
-            cmake_args += [
-                "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}".format(cfg.upper(), extdir)
-            ]
-            if sys.maxsize > 2**32:
-                cmake_args += ["-A", "x64"]
-            build_args += ["--", "/m"]
+            # cmake_args += [
+            #     "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}".format(cfg.upper(), extdir)
+            # ]
+            # if sys.maxsize > 2**32:
+            #     cmake_args += ["-A", "x64"]
+            # build_args += ["--parallel 1" ]
         else:
             cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
             build_args += ["--", "-j4"]
