diff --git a/setup.py b/setup.py
index 2996b62..14a4f3d 100644
--- a/setup.py
+++ b/setup.py
@@ -59,7 +59,7 @@ class CMakeBuild(build_ext):
             ]
             if sys.maxsize > 2**32:
                 cmake_args += ["-A", "x64"]
-            build_args += ["--", "/m"]
+            build_args += ["--", "/m:1"]
         else:
             cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
             build_args += ["--", "-j4"]
