diff --git a/scripts/install-tree-sitter-lib b/scripts/install-tree-sitter-lib
index 398386b..dfcc3ab 100755
--- a/scripts/install-tree-sitter-lib
+++ b/scripts/install-tree-sitter-lib
@@ -61,7 +61,7 @@ git clone --depth 1 --branch "$version" \
   # When building from source on homebrew sudo is not allowed
   # but also not needed
   if [[ -z "${HOMEBREW_SYSTEM+x}" ]]; then
-    sudo make install
+    make install
   else
     make install
   fi
