From f8ce29b346b9e3330014c089a3da6ad129194c21 Mon Sep 17 00:00:00 2001
From: Ahmed Fasih <wuzzyview@gmail.com>
Date: Fri, 16 Apr 2021 15:43:33 -0700
Subject: [PATCH] patch ocaml-tree-sitter submodule to remove sudo

---
 Makefile | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Makefile b/Makefile
index e3623369..8c630686 100644
--- a/Makefile
+++ b/Makefile
@@ -67,6 +67,13 @@ rebuild:
 .PHONY: setup
 setup:
 	git submodule update --init
+
+	# apply patch to remove sudo in ocaml-tree-sitter (a git submodule)
+	# --> scripts/install-tree-sitter-lib
+	cd ocaml-tree-sitter && ls
+	git apply ${RECIPE_DIR}/0001-no-sudo-on-Linux-CI.patch
+	cd ..
+
 	opam update -y
 	./scripts/install-ocaml-tree-sitter
 	opam install -y --deps-only ./semgrep-core/src/pfff
-- 
2.30.2

