diff -ur PySide-1.2.4.orig/sources/shiboken/ext/sparsehash/google/sparsehash/sparseconfig.h PySide-1.2.4/sources/shiboken/ext/sparsehash/google/sparsehash/sparseconfig.h
--- sources/shiboken/ext/sparsehash/google/sparsehash/sparseconfig.h	2016-10-01 11:50:45.455842332 +0200
+++ sources/shiboken/ext/sparsehash/google/sparsehash/sparseconfig.h	2016-10-01 11:58:18.889198432 +0200
@@ -8,7 +8,16 @@
 
 #ifdef _MSC_VER
     /* the location of the header defining hash functions */
-    #define HASH_FUN_H <hash_map>
+    #if (_MSC_VER >= 1800 )
+
+        #define HASH_FUN_H  <unordered_map>
+ 
+    #else /* Earlier than VSC++ 2013 */ 
+
+        #define HASH_FUN_H  <hash_map>
+ 
+    #endif
+
     /* the namespace of the hash<> function */
     #define HASH_NAMESPACE stdext
     /* The system-provided hash function including the namespace. */
