From e7b2db04c2e738c72fefdcac26313af13a427210 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Sun, 24 Mar 2024 08:54:54 -0500
Subject: [PATCH] depend on dataclasses_json

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 2599d8c..eabbc75 100644
--- a/setup.py
+++ b/setup.py
@@ -20,7 +20,7 @@
     install_requires=[
         "certifi>=2023.7.22",
         "charset-normalizer>=3.2.0",
-        "dataclasses-json-speakeasy>=0.5.11",
+        "dataclasses-json>=0.5.11",
         "idna>=3.4",
         "jsonpath-python>=1.0.6 ",
         "marshmallow>=3.19.0",
