--- setup.py	2019-05-13 21:18:50.533585605 -0400
+++ setup.py	2019-05-13 22:04:50.740553644 -0400
@@ -39,7 +39,7 @@
 setup(
     name='pyshacl',
     packages=['pyshacl', 'pyshacl.constraints', 'pyshacl.constraints.core', 'pyshacl.constraints.sparql', 'pyshacl.inference', 'pyshacl.rdfutil', 'pyshacl.monkey'],
-    scripts=['bin/pyshacl'],
+    # scripts=['bin/pyshacl'],
     package_dir={'pyshacl': './pyshacl'},
     package_data={'pyshacl': ['*.pickle']},
     #data_files=[('pyshacl', ['pyshacl/shacl-shacl.pickle'])],
@@ -75,4 +75,3 @@
     install_requires=requirements,
     dependency_links=dependency_links
 )
-
