diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3270f47..5cbd872 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,10 +1,12 @@
 
-PROJECT (xcdf CXX C)
+PROJECT (xcdf CXX)
 
 CMAKE_MINIMUM_REQUIRED (VERSION 2.8)
 
 ADD_DEFINITIONS ("-Wall -O2")
 
+SET (CMAKE_CXX_STANDARD 98)
+
 # ------------------------------------------------------------------------------
 # Prevent in-place builds
 # ------------------------------------------------------------------------------
@@ -74,15 +76,15 @@ IF (PYTHONLIBS_FOUND)
     PREFIX ""
     COMPILE_FLAGS "-fno-strict-aliasing"
     OUTPUTNAME pyxcdf)
-  INSTALL (TARGETS pyxcdf LIBRARY DESTINATION lib/xcdf)
+  INSTALL (TARGETS pyxcdf LIBRARY DESTINATION xcdf)
 
-  # Install __init__.py and other scripts into lib/xcdf
+  # Install __init__.py and other scripts into xcdf
   CONFIGURE_FILE (${CMAKE_CURRENT_SOURCE_DIR}/python/__init__.py.in
                   ${CMAKE_CURRENT_BINARY_DIR}/__init__.py
                   @ONLY)
   INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/__init__.py
                  ${CMAKE_CURRENT_SOURCE_DIR}/python/Histogram.py
-          DESTINATION lib/xcdf
+          DESTINATION xcdf
           PERMISSIONS OWNER_READ OWNER_WRITE
                       GROUP_READ GROUP_WRITE
                       WORLD_READ)
