--- lib/jxl.cmake
+++ lib/jxl.cmake
@@ -576,10 +576,11 @@
 
 # Check whether the linker support excluding libs
 set(LINKER_EXCLUDE_LIBS_FLAG "-Wl,--exclude-libs=ALL")
+set(CMAKE_REQUIRED_FLAGS_BEFORE "${CMAKE_REQUIRED_FLAGS}")
 include(CheckCSourceCompiles)
-list(APPEND CMAKE_EXE_LINKER_FLAGS ${LINKER_EXCLUDE_LIBS_FLAG})
+set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS_BEFORE} ${LINKER_EXCLUDE_LIBS_FLAG}")
 check_c_source_compiles("int main(){return 0;}" LINKER_SUPPORT_EXCLUDE_LIBS)
-list(REMOVE_ITEM CMAKE_EXE_LINKER_FLAGS ${LINKER_EXCLUDE_LIBS_FLAG})
+set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS_BEFORE}")
 
 # Add a jxl.version file as a version script to tag symbols with the
 # appropriate version number. This script is also used to limit what's exposed
