diff --git a/epics/extensions/src/SDDS/png/config.h b/epics/extensions/src/SDDS/png/config.h
index efbf6c3..4d8f812 100644
--- a/epics/extensions/src/SDDS/png/config.h
+++ b/epics/extensions/src/SDDS/png/config.h
@@ -72,13 +72,13 @@
 #define PACKAGE_VERSION "1.6.10"
 
 /* Turn on ARM Neon optimizations at run-time */
-/* #undef PNG_ARM_NEON_API_SUPPORTED */
+#define PNG_ARM_NEON_API_SUPPORTED 0
 
 /* Check for ARM Neon support at run-time */
-/* #undef PNG_ARM_NEON_CHECK_SUPPORTED */
+#define PNG_ARM_NEON_CHECK_SUPPORTED 0
 
 /* Enable ARM Neon optimizations */
-/* #undef PNG_ARM_NEON_OPT */
+#define PNG_ARM_NEON_OPT 0
 
 /* Define to 1 if you have the ANSI C header files. */
 #define STDC_HEADERS 1
