--- setup.py	2021-05-10 19:05:32.773165761 -0400
+++ setup.py	2021-05-10 19:05:02.125077494 -0400
@@ -1,7 +1,5 @@
 #!/usr/bin/env python3

-# std
-from distutils.core import setup
 # noinspection PyUnresolvedReferences
 import setuptools  # see below (1)
 from pathlib import Path
@@ -29,7 +27,7 @@
 with (this_dir / "randomfiletree" / "version.txt").open() as vf:
     version = vf.read()

-setup(
+setuptools.setup(
     name='RandomFileTree',
     version=version,
     packages=packages,
@@ -44,7 +42,9 @@
     },
     install_requires=[],
     license="MIT",
-    scripts=["randomfiletree/bin/randomfiletree"],
+    entry_points={
+        "console_scripts": ["randomfiletree=randomfiletree.cli:cli"]
+    },
     keywords=keywords,
     description=description,
     long_description=long_description,
