From 1d188ef7df8b919460c82b9512473f3c9914dbfa Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Mon, 14 Feb 2022 14:54:14 +0100
Subject: [PATCH] Update setup.py

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index ba67717..5013e98 100644
--- a/setup.py
+++ b/setup.py
@@ -40,7 +40,6 @@
         "flax",
         "jax>=0.2.17",
         "jaxlib>=0.1.65",
-        "opencv-python",
         "Pillow",
         "cma",
     ],
