From 7ab57006a38816c13a9ef37a5df7aad63b136459 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juan=20Luis=20Cano=20Rodr=C3=ADguez?= <juanlu001@gmail.com>
Date: Mon, 3 Sep 2018 11:30:08 +0200
Subject: [PATCH] Include License in source distribution

---
 MANIFEST.in | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 MANIFEST.in

diff --git MANIFEST.in MANIFEST.in
new file mode 100644
index 0000000..42eb410
--- /dev/null
+++ MANIFEST.in
@@ -0,0 +1 @@
+include LICENSE.txt
-- 
2.7.4

