--- src/app/main.cpp	2016-04-29 08:00:28.000000000 -0400
+++ src/app/main.cpp	2016-05-06 02:34:20.688609000 -0400
@@ -1014,6 +1014,20 @@
   QCoreApplication::addLibraryPath( QgsApplication::pluginPath() );
   QCoreApplication::addLibraryPath( myPath );
 #endif
+#ifdef Q_OS_LINUX
+  // Same as on Mac OSX -- check for location of "plugins"
+  QString myPath( QCoreApplication::applicationDirPath().append( "/../plugins" ) );
+  if ( !QFile::exists( myPath + "/imageformats" ) )
+  {
+    myPath = QT_PLUGINS_DIR;
+  }
+  QCoreApplication::addLibraryPath( myPath );
+
+  // Debug
+  QgsDebugMsg( QString( "QGIS library paths: " ) );
+  foreach ( const QString &path, QCoreApplication::libraryPaths() )
+    QgsDebugMsg( QString( "    %s" ).arg( path ) );
+#endif
 
   // set authentication database directory
   if ( !authdbdirectory.isEmpty() )
