From dee6054b23e8b0d9fb0c7e01c0a8799b473450fd Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Thu, 15 Nov 2018 19:11:44 +0000
Subject: [PATCH] Set CLANG_INCLUDE_DIRS before configuring clad

---
 interpreter/CMakeLists.txt | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/interpreter/CMakeLists.txt b/interpreter/CMakeLists.txt
index 3752a6e4237..eb22e8dff60 100644
--- a/interpreter/CMakeLists.txt
+++ b/interpreter/CMakeLists.txt
@@ -374,14 +374,6 @@ Please install Python or specify the PYTHON_EXECUTABLE CMake variable.")
   set( CLANG_BUILT_STANDALONE 1 )
   set(BACKEND_PACKAGE_STRING "LLVM ${LLVM_PACKAGE_VERSION}")
 
-
-if (builtin_clang)
-  add_subdirectory(llvm/src/tools/clang EXCLUDE_FROM_ALL)
-else()
-  add_subdirectory(cling)
-endif()
-endif(builtin_llvm)
-
 #---Export the include directories------------------------------------------------------------------
 if (builtin_clang)
   set(CLANG_INCLUDE_DIRS
@@ -390,6 +382,13 @@ if (builtin_clang)
     CACHE STRING "Clang include directories.")
 endif()
 
+if (builtin_clang)
+  add_subdirectory(llvm/src/tools/clang EXCLUDE_FROM_ALL)
+else()
+  add_subdirectory(cling)
+endif()
+endif(builtin_llvm)
+
 if(builtin_llvm)
   set(LLVM_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/interpreter/llvm/src/include
     ${CMAKE_BINARY_DIR}/interpreter/llvm/src/include
