diff --git a/gethost.c b/gethost.c
index b304bb9..872789a 100644
--- a/gethost.c
+++ b/gethost.c
@@ -259,7 +259,7 @@ struct addrlist *get_address_info (
            if (ai->ai_family == AF_INET) {
                struct sockaddr_in *sin = (struct sockaddr_in *)ai->ai_addr;
                src = &(sin->sin_addr);
-                if (*(const in_addr_t *) src == htonl(INADDR_LOOPBACK)) {
+                if (*(const uint32_t *) src == htonl(INADDR_LOOPBACK)) {
                     family = FamilyLocal;
                     if (get_local_hostname (buf, sizeof buf)) {
                         src = buf;