--- gethost.c	2022-04-20 11:04:52.000000000 +0000
+++ gethost.c	2024-04-29 11:04:52.000000000 +0000
@@ -259,7 +259,7 @@
 	    if (ai->ai_family == AF_INET) {
 		struct sockaddr_in *sin = (struct sockaddr_in *)ai->ai_addr;
 		src = &(sin->sin_addr);
-                if (*(const in_addr_t *) src == htonl(INADDR_LOOPBACK)) {
+                if (*(const uint32_t *) src == htonl(INADDR_LOOPBACK)) {
                     family = FamilyLocal;
                     if (get_local_hostname (buf, sizeof buf)) {
                         src = buf;
@@ -329,7 +329,7 @@
 #else
 	if (!get_inet_address (host, &hostinetaddr)) return NULL;
 	src = (char *) &hostinetaddr;
-        if (*(const in_addr_t *) src == htonl(INADDR_LOOPBACK)) {
+        if (*(const uint32_t *) src == htonl(INADDR_LOOPBACK)) {
             family = FamilyLocal;
             if (get_local_hostname (buf, sizeof buf)) {
                 src = buf;
