--- setup.py.orig	2017-10-01 08:58:08.751704067 +0200
+++ setup.py	2017-10-01 09:02:05.991715979 +0200
@@ -7,6 +7,7 @@
 
 from numpy.distutils.core import setup, Extension
 import os
+import sys
 from subprocess import call
 import numpy as np
 
@@ -15,6 +16,9 @@
 
 from Cython.Build import cythonize
 
+extra_compile_args=[]
+if not sys.platform.startswith('win'):
+    extra_compile_args.append('-std=c++11')
 
 ext = cythonize(
     Extension("smt.methods.rbfclib",
@@ -22,7 +26,7 @@
         'smt/src/rbf/rbf.cpp',
         'smt/src/rbf/rbfclib.pyx',
     ],
-    language="c++", extra_compile_args=['-std=c++11'],
+    language="c++", extra_compile_args=extra_compile_args,
     include_dirs=[np.get_include(),
 ])) + cythonize(
     Extension("smt.methods.idwclib",
@@ -30,7 +34,7 @@
         'smt/src/idw/idw.cpp',
         'smt/src/idw/idwclib.pyx',
     ],
-    language="c++", extra_compile_args=['-std=c++11'],
+    language="c++", extra_compile_args=extra_compile_args,
     include_dirs=[np.get_include(),
 ])) + cythonize(
     Extension("smt.methods.rmtsclib",
@@ -41,7 +45,7 @@
         'smt/src/rmts/rmtb.cpp',
         'smt/src/rmts/rmtc.cpp',
     ],
-    language="c++", extra_compile_args=['-std=c++11'],
+    language="c++", extra_compile_args=extra_compile_args,
     include_dirs=[np.get_include(),
 ]))
 
