diff --git a/smt/src/rmts/rmtc.cpp b/smt/src/rmts/rmtc.cpp
index 3ce22f5..0a8fb49 100644
--- a/smt/src/rmts/rmtc.cpp
+++ b/smt/src/rmts/rmtc.cpp
@@ -5,6 +5,7 @@
 #include <math.h>
 #include <iostream>
 #include <cstring>
+#include <algorithm>
 
 using namespace std;
 
diff --git a/smt/src/rmts/rmts.cpp b/smt/src/rmts/rmts.cpp
index 73928c2..b42b742 100644
--- a/smt/src/rmts/rmts.cpp
+++ b/smt/src/rmts/rmts.cpp
@@ -4,6 +4,7 @@
 #include <math.h>
 #include <iostream>
 #include <cstring>
+#include <algorithm>
 
 using namespace std;
 
