diff --git a/arcade/__init__.py b/arcade/__init__.py
index a5152680..9f410105 100644
--- a/arcade/__init__.py
+++ b/arcade/__init__.py
@@ -44,16 +44,16 @@ def configure_logging(level: int | None = None):
 # However it is setup to load ffmpeg if the binaries are present
 # on Windows and Linux. So if you need ffmpeg you can simply
 # drop the binaries in the "lib" folder of Arcade
-lib_location = Path(__file__).parent.absolute()
-lib_location = lib_location / "lib"
-
-if sys.platform == "darwin" or sys.platform.startswith("linux"):
-    if "LD_LIBRARY_PATH" in os.environ:
-        os.environ["LD_LIBRARY_PATH"] += ":" + str(lib_location)
-    else:
-        os.environ["LD_LIBRARY_PATH"] = str(lib_location)
-else:
-    os.environ["PATH"] += str(lib_location)
+# lib_location = Path(__file__).parent.absolute()
+# lib_location = lib_location / "lib"
+
+# if sys.platform == "darwin" or sys.platform.startswith("linux"):
+#     if "LD_LIBRARY_PATH" in os.environ:
+#         os.environ["LD_LIBRARY_PATH"] += ":" + str(lib_location)
+#     else:
+#         os.environ["LD_LIBRARY_PATH"] = str(lib_location)
+# else:
+#     os.environ["PATH"] += str(lib_location)

 import pyglet
