diff --git a/arcade/__init__.py b/arcade/__init__.py
index dbf9f2ba..5b196a29 100644
--- a/arcade/__init__.py
+++ b/arcade/__init__.py
@@ -44,16 +44,16 @@ def configure_logging(level: int | None = None):
 # However it is setup to load ffmpeg if the binaries are present
 # on Windows and Linux. So if you need ffmpeg you can simply
 # drop the binaries in the "lib" folder of Arcade
-lib_location = Path(__file__).parent.absolute()
-lib_location = lib_location / "lib"
-
-if sys.platform == "darwin" or sys.platform.startswith("linux"):
-    if "LD_LIBRARY_PATH" in os.environ:
-        os.environ["LD_LIBRARY_PATH"] += ":" + str(lib_location)
-    else:
-        os.environ["LD_LIBRARY_PATH"] = str(lib_location)
-else:
-    os.environ["PATH"] += str(lib_location)
+# lib_location = Path(__file__).parent.absolute()
+# lib_location = lib_location / "lib"
+
+# if sys.platform == "darwin" or sys.platform.startswith("linux"):
+#     if "LD_LIBRARY_PATH" in os.environ:
+#         os.environ["LD_LIBRARY_PATH"] += ":" + str(lib_location)
+#     else:
+#         os.environ["LD_LIBRARY_PATH"] = str(lib_location)
+# else:
+#     os.environ["PATH"] += str(lib_location)

 import pyglet

diff --git a/arcade/lib/libavcodec.58.dylib b/arcade/lib/libavcodec.58.dylib
deleted file mode 100755
index 032d6ad9..00000000
Binary files a/arcade/lib/libavcodec.58.dylib and /dev/null differ
diff --git a/arcade/lib/libavformat.58.dylib b/arcade/lib/libavformat.58.dylib
deleted file mode 100755
index 3932f16c..00000000
Binary files a/arcade/lib/libavformat.58.dylib and /dev/null differ
diff --git a/arcade/lib/libavutil.56.dylib b/arcade/lib/libavutil.56.dylib
deleted file mode 100755
index 635253f4..00000000
Binary files a/arcade/lib/libavutil.56.dylib and /dev/null differ
diff --git a/arcade/lib/libswresample.3.dylib b/arcade/lib/libswresample.3.dylib
deleted file mode 100755
index 080ab859..00000000
Binary files a/arcade/lib/libswresample.3.dylib and /dev/null differ
diff --git a/arcade/lib/libswscale.5.dylib b/arcade/lib/libswscale.5.dylib
deleted file mode 100755
index bb56224a..00000000
Binary files a/arcade/lib/libswscale.5.dylib and /dev/null differ
