--- pystac_core-1.15.1.orig/pystac/__init__.py	2020-02-01 21:00:00.000000000 -0300
+++ pystac_core-1.15.1/pystac/__init__.py	2026-07-01 10:43:41.489226407 -0300
@@ -91,56 +91,56 @@
 from pystac.utils import HREF
 
 import pystac.extensions.hooks
-import pystac.extensions.classification
-import pystac.extensions.datacube
-import pystac.extensions.eo
-import pystac.extensions.file
-import pystac.extensions.grid
-import pystac.extensions.item_assets
+# import pystac.extensions.classification
+# import pystac.extensions.datacube
+# import pystac.extensions.eo
+# import pystac.extensions.file
+# import pystac.extensions.grid
+# import pystac.extensions.item_assets
 
-with warnings.catch_warnings():
-    warnings.filterwarnings("ignore", category=DeprecationWarning)
-    import pystac.extensions.label
-import pystac.extensions.mgrs
-import pystac.extensions.mlm
-import pystac.extensions.pointcloud
-import pystac.extensions.projection
-import pystac.extensions.raster
-import pystac.extensions.sar
-import pystac.extensions.sat
-import pystac.extensions.scientific
-import pystac.extensions.storage
-import pystac.extensions.table
-import pystac.extensions.timestamps
-import pystac.extensions.version
-import pystac.extensions.view
-import pystac.extensions.xarray_assets
+# with warnings.catch_warnings():
+#     warnings.filterwarnings("ignore", category=DeprecationWarning)
+#     import pystac.extensions.label
+# import pystac.extensions.mgrs
+# import pystac.extensions.mlm
+# import pystac.extensions.pointcloud
+# import pystac.extensions.projection
+# import pystac.extensions.raster
+# import pystac.extensions.sar
+# import pystac.extensions.sat
+# import pystac.extensions.scientific
+# import pystac.extensions.storage
+# import pystac.extensions.table
+# import pystac.extensions.timestamps
+# import pystac.extensions.version
+# import pystac.extensions.view
+# import pystac.extensions.xarray_assets
 
-EXTENSION_HOOKS = pystac.extensions.hooks.RegisteredExtensionHooks(
-    [
-        pystac.extensions.classification.CLASSIFICATION_EXTENSION_HOOKS,
-        pystac.extensions.datacube.DATACUBE_EXTENSION_HOOKS,
-        pystac.extensions.eo.EO_EXTENSION_HOOKS,
-        pystac.extensions.file.FILE_EXTENSION_HOOKS,
-        pystac.extensions.grid.GRID_EXTENSION_HOOKS,
-        pystac.extensions.item_assets.ITEM_ASSETS_EXTENSION_HOOKS,
-        pystac.extensions.label.LABEL_EXTENSION_HOOKS,
-        pystac.extensions.mgrs.MGRS_EXTENSION_HOOKS,
-        pystac.extensions.mlm.MLM_EXTENSION_HOOKS,
-        pystac.extensions.pointcloud.POINTCLOUD_EXTENSION_HOOKS,
-        pystac.extensions.projection.PROJECTION_EXTENSION_HOOKS,
-        pystac.extensions.raster.RASTER_EXTENSION_HOOKS,
-        pystac.extensions.sar.SAR_EXTENSION_HOOKS,
-        pystac.extensions.sat.SAT_EXTENSION_HOOKS,
-        pystac.extensions.scientific.SCIENTIFIC_EXTENSION_HOOKS,
-        pystac.extensions.storage.STORAGE_EXTENSION_HOOKS,
-        pystac.extensions.table.TABLE_EXTENSION_HOOKS,
-        pystac.extensions.timestamps.TIMESTAMPS_EXTENSION_HOOKS,
-        pystac.extensions.version.VERSION_EXTENSION_HOOKS,
-        pystac.extensions.view.VIEW_EXTENSION_HOOKS,
-        pystac.extensions.xarray_assets.XARRAY_ASSETS_EXTENSION_HOOKS,
-    ]
-)
+# EXTENSION_HOOKS = pystac.extensions.hooks.RegisteredExtensionHooks(
+#     [
+#         pystac.extensions.classification.CLASSIFICATION_EXTENSION_HOOKS,
+#         pystac.extensions.datacube.DATACUBE_EXTENSION_HOOKS,
+#         pystac.extensions.eo.EO_EXTENSION_HOOKS,
+#         pystac.extensions.file.FILE_EXTENSION_HOOKS,
+#         pystac.extensions.grid.GRID_EXTENSION_HOOKS,
+#         pystac.extensions.item_assets.ITEM_ASSETS_EXTENSION_HOOKS,
+#         pystac.extensions.label.LABEL_EXTENSION_HOOKS,
+#         pystac.extensions.mgrs.MGRS_EXTENSION_HOOKS,
+#         pystac.extensions.mlm.MLM_EXTENSION_HOOKS,
+#         pystac.extensions.pointcloud.POINTCLOUD_EXTENSION_HOOKS,
+#         pystac.extensions.projection.PROJECTION_EXTENSION_HOOKS,
+#         pystac.extensions.raster.RASTER_EXTENSION_HOOKS,
+#         pystac.extensions.sar.SAR_EXTENSION_HOOKS,
+#         pystac.extensions.sat.SAT_EXTENSION_HOOKS,
+#         pystac.extensions.scientific.SCIENTIFIC_EXTENSION_HOOKS,
+#         pystac.extensions.storage.STORAGE_EXTENSION_HOOKS,
+#         pystac.extensions.table.TABLE_EXTENSION_HOOKS,
+#         pystac.extensions.timestamps.TIMESTAMPS_EXTENSION_HOOKS,
+#         pystac.extensions.version.VERSION_EXTENSION_HOOKS,
+#         pystac.extensions.view.VIEW_EXTENSION_HOOKS,
+#         pystac.extensions.xarray_assets.XARRAY_ASSETS_EXTENSION_HOOKS,
+#     ]
+# )
 
 
 def read_file(href: HREF, stac_io: StacIO | None = None) -> STACObject:
