--- "lpsolve55\\lpsolve.rc"	2016-09-18 21:34:29.000000000 +0000
+++ "lpsolve55\\lpsolve.rc"	2017-05-06 12:42:35.757868500 +0000
@@ -7,7 +7,7 @@
 //
 // Generated from the TEXTINCLUDE 2 resource.
 //
-#include "afxres.h"
+#include "windows.h"
 
 /////////////////////////////////////////////////////////////////////////////
 #undef APSTUDIO_READONLY_SYMBOLS
--- "lpsolve55\\cvc8msvcrt.bat"	2011-01-23 21:04:21.000000000 +0000
+++ "lpsolve55\\cvc8msvcrt.bat"	2017-05-07 12:25:11.409034700 +0000
@@ -20,7 +20,8 @@
 set c=cl
 
 REM determine platform (win32/win64)
-echo main(){printf("SET PLATFORM=win%%d\n", (int) (sizeof(void *)*8));}>platform.c
+echo #include ^<stdio.h^>>platform.c
+echo main(){printf("SET PLATFORM=win%%d\n", (int) (sizeof(void *)*8));}>>platform.c
 %c% /nologo platform.c /Feplatform.exe
 del platform.c
 platform.exe >platform.bat
--- "lp_solve\\cvc8.bat"	2009-01-23 23:00:37.000000000 +0000
+++ "lp_solve\\cvc8.bat"	2017-05-07 12:25:20.855167000 +0000
@@ -5,7 +5,8 @@
 set c=cl
 
 REM determine platform (win32/win64)
-echo main(){printf("SET PLATFORM=win%%d\n", (int) (sizeof(void *)*8));}>platform.c
+echo #include ^<stdio.h^>>platform.c
+echo main(){printf("SET PLATFORM=win%%d\n", (int) (sizeof(void *)*8));}>>platform.c
 %c% /nologo platform.c /Feplatform.exe
 del platform.c
 platform.exe >platform.bat
