--- lp_solve/ccc 2009-01-25 18:39:03.000000000 +0000
+++ lp_solve/ccc_patched 2017-02-26 00:55:56.722775232 +0000
@@ -20,7 +20,7 @@
 echo '#include <stdio.h>'>>/tmp/isnan.c
 echo '#include <stdlib.h>'>>/tmp/isnan.c
 echo '#include <math.h>'>>/tmp/isnan.c
-echo 'main(){isnan(0);}'>>/tmp/isnan.c
+echo 'main(){isnan(0.0);}'>>/tmp/isnan.c
 $c /tmp/isnan.c -o /tmp/isnan $math >/dev/null 2>&1
 if [ $? = 0 ]
 then NOISNAN=
--- lpsolve55/ccc 2017-02-26 00:53:20.594775232 +0000
+++ lpsolve55/ccc_patched 2017-02-26 00:53:16.942775232 +0000
@@ -18,7 +18,7 @@
 echo '#include <stdio.h>'>>/tmp/isnan.c
 echo '#include <stdlib.h>'>>/tmp/isnan.c
 echo '#include <math.h>'>>/tmp/isnan.c
-echo 'main(){isnan(0);}'>>/tmp/isnan.c
+echo 'main(){isnan(0.0);}'>>/tmp/isnan.c
 $c /tmp/isnan.c -o /tmp/isnan $math >/dev/null 2>&1
 if [ $? = 0 ]
 then NOISNAN=
