diff --git a/genesis.def b/genesis.def
index 826e68d..ca7bc9c 100644
--- a/genesis.def
+++ b/genesis.def
@@ -14,8 +14,8 @@ c
      +          f_sdds = 2 ,                 !indicator for sdds filetype
      +          npmax  = 1000001,             !# of particles
      +          nzmax  = 10000,              !# of integration steps
-     +          nsmax  = 150000,              !# of slices
-     +          nhmax  = 5,                  !maximum of harmonics
+     +          nsmax  = 50000,              !# of slices
+     +          nhmax  = 7,                  !maximum of harmonics
      +          ndmax  = 1250000,            !maximum of particle in imported distribution
      +          keepdist = 1,                !<> 0 keeps distribution in memory
      +          eev    = 510999.06d0,        !energy units (mc^2) in ev
@@ -27,7 +27,7 @@ c
      +          tiny   = 1.d-25,             !check i for precission 
      +          small  = 1.d-7,              !check ii for precission
      +          nrgrid = 1000,               !number of radial points for s. c.
-     +          ncmax  = 503)                !# of gridpoints of cartesian mesh
+     +          ncmax  = 261)                !# of gridpoints of cartesian mesh
 c
 c     error codes 
 c
diff --git a/timerec.cmn b/timerec.cmn
index 1284c7b..45f6260 100644
--- a/timerec.cmn
+++ b/timerec.cmn
@@ -7,7 +7,7 @@ c
       integer ntmax,nofile
 c
       parameter(
-     +          ntmax  = 501*501*500, !must be > than ncar*ncar*nslp(98)
+     +          ntmax  = 151*151*500, !must be > than ncar*ncar*nslp(98)
      +          nofile = 0)   !<>0 -> slippage is stored on disk
 c
 c     --------------------------------------------------------------------

