diff --git a/Makefile b/Makefile
index fb02979..6d52a3a 100644
--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,7 @@ EXECUTABLE = genesis2
 # targets
 #
 genesis2:	$(SOURCE)
-	$(COMPILER) -g -w -O -Wall -o $(EXECUTABLE) $(SOURCE) $(LIB)
+	$(COMPILER) -g -w -O -Wall -o $(EXECUTABLE) $(SOURCE) > build.log 2>&1
 	@echo ' ******* end of linking ****** '
 
 clean:
diff --git a/genesis.def b/genesis.def
index 826e68d..369acba 100644
--- a/genesis.def
+++ b/genesis.def
@@ -12,11 +12,11 @@ c
      +			platf  ='Unix',              !platform
      +          original = 1 ,               !indicator for original filetype
      +          f_sdds = 2 ,                 !indicator for sdds filetype
-     +          npmax  = 1000001,             !# of particles
-     +          nzmax  = 10000,              !# of integration steps
-     +          nsmax  = 150000,              !# of slices
+     +          npmax  = 10001,             !# of particles
+     +          nzmax  = 1000,              !# of integration steps
+     +          nsmax  = 15000,              !# of slices
      +          nhmax  = 5,                  !maximum of harmonics
-     +          ndmax  = 1250000,            !maximum of particle in imported distribution
+     +          ndmax  = 12500,            !maximum of particle in imported distribution
      +          keepdist = 1,                !<> 0 keeps distribution in memory
      +          eev    = 510999.06d0,        !energy units (mc^2) in ev
      +          vacimp = 376.73d0,           !vacuum impedence in ohms
