diff --git win32ctypes/tests/test_win32api.py win32ctypes/tests/test_win32api.py
index b6443bf..7728e6a 100644
--- a/win32ctypes/tests/test_win32api.py
+++ b/win32ctypes/tests/test_win32api.py
@@ -10,7 +10,7 @@ import sys
 import unittest

 import win32ctypes.win32api
-import win32api
+from win32ctypes import win32api

 from win32ctypes.pywintypes import error
 from win32ctypes.tests import compat
diff --git win32ctypes/tests/test_win32cred.py win32ctypes/tests/test_win32cred.py
index fe65d5d..0a8c880 100644
--- a/win32ctypes/tests/test_win32cred.py
+++ b/win32ctypes/tests/test_win32cred.py
@@ -10,7 +10,7 @@ from __future__ import absolute_import

 import unittest

-import win32cred
+from win32ctypes import win32cred

 from win32ctypes._winerrors import ERROR_NOT_FOUND
 from win32ctypes.pywintypes import error
