diff --git a/differential_privacy/algorithms/BUILD b/differential_privacy/algorithms/BUILD
index f2c93f7..07cccd1 100644
--- a/differential_privacy/algorithms/BUILD
+++ b/differential_privacy/algorithms/BUILD
@@ -50,6 +50,7 @@ cc_test(
     timeout = "eternal",
     srcs = ["algorithm-stochastic-dp_test.cc"],
     copts = ["-Wno-sign-compare"],
+    linkopts = ["-lrt"],
     shard_count = 16,
     deps = [
         ":approx-bounds",
