diff --git a/python/sentencepiece.pc.in b/python/sentencepiece.pc.in
new file mode 100644
index 0000000..ac7fef6
--- /dev/null
+++ b/python/sentencepiece.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: @PROJECT_NAME@
+Description: Unsupervised text tokenizer and detokenizer for Neural Network-based text generation.
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lsentencepiece -lsentencepiece_train @libprotobuf_lite@ @pkgconfiglibs@
+Cflags: -I${includedir} @pkgconfigcflags@
