diff --git a/sentencepiece.pc.in b/python/sentencepiece.pc.in
similarity index 99%
rename from sentencepiece.pc.in
rename to python/sentencepiece.pc.in
index ac7fef6..8a6d43e 100644
--- a/sentencepiece.pc.in
+++ b/python/sentencepiece.pc.in
@@ -8,3 +8,4 @@ Description: Unsupervised text tokenizer and detokenizer for Neural Network-base
 Version: @PROJECT_VERSION@
 Libs: -L${libdir} -lsentencepiece -lsentencepiece_train @libprotobuf_lite@ @pkgconfiglibs@
 Cflags: -I${includedir} @pkgconfigcflags@
+ 
