--- Makefile.mak
+++ Makefile.mak
@@ -336,16 +336,5 @@
 
-# Visual Studio redistributable files
-MSVCR = Microsoft.VC$(MSVSVER).CRT
-!if $(MSVSVER) <= 90
-# VS <= 2008 needs a manifest too
-bincrt = $(bindir)\msvcr$(MSVSVER).dll $(bindir)\$(MSVCR).manifest
-mexcrt = $(mexdir)\msvcr$(MSVSVER).dll $(mexdir)\$(MSVCR).manifest
-!else
-bincrt = $(bindir)\msvcr$(MSVSVER).dll
-mexcrt = $(mexdir)\msvcr$(MSVSVER).dll
-!endif
-
 !ifdef MATLABROOT
 all: $(bindir) $(objdir) $(mexdir) \
-     $(bincrt) $(bindir)\vl.lib $(bindir)\vl.dll \
+     $(bindir)\vl.lib $(bindir)\vl.dll \
      $(cmdexe) \
@@ -354,3 +343,3 @@
 all: $(bindir) $(objdir) \
-     $(bincrt) $(bindir)\vl.lib $(bindir)\vl.dll \
+     $(bindir)\vl.lib $(bindir)\vl.dll \
      $(cmdexe)
@@ -403,3 +392,2 @@
 	@echo ** MSVCR_PATH  = $(MSVCR_PATH)
-	@echo ** bincrt      = $(bincrt)
 	@echo ** mexcrt      = $(mexcrt)
@@ -451,9 +439,2 @@
 	@lib $(**) /OUT:"$(@)" /NOLOGO
-
-# redistributable: msvcr__.dll => bin/win{32,64}/msvcr__.dll
-$(bindir)\$(MSVCR).manifest : "$(MSVCR_PATH)\$(MSVCR).manifest"
-        copy $(**) "$(@)"
-
-$(bindir)\msvcr$(MSVSVER).dll: "$(MSVCR_PATH)\msvcr$(MSVSVER).dll"
-        copy $(**) "$(@)"
 
