diff --git home/pts08/Downloads/vlfeat-0.9.20/make/bin.mak home/pts08/Downloads/vlfeat-0.9.20/make/bin2.mak
index 81c878b..015facc 100644
--- make/bin.mak
+++ make/bin.mak
@@ -52,9 +52,15 @@ endif
 # At run time this causes the loader to search for a local copy of the
 # library for any binary which is linked against it. The install_name
 # can be modified later by install_name_tool.
-
 bin_src := $(wildcard $(VLDIR)/src/*.c)
 bin_tgt := $(addprefix $(BINDIR)/, $(patsubst %.c,%,$(notdir $(bin_src))))
+
+ifeq "$(NO_TESTS)" "yes"
+FILTER_OUT = $(foreach v,$(2),$(if $(findstring $(1),$(v)),,$(v)))
+bin_src := $(call FILTER_OUT,test,$(bin_src))
+bin_tgt := $(call FILTER_OUT,test,$(bin_tgt))
+endif
+
 bin_dep := $(addsuffix .d, $(bin_tgt))
 
 deps += $(bin_dep)
@@ -91,8 +97,8 @@ bin-distclean:
 
 bin-info:
 	$(call echo-title,Command line utilities)
-	$(call dump-var,bin_src)
-	$(call dump-var,bin_tgt)
+	$(call echo-var,bin_src)
+	$(call echo-var,bin_tgt)
 	$(call dump-var,bin_dep)
 	$(call echo-var,BIN_CFLAGS)
 	$(call echo-var,BIN_LDFLAGS)
