diff -urN qtwebkit.orig/Source/autotools/SetupLibtool.m4 qtwebkit/Source/autotools/SetupLibtool.m4
--- qtwebkit.orig/Source/autotools/SetupLibtool.m4	2016-07-02 02:13:06.198898198 +0100
+++ qtwebkit/Source/autotools/SetupLibtool.m4	2016-07-02 02:13:42.972232490 +0100
@@ -8,6 +8,7 @@
 if test -z "$AR_FLAGS"; then
     AR_FLAGS="cruT"
 fi
+AC_SUBST([AR_FLAGS])
 
 DOLT
 AC_DISABLE_STATIC
diff -urN qtwebkit.orig/Source/WebKit2/GNUmakefile.am qtwebkit/Source/WebKit2/GNUmakefile.am
--- qtwebkit.orig/Source/WebKit2/GNUmakefile.am	2016-07-02 02:13:06.148898197 +0100
+++ qtwebkit/Source/WebKit2/GNUmakefile.am	2016-07-02 02:13:42.972232490 +0100
@@ -588,7 +588,9 @@
 WebCoreLayerGtk2.a: $(webcore_layer_gtk2_deps)
 	$(AM_V_GEN)
 	$(AM_V_at)$(shell rm -f $@)
-	$(AM_V_at)$(foreach archive, $(webcore_layer_gtk2_archives), $(shell ar t $(archive) | xargs -n50 ar cruT $@))
+	$(AM_V_at)$(shell find . -name "*.o" > objects_list)
+	$(AM_V_at)$(foreach archive, $(webcore_layer_gtk2_archives), $(shell $(AR) t $(archive) | xargs -n1 basename | xargs -I obj_file grep -F obj_file objects_list | xargs -n50 $(AR) $(AR_FLAGS) $@))
+	$(AM_V_at)$(shell rm -f objects_list)
 
 DISTCLEANFILES += \
 	$(top_builddir)/WebCoreLayerGtk2.a
diff -urN qtwebkit.orig/Tools/qmake/mkspecs/features/default_post.prf qtwebkit/Tools/qmake/mkspecs/features/default_post.prf
--- qtwebkit.orig/Tools/qmake/mkspecs/features/default_post.prf	2016-07-02 02:13:06.972231552 +0100
+++ qtwebkit/Tools/qmake/mkspecs/features/default_post.prf	2016-07-02 02:19:56.392242212 +0100
@@ -276,7 +276,7 @@
         # Replace the hardcoded archiver command line
         # options to use GNU ar's thin archive format.
         AR_COMMAND = $$split(QMAKE_AR, " ")
-        QMAKE_AR = $$member(AR_COMMAND, 0) cruT
+        QMAKE_AR = $$member(AR_COMMAND, 0) cru
     }
 
     isEmpty(DESTDIR): DESTDIR = $$targetSubDir()
diff -urN qtwebkit.orig/Tools/TestWebKitAPI/GNUmakefile.am qtwebkit/Tools/TestWebKitAPI/GNUmakefile.am
--- qtwebkit.orig/Tools/TestWebKitAPI/GNUmakefile.am	2016-07-02 02:13:06.995564885 +0100
+++ qtwebkit/Tools/TestWebKitAPI/GNUmakefile.am	2016-07-02 02:13:42.972232490 +0100
@@ -122,7 +122,9 @@
 WebCoreLayer.a: $(webcore_layer_deps)
 	$(AM_V_GEN)
 	$(AM_V_at)$(shell rm -f $@)
-	$(AM_V_at)$(foreach archive, $(webcore_layer_archives), $(shell ar t $(archive) | xargs -n50 ar cruT $@))
+	$(AM_V_at)$(shell find . -name "*.o" > objects_list)
+	$(AM_V_at)$(foreach archive, $(webcore_layer_archives), $(shell $(AR) t $(archive) | xargs -n1 basename | xargs -I obj_file grep -F obj_file objects_list | xargs -n50 $(AR) $(AR_FLAGS) $@))
+	$(AM_V_at)$(shell rm -f objects_list)
 
 DISTCLEANFILES += \
 	$(top_builddir)/WebCoreLayer.a
