--- a/qtgamepad/src/plugins/gamepads/evdev/qevdevgamepadbackend.cpp	2018-03-01 15:27:43.264976205 +0000
+++ b/qtgamepad/src/plugins/gamepads/evdev/qevdevgamepadbackend.cpp	2018-03-01 15:38:48.145978825 +0000
@@ -46,6 +46,19 @@
 
 #include <cmath>
 
+#ifndef BTN_TRIGGER_HAPPY1
+#define BTN_TRIGGER_HAPPY1 0x2c0
+#endif
+#ifndef BTN_TRIGGER_HAPPY2
+#define BTN_TRIGGER_HAPPY2 0x2c1
+#endif
+#ifndef BTN_TRIGGER_HAPPY3
+#define BTN_TRIGGER_HAPPY3 0x2c2
+#endif
+#ifndef BTN_TRIGGER_HAPPY4
+#define BTN_TRIGGER_HAPPY4 0x2c3
+#endif
+
 QT_BEGIN_NAMESPACE
 
 Q_LOGGING_CATEGORY(lcEGB, "qt.gamepad")
