--- qtwebkit.orig/Source/JavaScriptCore/offlineasm/offsets.rb	2016-09-09 14:35:32.000000000 +0100
+++ qtwebkit/Source/JavaScriptCore/offlineasm/offsets.rb	2017-08-23 22:44:05.533550474 +0100
@@ -108,7 +108,11 @@
     File.open(file, "rb") {
         | inp |
         loop {
-            byte = inp.getbyte
+            if RUBY_VERSION >= '1.8.7'
+              byte = inp.getbyte
+            else
+              byte = inp.getc
+            end
             break unless byte
             fileBytes << byte
         }
