From a94c447505c1feea57dfc37741d2dfbdf56c6790 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 10 Dec 2018 15:05:47 -0600
Subject: [PATCH] qtwebengine: use CONDA_PREFIX/include for system_libjpeg

---
 chromium/third_party/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qtwebengine/src/3rdparty/chromium/third_party/BUILD.gn b/qtwebengine/src/3rdparty/chromium/third_party/BUILD.gn
index f0a8651e4e..9e21b2ae15 100644
--- a/qtwebengine/src/3rdparty/chromium/third_party/BUILD.gn
+++ b/qtwebengine/src/3rdparty/chromium/third_party/BUILD.gn
@@ -9,6 +9,7 @@ import("//third_party/libjpeg.gni")
 assert(!is_ios, "This is not used on iOS, don't drag it in unintentionally")
 
 config("system_libjpeg_config") {
+  include_dirs = [ getenv("CONDA_PREFIX")+"/include" ]
   libs = [ "jpeg" ]
   defines = [ "USE_SYSTEM_LIBJPEG" ]
 }
-- 
2.19.1

