From a2f0e848769735af9942cbb3b9ac766f75a3399e Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 23 Aug 2019 02:15:17 +0200
Subject: [PATCH 7/7] qtbase: Disable kTLSProtocol13 for macOS <= 10.12

---
 src/network/ssl/qsslsocket_mac.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git /a/qtbase/src/network/ssl/qsslsocket_mac.cpp /b/qtbase/src/network/ssl/qsslsocket_mac.cpp
new file mode 100644
index 7c5f4310f8..ba3705a067 100644
--- /a/qtbase/src/network/ssl/qsslsocket_mac.cpp
+++ /b/qtbase/src/network/ssl/qsslsocket_mac.cpp
@@ -506,8 +506,10 @@ QSsl::SslProtocol QSslSocketBackendPrivate::sessionProtocol() const
         return QSsl::TlsV1_1;
     case kTLSProtocol12:
         return QSsl::TlsV1_2;
+#if QT_DARWIN_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_13_4, __IPHONE_11_0, __TVOS_11_0, __WATCHOS_4_0)
     case kTLSProtocol13:
         return QSsl::TlsV1_3;
+#endif
     default:
         return QSsl::UnknownProtocol;
     }
-- 
2.20.1

