--- work/qtwebengine/src/3rdparty/gn/build/gen.py.orig	2019-01-29 21:38:44.139791640 +0000
+++ work/qtwebengine/src/3rdparty/gn/build/gen.py	2019-01-29 21:39:17.241393907 +0000
@@ -380,6 +380,7 @@
           # These are needed by libc++.
           '-ldl',
           '-lpthread',
+          '-lrt',
       ])
     elif platform.is_darwin():
       min_mac_version_flag = '-mmacosx-version-min=10.9'
