From 9d5ffb252b7d8bf089242c0bfebc5358669f3abe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Edgar=20Andr=C3=A9s=20Margffoy=20Tuay?= <andfoy@gmail.com>
Date: Wed, 17 Nov 2021 20:10:10 -0500
Subject: [PATCH] Make repc arm64-compatible

---
 qtremoteobjects/tools/repc/repc.pro | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/qtremoteobjects/tools/repc/repc.pro b/qtremoteobjects/tools/repc/repc.pro
index 1b47e5a..5459f88 100644
--- a/qtremoteobjects/tools/repc/repc.pro
+++ b/qtremoteobjects/tools/repc/repc.pro
@@ -8,6 +8,10 @@ DEFINES += QT_NO_CAST_TO_ASCII QT_NO_CAST_FROM_ASCII QT_NO_CAST_FROM_BYTEARRAY Q
 DEFINES += RO_INSTALL_HEADERS=$$shell_quote(\"$$clean_path($$[QT_INSTALL_HEADERS]/QtRemoteObjects)\")
 msvc: QMAKE_CXXFLAGS += /wd4129

+QMAKE_CXXFLAGS += -arch x86_64 -arch arm64
+QMAKE_CFLAGS += -arch x86_64 -arch arm64
+QMAKE_LFLAGS += -arch x86_64 -arch arm64
+
 CONFIG += qlalr
 QLALRSOURCES += $$QTRO_SOURCE_TREE/src/repparser/parser.g
 INCLUDEPATH += $$QTRO_SOURCE_TREE/src/repparser
--
2.25.1

