diff --git a/src/gui/Qt5GuiConfigExtras.cmake.in b/src/gui/Qt5GuiConfigExtras.cmake.in
index 84dbbfeb..ade39e30 100644
--- a/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in
+++ b/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in
@@ -171,11 +171,11 @@ endmacro()
 
 
 !!IF !isEmpty(CMAKE_EGL_LIBS)
-_qt5gui_find_extra_libs(EGL \"$$CMAKE_EGL_LIBS\" \"$$CMAKE_EGL_LIBDIR\" \"$$CMAKE_EGL_INCDIRS\")
+_qt5gui_find_extra_libs(EGL \"EGL\" \"\" \"\")
 !!ENDIF
 
 !!IF !isEmpty(CMAKE_OPENGL_LIBS)
-_qt5gui_find_extra_libs(OPENGL \"$$CMAKE_OPENGL_LIBS\" \"$$CMAKE_OPENGL_LIBDIR\" \"$$CMAKE_OPENGL_INCDIRS\")
+_qt5gui_find_extra_libs(OPENGL \"GL\" \"\" \"\")
 
 !!ENDIF
 
