--- ./work/qtwebengine/src/core/config/common.pri.orig  2019-01-30 10:06:04.776357953 +0000
+++ ./work/qtwebengine/src/core/config/common.pri	2019-01-30 09:46:04.170282106 +0000
@@ -2,8 +2,12 @@
 include($$QTWEBENGINE_OUT_ROOT/src/core/qtwebenginecore-config.pri)
 QT_FOR_CONFIG += webenginecore

+# This is horrible, sorry
+CONDA_PREFIX=$$dirname(QMAKE_INCDIR_WEBENGINE-ZLIB)
+
 gn_args += \
     use_qt=true \
+    conda_prefix=\"$$CONDA_PREFIX\" \
     is_component_build=false \
     is_shared=true \
     enable_message_center=false \

--- ./work/qtwebengine/src/3rdparty/chromium/build/config/features.gni.orig	2019-01-31 08:34:51.840450034 -0600
+++ ./work/qtwebengine/src/3rdparty/chromium/build/config/features.gni	2019-01-31 08:35:29.093030835 -0600
@@ -53,6 +53,8 @@
   use_gio = is_linux && !is_chromeos && !is_chromecast
 
   use_qt = false
+#
+#  conda_prefix = ""
 }
 #
 # =============================================
--- ./work/qtwebengine/src/3rdparty/chromium/.gn.orig	2019-01-31 09:52:02.961972105 -0600
+++ ./work/qtwebengine/src/3rdparty/chromium/.gn	2019-01-31 09:52:26.648977260 -0600
@@ -67,6 +67,8 @@
   # Changes some setup for the Crashpad build to set them to build against
   # Chromium's zlib, base, etc.
   crashpad_dependencies = "chromium"
+#
+#  conda_prefix = ""
 }
 
 # These are the targets to check headers for by default. The files in targets
--- ./work/qtwebengine/src/3rdparty/chromium/build/config/BUILDCONFIG.gn.orig	2019-01-31 10:55:29.239491859 -0600
+++ ./work/qtwebengine/src/3rdparty/chromium/build/config/BUILDCONFIG.gn	2019-01-31 10:57:28.775564393 -0600
@@ -172,6 +172,10 @@
 
 assert(!(is_debug && is_official_build), "Can't do official debug builds")
 
+declare_args() {
+  conda_prefix = ""
+}
+
 # ==============================================================================
 # TOOLCHAIN SETUP
 # ==============================================================================
