--- ./work/qtwebengine/src/3rdparty/chromium/build/linux/unbundle/zlib.gn.orig	2019-01-30 00:36:42.021657473 +0000
+++ ./work/qtwebengine/src/3rdparty/chromium/build/linux/unbundle/zlib.gn	2019-01-30 00:48:00.523433133 +0000
@@ -11,6 +11,8 @@
 
 config("system_zlib") {
   defines = [ "USE_SYSTEM_ZLIB=1" ]
+  include_dirs = [ conda_prefix+"/include" ]
+  lib_dirs = [ conda_prefix+"/lib" ]
 }
 
 source_set("zlib") {
--- ./work/qtwebengine/src/3rdparty/chromium/third_party/pdfium/third_party/BUILD.gn.orig	2019-01-30 15:01:16.272785985 +0000
+++ ./work/qtwebengine/src/3rdparty/chromium/third_party/pdfium/third_party/BUILD.gn	2019-01-30 15:01:10.137017538 +0000
 group("third_party") {
   deps = [
     ":bigint",
@@ -328,6 +332,8 @@
 config("system_zlib_config") {
   libs = [ "z" ]
   defines = [ "USE_SYSTEM_ZLIB" ]
+  include_dirs = [ conda_prefix+"/include" ]
+  lib_dirs = [ conda_prefix+"/lib" ]
 }
 
 group("zlib") {
