From 4c48c791b36300f305b223b5ae8731a403e83c65 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Edgar=20Andr=C3=A9s=20Margffoy=20Tuay?= <andfoy@gmail.com>
Date: Wed, 17 Nov 2021 20:55:05 -0500
Subject: [PATCH] Enable arm64 support for qscxmlc

---
 qtscxml/tools/qscxmlc/qscxmlc.pro | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/qtscxml/tools/qscxmlc/qscxmlc.pro b/qtscxml/tools/qscxmlc/qscxmlc.pro
index 7620d3d..fd0f87b 100644
--- a/qtscxml/tools/qscxmlc/qscxmlc.pro
+++ b/qtscxml/tools/qscxmlc/qscxmlc.pro
@@ -7,6 +7,10 @@ include(qscxmlc.pri)
 TARGET = qscxmlc
 CONFIG += console c++11

+QMAKE_CXXFLAGS += -arch x86_64 -arch arm64
+QMAKE_CFLAGS += -arch x86_64 -arch arm64
+QMAKE_LFLAGS += -arch x86_64 -arch arm64
+
 SOURCES += \
     main.cpp

--
2.25.1

