--- ./work/qtwebengine/mkspecs/features/configure.prf.orig	2019-01-31 03:41:39.286908403 -0600
+++ ./work/qtwebengine/mkspecs/features/configure.prf	2019-01-31 03:42:55.846157188 -0600
@@ -71,10 +71,11 @@
             return(false)
         }
 
-        !qtConfig(webengine-system-glibc) {
-            skipBuild("A suitable version of libc could not be found. See: https://sourceware.org/bugzilla/show_bug.cgi?id=14898")
-            return(false)
-        }
+# We may want to consider patching our glibc, but either way we need to disable this version number test.
+#        !qtConfig(webengine-system-glibc) {
+#            skipBuild("A suitable version of libc could not be found. See: https://sourceware.org/bugzilla/show_bug.cgi?id=14898")
+#            return(false)
+#        }
 
         QT_FOR_CONFIG += gui-private
 
