--- qtwebengine/src/3rdparty/chromium/third_party/libevdev/src/src/libevdev.c.bak	2019-07-24 15:16:31.791915512 -0500
+++ qtwebengine/src/3rdparty/chromium/third_party/libevdev/src/src/libevdev.c	2019-07-24 15:17:55.956287549 -0500
@@ -23,6 +23,10 @@
 #define EVIOCGMTSLOTS(len)  _IOC(_IOC_READ, 'E', 0x0a, len)
 #endif
 
+#ifndef EVIOCGPROP
+#define EVIOCGPROP(len)    _IOC(_IOC_READ, 'E', 0x09, len)
+#endif
+
 /* Set clockid to be used for timestamps */
 #ifndef EVIOCSCLOCKID
 #define EVIOCSCLOCKID  _IOW('E', 0xa0, int)
--- qtwebengine/src/3rdparty/chromium/ui/events/ozone/evdev/event_device_info.cc.bak	2019-07-24 15:16:22.763880860 -0500
+++ qtwebengine/src/3rdparty/chromium/ui/events/ozone/evdev/event_device_info.cc	2019-07-24 15:17:37.968200958 -0500
@@ -16,6 +16,10 @@
 #define EVIOCGMTSLOTS(len) _IOC(_IOC_READ, 'E', 0x0a, len)
 #endif
 
+#if !defined(EVIOCGPROP)
+#define EVIOCGPROP(len)    _IOC(_IOC_READ, 'E', 0x09, len)
+#endif
+
 namespace ui {
 
 namespace {
