--- qtwebengine/src/3rdparty/chromium/net/socket/udp_socket_posix.h.bak 2019-07-24 21:38:18.948273017 +0000
+++ qtwebengine/src/3rdparty/chromium/net/socket/udp_socket_posix.h     2019-07-24 21:38:31.776394748 +0000
@@ -34,7 +34,11 @@
 #if defined(__ANDROID__) && defined(__aarch64__)
 #define HAVE_SENDMMSG 1
 #elif defined(OS_LINUX)
+#if defined(__GLIBC__) && defined(__GLIBC_MINOR__) && (__GLIBC__ == 2) && (__GLIBC_MINOR__ < 14)
+#define HAVE_SENDMMSG 0
+#else
 #define HAVE_SENDMMSG 1
+#endif
 #else
 #define HAVE_SENDMMSG 0
 #endif

