--- a/qtbase/mkspecs/common/gcc-base.conf	2018-03-14 22:26:23.716612978 +0000
+++ b/qtbase/mkspecs/common/gcc-base.conf	2018-03-14 22:26:39.012700839 +0000
@@ -34,7 +34,7 @@
 QMAKE_CFLAGS_OPTIMIZE      = -O2
 QMAKE_CFLAGS_OPTIMIZE_FULL = -O3
 QMAKE_CFLAGS_OPTIMIZE_DEBUG = -Og
-QMAKE_CFLAGS_OPTIMIZE_SIZE = -Os
+QMAKE_CFLAGS_OPTIMIZE_SIZE = -Os
 
 !equals(QMAKE_HOST.os, Windows): QMAKE_CFLAGS += -pipe
 QMAKE_CFLAGS_DEPS          += -M
