--- a/qtbase/src/corelib/kernel/qcore_mac_objc.mm	2019-06-28 13:26:43.000000000 +0300
+++ b/qtbase/src/corelib/kernel/qcore_mac_objc.mm	2019-06-28 13:37:21.000000000 +0300
@@ -142,16 +142,20 @@
 
 #ifdef QT_DEBUG
     void *poolFrame = nullptr;
+#if QT_DARWIN_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_14, __IPHONE_12_0, __TVOS_12_0, __WATCHOS_5_0)
     if (__builtin_available(macOS 10.14, iOS 12.0, tvOS 12.0, watchOS 5.0, *)) {
         void *frame;
         if (backtrace_from_fp(__builtin_frame_address(0), &frame, 1))
             poolFrame = frame;
     } else {
+#endif
         static const int maxFrames = 3;
         void *callstack[maxFrames];
         if (backtrace(callstack, maxFrames) == maxFrames)
             poolFrame = callstack[maxFrames - 1];
+#if QT_DARWIN_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_14, __IPHONE_12_0, __TVOS_12_0, __WATCHOS_5_0)
     }
+#endif
 
     if (poolFrame) {
         Dl_info info;
