From de1949561d3827a0b7347f1e85d50f64862c48cb Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Tue, 13 Oct 2020 18:51:16 +0200
Subject: [PATCH] Test: consistent rt fix

---
 tests/CMakeLists.txt | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git tests/CMakeLists.txt tests/CMakeLists.txt
index fa3a5bd..2cd7122 100644
--- tests/CMakeLists.txt
+++ tests/CMakeLists.txt
@@ -26,9 +26,16 @@ ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK -DBOOST_TEST_MAIN)
 # This macro will create a binary from `NAME.cc', link it against
 # Boost and add it to the test suite.
 #
+
+find_library(RT_LIB rt)
+
 MACRO(ADD_TESTCASE NAME)
   ADD_UNIT_TEST(${NAME} ${NAME}.cc)
-  TARGET_LINK_LIBRARIES(${NAME} PRIVATE Boost::unit_test_framework ${PROJECT_NAME})
+  if(RT_LIB)
+    TARGET_LINK_LIBRARIES(${NAME} PRIVATE Boost::unit_test_framework ${PROJECT_NAME} ${RT_LIB})
+  else()
+    TARGET_LINK_LIBRARIES(${NAME} PRIVATE Boost::unit_test_framework ${PROJECT_NAME})
+  endif()
 ENDMACRO(ADD_TESTCASE)
 
 FIND_PACKAGE(OpenMP)
-- 
2.17.1

