diff --git a/setup.py b/setup.py
index 4aedcbfdda..2a4107d033 100644
--- a/setup.py
+++ b/setup.py
@@ -16,10 +16,7 @@ if sys.platform == "win32":
     #    system PATH when compiling.
     # 3. The code below will moneky-patch distutils to work.
     import distutils.cygwinccompiler
-    distutils.cygwinccompiler.get_msvcr = lambda: [] 
-    # Make sure that pthreads is linked statically, otherwise we run into problems
-    # on computers where it is not installed.
-    extra_link_args = ["-static"]
+    distutils.cygwinccompiler.get_msvcr = lambda: []
 
 
 def get_c_sources(include_headers=False):
