diff --git a/recipes/quickjs/setup.py b/recipes/quickjs/setup.py
index 4aedcbfdda..552ad04d2e 100644
--- a/recipes/quickjs/setup.py
+++ b/recipes/quickjs/setup.py
@@ -7,20 +7,6 @@ from setuptools import setup, Extension
 CONFIG_VERSION = open("upstream-quickjs/VERSION").read().strip()
 extra_link_args: List[str] = []
 
-if sys.platform == "win32":
-    # To build for Windows:
-    # 1. Install MingW-W64-builds from https://mingw-w64.org/doku.php/download
-    #    It is important to change the default to 64-bit when installing if a
-    #    64-bit Python is installed in windows.
-    # 2. Put the bin/ folder inside x86_64-8.1.0-posix-seh-rt_v6-rev0 in your
-    #    system PATH when compiling.
-    # 3. The code below will moneky-patch distutils to work.
-    import distutils.cygwinccompiler
-    distutils.cygwinccompiler.get_msvcr = lambda: [] 
-    # Make sure that pthreads is linked statically, otherwise we run into problems
-    # on computers where it is not installed.
-    extra_link_args = ["-static"]
-
 
 def get_c_sources(include_headers=False):
     sources = [
@@ -54,7 +40,7 @@ _quickjs = Extension(
     # HACK.
     # See https://github.com/pypa/packaging-problems/issues/84.
     sources=get_c_sources(include_headers=("sdist" in sys.argv)),
-    extra_compile_args=["-Werror=incompatible-pointer-types"],
+    extra_compile_args=[] if sys.platform == "win32" else ["-Werror=incompatible-pointer-types"],
     extra_link_args=extra_link_args)
 
 long_description = """
