--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -41,21 +41,6 @@
 
 include_directories(${CMAKE_BINARY_DIR})
 
-# Build and install the static library.
-
-add_library(libcsmith_a STATIC
-  ${CMAKE_BINARY_DIR}/config.h
-  volatile_runtime.c
-  volatile_runtime.h
-  )
-set_target_properties(libcsmith_a PROPERTIES OUTPUT_NAME "csmith")
-
-install(TARGETS libcsmith_a
-  ARCHIVE DESTINATION "${LIB_DIR}"
-  )
-
-#####
-
 # Build and install the shared library.
 
 add_library(libcsmith_so SHARED
