From 4c71969756a70d31eff7f36aa026ccfd19223168 Mon Sep 17 00:00:00 2001
From: Duncan Macleod <duncan.macleod@ligo.org>
Date: Wed, 5 May 2021 13:36:17 +0100
Subject: [PATCH] cmake: use gnuinstalldirs

---
 CMakeLists.txt | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9e44401..6658694 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,8 @@ option( BUILD_UNITTESTS "Build the scitokens-cpp unit tests" OFF )
 
 set( CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake )
 
+include(GNUInstallDirs)
+
 find_package( jwt-cpp REQUIRED )
 find_package( CURL REQUIRED )
 find_package( UUID REQUIRED )
@@ -66,14 +68,6 @@ target_link_libraries(scitokens-list-access SciTokens)
 add_executable(scitokens-create src/create.cpp)
 target_link_libraries(scitokens-create SciTokens)
 
-if (NOT DEFINED LIB_INSTALL_DIR)
-  SET(LIB_INSTALL_DIR "lib")
-endif()
-
-if (NOT DEFINED INCLUDE_INSTALL_DIR)
-  SET(INCLUDE_INSTALL_DIR "include")
-endif()
-
 if( BUILD_UNITTESTS )
 
 include(ExternalProject)
@@ -88,11 +82,11 @@ endif()
 
 install(
   TARGETS SciTokens
-  LIBRARY DESTINATION ${LIB_INSTALL_DIR} )
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} )
 
 install(
   FILES src/scitokens.h
-  DESTINATION ${INCLUDE_INSTALL_DIR}/scitokens )
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/scitokens )
 
 set_target_properties(
   SciTokens
-- 
2.30.2

