diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6658694..e2fd131 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,13 +69,6 @@ add_executable(scitokens-create src/create.cpp)
 target_link_libraries(scitokens-create SciTokens)
 
 if( BUILD_UNITTESTS )
-
-include(ExternalProject)
-ExternalProject_Add(gtest
-    PREFIX external/gtest
-    URL file://${PROJECT_SOURCE_DIR}/vendor/gtest
-    INSTALL_COMMAND :
-)
 enable_testing()
 add_subdirectory(test)
 endif()
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 9c9a21c..e9be288 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,13 +1,11 @@
 
 add_executable(scitokens-gtest main.cpp)
-add_dependencies(scitokens-gtest gtest)
-include_directories("${PROJECT_SOURCE_DIR}/vendor/gtest/googletest/include")
 
-target_link_libraries(scitokens-gtest SciTokens "${CMAKE_BINARY_DIR}/external/gtest/src/gtest-build/lib/libgtest.a" -lpthread)
+target_link_libraries(scitokens-gtest SciTokens gtest pthread)
 
 add_test(
   NAME
     unit
   COMMAND
-    ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/test/scitokens-gtest
+    ${CMAKE_BINARY_DIR}/test/scitokens-gtest
   )
