diff --git a/tools/nasm-filter.sh b/tools/nasm-filter.sh
index 5ec9ba3..b75ef5d 100755
--- a/tools/nasm-filter.sh
+++ b/tools/nasm-filter.sh
@@ -1,6 +1,6 @@
-#/bin/sh
+#/usr/bin/env bash
 
-# Filter out unnecessary options added by automake
+# Filter out unnecessary options added by automake and incompatible flags added by the conda build system.
 
 while [ -n "$*" ]; do
     case "$1" in
@@ -20,10 +20,17 @@ while [ -n "$*" ]; do
 	    options="$options $1"
 	    shift
 	    ;;
+	-isystem )  #-isystem *
+		shift
+		shift
+		;;
 	-I* | -i* )
 	    options="$options $1/"
 	    shift
 	    ;;
+  	-f*'='* ) # Long debug prefixes
+		shift
+		;;
 	-D* ) # For defines we need to remove spaces
 	    case "$1" in
 		*' '* ) ;;
@@ -44,4 +51,4 @@ while [ -n "$*" ]; do
     esac
 done
 
-nasm $options $args
+$BUILD_PREFIX/bin/nasm $options $args
