diff --git a/tools/nasm-filter.sh b/tools/nasm-filter.sh
index 5ec9ba3..d5a10c0 100755
--- a/tools/nasm-filter.sh
+++ b/tools/nasm-filter.sh
@@ -1,6 +1,6 @@
 #/bin/sh
 
-# Filter out unnecessary options added by automake
+# Filter out unnecessary options added by automake and incompatible flags added by the conda build system.
 
 while [ -n "$*" ]; do
     case "$1" in
@@ -20,10 +20,13 @@ while [ -n "$*" ]; do
 	    options="$options $1"
 	    shift
 	    ;;
-	-I* | -i* )
-	    options="$options $1/"
-	    shift
-	    ;;
+	-isystem )  #-isystem *
+		shift
+		shift
+		;;
+	-f*'='* ) # Long debug prefixes
+		shift
+		;;
 	-D* ) # For defines we need to remove spaces
 	    case "$1" in
 		*' '* ) ;;
