diff --git a/python/setup.py b/python/setup.py
index 3824844..1acf773 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -6,25 +6,6 @@ from setuptools import setup
 
 dbfile = 'xraydb.sqlite'
 
-required_modules = ('sqlalchemy', 'json', 'numpy', 'scipy')
-def check_dependencies(modules):
-    missing = []
-    for mod in modules:
-        try:
-            x = __import__(mod)
-        except ImportError:
-            missing.append(mod)
-
-    if len(missing) > 0:
-        print('The following python packages are needed to use xraydb.py:')
-        print(' ')
-        print('      %s ' % ('and '.join(missing)))
-        print(' ')
-        print(' Please install these and try again.')
-        sys.exit(1)
-
-check_dependencies(required_modules)
-
 # copy db from .. to xraydb
 shutil.copy(os.path.join('..', dbfile), os.path.join('xraydb', dbfile))
 
@@ -35,7 +16,7 @@ setup(name = 'xraydb',
       author_email = 'newville@cars.uchicago.edu',
       url          = 'http://github.com/scikit-beam/XrayDB',
       download_url = 'http://github.com/scikit-beam/XrayDB',
-      requires = ('numpy', 'sqlalchemy'),
+      requires = ('numpy', 'sqlalchemy', 'sqlalchemy', 'scipy')
       license = 'BSD',
       description = "X-ray Reference Data for the Elements",
       platforms = ('Windows', 'Linux', 'Mac OS X'),
