diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c10971..0e1ab2d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,7 +151,7 @@ endif()
 # Curl configuration
 if(CPR_FORCE_USE_SYSTEM_CURL)
     if(CPR_ENABLE_SSL)
-        find_package(CURL COMPONENTS HTTP HTTPS)
+        find_package(CURL REQUIRED)
         if(CURL_FOUND)
             message(STATUS "Curl ${CURL_VERSION_STRING} found on this system.")
             # To be able to load certificates under Windows when using OpenSSL:
