From 83bd28c4e36945f9be769e27e30852d54c687b67 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sat, 18 Jun 2022 19:51:57 -0400
Subject: [PATCH] Link to rt on Linux

---
 src/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d352152..33ef0c4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -63,6 +63,9 @@ target_compile_definitions(nvtop PRIVATE _GNU_SOURCE)
 
 target_link_libraries(nvtop
   PRIVATE ncurses m ${CMAKE_DL_LIBS})
+if(UNIX AND NOT APPLE)
+    target_link_libraries(nvtop PRIVATE rt)
+endif()
 
 install (TARGETS nvtop
   RUNTIME DESTINATION bin)
