diff --git a/src/extract_gpuinfo_amdgpu.c b/src/extract_gpuinfo_amdgpu.c
index 1b3ec4a..cf0e668 100644
--- a/src/extract_gpuinfo_amdgpu.c
+++ b/src/extract_gpuinfo_amdgpu.c
@@ -33,7 +33,14 @@
 #include <inttypes.h>
 #include <libdrm/amdgpu.h>
 #include <libdrm/amdgpu_drm.h>
-#include <linux/kcmp.h>
+// Workaround for compiling with older kernel modules
+// Without the appropriate kernel the syscall will likely just fail
+// These users likely don't even have AMD GPU
+// #include <linux/kcmp.h>
+// https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/unistd.h#L733
+#define SYS_kcmp 272
+// https://github.com/torvalds/linux/blob/master/include/uapi/linux/kcmp.h#L8
+#define KCMP_FILE 0
 #include <math.h>
 #include <stdarg.h>
 #include <stdbool.h>
