diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4cfc639..425f9ba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,23 +27,27 @@ set_default_configuration_release()
 msvc_use_static_runtime()
 
 # OpenMP check
-if(OPENMP_FOUND)
+if(OpenMP_FOUND)
   message("Got OpenMP")
   add_definitions(-DTREELITE_OPENMP_SUPPORT)
-else(OPENMP_FOUND)
+  if(OpenMP_CXX_INCLUDE_DIRS)
+    message("OpenMP_CXX_INCLUDE_DIRS ${OpenMP_CXX_INCLUDE_DIRS}")
+    include_directories(${OpenMP_CXX_INCLUDE_DIRS})
+  endif()
+else(OpenMP_FOUND)
   message("Don't have OpenMP")
   include(dmlc-core/cmake/Utils.cmake)
   dmlccore_option(USE_OPENMP "Build with OpenMP" OFF)
-endif(OPENMP_FOUND)
+endif(OpenMP_FOUND)
 
 # Compiler flags
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-if(OPENMP_FOUND)
+if(OpenMP_FOUND)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-else(OPENMP_FOUND)
+else(OpenMP_FOUND)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-endif(OPENMP_FOUND)
+endif(OpenMP_FOUND)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 if(MSVC)
   # Multithreaded compilation
diff --git a/runtime/CMakeLists.txt b/runtime/CMakeLists.txt
index 321e44f..aa2bbd2 100644
--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -13,9 +13,13 @@ if(MSVC)
   endif()
 endif()
 
-if(OPENMP_FOUND)
+if(OpenMP_FOUND)
   message("Got OpenMP")
   add_definitions(-DTREELITE_OPENMP_SUPPORT)
+  if(OpenMP_CXX_INCLUDE_DIRS)
+    message("OpenMP_CXX_INCLUDE_DIRS ${OpenMP_CXX_INCLUDE_DIRS}")
+    include_directories(${OpenMP_CXX_INCLUDE_DIRS})
+  endif()
 else()
   message("Don't have OpenMP")
 endif()
