From d7fb5e33bc3b39a349eef619d7d6cacd623abf4c Mon Sep 17 00:00:00 2001
From: Daeho Ro <40587651+daeho-ro@users.noreply.github.com>
Date: Mon, 1 Jun 2026 19:54:07 +0900
Subject: [PATCH] fix: guard SO_BINDTODEVICE

---
 src/brpc/socket.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/brpc/socket.cpp b/src/brpc/socket.cpp
index 0ca69504287ca0fcdf1b4a30ebafd29da51bf5c2..e740b6c096c141d3f9a4b42c89f6617994221f72 100644
--- a/src/brpc/socket.cpp
+++ b/src/brpc/socket.cpp
@@ -1272,12 +1272,18 @@ int Socket::Connect(const timespec* abstime,
     // We need to do async connect (to manage the timeout by ourselves).
     CHECK_EQ(0, butil::make_non_blocking(sockfd));
     if (!_device_name.empty()) {
+#ifdef SO_BINDTODEVICE
         if (setsockopt(sockfd, SOL_SOCKET, SO_BINDTODEVICE,
                        _device_name.c_str(), _device_name.size()) < 0) {
             PLOG(ERROR) << "Fail to set SO_BINDTODEVICE of fd=" << sockfd
                         << " to device_name=" << _device_name;
             return -1;
         }
+#else
+        LOG(ERROR) << "SO_BINDTODEVICE (device_name=" << _device_name
+                   << ") is not supported on this platform";
+        return -1;
+#endif
     }
     if (local_side().ip != butil::IP_ANY) {
         struct sockaddr_storage cli_addr;
