From f524410c251744b1c9a1a173e40f946d1328fd83 Mon Sep 17 00:00:00 2001
From: Tim Snyder <snyder.tim@gmail.com>
Date: Mon, 13 Mar 2023 06:06:54 +0000
Subject: [PATCH] examples/sysc/2.3/simple_async needs PTHREADS

---
 examples/sysc/2.3/simple_async/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/examples/sysc/2.3/simple_async/Makefile b/examples/sysc/2.3/simple_async/Makefile
index 1c4f59cc0..ac4e4a64e 100644
--- a/examples/sysc/2.3/simple_async/Makefile
+++ b/examples/sysc/2.3/simple_async/Makefile
@@ -4,4 +4,6 @@ include ../../../build-unix/Makefile.config
 PROJECT = simple_async
 OBJS    = main.o
 
+PTHREADS = 1
+
 include ../../../build-unix/Makefile.rules
-- 
2.39.1

