diff --git a/compile.sh b/compile.sh
index d91583d277..699b8ec773 100755
--- a/compile.sh
+++ b/compile.sh
@@ -63,7 +63,8 @@ display "."
 log "Building output/bazel"
 # We set host and target platform directly because we are building for the local
 # host.
-bazel_build "src:bazel_nojdk${EXE_EXT}" \
+#bazel_build "src:bazel_nojdk${EXE_EXT}" \
+bazel_build ${BAZEL_BUILD_OPTS} "src:bazel_nojdk${EXE_EXT}" \
   --action_env=PATH \
   --host_platform=@local_config_platform//:host \
   --platforms=@local_config_platform//:host \
