diff --git a/setup.py b/setup.py
index 41efc7f..29ffe8a 100644
--- a/setup.py
+++ b/setup.py
@@ -45,30 +45,6 @@ linux_distributions_help = {
     "centos": "wget -O gifsicle.rpm https://centos.pkgs.org/7/epel-x86_64/gifsicle-1.91-1.el7.x86_64.rpm.html\nrpm -i gifsicle.rpm"
 }
 
-if platform.system() == "Darwin":
-    subprocess.call(["brew", "install", "gifsicle"])
-elif platform.system() == "Linux":
-    if is_stdout_enabled():
-        print("Installing gifsicle on Linux requires sudo!")
-        distro = "default"
-        if os.path.exists("/etc/os-release"):
-            with open("/etc/os-release", "r") as f:
-                for line in f:
-                    if line.strip()[0:3] == "ID=":
-                        distro = line.strip().split("=")[-1].lower()
-                        break
-            print("The current system was detected to be {}".format(distro))
-            print("Please run the following command in your terminal:")
-        else:
-            print("The current distribution could not be detected")
-        print(linux_distributions_help.get(
-            distro, linux_distributions_help["default"]))
-        input("Press any key to continue with the installation of the python package.")
-elif platform.system() == "Windows":
-    if is_stdout_enabled():
-        print("Please install the current gifsickle version from the website:")
-        print("https://eternallybored.org/misc/gifsicle/")
-        input("Press any key to continue with the installation of the python package.")
 
 test_deps = [
     "pytest",
