Description: Fix several spelling errors in upstream source
Author: Regis Fernandes Gontijo <regisfg@gmail.com>
Last-Update: 2020-05-11
Index: num-utils-0.5/numgrep
===================================================================
--- numgrep	2020-05-11 02:11:32.150696932 -0300
+++ numgrep	2020-05-13 01:42:41.554225814 -0300
@@ -280,7 +280,7 @@
 
 =head1 DESCRIPTION
 
-B<numgrep> searches for different occurances of numbers through the use
+B<numgrep> searches for different occurrences of numbers through the use
 of numeric expressions.
 
 =head1 OPTIONS
@@ -309,7 +309,7 @@
          between, greater than or less than the numbers specified
          should be matched.
 
-    ,    Expression sepeartor.  The comma sepearates one complete
+    ,    Expression separator.  The comma separates one complete
          expression from another in a set enclosed by //.
 
     m<n> Multiples of <n>.  This operator, followed by a number
Index: num-utils-0.5/bound
===================================================================
--- bound	2020-05-11 03:41:41.000000000 -0300
+++ bound	2020-05-13 01:42:41.514227857 -0300
@@ -163,7 +163,7 @@
 B<numbound>
 will find boundary numbers (minimum and maximum) in files or STDIN.  By default
 it will find the upper bound in the set of numbers (the maximum number) in the
-files or on STDIN.  You can use the -l option for finding the lower bound (minumum
+files or on STDIN.  You can use the -l option for finding the lower bound (minimum
 number).
 
 =head1 OPTIONS
Index: num-utils-0.5/average
===================================================================
--- average	2020-05-11 03:41:33.000000000 -0300
+++ average	2020-05-13 01:42:41.470230104 -0300
@@ -225,7 +225,7 @@
     -i  Only return the integer portion of the final sum.
     -I  Only return the decimal portion of the final sum
 
-    -m  Find the mode (most occuring) of the list of numbers,
+    -m  Find the mode (most occurring) of the list of numbers,
         or when there's more than one mode, the first completed mode.
     -M  Find the median (middle number) of the list of numbers.
     -l  When finding the median and the count of numbers in the set is even,
Index: num-utils-0.5/Makefile
===================================================================
--- Makefile	2020-05-11 01:31:40.000000000 -0300
+++ Makefile	2020-05-13 01:53:37.832616034 -0300
@@ -32,7 +32,7 @@
 #	'/_topdir(?:\s+|\s+:\s+)(\/.*|%{_usrsrc}.*$$)/; \
 #	$$dir = $$1; $$dir =~ s|%{_usrsrc}|/usr/src|; print "$$1\n";')
 
-# Modify these as nessecary
+# Modify these as necessary
 PERL	= /usr/bin/perl
 BINDIR	= /usr/bin
 TOPDIR	= $(ROOT)/usr
Index: num-utils-0.5/GOALS
===================================================================
--- GOALS	2020-05-11 01:31:40.000000000 -0300
+++ GOALS	2020-05-13 02:09:40.307850786 -0300
@@ -19,7 +19,7 @@
 is interested in helping out in any way, please contact me at suso@suso.org.
 Any contributions are welcome.
 
-  I've put an 'X' at the begining of the line for features that have been
+  I've put an 'X' at the beginning of the line for features that have been
 completeled.
 
 -= All numeric utilities =-
@@ -110,7 +110,7 @@
     i    -- increment operator
     f    -- factor operator
     m    -- multiple operator
-    ,    -- expression seperation separator
+    ,    -- expression separation separator
 
     []   -- character grouping
     +    -- quantifier  (1 or more)
@@ -183,15 +183,15 @@
            line.   
 X  -c   -- Treat each line as a set of columns separated by white space, or a
            string if the -s option is used.  Sum up the values in each column
-           and print out the result of each column seperated by the seperation
+           and print out the result of each column separated by the separation
            character.  This is shown above in the advanced examples section.
 X  -s <string>  [for columns]
         -- Use <string> as the separator between each column.  This is allowed
            to be more than one character and possibly even a number.
 X  -r   -- Treat each line (by default) as a row of numbers to sum up.  The
-           results of the sums of each row will be printed on seperate lines.
+           results of the sums of each row will be printed on separated lines.
    -s <string>  [for rows] 
-        -- When used with the -r flag, this will specify the seperator for
+        -- When used with the -r flag, this will specify the separator for
            rows.  By default it is the new line character.  It could be a
            character, set of characters or even a number.
 
@@ -216,9 +216,9 @@
            
            $ numsum -c -x 10 access_log  
 
-           To get the total bytes transfered in an access_log.
+           To get the total bytes transferred in an access_log.
 
-           Maybe this could also be able to handle comma seperated values, so
+           Maybe this could also be able to handle comma separated values, so
            1,5,10 would sum up the 1st, 5th and 10th columns.
 
 
@@ -260,7 +260,7 @@
 
       numgrep  /f1024m12,f2333m9/ data.txt
 
-   o seach for numbers that are in the set 1, 4, 7, 10, 13 and 16
+   o search for numbers that are in the set 1, 4, 7, 10, 13 and 16
    
       numgrep /1..16i3/ data.txt
 
@@ -330,7 +330,7 @@
            the most frequently occuring value in the set.
 
 X  -M   -- Print out the median of the set of numbers entered.  The median is
-           the middle value all all numbers encountered.  So if the numbers
+           the middle value all numbers encountered.  So if the numbers
            88, 12, 2, 1, 9, 100 and 1000 are encountered, the median of that
            set is 12.  Illustrated:
 
@@ -342,16 +342,16 @@
            each line.   
    -c   -- Treat each line as a set of columns separated by white space, or a
            string if the -s option is used.  Average the values in each column
-           and print out the result of each column seperated by the seperation
+           and print out the result of each column separated by the separation
            character.  This is shown above in the advanced examples section.
    -s <string>  [for columns]
         -- Use <string> as the separator between each column.  This is allowed
            to be more than one character and possibly even a number.
    -r   -- Treat each line (by default) as a row of numbers to sum up.  The
-           results of the averages of each row will be printed on seperate
+           results of the averages of each row will be printed on separate
            lines.
    -s <string>  [for rows]
-        -- When used with the -r flag, this will specify the seperator for
+        -- When used with the -r flag, this will specify the separator for
            rows.  By default it is the new line character.  It could be a
            character, set of characters or even a number.
 
@@ -429,7 +429,7 @@
 X  -e <set>
       -- Exclude the numbers in <set> from the output.  This is so that if
          you want to do a complex range without including certain numbers.
-         <set> is a list of numbers seperated by a ','.
+         <set> is a list of numbers separated by a ','.
 
 -- numrandom --
 
@@ -450,8 +450,8 @@
 
  Usage options  (in addition to the standard options)
 
-   -n <n> -- Generate <n> random numbers seperated by a space.
-   -s <c> -- Use <c> as the seperation character.
+   -n <n> -- Generate <n> random numbers separated by a space.
+   -s <c> -- Use <c> as the separation character.
    -N     -- Shortcut for using a newline separation character.
 
 
