Description: calling with --help option locks stdin
Author: Regis Fernandes Gontijo <regisfg@gmail.com>
Last-Update: 2020-05-15
Index: num-utils-0.5/average
===================================================================
--- a/average	2020-05-15 15:37:53.473463799 -0300
+++ b/average	2020-05-15 20:10:31.584804435 -0300
@@ -27,6 +27,7 @@
 use Getopt::Std;
 use strict;
 use vars qw/ %opts $verbose /;
+$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 getopts('dhiIlmMqV', \%opts);
 
@@ -123,6 +124,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub process_filehandle {
     my $filehandle = shift;
     my $number_array_ref = shift;
Index: num-utils-0.5/bound
===================================================================
--- a/bound	2020-05-15 15:37:53.489462794 -0300
+++ b/bound	2020-05-15 20:10:31.592804051 -0300
@@ -27,6 +27,7 @@
 use Getopt::Std;
 use strict;
 use vars qw/ %opts $verbose /;
+$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 my ($file, @number_array);
 
@@ -104,6 +105,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub process_filehandle {
     my $filehandle = shift;
     my $number_array_ref = shift;
Index: num-utils-0.5/interval
===================================================================
--- a/interval	2020-05-15 15:37:53.497462292 -0300
+++ b/interval	2020-05-15 20:10:31.600803666 -0300
@@ -29,6 +29,7 @@
 use Getopt::Std;
 use strict;
 use vars qw/ %opts $verbose /;
+$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 getopts('hdVq', \%opts);
 
@@ -105,6 +106,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub process_filehandle {
     my $filehandle = shift;
     my $number_array_ref = shift;
Index: num-utils-0.5/normalize
===================================================================
--- a/normalize	2020-05-15 15:37:53.505461789 -0300
+++ b/normalize	2020-05-15 20:10:31.604803474 -0300
@@ -27,6 +27,7 @@
 use Getopt::Std;
 use strict;
 use vars qw/ %opts $verbose /;
++$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 my ($file, @number_array, @output_array, $number);
 
@@ -103,6 +104,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub process_filehandle {
     my $filehandle = shift;
     my $number_array_ref = shift;
Index: num-utils-0.5/numgrep
===================================================================
--- a/numgrep	2020-05-15 15:37:53.509461538 -0300
+++ b/numgrep	2020-05-15 20:10:31.616802896 -0300
@@ -28,6 +28,7 @@
 use Getopt::Std;
 use strict;
 use vars qw/ %opts $verbose /;
++$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 $| = 1;
 
@@ -120,6 +121,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub process_filehandle {
     my $filehandle = shift;
     my $returnvalue;
Index: num-utils-0.5/numprocess
===================================================================
--- a/numprocess	2020-05-15 15:37:53.517461035 -0300
+++ b/numprocess	2020-05-15 20:10:31.620802704 -0300
@@ -27,6 +27,7 @@
 use Getopt::Std;
 use strict;
 use vars qw/ %opts $verbose /;
++$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 my ($file);
 
@@ -117,6 +118,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub process_filehandle {
     my $filehandle = shift;
 
Index: num-utils-0.5/numsum
===================================================================
--- a/numsum	2020-05-15 15:37:53.525460532 -0300
+++ b/numsum	2020-05-15 20:10:31.628802319 -0300
@@ -29,6 +29,7 @@
 #use strict;
 use vars qw/ %opts /;
 use Scalar::Util qw/looks_like_number/;
++$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 my ($file, $finalsum, @number_array, @columns_to_print, @column_output, $output_line, $string_seperator, $verbose);
 
@@ -194,6 +195,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub decimals_friendly_sum
 {
         # try summing 1.99 and -1.90 to understand why
Index: num-utils-0.5/random
===================================================================
--- a/random	2020-05-15 15:37:53.529460282 -0300
+++ b/random	2020-05-15 20:10:31.636801935 -0300
@@ -27,6 +27,7 @@
 use Getopt::Std;
 use strict;
 use vars qw/ %opts $verbose /;
++$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 my ($expression, @random_array, $number);
 
@@ -144,6 +145,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub process_filehandle {
     my $filehandle = shift;
     my $number_array_ref = shift;
Index: num-utils-0.5/range
===================================================================
--- a/range	2020-05-15 15:37:53.537459779 -0300
+++ b/range	2020-05-15 20:10:31.644801550 -0300
@@ -27,6 +27,7 @@
 use Getopt::Std;
 use strict;
 use vars qw/ %opts $verbose /;
++$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 my ($expression, @output_array, $number);
 
@@ -186,7 +187,9 @@
 EOF
 }
 
-
+sub HELP_MESSAGE{
+    &help();
+}
 
 # Lay down some of that perl pod action.
 =pod
Index: num-utils-0.5/round
===================================================================
--- a/round	2020-05-15 15:37:53.545459276 -0300
+++ b/round	2020-05-15 20:10:31.652801165 -0300
@@ -28,6 +28,7 @@
 use strict;
 use vars qw/ %opts $verbose /;
 use POSIX;  # for rounding functions.
++$Getopt::Std::STANDARD_HELP_VERSION = 1;
 
 getopts('cfn:hdV', \%opts);
 
@@ -103,6 +104,10 @@
 EOF
 }
 
+sub HELP_MESSAGE{
+    &help();
+}
+
 sub process_filehandle {
     my $filehandle = shift;
     my $number_array_ref = shift;
