Description: Fix upstream URI in manpages
Author: Regis Fernandes Gontijo
Last-Update: 2020-05-15
Index: num-utils-0.5/average
===================================================================
--- average	2020-05-15 20:10:31.584804435 -0300
+++ average	2020-05-15 20:36:41.138070923 -0300
@@ -260,7 +260,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/bound
===================================================================
--- bound	2020-05-15 20:10:31.592804051 -0300
+++ bound	2020-05-15 20:37:44.652737961 -0300
@@ -203,7 +203,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/interval
===================================================================
--- interval	2020-05-15 20:10:31.600803666 -0300
+++ interval	2020-05-15 20:38:17.546039206 -0300
@@ -177,7 +177,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/normalize
===================================================================
--- normalize	2020-05-15 20:10:31.604803474 -0300
+++ normalize	2020-05-15 20:39:13.641526725 -0300
@@ -194,7 +194,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/numgrep
===================================================================
--- numgrep	2020-05-15 20:10:31.616802896 -0300
+++ numgrep	2020-05-15 20:40:02.421686957 -0300
@@ -392,7 +392,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/numprocess
===================================================================
--- numprocess	2020-05-15 20:10:31.620802704 -0300
+++ numprocess	2020-05-15 20:40:47.054235966 -0300
@@ -267,7 +267,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/numsum
===================================================================
--- numsum	2020-05-15 20:10:31.628802319 -0300
+++ numsum	2020-05-15 20:41:22.691519764 -0300
@@ -395,7 +395,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/random
===================================================================
--- random	2020-05-15 20:10:31.636801935 -0300
+++ random	2020-05-15 20:42:05.344311173 -0300
@@ -245,7 +245,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/range
===================================================================
--- range	2020-05-15 20:10:31.644801550 -0300
+++ range	2020-05-15 20:42:54.400673363 -0300
@@ -291,7 +291,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
Index: num-utils-0.5/round
===================================================================
--- round	2020-05-15 20:10:31.652801165 -0300
+++ round	2020-05-15 20:43:30.294044394 -0300
@@ -225,7 +225,7 @@
 
 =over 1
 
-=item B<http://suso.suso.org/programs/num-utils/>
+=item B<http://suso.suso.org/xulu/Num-utils>
 
 =back
 
