--- a/packages/dbgpt-core/pyproject.toml
+++ b/packages/dbgpt-core/pyproject.toml
@@ -12,9 +12,9 @@
 readme = "README.md"
 requires-python = ">= 3.10"
 dependencies = [
-    "aiohttp==3.8.4",
-    "chardet==5.1.0",
-    "importlib-resources==5.12.0",
+    "aiohttp>=3.8.4",
+    "chardet>=5.1.0",
+    "importlib-resources>=5.12.0",
     "cachetools",
     "pydantic>=2.6.0",
     # For AWEL type checking
@@ -40,14 +40,14 @@
 cli = [
     "prettytable",
     "click",
-    "psutil==5.9.4",
-    "colorama==0.4.6",
+    "psutil>=5.9.4",
+    "colorama>=0.4.6",
     "tomlkit",
     "rich",
 ]
 agent = [
     "termcolor",
-    "pandas==2.2.3",
+    "pandas>=2.2.3",
     # numpy should less than 2.0.0
     "numpy>=1.21.0,<2.0.0",
     "auto-gpt-plugin-template>=0.0.3",
@@ -68,27 +68,27 @@
     #  find a new toolkit.
     "pympler",
     "duckdb",
-    "duckdb-engine==0.9.1",
+    "duckdb-engine>=0.9.1",
     # lightweight python library for scheduling jobs
     "schedule",
     # For datasource subpackage
-    "sqlparse==0.4.4",
+    "sqlparse>=0.4.4",
     "python-multipart",
 ]
 framework = [
     "coloredlogs",
     "seaborn",
     "auto-gpt-plugin-template",
-    "gTTS==2.3.1",
+    "gTTS>=2.3.1",
     "pymysql",
     "jsonschema",
     "python-jsonpath",
     # TODO move transformers to default
     "tokenizers>=0.14",
-    "alembic==1.12.0",
+    "alembic>=1.12.0",
     # for excel
-    "openpyxl==3.1.2",
-    "xlrd==2.0.1",
+    "openpyxl>=3.1.2",
+    "xlrd>=2.0.1",
     "aiofiles",
     # for agent
     "GitPython",
