--- a/packages/dbgpt-ext/pyproject.toml
+++ b/packages/dbgpt-ext/pyproject.toml
@@ -26,7 +26,7 @@

 [project.optional-dependencies]
 rag = [
-    "spacy==3.7",
+    "spacy>=3.7",
     "markdown",
     "bs4",
     "python-pptx",
@@ -45,10 +45,10 @@
     # mysqlclient 2.2.x have pkg-config issue on 3.10+
     # If you want to install psycopg2 and mysqlclient in ubuntu, you should install
     # libpq-dev and libmysqlclient-dev first.
-    "mysqlclient==2.1.0",
+    "mysqlclient>=2.1.0",
 ]
 datasource_oracle = [
-    "oracledb==3.1.0",  # use python-oracledb，new driver for Oracle
+    "oracledb>=3.1.0",  # use python-oracledb，new driver for Oracle
 ]

 datasource_postgres = [
@@ -68,7 +68,7 @@
 datasource_vertica = ["vertica-python"]
 datasource_duckdb = [
     "duckdb",
-    "duckdb-engine==0.9.1",
+    "duckdb-engine>=0.9.1",
 ]
 # datasource_doris = ["pydoris>=1.0.2,<2.0.0"]
 storage_milvus = ["pymilvus"]
