--- a/Cargo.toml  1969-12-31 18:00:00.000000000 -0600
+++ b/Cargo.toml        2023-08-27 14:42:16.144242969 -0500
@@ -22,17 +22,3 @@
 [dependencies.pyo3]
 version = "0.11.1"
 features = ["extension-module"]
-
-[package.metadata.maturin]
-maintainer = "Stichting Polkascan (Polkascan Foundation)"
-maintainer-email = "legal@polkascan.org"
-classifier = [
-    "License :: OSI Approved :: Apache Software License",
-    "Intended Audience :: Developers",
-    "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.7",
-    "Programming Language :: Python :: 3.8",
-    "Programming Language :: Python :: 3.9",
-    "Programming Language :: Python :: 3.10",
-    "Programming Language :: Python :: 3.11"
-]
