diff --git a/pyproject.toml b/pyproject.toml
--- pyproject.toml
+++ pyproject.toml
@@ -1,4 +1,26 @@
 [build-system]
-requires = ["maturin>=0.12,<0.13"]
+requires = ["maturin"]
 build-backend = "maturin"

+[project]
+name = "py-sr25519-bindings"
+description = "Python bindings for sr25519 library"
+authors=[{name="Polkascan Foundation"}]
+version = "0.2.0"
+repository = "https://github.com/polkascan/py-sr25519-bindings"
+homepage = "https://github.com/polkascan/py-sr25519-bindings"
+license = "Apache-2.0"
+readme = "README.md"
+edition = "2018"
+maintainer = "Stichting Polkascan (Polkascan Foundation)"
+maintainer-email = "legal@polkascan.org"
+classifier = [
+    "License :: OSI Approved :: Apache Software License",
+    "Intended Audience :: Developers",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10"
+]
