--- a/config/readline.m4
+++ b/config/readline.m4
@@ -10,7 +10,7 @@
 #
 # This macro tries to link a test program using
 #
-#    -lreadline -ltermcap
+#    -lreadline
 #
 #
 #
@@ -22,7 +22,7 @@ AC_DEFUN([SWIN_LIB_READLINE],
   AC_MSG_CHECKING([for GNU readline installation])
 
   READLINE_CFLAGS=""
-  READLINE_LIBS="-lreadline -ltermcap"
+  READLINE_LIBS="-lreadline"
 
   ac_save_CFLAGS="$CFLAGS"
   ac_save_LIBS="$LIBS"
