From c58b8ff4101a1b260e17a8a74859b16a06c53152 Mon Sep 17 00:00:00 2001
From: AndresCasado <andres.casado.94@gmail.com>
Date: Tue, 20 Dec 2022 14:14:16 +0100
Subject: [PATCH] Delete unused and deprecated import

---
 chumpy/__init__.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/chumpy/__init__.py b/chumpy/__init__.py
index 21e366e..a49ac0e 100644
--- a/chumpy/__init__.py
+++ b/chumpy/__init__.py
@@ -8,7 +8,6 @@ from .version import version as __version__
 
 from .version import version as __version__
 
-from numpy import bool, int, float, complex, object, unicode, str, nan, inf
 
 def test():
     from os.path import split
-- 
2.51.0

