From 5db9369de4e160dc79689072ec4d1385417ee0d3 Mon Sep 17 00:00:00 2001
From: Ista Zahn <istazahn@gmail.com>
Date: Fri, 18 Mar 2022 07:31:53 -0400
Subject: [PATCH] don't use static linking

---
 src/cpp/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index d5b30b97dd..b4f2cebf3d 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -227,9 +227,9 @@ include_directories(SYSTEM "${YAML_CPP_INCLUDE_DIR}")
 # determine whether we should statically link boost. we always do this
 # unless we are building a non-packaged build on linux (in which case
 # boost dynamic libraries are presumed to be installed on the system ldpath)
-if(APPLE OR WIN32 OR RSTUDIO_PACKAGE_BUILD)
-   set(Boost_USE_STATIC_LIBS ON)
-endif()
+#if(APPLE OR WIN32 OR RSTUDIO_PACKAGE_BUILD)
+#   set(Boost_USE_STATIC_LIBS ON)
+#endif()
 
 # default Boost versions
 if(NOT RSTUDIO_BOOST_REQUESTED_VERSION)
-- 
2.35.1

