From aefed60c4a2b7c031a2561dc05a93d8891a1e352 Mon Sep 17 00:00:00 2001
From: Ista Zahn <istazahn@gmail.com>
Date: Tue, 22 Mar 2022 07:17:04 -0400
Subject: [PATCH] standard directory layout

---
 src/cpp/desktop/DesktopOptions.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/cpp/desktop/DesktopOptions.cpp b/src/cpp/desktop/DesktopOptions.cpp
index 26b7c579e8..e5243dc4d3 100644
--- a/src/cpp/desktop/DesktopOptions.cpp
+++ b/src/cpp/desktop/DesktopOptions.cpp
@@ -453,7 +453,7 @@ FilePath Options::supportingFilePath() const
             supportingFilePath_ = supportingFilePath_.completePath("Resources");
 #endif
    }
-   return supportingFilePath_;
+   return supportingFilePath_.completePath("share/rstudio");
 }
 
 FilePath Options::resourcesPath() const
@@ -465,9 +465,9 @@ FilePath Options::resourcesPath() const
       // part of the supporting files folder
       resourcesPath_ = supportingFilePath().completePath("resources");
 #else
-      // developer configuration: the 'resources' folder is
-      // a sibling of the RStudio executable
-      resourcesPath_ = scriptsPath().completePath("resources");
+      // conda-forge configuration: the 'resources' folder is
+      // still part of the supporting files folder
+      resourcesPath_ = supportingFilePath().completePath("resources");
 #endif
    }
 
-- 
2.35.1

