From 61f8d7c42b014793a31b0c26d512d079ebfa8809 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 17 Jul 2018 07:33:52 -0700
Subject: [PATCH 03/42] Qt5Sensors is not required

---
 src/cpp/desktop/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/cpp/desktop/CMakeLists.txt b/src/cpp/desktop/CMakeLists.txt
index 03d65cd2e3..9ad2104d0a 100644
--- a/src/cpp/desktop/CMakeLists.txt
+++ b/src/cpp/desktop/CMakeLists.txt
@@ -118,7 +118,6 @@ find_package(Qt5WebEngineWidgets REQUIRED)
 find_package(Qt5PrintSupport REQUIRED)
 find_package(Qt5Quick REQUIRED)
 find_package(Qt5QuickWidgets REQUIRED)
-find_package(Qt5Sensors REQUIRED)
 find_package(Qt5Svg REQUIRED)
 find_package(Qt5Xml REQUIRED)
 find_package(Qt5XmlPatterns REQUIRED)
@@ -338,7 +337,6 @@ set(RSTUDIO_QTMODULES
     Qt5::PrintSupport
     Qt5::Quick
     Qt5::QuickWidgets
-    Qt5::Sensors
     Qt5::Svg
     Qt5::Xml
     Qt5::XmlPatterns
-- 
2.28.0.windows.1

