From 474d4b95be837775e79f0aa0f403b29418e4a537 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 22 Oct 2016 18:37:40 +0100
Subject: [PATCH 05/42] Boost_USE_STATIC_LIBS when DEFINED ENV{CONDA_BUILD}

---
 src/cpp/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index 0637933da2..34f3a1aa10 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -192,7 +192,7 @@ endif()
 # determine whether we should statically link boost. we always do this
 # unless we are building a non-packaged build on linux (in which case
 # boost dynamic libraries are presumed to be installed on the system ldpath)
-if(APPLE OR WIN32 OR RSTUDIO_PACKAGE_BUILD)
+if(APPLE OR WIN32 OR RSTUDIO_PACKAGE_BUILD OR DEFINED ENV{CONDA_BUILD})
    set(Boost_USE_STATIC_LIBS ON)
 endif()
 
-- 
2.28.0.windows.1

