From 87d3389fa9c023ea8ca026ed43d2e1000c6bef5b Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 15 Jun 2018 12:01:26 +0000
Subject: [PATCH 39/42] Add support for Qt < 5.10

---
 src/cpp/desktop/DesktopDownloadItemHelper.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/cpp/desktop/DesktopDownloadItemHelper.cpp b/src/cpp/desktop/DesktopDownloadItemHelper.cpp
index 88252bbe7b..1e63f9dea3 100644
--- a/src/cpp/desktop/DesktopDownloadItemHelper.cpp
+++ b/src/cpp/desktop/DesktopDownloadItemHelper.cpp
@@ -38,8 +38,10 @@ DownloadHelper::DownloadHelper(QWebEngineDownloadItem* item)
    connect(item, &QWebEngineDownloadItem::finished,
            this, &DownloadHelper::onFinished);
    
+#if QT_VERSION >= QT_VERSION_CHECK(5,10,0)
    connect(item, &QWebEngineDownloadItem::isPausedChanged,
            this, &DownloadHelper::onPausedChanged);
+#endif
    
    connect(item, &QWebEngineDownloadItem::stateChanged,
            this, &DownloadHelper::onStateChanged);
-- 
2.28.0.windows.1

