diff -ru rstudio-2022.02.0-443.original/src/cpp/tests/cpp/tests/vendor/catch.hpp rstudio-2022.02.0-443/src/cpp/tests/cpp/tests/vendor/catch.hpp
--- a/src/cpp/tests/cpp/tests/vendor/catch.hpp	2022-02-17 08:26:03.000000000 +1300
+++ b/src/cpp/tests/cpp/tests/vendor/catch.hpp	2022-02-18 19:04:27.382806708 +1300
@@ -10824,7 +10824,7 @@
 
     // 32kb for the alternate stack seems to be sufficient. However, this value
     // is experimentally determined, so that's not guaranteed.
-    static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
+    static constexpr std::size_t sigStackSize = 32768;
 
     static SignalDefs signalDefs[] = {
         { SIGINT,  "SIGINT - Terminal interrupt signal" },
