--- rts/configure
+++ rts/configure
@@ -4055,2 +4055,5 @@
       # As far as I'm aware, none of these have relevant variants
+      osx)
+        BuildOS="darwin"
+        ;;
       freebsd|dragonfly|hpux|linuxaout|kfreebsdgnu|freebsd2|darwin|nextstep2|nextstep3|sunos4|ultrix|haiku)
@@ -4406,3 +4409,3 @@
             ;;
-        darwin|ios|watchos|tvos)
+        darwin|osx|ios|watchos|tvos)
             test -z "$2" || eval "$2=OSDarwin"
@@ -4603,2 +4606,5 @@
       # As far as I'm aware, none of these have relevant variants
+      osx)
+        HostOS="darwin"
+        ;;
       freebsd|dragonfly|hpux|linuxaout|kfreebsdgnu|freebsd2|darwin|nextstep2|nextstep3|sunos4|ultrix|haiku)
@@ -8272,3 +8278,3 @@
         ;;
-    darwin|ios|watchos|tvos)
+    darwin|osx|ios|watchos|tvos)
         RtsLinkerUseMmap=1
