diff --git a/hadrian/src/Rules/BinaryDist.hs b/hadrian/src/Rules/BinaryDist.hs
index 1234567..abcdefg 100644
--- a/hadrian/src/Rules/BinaryDist.hs
+++ b/hadrian/src/Rules/BinaryDist.hs
@@ -202,4 +202,8 @@
         -- (c.f. #20267).
-        ghcPkgName <- programName (vanillaContext Stage1 ghcPkg)
-        cmd_ (bindistFilesDir -/- "bin" -/- ghcPkgName) ["recache"]
+        cross <- flag CrossCompiling
+        unless cross $ do
+            -- N.B. the ghc-pkg executable may be prefixed with a target triple
+            -- (c.f. #20267).
+            ghcPkgName <- programName (vanillaContext Stage1 ghcPkg)
+            cmd_ (bindistFilesDir -/- "bin" -/- ghcPkgName) ["recache"]
 
