--- a/setup.py
+++ b/setup.py
@@ -1,56 +1,13 @@
 """Setup file for the sphinxcontrib_rust extension"""
 
-import subprocess
 from pathlib import Path
-from shutil import which
 
 from setuptools import setup, find_packages
-from setuptools.command.develop import develop
-from setuptools.command.install import install
 
 LONG_DESC = Path("index.rst").read_text(encoding="utf-8")
 
 __VERSION__ = "1.2.1"
-__THIS_DIR = Path(__file__).parent.absolute()
-# noinspection PyDeprecation
-__CARGO = which("cargo")
-if not __CARGO:
-    raise ValueError("Could not find cargo executable on path")
 
-
-# XXX: Use pip install -vvv -e to debug the build.
-# pip install does not print the output from cargo otherwise
-
-
-def run_cargo(command: str, *args: str) -> subprocess.CompletedProcess[str]:
-    """Run subprocess to build the sphinx-rustdocgen crate"""
-    command = [__CARGO, command, *args]
-    # noinspection PyTypeChecker
-    return subprocess.run(
-        command,
-        check=True,
-        text=True,
-        cwd=f"{__THIS_DIR}/sphinx-rustdocgen",
-    )
-
-
-class CargoBuild(develop):
-    """Command to install debug version of the sphinx-rustdocgen executable"""
-
-    def run(self):
-        run_cargo("install", "--debug", "--path", ".")
-        develop.run(self)
-
-
-class CargoInstall(install):
-    """Command to install release version of the sphinx-rustdocgen executable"""
-
-    def run(self):
-        """Run the install"""
-        run_cargo("install", "--path", ".")
-        install.run(self)
-
-
 setup(
     name="sphinxcontrib-rust",
     version=__VERSION__,
@@ -81,10 +38,6 @@
     packages=find_packages(),
     package_data={"sphinxcontrib_rust": ["sphinx-rustdocgen/**"]},
     include_package_data=True,
-    cmdclass={
-        "develop": CargoBuild,
-        "install": CargoInstall,
-    },
     url="https://gitlab.com/munir0b0t/sphinxcontrib-rust",
     author="Munir Contractor",
     license="GPL-3.0-or-later",
