From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jarrett Johnson <jarrett.johnson@schrodinger.com>
Date: Sat, 25 Jul 2026 00:00:00 +0000
Subject: [PATCH] Support system LZ4 installations that provide pkg-config

---
 CMakeLists.txt | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 88df32e46..8af484efe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -569,9 +569,16 @@ if(${SLANG_USE_SYSTEM_MINIZ})
 endif()
 
 if(${SLANG_USE_SYSTEM_LZ4})
-    find_package(lz4 REQUIRED)
-    if(NOT TARGET lz4_static)
+    find_package(lz4 CONFIG QUIET)
+    if(TARGET LZ4::lz4 AND NOT TARGET lz4_static)
         add_library(lz4_static ALIAS LZ4::lz4)
+    elseif(NOT TARGET lz4_static)
+        find_package(PkgConfig REQUIRED)
+        pkg_check_modules(
+            LZ4 REQUIRED IMPORTED_TARGET GLOBAL
+            liblz4
+        )
+        add_library(lz4_static ALIAS PkgConfig::LZ4)
     endif()
 endif()
 
-- 
2.50.1
