diff -ruN a/flowy/include/definitions.hpp b/flowy/include/definitions.hpp
--- a/flowy/include/definitions.hpp	2025-03-27 17:35:39.000000000 -0500
+++ b/flowy/include/definitions.hpp	2026-07-08 14:43:25.524788368 -0500
@@ -3,14 +3,14 @@
 // Copyright 2023--present Flowy developers
 
 #include "xtensor-blas/xlinalg.hpp"
-#include "xtensor/xbuilder.hpp"
-#include "xtensor/xmanipulation.hpp"
-#include "xtensor/xmath.hpp"
-#include "xtensor/xtensor_forward.hpp"
-#include <xtensor/xcsv.hpp>
-#include <xtensor/xfixed.hpp>
-#include <xtensor/xtensor.hpp>
-#include <xtensor/xview.hpp>
+#include "xtensor/generators/xbuilder.hpp"
+#include "xtensor/misc/xmanipulation.hpp"
+#include "xtensor/core/xmath.hpp"
+#include "xtensor/core/xtensor_forward.hpp"
+#include <xtensor/io/xcsv.hpp>
+#include <xtensor/containers/xfixed.hpp>
+#include <xtensor/containers/xtensor.hpp>
+#include <xtensor/views/xview.hpp>
 
 namespace Flowy
 {
diff -ruN a/flowy/include/dump_csv.hpp b/flowy/include/dump_csv.hpp
--- a/flowy/include/dump_csv.hpp	2025-03-27 17:35:39.000000000 -0500
+++ b/flowy/include/dump_csv.hpp	2026-07-08 14:43:25.520241617 -0500
@@ -8,8 +8,8 @@
 #include <string>
 #include <utility>
 
-#include <xtensor/xtensor.hpp>
-#include <xtensor/xtensor_config.hpp>
+#include <xtensor/containers/xtensor.hpp>
+#include <xtensor/core/xtensor_config.hpp>
 
 namespace Flowy::Utility
 {
diff -ruN a/flowy/include/topography.hpp b/flowy/include/topography.hpp
--- a/flowy/include/topography.hpp	2025-03-27 17:35:39.000000000 -0500
+++ b/flowy/include/topography.hpp	2026-07-08 14:43:25.499245983 -0500
@@ -3,9 +3,9 @@
 // Copyright 2023--present Flowy developers
 #include "flowy/include/definitions.hpp"
 #include "flowy/include/lobe.hpp"
-#include "xtensor/xbuilder.hpp"
-#include "xtensor/xmath.hpp"
-#include "xtensor/xsort.hpp"
+#include "xtensor/generators/xbuilder.hpp"
+#include "xtensor/core/xmath.hpp"
+#include "xtensor/misc/xsort.hpp"
 #include <iterator>
 #include <memory>
 #include <optional>
diff -ruN a/meson.build b/meson.build
--- a/meson.build	2025-03-27 17:35:39.000000000 -0500
+++ b/meson.build	2026-07-08 14:43:25.396580613 -0500
@@ -26,7 +26,7 @@
   message('Adding compiler flags', cpp_args)
 endif
 
-incdir = include_directories(['.', 'thirdparty/tsl'])
+incdir = include_directories(['.'])
 
 _sources = [
   'src/asc_file.cpp',
@@ -37,7 +37,7 @@
 
 # Library dependencies
 _deps = []
-pdflib_dep = dependency('pdf_cpplib', fallback : ['pdf_cpplib', 'pdflib_dep'])
+pdflib_dep = dependency('pdf_cpplib')
 
 with_netcdf = get_option('with_netcdf')
 ## Netcdf
diff -ruN a/src/config_parser.cpp b/src/config_parser.cpp
--- a/src/config_parser.cpp	2025-03-27 17:35:39.000000000 -0500
+++ b/src/config_parser.cpp	2026-07-08 14:43:25.473241272 -0500
@@ -5,7 +5,7 @@
 #include "flowy/include/config.hpp"
 #include "flowy/include/definitions.hpp"
 #include "flowy/include/netcdf_file.hpp"
-#include "thirdparty/toml.hpp"
+#include <toml++/toml.h>
 #include <fmt/format.h>
 #include <fmt/ranges.h>
 #include <fmt/std.h>
diff -ruN a/src/netcdf_file.cpp b/src/netcdf_file.cpp
--- a/src/netcdf_file.cpp	2025-03-27 17:35:39.000000000 -0500
+++ b/src/netcdf_file.cpp	2026-07-08 14:43:25.482249517 -0500
@@ -2,8 +2,8 @@
 // Copyright 2023--present Flowy developers
 #include "flowy/include/netcdf_file.hpp"
 #include "flowy/include/definitions.hpp"
-#include "xtensor/xmanipulation.hpp"
-#include "xtensor/xmath.hpp"
+#include "xtensor/misc/xmanipulation.hpp"
+#include "xtensor/core/xmath.hpp"
 #include <fmt/format.h>
 #include <fmt/ostream.h>
 #include <netcdf.h>
diff -ruN a/src/simulation.cpp b/src/simulation.cpp
--- a/src/simulation.cpp	2025-03-27 17:35:39.000000000 -0500
+++ b/src/simulation.cpp	2026-07-08 14:43:25.477797691 -0500
@@ -10,8 +10,8 @@
 #include "flowy/include/topography.hpp"
 #include "flowy/include/topography_file.hpp"
 #include "fmt/core.h"
-#include "xtensor/xmath.hpp"
-#include "xtensor/xsort.hpp"
+#include "xtensor/core/xmath.hpp"
+#include "xtensor/misc/xsort.hpp"
 #include <fmt/chrono.h>
 #include <fmt/format.h>
 #include <fmt/ranges.h>
diff -ruN a/src/topography.cpp b/src/topography.cpp
--- a/src/topography.cpp	2025-03-27 17:35:39.000000000 -0500
+++ b/src/topography.cpp	2026-07-08 14:43:25.468252428 -0500
@@ -3,8 +3,8 @@
 #include "flowy/include/topography.hpp"
 #include "flowy/include/definitions.hpp"
 #include "flowy/include/lobe.hpp"
-#include "thirdparty/tsl/robin_set.h"
-#include "xtensor/xbuilder.hpp"
+#include "tsl/robin_set.h"
+#include "xtensor/generators/xbuilder.hpp"
 #include <fmt/ranges.h>
 #include <fmt/std.h>
 #include <algorithm>
diff -ruN a/test/test_file_io.cpp b/test/test_file_io.cpp
--- a/test/test_file_io.cpp	2025-03-27 17:35:39.000000000 -0500
+++ b/test/test_file_io.cpp	2026-07-08 14:43:25.461253884 -0500
@@ -11,7 +11,7 @@
 #include "flowy/include/topography.hpp"
 #include "flowy/include/topography_file.hpp"
 #include "fmt/ostream.h"
-#include "xtensor/xbuilder.hpp"
+#include "xtensor/generators/xbuilder.hpp"
 #include <filesystem>
 
 void test_read_write(
diff -ruN a/test/test_lobe.cpp b/test/test_lobe.cpp
--- a/test/test_lobe.cpp	2025-03-27 17:35:39.000000000 -0500
+++ b/test/test_lobe.cpp	2026-07-08 14:43:25.457254715 -0500
@@ -3,7 +3,7 @@
 #include "flowy/include/definitions.hpp"
 #include "flowy/include/lobe.hpp"
 #include "flowy/include/math.hpp"
-#include "xtensor/xmath.hpp"
+#include "xtensor/core/xmath.hpp"
 #include <fmt/format.h>
 #include <fmt/ostream.h>
 #include <fmt/ranges.h>
diff -ruN a/test/test_topography.cpp b/test/test_topography.cpp
--- a/test/test_topography.cpp	2025-03-27 17:35:39.000000000 -0500
+++ b/test/test_topography.cpp	2026-07-08 14:43:25.441258042 -0500
@@ -6,7 +6,7 @@
 #include "flowy/include/math.hpp"
 #include "flowy/include/topography.hpp"
 #include "fmt/core.h"
-#include "xtensor/xbuilder.hpp"
+#include "xtensor/generators/xbuilder.hpp"
 #include "xtensor/xio.hpp"
 #include <fmt/format.h>
 #include <fmt/ostream.h>
diff -ruN a/test/test_truncated_normal.cpp b/test/test_truncated_normal.cpp
--- a/test/test_truncated_normal.cpp	2025-03-27 17:35:39.000000000 -0500
+++ b/test/test_truncated_normal.cpp	2026-07-08 14:43:25.437258873 -0500
@@ -7,8 +7,8 @@
 #include "flowy/include/definitions.hpp"
 #include "flowy/include/math.hpp"
 #include "pdf_cpplib/include/probability_dist.hpp"
-#include "xtensor/xbuilder.hpp"
-#include "xtensor/xmath.hpp"
+#include "xtensor/generators/xbuilder.hpp"
+#include "xtensor/core/xmath.hpp"
 #include <fmt/ostream.h>
 #include <cstddef>
 #include <filesystem>
