From d7ebde691ff1d13f7fb9e0633bf785f5f34bf6d4 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 2 May 2023 17:06:45 -0600
Subject: [PATCH] remove wheel and versioneer from requirementsRemo

---
 requirements.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/requirements.txt b/requirements.txt
index ef30108..f4c497c 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,3 @@
-wheel
-versioneer
 numpy
 scipy
 matplotlib
