diff --git a/triangle.h b/triangle.h
index 9df1f39..e5f5bb0 100644
--- a/triangle.h
+++ b/triangle.h
@@ -247,7 +247,10 @@
 /*  not used, `numberofsegments' will indicate the number of boundary edges. */
 /*                                                                           */
 /*****************************************************************************/
-
+#ifdef __cplusplus
+extern "C"
+{
+#endif
 struct triangulateio {
   REAL *pointlist;                                               /* In / out */
   REAL *pointattributelist;                                      /* In / out */
@@ -277,6 +280,14 @@ struct triangulateio {
   int *edgemarkerlist;            /* Not used with Voronoi diagram; out only */
   REAL *normlist;                /* Used only with Voronoi diagram; out only */
   int numberofedges;                                             /* Out only */
+
+#ifdef MWF_ADDED_FLAGS
+  /*mwf add a couple of flags to let me know when holelist and regionlist
+    have to be freed and aren't just copies of input pointers*/ 
+  int holelistmemflag;
+  int regionlistmemflag;
+#endif
+
 };
 
 #ifdef ANSI_DECLARATORS
@@ -287,3 +298,7 @@ void trifree(VOID *memptr);
 void triangulate();
 void trifree();
 #endif /* not ANSI_DECLARATORS */
+
+#ifdef __cplusplus
+}
+#endif
