diff --git a/INCHI_API/libinchi/gcc/makefile b/INCHI_API/libinchi/gcc/makefile
index 53d741d..6a378d3 100644
--- a/INCHI_API/libinchi/gcc/makefile
+++ b/INCHI_API/libinchi/gcc/makefile
@@ -182,11 +182,11 @@ sha2.o	\
 strutil.o	\
 util.o
 # === InChI Library link rule =========
-$(INCHI_LIB_PATHNAME).so$(VERSION): $(INCHI_LIB_OBJS)
-	$(SHARED_LINK) $(SHARED_LINK_PARM) -o $(INCHI_LIB_PATHNAME).so$(VERSION)	\
-$(INCHI_LIB_OBJS) -Wl$(LINUX_MAP)$(LINUX_Z_RELRO),-soname,$(INCHI_LIB_NAME).so$(MAIN_VERSION) -lm
-	ln -fs $(INCHI_LIB_NAME).so$(VERSION)	\
-$(INCHI_LIB_PATHNAME).so$(MAIN_VERSION)
+$(INCHI_LIB_PATHNAME)$(VERSION).dylib: $(INCHI_LIB_OBJS)
+	$(SHARED_LINK) $(SHARED_LINK_PARM) -o $(INCHI_LIB_PATHNAME)$(VERSION).dylib	\
+$(INCHI_LIB_OBJS) -Wl,-install_name,$(LIBDIR)/$(INCHI_LIB_NAME)$(MAIN_VERSION).dylib
+	ln -fs $(INCHI_LIB_NAME)$(VERSION).dylib \
+$(INCHI_LIB_PATHNAME)$(MAIN_VERSION).dylib
 # === InChI Library compile rule =========
 %.o: $(P_LIBR)%.c
 	$(C_COMPILER) $(C_SO_OPTIONS) $(C_OPTIONS) $<
