diff --git a/CMakeLists.txt b/CMakeLists.txt
index 997bef3..7fb14b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,16 +3,8 @@ project(cv_bridge)
 
 find_package(catkin REQUIRED COMPONENTS rosconsole sensor_msgs)
 
-if(NOT ANDROID)
-  find_package(PythonLibs)
-  if(PYTHONLIBS_VERSION_STRING VERSION_LESS 3)
-    find_package(Boost REQUIRED python)
-  else()
-    find_package(Boost REQUIRED python3)
-  endif()
-else()
-find_package(Boost REQUIRED)
-endif()
+find_package(Boost REQUIRED python)
+
 find_package(OpenCV 3 REQUIRED
   COMPONENTS
     opencv_core
