diff --git a/makefiles/Makefile.cpp.mk b/makefiles/Makefile.cpp.mk
index a44f310..874b9dc 100755
--- a/makefiles/Makefile.cpp.mk
+++ b/makefiles/Makefile.cpp.mk
@@ -605,34 +605,6 @@ $(OBJ_DIR)/sat/sat_runner.$O: $(EX_DIR)/cpp/sat_runner.cc $(EX_DIR)/cpp/opb_read
 $(BIN_DIR)/sat_runner$E: $(OR_TOOLS_LIBS) $(OBJ_DIR)/sat/sat_runner.$O
 	$(CCC) $(CFLAGS) $(OBJ_DIR)$Ssat$Ssat_runner.$O $(OR_TOOLS_LNK) $(OR_TOOLS_LD_FLAGS) $(EXE_OUT)$(BIN_DIR)$Ssat_runner$E
 
-# OR Tools unique library.
-
-$(LIB_DIR)/$(LIB_PREFIX)ortools.$(LIB_SUFFIX): \
-    $(BASE_LIB_OBJS) \
-    $(UTIL_LIB_OBJS) \
-    $(LP_DATA_LIB_OBJS) \
-    $(GLOP_LIB_OBJS) \
-    $(GRAPH_LIB_OBJS) \
-    $(ALGORITHMS_LIB_OBJS) \
-    $(SAT_LIB_OBJS) \
-    $(BOP_LIB_OBJS) \
-    $(LP_LIB_OBJS) \
-    $(CP_LIB_OBJS)
-	$(LINK_CMD) \
-	  $(LDOUT)$(LIB_DIR)$S$(LIB_PREFIX)ortools.$(LIB_SUFFIX) \
-	  $(BASE_LIB_OBJS) \
-	  $(UTIL_LIB_OBJS) \
-	  $(LP_DATA_LIB_OBJS) \
-	  $(GLOP_LIB_OBJS) \
-	  $(GRAPH_LIB_OBJS) \
-	  $(ALGORITHMS_LIB_OBJS) \
-	  $(SAT_LIB_OBJS) \
-	  $(BOP_LIB_OBJS) \
-	  $(LP_LIB_OBJS) \
-	  $(CP_LIB_OBJS) \
-	  $(DEPENDENCIES_LNK) \
-	  $(OR_TOOLS_LD_FLAGS)
-
 # compile and run C++ examples
 
 ccc: $(BIN_DIR)$S$(basename $(notdir $(EX)))$E
@@ -648,4 +620,4 @@ printdir:
 	@echo GEN_DIR = $(GEN_DIR)
 	@echo OBJ_DIR = $(OBJ_DIR)
 	@echo SRC_DIR = $(SRC_DIR)
-	@echo EX_DIR  = $(EX_DIR)
\ No newline at end of file
+	@echo EX_DIR  = $(EX_DIR)
diff --git a/makefiles/Makefile.unix b/makefiles/Makefile.unix
index 9a5597a..bf3c085 100755
--- a/makefiles/Makefile.unix
+++ b/makefiles/Makefile.unix
@@ -17,11 +17,11 @@ else # No need to keep the path in the lib, it is not stored there on linux.
 LIB_DIR = $(PREFIX)/lib
 endif
 BIN_DIR = $(PREFIX)/bin
-GEN_DIR = $(OR_ROOT)src/gen
+GEN_DIR = $(PREFIX)/include/ortools/gen
 OBJ_DIR = $(OR_ROOT)objs
-SRC_DIR = $(OR_ROOT)src
+SRC_DIR = $(PREFIX)/include/ortools
 EX_DIR  = $(OR_ROOT)examples
-INC_DIR = $(OR_ROOT)src
+INC_DIR = $(PREFIX)/include/ortools
 DEP_BIN_DIR = $(OR_ROOT)dependencies/install/bin
 
 
