--- a/tools/packaging/common_copy_licenses.sh
+++ b/tools/packaging/common_copy_licenses.sh
@@ -1 +1 @@
-#!/usr/bin/env bash
+#!/usr/bin/bash
@@ -102 +102 @@
-if [ $# -ne 2 ]; then
+if [ $# -ne 3 ]; then
@@ -106 +106 @@
-BASE="$(dirname $0)/../.."
+BASE="$3/../.."
@@ -170,5 +170 @@
-LICENSE_COUNT=$(ls $TARGET/*-license | wc -w)
-echo
-echo "The target directory for licenses ($TARGET) contains $LICENSE_COUNT license files:"
-ls $TARGET
-echo
