--- a/src/Emulator/Main/Tests/UnitTests/UnitTests_NET.csproj
+++ b/src/Emulator/Main/Tests/UnitTests/UnitTests_NET.csproj
@@ -18,5 +18,4 @@
     <PackageReference Include="Moq" Version="4.18.1" />
-    <Reference Include="IronPython">
-      <HintPath>..\..\..\..\..\..\..\lib\resources\libraries\ironpython-netcore\IronPython.dll</HintPath>
-    </Reference>
+    <PackageReference Include="IronPython" Version="2.7.12" />
+    <PackageReference Include="IronPython.StdLib" Version="2.7.12" />
     <PackageReference Include="NUnit" Version="3.13.1" />
--- a/src/Emulator/Main/Foreign/Player.cs
+++ b/src/Emulator/Main/Foreign/Player.cs
@@ -26,3 +26,4 @@
             this.stream = stream;
+            byte[] metadataBytes = null;
-            deserializer = new Serializer(new Settings(useBuffering: false, disableTypeStamping: true)).ObtainOpenStreamDeserializer(stream);
+            deserializer = new Serializer(new Settings(useBuffering: false, disableTypeStamping: true)).ObtainOpenStreamDeserializer(stream, out metadataBytes);
             handlersCache = new Dictionary<NameAndHandler, Delegate>();
