--- a/tools/building/createAssemblyInfo.ps1
+++ b/tools/building/createAssemblyInfo.ps1
@@ -4 +4 @@
-$CURRENT_INFORMATIONAL_VERSION = git rev-parse --short=8 HEAD
+$CURRENT_INFORMATIONAL_VERSION = "03756cb5"
--- a/tools/building/createAssemblyInfo.sh
+++ b/tools/building/createAssemblyInfo.sh
@@ -7 +7 @@
-CURRENT_INFORMATIONAL_VERSION="`git rev-parse --short=8 HEAD`"
+CURRENT_INFORMATIONAL_VERSION="03756cb5"
