--- a/src/Infrastructure/src/Emulator/Cores/cores.template_NET.csproj	2024-09-17 05:29:16.000000000 -0500
+++ b/src/Infrastructure/src/Emulator/Cores/cores.template_NET.csproj	2025-01-27 21:35:32.396468884 -0600
@@ -15,12 +16,2 @@

-  <Target Name="PrepareEmbeddedResources" BeforeTargets="ResolveReferences" >
-    <ItemGroup>
-      <EmbeddedResource Include="..\Cores\bin\$(Configuration)\lib\translate-$(Architecture)-le.so" Condition=" '%(Endianess.Identity)' == 'le' ">
-        <LogicalName>Antmicro.Renode.translate-$(Architecture)-le.so</LogicalName>
-      </EmbeddedResource>
-      <EmbeddedResource Include="..\Cores\bin\$(Configuration)\lib\translate-$(Architecture)-be.so" Condition=" '%(Endianess.Identity)' == 'be' ">
-        <LogicalName>Antmicro.Renode.translate-$(Architecture)-be.so</LogicalName>
-      </EmbeddedResource>
-    </ItemGroup>
-  </Target>
 </Project>