--- a/src/Infrastructure/src/Emulator/Cores/cores.template_NET.csproj	2024-09-17 05:29:16.000000000 -0500
+++ b/src/Infrastructure/src/Emulator/Cores/cores.template_NET.csproj	2025-01-27 21:35:32.396468884 -0600
@@ -1,4 +1,2 @@
 <Project DefaultTargets="Build">
-  <UsingTask AssemblyFile="$(MSBuildProjectDirectory)/../../../../../lib/cctask/CCTask/bin/Release/net6.0/CCTask.dll" TaskName="EnvironmentTask" />
-
   <PropertyGroup>
@@ -17,18 +15,9 @@
 
-  <Target Name="CompileTranslationLib" DependsOnTargets="VerifyProperties">
-    <!-- Due to a bug in some versions of xbuild we set default value of $(TargetWordSize) and $(TargetInsnStartExtraWords)
-             before passing it to 'translate.cproj' as it is impossible to overwrite them later. -->
-    <PropertyGroup>
-      <TargetWordSize Condition=" $(TargetWordSize) == '' ">32</TargetWordSize>
-      <TargetInsnStartExtraWords Condition=" $(TargetInsnStartExtraWords) == '' ">0</TargetInsnStartExtraWords>
-    </PropertyGroup>
-    <MSBuild Projects="./translate_NET.cproj" Properties="AdditionalCompilationFlags=$(AdditionalCompilationFlags);HostArchitecture=i386;HostEndianess=le;TargetEndianess=%(Endianess.Identity);TargetArchitecture=$(Architecture);EmulatedTarget=$(EmulatedTarget);Configuration=$(Configuration);TargetWordSize=$(TargetWordSize);TargetInsnStartExtraWords=$(TargetInsnStartExtraWords);CompilerPath=$(CompilerPath);LinkerPath=$(LinkerPath);ArPath=$(ArPath);CurrentPlatform=$(CurrentPlatform)">
-      <Output TaskParameter="TargetOutputs" ItemName="TranslationLibraries" />
-    </MSBuild>
-  </Target>
-
-  <Target Name="PrepareEmbeddedResources" BeforeTargets="ResolveReferences" DependsOnTargets="CompileTranslationLib">
+  <Target Name="PrepareEmbeddedResources" BeforeTargets="ResolveReferences" >
     <ItemGroup>
-      <EmbeddedResource Include="@(TranslationLibraries)">
-        <LogicalName>Antmicro.Renode.%(Filename)%(Extension)</LogicalName>
+      <EmbeddedResource Include="..\Cores\bin\$(Configuration)\lib\translate-$(Architecture)-le.so" Condition=" '%(Endianess.Identity)' == 'le' ">
+        <LogicalName>Antmicro.Renode.translate-$(Architecture)-le.so</LogicalName>
+      </EmbeddedResource>
+      <EmbeddedResource Include="..\Cores\bin\$(Configuration)\lib\translate-$(Architecture)-be.so" Condition=" '%(Endianess.Identity)' == 'be' ">
+        <LogicalName>Antmicro.Renode.translate-$(Architecture)-be.so</LogicalName>
       </EmbeddedResource>
@@ -36,11 +25,2 @@
   </Target>
-
-  <Target Name="ActualClean" DependsOnTargets="_PrepareProperties">
-    <MSBuild Projects="translate_NET.cproj" Targets="Clean" />
-  </Target>
-
-  <!-- It is intended to have 'Clean' and 'ActualClean' targets separated.
-         Thanks to this multiple calling of 'Clean' from MSBuild task will not cause
-         'ActualClean' to run more than once. -->
-  <Target Name="Clean" DependsOnTargets="ActualClean" />
-</Project>
+</Project>
\ No newline at end of file
