--- a/src/Renode/Renode_NET.csproj	2024-09-17 05:28:40.000000000 -0500
+++ b/src/Renode/Renode_NET.csproj	2025-01-30 11:47:34.648017897 -0600
@@ -14,1 +14,1 @@
-    <LangVersion>7.2</LangVersion>
+    <LangVersion>7.3</LangVersion>
@@ -45,2 +45,3 @@
   <ItemGroup>
+    <ProjectReference Include="DynamicLibraryLoader.csproj"/>
     <ProjectReference Include="..\Infrastructure\src\UI\UI_NET.csproj"/>
