diff --git a/jxrgluelib/JXRGlue.c b/jxrgluelib/JXRGlue.c
index 028d0c5..5ab9c53 100644
--- a/jxrgluelib/JXRGlue.c
+++ b/jxrgluelib/JXRGlue.c
@@ -398,6 +398,34 @@ Cleanup:
     return err;
 }
 
+ERR PKCodecFactory_CreateDecoderFromBytes(void* bytes, size_t len, PKImageDecode** ppDecoder)
+{
+	ERR err = WMP_errSuccess;
+
+	char *pExt = NULL;
+	const PKIID* pIID = NULL;
+
+	struct WMPStream* pStream = NULL;
+	PKImageDecode* pDecoder = NULL;
+
+	// get decode PKIID
+	Call(GetImageDecodeIID((const char *)".jxr", &pIID));
+
+	// create stream
+	Call(CreateWS_Memory(&pStream, bytes, len));
+
+	// Create decoder
+	Call(PKCodecFactory_CreateCodec(pIID, (void **)ppDecoder));
+	pDecoder = *ppDecoder;
+
+	// attach stream to decoder
+	Call(pDecoder->Initialize(pDecoder, pStream));
+	pDecoder->fStreamOwner = !0;
+
+Cleanup:
+	return err;
+}
+
 ERR PKCodecFactory_CreateDecoderFromFile(const char* szFilename, PKImageDecode** ppDecoder)
 {
     ERR err = WMP_errSuccess;
diff --git a/jxrgluelib/JXRGlue.h b/jxrgluelib/JXRGlue.h
index 1a38087..a8c96e7 100644
--- a/jxrgluelib/JXRGlue.h
+++ b/jxrgluelib/JXRGlue.h
@@ -586,6 +586,7 @@ ERR PKImageDecode_GetPhotoshopMetadata_WMP(PKImageDecode *pID, U8 *pbPhotoshopMe
 
 ERR PKImageDecode_Create(PKImageDecode** ppID);
 ERR PKCodecFactory_CreateDecoderFromFile(const char* szFilename, PKImageDecode** ppDecoder);
+ERR PKCodecFactory_CreateDecoderFromBytes(void* data, size_t size, PKImageDecode** ppDecoder);
 
 //================================================================
 typedef struct tagPKFormatConverter
