--- setup.py	2018-05-17 04:06:12.000000000 -0500
+++ setup.py.new	2019-03-16 17:35:10.662719065 -0500
@@ -6,11 +6,6 @@
 import os.path
 import subprocess
 
-setup_requires = ['numpy']
-FNULL = open(os.devnull, 'w')
-subprocess.check_call(
-    ['pip', 'install', *setup_requires], stdout=FNULL, stderr=subprocess.STDOUT
-)
 import numpy
 try:
     from distutils.command.build_py import build_py_2to3 \
