From de2ad9a6b65fe2d45a5b3e7621430aa1a68d9f73 Mon Sep 17 00:00:00 2001
From: "Matthew R. Becker" <beckermr@users.noreply.github.com>
Date: Tue, 5 Oct 2021 13:14:39 -0500
Subject: [PATCH] ENH move to setuptools

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 25d7426..995a87c 100644
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,5 @@
 import os
-from distutils.core import setup
+from setuptools import setup, find_packages

 scripts=[
     'des-make-image',
@@ -10,7 +10,7 @@

 setup(
     name="desimage",
-    packages=['desimage'],
+    packages=find_packages(),
     scripts=scripts,
     version="0.9.0",
     description="Make color images from DES coadds",
