From e36423dd54c4d1028295dcd28c0dedf40caca68f Mon Sep 17 00:00:00 2001
From: "Lori A. Burns" <lori.burns@gmail.com>
Date: Tue, 16 May 2023 15:41:52 -0400
Subject: [PATCH] lapack private to avoid overlinking

---
 src/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fb288aa..42cb0fe 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -106,6 +106,7 @@ if (NOT STATIC_ONLY)
       ambit-shared
       PUBLIC
         tgt::hdf5
+      PRIVATE
         tgt::lapack
       )
     set_target_properties (
@@ -124,6 +125,7 @@ if (NOT SHARED_ONLY)
       ambit-static
       PUBLIC
         tgt::hdf5
+      PRIVATE
         tgt::lapack
       )
     set_target_properties (
-- 
2.34.1

