From 0f0cb1ea2724e89480f29b0187204aaacc90488a Mon Sep 17 00:00:00 2001
From: ndana <nathan.dana@radiantsolutions.com>
Date: Tue, 14 Jul 2020 12:33:59 -0400
Subject: [PATCH] fix BERT.py

---
 sentence_transformers/models/BERT.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sentence_transformers/models/BERT.py b/sentence_transformers/models/BERT.py
index 6fe0e3a..0b7ab18 100755
--- a/sentence_transformers/models/BERT.py
+++ b/sentence_transformers/models/BERT.py
@@ -1,6 +1,6 @@
 from torch import Tensor
 from torch import nn
-from pytorch_transformers import  BertModel, BertTokenizer
+from transformers import  BertModel, BertTokenizer
 import json
 from typing import Union, Tuple, List, Dict
 import os
-- 
2.17.0

