From 7a561a10c3ca94afd73f184282b9c44f1ecd2e16 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Mon, 31 Oct 2022 10:14:07 +0100
Subject: [PATCH] Use CMAKE_SYSTEM_PROCESSOR instead of
 CMAKE_LIBRARY_ARCHITECTURE

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b5e8758..a1b2029 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -256,14 +256,14 @@ endif()
 if (NOT(${IS_X86_64_ARCH}))
   message(
     STATUS
-    "arch ${CMAKE_LIBRARY_ARCHITECTURE} does not match x86_64, "
+    "arch ${CMAKE_SYSTEM_PROCESSOR} does not match x86_64, "
     "skipping building SSE4.2 version of base64"
   )
   list(REMOVE_ITEM files ${FOLLY_DIR}/detail/base64_detail/Base64_SSE4_2.cpp)
 else()
   message(
     STATUS
-    "arch ${CMAKE_LIBRARY_ARCHITECTURE} matches x86_64, "
+    "arch ${CMAKE_SYSTEM_PROCESSOR} matches x86_64, "
     "building SSE4.2 version of base64"
   )
   # MSVC does not have a way to enable just sse4.2, only avx.
@@ -282,13 +282,13 @@ if (${LIBSODIUM_FOUND})
   if (NOT(${IS_X86_64_ARCH}))
     message(
       STATUS
-      "arch ${CMAKE_LIBRARY_ARCHITECTURE} does not match x86_64, "
+      "arch ${CMAKE_SYSTEM_PROCESSOR} does not match x86_64, "
       "skipping setting SSE2/AVX2 compile flags for LtHash SIMD code"
     )
   else()
     message(
       STATUS
-      "arch ${CMAKE_LIBRARY_ARCHITECTURE} matches x86_64, "
+      "arch ${CMAKE_SYSTEM_PROCESSOR} matches x86_64, "
       "setting SSE2/AVX2 compile flags for LtHash SIMD code"
     )
     set_source_files_properties(
-- 
2.39.2 (Apple Git-143)

