From bbab0928839f74992f26da21682aa0c89ff2db78 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 21 Nov 2021 16:27:21 +1100
Subject: [PATCH] don't hardcode compiler

---
 eval/ndeval/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eval/ndeval/Makefile b/eval/ndeval/Makefile
index f789269..dbab59a 100644
--- a/eval/ndeval/Makefile
+++ b/eval/ndeval/Makefile
@@ -1,4 +1,4 @@
 ndeval: ndeval.c
-	gcc -o ndeval ndeval.c -lm
+	${CC} -o ndeval ndeval.c -lm
 clean:
 	rm -f ndeval
\ No newline at end of file
-- 
2.32.0.windows.2

