--- setup.py	2021-04-11 14:49:34.000000000 -0400
+++ setup.py.new	2022-07-23 18:55:28.545527800 -0400
@@ -17,8 +17,6 @@
     author="Vincent D. Warmerdam",
     packages=find_packages(exclude=["notebooks"]),
     package_data={"drawdata": ["html/*.html"]},
-    long_description=read("readme.md"),
-    long_description_content_type="text/markdown",
     install_requires=base_packages,
     classifiers=[
         "Intended Audience :: Developers",
@@ -28,4 +26,4 @@
         "License :: OSI Approved :: MIT License",
         "Topic :: Scientific/Engineering",
     ],
-)
\ No newline at end of file
+)
