From ff6154a2c0c01ad80532c61c04df4f8f61d1a3c1 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 3 Oct 2021 13:53:29 -0600
Subject: [PATCH] Update requirements.txt

---
 requirements.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index a86eba71..a881fd31 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -4,7 +4,7 @@ graphviz>=0.16
 PyYAML>=5.4.1
 blessings>=1.7
 Pygments>=2.7.4
-rdflib~=4.2.1
+rdflib>=4.2.1
 semver>=2.8.1
 pydot>=1.4.1
 
