--- a/src/pni/arrays/index_map/index_map.hpp
+++ b/src/pni/arrays/index_map/index_map.hpp
@@ -257,7 +257,7 @@
                     >
             size_t offset(const ContainerT &index) const
             {
-                return implementation_type::template offset(_shape,index);
+                return implementation_type::template offset<>(_shape,index);
             }

             //-----------------------------------------------------------------
@@ -290,7 +290,7 @@
                     >
             size_t offset(const array_selection &s,const ContainerT &index) const
             {
-                return implementation_type::template offset(s,_shape,index);
+                return implementation_type::template offset<>(s,_shape,index);
             }

             //-----------------------------------------------------------------
@@ -313,7 +313,7 @@
             template<typename ContainerT> ContainerT index(size_t offset) const
             {
                 ContainerT index = container_utils<ContainerT>::create(rank()); 
-                implementation_type::template index(_shape,index,offset);
+                implementation_type::template index<>(_shape,index,offset);
                 return index;

             }
--- a/src/pni/arrays/index_map/static_index_map.hpp
+++ b/src/pni/arrays/index_map/static_index_map.hpp
@@ -273,7 +273,7 @@
                     > 
             size_t offset(const ContainerT &index) const
             {
-                return implementation_type::template offset(_shape,index);
+                return implementation_type::template offset<>(_shape,index);
             }
             
             //-----------------------------------------------------------------
@@ -297,7 +297,7 @@
             size_t offset(const array_selection &s,const ContainerT &index) 
             const
             {
-                return implementation_type::template offset(s,_shape,index);
+                return implementation_type::template offset<>(s,_shape,index);
             }

             //-----------------------------------------------------------------
@@ -323,7 +323,7 @@
             template<typename ContainerT> ContainerT index(size_t offset) const
             {
                 ContainerT index = container_utils<ContainerT>::create(rank()); 
-                implementation_type::template index(_shape,index,offset);
+                implementation_type::template index<>(_shape,index,offset);
                 return index;
             }

