diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 048d231..1d8e749 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -11,7 +11,10 @@ message(STATUS "-------- LLVM / Clang detection -------------")
 include(clang_detection)
 
 # fmt
-if(NOT TARGET fmt)
+if(NOT TARGET fmt::fmt)
+  find_package(fmt CONFIG QUIET)
+endif()
+if(NOT TARGET fmt::fmt)
   message(STATUS "-------- Fetch & Configure FMT -------------")
   include(FetchContent)
   FetchContent_Declare(
@@ -27,6 +30,9 @@ else()
 endif()
 
 # toml++
+if(NOT TARGET tomlplusplus::tomlplusplus)
+  find_package(tomlplusplus CONFIG QUIET)
+endif()
 if(NOT TARGET tomlplusplus::tomlplusplus)
   message(STATUS "-------- Fetch & Configure toml++ -------------")
   include(FetchContent)
