--- Makefile.in.orig	2016-10-06 14:50:05.955243132 +0000
+++ Makefile.in	2016-10-06 14:51:12.729258334 +0000
@@ -928,7 +928,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -956,7 +956,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
--- tests/Makefile.in.orig	2016-10-06 15:00:55.895372258 +0000
+++ tests/Makefile.in	2016-10-06 15:01:17.578376464 +0000
@@ -618,7 +618,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -646,7 +646,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
--- tests/scanner/Makefile.in.orig	2016-10-06 15:06:05.398430946 +0000
+++ tests/scanner/Makefile.in	2016-10-06 15:06:58.893440506 +0000
@@ -720,7 +720,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = --verbose -I$(top_srcdir) \
 	--add-include-path=$(srcdir) \
@@ -746,7 +746,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
--- tests/scanner/annotationparser/Makefile.in.orig	2016-10-06 15:06:09.883431748 +0000
+++ tests/scanner/annotationparser/Makefile.in	2016-10-06 15:07:16.782443702 +0000
@@ -521,7 +521,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -549,7 +549,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
--- tests/warn/Makefile.in.orig	2016-10-06 15:06:16.318432898 +0000
+++ tests/warn/Makefile.in	2016-10-06 15:07:30.891446224 +0000
@@ -520,7 +520,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -548,7 +548,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
--- tests/offsets/Makefile.in.orig	2016-10-06 15:06:01.111430180 +0000
+++ tests/offsets/Makefile.in	2016-10-06 15:06:39.637437065 +0000
@@ -619,7 +619,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -647,7 +647,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
