From 919dd99871eceb270821e191087939333376ac38 Mon Sep 17 00:00:00 2001
From: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>
Date: Wed, 4 May 2022 20:31:20 +0200
Subject: [PATCH] Don't include drivers

---
 MANIFEST.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/MANIFEST.in b/MANIFEST.in
index e089537..2137b0e 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -2,7 +2,6 @@ include LICENSE.txt
 include requirements.txt
 recursive-include qiskit_nature *.txt
 graft qiskit_nature/drivers/gaussiand/gauopen
-graft qiskit_nature/drivers/second_quantization/gaussiand/gauopen
 graft test
 prune docs
 prune tools
-- 
2.35.3

