From b575ff9f690ee3076c5fe9f335dad4b4fd81712d Mon Sep 17 00:00:00 2001
From: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>
Date: Wed, 4 May 2022 19:31:57 +0200
Subject: [PATCH] Don't include drivers

---
 MANIFEST.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/MANIFEST.in b/MANIFEST.in
index 4114105..89cbe8c 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,7 +1,6 @@
 include LICENSE.txt
 include requirements.txt
 recursive-include qiskit_nature *.txt
-graft qiskit_nature/drivers/second_quantization/gaussiand/gauopen
 graft test
 prune docs
 prune tools
-- 
2.35.3

