--- pyproject.toml	2025-04-04 08:13:16.671019001 +0000
+++ pyproject.toml	2025-04-04 09:22:12.264478013 +0000
@@ -4,7 +4,7 @@
 requires = [ "cython", "numpy", "scikit-build-core" ]
 
 [project]
-name = "fastemriwaveforms" #@NAMESUFFIX@
+name = "fastemriwaveforms-cuda" #@NAMESUFFIX@
 
 description = "Fast and accurate EMRI Waveforms."
 
@@ -39,7 +39,6 @@
   "Programming Language :: Python :: 3.13",
 ]
 # version is deduced from the lattest git tag
-# description is the docstring at the top of src/few/__init__.py
 dynamic = [ "version" ]
 
 dependencies = [
@@ -91,9 +90,6 @@
 urls.Source = "https://github.com/BlackHolePerturbationToolkit/FastEMRIWaveforms"
 urls.Tracker = "https://github.com/BlackHolePerturbationToolkit/FastEMRIWaveforms/issues"
 
-scripts.few_citations = "few.cmd.citations:main" #@SKIP_PLUGIN@
-scripts.few_files = "few.cmd.files:main"         #@SKIP_PLUGIN@
-
 [tool.setuptools_scm]
 version_scheme = "no-guess-dev"
 local_scheme = "node-and-date"
@@ -142,8 +138,6 @@
 ]
 
 [tool.scikit-build.wheel.packages]
-few = "src/few"       #@SKIP_PLUGIN@
-"few/tests" = "tests" #@SKIP_PLUGIN@
 
 [tool.ruff]
 
