--- pyproject.toml    2025-02-20 15:05:00.915214008 +0000
+++ pyproject.toml    2025-02-20 15:05:40.883796013 +0000
@@ -83,8 +83,9 @@
 scripts.few_files = "few.cmd.files:main"
 
 [tool.setuptools_scm]
-version_scheme = "no-guess-dev"
-local_scheme = "node-and-date"
+version_scheme = "only-version"
+local_scheme = "no-local-version"
+fallback_version = "2.0.0rc1"
 version_file = "src/few/_version.py"
 
 [tool.scikit-build]
