diff --git a/CMakeLists.txt b/CMakeLists.txt
index ab37d6c..30fc98e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,14 +35,30 @@ if(NOT TARGET OpenMP::OpenMP_Fortran)
   )
 endif()
 
-find_package(LAPACK REQUIRED)
+# FIXME: We currently only support MKL here:
+set(BLA_VENDOR "Intel10_64lp" CACHE STRING "Search for BLAS only from the specified vendor." FORCE)
+if(NOT TARGET BLAS::BLAS)
+  find_package(BLAS REQUIRED)
+  if(NOT TARGET BLAS::BLAS AND BLAS_FOUND)
+    add_library(BLAS::BLAS INTERFACE IMPORTED)
+    target_link_libraries(BLAS::BLAS INTERFACE "${BLAS_LIBRARIES}")
+    target_link_options(BLAS::BLAS INTERFACE "${BLAS_LINKER_FLAGS}")
+  endif()
+endif()
+if(NOT TARGET LAPACK::LAPACK)
+  find_package(LAPACK REQUIRED)
+  if(NOT TARGET LAPACK::LAPACK AND LAPACK_FOUND)
+    add_library(LAPACK::LAPACK INTERFACE IMPORTED)
+    target_link_libraries(LAPACK::LAPACK INTERFACE "${LAPACK_LIBRARIES}")
+    target_link_options(LAPACK::LAPACK INTERFACE "${LAPACK_LINKER_FLAGS}")
+  endif()
+endif()
 list(
   APPEND exe-deps
   LAPACK::LAPACK
+  BLAS::BLAS
 )
 
-set(exe-deps "${exe-deps}" PARENT_SCOPE)
-
 # General configuration information
 set(prog)
 set(srcs)
