diff --git a/ci/download-wasmtime.py b/ci/download-wasmtime.py
index 1bf95af..c2ec020 100644
--- a/ci/download-wasmtime.py
+++ b/ci/download-wasmtime.py
@@ -103,6 +103,7 @@ def main(platform, arch):
 
 
 if __name__ == '__main__':
+    sys.exit(0)
     if len(sys.argv) > 2:
         main(sys.argv[1], sys.argv[2])
     else:
diff --git a/wasmtime/_ffi.py b/wasmtime/_ffi.py
index 2f277b3..b1ac411 100644
--- a/wasmtime/_ffi.py
+++ b/wasmtime/_ffi.py
@@ -35,7 +35,7 @@ if machine != 'x86_64' and machine != 'aarch64':
 
 filename = Path(__file__).parent / (sys_platform + '-' + machine) / libname
 
-dll = cdll.LoadLibrary(str(filename))
+dll = cdll.LoadLibrary(filename.name[1:])
 
 WASM_I32 = c_uint8(0)
 WASM_I64 = c_uint8(1)
