diff --git a/app/app/settings.py b/app/app/settings.py
index b55abca..5fd7a8e 100644
--- a/app/app/settings.py
+++ b/app/app/settings.py
@@ -60,8 +60,7 @@ INSTALLED_APPS = [
     'social_django',
     'polymorphic',
     'webpack_loader',
-    'corsheaders',
-    'drf_yasg'
+    'corsheaders'
 ]
 
 CLOUD_BROWSER_APACHE_LIBCLOUD_PROVIDER = env('CLOUD_BROWSER_LIBCLOUD_PROVIDER', None)
diff --git a/app/app/urls.py b/app/app/urls.py
index a71ac81..9c8d56a 100644
--- a/app/app/urls.py
+++ b/app/app/urls.py
@@ -17,22 +17,8 @@ from django.conf import settings
 from django.contrib import admin
 from django.urls import path, include
 from django.contrib.auth.views import PasswordResetView, LogoutView
-from drf_yasg.views import get_schema_view
-from drf_yasg import openapi
-
 from server.views import LoginView
 
-# TODO: adds AnnotationList and AnnotationDetail endpoint.
-schema_view = get_schema_view(
-   openapi.Info(
-      title="doccano API",
-      default_version='v1',
-      description="doccano API description",
-      license=openapi.License(name="MIT License"),
-   ),
-   public=True,
-)
-
 urlpatterns = [
     path('', include('authentification.urls')),
     path('', include('server.urls')),
