diff --git a/CMakeLists.txt b/CMakeLists.txt
index b70ec44..2d223d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,8 @@ set(LIBFRV_VERSION_STRING "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${P
 find_package(PkgConfig REQUIRED)
 
 set(FRV_DEPENDENCIES "framel >= 8.38.1" "fftw3 >= 3.3.8" "fftw3f")
-pkg_check_modules(FRVDEPS REQUIRED IMPORTED_TARGET ${FRV_DEPENDENCIES})
+list(TRANSFORM FRV_DEPENDENCIES REPLACE " " "" OUTPUT_VARIABLE FRV_DEPENDENCIES_PKGCONFIG)
+pkg_check_modules(FRVDEPS REQUIRED IMPORTED_TARGET ${FRV_DEPENDENCIES_PKGCONFIG})
 
 add_subdirectory(src)
 
