--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,58 @@
+cmake_minimum_required( VERSION 3.5 )
+ 
+cmake_policy( SET CMP0048 NEW )
+ 
+project( xdr VERSION 4.9.1 )
+ 
+if( NOT WIN32 )
+  set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -fno-strict-aliasing -std=gnu89 -Wno-old-style-definition -Wno-strict-prototypes" )
+endif()
+ 
+include( TestBigEndian )
+ 
+if( WIN32 )
+  if( CMAKE_CXX_COMPILER_ID STREQUAL "GNU" )
+    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu89 -Wno-old-style-definition -Wno-strict-prototypes" )
+  endif()
+ 
+  add_definitions( -DBIG_ENDIAN=1 -DLITTLE_ENDIAN=2 )
+  test_big_endian( BIG_ENDIAN )
+ 
+  if( BIG_ENDIAN )
+    add_definitions( -DBYTE_ORDER=1 )
+  else()
+    add_definitions( -DBYTE_ORDER=2 )
+  endif()
+endif()
+ 
+set( SOURCES xdr_array.c xdr_float.c xdr_mem.c xdr_reference.c xdr.c xdr_intXX_t.c xdr_rec.c xdr_stdio.c )
+ 
+configure_file(
+  config.h.in
+  ${CMAKE_BINARY_DIR}/config.h
+)
+ 
+include_directories( ${CMAKE_BINARY_DIR} ${CMAKE_SOURCE_DIR} )
+ 
+option( BUILD_SHARED_LIBS "build xdrlib as a shared library" ON )
+if( BUILD_SHARED_LIBS )
+    add_library( ${PROJECT_NAME} SHARED ${SOURCES} )
+else()
+    add_library( ${PROJECT_NAME} STATIC ${SOURCES} )
+endif()
+ 
+if( CMAKE_C_COMPILER_ID MATCHES "GNU|Clang" )
+    target_compile_options( ${PROJECT_NAME} PRIVATE -Wno-incompatible-pointer-types )
+endif()
+ 
+install(
+    FILES rpc/rpc.h rpc/types.h rpc/xdr.h ${CMAKE_BINARY_DIR}/config.h
+    DESTINATION include/rpc
+)
+ 
+install(
+  TARGETS ${PROJECT_NAME}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
