From 258a23b7e63711f9ba2f7a75b64e6c31a7f5ea21 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 9 Oct 2017 21:54:35 +0100
Subject: [PATCH 01/41] macOS: Allow disabling R framework

---
 cmake/modules/FindLibR.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmake/modules/FindLibR.cmake b/cmake/modules/FindLibR.cmake
index 31bff35a78..afb6f5d175 100644
--- a/cmake/modules/FindLibR.cmake
+++ b/cmake/modules/FindLibR.cmake
@@ -17,8 +17,10 @@
 # LIBR_DOC_DIR
 # LIBR_LIBRARIES
 
+set(USE_MACOS_R_FRAMEWORK TRUE CACHE BOOL "Look for R framework on macOS")
+
 # detection for OSX (look for R framework)
-if(APPLE)
+if(APPLE AND USE_MACOS_R_FRAMEWORK)
 
    find_library(LIBR_LIBRARIES R)
    
-- 
2.29.2.windows.3

