From 5a6fd36414a477a93ff6d56018c4725bb592299a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 10 Aug 2016 14:34:20 +0100
Subject: [PATCH 02/41] Qt5Positioning is not required

---
 src/cpp/desktop/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/cpp/desktop/CMakeLists.txt b/src/cpp/desktop/CMakeLists.txt
index 3854a0bd96..ff89826715 100644
--- a/src/cpp/desktop/CMakeLists.txt
+++ b/src/cpp/desktop/CMakeLists.txt
@@ -121,7 +121,6 @@ find_package(Qt5WebEngineWidgets REQUIRED)
 find_package(Qt5PrintSupport REQUIRED)
 find_package(Qt5Quick REQUIRED)
 find_package(Qt5QuickWidgets REQUIRED)
-find_package(Qt5Positioning REQUIRED)
 find_package(Qt5Sensors REQUIRED)
 find_package(Qt5Svg REQUIRED)
 find_package(Qt5Xml REQUIRED)
@@ -345,7 +344,6 @@ set(RSTUDIO_QTMODULES
     Qt5::PrintSupport
     Qt5::Quick
     Qt5::QuickWidgets
-    Qt5::Positioning
     Qt5::Sensors
     Qt5::Svg
     Qt5::Xml
-- 
2.29.2.windows.3

