From 9f267d578e040714fd8903ac2d37d2b9a1b58075 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 9 Nov 2016 17:52:56 +0000
Subject: [PATCH 21/41] Conda: Add arch suffix to the QSettings file

---
 src/cpp/desktop/DesktopOptions.hpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/cpp/desktop/DesktopOptions.hpp b/src/cpp/desktop/DesktopOptions.hpp
index a11fb01e26..286d797e88 100644
--- a/src/cpp/desktop/DesktopOptions.hpp
+++ b/src/cpp/desktop/DesktopOptions.hpp
@@ -37,6 +37,18 @@
 #define FORMAT QSettings::IniFormat
 #endif
 
+#if defined(CONDA_BUILD)
+// We do not want the x64 settings being loaded by i686 RStudio and vice-versa because
+// they will point at different R installations.
+#if defined(__x86_64) || defined(__x86_64__) || defined(_M_X64) || defined(_M_AMD64)
+#define ARCH_SUFFIX "-x64"
+#else
+#define ARCH_SUFFIX "-i686"
+#endif
+#else
+#define ARCH_SUFFIX ""
+#endif
+
 namespace rstudio {
 namespace desktop {
 
-- 
2.29.2.windows.3

