From a630856a3b88d3320e94def40d51354542f1de8b Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 24 Mar 2018 17:30:05 +0000
Subject: [PATCH 27/41] Use find_program to find qmake as a last-ditch fallback

---
 src/cpp/desktop/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/cpp/desktop/CMakeLists.txt b/src/cpp/desktop/CMakeLists.txt
index 269ef9fa18..2055911f6b 100644
--- a/src/cpp/desktop/CMakeLists.txt
+++ b/src/cpp/desktop/CMakeLists.txt
@@ -94,6 +94,10 @@ else()
    endif()
 endif()
 
+if(NOT QT_QMAKE_EXECUTABLE)
+   find_program(QT_QMAKE_EXECUTABLE qmake)
+endif()
+
 if(QT_QMAKE_EXECUTABLE)
    message(STATUS "Found Qt: ${QT_QMAKE_EXECUTABLE}")
 else()
-- 
2.29.2.windows.3

