From 1e08662ceba14b75267333fba0826f2b6fedb2ed Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 25 Mar 2018 16:11:37 +0000
Subject: [PATCH 29/41] Add missing std::mutex header

---
 src/cpp/desktop/DesktopWebPage.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpp/desktop/DesktopWebPage.cpp b/src/cpp/desktop/DesktopWebPage.cpp
index 4c05e11a2d..6ee3535de4 100644
--- a/src/cpp/desktop/DesktopWebPage.cpp
+++ b/src/cpp/desktop/DesktopWebPage.cpp
@@ -16,7 +16,7 @@
 #include "DesktopWebPage.hpp"
 
 #include <boost/algorithm/string.hpp>
-#include <boost/thread/once.hpp>
+#include <mutex>
 
 #include <core/http/URL.hpp>
 #include <core/Thread.hpp>
-- 
2.29.2.windows.3

