From 5e2259679d7e4414e57510146130cd6972006874 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 27 Mar 2018 04:58:02 -0700
Subject: [PATCH 30/41] Force overwrite when unzipping dependencies on Windows

---
 dependencies/windows/install-dependencies.cmd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dependencies/windows/install-dependencies.cmd b/dependencies/windows/install-dependencies.cmd
index 36306ff1cb..e1e0a34979 100644
--- a/dependencies/windows/install-dependencies.cmd
+++ b/dependencies/windows/install-dependencies.cmd
@@ -5,7 +5,7 @@ setlocal
 set PATH=%CD%\tools;%PATH%
 
 set WGET_ARGS=-c --no-check-certificate
-set UNZIP_ARGS=-q
+set UNZIP_ARGS=-q -o
 
 set BASEURL=https://s3.amazonaws.com/rstudio-buildtools/
 set GIN_FILE=gin-2.1.2.zip
-- 
2.29.2.windows.3

