From dbbc0cc29eacdad504a251f5f8ee3fcf37321883 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 27 Mar 2018 04:58:02 -0700
Subject: [PATCH 30/41] Force overwrite when unzipping dependencies on Windows

---
 dependencies/windows/install-dependencies.cmd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dependencies/windows/install-dependencies.cmd b/dependencies/windows/install-dependencies.cmd
index f91582901f..c98069b0e6 100644
--- a/dependencies/windows/install-dependencies.cmd
+++ b/dependencies/windows/install-dependencies.cmd
@@ -5,7 +5,7 @@ setlocal
 set PATH=%CD%\tools;%PATH%
 
 set WGET_ARGS=-c --no-check-certificate
-set UNZIP_ARGS=-q
+set UNZIP_ARGS=-q -o
 
 set BASEURL=https://s3.amazonaws.com/rstudio-buildtools/
 set GIN_FILE=gin-2.1.2.zip
-- 
2.29.2.windows.3

