From 2351ce48009e71cc4fb152a414d70c4fa0076848 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 15 Jun 2018 12:01:26 +0000
Subject: [PATCH 39/41] Add support for Qt < 5.10

---
 src/cpp/desktop/DesktopDownloadItemHelper.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/cpp/desktop/DesktopDownloadItemHelper.cpp b/src/cpp/desktop/DesktopDownloadItemHelper.cpp
index 717e44acfd..3269563210 100644
--- a/src/cpp/desktop/DesktopDownloadItemHelper.cpp
+++ b/src/cpp/desktop/DesktopDownloadItemHelper.cpp
@@ -38,8 +38,10 @@ DownloadHelper::DownloadHelper(QWebEngineDownloadItem* item)
    connect(item, &QWebEngineDownloadItem::finished,
            this, &DownloadHelper::onFinished);
    
+#if QT_VERSION >= QT_VERSION_CHECK(5,10,0)
    connect(item, &QWebEngineDownloadItem::isPausedChanged,
            this, &DownloadHelper::onPausedChanged);
+#endif
    
    connect(item, &QWebEngineDownloadItem::stateChanged,
            this, &DownloadHelper::onStateChanged);
-- 
2.29.2.windows.3

