diff --git a/Users/schiotz/tmp/pyqstem-1.0.2/setup.py b/setup.py
index c59f3e6..4d7ad21 100644
--- a/Users/schiotz/tmp/pyqstem-1.0.2/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ sources += ['pyqstem/qstem_interface.pyx','pyqstem/QSTEM.cpp']
 if is_windows:
     compargs = ['-D MS_WIN64']
 else:
-    compargs = ['-std=c++11', '-stdlib=libc++']
+    compargs = ['-std=c++11']
 
 setup(name='pyqstem',
       packages = find_packages(),
