diff --git a/CMakeLists.txt b/CMakeLists.txt
index efc49a2..097aaf8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,14 +78,14 @@ endif()
 install(
   TARGETS jwt-cpp
   EXPORT jwt-cpp
-  DESTINATION jwt-cpp)
+  DESTINATION lib/cmake/jwt-cpp)
 install(
   EXPORT jwt-cpp
   NAMESPACE jwt-cpp::
-  DESTINATION jwt-cpp)
+  DESTINATION lib/cmake/jwt-cpp)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/jwt-cpp-config.cmake
               ${CMAKE_CURRENT_BINARY_DIR}/jwt-cpp-config-version.cmake
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/jwt-cpp)
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/cmake/jwt-cpp)
 
 if(BUILD_TESTS)
   add_subdirectory(tests)
