Index: setup.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup.py	(date 1566248322000)
+++ setup.py	(date 1567411504000)
@@ -75,7 +75,7 @@
     # All libraries to link to all modules:
     base_libs = ['c', 'rt', 'dl']
     # No "no reproducible builds" warning:
-    base_compile_args = ['-Wno-date-time']
+    base_compile_args = ['-Wno-date-time', '-I/usr/include/alsa']
     # Extra OS specific libs for PsychPortAudio:
     audio_libdirs = []
     audio_extralinkargs = []
