diff --git a/setup.py b/setup.py
index 1571d3c..f868a37 100755
--- a/setup.py
+++ b/setup.py
@@ -450,15 +450,9 @@ class tkwant_sdist(sdist):
 ################ requirements
 
 requirements = (
-    "cython>=0.21.1",
-    "numpy>=1.8.2",
-    "scipy>=0.14.0",
-    "sympy",
-    "kwant>=1.3,<2.0",
-    "mpi4py>=2.0",
-    "dill",
-    "tinyarray",
-    "kwantspectrum",
+    "cython",
+    "numpy",
+    "scipy"
 )
 
 test_requirements = (
-- 
2.17.1

