diff --git a/vendor/github.com/containers/common/pkg/config/config.go b/vendor/github.com/containers/common/pkg/config/config.go
index 18fab548..e9e4d7a6 100644
--- a/vendor/github.com/containers/common/pkg/config/config.go
+++ b/vendor/github.com/containers/common/pkg/config/config.go
@@ -17,6 +17,33 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// const char *condaPrefix = "";
+import "C"
+
+// For Conda environments we transform default configuration paths like so:
+//   /etc/path -> PREFIX/etc/path
+//   /usr/share/path -> PREFIX/share/path
+func PathInPrefix(path string) string {
+	condaPrefix := C.GoString(C.condaPrefix)
+	if condaPrefix == "" {
+		return path
+	}
+	relativePath := strings.TrimPrefix(path, "/usr")
+	return filepath.Join(condaPrefix, relativePath)
+}
+
+func FallbackToPathInPrefix(path string) string {
+	condaPrefix := C.GoString(C.condaPrefix)
+	if condaPrefix == "" {
+		return path
+	}
+	if _, err := os.Stat(path); err == nil {
+		return path
+	}
+	relativePath := strings.TrimPrefix(path, "/usr")
+	return filepath.Join(condaPrefix, relativePath)
+}
+
 const (
 	// _configPath is the path to the containers/containers.conf
 	// inside a given config directory.
@@ -501,8 +528,8 @@ func systemConfigs() ([]string, error) {
 		}
 		return append(configs, path), nil
 	}
-	if _, err := os.Stat(DefaultContainersConfig); err == nil {
-		configs = append(configs, DefaultContainersConfig)
+	if _, err := os.Stat(PathInPrefix(DefaultContainersConfig)); err == nil {
+		configs = append(configs, PathInPrefix(DefaultContainersConfig))
 	}
 	if _, err := os.Stat(OverrideContainersConfig); err == nil {
 		configs = append(configs, OverrideContainersConfig)
diff --git a/vendor/github.com/containers/common/pkg/config/default.go b/vendor/github.com/containers/common/pkg/config/default.go
index 5f8f4999..5bcdbe75 100644
--- a/vendor/github.com/containers/common/pkg/config/default.go
+++ b/vendor/github.com/containers/common/pkg/config/default.go
@@ -113,9 +113,9 @@ const (
 	DefaultPidsLimit = 2048
 	// DefaultPullPolicy pulls the image if it does not exist locally
 	DefaultPullPolicy = "missing"
-	// DefaultSignaturePolicyPath is the default value for the
+	// defaultSignaturePolicyPath is the default value for the
 	// policy.json file.
-	DefaultSignaturePolicyPath = "/etc/containers/policy.json"
+	defaultSignaturePolicyPath = "/etc/containers/policy.json"
 	// DefaultRootlessSignaturePolicyPath is the location within
 	// XDG_CONFIG_HOME of the rootless policy.json file.
 	DefaultRootlessSignaturePolicyPath = "containers/policy.json"
@@ -143,7 +143,8 @@ func DefaultConfig() (*Config, error) {
 
 	cniConfig := _cniConfigDir
 
-	defaultEngineConfig.SignaturePolicyPath = DefaultSignaturePolicyPath
+	defaultSigPath := FallbackToPathInPrefix(defaultSignaturePolicyPath)
+	defaultEngineConfig.SignaturePolicyPath = defaultSigPath
 	if unshare.IsRootless() {
 		configHome, err := homedir.GetConfigHome()
 		if err != nil {
@@ -152,8 +153,8 @@ func DefaultConfig() (*Config, error) {
 		sigPath := filepath.Join(configHome, DefaultRootlessSignaturePolicyPath)
 		defaultEngineConfig.SignaturePolicyPath = sigPath
 		if _, err := os.Stat(sigPath); err != nil {
-			if _, err := os.Stat(DefaultSignaturePolicyPath); err == nil {
-				defaultEngineConfig.SignaturePolicyPath = DefaultSignaturePolicyPath
+			if _, err := os.Stat(defaultSigPath); err == nil {
+				defaultEngineConfig.SignaturePolicyPath = defaultSigPath
 			}
 		}
 		netns = "slirp4netns"
diff --git a/vendor/github.com/containers/image/v5/docker/lookaside.go b/vendor/github.com/containers/image/v5/docker/lookaside.go
index 06d616d0..d8c4dfd3 100644
--- a/vendor/github.com/containers/image/v5/docker/lookaside.go
+++ b/vendor/github.com/containers/image/v5/docker/lookaside.go
@@ -9,6 +9,7 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/containers/common/pkg/config"
 	"github.com/containers/image/v5/docker/reference"
 	"github.com/containers/image/v5/internal/rootless"
 	"github.com/containers/image/v5/types"
@@ -107,7 +108,7 @@ func registriesDirPath(sys *types.SystemContext) string {
 		return filepath.Join(sys.RootForImplicitAbsolutePaths, systemRegistriesDirPath)
 	}
 
-	return systemRegistriesDirPath
+	return config.FallbackToPathInPrefix(systemRegistriesDirPath)
 }
 
 // builtinDefaultSignatureStorageDir returns default signature storage URL as per euid
diff --git a/vendor/github.com/containers/image/v5/pkg/sysregistriesv2/system_registries_v2.go b/vendor/github.com/containers/image/v5/pkg/sysregistriesv2/system_registries_v2.go
index ea2b2157..a3c5f343 100644
--- a/vendor/github.com/containers/image/v5/pkg/sysregistriesv2/system_registries_v2.go
+++ b/vendor/github.com/containers/image/v5/pkg/sysregistriesv2/system_registries_v2.go
@@ -10,6 +10,7 @@ import (
 	"sync"
 
 	"github.com/BurntSushi/toml"
+	"github.com/containers/common/pkg/config"
 	"github.com/containers/image/v5/docker/reference"
 	"github.com/containers/image/v5/types"
 	"github.com/containers/storage/pkg/homedir"
@@ -389,7 +390,7 @@ func newConfigWrapper(ctx *types.SystemContext) configWrapper {
 	} else if ctx != nil && ctx.RootForImplicitAbsolutePaths != "" {
 		wrapper.configPath = filepath.Join(ctx.RootForImplicitAbsolutePaths, systemRegistriesConfPath)
 	} else {
-		wrapper.configPath = systemRegistriesConfPath
+		wrapper.configPath = config.FallbackToPathInPrefix(systemRegistriesConfPath)
 	}
 
 	// potentially use both system and per-user dirs if not using per-user config file
@@ -400,7 +401,7 @@ func newConfigWrapper(ctx *types.SystemContext) configWrapper {
 		wrapper.configDirPath = filepath.Join(ctx.RootForImplicitAbsolutePaths, systemRegistriesConfDirPath)
 		wrapper.userConfigDirPath = userRegistriesDirPath
 	} else {
-		wrapper.configDirPath = systemRegistriesConfDirPath
+		wrapper.configDirPath = config.FallbackToPathInPrefix(systemRegistriesConfDirPath)
 		wrapper.userConfigDirPath = userRegistriesDirPath
 	}
 
diff --git a/vendor/github.com/containers/image/v5/signature/policy_config.go b/vendor/github.com/containers/image/v5/signature/policy_config.go
index a4873e9f..d9331b0b 100644
--- a/vendor/github.com/containers/image/v5/signature/policy_config.go
+++ b/vendor/github.com/containers/image/v5/signature/policy_config.go
@@ -20,6 +20,7 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/containers/common/pkg/config"
 	"github.com/containers/image/v5/docker/reference"
 	"github.com/containers/image/v5/transports"
 	"github.com/containers/image/v5/types"
@@ -68,7 +69,7 @@ func defaultPolicyPath(sys *types.SystemContext) string {
 	if sys != nil && sys.RootForImplicitAbsolutePaths != "" {
 		return filepath.Join(sys.RootForImplicitAbsolutePaths, systemDefaultPolicyPath)
 	}
-	return systemDefaultPolicyPath
+	return config.FallbackToPathInPrefix(systemDefaultPolicyPath)
 }
 
 // NewPolicyFromFile returns a policy configured in the specified file.
