diff --git a/Makefile.in.orig b/Makefile.in
index 1faabd7..98203d6 100644
--- a/Makefile.in.orig
+++ b/Makefile.in
@@ -202,17 +202,17 @@ install.inc:
 		    $(RM) $(AFTER_INC_DIR)/*; \
 		 else \
 		    if [ -d $(INCLUDEDIR) ]; then \
-		       echo "$(MKDIR) $(AFTER_INC_DIR)"; \
-		       if $(MKDIR) $(AFTER_INC_DIR); then \
+		       echo "$(MKDIR) -p $(AFTER_INC_DIR)"; \
+		       if $(MKDIR) -p $(AFTER_INC_DIR); then \
 		          echo " ">/dev/null; \
 		       else \
 		    	  echo "failed to create include directory: $(AFTER_INC_DIR)"; \
 		       fi; \
 		    else \
-		       echo "$(MKDIR) $(INCLUDEDIR)"; \
-		       if $(MKDIR) $(INCLUDEDIR); then \
-		          echo "$(MKDIR) $(AFTER_INC_DIR)"; \
-		          if $(MKDIR) $(AFTER_INC_DIR) >/dev/null; then \
+		       echo "$(MKDIR) -p $(INCLUDEDIR)"; \
+		       if $(MKDIR) -p $(INCLUDEDIR); then \
+		          echo "$(MKDIR) -p $(AFTER_INC_DIR)"; \
+		          if $(MKDIR) -p $(AFTER_INC_DIR) >/dev/null; then \
 		             echo " ">/dev/null; \
 		          else \
 		             echo "failed to create include directory: $(AFTER_INC_DIR)"; \
