diff -Naur python-magic-0.4.13/magic.py python-magic-0.4.13-new/magic.py
--- python-magic-0.4.13/magic.py	2017-03-20 01:23:50.000000000 -0500
+++ python-magic-0.4.13-new/magic.py	2017-03-26 21:04:47.000000000 -0500
@@ -158,7 +158,8 @@
     platform_to_lib = {'darwin': ['/opt/local/lib/libmagic.dylib',
                                   '/usr/local/lib/libmagic.dylib'] +
                          # Assumes there will only be one version installed
-                         glob.glob('/usr/local/Cellar/libmagic/*/lib/libmagic.dylib'),
+                         glob.glob('/usr/local/Cellar/libmagic/*/lib/libmagic.dylib') +
+                         glob.glob(os.path.expandvars('$CONDA_PREFIX/lib/libmagic.dylib')),
                        'win32': windows_dlls,
                        'cygwin': windows_dlls,
                        'linux': ['libmagic.so.1'],    # fallback for some Linuxes (e.g. Alpine) where library search does not work
