diff --git a/setup.py b/setup.py
index fa659413..388bb023 100644
--- a/setup.py
+++ b/setup.py
@@ -325,10 +325,29 @@ EXT_MODULES = [setuptools.Extension(name="onnx.onnx_cpp2py_export", sources=[])]
 ################################################################################
 
 setuptools.setup(
-    ext_modules=EXT_MODULES,
-    cmdclass=CMD_CLASS,
+    name="onnx",
     version=VERSION_INFO["version"],
-    options={"bdist_wheel": {"plat_name": ONNX_WHEEL_PLATFORM_NAME}}
-    if ONNX_WHEEL_PLATFORM_NAME is not None
-    else {},
+    description="Open Neural Network Exchange",
+    long_description=pathlib.Path("README.md").read_text(),
+    long_description_content_type="text/markdown",
+    ext_modules=EXT_MODULES,
+    cmdclass=CMDCLASS,
+    packages=packages,
+    license="Apache License v2.0",
+    include_package_data=True,
+    package_data={"onnx": ["py.typed", "*.pyi"]},
+    install_requires=install_requires,
+    setup_requires=setup_requires,
+    tests_require=tests_require,
+    extras_require=extras_require,
+    author="ONNX",
+    author_email="onnx-technical-discuss@lists.lfaidata.foundation",
+    url="https://github.com/onnx/onnx",
+    entry_points={
+        "console_scripts": [
+            "check-model = onnx.bin.checker:check_model",
+            "check-node = onnx.bin.checker:check_node",
+            "backend-test-tools = onnx.backend.test.cmd_tools:main",
+        ]
+    },
 )
