diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0372c54b..dc5cd524 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,17 +55,6 @@ else()
   set(Protobuf_USE_STATIC_LIBS ON)
 endif()
 
-# Required to use C++17 or higher on Windows
-# For other platforms, set C++14 as standard for the whole project
-if(NOT DEFINED CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
-endif()
-if(MSVC AND CMAKE_CXX_STANDARD LESS 17)
-  set(CMAKE_CXX_STANDARD 17)
-elseif(CMAKE_CXX_STANDARD LESS 14)
-  set(CMAKE_CXX_STANDARD 14)
-endif()
-
 include(GNUInstallDirs)
 
 set(ONNX_ROOT ${PROJECT_SOURCE_DIR})
