diff --git a/pyproject.toml b/pyproject.toml
index c5be759..f52f2e6 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,3 @@
 [build-system]
 # Minimum requirements for the build system to execute.
-requires = ["setuptools", "wheel", "numpy", "cython", "lalsuite"]
+requires = ["setuptools", "numpy", "cython"]
diff --git a/setup.cfg b/setup.cfg
index a95139e..48251d5 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -21,7 +21,6 @@ classifiers =
 setup_requires = setuptools >= 30.3.0
 install_requires =
 	h5py
-	lalsuite
 	lscsoft-glue
 	matplotlib
 	numpy
