Subject: [PATCH] remove hardcoded osx sysroot and deployment target
---
Index: cmake/pre_checks.cmake
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cmake/pre_checks.cmake b/cmake/pre_checks.cmake
--- a/cmake/pre_checks.cmake	(revision 17a2a88f19fc872ade4e557c2ab9636ec17a161e)
+++ b/cmake/pre_checks.cmake	(revision 03aef299b4a61103a92440b1b10c9ee6625f8544)
@@ -1,6 +1,6 @@
 if (APPLE)
-    set(CMAKE_OSX_SYSROOT "/Users/runner/work/MacOSX10.15.sdk" CACHE PATH "macOS SDK path" FORCE)
-    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "macOS deployment target" FORCE)
+#    set(CMAKE_OSX_SYSROOT "/Users/runner/work/MacOSX10.15.sdk" CACHE PATH "macOS SDK path" FORCE)
+#    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "macOS deployment target" FORCE)
 
     if (DEFINED CMAKE_OSX_SYSROOT)
         message(STATUS "Setting macOS sysroot to ${CMAKE_OSX_SYSROOT}")
