diff --git a/src/python/test/graph/test_grid_graph.py b/src/python/test/graph/test_grid_graph.py
index d33d9119..22e9f675 100644
--- a/src/python/test/graph/test_grid_graph.py
+++ b/src/python/test/graph/test_grid_graph.py
@@ -26,9 +26,9 @@ class TestUndirectedGridGraph(unittest.TestCase):
                 self.assertEqual(u, counter)
                 coord = g.nodeToCoordinate(u)
 
-                self.assertEqual(g.coordianteToNode(coord),u)
-                self.assertEqual(g.coordianteToNode(coord),counter)
-                self.assertEqual(g.coordianteToNode([x0,x1]),counter)
+                self.assertEqual(g.coordinateToNode(coord),u)
+                self.assertEqual(g.coordinateToNode(coord),counter)
+                self.assertEqual(g.coordinateToNode([x0,x1]),counter)
 
 
                 self.assertEqual(coord[0], x0)
@@ -57,7 +57,7 @@ class TestUndirectedGridGraph(unittest.TestCase):
         self.assertEqual(eCounter,g.numberOfEdges)
         for node in g.nodes():
             coord = g.nodeToCoordinate(node)
-            self.assertEqual(g.coordianteToNode(coord),node)
+            self.assertEqual(g.coordinateToNode(coord),node)
 
         for edge in g.nodes():
             uv = g.uv(edge)
