diff --git a/Cargo.toml b/Cargo.toml
index 8c0ff83..b9e42d2 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,14 +1,14 @@
 
-[patch.crates-io]
-dynamic-dispatch-proc-macro = { path = "libs-crates/dynamic-dispatch-rs/dynamic-dispatch-proc-macro" }
-dynamic-dispatch = { path = "libs-crates/dynamic-dispatch-rs" }
-mt-debug-counters = { path = "libs-crates/mt-debug-counters-rs" }
-streaming-libdeflate-rs = { path = "libs-crates/streaming-libdeflate-rs" }
-papi-bindings = { path = "libs-crates/papi-bindings-rs" }
-nightly-quirks = { path = "libs-crates/nightly-quirks-rs" }
-instrumenter-proc-macro = { path = "libs-crates/instrumenter-rs/instrumenter-proc-macro-rs" }
-instrumenter = { path = "libs-crates/instrumenter-rs" }
-parallel-processor = { path = "libs-crates/parallel-processor-rs" }
+#[patch.crates-io]
+#dynamic-dispatch-proc-macro = { path = "libs-crates/dynamic-dispatch-rs/dynamic-dispatch-proc-macro" }
+#dynamic-dispatch = { path = "libs-crates/dynamic-dispatch-rs" }
+#mt-debug-counters = { path = "libs-crates/mt-debug-counters-rs" }
+#streaming-libdeflate-rs = { path = "libs-crates/streaming-libdeflate-rs" }
+#papi-bindings = { path = "libs-crates/papi-bindings-rs" }
+#nightly-quirks = { path = "libs-crates/nightly-quirks-rs" }
+#instrumenter-proc-macro = { path = "libs-crates/instrumenter-rs/instrumenter-proc-macro-rs" }
+#instrumenter = { path = "libs-crates/instrumenter-rs" }
+#parallel-processor = { path = "libs-crates/parallel-processor-rs" }
 
 
 #ggat_cmdline = { path = "crates/cmdline" }
@@ -32,15 +32,15 @@ resolver = "2"
 
 members = [
     # Library crates added as submodules
-    "libs-crates/dynamic-dispatch-rs/dynamic-dispatch-proc-macro",
-    "libs-crates/dynamic-dispatch-rs",
-    "libs-crates/mt-debug-counters-rs",
-    "libs-crates/streaming-libdeflate-rs",
-    "libs-crates/papi-bindings-rs",
-    "libs-crates/nightly-quirks-rs",
-    "libs-crates/instrumenter-rs/instrumenter-proc-macro-rs",
-    "libs-crates/instrumenter-rs",
-    "libs-crates/parallel-processor-rs",
+    #"libs-crates/dynamic-dispatch-rs/dynamic-dispatch-proc-macro",
+    #"libs-crates/dynamic-dispatch-rs",
+    #"libs-crates/mt-debug-counters-rs",
+    #"libs-crates/streaming-libdeflate-rs",
+    #"libs-crates/papi-bindings-rs",
+    #"libs-crates/nightly-quirks-rs",
+    #"libs-crates/instrumenter-rs/instrumenter-proc-macro-rs",
+    #"libs-crates/instrumenter-rs",
+    #"libs-crates/parallel-processor-rs",
 
     # Main crate for the command line tools
     "crates/cmdline",
