--- a/python/pyproject.toml
+++ b/python/pyproject.toml
@@ -4,7 +4,7 @@ requires = ["scikit-build-core >=0.4.3", "nanobind >=1.3.2"]
 
 [project]
 name = "voyager"
-version = "2.0.10"
+version = "2.1.1"
 description = "Easy-to-use, fast, simple multi-platform approximate nearest-neighbor search library."
 readme = "README.md"
 requires-python = ">=3.8"
--- a/python/src/PythonInputStream.h
+++ b/python/src/PythonInputStream.h
@@ -18,7 +18,7 @@
 #include <optional>
 
 #include "PythonFileLike.h"
-#include "cpp/src/StreamUtils.h"
+#include "../../cpp/src/StreamUtils.h"
 
 namespace nb = nanobind;
 
--- a/python/src/PythonOutputStream.h
+++ b/python/src/PythonOutputStream.h
@@ -18,7 +18,7 @@
 #include <optional>
 
 #include "PythonFileLike.h"
-#include "cpp/src/StreamUtils.h"
+#include "../../cpp/src/StreamUtils.h"
 
 namespace nb = nanobind;
 
--- a/python/src/bindings.cpp
+++ b/python/src/bindings.cpp
@@ -32,7 +32,7 @@
 
 #include "PythonInputStream.h"
 #include "PythonOutputStream.h"
-#include "cpp/src/TypedIndex.h"
+#include "../../cpp/src/TypedIndex.h"
 
 namespace nb = nanobind;
 using namespace nanobind::literals; // needed to bring in _a literal
@@ -206,7 +206,7 @@ template <typename dist_t, typename data_t,
 NB_MODULE(voyager_ext, m) {
   nb::exception<RecallError>(m, "RecallError");
 
-  m.attr("version") = nb::make_tuple(2, 1, 0);
+  m.attr("version") = nb::make_tuple(2, 1, 1);
 
   init_LabelSetView(m);
 
