Avoid setenv(3) with zero-length name parameter.
--- tiff.c.orig
+++ tiff.c
@@ -609,7 +609,7 @@ tvips_ctime_tiff(
      * where the image was written.  According to Peter Sparlinek, the
      * timestamp is nowadays always recorded in localtime.
      */
-    if (zone != NULL) {
+    if (zone != NULL && zone[0] != '\0') {
         zone_old = getenv("TZ");
         if (setenv("TZ", zone, 1) != 0)
             return (-1);
@@ -679,7 +679,7 @@ tvips_ctime_tiff(
 
     /* Reset the timezone if it was changed.
      */
-    if (zone != NULL) {
+    if (zone != NULL && zone[0] != '\0') {
         if (zone_old != NULL)
             setenv("TZ", zone_old, 1);
         else
