--- Makefile.orig
+++ Makefile
@@ -20,20 +20,11 @@
 .SUFFIXES: .fo .pdf
 
 #CPPFLAGS += -DHAVE_GETOPT_H -DHAVE_GETOPT_LONG
-CPPFLAGS += -DGIT_BRANCH=$(git_branch) -DGIT_COMMIT=$(git_commit)
 CFLAGS += -Wall
 CXXFLAGS += -Wall
 xsl_ns = http://docbook.sourceforge.net/release/xsl-ns/current
 
 
-# XXX The commands to generate the GIT_BRANCH and GIT_COMMIT
-# preprocessor definitions on a git installation and require make(1)
-# to be run from a local repository.  Once the repository contains
-# some tags, see GIT-VERSION-GEN from the git sources.
-git_branch = "master"
-git_commit = "6d546f9"
-
-
 all: dm2smv tiff2smv tvips2smv
 
 dm2smv: delta_t.o dm2smv.o frame.o util.o
--- dm2smv.c.orig
+++ dm2smv.c
@@ -1412,7 +1412,7 @@ version()
 {
     extern char *__progname;
     fprintf(stdout,
-            "%s (TVIPS tools) " xstr(GIT_BRANCH) "." xstr(GIT_COMMIT) "\n",
+            "%s (TVIPS tools) " xstr(PACKAGE_VERSION) "\n",
             __progname);
 }
 
--- tiff2smv.c.orig
+++ tiff2smv.c
@@ -119,7 +119,7 @@ version()
 {
     extern char *__progname;
     fprintf(stdout,
-            "%s (TVIPS tools) " xstr(GIT_BRANCH) "." xstr(GIT_COMMIT) "\n",
+            "%s (TVIPS tools) " xstr(PACKAGE_VERSION) "\n",
             __progname);
 }
 
--- tvips2smv.c.orig
+++ tvips2smv.c
@@ -550,7 +550,7 @@ version()
 {
     extern char *__progname;
     fprintf(stdout,
-            "%s (TVIPS tools) " xstr(GIT_BRANCH) "." xstr(GIT_COMMIT) "\n",
+            "%s (TVIPS tools) " xstr(PACKAGE_VERSION) "\n",
             __progname);
 }
 
