From 26b8cc08324cb93554a112d699709267dc059841 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Sun, 26 Jul 2026 21:32:52 +0200
Subject: [PATCH] patch opencv

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 3352eff..7cfbbf7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -33,7 +33,7 @@ dependencies = [
      "scipy==1.15.2",
      "comm==0.2.2",
      "ipywidgets>=7.0.0",
-     "opencv-python-headless==4.11.0.86",
+     "opencv-python-headless==4.11.0",
      "matplotlib==3.10.1",
      "chaldene-super-resolution",
      "im2im",
