From df459315cd8efed65b4da346fb15904fe2922999 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 11 Feb 2022 21:12:39 +1100
Subject: [PATCH 1/4] allow setting CUDA_ARCHES through cmake itself

---
 CMakeLists.txt | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 980d428..9cbeced 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,12 +25,12 @@ find_package(pybind11 REQUIRED)
 # CUDA_ROOT & CUDA_ARCH
 get_filename_component(CUDA_ROOT ${CMAKE_CUDA_COMPILER} DIRECTORY)
 get_filename_component(CUDA_ROOT ${CUDA_ROOT} DIRECTORY)
-if (NOT ALL_ARCH)
-    cuda_select_nvcc_arch_flags(CUDA_ARCH Auto)
-else()
-    cuda_select_nvcc_arch_flags(CUDA_ARCH 3.5 5.0 6.0 7.0)
-endif()
-string(REPLACE ";" " " CUDA_ARCH "${CUDA_ARCH}")
+# if (NOT ALL_ARCH)
+#     cuda_select_nvcc_arch_flags(CUDA_ARCH Auto)
+# else()
+#     cuda_select_nvcc_arch_flags(CUDA_ARCH 3.5 5.0 6.0 7.0)
+# endif()
+# string(REPLACE ";" " " CUDA_ARCH "${CUDA_ARCH}")
 
 # add externals
 include(ExternalProject)
@@ -125,7 +125,7 @@ set(CMAKE_CXX_FLAGS_RELEASE "-Ofast -D NDEBUG")
 set(CMAKE_CUDA_FLAGS "-std=c++11")
 set(CMAKE_CUDA_FLAGS_DEBUG "-g -G")
 set(CMAKE_CUDA_FLAGS_RELEASE "-O3")
-set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} ${CUDA_ARCH}")
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}")
 
 foreach (MACRO FAST_COMPILE NO_FAISS)
     if (${${MACRO}})
-- 
2.32.0.windows.2

