From 6e05e83fd8465ff52747975b15dfe5a78e1e34d6 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 11 Feb 2022 21:13:06 +1100
Subject: [PATCH 2/4] compile with c++17

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9cbeced..28f8b7c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,11 +118,11 @@ if (NOT CMAKE_BUILD_TYPE)
 endif()
 
 # compilation flags
-set(CMAKE_CXX_FLAGS "-std=c++11 -Wall")
+set(CMAKE_CXX_FLAGS "-std=c++17 -Wall")
 set(CMAKE_CXX_FLAGS_DEBUG "-g -Og")
 set(CMAKE_CXX_FLAGS_RELEASE "-Ofast -D NDEBUG")
 
-set(CMAKE_CUDA_FLAGS "-std=c++11")
+set(CMAKE_CUDA_FLAGS "-std=c++17")
 set(CMAKE_CUDA_FLAGS_DEBUG "-g -G")
 set(CMAKE_CUDA_FLAGS_RELEASE "-O3")
 set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}")
-- 
2.32.0.windows.2

