From b53c419a8b30f0633f616b37132bc5821d8e0659 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 12 Feb 2022 23:29:22 +1100
Subject: [PATCH 4/4] add install step

---
 src/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5d32062..1061253 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,4 +10,6 @@ else ()
 
     target_link_libraries(graphvite pthread curand glog.so libfaiss.so)
     target_compile_options(graphvite PRIVATE "-Xcompiler=-fno-fat-lto-objects") # -flto
-endif ()
\ No newline at end of file
+endif ()
+
+install(TARGETS graphvite LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-- 
2.32.0.windows.2

