--- a/mklib
+++ b/mklib
@@ -34,7 +34,7 @@
 PATCH=""
 DEPS=""
 LINK=""
-LDFLAGS=""
+LDFLAGS=${LDFLAGS}
 CPLUSPLUS=0
 STATIC=0
 DLOPEN=0
@@ -210,11 +210,11 @@

 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
-            if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
-            else
-                LINK=gcc
-            fi
+        if [ $CPLUSPLUS = 1 ] ; then
+            LINK=${CXX}
+        else
+            LINK=${CC}
+        fi
 	fi

 	if [ $NOPREFIX = 1 ] ; then
@@ -615,7 +615,7 @@
         if [ $STATIC = 1 ] ; then
             LIBNAME="lib${LIBNAME}.a"
             echo "mklib: Making Darwin static library: " ${LIBNAME}
-            LINK="ar"
+            LINK=${AR}
             OPTS="-ruvs"
             if [ "${ALTOPTS}" ] ; then
                 OPTS=${ALTOPTS}
@@ -668,9 +668,9 @@

 	    # determine linker
 	    if [ $CPLUSPLUS = 1 ] ; then
-		LINK="g++"
+		LINK=${CXX}
 	    else
-		LINK="cc"
+		LINK=${CC}
 	    fi

             echo "mklib: Making Darwin shared library: " ${LIBNAME}
