From 89d05671dff11d52dd135d188e8b2149175614b7 Mon Sep 17 00:00:00 2001
From: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>
Date: Thu, 5 May 2022 10:39:31 +0200
Subject: [PATCH] Don't include docs in discovered packages

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 3713cbc90..c2c22e0c9 100644
--- a/setup.py
+++ b/setup.py
@@ -76,7 +76,7 @@
         "Topic :: Scientific/Engineering",
     ],
     keywords="qiskit sdk quantum",
-    packages=setuptools.find_namespace_packages(exclude=['test*']),
+    packages=setuptools.find_namespace_packages(exclude=['test*', 'docs']),
     extras_require={
         'visualization': ['matplotlib>=2.1'],
         'cvx': ['cvxpy>=1.0.15'],
