--- setup.py.old	2019-06-03 12:13:23.343081160 -0500
+++ setup.py	2019-06-03 12:14:05.069115376 -0500
@@ -52,8 +52,6 @@
     extra_link_args = ['-lcblas']
     dist = platform.linux_distribution(full_distribution_name=0)[0]
     redhat_dists = set(["redhat", "fedora", "centos"])
-    if dist in redhat_dists:
-        extra_link_args = ['-lsatlas']
     
     # LINUX
     ext_modules = [Extension(name='bh_sne',
