=== modified file 'src/cas/io/bsdSocket/casIntfIO.cc'
--- src/cas/io/bsdSocket/casIntfIO.cc	2010-08-10 21:05:46 +0000
+++ src/cas/io/bsdSocket/casIntfIO.cc	2017-01-17 00:13:23 +0000
@@ -60,8 +60,9 @@
 	status = bind ( this->sock,
                       reinterpret_cast <sockaddr *> (&this->addr),
                       sizeof(this->addr) );
 	if (status<0) {
-		if (SOCKERRNO == SOCK_EADDRINUSE) {
+		if (SOCKERRNO == SOCK_EADDRINUSE ||
+		    SOCKERRNO == SOCK_EACCES) {
 			//
 			// enable assignment of a default port
 			// (so the getsockname() call below will

=== modified file 'src/libCom/osi/os/Darwin/osdSock.h'
--- src/libCom/osi/os/Darwin/osdSock.h	2016-05-22 03:43:09 +0000
+++ src/libCom/osi/os/Darwin/osdSock.h	2017-01-18 17:09:52 +0000
@@ -38,6 +38,7 @@
 #define SOCK_ENOBUFS ENOBUFS
 #define SOCK_ECONNRESET ECONNRESET
 #define SOCK_ETIMEDOUT ETIMEDOUT
+#define SOCK_EACCES EACCES
 #define SOCK_EADDRINUSE EADDRINUSE
 #define SOCK_ECONNREFUSED ECONNREFUSED
 #define SOCK_ECONNABORTED ECONNABORTED

=== modified file 'src/libCom/osi/os/Linux/osdSock.h'
--- src/libCom/osi/os/Linux/osdSock.h	2011-09-09 22:14:35 +0000
+++ src/libCom/osi/os/Linux/osdSock.h	2017-01-18 17:10:40 +0000
@@ -53,6 +53,7 @@
 #define SOCK_ENOBUFS ENOBUFS
 #define SOCK_ECONNRESET ECONNRESET
 #define SOCK_ETIMEDOUT ETIMEDOUT
+#define SOCK_EACCES EACCES
 #define SOCK_EADDRINUSE EADDRINUSE
 #define SOCK_ECONNREFUSED ECONNREFUSED
 #define SOCK_ECONNABORTED ECONNABORTED

=== modified file 'src/libCom/osi/os/RTEMS/osdSock.h'
--- src/libCom/osi/os/RTEMS/osdSock.h	2016-05-22 03:43:09 +0000
+++ src/libCom/osi/os/RTEMS/osdSock.h	2017-01-18 17:10:44 +0000
@@ -49,6 +49,7 @@
 #define SOCK_ENOBUFS ENOBUFS
 #define SOCK_ECONNRESET ECONNRESET
 #define SOCK_ETIMEDOUT ETIMEDOUT
+#define SOCK_EACCES EACCES
 #define SOCK_EADDRINUSE EADDRINUSE
 #define SOCK_ECONNREFUSED ECONNREFUSED
 #define SOCK_ECONNABORTED ECONNABORTED

=== modified file 'src/libCom/osi/os/WIN32/osdSock.h'
--- src/libCom/osi/os/WIN32/osdSock.h	2011-09-09 22:14:35 +0000
+++ src/libCom/osi/os/WIN32/osdSock.h	2017-01-18 17:22:44 +0000
@@ -48,6 +48,7 @@
 #define SOCK_ENOBUFS WSAENOBUFS
 #define SOCK_ECONNRESET WSAECONNRESET
 #define SOCK_ETIMEDOUT WSAETIMEDOUT
+#define SOCK_EACCES WSAEACCES
 #define SOCK_EADDRINUSE WSAEADDRINUSE
 #define SOCK_ECONNREFUSED WSAECONNREFUSED
 #define SOCK_ECONNABORTED WSAECONNABORTED

=== modified file 'src/libCom/osi/os/cygwin32/osdSock.h'
--- src/libCom/osi/os/cygwin32/osdSock.h	2015-04-29 19:01:07 +0000
+++ src/libCom/osi/os/cygwin32/osdSock.h	2017-01-18 17:09:37 +0000
@@ -49,6 +49,7 @@
 #define SOCK_ENOBUFS ENOBUFS
 #define SOCK_ECONNRESET ECONNRESET
 #define SOCK_ETIMEDOUT ETIMEDOUT
+#define SOCK_EACCES EACCES
 #define SOCK_EADDRINUSE EADDRINUSE
 #define SOCK_ECONNREFUSED ECONNREFUSED
 #define SOCK_ECONNABORTED ECONNABORTED

=== modified file 'src/libCom/osi/os/freebsd/osdSock.h'
--- src/libCom/osi/os/freebsd/osdSock.h	2011-09-09 22:14:35 +0000
+++ src/libCom/osi/os/freebsd/osdSock.h	2017-01-18 17:10:32 +0000
@@ -43,6 +43,7 @@
 #define SOCK_ENOBUFS ENOBUFS
 #define SOCK_ECONNRESET ECONNRESET
 #define SOCK_ETIMEDOUT ETIMEDOUT
+#define SOCK_EACCES EACCES
 #define SOCK_EADDRINUSE EADDRINUSE
 #define SOCK_ECONNREFUSED ECONNREFUSED
 #define SOCK_ECONNABORTED ECONNABORTED

=== modified file 'src/libCom/osi/os/iOS/osdSock.h'
--- src/libCom/osi/os/iOS/osdSock.h	2016-05-22 03:43:09 +0000
+++ src/libCom/osi/os/iOS/osdSock.h	2017-01-18 17:10:37 +0000
@@ -39,6 +39,7 @@
 #define SOCK_ENOBUFS ENOBUFS
 #define SOCK_ECONNRESET ECONNRESET
 #define SOCK_ETIMEDOUT ETIMEDOUT
+#define SOCK_EACCES EACCES
 #define SOCK_EADDRINUSE EADDRINUSE
 #define SOCK_ECONNREFUSED ECONNREFUSED
 #define SOCK_ECONNABORTED ECONNABORTED

=== modified file 'src/libCom/osi/os/solaris/osdSock.h'
--- src/libCom/osi/os/solaris/osdSock.h	2011-09-09 22:14:35 +0000
+++ src/libCom/osi/os/solaris/osdSock.h	2017-01-18 17:22:37 +0000
@@ -51,6 +51,7 @@
 #define SOCK_ENOBUFS ENOBUFS
 #define SOCK_ECONNRESET ECONNRESET
 #define SOCK_ETIMEDOUT ETIMEDOUT
+#define SOCK_EACCES EACCES
 #define SOCK_EADDRINUSE EADDRINUSE
 #define SOCK_ECONNREFUSED ECONNREFUSED
 #define SOCK_ECONNABORTED ECONNABORTED

=== modified file 'src/libCom/osi/os/vxWorks/osdSock.h'
--- src/libCom/osi/os/vxWorks/osdSock.h	2011-09-09 22:14:35 +0000
+++ src/libCom/osi/os/vxWorks/osdSock.h	2017-01-18 17:22:40 +0000
@@ -72,6 +72,7 @@
 #define SOCK_ENOBUFS ENOBUFS
 #define SOCK_ECONNRESET ECONNRESET
 #define SOCK_ETIMEDOUT ETIMEDOUT
+#define SOCK_EACCES EACCES
 #define SOCK_EADDRINUSE EADDRINUSE
 #define SOCK_ECONNREFUSED ECONNREFUSED
 #define SOCK_ECONNABORTED ECONNABORTED

=== modified file 'src/rsrv/caservertask.c'
--- src/rsrv/caservertask.c	2016-05-22 03:43:09 +0000
+++ src/rsrv/caservertask.c	2017-01-17 00:13:24 +0000
@@ -106,7 +106,8 @@
     /* get server's Internet address */
     status = bind ( IOC_sock, (struct sockaddr *) &serverAddr, sizeof ( serverAddr ) );
 	if ( status < 0 ) {
-		if ( SOCKERRNO == SOCK_EADDRINUSE ) {
+		if ( SOCKERRNO == SOCK_EADDRINUSE ||
+		     SOCKERRNO == SOCK_EACCES ) {
 			/*
 			 * enable assignment of a default port
 			 * (so the getsockname() call below will

