--- configure/CONFIG_COMMON 2017-12-15 23:06:08.000000000 +0100
+++ configure/CONFIG_COMMON 2018-03-08 09:44:15.133490475 +0100
@@ -198,7 +198,7 @@                                                                                      
 OP_SYS_INCLUDES =                                                                                       
                                                                                                         
 # Makefile specific flags                                                                               
-USR_INCLUDES =
+USR_INCLUDES = -I$(PREFIX)/include
 USR_CFLAGS =
 USR_CXXFLAGS =
 USR_LDFLAGS =
@@ -382,10 +382,10 @@

 #---------------------------------------------------------------
 # Installed file permissions
-BIN_PERMISSIONS = 555
-LIB_PERMISSIONS = 444
-SHRLIB_PERMISSIONS = 555
-INSTALL_PERMISSIONS = 444
+BIN_PERMISSIONS = 755
+LIB_PERMISSIONS = 644
+SHRLIB_PERMISSIONS = 755
+INSTALL_PERMISSIONS = 644

 #---------------------------------------------------------------
 #
