--- ../orig/base-3.14.12.4/configure/CONFIG_SITE	2013-12-16 16:56:27.000000000 -0500
+++ configure/CONFIG_SITE	2015-01-28 07:59:09.784000000 -0500
@@ -171,7 +171,7 @@
 #  Installation directory
 # If you don't want to install into $(TOP) dir then
 # define INSTALL_LOCATION here
-#INSTALL_LOCATION=<fullpathname>
+INSTALL_LOCATION=$(PREFIX)/epics

 # Use POSIX thread priority scheduling (YES or NO)
 USE_POSIX_THREAD_PRIORITY_SCHEDULING = NO
