--- configure/os/CONFIG_SITE.Common.linux-x86
+++ configure/os/CONFIG_SITE.Common.linux-x86
@@ -22,7 +22,7 @@
 # comment them all out to build without readline support.
 
 # No other libraries needed (recent Fedora, Ubuntu etc.):
-COMMANDLINE_LIBRARY = READLINE
+#COMMANDLINE_LIBRARY = READLINE
 
 # Needs -lncurses (RHEL 5 etc.):
 #COMMANDLINE_LIBRARY = READLINE_NCURSES

--- configure/os/CONFIG_SITE.Common.linux-x86_64
+++ configure/os/CONFIG_SITE.Common.linux-x86_64
@@ -22,7 +22,7 @@
 # comment them all out to build without readline support.
 
 # No other libraries needed (recent Fedora, Ubuntu etc.):
-COMMANDLINE_LIBRARY = READLINE
+#COMMANDLINE_LIBRARY = READLINE
 
 # Needs -lncurses (RHEL 5 etc.):
 #COMMANDLINE_LIBRARY = READLINE_NCURSES

