From 26ddece8eaac32dae277dca98283b5a90f3340ea Mon Sep 17 00:00:00 2001
From: Pat Schloss <pschloss@umich.edu>
Date: Tue, 27 Sep 2022 17:04:32 -0400
Subject: [PATCH] Force so extension on libconnection

---
 R/zzz.R | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/R/zzz.R b/R/zzz.R
index 30f61ed..e27d9d5 100644
--- a/R/zzz.R
+++ b/R/zzz.R
@@ -1,5 +1,11 @@
 .onLoad <- function(libname, pkgname) {
   lib_path <- system.file("lib", .Platform$r_arch, paste0("libconnection", .Platform$dynlib.ext), package = "archive")
+
+  if(lib_path == ""){
+    lib_path <- system.file("lib", .Platform$r_arch, paste0("libconnection", ".so"), package = 
+"archive")
+  }
+
   res <- dyn.load(lib_path)
 
   rchive_init(res$new_connection$address, res$read_connection$address)
-- 
2.37.0 (Apple Git-136)

