From 4a7cc89109a24ee6730bd2b20ffe97eebabc349d Mon Sep 17 00:00:00 2001
From: Pat Schloss <pschloss@umich.edu>
Date: Tue, 27 Sep 2022 17:04:32 -0400
Subject: [PATCH] Use correct extension on libconnection

---
 R/zzz.R | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/R/zzz.R b/R/zzz.R
index 30f61ed..c18c3dc 100644
--- a/R/zzz.R
+++ b/R/zzz.R
@@ -1,11 +1,10 @@
 .onLoad <- function(libname, pkgname) {
-  lib_path <- system.file("lib", .Platform$r_arch, paste0("libconnection", .Platform$dynlib.ext), package = "archive")
+  lib_path <- list.files(paste0(libname, "/archive/lib"), pattern="libconnection.*", full.names=TRUE)
   res <- dyn.load(lib_path)
-
   rchive_init(res$new_connection$address, res$read_connection$address)
 }
 
 .onUnload <- function(libname) {
-  lib_path <- system.file("lib", .Platform$r_arch, paste0("libconnection", .Platform$dynlib.ext), package = "archive")
+  lib_path <- list.files(paste0(libname, "/archive/lib"), pattern="libconnection.*", full.names=TRUE)
   dyn.unload(lib_path)
 }
-- 
2.37.0 (Apple Git-136)

