--- v4-client-cpp/lib/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -1,7 +1,10 @@
 # Dependencies
-find_package(Boost COMPONENTS beast REQUIRED)
+find_package(Boost 1.82.0 REQUIRED)
-find_package(OpenSSL REQUIRED)
+find_package(OpenSSL 3 REQUIRED)
-find_package(bip3x REQUIRED)
+find_package(bip3x 3 REQUIRED)
-find_package(fmt REQUIRED)
+find_package(fmt REQUIRED)
-find_package(json REQUIRED)
+find_package(nlohmann_json 3.11.3 REQUIRED)
+message("OpenSSL libraries: ${OPENSSL_LIBRARIES} ${OPENSSL_SSL_LIBRARIES} ${OPENSSL_CRYPTO_LIBRARIES} ${OPENSSL_FOUND} ${OPENSSL_ROOT_DIR}")
+message("bip3x libraries: ${bip3x_FOUND} ${bip3x_INCLUDE_DIRS} ${bip3x_LIBRARIES}")
+message("nlohmann_json libraries: ${nlohmann_json_LIBRARIES}")

@@ -14,3 +14,4 @@
 target_include_directories(dydx_v4_client_lib PUBLIC ${INCLUDE_DIR})
-target_link_libraries(dydx_v4_client_lib PUBLIC dydx_v4_proto Boost::beast OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
+target_link_libraries(dydx_v4_client_lib PUBLIC dydx_v4_proto OpenSSL::SSL bip3x fmt nlohmann_json::nlohmann_json)
+message("OpenSSL libraries: ${OPENSSL_LIBRARIES} ${OPENSSL_SSL_LIBRARIES} ${OPENSSL_CRYPTO_LIBRARIES} ${OPENSSL_FOUND} ${OPENSSL_ROOT_DIR}")

