--- v4-client-cpp/lib/proto/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -3,6 +3,16 @@
 file(GLOB_RECURSE proto_files RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.proto)
-protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS ${proto_files})
+if(protobuf_MODULE_COMPATIBLE)
+    # Legacy Support
+    protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS ${proto_files})
+endif()

 add_library(dydx_v4_proto ${BALOO_BIN_TYPE} ${PROTO_SRCS} ${PROTO_HDRS})
-target_include_directories(dydx_v4_proto PUBLIC ${Protobuf_INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(dydx_v4_proto PUBLIC ${Protobuf_LIBRARIES})
\ No newline at end of file
+
+if(protobuf_MODULE_COMPATIBLE)
+    target_include_directories(dydx_v4_proto PUBLIC ${Protobuf_INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR})
+    target_link_libraries(dydx_v4_proto PUBLIC ${Protobuf_LIBRARIES})
+else ()
+    target_sources(dydx_v4_proto PRIVATE ${proto_files})
+    target_link_libraries(dydx_v4_proto PUBLIC protobuf::libprotobuf)
+    protobuf_generate(TARGET dydx_v4_proto)
+endif()
