From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Wed, 25 Oct 2023 15:45:50 -0400
Subject: [PATCH] Use C++ std::complex type for portability

---
 lib/optimize_c.cc | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/lib/optimize_c.cc b/lib/optimize_c.cc
index dd35b6a..45ae569 100644
--- a/lib/optimize_c.cc
+++ b/lib/optimize_c.cc
@@ -11,6 +11,8 @@
 #include "config.h"
 #endif
 
+#include <complex>
+
 #include <gnuradio/io_signature.h>
 #include <gnuradio/iqbalance/optimize_c.h>
 
@@ -20,11 +22,6 @@
 		__GNUC_PATCHLEVEL__	\
 	)
 
-#if GCC_VERSION >= 40800 || defined(__clang__)
-# define complex _Complex
-# undef _GLIBCXX_HAVE_COMPLEX_H
-#endif
-
 extern "C" {
 #include <osmocom/dsp/cxvec.h>
 #include <osmocom/dsp/iqbal.h>
@@ -109,7 +106,7 @@ iqbalance::optimize_c::work(int noutput_items,
 	p[0] = this->d_mag;
 	p[1] = this->d_phase;
 
-	osmo_cxvec_init_from_data(sig, (float complex *)in, N);
+	osmo_cxvec_init_from_data(sig, (std::complex<float> *)in, N);
 	osmo_iqbal_cxvec_optimize(sig, &p[0], &p[1], &opts);
 
 	if (this->d_first) {
-- 
2.40.0

