From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Tue, 24 Oct 2023 12:59:47 -0400
Subject: [PATCH] Build libosmo-dsp sources as C++ with MSVC.

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 567aefb..d91c503 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,6 +102,10 @@ if(NOT LIBOSMODSP_FOUND)
     set(LIBOSMODSP_SEL_INCLUDE_DIRS ${FFTW3f_INCLUDE_DIRS} "${CMAKE_CURRENT_SOURCE_DIR}/libosmo-dsp/include/")
     set(LIBOSMODSP_SEL_LIBRARIES ${FFTW3f_LIBRARIES})
     set(LIBOSMODSP_SEL_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/libosmo-dsp/src/cxvec.c ${CMAKE_CURRENT_SOURCE_DIR}/libosmo-dsp/src/iqbal.c)
+    if(MSVC)
+      # compile the sources as C++ due to non-standard complex types under MSVC
+      set_source_files_properties(${LIBOSMODSP_SEL_SOURCES} PROPERTIES LANGUAGE CXX)
+    endif()
 endif()
 
 ########################################################################
-- 
2.40.0

