diff --git a/sio/DNSUtil.c b/sio/DNSUtil.c
index 0d542bb..fac1e79 100644
--- a/sio/DNSUtil.c
+++ b/sio/DNSUtil.c
@@ -12,11 +12,6 @@
 #	define Strncpy(a,b,s) strncpy(a, b, s); a[s - 1] = '\0' 
 #endif
 
-#if (((defined(MACOSX)) && (MACOSX < 10300)) || (defined(AIX) && (AIX < 430)) || (defined(DIGITAL_UNIX)) || (defined(SOLARIS)) || (defined(SCO)) || (defined(HPUX)))
-extern int getdomainname(char *name, gethostname_size_t namelen);
-#endif
-
-
 int
 GetHostByName(struct hostent *const hp, const char *const name, char *const hpbuf, size_t hpbufsize)
 {
