diff --git a/setup.py b/setup.py
index d63b8d5..e9fff9a 100644
--- a/setup.py
+++ b/setup.py
@@ -38,12 +38,13 @@ from setuptools import setup, find_packages  # Always prefer setuptools over dis
 from os import path
 import io
 
-here = path.abspath(path.dirname(__file__))
 
-# get the log description
-with io.open(path.join(here, "DESCRIPTION.rst"), encoding="utf-8") as f:
-    long_description = f.read()
+long_description = """DE2120_Py
+==========================
 
+This is python package based off of the SparkFun DE2120 Arduino Library.
+It contains functions to read barcodes from the module and set configurations.
+"""
 
 setup(
 
