--- scikits/umfpack/interface.py	2015-09-11 16:19:05.000000000 -0400
+++ scikits/umfpack/interface_new.py	2016-05-27 01:40:48.000000000 -0400
@@ -19,6 +19,7 @@
 from __future__ import division, print_function, absolute_import

 from warnings import warn
+import sys
 import numpy as np
 from numpy import asarray, empty, ravel, nonzero
 from scipy.sparse import (isspmatrix_csc, isspmatrix_csr, isspmatrix,
@@ -26,9 +27,14 @@

 from .umfpack import UmfpackContext, UMFPACK_A

+
 __all__ = ['spsolve', 'splu', 'UmfpackLU']


+if sys.version_info[0] >= 3:
+    xrange = range
+
+
 def spsolve(A, b):
     """Solve the sparse linear system Ax=b, where b may be a vector or a matrix.

@@ -104,6 +110,7 @@
     Methods
     -------
     solve
+    solve_sparse

     Examples
     --------
