--- scikits/umfpack/setup.py	2016-05-27 00:36:10.000000000 -0400
+++ scikits/umfpack/setup_new.py	2016-05-27 00:39:44.000000000 -0400
@@ -11,11 +11,6 @@

     umf_info = get_info('umfpack', notfound_action=1)

-    ## The following addition is needed when linking against a umfpack built
-    ## from the latest SparseSuite. Not (strictly) needed when linking against
-    ## the version in the ubuntu repositories.
-    umf_info['libraries'].insert(0, 'rt')
-
     umfpack_i_file = config.paths('umfpack.i')[0]

     def umfpack_i(ext, build_dir):
