From 2831ad5ee4b8fd1c8162b44b1d396c8b136146dc Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 23 Jul 2024 07:08:16 +0200
Subject: [PATCH] Update setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 397d6a3..6c6b303 100644
--- a/setup.py
+++ b/setup.py
@@ -45,7 +45,7 @@
     # And probably include paths by option: --include-dirs /usr/include/zstd
     # And probably library paths by option: --library-dirs /usr/lib/i386-linux-gnu
     # We need pkg-config here!
-    pkgconf = "/usr/bin/pkg-config"
+    pkgconf = os.path.join(os.environ["BUILD_PREFIX"], "bin", "pkg-config")
     if os.path.exists(pkgconf):
         cmd = [pkgconf, "libzstd", "--modversion"]
         if sys.hexversion >= 0x02070000:
