From 513cb55a1299453c9163a36d86fa078c82abe1aa Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 8 May 2020 20:12:30 +0200
Subject: [PATCH 2/2] remove -fPIC flag for msvc-compat

---
 makefile.inc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile.inc.in b/makefile.inc.in
index 19d8511..1c4e976 100644
--- a/makefile.inc.in
+++ b/makefile.inc.in
@@ -6,7 +6,7 @@
 CXX          = @CXX@
 CXXCPP       = @CXXCPP@
 CPPFLAGS     = -DFINTEGER=int @CPPFLAGS@ @OPENMP_CXXFLAGS@ @NVCC_CPPFLAGS@
-CXXFLAGS     = -fPIC @ARCH_CXXFLAGS@ -Wno-sign-compare @CXXFLAGS@
+CXXFLAGS     = @ARCH_CXXFLAGS@ -Wno-sign-compare @CXXFLAGS@
 CPUFLAGS     = @ARCH_CPUFLAGS@
 LDFLAGS      = @OPENMP_LDFLAGS@ @LDFLAGS@ @NVCC_LDFLAGS@
 LIBS         = @BLAS_LIBS@ @LAPACK_LIBS@ @LIBS@ @NVCC_LIBS@
-- 
2.26.2.windows.1

