diff --git a/setup.cfg b/setup.cfg
index ebc5946..eea1192 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,6 +1,6 @@
 [metadata]
 name = pyfastspm
-version = attr: pyfastspm.version.__version__
+version = 0.11.0
 description = A python package for loading, processing and exporting scanning probe microscopy movies acquired with the FAST unit.
 long_description = file: README.md, CHANGELOG.md, LICENSE
 long_description_content_type = text/markdown
