diff --git a/src/ipopt.pc.in b/src/ipopt.pc.in
index 15a7305f..2220edd3 100644
--- a/src/ipopt.pc.in
+++ b/src/ipopt.pc.in
@@ -10,6 +10,5 @@ URL: @PACKAGE_URL@
 Version: @PACKAGE_VERSION@
 Cflags: -I${includedir}
 @COIN_STATIC_BUILD_FALSE@Libs: -L${libdir} -lipopt
-@COIN_STATIC_BUILD_FALSE@Requires.private: @IPOPTLIB_PCFILES@
 @COIN_STATIC_BUILD_TRUE@Libs: -L${libdir} -lipopt @IPOPTLIB_LFLAGS_NOPC@
 @COIN_STATIC_BUILD_TRUE@Requires: @IPOPTLIB_PCFILES@
