From 15f6ababcf75a29cdd339aee1578a5c474078d7f Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Thu, 13 Dec 2018 15:52:48 +0000
Subject: [PATCH] Fix building clock_gettime for older MacOS SDKs

---
 src/XrdSys/XrdSysPthread.hh | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/src/XrdSys/XrdSysPthread.hh b/src/XrdSys/XrdSysPthread.hh
index 5dec3490fce..e4e0bb751f3 100644
--- a/src/XrdSys/XrdSysPthread.hh
+++ b/src/XrdSys/XrdSysPthread.hh
@@ -41,19 +41,14 @@
 #include <semaphore.h>
 #endif
 
-#ifdef __APPLE__
-#ifndef CLOCK_REALTIME
+#if __APPLE__ && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 101200
 #include <mach/clock.h>
 #include <mach/mach.h>
-#endif
 namespace
 {
   template< typename TYPE >
   void get_apple_realtime( TYPE & wait )
   {
-#ifdef CLOCK_REALTIME
-    clock_gettime(CLOCK_REALTIME, &wait);
-#else
     clock_serv_t cclock;
     mach_timespec_t mts;
     host_get_clock_service(mach_host_self(), CALENDAR_CLOCK, &cclock);
@@ -61,7 +56,6 @@ namespace
     mach_port_deallocate(mach_task_self(), cclock);
     wait.tv_sec  = mts.tv_sec;
     wait.tv_nsec = mts.tv_nsec;
-#endif
   }
 }
 #endif
@@ -170,7 +164,7 @@ inline int CondLock()
        {if (pthread_mutex_trylock( &cs )) return 0;
         return 1;
        }
-#ifdef __APPLE__
+#if __APPLE__ && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 101200
 inline int TimedLock( int wait_ms )
 {
   struct timespec wait, cur, dur;
