From 72a8e3e01effe22ac0f4e29c14153743172efcb5 Mon Sep 17 00:00:00 2001
From: Cheney-Wang <v-xincwa@microsoft.com>
Date: Wed, 31 Oct 2018 19:21:26 -0700
Subject: [PATCH 4/4] Add missing <stdexcept> include to use std::runtime_error

---
 docopt_value.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/docopt_value.h b/docopt_value.h
index a923219..829ee55 100644
--- a/docopt_value.h
+++ b/docopt_value.h
@@ -9,6 +9,7 @@
 #ifndef docopt__value_h_
 #define docopt__value_h_
 
+#include <stdexcept>
 #include <string>
 #include <vector>
 #include <functional> // std::hash
-- 
2.21.0 (Apple Git-122.2)

