Binary files a/cppinclude and b/cppinclude differ
diff -uNr a/src/tools/std_hash_fs_path.hpp b/src/tools/std_hash_fs_path.hpp
--- a/src/tools/std_hash_fs_path.hpp	2021-08-11 02:32:51.000000000 +0800
+++ b/src/tools/std_hash_fs_path.hpp	2025-02-07 14:34:40.966798421 +0800
@@ -6,13 +6,13 @@
 
 namespace std
 {
-template<> struct hash< stdfs::path >
-{
-	std::size_t operator()( const stdfs::path & _file ) const noexcept
-	{
-		return stdfs::hash_value( _file );
-	}
-};
+// template<> struct hash< stdfs::path >
+// {
+// 	std::size_t operator()( const stdfs::path & _file ) const noexcept
+// 	{
+// 		return stdfs::hash_value( _file );
+// 	}
+// };
 }
 
 //------------------------------------------------------------------------------
