Index: weightwatcher/constants.py
===================================================================
--- weightwatcher/constants.py
+++ weightwatcher/constants.py
@@ -263,5 +263,7 @@
 
 
 class XMIN(IntFlag):
     UNKNOWN = auto()
-    AUTO = auto()
\ No newline at end of file
+    AUTO = auto()
+
+DEFAULT_PEFT = False # | True | 'peft' | 'peft_onlu' ## should be removed after latest master is tagged
\ No newline at end of file
