From 267c3314b6ce6298d58dd2161ab3be303e8ca9c6 Mon Sep 17 00:00:00 2001
From: Joshua Downer <joshua.downer@gmail.com>
Date: Tue, 11 Aug 2020 12:47:23 -0400
Subject: [PATCH 3/3] makefile.am: remove installation of udev rules

---
 Makefile.am | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 0e69cd4..4f6c5e7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,11 +25,6 @@ dist_pkgconfig_DATA=libuldaq.pc
 # fpga/USB_CTR.bin\
 # fpga/USB_DIO32HS.bin
 
-if OS_LINUX
-rulesdatadir=/lib/udev/rules.d/
-dist_rulesdata_DATA=rules/50-uldaq.rules
-endif
-
 dist-hook:
 	chmod u+w $(distdir)/examples/*.c
 	chmod u+w $(distdir)/examples/*.h
@@ -41,17 +36,3 @@ dist-up: dist
 	rm -rf $(reldir)
 	mkdir -p $(reldir)
 	cp $(distdir).tar.bz2 $(reldir)
-
-if OS_LINUX
-install-data-hook:
-	udevadm control --reload-rules && \
-	test -f "/etc/redhat-release" && echo "/usr/local/lib" > /etc/ld.so.conf.d/uldaq.conf || echo "" && \
-	test -f "/etc/arch-release" && echo "/usr/local/lib" > /etc/ld.so.conf.d/uldaq.conf || echo "" && \
-	ldconfig
-endif
-
-if OS_LINUX
-uninstall-hook:
-	test -f "/etc/ld.so.conf.d/uldaq.conf" &&  rm "/etc/ld.so.conf.d/uldaq.conf" || echo ""
-endif
-
-- 
2.17.1

