From 7d891b754b0471040b7c61941239217456abbe71 Mon Sep 17 00:00:00 2001
From: Joshua Downer <joshua.downer@gmail.com>
Date: Mon, 10 Aug 2020 21:57:41 -0400
Subject: [PATCH 2/2] examples: add -ludev to LDADD

---
 examples/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/Makefile.am b/examples/Makefile.am
index 3b2d4b7..07001e1 100644
--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -1,5 +1,5 @@
 AM_CPPFLAGS = -I$(top_srcdir)/src
-LDADD = ../src/libuldaq.la
+LDADD = ../src/libuldaq.la -ludev -lusb-1.0
 
 noinst_PROGRAMS = AIn AInScan AInScanWithEvents AInScanWithQueue AInScanWithTrigger AInScanIepe\
 AOut AOutScan\
-- 
2.17.1

