From 690d4bd9f29a805999a3ce4651dac9585ccc9917 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Wed, 11 May 2011 16:46:57 +0200
Subject: [PATCH 1/2] findutils-4.5.7-warnings.patch

---
 xargs/xargs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/xargs/xargs.c b/xargs/xargs.c
index 5e373f2..c0a8676 100644
--- a/xargs/xargs.c
+++ b/xargs/xargs.c
@@ -1289,7 +1289,8 @@ xargs_do_exec (struct buildcmd_control *ctl, void *usercontext, int argc, char *
 		 * utility if we run it, for POSIX compliance on the
 		 * handling of exit values.
 		 */
-		write (fd[1], &errno, sizeof (int));
+		int sink = write (fd[1], &errno, sizeof (int));
+		(void) sink;
 	      }
 
 	    close (fd[1]);
-- 
1.7.1


From c5654b9ca5f50daa1ca406ebd7b4546f24d00db6 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Mon, 23 Sep 2013 15:04:03 +0200
Subject: [PATCH 2/2] parser: silence a [-Wmaybe-uninitialized] GCC warning

... caused by a missing model of error()
---
 find/parser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/find/parser.c b/find/parser.c
index 89d8bcf..8c399d7 100644
--- a/find/parser.c
+++ b/find/parser.c
@@ -2723,7 +2723,7 @@ insert_type (char **argv, int *arg_ptr,
 	     const struct parser_table *entry,
 	     PRED_FUNC which_pred)
 {
-  mode_t type_cell;
+  mode_t type_cell /* to silence GCC warning */ = 0;
   struct predicate *our_pred;
   float rate = 0.5;
   const char *typeletter;
-- 
1.9.3

