--- a/khronos-headers/OpenCL/cl_platform.h
+++ b/khronos-headers/OpenCL/cl_platform.h
@@ -61,7 +61,16 @@ extern "C" {
     #define GCL_API_SUFFIX__VERSION_1_1                 AVAILABLE_MAC_OS_X_VERSION_10_7_AND_LATER
     #define CL_EXT_SUFFIX__VERSION_1_1                  CL_EXTENSION_WEAK_LINK AVAILABLE_MAC_OS_X_VERSION_10_7_AND_LATER
     #define CL_EXT_SUFFIX__VERSION_1_0_DEPRECATED       CL_EXTENSION_WEAK_LINK AVAILABLE_MAC_OS_X_VERSION_10_6_AND_LATER_BUT_DEPRECATED_IN_MAC_OS_X_VERSION_10_7
-    
+
+    #define CL_API_SUFFIX__VERSION_2_0 /* nothing */
+    #define CL_EXT_SUFFIX__VERSION_2_0 /* nothing */
+    #define CL_API_SUFFIX__VERSION_2_1 /* nothing */
+    #define CL_EXT_SUFFIX__VERSION_2_1 /* nothing */
+
+    #define CL_EXT_SUFFIX__VERSION_1_2_DEPRECATED /* nothing */
+    #define CL_EXT_SUFFIX__VERSION_2_0_DEPRECATED /* nothing */
+    #define CL_EXT_PREFIX__VERSION_2_0_DEPRECATED /* nothing */
+
     #ifdef AVAILABLE_MAC_OS_X_VERSION_10_8_AND_LATER
         #define CL_API_SUFFIX__VERSION_1_2              AVAILABLE_MAC_OS_X_VERSION_10_8_AND_LATER
         #define GCL_API_SUFFIX__VERSION_1_2             AVAILABLE_MAC_OS_X_VERSION_10_8_AND_LATER
--- a/khronos-headers/CL/cl_platform.h
+++ b/khronos-headers/CL/cl_platform.h
@@ -61,7 +61,16 @@ extern "C" {
     #define GCL_API_SUFFIX__VERSION_1_1                 AVAILABLE_MAC_OS_X_VERSION_10_7_AND_LATER
     #define CL_EXT_SUFFIX__VERSION_1_1                  CL_EXTENSION_WEAK_LINK AVAILABLE_MAC_OS_X_VERSION_10_7_AND_LATER
     #define CL_EXT_SUFFIX__VERSION_1_0_DEPRECATED       CL_EXTENSION_WEAK_LINK AVAILABLE_MAC_OS_X_VERSION_10_6_AND_LATER_BUT_DEPRECATED_IN_MAC_OS_X_VERSION_10_7
-    
+
+    #define CL_API_SUFFIX__VERSION_2_0 /* nothing */
+    #define CL_EXT_SUFFIX__VERSION_2_0 /* nothing */
+    #define CL_API_SUFFIX__VERSION_2_1 /* nothing */
+    #define CL_EXT_SUFFIX__VERSION_2_1 /* nothing */
+
+    #define CL_EXT_SUFFIX__VERSION_1_2_DEPRECATED /* nothing */
+    #define CL_EXT_SUFFIX__VERSION_2_0_DEPRECATED /* nothing */
+    #define CL_EXT_PREFIX__VERSION_2_0_DEPRECATED /* nothing */
+
     #ifdef AVAILABLE_MAC_OS_X_VERSION_10_8_AND_LATER
         #define CL_API_SUFFIX__VERSION_1_2              AVAILABLE_MAC_OS_X_VERSION_10_8_AND_LATER
         #define GCL_API_SUFFIX__VERSION_1_2             AVAILABLE_MAC_OS_X_VERSION_10_8_AND_LATER
