--- mfi-prev	2016-10-09 14:46:40.660144056 -0500
+++ Makefile.in	2016-10-09 14:47:04.488145193 -0500
@@ -111,7 +111,8 @@
 am__configure_deps = $(am__aclocal_m4_deps) $(CONFIGURE_DEPENDENCIES) \
 	$(ACLOCAL_M4)
 DIST_COMMON = $(srcdir)/Makefile.am $(top_srcdir)/configure \
-	$(am__configure_deps) $(am__DIST_COMMON)
+	$(am__configure_deps) $(clinclude_HEADERS) \
+	$(openclinclude_HEADERS) $(am__DIST_COMMON)
 am__CONFIG_DISTCLEAN_FILES = config.status config.cache config.log \
  configure.lineno config.status.lineno
 mkinstalldirs = $(install_sh) -d
@@ -147,7 +148,8 @@
   }
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(pkglibdir)" \
 	"$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(pkgexampledir)" \
-	"$(DESTDIR)$(includedir)"
+	"$(DESTDIR)$(clincludedir)" "$(DESTDIR)$(includedir)" \
+	"$(DESTDIR)$(openclincludedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES) $(pkglib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 libOpenCL_la_DEPENDENCIES = $(am__DEPENDENCIES_1)
@@ -273,7 +275,8 @@
     *) (install-info --version) >/dev/null 2>&1;; \
   esac
 DATA = $(pkgconfig_DATA) $(pkgexample_DATA)
-HEADERS = $(nodist_include_HEADERS)
+HEADERS = $(clinclude_HEADERS) $(nodist_include_HEADERS) \
+	$(openclinclude_HEADERS)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 am__recursive_targets = \
@@ -521,6 +524,32 @@
 	$(am__append_1)
 libOpenCL_la_LIBADD = $(PTHREAD_LIBS)
 @USE_MAP_TRUE@libOpenCL_la_DEPENDS = ocl_icd_loader_gen.map
+clincludedir = $(includedir)/CL
+clinclude_HEADERS = \
+    khronos-headers/CL/cl_d3d10.h \
+    khronos-headers/CL/cl_d3d11.h \
+    khronos-headers/CL/cl_dx9_media_sharing.h \
+    khronos-headers/CL/cl_egl.h \
+    khronos-headers/CL/cl_ext.h \
+    khronos-headers/CL/cl_gl_ext.h \
+    khronos-headers/CL/cl_gl.h \
+    khronos-headers/CL/cl.h \
+    khronos-headers/CL/cl_platform.h \
+    khronos-headers/CL/opencl.h
+
+openclincludedir = $(includedir)/OpenCL
+openclinclude_HEADERS = \
+    khronos-headers/OpenCL/cl_d3d10.h \
+    khronos-headers/OpenCL/cl_d3d11.h \
+    khronos-headers/OpenCL/cl_dx9_media_sharing.h \
+    khronos-headers/OpenCL/cl_egl.h \
+    khronos-headers/OpenCL/cl_ext.h \
+    khronos-headers/OpenCL/cl_gl_ext.h \
+    khronos-headers/OpenCL/cl_gl.h \
+    khronos-headers/OpenCL/cl.h \
+    khronos-headers/OpenCL/cl_platform.h \
+    khronos-headers/OpenCL/opencl.h
+
 ocl_test_SOURCES = ocl_test.c
 ocl_test_LDADD = libOpenCL.la
 ocl_test_icdl_SOURCES = ocl_test_icdl.c
@@ -961,6 +990,27 @@
 	@list='$(pkgexample_DATA)'; test -n "$(pkgexampledir)" || list=; \
 	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
 	dir='$(DESTDIR)$(pkgexampledir)'; $(am__uninstall_files_from_dir)
+install-clincludeHEADERS: $(clinclude_HEADERS)
+	@$(NORMAL_INSTALL)
+	@list='$(clinclude_HEADERS)'; test -n "$(clincludedir)" || list=; \
+	if test -n "$$list"; then \
+	  echo " $(MKDIR_P) '$(DESTDIR)$(clincludedir)'"; \
+	  $(MKDIR_P) "$(DESTDIR)$(clincludedir)" || exit 1; \
+	fi; \
+	for p in $$list; do \
+	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
+	  echo "$$d$$p"; \
+	done | $(am__base_list) | \
+	while read files; do \
+	  echo " $(INSTALL_HEADER) $$files '$(DESTDIR)$(clincludedir)'"; \
+	  $(INSTALL_HEADER) $$files "$(DESTDIR)$(clincludedir)" || exit $$?; \
+	done
+
+uninstall-clincludeHEADERS:
+	@$(NORMAL_UNINSTALL)
+	@list='$(clinclude_HEADERS)'; test -n "$(clincludedir)" || list=; \
+	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
+	dir='$(DESTDIR)$(clincludedir)'; $(am__uninstall_files_from_dir)
 install-nodist_includeHEADERS: $(nodist_include_HEADERS)
 	@$(NORMAL_INSTALL)
 	@list='$(nodist_include_HEADERS)'; test -n "$(includedir)" || list=; \
@@ -982,6 +1032,27 @@
 	@list='$(nodist_include_HEADERS)'; test -n "$(includedir)" || list=; \
 	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
 	dir='$(DESTDIR)$(includedir)'; $(am__uninstall_files_from_dir)
+install-openclincludeHEADERS: $(openclinclude_HEADERS)
+	@$(NORMAL_INSTALL)
+	@list='$(openclinclude_HEADERS)'; test -n "$(openclincludedir)" || list=; \
+	if test -n "$$list"; then \
+	  echo " $(MKDIR_P) '$(DESTDIR)$(openclincludedir)'"; \
+	  $(MKDIR_P) "$(DESTDIR)$(openclincludedir)" || exit 1; \
+	fi; \
+	for p in $$list; do \
+	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
+	  echo "$$d$$p"; \
+	done | $(am__base_list) | \
+	while read files; do \
+	  echo " $(INSTALL_HEADER) $$files '$(DESTDIR)$(openclincludedir)'"; \
+	  $(INSTALL_HEADER) $$files "$(DESTDIR)$(openclincludedir)" || exit $$?; \
+	done
+
+uninstall-openclincludeHEADERS:
+	@$(NORMAL_UNINSTALL)
+	@list='$(openclinclude_HEADERS)'; test -n "$(openclincludedir)" || list=; \
+	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
+	dir='$(DESTDIR)$(openclincludedir)'; $(am__uninstall_files_from_dir)
 
 # This directory's subdirectories are mostly independent; you can cd
 # into them and run 'make' without going through this Makefile.
@@ -1291,7 +1362,7 @@
 		config.h
 installdirs: installdirs-recursive
 installdirs-am:
-	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(pkglibdir)" "$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(pkgexampledir)" "$(DESTDIR)$(includedir)"; do \
+	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(pkglibdir)" "$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(pkgexampledir)" "$(DESTDIR)$(clincludedir)" "$(DESTDIR)$(includedir)" "$(DESTDIR)$(openclincludedir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
 install: $(BUILT_SOURCES)
@@ -1352,8 +1423,9 @@
 
 info-am:
 
-install-data-am: install-nodist_includeHEADERS install-pkgconfigDATA \
-	install-pkgexampleDATA
+install-data-am: install-clincludeHEADERS \
+	install-nodist_includeHEADERS install-openclincludeHEADERS \
+	install-pkgconfigDATA install-pkgexampleDATA
 
 install-dvi: install-dvi-recursive
 
@@ -1402,7 +1474,8 @@
 
 ps-am:
 
-uninstall-am: uninstall-libLTLIBRARIES uninstall-nodist_includeHEADERS \
+uninstall-am: uninstall-clincludeHEADERS uninstall-libLTLIBRARIES \
+	uninstall-nodist_includeHEADERS uninstall-openclincludeHEADERS \
 	uninstall-pkgconfigDATA uninstall-pkgexampleDATA \
 	uninstall-pkglibLTLIBRARIES
 
@@ -1418,19 +1491,21 @@
 	distcheck distclean distclean-compile distclean-generic \
 	distclean-hdr distclean-libtool distclean-tags distcleancheck \
 	distdir distuninstallcheck dvi dvi-am html html-am info \
-	info-am install install-am install-data install-data-am \
-	install-dvi install-dvi-am install-exec install-exec-am \
-	install-exec-hook install-html install-html-am install-info \
-	install-info-am install-libLTLIBRARIES install-man \
-	install-nodist_includeHEADERS install-pdf install-pdf-am \
-	install-pkgconfigDATA install-pkgexampleDATA \
-	install-pkglibLTLIBRARIES install-ps install-ps-am \
-	install-strip installcheck installcheck-am installdirs \
-	installdirs-am maintainer-clean maintainer-clean-generic \
-	mostlyclean mostlyclean-compile mostlyclean-generic \
-	mostlyclean-libtool pdf pdf-am ps ps-am tags tags-am uninstall \
-	uninstall-am uninstall-libLTLIBRARIES \
-	uninstall-nodist_includeHEADERS uninstall-pkgconfigDATA \
+	info-am install install-am install-clincludeHEADERS \
+	install-data install-data-am install-dvi install-dvi-am \
+	install-exec install-exec-am install-exec-hook install-html \
+	install-html-am install-info install-info-am \
+	install-libLTLIBRARIES install-man \
+	install-nodist_includeHEADERS install-openclincludeHEADERS \
+	install-pdf install-pdf-am install-pkgconfigDATA \
+	install-pkgexampleDATA install-pkglibLTLIBRARIES install-ps \
+	install-ps-am install-strip installcheck installcheck-am \
+	installdirs installdirs-am maintainer-clean \
+	maintainer-clean-generic mostlyclean mostlyclean-compile \
+	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
+	tags tags-am uninstall uninstall-am uninstall-clincludeHEADERS \
+	uninstall-libLTLIBRARIES uninstall-nodist_includeHEADERS \
+	uninstall-openclincludeHEADERS uninstall-pkgconfigDATA \
 	uninstall-pkgexampleDATA uninstall-pkglibLTLIBRARIES
 
 .PRECIOUS: Makefile
