diff --git a/Makefile.am b/Makefile.am
index 845e89d..0ca0eb9 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -52,6 +52,32 @@ libOpenCL_la_LDFLAGS += -Wl,--version-script,ocl_icd_loader_gen.map
 libOpenCL_la_DEPENDS = ocl_icd_loader_gen.map
 endif
 
+clincludedir = $(includedir)/CL
+clinclude_HEADERS = \
+    khronos-headers/CL/cl_d3d10.h \
+    khronos-headers/CL/cl_d3d11.h \
+    khronos-headers/CL/cl_dx9_media_sharing.h \
+    khronos-headers/CL/cl_egl.h \
+    khronos-headers/CL/cl_ext.h \
+    khronos-headers/CL/cl_gl_ext.h \
+    khronos-headers/CL/cl_gl.h \
+    khronos-headers/CL/cl.h \
+    khronos-headers/CL/cl_platform.h \
+    khronos-headers/CL/opencl.h
+
+openclincludedir = $(includedir)/OpenCL
+openclinclude_HEADERS = \
+    khronos-headers/OpenCL/cl_d3d10.h \
+    khronos-headers/OpenCL/cl_d3d11.h \
+    khronos-headers/OpenCL/cl_dx9_media_sharing.h \
+    khronos-headers/OpenCL/cl_egl.h \
+    khronos-headers/OpenCL/cl_ext.h \
+    khronos-headers/OpenCL/cl_gl_ext.h \
+    khronos-headers/OpenCL/cl_gl.h \
+    khronos-headers/OpenCL/cl.h \
+    khronos-headers/OpenCL/cl_platform.h \
+    khronos-headers/OpenCL/opencl.h
+
 ####################################
 # A very small program test
 check_PROGRAMS=ocl_test ocl_test_icdl
