diff --git libtiff/tif_config.vc.h libtiff/tif_config.vc.h
index 7e6c5c2..e994abd 100644
--- libtiff/tif_config.vc.h
+++ libtiff/tif_config.vc.h
@@ -107,6 +107,8 @@
 /* Visual Studio 2015 / VC 14 / MSVC 19.00 finally has snprintf() */
 #if defined(_MSC_VER) && _MSC_VER < 1900
 #define snprintf _snprintf
+#else
+#define HAVE_SNPRINTF
 #endif
 
 /* Define to 1 if your processor stores words with the most significant byte
