From e297cb0366bc70daf807a4b3a5053d6d806f61fe Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 12 Oct 2024 19:59:02 +1100
Subject: [PATCH] fix corrupted line endings

---
 build.i | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/build.i b/build.i
index 08fd7bc..5f2fa2e 100644
--- a/build.i
+++ b/build.i
@@ -1 +1,6 @@
-      character Revision*4      character Build*19      character compdate*100      Revision='657'      Build='2012/11/28 12:09:23'      compdate='Revision: '//Revision//' Build: ' // Build
\ No newline at end of file
+      character Revision*4
+      character Build*19
+      character compdate*100
+      Revision='657'
+      Build='2012/11/28 12:09:23'
+      compdate='Revision: '//Revision//' Build: ' // Build
