From d2c781a6ef18667f0465219046ff7c64e9ff72f7 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 12 Oct 2024 21:26:50 +1100
Subject: [PATCH] make linker invocation compatible with MSFT syntax

---
 makefile.gf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile.gf b/makefile.gf
index 70c6739..7e620af 100644
--- a/makefile.gf
+++ b/makefile.gf
@@ -290,7 +290,7 @@ SRCS            = aaamain.f abend.f acf.f acfar.f acfdgn.f \
                   testodf.f
 
 $(PROGRAM):     $(OBJS) $(LIBS)
-	$(LINKER) -static -o $@ $(OBJS) $(LDMAP) $(LIBS) $(LDFLAGS)
+	$(LINKER) /OUT:$@ $(OBJS) $(LDMAP) $(LIBS) $(LDFLAGS)
 
 clean:;         @rm -f $(OBJS)
 
