From edf5db93996b2dbe9f97cd98797af220778e7770 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 12 Oct 2024 20:14:42 +1100
Subject: [PATCH] fix corrupted line endings

---
 build.i | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/build.i b/build.i
index b1bdf70..5f2fa2e 100644
--- a/build.i
+++ b/build.i
@@ -1,6 +1,6 @@
-      character Revision*4
-      character Build*19
-      character compdate*100
-      Revision='657'
-      Build='2012/11/28 12:09:23'
-      compdate='Revision: '//Revision//' Build: ' // Build
+      character Revision*4
+      character Build*19
+      character compdate*100
+      Revision='657'
+      Build='2012/11/28 12:09:23'
+      compdate='Revision: '//Revision//' Build: ' // Build
