From d8cccb4923fb3df2071750e2b1f61b018934e923 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 12 Oct 2024 21:48:38 +1100
Subject: [PATCH 2/2] avoid installion in Makefile

---
 makefile.gf | 2 --
 1 file changed, 2 deletions(-)

diff --git a/makefile.gf b/makefile.gf
index 7e620af..1a08f4c 100644
--- a/makefile.gf
+++ b/makefile.gf
@@ -295,8 +295,6 @@ $(PROGRAM):     $(OBJS) $(LIBS)
 clean:;         @rm -f $(OBJS)
 
 install:   $(PROGRAM)
-	@echo Installing $(PROGRAM) in $(DEST)
-	@if not $(DEST)x==.x copy $(PROGRAM) $(DEST)
 ### OPUS MKMF:  Do not remove this line!  Automatic dependencies follow.
 
 aaamain.o:  build.prm cchars.i chrt.cmn error.cmn hiddn.cmn lex.i \
