From 4d1cf77ef42b303f179cfa67eefcd1d2135c535b Mon Sep 17 00:00:00 2001
From: Enrico Bertolazzi <enrico.bertolazzi@unitn.it>
Date: Thu, 6 May 2021 17:25:24 +0200
Subject: [PATCH 02/13] updated doc

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 2f56388..b9db43d 100644
--- a/.gitignore
+++ b/.gitignore
@@ -12,3 +12,4 @@ docs_build/sphinx/api
 docs_build/xml
 docs_build/sphinx/api-cpp
 docs_build/xml-cpp
+docs_build/sphinx/_doxygen
-- 
2.34.1

