From 948b13a8db2916f5a762148b0a3140648c02cd86 Mon Sep 17 00:00:00 2001
From: Enrico Bertolazzi <enrico.bertolazzi@unitn.it>
Date: Fri, 15 Oct 2021 17:35:08 +0200
Subject: [PATCH 07/13] Modified Rakefile for the new cmake.

---
 Rakefile | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/Rakefile b/Rakefile
index 7781aa1..97e35fc 100755
--- a/Rakefile
+++ b/Rakefile
@@ -9,6 +9,8 @@ end
 
 require_relative "./Rakefile_common.rb"
 
+file_base = File.expand_path(File.dirname(__FILE__)).to_s+'/lib'
+
 task :default => [:build]
 
 TESTS = [
@@ -62,12 +64,13 @@ task :build_win, [:year, :bits] do |t, args|
   FileUtils.mkdir_p dir
   FileUtils.cd      dir
 
-  cmake_cmd = win_vs(args.bits,args.year)
+  cmd_cmake = win_vs(args.bits,args.year)
   if COMPILE_EXECUTABLE then
-    cmake_cmd += ' -DBUILD_EXECUTABLE:VAR=true '
+    cmd_cmake += ' -DBUILD_EXECUTABLE:VAR=true '
   else
-    cmake_cmd += ' -DBUILD_EXECUTABLE:VAR=false '
+    cmd_cmake += ' -DBUILD_EXECUTABLE:VAR=false '
   end
+  cmd_cmake += " -DCMAKE_INSTALL_PREFIX=\"#{file_base}\" "
 
   FileUtils.mkdir_p "../lib/lib"
   FileUtils.mkdir_p "../lib/bin"
@@ -76,10 +79,10 @@ task :build_win, [:year, :bits] do |t, args|
   FileUtils.mkdir_p "../lib/include"
 
   if COMPILE_DEBUG then
-    sh cmake_cmd + ' -DCMAKE_BUILD_TYPE:VAR=Debug --loglevel=WARNING ..'
+    sh cmd_cmake + ' -DCMAKE_BUILD_TYPE:VAR=Debug --loglevel=WARNING ..'
     sh 'cmake --build . --config Debug --target install '+PARALLEL+QUIET
   else
-    sh cmake_cmd + ' -DCMAKE_BUILD_TYPE:VAR=Release --loglevel=WARNING ..'
+    sh cmd_cmake + ' -DCMAKE_BUILD_TYPE:VAR=Release --loglevel=WARNING ..'
     sh 'cmake  --build . --config Release  --target install '+PARALLEL+QUIET
   end
 
@@ -102,6 +105,7 @@ task :build_osx do |t, args|
   else
     cmd_cmake += 'false '
   end
+  cmd_cmake += " -DCMAKE_INSTALL_PREFIX=\"#{file_base}\" "
 
   if COMPILE_DEBUG then
     sh cmd_cmake + '-DCMAKE_BUILD_TYPE:VAR=Debug --loglevel=WARNING ..'
@@ -129,6 +133,7 @@ task :build_linux do |t, args|
   else
     cmd_cmake += 'false '
   end
+  cmd_cmake += " -DCMAKE_INSTALL_PREFIX=\"#{file_base}\" "
 
   if COMPILE_DEBUG then
     sh cmd_cmake + '-DCMAKE_BUILD_TYPE:VAR=Debug --loglevel=WARNING ..'
-- 
2.34.1

