From 06f6e576b2ff317d68aa86192d895f9bdff89d62 Mon Sep 17 00:00:00 2001
From: Enrico Bertolazzi <enrico.bertolazzi@unitn.it>
Date: Sat, 16 Oct 2021 00:20:09 +0200
Subject: [PATCH 09/13] corrected INSTALL_HERE coomand in CMakeLIsts.txt

---
 CMakeLists.txt | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33c38ff..760f03e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -112,9 +112,10 @@ endif()
 # FIXME: CMakeLists should not override CMAKE_INSTALL_PREFIX
 # ref: https://gitlab.kitware.com/cmake/cmake/-/issues/19322
 if(INSTALL_HERE)
-  set(CMAKE_INSTALL_PREFIX "${CMAKE_CURRENT_SOURCE_DIR}/lib")
-  set(PROJECT_PUBLIC_HEADER_DESTINATION "${CMAKE_INSTALL_PREFIX}/include")
+  set(INSTALL_PREFIX "${CMAKE_CURRENT_SOURCE_DIR}/lib")
+  set(PROJECT_PUBLIC_HEADER_DESTINATION "${INSTALL_PREFIX}/include")
 else()
+  set(INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}/lib")
   set(PROJECT_PUBLIC_HEADER_DESTINATION "${CMAKE_INSTALL_PREFIX}/include/${PROJECT_NAME}")
 endif()
 
@@ -124,12 +125,19 @@ if(BUILD_SHARED)
 endif()
 install(
   TARGETS ${INSTALL_TARGETS}
-  RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin"
-  ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/lib"
-  LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/dll"
+  RUNTIME DESTINATION "${INSTALL_PREFIX}/bin"
+  ARCHIVE DESTINATION "${INSTALL_PREFIX}/lib"
+  LIBRARY DESTINATION "${INSTALL_PREFIX}/dll"
   PUBLIC_HEADER DESTINATION "${PROJECT_PUBLIC_HEADER_DESTINATION}"
 )
 
+if(INSTALL_HERE)
+  add_custom_command(
+    TARGET ${STATIC_PROJECT} POST_BUILD
+    COMMAND cmake  --build . --config Release  --target install
+  )
+endif()
+
 message(STATUS "Using ${SSE_FLAGS} extensions")
 message(STATUS "C compiler                  = ${CMAKE_C_COMPILER_ID} ${CMAKE_C_COMPILER}")
 message(STATUS "C++ compiler                = ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER}")
-- 
2.34.1

