From e30c840e52527488d4a981c02e1ed11138eb958b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jo=C3=A3o=20Paulo=20Taylor=20Ienczak=20Zanette?=
 <jpaulotiz@gmail.com>
Date: Thu, 16 Dec 2021 21:41:22 -0300
Subject: [PATCH 16/17] Use standard generated library filename

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d4b525..06a4a50 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,7 +69,6 @@ target_include_directories(
 )
 set_target_properties(
   ${STATIC_PROJECT} PROPERTIES
-  OUTPUT_NAME "${PROJECT_NAME}${ARTIFACTS_STATIC_SUFFIX}"
   PUBLIC_HEADER "${HEADERS}"
   POSITION_INDEPENDENT_CODE ON
 )
-- 
2.34.1

