diff --git i/config.mk w/config.mk
index 9f08429..16ab9e9 100644
--- i/config.mk
+++ w/config.mk
@@ -12,12 +12,12 @@
 #  - Start the simulation using `./poiseuille2d`
 
 # Compiler to use for C++ files, change to `mpic++` when using OpenMPI and GCC
-CXX             := g++
+CXX             ?= g++
 # Compiler to use for C files (used for emebedded dependencies)
-CC              := gcc
+CC              ?= gcc
 
 # Suggested optimized build flags for GCC, consult `config/` for further examples
-CXXFLAGS        := -O3 -Wall -march=native -mtune=native
+CXXFLAGS        ?= -O3 -Wall -march=native -mtune=native
 # Uncomment to add debug symbols and enable runtime asserts
 #CXXFLAGS        += -g -DOLB_DEBUG
 
@@ -34,7 +34,7 @@ LDFLAGS         :=
 # Parallelization mode, must be one of: OFF, MPI, OMP, HYBRID
 # Note that for MPI and HYBRID the compiler also needs to be adapted.
 # See e.g. `config/cpu_gcc_openmpi.mk`
-PARALLEL_MODE   := OFF
+PARALLEL_MODE   := OMP
 
 # optional MPI and OpenMP flags
 MPIFLAGS        :=
@@ -55,4 +55,4 @@ FLOATING_POINT_TYPE := double
 FEATURES        :=
 
 # Set to OFF if libz and tinyxml are provided by the system (optional)
-USE_EMBEDDED_DEPENDENCIES := ON
+USE_EMBEDDED_DEPENDENCIES := OFF
