diff --git a/medm/Makefile b/medm/Makefile
index 223f212..5820ac3 100644
--- a/medm/Makefile
+++ b/medm/Makefile
@@ -179,11 +179,7 @@ USR_LIBS_Linux = Xm Xt Xp Xmu X11 Xext
 #USR_LIBS_Linux = Xm Xt Xmu X11 Xext
 
 USR_LIBS_cygwin32 = Xm Xt Xmu X11 Xext SM ICE
-USR_LIBS_Darwin = -nil-
-PROD_SYS_LIBS_Darwin = Xt Xmu X11 Xext Xft Xp fontconfig png
-
-#Do not use PROD_LDFLAGS_Darwin because it goes after OP_SYS_LDFLAGS
-USR_LDFLAGS_Darwin = -L/usr/X11/lib /opt/local/lib/libXm.a /opt/local/lib/libjpeg.a /opt/local/lib/libiconv.a
+USR_LIBS_Darwin = Xm Xt Xp Xmu X11 Xext
 
 WIN32_RUNTIME=MD
 USR_CFLAGS_WIN32 += /DWIN32 /D_WINDOWS
