diff --git a/setup.py b/setup.py
index 598e8e7..01ba679 100644
--- a/setup.py
+++ b/setup.py
@@ -67,9 +67,7 @@ setup(
         "Topic :: Scientific/Engineering :: Physics",
     ],
     packages=find_packages(),
-    package_data={"": ["LICENSE", "README.md"], "pguresvt": ["*.py"]},
-    python_requires=">=3.6",
+    package_data={"": ["LICENSE", "README.md"], "pguresvt": ["*.py", "tests/*"]},
     install_requires=["numpy"],
-    setup_requires=["wheel", "auditwheel"],
     ext_modules=cythonize(extensions),
 )

