From 0c03aab13f4510e3e7fb16424312e2f78a964c1a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Sat, 13 Nov 2021 20:27:45 -0600
Subject: [PATCH] Fix AX_CXX_CHECK_LIB test name mangling on Windows

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 355ff5d..6d73866 100644
--- a/configure.ac
+++ b/configure.ac
@@ -31,7 +31,7 @@ AX_CXX_COMPILE_STDCXX(17)
 # GMPXX does not contain anything that we can check for with AX_CXX_CHECK_LIB
 # so we just check for something from the standard library, i.e., that -lgmpxx
 # goes through
-AX_CXX_CHECK_LIB([gmpxx], [std::runtime_error::what () const], [have_gmpxx=yes], AC_MSG_ERROR([GMP library for C++ not found]), [-lgmp])
+AX_CXX_CHECK_LIB([gmpxx], [std::runtime_error::what () const noexcept], [have_gmpxx=yes], AC_MSG_ERROR([GMP library for C++ not found]), [-lgmp])
 
 dnl Our benchmarks use Google's C++ benchmark library.
 dnl We fail if they cannot be found but let the user disable it explicitly.
-- 
2.33.1

