--- a/pyproject.toml
+++ b/pyproject.toml
@@ -11,7 +11,7 @@
 
 [tool.poetry.dependencies]
 python = "^3.6"
-pybase62 = "0.4.3"
+pybase62 = ">=0.4.3"
 
 [tool.poetry.dev-dependencies]
 pytest = "^6.2.5"
--- a/PKG-INFO
+++ b/PKG-INFO
@@ -15,7 +15,7 @@
 Classifier: Programming Language :: Python :: 3.7
 Classifier: Programming Language :: Python :: 3.8
 Classifier: Programming Language :: Python :: 3.9
-Requires-Dist: pybase62 (==0.4.3)
+Requires-Dist: pybase62 (>=0.4.3)
 Project-URL: Repository, https://github.com/sashahilton00/python-pksuid
 Description-Content-Type: text/markdown
 
--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,7 @@
 {'': ['*']}
 
 install_requires = \
-['pybase62==0.4.3']
+['pybase62>=0.4.3']
 
 setup_kwargs = {
     'name': 'pksuid',
