From 7a7e28354f659b11a94dd64eb9a2b23f4d4c520e Mon Sep 17 00:00:00 2001
From: Chris Barker <Chris.Barker@noaa.gov>
Date: Fri, 2 Oct 2020 16:15:25 -0700
Subject: [PATCH] removed setup_requires

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 84cc14e..14ef749 100644
--- a/setup.py
+++ b/setup.py
@@ -88,7 +88,6 @@ setup(
     long_description_content_type='text/markdown',
     license='ISC',
     ext_modules=[CMakeExtension('mapbox_earcut')],
-    setup_requires=['setuptools_scm'],
     install_requires=['numpy'],
     cmdclass=dict(build_ext=CMakeBuild),
     zip_safe=False,
-- 
2.17.2 (Apple Git-113)

