From 30ff8148966fae6b0d1a6a367f8c936c6970ef91 Mon Sep 17 00:00:00 2001
From: "jay.hennen" <jay.hennen@noaa.gov>
Date: Mon, 30 Nov 2020 12:02:06 -0800
Subject: [PATCH 2/2] removed -A flag for Windows builds

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 84cc14e..fc6d1d0 100644
--- a/setup.py
+++ b/setup.py
@@ -51,8 +51,8 @@ class CMakeBuild(build_ext):
                 '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}'.format(
                     cfg.upper(), extdir)
             ]
-            if sys.maxsize > 2**32:
-                cmake_args += ['-A', 'x64']
+            #if sys.maxsize > 2**32:
+            #    cmake_args += ['-A', 'x64']
             build_args += ['--', '/m']
         else:
             cmake_args += ['-DCMAKE_BUILD_TYPE=' + cfg]
-- 
2.13.0.windows.1

