From 668c729a3b5580e7f61d5c9230e759cc3d940bb4 Mon Sep 17 00:00:00 2001
From: Chun Cai <amoycaic@gmail.com>
Date: Thu, 7 Mar 2024 16:26:08 +0800
Subject: [PATCH] Build: disable sm_90 arch for forward compatibility on CUDA
 11.2 (#95)

---
 unidock/CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/unidock/CMakeLists.txt b/unidock/CMakeLists.txt
index 861d4a8..c5a47ca 100644
--- a/unidock/CMakeLists.txt
+++ b/unidock/CMakeLists.txt
@@ -26,12 +26,10 @@ if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
   # https://en.wikipedia.org/wiki/CUDA#GPUs_supported
   set(CMAKE_CUDA_ARCHITECTURES
     70 # V100
-
     # 75 # RTX 20, T4
     80 # A100, A800
-
     # 86 # RTX 30 89 # RTX 40, L40
-    90 # H100
+    # 90 # H100
   )
 endif()
 
-- 
2.39.3 (Apple Git-146)

