--- qemu/configure
+++ qemu/configure
@@ -1656,3 +1656,3 @@
     *-linux-user|*-bsd-user)
-      qemu="qemu-$arch"
+      qemu="qemu-execve-$arch"
       ;;
--- qemu/linux-user/main.c
+++ qemu/linux-user/main.c
@@ -1656,3 +1656,3 @@
 {
-    printf("qemu-" TARGET_NAME " version " QEMU_FULL_VERSION
+    printf("qemu-execve-" TARGET_NAME " version " QEMU_FULL_VERSION
            "\n" QEMU_COPYRIGHT "\n");
