From 0393767bf2bc217fa6a7b2eedc9918bbc6dc0d82 Mon Sep 17 00:00:00 2001
From: dillonroach <dillon.roach@gmailcom>
Date: Mon, 10 Oct 2022 09:33:06 -0700
Subject: [PATCH] fix imports

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index db805bf..ec71597 100644
--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ from pathlib import Path
 from setuptools import find_packages, setup
 
 dependencies = ['matplotlib', 'numpy', 'pandas',
-                'seaborn', 'sklearn', 'scipy', 'sklearn']
+                'seaborn', 'scikit-learn', 'scipy']
 
 # read the contents of README file
 this_directory = Path(__file__).parent
-- 
2.25.1

