diff --git /dev/null cmake/GeneratePkgConfigFile.cmake
index e69de29..ff9b9e0 100644
--- /dev/null
+++ cmake/GeneratePkgConfigFile.cmake
@@ -0,0 +1,8 @@
+function(generate_pkg_config_file in_file)
+    set(PACKAGE_NAME "${PROJECT_NAME}")
+    set(PACKAGE_DESCRIPTION "${PROJECT_DESCRIPTION}")
+    set(PACKAGE_URL "${PROJECT_HOMEPAGE_URL}")
+    set(PACKAGE_VERSION "${PROJECT_VERSION}")
+    set(PACKAGE_LIBNAME "${PROJECT_LIBNAME}")
+    configure_file(${in_file} ${PROJECT_NAME}.pc @ONLY)
+endfunction()
