diff --git a/core/vtk/CMakeLists.txt b/core/vtk/CMakeLists.txt
index a4d4d661..0f4eb20f 100644
--- a/core/vtk/CMakeLists.txt
+++ b/core/vtk/CMakeLists.txt
@@ -146,8 +146,7 @@ if (TTK_BUILD_VTK_PYTHON_MODULE)
   target_link_libraries(${module}PythonD ${VTKWRAPPER_LIB_LIST})
   target_link_libraries(${module}PythonD
     ${VTK_LIBRARIES}
-    vtkWrappingPythonCore
-    ${VTK_PYTHON_LIBRARIES})
+    vtkWrappingPythonCore)
 
   # link against VTK PythonD modules
   set(_pythond_libs)
