From 64de5a479b94beba357e73c926cdc6ee9114ceec Mon Sep 17 00:00:00 2001
From: Pavel Demin <pavel.demin@uclouvain.be>
Date: Sun, 24 Sep 2023 10:01:36 +0200
Subject: [PATCH] remove binary_function from puppiParticle.hh

---
 external/PUPPI/puppiParticle.hh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/external/PUPPI/puppiParticle.hh b/external/PUPPI/puppiParticle.hh
index 2d0a34ba1..baf7972d1 100644
--- a/external/PUPPI/puppiParticle.hh
+++ b/external/PUPPI/puppiParticle.hh
@@ -60,7 +60,7 @@ class puppiParticle {
 
 };
 
-class puppiValSort : public std::binary_function<int,int,bool> {
+class puppiValSort {
  public:
 
   puppiValSort(){};
