From 06a6a6bf45fbd039cf2d0e979220132f7328bc89 Mon Sep 17 00:00:00 2001
From: Joris Vaillant <joris.vaillant@inria.fr>
Date: Tue, 10 Sep 2024 15:46:53 +0200
Subject: [PATCH] Don't link against Python library in Python module

---
 python/CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index a4a8978e..5954d28d 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -14,9 +14,8 @@ endif()
 set(BUFFER_SIZE 128 CACHE STRING "The size of the pointer buffer")
 set(NAME_LENGTH 128 CACHE STRING "The length of a name in the buffer")
 
-pybind11_add_module(TracyClientBindings SHARED bindings/Module.cpp)
+pybind11_add_module(TracyClientBindings THIN_LTO bindings/Module.cpp)
 target_link_libraries(TracyClientBindings PUBLIC TracyClient)
-target_link_libraries(TracyClientBindings PUBLIC ${Python_LIBRARIES})
 target_compile_definitions(TracyClientBindings PUBLIC BUFFER_SIZE=${BUFFER_SIZE})
 target_compile_definitions(TracyClientBindings PUBLIC NAME_LENGTH=${NAME_LENGTH})
 
-- 
2.34.1

