From d2a5747a7988f913ac6cd0e12d2d4a5336e9df75 Mon Sep 17 00:00:00 2001
From: Mervin Fansler <mervin.fansler@bric.ku.dk>
Date: Thu, 11 Jun 2026 14:03:11 +0200
Subject: [PATCH] append RUSTFLAGS

---
 src/Makevars.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makevars.in b/src/Makevars.in
index 3d50e8f..effdd71 100644
--- a/src/Makevars.in
+++ b/src/Makevars.in
@@ -47,7 +47,7 @@ $(STATLIB):
 	      export CARGO_HOME=$(CARGOTMP);  \
 	      export R_HOME=$(R_HOME);  \
 	      export PATH="$(PATH):$(HOME)/.cargo/bin";  \
-	      export RUSTFLAGS="-C linker=${CC_ALONE}"; \
+	      export RUSTFLAGS="-C linker=${CC_ALONE} ${RUSTFLAGS}"; \
 	      cargo build $(CARGO_BUILD_ARGS) --release --offline; \
 	    else \
 	      cargo build $(CARGO_BUILD_ARGS); \
