Subject: [PATCH] dont install commands
---
Index: setup.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/setup.py b/setup.py
--- a/setup.py	(revision f07211d2f12d2cb09b91de9e7301eff822b93f02)
+++ b/setup.py	(revision d720be24bfa30610ac2edc6b1b0c897bb3b9dfaf)
@@ -196,13 +196,5 @@
         'bmf.server',
     ],
     ext_modules=[CMakeExtension("bmf")],
-    cmdclass={"build_ext": CMakeBuild},
-    entry_points={
-        'console_scripts': [
-           'run_bmf_graph = bmf.cmd.python_wrapper.wrapper:run_bmf_graph',
-           'trace_format_log = bmf.cmd.python_wrapper.wrapper:trace_format_log',
-           'module_manager = bmf.cmd.python_wrapper.wrapper:module_manager',
-           "bmf_env = bmf.cmd.python_wrapper.wrapper:bmf_env",
-        ],
-    }
+    cmdclass={"build_ext": CMakeBuild}
 )
