Subject: [PATCH] Make CMake standard requirement optional
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision d0094a259dcf4f9d1b2ae91a5239e0b9627e8bad)
+++ b/CMakeLists.txt	(date 1701761181081)
@@ -48,7 +48,6 @@
 
 ### general settings
 set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
 if (${COVERAGE})
