Subject: [PATCH] Search library in binary dir
---
Index: bmf/sdk/cpp_sdk/include/bmf/sdk/shared_library.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bmf/sdk/cpp_sdk/include/bmf/sdk/shared_library.h b/bmf/sdk/cpp_sdk/include/bmf/sdk/shared_library.h
--- a/bmf/sdk/cpp_sdk/include/bmf/sdk/shared_library.h	(revision c16ec0d0a41445fea9629a7996f670d3967f8b89)
+++ b/bmf/sdk/cpp_sdk/include/bmf/sdk/shared_library.h	(date 1700744400048)
@@ -82,7 +82,7 @@
 #elif defined(__APPLE__)
         return "lib";
 #elif defined(__MINGW32__) || defined(_WIN32)
-        return "lib";
+        return "bin";
 #else
 #error "unsupported os"
 #endif
