Subject: [PATCH] Remove leading dash
---
Index: Makefile
===================================================================
diff --git a/Makefile b/Makefile
--- a/Makefile	(revision 79af0071ffe32705d573b38e2af0b2afdaf1c652)
+++ b/Makefile	(revision 5d0abd9708e0f5fe5f59d08059f9bc46dc268599)
@@ -1,7 +1,7 @@
 BUILD    := ./build
 BIN    	 := ./bin
 TARGET   := DiveMESH
-CXX      := -g++
+CXX      := g++
 CXXFLAGS := -w -O3
 LDFLAGS  :=

