Subject: [PATCH] Remove CXX in Makefile
---
Index: Makefile
===================================================================
diff --git a/Makefile b/Makefile
--- a/Makefile	(revision 79af0071ffe32705d573b38e2af0b2afdaf1c652)
+++ b/Makefile	(revision 791be992a0588394bfdc60c32528298fcf8181e8)
@@ -1,7 +1,6 @@
 BUILD    := ./build
 BIN    	 := ./bin
 TARGET   := DiveMESH
-CXX      := -g++
 CXXFLAGS := -w -O3
 LDFLAGS  :=
