diff --git a/CMakeLists.txt b/CMakeLists.txt
index b8fe86ba9..c173146ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -171,9 +171,9 @@ if("${isSystemDir}" STREQUAL "-1")
    set(CMAKE_INSTALL_RPATH "${dismod_at_pefix}/${cmake_libdir}")
 endif("${isSystemDir}" STREQUAL "-1")
 # ----------------------------------------------------------------------------
-# pkg-config information for: sqlite3, gsl, eigen, ipopt, cppad_mixed, cppad
+# pkg-config information for: gsl, eigen, ipopt, cppad_mixed, cppad
 FIND_PACKAGE(PkgConfig)
-FOREACH(pkg sqlite3 gsl eigen3 ipopt cppad_mixed cppad)
+FOREACH(pkg gsl eigen3 ipopt cppad_mixed cppad)
    pkg_check_modules( ${pkg} QUIET ${pkg} )
    IF( ${pkg}_FOUND )
       MESSAGE(STATUS "Found ${pkg}.pc file")
@@ -190,6 +190,7 @@ FOREACH(pkg sqlite3 gsl eigen3 ipopt cppad_mixed cppad)
    # LINK_DIRECTORIES
    LINK_DIRECTORIES( ${${pkg}_LIBRARY_DIRS} )
 ENDFOREACH(pkg)
+SET(sqlite3_LIBRARIES sqlite3)
 # ----------------------------------------------------------------------------
 # no pkg-config for SuiteSparse so explicitly set
 # suitesparse_INCLUDE_DIR
diff --git a/bin/user_test.sh.in b/bin/user_test.sh.in
index c2576331e..321b5fd89 100755
--- a/bin/user_test.sh.in
+++ b/bin/user_test.sh.in
@@ -29,7 +29,7 @@ echo "bin/user_test.sh $user_test"
 source bin/grep_and_sed.sh
 #
 # try
-for try in {1..3}
+for try in {1..4}
 do
    # user_test.$$
    if @python3_executable@ $user_test >& user_test.$$
@@ -54,5 +54,5 @@ do
 done
 cat user_test.$$
 rm user_test.$$
-echo "user_test.sh: $user_test: Error or warning 3 times in a row."
+echo "user_test.sh: $user_test: Error or warning 4 times in a row."
 exit 1
diff --git a/devel/cmd/init_command.cpp b/devel/cmd/init_command.cpp
index 443977942..0517a7369 100644
--- a/devel/cmd/init_command.cpp
+++ b/devel/cmd/init_command.cpp
@@ -320,7 +320,7 @@ void init_command(
       {
          size_t index        = size_t(data_id) * n_covariate + id;
          double cov_value    = db_input.data_cov_value[index];
-         if( not std::isnan( cov_value ) )
+         if( ! std::isnan( cov_value ) )
          {  double reference    = db_input.covariate_table[id].reference;
             double abs_diff     = std::fabs(cov_value - reference);
             //
diff --git a/python/bin/dismodat.py.in b/python/bin/dismodat.py.in
index 9d58b3a73..fa83c22b1 100755
--- a/python/bin/dismodat.py.in
+++ b/python/bin/dismodat.py.in
@@ -1,4 +1,4 @@
-#! @python3_executable@
+#! /usr/bin/env python
 # $Id:$
 # SPDX-License-Identifier: AGPL-3.0-or-later
 # SPDX-FileCopyrightText: University of Washington <https://www.washington.edu>
