diff --git a/devel/table/get_db_input.cpp b/devel/table/get_db_input.cpp
index 0beae7a99..e7ee53e8d 100644
--- a/devel/table/get_db_input.cpp
+++ b/devel/table/get_db_input.cpp
@@ -368,6 +368,7 @@ void get_db_input(sqlite3* db, db_input_struct& db_input)
       db_input.rate_eff_cov_table    ,
       db_input.option_table
    );
+   sqlite3_close(db_other);
    return;
 }
 
diff --git a/example/devel/table/blob_table_xam.cpp b/example/devel/table/blob_table_xam.cpp
index 538c34526..b1a9a1bdb 100644
--- a/example/devel/table/blob_table_xam.cpp
+++ b/example/devel/table/blob_table_xam.cpp
@@ -54,6 +54,7 @@ bool blob_table_xam(void)
    ok &= one.x == two.x;
    ok &= one.y == two.y;
    //
+   sqlite3_close(db);
    return ok;
 }
 // END C++
diff --git a/example/devel/table/check_pini_n_age_xam.cpp b/example/devel/table/check_pini_n_age_xam.cpp
index ff22d0f27..e76dfbcf7 100644
--- a/example/devel/table/check_pini_n_age_xam.cpp
+++ b/example/devel/table/check_pini_n_age_xam.cpp
@@ -54,6 +54,7 @@ bool check_pini_n_age_xam(void)
    //
    dismod_at::check_pini_n_age(db, rate_table, smooth_table);
    //
+   sqlite3_close(db);
    return ok;
 }
 // END C++
diff --git a/example/user/CMakeLists.txt b/example/user/CMakeLists.txt
index e452ab61d..c5d815798 100644
--- a/example/user/CMakeLists.txt
+++ b/example/user/CMakeLists.txt
@@ -73,7 +73,7 @@ FOREACH(user_case
 # END_SORT_THIS_LINE_MINUS_2
    ADD_CUSTOM_TARGET(
       check_example_user_${user_case}
-      bin/user_test.py example/user/${user_case}.py
+      ${python3_executable} bin/user_test.py example/user/${user_case}.py
       WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
       DEPENDS dismod_at
    )
diff --git a/test/user/CMakeLists.txt b/test/user/CMakeLists.txt
index f0b02d788..785914e88 100644
--- a/test/user/CMakeLists.txt
+++ b/test/user/CMakeLists.txt
@@ -41,7 +41,7 @@ FOREACH(user_case
 #
    ADD_CUSTOM_TARGET(
       check_test_user_${user_case}
-      bin/user_test.py test/user/${user_case}.py
+      ${python3_executable} bin/user_test.py test/user/${user_case}.py
       WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
       DEPENDS dismod_at
    )
