diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca96d4266..21bf7bc93 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,7 +94,7 @@ MACRO( set_library_path_list
    # MESSAGE(STATUS "${library_path_list}=${${library_path_list}}"  )
 ENDMACRO( set_library_path_list )
 # ============================================================================
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
 PROJECT(dismod_at)
 SET(dismod_at_version "2025.4.11")
 INCLUDE_DIRECTORIES( ${dismod_at_SOURCE_DIR}/include )
