diff --git a/CMakeLists.txt b/CMakeLists.txt
index a08247a..990b355 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -298,8 +298,11 @@ if(WIN32)
   target_compile_definitions(srtp2 PUBLIC _CRT_SECURE_NO_WARNINGS)
 endif()

-install(TARGETS srtp2 DESTINATION lib
+install(TARGETS srtp2
   EXPORT libSRTPTargets
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
 )

 install(FILES include/srtp.h crypto/include/auth.h
