From 794da8a8c3d4b255a993a8f29d4295515b48f5ea Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Tue, 13 Aug 2019 07:58:05 -0400
Subject: [PATCH 1/2] Do not assume menuinst wants to operate on sys.prefix

---
 menuinst/__init__.py | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/menuinst/__init__.py b/menuinst/__init__.py
index a21240a..1d14731 100644
--- a/menuinst/__init__.py
+++ b/menuinst/__init__.py
@@ -53,20 +53,20 @@ def install(path, remove=False, prefix=sys.prefix, recursing=False):
     install Menu and shortcuts
     """
     # this sys.prefix is intentional.  We want to reflect the state of the root installation.
-    if sys.platform == 'win32' and not exists(join(sys.prefix, '.nonadmin')):
+    if sys.platform == 'win32' and not exists(join(prefix, '.nonadmin')):
         if isUserAdmin():
             _install(path, remove, prefix, mode='system')
         else:
             from pywintypes import error
+            retcode = 1
             try:
                 if not recursing:
-                    retcode = runAsAdmin([join(sys.prefix, 'python'), '-c',
-                                          "import menuinst; menuinst.install(%r, %r, %r, %r)" % (
-                                              path, bool(remove), prefix, True)])
-                else:
-                    retcode = 1
+                    if exists(join(prefix, 'python.exe')):
+                        retcode = runAsAdmin([join(prefix, 'python.exe'), '-c',
+                            "import menuinst; menuinst.install(%r, %r, %r, %r)" % (
+                                path, bool(remove), prefix, True)])
             except error:
-                retcode = 1
+                pass
 
             if retcode != 0:
                 logging.warn("Insufficient permissions to write menu folder.  "
-- 
2.22.0

