From 7a1e02ef9b9c6e932c3d32669423585794004592 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Tue, 13 Aug 2019 07:58:45 -0400
Subject: [PATCH 2/2] Allow menuinst to operate on non base prefix

---
 menuinst/win32.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/menuinst/win32.py b/menuinst/win32.py
index f95abe2..a8ffb8f 100644
--- a/menuinst/win32.py
+++ b/menuinst/win32.py
@@ -170,7 +170,7 @@ def substitute_env_variables(text, dir):
 
     for a, b in (
         (u'${PREFIX}', env_prefix),
-        (u'${ROOT_PREFIX}', unicode_root_prefix),
+        (u'${ROOT_PREFIX}', to_unicode(env_prefix)),
         (u'${PYTHON_SCRIPTS}',
           os.path.normpath(join(env_prefix, u'Scripts')).replace(u"\\", u"/")),
         (u'${MENU_DIR}', join(env_prefix, u'Menu')),
@@ -284,6 +284,7 @@ class ShortCut(object):
         args = []
         fix_win_slashes = [0]
         prefix = self.menu.prefix.replace('/', '\\')
+        unicode_root_prefix = to_unicode(prefix)
         root_py  = join(unicode_root_prefix, u"python.exe")
         root_pyw = join(unicode_root_prefix, u"pythonw.exe")
         env_py  = join(prefix, u"python.exe")
-- 
2.22.0

