From fd51f435d16f894be7ed63fa24edab52798e3cd4 Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Sun, 17 Jan 2021 23:36:38 +0100
Subject: [PATCH] cmake-fixup

---
 CMakeLists.txt | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9de002f..802a727 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,14 +13,6 @@ project(xeus-sql)
 set(XEUS_SQL_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
 set(XEUS_TEST_DIR ${CMAKE_CURRENT_SOURCE_DIR}/test)
 
-set(CMAKE_FIND_ROOT_PATH ${CMAKE_PREFIX_PATH})
-set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
-set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
-set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
-set(CMAKE_FIND_FRAMEWORK NEVER)
-set(CMAKE_FIND_APPBUNDLE NEVER)
-set(CMAKE_PROGRAM_PATH ${CMAKE_PREFIX_PATH})
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 # Versioning
-- 
2.30.0

