From 0c07801350ff408b1a0c854db1eb62ba33af4d71 Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Mon, 18 Jan 2021 10:41:48 +0100
Subject: [PATCH] Patch FindSoci

---
 cmake/FindSoci.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindSoci.cmake b/cmake/FindSoci.cmake
index f4c1a0d..4e04b81 100644
--- a/cmake/FindSoci.cmake
+++ b/cmake/FindSoci.cmake
@@ -46,7 +46,7 @@ SET(SOCI_INCLUDE_DIRS ${SOCI_INCLUDE_DIR})
 #
 FIND_LIBRARY(
     SOCI_LIBRARY
-    NAMES soci_core
+    NAMES soci_core_4_0
     HINTS ${SOCI_INCLUDE_DIR}/../../lib
     PATH_SUFFIXES lib${LIB_SUFFIX})
 MARK_AS_ADVANCED(SOCI_LIBRARY)
-- 
2.30.0

