From 7f99e09ffae342892db8bb654ec930ddbc562b32 Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Mon, 28 Dec 2020 18:54:13 +0100
Subject: [PATCH] Link with xvega-bindings

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d90d00..e26dc3e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -221,6 +221,7 @@ macro(xsql_create_target target_name linkage output_name)
 
     target_link_libraries(${target_name} PUBLIC 
       ${XSQL_XEUS_TARGET}
+      xvega-bindings
       xvega
       ${SOCI_LIBRARY}
     )
-- 
2.23.0

