From 30b11e538d95790445ebf46aa6e5a1962429f60b Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Thu, 24 Dec 2020 08:53:35 +0100
Subject: [PATCH] Fixup xvega overlinking

---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d90d00..38b5f00 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,7 +72,6 @@ set(xeus_REQUIRED_VERSION 0.23.10)
 set(xvega_bindings_REQUIRED_VERSION 0.0.3)
 
 find_package(xeus ${xeus_REQUIRED_VERSION} REQUIRED)
-find_package(xvega REQUIRED)
 find_package(xvega-bindings ${xvega_bindings_REQUIRED_VERSION} REQUIRED)
 
 find_package(tabulate REQUIRED)
@@ -221,7 +220,7 @@ macro(xsql_create_target target_name linkage output_name)
 
     target_link_libraries(${target_name} PUBLIC 
       ${XSQL_XEUS_TARGET}
-      xvega
+      xvega-bindings
       ${SOCI_LIBRARY}
     )
 
-- 
2.23.0

