From 8c2def7e71893d4db62ff105dc8d094f71d78f80 Mon Sep 17 00:00:00 2001
From: Mariana Meireles <mariana@psychonautgirl.space>
Date: Mon, 4 Jan 2021 12:22:03 +0000
Subject: [PATCH 4/4] Maybe this flag is no longer needed?

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4311aa3..bff06c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,7 +121,7 @@ set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib:${CMAKE_INSTALL_PREFIX}/${C
 if (NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     message(STATUS "🔮")
     message(STATUS CMAKE_CXX_COMPILER_ID)
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --disable-new-dtags")
+    # set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --disable-new-dtags")
 endif()
 
 macro(xsql_set_common_options target_name)
-- 
2.25.1

