diff --git a/casaconfig/config.py b/casaconfig/config.py
index ddbdec0..1245e9e 100644
--- a/casaconfig/config.py
+++ b/casaconfig/config.py
@@ -42,8 +42,9 @@ from .private.get_argparser import get_argparser as __get_argparser
 __errors_encountered = { }
 
 def _standard_config_path( ):
-    standard_siteconfig_paths = [ '/opt/casa/casasiteconfig.py',
-                                  '/home/casa/casasiteconfig.py' ]
+    standard_siteconfig_paths = [
+        '@PREFIX@/share/casa/casasiteconfig.py'
+    ]
 
     if 'CASASITECONFIG' in __os.environ:
         f = __os.environ.get('CASASITECONFIG')
