diff -Naur pypy3.6-v7.3.0-src.orig/lib-python/3/distutils/tests/test_unixccompiler.py pypy3.6-v7.3.0-src/lib-python/3/distutils/tests/test_unixccompiler.py
--- pypy3.6-v7.3.0-src.orig/lib-python/3/distutils/tests/test_unixccompiler.py	2019-12-23 07:46:56.000000000 -0300
+++ pypy3.6-v7.3.0-src/lib-python/3/distutils/tests/test_unixccompiler.py	2020-03-11 16:11:47.650413549 -0300
@@ -67,7 +67,7 @@
             elif v == 'GNULD':
                 return 'yes'
         sysconfig.get_config_var = gcv
-        self.assertEqual(self.cc.rpath_foo(), '-Wl,--enable-new-dtags,-R/foo')
+        self.assertEqual(self.cc.rpath_foo(), '-Wl,-R/foo')
 
         # GCC non-GNULD
         sys.platform = 'bar'
@@ -88,7 +88,7 @@
             elif v == 'GNULD':
                 return 'yes'
         sysconfig.get_config_var = gcv
-        self.assertEqual(self.cc.rpath_foo(), '-Wl,--enable-new-dtags,-R/foo')
+        self.assertEqual(self.cc.rpath_foo(), '-Wl,-R/foo')
 
         # non-GCC GNULD
         sys.platform = 'bar'
diff -Naur pypy3.6-v7.3.0-src.orig/lib-python/3/distutils/unixccompiler.py pypy3.6-v7.3.0-src/lib-python/3/distutils/unixccompiler.py
--- pypy3.6-v7.3.0-src.orig/lib-python/3/distutils/unixccompiler.py	2019-12-23 07:46:56.000000000 -0300
+++ pypy3.6-v7.3.0-src/lib-python/3/distutils/unixccompiler.py	2020-03-11 16:11:44.062350279 -0300
@@ -260,12 +260,12 @@
                 # use it anyway.  Since distutils has always passed in
                 # -Wl whenever gcc was used in the past it is probably
                 # safest to keep doing so.
-                if sysconfig.get_config_var("GNULD") == "yes":
+                if sysconfig.get_config_var("GNULD") == "yes" or sys.platform == 'win32':
                     # GNU ld needs an extra option to get a RUNPATH
                     # instead of just an RPATH.
-                    return "-Wl,--enable-new-dtags,-R" + dir
-                else:
                     return "-Wl,-R" + dir
+                else:
+                    return "-Wl,--disable-new-dtags,-R" + dir
             else:
                 # No idea how --enable-new-dtags would be passed on to
                 # ld if this system was using GNU ld.  Don't know if a
