--- ./pypy3.6-v7.3.0-src.orig/lib-python/3/distutils/sysconfig_pypy.py	2019-12-23 07:46:56.000000000 -0300
+++ ./pypy3.6-v7.3.0-src/lib-python/3/distutils/sysconfig_pypy.py	2020-03-11 16:15:05.544853475 -0300
@@ -192,6 +192,9 @@
             linker_exe=cc,
             archiver=archiver)
 
+        if 'RANLIB' in os.environ and 'ranlib' in compiler.executables:
+            compiler.set_executables(ranlib=os.environ['RANLIB'])
+
         compiler.shared_lib_extension = shlib_suffix
 
 
