--- pypy3/rpython/rlib/clibffi.py	2019-02-27 14:04:22.000000000 +0100
+++ pypy3/rpython/rlib/clibffi.py	2019-02-27 14:04:41.000000000 +0100
@@ -82,6 +82,7 @@
         include_dirs = platform.include_dirs_for_libffi(),
         library_dirs = platform.library_dirs_for_libffi(),
         link_files = link_files,
+        link_extra = ['-Wl,-rpath,{}/lib'.format(os.environ['PREFIX'])],
         testonly_libraries = ['ffi'],
     )
 elif _MINGW:
