diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e86afe..6419d74 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,8 +111,14 @@ else()
     message("Google storage IO will not be available")
 endif()
 
+#linux
+if(UNIX AND NOT APPLE)
+    set(PDAL_TARGET "pdal_base")
+else()
+    set(PDAL_TARGET "pdalcpp")
+endif()
 
-get_target_property(PDALCPP_INCLUDE_DIRS pdalcpp INTERFACE_INCLUDE_DIRECTORIES)
+get_target_property(PDALCPP_INCLUDE_DIRS ${PDAL_TARGET} INTERFACE_INCLUDE_DIRECTORIES)
 if (PDALCPP_INCLUDE_DIRS)
     message("Including from PDAL: ${PDALCPP_INCLUDE_DIRS}")
     include_directories(${PDALCPP_INCLUDE_DIRS})
@@ -151,10 +157,17 @@ else()
     add_library(entwine SHARED ${OBJS})
 endif()
 
+#linux
+if(UNIX AND NOT APPLE)
+    set(PDAL_LIBRARIES "pdal_base;pdal_util")
+else()
+    set(PDAL_LIBRARIES "pdalcpp;pdal_util")
+endif()
+
 target_link_libraries(
     entwine
     PUBLIC
-    pdalcpp pdal_util
+    ${PDAL_LIBRARIES}
     ${LZMA_LIBRARY}
     ${CMAKE_DL_LIBS})
 
