diff --git a/setup.py b/setup.py
index c804524526d13257b864ec25f0510546af7dba59..d6c6dddaa6f36e7f72ec6a41899adf61e043fe8d 100755
--- a/setup.py
+++ b/setup.py
@@ -216,7 +216,7 @@ def extensions():
     result.append((['tkwant.onebody.solvers',
       ['tkwant/onebody/solvers.pyx']],
      {'libraries': [SCIPY_SOLVER_LIB],
-      'extra_link_args': ['-Wl,-rpath={}'.format(SCIPY_SOLVER_DIR)],
+      'extra_link_args': ['-Wl,-rpath,{}'.format(SCIPY_SOLVER_DIR)],
       'library_dirs': [SCIPY_SOLVER_DIR],
       'include_dirs': ['tkwant/onebody'],
       'depends': ['tkwant/onebody/kernels.pxd'],
