--- setup.py
+++ setup.py
@@ -379,13 +379,11 @@
         # Barebones packages needed to run symforce
         install_requires=maybe_rewrite_local_dependencies(
             [
-                "ruff",
-                "clang-format",
                 "graphviz",
                 "jinja2",
                 # numpy 2.0 isn't supported until SymPy 1.13.0
                 # https://github.com/sympy/sympy/wiki/Release-Notes-for-1.13.0
-                "numpy<2.0",
+                "numpy",
                 "scipy",
                 f"skymarshal @ file://localhost/{ESCAPED_SOURCE_DIR}/third_party/skymarshal",
-                "sympy~=1.11.1",
+                "sympy",
                 f"symforce-sym @ file://localhost/{ESCAPED_SOURCE_DIR}/gen/python",
--- third_party/skymarshal/pyproject.toml
+++ third_party/skymarshal/pyproject.toml
@@ -14,7 +14,7 @@
   "Operating System :: OS Independent",
 ]
 requires-python = ">=3.8"
-dependencies = ["argh", "jinja2", "numpy", "ply"]
+dependencies = ["jinja2", "numpy", "ply"]
 dynamic = ["version"]
 
 [project.urls]
