diff --git a/third_party/symenginepy/symengine/lib/symengine_wrapper.pyx b/third_party/symenginepy/symengine/lib/symengine_wrapper.pyx
index c2e72b0..5251426 100644
--- a/third_party/symenginepy/symengine/lib/symengine_wrapper.pyx
+++ b/third_party/symenginepy/symengine/lib/symengine_wrapper.pyx
@@ -1176,7 +1176,7 @@ cdef class Basic(object):
         return int(float(self))
 
     def __long__(self):
-        return long(float(self))
+        return int(float(self))
 
     def __complex__(self):
         f = self.n(real=False)
@@ -1559,7 +1559,9 @@ class Equality(Relational):
     def is_Equality(self):
         return True
 
-    func = __class__
+    @property
+    def func(self):
+        return self.__class__
 
 
 Eq = Equality
@@ -1580,7 +1582,9 @@ class Unequality(Relational):
         s = self.args_as_sage()
         return sage.ne(*s)
 
-    func = __class__
+    @property
+    def func(self):
+        return self.__class__
 
 
 Ne = Unequality
@@ -2846,35 +2850,35 @@ class UndefFunction(object):
         return function_symbol(self.name, *values)
 
 
-cdef rcp_const_basic pynumber_to_symengine(PyObject* o1):
+cdef rcp_const_basic pynumber_to_symengine(PyObject* o1) noexcept:
     cdef Basic X = sympify(<object>o1)
     return X.thisptr
 
-cdef PyObject* symengine_to_sage(rcp_const_basic o1):
+cdef PyObject* symengine_to_sage(rcp_const_basic o1) noexcept:
     import sage.all as sage
     t = sage.SR(c2py(o1)._sage_())
     Py_XINCREF(<PyObject*>t)
     return <PyObject*>(t)
 
-cdef PyObject* symengine_to_sympy(rcp_const_basic o1):
+cdef PyObject* symengine_to_sympy(rcp_const_basic o1) noexcept:
     t = c2py(o1)._sympy_()
     Py_XINCREF(<PyObject*>t)
     return <PyObject*>(t)
 
-cdef RCP[const symengine.Number] sympy_eval(PyObject* o1, long bits):
+cdef RCP[const symengine.Number] sympy_eval(PyObject* o1, long bits) noexcept:
     prec = max(1, int(round(bits/3.3219280948873626)-1))
     cdef Number X = sympify((<object>o1).n(prec))
     return symengine.rcp_static_cast_Number(X.thisptr)
 
-cdef RCP[const symengine.Number] sage_eval(PyObject* o1, long bits):
+cdef RCP[const symengine.Number] sage_eval(PyObject* o1, long bits) noexcept:
     cdef Number X = sympify((<object>o1).n(bits))
     return symengine.rcp_static_cast_Number(X.thisptr)
 
-cdef rcp_const_basic sage_diff(PyObject* o1, rcp_const_basic symbol):
+cdef rcp_const_basic sage_diff(PyObject* o1, rcp_const_basic symbol) noexcept:
     cdef Basic X = sympify((<object>o1).diff(c2py(symbol)._sage_()))
     return X.thisptr
 
-cdef rcp_const_basic sympy_diff(PyObject* o1, rcp_const_basic symbol):
+cdef rcp_const_basic sympy_diff(PyObject* o1, rcp_const_basic symbol) noexcept:
     cdef Basic X = sympify((<object>o1).diff(c2py(symbol)._sympy_()))
     return X.thisptr
 
