diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4fa8ff9..a5aa882 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,8 +28,10 @@ add_library(rosbag
   src/recorder.cpp
   src/time_translator.cpp)
 
+find_package(Threads)
+
 target_link_libraries(rosbag ${catkin_LIBRARIES} ${Boost_LIBRARIES}
-  ${BZIP2_LIBRARIES}
+  ${BZIP2_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}
 )
 
 add_executable(record src/record.cpp)
