From 34b869ef91567bd0fa6d8048517a4cc1eeeb0dc6 Mon Sep 17 00:00:00 2001
From: Maksim Rakitin <mrakitin@bnl.gov>
Date: Mon, 19 Jul 2021 12:03:17 -0400
Subject: [PATCH] Fix defaults for username

---
 pyOlog/conf.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyOlog/conf.py b/pyOlog/conf.py
index 21b269e..04acb9c 100644
--- a/pyOlog/conf.py
+++ b/pyOlog/conf.py
@@ -22,7 +22,7 @@ logger = logging.getLogger(__name__)
 
 class Config(object):
     defaults = {'url': 'http://localhost:8181/Olog',
-                'username': os.environ['USER']}
+                'username': getpass.getuser()}
     conf_files = ['/etc/pyOlog.conf',
                   os.path.expanduser('~/pyOlog.conf'),
                   os.path.expanduser('~/.pyOlog.conf'),
-- 
2.16.1

