From 185713729262af93e4a6b9f9184d0fc91c144dd2 Mon Sep 17 00:00:00 2001
From: ktro2828 <kotaro.uetake@tier4.jp>
Date: Thu, 20 Nov 2025 23:13:51 +0900
Subject: [PATCH] chore: update settings for setuptools

Signed-off-by: ktro2828 <kotaro.uetake@tier4.jp>
---
 pyproject.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index e4998de0..5fd6d5e0 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -82,8 +82,8 @@ train = [
 "Homepage" = "https://github.com/facebookresearch/sam3"
 "Bug Tracker" = "https://github.com/facebookresearch/sam3/issues"
 
-[tool.setuptools]
-packages = ["sam3", "sam3.model"]
+[tool.setuptools.packages.find]
+include = ["sam3*"]
 
 [tool.setuptools.dynamic]
 version = {attr = "sam3.__version__"}
