From af3bb774a82c56cda5b3ab77dc20cea3b5d5a8c5 Mon Sep 17 00:00:00 2001
From: Xylar Asay-Davis <xylarstorm@gmail.com>
Date: Tue, 17 Nov 2020 10:23:30 +0100
Subject: [PATCH] Update default.make for conda-forge

---
 mk/system/default.make | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/mk/system/default.make b/mk/system/default.make
index b217566..507583a 100644
--- a/mk/system/default.make
+++ b/mk/system/default.make
@@ -1,19 +1,18 @@
-# Copyright (c) 2020 Paul Ullrich 
+# Copyright (c) 2020 Paul Ullrich
 #
-# Distributed under the Boost Software License, Version 1.0. (See accompanying 
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-# UC Davis Agri 
+# UC Davis Agri
 
 # C++ compiler without and with MPI
-CXX=               g++
-MPICXX=            mpiCC
+MPICXX=            mpicxx
 
 # Additional C++ command line flags
 CXXFLAGS+=         -fPIC
 
 # NetCDF C library arguments
-NETCDF_ROOT=       $(NETCDF_HOME)
+NETCDF_ROOT=       $(PREFIX)
 NETCDF_CXXFLAGS=   -I$(NETCDF_ROOT)/include
 NETCDF_LIBRARIES=  -lnetcdf_c++ -lnetcdf
 NETCDF_LDFLAGS=    -L$(NETCDF_ROOT)/lib
-- 
2.29.1

