From 31cab5b75710240323a33d6280c41b155b7b2e87 Mon Sep 17 00:00:00 2001
From: Xylar Asay-Davis <xylarstorm@gmail.com>
Date: Tue, 17 Nov 2020 11:03:35 +0100
Subject: [PATCH] Update macosx.make for conda-forge

---
 mk/system/macosx.make | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/mk/system/macosx.make b/mk/system/macosx.make
index 4c9916e..20c95fb 100644
--- a/mk/system/macosx.make
+++ b/mk/system/macosx.make
@@ -1,18 +1,17 @@
-# Copyright (c) 2020 Paul Ullrich 
+# Copyright (c) 2020 Paul Ullrich
 #
-# Distributed under the Boost Software License, Version 1.0. (See accompanying 
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 # Mac OSX (Darwin)
 
 # C++ compiler without and with MPI
-CXX=               g++
 MPICXX=            mpicxx
 
 # Additional C++ command line flags
 
 # NetCDF C library arguments
-NETCDF_ROOT=       /opt/local
+NETCDF_ROOT=       $(PREFIX)
 NETCDF_CXXFLAGS=   -I$(NETCDF_ROOT)/include
 NETCDF_LIBRARIES=  -lnetcdf -lnetcdf_c++
 NETCDF_LDFLAGS=    -L$(NETCDF_ROOT)/lib -Wl,-rpath,$(NETCDF_CXX_ROOT)/lib
-- 
2.29.1

