From 993692410b918b4c30a67353505f943bd1fef527 Mon Sep 17 00:00:00 2001
From: Xylar Asay-Davis <xylarstorm@gmail.com>
Date: Tue, 17 Nov 2020 10:23:55 +0100
Subject: [PATCH] Disable MPI build

---
 mk/config.make | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mk/config.make b/mk/config.make
index a622eaf..87ef9aa 100644
--- a/mk/config.make
+++ b/mk/config.make
@@ -13,7 +13,7 @@
 
 DEBUG=    FALSE
 OPT=      TRUE
-PARALLEL= MPIOMP
+PARALLEL= NONE
 NETCDF=   TRUE
 
 # DO NOT DELETE
-- 
2.29.1

