From 5f446e7b0d4941132e7acdcf39f6d52804d810a7 Mon Sep 17 00:00:00 2001
From: "Dougal J. Sutherland" <dougal@gmail.com>
Date: Tue, 30 Jan 2018 17:39:13 +0000
Subject: [PATCH] try fixing __mm_undefined_ps definition

---
 include/adept/Packet.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/adept/Packet.h b/include/adept/Packet.h
index ebfadad..f3a1e11 100644
--- a/include/adept/Packet.h
+++ b/include/adept/Packet.h
@@ -30,7 +30,7 @@
 #define GCC_VERSION (__GNUC__ * 10000 \
                      + __GNUC_MINOR__ * 100 \
                      + __GNUC_PATCHLEVEL__)
-#if GCC_VERSION < 40901
+#ifndef _mm_undefined_ps
 #define _mm_undefined_ps _mm_setzero_ps
 #endif
 #endif // __clang__
-- 
2.7.4

