From 034671a4c8794e013573bf90e545492d6eb0409e Mon Sep 17 00:00:00 2001
From: "Dougal J. Sutherland" <dougal@gmail.com>
Date: Tue, 30 Jan 2018 18:37:59 +0000
Subject: [PATCH] try working around _mm_undefined_ps problem

---
 include/adept/Packet.h | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/include/adept/Packet.h b/include/adept/Packet.h
index ebfadad..f720eab 100644
--- a/include/adept/Packet.h
+++ b/include/adept/Packet.h
@@ -24,16 +24,11 @@
 #ifdef __SSE2__
 #include <xmmintrin.h> // SSE
 #include <emmintrin.h> // SSE2
-// GCC < 4.9.1 does not define _mm_undefined_ps in xmmintrin.h so we
+// GCC < 4.9.1 and old clang don't define _mm_undefined_ps in xmmintrin.h so we
 // use _mm_setzero_ps instead
-#ifndef __clang__
-#define GCC_VERSION (__GNUC__ * 10000 \
-                     + __GNUC_MINOR__ * 100 \
-                     + __GNUC_PATCHLEVEL__)
-#if GCC_VERSION < 40901
+#ifndef _mm_undefined_ps
 #define _mm_undefined_ps _mm_setzero_ps
 #endif
-#endif // __clang__
 #endif // __SSE2__
 
 #ifdef __AVX__
-- 
2.7.4

