diff --git a/setup.py b/setup.py
index 1c6a555..ca990d1 100755
--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@ except ImportError:
 from distutils.dir_util import remove_tree
 
 MODULE_NAME = "binwalk"
-MODULE_VERSION = "2.3.3"
+MODULE_VERSION = os.environ.get("PKG_VERSION")  # conda-forge
 SCRIPT_NAME = MODULE_NAME
 MODULE_DIRECTORY = os.path.dirname(os.path.realpath(__file__))
 
@@ -23,17 +23,6 @@ try:
 except ImportError:
     DEVNULL = open(os.devnull, 'wb')
 
-# If this version of binwalk was checked out from the git repository,
-# include the git commit hash as part of the version number reported
-# by binwalk.
-try:
-    label = subprocess.check_output(["git", "rev-parse", "--short", "HEAD"], stderr=DEVNULL).decode('utf-8')
-    MODULE_VERSION = "%s+%s" % (MODULE_VERSION, label.strip())
-except KeyboardInterrupt as e:
-    raise e
-except Exception:
-    pass
-
 # Python2/3 compliance
 try:
     raw_input
@@ -340,11 +329,6 @@ setup(
     package_dir={"": "src"},
     packages=[MODULE_NAME],
     package_data={MODULE_NAME: install_data_files},
-    scripts=[
-        os.path.join(
-            "src",
-            "scripts",
-            SCRIPT_NAME)],
     cmdclass={
         'clean': CleanCommand,
         'uninstall': UninstallCommand,
