From 3c84d3fb0ddd381f7c443ea6bf8818a7cdc00b10 Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Fri, 18 Sep 2020 11:42:16 +0200
Subject: [PATCH] Try getting CMAKE_CXX_COMPILER from os.environ

---
 setup.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 57c7868fa..37803b638 100644
--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,10 @@ def build_extension(self, ext):
            compiler_path = self.compiler.compiler_cxx[0]
            cmake_args.append("-DCMAKE_CXX_COMPILER={0}".format(compiler_path))
         except AttributeError:
-            print("Not able to access compiler path (on Windows), using CMake default")
+            if "CXX" in os.environ:
+                cmake_args.append("-DCMAKE_CXX_COMPILER={0}".format(os.environ["CXX"]))
+            else:
+                print("Not able to access compiler path and CXX not set (on Windows), using CMake default")
 
         cfg = "Debug" if self.debug else "Release"
         build_args = ["--config", cfg]
