From cf1b62b0a7dd0055584ca14792f3cbe5f0fead89 Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Fri, 18 Sep 2020 12:55:06 +0200
Subject: [PATCH] Try again

---
 setup.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 57c7868fa..61c85fc8d 100644
--- a/setup.py
+++ b/setup.py
@@ -67,14 +67,11 @@ def build_extension(self, ext):
                 "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{0}={1}".format(cfg.upper(), extdir),
                 "-DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=TRUE",
             ]
-            if sys.maxsize > 2**32:
-                cmake_args += ["-A", "x64"]
-            build_args += ["--", "/m"]
         else:
             cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
 
         if platform.system() == "Windows":
-            build_args += ["/m"]
+            pass
         else:
             build_args += ["-j", str(multiprocessing.cpu_count())]
 
