From b1db89598b72bf550e43d1b21c06a36214ce0f79 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Fri, 18 Sep 2020 15:15:34 +0200
Subject: [PATCH] Try using CMAKE_GENERATOR_PLATFORM

---
 setup.py | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/setup.py b/setup.py
index 57c7868..d0b2781 100644
--- a/setup.py
+++ b/setup.py
@@ -68,15 +68,11 @@ class CMakeBuild(setuptools.command.build_ext.build_ext):
                 "-DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=TRUE",
             ]
             if sys.maxsize > 2**32:
-                cmake_args += ["-A", "x64"]
-            build_args += ["--", "/m"]
+                cmake_args += ["-DCMAKE_GENERATOR_PLATFORM=x64"]
         else:
             cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
 
-        if platform.system() == "Windows":
-            build_args += ["/m"]
-        else:
-            build_args += ["-j", str(multiprocessing.cpu_count())]
+        build_args += ["-j", str(multiprocessing.cpu_count())]
 
         if not os.path.exists(self.build_temp):
              os.makedirs(self.build_temp)
-- 
2.28.0

