From 1cd0c5ed3983f56a867f31e07e8a70d79226f237 Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Fri, 18 Sep 2020 12:07:46 +0200
Subject: [PATCH] Try to set CMAKE_BUILD_TYPE on Windows

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 57c7868fa..0e8701914 100644
--- a/setup.py
+++ b/setup.py
@@ -70,8 +70,8 @@ def build_extension(self, ext):
             if sys.maxsize > 2**32:
                 cmake_args += ["-A", "x64"]
             build_args += ["--", "/m"]
-        else:
-            cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
+
+        cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
 
         if platform.system() == "Windows":
             build_args += ["/m"]
