diff --git a/setup.cfg b/setup.cfg.edit
index 11bf892..0ea3949 100644
--- a/setup.cfg
+++ b/setup.cfg.edit
@@ -13,6 +13,8 @@
 # to make it able to import numpy.
 
 #-------------------------------------------------------------------------------
+import os
+prefisso = os.environ.get('PREFIX', '')
 # Code_Aster toplevel directory (ex: /aster, /opt/aster...)
 ASTER_ROOT='/opt/aster'
 
@@ -104,9 +106,9 @@ USE_LOCATE = False
 #-------------------------------------------------------------------------------
 # You may want to add no standard directories in searched paths (as python list)
 # to search respectively for binaries, libraries and include files :
-#BINDIR=['/myprefix/bin', ]
-#LIBDIR=['/myprefix/lib', ]
-#INCLUDEDIR=['/myprefix/include', ]
+BINDIR=[prefisso+'/bin', ]
+LIBDIR=[prefisso+'/lib', ]
+INCLUDEDIR=[prefisso+'/include', ]
 
 # To search for shared libraries first
 PREFER_SHARED_LIBS=False   # False/True
