diff --git a/setup.cfg b/setup.cfg.edit
index 11bf892..cd7b5b6 100644
--- a/setup.cfg
+++ b/setup.cfg.edit
@@ -13,6 +13,8 @@
 # to make it able to import numpy.
 
 #-------------------------------------------------------------------------------
+import os
+prefisso = os.environ.get('PREFIX', '')
 # Code_Aster toplevel directory (ex: /aster, /opt/aster...)
 ASTER_ROOT='/opt/aster'
 
@@ -78,7 +80,7 @@ USE_LOCATE = False
 # A value defined here will be used for all compilers.
 #CC='/usr/bin/gcc'
 #CXX='/usr/bin/g++'
-#F90='/usr/bin/gfortran'
+F90=prefisso+'/gfortran'
 #LD=F90
 #CFLAGS="-fno-stack-protector"      # for gcc 4.x
