diff --git a/CMakeLists.txt b/CMakeLists.txt
index be1354e..e4fab34 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,9 +30,9 @@ set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/_version.cpp PROPERTIES
 add_library(libsmina
     ${sminasrc}
 )
-set_target_properties(libsmina PROPERTIES OUTPUT_NAME "smina")
+set_target_properties(libsmina PROPERTIES)
 target_include_directories(libsmina PUBLIC src/lib/)
-set(Boost_NO_BOOST_CMAKE ON)  #apparently required for newer boosts?
+# set(Boost_NO_BOOST_CMAKE ON)  #apparently required for newer boosts?
 find_package(Boost REQUIRED  COMPONENTS 
   iostreams 
   timer 
