--- setuporig.cfg	2024-07-01 13:18:32
+++ setupnew.cfg	2024-07-01 13:18:02
@@ -0,0 +1,121 @@
+[metadata]
+name = biogeme
+version = attr: biogeme.version.__version__
+description = Estimation and application of discrete choice models
+long_description = file: README.md
+long_description_content_type = text/markdown
+author = Michel Bierlaire
+author_email = michel.bierlaire@epfl.ch
+url = http://biogeme.epfl.ch
+download_urls = https://pypi.org/project/biogeme/
+project_urls =
+	Documentation = http://biogeme.epfl.ch/documents.html
+	Code = https://github.com/michelbierlaire/biogeme
+classifiers =
+	Intended Audience :: Education
+	Intended Audience :: Science/Research
+	Topic :: Scientific/Engineering
+	Topic :: Scientific/Engineering :: Mathematics
+	Natural Language :: English
+	Operating System :: OS Independent
+	Programming Language :: C++
+	Programming Language :: Cython
+	Programming Language :: Python
+	Programming Language :: Python :: 3
+	Programming Language :: Python :: 3.4
+	Programming Language :: Python :: 3.5
+	Programming Language :: Python :: 3.6
+	Programming Language :: Python :: 3.7
+	Programming Language :: Python :: 3 :: Only
+	Programming Language :: Python :: Implementation :: CPython
+
+[options]
+package_dir = =src
+packages = find:
+install_requires =
+    cython >= 0.29.16
+    pandas >= 1.3.5
+    scipy >= 1.7.3
+    tqdm >= 4.64.1
+    tomlkit >= 0.11.5
+
+[options.packages.find]
+where=src
+
+[options.extras_require]
+testing = cython >= 0.29.32
+    numpy >= 1.23.4
+    pytest >= 7.2.0
+    pytest-cov >= 4.0.0
+    tox >= 3.27.1
+
+[build]
+compiler=mingw32
+
+[build_ext]
+compiler=mingw32
+
+[cython-defaults]
+include_dirs = src/
+
+[cython-module: biogeme.cythonbiogeme]
+sources = src/biogeme/cpp/cythonbiogeme.pyx
+          src/biogeme/cpp/biogeme.cc
+          src/biogeme/cpp/evaluateExpressions.cc
+          src/biogeme/cpp/bioMemoryManagement.cc
+          src/biogeme/cpp/bioNormalCdf.cc
+          src/biogeme/cpp/bioFormula.cc
+          src/biogeme/cpp/bioSeveralFormulas.cc
+          src/biogeme/cpp/bioThreadMemory.cc
+          src/biogeme/cpp/bioThreadMemoryOneExpression.cc
+          src/biogeme/cpp/bioThreadMemorySimul.cc
+          src/biogeme/cpp/bioString.cc
+          src/biogeme/cpp/bioExprNormalCdf.cc
+          src/biogeme/cpp/bioExprIntegrate.cc
+          src/biogeme/cpp/bioExprGaussHermite.cc
+          src/biogeme/cpp/bioExprRandomVariable.cc
+          src/biogeme/cpp/bioExprMontecarlo.cc
+          src/biogeme/cpp/bioExprPanelTrajectory.cc
+          src/biogeme/cpp/bioExprDraws.cc
+          src/biogeme/cpp/bioExprDerive.cc
+          src/biogeme/cpp/bioExprMin.cc
+          src/biogeme/cpp/bioExprMax.cc
+          src/biogeme/cpp/bioExprAnd.cc
+          src/biogeme/cpp/bioExprOr.cc
+          src/biogeme/cpp/bioExprEqual.cc
+          src/biogeme/cpp/bioExprNotEqual.cc
+          src/biogeme/cpp/bioExprLessOrEqual.cc
+          src/biogeme/cpp/bioExprLess.cc
+          src/biogeme/cpp/bioExprGreaterOrEqual.cc
+          src/biogeme/cpp/bioExprGreater.cc
+          src/biogeme/cpp/bioExprElem.cc
+          src/biogeme/cpp/bioExprMultSum.cc
+          src/biogeme/cpp/bioExprLiteral.cc
+          src/biogeme/cpp/bioExprFreeParameter.cc
+          src/biogeme/cpp/bioExprFixedParameter.cc
+          src/biogeme/cpp/bioExprVariable.cc
+          src/biogeme/cpp/bioExprPlus.cc
+          src/biogeme/cpp/bioExprMinus.cc
+          src/biogeme/cpp/bioExprTimes.cc
+          src/biogeme/cpp/bioExprDivide.cc
+          src/biogeme/cpp/bioExprPower.cc
+          src/biogeme/cpp/bioExprUnaryMinus.cc
+          src/biogeme/cpp/bioExprExp.cc
+          src/biogeme/cpp/bioExprLog.cc
+          src/biogeme/cpp/bioExprLogzero.cc
+          src/biogeme/cpp/bioExprNumeric.cc
+          src/biogeme/cpp/bioExprLogLogit.cc
+          src/biogeme/cpp/bioExprLogLogitFullChoiceSet.cc
+          src/biogeme/cpp/bioExprLinearUtility.cc
+          src/biogeme/cpp/bioExpression.cc
+          src/biogeme/cpp/bioSeveralExpressions.cc
+          src/biogeme/cpp/bioExceptions.cc
+          src/biogeme/cpp/bioDerivatives.cc
+          src/biogeme/cpp/bioVectorOfDerivatives.cc
+          src/biogeme/cpp/bioGaussHermite.cc
+          src/biogeme/cpp/bioGhFunction.cc
+
+include_dirs = eval(__import__('numpy').get_include())
+language = c++
+extra_compile_args = -std=c++11 -DMS_WIN64
+extra_link_args = -std=c++11 -static-libstdc++ -static-libgcc -Bstatic -lpthread -mms-bitfields -mwindows
\ No newline at end of file
