--- setup.cfg
+++ setup.cfg
@@ -26,8 +26,8 @@
 include_dirs = eval(__import__('numpy').get_include())
 language = c++
 
-extra_compile_args = -std=c++11
-extra_link_args = -std=c++11
+extra_compile_args = -std=c++11 -DMS_WIN64
+extra_link_args = -static -std=c++11 -static-libstdc++ -static-libgcc -Bstatic -lpthread -mms-bitfields -mwindows -Wl,-Bstatic,--whole-archive -Wl,--no-whole-archive
 
 # For Windows