--- setuporig.cfg	2024-07-01 17:23:39
+++ setupnew.cfg	2024-07-01 17:23:35
@@ -33,8 +33,8 @@
     pandas >= 1.3.5
 
 # For Windows
-#[build]
-#compiler=mingw32
+[build]
+compiler=mingw32
 
 
 [options.packages.find]
\ No newline at end of file
@@ -115,9 +115,9 @@
 include_dirs = eval(__import__('numpy').get_include())
 language = c++
 
-extra_compile_args = -std=c++11
-extra_link_args = -std=c++11
+#extra_compile_args = -std=c++11
+#extra_link_args = -std=c++11
 
 # For Windows
-#extra_compile_args = -std=c++11 -DMS_WIN64
-#extra_link_args = -static -std=c++11 -static-libstdc++ -static-libgcc -Bstatic -lpthread -mms-bitfields -mwindows -Wl,-Bstatic,--whole-archive -Wl,--no-whole-archive
\ No newline at end of file
+extra_compile_args = -std=c++11 -DMS_WIN64
+extra_link_args = -std=c++11 -static-libstdc++ -static-libgcc -Bstatic -lpthread -mms-bitfields -mwindows
\ No newline at end of file
