From e3fa5affa2736c0f70fbb89f6d2ef808ab992bcb Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Fri, 25 Mar 2022 12:16:02 +0100
Subject: [PATCH 2/2] Remove test-only dependencies

---
 setup.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/setup.py b/setup.py
index 1c9465f..d4c6784 100644
--- a/setup.py
+++ b/setup.py
@@ -133,11 +133,7 @@ setup(
                       "networkx",
                       "numpy",
                       "vtk",
-                      "cython",
-                      "GitPython",
                       "configparser",
-                      "testtools",
-                      "pypandoc",
                       "sympy>=1.7"],
     ext_modules=exts,
     python_requires=">=3.7.1",
-- 
2.35.1

