--- setup.py	2019-06-07 00:13:19.932861859 -0400
+++ setup.py	2019-06-07 00:14:07.041155149 -0400
@@ -1,5 +1,6 @@
 import json
 import os
+import sys
 from setuptools import setup, find_packages
 from setuptools.command.install import install

@@ -14,7 +15,7 @@
         from jupyter_client.kernelspec import KernelSpecManager
         from IPython.utils.tempdir import TemporaryDirectory
         kernel_json = {
-            "argv": ["python", "-m", "ansible_kernel", "-f", "{connection_file}"],
+            "argv": [sys.executable, "-m", "ansible_kernel", "-f", "{connection_file}"],
             "codemirror_mode": "yaml",
             "display_name": "Ansible",
             "language": "ansible"
