=== modified file 'medm3_1_7/medm/medmCdev.h'
--- medm3_1_7/medm/medmCdev.h	2005-08-03 09:56:55.000000000 -0500
+++ medm3_1_7/medm/medmCdev.h	2012-11-02 10:08:59.398056800 -0500
@@ -119,7 +119,7 @@ typedef cdev_TS_STAMP           TS_STAMP
 
 #define MAX_ENUM_STATES		16
 #define db_state_dim		MAX_ENUM_STATES
-#define MAX_STRING_SIZE         40
+#define MAX_STRING_SIZE         60
 
 typedef struct _pv_type_name_
 {

