From 698e33a0d0ddcae341cfe817c010cb51dc368a8e Mon Sep 17 00:00:00 2001
From: Jonathan Hanks <jonathan.hanks@ligo.org>
Date: Mon, 29 Jun 2015 09:37:34 -0700
Subject: [PATCH] Updated version string

---
 extensions/src/medm3_1_9/medm/medm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/medm/medm.c b/extensions/src/medm3_1_9/medm/medm.c
index 33e9de5..030cd3b 100644
--- a/medm/medm.c
+++ b/medm/medm.c
@@ -29,7 +29,7 @@
 #define FULLPATHNAME_SIZE 100
 #define DEBUG_ERRORHANDLER 0
 
-#define LIGO_BUILD_NOTE "Unstable development build 2015-01-05+"
+#define LIGO_BUILD_NOTE "Unstable development build 2015-06-29+"
 
 #define ALLOCATE_STORAGE
 #include "medm.h"
-- 
1.9.5 (Apple Git-50.3)


