Index: medm/medm/Makefile
===================================================================
--- medm.orig/medm/Makefile
+++ medm/medm/Makefile
@@ -70,7 +70,7 @@ endif
 # Use to test SciPlot or JPT, otherwise should come from CONFIG
 # Set the one you want and reset any that come from CONFIG
 #XRTGRAPH = 
-#SCIPLOT = YES
+SCIPLOT = YES
 #JPT = YES
 
 # Use to test CDEV, Otherwise should come from CONFIG
