Index: medm/medm/medmWidget.c
===================================================================
--- medm.orig/medm/medmWidget.c
+++ medm/medm/medmWidget.c
@@ -228,39 +228,39 @@ void medmInit(char *displayFont)
    consistent with medmWidget.h definitions. */
     executePopupMenuButtonType[0] = XmPUSHBUTTON;
     executePopupMenuButtons[0] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_PRINT);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_CLOSE);
 
     executePopupMenuButtonType[1] = XmPUSHBUTTON;
     executePopupMenuButtons[1] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_CLOSE);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_PVINFO);
 
     executePopupMenuButtonType[2] = XmPUSHBUTTON;
     executePopupMenuButtons[2] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_PVINFO);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_PVLIMITS);
 
     executePopupMenuButtonType[3] = XmPUSHBUTTON;
     executePopupMenuButtons[3] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_PVLIMITS);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_MAIN);
 
     executePopupMenuButtonType[4] = XmPUSHBUTTON;
     executePopupMenuButtons[4] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_MAIN);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_DISPLAY_LIST);
 
     executePopupMenuButtonType[5] = XmPUSHBUTTON;
     executePopupMenuButtons[5] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_DISPLAY_LIST);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_FLASH_HIDDEN);
 
    executePopupMenuButtonType[6] = XmPUSHBUTTON;
     executePopupMenuButtons[6] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_FLASH_HIDDEN);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_REFRESH);
 
     executePopupMenuButtonType[7] = XmPUSHBUTTON;
     executePopupMenuButtons[7] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_REFRESH);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_RETRY);
 
     executePopupMenuButtonType[8] = XmPUSHBUTTON;
     executePopupMenuButtons[8] =
-      XmStringCreateLocalized(EXECUTE_POPUP_MENU_RETRY);
+      XmStringCreateLocalized(EXECUTE_POPUP_MENU_PRINT);
 
   /* Note that the Execute Menu is a cascade button */
     executePopupMenuButtonType[9] = XmCASCADEBUTTON;
Index: medm/medm/medmWidget.h
===================================================================
--- medm.orig/medm/medmWidget.h
+++ medm/medm/medmWidget.h
@@ -97,15 +97,15 @@ extern void popupValuatorKeyboardEntry(W
 #define EXECUTE_POPUP_MENU_REFRESH "Refresh"
 #define EXECUTE_POPUP_MENU_RETRY "Retry Connections"
 #define EXECUTE_POPUP_MENU_EXECUTE "Execute"
-#define EXECUTE_POPUP_MENU_PRINT_ID 0
-#define EXECUTE_POPUP_MENU_CLOSE_ID 1
-#define EXECUTE_POPUP_MENU_PVINFO_ID 2
-#define EXECUTE_POPUP_MENU_PVLIMITS_ID 3
-#define EXECUTE_POPUP_MENU_MAIN_ID 4
-#define EXECUTE_POPUP_MENU_DISPLAY_LIST_ID 5
-#define EXECUTE_POPUP_MENU_FLASH_HIDDEN_ID 6
-#define EXECUTE_POPUP_MENU_REFRESH_ID 7
-#define EXECUTE_POPUP_MENU_RETRY_ID 8
+#define EXECUTE_POPUP_MENU_CLOSE_ID 0
+#define EXECUTE_POPUP_MENU_PVINFO_ID 1
+#define EXECUTE_POPUP_MENU_PVLIMITS_ID 2
+#define EXECUTE_POPUP_MENU_MAIN_ID 3
+#define EXECUTE_POPUP_MENU_DISPLAY_LIST_ID 4
+#define EXECUTE_POPUP_MENU_FLASH_HIDDEN_ID 5
+#define EXECUTE_POPUP_MENU_REFRESH_ID 6
+#define EXECUTE_POPUP_MENU_RETRY_ID 7
+#define EXECUTE_POPUP_MENU_PRINT_ID 8
 /* The following must be the last item */
 #define EXECUTE_POPUP_MENU_EXECUTE_ID 9
 #define NUM_EXECUTE_POPUP_ENTRIES 10
