diff --git src/Output/reportwriter.cpp src/Output/reportwriter.cpp
index 452ecc6..de63a32 100644
--- src/Output/reportwriter.cpp
+++ src/Output/reportwriter.cpp
@@ -139,7 +139,7 @@ int  ReportWriter::writeReport(string inpFileName, OutputFile* outFile)
     if ( rptFileName2.length() > 0 )
     {
         // ... write report to this file
-        sout.open(rptFileName2);
+        sout.open(rptFileName2.c_str());
         if (!sout.is_open()) return FileError::CANNOT_OPEN_REPORT_FILE;
         usingRptFile2 = true;
