diff --git a/setup.py b/setup.py
index da4136f..fea3897 100644
--- a/setup.py
+++ b/setup.py
@@ -30,15 +30,7 @@ setup(
     license='MIT',
     zip_safe=True,
     py_modules=['sphinx_autodoc_typehints'],
-    setup_requires=[
-        'setuptools_scm >= 1.7.0'
-    ],
     install_requires=[
-        'Sphinx >= 1.4'
+        'sphinx >= 1.4'
     ],
-    extras_require={
-        'test': 'pytest >= 3.1.0',
-        ':python_version == "3.3"': 'typing >= 3.5',
-        ':python_version == "3.4"': 'typing >= 3.5'
-    }
 )
