--- trottersuzuki-1.6.2/setup.py
+++ trottersuzuki-1.6.2/setup.py
@@ -107,9 +107,6 @@
     if sys.platform.startswith('win'):
         extra_compile_args = ['-openmp', '-DWIN32']
         libraries = None
-    elif sys.platform.startswith('darwin') and 'CC' not in os.environ:
-        extra_compile_args = {'cc': []}
-        libraries = None
     else:
         extra_compile_args = {'cc': ['-fopenmp']}
         if 'CC' in os.environ and 'clang-omp' in os.environ['CC']:
