--- setup.py	2015-06-22 17:28:24.000000000 -0300
+++ setup.py	2015-09-07 20:02:02.594116419 -0300
@@ -21,7 +21,6 @@
     #version='0.0.2',
     version=__version__,
 
-    setup_requires=['setuptools-markdown'],
     description='Python Matplotlib, Numpy library to manage wind data, draw windrose (also known as a polar rose plot)',
     long_description_markdown_filename='README.md',
 
@@ -101,13 +100,4 @@
     # see http://docs.python.org/3.4/distutils/setupscript.html#installing-additional-files
     # In this case, 'data_file' will be installed into '<sys.prefix>/my_data'
     #data_files=[('my_data', ['data/data_file'])],
-
-    # To provide executable scripts, use entry points in preference to the
-    # "scripts" keyword. Entry points provide cross-platform support and allow
-    # pip to create the appropriate form of executable for the target platform.
-    entry_points={
-        'console_scripts': [
-            'sample=sample:main',
-        ],
-    },
 )
