diff --git a/setup.py b/setup.py
index 0f30055..23a1f95 100755
--- a/setup.py
+++ b/setup.py
@@ -8,8 +8,8 @@ from setuptools.command.build_py import build_py
 from subprocess import check_call
 import sys
 import os
+
 def enable_visual_interface():
-    check_call(f"{sys.executable} -m pip install jupyter", shell=True)
     import notebook
     notebook.nbextensions.install_nbextension_python(
         "checklist.viewer", user=True, overwrite=True)
