diff --git a/setup.py b/setup.py
index da79d0e..7b84c33 100644
--- a/setup.py
+++ b/setup.py
@@ -26,7 +26,8 @@ if __name__ == "__main__":
         install_requires=["cffi>=1.12.0"],
         # setup_requires=["cffi>=1.12.0", "wheel"],
         tests_require=[],
-        test_suite="setup.miniaudio_test_suite"
+        test_suite="setup.miniaudio_test_suite",
+        options={"bdist_wheel": {"py_limited_api": "cp310"}},
     )
 
 
