diff --git a/setup.py b/setup.py
index c7ec2887..1798cd1c 100644
--- a/setup.py
+++ b/setup.py
@@ -186,7 +186,7 @@ def ext_modules():
     if (SYS_NAME == "Windows") or (SYS_NAME == "Darwin"):
         num_jobs = 0
     else:
-        num_jobs = os.cpu_count()
+        num_jobs = os.getenv('CPU_COUNT') or os.cpu_count()
 
     # See https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#compiler-directives
     # for in-depth details on the options for compiler directives
@@ -296,7 +296,7 @@ def parallelise_build():
         """Function to monkey-patch Cython.Build.build_ext.build_extensions to build
         extensions in parallel."""
         self.check_extensions_list(self.extensions)
-        num_jobs = os.cpu_count()
+        num_jobs = os.getenv('CPU_COUNT') or os.cpu_count()
         try:
             with ProcessPoolExecutor(max_workers=num_jobs) as pool:
                 # results = pool.map(self.build_extension, self.extensions)
