diff --git a/include/mutils___float128.h b/include/mutils___float128.h
index 69e0cfb..7ff2c76 100644
--- a/include/mutils___float128.h
+++ b/include/mutils___float128.h
@@ -38,11 +38,9 @@ extern "C" {
 #endif
 
 namespace std {
-#ifndef _GLIBCXX_BITS_STD_ABS_H
 inline __float128 abs(const __float128 &a) {
    return fabsq(a);
 }
-#endif
 
 inline __float128 sqrt(const __float128 &a) {
    return sqrtq(a);
