From 602721753bd0fdfd02d293effe0186b892bfe72a Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Thu, 11 Apr 2024 16:00:36 -0500
Subject: [PATCH] remove rdkit

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 91a3bcf..b8740cf 100644
--- a/setup.py
+++ b/setup.py
@@ -58,7 +58,6 @@
             'scikit-learn',
             'matplotlib',
             'python-Levenshtein',
-            'rdkit-pypi',
             'scikit-optimize'
         ],
     include_package_data=True,
