From e789ed2ad6a28e1ae1e7fa8188a08afdac42267c Mon Sep 17 00:00:00 2001
From: Tim Paine <t.paine154@gmail.com>
Date: Sat, 18 Mar 2023 23:09:54 -0400
Subject: [PATCH] remove-static-stdc++

---
 cocotb_build_libs.py | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/cocotb_build_libs.py b/cocotb_build_libs.py
index 7e84ecf7..51edae7f 100755
--- a/cocotb_build_libs.py
+++ b/cocotb_build_libs.py
@@ -320,15 +320,6 @@ class build_ext(_build_ext):
                             "-Wl,-install_name,@rpath/%s.so" % install_name
                         ]
 
-                if sys.platform == "linux":
-                    # Avoid a runtime dependency on libstdc++. Some simulators
-                    # ship a version of libstdc++6.so which is older than the
-                    # one cocotb has been compiled with, which will then lead to
-                    # load-time errors like "libstdc++.so.6: version
-                    # `GLIBCXX_3.4.29' not found (required by
-                    # /path/to/libcocotbvhpi_modelsim.so)."
-                    ext.extra_link_args += ["-static-libstdc++"]
-
                 ext.extra_link_args += ["-Wl,-rpath,%s" % rpath for rpath in rpaths]
 
         # vpi_user.h and vhpi_user.h require that WIN32 is defined
-- 
2.40.0

