Index: setup.py
===================================================================
--- setup.py
+++ setup.py
@@ -31,9 +31,9 @@
       license = 'MIT License',
       
       py_modules = ['pyRMT'],
       platforms = 'ALL',
-      install_requires = ['numpy', 'pandas', 'sklearn'],
+      install_requires = ['numpy', 'pandas', 'scikit-learn'],
       setup_requires = ['numpy'],
                           
       classifiers = ['Development Status :: 4 - Beta',
                    'Environment :: Console',
