diff --git a/CMakeLists.txt b/CMakeLists.txt
index fed418f..523263b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,7 @@ add_test(NAME PyStringTest COMMAND pystring_test)
 include(GNUInstallDirs)
 
 install(TARGETS pystring
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}   # pystring.dll (Windows)
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}   # .so / .dylib
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}   # pystring.lib / static
 )
-