diff --git a/CMakeLists.txt b/CMakeLists.txt
index fed418f..1543bfc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,9 +1,9 @@
 cmake_minimum_required(VERSION 3.2)
 project(pystring CXX)
 
-set(BUILD_SHARED_LIBS YES)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
-add_library(pystring
+add_library(pystring SHARED
     pystring.cpp
     pystring.h
 )
@@ -17,6 +17,12 @@ add_test(NAME PyStringTest COMMAND pystring_test)
 include(GNUInstallDirs)
 
 install(TARGETS pystring
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}   # pystring.dll (Windows)
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}   # .so / .dylib
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}   # pystring.lib / static
 )
 
+install(
+    FILES pystring.h
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)