diff --git a/cmake/ImportJSON.cmake b/cmake/ImportJSON.cmake
index 7d93820..1a15583 100644
--- a/cmake/ImportJSON.cmake
+++ b/cmake/ImportJSON.cmake
@@ -5,6 +5,11 @@
 #
 
 macro(import_json)
+  if(NOT TARGET json)
+    find_package(nlohmann_json)
+  endif()
+
+  if(NOT TARGET json)
   set(JSON_VERSION "3.9.1")
   set(JSON_LICENSE_FILE "${CMAKE_CURRENT_BINARY_DIR}/nlohmann/LICENSE")
   set(JSON_HPP_FILE "${CMAKE_CURRENT_BINARY_DIR}/nlohmann/json.hpp")
@@ -31,4 +36,5 @@ macro(import_json)
     DESTINATION share/doc/molassembler/licenses
     RENAME nlohmann-json.txt
   )
+  endif()
 endmacro()
