diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 379e3898..d244e8de 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -88,16 +88,7 @@ if(PY_PIP)
       )
     endif()
 
-    # Copy Utils and Core (this works regardless of whether they are imported
-    # or build in-tree)
-    add_custom_command(TARGET scine_molassembler POST_BUILD
-      COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:Scine::UtilsOS> ${CMAKE_CURRENT_BINARY_DIR}/scine_molassembler
-      COMMENT "Copying dependent shared libraries into python package directory"
-    )
-    target_lib_filename(Scine::UtilsOS _utils_name)
-    set(molassembler_PY_DEPS ", \"${_utils_name}\"")
-    unset(_utils_name)
-    unset(_core_name)
+    set(molassembler_PY_DEPS "")
   endif()
   unset(_utils_libtype)
 
