diff --git a/requirements.txt b/requirements.txt
new file mode 100644
index 0000000..1631925
--- /dev/null
+++ b/requirements.txt
@@ -0,0 +1,13 @@
+albumentations==0.5.*
+cython==0.28.*
+future==0.18.*
+numpy<1.17
+pillow==5.0.*
+psutil==5.8.*
+pycocotools==2.0.*
+python
+rastervision-core==0.13.1
+rastervision-pipeline==0.13.1
+tensorboard==1.15.*
+torch==1.7.*
+torchvision==0.8.*
\ No newline at end of file
diff --git a/setup.py b/setup.py
index 4fc22bd..a2a6a91 100644
--- a/setup.py
+++ b/setup.py
@@ -31,4 +31,10 @@ setup(
     'raster deep-learning ml computer-vision earth-observation geospatial geospatial-processing',
     packages=find_namespace_packages(exclude=['integration_tests*', 'tests*']),
     install_requires=install_requires,
-    zip_safe=False)
+    zip_safe=False,
+    entry_points={
+        'console_scripts': [
+            'rastervision=rastervision.pytorch_learner.learner:main',
+        ],
+    }
+)
