diff --git a/ncempy/io/emdVelox.py b/ncempy/io/emdVelox.py
index f1b1596..0b3d4a1 100644
--- a/ncempy/io/emdVelox.py
+++ b/ncempy/io/emdVelox.py
@@ -12,8 +12,6 @@ import json
 import numpy as np
 import h5py
 
-import h5py_cache
-
 class fileEMDVelox:
     '''Class to represent Velox EMD files. It uses the h5py_cache module
     to increase the default cache size from 1MB to 10MB. This significantly
@@ -70,8 +68,7 @@ class fileEMDVelox:
 
         # try opening the file
         try:
-            #self.file_hdl = h5py.File(filename, 'r')
-            self.file_hdl = h5py_cache.File(filename, 'r', chunk_cache_mem_size=10*1024**2)
+            self.file_hdl = h5py.File(filename, 'r', rdcc_nbytes=10*1024**2)
         except:
             print('Error opening file for readonly: "{}"'.format(filename))
             raise
@@ -185,4 +182,4 @@ class fileEMDVelox:
         md['detectorName'] = detectorName
         
         return md
-    
\ No newline at end of file
+    
diff --git a/setup.py b/setup.py
index 88c30e4..02e9a64 100644
--- a/setup.py
+++ b/setup.py
@@ -73,7 +73,7 @@ setup(
     # your project is installed. For an analysis of "install_requires" vs pip's
     # requirements files see:
     # https://packaging.python.org/en/latest/requirements.html
-    install_requires=['numpy', 'scipy', 'matplotlib', 'h5py', 'h5py_cache'],       
+    install_requires=['numpy', 'scipy', 'matplotlib', 'h5py'],
 
     # List additional groups of dependencies here (e.g. development
     # dependencies). You can install these using the following syntax,
