Index: CMakeLists.txt
===================================================================
diff --git a/Omicron/trunk/CMakeLists.txt b/Omicron/trunk/CMakeLists.txt
--- a/Omicron/trunk/CMakeLists.txt	(revision 82557)
+++ b/Omicron/trunk/CMakeLists.txt	(working copy)
@@ -24,7 +24,7 @@
 find_package(PkgConfig REQUIRED)

 # specify the C++ standard
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)

 # -- set install paths ------
