Index: src/CMakeLists.txt
===================================================================
diff --git a/Omicron/trunk/src/CMakeLists.txt b/Omicron/trunk/src/CMakeLists.txt
--- a/Omicron/trunk/src/CMakeLists.txt	(revision 82562)
+++ b/Omicron/trunk/src/CMakeLists.txt	(working copy)
@@ -1,5 +1,8 @@
 # -- dependencies -----------

+# FFTW
+pkg_check_modules(FFTW REQUIRED "fftw3")
+
 # FrameL
 pkg_check_modules(FRAMEL REQUIRED "libframe")

@@ -22,7 +25,7 @@
 message(STATUS "HDF5 libraries: ${HDF5_INCLUDE_DIRS}")

 # add link paths
-link_directories(${FRAMEL_LIBRARY_DIRS} ${GWOLLUM_LIBRARY_DIRS})
+link_directories(${FFTW_LIBRARY_DIRS} ${FRAMEL_LIBRARY_DIRS} ${GWOLLUM_LIBRARY_DIRS})

 # versioning
 configure_file(Oconfig.h.in ${CMAKE_CURRENT_BINARY_DIR}/Oconfig.h @ONLY)
@@ -81,7 +84,8 @@
   ${ROOT_Core_LIBRARY}
   ${ROOT_Hist_LIBRARY}
   ${ROOT_MathCore_LIBRARY}
   ${ROOT_RIO_LIBRARY}
+  ${FFTW_LIBRARIES}
   CUtils
   Inject
   RootUtils
Index: src/Oomicron.h
===================================================================
diff --git a/Omicron/trunk/src/Oomicron.h b/Omicron/trunk/src/Oomicron.h
--- a/Omicron/trunk/src/Oomicron.h	(revision 82562)
+++ b/Omicron/trunk/src/Oomicron.h	(working copy)
@@ -11,6 +11,7 @@
 #include <Date.h>
 #include <InjEct.h>
 #include <ffl.h>
+#include <FFT.h>
 #include <TRandom.h>
 #include <TRandom3.h>
 #include <TriggerBuffer.h>
