diff --git a/wrappers/pyAgrum/CMakeLists.txt b/wrappers/pyAgrum/CMakeLists.txt
index 13f4304..dc720ea 100644
--- wrappers/pyAgrum/CMakeLists.txt
+++ wrappers/pyAgrum/CMakeLists.txt
@@ -13,41 +13,19 @@ else(WIN32)
   endif(APPLE)
 endif(WIN32)
 
-option(FOR_PYTHON2 "ON if pyAgrum will be compiler agains python2. Python3 is uses by default" "OFF")
-
-# PYTHON 2 OR 3
-if (FOR_PYTHON2) # we use 3 by default
-  set(PYTHONVERSION "2")
-  # hackish
-  if( WIN32)
-	set(PYTHON_EXECUTABLE "python.exe")
-  else(WIN32)
-	set(PYTHON_EXECUTABLE "python2")
-  endif(WIN32)
-  
-  find_package(PythonInterp REQUIRED)
-  set(CMAKE_SWIG_FLAGS  "-Wall" )
-  
-  #change in module naming from inside module between python2 and python3
-  set(PYAGRUM_MODULE "pyAgrum")
-else (FOR_PYTHON2) 
-  set(PYTHONVERSION "3")
-  # hackish
-  if(WIN32)
-	set(PYTHON_EXECUTABLE "python.exe")
-  else(WIN32)
-	set(PYTHON_EXECUTABLE "python3")
-  endif(WIN32)
-  
-  find_package(PythonInterp 3 REQUIRED)
-  
-  set(CMAKE_SWIG_FLAGS  "-Wall" )
+
+find_package(PythonInterp REQUIRED)
+
+set(CMAKE_SWIG_FLAGS  "-Wall" )
+set(PYAGRUM_MODULE "pyAgrum")
+
+if (PYTHON_VERSION_MAJOR VERSION_GREATER 2)
   set(CMAKE_SWIG_FLAGS  "${CMAKE_SWIG_FLAGS};-py3" )
-  
   #change in module naming from inside module between python2 and python3
   set(PYAGRUM_MODULE ".pyAgrum")
-endif (FOR_PYTHON2)
-set(GENERATED_PYTHON "${CMAKE_CURRENT_SOURCE_DIR}/generated-files${PYTHONVERSION}")
+endif ()
+
+set(GENERATED_PYTHON "${CMAKE_CURRENT_SOURCE_DIR}/generated-files${PYTHON_VERSION_MAJOR}")
 set(_PYAGRUMLIB "_pyAgrum")
 
 
