--- fast3tree/make.py
+++ fast3tree/make.py
@@ -51,11 +51,11 @@
             )
         )

-    call_args = "gcc -m64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_DEFAULT_SOURCE -D_POSIX_SOURCE -D_POSIX_C_SOURCE=200809L -D_SVID_SOURCE -D_DARWIN_C_SOURCE -Wall -fno-math-errno -fPIC -shared {0} -o {1} -lm -O3 -std=c99".format(
+    call_args = "${{CC}} ${{CFLAGS}} ${{LDFLAGS}} -m64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_DEFAULT_SOURCE -D_POSIX_SOURCE -D_POSIX_C_SOURCE=200809L -D_SVID_SOURCE -D_DARWIN_C_SOURCE -Wall -fno-math-errno -fPIC -shared {0} -o {1} -lm -O3 -std=c99".format(
         lib_c_path, lib_so_path
-    ).split()
+    )
     try:
-        check_call(call_args)
+        check_call(call_args, shell=True)
     except CalledProcessError:
         if raise_if_fall:
             raise
