--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -203,7 +203,10 @@ set(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
-add_subdirectory("${3RDPARTY_DIR}/tracy" "tracy" EXCLUDE_FROM_ALL)
+if(EXISTS "${3RDPARTY_DIR}/tracy/CMakeLists.txt")
+	add_subdirectory("${3RDPARTY_DIR}/tracy" "tracy" EXCLUDE_FROM_ALL)
+	disable_warnings_for_all_targets_in("${3RDPARTY_DIR}/tracy")
+endif()
-disable_warnings_for_all_targets_in("${3RDPARTY_DIR}/tracy")
 message(STATUS "Tracy: ${TRACY_ENABLE}")