From 9d9a7cd6d0db52d3e77b9e4f18cff7d490133201 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 23 Jul 2022 23:42:29 -0600
Subject: [PATCH] Remove unused entry point

---
 setup.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/setup.py b/setup.py
index c52f910..f245bd0 100644
--- a/setup.py
+++ b/setup.py
@@ -62,9 +62,4 @@
     install_requires=['python-snappy', 'multitasking', 'toolz', 'partd',
                       'cloudpickle', 'distributed', 'pandas', 'numpy',
                       'fastparquet', 'dask'],
-    entry_points={
-        'console_scripts': [
-            'sample=sample:main',
-        ],
-    },
 )
