indexPos was renamed to indexPosInt in LTL at 2010-07-29.

diff --git a/fitstools/comparefits.cpp b/fitstools/comparefits.cpp
index ed61611..82ff77b 100644
--- a/fitstools/comparefits.cpp
+++ b/fitstools/comparefits.cpp
@@ -264,7 +264,7 @@ int main(int argc, char* argv[])
    // circle stencil
    MArray<float, 2> circ( Range(-flagVal.length, flagVal.length),
                           Range(-flagVal.length, flagVal.length) );
-   circ = (pow2(indexPos(circ, 1)) + pow2(indexPos(circ, 2))) <
+   circ = (pow2(indexPosInt(circ, 1)) + pow2(indexPosInt(circ, 2))) <
       (flagVal.length * (flagVal.length + 1));
 
    // place holder for reference / file with additional length border
diff --git a/fitstools/convolvefits.cpp b/fitstools/convolvefits.cpp
index c9f7ab6..035a194 100644
--- a/fitstools/convolvefits.cpp
+++ b/fitstools/convolvefits.cpp
@@ -71,7 +71,7 @@ MArray<T, 2> getCircle(const int radius)
 {
    const int horizon2 = radius * radius + 1;
    MArray<T, 2> circle(Range(-radius, radius), Range(-radius, radius));
-   circle = ( (pow2(indexPos(circle, 1)) + pow2(indexPos(circle, 2))) <= horizon2 );
+   circle = ( (pow2(indexPosInt(circle, 1)) + pow2(indexPosInt(circle, 2))) <= horizon2 );
    return circle;
 }
 
diff --git a/fitstools/maskfits.cpp b/fitstools/maskfits.cpp
index a097bc9..608d6b7 100644
--- a/fitstools/maskfits.cpp
+++ b/fitstools/maskfits.cpp
@@ -251,7 +251,7 @@ MArray<T, 2> stencil<T>::getCircle(const int radius) const
 {
    const int horizon2 = radius * radius + 1;
    MArray<T, 2> circle(Range(-radius, radius), Range(-radius, radius));
-   circle = ( (pow2(indexPos(circle, 1)) + pow2(indexPos(circle, 2))) <= horizon2 );
+   circle = ( (pow2(indexPosInt(circle, 1)) + pow2(indexPosInt(circle, 2))) <= horizon2 );
    return circle;
 }
 
diff --git a/fitstools/smoothfits.cpp b/fitstools/smoothfits.cpp
index 9a11cd9..3218d86 100644
--- a/fitstools/smoothfits.cpp
+++ b/fitstools/smoothfits.cpp
@@ -159,8 +159,8 @@ int main( int argc, char *argv[] )
                                                             B.maxIndex(2)) ) ));
                      C.setBase( max((0 - flagVal.radius), (1 - ii(1))), 
                                 max((0 - flagVal.radius), (1 - ii(2))) );
-                     IndexList<2> l = where( pow2(indexPos(C,1)) +
-                                                   pow2(indexPos(C,2) <= limit) );
+                     IndexList<2> l = where( pow2(indexPosInt(C,1)) +
+                                                   pow2(indexPosInt(C,2) <= limit) );
                      B(ii(1), ii(2)) =
                         median_exact(C(l));
                      ++ii;
diff --git a/fitstools/smoothing.h b/fitstools/smoothing.h
index 1d2710d..c6fb7ec 100644
--- a/fitstools/smoothing.h
+++ b/fitstools/smoothing.h
@@ -1916,7 +1916,7 @@ MArray<T, 2> stencil<T>::getCircle(const int radius) const
 {
    const int horizon2 = radius * radius + 1;
    MArray<T, 2> circle(Range(-radius, radius), Range(-radius, radius));
-   circle = ( (pow2(indexPos(circle, 1)) + pow2(indexPos(circle, 2))) <= horizon2 );
+   circle = ( (pow2(indexPosInt(circle, 1)) + pow2(indexPosInt(circle, 2))) <= horizon2 );
    return circle;
 }
 
