diff --git a/fitstools/jpeg2fits.cpp.orig b/fitstools/jpeg2fits.cpp
index 5de30e2..069b053 100644
--- a/fitstools/jpeg2fits.cpp.orig
+++ b/fitstools/jpeg2fits.cpp
@@ -156,7 +156,7 @@ int main(int argc, char* argv[])
 void print_usage(OptionParser& flags)
 {
    const string usage =
-      string("Usage: fits2jpeg [-FLAGS [option]] FILES\n") +
+      string("Usage: jpeg2fits [-FLAGS [option]] FILES\n") +
       string("Convert JPEG FILES to FITS images.\n\n");
    const string copyright = 
       string("Copyright (C) 2007-2008 Claus A. Goessl <cag@usm.uni-muenchen.de>\n");
