diff --git a/fitstools/jpeg2fits.cpp b/fitstools/jpeg2fits.cpp
index 069b053..e6fb557 100644
--- a/fitstools/jpeg2fits.cpp
+++ b/fitstools/jpeg2fits.cpp
@@ -124,7 +124,7 @@ int main(int argc, char* argv[])
         // prepare save file
         const int bitpix = 8;
         const int naxis = 2;
-        const int naxis_i[2] = { cinfo.image_width, cinfo.image_height };
+        const int naxis_i[2] = { static_cast<int>(cinfo.image_width), static_cast<int>(cinfo.image_height) };
         const string outfilename = fits_filename(*file_i);
         FitsOut outfile(outfilename);
         outfile.openData(bitpix, naxis, naxis_i);
