From d33ccd835050e0c834a1e8ae0d51489c5114f6ac Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Mon, 6 May 2024 12:10:24 +0200
Subject: [PATCH] Do not include unnecessary files in wheel

---
 setup.py | 23 +++++++++++++++++++++--
 1 file changed, 21 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 9722581..841f3c7 100644
--- a/setup.py
+++ b/setup.py
@@ -16,6 +16,14 @@ def __init__(self, name, sourcedir=''):
 
 
 class CMakeBuild(build_ext):
+    def safe_delete_dir(self, dir_to_delete):
+        if os.path.exists(dir_to_delete):
+            shutil.rmtree(dir_to_delete)
+
+    def safe_delete_file(self, file_to_delete):
+        if os.path.exists(file_to_delete):
+            os.remove(file_to_delete)
+
     def run(self):
         try:
             out = subprocess.check_output(['cmake', '--version'])
@@ -30,13 +38,24 @@ def build_extension(self, ext):
 
         # make build directory
         build_dir = os.path.join(extdir, "egl_probe", "build")
-        if os.path.exists(build_dir):
-            shutil.rmtree(build_dir)
+        self.safe_delete_dir(build_dir)
         os.mkdir(build_dir)
 
         # build using cmake
         subprocess.check_call("cmake ..; make -j", cwd=build_dir, shell=True)
 
+        # Cleanup build directory
+        self.safe_delete_dir(os.path.join(build_dir, "CMakeFiles"))
+        self.safe_delete_file(os.path.join(build_dir, "CMakeCache.txt"))
+        self.safe_delete_file(os.path.join(build_dir, "Makefile"))
+        self.safe_delete_file(os.path.join(build_dir, "cmake_install.cmake"))
+
+        # Cleanup sources not needed at runtime
+        self.safe_delete_dir(os.path.join(extdir, "egl_probe", "glad"))
+        self.safe_delete_file(os.path.join(extdir, "egl_probe", "CMakeLists.txt"))
+        self.safe_delete_file(os.path.join(extdir, "egl_probe", "query_devices.cpp"))
+        self.safe_delete_file(os.path.join(extdir, "egl_probe", "test_device.cpp"))
+
 
 if sys.version_info.major == 3:
     with open("README.md", "r", encoding="utf-8") as fh:
