--- v4-client-cpp/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -19,2 +19,3 @@

-find_package(Catch2 REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(Catch2 REQUIRED IMPORTED_TARGET Catch2)
@@ -24,3 +24,2 @@

-add_subdirectory(lib)
-add_subdirectory(examples)
\ No newline at end of file
+add_subdirectory(lib/proto)
