--- v4-client-cpp/lib/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -1,7 +1,8 @@
 # Dependencies
-find_package(Boost COMPONENTS beast REQUIRED)
-find_package(OpenSSL REQUIRED)
-find_package(bip3x REQUIRED)
+find_package(Boost 1.82.0 REQUIRED)
+find_package(OpenSSL 3 REQUIRED)
+find_package(toolbox REQUIRED)
+find_package(bip3x 3 REQUIRED)
 find_package(fmt REQUIRED)
-find_package(json REQUIRED)
+find_package(nlohmann_json 3.11.3 REQUIRED)

--- v4-client-cpp/cmake/modules/Findbip3x.cmake	2021-10-14 15:00:00.000000000 +0000
+++ /dev/null        	2021-10-14 15:00:00.000000000 +0000
@@ -1,12 +0,0 @@
-Include(FetchContent)
-
-set(FETCHCONTENT_QUIET OFF)
-
-FetchContent_Declare(
-        bip3x
-        GIT_REPOSITORY https://github.com/asnefedovv/dydx-v4-cpp-dep-bip3x.git
-        GIT_TAG 3.0.0
-        GIT_PROGRESS TRUE
-)
-
-FetchContent_MakeAvailable(bip3x)
