--- v4-client-cpp/lib/proto/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/proto/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -3,6 +3,12 @@
 file(GLOB_RECURSE proto_files RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.proto)
-protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS ${proto_files})

-add_library(dydx_v4_proto ${BALOO_BIN_TYPE} ${PROTO_SRCS} ${PROTO_HDRS})
-target_include_directories(dydx_v4_proto PUBLIC ${Protobuf_INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(dydx_v4_proto PUBLIC ${Protobuf_LIBRARIES})
\ No newline at end of file
+include(GNUInstallDirs)
+set(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include/dydx_v4_proto")
+
+include($ENV{RECIPE_DIR}/cmake/1_GenerateProtobufFiles.txt)
+include($ENV{RECIPE_DIR}/cmake/2_AddExportMacroHeaders.txt)
+include($ENV{RECIPE_DIR}/cmake/3_BuildDYDXLib.txt)
+include($ENV{RECIPE_DIR}/cmake/4_InstallLibrary.txt)
+include($ENV{RECIPE_DIR}/cmake/5_InstallExportHeaders.txt)
+include($ENV{RECIPE_DIR}/cmake/6_InstallCmakeConfigFiles.txt)
+include($ENV{RECIPE_DIR}/cmake/7_InstallPkgConfigFiles.txt)

