From 14e896a3e62ad4d0a425e9822abeb885d9730a5c Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Tue, 26 May 2020 12:10:09 -0400
Subject: [PATCH] Require CMake version 3.2.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index caa894f..5c42604 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.11)
+cmake_minimum_required(VERSION 3.2)
 
 # Project name
 project(gqrx)
-- 
2.20.1

