From 482d7d108a2d6e29687f6cb96fd05ef6532c60cf Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Fri, 22 May 2020 18:01:04 -0400
Subject: [PATCH] Add missing project directive to src/CMakeLists.txt.

---
 src/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ffe3220..815b4cf 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,3 +1,4 @@
+project(gqrx)
 #######################################################################################################################
 # Add the source subdirectories
 add_subdirectory(applications)
-- 
2.20.1

