From e4ee00008b3664d7b1fc40fda725ed9a992ade34 Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Fri, 22 May 2020 16:07:22 -0400
Subject: [PATCH] Don't set CMAKE_OSX_DEPLOYMENT_TARGET (conda-forge handles
 things).

---
 src/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ffe3220..a1490ac 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -13,7 +13,6 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
         add_subdirectory(osxaudio)
     endif()
     add_definitions(-DGQRX_OS_MACX)
-    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.9)
 elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     if(${LINUX_AUDIO_BACKEND} MATCHES "Pulseaudio")
         add_subdirectory(pulseaudio)
-- 
2.20.1

