From dda703aa6d5336fdf353da20884887a783f5d610 Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Wed, 20 May 2020 11:16:20 -0400
Subject: [PATCH 1/2] Add a definition for timeval struct when using MSVC (it
 was missing).

---
 src/qtgui/plotter.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/qtgui/plotter.cpp b/src/qtgui/plotter.cpp
index a9d9cff..fa20f32 100644
--- a/src/qtgui/plotter.cpp
+++ b/src/qtgui/plotter.cpp
@@ -35,6 +35,11 @@
 #include <Windows.h>
 #include <cstdint>
 
+typedef struct timeval {
+    long tv_sec;
+    long tv_usec;
+} timeval;
+
 int gettimeofday(struct timeval * tp, struct timezone * tzp)
 {
     // Note: some broken versions only have 8 trailing zero's, the correct epoch has 9 trailing zero's
-- 
2.20.1

