From 0b53a4ff437796a964160107b4f4412082be1f65 Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Tue, 26 May 2020 11:02:59 -0400
Subject: [PATCH 2/2] Remove project command from top-level CMakeLists.txt.

---
 CMakeLists.txt | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index caa894f..d0634ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,12 +1,12 @@
 cmake_minimum_required(VERSION 2.8.11)
 
 # Project name
-project(gqrx)
-set(${PROJECT_NAME}_MAJOR "2")
-set(${PROJECT_NAME}_MINOR "12")
-set(${PROJECT_NAME}_PATCH "1")
-##set(VERSION "${${PROJECT_NAME}_MAJOR}.${${PROJECT_NAME}_MINOR}.${${PROJECT_NAME}_PATCH}")
-set(VERSION "${${PROJECT_NAME}_MAJOR}.${${PROJECT_NAME}_MINOR}")
+set(PACKAGE "gqrx")
+set(${PACKAGE}_MAJOR "2")
+set(${PACKAGE}_MINOR "12")
+set(${PACKAGE}_PATCH "1")
+##set(VERSION "${${PACKAGE}_MAJOR}.${${PACKAGE}_MINOR}.${${PACKAGE}_PATCH}")
+set(VERSION "${${PACKAGE}_MAJOR}.${${PACKAGE}_MINOR}")
 add_definitions(-DVERSION="${VERSION}")
 
 # development version
@@ -18,8 +18,6 @@ execute_process(
 )
 #add_definitions(-DVERSION="${GITVERSION}")
 
-set(PACKAGE ${PROJECT_NAME})
-
 ########### Main global variables ###########
 if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: Debug GProf Valgrind Release" FORCE)
@@ -30,7 +28,7 @@ string(TOUPPER ${BUILDTYPE} BUILDTYPE)
 add_definitions(-D${BUILDTYPE})
 
 # We have some custom .cmake scripts not in the official distribution.
-set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 
 # Add valgrind build options if necessary
 if(${CMAKE_BUILD_TYPE} MATCHES "Valgrind")
-- 
2.20.1

