--- deps.edn    2024-05-01 13:10:35.000000000 -0500
+++ deps.edn    2024-06-13 21:11:11.017145256 -0500
@@ -7,2 +7,5 @@

+ :mvn/repos
+ {"local" {:url "file:C:/Users/VssAdministrator/.m2/repository"}}
+
  :aliases
@@ -11,3 +14,3 @@
   :build
-  {:deps {io.github.clojure/tools.build {:git/tag "v0.10.3" :git/sha "15ead66"}}
+  {:deps {io.github.clojure/tools.build {:mvn/version "0.10.3"}}
    :paths ["script"]
