diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c90f5d..01bdc72 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,4 +128,5 @@ set(RE2_HEADERS
     )

 install(FILES ${RE2_HEADERS} DESTINATION include/re2)
-install(TARGETS re2 ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
+install(TARGETS re2 ARCHIVE DESTINATION lib LIBRARY DESTINATION lib
+                    RUNTIME DESTINATION bin)
