From 5b7b2a5519f712a7831053865a4ae6082dac4ed5 Mon Sep 17 00:00:00 2001
From: Peter Majchrak <petoknm@gmail.com>
Date: Sat, 22 Jul 2017 19:37:05 +0200
Subject: [PATCH] Add 'include' to the install path of the headers

---
 projects/CMake/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/projects/CMake/CMakeLists.txt b/projects/CMake/CMakeLists.txt
index 6130d0591..7230ca782 100644
--- a/projects/CMake/CMakeLists.txt
+++ b/projects/CMake/CMakeLists.txt
@@ -133,5 +133,5 @@ SET_TARGET_PROPERTIES(RxCpp PROPERTIES LINKER_LANGUAGE CXX)
 
 set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY TRUE CACHE BOOL "Don't require all projects to be built in order to install" FORCE)
 
-install(DIRECTORY ${RXCPP_DIR}/Rx/v2/src/rxcpp/ DESTINATION rxcpp
+install(DIRECTORY ${RXCPP_DIR}/Rx/v2/src/rxcpp/ DESTINATION include/rxcpp
         FILES_MATCHING PATTERN "*.hpp")
