From 8ee148ed8f2ee690a57bb4d774f130f54a19c9ee Mon Sep 17 00:00:00 2001
From: HeavyTony2 <97621927+HeavyTony2@users.noreply.github.com>
Date: Tue, 18 Jan 2022 23:18:45 -0500
Subject: [PATCH] Update disc_win32.c

---
 src/disc_win32.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/disc_win32.c b/src/disc_win32.c
index df9409d..e1222a5 100644
--- a/src/disc_win32.c
+++ b/src/disc_win32.c
@@ -31,13 +31,9 @@
 #include <string.h>
 #include <stdio.h>
 
-#if defined(__CYGWIN__)
-#include <ntddcdrm.h>
-#elif defined(__MINGW32__)
-#include <ddk/ntddcdrm.h>
-#else
+
 #include "ntddcdrm.h"
-#endif
+
 
 #include "discid/discid.h"
 #include "discid/discid_private.h"
-- 
2.34.1.windows.1

