diff --git a/CMakeLists.txt b/CMakeLists.txt
index 72bdbd5..fefcfca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,6 +64,8 @@ ELSEIF(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
     SET(libdiscid_OSDEP_LIBS util) # for getrawpartition
 ELSEIF(CMAKE_SYSTEM_NAME MATCHES "SunOS")
     SET(libdiscid_OS "solaris")
+ELSEIF(CMAKE_SYSTEM_NAME MATCHES "Haiku")
+    SET(libdiscid_OS "haiku")
 ELSEIF(CMAKE_SYSTEM_NAME MATCHES "Windows")
     SET(libdiscid_OS "win32")
 ELSEIF(CMAKE_SYSTEM_NAME MATCHES "CYGWIN")
@@ -74,17 +76,18 @@ ELSE()
 ENDIF()
 
 # choose platform dependent source files
-IF(libdiscid_OS EQUAL "win32")
+IF(libdiscid_OS STREQUAL "win32")
     SET(libdiscid_OSDEP_SRCS src/toc.c src/disc_win32.c)
     SET(libdiscid_RCS ${CMAKE_CURRENT_BINARY_DIR}/versioninfo.rc)
-ELSEIF(libdiscid_OS EQUAL "generic")
+ELSEIF(libdiscid_OS STREQUAL "generic")
     SET(libdiscid_OSDEP_SRCS src/disc_${libdiscid_OS}.c)
 ELSE()
     # unix platforms are the standard/default case
     SET(libdiscid_OSDEP_SRCS src/toc.c src/unix.c src/disc_${libdiscid_OS}.c)
-    IF(libdiscid_OS EQUAL "darwin") # Extra libraries needed
+    IF(libdiscid_OS STREQUAL "darwin") # Extra libraries needed
         FIND_LIBRARY(COREFOUNDATION_LIBRARY CoreFoundation)
-        SET(libdiscid_OSDEP_LIBS ${COREFOUNDATION_LIBRARY} IOKit)
+        FIND_LIBRARY(IOKIT_LIBRARY IOKit)
+        SET(libdiscid_OSDEP_LIBS ${COREFOUNDATION_LIBRARY} ${IOKIT_LIBRARY})
     ENDIF()
 ENDIF()
 
