From a28f4c1ce9d22fe2cc7e59c28fe40a8506bd4441 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 30 Jan 2022 22:27:31 -0500
Subject: [PATCH] Fix destination directory

---
 CMakeLists.txt | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4dba035..3b0f75e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -266,8 +266,7 @@ endif()
 # -------------------------------------------------------- install library
 if (HWY_ENABLE_INSTALL)
 
-install(TARGETS hwy
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS hwy)
 # Install all the headers keeping the relative path to the current directory
 # when installing them.
 foreach (source ${HWY_SOURCES})
@@ -278,8 +277,7 @@ foreach (source ${HWY_SOURCES})
   endif()
 endforeach()
 
-install(TARGETS hwy_contrib
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS hwy_contrib)
 # Install all the headers keeping the relative path to the current directory
 # when installing them.
 foreach (source ${HWY_CONTRIB_SOURCES})
@@ -290,8 +288,7 @@ foreach (source ${HWY_CONTRIB_SOURCES})
   endif()
 endforeach()
 
-install(TARGETS hwy_test
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS hwy_test)
 # Install all the headers keeping the relative path to the current directory
 # when installing them.
 foreach (source ${HWY_TEST_SOURCES})
-- 
2.32.0

