From 16aff81a303e50e2622f3541977fabbc00d394bd Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 30 Jan 2022 09:43:39 -0500
Subject: [PATCH] Export chosen_target correctly

---
 hwy/targets.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hwy/targets.cc b/hwy/targets.cc
index daab3a6..f302b13 100644
--- a/hwy/targets.cc
+++ b/hwy/targets.cc
@@ -345,7 +345,7 @@ uint32_t SupportedTargets() {
 }
 
 // Declared in targets.h
-ChosenTarget chosen_target;
+HWY_DLLEXPORT ChosenTarget chosen_target;
 
 void ChosenTarget::Update() {
   // The supported variable contains the current CPU supported targets shifted
-- 
2.32.0

