diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4dba035..833a4bc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -221,7 +221,8 @@ target_include_directories(hwy PUBLIC ${CMAKE_CURRENT_LIST_DIR})
 target_compile_features(hwy PUBLIC cxx_std_11)
 set_target_properties(hwy PROPERTIES
   LINK_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/hwy/hwy.version)
-if(UNIX)
+if(APPLE)
+elseif(UNIX)
   set_property(TARGET hwy APPEND_STRING PROPERTY
     LINK_FLAGS " -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/hwy/hwy.version")
 endif()
