From 9d27afc39aba1c96dd70222542e7ffa3cbde43d7 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 30 Jan 2022 23:03:12 -0500
Subject: [PATCH 5/5] Fix destination directory

---
 CMakeLists.txt | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7c43cb1..4d69b3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -249,8 +249,7 @@ add_custom_command(TARGET hwy_list_targets POST_BUILD
 # -------------------------------------------------------- install library
 if (HWY_ENABLE_INSTALL)
 
-install(TARGETS hwy
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS hwy)
 # Install all the headers keeping the relative path to the current directory
 # when installing them.
 foreach (source ${HWY_SOURCES})
@@ -261,8 +260,7 @@ foreach (source ${HWY_SOURCES})
   endif()
 endforeach()
 
-install(TARGETS hwy_contrib
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS hwy_contrib)
 # Install all the headers keeping the relative path to the current directory
 # when installing them.
 foreach (source ${HWY_CONTRIB_SOURCES})
@@ -273,8 +271,7 @@ foreach (source ${HWY_CONTRIB_SOURCES})
   endif()
 endforeach()
 
-install(TARGETS hwy_test
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS hwy_test)
 # Install all the headers keeping the relative path to the current directory
 # when installing them.
 foreach (source ${HWY_TEST_SOURCES})
-- 
2.32.0

