diff --git a/setup.py b/setup.py
index ada966a..ed1faf3 100644
--- setup.py
+++ setup.py
@@ -5,8 +5,8 @@ https://packaging.python.org/en/latest/distributing.html
 https://github.com/pypa/sampleproject
 """
 
-import ez_setup
-ez_setup.use_setuptools()
+#import ez_setup
+#ez_setup.use_setuptools()
 
 # Always prefer setuptools over distutils
 from setuptools import setup, find_packages
@@ -112,4 +112,4 @@ setup(
     #        'sample=sample:main',
     #    ],
     #},
-)
\ No newline at end of file
+)
