diff --git a/libtiledbsoma/src/CMakeLists.txt b/libtiledbsoma/src/CMakeLists.txt
index a6d233df..e797559b 100644
--- a/libtiledbsoma/src/CMakeLists.txt
+++ b/libtiledbsoma/src/CMakeLists.txt
@@ -258,6 +258,13 @@ install(FILES
   DESTINATION "include/tiledbsoma/nlohmann"
 )
 
+install(FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/external/include/thread_pool/producer_consumer_queue.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/external/include/thread_pool/status.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/external/include/thread_pool/thread_pool.h
+  DESTINATION "include/tiledbsoma/thread_pool"
+)
+
 
 # ###########################################################
 # API symbol exports
@@ -300,6 +307,10 @@ endif()
 # CLI executable target
 # ###########################################################
 if(TILEDBSOMA_BUILD_CLI)
+
+  # Add this before including spdlog headers
+  add_definitions(-DSPDLOG_COMPILED_LIB)
+
   add_executable(tiledbsoma-cli
     ${CMAKE_CURRENT_SOURCE_DIR}/cli/cli.cc
     $<TARGET_OBJECTS:TILEDB_SOMA_OBJECTS>
@@ -335,6 +346,13 @@ if(TILEDBSOMA_BUILD_CLI)
     ${TILEDB_SOMA_EXPORT_HEADER_DIR}
     ${pybind11_INCLUDE_DIRS}
   )
+
+  # Ensure spdlog is properly linked
+  target_link_libraries(tiledbsoma PRIVATE spdlog::spdlog)
+
+  # Ensure proper visibility settings
+  set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+
 endif()
 
 # ###########################################################
diff --git a/libtiledbsoma/src/utils/fastercsx.h b/libtiledbsoma/src/utils/fastercsx.h
index 7cb460ae..7584b65c 100644
--- a/libtiledbsoma/src/utils/fastercsx.h
+++ b/libtiledbsoma/src/utils/fastercsx.h
@@ -35,7 +35,7 @@
 #include <cstdint>
 #include <format>
 #include <numeric>
-#include "span/span.hpp"
+#include "tiledbsoma/soma/span/span.hpp"
 
 #include "parallel_functions.h"
 
diff --git a/libtiledbsoma/src/utils/parallel_functions.h b/libtiledbsoma/src/utils/parallel_functions.h
index 77d3ad41..b408b5f7 100644
--- a/libtiledbsoma/src/utils/parallel_functions.h
+++ b/libtiledbsoma/src/utils/parallel_functions.h
@@ -33,7 +33,7 @@
 #ifndef TILEDB_PARALLEL_FUNCTIONS_H
 #define TILEDB_PARALLEL_FUNCTIONS_H
 
-#include <thread_pool/thread_pool.h>
+#include "tiledbsoma/thread_pool/thread_pool.h"
 
 #include <algorithm>
 #include <cassert>
