diff --git a/app/main.f90 b/app/main.f90
index 512e848..51c138f 100644
--- a/app/main.f90
+++ b/app/main.f90
@@ -16,8 +16,8 @@
 
 program main
    use, intrinsic :: iso_fortran_env, only : output_unit, error_unit, input_unit
-   use mctc_env
-   use mctc_io
+   use mctc_env, only : error_type, fatal_error, wp
+   use mctc_io, only : structure_type, read_structure, filetype, get_filetype
    use gcp, only : gcp_call, wregrad_tm, get_gcp_version
    use gcp_strings, only : lowercase
    implicit none
diff --git a/fpm.toml b/fpm.toml
index 6b02906..68c68d9 100644
--- a/fpm.toml
+++ b/fpm.toml
@@ -10,10 +10,12 @@ keywords = ["counterpoise", "computational-chemistry"]
 [dependencies]
 [dependencies.mctc-lib]
 git = "https://github.com/grimme-lab/mctc-lib"
+tag = "v0.2.0"
 
 [dev-dependencies]
 [dev-dependencies.mstore]
 git = "https://github.com/grimme-lab/mstore"
+tag = "v0.1.1"
 
 [[executable]]
 name = "mctc-gcp"
diff --git a/subprojects/mctc-lib.wrap b/subprojects/mctc-lib.wrap
index 71ad212..2d793a1 100644
--- a/subprojects/mctc-lib.wrap
+++ b/subprojects/mctc-lib.wrap
@@ -1,4 +1,4 @@
 [wrap-git]
 directory = mctc-lib
 url = https://github.com/grimme-lab/mctc-lib
-revision = head
+revision = v0.2.0
diff --git a/subprojects/mstore.wrap b/subprojects/mstore.wrap
index 37e72cf..ebee249 100644
--- a/subprojects/mstore.wrap
+++ b/subprojects/mstore.wrap
@@ -1,4 +1,4 @@
 [wrap-git]
 directory = mstore
 url = https://github.com/grimme-lab/mstore
-revision = head
+revision = v0.1.1
