diff --git a/tools/clang/tools/dxcompiler/CMakeLists.txt b/tools/clang/tools/dxcompiler/CMakeLists.txt
--- a/tools/clang/tools/dxcompiler/CMakeLists.txt
+++ b/tools/clang/tools/dxcompiler/CMakeLists.txt
@@ -153,13 +153,15 @@ set_target_properties(dxcompiler
 
 if (WIN32)
-  set(install_dest RUNTIME DESTINATION bin)
+  set(install_dest
+    RUNTIME DESTINATION bin COMPONENT dxcompiler
+    ARCHIVE DESTINATION lib COMPONENT dxcompiler)
 else()
-  set(install_dest LIBRARY DESTINATION lib${LLVM_LIBDIR_SUFFIX})
+  set(install_dest
+    LIBRARY DESTINATION lib${LLVM_LIBDIR_SUFFIX} COMPONENT dxcompiler)
 endif()
 
 install(TARGETS dxcompiler
-  ${install_dest}
-  COMPONENT dxcompiler)
+  ${install_dest})
 
 add_custom_target(install-dxcompiler
   DEPENDS dxcompiler
