diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -683,6 +683,9 @@ if(EXISTS "${LLVM_MAIN_SRC_DIR}/external")
   add_subdirectory(external) # SPIRV change
 endif()
-include_directories(AFTER ${DIRECTX_HEADER_INCLUDE_DIR}/directx ${DIRECTX_HEADER_INCLUDE_DIR}/wsl/stubs)
+include_directories(AFTER ${DIRECTX_HEADER_INCLUDE_DIR}/directx)
+if(NOT WIN32)
+  include_directories(AFTER ${DIRECTX_HEADER_INCLUDE_DIR}/wsl/stubs)
+endif()
 
 # HLSL - Change End
 
