diff --git a/scripts/preinstall.sh b/scripts/preinstall.sh
index b377228..8a5e383 100755
--- a/scripts/preinstall.sh
+++ b/scripts/preinstall.sh
@@ -44,6 +44,7 @@ else
 
     (
         # Build aws-lambda-cpp
+        cp patches/backward.h aws-lambda-cpp-0.2.6/include
         mkdir -p ./aws-lambda-cpp-$AWS_LAMBDA_CPP_RELEASE/build && \
             cd ./aws-lambda-cpp-$AWS_LAMBDA_CPP_RELEASE/build
 
@@ -54,4 +55,4 @@ else
                 -DCMAKE_MODULE_PATH="$ARTIFACTS_DIR"/lib/pkgconfig && \
             make && make install
     )
-fi
\ No newline at end of file
+fi
