diff --git a/setup.py b/setup.py
index 1c5ed51b..43583678 100644
--- a/setup.py
+++ b/setup.py
@@ -25,13 +25,15 @@ if not isfile('site.cfg'):
       exit(1)
 config = ConfigParser()
 config.read('site.cfg')
-try:
-      libraries.extend([config['openblas']['library']])
-      library_dirs.extend([config['openblas']['library_dir']])
-      include_dirs.extend([config['openblas']['include_dir']])
-except KeyError as err:
-      print(f'\033[31mKeyError: cannot find the {err} key in the site.cfg file. See the site.cfg.example file for documentation\033[0m')
-      exit(1)
+
+for lib in ('blas', 'lapack', 'openblas'):
+    if lib in config:
+        libraries.extend([config[lib]['library']])
+        library_dirs.extend([config[lib]['library_dir']])
+        include_dirs.extend([config[lib]['include_dir']])
+if len(libraries) == 0:
+    print(f'\033[31mKeyError: cannot find LAPACK/BLAS/OpenBLAS paths in the site.cfg file. See the site.cfg.example file for documentation\033[0m')
+    exit(1)
 
 if sys.platform.startswith('win32'):
       extra_compile_args.extend(['/std:c++14', '/fp:fast'])
