diff --git a/setup.py b/setup.py
index de79b25..25a1826 100644
--- a/setup.py
+++ b/setup.py
@@ -3,10 +3,10 @@ from setuptools import setup, find_packages
 with open("magicclass/__init__.py", encoding="utf-8") as f:
     line = next(iter(f))
     VERSION = line.strip().split()[-1][1:-1]
-      
+
 with open("README.md", "r") as f:
     readme = f.read()
-    
+
 setup(
     name="magic-class",
     version=VERSION,
@@ -17,7 +17,7 @@ setup(
     author_email="liuhanjin-sc@g.ecc.u-tokyo.ac.jp",
     license="BSD 3-Clause",
     download_url="https://github.com/hanjinliu/magic-class",
-    packages=find_packages(exclude=["docs", "examples", "rst"]),
+    packages=find_packages(exclude=["docs", "examples", "rst", "tests", "tests.*"]),
     package_data={"magicclass": ["**/*.pyi", "*.pyi"]},
     install_requires=[
           "magicgui>=0.3.4",
