diff --git a/src/libasr/runtime/lfortran_intrinsics.c b/src/libasr/runtime/lfortran_intrinsics.c
index ec38ae64d..a62e51974 100644
--- a/src/libasr/runtime/lfortran_intrinsics.c
+++ b/src/libasr/runtime/lfortran_intrinsics.c
@@ -1267,6 +1267,9 @@ LFORTRAN_API double _lfortran_time()
     uli.LowPart = ft.dwLowDateTime;
     uli.HighPart = ft.dwHighDateTime;
     return (double)uli.QuadPart / 10000000.0 - 11644473600.0;
+#elif defined(__MACH__)
+    // CLOCK_REALTIME is sometimes not defined
+    return 0.0;
 #else
     struct timespec ts;
     clock_gettime(CLOCK_REALTIME, &ts);
