diff --git a/CMakeLists.txt b/CMakeLists.txt
index d51da30de..65b9d40c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Clang)
     set(LPYTHON_CXX_NO_RTTI_FLAG "-fno-rtti")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    set(LPYTHON_CXX_NO_RTTI_FLAG "/GR-")
+    #set(LPYTHON_CXX_NO_RTTI_FLAG "/GR-")
 endif ()
