diff --git a/build0.sh b/build0.sh
index d1f1c279e..e82bb33d0 100755
--- a/build0.sh
+++ b/build0.sh
@@ -3,9 +3,6 @@
 set -e
 set -x
 
-# Generate the `version` file
-ci/version.sh
-
 # Generate a Python AST from Python.asdl (Python)
 python grammar/asdl_py.py
 # Generate a Python AST from Python.asdl (C++)
diff --git a/build0_win.xsh b/build0_win.xsh
index 8530d835b..99ee6e260 100644
--- a/build0_win.xsh
+++ b/build0_win.xsh
@@ -1,9 +1,5 @@
 #!/usr/bin/env xonsh
 
-#bash ci/version.sh
-version=$(git describe --tags --dirty).strip()[1:]
-echo @(version) > version
-
 # Generate a Python AST from Python.asdl (Python)
 python grammar/asdl_py.py
 # Generate a Python AST from Python.asdl (C++)
