From eb96a08fb751baef1eeebc18c9b508173c110aa2 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 25 Jun 2023 12:33:01 -0400
Subject: [PATCH 2/3] install the library

---
 CMakeLists.txt | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 25490ef9..bc83c306 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,7 @@ include(FetchContent)
 
 set(CMAKE_C_STANDARD)
 set(CMAKE_C_STANDARD_REQUIRED True)
+include(GNUInstallDirs)
 
 
 # Options
@@ -432,13 +433,17 @@ file(GLOB imgui_sources
     "external/imgui/backends/imgui_impl_vulkan.*"
 )
 
+file(GLOB DATOVIZ_HEADERS
+    "include/datoviz/*.h"
+)
+
 set_source_files_properties(${imgui_sources} PROPERTIES COMPILE_FLAGS -w)
 add_library(datoviz SHARED ${sources} ${path_shadersc} ${path_colortex} ${path_fonts})
 
 set_target_properties(datoviz PROPERTIES
     VERSION ${DATOVIZ_VERSION}
     SOVERSION 1
-    PUBLIC_HEADER include/datoviz/datoviz.h)
+    PUBLIC_HEADER "${DATOVIZ_HEADERS}")
 add_dependencies(datoviz shaders)
 
 target_compile_definitions(datoviz PUBLIC ${COMPILE_DEFINITIONS})
@@ -490,6 +495,16 @@ if (DATOVIZ_WITH_CLI)
     add_test(NAME datoviz_test COMMAND datovizcli test)
 endif()
 
+install(
+    TARGETS datoviz
+    EXPORT  datoviz
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/datoviz
+)
+
+
 
 # -------------------------------------------------------------------------------------------------
 # Leftovers
-- 
2.39.2

