From 71cc0d82823ca1a4f637bea267a7e1b1e62921ff Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 25 Jun 2023 16:42:51 -0400
Subject: [PATCH] Remove unused requirements from generate_cython

---
 utils/generate_cython.py | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/utils/generate_cython.py b/utils/generate_cython.py
index fbeb1ee9..12ea0f5e 100644
--- a/utils/generate_cython.py
+++ b/utils/generate_cython.py
@@ -4,21 +4,9 @@
 # -------------------------------------------------------------------------------------------------
 
 from functools import lru_cache
-import json
-from pathlib import Path
-import os
-from pprint import pprint
 import re
-import sys
 from textwrap import indent
 
-from tqdm import tqdm
-import pyparsing as pp
-from pyparsing import (
-    Suppress, Word, alphas, alphanums, nums, Optional, Group, ZeroOrMore, empty, restOfLine,
-    Keyword, cStyleComment, Empty, Literal
-)
-
 from parse_headers import ROOT_DIR, iter_vars
 
 
