diff --git a/setup.py b/setup.py
index b59de0a..6cb8d16 100644
--- a/setup.py
+++ b/setup.py
@@ -13,7 +13,6 @@
 console_scripts			= [
     'enip_server	= cpppo.server.enip.main:main',
     'enip_client	= cpppo.server.enip.client:main',
-    'enip_getattr	= cpppo.server.enip.getattr:main',	# pending deprecation
     'enip_get_attribute	= cpppo.server.enip.get_attribute:main',
 ]
 if sys.version_info[0:2] < (3,0):
