--- a/sample/test/TestRange.cs
+++ b/sample/test/TestRange.cs
@@ -35 +35 @@
-			((Range) hscale).UpdatePolicy = UpdateType.Delayed;
+			((Gtk.Range) hscale).UpdatePolicy = UpdateType.Delayed;
@@ -42 +42 @@
-			((Range) hscrollbar).UpdatePolicy = UpdateType.Continuous;
+			((Gtk.Range) hscrollbar).UpdatePolicy = UpdateType.Continuous;
@@ -62 +62 @@
-			((Range) vscale).Inverted = true;
+			((Gtk.Range) vscale).Inverted = true;
