--- a/Makefile.include	2017-05-31 04:31:41.000000000 -0500
+++ b/Makefile.include	2025-01-17 09:22:08.723319987 -0600
@@ -25,5 +25,4 @@
 	chmod u+w $(API)
-	@if test -n '$(METADATA)'; then		\
-		echo "$(RUNTIME) $(top_builddir)/parser/gapi-fixup.exe --api=$(API) --metadata=$(srcdir)/$(METADATA) $(build_symbols)";	\
-		$(RUNTIME) $(top_builddir)/parser/gapi-fixup.exe --api=$(API) --metadata=$(srcdir)/$(METADATA) $(build_symbols); \
+	@if test -n '$(METADATA)'; then \
+		$(RUNTIME) run --project $(top_builddir)/parser/gapi-fixup.csproj --api=$(API) --metadata=$(srcdir)/$(METADATA) $(build_symbols); \
 	fi
@@ -34,8 +33,8 @@
 generated-stamp: $(API) $(INCLUDE_API) $(top_builddir)/generator/gapi_codegen.exe $(build_customs)
-	rm -f generated/* &&	\
-	$(RUNTIME) $(top_builddir)/generator/gapi_codegen.exe --generate $(API) \
-	$(api_includes)	\
-	--outdir=generated --customdir=$(srcdir) --assembly-name=$(ASSEMBLY_NAME)	\
-	--gluelib-name=$(pkg)sharpglue-2 --glue-filename=glue/generated.c		\
-	--glue-includes=$(glue_includes)						\
+	rm -f generated/* && \
+	$(RUNTIME) run --project $(top_builddir)/generator/generator.csproj --generate $(API) \
+	$(api_includes) \
+	--outdir=generated --customdir=$(srcdir) --assembly-name=$(ASSEMBLY_NAME) \
+	--gluelib-name=$(pkg)sharpglue-2 --glue-filename=glue/generated.c \
+	--glue-includes=$(glue_includes) \
 	&& touch generated-stamp
@@ -51,4 +50,4 @@
 
-$(POLICY_ASSEMBLIES) : policy.%.$(ASSEMBLY): policy.%.config $(SNK)
-	$(AL) -link:policy.$*.config -version:$* -out:$@ -keyfile:$(SNK)
+$(POLICY_ASSEMBLIES): policy.%.$(ASSEMBLY): policy.%.config $(SNK)
+	$(CSC) -c Release --framework net8.0 /p:SignAssembly=true /p:AssemblyOriginatorKeyFile=$(SNK) /t:library /out:$@ policy.$*.config
 
@@ -59,14 +58,19 @@
 	@rm -f $(ASSEMBLY).mdb
-	$(CSC) $(CSFLAGS) -nowarn:0169,0612,0618 -unsafe -out:$(ASSEMBLY) -target:library $(build_references) $(GENERATED_SOURCES) $(build_sources)
+	$(CSC) -c Release --framework net8.0 -o $(ASSEMBLY) \
+		/p:AllowUnsafeBlocks=true \
+		/p:NoWarn=0169,0612,0618 \
+		/p:SignAssembly=true \
+		/p:AssemblyOriginatorKeyFile=$(SNK) \
+		$(build_references) $(GENERATED_SOURCES) $(build_sources)
 
 install-data-local:
-	@if test -n '$(pkg)'; then							\
-	  echo "$(GACUTIL) -i $(ASSEMBLY) -f $(GACUTIL_FLAGS)";				\
-	  $(GACUTIL) -i $(ASSEMBLY) -f $(GACUTIL_FLAGS) || exit 1;			\
-	  if test -n '$(POLICY_VERSIONS)'; then						\
-	    for i in $(POLICY_VERSIONS); do						\
-	      echo "$(GACUTIL) -i policy.$$i.$(ASSEMBLY) -f $(GACUTIL_FLAGS)";		\
-	      $(GACUTIL) -i policy.$$i.$(ASSEMBLY) -f $(GACUTIL_FLAGS) || exit 1;	\
-	    done									\
-	  fi										\
+	@if test -n '$(pkg)'; then \
+		echo "$(DOTNET) nuget add $(ASSEMBLY) -s $(DESTDIR)$(prefix)/lib"; \
+		$(DOTNET) nuget add $(ASSEMBLY) -s $(DESTDIR)$(prefix)/lib || exit 1; \
+		if test -n '$(POLICY_VERSIONS)'; then \
+			for i in $(POLICY_VERSIONS); do \
+				echo "$(DOTNET) nuget add policy.$$i.$(ASSEMBLY) -s $(DESTDIR)$(prefix)/lib"; \
+				$(DOTNET) nuget add policy.$$i.$(ASSEMBLY) -s $(DESTDIR)$(prefix)/lib || exit 1; \
+			done \
+		fi \
 	fi
@@ -74,11 +78,11 @@
 uninstall-local:
-	@if test -n '$(pkg)'; then							\
-	  echo "$(GACUTIL) -u $(ASSEMBLY_NAME) $(GACUTIL_FLAGS)";			\
-	  $(GACUTIL) -u $(ASSEMBLY_NAME) $(GACUTIL_FLAGS) || exit 1;			\
-	  if test -n '$(POLICY_VERSIONS)'; then						\
-	    for i in $(POLICY_VERSIONS); do						\
-	      echo "$(GACUTIL) -u policy.$$i.$(ASSEMBLY_NAME) $(GACUTIL_FLAGS)";	\
-	      $(GACUTIL) -u policy.$$i.$(ASSEMBLY_NAME) $(GACUTIL_FLAGS) || exit 1;	\
-	    done									\
-	  fi										\
+	@if test -n '$(pkg)'; then \
+		echo "$(DOTNET) nuget remove $(ASSEMBLY_NAME) -s $(DESTDIR)$(prefix)/lib"; \
+		$(DOTNET) nuget remove $(ASSEMBLY_NAME) -s $(DESTDIR)$(prefix)/lib || exit 1; \
+		if test -n '$(POLICY_VERSIONS)'; then \
+			for i in $(POLICY_VERSIONS); do \
+				echo "$(DOTNET) nuget remove policy.$$i.$(ASSEMBLY_NAME) -s $(DESTDIR)$(prefix)/lib"; \
+				$(DOTNET) nuget remove policy.$$i.$(ASSEMBLY_NAME) -s $(DESTDIR)$(prefix)/lib || exit 1; \
+			done \
+		fi \
 	fi
