commit f7157ece34df9a2fc61c153e276016f1cba29daf
Author: Fabrice Salvaire <fabrice.salvaire@orange.fr>
Date:   Mon Mar 11 19:13:30 2019 +0100

    fixed release tasks

diff --git a/setup_data.py b/setup_data.py
index 6ef175e..6012e0a 100644
--- a/setup_data.py
+++ b/setup_data.py
@@ -31,7 +31,7 @@ if pyspice_path.name == 'conf.py':
     pyspice_path = pyspice_path.parents[3]
 else:
     pyspice_path = pyspice_path.parent
-init_path = pyspice_path.joinpath('PySpice', '__init__.py.in')
+init_path = pyspice_path.joinpath('PySpice', '__init__.py')
 with open(init_path) as fh:
     try:
         version = re.findall(r"^__version__ = '([^']+)'\r?$", fh.read(), re.M)[0]
diff --git a/tasks/release.py b/tasks/release.py
index 1fb1239..56af68c 100644
--- a/tasks/release.py
+++ b/tasks/release.py
@@ -107,10 +107,12 @@ def update_git_sha(ctx):
 
 ####################################################################################################
 
+@task()
 def clean(ctx):
     for directory in ('build', 'dist'):
-        shutil.rmtree(directory)
+        shutil.rmtree(directory, ignore_errors=True)
 
+@task()
 def show_python_site(ctx):
     ctx.run('python3 -m site')
 
@@ -122,10 +124,6 @@ def build(ctx):
 def install(ctx):
     ctx.run('python3 setup.py install')
 
-@task(build)
-def install(ctx):
-    ctx.run('python3 setup.py install')
-
 @task(clean, build)
 def wheel(ctx):
     ctx.run('python3 setup.py bdist_wheel')
