--- CMakeLists.txt
+++ CMakeLists.txt
@@ -29,8 +29,8 @@
 
 # include directories
 include_directories("include")
-include_directories("3rdparty/dlpack/include")
-include_directories("3rdparty/dmlc-core/include")
+include_directories("$ENV{PREFIX}/include")
+include_directories("$ENV{PREFIX}/include")
 
 # initial variables
 set(DECORD_LINKER_LIBS "")
@@ -85,7 +85,7 @@
 assign_source_group("Include" ${GROUP_INCLUDE})
 
 # Source file lists
-file(GLOB DECORD_CORE_SRCS src/*.cc src/runtime/*.cc src/video/*.cc src/sampler/*.cc src/audio/*.cc src/av_wrapper/*.cc)
+file(GLOB DECORD_CORE_SRCS src/*.cc src/runtime/*.cc src/video/*.cc src/sampler/*.cc src/av_wrapper/*.cc)
 
 # Module rules
 include(cmake/modules/FFmpeg.cmake)
