--- include/decord/audio_interface.h
+++ include/decord/audio_interface.h
@@ -33,7 +33,7 @@
         virtual void GetInfo() = 0;
     };
 
-    DECORD_DLL AudioReaderPtr GetAudioReader(std::string fname, int sampleRate, DLContext ctx, int io_type, bool mono);
+    DECORD_DLL AudioReaderPtr GetAudioReader(std::string fname, int sampleRate, DLDevice ctx, int io_type, bool mono);
 }
 
 #endif //DECORD_AUDIO_INTERFACE_H
--- include/decord/base.h
+++ include/decord/base.h
@@ -28,8 +28,8 @@
   return (d1.bits == d2.bits && d1.code == d2.code && d1.lanes == d2.lanes);
 }
 
-static const DLContext kCPU = {kDLCPU, 0};
-static const DLContext kGPU = {kDLGPU, 0};
+static const DLDevice kCPU = {kDLCPU, 0};
+static const DLDevice kGPU = {kDLCUDA, 0};
 
 /*! \brief performance flags */
 int constexpr kCPUAlignment = 32;  // For video width alignment, not comply to this will result in sparse arrays
--- include/decord/runtime/c_runtime_api.h
+++ include/decord/runtime/c_runtime_api.h
@@ -112,7 +112,7 @@
 /*!
  * \brief The Device information, abstract away common device types.
  */
-typedef DLContext DECORDContext;
+typedef DLDevice DECORDContext;
 
 /*!
  * \brief The tensor array stucture to DECORD API.
--- include/decord/runtime/device_api.h
+++ include/decord/runtime/device_api.h
@@ -187,7 +187,7 @@
 inline const char* DeviceName(int type) {
   switch (type) {
     case kDLCPU: return "cpu";
-    case kDLGPU: return "gpu";
+    case kDLCUDA: return "gpu";
     case kDLOpenCL: return "opencl";
     case kDLSDAccel: return "sdaccel";
     case kDLAOCL: return "aocl";
@@ -202,7 +202,7 @@
 }
 
 #ifndef _LIBCPP_SGX_NO_IOSTREAMS
-inline std::ostream& operator<<(std::ostream& os, DLContext ctx) {  // NOLINT(*)
+inline std::ostream& operator<<(std::ostream& os, DLDevice ctx) {  // NOLINT(*)
   int device_type = static_cast<int>(ctx.device_type);
   if (device_type > kRPCSessMask) {
     os << "remote[" << (device_type / kRPCSessMask) << "]-";
--- include/decord/runtime/ndarray.h
+++ include/decord/runtime/ndarray.h
@@ -141,7 +141,7 @@
    * \param ctx The target context.
    * \return The array under another context.
    */
-  inline NDArray CopyTo(const DLContext& ctx) const;
+  inline NDArray CopyTo(const DLDevice& ctx) const;
   /*!
    * \brief Load NDArray from stream
    * \param stream The input data stream
@@ -185,7 +185,7 @@
    */
   DECORD_DLL static NDArray Empty(std::vector<int64_t> shape,
                                DLDataType dtype,
-                               DLContext ctx);
+                               DLDevice ctx);
   /*!
    * \brief Create a NDArray backed by a dlpack tensor.
    *
@@ -344,10 +344,10 @@
   // Create view as DLTensor
   DLTensor dlt;
   dlt.data = dmlc::BeginPtr(other);
-  DLContext cpu_ctx;
+  DLDevice cpu_ctx;
   cpu_ctx.device_type = kDLCPU;
   cpu_ctx.device_id = 0;
-  dlt.ctx = cpu_ctx;
+  dlt.device = cpu_ctx;
   DLDataType dtype;
   dtype.bits = 8U * sizeof(T);
   if (std::is_floating_point<T>::value) {
@@ -410,7 +410,7 @@
   CopyFromTo(&(data_->dl_tensor), &dlt);
 }
 
-inline NDArray NDArray::CopyTo(const DLContext& ctx) const {
+inline NDArray NDArray::CopyTo(const DLDevice& ctx) const {
   CHECK(data_ != nullptr);
   const DLTensor* dptr = operator->();
   NDArray ret = Empty(std::vector<int64_t>(dptr->shape, dptr->shape + dptr->ndim),
@@ -453,7 +453,7 @@
   //
   // We can always do array.CopyTo(target_ctx) to get a corresponding
   // array in the target context.
-  DLContext cpu_ctx;
+  DLDevice cpu_ctx;
   cpu_ctx.device_type = kDLCPU;
   cpu_ctx.device_id = 0;
   strm->Write(cpu_ctx);
@@ -470,7 +470,7 @@
   strm->Write(data_byte_size);
 
   if (DMLC_IO_NO_ENDIAN_SWAP &&
-      tensor->ctx.device_type == kDLCPU &&
+      tensor->device.device_type == kDLCPU &&
       tensor->strides == nullptr &&
       tensor->byte_offset == 0) {
     // quick path
@@ -500,7 +500,7 @@
       << "Invalid DLTensor file format";
   CHECK(header == kDECORDNDArrayMagic)
       << "Invalid DLTensor file format";
-  DLContext ctx;
+  DLDevice ctx;
   int ndim;
   DLDataType dtype;
   CHECK(strm->Read(&ctx))
--- include/decord/runtime/serializer.h
+++ include/decord/runtime/serializer.h
@@ -2,7 +2,7 @@
  *  Copyright (c) 2019 by Contributors if not otherwise specified
  * \file decord/runtime/serializer.h
  * \brief Serializer extension to support DECORD data types
- *  Include this file to enable serialization of DLDataType, DLContext
+ *  Include this file to enable serialization of DLDataType, DLDevice
  */
 #ifndef DECORD_RUNTIME_SERIALIZER_H_
 #define DECORD_RUNTIME_SERIALIZER_H_
@@ -31,13 +31,13 @@
 };
 
 template<>
-struct Handler<DLContext> {
-  inline static void Write(Stream *strm, const DLContext& ctx) {
+struct Handler<DLDevice> {
+  inline static void Write(Stream *strm, const DLDevice& ctx) {
     int32_t device_type = static_cast<int32_t>(ctx.device_type);
     Handler<int32_t>::Write(strm, device_type);
     Handler<int32_t>::Write(strm, ctx.device_id);
   }
-  inline static bool Read(Stream *strm, DLContext* ctx) {
+  inline static bool Read(Stream *strm, DLDevice* ctx) {
     int32_t device_type = 0;
     if (!Handler<int32_t>::Read(strm, &(device_type))) return false;
     ctx->device_type = static_cast<DLDeviceType>(device_type);
--- include/decord/video_interface.h
+++ include/decord/video_interface.h
@@ -87,7 +87,7 @@
 };  // class VideoReader
 
 
-DECORD_DLL VideoReaderPtr GetVideoReader(std::string fname, DLContext ctx,
+DECORD_DLL VideoReaderPtr GetVideoReader(std::string fname, DLDevice ctx,
                                          int width=-1, int height=-1, int nb_thread=0,
                                          int io_type=kNormal);
 
--- src/audio/audio_interface.cc
+++ src/audio/audio_interface.cc
@@ -11,7 +11,7 @@
 
 namespace decord {
 
-    AudioReaderPtr GetAudioReader(std::string fn, int sampleRate, DLContext ctx, int io_type, bool mono) {
+    AudioReaderPtr GetAudioReader(std::string fn, int sampleRate, DLDevice ctx, int io_type, bool mono) {
         std::shared_ptr<AudioReaderInterface> ptr;
         ptr = std::make_shared<AudioReader>(fn, sampleRate, ctx, io_type, mono);
         return ptr;
@@ -29,7 +29,7 @@
             bool mono = int(args[5]) == 1 ? true : false;
 
             // TODO: add io type
-            DLContext ctx;
+            DLDevice ctx;
             ctx.device_type = static_cast<DLDeviceType>(device_type);
             ctx.device_id = device_id;
             auto reader = new AudioReader(fn, sampleRate, ctx, io_type, mono);
--- src/audio/audio_reader.cc
+++ src/audio/audio_reader.cc
@@ -11,7 +11,7 @@
     // AVIO buffer size when reading from raw bytes
     static const int AVIO_BUFFER_SIZE = std::stoi(runtime::GetEnvironmentVariableOrDefault("DECORD_AVIO_BUFFER_SIZE", "40960"));
 
-    AudioReader::AudioReader(std::string fn, int sampleRate, DLContext ctx, int io_type, bool mono)
+    AudioReader::AudioReader(std::string fn, int sampleRate, DLDevice ctx, int io_type, bool mono)
     : ctx(ctx), io_ctx_(), pFormatContext(nullptr), swr(nullptr), pCodecParameters(nullptr),
       pCodecContext(nullptr), audioStreamIndex(-1), outputVector(), output(), padding(-1.0), filename(fn), originalSampleRate(0),
       targetSampleRate(sampleRate), numChannels(0), mono(mono), totalSamplesPerChannel(0), totalConvertedSamplesPerChannel(0),
--- src/audio/audio_reader.h
+++ src/audio/audio_reader.h
@@ -13,7 +13,7 @@
 
     class AudioReader: public AudioReaderInterface {
     public:
-        AudioReader(std::string fn, int sampleRate, DLContext ctx, int io_type=kNormal, bool mono=true);
+        AudioReader(std::string fn, int sampleRate, DLDevice ctx, int io_type=kNormal, bool mono=true);
         ~AudioReader();
         NDArray GetNDArray();
         int GetNumPaddingSamples();
@@ -30,7 +30,7 @@
         void ToNDArray();
         void SaveToVector(float** buffer, int numChannels, int numSamples);
 
-        DLContext ctx;
+        DLDevice ctx;
         std::unique_ptr<ffmpeg::AVIOBytesContext> io_ctx_;  // avio context for raw memory access
         AVFormatContext *pFormatContext;
         struct SwrContext* swr;
--- src/runtime/cuda/cuda_device_api.cc
+++ src/runtime/cuda/cuda_device_api.cc
@@ -134,7 +134,7 @@
     cudaStream_t cu_stream = static_cast<cudaStream_t>(stream);
     from = static_cast<const char*>(from) + from_offset;
     to = static_cast<char*>(to) + to_offset;
-    if (ctx_from.device_type == kDLGPU && ctx_to.device_type == kDLGPU) {
+    if (ctx_from.device_type == kDLCUDA && ctx_to.device_type == kDLCUDA) {
       CUDA_CALL(cudaSetDevice(ctx_from.device_id));
       if (ctx_from.device_id == ctx_to.device_id) {
         GPUCopy(from, to, size, cudaMemcpyDeviceToDevice, cu_stream);
@@ -143,10 +143,10 @@
                             from, ctx_from.device_id,
                             size, cu_stream);
       }
-    } else if (ctx_from.device_type == kDLGPU && ctx_to.device_type == kDLCPU) {
+    } else if (ctx_from.device_type == kDLCUDA && ctx_to.device_type == kDLCPU) {
       CUDA_CALL(cudaSetDevice(ctx_from.device_id));
       GPUCopy(from, to, size, cudaMemcpyDeviceToHost, cu_stream);
-    } else if (ctx_from.device_type == kDLCPU && ctx_to.device_type == kDLGPU) {
+    } else if (ctx_from.device_type == kDLCPU && ctx_to.device_type == kDLCUDA) {
       CUDA_CALL(cudaSetDevice(ctx_to.device_id));
       GPUCopy(from, to, size, cudaMemcpyHostToDevice, cu_stream);
     } else {
@@ -219,7 +219,7 @@
 typedef dmlc::ThreadLocalStore<CUDAThreadEntry> CUDAThreadStore;
 
 CUDAThreadEntry::CUDAThreadEntry()
-    : pool(kDLGPU, CUDADeviceAPI::Global()) {
+    : pool(kDLCUDA, CUDADeviceAPI::Global()) {
 }
 
 CUDAThreadEntry* CUDAThreadEntry::ThreadLocal() {
--- src/runtime/ndarray.cc
+++ src/runtime/ndarray.cc
@@ -40,8 +40,8 @@
     if (ptr->manager_ctx != nullptr) {
       static_cast<NDArray::Container*>(ptr->manager_ctx)->DecRef();
     } else if (ptr->dl_tensor.data != nullptr) {
-      decord::runtime::DeviceAPI::Get(ptr->dl_tensor.ctx)->FreeDataSpace(
-          ptr->dl_tensor.ctx, ptr->dl_tensor.data);
+      decord::runtime::DeviceAPI::Get(ptr->dl_tensor.device)->FreeDataSpace(
+          ptr->dl_tensor.device, ptr->dl_tensor.data);
     }
     delete ptr;
   }
@@ -61,7 +61,7 @@
   // but does not allocate space for the data.
   static NDArray Create(std::vector<int64_t> shape,
                         DLDataType dtype,
-                        DLContext ctx) {
+                        DLDevice ctx) {
     VerifyDataType(dtype);
     // critical zone
     NDArray::Container* data = new NDArray::Container();
@@ -76,7 +76,7 @@
     // setup dtype
     data->dl_tensor.dtype = dtype;
     // setup ctx
-    data->dl_tensor.ctx = ctx;
+    data->dl_tensor.device = ctx;
     return ret;
   }
   // Implementation of API function
@@ -103,7 +103,7 @@
   CHECK(data_ != nullptr);
   CHECK(data_->dl_tensor.strides == nullptr)
       << "Can only create view for compact tensor";
-  NDArray ret = Internal::Create(shape, dtype, data_->dl_tensor.ctx);
+  NDArray ret = Internal::Create(shape, dtype, data_->dl_tensor.device);
   ret.data_->dl_tensor.byte_offset =
       this->data_->dl_tensor.byte_offset;
   size_t curr_size = GetDataSize(this->data_->dl_tensor);
@@ -122,7 +122,7 @@
   CHECK(data_ != nullptr);
   CHECK(data_->dl_tensor.strides == nullptr)
       << "Can only create offset view for compact tensor";
-  NDArray ret = Internal::Create(shape, dtype, data_->dl_tensor.ctx);
+  NDArray ret = Internal::Create(shape, dtype, data_->dl_tensor.device);
   ret.data_->dl_tensor.byte_offset =
       this->data_->dl_tensor.byte_offset;
   size_t curr_size = GetDataSize(this->data_->dl_tensor);
@@ -143,14 +143,14 @@
 
 NDArray NDArray::Empty(std::vector<int64_t> shape,
                        DLDataType dtype,
-                       DLContext ctx) {
+                       DLDevice ctx) {
   NDArray ret = Internal::Create(shape, dtype, ctx);
   // setup memory content
   size_t size = GetDataSize(ret.data_->dl_tensor);
   size_t alignment = GetDataAlignment(ret.data_->dl_tensor);
   ret.data_->dl_tensor.data =
-      DeviceAPI::Get(ret->ctx)->AllocDataSpace(
-          ret->ctx, size, alignment, ret->dtype);
+      DeviceAPI::Get(ret->device)->AllocDataSpace(
+          ret->device, size, alignment, ret->dtype);
   return ret;
 }
 
@@ -171,19 +171,19 @@
   CHECK_EQ(from_size, to_size)
     << "DECORDArrayCopyFromTo: The size must exactly match";
 
-  CHECK(from->ctx.device_type == to->ctx.device_type
-        || from->ctx.device_type == kDLCPU
-        || to->ctx.device_type == kDLCPU)
+  CHECK(from->device.device_type == to->device.device_type
+        || from->device.device_type == kDLCPU
+        || to->device.device_type == kDLCPU)
     << "Can not copy across different ctx types directly";
 
   // Use the context that is *not* a cpu context to get the correct device
   // api manager.
-  DECORDContext ctx = from->ctx.device_type != kDLCPU ? from->ctx : to->ctx;
+  DECORDContext ctx = from->device.device_type != kDLCPU ? from->device : to->device;
 
   DeviceAPI::Get(ctx)->CopyDataFromTo(
     from->data, static_cast<size_t>(from->byte_offset),
     to->data, static_cast<size_t>(to->byte_offset),
-    from_size, from->ctx, to->ctx, from->dtype, stream);
+    from_size, from->device, to->device, from->dtype, stream);
 }
 
 }  // namespace runtime
@@ -209,7 +209,7 @@
   dtype.code = static_cast<uint8_t>(dtype_code);
   dtype.bits = static_cast<uint8_t>(dtype_bits);
   dtype.lanes = static_cast<uint16_t>(dtype_lanes);
-  DLContext ctx;
+  DLDevice ctx;
   ctx.device_type = static_cast<DLDeviceType>(device_type);
   ctx.device_id = device_id;
   *out = NDArray::Internal::MoveAsDLTensor(
@@ -259,10 +259,10 @@
   size_t arr_size = GetDataSize(*handle);
   CHECK_EQ(arr_size, nbytes)
       << "DECORDArrayCopyFromBytes: size mismatch";
-  DeviceAPI::Get(handle->ctx)->CopyDataFromTo(
+  DeviceAPI::Get(handle->device)->CopyDataFromTo(
       data, 0,
       handle->data, static_cast<size_t>(handle->byte_offset),
-      nbytes, cpu_ctx, handle->ctx, handle->dtype, nullptr);
+      nbytes, cpu_ctx, handle->device, handle->dtype, nullptr);
   API_END();
 }
 
@@ -276,9 +276,9 @@
   size_t arr_size = GetDataSize(*handle);
   CHECK_EQ(arr_size, nbytes)
       << "DECORDArrayCopyToBytes: size mismatch";
-  DeviceAPI::Get(handle->ctx)->CopyDataFromTo(
+  DeviceAPI::Get(handle->device)->CopyDataFromTo(
       handle->data, static_cast<size_t>(handle->byte_offset),
       data, 0,
-      nbytes, handle->ctx, cpu_ctx, handle->dtype, nullptr);
+      nbytes, handle->device, cpu_ctx, handle->dtype, nullptr);
   API_END();
 }
--- src/video/ffmpeg/ffmpeg_common.h
+++ src/video/ffmpeg/ffmpeg_common.h
@@ -200,10 +200,10 @@
         << "AVFrame data is not a compact array. linesize: " << p->linesize[0]
         << " width: " << p->width;
 
-	DLContext ctx;
+	DLDevice ctx;
 	if (p->hw_frames_ctx) {
         LOG(FATAL) << "HW ctx not supported";
-		ctx = DLContext({ kDLGPU, 0 });
+		ctx = DLDevice({ kDLCUDA, 0 });
 	}
 	else {
 		ctx = kCPU;
@@ -215,7 +215,7 @@
 	shape[1] = p->width;
 	shape[2] = p->linesize[0] / p->width;
 	dlt.data = p->data[0];
-	dlt.ctx = ctx;
+	dlt.device = ctx;
 	dlt.ndim = 3;
 	dlt.dtype = kUInt8;
 	dlt.shape = shape;
--- src/video/ffmpeg/threaded_decoder.cc
+++ src/video/ffmpeg/threaded_decoder.cc
@@ -245,7 +245,7 @@
     //     << "AVFrame data is not a compact array. linesize: " << p->linesize[0]
     //     << " width: " << p->width;
 
-    DLContext ctx;
+    DLDevice ctx;
     CHECK(!p->hw_frames_ctx) << "Not supported hw_frames_ctx";
     ctx = kCPU;
     NDArray arr = NDArray::Empty({p->height, p->width, channel}, kUInt8, ctx);
--- src/video/storage_pool.cc
+++ src/video/storage_pool.cc
@@ -12,7 +12,7 @@
 
 }
 
-NDArrayPool::NDArrayPool(std::size_t sz, std::vector<int64_t> shape, DLDataType dtype, DLContext ctx)
+NDArrayPool::NDArrayPool(std::size_t sz, std::vector<int64_t> shape, DLDataType dtype, DLDevice ctx)
     : size_(sz), shape_(shape), dtype_(dtype), ctx_(ctx), init_(true) {
 }
 
@@ -44,16 +44,16 @@
     if (ptr->manager_ctx != nullptr) {
         auto pool = static_cast<NDArrayPool*>(ptr->manager_ctx);
         if (pool->size_ <= pool->queue_.size()) {
-            decord::runtime::DeviceAPI::Get(ptr->dl_tensor.ctx)->FreeDataSpace(
-          ptr->dl_tensor.ctx, ptr->dl_tensor.data);
+            decord::runtime::DeviceAPI::Get(ptr->dl_tensor.device)->FreeDataSpace(
+          ptr->dl_tensor.device, ptr->dl_tensor.data);
             delete ptr;
             ptr = nullptr;
         } else {
             static_cast<NDArrayPool*>(ptr->manager_ctx)->queue_.push(NDArray(ptr));
         }
     } else if (ptr->dl_tensor.data != nullptr) {
-        decord::runtime::DeviceAPI::Get(ptr->dl_tensor.ctx)->FreeDataSpace(
-          ptr->dl_tensor.ctx, ptr->dl_tensor.data);
+        decord::runtime::DeviceAPI::Get(ptr->dl_tensor.device)->FreeDataSpace(
+          ptr->dl_tensor.device, ptr->dl_tensor.data);
         delete ptr;
         ptr = nullptr;
     }
--- src/video/storage_pool.h
+++ src/video/storage_pool.h
@@ -102,7 +102,7 @@
     using NDArray = runtime::NDArray;
     public:
         NDArrayPool();
-        NDArrayPool(std::size_t sz, std::vector<int64_t> shape, DLDataType dtype, DLContext ctx);
+        NDArrayPool(std::size_t sz, std::vector<int64_t> shape, DLDataType dtype, DLDevice ctx);
         NDArray Acquire();
         ~NDArrayPool();
         static void Deleter(NDArray::Container* ptr);
@@ -112,7 +112,7 @@
         std::size_t size_;
         std::vector<int64_t> shape_;
         DLDataType dtype_;
-        DLContext ctx_;
+        DLDevice ctx_;
         std::queue<runtime::NDArray> queue_;
         bool init_;
 };  // NDArrayPool
--- src/video/video_interface.cc
+++ src/video/video_interface.cc
@@ -17,7 +17,7 @@
 
 namespace decord {
 
-VideoReaderPtr GetVideoReader(std::string fn, DLContext ctx, int width, int height, int nb_thread,
+VideoReaderPtr GetVideoReader(std::string fn, DLDevice ctx, int width, int height, int nb_thread,
                               int io_type, std::string fault_tol) {
     std::shared_ptr<VideoReaderInterface> ptr;
     ptr = std::make_shared<VideoReader>(fn, ctx, width, height, nb_thread, io_type, fault_tol);
@@ -35,7 +35,7 @@
     int num_thread = args[5];
     int io_type = args[6];
     std::string fault_tol = args[7];
-    DLContext ctx;
+    DLDevice ctx;
     ctx.device_type = static_cast<DLDeviceType>(device_type);
     ctx.device_id = device_id;
     auto reader = new VideoReader(fn, ctx, width, height, num_thread, io_type, fault_tol);
@@ -153,12 +153,12 @@
     device_types.CopyTo(dev_types);
     std::vector<long int> dev_ids;
     device_ids.CopyTo(dev_ids);
-    std::vector<DLContext> ctxs;
+    std::vector<DLDevice> ctxs;
     ctxs.reserve(dev_ids.size());
     CHECK(dev_types.size() > 0);
     CHECK_EQ(dev_types.size(), dev_ids.size());
     for (std::size_t i = 0; i < dev_types.size(); ++i) {
-      DLContext ctx;
+      DLDevice ctx;
       ctx.device_type = static_cast<DLDeviceType>(dev_types[i]);
       ctx.device_id = static_cast<int>(dev_ids[i]);
       ctxs.emplace_back(ctx);
--- src/video/video_loader.cc
+++ src/video/video_loader.cc
@@ -14,7 +14,7 @@
 
 namespace decord {
 
-VideoLoader::VideoLoader(std::vector<std::string> filenames, std::vector<DLContext> ctxs,
+VideoLoader::VideoLoader(std::vector<std::string> filenames, std::vector<DLDevice> ctxs,
                          std::vector<int> shape, int interval,
                          int skip, int shuffle, int prefetch)
     : readers_(), shape_(shape), intvl_(interval), skip_(skip), shuffle_(shuffle),
--- src/video/video_loader.h
+++ src/video/video_loader.h
@@ -26,7 +26,7 @@
 
 class VideoLoader : public VideoLoaderInterface {
 public:
-        VideoLoader(std::vector<std::string> filenames, std::vector<DLContext> ctxs,
+        VideoLoader(std::vector<std::string> filenames, std::vector<DLDevice> ctxs,
                           std::vector<int> shape, int interval,
                           int skip, int shuffle,
                           int prefetch);
@@ -62,7 +62,7 @@
         // std::vector<std::size_t> visit_bounds_;
         // std::vector<std::vector<std::pair<std::size_t, int64_t> > > visit_buffer_;
         // std::size_t curr_;
-        std::vector<DLContext> ctxs_;
+        std::vector<DLDevice> ctxs_;
         NDArrayPool ndarray_pool_;
 };  // class VideoLoader
 }  // namespace decord
--- src/video/video_reader.cc
+++ src/video/video_reader.cc
@@ -31,7 +31,7 @@
 // (corrupted video only): The warning threshold(0.0 - 1.0) when multiple frames are unavailable and fallbacked to cached frames
 static const float DUPLICATE_WARNING_THRESHOLD = std::stof(runtime::GetEnvironmentVariableOrDefault("DECORD_DUPLICATE_WARNING_THRESHOLD", "0.25"));
 
-VideoReader::VideoReader(std::string fn, DLContext ctx, int width, int height, int nb_thread, int io_type, std::string fault_tol)
+VideoReader::VideoReader(std::string fn, DLDevice ctx, int width, int height, int nb_thread, int io_type, std::string fault_tol)
      : ctx_(ctx), key_indices_(), pts_frame_map_(), tmp_key_frame_(), overrun_(false), frame_ts_(), codecs_(),
      actv_stm_idx_(-1), fmt_ctx_(nullptr), decoder_(nullptr), curr_frame_(0),
      nb_thread_decoding_(nb_thread), width_(width), height_(height), eof_(false), io_ctx_(),
@@ -158,7 +158,7 @@
         << "Error copy stream->codecpar to buffer codecpar";
     if (kDLCPU == ctx_.device_type) {
         decoder_ = std::unique_ptr<ThreadedDecoderInterface>(new FFMPEGThreadedDecoder());
-    } else if (kDLGPU == ctx_.device_type) {
+    } else if (kDLCUDA == ctx_.device_type) {
 #ifdef DECORD_USE_CUDA
         // note: cuda threaded decoder will modify codecpar
         decoder_ = std::unique_ptr<ThreadedDecoderInterface>(new cuda::CUThreadedDecoder(
@@ -198,7 +198,7 @@
     int original_width = codecpar->width;
     int original_height = codecpar->height;
 
-    if ((rotation == 90 || rotation == 270) && ctx_.device_type != kDLGPU) {
+    if ((rotation == 90 || rotation == 270) && ctx_.device_type != kDLCUDA) {
         std::swap(original_width, original_height);
     }
 
@@ -209,7 +209,7 @@
         height_ = original_height;
     }
 
-    if (ctx_.device_type == kDLGPU) {
+    if (ctx_.device_type == kDLCUDA) {
         ndarray_pool_ = NDArrayPool(0, {height_, width_, 3}, kUInt8, ctx_);
     }
 
@@ -375,7 +375,7 @@
             if (ret == AVERROR_EOF) {
                 eof_ = true;
                 // flush buffer
-                if (ctx_.device_type != kDLGPU) {
+                if (ctx_.device_type != kDLCUDA) {
                     // no preallocated memory and memory pool, use FFMPEG AVFrame pool
                     decoder_->Push(nullptr, NDArray());
                 } else {
@@ -389,7 +389,7 @@
             return;
         }
         if (packet->stream_index == actv_stm_idx_) {
-            if (ctx_.device_type != kDLGPU) {
+            if (ctx_.device_type != kDLCUDA) {
                     // no preallocated memory and memory pool, use FFMPEG AVFrame pool
                     decoder_->Push(packet, NDArray());
                 } else {
--- src/video/video_reader.h
+++ src/video/video_reader.h
@@ -34,7 +34,7 @@
     using ThreadedDecoderPtr = std::unique_ptr<ThreadedDecoderInterface>;
     using NDArray = runtime::NDArray;
     public:
-        VideoReader(std::string fn, DLContext ctx, int width=-1, int height=-1,
+        VideoReader(std::string fn, DLDevice ctx, int width=-1, int height=-1,
                     int nb_thread=0, int io_type=kNormal, std::string fault_tol="-1");
         /*! \brief Destructor, note that FFMPEG resources has to be managed manually to avoid resource leak */
         ~VideoReader();
@@ -66,7 +66,7 @@
         void CacheFrame(NDArray frame);
         bool FetchCachedFrame(NDArray &frame, int64_t pos);
 
-        DLContext ctx_;
+        DLDevice ctx_;
         std::vector<int64_t> key_indices_;
         std::map<int64_t, int64_t> pts_frame_map_;
         NDArray tmp_key_frame_;
