diff --git a/setup.py b/setup.py
index 42d4c36..dcfa8c2 100644
--- a/setup.py
+++ b/setup.py
@@ -8,12 +8,7 @@ from subprocess import check_call
 
 class CustomBuild(build_ext):
     def run(self):
-        if system() == "Windows":
-            try:
-                check_call(["blst\\build.bat"])
-            except Exception:
-                pass
-        check_call(["make", "-C", "src", "blst"])
+        check_call(["make", "-C", "src", "blst", f"CC={os.environ['CC']}"])
         super().run()
 
 
