diff -ruN ckzg-1.0.1.orig/setup.py ckzg-1.0.1/setup.py
--- ckzg-1.0.1.orig/setup.py	2024-04-17 02:53:10.000000000 +0200
+++ ckzg-1.0.1/setup.py	2024-05-08 10:59:56.713791728 +0200
@@ -4,22 +4,15 @@
 from setuptools.command.build_ext import build_ext
 from subprocess import check_call
 
-this_dir = Path(__file__).parent
-long_description = (this_dir / "bindings/python/README.md").read_text()
-
-
-def f(path_str):
-    return str(this_dir / path_str)
-
 
 class CustomBuild(build_ext):
     def run(self):
         if system() == "Windows":
             try:
-                check_call([f("blst\\build.bat")])
+                check_call(["blst\\build.bat"])
             except Exception:
                 pass
-        check_call(["make", "-C", f("src"), "blst"])
+        check_call(["make", "-C", "src", "blst"])
         super().run()
 
 
@@ -31,15 +24,15 @@
         author_email="security@ethereum.org",
         url="https://github.com/ethereum/c-kzg-4844",
         description="Python bindings for C-KZG-4844",
-        long_description=long_description,
+        long_description=Path("bindings/python/README.md").read_text(),
         long_description_content_type="text/markdown",
         license="Apache-2.0",
         ext_modules=[
             Extension(
                 "ckzg",
-                sources=[f("bindings/python/ckzg.c"), f("src/c_kzg_4844.c")],
-                include_dirs=[f("inc"), f("src")],
-                library_dirs=[f("lib")],
+                sources=["bindings/python/ckzg.c", "src/c_kzg_4844.c"],
+                include_dirs=["inc", "src"],
+                library_dirs=["lib"],
                 libraries=["blst"]
             )
         ],
