diff --git a/python/pixcorrect/mask_saturation.py b/python/pixcorrect/mask_saturation.py
index 7a73a23..4d24152 100755
--- a/python/pixcorrect/mask_saturation.py
+++ b/python/pixcorrect/mask_saturation.py
@@ -20,7 +20,12 @@ config_section = 'mask_saturation'
 # classes

 # Lowest level access to the C library function
-masksatr_lib = load_shlib('libmasksatr')
+import os
+masksatr_lib = load_shlib(os.path.join(
+    os.environ["PIXCORRECT_DIR"],
+    'lib',
+    'libmasksatr',
+))
 mask_saturation_c = masksatr_lib.mask_saturation
 mask_saturation_c.restype = ctypes.c_int
 mask_saturation_c.argtypes = [DESImageCStruct, ctypes.POINTER(ctypes.c_int)]
