From 6c74f9f9e937f81a6a9e9fa70150a502a34e43a0 Mon Sep 17 00:00:00 2001
From: Ghislain Antony Vaillant <ghisvail@gmail.com>
Date: Mon, 31 May 2021 22:29:50 +0200
Subject: [PATCH] No MorphologicalContourInterpolation ITK module

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 65b8889..54ea7a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,6 @@ IF(NOT CONVERT3D_BUILD_AS_SUBPROJECT)
     ITKThresholding
     ITKTransform
     ITKTransformFactory
-    MorphologicalContourInterpolation
     ITKImageIO
     ITKTransformIO)
 
-- 
2.30.2

