diff --git qutip/orig.__init__.py qutip/__init__.py
index 8b98ce6..b4d1b0e 100644
--- qutip/orig.__init__.py
+++ qutip/__init__.py
@@ -103,26 +103,6 @@ else:
 del top_path
 
 # -----------------------------------------------------------------------------
-# setup the cython environment
-#
-_cython_requirement = "0.15.0"
-try:
-    import Cython
-    if _version2int(Cython.__version__) < _version2int(_cython_requirement):
-        print("QuTiP warning: old version of cython detected " +
-              ("(%s), requiring %s." %
-               (Cython.__version__, _cython_requirement)))
-
-    import pyximport
-    os.environ['CFLAGS'] = '-O3 -w -ffast-math -march=native -mfpmath=sse'
-    pyximport.install(setup_args={'include_dirs': [numpy.get_include()]})
-
-except Exception as e:
-    print("QuTiP warning: Cython setup failed: " + str(e))
-else:
-    del Cython, pyximport
-
-# -----------------------------------------------------------------------------
 # Load user configuration if present: override defaults.
 #
 try:
