diff --git a/cmake/GtsamTesting.cmake b/cmake/GtsamTesting.cmake
index 573fb696a..db2452848 100644
--- a/cmake/GtsamTesting.cmake
+++ b/cmake/GtsamTesting.cmake
@@ -284,7 +284,7 @@ macro(gtsamAddExesGlob_impl globPatterns excludedFiles linkLibraries groupName b
 
 		# Add executable
 		add_executable(${script_name} ${script_src} ${script_headers})
-		target_link_libraries(${script_name} ${linkLibraries})
+		target_link_libraries(${script_name} ${linkLibraries} rt)
 
 		# Apply user build flags from CMake cache variables:
 		gtsam_apply_build_flags(${script_name})
diff --git a/gtsam/base/FastList.h b/gtsam/base/FastList.h
index 380836d1d..525ceb5d5 100644
--- a/gtsam/base/FastList.h
+++ b/gtsam/base/FastList.h
@@ -21,6 +21,7 @@
 #include <gtsam/base/FastDefaultAllocator.h>
 #include <list>
 #include <boost/utility/enable_if.hpp>
+#include <boost/serialization/library_version_type.hpp>
 #include <boost/serialization/nvp.hpp>
 #include <boost/serialization/list.hpp>
 
diff --git a/gtsam/linear/LossFunctions.h b/gtsam/linear/LossFunctions.h
index 4705721a0..f54ea846b 100644
--- a/gtsam/linear/LossFunctions.h
+++ b/gtsam/linear/LossFunctions.h
@@ -23,6 +23,7 @@
 #include <gtsam/base/Testable.h>
 #include <gtsam/dllexport.h>
 
+#include <boost/serialization/version.hpp>
 #include <boost/serialization/extended_type_info.hpp>
 #include <boost/serialization/nvp.hpp>
 #include <boost/serialization/optional.hpp>
diff --git a/gtsam/linear/SubgraphBuilder.h b/gtsam/linear/SubgraphBuilder.h
index 5247ea2a2..84a477a5e 100644
--- a/gtsam/linear/SubgraphBuilder.h
+++ b/gtsam/linear/SubgraphBuilder.h
@@ -21,6 +21,7 @@
 #include <gtsam/base/types.h>
 #include <gtsam/dllexport.h>
 
+#include <boost/serialization/version.hpp>
 #include <boost/serialization/nvp.hpp>
 #include <boost/shared_ptr.hpp>
 