diff --git a/gtsam/base/serialization.h b/gtsam/base/serialization.h
index 24355c684..e615afe83 100644
--- a/gtsam/base/serialization.h
+++ b/gtsam/base/serialization.h
@@ -25,6 +25,7 @@
 #include <string>
 
 // includes for standard serialization types
+#include <boost/serialization/version.hpp>
 #include <boost/serialization/optional.hpp>
 #include <boost/serialization/shared_ptr.hpp>
 #include <boost/serialization/vector.hpp>
diff --git a/gtsam/linear/NoiseModel.h b/gtsam/linear/NoiseModel.h
index 54b9d955f..a737c5cc8 100644
--- a/gtsam/linear/NoiseModel.h
+++ b/gtsam/linear/NoiseModel.h
@@ -27,6 +27,7 @@
 #include <boost/serialization/extended_type_info.hpp>
 #include <boost/serialization/singleton.hpp>
 #include <boost/serialization/shared_ptr.hpp>
+#include <boost/serialization/version.hpp>
 #include <boost/serialization/optional.hpp>
 
 namespace gtsam {
diff --git a/gtsam/slam/SmartFactorBase.h b/gtsam/slam/SmartFactorBase.h
index ddf56b289..f128bd124 100644
--- a/gtsam/slam/SmartFactorBase.h
+++ b/gtsam/slam/SmartFactorBase.h
@@ -30,6 +30,7 @@
 #include <gtsam/geometry/CameraSet.h>
 
 #include <boost/optional.hpp>
+#include <boost/serialization/version.hpp>
 #include <boost/serialization/optional.hpp>
 #include <boost/make_shared.hpp>
 #include <vector>
