diff --git a/gtsam/3rdparty/CMakeLists.txt b/gtsam/3rdparty/CMakeLists.txt
index c8fecc339..419100863 100644
--- a/gtsam/3rdparty/CMakeLists.txt
+++ b/gtsam/3rdparty/CMakeLists.txt
@@ -53,7 +53,9 @@ if(GTSAM_SUPPORT_NESTED_DISSECTION)
 	add_subdirectory(metis)
 endif()
 
-add_subdirectory(ceres)
+# add_subdirectory(ceres)
+find_package(Ceres REQUIRED)
+include_directories(${CERES_INCLUDE_DIRS})
 
 ############ NOTE:  When updating GeographicLib be sure to disable building their examples
 ############        and unit tests by commenting out their lines:
diff --git a/gtsam/3rdparty/ceres/example.h b/gtsam/3rdparty/ceres/example.h
index 6b051fce0..a3168fdb4 100644
--- a/gtsam/3rdparty/ceres/example.h
+++ b/gtsam/3rdparty/ceres/example.h
@@ -33,7 +33,7 @@
 
 #pragma once
 
-#include <gtsam/3rdparty/ceres/rotation.h>
+#include <ceres/rotation.h>
 
 // Templated pinhole camera model for used with Ceres. The camera is
 // parameterized using 9 parameters: 3 for rotation, 3 for translation, 1 for
diff --git a/gtsam/nonlinear/AdaptAutoDiff.h b/gtsam/nonlinear/AdaptAutoDiff.h
index 682cca29a..a8ce86dd4 100644
--- a/gtsam/nonlinear/AdaptAutoDiff.h
+++ b/gtsam/nonlinear/AdaptAutoDiff.h
@@ -20,7 +20,7 @@
 
 #include <gtsam/base/VectorSpace.h>
 #include <gtsam/base/OptionalJacobian.h>
-#include <gtsam/3rdparty/ceres/autodiff.h>
+#include <ceres/internal/autodiff.h>
 
 #include <boost/static_assert.hpp>
 #include <boost/type_traits/is_base_of.hpp>
