diff --git a/.bazelrc b/.bazelrc
index 379fc832..7f87e94b 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Build with C++17.
+build --cxxopt=-std=c++17
+
 # Display the current git revision in the info block.
 build --stamp --workspace_status_command tools/workspace_status.sh
 
diff --git a/WORKSPACE b/WORKSPACE
index 0ad2bb17..382f6cf9 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -24,7 +24,7 @@ load("@io_bazel_rules_go//go:deps.bzl", "go_rules_dependencies", "go_register_to
 go_rules_dependencies()
 
 go_register_toolchains(
-    go_version = "1.13.4",
+    go_version = "host",
     nogo = "@//:nogo",
 )
 
@@ -62,11 +62,11 @@ protobuf_deps()
 # See releases at https://releases.bazel.build/bazel-toolchains.html
 http_archive(
     name = "bazel_toolchains",
-    sha256 = "a019fbd579ce5aed0239de865b2d8281dbb809efd537bf42e0d366783e8dec65",
-    strip_prefix = "bazel-toolchains-0.29.2",
+    sha256 = "aa0df93b37fae9fe88de4ab7da3bed6396561a3f03d70c562d60f306ddacff43",
+    strip_prefix = "bazel-toolchains-2.0.5",
     urls = [
-        "https://mirror.bazel.build/github.com/bazelbuild/bazel-toolchains/archive/0.29.2.tar.gz",
-        "https://github.com/bazelbuild/bazel-toolchains/archive/0.29.2.tar.gz",
+        "https://mirror.bazel.build/github.com/bazelbuild/bazel-toolchains/archive/2.0.5.tar.gz",
+        "https://github.com/bazelbuild/bazel-toolchains/archive/2.0.5.tar.gz",
     ],
 )
 
