diff --git a/libcontainer/nsenter/cloned_binary.c b/libcontainer/nsenter/cloned_binary.c
index 24cc8c6e..c644b7fb 100644
--- a/libcontainer/nsenter/cloned_binary.c
+++ b/libcontainer/nsenter/cloned_binary.c
@@ -98,6 +98,10 @@ int memfd_create(const char *name, unsigned int flags)
 #define RUNC_MEMFD_SEALS \
 	(F_SEAL_SEAL | F_SEAL_SHRINK | F_SEAL_GROW | F_SEAL_WRITE)
 
+#ifndef O_PATH
+#	define O_PATH 010000000
+#endif
+
 static void *must_realloc(void *ptr, size_t size)
 {
 	void *old = ptr;
