--- Lib/distutils/cygwinccompiler.py.orig	2017-05-09 23:32:15.046506800 +0100
+++ Lib/distutils/cygwinccompiler.py	2017-05-09 23:36:41.868962800 +0100
@@ -299,9 +299,13 @@
             raise CCompilerError(
                 'Cygwin gcc cannot be used with --compiler=mingw32')
 
-        self.set_executables(compiler='gcc -O -Wall',
-                             compiler_so='gcc -mdll -O -Wall',
-                             compiler_cxx='g++ -O -Wall',
+        if sys.maxsize == 2**31 - 1:
+            ms_win=' -DMS_WIN32'
+        else:
+            ms_win=' -DMS_WIN64'
+        self.set_executables(compiler='gcc -O -Wall'+ms_win,
+                             compiler_so='gcc -mdll -O -Wall'+ms_win,
+                             compiler_cxx='g++ -O -Wall'+ms_win,
                              linker_exe='gcc',
                              linker_so='%s %s %s'
                                         % (self.linker_dll, shared_option,
