diff --git Lib/distutils/unixccompiler.py Lib/distutils/unixccompiler.py
index 094a2f0..4252605 100644
--- Lib/distutils/unixccompiler.py
+++ Lib/distutils/unixccompiler.py
@@ -191,6 +191,7 @@ class UnixCCompiler(CCompiler):
                     linker[i] = self.compiler_cxx[i]
 
                 if sys.platform == 'darwin':
+                    ld_args = ['-arch', 'x86_64'] + ld_args
                     linker = _osx_support.compiler_fixup(linker, ld_args)
 
                 self.spawn(linker + ld_args)
