--- Lib/distutils/cygwinccompiler.py	2016-10-13 19:28:33.557226300 +0100
+++ Lib/distutils/cygwinccompiler.py	2016-10-13 19:29:27.686231200 +0100
@@ -82,6 +82,9 @@
         elif msc_ver == '1600':
             # VS2010 / MSVC 10.0
             return ['msvcr100']
+        elif int(msc_ver) >= 1900:
+            # VS2015 / MSVC 14.0
+            return ['msvcr140']
         else:
             raise ValueError("Unknown MS Compiler version %s " % msc_ver)
 
