Subject: [PATCH] new path include for conda package
---
Index: nrlmsise-00_test.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nrlmsise-00_test.c b/nrlmsise-00_test.c
--- a/nrlmsise-00_test.c	(revision bc9a2feba4344e74201281e563332688a4d09cc3)
+++ b/nrlmsise-00_test.c	(date 1721298382975)
@@ -23,7 +23,7 @@
 /* ------------------------------------------------------------------- */
 
 #include <stdio.h>
-#include "nrlmsise-00.h"
+#include "nrlmsise-00/nrlmsise-00.h"
 
 
 
