From 002708e4d7c45f26c4f4e31107f992df809cc0ce Mon Sep 17 00:00:00 2001
From: Tim Snyder <snyder.tim@gmail.com>
Date: Mon, 21 Mar 2022 16:17:08 +0000
Subject: [PATCH] rm unneeded unistd.h include

---
 execs.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/execs.c b/execs.c
index ca2d6cd34..feec95543 100644
--- a/execs.c
+++ b/execs.c
@@ -20,7 +20,6 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <unistd.h>
 #include <execs.h>
 
 //char *sn[] = { "END", "SPACE", "CHAR", "SGLQ", "DBLQ", "ESCAPE", "SEMIC", "VAR", "ESCVAR", "DBLESC" };
-- 
2.35.0

