From 0e6ce9a24440b24a35a182e2a1f24c6d3a4b1668 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Sat, 11 Apr 2026 20:39:26 -0600
Subject: [PATCH] fix: Add missing requirements.txt

---
 requirements.txt | 3 +++
 1 file changed, 3 insertions(+)
 create mode 100644 requirements.txt

diff --git a/requirements.txt b/requirements.txt
new file mode 100644
index 0000000..23aab03
--- /dev/null
+++ b/requirements.txt
@@ -0,0 +1,3 @@
+equinox >= 0.11.0, < 0.14
+jax >= 0.4.36, < 0.10
+numpy >= 1.20.0, < 2.5
-- 
2.53.0

