From d13a8e0296c72f5efb1d93edc4f7ee837df2cb6e Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Wed, 17 Oct 2018 15:36:01 -0500
Subject: [PATCH] remove march=native

---
 setup.py | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/setup.py b/setup.py
index 0e68f80..6e2d86e 100644
--- a/setup.py
+++ b/setup.py
@@ -15,14 +15,6 @@ from lightfm import __version__ as version  # NOQA
 def define_extensions(use_openmp):
     compile_args = ['-ffast-math']
 
-    # There are problems with illegal ASM instructions
-    # when using the Anaconda distribution (at least on OSX).
-    # This could be because Anaconda uses its own assembler?
-    # To work around this we do not add -march=native if we
-    # know we're dealing with Anaconda
-    if 'anaconda' not in sys.version.lower():
-        compile_args.append('-march=native')
-
     if not use_openmp:
         print('Compiling without OpenMP support.')
         return [Extension("lightfm._lightfm_fast_no_openmp",
-- 
2.18.0

