diff --git a/src/platform_specifics.cpp b/src/platform_specifics.cpp
--- a/src/platform_specifics.cpp
+++ b/src/platform_specifics.cpp
@@ -8,7 +8,7 @@
 #ifdef _MSC_VER
 #include <intrin.h>
 #endif
-#ifdef __GNUC__
+#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
 #include <x86intrin.h>
 #endif
 #include <string>
@@ -42,10 +42,12 @@
 Sint64 PractRand::Internals::high_resolution_time() {
 #if defined _MSC_VER && ((defined _M_IX86 && _M_IX86 >= 500) || defined _M_X64 || defined _M_AMD64)
 	return __rdtsc();
-#elif defined __GNUC__
+#elif defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
 	return __rdtsc();
 #else
-	return std::chrono::high_resolution_clock();
+	return static_cast<Sint64>(
+		std::chrono::high_resolution_clock::now().time_since_epoch().count()
+	);
 #endif
 }
 bool PractRand::Internals::add_entropy_automatically( PractRand::RNGs::vRNG *rng, int milliseconds ) {
