diff --git a/setup.py b/setup.py
index d31a24f92..5632fe1f7 100644
--- a/setup.py
+++ b/setup.py
@@ -196,6 +196,7 @@ class BuildSphinxCommand(Command):
         import build_sphinx
         build_sphinx.rebuild()
 
+_ = subprocess.call([sys.executable, "src/sas/qtgui/convertUI.py"])
 
 # sas module
 package_dir["sas"] = os.path.join("src", "sas")
@@ -453,6 +454,18 @@ package_data['sas.sasview'] = ['images/*',
                                'test/upcoming_formats/*',
                                ]
 packages.append("sas.sasview")
+package_data['sas.qtgui'] = ['Calculators/UI/*',
+                             'MainWindow/UI/*',
+                             'Perspectives/Corfunc/UI/*',
+                             'Perspectives/Fitting/UI/*',
+                             'Perspectives/Invariant/UI/*',
+                             'Perspectives/Inversion/UI/*',
+                             'Plotting/UI/*',
+                             'Utilities/UI/*',
+                             'UI/*',
+                             'UI/res/*',
+                             ]
+packages.append("sas.qtgui")
 
 required = [
     'bumps>=0.7.5.9', 'periodictable>=1.5.0', 'pyparsing>=2.0.0',
@@ -489,7 +502,7 @@ setup(
     zip_safe=False,
     entry_points={
         'console_scripts': [
-            "sasview = sas.run",
+            "sasview=sas.qtgui.MainWindow.MainWindow:run_sasview",
         ]
     },
     cmdclass={'build_ext': build_ext_subclass,
