--- setup.py
+++ setup.py
@@ -6,11 +6,10 @@ except ImportError:
     from distutils.core import setup

 setup(name='parcels',
-      version='0.0.1',
+      version='1.0',
       description="""Framework for Lagrangian tracking of virtual
       ocean particles in the petascale age.""",
       author="Imperial College London",
-      use_scm_version=True,
       setup_requires=['setuptools_scm'],
       packages=['parcels'],
       package_data={'parcels': ['parcels/include/*',
                                 'parcels/examples/*']},
