--- setup.py
+++ setup.py
@@ -1,18 +1,17 @@
 """Install Parcels and dependencies."""

 try:
-    from setuptools import setup
+    from setuptools import setup, find_packages
 except ImportError:
-    from distutils.core import setup
+    from distutils.core import setup, find_packages

 setup(name='parcels',
       version='0.0.1',
       description="""Framework for Lagrangian tracking of virtual
       ocean particles in the petascale age.""",
       author="Imperial College London",
-      use_scm_version=True,
       setup_requires=['setuptools_scm'],
-      packages=['parcels'],
+      packages=find_packages(),
       package_data={'parcels': ['parcels/include/*',
                                 'parcels/examples/*']},
       include_package_data=True,
