diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee2d3f255..82ff928ca 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3649,7 +3649,7 @@ IF(NOT TARGET fxdiv)
     IF(NOT FXDIV_HDR)
       MESSAGE(FATAL_ERROR "Cannot find fxdiv")
     ENDIF()
-    ADD_LIBRARY(fxdiv STATIC "${FXDIV_HDR}")
+    ADD_LIBRARY(fxdiv INTERFACE "${FXDIV_HDR}")
     SET_PROPERTY(TARGET fxdiv PROPERTY LINKER_LANGUAGE C)
   ENDIF()
 ENDIF()
@@ -3668,7 +3668,7 @@ IF(NOT TARGET fp16)
     IF(NOT FP16_HDR)
       MESSAGE(FATAL_ERROR "Cannot find fp16")
     ENDIF()
-    ADD_LIBRARY(fp16 STATIC "${FP16_HDR}")
+    ADD_LIBRARY(fp16 INTERFACE "${FP16_HDR}")
     SET_PROPERTY(TARGET fp16 PROPERTY LINKER_LANGUAGE C)
   ENDIF()
 ENDIF()

