diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b7c6a5867..b88c9467a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -75,29 +75,7 @@ endif()
 ### Enter subdirectories
 
 # VCGLib -- required
-if (VCGDIR) # VCGDIR exists - using custom user vcglib path
-	if(EXISTS ${VCGDIR})
-		add_subdirectory(${VCGDIR} {CMAKE_CURRENT_BINARY_DIR}/vcglib)
-		message(STATUS "- VCGLib - using custom VCGDIR path library")
-	else()
-		set(VCGDIR NOTFOUND)
-	endif()
-else()
-	get_filename_component(VCGDIR "${CMAKE_CURRENT_LIST_DIR}/vcglib" ABSOLUTE)
-	if(EXISTS ${VCGDIR})
-		add_subdirectory(${VCGDIR})
-		message(STATUS "- VCGLib - using bundled source")
-	else()
-		set(VCGDIR NOTFOUND)
-	endif()
-endif()
-set(VCGDIR "${VCGDIR}")
-
-if(NOT VCGDIR)
-	message(FATAL_ERROR "VCGLib not found. Please clone recursively the MeshLab repo.")
-else ()
-	message(STATUS "- VCGLib directory: " ${VCGDIR})
-endif()
+find_package(vcglib CONFIG REQUIRED)
 
 # External
 set(EXTERNAL_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external)
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 0e640fdd4..33d8b8607 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -211,6 +211,7 @@ target_link_libraries(
 		vcglib
 		external-glew
 		external-easyexif
+		vcglib
 )
 
 set_property(TARGET meshlab-common PROPERTY FOLDER Core)
