--- a/libQGLViewer-2.7.2.pro	2019-11-28 03:00:03.000000000 +1000
+++ b/libQGLViewer-2.7.2.pro	2021-08-03 09:14:06.232237970 +1000
@@ -1,6 +1,6 @@
 CONFIG       += ordered
 TEMPLATE      = subdirs
-SUBDIRS       = QGLViewer examples designerPlugin
+SUBDIRS       = QGLViewer # examples designerPlugin
 
 QT_VERSION=$$[QT_VERSION]
 
--- a/QGLViewer/QGLViewer.pro	2019-11-28 02:59:59.000000000 +1000
+++ b/QGLViewer/QGLViewer.pro	2021-08-03 09:13:03.172237203 +1000
@@ -148,7 +148,7 @@
 	macx|darwin-g++ {
 		# GLU is part of the OpenGL framework
 	} else {
-		QMAKE_LIBS_OPENGL *= -lGLU
+		QMAKE_LIBS_OPENGL *= -L$${LIB_DIR} -lGLU
 
 		isEmpty( NO_QT_VERSION_SUFFIX ) {
 			equals (QT_MAJOR_VERSION, 4) {
