diff --git a/test/C/Makefile.in b/test/C/Makefile.in
index 7041dd5..0a40ad9 100644
--- a/test/C/Makefile.in
+++ b/test/C/Makefile.in
@@ -102,12 +102,11 @@ CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
 am__EXEEXT_1 = test-arrow$(EXEEXT) test-band$(EXEEXT) \
-	test-box$(EXEEXT) test-cairo-xw$(EXEEXT) \
 	test-change-page$(EXEEXT) test-circle$(EXEEXT) \
 	test-colour-index$(EXEEXT) test-environment$(EXEEXT) \
 	test-error-bars$(EXEEXT) test-format-number$(EXEEXT) \
-	test-giza-xw$(EXEEXT) test-line-cap$(EXEEXT) \
-	test-line-style$(EXEEXT) test-openclose$(EXEEXT) \
+	test-line-cap$(EXEEXT) \
+	test-line-style$(EXEEXT) \
 	test-pdf$(EXEEXT) test-png$(EXEEXT) test-points$(EXEEXT) \
 	test-qtext$(EXEEXT) test-rectangle$(EXEEXT) \
 	test-set-line-width$(EXEEXT) test-svg$(EXEEXT) \
