diff --git a/setup.py b/setup.py
index a41a5d7..c13caed 100644
--- a/setup.py
+++ b/setup.py
@@ -82,6 +82,8 @@ class CMakeBuild(build_ext):
             # generator name.
             if not single_config and not contains_arch:
                 cmake_args += ["-A", PLAT_TO_CMAKE[self.plat_name]]
+            else:
+                cmake_args += ["-DCMAKE_GENERATOR_PLATFORM=''", "-DCMAKE_GENERATOR_TOOLSET=''"]
 
             # Multi-config generators have a different way to specify configs
             if not single_config:
