diff --git a/CMakeLists.txt b/CMakeLists.txt
index dd16157..c0997b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,6 +127,7 @@ set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
     --avoid-protected-hack --enable-pyside-extensions
     ${qt_includes} ${ads_includes}
     -I${CMAKE_SOURCE_DIR}/src
+    -I$ENV{BUILD_PREFIX}/include/c++/v1/
     -T${PYSIDE_TYPESYSTEMS}
     --output-directory=${CMAKE_CURRENT_BINARY_DIR}
     )
