From d15a54e320aa8ab367f196f6d0109e5501bef6ef Mon Sep 17 00:00:00 2001
From: terraputix <terraputix@mailbox.org>
Date: Wed, 11 Mar 2026 22:12:23 +0100
Subject: [PATCH] improve upstream build script

---
 Cargo.lock | 4 ++--
 Cargo.toml | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index a4c6023..1de80d9 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -837,7 +837,7 @@ dependencies = [
 [[package]]
 name = "om-file-format-sys"
 version = "0.0.0"
-source = "git+https://github.com/open-meteo/om-file-format?rev=9feec4b8f01fee2843ddb940ddc4dd92b2981437#9feec4b8f01fee2843ddb940ddc4dd92b2981437"
+source = "git+https://github.com/open-meteo/om-file-format?rev=7e688eca60ce9600d9111647d3e4ed9bb3c6e8f1#7e688eca60ce9600d9111647d3e4ed9bb3c6e8f1"
 dependencies = [
  "bindgen",
  "cc",
@@ -847,7 +847,7 @@ dependencies = [
 [[package]]
 name = "omfiles"
 version = "0.1.0"
-source = "git+https://github.com/open-meteo/rust-omfiles?rev=a96adaf2c0808821452eaa5e54ea5032d5834184#a96adaf2c0808821452eaa5e54ea5032d5834184"
+source = "git+https://github.com/open-meteo/rust-omfiles?rev=f9d139d9dfe4aeb5b25afc01aff0b0abcedc9644#f9d139d9dfe4aeb5b25afc01aff0b0abcedc9644"
 dependencies = [
  "async-executor",
  "async-lock",
diff --git a/Cargo.toml b/Cargo.toml
index 5231af8..c3ffbcf 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,8 +24,8 @@ async-lock = "3.4.2"
 numpy = "0.27"
 num-traits = "0.2"
 delegate = "0.13"
-omfiles-rs = { git = "https://github.com/open-meteo/rust-omfiles", rev = "a96adaf2c0808821452eaa5e54ea5032d5834184", package = "omfiles", features = ["metadata-tree"] }
-om-file-format-sys = { git = "https://github.com/open-meteo/om-file-format", rev = "9feec4b8f01fee2843ddb940ddc4dd92b2981437" }
+omfiles-rs = { git = "https://github.com/open-meteo/rust-omfiles", rev = "f9d139d9dfe4aeb5b25afc01aff0b0abcedc9644", package = "omfiles", features = ["metadata-tree"] }
+om-file-format-sys = { git = "https://github.com/open-meteo/om-file-format", rev = "7e688eca60ce9600d9111647d3e4ed9bb3c6e8f1" }
 thiserror = "2.0.18"
 
 [features]
