diff --git a/MPC/clone_build_tree.pl b/MPC/clone_build_tree.pl
index a1e757b..7bcd4b4 100755
--- a/MPC/clone_build_tree.pl
+++ b/MPC/clone_build_tree.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/combine_dsw.pl b/MPC/combine_dsw.pl
index d5b4d95..664a31b 100755
--- a/MPC/combine_dsw.pl
+++ b/MPC/combine_dsw.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/create_base.pl b/MPC/create_base.pl
index 8991f17..19d7b60 100755
--- a/MPC/create_base.pl
+++ b/MPC/create_base.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/depgen.pl b/MPC/depgen.pl
index 4c54f8c..1feb794 100755
--- a/MPC/depgen.pl
+++ b/MPC/depgen.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/devtools/document_template.pl b/MPC/devtools/document_template.pl
index ffbdcbf..96e680a 100755
--- a/MPC/devtools/document_template.pl
+++ b/MPC/devtools/document_template.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/devtools/highlight_template.pl b/MPC/devtools/highlight_template.pl
index 8a94bcb..6c49e64 100755
--- a/MPC/devtools/highlight_template.pl
+++ b/MPC/devtools/highlight_template.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/generate_export_header.pl b/MPC/generate_export_header.pl
index 8338330..d0fab89 100755
--- a/MPC/generate_export_header.pl
+++ b/MPC/generate_export_header.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/mpc.pl b/MPC/mpc.pl
index aea4628..d5a9ad5 100755
--- a/MPC/mpc.pl
+++ b/MPC/mpc.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/mwc.pl b/MPC/mwc.pl
index f03ffd5..6374fb6 100755
--- a/MPC/mwc.pl
+++ b/MPC/mwc.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/prj_install.pl b/MPC/prj_install.pl
index 9bcb23a..9ab8e23 100755
--- a/MPC/prj_install.pl
+++ b/MPC/prj_install.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/registry.pl b/MPC/registry.pl
index d221b1b..30fdf00 100755
--- a/MPC/registry.pl
+++ b/MPC/registry.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/MPC/vs_postclean.pl b/MPC/vs_postclean.pl
index 4f44809..7ce541a 100755
--- a/MPC/vs_postclean.pl
+++ b/MPC/vs_postclean.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/bin/PerlACE/ConfigList.pm b/bin/PerlACE/ConfigList.pm
index 612c94c..522f586 100644
--- a/bin/PerlACE/ConfigList.pm
+++ b/bin/PerlACE/ConfigList.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::ConfigList;
 use strict;
diff --git a/bin/PerlACE/MSProject.pm b/bin/PerlACE/MSProject.pm
index 4c65006..7cb53c5 100644
--- a/bin/PerlACE/MSProject.pm
+++ b/bin/PerlACE/MSProject.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::MSProject;
 
diff --git a/bin/PerlACE/Process.pm b/bin/PerlACE/Process.pm
index 4bccbe7..0f2aa97 100644
--- a/bin/PerlACE/Process.pm
+++ b/bin/PerlACE/Process.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::Process;
 
diff --git a/bin/PerlACE/ProcessAndroid.pm b/bin/PerlACE/ProcessAndroid.pm
index a00864a..a5da680 100644
--- a/bin/PerlACE/ProcessAndroid.pm
+++ b/bin/PerlACE/ProcessAndroid.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::ProcessAndroid;
 
diff --git a/bin/PerlACE/ProcessLVRT.pm b/bin/PerlACE/ProcessLVRT.pm
index f5ee2de..9865941 100644
--- a/bin/PerlACE/ProcessLVRT.pm
+++ b/bin/PerlACE/ProcessLVRT.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 #
 # ProcessLVRT - how to run ACE+TAO tests on a LabVIEW RT target.
 # Tests on LabVIEW RT are not executables - LabVIEW RT can't start plain
diff --git a/bin/PerlACE/ProcessVX.pm b/bin/PerlACE/ProcessVX.pm
index 808c6ac..570e12d 100644
--- a/bin/PerlACE/ProcessVX.pm
+++ b/bin/PerlACE/ProcessVX.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::ProcessVX;
 
diff --git a/bin/PerlACE/ProcessVX_Unix.pm b/bin/PerlACE/ProcessVX_Unix.pm
index 56b5641..3ce07a3 100644
--- a/bin/PerlACE/ProcessVX_Unix.pm
+++ b/bin/PerlACE/ProcessVX_Unix.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::ProcessVX;
 
diff --git a/bin/PerlACE/ProcessVX_Win32.pm b/bin/PerlACE/ProcessVX_Win32.pm
index d651d38..d457fcd 100644
--- a/bin/PerlACE/ProcessVX_Win32.pm
+++ b/bin/PerlACE/ProcessVX_Win32.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::ProcessVX;
 
diff --git a/bin/PerlACE/ProcessWinCE.pm b/bin/PerlACE/ProcessWinCE.pm
index f0784e4..74eecb3 100644
--- a/bin/PerlACE/ProcessWinCE.pm
+++ b/bin/PerlACE/ProcessWinCE.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::ProcessVX;
 
diff --git a/bin/PerlACE/ProcessWinCE_Unix.pm b/bin/PerlACE/ProcessWinCE_Unix.pm
index 570084a..1a15361 100644
--- a/bin/PerlACE/ProcessWinCE_Unix.pm
+++ b/bin/PerlACE/ProcessWinCE_Unix.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::ProcessVX;
 
diff --git a/bin/PerlACE/Process_Unix.pm b/bin/PerlACE/Process_Unix.pm
index 6202626..e40f9fb 100644
--- a/bin/PerlACE/Process_Unix.pm
+++ b/bin/PerlACE/Process_Unix.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::Process;
 
diff --git a/bin/PerlACE/Process_VMS.pm b/bin/PerlACE/Process_VMS.pm
index 9affb6d..1056288 100644
--- a/bin/PerlACE/Process_VMS.pm
+++ b/bin/PerlACE/Process_VMS.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::Process;
 
diff --git a/bin/PerlACE/Process_Win32.pm b/bin/PerlACE/Process_Win32.pm
index 3fac894..ddc365e 100644
--- a/bin/PerlACE/Process_Win32.pm
+++ b/bin/PerlACE/Process_Win32.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 use PerlACE::Run_Test;
 
diff --git a/bin/PerlACE/Run_Test.pm b/bin/PerlACE/Run_Test.pm
index 4ae912f..f59d956 100644
--- a/bin/PerlACE/Run_Test.pm
+++ b/bin/PerlACE/Run_Test.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 # This module contains a few miscellanous functions and some
 # startup ARGV processing that is used by all tests.
diff --git a/bin/PerlACE/TestTarget.pm b/bin/PerlACE/TestTarget.pm
index de838de..1cf65db 100644
--- a/bin/PerlACE/TestTarget.pm
+++ b/bin/PerlACE/TestTarget.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 #
 # The TestTarget class is for operations that are per-target while testing.
 # They can be overridden for specific needs like embedded systems, etc.
diff --git a/bin/PerlACE/TestTarget_Android.pm b/bin/PerlACE/TestTarget_Android.pm
index b6d0859..29587a9 100644
--- a/bin/PerlACE/TestTarget_Android.pm
+++ b/bin/PerlACE/TestTarget_Android.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 package PerlACE::TestTarget_Android;
 
 # ******************************************************************
diff --git a/bin/PerlACE/TestTarget_LVRT.pm b/bin/PerlACE/TestTarget_LVRT.pm
index 79371d9..1726290 100644
--- a/bin/PerlACE/TestTarget_LVRT.pm
+++ b/bin/PerlACE/TestTarget_LVRT.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 #
 # TestTarget_LVRT - how to manage the test environment on a LabVIEW RT target.
 #
diff --git a/bin/PerlACE/TestTarget_VxWorks.pm b/bin/PerlACE/TestTarget_VxWorks.pm
index c46eb88..b6e01d0 100644
--- a/bin/PerlACE/TestTarget_VxWorks.pm
+++ b/bin/PerlACE/TestTarget_VxWorks.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 
 package PerlACE::TestTarget_VxWorks;
 
diff --git a/bin/PerlACE/TestTarget_WinCE.pm b/bin/PerlACE/TestTarget_WinCE.pm
index 2aba983..3f12390 100644
--- a/bin/PerlACE/TestTarget_WinCE.pm
+++ b/bin/PerlACE/TestTarget_WinCE.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 package PerlACE::TestTarget_WinCE;
 
 # ******************************************************************
diff --git a/bin/Uniqueid.pm b/bin/Uniqueid.pm
index 5ff9540..e1638ed 100644
--- a/bin/Uniqueid.pm
+++ b/bin/Uniqueid.pm
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 sub uniqueid
 {
   if ($^O eq "MSWin32")
diff --git a/bin/depgen.pl b/bin/depgen.pl
index 2ebecf9..4e6d536 100755
--- a/bin/depgen.pl
+++ b/bin/depgen.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/bin/generate_export_file.pl b/bin/generate_export_file.pl
index 99b36d2..735f1d5 100755
--- a/bin/generate_export_file.pl
+++ b/bin/generate_export_file.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
     & eval 'exec perl -S $0 $argv:q'
     if 0;
diff --git a/bin/mpc.pl b/bin/mpc.pl
index c1cfc2c..b186091 100755
--- a/bin/mpc.pl
+++ b/bin/mpc.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/bin/mwc.pl b/bin/mwc.pl
index c2afdb7..c43870b 100755
--- a/bin/mwc.pl
+++ b/bin/mwc.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/bin/split-cpp.pl b/bin/split-cpp.pl
index 19fee2a..9d5ca86 100755
--- a/bin/split-cpp.pl
+++ b/bin/split-cpp.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/examples/IPC_SAP/SOCK_SAP/summarize.pl b/examples/IPC_SAP/SOCK_SAP/summarize.pl
index c4c0935..4f45686 100755
--- a/examples/IPC_SAP/SOCK_SAP/summarize.pl
+++ b/examples/IPC_SAP/SOCK_SAP/summarize.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/examples/IPC_SAP/SSL_SAP/summarize.pl b/examples/IPC_SAP/SSL_SAP/summarize.pl
index c4c0935..4f45686 100755
--- a/examples/IPC_SAP/SSL_SAP/summarize.pl
+++ b/examples/IPC_SAP/SSL_SAP/summarize.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#! /usr/bin/env perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
     if 0;
diff --git a/protocols/ace/HTBP/HTID_Generator.cgi b/protocols/ace/HTBP/HTID_Generator.cgi
index 0d57a55..5c10fba 100644
--- a/protocols/ace/HTBP/HTID_Generator.cgi
+++ b/protocols/ace/HTBP/HTID_Generator.cgi
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#! /usr/bin/env perl
 
 # This is a sample script that may be used to create domain specific
 # unique ID's rather than using UUID. In a typical scenario, this HTID
