From dac24092e36a23c69842350506a5a7dd25b1bfed Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 25 Oct 2020 13:03:08 +0100
Subject: [PATCH] Update QE_path

---
 QE_path | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/QE_path b/QE_path
index 40632d7..0de3606 100755
--- a/QE_path
+++ b/QE_path
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-PWDIR='/path_to_quantum_espresso/'
+PWDIR={PREFIX}
