Subject: [PATCH] add bigobj compile option
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision e05b03fc156f4e10214ece709c01874365d43f63)
+++ b/CMakeLists.txt	(date 1703863325322)
@@ -111,6 +111,7 @@
         "MEDCOUPLING_ENABLE_PARTITIONER;MEDCOUPLING_USE_MPI" OFF)
 SET(XDR_DEFAULT_OPTION ON)
 IF(WIN32)
+    add_compile_options(/bigobj)
     SET(XDR_DEFAULT_OPTION OFF)
 ENDIF()
 CMAKE_DEPENDENT_OPTION(MEDCOUPLING_MEDLOADER_USE_XDR "Enable use of XDR for SauvReader." ${XDR_DEFAULT_OPTION} "NOT MEDCOUPLING_MICROMED" OFF)
