diff --git a/cmake/FindSalomeMEDFile.cmake b/cmake/FindSalomeMEDFile.cmake
index 960291a..3cc4de6 100644
--- a/cmake/FindSalomeMEDFile.cmake
+++ b/cmake/FindSalomeMEDFile.cmake
@@ -47,10 +47,11 @@ IF(MEDFile_FOUND)
     SET(MED_INT_SIZE UNKNOWN)
   ENDIF()
   IF(MED_INT_SIZE EQUAL 8)
-    SET(MED_INT_IS_LONG TRUE)
+    SET(_MED_INT_IS_LONG TRUE)
   ELSE()
-    SET(MED_INT_IS_LONG FALSE)
+    SET(_MED_INT_IS_LONG FALSE)
   ENDIF()
+  set(MED_INT_IS_LONG ${_MED_INT_IS_LONG} CACHE BOOL "MED_INT_IS_LONG")
   MESSAGE(STATUS "MEDFile: size of med_int is ${MED_INT_SIZE}")
   UNSET(_med_int_cxx)
   UNSET(_med_int_run_result)
