Subject: [PATCH] find threads
---
Index: src/MEDLoader/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/MEDLoader/CMakeLists.txt b/src/MEDLoader/CMakeLists.txt
--- a/src/MEDLoader/CMakeLists.txt	(revision e05b03fc156f4e10214ece709c01874365d43f63)
+++ b/src/MEDLoader/CMakeLists.txt	(date 1703854529682)
@@ -42,6 +42,7 @@
     SET(MEDLOADER_XDR_LIBRARIES ${XDR_LIBRARIES})
   ENDIF(MEDCOUPLING_MEDLOADER_USE_XDR)
 ELSE(WIN32)
+  find_package(Threads REQUIRED)
   SET(MEDLOADER_XDR_INCLUDE_DIRS ${XDR_INCLUDE_DIRS})
   SET(MEDLOADER_XDR_LIBRARIES ${XDR_LIBRARIES})
 ENDIF(WIN32)
@@ -92,8 +93,8 @@
   )

 ADD_LIBRARY(medloader ${medloader_SOURCES})
-SET_TARGET_PROPERTIES(medloader PROPERTIES COMPILE_FLAGS "")
-TARGET_LINK_LIBRARIES(medloader medcouplingcpp ${MEDFILE_C_LIBRARIES} ${HDF5_LIBRARIES} ${MEDLOADER_XDR_LIBRARIES})
+#SET_TARGET_PROPERTIES(medloader PROPERTIES COMPILE_FLAGS "")
+TARGET_LINK_LIBRARIES(medloader medcouplingcpp ${MEDFILE_C_LIBRARIES} ${HDF5_LIBRARIES} ${MEDLOADER_XDR_LIBRARIES} Threads::Threads)
 INSTALL(TARGETS medloader EXPORT ${PROJECT_NAME}TargetGroup DESTINATION ${MEDCOUPLING_INSTALL_LIBS})

 FILE(GLOB medloader_HEADERS_HXX "${CMAKE_CURRENT_SOURCE_DIR}/*.hxx")
