Subject: [PATCH] always set med int to 8
---
Index: cmake/FindSalomeMEDFile.cmake
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cmake/FindSalomeMEDFile.cmake b/cmake/FindSalomeMEDFile.cmake
--- a/cmake/FindSalomeMEDFile.cmake	(revision e05b03fc156f4e10214ece709c01874365d43f63)
+++ b/cmake/FindSalomeMEDFile.cmake	(date 1703857971142)
@@ -32,25 +32,9 @@
   SALOME_ACCUMULATE_HEADERS(MEDFILE_INCLUDE_DIRS)
   SALOME_ACCUMULATE_ENVIRONMENT(LD_LIBRARY_PATH ${MEDFILE_LIBRARIES})
 
-  # Check size of med_int
-  SET(_med_int_cxx ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/check_med_int_size.cxx)
-  FILE(WRITE ${_med_int_cxx}
-       "#include <med.h>\n#include <stdio.h>\nint main(){printf(\"%d\", sizeof(med_int)); return 0;}")
-  TRY_RUN(_med_int_run_result _med_int_compile_results
-          ${CMAKE_BINARY_DIR} ${_med_int_cxx}
-          CMAKE_FLAGS "-DINCLUDE_DIRECTORIES:STRING=${MEDFILE_INCLUDE_DIRS};${HDF5_INCLUDE_DIR}"
-	  LINK_LIBRARIES ${MEDFILE_C_LIBRARIES} ${HDF5_LIBRARIES}
-	  RUN_OUTPUT_VARIABLE _med_int_output)
-  IF(_med_int_compile_results)
-    SET(MED_INT_SIZE ${_med_int_output})
-  ELSE()
-    SET(MED_INT_SIZE UNKNOWN)
-  ENDIF()
-  IF(MED_INT_SIZE EQUAL 8)
-    SET(MED_INT_IS_LONG TRUE)
-  ELSE()
-    SET(MED_INT_IS_LONG FALSE)
-  ENDIF()
+  # Always set med_int to 8
+  SET(MED_INT_SIZE 8)
+  SET(MED_INT_IS_LONG TRUE)
   MESSAGE(STATUS "MEDFile: size of med_int is ${MED_INT_SIZE}")
   UNSET(_med_int_cxx)
   UNSET(_med_int_run_result)
