Subject: [PATCH] swig big file compile flag
---
Index: src/RENUMBER_Swig/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/RENUMBER_Swig/CMakeLists.txt b/src/RENUMBER_Swig/CMakeLists.txt
--- a/src/RENUMBER_Swig/CMakeLists.txt	(revision 5cb0b9f7852dbb4d68059c363cc38377ed5b3ddb)
+++ b/src/RENUMBER_Swig/CMakeLists.txt	(date 1703937028960)
@@ -70,6 +70,7 @@

 IF(WIN32)
   SET_TARGET_PROPERTIES(_MEDRenumber PROPERTIES DEBUG_OUTPUT_NAME _MEDRenumber_d)
+  TARGET_COMPILE_OPTIONS(_MEDRenumber PRIVATE /bigobj)
 ENDIF(WIN32)
 INSTALL(TARGETS ${SWIG_MODULE_MEDRenumber_REAL_NAME} DESTINATION ${MEDCOUPLING_INSTALL_PYTHON})

Index: src/MEDCoupling_Swig/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/MEDCoupling_Swig/CMakeLists.txt b/src/MEDCoupling_Swig/CMakeLists.txt
--- a/src/MEDCoupling_Swig/CMakeLists.txt	(revision 5cb0b9f7852dbb4d68059c363cc38377ed5b3ddb)
+++ b/src/MEDCoupling_Swig/CMakeLists.txt	(date 1703937272130)
@@ -123,6 +123,7 @@

 IF(WIN32 OR CYGWIN)
   SWIG_LINK_LIBRARIES(${MEDCouling_target_name} ${PYTHON_LIBRARIES} ${PLATFORM_LIBS} medcouplingcpp)
+  TARGET_COMPILE_OPTIONS(medcouplingcpp PRIVATE /bigobj)
 ELSE(WIN32)
   # ${PYTHON_LIBRARIES} not needed see https://www.python.org/dev/peps/pep-0513/#libpythonx-y-so-1
   SWIG_LINK_LIBRARIES(${MEDCouling_target_name} ${PLATFORM_LIBS} medcouplingcpp)
@@ -165,6 +166,8 @@
 IF(WIN32)
   SET_TARGET_PROPERTIES(_MEDCouplingRemapper PROPERTIES DEBUG_OUTPUT_NAME _MEDCouplingRemapper_d)
   SET_TARGET_PROPERTIES(_${MEDCouling_target_name} PROPERTIES DEBUG_OUTPUT_NAME _${MEDCouling_target_name}_d)
+  TARGET_COMPILE_OPTIONS(_MEDCouplingRemapper PRIVATE /bigobj)
+  TARGET_COMPILE_OPTIONS(_${MEDCouling_target_name} PRIVATE /bigobj)
 ENDIF(WIN32)
 INSTALL(TARGETS ${SWIG_MODULE_${MEDCouling_target_name}_REAL_NAME} ${SWIG_MODULE_MEDCouplingRemapper_REAL_NAME} DESTINATION ${MEDCOUPLING_INSTALL_PYTHON})

