--- setup.py.orig	2021-07-06 17:17:07.000000000 -0600
+++ setup.py	2021-07-06 17:19:48.000000000 -0600
@@ -6,9 +6,9 @@
 from setuptools import Extension, setup
 from setuptools.command.build_ext import build_ext
 
-src_root = pathlib.Path(__file__).parent / "lib"
+src_root = pathlib.Path("lib")
 sources = [
-    src_root.joinpath(s).as_posix() for s in ["Ppmd7.c", "Ppmd7Dec.c", "Ppmd7Enc.c", "Ppmd8.c", "Ppmd8Dec.c", "Ppmd8Enc.c"]
+    src_root.joinpath(s).relative_to(".") for s in ["Ppmd7.c", "Ppmd7Dec.c", "Ppmd7Enc.c", "Ppmd8.c", "Ppmd8Dec.c", "Ppmd8Enc.c"]
 ]
 
 _ppmd_extension = Extension("pyppmd._ppmd", sources)
@@ -51,7 +51,7 @@
     def build_extensions(self):
         if "msvc" in self.compiler.compiler_type.lower():
             for extension in self.extensions:
-                more_options = ["/Ob3", "/GF", "/Gy"]
+                more_options = ["/GF", "/Gy"]
                 extension.extra_compile_args.extend(more_options)
         super().build_extensions()
 
