diff --git a/cling/setup.py b/cling/setup.py
index d7c6595..208a708 100755
--- a/cling/setup.py
+++ b/cling/setup.py
@@ -133,7 +133,16 @@ class my_cmake_build(_build):
             if has_avx: extra_args += ' -mavx'
             os.putenv('EXTRA_CLING_ARGS', extra_args)
 
+        platform_flags = os.getenv('CMAKE_PLATFORM_FLAGS')
+        root_flags = os.getenv('CMAKE_ROOT_FLAGS')
+
         CMAKE_COMMAND = ['cmake', srcdir,
+                # Include flags determined in build.sh
+                platform_flags,
+                # Some flags that root-feedstock sets when invoking cmake. Most
+                # importantly, do not build llvm & clang but use the ones
+                # package in conda-forge.
+                root_flags,
                 stdcxx, '-DLLVM_ENABLE_TERMINFO=0',
                 '-Dminimal=ON', '-Dasimage=OFF', '-Droot7=OFF', '-Dhttp=OFF',
                 '-Dbuiltin_pcre=ON', '-Dbuiltin_freetype=ON', '-Dbuiltin_zlib=ON', '-Dbuiltin_xxhash=ON']

