diff --git a/cling/setup.py b/cling/setup.py
index d7c6595..6308ef2 100755
--- a/setup.py
+++ b/setup.py
@@ -14,6 +14,7 @@ except ImportError:
     has_wheel = False
 from distutils.errors import DistutilsSetupError
 
+import shlex
 
 requirements = []
 setup_requirements = ['wheel']+requirements
@@ -137,6 +138,12 @@ class my_cmake_build(_build):
                 stdcxx, '-DLLVM_ENABLE_TERMINFO=0',
                 '-Dminimal=ON', '-Dasimage=OFF', '-Droot7=OFF', '-Dhttp=OFF',
                 '-Dbuiltin_pcre=ON', '-Dbuiltin_freetype=ON', '-Dbuiltin_zlib=ON', '-Dbuiltin_xxhash=ON']
+
+        # Flags that root-feedstock sets when invoking cmake. Most
+        # importantly, do not build llvm & clang but use the ones
+        # package in conda-forge.
+        CMAKE_COMMAND.extend(shlex.split(os.getenv('CMAKE_ROOT_FLAGS')))
+
         if 'darwin' in sys.platform:
             CMAKE_COMMAND.append('-Dlibcxx=ON')
         CMAKE_COMMAND.append('-DCMAKE_BUILD_TYPE='+get_build_type())
