diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4732a45..3118999 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -339,11 +339,11 @@ if (YAML_CPP_INSTALL)
 		DESTINATION "${INSTALL_CMAKE_DIR}" COMPONENT dev)
 	install(EXPORT yaml-cpp-targets DESTINATION ${INSTALL_CMAKE_DIR})

-	if(UNIX)
+	# if(UNIX)
 		set(PC_FILE ${CMAKE_BINARY_DIR}/yaml-cpp.pc)
 		configure_file("yaml-cpp.pc.cmake" ${PC_FILE} @ONLY)
 		install(FILES ${PC_FILE} DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
-	endif()
+	# endif()

 endif()
