From 3707dc6e4e4b005fd1d783ef2c398bed0d67fa1d Mon Sep 17 00:00:00 2001
From: Christopher Burr <christopher.burr@cern.ch>
Date: Sat, 2 Mar 2019 20:39:52 +0000
Subject: [PATCH] Prevent astropy_helpers trying to auto-update

---
 ah_bootstrap.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ah_bootstrap.py b/ah_bootstrap.py
index 2cea5bd..4b0fcfd 100644
--- a/ah_bootstrap.py
+++ b/ah_bootstrap.py
@@ -146,7 +146,7 @@ else:
 DOWNLOAD_IF_NEEDED = True
 INDEX_URL = 'https://pypi.python.org/simple'
 USE_GIT = True
-OFFLINE = False
+OFFLINE = True
 AUTO_UPGRADE = True
 
 # A list of all the configuration options and their required types
-- 
2.20.0

