*** pyswmm/lib/__init__.py	2020-06-13 22:54:21.237925940 -0500
--- pyswmm/lib/__init__.py2	2020-06-13 22:54:21.237925940 -0500
***************
*** 30,37 ****
  
  
  # Library paths
  if os.name == 'nt':
!     if sys.maxsize > 2**32:
          LIB_SWMM = os.path.join(
              HERE, _platform(), 'swmm5-x64.dll').replace('\\', '/')
      else:
--- 30,42 ----
  
  
  # Library paths
+ # First search in prefix then default to packaged binaries
+ lib_path = os.path.join(sys.prefix, 'lib')
+ has_swmm_lib = any([l for l in os.listdir(lib_path) if 'libswmm5' in l])
  if os.name == 'nt':
!     if any([l for l in os.listdir(sys.prefix) if 'libswmm5' in l]):
!         LIB_SWMM = os.path.join(sys.prefix, 'libswmm5.dll').replace('\\', '/')
!     elif sys.maxsize > 2**32:
          LIB_SWMM = os.path.join(
              HERE, _platform(), 'swmm5-x64.dll').replace('\\', '/')
      else:
***************
*** 42,55 ****
              '\\',
              '/')
  elif sys.platform == 'darwin':
!     LIB_SWMM = os.path.join(
!         HERE,
!         _platform(),
!         'swmm5.dylib').replace(
!         '\\',
!         '/')
  elif sys.platform.startswith('linux'):
!     LIB_SWMM = os.path.join(HERE, _platform(), 'swmm5.so').replace('\\', '/')
  else:
      LIB_SWMM = ''
  
--- 47,66 ----
              '\\',
              '/')
  elif sys.platform == 'darwin':
!     if has_swmm_lib:
!         LIB_SWMM = os.path.join(lib_path, 'libswmm5.dylib').replace('\\', '/')
!     else:
!         LIB_SWMM = os.path.join(
!             HERE,
!             _platform(),
!             'swmm5.dylib').replace(
!             '\\',
!             '/')
  elif sys.platform.startswith('linux'):
!     if has_swmm_lib:
!         LIB_SWMM = os.path.join(lib_path, 'libswmm5.so').replace('\\', '/')
!     else:
!         LIB_SWMM = os.path.join(HERE, _platform(), 'swmm5.so').replace('\\', '/')
  else:
      LIB_SWMM = ''
  
