--- setup.py	2019-05-13 20:20:34.383626601 -0400
+++ setup.py.new	2019-05-13 20:53:35.771371301 -0400
@@ -38,7 +38,7 @@
 setup(
     name='owlrl',
     packages=['owlrl'],
-    scripts=['scripts/owlrl.py', 'scripts/RDFConvertService.py'],
+    scripts=['scripts/RDFConvertService.py'],
     package_dir={'owlrl': './owlrl'},
     version=version,
     description='OWL-RL and RDFS based RDF Closure inferencing',
@@ -75,4 +75,3 @@
     install_requires=requirements,
     dependency_links=dependency_links
 )
-
