From 7c2f0b813c6eb47ef7fdc7244d7b650b2cee1a91 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 29 Mar 2020 23:18:34 +0200
Subject: [PATCH] comment out requires in setup.py

---
 setup.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/setup.py b/setup.py
index 9f50f5e..95d0b31 100644
--- a/setup.py
+++ b/setup.py
@@ -27,11 +27,11 @@ setup(name='gym',
       packages=[package for package in find_packages()
                 if package.startswith('gym')],
       zip_safe=False,
-      install_requires=[
-          'scipy', 'numpy>=1.10.4', 'six', 'pyglet>=1.4.0,<=1.5.0', 'cloudpickle>=1.2.0,<1.4.0',
-          'enum34~=1.1.6;python_version<"3.4"',
-      ],
-      extras_require=extras,
+      # install_requires=[
+      #     'scipy', 'numpy>=1.10.4', 'six', 'pyglet>=1.4.0,<=1.5.0', 'cloudpickle>=1.2.0,<1.4.0',
+      #     'enum34~=1.1.6;python_version<"3.4"',
+      # ],
+      # extras_require=extras,
       package_data={'gym': [
         'envs/mujoco/assets/*.xml',
         'envs/classic_control/assets/*.png',
@@ -42,7 +42,7 @@ setup(name='gym',
         'envs/robotics/assets/stls/hand/*.stl',
         'envs/robotics/assets/textures/*.png']
       },
-      tests_require=['pytest', 'mock'],
+      # tests_require=['pytest', 'mock'],
       python_requires='>=3.5',
       classifiers=[
           'Programming Language :: Python :: 3',
-- 
2.24.0.windows.2

