--- rtree/core.py	2014-11-01 00:57:35.000000000 -0300
+++ rtree/core.py	2015-09-24 14:06:24.679228624 -0300
@@ -102,8 +102,11 @@
 
 elif os.name == 'posix':
     platform = os.uname()[0]
-    lib_name = find_library('spatialindex_c')
-    rt = ctypes.CDLL(lib_name)
+    if platform == 'Linux':
+        lib_name = 'libspatialindex_c.so'
+    else:
+        lib_name = 'libspatialindex_c.dylib'
+    rt = ctypes.CDLL(os.path.join(sys.prefix, 'lib', lib_name))
 else:
     raise RTreeError('Unsupported OS "%s"' % os.name)
 
