diff -Naur Rtree-0.8.2.orig/rtree/core.py Rtree-0.8.2/rtree/core.py
--- rtree/core.py	2014-11-01 00:57:35.000000000 -0300
+++ rtree/core.py	2015-09-24 14:44:59.628343915 -0300
@@ -65,41 +65,8 @@
     
 
 if os.name == 'nt':
-
-    def _load_library(dllname, loadfunction, dllpaths=('', )):
-        """Load a DLL via ctypes load function. Return None on failure.
-
-        Try loading the DLL from the current package directory first,
-        then from the Windows DLL search path.
-
-        """
-        try:
-            dllpaths = (os.path.abspath(os.path.dirname(__file__)),
-                        ) + dllpaths
-        except NameError:
-            pass # no __file__ attribute on PyPy and some frozen distributions
-        for path in dllpaths:
-            if path:
-                # temporarily add the path to the PATH environment variable
-                # so Windows can find additional DLL dependencies.
-                try:
-                    oldenv = os.environ['PATH']
-                    os.environ['PATH'] = path + ';' + oldenv
-                except KeyError:
-                    oldenv = None
-            try:
-                return loadfunction(os.path.join(path, dllname))
-            except (WindowsError, OSError):
-                pass
-            finally:
-                if path and oldenv is not None:
-                    os.environ['PATH'] = oldenv
-        return None
-
-    rt = _load_library('spatialindex_c.dll', ctypes.cdll.LoadLibrary)
-    if not rt:
-        raise OSError("could not find or load spatialindex_c.dll")
-
+    os.environ['PATH'] = "{};{}".format(os.environ['PATH'], os.path.join(sys.prefix, "Library", "bin"))
+    rt = ctypes.CDLL("spatialindex_c-64.dll")
 elif os.name == 'posix':
     platform = os.uname()[0]
     lib_name = find_library('spatialindex_c')
diff -Naur Rtree-0.8.2.orig/setup.py Rtree-0.8.2/setup.py
--- setup.py	2014-07-13 02:19:37.000000000 -0300
+++ setup.py	2015-09-24 14:50:25.352360137 -0300
@@ -7,14 +7,7 @@
 readme_text = open('docs/source/README.txt', 'r').read()
 
 import os
-
-if os.name == 'nt':
-    data_files=[('Lib/site-packages/rtree',
-                 [r'D:\libspatialindex\bin\spatialindex.dll',
-                  r'D:\libspatialindex\bin\spatialindex_c.dll',]),]
-else:
-    data_files = None
-    
+  
 setup(name          = 'Rtree',
       version       = rtree.__version__,
       description   = 'R-Tree spatial index for Python GIS',
@@ -29,7 +22,6 @@
       packages      = ['rtree'],
       install_requires = ['setuptools'],
       test_suite = 'tests.test_suite',
-      data_files = data_files,
       zip_safe = False,
       classifiers   = [
         'Development Status :: 5 - Production/Stable',
