diff --git a/MANIFEST.in b/MANIFEST.in
index affa175..e69de29 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1 +0,0 @@
-include tabula/*.jar
diff --git a/tabula/wrapper.py b/tabula/wrapper.py
index 2b17bfc..0d72d0a 100644
--- a/tabula/wrapper.py
+++ b/tabula/wrapper.py
@@ -11,6 +11,7 @@ Todo:
 import io
 import json
 import os
+import platform
 import shlex
 import subprocess
 import requests
@@ -18,9 +19,11 @@ import pandas as pd
 import numpy as np
 from .util import deprecated_option
 
-JAR_NAME = "tabula-1.0.1-jar-with-dependencies.jar"
-JAR_DIR = os.path.abspath(os.path.dirname(__file__))
-JAR_PATH = os.path.join(JAR_DIR, JAR_NAME)
+JAR_NAME = "tabula.jar"
+JAR_DIR = os.environ['CONDA_PREFIX']
+if platform.system() == "Windows":
+    JAR_DIR = os.path.join(JAR_DIR, 'Library')
+JAR_PATH = os.path.join(JAR_DIR, 'lib', JAR_NAME)
 
 JAVA_NOT_FOUND_ERROR = "`java` command is not found from this Python process. Please ensure Java is installed and PATH is set for `java`"
 
