From 5f6ab11cbb199e5349874fd3014ff74c783a150a Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Mon, 17 Oct 2016 12:00:34 +0200
Subject: [PATCH] compatibility patch for Xcode 6

---
 dolfin/mesh/MeshEntity.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dolfin/mesh/MeshEntity.h b/dolfin/mesh/MeshEntity.h
index f0c7078..2545c1f 100644
--- a/dolfin/mesh/MeshEntity.h
+++ b/dolfin/mesh/MeshEntity.h
@@ -222,8 +222,10 @@ namespace dolfin
       const std::map<std::int32_t, std::set<unsigned int>>& sharing_map
         = _mesh->topology().shared_entities(_dim);
       const auto map_it = sharing_map.find(_local_index);
-      if (map_it == sharing_map.end())
-        return {};
+      if (map_it == sharing_map.end()) {
+          const std::set<unsigned int> x = std::set<unsigned int> {};
+          return x;
+      }
       else
         return map_it->second;
     }
-- 
2.10.1

