From 84472d7a376c4614b1564b515aa6120da5b32abc Mon Sep 17 00:00:00 2001
From: Jan Blechta <blechta@karlin.mff.cuni.cz>
Date: Wed, 20 Jul 2016 16:03:10 +0200
Subject: [PATCH] Eliminate Point.__[i]truediv__ workaround in new SWIG

---
 dolfin/swig/mesh/post.i | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/dolfin/swig/mesh/post.i b/dolfin/swig/mesh/post.i
index 6f9a975..007caca 100644
--- a/dolfin/swig/mesh/post.i
+++ b/dolfin/swig/mesh/post.i
@@ -34,8 +34,13 @@
 //-----------------------------------------------------------------------------
 %extend dolfin::Point {
 %pythoncode %{
-__truediv__ = __div__
-__itruediv__ = __idiv__
+try:
+    # Workaround for SWIG < 3.0.9
+    __truediv__ = __div__
+    __itruediv__ = __idiv__
+except NameError:
+    # SWIG >= 3.0.9
+    pass
 %}
 }
 
-- 
2.10.1

