diff --git a/src/RingDecomposerLib/CMakeLists.txt b/src/RingDecomposerLib/CMakeLists.txt
index f032c12..3e111ac 100644
--- a/src/RingDecomposerLib/CMakeLists.txt
+++ b/src/RingDecomposerLib/CMakeLists.txt
@@ -18,10 +18,12 @@ function(build_library LIBRARY_NAME SPECIFIER)
   if(NOT WIN32)
     target_link_libraries("${LIBRARY_NAME}" m)
   endif()
+  if("${SPECIFIER}" STREQUAL "SHARED")
   install(TARGETS ${LIBRARY_NAME}
           ARCHIVE DESTINATION lib
           LIBRARY DESTINATION lib
           RUNTIME DESTINATION bin)
+  endif()
 endfunction()
 
 build_library("${SHARED_LIBRARY}" "SHARED")
