--- a/build/Makefile.Linux
+++ b/build/Makefile.Linux
@@ -7,11 +7,11 @@

 #### Compilers: ####
 ifdef clang
-CXX           = clang++
-CC            = clang
+CXX           ?= clang++
+CC            ?= clang
 else
-CXX           = g++
-CC            = gcc
+CXX           ?= g++
+CC            ?= gcc
 endif

 #### Compiler optimization flags: ####
@@ -37,11 +37,11 @@
 endif

 ifdef clang
-LD            = clang++
+LD            = $(CXX)
 LDFLAGS       += $(LDFLAGS_RPATH) $(LDFLAGS_EXTRA) -O
 SOFLAGS       = -shared -Wl,-soname,
 else
-LD            = g++
+LD            = $(CXX)
 LDFLAGS       += $(LDFLAGS_RPATH) -Wl,--no-as-needed $(LDFLAGS_EXTRA) -O
 SOFLAGS       = -shared -Wl,-soname,
 endif
