From 798dcc57e84d1dda9de9a042cf877eb14930dae3 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 24 Jul 2022 00:45:15 -0600
Subject: [PATCH] Update setup.py

---
 setup.py | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index aeefc01..72ccc72 100644
--- a/setup.py
+++ b/setup.py
@@ -11,20 +11,20 @@
 
 install_requires = [
     'arrow',
-    'base58==0.2.2',
-    'click==6.6',
-    'docker-py==1.8.0',
+    'base58>=0.2.2',
+    'click>=6.6',
     'flake8',
-    'jsonrpcclient==2.0.1',
-    'jsonrpcserver==3.1.1',
+    'jsonrpcclient>=2.0.1',
+    'jsonrpcserver>=3.1.1',
     'mnemonic==0.13',
-    'path.py',
     'pexpect',
-    'protobuf==3.0.0a3',
+    'protobuf>=3.0.0a3',
     'pyaes',
     'pytest',
     'pyyaml',
-    'requests<=2.11.1',
+    'requests>=2.11.1',
     'sha256',
     'tabulate',
 ]
