From b93b8f11179f7118a8616564419737cba7db5532 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 24 Jul 2022 08:11:13 -0600
Subject: [PATCH] Update version.py

---
 two1/commands/util/version.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/two1/commands/util/version.py b/two1/commands/util/version.py
index 49b6e81..1d18081 100644
--- a/two1/commands/util/version.py
+++ b/two1/commands/util/version.py
@@ -2,7 +2,7 @@
 import requests
 import urllib.parse as parse
 from pkg_resources import parse_version
-from pkg_resources import SetuptoolsVersion
+from packaging.version import Version
 from distutils.version import LooseVersion
 
 import two1
@@ -36,7 +36,7 @@ def is_version_gte(actual, expected):
         ValueError: if expected ot actual version is not in Major.Minor.Patch
             format.
     """
-    if isinstance(parse_version(actual), SetuptoolsVersion):
+    if isinstance(parse_version(actual), Version):
         # This handles versions that end in things like `rc0`
         return parse_version(actual) >= parse_version(expected)
     else:
