From c2f65ae803db192b8fb946e08ca9de96bad956cd Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Fri, 30 Oct 2020 17:32:34 +0100
Subject: [PATCH] Soft missing header fix boost serialization

---
 src/serialization.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git src/serialization.cc src/serialization.cc
index 9b509f0..368e81f 100644
--- src/serialization.cc
+++ src/serialization.cc
@@ -17,6 +17,9 @@
 // <http://www.gnu.org/licenses/>.
 
 #include <boost/serialization/serialization.hpp>
+#if __has_include(<boost/serialization/library_version_type.hpp>)
+  #include <boost/serialization/library_version_type.hpp>
+#endif
 #include <boost/serialization/set.hpp>
 #include <boost/serialization/list.hpp>
 #include <boost/serialization/weak_ptr.hpp>
-- 
2.17.1

