From 509491ebb71768d851542a73740d6a98c490bad2 Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Mon, 19 Oct 2020 17:18:46 +0200
Subject: [PATCH] Add RT test fix

---
 tests/CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git tests/CMakeLists.txt tests/CMakeLists.txt
index 35e7c1c..593b3ad 100644
--- tests/CMakeLists.txt
+++ tests/CMakeLists.txt
@@ -34,6 +34,8 @@ ADD_DEFINITIONS(-DTEST_DIRECTORY="${CMAKE_SOURCE_DIR}/tests/data")
 # This macro will create a binary from `NAME.cc', link it against
 # Boost and add it to the test suite.
 #
+find_library(RT_LIB rt)
+
 MACRO(ADD_TESTCASE NAME GENERATED)
   IF (${GENERATED} STREQUAL TRUE)
     ADD_UNIT_TEST(${NAME} ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.cc)
@@ -41,7 +43,11 @@ MACRO(ADD_TESTCASE NAME GENERATED)
     ADD_UNIT_TEST(${NAME} ${NAME}.cc)
   ENDIF()
   TARGET_INCLUDE_DIRECTORIES(${NAME} PRIVATE ../src)
-  TARGET_LINK_LIBRARIES(${NAME} Boost::unit_test_framework ${PROJECT_NAME}-gpl)
+  if(RT_LIB)
+    TARGET_LINK_LIBRARIES(${NAME} Boost::unit_test_framework ${PROJECT_NAME}-gpl ${RT_LIB})
+  else()
+    TARGET_LINK_LIBRARIES(${NAME} Boost::unit_test_framework ${PROJECT_NAME}-gpl)
+  endif()
 ENDMACRO(ADD_TESTCASE)
 
 FIND_PACKAGE(OpenMP)
-- 
2.17.1

