From 9f713e328583722c79cb9458f6a6d93e05b9d7b1 Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Mon, 19 Oct 2020 17:07:13 +0200
Subject: [PATCH] Patch missing include in boost/serialization/set.hpp of
 version 1.74

---
 src/serialization.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git src/serialization.cc src/serialization.cc
index 9b509f0..9015418 100644
--- src/serialization.cc
+++ src/serialization.cc
@@ -17,6 +17,7 @@
 // <http://www.gnu.org/licenses/>.
 
 #include <boost/serialization/serialization.hpp>
+#include <boost/serialization/library_version_type.hpp>
 #include <boost/serialization/set.hpp>
 #include <boost/serialization/list.hpp>
 #include <boost/serialization/weak_ptr.hpp>
-- 
2.17.1

