--- v4-client-cpp/lib/CMakeLists.txt.orig	2021-10-14 15:00:00.000000000 +0000
+++ v4-client-cpp/lib/CMakeLists.txt	2021-10-14 15:00:00.000000000 +0000
@@ -1,7 +1,9 @@
 # Dependencies
-find_package(Boost COMPONENTS beast REQUIRED)
-find_package(OpenSSL REQUIRED)
-find_package(bip3x REQUIRED)
+cmake_policy(SET CMP0167 NEW)
+find_package(Boost REQUIRED)
+find_package(OpenSSL 3 REQUIRED)
+find_package(toolbox REQUIRED)
+find_package(bip3x 3 REQUIRED)
 find_package(fmt REQUIRED)
-find_package(json REQUIRED)
+find_package(nlohmann_json 3.11.3 REQUIRED)

