--- setup.py.bak	2022-03-23 21:42:07.212278572 -0500
+++ setup.py	2022-03-23 21:46:08.367799027 -0500
@@ -90,21 +90,15 @@
     library_dirs = []
     extra_compile_args = ['-std=c++11', '-Wno-format-security']
     extra_link_args = None
-    if platform.system() == 'Darwin':
-        libraries = None
-        extra_link_args = ['-Wl,-framework,OpenCL']
-    elif platform.system() == "Windows":
+    if platform.system() == "Windows":
         # Add include & lib dirs if possible from usual nvidia and AMD paths
-        for path in ["CUDA_HOME", "CUDAHOME", "CUDA_PATH"]:
-            if path in os.environ:
-                include_dirs.append(pjoin(os.environ[path], 'include'))
-                library_dirs.append(pjoin(os.environ[path], 'lib', 'x64'))
+        include_dirs.append(os.environ["LIBRARY_INC"])
+        library_dirs.append(os.environ["LIBRARY_LIB"])
         libraries = ['OpenCL']
         extra_compile_args = None
     else:
-        # Linux
+        # Linux and OSX
         libraries = ['OpenCL']
-        extra_link_args = ['--shared']
 
     opencl_config = {'libraries': libraries, 'extra_link_args': extra_link_args,
                      'include_dirs': include_dirs, 'library_dirs': library_dirs,
