--- bin/nvcc.profile	2023-03-23 15:56:49.535384866 -0400
+++ bin/nvcc.profile.new	2023-03-23 19:21:08.071151138 -0400
@@ -5,9 +5,9 @@
 
 PATH            += $(TOP)/$(_NVVM_BRANCH_)/bin;$(_HERE_);$(TOP)/lib;
 
-INCLUDES        +=  "-I$(TOP)/include" $(_SPACE_)
+INCLUDES        +=  "-I$(TOP)/include" $(_SPACE_) "-I$(TOP)/include/targets/$(_WIN_PLATFORM_)" $(_SPACE_)
 
-LIBRARIES        =+ $(_SPACE_) "/LIBPATH:$(TOP)/lib/$(_WIN_PLATFORM_)"
+LIBRARIES        =+ $(_SPACE_) "/LIBPATH:$(TOP)/lib/$(_WIN_PLATFORM_)" $(_SPACE_) "/LIBPATH:$(TOP)/lib"
 
 CUDAFE_FLAGS    +=
 PTXAS_FLAGS     +=
