--- bin/nvcc.profile        2021-07-06 11:22:41.714913066 -0700
+++ bin/nvcc.profile.new    2021-07-06 11:22:33.467108854 -0700
@@ -6,9 +6,9 @@
 LD_LIBRARY_PATH += $(TOP)/lib:
 PATH            += $(TOP)/$(_NVVM_BRANCH_)/bin:$(_HERE_):

-INCLUDES        +=  "-I$(TOP)/$(_TARGET_DIR_)/include" $(_SPACE_)
+INCLUDES        +=  "-I$(TOP)/include" $(_SPACE_)

-LIBRARIES        =+ $(_SPACE_) "-L$(TOP)/$(_TARGET_DIR_)/lib$(_TARGET_SIZE_)/stubs" "-L$(TOP)/$(_TARGET_DIR_)/lib$(_TARGET_SIZE_)"
+LIBRARIES        =+ $(_SPACE_) "-L$(TOP)/lib/stubs" "-L$(TOP)/lib"

 CUDAFE_FLAGS    +=
 PTXAS_FLAGS     +=
