From 30c62ac8073b26f1724c075f6a4dbda1b6c0078f Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 2 May 2023 17:03:00 -0600
Subject: [PATCH] remove wheel and versioneer from requirements

---
 requirements.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/requirements.txt b/requirements.txt
index f0404e8..bfeb553 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,3 @@
-wheel
-versioneer
 torch >= 1.9.0
 scipy
 numpy
