--- linux.cf.orig	2017-02-06 15:13:43.029920163 -0500
+++ linux.cf	2017-02-06 15:15:13.206667177 -0500
@@ -578,11 +578,7 @@
 #define MkdirHierCmd		mkdir -p
 
 #ifndef HaveLib64
-# if defined (AMD64Architecture) || defined (s390xArchitecture) || defined (Ppc64Architecture) || defined (AArch64Architecture)
-#  define HaveLib64	YES
-# else
-#  define HaveLib64	NO
-# endif
+# define HaveLib64	NO
 #endif
 
 #if UseElfFormat
@@ -1072,7 +1068,7 @@
 
 #if HaveLib64
 # ifndef LibDirName
-#  define LibDirName		lib64
+#  define LibDirName		lib
 # endif
 # ifndef SystemUsrLibDir
 #  define SystemUsrLibDir	/usr/lib64
