--- src/libscotch/common.c	(revision 22563)
+++ src/libscotch/common.c	(working copy)
@@ -104,7 +104,14 @@
 #else /* COMMON_TIMING_OLD */
   struct timespec     tp;

-  clock_gettime (CLOCK_REALTIME, &tp);            /* Elapsed time */
+#if defined(_POSIX_TIMERS) && _POSIX_TIMERS > 0
+    clock_gettime(CLOCK_REALTIME, &tp);
+#else
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    tp.tv_sec = tv.tv_sec;
+    tp.tv_nsec = tv.tv_usec * 1000;
+#endif

   return ((double) tp.tv_sec + (double) tp.tv_nsec * 1.0e-9L);
 #endif /* COMMON_TIMING_OLD */
