diff --git a/src/layer/CMakeLists.txt b/src/layer/CMakeLists.txt
index 5afff80..cd89139 100644
--- a/src/layer/CMakeLists.txt
+++ b/src/layer/CMakeLists.txt
@@ -5,7 +5,8 @@
 # SPDX-License-Identifier: Apache-2.0
 set(CMAKE_FOLDER "${CMAKE_FOLDER}/VulkanLayerSettings")
 
-add_library(VulkanLayerSettings STATIC)
+add_library(VulkanLayerSettings)
+set_target_properties(VulkanLayerSettings PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 add_library(Vulkan::LayerSettings ALIAS VulkanLayerSettings)
 
 target_sources(VulkanLayerSettings PRIVATE
diff --git a/src/vulkan/CMakeLists.txt b/src/vulkan/CMakeLists.txt
index 04e7f0a..a5a268c 100644
--- a/src/vulkan/CMakeLists.txt
+++ b/src/vulkan/CMakeLists.txt
@@ -5,7 +5,8 @@
 # SPDX-License-Identifier: Apache-2.0
 set(CMAKE_FOLDER "${CMAKE_FOLDER}/VulkanSafeStruct")
 
-add_library(VulkanSafeStruct STATIC)
+add_library(VulkanSafeStruct)
+set_target_properties(VulkanSafeStruct PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 add_library(Vulkan::SafeStruct ALIAS VulkanSafeStruct)
 
 target_sources(VulkanSafeStruct PRIVATE
