Index: torchfsm/operator/_base.py
===================================================================
--- torchfsm/operator/_base.py
+++ torchfsm/operator/_base.py
@@ -960,9 +960,9 @@
     def __call__(
         self,
         u_fft: FourierTensor["B C H ..."],
         f_mesh: FourierMesh,
-        u: Optional[SpatialTensor["B C H ..."], None],
+        u: Optional[SpatialTensor["B C H ..."]] = None,
     ) -> FourierTensor["B C H ..."]:
         if self.source.device != f_mesh.device:
             self.source = self.source.to(f_mesh.device)
         return self.source
