diff --git bob/learn/em/linear_scoring.cpp bob/learn/em/linear_scoring.cpp
index c475ff0..e5b592d 100644
--- bob/learn/em/linear_scoring.cpp
+++ bob/learn/em/linear_scoring.cpp
@@ -69,7 +69,7 @@ static inline bool f(PyObject* o){return o != 0 && PyObject_IsTrue(o) > 0;}
 
 
 /*** linear_scoring ***/
-bob::extension::FunctionDoc linear_scoring1 = bob::extension::FunctionDoc(
+static auto linear_scoring1 = bob::extension::FunctionDoc(
   "linear_scoring",
   "",
   0,
@@ -84,7 +84,7 @@ bob::extension::FunctionDoc linear_scoring1 = bob::extension::FunctionDoc(
 .add_return("output","array_like<float,1>","Score");
 
 
-bob::extension::FunctionDoc linear_scoring2 = bob::extension::FunctionDoc(
+static auto linear_scoring2 = bob::extension::FunctionDoc(
   "linear_scoring",
   "",
   0,
@@ -101,7 +101,7 @@ bob::extension::FunctionDoc linear_scoring2 = bob::extension::FunctionDoc(
 
 
 
-bob::extension::FunctionDoc linear_scoring3 = bob::extension::FunctionDoc(
+static auto linear_scoring3 = bob::extension::FunctionDoc(
   "linear_scoring",
   "",
   0,
diff --git a/bob/learn/em/main.cpp b/bob/learn/em/main.cpp
index dc413e8..49e4c3a 100644
--- bob/learn/em/main.cpp
+++ bob/learn/em/main.cpp
@@ -10,6 +10,10 @@
 #endif
 #include "main.h"
 
+// since their variables are static and need to be used here.
+#include "ztnorm.cpp"
+#include "linear_scoring.cpp"
+
 static PyMethodDef module_methods[] = {
   {
     zt_norm.name(),
diff --git a/bob/learn/em/main.h b/bob/learn/em/main.h
index 1d61e87..f22cae1 100644
--- bob/learn/em/main.h
+++ bob/learn/em/main.h
@@ -265,21 +265,21 @@ bool init_BobLearnEMEMPCATrainer(PyObject* module);
 int PyBobLearnEMEMPCATrainer_Check(PyObject* o);
 
 
-//ZT Normalization
-PyObject* PyBobLearnEM_ztNorm(PyObject*, PyObject* args, PyObject* kwargs);
-extern bob::extension::FunctionDoc zt_norm;
+// //ZT Normalization
+// PyObject* PyBobLearnEM_ztNorm(PyObject*, PyObject* args, PyObject* kwargs);
+// extern bob::extension::FunctionDoc zt_norm;
 
-PyObject* PyBobLearnEM_tNorm(PyObject*, PyObject* args, PyObject* kwargs);
-extern bob::extension::FunctionDoc t_norm;
+// PyObject* PyBobLearnEM_tNorm(PyObject*, PyObject* args, PyObject* kwargs);
+// extern bob::extension::FunctionDoc t_norm;
 
-PyObject* PyBobLearnEM_zNorm(PyObject*, PyObject* args, PyObject* kwargs);
-extern bob::extension::FunctionDoc z_norm;
+// PyObject* PyBobLearnEM_zNorm(PyObject*, PyObject* args, PyObject* kwargs);
+// extern bob::extension::FunctionDoc z_norm;
 
 
-//Linear scoring
-PyObject* PyBobLearnEM_linear_scoring(PyObject*, PyObject* args, PyObject* kwargs);
-extern bob::extension::FunctionDoc linear_scoring1;
-extern bob::extension::FunctionDoc linear_scoring2;
-extern bob::extension::FunctionDoc linear_scoring3;
+// //Linear scoring
+// PyObject* PyBobLearnEM_linear_scoring(PyObject*, PyObject* args, PyObject* kwargs);
+// extern bob::extension::FunctionDoc linear_scoring1;
+// extern bob::extension::FunctionDoc linear_scoring2;
+// extern bob::extension::FunctionDoc linear_scoring3;
 
 #endif // BOB_LEARN_EM_MAIN_H
diff --git a/bob/learn/em/ztnorm.cpp b/bob/learn/em/ztnorm.cpp
index adc8a6f..3fce1ab 100644
--- bob/learn/em/ztnorm.cpp
+++ bob/learn/em/ztnorm.cpp
@@ -10,7 +10,7 @@
 #include "main.h"
 
 /*** zt_norm ***/
-bob::extension::FunctionDoc zt_norm = bob::extension::FunctionDoc(
+static auto zt_norm = bob::extension::FunctionDoc(
   "ztnorm",
   "Normalise raw scores with ZT-Norm."
   "Assume that znorm and tnorm have no common subject id.",
@@ -70,7 +70,7 @@ PyObject* PyBobLearnEM_ztNorm(PyObject*, PyObject* args, PyObject* kwargs) {
 
 
 /*** t_norm ***/
-bob::extension::FunctionDoc t_norm = bob::extension::FunctionDoc(
+static auto t_norm = bob::extension::FunctionDoc(
   "tnorm",
   "Normalise raw scores with T-Norm",
   0,
@@ -107,7 +107,7 @@ PyObject* PyBobLearnEM_tNorm(PyObject*, PyObject* args, PyObject* kwargs) {
 
 
 /*** z_norm ***/
-bob::extension::FunctionDoc z_norm = bob::extension::FunctionDoc(
+static auto z_norm = bob::extension::FunctionDoc(
   "znorm",
   "Normalise raw scores with Z-Norm",
   0,
diff --git a/setup.py b/setup.py
index 3f56083..5a0573c 100644
--- setup.py
+++ setup.py
@@ -114,9 +114,9 @@ setup(
 
           "bob/learn/em/plda_trainer.cpp",
 
-          "bob/learn/em/ztnorm.cpp",
-
-          "bob/learn/em/linear_scoring.cpp",
+          # include the module methods directly in main.cpp
+          # "bob/learn/em/ztnorm.cpp",
+          # "bob/learn/em/linear_scoring.cpp",
 
           "bob/learn/em/main.cpp",
         ],
