diff --git a/setup.py b/setup.py
index 1fe1da5..0687d56 100644
--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@ setup(
     description='a generalist algorithm for anatomical segmentation',
     long_description=long_description,
     long_description_content_type='text/markdown',
-    packages=find_packages(),
+    packages=find_packages(exclude=("tests", "tests.*")),
     python_requires='>=3.6',
     use_scm_version=True,
     install_requires=install_deps,
