--- a/ESMP_LoadESMF.py  2014-01-14 10:00:22.000000000 -0500
+++ b/ESMP_LoadESMF.py  2014-01-14 10:40:57.000000000 -0500
@@ -64,6 +64,13 @@
 #      esmfmk = c[2]

   try:
+
+    # If we are not dealing with an absolute path treat it a relative to the
+    # current Python module.
+    if not os.path.isabs(esmfmk):
+      # Get the directory for this module
+      esmfmk = os.path.abspath(os.path.join(sys.prefix,'lib', esmfmk))
+
     MKFILE = open(esmfmk, 'r')
   except:
     raise IOError("File not found\n  %s") % esmfmk
@@ -72,11 +79,12 @@
   libsdir = 0
   esmfos = 0
   esmfabi = 0
+
+  libsdir = os.path.dirname(esmfmk)
+
 #  MKFILE = open(esmfmk,'r')
   for line in MKFILE:
-    if 'ESMF_LIBSDIR' in line:
-      libsdir = line.split("=")[1]
-    elif 'ESMF_OS:' in line:
+    if 'ESMF_OS:' in line:
       esmfos = line.split(":")[1]
     elif 'ESMF_ABI:' in line:
       esmfabi = line.split(":")[1]
