From 72c8e9acd1a28c501b2bcb2306b8f242b5b73fb6 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <xhochy@users.noreply.github.com>
Date: Sun, 6 Jan 2019 23:20:40 +0100
Subject: [PATCH] Remove forced static linkage

---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d3fc4f1..c689967 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,10 +93,6 @@ endif ()
 include (cmake/find_ltdl.cmake) # for odbc
 include (cmake/find_odbc.cmake)
 
-if (NOT WIN32)
-    set (OPENSSL_LIBRARIES "-Wl,-Bdynamic ${OPENSSL_LIBRARIES} -Wl,-Bstatic")
-endif ()
-
 include (cmake/find_poco.cmake)
 include (cmake/find_nanoodbc.cmake)
 include (cmake/find_ccache.cmake)
