From cac38224ae9ac657c80f56410e701b1509d03333 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <xhochy@users.noreply.github.com>
Date: Sun, 6 Jan 2019 23:06:57 +0100
Subject: [PATCH] Only set OPENSSL_ROOT_DIR if not defined

---
 cmake/find_ssl.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmake/find_ssl.cmake b/cmake/find_ssl.cmake
index 08c0c5b..9431ae7 100644
--- a/cmake/find_ssl.cmake
+++ b/cmake/find_ssl.cmake
@@ -13,7 +13,9 @@ endif ()
 
 if (NOT USE_INTERNAL_SSL_LIBRARY)
     if (APPLE)
-        set (OPENSSL_ROOT_DIR "/usr/local/opt/openssl")
+        if (NOT DEFINED OPENSSL_ROOT_DIR)
+            set (OPENSSL_ROOT_DIR "/usr/local/opt/openssl")
+         endif ()
         # https://rt.openssl.org/Ticket/Display.html?user=guest&pass=guest&id=2232
         if (USE_STATIC_LIBRARIES)
             message(WARNING "Disable USE_STATIC_LIBRARIES if you have linking problems with OpenSSL on MacOS")
