Index: src/metapensiero/pj/transformations/obvious.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/metapensiero/pj/transformations/obvious.py	(revision e8a9727d87f0b83af669481c51a773fe98d73efc)
+++ src/metapensiero/pj/transformations/obvious.py	(revision 456030c0ff1b7037223aa84926baed04f1fa5462)
@@ -278,6 +278,9 @@
     return cls()
 
 
+Constant = NameConstant
+
+
 def Yield(t, x):
     return JSYield(x.value)
 
