--- pydip/CMakeLists.txt	2022-07-22 22:38:01.000000000 +0200
+++ pydip/CMakeLists.txt.mod	2022-07-22 22:37:43.000000000 +0200
@@ -1,5 +1,6 @@
 set(PYBIND11_PYTHON_VERSION ${PYTHON_VERSION}) # Avoid a warning message
-add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/pybind11" "${PROJECT_BINARY_DIR}/pybind11" EXCLUDE_FROM_ALL)
+# add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/pybind11" "${PROJECT_BINARY_DIR}/pybind11" EXCLUDE_FROM_ALL)
+find_package(pybind11 REQUIRED)
 
 # Find sources
 file(GLOB DIP_PYTHON_SRC "${CMAKE_CURRENT_LIST_DIR}/src/*.cpp" "${CMAKE_CURRENT_LIST_DIR}/src/*.h")
