--- src/CMakeLists.txt	2022-07-22 22:23:44.000000000 +0200
+++ src/CMakeLists.txt	2022-07-23 00:55:30.000000000 +0200
@@ -95,7 +95,8 @@
 set(HAS_128_INT ${HAS_128_INT} PARENT_SCOPE)
 
 # Eigen
-target_include_directories(DIP PRIVATE "${PROJECT_SOURCE_DIR}/dependencies/eigen3")
+find_package(Eigen3 REQUIRED)
+target_include_directories(DIP PRIVATE ${EIGEN3_INCLUDE_DIRS})
 target_compile_definitions(DIP PRIVATE
                            EIGEN_MPL2_ONLY # This makes sure we only use parts of the Eigen library that use the MPL2 license or more permissive ones.
                            EIGEN_DONT_PARALLELIZE) # This to prevent Eigen algorithms trying to run in parallel -- we parallelize at a larger scale.
@@ -103,13 +104,15 @@
 # zlib (for use in libics and libtiff)
 set(DIP_ENABLE_ZLIB ON CACHE BOOL "Enable zlib compression in ICS and TIFF (deflate)")
 if(DIP_ENABLE_ZLIB)
-   add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/zlib" "${PROJECT_BINARY_DIR}/zlib" EXCLUDE_FROM_ALL)
+   # add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/zlib" "${PROJECT_BINARY_DIR}/zlib" EXCLUDE_FROM_ALL)
+   find_package(ZLIB REQUIRED)
 endif()
 
 # libjpeg (for use in libtiff)
 set(DIP_ENABLE_JPEG ON CACHE BOOL "Enable JPEG file support and compression in TIFF")
 if(DIP_ENABLE_JPEG)
-   add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/libjpeg" "${PROJECT_BINARY_DIR}/libjpeg" EXCLUDE_FROM_ALL)
+   # add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/libjpeg" "${PROJECT_BINARY_DIR}/libjpeg" EXCLUDE_FROM_ALL)
+   find_package(JPEG REQUIRED)
    target_link_libraries(DIP PRIVATE jpeg)
    target_compile_definitions(DIP PRIVATE DIP_CONFIG_HAS_JPEG)
 endif()
@@ -118,7 +121,8 @@
 set(DIP_ENABLE_ICS ON CACHE BOOL "Enable ICS file support")
 if(DIP_ENABLE_ICS)
    set(LIBICS_INCLUDE_CPP Off) # TODO: we should start using the C++ interface
-   add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/libics" "${PROJECT_BINARY_DIR}/libics" EXCLUDE_FROM_ALL)
+   # add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/libics" "${PROJECT_BINARY_DIR}/libics" EXCLUDE_FROM_ALL)
+   find_package(libics REQUIRED)
    target_link_libraries(DIP PRIVATE libics)
    target_compile_definitions(DIP PRIVATE DIP_CONFIG_HAS_ICS)
 endif()
@@ -126,7 +130,8 @@
 # libtiff
 set(DIP_ENABLE_TIFF ON CACHE BOOL "Enable TIFF file support")
 if(DIP_ENABLE_TIFF)
-   add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/libtiff" "${PROJECT_BINARY_DIR}/libtiff" EXCLUDE_FROM_ALL)
+   # add_subdirectory("${PROJECT_SOURCE_DIR}/dependencies/libtiff" "${PROJECT_BINARY_DIR}/libtiff" EXCLUDE_FROM_ALL)
+   find_package(TIFF REQUIRED)
    target_link_libraries(DIP PRIVATE tiff)
    target_compile_definitions(DIP PRIVATE DIP_CONFIG_HAS_TIFF)
 endif()
