diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 21e3598..d4a49bf 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -39,7 +39,7 @@ endif(GIT_FOUND)
 
 # global defines
 list (APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/)
-list (APPEND CMAKE_CXX_FLAGS "-std=c++11 -Wno-ignored-attributes")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wno-ignored-attributes")
 
 # define the abi
 if (NOT DEFINED TF_GOOGLE_BIN) 
