diff --git a/pyproject.toml b/pyproject.toml
index 7401026..d38e72f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -38,9 +38,7 @@ Source = "https://github.com/frankie567/hatch-regex-commit"
 regex-commit = "hatch_regex_commit.hooks"
 
 [tool.hatch.version]
-source = "regex_commit"
 path = "hatch_regex_commit/__about__.py"
-commit_extra_args = ["-e"]
 
 [tool.hatch.envs.default]
 dependencies = [
