--- maya-0.1.5.orig/setup.py	2016-12-25 21:47:20.000000000 -0300
+++ maya-0.1.5/setup.py	2017-01-17 14:38:59.848981695 -0300
@@ -37,7 +37,7 @@
     name='maya',
     version='0.1.5',
     description='Datetimes for Humans.',
-    long_description= '\n' + read('README.rst'),
+    long_description='',
     author='Kenneth Reitz',
     author_email='me@kennethreitz.com',
     url='https://github.com/kennethreitz/maya',
