diff --git a/CMakeLists.txt b/CMakeLists.txt
index 494afbdf..2ffdc210 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,9 +51,17 @@ tvm_option(USE_ANTLR "Build with ANTLR for Relay parsing" OFF)
 
 # include directories
 include_directories("include")
-include_directories("3rdparty/dlpack/include")
-include_directories("3rdparty/dmlc-core/include")
-include_directories("3rdparty/rang/include")
+include_directories("${CMAKE_SOURCE_DIR}/include")
+include_directories("${CMAKE_SOURCE_DIR}/include/tvm")
+include_directories("${CMAKE_PREFIX_PATH}/include")
+include_directories("${CMAKE_PREFIX_PATH}/include/dlpack")
+include_directories("${CMAKE_PREFIX_PATH}/include/dmlc")
+include_directories("${CMAKE_PREFIX_PATH}/include/ir")
+include_directories("${CMAKE_PREFIX_PATH}/include/tvm")
+include_directories("${CMAKE_PREFIX_PATH}/include/tvm/node")
+include_directories("${CMAKE_PREFIX_PATH}/include/arithmetic")
+include_directories("${CMAKE_PREFIX_PATH}/include/base")
+include_directories("${CMAKE_PREFIX_PATH}/include/compiler-rt")
 include_directories("3rdparty/compiler-rt")
 
 # initial variables
@@ -210,7 +218,7 @@ target_link_libraries(nnvm_compiler tvm)
 # Related headers
 target_include_directories(
   tvm
-  PUBLIC "3rdparty/HalideIR/src"
+  PUBLIC "${CMAKE_PREFIX_PATH}/include"
   PUBLIC "topi/include")
 target_include_directories(
   tvm_topi
