diff --git a/cmake/util/FindLLVM.cmake b/cmake/util/FindLLVM.cmake
index 8497761a..515879db 100644
--- a/cmake/util/FindLLVM.cmake
+++ b/cmake/util/FindLLVM.cmake
@@ -40,9 +40,9 @@ macro(find_llvm use_llvm)
       COMMAND cut -b 1,3
       OUTPUT_VARIABLE TVM_LLVM_VERSION)
     # definitions
-    string(REGEX MATCHALL "(^| )-D[A-Za-z0-9_]*" LLVM_DEFINITIONS ${__llvm_cxxflags})
+    string(REGEX MATCHALL "(^| )-D[A-Za-z0-9_]*" LLVM_DEFINITIONS "${__llvm_cxxflags}")
     # include dir
-    string(REGEX MATCHALL "(^| )-I[^ ]*" __llvm_include_flags ${__llvm_cxxflags})    
+    string(REGEX MATCHALL "(^| )-I[^ ]*" __llvm_include_flags "${__llvm_cxxflags}")    
     set(LLVM_INCLUDE_DIRS "")
     foreach(__flag IN ITEMS ${__llvm_include_flags})
       string(REGEX REPLACE "(^| )-I" "" __dir "${__flag}")
@@ -50,11 +50,11 @@ macro(find_llvm use_llvm)
     endforeach()
     message(STATUS ${LLVM_INCLUDE_DIRS})
     # libfiles
-    string(STRIP ${__llvm_libfiles} __llvm_libfiles)
-    string(STRIP ${__llvm_system_libs} __llvm_system_libs)
+    string(STRIP "${__llvm_libfiles}" __llvm_libfiles)
+    string(STRIP "${__llvm_system_libs}" __llvm_system_libs)
     set(LLVM_LIBS "${__llvm_libfiles} ${__llvm_system_libs}")
     separate_arguments(LLVM_LIBS)
-    string(STRIP ${TVM_LLVM_VERSION} TVM_LLVM_VERSION)
+    string(STRIP "${TVM_LLVM_VERSION}" TVM_LLVM_VERSION)
   endif()
   if(NOT LLVM_CONFIG STREQUAL "OFF")
     message(STATUS "Found LLVM_INCLUDE_DIRS=" ${LLVM_INCLUDE_DIRS})
