diff --git a/CMakeLists.txt b/CMakeLists.txt
index 494afbdf..37194244 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,11 +50,10 @@ tvm_option(USE_RANDOM "Build with random support" OFF)
 tvm_option(USE_ANTLR "Build with ANTLR for Relay parsing" OFF)
 
 # include directories
-include_directories("include")
-include_directories("3rdparty/dlpack/include")
-include_directories("3rdparty/dmlc-core/include")
-include_directories("3rdparty/rang/include")
-include_directories("3rdparty/compiler-rt")
+include_directories("${CMAKE_PREFIX_PATH}/include/dlpack")
+include_directories("${CMAKE_PREFIX_PATH}/include/dmlc")
+include_directories("${CMAKE_PREFIX_PATH}/include")
+include_directories("${CMAKE_PREFIX_PATH}/include/compiler-rt")
 
 # initial variables
 set(TVM_LINKER_LIBS "")
@@ -210,7 +209,7 @@ target_link_libraries(nnvm_compiler tvm)
 # Related headers
 target_include_directories(
   tvm
-  PUBLIC "3rdparty/HalideIR/src"
+  PUBLIC "${CMAKE_PREFIX_PATH}/include"
   PUBLIC "topi/include")
 target_include_directories(
   tvm_topi
