diff --git a/nengo_loihi/hardware/interface.py b/nengo_loihi/hardware/interface.py
index 9816a6396..c86f65c4f 100644
--- a/nengo_loihi/hardware/interface.py
+++ b/nengo_loihi/hardware/interface.py
@@ -62,7 +62,7 @@ class HardwareInterface:
 
     connection_retries = 3
     min_nxsdk_version = parse_nxsdk_version("0.9.0")
-    max_nxsdk_version = parse_nxsdk_version("0.9.9")
+    max_nxsdk_version = parse_nxsdk_version("1.0.0")
 
     def __init__(
         self,
diff --git a/nengo_loihi/hardware/nxsdk_shim.py b/nengo_loihi/hardware/nxsdk_shim.py
index df37bf657..f1bce7f81 100644
--- a/nengo_loihi/hardware/nxsdk_shim.py
+++ b/nengo_loihi/hardware/nxsdk_shim.py
@@ -129,7 +129,13 @@ def createSnip(self, phase, *args, **kwargs):
             b"bnhzZGsuY29tcGlsZXIudHJhY2VjZmdnZW4udHJhY2VjZmdnZW4=", b"VHJhY2VDZmdHZW4="
         )
 
-    NxsdkBoard = d_import(b"bnhzZGsuZ3JhcGgubnhib2FyZA==", b"TjJCb2FyZA==")
+    try:
+        # try new location (nxsdk >= 1.0.0)
+        NxsdkBoard = d_import(b"bnhzZGsuYXJjaC5uMmEubjJib2FyZA==", b"TjJCb2FyZA==")
+    except ImportError:  # pragma: no cover
+        # try old location (nxsdk < 1.0.0)
+        NxsdkBoard = d_import(b"bnhzZGsuZ3JhcGgubnhib2FyZA==", b"TjJCb2FyZA==")
+
     SpikeGen = d_import(
         b"bnhzZGsuZ3JhcGgubnhpbnB1dGdlbi5ueGlucHV0Z2Vu", b"QmFzaWNTcGlrZUdlbmVyYXRvcg=="
     )
