From 6dd361818eee70e770f3f098eb61533741de68b9 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 5 Sep 2021 11:35:43 -0600
Subject: [PATCH] Update setup.py

---
 setup.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/setup.py b/setup.py
index 8a31a08..11d459b 100644
--- a/setup.py
+++ b/setup.py
@@ -2,14 +2,12 @@
 
 url = ""
 version = "0.1.0"
-readme = open('README.rst').read()
 
 setup(
     name="dtool_utils",
     packages=["dtool_utils"],
     version=version,
     description="Dtool utility functions and classes",
-    long_description=readme,
     include_package_data=True,
     author="Matthew Hartley",
     author_email="matthew.hartley@jic.ac.uk",
