diff --git i/SuperBuild/External_CTK.cmake w/SuperBuild/External_CTK.cmake
index fb41007b04..f2e297de27 100644
--- i/SuperBuild/External_CTK.cmake
+++ w/SuperBuild/External_CTK.cmake
@@ -82,6 +82,7 @@ if(NOT DEFINED CTK_DIR AND NOT Slicer_USE_SYSTEM_${proj})
     ${${proj}_EP_ARGS}
     GIT_REPOSITORY "${Slicer_${proj}_GIT_REPOSITORY}"
     GIT_TAG "${Slicer_${proj}_GIT_TAG}"
+    PATCH_COMMAND patch -p1 < $ENV{RECIPE_DIR}/ctk-pythonqt-threads.patch
     SOURCE_DIR ${EP_SOURCE_DIR}
     BINARY_DIR ${EP_BINARY_DIR}
     CMAKE_CACHE_ARGS
diff --git i/SuperBuild/External_SimpleITK.cmake w/SuperBuild/External_SimpleITK.cmake
index 694a241408..8fb53b6413 100644
--- i/SuperBuild/External_SimpleITK.cmake
+++ w/SuperBuild/External_SimpleITK.cmake
@@ -173,9 +173,6 @@ ExternalProject_Execute(${proj} \"install\" \"${PYTHON_EXECUTABLE}\" \"-m\" \"pi
       -DCMAKE_CXX_FLAGS:STRING=${ep_common_cxx_flags}
       -DCMAKE_C_COMPILER:FILEPATH=${CMAKE_C_COMPILER}
       -DCMAKE_C_FLAGS:STRING=${ep_common_c_flags}
-      -DCMAKE_CXX_STANDARD:STRING=${CMAKE_CXX_STANDARD}
-      -DCMAKE_CXX_STANDARD_REQUIRED:BOOL=${CMAKE_CXX_STANDARD_REQUIRED}
-      -DCMAKE_CXX_EXTENSIONS:BOOL=${CMAKE_CXX_EXTENSIONS}
       -DCMAKE_CXX_VISIBILITY_PRESET:BOOL=default
       -DBUILD_SHARED_LIBS:BOOL=${Slicer_USE_SimpleITK_SHARED}
       -DBUILD_EXAMPLES:BOOL=OFF
