--- CTK/CMakeExternals/PythonQt.cmake	2022-07-13 18:42:03.304128760 -0500
+++ CTK/CMakeExternals/PythonQt.cmake.new	2022-07-13 18:41:59.961133207 -0500
@@ -109,6 +109,7 @@
     PREFIX ${proj}${ep_suffix}
     ${location_args}
     BUILD_COMMAND ""
+    PATCH_COMMAND patch -p1 < $ENV{RECIPE_DIR}/pythonqt-threads.patch
     CMAKE_CACHE_ARGS
       ${ep_common_cache_args}
       -DPythonQt_QT_VERSION:STRING=${CTK_QT_VERSION}
