diff --git a/corelib/CMakeLists.txt b/corelib/CMakeLists.txt
index ac8b268f..b2e9b8eb 100644
--- a/corelib/CMakeLists.txt
+++ b/corelib/CMakeLists.txt
@@ -552,7 +552,7 @@ if ( ${SIRE_CXX_COMPILER_ID} MATCHES "GNU" )
       else()
         # OpenMM with conda uses the old C++ binary API!
         # Tell GCC 5 to respect the old API
-        set( SIRE_PLATFORM_FLAGS "${SIRE_PLATFORM_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0" )
+        set( SIRE_PLATFORM_FLAGS "${SIRE_PLATFORM_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=1" )
       endif()
     endif()
 
diff --git a/corelib/src/apps/test_system/CMakeLists.txt b/corelib/src/apps/test_system/CMakeLists.txt
index 0be975b0..1f29552b 100644
--- a/corelib/src/apps/test_system/CMakeLists.txt
+++ b/corelib/src/apps/test_system/CMakeLists.txt
@@ -24,7 +24,7 @@ endif()
 if (SIRE_FOUND_CPUID)
   add_definitions( "-DSIRE_FOUND_CPUID" )
   add_definitions( "${CPUID_DEFINITIONS}" )
-  include_directories( ${CPUID_INCLUDE_DIR} )
+  unset(CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES)
 
   add_executable( get_cpuid get_cpuid.c )
 
diff --git a/wrapper/CMakeLists.txt b/wrapper/CMakeLists.txt
index 73f887a4..74f7e86c 100644
--- a/wrapper/CMakeLists.txt
+++ b/wrapper/CMakeLists.txt
@@ -259,7 +259,7 @@ if (DEFINED SIRE_APP)
   # For some reason, when using the conda GNU compilers,  CMake stripped
   # "${SIRE_APP}/include" from the directories searched for header files
   # If this happens, uncomment the following line
-  # unset(CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES)
+  unset(CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES)
   include_directories( "${SIRE_APP}/include" "${SIRE_APP}/Library/include" )
 endif()
 
