diff --git a/qwtbuild.pri b/qwtbuild.pri
index c1ad8a8..30bf966 100644
--- a/qwtbuild.pri
+++ b/qwtbuild.pri
@@ -27,8 +27,8 @@ win32 {
     # might need a debug version. 
     # Enable debug_and_release + build_all if you want to build both.
 
-    CONFIG           += debug_and_release
-    CONFIG           += build_all
+    CONFIG           += release
+#    CONFIG           += build_all
 }
 else {
 
