--- setup.py	2021-01-25 19:42:19.000000000 -0500
+++ setup.py	2022-12-16 15:27:12.923936600 -0500
@@ -27,8 +27,8 @@
     # long_description=open("./README.rst").read(),
     # long_description_content_type="text/x-rst",
 
-    long_description=open("./README.md").read(),
-    long_description_content_type="text/markdown",
+    # long_description=open("./README.md").read(),
+    # long_description_content_type="text/markdown",
 
     license = __license__,
     url = __uri__,
