diff --git a/main.c b/main.c
index f0631c4..bee4848 100644
--- a/main.c
+++ b/main.c
@@ -146,6 +146,11 @@ open_by_handle_at (int mount_fd, struct file_handle *handle, int flags)
 # define FICLONE _IOW (0x94, 9, int)
 #endif
 
+/* NB: renameat2 syscall was introduced in Linux 3.15 and got backported to Centos 7. */
+#if !defined(SYS_renameat2) && defined(__linux__) && defined(__x86_64__)
+# define SYS_renameat2 316
+#endif
+
 #if defined(__GNUC__) && (__GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ >= 6) && !defined __cplusplus
 _Static_assert (sizeof (fuse_ino_t) >= sizeof (uintptr_t),
                "fuse_ino_t too small to hold uintptr_t values!");
diff --git a/utils.h b/utils.h
index 99e911a..58ef5e2 100644
--- a/utils.h
+++ b/utils.h
@@ -56,6 +56,11 @@ int open_fd_or_get_path (struct ovl_layer *l, const char *path, char *out, int *
 void statx_to_stat (struct statx *stx, struct stat *st);
 # endif
 
+/* NB: O_PATH was introduced in Linux 2.6.39 and is not available on CentOS 6. */
+#if !defined(O_PATH) && defined(__linux__)
+# define O_PATH 010000000
+#endif
+
 int safe_openat (int dirfd, const char *pathname, int flags, mode_t mode);
 
 int override_mode (struct ovl_layer *l, int fd, const char *abs_path, const char *path, struct stat *st);
