From 6fc06a9b42e73df6fa73b6cf28cb77b37921ae34 Mon Sep 17 00:00:00 2001
From: Joshua Lynch <jlynch@bnl.gov>
Date: Thu, 22 Jul 2021 17:28:29 -0400
Subject: [PATCH] import bluesky_kafka only when needed

---
 nslsii/__init__.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/nslsii/__init__.py b/nslsii/__init__.py
index 54bfe21..999871e 100644
--- a/nslsii/__init__.py
+++ b/nslsii/__init__.py
@@ -13,7 +13,6 @@ import msgpack_numpy as mpn
 
 from IPython import get_ipython
 
-from bluesky_kafka import Publisher
 from event_model import RunRouter
 
 from ._version import get_versions
@@ -578,6 +577,8 @@ def subscribe_kafka_publisher(RE, beamline_name, bootstrap_servers, producer_con
         by this function
 
     """
+    from bluesky_kafka import Publisher
+
     topic = f"{beamline_name.lower()}.bluesky.runengine.documents"
 
     def kafka_publisher_factory(name, start_doc):
-- 
2.16.1

