--- ase/visualize/nglview.py	2017-12-17 17:22:19.871994011 +0100
+++ ase/visualize/nglview.py	2017-12-17 17:40:12.669979238 +0100
@@ -33,7 +33,9 @@
                                args=['%dpx' % (xsize,), '%dpx' % (ysize,)])
         self.view.add_unitcell()
         self.view.add_spacefill()
+        self.view.remove_ball_and_stick()
         self.view.camera = 'orthographic'
+        self.view.parameters = { "clipDist": 0 }
         self.view.update_spacefill(radiusType='covalent', scale=0.7)
         self.view.center()
