diff --git a/src/EGL/meson.build b/src/EGL/meson.build
index c90f0ca..3cf618a 100644
--- a/src/EGL/meson.build
+++ b/src/EGL/meson.build
@@ -39,9 +39,12 @@ if host_machine.system() in ['haiku']
   egl_vendor_config_dirs += ':/boot/system/non-packaged/add-ons/opengl/egl_vendor.d'
   egl_vendor_config_dirs += ':/boot/system/add-ons/opengl/egl_vendor.d'
 else
+  # conda-forge specific patch: in conda-forge we do not ship any EGL implementation
+  # as a conda-forge package, so we patch the DEFAULT_EGL_VENDOR_CONFIG_DIRS to point
+  # to the system one, where we assume that glvnd is installed with /usr prefix
   egl_vendor_config_dirs = '@0@/glvnd/egl_vendor.d:@1@/glvnd/egl_vendor.d'.format(
-    join_paths(get_option('prefix'),get_option('sysconfdir')),
-    join_paths(get_option('prefix'),get_option('datadir')))
+    '/etc',
+    join_paths('/usr',get_option('datadir')))
 endif
 
 libEGL = shared_library(
