diff --git src/tools/msvc/Solution.pm src/tools/msvc/Solution.pm
index 24dcd99..bfa5cf4 100644
--- src/tools/msvc/Solution.pm
+++ src/tools/msvc/Solution.pm
@@ -491,15 +491,15 @@ sub AddProject
 	if ($self->{options}->{zlib})
 	{
 		$proj->AddIncludeDir($self->{options}->{zlib} . '\include');
-		$proj->AddLibrary($self->{options}->{zlib} . '\lib\zdll.lib');
+		$proj->AddLibrary($self->{options}->{zlib} . '\lib\zlib.lib');
 	}
 	if ($self->{options}->{openssl})
 	{
 		$proj->AddIncludeDir($self->{options}->{openssl} . '\include');
 		$proj->AddLibrary(
-			$self->{options}->{openssl} . '\lib\VC\ssleay32.lib', 1);
+			$self->{options}->{openssl} . '\lib\ssleay32.lib');
 		$proj->AddLibrary(
-			$self->{options}->{openssl} . '\lib\VC\libeay32.lib', 1);
+			$self->{options}->{openssl} . '\lib\libeay32.lib');
 	}
 	if ($self->{options}->{nls})
 	{
