--- tests/slow/Makefile.am	2017-05-27 07:31:00.000000000 +0200
+++ tests/slow/Makefile.am	2017-09-12 10:06:10.520007748 +0200
@@ -56,7 +56,7 @@
 check_PROGRAMS = $(ctests) cipher-test cipher-override mac-override cipher-override2 hash-large
 TESTS = $(ctests) test-ciphers.sh override-ciphers test-hash-large.sh
 
-if HAVE_LIBCRYPTO
+if ENABLE_OPENSSL_TRUE
 cipher_openssl_compat_LDFLAGS = $(LDADD) $(LIBCRYPTO)
 
 dist_check_SCRIPTS += test-ciphers-openssl.sh
