--- tests/pkgconfig.sh	2017-09-03 17:23:50.943352157 +0200
+++ tests/pkgconfig.sh	2017-09-03 17:23:31.247394776 +0200
@@ -31,7 +31,7 @@
 
 ${PKGCONFIG} --version >/dev/null || exit 77
 
-PKG_CONFIG_PATH=${top_builddir}/lib
+PKG_CONFIG_PATH=${PREFIX}/lib/pkgconfig/
 export PKG_CONFIG_PATH
 
 set -e
@@ -45,16 +45,16 @@
 }
 __EOF__
 
-COMMON="-I${PKG_CONFIG_PATH}/includes -L${PKG_CONFIG_PATH}/.libs -I${srcdir}/../lib/includes"
+COMMON="-L${PREFIX}/lib/ -I${PREFIX}/include/"
 echo "Trying dynamic linking with:"
 echo "  * flags: $(${PKGCONFIG} --libs gnutls)"
 echo "  * common: ${COMMON}"
 echo "  * lib: ${CFLAGS}"
-cc ${TMPFILE} -o ${TMPFILE_O} $(${PKGCONFIG} --libs gnutls) $(${PKGCONFIG} --cflags gnutls) ${COMMON}
+gcc ${TMPFILE} -o ${TMPFILE_O} $(${PKGCONFIG} --libs gnutls) $(${PKGCONFIG} --cflags gnutls) ${COMMON} -std=gnu99
 
 echo ""
 echo "Trying static linking with $(${PKGCONFIG} --libs --static gnutls)"
-cc ${TMPFILE} -o ${TMPFILE_O} $(${PKGCONFIG} --static --libs gnutls) $(${PKGCONFIG} --cflags gnutls) ${COMMON}
+gcc ${TMPFILE} -o ${TMPFILE_O} $(${PKGCONFIG} --static --libs gnutls) $(${PKGCONFIG} --cflags gnutls) ${COMMON} -std=gnu99
 
 rm -f ${TMPFILE} ${TMPFILE_O}
 
