--- a/daemon/runner.py
+++ b/daemon/runner.py
@@ -120,8 +120,7 @@ class DaemonRunner:
         self.daemon_context = DaemonContext()
         self.daemon_context.stdin = open(app.stdin_path, 'rt')
         self.daemon_context.stdout = open(app.stdout_path, 'w+t')
-        self.daemon_context.stderr = open(
-                app.stderr_path, 'w+t', buffering=0)
+        self.daemon_context.stderr = open(app.stderr_path, 'w+t')
 
         self.pidfile = None
         if app.pidfile_path is not None:
