diff --git a/lib/kernel/vecmathlib/floattypes.h b/lib/kernel/vecmathlib/floattypes.h
index fa4cc44..cb9071b 100644
--- a/lib/kernel/vecmathlib/floattypes.h
+++ b/lib/kernel/vecmathlib/floattypes.h
@@ -40,6 +40,7 @@ using namespace std;
 #endif
 
 // Capture libc macros, then undefine them
+/*
 #ifndef isfinite
 #error "isfinite is not a macro"
 #endif
@@ -55,6 +56,7 @@ using namespace std;
 #ifndef signbit
 #error "signbit is not a macro"
 #endif
+*/
 
 namespace {
 template <typename T> inline int libc_isfinite(T x) { return isfinite(x); }
