diff --git a/lib/CL/pocl_llvm_api.cc b/lib/CL/pocl_llvm_api.cc
index ac4ec4ba..8a56526c 100644
--- a/lib/CL/pocl_llvm_api.cc
+++ b/lib/CL/pocl_llvm_api.cc
@@ -853,6 +864,10 @@ int pocl_llvm_get_kernel_metadata(cl_program program,

 char* get_cpu_name() {
   StringRef r = llvm::sys::getHostCPUName();
+  if (r.str() == "skylake")
+  {
+    r = llvm::StringRef("haswell");
+  }
   assert(r.size() > 0);
   char* cpu_name = (char*) malloc (r.size()+1);
   strncpy(cpu_name, r.data(), r.size());
