From 088d66bce3f090ee6883c91983225dd8e62a5d02 Mon Sep 17 00:00:00 2001
From: Levi Naden <lnaden@vt.edu>
Date: Tue, 19 Feb 2019 10:09:39 -0500
Subject: [PATCH] Set setup.py to zip_safe False for Windows testing

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index bd05bcc..7299221 100644
--- a/setup.py
+++ b/setup.py
@@ -61,5 +61,5 @@ if __name__ == "__main__":
             'Intended Audience :: Science/Research',
             'Programming Language :: Python :: 3',
         ],
-        zip_safe=True,
+        zip_safe=False,
     )
-- 
2.19.1

