diff -ruN simple-binary-encoding-1.33.1.orig/build.gradle simple-binary-encoding-1.33.1/build.gradle
--- simple-binary-encoding-1.33.1.orig/build.gradle	2024-09-23 13:41:03.000000000 +0200
+++ simple-binary-encoding-1.33.1/build.gradle	2024-12-03 16:21:16.304745211 +0100
@@ -21,6 +21,7 @@
 plugins {
     id 'java-library'
     id 'jvm-test-suite'
+    id 'com.github.jk1.dependency-license-report' version '2.9'
     alias(libs.plugins.shadow).apply(false)
     alias(libs.plugins.versions)
 }
