diff --git a/pybind/CMakeLists.txt b/pybind/CMakeLists.txt
index 923a6da..7864925 100644
--- a/pybind/CMakeLists.txt
+++ b/pybind/CMakeLists.txt
@@ -9,8 +9,4 @@ pybind11_add_module(pyposelib MODULE pyposelib.cc)
 target_link_libraries(pyposelib PUBLIC PoseLib::PoseLib Eigen3::Eigen)
 target_compile_features(pyposelib PRIVATE cxx_std_17)
 set_target_properties(pyposelib PROPERTIES OUTPUT_NAME poselib)
-if (MARCH_NATIVE)
-    target_compile_options(pyposelib PRIVATE -march=native -Wall -Werror -fPIC -Wno-ignored-optimization-argument)
-else()
-    target_compile_options(pyposelib PRIVATE -Wall -Werror -fPIC)
-endif()
+target_compile_options(pyposelib PRIVATE -fPIC)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 68e8e53..0c9e907 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,13 +26,7 @@ if(MSVC)
 	target_compile_options(${LIBRARY_NAME} PRIVATE /bigobj /fp:fast)
 else()
 	# If you change this, make sure to update the corresponding line in the pybind CMakeLists
-	if (MARCH_NATIVE)
-		target_compile_options(${LIBRARY_NAME} PRIVATE
-			-march=native -Wall -Werror -fPIC -Wno-ignored-optimization-argument)
-	else()
-		target_compile_options(${LIBRARY_NAME} PRIVATE
-			-Wall -Werror -fPIC)
-	endif()
+	target_compile_options(${LIBRARY_NAME} PRIVATE -fPIC)
 	if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 		target_compile_options(${LIBRARY_NAME} PRIVATE
 				-Wno-maybe-uninitialized)
