--- tests/Makefile.in.old	2016-09-04 18:50:28.407571370 -0500
+++ tests/Makefile.in	2016-09-04 18:51:16.703573674 -0500
@@ -152,8 +152,8 @@
 CONFIG_CLEAN_FILES = wrapper.sh
 CONFIG_CLEAN_VPATH_FILES =
 @HWLOC_HAVE_LINUX_LIBNUMA_TRUE@am__EXEEXT_1 = linux-libnuma$(EXEEXT)
-@HWLOC_HAVE_SCHED_SETAFFINITY_TRUE@am__EXEEXT_2 =  \
-@HWLOC_HAVE_SCHED_SETAFFINITY_TRUE@	glibc-sched$(EXEEXT)
+# @HWLOC_HAVE_SCHED_SETAFFINITY_TRUE@am__EXEEXT_2 =  \
+# @HWLOC_HAVE_SCHED_SETAFFINITY_TRUE@	glibc-sched$(EXEEXT)
 @HWLOC_HAVE_LIBIBVERBS_TRUE@am__EXEEXT_3 = openfabrics-verbs$(EXEEXT)
 @HWLOC_HAVE_MYRIEXPRESS_TRUE@am__EXEEXT_4 = myriexpress$(EXEEXT)
 @HWLOC_HAVE_OPENCL_TRUE@am__EXEEXT_5 = opencl$(EXEEXT)
--- tests/linux/Makefile.in.old	2016-08-28 15:29:28.000000000 -0500
+++ tests/linux/Makefile.in	2016-09-04 19:09:37.043626167 -0500
@@ -596,7 +596,6 @@
 # By default, it shows the output of `lstopo - -v'.
 sysfs_outputs = \
 		2amd64-2n.output \
-		2i386-2t-hugepagesizecount.xml.output \
 		2i386-2c-nohugepage.xml.output \
 		2ps3-2t.output \
 		4em64t-2c2t-ignore-reorder.output \
@@ -655,8 +654,6 @@
 # or a corresponding `xyz.source' specifying which tarball to use
 sysfs_tarballs = \
 		2amd64-2n.tar.bz2 \
-		2i386-2t-hugepagesizecount.tar.bz2 \
-		2i386-2t-hugepagesizecount.xml.source \
 		2i386-2c-nohugepage.tar.bz2 \
 		2i386-2c-nohugepage.xml.source \
 		2ps3-2t.tar.bz2 \
@@ -721,7 +718,6 @@
 # Each output `xyz.output' may have a corresponding `xyz.options'
 # file modifying the behavior of lstopo
 sysfs_options = \
-		2i386-2t-hugepagesizecount.xml.options \
 		2i386-2c-nohugepage.xml.options \
 		4em64t-2c2t-ignore-reorder.options \
 		16amd64-4n4c-cgroup-distance-merge.options \
@@ -742,7 +738,6 @@
 # Each output `xyz.output' may have a corresponding `xyz.env'
 # modifying the environment of lstopo
 sysfs_envs = \
-		2i386-2t-hugepagesizecount.xml.env \
 		16amd64-8n2c-cpusets.xml.env \
 		32amd64-4s2n4c-cgroup.env \
 		32amd64-4s2n4c-cgroup.xml.env \
@@ -1042,13 +1037,6 @@
 	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-2i386-2t-hugepagesizecount.xml.output.log: 2i386-2t-hugepagesizecount.xml.output
-	@p='2i386-2t-hugepagesizecount.xml.output'; \
-	b='2i386-2t-hugepagesizecount.xml.output'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
 2i386-2c-nohugepage.xml.output.log: 2i386-2c-nohugepage.xml.output
 	@p='2i386-2c-nohugepage.xml.output'; \

