--- setup.py
+++ setup.py
@@ -1,7 +1,10 @@
 #!/usr/bin/python
 
+import os
 from distutils.core import setup
 
+PREFIX = os.environ['PREFIX']
+
 setup(name         = 'go2',
       version      = '2.20170703',
       description  = 'go2 directory finder',
@@ -9,7 +12,7 @@
       author_email = 'David.Villa@gmail.com',
       url          = 'http://savannah.nongnu.org/projects/go2/',
       license      = 'GPL v2 or later',
-      data_files   = [('/usr/lib/go2', ['go2.sh', 'go2.py', 'osfs.py']),
-                      ('/usr/share/man/man1', ['go2.1']),
-                      ('/usr/bin', ['go2'])],
+      data_files   = [(os.path.join(PREFIX, 'lib/go2'), ['go2.sh', 'go2.py', 'osfs.py']),
+                      (os.path.join(PREFIX, 'share/man/man1'), ['go2.1']),
+                      (os.path.join(PREFIX, 'bin'), ['go2'])],
       )
