--- setup.py
+++ setup.py
@@ -9,7 +9,7 @@
       author_email = 'David.Villa@gmail.com',
       url          = 'http://savannah.nongnu.org/projects/go2/',
       license      = 'GPL v2 or later',
-      data_files   = [('/usr/lib/go2', ['go2.sh', 'go2.py', 'osfs.py']),
-                      ('/usr/share/man/man1', ['go2.1']),
-                      ('/usr/bin', ['go2'])],
+      data_files   = [('lib/go2', ['go2.sh', 'go2.py', 'osfs.py']),
+                      ('share/man/man1', ['go2.1']),
+                      ('bin', ['go2'])],
       )
