--- setup.py	2019-05-04 16:21:33.815260676 -0400
+++ setup.py	2019-05-04 16:23:52.613052860 -0400
@@ -81,7 +81,7 @@

     extras = {
         'web': ['regex', 'flask'],
-        'plot': ['graphviz', 'regex', 'flask', 'Pygments', 'lxml', 'bs4',
+        'plot': ['graphviz', 'regex', 'flask', 'Pygments', 'lxml', 'beautifulsoup4',
                  'jinja2', 'docutils'],
         'parallel': ['multiprocess'],
     }
