From bcf6040f2318c0d30003fd5df085909dabf8b332 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sun, 11 Feb 2018 19:45:43 +0000
Subject: [PATCH] Disable tests which require additional deps

---
 Makefile.in | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 2c107f5..fa1a953 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -226,10 +226,7 @@ libsecret_testable_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC \
 am__EXEEXT_2 = test-hex$(EXEEXT) test-secmem$(EXEEXT) $(am__EXEEXT_1)
 am__EXEEXT_3 =
 am__EXEEXT_4 = test-attributes$(EXEEXT) test-value$(EXEEXT) \
-	test-prompt$(EXEEXT) test-service$(EXEEXT) \
-	test-session$(EXEEXT) test-paths$(EXEEXT) \
-	test-methods$(EXEEXT) test-password$(EXEEXT) \
-	test-item$(EXEEXT) test-collection$(EXEEXT) $(am__EXEEXT_3)
+	$(am__EXEEXT_3)
 @HAVE_VALAC_TRUE@am__EXEEXT_5 = test-vala-lang$(EXEEXT) \
 @HAVE_VALAC_TRUE@	test-vala-unstable$(EXEEXT) $(am__EXEEXT_3)
 @ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@@HAVE_VALAC_TRUE@am__EXEEXT_6 = $(am__EXEEXT_5)
@@ -557,10 +554,6 @@ am__set_TESTS_bases = \
   bases=`for i in $$bases; do echo $$i; done | sed 's/\.log$$//'`; \
   bases=`echo $$bases`
 RECHECK_LOGS = $(TEST_LOGS)
-am__EXEEXT_7 = libsecret/test-py-lookup.py libsecret/test-py-clear.py \
-	libsecret/test-py-store.py $(am__EXEEXT_3)
-am__EXEEXT_8 = libsecret/test-js-lookup.js libsecret/test-js-clear.js \
-	libsecret/test-js-store.js $(am__EXEEXT_3)
 TEST_SUITE_LOG = test-suite.log
 LOG_COMPILE = $(LOG_COMPILER) $(AM_LOG_FLAGS) $(LOG_FLAGS)
 am__set_b = \
-- 
2.16.1

