From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Sat, 24 Dec 2022 14:58:54 -0500
Subject: [PATCH] Install perl bindings to vendor location

---
 bindings/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bindings/Makefile.am b/bindings/Makefile.am
index 7e264794..72edebd3 100644
--- a/bindings/Makefile.am
+++ b/bindings/Makefile.am
@@ -73,7 +73,7 @@ distcheck-perl: Hamlib-pl.mk
 	$(AM_V_at)$(MAKE) $(AM_MAKEFLAGS) -f Hamlib-pl.mk distcheck
 
 install-perl: Hamlib-pl.mk
-	$(AM_V_at)$(MAKE) $(AM_MAKEFLAGS) -f Hamlib-pl.mk install_site
+	$(AM_V_at)$(MAKE) $(AM_MAKEFLAGS) -f Hamlib-pl.mk install_vendor
 
 ## ExtUtils::MakeMaker says uninstall is deprecated and won't remove the files
 ## anyway, so we'll have to figure out another way to remove the installed files.
-- 
2.40.0

