From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Sat, 24 Dec 2022 15:00:07 -0500
Subject: [PATCH] Add include guards to hamlib/config.h

---
 configure.ac | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/configure.ac b/configure.ac
index c92e77cc..5bdd6a3e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -811,6 +811,15 @@ AC_SUBST([AMP_BACKENDEPS])
 ## Final output ##
 ## ------------ ##
 
+# Add include guards to hamlib/config.h
+AH_TOP([
+#ifndef _HAMLIB_CONFIG_H
+#define _HAMLIB_CONFIG_H
+])
+AH_BOTTOM([
+#endif /* _HAMLIB_CONFIG_H */
+])
+
 dnl Output accumulated flags to the Makefile files.
 AC_SUBST([AM_CPPFLAGS])
 AC_SUBST([AM_CFLAGS])
-- 
2.34.1

