From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Sun, 25 Dec 2022 04:28:13 -0500
Subject: [PATCH] rig.h: Define hamlib_port using HAVE_WINDOWS_H to match
 src/iofunc.c.

---
 include/hamlib/rig.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/hamlib/rig.h b/include/hamlib/rig.h
index 37f85267..cbbda8f5 100644
--- a/include/hamlib/rig.h
+++ b/include/hamlib/rig.h
@@ -44,6 +44,7 @@
 #endif
 #include <pthread.h>
 
+#include <hamlib/config.h>
 /* Rig list is in a separate file so as not to mess up w/ this one */
 #include <hamlib/riglist.h>
 
@@ -2283,7 +2284,7 @@ typedef struct hamlib_port {
     int client_port;      /*!< client socket port for tcp connection */
     RIG *rig;             /*!< our parent RIG device */
     int asyncio;            /*!< enable asynchronous data handling if true -- async collides with python keyword so _async is used */
-#if defined(_WIN32)
+#if defined(_WIN32) || defined(HAVE_WINDOWS_H)
     hamlib_async_pipe_t *sync_data_pipe;         /*!< pipe data structure for synchronous data */
     hamlib_async_pipe_t *sync_data_error_pipe;   /*!< pipe data structure for synchronous data error codes */
 #else
-- 
2.40.0

