diff --git a/resp/__init__.py b/resp/__init__.py
index 7c97dcb..45fe221 100644
--- a/resp/__init__.py
+++ b/resp/__init__.py
@@ -3,7 +3,7 @@
 """Top-level package for RESP."""
 
 __authors__ =  "Asem Alenaizan"
-__version__ = '0.8'
+__version__ = '1.0.0'
 __license__ = "BSD-3-Clause"
 __date__    = "2019-08-07"
 
diff --git a/setup.py b/setup.py
index d76b67f..9b68f0e 100644
--- a/setup.py
+++ b/setup.py
@@ -2,6 +2,7 @@ from setuptools import setup
 
 setup(
     name='resp',
+    version='1.0.0',
     url='https://github.com/cdsgroup/resp',
     license='BSD license',
     packages=['resp'],
