diff --git src/Makefile src/Makefile
index d71c75c..adb4980 100644
--- src/Makefile
+++ src/Makefile
@@ -11,6 +11,9 @@ CFLAGS= -O2 -Wall -Wextra -DLUA_COMPAT_5_2 $(SYSCFLAGS) $(MYCFLAGS)
 LDFLAGS= $(SYSLDFLAGS) $(MYLDFLAGS)
 LIBS= -lm $(SYSLIBS) $(MYLIBS)
 
+# Sets the default path for lua modules
+INSTALL_TOP=/usr/local
+
 AR= ar rcu
 RANLIB= ranlib
 RM= rm -f
@@ -19,7 +22,7 @@ SYSCFLAGS=
 SYSLDFLAGS=
 SYSLIBS=
 
-MYCFLAGS=
+MYCFLAGS= -DLUA_USER_DEFAULT_PATH='"$(INSTALL_TOP)/"'
 MYLDFLAGS=
 MYLIBS=
 MYOBJS=
diff --git src/luaconf.h src/luaconf.h
index b4ed500..797ae66 100644
--- src/luaconf.h
+++ src/luaconf.h
@@ -188,7 +188,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	LUA_USER_DEFAULT_PATH
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 #define LUA_PATH_DEFAULT  \
