--- config\cmake_ext_mod\HDFTests.c.orig	2016-01-22 09:14:05.261031500 +1000
+++ config\cmake_ext_mod\HDFTests.c	2016-01-22 09:14:12.725278500 +1000
@@ -213,7 +213,7 @@
 
 int main(void)
 {
-  char *llwidthArgs[] = { "l64", "l", "L", "q", "ll", NULL };
+  char *llwidthArgs[] = { "I64", "l64", "l", "L", "q", "ll", NULL };
   char *s = malloc(128);
   char **currentArg = NULL;
   LL_TYPE x = (LL_TYPE)1048576 * (LL_TYPE)1048576;
