From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Tue, 29 Nov 2022 12:26:34 -0500
Subject: [PATCH] Remove "-fPIC" from forced CFLAGS, errors on Windows with
 Clang.

---
 makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile.in b/makefile.in
index 954b0006..09416d87 100644
--- a/makefile.in
+++ b/makefile.in
@@ -72,7 +72,7 @@ INCLUDE_CFLAGS	= $(addprefix -I,$(include_dirs))
 COVERAGE_FLAGS  = @COVERAGE_OPTION@ # dynamic library linker needs separate flag
 CONFIG_CFLAGS	= @CFLAGS@ @DEBUG_MSG_OPTION@ @ARCH_OPTION@ ${COVERAGE_FLAGS}
 CPPFLAGS	= @CPPFLAGS@ $(INCLUDE_CFLAGS)
-CFLAGS		= $(CONFIG_CFLAGS) -Wall -fPIC -Wno-deprecated -Wno-deprecated-declarations
+CFLAGS		= $(CONFIG_CFLAGS) -Wall -Wno-deprecated -Wno-deprecated-declarations
 CXXFLAGS	= $(CFLAGS) -std=c++14 -Wno-return-type-c-linkage
 LDFLAGS		= @LDFLAGS@
 LIBS		= @LIBS@
-- 
2.38.1

