diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e7d127..613b2f7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,16 +28,6 @@ MACRO(FP16_TARGET_ENABLE_CXX11 target)
 ENDMACRO()

 # ---[ Download deps
-IF(NOT DEFINED PSIMD_SOURCE_DIR)
-  MESSAGE(STATUS "Downloading PSimd to ${CMAKE_BINARY_DIR}/psimd-source (define PSIMD_SOURCE_DIR to avoid it)")
-  CONFIGURE_FILE(cmake/DownloadPSimd.cmake "${CMAKE_BINARY_DIR}/psimd-download/CMakeLists.txt")
-  EXECUTE_PROCESS(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
-    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/psimd-download")
-  EXECUTE_PROCESS(COMMAND "${CMAKE_COMMAND}" --build .
-    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/psimd-download")
-  SET(PSIMD_SOURCE_DIR "${CMAKE_BINARY_DIR}/psimd-source" CACHE STRING "PSimd source directory")
-ENDIF()
-
 IF(FP16_BUILD_TESTS AND NOT DEFINED GOOGLETEST_SOURCE_DIR)
   MESSAGE(STATUS "Downloading Google Test to ${CMAKE_BINARY_DIR}/googletest-source (define GOOGLETEST_SOURCE_DIR to avoid it)")
   CONFIGURE_FILE(cmake/DownloadGoogleTest.cmake "${CMAKE_BINARY_DIR}/googletest-download/CMakeLists.txt")
@@ -60,7 +50,7 @@ ENDIF()

 # ---[ FP16 library
 IF(${CMAKE_VERSION} VERSION_LESS "3.0")
-  ADD_LIBRARY(fp16 STATIC
+  ADD_LIBRARY(fp16 SHARED
     include/fp16.h
     include/fp16/fp16.h
     include/fp16/bitcasts.h
@@ -85,12 +75,6 @@ INSTALL(FILES
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/fp16)

 # ---[ Configure psimd
-IF(NOT TARGET psimd)
-  ADD_SUBDIRECTORY(
-    "${PSIMD_SOURCE_DIR}"
-    "${CMAKE_BINARY_DIR}/psimd")
-ENDIF()
-
 IF(FP16_BUILD_TESTS)
   # ---[ Build google test
   IF(NOT TARGET gtest)
