diff --git a/setup.py b/setup.py
index a5155a7..7ddd550 100644
--- a/setup.py
+++ b/setup.py
@@ -31,4 +31,10 @@ setup(
     'raster deep-learning ml computer-vision earth-observation geospatial geospatial-processing',
     packages=find_namespace_packages(exclude=['integration_tests*', 'tests*']),
     install_requires=install_requires,
-    zip_safe=False)
+    zip_safe=False,
+    entry_points={
+        'console_scripts': [
+            'rastervision=rastervision.aws_s3.s3_file_system',
+        ],
+    }
+)
\ No newline at end of file
