diff --git a/tests/test_cli.py b/tests/test_cli.py
index c919f2e..8ae54f6 100644
--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -11,7 +11,6 @@ from icnsutil import RawData, PIL_ENABLED, __version__
 
 def main():
     # ensure working dir is correct
-    os.chdir(os.path.join(os.path.dirname(__file__), 'fixtures'))
     print('Running tests with PIL_ENABLED =', PIL_ENABLED)
     unittest.main()
     exit()
@@ -22,8 +21,8 @@ def main():
 ###############
 
 def run_cli(args):  # relative to fixtures folder
-    exec_path = os.path.join(os.pardir, os.pardir, 'icnsutil', 'cli.py')
-    return run([sys.executable, exec_path] + args, stdout=PIPE, stderr=PIPE)
+    exec_path = os.path.join(os.path.dirname(__file__), 'fixtures')
+    return run(['icnsutil'] + args, stdout=PIPE, stderr=PIPE, cwd=exec_path)
 
 
 class TestCLI(unittest.TestCase):
