diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9eceab7..62cad56 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,8 +113,8 @@ MED_FIND_MPI()
 IF(MEDFILE_BUILD_PYTHON)
   FIND_PACKAGE(SWIG REQUIRED)
   INCLUDE(${SWIG_USE_FILE})
-  FIND_PACKAGE(PythonLibs REQUIRED)
   FIND_PACKAGE(PythonInterp REQUIRED)  # to get version string ...
+  FIND_PACKAGE(PythonLibs REQUIRED)
 ENDIF()
 
 ## Creation of files med_config.h.cmake
