From a2d1f3f3aca8b47bf696ca1a3f1d182352b9ee94 Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Thu, 22 Sep 2022 11:17:52 -0500
Subject: [PATCH 1/2] find pybind

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 74a2820..e4bbd1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -157,8 +157,8 @@ if(BGS_PYTHON_SUPPORT)
 
   #set(PYBIND11_PYTHON_VERSION 2.7 3.5 3.6)
   set(PYBIND11_PYTHON_VERSION ${BGS_PYTHON_VERSION})
-  #find_package(pybind11 REQUIRED)
-  add_subdirectory(modules/pybind11)
+  find_package(pybind11 REQUIRED)
+  #add_subdirectory(modules/pybind11)
 
   #find_package(PythonInterp ${BGS_PYTHON_VERSION} REQUIRED)
   #find_package(PythonLibs ${BGS_PYTHON_VERSION} REQUIRED)
-- 
2.34.1

