From d53ce24ecd7f1788c43baa431bee0714674bb4f1 Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Fri, 23 Sep 2022 12:48:36 -0500
Subject: [PATCH 2/2] fix unsupported cmake flag

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index c0211cb..6903b90 100644
--- a/setup.py
+++ b/setup.py
@@ -179,7 +179,7 @@ class BuildCMakeExt(build_ext, object):
                         "--config", "Release", "--", "-m"])
         else:
             self.spawn(["cmake", "--build", self.build_temp, 
-                        "--config", "Release", "--", "-j8"])
+                        "--config", "Release", "--parallel"])
 
         # Build finished, now copy the files into the copy directory
         # The copy directory is the parent directory of the extension (.pyd)
-- 
2.34.1

