From 838c95fccb58a42942da3cdd61122ab1ecf76e9d Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Fri, 23 Sep 2022 13:09:52 -0500
Subject: [PATCH 3/3] add reference to FindPythonPyEnv hack for mac

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e4bbd1a..e7b97d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -155,6 +155,10 @@ if(BGS_PYTHON_SUPPORT)
   # faster.
   set(PYBIND11_LTO_CXX_FLAGS "")
 
+  # Mac finds the wrong python library. This is a workaround to redirect it.
+  list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
+  include(FindPythonPyEnv)
+
   #set(PYBIND11_PYTHON_VERSION 2.7 3.5 3.6)
   set(PYBIND11_PYTHON_VERSION ${BGS_PYTHON_VERSION})
   find_package(pybind11 REQUIRED)
-- 
2.34.1

