--- /a/py2neo/cli/console.py	2017-06-08 13:53:23.000000000 +0200
+++ /b/py2neo/cli/console.py	2017-06-08 14:32:32.144588073 +0200
@@ -17,6 +17,8 @@
 
 
 from sys import stdin, stdout, stderr
+import sys
+import codecs
 
 from py2neo.compat import number
 
@@ -177,8 +179,13 @@
         return self.use_colour and self.err_stream.isatty()
 
     def write(self, s="", end="\n"):
-        self.out_stream.write(s.encode(self.out_encoding))
-        self.out_stream.write(end.encode(self.out_encoding))
+        output = s + end
+        if self.out_stream.encoding == self.out_encoding:
+            self.out_stream.write(output)
+        elif sys.version_info[0] >= 3:
+            self.out_stream.buffer.write(bytes(output.encode(self.out_encoding)))
+        else:
+            codecs.getwriter(self.out_encoding).write(output)
 
     def write_metadata(self, s="", end="\n"):
         if self.can_write_colour_out():
