diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e88c65..f4a033d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,10 +21,15 @@ if(WITH_CUDA)
   add_definitions(-DWITH_CUDA)
   set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --expt-relaxed-constexpr")
 
-  set(CUTLASS_DIR third_party/cutlass/include)
-  include_directories(${CUTLASS_DIR})
-  set(CUTLASS_UTIL_DIR third_party/cutlass/tools/util/include)
-  include_directories(${CUTLASS_UTIL_DIR})
+  if (NOT "$ENV{EXTERNAL_CUTLASS_INCLUDE_DIR}" STREQUAL "")
+    include_directories($ENV{EXTERNAL_CUTLASS_INCLUDE_DIR})
+  else()
+    set(CUTLASS_DIR third_party/cutlass/include)
+    include_directories(${CUTLASS_DIR})
+    set(CUTLASS_UTIL_DIR third_party/cutlass/tools/util/include)
+    include_directories(${CUTLASS_UTIL_DIR})
+  endif()
+
 endif()
 
 set(CSRC pyg_lib/csrc)
@@ -36,8 +41,14 @@ file(GLOB_RECURSE ALL_HEADERS ${CSRC}/*.h)
 add_library(${PROJECT_NAME} SHARED ${ALL_SOURCES})
 target_include_directories(${PROJECT_NAME} PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
 
-set(PHMAP_DIR third_party/parallel-hashmap)
-target_include_directories(${PROJECT_NAME} PRIVATE ${PHMAP_DIR})
+message("================================================================")
+if (NOT "$ENV{EXTERNAL_PHMAP_INCLUDE_DIR}" STREQUAL "")
+  include_directories($ENV{EXTERNAL_PHMAP_INCLUDE_DIR})
+  message("EXTERNAL_PHMAP_INCLUDE_DIR = $ENV{EXTERNAL_PHMAP_INCLUDE_DIR}")
+else()
+  set(PHMAP_DIR third_party/parallel-hashmap)
+  target_include_directories(${PROJECT_NAME} PRIVATE ${PHMAP_DIR})
+endif()
 
 find_package(Torch REQUIRED)
 target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIBRARIES})
