diff --git a/cmake/extlib_utf8.git.cmake b/cmake/extlib_utf8.git.cmake
index f35e1e9..26c4d9c 100644
--- a/cmake/extlib_utf8.git.cmake
+++ b/cmake/extlib_utf8.git.cmake
@@ -4,9 +4,9 @@ message(STATUS "entering in extlib_utf8.cmake")
 set(ext_name "utf8")
 
 if(USE_SYSTEM_DEPS)
-    find_package(utf8cpp REQUIRED)
+    # find_package(utf8cpp REQUIRED)
     add_library(${ext_name} INTERFACE IMPORTED)
-    add_dependencies(${ext_name} utf8cpp)
+    add_dependencies(${ext_name} utfcpp)
 
 else()
 
