diff --git a/pyproject.toml b/pyproject.toml
index 23d063d..dcedc2e 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -27,11 +27,12 @@ python = "^3.9"
 docling-core = "^2.0"
 deepsearch-toolkit = { version = "^1.1.0", optional = true }
 tabulate = ">=0.8.9"
-numpy = [
-    { version = "^2.0.2", markers = 'python_version >= "3.13"' },
-    { version = "^1.26.4", markers = 'python_version >= "3.9" and python_version < "3.13"' },
-    { version = ">=1.24,<1.26.0", markers = 'python_version < "3.9"' }
-]
+numpy = "*"  # it seems like there is nothing specific about numpy in the code
+# numpy = [
+#     { version = "^2.0.2", markers = 'python_version >= "3.13"' },
+#     { version = "^1.26.4", markers = 'python_version >= "3.9" and python_version < "3.13"' },
+#     { version = ">=1.24,<1.26.0", markers = 'python_version < "3.9"' }
+# ]
 pandas = [
     { version = "^2.1.4", markers = 'python_version >= "3.9"' },
     { version = ">=1.5.1", markers = 'python_version < "3.9"' },
