From bb4f8ca7caba9e027706566a068d31c612b7a806 Mon Sep 17 00:00:00 2001
From: Stefan Harris <1240279+stfnhh@users.noreply.github.com>
Date: Tue, 29 Aug 2023 16:31:03 -0400
Subject: [PATCH] update pins.h with new ssl cert

---
 pins.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/pins.h b/pins.h
index 74555743..83ae193c 100644
--- a/pins.h
+++ b/pins.h
@@ -2,8 +2,10 @@
 #define PINS_H
 const char *PK_PINS[] = {
 	/* current lastpass.com primary (Thawte) */
+	"YDjIAXSYj+mh+25FGifAiKN4oNOAj+as6gQv4naQG0M=",
+	/* previous lastpass.com primary (Thawte) */
 	"HXXQgxueCIU5TTLHob/bPbwcKOKw6DkfsTWYHbxbqTY=",
-	/* current lastpass.eu primary (AddTrust) */
+	/* previous lastpass.eu primary (AddTrust) */
 	"lCppFqbkrlJ3EcVFAkeip0+44VaoJUymbnOaEUk7tEU=",
 	/* future lastpass root CA (GlobalSign R1) */
 	"K87oWBWM9UZfyddvDfoxL+8lpNyoUB2ptGtn0fv6G2Q=",
