diff --git a/setup.py b/setup.py
index ab7da8a..72bb073 100644
--- a/setup.py
+++ b/setup.py
@@ -29,7 +29,7 @@
     long_description=long_description,
     long_description_content_type="text/markdown",
     license="Apache V2.0",
-    packages=setuptools.find_packages(),
+    packages=setuptools.find_packages(exclude=("tests", "tests.*")),
     classifiers=[
         "Development Status :: 3 - Alpha",
         "Programming Language :: Python :: 3",
