diff --git a/ext/cool.io/ev_wrap.h b/ext/cool.io/ev_wrap.h
index b19016f..6470535 100644
--- a/ext/cool.io/ev_wrap.h
+++ b/ext/cool.io/ev_wrap.h
@@ -6,5 +6,4 @@
 #define EV_USE_REALTIME 0
 #endif
 
-#include "../libev/ev.h"
-
+#require libev
diff --git a/ext/cool.io/libev.c b/ext/cool.io/libev.c
index 72cb654..2cef93f 100644
--- a/ext/cool.io/libev.c
+++ b/ext/cool.io/libev.c
@@ -5,4 +5,4 @@
  */
 
 #include "ev_wrap.h"
-#include "../libev/ev.c"
+#require libev
