diff --git a/lib/pegasus/python/Pegasus/tools/utils.py b/lib/pegasus/python/Pegasus/tools/utils.py
index 2667d2231..5709acceb 100644
--- a/lib/pegasus/python/Pegasus/tools/utils.py
+++ b/lib/pegasus/python/Pegasus/tools/utils.py
@@ -29,7 +29,6 @@ import errno
 import shutil
 import logging
 import calendar
-import commands
 import datetime
 import traceback
 import subprocess
@@ -394,7 +393,7 @@ def version():
     """
     Obtains Pegasus version
     """
-    my_output = commands.getstatusoutput("pegasus-version")
+    my_output = subprocess.check_output("pegasus-version")

     return my_output[1]

