diff --git a/lib/pegasus/python/Pegasus/service/__init__.py b/lib/pegasus/python/Pegasus/service/__init__.py
index aebc0a821..a5069a466 100644
--- a/lib/pegasus/python/Pegasus/service/__init__.py
+++ b/lib/pegasus/python/Pegasus/service/__init__.py
@@ -36,7 +36,7 @@ def get_pegasus_home():
     return None


-from flask.ext.cache import Cache
+from flask_cache import Cache
 cache = Cache(app)

 #
