diff --git a/lib/pegasus/python/Pegasus/service/auth.py b/lib/pegasus/python/Pegasus/service/auth.py
index 1589144e8..9b0619a42 100644
--- a/lib/pegasus/python/Pegasus/service/auth.py
+++ b/lib/pegasus/python/Pegasus/service/auth.py
@@ -1,7 +1,7 @@
 import logging
 import os

-import pam
+import pamela as pam
 from flask import Response, abort, g, make_response, request, url_for
 from Pegasus import user
 from Pegasus.service import app
diff --git a/lib/pegasus/python/Pegasus/service/ensembles/auth.py b/lib/pegasus/python/Pegasus/service/ensembles/auth.py
index e13301b96..4e1b26d62 100644
--- a/lib/pegasus/python/Pegasus/service/ensembles/auth.py
+++ b/lib/pegasus/python/Pegasus/service/ensembles/auth.py
@@ -2,7 +2,7 @@ import os
 import logging

 from flask import request, Response, g, abort
-import pam
+import pamela as pam

 from Pegasus import user
 from Pegasus.service.ensembles import emapp
