commit 9b5533b971aaa50fe47d187964f86bfd3b549d1d
Author: Axel Huebl <axel.huebl@plasma.ninja>
Date:   Mon Apr 24 12:23:51 2023 -0700

    Remove Conda Recipe

diff --git a/conda_recipe/Dockerfile b/conda_recipe/Dockerfile
deleted file mode 100644
index cce19c0..0000000
--- a/conda_recipe/Dockerfile
+++ /dev/null
@@ -1,25 +0,0 @@
-FROM continuumio/miniconda
-MAINTAINER Axel Huebl <axelhuebl@lbl.gov>
-
-# Arguments that need to be passed at runtime via -e platform=linux-64
-# linux-64 osx-64 linux-ppc64le win-64
-ENV platform linux-64
-
-ENV pyvs "3.6 3.7 3.8 3.9 3.10"
-
-RUN conda install --yes conda conda-build anaconda-client
-
-CMD /bin/echo "Building for $platform" \
-    && cd /home/ \
-    && mkdir -p $platform \
-    && for pyv in $pyvs; do \
-      conda build --python=$pyv . \
-      && base=$(conda build --python=$pyv . --output) \
-      && conda convert $base -p $platform; \
-    done \
-    && if test "$platform" = "linux-64"; then \
-      cp -R /opt/conda/conda-bld/linux-64/openpmd_validator* $platform/ ; \
-    fi \
-    && chmod a+w -R $platform
-
-
diff --git a/conda_recipe/meta.yaml b/conda_recipe/meta.yaml
deleted file mode 100644
index 83c1c10..0000000
--- a/conda_recipe/meta.yaml
+++ /dev/null
@@ -1,36 +0,0 @@
-{% set version = "1.1.0.4" %}
-
-package:
-  name: openpmd_validator
-  version: {{ version }}
-
-source:
-  git_rev: {{ version }}
-  git_url: https://github.com/openPMD/openPMD-validator.git
-
-build:
-  script: python setup.py install
-
-requirements:
-  build:
-    - python
-    - setuptools
-    - python-dateutil
-    - numpy
-    - h5py
-  run:
-    - python
-    - python-dateutil
-    - numpy
-    - h5py
-
-test:
-  imports:
-    - openpmd_validator
-
-about:
-  home: https://github.com/openPMD/openPMD-validator
-  license: ISC
-  license_file: LICENSE.txt
-  summary: "Validator and example scripts for openPMD files"
-
