From bf3f559d0019db4675473d2ed9c88da6aa111683 Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Sat, 22 Jan 2022 18:55:03 -0500
Subject: [PATCH 1/5] use existing onednn and make shared

---
 CMakeLists.txt | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 196e0078a..780392ea1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -290,7 +290,7 @@ if(USE_ONEDNN)
     set(DNNL_BUILD_EXAMPLES OFF CACHE INTERNAL "" FORCE)
     set(DNNL_ARCH_OPT_FLAGS "" CACHE INTERNAL "" FORCE)
     set(DNNL_ENABLE_JIT_PROFILING OFF CACHE INTERNAL "" FORCE)
-    set(DNNL_LIBRARY_TYPE STATIC CACHE INTERNAL "" FORCE)
+    set(DNNL_LIBRARY_TYPE SHARED CACHE INTERNAL "" FORCE)
     set(DNNL_ENABLE_CONCURRENT_EXEC ON CACHE INTERNAL "" FORCE)
     set(DNNL_ENABLE_PRIMITIVE_CACHE ON CACHE INTERNAL "" FORCE)
 
@@ -298,15 +298,11 @@ if(USE_ONEDNN)
       set(DNNL_CPU_RUNTIME SEQ CACHE INTERNAL "" FORCE)
     endif()
 
-    set(CMAKE_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}/onednn")
-    add_subdirectory(3rdparty/onednn)
+    set(CMAKE_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_PREFIX}/include")
   endfunction()
   load_onednn()
-  include_directories(3rdparty/onednn/include)
-  include_directories(${PROJECT_BINARY_DIR}/3rdparty/onednn/include)
   add_definitions(-DMXNET_USE_ONEDNN=1)
   list(APPEND mxnet_LINKER_LIBS dnnl)
-  set_target_properties(dnnl PROPERTIES CXX_CLANG_TIDY "")  # don't lint 3rdparty dependency
 endif()
 
 if(USE_CPP_PACKAGE)
@@ -673,7 +669,7 @@ if(UNIX)
   if(MXNET_BUILD_SHARED_LIBS)
     add_library(mxnet SHARED ${SOURCE})
   else()
-    add_library(mxnet STATIC ${SOURCE})
+    add_library(mxnet SHARED ${SOURCE})
   endif()
   target_link_libraries(mxnet PUBLIC mshadow)
   target_link_libraries(mxnet PUBLIC ${CMAKE_DL_LIBS})
@@ -856,9 +852,9 @@ endif()
 if(USE_ONEDNN)
     add_custom_command(TARGET mxnet POST_BUILD
       COMMAND ${CMAKE_COMMAND} -E copy
-      ${CMAKE_BINARY_DIR}/3rdparty/onednn/include/oneapi/dnnl/dnnl_config.h  ${CMAKE_SOURCE_DIR}/include/onednn/oneapi/dnnl/
+      ${CMAKE_INSTALL_PREFIX}/include/dnnl_config.h  ${CMAKE_SOURCE_DIR}/include/onednn/oneapi/dnnl/
       COMMAND ${CMAKE_COMMAND} -E copy
-      ${CMAKE_BINARY_DIR}/3rdparty/onednn/include/oneapi/dnnl/dnnl_version.h  ${CMAKE_SOURCE_DIR}/include/onednn/oneapi/dnnl/)
+      ${CMAKE_INSTALL_PREFIX}/include/dnnl_version.h  ${CMAKE_SOURCE_DIR}/include/onednn/oneapi/dnnl/)
 endif()
 
 if(USE_INTGEMM)
-- 
2.34.1

