From 67375a7ba098885591eed781c8c38c1e7d0ff356 Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Sat, 22 Jan 2022 20:36:55 -0500
Subject: [PATCH 2/5] edit setup py

---
 python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index ce1130ba8..08769aca9 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -121,7 +121,7 @@ setup(name='mxnet',
       version=__version__,
       description=open(os.path.join(CURRENT_DIR, 'README.md')).read(),
       packages=find_packages(),
-      data_files=[('mxnet', [LIB_PATH[0]])],
+      data_files=[('mxnet', [])],
       url='https://github.com/apache/incubator-mxnet',
       ext_modules=config_cython(),
       classifiers=[
-- 
2.34.1

