From 6fb748ca325387ca38849d15742bee2fbe0cad29 Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Sat, 22 Jan 2022 20:55:20 -0500
Subject: [PATCH 3/5] put dlpack away

---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 780392ea1..4c6da9af4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -342,7 +342,6 @@ include_directories("include")
 include_directories("3rdparty/tvm/nnvm/include")
 include_directories("3rdparty/tvm/include")
 include_directories("3rdparty/dmlc-core/include")
-include_directories("3rdparty/dlpack/include")
 
 if(UNIX)
   find_library(RTLIB rt)
@@ -939,7 +938,7 @@ install(TARGETS mxnet
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
           COMPONENT   MXNET_Development
 )
-install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/dlpack/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(DIRECTORY ${CMAKE_INSTALL_PREFIX}/include/dlpack/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/dmlc-core/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/mshadow/mshadow/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mshadow)
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/mxnet/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mxnet)
-- 
2.34.1

