From 1ff549889e3530edaf52a71f927b239e6a522000 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan.janssen@outlook.com>
Date: Mon, 11 Jul 2022 18:39:11 -0600
Subject: [PATCH] use external from system

---
 tools/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tools/Makefile b/tools/Makefile
index 83786c6..6f1ea1d 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -3,7 +3,6 @@
 
 all:
 	make -C src
-	make -C external
 
 clean:
 	make -C src clean
