From 37b72522e1eefb4a9a342296101022375dc63ca1 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan.janssen@outlook.com>
Date: Mon, 11 Jul 2022 18:53:52 -0600
Subject: [PATCH] Use conda environment variables

---
 Makefile.defs | 148 +++++++++++++++++++++++++-------------------------
 1 file changed, 74 insertions(+), 74 deletions(-)

diff --git a/Makefile.defs b/Makefile.defs
index 358c5ed..4cc345b 100644
--- a/Makefile.defs
+++ b/Makefile.defs
@@ -25,16 +25,16 @@
 # Makefile definitions for OpenPhase project.
 
 MAKE := make SETTINGS="$(SETTINGS)"
-CFLAGS = -Ofast -fopenmp -Wall -std=c++11 -pedantic-errors -DNDEBUG
+# CFLAGS = -Ofast -fopenmp -Wall -std=c++11 -pedantic-errors -DNDEBUG
 
 # Adding Thermodynamic Interfaces Flags
-TCIPATH = /opt/netapps/thermocalc/2017a/SDK/TQ/source/C
-TCLPATH = /opt/netapps/thermocalc/2017a/SDK/TQ/libtq-linux-x86_64-gfortran-9.6.10716.so
-#TCIPATH = /opt/netapps/64bit/thermocalc/4.0/SDK/TQ/source/C
-#TCLPATH = /opt/netapps/64bit/thermocalc/4.0/SDK/TQ/libtq-linux-x86_64-gfortran-9.6.10716.so
+# TCIPATH = /opt/netapps/thermocalc/2017a/SDK/TQ/source/C
+# TCLPATH = /opt/netapps/thermocalc/2017a/SDK/TQ/libtq-linux-x86_64-gfortran-9.6.10716.so
+# TCIPATH = /opt/netapps/64bit/thermocalc/4.0/SDK/TQ/source/C
+# TCLPATH = /opt/netapps/64bit/thermocalc/4.0/SDK/TQ/libtq-linux-x86_64-gfortran-9.6.10716.so
 # TQPATH = /cluster/thermocalc/4.0/SDK/TQ/source/C
-OCIPATH = /scratch/OpenCalphad/opencalphad/
-OCLPATH = /scratch/OpenCalphad/opencalphad/liboctq-isoc.a -lgfortran -lm
+# OCIPATH = /scratch/OpenCalphad/opencalphad/
+# OCLPATH = /scratch/OpenCalphad/opencalphad/liboctq-isoc.a -lgfortran -lm
 
 STDLIBS = -lfftw3_threads -lfftw3 -lgomp -lm -lvoro++ -lgfortran -lstdc++
 TCINTERFACE = TC
@@ -43,72 +43,72 @@ OCINTERFACE = OC
 OBJDIR = /obj
 LIBNAME = libOPhase.a
 
-ifeq ($(CC),g++)
-	COMPVER = $(shell $(CC) -dumpversion)
-	ifneq ($(findstring debug, $(SETTINGS)),)
-		CFLAGS += -g -DDEBUG
-    endif
-	ifneq ($(findstring silent, $(SETTINGS)),)
-		CFLAGS += -DSILENT
-    endif
-    ifneq ($(findstring serial, $(SETTINGS)),)
-        CFLAGS += -DSERIAL
-        CFLAGS := $(filter-out -fopenmp, $(CFLAGS))
-        CFLAGS += -Wno-unknown-pragmas
-    endif
-    ifneq ($(findstring parallel, $(SETTINGS)),)
-        CC = mpicxx
-        OBJDIR = /objmpi
-        LIBNAME = libOPhaseMPI.a
-	    CFLAGS += -DPARALLEL
-    endif
-    ifneq ($(findstring vulcan, $(SETTINGS)),)
-        CFLAGS += -march=corei7 -mtune=corei7-avx
-    else 
-        CFLAGS += -march=native -mtune=native
-    endif
-    ifneq ($(findstring $(TCINTERFACE), $(SETTINGS)),)
-        CFLAGS += -I$(TCIPATH)
-        CFLAGS += -DTQINTERFACE_TC -DTCTQ
-    endif
-    ifneq ($(findstring $(OCINTERFACE), $(SETTINGS)),)
-        CFLAGS += -I$(OCIPATH)
-        CFLAGS += -DTQINTERFACE_OC -DOCTQ
-    endif
-else
-	CC = g++
-	COMPVER = $(shell $(CC) -dumpversion)
-	ifneq ($(findstring debug, $(SETTINGS)),)
-		CFLAGS += -g -DDEBUG
-    endif
-	ifneq ($(findstring silent, $(SETTINGS)),)
-		CFLAGS += -DSILENT
-    endif
-    ifneq ($(findstring serial, $(SETTINGS)),)
-        CFLAGS += -DSERIAL
-		CFLAGS := $(filter-out -fopenmp, $(CFLAGS))
-		CFLAGS += -Wno-unknown-pragmas
-    endif
-    ifneq ($(findstring parallel, $(SETTINGS)),)
-        CC = mpicxx
-        OBJDIR = /objmpi
-	    CFLAGS += -DPARALLEL
-	    LIBNAME = libOPhaseMPI.a
-    endif
-    ifneq ($(findstring vulcan, $(SETTINGS)),)
-        CFLAGS += -march=corei7 -mtune=corei7-avx
-    else 
-        CFLAGS += -march=native -mtune=native
-    endif
-    ifneq ($(findstring $(TCINTERFACE), $(SETTINGS)),)
-        CFLAGS += -I$(TCIPATH)
-        CFLAGS += -DTQINTERFACE_TC -DTCTQ
-    endif
-    ifneq ($(findstring $(OCINTERFACE), $(SETTINGS)),)
-        CFLAGS += -I$(OCIPATH)
-        CFLAGS += -DTQINTERFACE_OC -DOCTQ
-    endif
-endif
+# ifeq ($(CC),g++)
+#	COMPVER = $(shell $(CC) -dumpversion)
+#	ifneq ($(findstring debug, $(SETTINGS)),)
+#		CFLAGS += -g -DDEBUG
+#    endif
+#	ifneq ($(findstring silent, $(SETTINGS)),)
+#		CFLAGS += -DSILENT
+#    endif
+#    ifneq ($(findstring serial, $(SETTINGS)),)
+#        CFLAGS += -DSERIAL
+#        CFLAGS := $(filter-out -fopenmp, $(CFLAGS))
+#        CFLAGS += -Wno-unknown-pragmas
+#    endif
+#    ifneq ($(findstring parallel, $(SETTINGS)),)
+#        CC = mpicxx
+#        OBJDIR = /objmpi
+#        LIBNAME = libOPhaseMPI.a
+# 	    CFLAGS += -DPARALLEL
+#    endif
+#    ifneq ($(findstring vulcan, $(SETTINGS)),)
+#        CFLAGS += -march=corei7 -mtune=corei7-avx
+#    else
+#        CFLAGS += -march=native -mtune=native
+#    endif
+#    ifneq ($(findstring $(TCINTERFACE), $(SETTINGS)),)
+#        CFLAGS += -I$(TCIPATH)
+#        CFLAGS += -DTQINTERFACE_TC -DTCTQ
+#    endif
+#    ifneq ($(findstring $(OCINTERFACE), $(SETTINGS)),)
+#        CFLAGS += -I$(OCIPATH)
+#        CFLAGS += -DTQINTERFACE_OC -DOCTQ
+#    endif
+# else
+# 	CC = g++
+#	COMPVER = $(shell $(CC) -dumpversion)
+#	ifneq ($(findstring debug, $(SETTINGS)),)
+#		CFLAGS += -g -DDEBUG
+#    endif
+#	ifneq ($(findstring silent, $(SETTINGS)),)
+#		CFLAGS += -DSILENT
+#    endif
+#    ifneq ($(findstring serial, $(SETTINGS)),)
+#        CFLAGS += -DSERIAL
+#		CFLAGS := $(filter-out -fopenmp, $(CFLAGS))
+#		CFLAGS += -Wno-unknown-pragmas
+#    endif
+#    ifneq ($(findstring parallel, $(SETTINGS)),)
+#        CC = mpicxx
+#        OBJDIR = /objmpi
+#	    CFLAGS += -DPARALLEL
+#	    LIBNAME = libOPhaseMPI.a
+#    endif
+#    ifneq ($(findstring vulcan, $(SETTINGS)),)
+#        CFLAGS += -march=corei7 -mtune=corei7-avx
+#    else
+#        CFLAGS += -march=native -mtune=native
+#    endif
+#    ifneq ($(findstring $(TCINTERFACE), $(SETTINGS)),)
+#        CFLAGS += -I$(TCIPATH)
+#        CFLAGS += -DTQINTERFACE_TC -DTCTQ
+#    endif
+#    ifneq ($(findstring $(OCINTERFACE), $(SETTINGS)),)
+#        CFLAGS += -I$(OCIPATH)
+#        CFLAGS += -DTQINTERFACE_OC -DOCTQ
+#    endif
+# endif
 
 AROPTIONS = -rv
 
@@ -120,7 +120,7 @@ define print_header
 @echo "---------------------------------"
 @echo "Compiling OpenPhase..."
 @echo "Start: " $(shell date)
-@echo "Compiler: " $(CC) $(COMPVER)
+@echo "Compiler: " $(CC)
 @echo "Settings: " $(SETTINGS)
 @echo "CFLAGS: " $(CFLAGS)
 @echo "Mode: $(1)"
