From 8095ff52df924f76c94d074c3570e5dac760bb62 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Sun, 19 Feb 2023 22:28:36 -0700
Subject: [PATCH] include conda path

---
 Makefile.defs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.defs b/Makefile.defs
index 3b9ba6d..b8c10e0 100644
--- a/Makefile.defs
+++ b/Makefile.defs
@@ -32,6 +32,7 @@ endif
 
 CXXFLAGS = -fopenmp -Wall -fPIC -std=c++17 
 STDLIBS = -lfftw3_omp -lfftw3 -lgomp -lm -lstdc++ -ldl
+INCLUDES = -I$(PREFIX)/include
 
 COMPVER = $(shell $(CXX) -dumpversion)
 ifneq ($(findstring debug, $(SETTINGS)),)
