From 766230150f740428709f266be791d10652356201 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 20 Feb 2023 07:18:42 -0700
Subject: [PATCH] Use CXXFLAGS from conda forge

---
 Makefile.defs | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Makefile.defs b/Makefile.defs
index 3b9ba6d..03b81bf 100644
--- a/Makefile.defs
+++ b/Makefile.defs
@@ -30,8 +30,13 @@ ifneq ($(CXX),)
     CXX = g++
 endif
 
-CXXFLAGS = -fopenmp -Wall -fPIC -std=c++17 
+ifneq ($(CXXFLAGS),)
+    CXXFLAGS = -fPIC
+endif
+
+CXXFLAGS += -fopenmp -Wall -std=c++17 
 STDLIBS = -lfftw3_omp -lfftw3 -lgomp -lm -lstdc++ -ldl
+INCLUDES = -I$(PREFIX)/include
 
 COMPVER = $(shell $(CXX) -dumpversion)
 ifneq ($(findstring debug, $(SETTINGS)),)
@@ -64,11 +69,6 @@ ifneq ($(findstring gpu, $(SETTINGS)),)
     OBJDIR    = /objgpu
     LIBNAME   = lib$(LIBPHASE)GPU.a
 endif
-ifneq ($(findstring generic, $(SETTINGS)),)
-    CXXFLAGS += -march=x86-64 -mtune=generic
-else
-	CXXFLAGS += -march=native -mtune=native
-endif
 ifneq ($(findstring shared, $(SETTINGS)),)
     LIBNAME = lib$(LIBPHASE).so
     CXXFLAGS += -fPIC
