--- src/xshmfence_alloc.c
+++ src/xshmfence_alloc.c
@@ -97,11 +97,7 @@ xshmfence_alloc_shm(void)
 				return fd;
 			unlink(template);
 #ifndef HAVE_MKOSTEMP
-			flags = fcntl(fd, F_GETFD);
-			if (flags != -1) {
-				flags |= FD_CLOEXEC;
-				fcntl(fd, F_SETFD, &flags);
-			}
+			flags = 0;
 #endif
 		}
 	}
@@ -124,13 +120,7 @@ xshmfence_alloc_shm(void)
 struct xshmfence *
 xshmfence_map_shm(int fd)
 {
-	struct xshmfence *addr;
-	addr = mmap (NULL, sizeof (struct xshmfence) , PROT_READ|PROT_WRITE, MAP_SHARED, fd, 0);
-	if (addr == MAP_FAILED) {
-		close (fd);
-		return 0;
-	}
-	return addr;
+	return 0;
 }
 
 /**
@@ -141,5 +131,5 @@ xshmfence_map_shm(int fd)
 void
 xshmfence_unmap_shm(struct xshmfence *f)
 {
-        munmap(f, sizeof (struct xshmfence));
+        return;
 }
--- src/xshmfenceint.h
+++ src/xshmfenceint.h
@@ -25,7 +25,6 @@
 
 #include <stdlib.h>
 #include <unistd.h>
-#include <sys/mman.h>
 #include "xshmfence.h"
 
 #if HAVE_FUTEX
