From 940ed92c475e6b14e94acba9bd376dde9c47733a Mon Sep 17 00:00:00 2001
From: Sherman Siu <s8siu@uwaterloo.ca>
Date: Mon, 21 Jul 2025 05:08:38 -0400
Subject: [PATCH 4/4] Factor in the cmake args when building, e.g. for
 cross-compilation

---
 setup.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/setup.py b/setup.py
index b0cc2f481..4068094eb 100755
--- a/setup.py
+++ b/setup.py
@@ -204,6 +204,9 @@ class cmake_build_ext(build_ext):
         # Make sure we use the nvcc from CUDA_HOME
         if _is_cuda():
             cmake_args += [f'-DCMAKE_CUDA_COMPILER={CUDA_HOME}/bin/nvcc']
+        user_cmake_args = os.environ.get("CMAKE_ARGS")
+        if user_cmake_args:
+            cmake_args += user_cmake_args.split()
         subprocess.check_call(
             ['cmake', ext.cmake_lists_dir, *build_tool, *cmake_args],
             cwd=self.build_temp)
