diff --git a/setup.py b/setup.py
index a4043c43..a10cb8d6 100755
--- a/setup.py
+++ b/setup.py
@@ -183,6 +183,11 @@ class cmake_build_ext(build_ext):
         else:
             # Default build tool to whatever cmake picks.
             build_tool = []
+
+        user_cmake_args = os.environ.get("VLLM_CMAKE_ARGS")
+        if user_cmake_args:
+            cmake_args.extend(user_cmake_args.split())
+
         subprocess.check_call(
             ['cmake', ext.cmake_lists_dir, *build_tool, *cmake_args],
             cwd=self.build_temp)
