diff --git a/assimulo/solvers/__init__.py b/assimulo/solvers/__init__.py
index 8a87937..867a22d 100644
--- a/assimulo/solvers/__init__.py
+++ b/assimulo/solvers/__init__.py
@@ -15,19 +15,12 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-__all__ = ["euler","radau5","sundials","runge_kutta","rosenbrock",
-           "glimda","odepack","radar5","dasp3","odassl"]
+__all__ = ["euler","sundials"]
 
 #Import all the solvers from the different modules
 from .euler import ExplicitEuler, ImplicitEuler
-from .radau5 import Radau5ODE, Radau5DAE, _Radau5ODE, _Radau5DAE 
 from .sundials import IDA, CVode
 from .kinsol import KINSOL
-from .runge_kutta import RungeKutta34, RungeKutta4, Dopri5
-from .rosenbrock import RodasODE
-from .odassl import ODASSL
-from .odepack import LSODAR
-from .radar5 import Radar5ODE
 try:
     from .dasp3 import DASP3ODE
 except ImportError as ie:
diff --git a/setup.py b/setup.py
index b3153c2..ddc1c9f 100644
--- a/setup.py
+++ b/setup.py
@@ -518,7 +518,7 @@ for e in ext_list:
     e.extra_f77_compile_args = []
     e.extra_f90_compile_args = []
 
-ext_list += prepare.fortran_extensionlists()
+#ext_list += prepare.fortran_extensionlists()
 
 # distutils part
 
