diff --git a/assimulo/solvers/glimda.py b/assimulo/solvers/glimda.py
index b7aa43c..08bb7c8 100644
--- a/assimulo/solvers/glimda.py
+++ b/assimulo/solvers/glimda.py
@@ -15,6 +15,7 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
+import sys
 import numpy as N
 
 from assimulo.exception import *
@@ -25,7 +26,7 @@ from assimulo.implicit_ode import Implicit_ODE
 try:
     from assimulo.lib.glimda import glimda
 except ImportError:
-    print("Could not find GLIMDA")
+    sys.stderr.write("Could not find GLIMDA\n")
 
 class GLIMDA(Implicit_ODE):
     """
