diff --git a/src/Makefile b/src/Makefile
index a599ec7..ee6d0a8 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -24,7 +24,8 @@ _p4p_SRCS += pvxs_client.cpp
 
 _p4p_CPPFLAGS += -DPVXS_ENABLE_EXPERT_API
 
-_p4p_LIBS += pvxs Com
+_p4p_LIBS += Com
+_p4p_SYS_LIBS += pvxs
 
 LOADABLE_LIBRARY_HOST += _gw
 
@@ -34,7 +35,8 @@ _gw_SRCS += _gw.cpp
 _gw_SRCS += pvxs_gw.cpp
 _gw_SRCS += pvxs_odometer.cpp
 
-_gw_LIBS += pvxs Com
+_gw_LIBS += Com
+_gw_SYS_LIBS += pvxs
 
 PY += p4p/__init__.py
 PY += p4p/disect.py
