diff --git a/btrack/include/inference.h b/btrack/include/inference.h
index 640133d..0c406ff 100644
--- a/btrack/include/inference.h
+++ b/btrack/include/inference.h
@@ -17,7 +17,7 @@
 #ifndef _INFERENCE_H_INCLUDED_
 #define _INFERENCE_H_INCLUDED_
 
-#include "eigen/Eigen/Dense"
+#include "eigen3/Eigen/Dense"
 #include "types.h"
 
 #include <vector>
diff --git a/btrack/include/motion.h b/btrack/include/motion.h
index 0fdd7fa..d55e19d 100644
--- a/btrack/include/motion.h
+++ b/btrack/include/motion.h
@@ -17,7 +17,7 @@
 #ifndef _MOTION_H_INCLUDED_
 #define _MOTION_H_INCLUDED_
 
-#include "eigen/Eigen/Dense"
+#include "eigen3/Eigen/Dense"
 #include "types.h"
 
 #include <vector>
diff --git a/btrack/include/tracker.h b/btrack/include/tracker.h
index accd237..f36d215 100644
--- a/btrack/include/tracker.h
+++ b/btrack/include/tracker.h
@@ -17,7 +17,7 @@
 #ifndef _TRACKER_H_INCLUDED_
 #define _TRACKER_H_INCLUDED_
 
-#include "eigen/Eigen/Dense"
+#include "eigen3/Eigen/Dense"
 #include <vector>
 #include <iostream>
 #include <map>
diff --git a/btrack/include/tracklet.h b/btrack/include/tracklet.h
index 27752cf..2b86c35 100644
--- a/btrack/include/tracklet.h
+++ b/btrack/include/tracklet.h
@@ -17,7 +17,7 @@
 #ifndef _TRACKLET_H_INCLUDED_
 #define _TRACKLET_H_INCLUDED_
 
-#include "eigen/Eigen/Dense"
+#include "eigen3/Eigen/Dense"
 #include <vector>
 
 #include "types.h"
diff --git a/btrack/include/types.h b/btrack/include/types.h
index 3163c91..09b411f 100644
--- a/btrack/include/types.h
+++ b/btrack/include/types.h
@@ -17,7 +17,7 @@
 #ifndef _TYPES_H_INCLUDED_
 #define _TYPES_H_INCLUDED_
 
-#include "eigen/Eigen/Dense"
+#include "eigen3/Eigen/Dense"
 #include <vector>
 #include <memory>
 #include <iostream>
