--- buildSrc/linux.gradle	2019-10-01 10:33:04.608065068 +0200
+++ buildSrc/linux.gradle	2019-10-01 10:34:53.676059336 +0200
@@ -200,8 +200,8 @@
     }
 )
 
-def compiler = IS_COMPILE_PARFAIT ? "parfait-gcc" : "${toolchainDir}gcc";
-def linker = IS_COMPILE_PARFAIT ? "parfait-g++" : "${toolchainDir}g++";
+def compiler = IS_COMPILE_PARFAIT ? "parfait-gcc" : System.getenv('GCC')
+def linker = IS_COMPILE_PARFAIT ? "parfait-g++" : System.getenv('GXX')
 
 LINUX.glass = [:]
 LINUX.glass.variants = ["glass", "glassgtk2"]
