--- buildSrc/mac.gradle	2019-10-01 10:33:11.732064693 +0200
+++ buildSrc/mac.gradle	2019-10-01 10:35:52.424056249 +0200
@@ -136,8 +136,8 @@
     toolchainDir = ""
 }
 
-def compiler = IS_COMPILE_PARFAIT ? "parfait-clang" : "${toolchainDir}clang";
-def linker = IS_COMPILE_PARFAIT ? "parfait-clang++" : "${toolchainDir}clang++";
+def compiler = IS_COMPILE_PARFAIT ? "parfait-clang" : System.getenv('CC')
+def linker = IS_COMPILE_PARFAIT ? "parfait-clang++" : System.getenv('CXX')
 
 MAC.glass = [:]
 MAC.glass.javahInclude = [
