--- buildSrc/mac.gradle	2019-10-01 10:33:11.732064693 +0200
+++ buildSrc/mac.gradle	2019-10-01 12:57:51.206635789 +0200
@@ -47,7 +47,7 @@
  * override these settings or pass them on the command line.
  */
 def prefSdkVersion = "10.11"
-def defaultSdkPath = "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX${prefSdkVersion}.sdk";
+def defaultSdkPath = System.getenv('CONDA_BUILD_SYSROOT')
 
 // Set the minimum API version that we require (developers do not need to override this)
 // Note that this is not necessarily the same as the preferred SDK version
@@ -136,8 +136,8 @@
     toolchainDir = ""
 }
 
-def compiler = IS_COMPILE_PARFAIT ? "parfait-clang" : "${toolchainDir}clang";
-def linker = IS_COMPILE_PARFAIT ? "parfait-clang++" : "${toolchainDir}clang++";
+def compiler = IS_COMPILE_PARFAIT ? "parfait-clang" : System.getenv('CC')
+def linker = IS_COMPILE_PARFAIT ? "parfait-clang++" : System.getenv('CXX')
 
 MAC.glass = [:]
 MAC.glass.javahInclude = [
