From d2f0de03a4c16fea71f7d7bbed5080452c1d0040 Mon Sep 17 00:00:00 2001
From: Christopher Burr <christopher.burr@cern.ch>
Date: Tue, 19 Feb 2019 19:34:12 +0000
Subject: [PATCH] Avoid using entire std namespace

---
 gsoap/wsdl/includes.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/gsoap/wsdl/includes.h b/gsoap/wsdl/includes.h
index af3e5eb..566e1b7 100755
--- a/gsoap/wsdl/includes.h
+++ b/gsoap/wsdl/includes.h
@@ -53,7 +53,15 @@ A commercial use license is available from Genivia, Inc., contact@genivia.com
 #include <set>
 #include <map>
 
-using namespace std;
+using std::map;
+using std::vector;
+using std::set;
+using std::pair;
+using std::string;
+using std::istream;
+using std::ostream;
+using std::cerr;
+using std::endl;
 
 struct ltstr
 { bool operator()(const char *s1, const char *s2) const
-- 
2.20.0

