diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..85dc3ec
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,3 @@
+*.pyc
+__pycache__/
+build/
\ No newline at end of file
diff --git a/xppy/parser/run.py b/xppy/parser/run.py
index 24bf383..355eba5 100644
--- a/xppy/parser/run.py
+++ b/xppy/parser/run.py
@@ -34,6 +34,13 @@
 tmp_name = '__tmp__'
 tmp_ode  = tmp_name+'.ode'
 tmp_set  = tmp_name+'.set'
+c_g = 'xppaut'
+
+def set_cmd(xpp_path):
+    global c_g
+    c_g = os.path.join(xpp_path, c_g)
+    print "XPP Path set: %s" % (c_g,)
+    return c_g
 
 def run(ode_file=tmp_ode, set_file=tmp_set, verbose=False):
     ''' 
@@ -44,7 +51,7 @@ def run(ode_file=tmp_ode, set_file=tmp_set, verbose=False):
     if not os.path.exists(ode_file):
         raise IOError('No such file or directory: '+ode_file)
 
-    c = 'xppaut '+ode_file+' -silent'
+    c = str(c_g) + ' '+ode_file+' -silent'
     if os.path.exists(set_file):
         c = c+' -setfile '+set_file
     # By default XPP stdio is not displayed
diff --git a/xppyDemo.py b/xppyDemo.py
new file mode 100644
index 0000000..94254da
--- /dev/null
+++ b/xppyDemo.py
@@ -0,0 +1,10 @@
+# coding: utf-8
+import xppy
+xppy.set_cmd('/PATH_TO_XPPAUT/')
+subHopf=xppy.run('SOME_ODE_FILE.ode')
+subHopf.getDesc()
+sHData=subHopf.getRawData()
+sHData.shape
+from xppy.utils import plot
+plot.plotLC(subHopf.getRawData())
+plot.pl.show()
