diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,17 +3,17 @@ EXTRA_DIST = doc README.md
 lib_LTLIBRARIES = libnmt.la
 libnmt_la_SOURCES = src/utils.c src/flatsky_utils.c src/healpix_extra.c src/nmt_bins.c src/nmt_bins_flat.c src/nmt_field.c src/nmt_field_flat.c src/nmt_mask.c src/nmt_mask_flat.c src/nmt_master.c src/nmt_covar.c src/nmt_master_flat.c src/nmt_covar_flat.c
 libnmt_la_CPPFLAGS = $(OPENMP_CFLAGS) -I./src/
-libnmt_la_LDFLAGS = $(OPENP_CFLAGS) -lfftw3 -lfftw3_omp -lgsl -lgslcblas -lcfitsio -lsharp -lfftpack -lc_utils -lm
+libnmt_la_LDFLAGS = $(OPENP_CFLAGS) -lfftw3 -lfftw3_threads -lpthread -lgsl -lgslcblas -lcfitsio -lsharp -lfftpack -lc_utils -lm

 include_HEADERS = src/namaster.h

 bin_PROGRAMS = namaster
 namaster_SOURCES = src/nmt_main.c
 namaster_CPPFLAGS = $(OPENMP_CFLAGS) -I./src/
-namaster_LDFLAGS = $(OPENMP_CFLAGS) -L./ -lnmt -lfftw3 -lfftw3_omp -lgsl -lgslcblas -lcfitsio -lsharp -lfftpack -lc_utils -lm
+namaster_LDFLAGS = $(OPENMP_CFLAGS) -L./ -lnmt -lfftw3 -lfftw3_threads -lpthread -lgsl -lgslcblas -lcfitsio -lsharp -lfftpack -lc_utils -lm

 noinst_PROGRAMS = test/check_nmt
 TESTS = test/check_nmt
 test_check_nmt_SOURCES = test/nmt_test.c test/nmt_test_utils.c test/nmt_test_fsk.c test/nmt_test_hpex.c test/nmt_test_bins.c test/nmt_test_field.c test/nmt_test_master.c test/nmt_test_covar.c test/nmt_test_bins_flat.c test/nmt_test_field_flat.c test/nmt_test_master_flat.c test/nmt_test_covar_flat.c test/nmt_test_mask.c
 test_check_nmt_CPPFLAGS = $(OPENMP_CFLAGS) -I./test/ -I./src/
-test_check_nmt_LDFLAGS = $(OPENP_CFLAGS) -L./ -lnmt -lfftw3 -lfftw3_omp -lgsl -lgslcblas -lcfitsio -lsharp -lfftpack -lc_utils -lm
+test_check_nmt_LDFLAGS = $(OPENP_CFLAGS) -L./ -lnmt -lfftw3 -lfftw3_threads -lpthread -lgsl -lgslcblas -lcfitsio -lsharp -lfftpack -lc_utils -lm
