diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -14,11 +14,11 @@ except AttributeError:

 use_icc=False #Set to True if you compiled libsharp with icc
 if use_icc :
-    libs=['nmt','fftw3','fftw3_omp','sharp','fftpack','c_utils','chealpix','cfitsio','gsl','gslcblas','m','gomp','iomp5']
-    extra=['-openmp',]
+    libs=['nmt','fftw3','fftw3_threads', 'pthread','sharp','fftpack','c_utils','chealpix','cfitsio','gsl','gslcblas','m','gomp','iomp5']
+    extra=[]
 else :
-    libs=['nmt','fftw3','fftw3_omp','sharp','fftpack','c_utils','chealpix','cfitsio','gsl','gslcblas','m','gomp']
-    extra=['-O4', '-fopenmp',]
+    libs=['nmt','fftw3','fftw3_threads', 'pthread','sharp','fftpack','c_utils','chealpix','cfitsio','gsl','gslcblas','m','gomp']
+    extra=['-O4']
