diff --git a/test/nmt_test_bins_flat.c b/test/nmt_test_bins_flat.c
index ca65558..3fa3ce4 100644
--- a/test/nmt_test_bins_flat.c
+++ b/test/nmt_test_bins_flat.c
@@ -68,7 +68,7 @@ CTEST(nmt,bins_flat_binning) {
   nmt_ell_eff_flat(bin,leff);
   nmt_unbin_cls_flat(bin,&cl_binned,lmax+1,cl,&cl,1);
 
-  for(ii=0;ii<=bin->n_bands;ii++) {
+  for(ii=0;ii<bin->n_bands;ii++) {
     ASSERT_DBL_NEAR_TOL(0.5*(bin->ell_0_list[ii]+bin->ell_f_list[ii]),leff[ii],1E-10);
     ASSERT_DBL_NEAR_TOL(leff[ii],cl_binned[ii],1E-5);
   }
