diff --git a/cmake/check_user_env.cmake b/cmake/check_user_env.cmake
index f4636c7..80d6b29 100644
--- a/cmake/check_user_env.cmake
+++ b/cmake/check_user_env.cmake
@@ -6,11 +6,6 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "XL"
         message(FATAL_ERROR "ExaChem cannot be currently built with ${CMAKE_CXX_COMPILER_ID} compilers.")
 endif()
 
-if (DEFINED ENV{CONDA_PREFIX}) #VIRTUAL_ENV
-  message(FATAL_ERROR "ExaChem cannot be currently built with CONDA. \
-          Please deactivate CONDA environment.")
-endif()
-
 if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Darwin")
     if (USE_CUDA)
         message(FATAL_ERROR "ExaChem does not support building with GPU support \
