From d795e64a96534118d24cad7cc4601c7360947c93 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 9 May 2022 18:23:52 -0600
Subject: [PATCH] Update requirements.txt

---
 requirements.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index 898aa13..7b13caa 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -3,7 +3,6 @@ numpy>=1.19
 six
 setuptools
 matplotlib
-cython
 h5py
 scipy
 hdf5plugin
