From 356e7eae8d4342c9cc617ab33242c7a461b92ab3 Mon Sep 17 00:00:00 2001
From: Klaus Zimmermann <klaus.zimmermann@smhi.se>
Date: Fri, 17 Jun 2022 17:25:31 +0200
Subject: [PATCH] Remove reading of CHANGELOG.rst from setup.py

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 390d220..65bcd52 100644
--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,6 @@ from setuptools import setup
 
 here = os.path.abspath(os.path.dirname(__file__))
 README = open(os.path.join(here, 'README.rst')).read()
-CHANGES = open(os.path.join(here, 'CHANGELOG.rst')).read()
 
 about = {}
 with open(os.path.join(here, 'era5cli', '__version__.py'), 'r') as f:
-- 
2.35.1

