diff --git a/_pyceres/glog.cc b/_pyceres/glog.cc
index 44c65b0..12b18f6 100644
--- a/_pyceres/glog.cc
+++ b/_pyceres/glog.cc
@@ -30,7 +30,7 @@ void PyBindLogStack(const char* data, T size) {
   std::cerr << std::endl << std::endl << std::endl;
 }
 
-void PyBindLogTermination() {
+void PyBindLogTermination(void) {
   std::chrono::milliseconds timespan(2000);  // or whatever
   py::scoped_estream_redirect stream(
       std::cerr,                                // std::ostream&
