diff --git a/chromium/build/config/compiler/BUILD.gn b/chromium/build/config/compiler/BUILD.gn
index 6a58d21cf07..4cabf5237a3 100644
--- a/chromium/build/config/compiler/BUILD.gn
+++ b/chromium/build/config/compiler/BUILD.gn
@@ -64,7 +64,7 @@ declare_args() {
 
   # Enable fatal linker warnings. Building Chromium with certain versions
   # of binutils can cause linker warning.
-  fatal_linker_warnings = true
+  fatal_linker_warnings = false
 
   # Build with C++ RTTI enabled. Chromium builds without RTTI by default,
   # but some sanitizers are known to require it, like CFI diagnostics
@@ -300,11 +300,6 @@ config("compiler") {
     if (fatal_linker_warnings && !is_apple && current_os != "aix") {
       ldflags += [ "-Wl,--fatal-warnings" ]
     }
-    if (fatal_linker_warnings && is_apple &&
-        # TODO(https://crbug.com/1107955): Enable in arm64 component builds.
-        !(is_mac && target_cpu == "arm64" && is_component_build)) {
-      ldflags += [ "-Wl,-fatal_warnings" ]
-    }
   } else {
   }
 
