diff --git a/chromium/build/config/BUILDCONFIG.gn b/chromium/build/config/BUILDCONFIG.gn
index 3815dd6f941..2f71d68a167 100644
--- a/chromium/build/config/BUILDCONFIG.gn
+++ b/chromium/build/config/BUILDCONFIG.gn
@@ -50,6 +50,10 @@ if (target_os == "") {
   target_os = host_os
 }
 
+current_cpu = "arm64"
+target_cpu = "arm64"
+host_cpu = "x64"
+
 if (target_cpu == "") {
   if (target_os == "android") {
     # If we're building for Android, we should assume that we want to
