From ca5e7fbc5b0ade4a990b3646945beeda8e869dd4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Edgar=20Andr=C3=A9s=20Margffoy=20Tuay?= <andfoy@gmail.com>
Date: Mon, 29 Nov 2021 18:29:43 -0500
Subject: [PATCH] Use CONDA_BUILD_SYSROOT

---
 gn/build/gen.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gn/build/gen.py b/gn/build/gen.py
index 8c646be8be3..2a37cb4e1ac 100755
--- a/gn/build/gen.py
+++ b/gn/build/gen.py
@@ -110,6 +110,7 @@ def main(argv):
                     help='The path to the macOS SDK sysroot to be used.')
   options, args = parser.parse_args(argv)

+  options.isysroot = os.environ.get('CONDA_BUILD_SYSROOT', None)
   if args:
     parser.error('Unrecognized command line arguments: %s.' % ', '.join(args))

--
2.25.1

