From 87fce9d22a4c6c4ccab4b2b096381f057cf94d6a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Edgar=20Andr=C3=A9s=20Margffoy=20Tuay?= <andfoy@gmail.com>
Date: Mon, 6 Dec 2021 12:57:45 -0500
Subject: [PATCH] Add use_system_libjpeg to gn args

---
 src/buildtools/config/common.pri | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/buildtools/config/common.pri b/src/buildtools/config/common.pri
index cf990c79..73196a8b 100644
--- a/src/buildtools/config/common.pri
+++ b/src/buildtools/config/common.pri
@@ -30,7 +30,8 @@ gn_args += \
     use_allocator=\"none\" \
     use_custom_libcxx=false \
     chrome_pgo_phase=0 \
-    enable_hangout_services_extension=true
+    enable_hangout_services_extension=true \
+    use_system_libjpeg=true
 
 # Closure compile requires Java
 gn_args += \
-- 
2.25.1

