diff -up grib2/wgrib2/geo.c.nogctpc grib2/wgrib2/geo.c
--- grib2/wgrib2/geo.c.nogctpc	2016-05-04 11:29:57.000000000 -0600
+++ grib2/wgrib2/geo.c	2016-12-04 10:35:45.012618131 -0700
@@ -900,10 +900,6 @@ int closest_init(unsigned char **sec) {
     int grid_type;
 
 
-    if (use_gctpc && output_order == wesn && nx > 0 && ny > 0) {
-       if (gctpc_ll2xy_init(sec, lon, lat) == 0) return 0;
-    }
-
     grid_type = code_table_3_1(sec);
 
     if  (!GDS_Scan_staggered(scan) && nx > 0 && ny > 0) {
@@ -951,11 +947,6 @@ long int closest(unsigned char **sec, do
     double t, xx, yy, zz, small;
     unsigned int k;
 
-    if (use_gctpc && output_order == wesn && nx > 0 && ny > 0) {
-	/* will fix it so that everything is 0 for out of bounds */
-	if (gctpc_ll2i(1, &plon, &plat, &k) == 0) return ((long int) k) - 1;
-    }
-
     grid_type = code_table_3_1(sec);
     // if grid with (lat,lon) -> (i,j) /l.. insert code here
     if (grid_type == 0 && nx > 0 && ny > 0 && output_order == wesn) return latlon_closest(sec, plat, plon);
diff -up grib2/wgrib2/rotll.c.nogctpc grib2/wgrib2/rotll.c
--- grib2/wgrib2/rotll.c.nogctpc	2016-12-04 10:35:02.455467582 -0700
+++ grib2/wgrib2/rotll.c	2016-12-04 10:36:13.292718174 -0700
@@ -2,7 +2,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
-#include "proj.h"
 #include "grb2.h"
 #include "wgrib2.h"
 #include "fnlist.h"
diff -up grib2/wgrib2/wgrib2.c.nogctpc grib2/wgrib2/wgrib2.c
--- grib2/wgrib2/wgrib2.c.nogctpc	2016-07-25 09:04:34.000000000 -0600
+++ grib2/wgrib2/wgrib2.c	2016-12-04 10:36:39.519810954 -0700
@@ -159,8 +159,6 @@ int wgrib2(int argc, const char **argv)
     if (initial_call) {		/* only done 1st time */
 	setup_user_gribtable();
 //      jas_init();
-//      gctpc initialiation
-        init(-1,-1,"gctpc_error,txt", "gctpc_param.txt");
         initial_call = 0;
     }
 
@@ -548,11 +546,6 @@ int wgrib2(int argc, const char **argv)
 		}
 #endif
 
-		if (use_gctpc && i != 0) {				/* use gctpc to get lat lon values */
-		    i = gctpc_get_latlon(sec, &lon, &lat);
-//		    if (i == 0) fprintf(stderr,"gctpc_get_lat used\n");
-		}
-
 		if (i != 0) get_latlon(sec, &lon, &lat);		 /* get lat lon of grid points using built-in code */
 	    }
 	}
