diff --git a/CMakeLists.txt b/CMakeLists.txt
index 85996167..89d313d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,11 @@ add_library(libb64 OBJECT libb64/cencode.c libb64/cdecode.c)
 target_include_directories(libb64 PRIVATE libb64)
 
 # --- sc
-add_library(sc $<TARGET_OBJECTS:iniparser> $<TARGET_OBJECTS:libb64>)
+if (SC_BUILD_SHARED_LIBS)
+  add_library(sc SHARED $<TARGET_OBJECTS:iniparser> $<TARGET_OBJECTS:libb64>)
+else()
+  add_library(sc STATIC $<TARGET_OBJECTS:iniparser> $<TARGET_OBJECTS:libb64>)
+endif()
 set_property(TARGET sc PROPERTY EXPORT_NAME SC)
 set_property(TARGET sc PROPERTY SOVERSION ${SC_SOVERSION})
 target_include_directories(sc
