--- a/pyproject.toml      2022-12-13 04:52:23.000000000 -0600
+++ b/pyproject.toml    2023-08-27 14:42:23.792305847 -0500
@@ -1,4 +1,27 @@
 [build-system]
-requires = ["maturin>=0.13,<0.14"]
+requires = ["maturin"]
 build-backend = "maturin"

+[project]
+name = "py-ed25519-zebra-bindings"
+description = "Python bindings for the ed25519-zebra RUST crate"
+authors=[{name="Stichting Polkascan (Polkascan Foundation)"}]
+version = "1.0.1"
+repository = "https://github.com/polkascan/py-ed25519-zebra-bindings"
+homepage = "https://github.com/polkascan/py-ed25519-zebra-bindings"
+license = "Apache-2.0"
+readme = "README.md"
+edition = "2018"
+maintainer = "Stichting Polkascan (Polkascan Foundation)"
+maintainer-email = "legal@polkascan.org"
+classifier = [
+    "License :: OSI Approved :: Apache Software License",
+    "Intended Audience :: Developers",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.5",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10"
+]
