--- UGrid-79841fe3af2cf60eece683e6cff2d5dc31690698.orig/include/UGrid/Constants.hpp	2022-05-03 12:07:03.000000000 -0300
+++ UGrid-79841fe3af2cf60eece683e6cff2d5dc31690698/include/UGrid/Constants.hpp	2022-06-27 14:53:46.091974807 -0300
@@ -25,6 +25,9 @@
 //
 //------------------------------------------------------------------------------
 
+#include <stdexcept>
+#include <limits>
+
 #pragma once
 
 /// \namespace ugrid
