--- UGrid-79841fe3af2cf60eece683e6cff2d5dc31690698.orig/tests/utils/include/TestUtils/Utils.hpp	2022-05-03 12:07:03.000000000 -0300
+++ UGrid-79841fe3af2cf60eece683e6cff2d5dc31690698/tests/utils/include/TestUtils/Utils.hpp	2022-06-27 15:11:28.239944112 -0300
@@ -70,11 +70,11 @@
 {
     if (value.empty())
     {
-        throw std::exception("string_to_char_array: value string is empty.");
+        throw std::runtime_error("string_to_char_array: value string is empty.");
     }
     if (char_array == nullptr)
     {
-        throw std::exception("string_to_char_array: char_array is nullptr.");
+        throw std::runtime_error("string_to_char_array: char_array is nullptr.");
     }
     for (auto i = 0; i < value.size(); ++i)
     {
