diff --git a/setup.py b/setup.py
index 622ac76..eaf9a29 100644
--- a/setup.py
+++ b/setup.py
@@ -17,13 +17,13 @@ import numpy as np
 # NOTE: `short_description` gets first line of `__doc__` only (linebreaks not allowed by setuptools)
 short_description = __doc__.strip().split('\n')[0]
 
-with open("README.md", "r") as handle:
+with open("README.md", "r",  encoding='utf-8') as handle:
     long_description = handle.read()
 
-with open("requirements.txt", "r") as handle:
+with open("requirements.txt", "r", encoding='ascii') as handle:
     requirements = handle.read()
 
-with open('holodeck/version.txt') as handle:
+with open('holodeck/version.txt', encoding='ascii') as handle:
     version = handle.read().strip()
 
 
@@ -44,7 +44,6 @@ ext_cyutils = Extension(
 
     # Silence some undesired warnings
     define_macros=[('NPY_NO_DEPRECATED_API', 0)],
-    extra_compile_args=['-Wno-unreachable-code-fallthrough', '-Wno-unused-function'],
 )
 
 ext_sam_cython = Extension(
@@ -62,7 +61,6 @@ ext_sam_cython = Extension(
 
     # Silence some undesired warnings
     define_macros=[('NPY_NO_DEPRECATED_API', 0)],
-    extra_compile_args=['-Wno-unreachable-code-fallthrough', '-Wno-unused-function'],
 )
 
 cython_modules = cythonize(
