commit cf44a6c66dbc9c572ae411c75cd5ea8e3880b754
Author: Tom Schoonjans <Tom.Schoonjans@rfi.ac.uk>
Date:   Mon Jul 27 09:57:01 2020 +0100

    soup-brotli-decompressor.h: remove superfluous get_type declaration

diff --git a/libsoup/soup-brotli-decompressor.h b/libsoup/soup-brotli-decompressor.h
index a730109..055d5df 100644
--- a/libsoup/soup-brotli-decompressor.h
+++ b/libsoup/soup-brotli-decompressor.h
@@ -31,7 +31,4 @@ G_DECLARE_FINAL_TYPE (SoupBrotliDecompressor, soup_brotli_decompressor, SOUP, BR
 SOUP_AVAILABLE_IN_2_68
 SoupBrotliDecompressor *soup_brotli_decompressor_new (void);
 
-SOUP_AVAILABLE_IN_2_68
-GType soup_brotli_decompressor_get_type (void) G_GNUC_CONST;
-
 G_END_DECLS
