commit 6c54e57d581045e049d8a343913634c84e12695d
Author: Tom Schoonjans <Tom.Schoonjans@rfi.ac.uk>
Date:   Mon Jul 27 06:27:55 2020 +0100

    Do not search for unused sqlite header

diff --git a/meson.build b/meson.build
index acfc798..8b87fd1 100644
--- a/meson.build
+++ b/meson.build
@@ -95,7 +95,7 @@ sqlite_dep = dependency('sqlite3', required: false)
 # Fallback check for sqlite, not all platforms ship pkg-config file
 if not sqlite_dep.found()
   sqlite_dep = cc.find_library('sqlite3',
-    has_headers : ['sqlite3.h', 'sqlite3ext.h'],
+    has_headers : ['sqlite3.h'],
     required: false)
 endif
 
