From 8e8dc98490c10aff3d1cb535b4f85797653813ca Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Sun, 23 Feb 2025 23:55:44 -0600
Subject: [PATCH] fix: Add python_requires to setup.py

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 5ee7690..2ce71a2 100755
--- a/setup.py
+++ b/setup.py
@@ -14,6 +14,7 @@ setup(name = 'tex2pix',
       url = None,
       description = 'Lightweight renderer of LaTeX to a variety of graphics formats',
       long_description = __doc__,
+      python_requires = ">=3.6",
       keywords = 'tex latex bibtex pdf eps postscript png graphics renderer',
       license = 'GPL',
       )
-- 
2.48.1

