From 4fc1521950123e9d97494ae65e5a4220320d5dec Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Sun, 23 Feb 2025 23:47:04 -0600
Subject: [PATCH] fix: Use setuptools over deprecated distutils

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 5ee7690..33454a1 100755
--- a/setup.py
+++ b/setup.py
@@ -1,6 +1,6 @@
 #! /usr/bin/env python
 
-from distutils.core import setup
+from setuptools import setup
 
 ## Setup definition
 import tex2pix
-- 
2.48.1

