diff -ruN firefox-91.8.0.orig/moz.configure firefox-91.8.0/moz.configure
--- firefox-91.8.0.orig/moz.configure	2022-03-30 21:25:49.000000000 +0200
+++ firefox-91.8.0/moz.configure	2022-04-29 16:39:19.567752522 +0200
@@ -759,7 +759,7 @@
     @checking("for full perl installation")
     @imports("subprocess")
     def has_full_perl_installation(perl):
-        ret = subprocess.call([perl, "-e", "use Config; exit(!-d $Config{archlib})"])
+        ret = subprocess.call([perl, "-e", "use Config"])
         return ret == 0
 
     @depends(has_full_perl_installation)
