diff -ruN -x __pycache__ -x obj -x '*~' -x old-configure -x moz.configure firefox-91.8.0.orig/mfbt/MacroArgs.h firefox-91.8.0/mfbt/MacroArgs.h
--- firefox-91.8.0.orig/mfbt/MacroArgs.h	2022-03-30 21:26:09.000000000 +0200
+++ firefox-91.8.0/mfbt/MacroArgs.h	2022-05-02 11:32:36.685179520 +0200
@@ -40,7 +40,7 @@
   MOZ_MACROARGS_ARG_COUNT_HELPER2(MOZ_MACROARGS_ARG_COUNT_HELPER(__VA_ARGS__))
 
 #define MOZ_MACROARGS_ARG_COUNT_HELPER(...)                                    \
-  (_, ##__VA_ARGS__, 50, 49, 48, 47, 46, 45, 44, 43, 42, 41, 40, 39, 38, 37,   \
+  (_ __VA_OPT__(,) __VA_ARGS__, 50, 49, 48, 47, 46, 45, 44, 43, 42, 41, 40, 39, 38, 37, \
    36, 35, 34, 33, 32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, \
    17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)
 
diff -ruN -x __pycache__ -x obj -x '*~' -x old-configure -x moz.configure firefox-91.8.0.orig/mfbt/ResultExtensions.h firefox-91.8.0/mfbt/ResultExtensions.h
--- firefox-91.8.0.orig/mfbt/ResultExtensions.h	2022-03-30 21:25:54.000000000 +0200
+++ firefox-91.8.0/mfbt/ResultExtensions.h	2022-05-02 11:38:25.880095230 +0200
@@ -317,10 +317,10 @@
 //
 //     nsCOMPtr<nsIFile> file;
 //     auto existsOrErr = MOZ_TO_RESULT_INVOKE(file, Exists);
-#define MOZ_TO_RESULT_INVOKE(obj, methodname, ...)                       \
-  ::mozilla::ToResultInvoke(                                             \
-      (obj), &::mozilla::detail::DerefedType<decltype(obj)>::methodname, \
-      ##__VA_ARGS__)
+#define MOZ_TO_RESULT_INVOKE(obj, methodname, ...)                      \
+  ::mozilla::ToResultInvoke(                                            \
+      (obj), &::mozilla::detail::DerefedType<decltype(obj)>::methodname \
+      __VA_OPT__(,) __VA_ARGS__)
 
 // Macro version of ToResultInvoke for member functions, where the result type
 // does not match the output parameter type. The macro has the advantage of not
@@ -333,7 +333,7 @@
   ::mozilla::ToResultInvoke<resultType>(                               \
       ::std::mem_fn(                                                   \
           &::mozilla::detail::DerefedType<decltype(obj)>::methodname), \
-      (obj), ##__VA_ARGS__)
+      (obj) __VA_OPT__(,) __VA_ARGS__)
 
 }  // namespace mozilla
 
