diff --git a/camb_python.f90 b/camb_python.f90
index 23e6b8c..898c00a 100644
--- a/camb_python.f90
+++ b/camb_python.f90
@@ -115,11 +115,11 @@
     Type(c_MatterTransferData) :: cData

     cData%num_q_trans = data%MTrans%num_q_trans
-    cData%q_trans = c_loc(data%MTrans%q_trans)
-    cData%sigma_8 = c_loc(data%MTrans%sigma_8)
-    cData%sigma2_vdelta_8 = c_loc(data%MTrans%sigma2_vdelta_8)
-    cData%TransferData = c_loc(data%MTrans%TransferData)
-    cData%q_trans = c_loc(data%MTrans%q_trans)
+    cData%q_trans = c_loc(data%MTrans%q_trans(1))
+    cData%sigma_8 = c_loc(data%MTrans%sigma_8(1, 1))
+    cData%sigma2_vdelta_8 = c_loc(data%MTrans%sigma2_vdelta_8(1, 1))
+    cData%TransferData = c_loc(data%MTrans%TransferData(1, 1, 1))
+    cData%q_trans = c_loc(data%MTrans%q_trans(1))
     cData%sigma_8_size = shape(data%MTrans%sigma_8)
     cData%sigma2_vdelta_8_size = shape(data%MTrans%sigma2_vdelta_8)
     cData%TransferData_size = shape(data%MTrans%TransferData)
@@ -150,18 +150,18 @@
     cData%NumSources = data%NumSources
     if (associated(Data%q%points)) then
         cData%q_size = size(data%q%points)
-        cData%q = c_loc(data%q%points)
+        cData%q = c_loc(data%q%points(1))
     else
         cData%q_size = 0
     end if
     if (associated(data%Delta_p_l_k)) then
         cData%delta_size = shape(Data%Delta_p_l_k)
-        cData%delta_p_l_k = c_loc(Data%Delta_p_l_k)
+        cData%delta_p_l_k = c_loc(Data%Delta_p_l_k(1, 1, 1))
     else
         cData%delta_size = 0
     end if
     cdata%l_size = Data%ls%l0
-    cdata%ls = c_loc(Data%ls%l)
+    cdata%ls = c_loc(Data%ls%l(1))

     end subroutine Convert_ClTransferData
