diff --git a/modules.f90 b/modules.f90
--- a/modules.f90
+++ b/modules.f90
@@ -231,7 +231,7 @@
     !     lmax is max possible number of l's evaluated
     integer, parameter :: lmax_arr = l0max

-    character(LEN=1024) :: highL_unlensed_cl_template = 'HighLExtrapTemplate_lenspotentialCls.dat'
+    character(LEN=1024) :: highL_unlensed_cl_template
     !fiducial high-accuracy high-L C_L used for making small cosmology-independent numerical corrections
     !to lensing and C_L interpolation. Ideally close to models of interest, but dependence is weak.
     logical :: use_spline_template = .true.
@@ -254,6 +254,13 @@

     contains

+    subroutine CAMB_INIT_CONDA()
+    character(LEN=1024) :: conda_prefix
+
+    call getenv( 'CONDA_PREFIX', conda_prefix )
+    write(highL_unlensed_cl_template, '(a,a)') trim(adjustl(conda_prefix)), '/share/camb/HighLExtrapTemplate_lenspotentialCls.dat'
+
+    end subroutine CAMB_INIT_CONDA

     subroutine CAMBParams_Set(P, error, DoReion)
     use constants
