diff --git a/pycamb/setup.py b/pycamb/setup.py
index 7871812..6eb2b94 100644
--- a/pycamb/setup.py
+++ b/pycamb/setup.py
@@ -29,7 +29,7 @@ os.chdir(file_dir)

 is32Bit = struct.calcsize("P") == 4

-gfortran_min = '4.9'
+gfortran_min = '4.8'


 def get_long_description():
@@ -186,13 +186,11 @@ if __name__ == "__main__":
           long_description=get_long_description(),
           author='Antony Lewis',
           url="https://camb.info/",
-          cmdclass={'build': SharedLibrary, 'build_cluster': SharedLibraryCluster,
-                    'install': CustomInstall, 'sdist': CustomSdist},
+          cmdclass={'build': SharedLibrary, 'build_cluster': SharedLibraryCluster},
           packages=['camb', 'camb_tests'],
           package_data={'camb': [DLLNAME, 'HighLExtrapTemplate_lenspotentialCls.dat',
                                  'PArthENoPE_880.2_marcucci.dat', 'PArthENoPE_880.2_standard.dat',
                                  'PRIMAT_Yp_DH_Error.dat']},
-          test_suite='camb_tests',
           classifiers=[
               "Programming Language :: Python :: 2",
               'Programming Language :: Python :: 2.7',
