From 4756cec18c27922c02801209ed9cb4b71c72d528 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sat, 5 Aug 2017 11:11:36 -0500
Subject: [PATCH] add qobject header

---
 libcore/utils.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libcore/utils.h b/libcore/utils.h
index fa4f520..1b0452c 100644
--- a/libcore/utils.h
+++ b/libcore/utils.h
@@ -23,6 +23,7 @@
 #ifndef UTILS_H
 #define UTILS_H
 
+#include <qobject.h>
 #include <qstring.h>
 
 class QIODevice;
-- 
1.8.2.3

