diff --git a/reacnetgenerator/static/webpack/webpack.config.js b/reacnetgenerator/static/webpack/webpack.config.js
index d28d626..879c8c1 100644
--- a/reacnetgenerator/static/webpack/webpack.config.js
+++ b/reacnetgenerator/static/webpack/webpack.config.js
@@ -93,7 +93,7 @@ module.exports = {
   ],
   optimization: {
     minimizer: [
-      new TerserPlugin(),
+      new TerserPlugin({cache: false,}),
     ],
   },
   performance: {
