diff --git a/src/pdbreaderlite/PDBReaderObject.cpp b/src/pdbreaderlite/PDBReaderObject.cpp
index db5fd0b..4473422 100644
--- a/src/pdbreaderlite/PDBReaderObject.cpp
+++ b/src/pdbreaderlite/PDBReaderObject.cpp
@@ -84,7 +84,7 @@ static PyObject* pdbreader_read(pdbreader* self, PyObject *args){
 //		delete [] self->coordinates;
 //	}
 
-	npy_intp dims[] = {self->reader->all_coordinates.size()};
+	npy_intp dims[] = {static_cast<npy_intp>(self->reader->all_coordinates.size())};
 //	self->coordinates = new double[dims[0]];
 //	double* remote_coords = &(self->reader->all_coordinates[0]);
 //	copy(remote_coords,remote_coords+dims[0],self->coordinates);
diff --git a/src/python/pyRMSD.cpp b/src/python/pyRMSD.cpp
index fa469e1..ca75b22 100644
--- a/src/python/pyRMSD.cpp
+++ b/src/python/pyRMSD.cpp
@@ -155,7 +155,7 @@ void parse_params(PyObject *args, Params* params){
 }
 
 PyArrayObject* embed_rmsd_data(vector<double>& rmsd){
-	npy_intp dims[1] = {rmsd.size()};
+	npy_intp dims[1] = {static_cast<npy_intp>(rmsd.size())};
 	PyArrayObject* rmsds_list_obj = (PyArrayObject *) PyArray_SimpleNew(1, dims, NPY_DOUBLE);
 
 	double* rmsd_data = (double*) (rmsds_list_obj->data);
