diff --git a/setup.py b/setup.py
index dde242b..84cdd99 100644
--- a/setup.py
+++ b/setup.py
@@ -72,8 +72,8 @@ setup(
                                         'src/calculators/NOSUP/NOSUPOmpKernel.cpp',
 
                              ],
-                             extra_compile_args = ['-fopenmp','-O3'],
-                             extra_link_args = ['-fopenmp']
+                             extra_compile_args = ['/openmp'],
+                             extra_link_args = ['/openmp']
                    )
       ],
       license = 'LICENSE.txt',
diff --git a/src/calculators/RMSDCalculator.cpp b/src/calculators/RMSDCalculator.cpp
index 01777a9..95a8c35 100644
--- a/src/calculators/RMSDCalculator.cpp
+++ b/src/calculators/RMSDCalculator.cpp
@@ -270,7 +270,7 @@ void RMSDCalculator::iterativeSuperposition(double rmsd_diff_to_stop, double* it
 		}
 		current_iteration++;
 	}
-	while(rmsd_difference > rmsd_diff_to_stop and current_iteration < MAX_ITERATIONS);
+	while(rmsd_difference > rmsd_diff_to_stop && current_iteration < MAX_ITERATIONS);
 
 	// One last superposition is performed, and the other rotation coordinates are moved here
 	superposition_with_external_reference(reference_coords);
