From 2002b465ae970a6272d333161667b259c4f19b74 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Sat, 15 Jun 2024 12:57:04 +0200
Subject: [PATCH] patch pyproject.toml

---
 pyproject.toml | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index d5cdb91..37b7db7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,12 +16,12 @@ repository = "https://gitlab.com/classroomcode/py2cfg"

 [tool.poetry.dependencies]
 python = ">= 3.7"
-astor = "^>= 0.8"
-pudb = "^>= 2022"
+astor = ">= 0.8"
+pudb = ">= 2022"
 websocket-server = ">= 0.4"
 graphviz = ">= 0.1"
-flask = "^>= 1"
-requests = "^>= 2"
+flask = ">= 1"
+requests = ">= 2"

 [tool.poetry.dev-dependencies]
