---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index c020fc9..ada9761 100644
--- a/setup.py
+++ b/setup.py
@@ -1,8 +1,8 @@
 from setuptools import setup
-import chembl_structure_pipeline
+#import chembl_structure_pipeline
 
 setup(name='chembl_structure_pipeline',
-      version=chembl_structure_pipeline.__version__,
+      version='1.0.0',
       description='ChEMBL Structure Pipeline',
       url='https://www.ebi.ac.uk/chembl/',
       author='Greg Landrum',
-- 

