diff -ruN google-cloud-workflows-0.2.0/setup.py google-cloud-workflows-0.2.0-patch/setup.py
--- setup.py	2021-01-21 17:48:51.000000000 +0100
+++ setup.py	2021-02-23 21:10:01.684688068 +0100
@@ -27,6 +27,12 @@
 with io.open(readme_filename, encoding="utf-8") as readme_file:
     readme = readme_file.read()
 
+packages = [
+    package
+    for package in setuptools.PEP420PackageFinder.find()
+    if package.startswith("google")
+]
+
 setuptools.setup(
     name="google-cloud-workflows",
     version=version,
@@ -35,7 +41,7 @@
     author_email="googleapis-packages@google.com",
     license="Apache 2.0",
     url="https://github.com/googleapis/python-workflows",
-    packages=setuptools.PEP420PackageFinder.find(),
+    packages=packages,
     namespace_packages=("google", "google.cloud"),
     platforms="Posix; MacOS X; Windows",
     include_package_data=True,
