--- config/install.py
+++ config/install.py
@@ -25,6 +25,7 @@ class Installer:
 
   def setupDirectories(self):
     self.installDir        = self.destDir
+    self.archDir           = os.path.join(self.rootDir, self.arch)
     self.rootIncludeDir    = os.path.join(self.rootDir, 'include')
     self.archIncludeDir    = os.path.join(self.rootDir, self.arch, 'include')
     self.rootConfDir       = os.path.join(self.rootDir, 'lib','slepc','conf')
@@ -220,7 +221,7 @@ for dir in dirs:
     if os.path.splitext(dst)[1] == '.dylib' and os.path.isfile('/usr/bin/install_name_tool'):
       (result, output) = commands.getstatusoutput('otool -D '+src)
       oldname = output[output.find("\n")+1:]
-      installName = oldname.replace(self.destDir, self.installDir)
+      installName = oldname.replace(self.archDir, self.installDir)
       (result, output) = commands.getstatusoutput('/usr/bin/install_name_tool -id ' + installName + ' ' + dst)
     # preserve the original timestamps - so that the .a vs .so time order is preserved
     shutil.copystat(src,dst)
