--- setup.py.orig	2025-03-11 15:52:53.742159000 -0400
+++ setup.py	2025-03-11 15:54:18.097081500 -0400
@@ -4,10 +4,10 @@
 import subprocess
 import sys
 
-subprocess.check_call(
-    [sys.executable, '-m', 'pip', 'install', '--upgrade', 'pip'])
+#subprocess.check_call(
+#    [sys.executable, '-m', 'pip', 'install', '--upgrade', 'pip'])
 
-subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'wheel'])
+#subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'wheel'])
 
 try:
     from dunamai import Version
