diff --git a/setup.py b/setup.py
index 1ee7670..efe47e7 100644
--- a/setup.py
+++ b/setup.py
@@ -268,23 +268,18 @@ def customize_build_ci(EXTENSIONS, OPTIONS):
 def customize_build_cf(EXTENSIONS, OPTIONS):
     """Customize build for conda-forge."""
 
-    library_inc = os.environ.get('LIBRARY_INC', '')
-
     del EXTENSIONS['jpeg12']
     del EXTENSIONS['jpegxl']
     del EXTENSIONS['lerc']
     del EXTENSIONS['zfp']
 
-    # build jpeg8 against libjpeg instead of libjpeg_turbo
+    # build jpeg8 or jpeg9 against libjpeg instead of libjpeg_turbo
     OPTIONS['cythonize'] = True
     EXTENSIONS['jpeg8']['cython_compile_env']['HAVE_LIBJPEG_TURBO'] = False
 
-    EXTENSIONS['jpegxr']['libraries'] = ['libjpegxr', 'libjxrglue']
-    EXTENSIONS['jpegxr']['include_dirs'] = [
-        os.path.join(library_inc, 'jpegxr')
-    ]
 
     if sys.platform == 'win32':
+        library_inc = os.environ.get('LIBRARY_INC', '')
         EXTENSIONS['bz2']['libraries'] = ['bzip2']
         EXTENSIONS['jpeg2k']['include_dirs'] = [
             os.path.join(
@@ -296,6 +291,15 @@ def customize_build_cf(EXTENSIONS, OPTIONS):
         EXTENSIONS['lzma']['libraries'] = ['liblzma']
         EXTENSIONS['png']['libraries'] = ['libpng', 'z']
         EXTENSIONS['webp']['libraries'] = ['libwebp']
+        EXTENSIONS['jpegxr']['include_dirs'] = [
+            os.path.join(os.environ['LIBRARY_INC'], 'jxrlib')
+        ]
+        EXTENSIONS['jpegxr']['libraries'] = ['libjpegxr', 'libjxrglue']
+    else:
+        EXTENSIONS['jpegxr']['include_dirs'] = [
+            os.path.join(os.environ['PREFIX'], 'include', 'jxrlib')
+        ]
+        EXTENSIONS['jpegxr']['libraries'] = ['jpegxr', 'jxrglue']
 
 
 # customize builds based on environment
@@ -306,7 +310,7 @@ except ImportError:
         customize_build = customize_build_cg
     elif os.environ.get('LD_LIBRARY_PATH', os.environ.get('LIBRARY_PATH', '')):
         customize_build = customize_build_ci
-    elif os.environ.get('LIBRARY_INC', ''):
+    elif os.environ.get('CONDA_BUILD', ''):
         customize_build = customize_build_cf
     else:
         customize_build = customize_build_default
