diff --git a/setup.py b/setup.py
index f40bad7..ab649f5 100644
--- a/setup.py
+++ b/setup.py
@@ -55,7 +55,7 @@ library_dirs = []
 if 'CG-' in os.environ.get('COMPUTERNAME', ''):
     # Windows development environment
     libraries = [
-        'zlib', 'lz4', 'webp', 'png', 'jxrlib', 'jpeg', 'lzf', 'libbz2',
+        'zlib', 'lz4', 'webp', 'png', 'jpeg', 'lzf', 'libbz2',
         'libblosc', 'snappy', 'zstd_static', 'lzma-static', 'openjp2',
         'libaec', 'lcms2',
         'brotlienc-static', 'brotlidec-static', 'brotlicommon-static'
@@ -90,7 +90,7 @@ elif os.environ.get('LD_LIBRARY_PATH', os.environ.get('LIBRARY_PATH', '')):
     # Czaki's CI environment
     libraries = [
         'jpeg', 'lz4', 'zstd', 'lzma', 'bz2', 'png', 'webp', 'blosc',
-        'openjp2', 'jxrglue', 'jpegxr', 'aec', 'lcms2', 'z',
+        'openjp2', 'aec', 'lcms2', 'z',
         'brotlienc', 'brotlidec', 'brotlicommon',
         'm'
     ]
@@ -143,12 +143,11 @@ else:
     # Most recent Debian
     libraries = [
         'jpeg', 'lz4', 'zstd', 'lzma', 'bz2', 'png', 'webp', 'blosc',
-        'openjp2', 'jxrglue', 'jpegxr', 'aec', 'lcms2', 'z',
+        'openjp2', 'jxrglue', 'aec', 'lcms2', 'z',
         'brotlienc', 'brotlidec', 'brotlicommon'
     ]
     include_dirs.extend(
         [
-            '/usr/include/jxrlib',
             '/usr/include/openjpeg-2.1',
             '/usr/include/openjpeg-2.2',
             '/usr/include/openjpeg-2.3',
