diff --git a/src/c/file.d b/src/c/file.d
index 0f739b0..dccc0cf 100755
--- a/src/c/file.d
+++ b/src/c/file.d
@@ -3351,7 +3351,7 @@ output_stream_write_byte8(cl_object strm, unsigned char *c, cl_index n)
   ecl_disable_interrupts();
   do {
     out = fwrite(c, sizeof(char), n, IO_STREAM_FILE(strm));
-  } while (out < n && restartable_io_error(strm, "fwrite"));
+  } while (out < n && (IO_STREAM_FILE(strm) != stderr) && restartable_io_error(strm, "fwrite"));
   ecl_enable_interrupts();
   return out;
 }

