From 24c18d80acb8d243ffadc2221700ad788d6c46b0 Mon Sep 17 00:00:00 2001
From: kiwi0fruit <peter.zagubisalo@gmail.com>
Date: Mon, 21 Jan 2019 17:33:39 +0700
Subject: [PATCH] patch

---
 setup.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/setup.py b/setup.py
index 62dc579..9d3d78a 100644
--- a/setup.py
+++ b/setup.py
@@ -13,11 +13,6 @@ setup(
     vcversioner={},
 
     packages=find_packages(),
-    setup_requires=[
-        'pytest-runner',
-        'pytest-asyncio; python_version > "3.4"',
-        'vcversioner'
-    ],
     tests_require=['twisted'],
     include_package_data=True,
 
-- 
2.19.2.windows.1

