diff --git a/cmake/extlib_utf8.git.cmake b/cmake/extlib_utf8.git.cmake
index daf3588..37789e4 100644
--- a/cmake/extlib_utf8.git.cmake
+++ b/cmake/extlib_utf8.git.cmake
@@ -7,10 +7,10 @@ if(USE_SYSTEM_DEPS)
     message(STATUS "using system-deps in extlib_utf8.cmake")
 
     # this will define the utf8cpp target
-    find_package(utf8cpp REQUIRED)
+    #find_package(utfcpp REQUIRED)
 
     add_library(${ext_name} INTERFACE IMPORTED)
-    add_dependencies(${ext_name} utf8cpp)
+    add_dependencies(${ext_name} utfcpp)
 
 else()
     message(STATUS "ignoring system-deps extlib_utf8.cmake")
