diff --git a/config.mk b/config.mk
index 4f74162..50e86de 100644
--- a/config.mk
+++ b/config.mk
@@ -8,17 +8,17 @@
 # the Makefiles.

 # C++ compiler
-CXX=g++
+#CXX=g++

 # Flags for the C++ compiler
-CFLAGS=-Wall -ansi -pedantic -O3
+CFLAGS+=-Wall -ansi -pedantic -O3

 # Relative include and library paths for compilation of the examples
 E_INC=-I../../src
 E_LIB=-L../../src

 # Installation directory
-PREFIX=/usr/local
+#PREFIX=/usr/local

 # Install command
 INSTALL=install
