--- setup.cfg.bak	2023-07-20 19:55:53.075638407 -0500
+++ setup.cfg	2023-07-20 19:56:02.392624262 -0500
@@ -24,10 +24,6 @@
 	pytest>=5.4.3
 	tox-travis>=0.12
 
-[options.entry_points]
-console_scripts = 
-	demultiplex = demultiplex.cli:main
-
 [egg_info]
 tag_build = 
 tag_date = 0
--- jit_open.egg-info/entry_points.txt	2021-11-13 10:07:23.000000000 -0600
+++ /dev/null	2023-05-17 21:22:02.983000000 -0500
@@ -1,3 +0,0 @@
-[console_scripts]
-demultiplex = demultiplex.cli:main
-
