From d49190aa5edb7104924f4ded84126b5efa7b5bec Mon Sep 17 00:00:00 2001
From: Justin Carpentier <justin.carpentier@inria.fr>
Date: Tue, 23 Jun 2020 17:40:00 +0200
Subject: [PATCH 1/2] cmake: sync submodule

---
 CMakeLists.txt | 5 ++++-
 cmake          | 2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f7e729..cbb1f5a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,9 @@ SET(PROJECT_URL https://github.com/${PROJECT_ORG}/${PROJECT_NAME})
 OPTION(BUILD_PYTHON_INTERFACE "Build the python unit tests and helpers" ON)
 OPTION(INSTALL_PYTHON_INTERFACE_ONLY "Install *ONLY* the python interface" OFF)
 
+MESSAGE(STATUS "python: ${PYTHON_SITELIB}")
+#STRING(REPLACE "\\" "/" TEST ${PYTHON_SITELIB})
+#MESSAGE(STATUS "python: ${TEST}")
 # Project configuration
 IF(NOT INSTALL_PYTHON_INTERFACE_ONLY)
   SET(PROJECT_USE_CMAKE_EXPORT TRUE)
@@ -31,8 +34,8 @@ COMPUTE_PROJECT_ARGS(PROJECT_ARGS LANGUAGES CXX)
 PROJECT(${PROJECT_NAME} ${PROJECT_ARGS})
 
 IF(BUILD_PYTHON_INTERFACE)
-  ADD_PROJECT_DEPENDENCY(pinocchio)
   FINDPYTHON()
+  ADD_PROJECT_DEPENDENCY(pinocchio)
   STRING(REGEX REPLACE "-" "_" PY_NAME ${PROJECT_NAME})
   ADD_SUBDIRECTORY(python)
   IF(BUILD_TESTING)
Submodule cmake d12c55e...d63b949:
diff --git a/cmake/python.cmake b/cmake/python.cmake
index 12ef0d6..3286da3 100644
--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -203,6 +203,12 @@ MACRO(FINDPYTHON)
     IF(PYTHON_PACKAGES_DIR)
       STRING(REGEX REPLACE "(site-packages|dist-packages)" "${PYTHON_PACKAGES_DIR}" PYTHON_SITELIB "${PYTHON_SITELIB}")
     ENDIF(PYTHON_PACKAGES_DIR)
+  ELSE(NOT PYTHON_SITELIB)
+    IF(WIN32)
+      STRING(REPLACE "\\" "/" PYTHON_SITELIB ${PYTHON_SITELIB})
+      MESSAGE(STATUS "python: ${PYTHON_SITELIB}")
+      SET(PYTHON_SITELIB ${PYTHON_SITELIB} PARENT_SCOPE)
+    ENDIF(WIN32)
   ENDIF(NOT PYTHON_SITELIB)
 
   MESSAGE(STATUS "Python site lib: ${PYTHON_SITELIB}")
@@ -381,7 +387,7 @@ MACRO(PYTHON_INSTALL_ON_SITE MODULE FILE)
     MESSAGE(FATAL_ERROR "Python has not been found.")
   ENDIF()
 
-  PYTHON_INSTALL("${MODULE}" "${FILE}" "${PYTHON_SITELIB}")
+  PYTHON_INSTALL("${MODULE}" "${FILE}" ${PYTHON_SITELIB})
 
 ENDMACRO()
 
diff --git a/cmake/test.cmake b/cmake/test.cmake
index 1ab74cd..852c2d7 100644
--- a/cmake/test.cmake
+++ b/cmake/test.cmake
@@ -107,9 +107,9 @@ MACRO(ADD_PYTHON_UNIT_TEST NAME SOURCE)
   SET(MODULES "${ARGN}")  # ARGN is not a variable
   FOREACH(MODULE_PATH IN LISTS MODULES)
     LIST(APPEND PYTHONPATH "${CMAKE_BINARY_DIR}/${MODULE_PATH}")
-    IF(WIN32)
+    IF(CMAKE_GENERATOR MATCHES "Visual Studio")
       LIST(APPEND PYTHONPATH "${CMAKE_BINARY_DIR}/${MODULE_PATH}/$<CONFIG>")
-    ENDIF(WIN32)
+    ENDIF(CMAKE_GENERATOR MATCHES "Visual Studio")
   ENDFOREACH(MODULE_PATH IN LISTS MODULES)
 
   IF(DEFINED ENV{PYTHONPATH})
-- 
2.21.1 (Apple Git-122.3)

