From 1c02745cb8e3d1eb34218228a18c791e7d066712 Mon Sep 17 00:00:00 2001
From: Justin Carpentier <justin.carpentier@inria.fr>
Date: Tue, 23 Jun 2020 17:40:34 +0200
Subject: [PATCH 2/2] cmake: fix Windows issue

---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cbb1f5a..2dae265 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,9 +10,6 @@ SET(PROJECT_URL https://github.com/${PROJECT_ORG}/${PROJECT_NAME})
 OPTION(BUILD_PYTHON_INTERFACE "Build the python unit tests and helpers" ON)
 OPTION(INSTALL_PYTHON_INTERFACE_ONLY "Install *ONLY* the python interface" OFF)
 
-MESSAGE(STATUS "python: ${PYTHON_SITELIB}")
-#STRING(REPLACE "\\" "/" TEST ${PYTHON_SITELIB})
-#MESSAGE(STATUS "python: ${TEST}")
 # Project configuration
 IF(NOT INSTALL_PYTHON_INTERFACE_ONLY)
   SET(PROJECT_USE_CMAKE_EXPORT TRUE)
-- 
2.21.1 (Apple Git-122.3)

