--- pc/gawkmisc.pc	2017-12-29 16:27:03.081713200 -0500
+++ pc/gawkmisc.pc	2017-12-29 16:27:08.603741408 -0500
@@ -678,6 +678,8 @@
   return setlocale (category, new_locale);
 }

+#ifndef __MINGW32__
+
 /*
  * On MS-Windows with MinGW, execvp causes the shell and the re-exec'ed
  * dgawk to compete for the keyboard input.
@@ -692,6 +694,8 @@
 	return -1;
 }

+#endif
+
 #ifdef DYNAMIC

 #include <dlfcn.h>
