diff --git c-blosc/hdf5/blosc_filter.c c-blosc/hdf5/blosc_filter.c
index aa35215..8f33f7d 100644
--- c-blosc/hdf5/blosc_filter.c
+++ c-blosc/hdf5/blosc_filter.c
@@ -20,7 +20,7 @@
 
 #if H5Epush_vers == 2
 /* 1.8.x */
-#define PUSH_ERR(func, minor, str...) H5Epush(H5E_DEFAULT, __FILE__, func, __LINE__, H5E_ERR_CLS, H5E_PLINE, minor, str)
+#define PUSH_ERR(func, minor, str, ...) H5Epush(H5E_DEFAULT, __FILE__, func, __LINE__, H5E_ERR_CLS, H5E_PLINE, minor, str)
 #else
 /* 1.6.x */
 #define PUSH_ERR(func, minor, str) H5Epush(__FILE__, func, __LINE__, H5E_PLINE, minor, str)
