diff --git setup.py setup.py
index 7988dd7..18bd06a 100755
--- setup.py
+++ setup.py
@@ -134,7 +134,6 @@ debug = '--debug' in sys.argv
 lib_dirs = []
 inc_dirs = [os.path.join('hdf5-blosc', 'src')]
 optional_libs = []
-data_files = []    # list of data files to add to packages (mainly for DLL's)
 
 default_header_dirs = None
 default_library_dirs = None
@@ -388,7 +387,7 @@ if os.name == 'posix':
 elif os.name == 'nt':
     _Package = WindowsPackage
     _platdep = {  # package tag -> platform-dependent components
-        'HDF5': ['hdf5dll', 'hdf5dll'],
+        'HDF5': ['hdf5', 'hdf5'],
         'LZO2': ['lzo2', 'lzo2'],
         'LZO': ['liblzo', 'lzo1'],
         'BZ2': ['bzip2', 'bzip2'],
@@ -397,9 +396,7 @@ elif os.name == 'nt':
 
     # Copy the next DLL's to binaries by default.
     # Update these paths for your own system!
-    dll_files = ['\\windows\\system\\zlib1.dll',
-                 '\\windows\\system\\szip.dll',
-                 ]
+    dll_files = []
     if '--debug' in sys.argv:
         _platdep['HDF5'] = ['hdf5ddll', 'hdf5ddll']
 
@@ -751,11 +748,6 @@ def find_name(base='tables'):
 
 name = find_name()
 
-if os.name == "nt":
-    # Add DLL's to the final package for windows
-    data_files.extend([
-        ('Lib/site-packages/%s' % name, dll_files),
-    ])
 
 ADDLIBS = [hdf5_package.library_name]
 
@@ -967,6 +959,5 @@ interactively save and retrieve large amounts of data.
     platforms=['any'],
     ext_modules=extensions,
     cmdclass=cmdclass,
-    data_files=data_files,
     **setuptools_kwargs
 )
