diff --git tables/__init__.py tables/__init__.py
index a1af81b..549cefe 100644
--- tables/__init__.py
+++ tables/__init__.py
@@ -21,6 +21,7 @@ to efficiently cope with extremely large amounts of data.
 
 
 import os
+import sys
 
 # On Windows, pre-load the HDF5 DLLs into the process via Ctypes
 # to improve diagnostics and avoid issues when loading DLLs during runtime.
@@ -60,9 +61,10 @@ if os.name == 'nt':
     # In order to improve diagnosis of a common Windows dependency
     # issue, we explicitly test that we can load the HDF5 dll before
     # loading tables.utilsextensions.
-    if not _load_library('hdf5dll.dll', ctypes.cdll.LoadLibrary):
+    if not _load_library('hdf5.dll', ctypes.cdll.LoadLibrary,
+                         dllpaths=(r'%s\Library\bin' % sys.prefix, "")):
         raise ImportError(
-            'Could not load "hdf5dll.dll", please ensure'
+            'Could not load "hdf5.dll", please ensure'
             ' that it can be found in the system path')
 
     # Some PyTables binary distributions place the dependency DLLs in the
