diff --git c-blosc/blosc/shuffle.c c-blosc/blosc/shuffle.c
index e8983bd..22a3713 100644
--- c-blosc/blosc/shuffle.c
+++ c-blosc/blosc/shuffle.c
@@ -123,10 +123,9 @@ static uint64_t _xgetbv(uint32_t xcr) {
     which means we can get away with returning a hard-coded value from
     this implementation of _xgetbv. */
 
-static inline uint64_t
-_xgetbv(uint32_t xcr) {
+static inline uint64_t _xgetbv(uint32_t xcr) {
     /* A 64-bit OS must have XMM save support. */
-    return xcr == 0 ? (1UL << 1) : 0UL;
+    return (xcr == 0 ? (1UL << 1) : 0UL);
 }
 
 #else
diff --git hdf5-blosc/src/blosc_filter.c hdf5-blosc/src/blosc_filter.c
index 3ab5237..5275a1d 100644
--- hdf5-blosc/src/blosc_filter.c
+++ hdf5-blosc/src/blosc_filter.c
@@ -20,7 +20,7 @@
 
 #if H5Epush_vers == 2
 /* 1.8.x */
-#define PUSH_ERR(func, minor, str...) H5Epush(H5E_DEFAULT, __FILE__, func, __LINE__, H5E_ERR_CLS, H5E_PLINE, minor, str)
+#define PUSH_ERR(func, minor, str, ...) H5Epush(H5E_DEFAULT, __FILE__, func, __LINE__, H5E_ERR_CLS, H5E_PLINE, minor, str)
 #else
 /* 1.6.x */
 #define PUSH_ERR(func, minor, str) H5Epush(__FILE__, func, __LINE__, H5E_PLINE, minor, str)
