diff -ru a/configure b/configure
--- a/configure	2018-09-22 06:25:29.000000000 -0400
+++ b/configure	2020-05-12 22:57:44.677739700 -0400
@@ -12120,9 +12120,9 @@
     HSYS=solaris
     HCPU="${host_cpu}"
     ;;
-  *mingw32*)
+  *mingw*)
     PANO_WIN32_RESOURCE=pano13.lo
-    HSYS=mingw32
+    HSYS=mingw
     HCPU="${host_cpu}"
     ;;
   *cygwin*)
@@ -12173,7 +12173,7 @@
   HAVE_MAC_FALSE=
 fi
 
- if test "${HSYS}" = "mingw32"; then
+ if test "${HSYS}" = "mingw"; then
   HAVE_MINGW_TRUE=
   HAVE_MINGW_FALSE='#'
 else
@@ -12292,7 +12292,7 @@
     darwin*)
       java_extra_inc=darwin
       ;;
-    *mingw32*)
+    *mingw*)
       java_extra_inc=win32
       ;;
     *cygwin*)
@@ -12652,7 +12652,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng -lz -lm $LIBS"
+LIBS="-lpng -lz $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -16373,4 +16373,3 @@
 echo "  LIB_ZLIB       = $LIB_ZLIB"
 echo "  JAVA_FLAGS     = $JAVA_FLAGS"
 echo "  ENDIAN_FLAG    = $ENDIAN_FLAG"
-
diff -ru a/sys_X11.h b/sys_X11.h
--- a/sys_X11.h	2014-02-17 17:04:42.000000000 -0500
+++ b/sys_X11.h	2020-05-12 22:01:03.725795700 -0400
@@ -22,7 +22,6 @@
 #ifndef SYS_X11_H
 #define SYS_X11_H
 
-#include <libgimp/gimp.h>
 #include <gtk/gtk.h>
 
 
@@ -258,7 +257,6 @@
     argv[0] 		= g_strdup("dummy");										\
 																				\
     gtk_init(&argc, &argv);														\
-    gtk_rc_parse(gimp_gtkrc());													\
 																				\
 	WinFunc( NULL, (gpointer) thePrefs );										\
  																				\
