diff --git a/src/Database/CMakeLists.txt b/src/Database/CMakeLists.txt
index 94c7086..21f1a2b 100644
--- a/src/Database/CMakeLists.txt
+++ b/src/Database/CMakeLists.txt
@@ -6,6 +6,8 @@ include(Files.cmake)
 include(ImportUtilsOS)
 import_utils_os()
 include(AddEigen)
+
+if(NOT SKIP_LIBRARY_BUILD)
 include(ImportMongoCxx)
 import_mongocxx()
 
@@ -49,6 +51,10 @@ scine_install_component_cmake_files(
   COMPONENT  ScineDatabase
   EXPORT_NAME databaseTargets
 )
+else()
+  include(ImportDatabase)
+  import_database()
+endif()
 
 # Tests
 if(SCINE_BUILD_TESTS)
@@ -131,11 +137,6 @@ if(SCINE_BUILD_PYTHON_BINDINGS)
     string(APPEND database_PY_DEPS ", \"${_target_filename}\"")
   endforeach()
 
-  add_custom_command(TARGET Database POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy ${_py_target_gen_exprs} ${CMAKE_CURRENT_BINARY_DIR}/scine_database
-    COMMENT "Copying dependent shared libraries into python package directory"
-  )
-
   # Copy the setup file and README.rst into the binary dir so you can install
   # using pip or make a distribution with the compiled binary
   configure_file(
