From e4c0b734aa580064dc57501114f857aaab2a7039 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwelk@xhochy.com>
Date: Sun, 30 Jun 2019 18:11:44 +0200
Subject: [PATCH] Support building with -std=c++14

---
 src/include/common/helper.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/include/common/helper.hpp b/src/include/common/helper.hpp
index fcf9b564..33e9ad34 100644
--- a/src/include/common/helper.hpp
+++ b/src/include/common/helper.hpp
@@ -19,7 +19,7 @@
 #endif
 
 namespace duckdb {
-#ifndef _MSC_VER
+#if !defined(_MSC_VER) && (__cplusplus < 201402L)
 template <typename T, typename... Args> unique_ptr<T> make_unique(Args &&... args) {
 	return unique_ptr<T>(new T(std::forward<Args>(args)...));
 }
-- 
2.22.0

