diff --git a/CMakeLists.txt b/CMakeLists.txt
index efe45a0..9872550 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,44 +20,46 @@ set(UAMMD_INCLUDE include/third_party/uammd/src include/third_party/uammd/src/th
 # find_package(UAMMD REQUIRED)
 # set(UAMMD_INCLUDE ${UAMMD_INCLUDE_DIR})
 set(BLA_STATIC OFF)
-set(BLA_VENDOR Intel10_64lp)
 find_package(BLAS)
-if(BLAS_FOUND)
-  message("mkl environment detected")
-  add_compile_definitions(PUBLIC USE_MKL)
-  link_libraries(
-    BLAS::BLAS
-  )
-  find_path(BLAS_INCLUDE_DIRS mkl.h
-    $ENV{CONDA_PREFIX}/include
-    /usr/include
-    /usr/local/include
-    $ENV{MKLROOT}/include
-    $ENV{BLAS_HOME}/include
-  )
+find_package(LAPACK)
+if(BLAS_FOUND AND LAPACK_FOUND)
+  message(STATUS "BLAS and LAPACK found")
+  # Additional check for MKL
+  find_path(MKL_INCLUDE_DIR NAMES mkl.h PATHS $ENV{MKLROOT}/include)
+  if(MKL_INCLUDE_DIR)
+    message(STATUS "Intel MKL environment detected")
+    add_compile_definitions(PUBLIC USE_MKL)
+    include_directories(${MKL_INCLUDE_DIR})
+  else()
+    find_path(BLAS_INCLUDE_DIR NAMES lapacke.h PATHS
+      $ENV{CONDA_PREFIX}/include
+      $ENV{PREFIX}/include
+      $ENV{MKLROOT}/include
+      $ENV{BLAS_HOME}/include
+      /usr/include
+      /usr/local/include
+    )
+    find_path(LAPACKE_INCLUDE_DIR lapacke.h)
+    find_library(LAPACKE_LIBRARY NAMES lapacke)
+    if(LAPACKE_INCLUDE_DIR AND LAPACKE_LIBRARY)
+      set(LAPACKE_FOUND TRUE)
+      set(LAPACKE_LIBRARIES ${LAPACKE_LIBRARY})
+      set(LAPACKE_INCLUDE_DIRS ${LAPACKE_INCLUDE_DIR})
+    else()
+      message(FATAL_ERROR "LAPACKE not found.")
+    endif()
+    include_directories(${LAPACKE_INCLUDE_DIRS})
+    include_directories(${BLAS_INCLUDE_DIR})
+    link_libraries(${LAPACKE_LIBRARIES})
+  endif()
+  link_libraries(BLAS::BLAS LAPACK::LAPACK)
 else()
-  unset(BLA_VENDOR)
-  find_package(LAPACK REQUIRED)
-  find_package(LAPACKE REQUIRED)
-  find_package(BLAS REQUIRED)
-  link_libraries(${LAPACK_LIBRARIES} ${LAPACKE_LIBRARIES})
-  find_path(BLAS_INCLUDE_DIRS cblas.h
-    $ENV{CONDA_PREFIX}/include
-    /usr/include
-    /usr/local/include
-    $ENV{MKLROOT}/include
-    $ENV{BLAS_HOME}/include
-  )
-  find_path(LAPACKE_INCLUDE_DIRS lapacke.h
-    $ENV{CONDA_PREFIX}/include
-    /usr/include
-    /usr/local/include
-    $ENV{MKLROOT}/include
-    $ENV{LAPACKE_HOME}/include
-  )
-
+  message(FATAL_ERROR "BLAS and LAPACK not found. Please install them.")
 endif()
+
 find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
+find_package(pybind11 CONFIG)
+include_directories(${pybind11_INCLUDE_DIRS})
 include_directories(${Python3_INCLUDE_DIRS})
 include_directories(${UAMMD_INCLUDE} ${BLAS_INCLUDE_DIRS} ${LAPACKE_INCLUDE_DIRS})
 include_directories(include/third_party/LanczosAlgorithm/include)
diff --git a/solvers/DPStokes/CMakeLists.txt b/solvers/DPStokes/CMakeLists.txt
index 9b66b7e..24df824 100644
--- a/solvers/DPStokes/CMakeLists.txt
+++ b/solvers/DPStokes/CMakeLists.txt
@@ -3,12 +3,12 @@ add_library(DPStokes.${Python3_SOABI} SHARED python_wrapper.cpp)
 # Remove the "lib" prefix from the shared library
 set_target_properties(DPStokes.${Python3_SOABI} PROPERTIES PREFIX "")
 target_include_directories(libMobility_DPStokes PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
-target_link_libraries(libMobility_DPStokes cufft cublas cusolver curand)
+target_link_libraries(libMobility_DPStokes cufft cublas)
 set_target_properties(libMobility_DPStokes PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
 set_target_properties(DPStokes.${Python3_SOABI} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/python")
 target_link_libraries(DPStokes.${Python3_SOABI} libMobility_DPStokes)
 install(TARGETS DPStokes.${Python3_SOABI} DESTINATION ${Python3_SITEARCH}/libMobility)
-install(TARGETS libMobility_DPStokes DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS libMobility_DPStokes DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/mobility.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/libMobility/solvers/DPStokes)
 file(GLOB EXTRA_HEADERS extra/*.h)
 install(FILES ${EXTRA_HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/include/libMobility/solvers/DPStokes/extra)
diff --git a/solvers/NBody/CMakeLists.txt b/solvers/NBody/CMakeLists.txt
index d060c38..291ac6e 100644
--- a/solvers/NBody/CMakeLists.txt
+++ b/solvers/NBody/CMakeLists.txt
@@ -7,7 +7,7 @@ set_target_properties(libMobility_NBody PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${C
 set_target_properties(NBody.${Python3_SOABI} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/python")
 target_link_libraries(NBody.${Python3_SOABI} libMobility_NBody)
 install(TARGETS NBody.${Python3_SOABI} DESTINATION ${Python3_SITEARCH}/libMobility)
-install(TARGETS libMobility_NBody DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS libMobility_NBody DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/mobility.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/libMobility/solvers/NBody)
 file(GLOB EXTRA_HEADERS extra/*.h)
 install(FILES ${EXTRA_HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/include/libMobility/solvers/NBody/extra)
diff --git a/solvers/PSE/CMakeLists.txt b/solvers/PSE/CMakeLists.txt
index e741616..082fca1 100644
--- a/solvers/PSE/CMakeLists.txt
+++ b/solvers/PSE/CMakeLists.txt
@@ -3,12 +3,12 @@ add_library(PSE.${Python3_SOABI} SHARED python_wrapper.cpp)
 # Remove the "lib" prefix from the shared library
 set_target_properties(PSE.${Python3_SOABI} PROPERTIES PREFIX "")
 target_include_directories(libMobility_PSE PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
-target_link_libraries(libMobility_PSE cufft cublas cusolver curand)
+target_link_libraries(libMobility_PSE cufft cublas)
 set_target_properties(libMobility_PSE PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
 set_target_properties(PSE.${Python3_SOABI} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/python")
 target_link_libraries(PSE.${Python3_SOABI} libMobility_PSE)
 install(TARGETS PSE.${Python3_SOABI} DESTINATION ${Python3_SITEARCH}/libMobility)
-install(TARGETS libMobility_PSE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS libMobility_PSE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/mobility.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/libMobility/solvers/PSE)
 file(GLOB EXTRA_HEADERS extra/*.h)
 install(FILES ${EXTRA_HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/include/libMobility/solvers/PSE/extra)
diff --git a/solvers/SelfMobility/CMakeLists.txt b/solvers/SelfMobility/CMakeLists.txt
index 1c728e5..d4378a6 100644
--- a/solvers/SelfMobility/CMakeLists.txt
+++ b/solvers/SelfMobility/CMakeLists.txt
@@ -7,5 +7,5 @@ set_target_properties(libMobility_SelfMobility PROPERTIES RUNTIME_OUTPUT_DIRECTO
 set_target_properties(SelfMobility.${Python3_SOABI} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/python")
 target_link_libraries(SelfMobility.${Python3_SOABI} libMobility_SelfMobility)
 install(TARGETS SelfMobility.${Python3_SOABI} DESTINATION ${Python3_SITEARCH}/libMobility)
-install(TARGETS libMobility_SelfMobility DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS libMobility_SelfMobility DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/mobility.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/libMobility/solvers/SelfMobility)
