diff --git a/CMakeLists.txt b/CMakeLists.txt
index efe45a0..de602b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,44 +20,44 @@ set(UAMMD_INCLUDE include/third_party/uammd/src include/third_party/uammd/src/th
 # find_package(UAMMD REQUIRED)
 # set(UAMMD_INCLUDE ${UAMMD_INCLUDE_DIR})
 set(BLA_STATIC OFF)
-set(BLA_VENDOR Intel10_64lp)
+# if(NOT DEFINED BLA_VENDOR)
+#   set(BLA_VENDOR Intel10_64lp)  # Default to Intel MKL LP64
+# endif()
+
+# Try to find BLAS and LAPACK
 find_package(BLAS)
-if(BLAS_FOUND)
-  message("mkl environment detected")
-  add_compile_definitions(PUBLIC USE_MKL)
-  link_libraries(
-    BLAS::BLAS
-  )
-  find_path(BLAS_INCLUDE_DIRS mkl.h
-    $ENV{CONDA_PREFIX}/include
-    /usr/include
-    /usr/local/include
-    $ENV{MKLROOT}/include
-    $ENV{BLAS_HOME}/include
-  )
-else()
-  unset(BLA_VENDOR)
-  find_package(LAPACK REQUIRED)
-  find_package(LAPACKE REQUIRED)
-  find_package(BLAS REQUIRED)
-  link_libraries(${LAPACK_LIBRARIES} ${LAPACKE_LIBRARIES})
-  find_path(BLAS_INCLUDE_DIRS cblas.h
+find_package(LAPACK)
+
+if(BLAS_FOUND AND LAPACK_FOUND)
+  message(STATUS "BLAS and LAPACK found")
+  # Additional check for MKL
+  find_path(MKL_INCLUDE_DIR NAMES mkl.h PATHS $ENV{MKLROOT}/include)
+  if(MKL_INCLUDE_DIR)
+    message(STATUS "Intel MKL environment detected")
+    add_compile_definitions(USE_MKL)
+    include_directories(${MKL_INCLUDE_DIR})
+  endif()
+  message(STATUS "BLAS include dirs: ${BLAS_INCLUDE_DIRS}")
+  message(STATUS "LAPACK include dirs: ${LAPACKE_INCLUDE_DIRS}")
+  find_path(BLAS_INCLUDE_DIR NAMES cblas.h mkl.h PATHS
     $ENV{CONDA_PREFIX}/include
-    /usr/include
-    /usr/local/include
+    $ENV{PREFIX}/include
     $ENV{MKLROOT}/include
     $ENV{BLAS_HOME}/include
-  )
-  find_path(LAPACKE_INCLUDE_DIRS lapacke.h
-    $ENV{CONDA_PREFIX}/include
     /usr/include
     /usr/local/include
-    $ENV{MKLROOT}/include
-    $ENV{LAPACKE_HOME}/include
+    NO_DEFAULT_PATH  # Avoids searching in default paths
   )
+  include_directories(${BLAS_INCLUDE_DIR})
+  # Linking BLAS and LAPACK libraries
+  link_libraries(${BLAS_LIBRARIES} ${LAPACK_LIBRARIES})
 
+else()
+  message(FATAL_ERROR "BLAS and LAPACK not found. Please install them.")
 endif()
 find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
+find_package(pybind11 CONFIG)
+include_directories(${pybind11_INCLUDE_DIRS})
 include_directories(${Python3_INCLUDE_DIRS})
 include_directories(${UAMMD_INCLUDE} ${BLAS_INCLUDE_DIRS} ${LAPACKE_INCLUDE_DIRS})
 include_directories(include/third_party/LanczosAlgorithm/include)
