From f21b57a87ebfa5a875c4bf02ab5490d7af34894d Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Mon, 14 Jun 2021 14:13:08 +0200
Subject: [PATCH] Remove Werror from compile arguments

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index d15cb17..6ca8e1e 100644
--- a/setup.py
+++ b/setup.py
@@ -242,7 +242,7 @@ def awscrt_ext():
         libraries += ['rt']
 
     if distutils.ccompiler.get_default_compiler() != 'msvc':
-        extra_compile_args += ['-Wextra', '-Werror', '-Wno-strict-aliasing', '-std=gnu99']
+        extra_compile_args += ['-Wextra', '-Wno-strict-aliasing', '-std=gnu99']
 
     return setuptools.Extension(
         '_awscrt',
-- 
2.30.1 (Apple Git-130)

