From dd8a8d1118ce1e31d675331398655de2e220fae7 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Thu, 9 May 2019 19:11:20 +0200
Subject: [PATCH] Patch rootPath to be $CONDA_PREFIX

---
 __init__.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/__init__.py b/__init__.py
index 95e166024..72b321ca6 100755
--- a/__init__.py
+++ b/__init__.py
@@ -69,6 +69,7 @@ import sys
 import os
 import platform as pyPlatform
 from pkgutil import extend_path
+import distutils.sysconfig
 __path__ = extend_path(__path__, __name__)
 
 # Set the environment variable such that openssl accepts proxy cert
@@ -125,7 +126,7 @@ alarmMail = "dirac.alarms@gmail.com"
 # Set rootPath of DIRAC installation
 
 pythonPath = os.path.realpath(__path__[0])
-rootPath = os.path.dirname(pythonPath)
+rootPath = os.path.dirname(distutils.sysconfig.project_base)
 
 # Import DIRAC.Core.Utils modules
 
-- 
2.21.0

