diff --git a/CMakeLists.txt b/CMakeLists.txt
index da1654d..8ed271b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ set( CMAKE_INSTALL_LIBDIR ${CMAKE_BINARY_DIR}/lib )  # Needed by ROOT_GENERATE_D
 set(FastFrames_INCLUDE_DIRS "${CMAKE_INSTALL_PREFIX}/include" CACHE STRING "Path to the includes")

 # Set the warning flag(s) to use.
-set( CMAKE_CXX_FLAGS "-Wall -Wextra -Wshadow -pedantic -O3 -g" )
+set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wshadow -pedantic -O3 -g" )

 set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)

diff --git a/FastFrames/MainFrame.h b/FastFrames/MainFrame.h
--- a/FastFrames/MainFrame.h
+++ b/FastFrames/MainFrame.h
@@ -17,6 +17,8 @@
 #include "ROOT/RDataFrame.hxx"
 #include "TClass.h"
+#include "TChain.h"
+#include "TFile.h"

 #include <memory>
 #include <string>
