From 0e0a52be6902d2ac370f9cb6a9027ed7a801f734 Mon Sep 17 00:00:00 2001
From: Tim Snyder <snyder.tim@gmail.com>
Date: Thu, 21 Jul 2022 22:42:27 +0000
Subject: [PATCH] include omp.h in libzpaq to enable -fopenmp

---
 libzpaq.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libzpaq.cpp b/libzpaq.cpp
index 523e363ec..3cc459d22 100644
--- a/libzpaq.cpp
+++ b/libzpaq.cpp
@@ -29,6 +29,7 @@ See libzpaq.h for additional documentation.
 #include <stdio.h>
 
 #ifdef unix
+#include <omp.h>
 #ifndef NOJIT
 #include <sys/mman.h>
 #endif
-- 
2.35.1

