diff --git a/CMakeLists.txt b/CMakeLists.txt
index 194260df16..65f4e0d18a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,10 +184,10 @@ if (ENABLE_BSON STREQUAL SYSTEM)
            message (FATAL_ERROR "System libbson built without shared library target")
    endif ()
    set (BSON_LIBRARIES mongo::bson_shared)
-   if (NOT TARGET mongo::bson_static)
-           message (FATAL_ERROR "System libbson built without static library target")
-   endif ()
-   set (BSON_STATIC_LIBRARIES mongo::bson_static)
+   # if (NOT TARGET mongo::bson_static)
+   #         message (FATAL_ERROR "System libbson built without static library target")
+   # endif ()
+   # set (BSON_STATIC_LIBRARIES mongo::bson_static)
 endif ()
 
 unset (dist_generated CACHE)

