diff --git a/src/esp/core/managedContainers/ManagedFileBasedContainer.h b/src/esp/core/managedContainers/ManagedFileBasedContainer.h
index c6fba93..5645dc6 100644
--- a/src/esp/core/managedContainers/ManagedFileBasedContainer.h
+++ b/src/esp/core/managedContainers/ManagedFileBasedContainer.h
@@ -76,7 +76,7 @@ class ManagedFileBasedContainer : public ManagedContainer<T, Access> {
       return nullptr;
     }
     // convert doc to const value
-    const io::JsonGenericValue config = docConfig->GetObject();
+    const io::JsonGenericValue& config = *docConfig;
     ManagedFileIOPtr attr = this->buildManagedObjectFromDoc(filename, config);
     return this->postCreateRegister(std::move(attr), registerObject);
   }  // ManagedFileBasedContainer::createObjectFromJSONFile
@@ -106,7 +106,7 @@ class ManagedFileBasedContainer : public ManagedContainer<T, Access> {
       return nullptr;
     }
     // convert doc to const value
-    const io::JsonGenericValue config = docConfig->GetObject();
+    const io::JsonGenericValue& config = *docConfig;
     ManagedFileIOPtr attr = this->buildManagedObjectFromDoc(docName, config);
     return this->postCreateRegister(std::move(attr), registerObject);
   }  // ManagedFileBasedContainer::createObjectFromJSONString
