diff --git a/src/esp/core/managedContainers/ManagedFileBasedContainer.h b/src/esp/core/managedContainers/ManagedFileBasedContainer.h
index 4e5abe8..b052798 100644
--- a/src/esp/core/managedContainers/ManagedFileBasedContainer.h
+++ b/src/esp/core/managedContainers/ManagedFileBasedContainer.h
@@ -78,7 +78,7 @@ class ManagedFileBasedContainer : public ManagedContainer<T, Access> {
       return nullptr;
     }
     // convert doc to const value
-    const io::JsonGenericValue config = docConfig->GetObject();
+    const io::JsonGenericValue& config = *docConfig;
     ManagedFileIOPtr attr = this->buildManagedObjectFromDoc(filename, config);
     attr->setActualFilename(filename);
     // Set attributes' status to saved (i.e. it matches the version on disk)
@@ -111,7 +111,7 @@ class ManagedFileBasedContainer : public ManagedContainer<T, Access> {
       return nullptr;
     }
     // convert doc to const value
-    const io::JsonGenericValue config = docConfig->GetObject();
+    const io::JsonGenericValue& config = *docConfig;
     ManagedFileIOPtr attr = this->buildManagedObjectFromDoc(docName, config);
     // Set attributes' status to saved (i.e. it matches the version on disk)
     // since it was just built from an existing JSON string.
