diff --git a/src/cmake/FindMagnumPlugins.cmake b/src/cmake/FindMagnumPlugins.cmake
index 0eb35e9..a6d38f7 100644
--- a/src/cmake/FindMagnumPlugins.cmake
+++ b/src/cmake/FindMagnumPlugins.cmake
@@ -175,7 +175,7 @@ set(_MAGNUMPLUGINS_PLUGIN_COMPONENTS
     SpirvToolsShaderConverter SpngImporter StanfordImporter
     StanfordSceneConverter StbDxtImageConverter StbImageConverter
     StbImageImporter StbResizeImageConverter StbTrueTypeFont
-    StbVorbisAudioImporter StlImporter UfbxImporter WebPImporter)
+    StbVorbisAudioImporter StlImporter TinyGltfImporter UfbxImporter WebPImporter)
 # Nothing is enabled by default right now
 set(_MAGNUMPLUGINS_IMPLICITLY_ENABLED_COMPONENTS )
 
diff --git a/src/cmake/dependencies.cmake b/src/cmake/dependencies.cmake
index 62e8d4e..ee57526 100644
--- a/src/cmake/dependencies.cmake
+++ b/src/cmake/dependencies.cmake
@@ -361,7 +361,7 @@ if(NOT USE_SYSTEM_MAGNUM)
       Magnum::AnySceneImporter
       MagnumPlugins::AssimpImporter
       MagnumPlugins::BasisImporter
-      MagnumPlugins::GltfImporter
+      MagnumPlugins::TinyGltfImporter
       MagnumPlugins::StbImageConverter
       MagnumPlugins::StbImageImporter
     )
diff --git a/src/esp/assets/CMakeLists.txt b/src/esp/assets/CMakeLists.txt
index 272cbf6..fa05093 100644
--- a/src/esp/assets/CMakeLists.txt
+++ b/src/esp/assets/CMakeLists.txt
@@ -42,7 +42,7 @@ find_package(
   MagnumPlugins
   REQUIRED
   BasisImporter
-  GltfImporter
+  TinyGltfImporter
   PrimitiveImporter
   StanfordImporter
   StbImageImporter
@@ -80,7 +80,7 @@ target_link_libraries(
          Magnum::Trade
          Magnum::Primitives
          MagnumPlugins::BasisImporter
-         MagnumPlugins::GltfImporter
+         MagnumPlugins::TinyGltfImporter
          MagnumPlugins::PrimitiveImporter
          MagnumPlugins::StanfordImporter
          MagnumPlugins::StbImageImporter
diff --git a/src/esp/gfx/CMakeLists.txt b/src/esp/gfx/CMakeLists.txt
index 08c257b..880cfef 100644
--- a/src/esp/gfx/CMakeLists.txt
+++ b/src/esp/gfx/CMakeLists.txt
@@ -92,7 +92,7 @@ find_package(
   AnyImageConverter
 )
 
-find_package(MagnumPlugins REQUIRED GltfImporter StbImageImporter StbImageConverter)
+find_package(MagnumPlugins REQUIRED TinyGltfImporter StbImageImporter StbImageConverter)
 
 find_package(MagnumIntegration REQUIRED Eigen)
 
@@ -125,7 +125,7 @@ target_link_libraries(
          Magnum::SceneGraph
          Magnum::Shaders
          Magnum::Trade
-         MagnumPlugins::GltfImporter
+         MagnumPlugins::TinyGltfImporter
          MagnumPlugins::StbImageImporter
          MagnumPlugins::StbImageConverter
          MagnumIntegration::Eigen
diff --git a/src/esp/physics/CMakeLists.txt b/src/esp/physics/CMakeLists.txt
index 02cddcf..be25506 100644
--- a/src/esp/physics/CMakeLists.txt
+++ b/src/esp/physics/CMakeLists.txt
@@ -2,7 +2,7 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-find_package(MagnumPlugins REQUIRED GltfImporter StbImageImporter StbImageConverter)
+find_package(MagnumPlugins REQUIRED TinyGltfImporter StbImageImporter StbImageConverter)
 
 set(ESP_DEFAULT_PHYSICS_CONFIG_REL_PATH ./data/default.physics_config.json)
 set(ESP_DEFAULT_PHYSICS_CONFIG
@@ -50,7 +50,7 @@ target_link_libraries(
   PUBLIC core
          scene
          assets
-         MagnumPlugins::GltfImporter
+         MagnumPlugins::TinyGltfImporter
          MagnumPlugins::StbImageImporter
          MagnumPlugins::StbImageConverter
 )
