From 16c3776fffcf0447e21f65d1b661f88042b446e2 Mon Sep 17 00:00:00 2001
From: past-due <30942300+past-due@users.noreply.github.com>
Date: Fri, 27 Apr 2018 12:55:53 -0400
Subject: [PATCH 1/4] [CMake] Use target_compile_definitions to avoid affecting
 global definitions (#121)

---
 CMakeLists.txt | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ea3bd1..cbc8d93 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required (VERSION 2.8.12)
 
 include (utils.cmake)
 
@@ -13,10 +13,6 @@ set(SO_MAJOR 2)
 set(SO_MINOR 1)
 set(SO_PATCH 1)
 
-add_definitions (
-  -DUTF8PROC_EXPORTS
-)
-
 if (NOT MSVC)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2 -std=c99 -pedantic -Wall")
 endif ()
@@ -26,6 +22,8 @@ add_library (utf8proc
   utf8proc.h
 )
 
+target_compile_definitions(utf8proc PRIVATE "UTF8PROC_EXPORTS")
+
 set_target_properties (utf8proc PROPERTIES
   POSITION_INDEPENDENT_CODE ON
   VERSION "${SO_MAJOR}.${SO_MINOR}.${SO_PATCH}"
-- 
2.17.0

