From 70170aa40a9ed71bd2fd74f99ce59a9e60015cdd Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 30 Sep 2019 11:28:02 +0200
Subject: [PATCH 2/2] use cx_freeze on windows

---
 setup.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 0860844..36f31df 100644
--- a/setup.py
+++ b/setup.py
@@ -66,11 +66,7 @@ if "HOME" not in os.environ and "HOMEPATH" in os.environ:
     os.environ.setdefault("HOME", os.environ.get("HOMEPATH", ""))
     print("Initializing HOME environment variable to '{}'".format(os.environ["HOME"]))
 
-use_cx_freeze = False
-for cmd in ["bdist_msi"]:
-    if cmd in sys.argv:
-        use_cx_freeze = True
-        break
+use_cx_freeze = (os.name == "nt")
 
 # CherryPy is required for the tests and benchmarks. It is also the preferrred
 # server for the stand-alone mode (`wsgidav.server.server_cli.py`).
-- 
2.21.0.windows.1

