From 5b5083cfb360bfccf38ef8abf8dd391900fa9f55 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Sat, 3 Apr 2021 18:51:11 +0200
Subject: [PATCH] Fix BUILD_PYTHON_MODULE option on Windows/MSVC

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb128c3..285d91e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,12 @@ if(BUILD_PYTHON_MODULE)
   pybind11_add_module(python_ruckig src/python.cpp)
   target_compile_features(python_ruckig PUBLIC cxx_std_17)
   target_link_libraries(python_ruckig PUBLIC ruckig)
-  set_target_properties(python_ruckig PROPERTIES OUTPUT_NAME ruckig)
+  if(NOT MSVC)
+    set_target_properties(python_ruckig PROPERTIES OUTPUT_NAME ruckig)
+  else()
+    set_target_properties(python_ruckig PROPERTIES RUNTIME_OUTPUT_NAME ruckig)
+    set_target_properties(python_ruckig PROPERTIES LIBRARY_OUTPUT_NAME ruckig)
+  endif()
 endif()
 
 
 
