From c98d58701a36bbed58eecff51c71851fdafa27e1 Mon Sep 17 00:00:00 2001
From: mattkram <mkramer@anaconda.com>
Date: Mon, 18 Sep 2023 14:40:22 -0500
Subject: [PATCH] Replace lib HYPRE_DIR with PREFIX, and patch EIGEN_DIR to use
 conda path

Replace prefix
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 0383bc78..5855fc2c 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,8 @@ CXX      := mpicxx
 
 OBJ_DIR   := $(BUILD)
 APP_DIR   := $(BIN)
-HYPRE_DIR := /usr/local/hypre
-EIGEN_DIR := ThirdParty/eigen-3.3.8 
+HYPRE_DIR := ${PREFIX}
+EIGEN_DIR := ${PREFIX}/include/eigen3
 CXXFLAGS := -w -std=c++11 -O3 
 LDFLAGS  := -L ${HYPRE_DIR}/lib/ -lHYPRE
 INCLUDE  := -I ${HYPRE_DIR}/include -I ${EIGEN_DIR} -DEIGEN_MPL2_ONLY 
-- 
2.39.2 (Apple Git-143)

