From d33c674bc85ad5937d6cd668efe7be292fb37bef Mon Sep 17 00:00:00 2001
From: Marius van Niekerk <marius.v.niekerk@gmail.com>
Date: Thu, 17 Sep 2020 13:42:34 -0400
Subject: [PATCH] Removed ez_setup from the setup process

---
 setup.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/setup.py b/setup.py
index 05e253e..45e692e 100644
--- a/setup.py
+++ b/setup.py
@@ -330,10 +330,6 @@ extra = {
     'use_2to3': False
 }
 
-from ez_setup import use_setuptools
-
-use_setuptools()
-
 from setuptools import setup
 
 VERSION = "0.9.17"
-- 
2.27.0

