--- msgpack-c-cpp-2.1.0/test/msgpack_c.cpp.orig	2017-01-11 01:07:17.261234842 +0100
+++ msgpack-c-cpp-2.1.0/test/msgpack_c.cpp	2017-01-11 01:08:07.919005825 +0100
@@ -1,5 +1,6 @@
 #include "msgpack.h"
 
+#include <cmath>
 #include <math.h>
 #include <vector>
 #include <limits>
@@ -226,16 +227,16 @@
             msgpack_unpack(sbuf.data, sbuf.size, NULL, &z, &obj);
         EXPECT_EQ(MSGPACK_UNPACK_SUCCESS, ret);
         EXPECT_EQ(MSGPACK_OBJECT_FLOAT32, obj.type);
-        if (isnan(val)) {
-            EXPECT_TRUE(isnan(obj.via.f64));
+        if (std::isnan(val)) {
+            EXPECT_TRUE(std::isnan(obj.via.f64));
 #if defined(MSGPACK_USE_LEGACY_NAME_AS_FLOAT)
-            EXPECT_TRUE(isnan(obj.via.dec));
+            EXPECT_TRUE(std::isnan(obj.via.dec));
 #endif // MSGPACK_USE_LEGACY_NAME_AS_FLOAT
         }
-        else if (isinf(val)) {
-            EXPECT_TRUE(isinf(obj.via.f64));
+        else if (std::isinf(val)) {
+            EXPECT_TRUE(std::isinf(obj.via.f64));
 #if defined(MSGPACK_USE_LEGACY_NAME_AS_FLOAT)
-            EXPECT_TRUE(isinf(obj.via.dec));
+            EXPECT_TRUE(std::isinf(obj.via.dec));
 #endif // MSGPACK_USE_LEGACY_NAME_AS_FLOAT
         }
         else {
@@ -292,16 +293,16 @@
 #if defined(MSGPACK_USE_LEGACY_NAME_AS_FLOAT)
         EXPECT_EQ(MSGPACK_OBJECT_DOUBLE, obj.type);
 #endif // MSGPACK_USE_LEGACY_NAME_AS_FLOAT
-        if (isnan(val)) {
-            EXPECT_TRUE(isnan(obj.via.f64));
+        if (std::isnan(val)) {
+            EXPECT_TRUE(std::isnan(obj.via.f64));
 #if defined(MSGPACK_USE_LEGACY_NAME_AS_FLOAT)
-            EXPECT_TRUE(isnan(obj.via.dec));
+            EXPECT_TRUE(std::isnan(obj.via.dec));
 #endif // MSGPACK_USE_LEGACY_NAME_AS_FLOAT
         }
-        else if (isinf(val)) {
-            EXPECT_TRUE(isinf(obj.via.f64));
+        else if (std::isinf(val)) {
+            EXPECT_TRUE(std::isinf(obj.via.f64));
 #if defined(MSGPACK_USE_LEGACY_NAME_AS_FLOAT)
-            EXPECT_TRUE(isinf(obj.via.dec));
+            EXPECT_TRUE(std::isinf(obj.via.dec));
 #endif // MSGPACK_USE_LEGACY_NAME_AS_FLOAT
         }
         else {
