--- src/pj_strtod.c	2016-01-05 17:01:09.000000000 -0300
+++ src/pj_strtod.c	2016-01-06 17:21:44.144281743 -0300
@@ -76,7 +76,13 @@
                                               char* pszWorkBuffer)
 {
 #if !defined(HAVE_LOCALECONV) || defined(_WIN32_WCE)
+
+#if defined(_MSC_VER)  /* Visual C++ */
+#pragma message("localeconv not available")
+#else
 #warning "localeconv not available"
+#endif
+
     static char byPoint = 0;
     if (byPoint == 0)
     {
