--- configure	2018-09-28 09:05:20.000000000 +0100
+++ configure	2019-02-06 16:21:17.000000000 +0000
@@ -14787,24 +14787,8 @@
 # on Debian the file is /usr/lib/x86_64-linux-gnu/libpython2.7.so
 # on SL7 the file is /usr/lib64/libpython2.7.so

-PYTHON_LIB_SUFFIX=so
-
-if test -a "/usr/lib/x86_64-linux-gnu/libpython2.7.so" ; then
-	PYTHON_LIB_LOC=/usr/lib/x86_64-linux-gnu
-
-elif test -a "/usr/lib64/libpython2.7.so" ; then
-	PYTHON_LIB_LOC=/usr/lib64
-
-elif test -a "/opt/local/lib/libpython2.7.dylib" ; then
-	PYTHON_LIB_LOC=/opt/local/lib
-
-	PYTHON_LIB_SUFFIX=dylib
-
-	# FIXME:  .pc file broken on macs, remove when fixed
-	PYGOBJECT_LIBS="${PYGOBJECT_LIBS} -lgmodule-2.0"
-else
-	as_fn_error $? "libpython2.7.so not found" "$LINENO" 5
-fi
+PYTHON_LIB_SUFFIX=`${SHLIB_EXT} | cut -c2-`
+PYTHON_LIB_LOC=${PREFIX}/lib


 #
