From 7efe0e41c60da2d885e540a69f45821519064679 Mon Sep 17 00:00:00 2001
From: Tom de Geus <tom@geus.me>
Date: Sun, 24 Nov 2019 15:18:32 +0100
Subject: [PATCH] Making simd optional for Python installation

---
 python/main.cpp |  2 --
 setup.py        | 27 +++++++++++++++++++--------
 2 files changed, 19 insertions(+), 10 deletions(-)

diff --git a/python/main.cpp b/python/main.cpp
index ab985d4..038362e 100644
--- a/python/main.cpp
+++ b/python/main.cpp
@@ -4,8 +4,6 @@
 
 ================================================================================================= */
 
-#define XTENSOR_USE_XSIMD
-
 #include <pybind11/pybind11.h>
 #include <pyxtensor/pyxtensor.hpp>
 
diff --git a/setup.py b/setup.py
index 4cfd15e..534fb6a 100644
--- a/setup.py
+++ b/setup.py
@@ -17,17 +17,28 @@ patch = re.split(r'(.*)(\#define GOOSEEYE_VERSION_PATCH\ )([0-9]+)(.*)', header)
 
 __version__ = '.'.join([major, minor, patch])
 
+include_dirs = [
+  os.path.abspath('include/'),
+  pyxtensor.find_pyxtensor(),
+  pyxtensor.find_pybind11(),
+  pyxtensor.find_xtensor(),
+  pyxtensor.find_xtl()]
+
+build = pyxtensor.BuildExt
+
+xsimd = pyxtensor.find_xsimd()
+if xsimd:
+  if len(xsimd) > 0:
+    include_dirs += [xsimd]
+    build.c_opts['unix'] += ['-march=native', '-DXTENSOR_USE_XSIMD']
+    build.c_opts['msvc'] += ['/DXTENSOR_USE_XSIMD']
+
 ext_modules = [
   Extension(
     'GooseEYE',
     ['python/main.cpp'],
-    include_dirs=[
-      os.path.abspath('include/'),
-      pyxtensor.find_pyxtensor(),
-      pyxtensor.find_pybind11(),
-      pyxtensor.find_xtensor(),
-      pyxtensor.find_xtl()],
-    language='c++'
+    include_dirs = include_dirs,
+    language = 'c++'
   ),
 ]
 
@@ -43,6 +54,6 @@ setup(
   url = 'https://github.com/tdegeus/GooseEYE',
   ext_modules = ext_modules,
   install_requires = ['pybind11>=2.2.0', 'pyxtensor>=0.1.1'],
-  cmdclass = {'build_ext': pyxtensor.BuildExt},
+  cmdclass = {'build_ext': build},
   zip_safe = False,
 )
-- 
2.21.0 (Apple Git-122.2)

