diff --git a/Molecule/Makefile b/Molecule/Makefile
index f769b37..18edbfc 100644
--- a/Molecule/Makefile
+++ b/Molecule/Makefile
@@ -1,18 +1,5 @@
-#Intel Compiler
+LD = $(CXX)
 
-#CXX = icpc
-#CC = icc
-#LD = icpc
-#CXXFLAGS = -O2 -I . -I ../include -ffast-math -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-
-# Clang
-# CXX=clang++-9
-# CC=clang-9
-# LD=clang++-9
-
-CC=gcc
-CXX = g++
-LD = g++
 CXXFLAGS = -O2 -I . -I ../include -ffast-math -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
 
 CXXFLAGS = -O2 -I . -I ../include -ffast-math -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DIW_IMPLEMENTATIONS_EXPOSED
@@ -28,7 +15,7 @@ COMMON_OBJECTS = molecule.o moleculeb.o moleculeh.o element.o atom.o bond.o bond
 	rwsubstructure.o substructure_nmab.o molecule_to_query.o is_actually_chiral.o tokenise_atomic_smarts.o temp_detach_atoms.o path_scoring.o \
 	element_hits_needed.o numass.o misc2.o standardise.o toggle_kekule_form.o
 
-LIBDIRS = -L../lib/ -L../supporting_libraries
+LIBDIRS = -L../lib/ -L../supporting_libraries -L$(PREFIX)/lib
 
 MC_FIRST_PASS_OBJECTS = mc_first_pass.o $(COMMON_OBJECTS)
 
diff --git a/supporting_libraries/Makefile b/supporting_libraries/Makefile
index 741c793..659876f 100644
--- a/supporting_libraries/Makefile
+++ b/supporting_libraries/Makefile
@@ -1,14 +1,3 @@
-#CXX = icpc
-#CC = icc
-#CXXFLAGS = -O2 -I . -I ../include -Wall -ffast-math -DLINUX -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-
-# CXX=clang++-9
-# CC=clang-9
-# LD=clang++-9
-
-CXX = g++
-CC = gcc
-
 CXXFLAGS = -O2 -I . -I ../include -Wall -ffast-math -DLINUX -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DIW_IMPLEMENTATIONS_EXPOSED
 
 
