--- setup.py	2016-01-29 00:18:58.000000000 +0000
+++ setup.py	2016-03-01 22:35:55.000000000 +0000
@@ -62,7 +62,7 @@
             self.openssl = 'c:\\pkg'
         else:
             self.libraries = ['ssl', 'crypto']
-            self.openssl = '/usr'
+            self.openssl = os.environ["OPENSSL_PATH"]

     def finalize_options(self):
         '''Overloaded build_ext implementation to append custom openssl
