From 832d6a7d64c245266c419a2d63937b68d91ccd52 Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Thu, 18 Dec 2025 19:26:42 +0100
Subject: [PATCH] fix installation of configured config.hpp file

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 502a47d..52ff847 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,14 +102,14 @@ add_library(tight_inclusion::tight_inclusion ALIAS tight_inclusion)
 # Fill in configuration options
 configure_file(
   "${TIGHT_INCLUSION_SOURCE_DIR}/config.hpp.in"
-  "${TIGHT_INCLUSION_SOURCE_DIR}/config.hpp")
+  "${CMAKE_CURRENT_BINARY_DIR}/src/tight_inclusion/config.hpp" @ONLY)
 
 # Add source and header files to tight_inclusion
 add_subdirectory("${TIGHT_INCLUSION_SOURCE_DIR}")
 
 # Public include directory for Tight Inclusion
 target_include_directories(tight_inclusion PUBLIC $<BUILD_INTERFACE:${TIGHT_INCLUSION_INCLUDE_DIR}>)
-target_include_directories(tight_inclusion PUBLIC $<INSTALL_INTERFACE:include>) #Include the generated metis file
+target_include_directories(tight_inclusion PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/src>)
 
 # Configure cmake config file
 include(CMakePackageConfigHelpers)
@@ -181,6 +181,7 @@ install(TARGETS tight_inclusion
         RUNTIME DESTINATION "bin" COMPONENT applications
         LIBRARY DESTINATION "lib" COMPONENT libraries
         ARCHIVE DESTINATION "lib" COMPONENT libraries)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/src/tight_inclusion/config.hpp DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tight_inclusion)        
 install(EXPORT ${PROJECT_NAME}Targets FILE ${PROJECT_NAME}Targets.cmake DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
 install(FILES
           ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
-- 
2.43.0

