diff --git a/cmake/templates/pkgConfig.cmake.in b/cmake/templates/pkgConfig.cmake.in
index 9c0e46d..65b228c 100644
--- a/cmake/templates/pkgConfig.cmake.in
+++ b/cmake/templates/pkgConfig.cmake.in
@@ -119,7 +119,7 @@ endif()
 set(libraries "@PKG_CONFIG_LIBRARIES@")
 foreach(library ${libraries})
   # keep build configuration keywords, target names and absolute libraries as-is
-  if("${library}" MATCHES "^(debug|optimized|general)$")
+  if("${library}" MATCHES "^(debug|optimized|general|rt|pthread|dl)$")
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
   elseif(${library} MATCHES "^-l")
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
diff --git a/cmake/tools/rt.cmake b/cmake/tools/rt.cmake
index 3e5b708..d6141ee 100644
--- a/cmake/tools/rt.cmake
+++ b/cmake/tools/rt.cmake
@@ -38,8 +38,8 @@ if(NOT (APPLE OR WIN32 OR MINGW OR ANDROID))
     # depend on the linker to find it for us.
     set(RT_LIBRARY rt CACHE FILEPATH "Hacked find of rt for cmake < 2.8.4")
   else()
-    find_library(RT_LIBRARY rt)
-    assert_file_exists(${RT_LIBRARY} "RT Library")
+    set(RT_LIBRARY rt)
+    # assert_file_exists(${RT_LIBRARY} "RT Library")
   endif()
   #message(STATUS "RT_LIBRARY: ${RT_LIBRARY}")
 endif()