diff --git a/setup.sh b/setup.sh
index 6246eef..5784020 100644
--- a/setup.sh
+++ b/setup.sh
@@ -11,8 +11,17 @@
 
 # since this file is sourced either use the provided _CATKIN_SETUP_DIR
 # or fall back to the destination set at configure time
-: ${_CATKIN_SETUP_DIR:=ABCABCPLACEHOLDERABCABC}
-_SETUP_UTIL="$_CATKIN_SETUP_DIR/_setup_util.py"
+
+if [ -n "`$SHELL -c 'echo $ZSH_VERSION'`" ]; then
+  CATKIN_SHELL=zsh
+elif [ -n "`$SHELL -c 'echo $BASH_VERSION'`" ]; then
+  CATKIN_SHELL=bash
+else
+  echo "could not detect your shell"
+fi
+
+: ${_CATKIN_SETUP_DIR:=$CONDA_PREFIX/etc/conda/activate.d/}
+_SETUP_UTIL="$_CATKIN_SETUP_DIR/ros_setup_util.py"
 unset _CATKIN_SETUP_DIR
 
 if [ ! -f "$_SETUP_UTIL" ]; then
