diff --git a/_setup_util.py b/_setup_util.py
index 3c23425..df04ed0 100755
--- a/_setup_util.py
+++ b/_setup_util.py
@@ -269,14 +269,14 @@ if __name__ == '__main__':
             # don't consider any other prefix path than this one
             CMAKE_PREFIX_PATH = []
         # prepend current workspace if not already part of CPP
-        base_path = os.path.dirname(__file__)
+        # base_path = os.path.dirname(__file__)
         # CMAKE_PREFIX_PATH uses forward slash on all platforms, but __file__ is platform dependent
         # base_path on Windows contains backward slashes, need to be converted to forward slashes before comparison
-        if os.path.sep != '/':
-            base_path = base_path.replace(os.path.sep, '/')
+        # if os.path.sep != '/':
+        #     base_path = base_path.replace(os.path.sep, '/')
 
-        if base_path not in CMAKE_PREFIX_PATH:
-            CMAKE_PREFIX_PATH.insert(0, base_path)
+        # if base_path not in CMAKE_PREFIX_PATH:
+        #    CMAKE_PREFIX_PATH.insert(0, base_path)
         CMAKE_PREFIX_PATH = os.pathsep.join(CMAKE_PREFIX_PATH)
 
         environ = dict(os.environ)
