diff --git a/cmake/FindGSL.cmake b/cmake/FindGSL.cmake
index c146f5d..081fa15 100755
--- a/cmake/FindGSL.cmake
+++ b/cmake/FindGSL.cmake
@@ -164,7 +164,9 @@ else()
         if("${lib_arg_begin}" STREQUAL "-l")
           string(REPLACE "-l" "" lib_name "${lib_arg}")
           find_library(GSL_LIB "${lib_name}" HINTS "${GSL_LIBRARY_DIRS}")
-          list(APPEND GSL_LIBRARIES "${GSL_LIB}")
+          if (GSL_LIB)
+            list(APPEND GSL_LIBRARIES "${GSL_LIB}")
+          endif(GSL_LIB)
           unset(GSL_LIB CACHE)
         endif()
       endforeach()
