diff --git a/interfaces/Python/CMakeLists.txt b/interfaces/Python/CMakeLists.txt
index 3745413..25ffa22 100644
--- a/interfaces/Python/CMakeLists.txt
+++ b/interfaces/Python/CMakeLists.txt
@@ -7,7 +7,12 @@ if(DAKOTA_ENABLE_TESTS AND PYTHONINTERP_FOUND AND
 	  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
   set_property(TEST dakota_interfacing_test PROPERTY LABELS UnitTest)
 endif()
-
-install(DIRECTORY "dakota" DESTINATION "share/dakota/Python/")
+if(PYTHON_EXECUTABLE)
+  EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c 
+  "from distutils.sysconfig import get_python_lib; print(get_python_lib())"
+  OUTPUT_VARIABLE python_libs OUTPUT_STRIP_TRAILING_WHITESPACE )
+  SET(PYTHON_INSTALL_DIR ${python_libs})
+  install(DIRECTORY "dakota" DESTINATION ${PYTHON_INSTALL_DIR})
+endif(PYTHON_EXECUTABLE)
 install(PROGRAMS "mpitile" DESTINATION "bin/")
 
