diff --git a/reproc/CMakeLists.txt b/reproc/CMakeLists.txt
index 30c6e22..00cb951 100644
--- a/reproc/CMakeLists.txt
+++ b/reproc/CMakeLists.txt
@@ -4,6 +4,11 @@ endif()
 
 reproc_library(reproc C)
 
+if(UNIX AND NOT APPLE)
+  target_link_libraries(reproc PRIVATE rt)
+endif()
+
+
 if(REPROC_MULTITHREADED)
   target_compile_definitions(reproc PRIVATE REPROC_MULTITHREADED)
   target_link_libraries(reproc PRIVATE Threads::Threads)
