diff --git a/CMakeLists.txt b/CMakeLists.txt
index 57b3bab..a3e53ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,10 +111,6 @@ if(WIN32)
   find_package(OpenSSL REQUIRED)
 endif()
 
-# Add boost as a dependency
-# We use the header-only version of boost::json: https://www.boost.org/doc/libs/1_87_0/libs/json/doc/html/json/overview.html
-find_package(Boost 1.85 REQUIRED COMPONENTS json)
-
 # ----------------------------- Download Apache Datasketches -------------------------
 include(ExternalProject)
 ExternalProject_Add(datasketches
