--- a/setup.py
+++ b/setup.py
@@ -9,18 +9,6 @@ import sys


 def _compile_ccl():
-    call(["mkdir", "-p", "build"])
-    v = sys.version_info
-    if call(["cmake", "-H.", "-Bbuild",
-             "-DPYTHON_VERSION=%d.%d.%d" % (
-                v.major, v.minor, v.micro)]) != 0:
-        raise Exception(
-            "Could not run CMake configuration. Make sure "
-            "CMake is installed !")
-
-    if call(["make", "-Cbuild", "_ccllib"]) != 0:
-        raise Exception("Could not build CCL")
-
     # Finds the library under its different possible names
     if os.path.exists("build/pyccl/_ccllib.so"):
         call(["cp", "build/pyccl/_ccllib.so", "pyccl/"])
@@ -61,8 +62,6 @@
     provides=['pyccl'],
     package_data={'pyccl': ['_ccllib.so']},
     include_package_data=True,
-    use_scm_version=True,
-    setup_requires=['setuptools_scm'],
     install_requires=['numpy', 'pyyaml'],
     cmdclass={'build_py': build, 'develop': develop},
     classifiers=[
