From d49cdb06da217d3849f7c7dd0b9bb760658990a1 Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Sat, 8 Jan 2022 15:45:29 -0500
Subject: [PATCH 3/4] add ldflags in lib

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 446573b..b584776 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -47,7 +47,7 @@ solib: ../libqrupdate$(SOEXT)
 	ar -cr $@ $(OBJS)
 
 ../libqrupdate.so: $(OBJS)
-	$(FC) $(FFLAGS) -shared -o $@ -Wl,-soname=libqrupdate.so.$(MAJOR) $(OBJS) \
+	$(FC) $(FFLAGS) $(LDFLAGS) -shared -o $@ -Wl,-soname=libqrupdate.so.$(MAJOR) $(OBJS) \
 		$(BLAS) $(LAPACK)
 
 ../libqrupdate.dylib: $(OBJS)
-- 
2.34.1

