From 623179e6e2e57f1179b5b4ece7a454e0cf461f37 Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Sat, 8 Jan 2022 15:55:39 -0500
Subject: [PATCH 4/4] point to lib

---
 Makeconf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makeconf b/Makeconf
index a3722ea..1006a93 100644
--- a/Makeconf
+++ b/Makeconf
@@ -15,7 +15,7 @@ VERSION=1.1
 MAJOR=1
 
 # The default library dir
-LIBDIR=lib
+LIBDIR=lib  # test
 
 # Destination installation offset
 DESTDIR=
-- 
2.34.1

