From 13cafa4bdcff43d197c5e6f6f88af9f1d2171b27 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Fri, 18 Oct 2019 15:42:31 +0200
Subject: [PATCH] search for libclang.lib on Windows

---
 external/external.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/external/external.cmake b/external/external.cmake
index 8a74eb8..d6b08f6 100644
--- a/external/cppast/external/external.cmake
+++ b/external/cppast/external/external.cmake
@@ -176,7 +176,7 @@ function(_cppast_find_libclang config_tool min_version force)
     if(NOT LIBCLANG_LIBRARY)
         execute_process(COMMAND ${config_tool} --libdir
                         OUTPUT_VARIABLE llvm_library_dir OUTPUT_STRIP_TRAILING_WHITESPACE)
-        find_library(LIBCLANG_LIBRARY "clang" "${llvm_library_dir}" NO_DEFAULT_PATH)
+        find_library(LIBCLANG_LIBRARY NAMES clang libclang HINTS "${llvm_library_dir}")
 
         if(NOT LIBCLANG_LIBRARY)
             message(FATAL_ERROR "libclang library not found")
-- 
2.23.0

