--- configure	2019-02-07 10:47:47.896493760 +0300
+++ configure.new	2019-02-07 10:55:19.239300776 +0300
@@ -28,8 +28,12 @@
     IFS=$sifs
 }
 
-findprog python
-python=$prog
+if test -n "${PYTHON}"; then
+    python="${PYTHON}"
+else
+    findprog python
+    python=$prog
+fi
 echo "checking for Python ... $python"
 if test "$python" = "no"; then
     fatal_error "Please install Python before proceeding: http://www.python.org"
