--- framework/configure.py	2018-08-13 15:21:32.938119682 +0300
+++ framework/configure.py_new	2018-08-13 15:24:25.334921430 +0300
@@ -2186,7 +2186,7 @@
         'that cannot be built with -static-intel',
         'yellow_on_red')
     context.Message("checking for SWIG ... ")
-    if 'swig' in Environment().get('TOOLS'):
+    if 'swig' in context.env.get('TOOLS'):
         swigx = WhereIs('swig')
         context.Result(swigx)
         context.env['SWIG'] = swigx
