--- SConstruct	2018-07-09 15:30:42.000000000 +0300
+++ SConstruct_new	2018-08-13 15:34:49.574086908 +0300
@@ -4,8 +4,6 @@
 sys.path.insert(0,'./framework')
 import bldutil, configure, setenv, rsf.doc
 
-env = Environment()
-
 if os.path.isfile('config.py'):
     import config
     root = config.RSFROOT
@@ -31,7 +29,8 @@
 
 opts = configure.options('config.py')
 opts.Add('RSFROOT','RSF installation root',root)
-opts.Update(env)
+
+env = Environment(variables=opts)
 
 if not os.path.isfile('config.py'):
     conf = Configure(env,custom_tests={'CheckAll':configure.check_all})
