diff --git a/cmake_modules/FindGLog.cmake b/cmake_modules/FindGLog.cmake
index c9876b7af..f6cb39748 100644
--- a/cmake_modules/FindGLog.cmake
+++ b/cmake_modules/FindGLog.cmake
@@ -22,16 +22,9 @@
 #  GLOG_STATIC_LIB, path to libglog's static library
 #  GLOG_FOUND, whether glog has been found
 
-find_path(GLOG_INCLUDE_DIR glog/logging.h
-  # make sure we don't accidentally pick up a different version
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-find_library(GLOG_SHARED_LIB glog
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-find_library(GLOG_STATIC_LIB libglog.a
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
+find_path(GLOG_INCLUDE_DIR glog/logging.h)
+find_library(GLOG_SHARED_LIB glog)
+find_library(GLOG_STATIC_LIB libglog.a)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(GLOG REQUIRED_VARS
