diff --git a/cmake_modules/FindGMock.cmake b/cmake_modules/FindGMock.cmake
index 7b6eb70..7531f84 100644
--- a/cmake_modules/FindGMock.cmake
+++ b/cmake_modules/FindGMock.cmake
@@ -31,26 +31,10 @@
 # GMOCK_STATIC_LIBRARY, Location of libgmock's static library
 # GMOCK_FOUND, If false, do not try to use gmock.
 
-find_path(GMOCK_INCLUDE_DIR gmock/gmock.h
-          DOC   "Path to the gmock header file"
-          NO_CMAKE_SYSTEM_PATH
-          NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_path(GTEST_INCLUDE_DIR gtest/gtest.h
-          DOC   "Path to the gtest header file"
-          NO_CMAKE_SYSTEM_PATH
-          NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_library(GMOCK_SHARED_LIBRARY gmock
-             DOC   "Google's framework for writing C++ tests (gmock)"
-             NO_CMAKE_SYSTEM_PATH
-             NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_library(GMOCK_STATIC_LIBRARY libgmock.a
-             DOC   "Google's framework for writing C++ tests (gmock) static"
-             NO_CMAKE_SYSTEM_PATH
-             NO_SYSTEM_ENVIRONMENT_PATH)
-
+find_path(GMOCK_INCLUDE_DIR gmock/gmock.h)
+find_path(GTEST_INCLUDE_DIR gtest/gtest.h)
+find_library(GMOCK_SHARED_LIBRARY gmock)
+find_library(GMOCK_STATIC_LIBRARY libgmock.a)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(GMOCK REQUIRED_VARS
