diff --git a/cmake_modules/FindProtobuf.cmake b/cmake_modules/FindProtobuf.cmake
index 08332ea..a1d8914 100644
--- a/cmake_modules/FindProtobuf.cmake
+++ b/cmake_modules/FindProtobuf.cmake
@@ -194,34 +194,12 @@ function(PROTOBUF_GENERATE_CPP SRCS HDRS TGTS)
 endfunction()
 
 
-find_path(PROTOBUF_INCLUDE_DIR google/protobuf/service.h
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_library(PROTOBUF_SHARED_LIBRARY protobuf
-             DOC "The Google Protocol Buffers Library"
-             NO_CMAKE_SYSTEM_PATH
-             NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_library(PROTOBUF_STATIC_LIBRARY libprotobuf.a
-             DOC "Static version of the Google Protocol Buffers Library"
-             NO_CMAKE_SYSTEM_PATH
-             NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_library(PROTOBUF_PROTOC_SHARED_LIBRARY protoc
-             DOC "The Google Protocol Buffers Compiler Library"
-             NO_CMAKE_SYSTEM_PATH
-             NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_library(PROTOBUF_PROTOC_STATIC_LIBRARY libprotoc.a
-             DOC "Static version of the Google Protocol Buffers Compiler Library"
-             NO_CMAKE_SYSTEM_PATH
-             NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_program(PROTOBUF_PROTOC_EXECUTABLE protoc
-             DOC "The Google Protocol Buffers Compiler"
-             NO_CMAKE_SYSTEM_PATH
-             NO_SYSTEM_ENVIRONMENT_PATH)
+find_path(PROTOBUF_INCLUDE_DIR google/protobuf/service.h)
+find_library(PROTOBUF_SHARED_LIBRARY protobuf)
+find_library(PROTOBUF_STATIC_LIBRARY libprotobuf.a)
+find_library(PROTOBUF_PROTOC_SHARED_LIBRARY protoc)
+find_library(PROTOBUF_PROTOC_STATIC_LIBRARY libprotoc.a)
+find_program(PROTOBUF_PROTOC_EXECUTABLE protoc)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(PROTOBUF REQUIRED_VARS
