diff --git a/cmake_modules/FindThrift.cmake b/cmake_modules/FindThrift.cmake
index 3b9b217..5acb98b 100644
--- a/cmake_modules/FindThrift.cmake
+++ b/cmake_modules/FindThrift.cmake
@@ -142,24 +142,10 @@ function(THRIFT_GENERATE_CPP SRCS HDRS TGTS)
   set(${TGTS} ${${TGTS}} PARENT_SCOPE)
 endfunction()
 
-find_path(THRIFT_INCLUDE_DIR thrift/Thrift.h
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_library(THRIFT_SHARED_LIBRARY thrift
-             DOC "The Thrift Library"
-             NO_CMAKE_SYSTEM_PATH
-             NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_library(THRIFT_STATIC_LIBRARY libthrift.a
-  DOC "Static version of the Thrift Library"
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-
-find_program(THRIFT_EXECUTABLE thrift
-  DOC "The Thrift Compiler"
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
+find_path(THRIFT_INCLUDE_DIR thrift/Thrift.h)
+find_library(THRIFT_SHARED_LIBRARY thrift)
+find_library(THRIFT_STATIC_LIBRARY libthrift.a)
+find_program(THRIFT_EXECUTABLE thrift)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(THRIFT REQUIRED_VARS
