diff --git a/cmake_modules/FindGPerf.cmake b/cmake_modules/FindGPerf.cmake
index d3ed697..21e2133 100644
--- a/cmake_modules/FindGPerf.cmake
+++ b/cmake_modules/FindGPerf.cmake
@@ -28,21 +28,11 @@
 #  PROFILER_SHARED_LIB, path to libprofiler's shared library
 #  PROFILER_STATIC_LIB, path to libprofiler's static library
 
-FIND_PATH(GOOGLE_PERFTOOLS_INCLUDE_DIR google/heap-profiler.h
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-FIND_LIBRARY(TCMALLOC_SHARED_LIB tcmalloc
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-FIND_LIBRARY(TCMALLOC_STATIC_LIB libtcmalloc.a
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-FIND_LIBRARY(PROFILER_SHARED_LIB profiler
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-FIND_LIBRARY(PROFILER_STATIC_LIB libprofiler.a
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
+FIND_PATH(GOOGLE_PERFTOOLS_INCLUDE_DIR google/heap-profiler.h)
+FIND_LIBRARY(TCMALLOC_SHARED_LIB tcmalloc)
+FIND_LIBRARY(TCMALLOC_STATIC_LIB libtcmalloc.a)
+FIND_LIBRARY(PROFILER_SHARED_LIB profiler)
+FIND_LIBRARY(PROFILER_STATIC_LIB libprofiler.a)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(GOOGLE_PERFTOOLS REQUIRED_VARS
