diff --git a/cmake_modules/FindLibEv.cmake b/cmake_modules/FindLibEv.cmake
index c2cfbd3..505b846 100644
--- a/cmake_modules/FindLibEv.cmake
+++ b/cmake_modules/FindLibEv.cmake
@@ -22,16 +22,9 @@
 #  LIBEV_STATIC_LIB, path to libev's static library
 #  LIBEV_FOUND, whether libev has been found
 
-find_path(LIBEV_INCLUDE_DIR ev++.h
-  # make sure we don't accidentally pick up a different version
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-find_library(LIBEV_SHARED_LIB ev
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-find_library(LIBEV_STATIC_LIB libev.a
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
+find_path(LIBEV_INCLUDE_DIR ev++.h)
+find_library(LIBEV_SHARED_LIB ev)
+find_library(LIBEV_STATIC_LIB libev.a)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(LIBEV REQUIRED_VARS
