diff --git a/cmake_modules/FindLibUnwind.cmake b/cmake_modules/FindLibUnwind.cmake
index b6a8622..5409a37 100644
--- a/cmake_modules/FindLibUnwind.cmake
+++ b/cmake_modules/FindLibUnwind.cmake
@@ -22,15 +22,9 @@
 #  UNWIND_SHARED_LIB, path to libunwind's shared library
 #  UNWIND_STATIC_LIB, path to libunwind's static library
 
-find_path(UNWIND_INCLUDE_DIR libunwind.h
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-find_library(UNWIND_SHARED_LIB unwind
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
-find_library(UNWIND_STATIC_LIB libunwind.a
-  NO_CMAKE_SYSTEM_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH)
+find_path(UNWIND_INCLUDE_DIR libunwind.h)
+find_library(UNWIND_SHARED_LIB unwind)
+find_library(UNWIND_STATIC_LIB libunwind.a)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(UNWIND REQUIRED_VARS
