From 64b5954390f9c88bb13ea66446a52f9a71c97767 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 17 Jul 2022 13:48:31 +0200
Subject: [PATCH 1/6] unconditionally use ninja

---
 setup.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index ff1b1c5..46ca75b 100644
--- a/setup.py
+++ b/setup.py
@@ -44,10 +44,11 @@ class CMakeBuild(build_ext):
             "-DSDL_DYNLOAD=ON",
             "-DBUILD_CPP_LIB=OFF",
             "-DBUILD_PYTHON_LIB=ON",
+            "-GNinja",
         ]
         build_args = []
 
-        if self.compiler.compiler_type != "msvc":
+        if False:
             # Using Ninja-build since it a) is available as a wheel and b)
             # multithreads automatically. MSVC would require all variables be
             # exported for Ninja to pick it up, which is a little tricky to do.
@@ -65,7 +66,7 @@ class CMakeBuild(build_ext):
                 except ImportError:
                     pass
 
-        else:
+        elif False:
             # Single config generators are handled "normally"
             single_config = any(x in cmake_generator for x in {"NMake", "Ninja"})
 
-- 
2.38.1.windows.1

