From 0da54df2970342fb05f70ee5bb38b7e9294227d4 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Tue, 6 Dec 2022 09:16:59 +1000
Subject: [PATCH 2/6] unvendor pybind11

---
 src/python/CMakeLists.txt | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
index 911e280..6552462 100644
--- a/src/python/CMakeLists.txt
+++ b/src/python/CMakeLists.txt
@@ -1,11 +1,7 @@
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
 find_package(Python3 COMPONENTS Interpreter Development.Module REQUIRED)
 
-include(FetchContent)
-FetchContent_Declare(
-    pybind11
-    GIT_REPOSITORY https://github.com/pybind/pybind11
-    GIT_TAG v2.10.0)
-FetchContent_MakeAvailable(pybind11)
+find_package(pybind11 CONFIG REQUIRED)
 
 add_library(ale-py MODULE ale_python_interface.cpp)
 # Depend on the ALE and pybind11 module
-- 
2.38.1.windows.1

