From d7bae0a5992ce88c44beeb5b0cd8eccd1a693de8 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Tue, 6 Dec 2022 09:16:59 +1000
Subject: [PATCH 3/6] build shared libs

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 46ca75b..8d37d57 100644
--- a/setup.py
+++ b/setup.py
@@ -45,6 +45,7 @@ class CMakeBuild(build_ext):
             "-DBUILD_CPP_LIB=OFF",
             "-DBUILD_PYTHON_LIB=ON",
             "-GNinja",
+            "-DBUILD_SHARED_LIBS=ON",
         ]
         build_args = []
 
-- 
2.38.1.windows.1

