From aec109d22ceb98287d290f62aa04308aac9e262a Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Wed, 7 Dec 2022 06:34:34 +1000
Subject: [PATCH 4/6] avoid vcpkg toolchain

---
 CMakeLists.txt | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b2032e..cefe88c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,21 +30,6 @@ if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
       CACHE STRING "")
 endif()
 
-# Discover VCPKG toolchain
-if (NOT DEFINED CMAKE_TOOLCHAIN_FILE)
-  # VCPKG_ROOT is what Microsoft recommends,
-  # VCPKG_INSTALLATION_ROOT is what's defined on Azure
-  if(DEFINED ENV{VCPKG_ROOT})
-    set(CMAKE_TOOLCHAIN_FILE
-        "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
-        CACHE STRING "")
-  elseif(DEFINED ENV{VCPKG_INSTALLATION_ROOT})
-    set(CMAKE_TOOLCHAIN_FILE
-        "$ENV{VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake"
-        CACHE STRING "")
-  endif()
-endif()
-
 # Don't allow running cmake in root directory
 if (CMAKE_BINARY_DIR STREQUAL CMAKE_SOURCE_DIR)
   message(FATAL_ERROR [=[
-- 
2.38.1.windows.1

