diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b2032e5..cefe88c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,21 +30,6 @@ if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
       CACHE STRING "")
 endif()
 
-# Discover VCPKG toolchain
-if (NOT DEFINED CMAKE_TOOLCHAIN_FILE)
-  # VCPKG_ROOT is what Microsoft recommends,
-  # VCPKG_INSTALLATION_ROOT is what's defined on Azure
-  if(DEFINED ENV{VCPKG_ROOT})
-    set(CMAKE_TOOLCHAIN_FILE
-        "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
-        CACHE STRING "")
-  elseif(DEFINED ENV{VCPKG_INSTALLATION_ROOT})
-    set(CMAKE_TOOLCHAIN_FILE
-        "$ENV{VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake"
-        CACHE STRING "")
-  endif()
-endif()
-
 # Don't allow running cmake in root directory
 if (CMAKE_BINARY_DIR STREQUAL CMAKE_SOURCE_DIR)
   message(FATAL_ERROR [=[
