diff --git a/setup.py b/setup.py
index ff1b1c59..a084a397 100644
--- a/setup.py
+++ b/setup.py
@@ -38,6 +38,7 @@ def build_extension(self, ext):
 
         cmake_args = [
             f"-DCMAKE_BUILD_TYPE={config}",
+            f"-DPython_EXECUTABLE={sys.executable}",
             f"-DPython3_EXECUTABLE={sys.executable}",
             f"-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={extdir}",
             "-DSDL_SUPPORT=ON",
