From a32b4e288a7ca050e0ecbe7022bbcc28d5561963 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Wed, 7 Dec 2022 12:15:48 +1000
Subject: [PATCH 5/6] fix python executable

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 8d37d57..5cf8a39 100644
--- a/setup.py
+++ b/setup.py
@@ -38,6 +38,7 @@ class CMakeBuild(build_ext):
 
         cmake_args = [
             f"-DCMAKE_BUILD_TYPE={config}",
+            f"-DPython_EXECUTABLE={sys.executable}",
             f"-DPython3_EXECUTABLE={sys.executable}",
             f"-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={extdir}",
             "-DSDL_SUPPORT=ON",
-- 
2.38.1.windows.1

