diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a4de95c3..ea89e7ea 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -66,7 +66,7 @@ if (BUILD_PYTHON_LIB)
 endif()
 
 # Install cpp-lib and CMake config & pkg-config
-if (UNIX AND BUILD_CPP_LIB)
+if (BUILD_CPP_LIB)
   include(GNUInstallDirs)
   include(CMakePackageConfigHelpers)
 
diff --git a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
index 911e2805..7c5e8980 100644
--- a/src/python/CMakeLists.txt
+++ b/src/python/CMakeLists.txt
@@ -19,19 +19,6 @@ set_target_properties(ale-py PROPERTIES
     PREFIX "${PYTHON_MODULE_PREFIX}"
     SUFFIX "${PYTHON_MODULE_EXTENSION}")
 
-# To test ale-py we need to make sure that all generated resources
-# are in CMAKE_CURRENT_BINARY_DIR, i.e., build/src/python.
-# MSVC adds the config to the path, i.e., Debug/Release.
-# There's a hack in CMake where if a generator expression is used
-# this Debug/Release suffix won't be added. We can append a null generator
-# expression to the path to force all outputs CMAKE_CURRENT_BINARY_DIR.
-if (MSVC)
-  set_target_properties(ale-py PROPERTIES
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}$<0:>
-    RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}$<0:>
-    ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}$<0:>)
-endif()
-
 # Copy over the Python source files
 file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/
      DESTINATION ${CMAKE_CURRENT_BINARY_DIR}
