diff --git code/cmt/python/tests/mcgsm_test.py code/cmt/python/tests/mcgsm_test.py
index fa02aa6..ff212f7 100644
--- code/cmt/python/tests/mcgsm_test.py
+++ code/cmt/python/tests/mcgsm_test.py
@@ -253,6 +253,9 @@ class Tests(unittest.TestCase):
 
 
 
+        @unittest.skip("Skipping test as it has a sporadic numerical error."
+                       "See issue ( https://github.com/lucastheis/cmt/issues/16 )."
+        )
 	def test_gradient(self):
 		mcgsm = MCGSM(5, 2, 2, 4, 10)
 
