diff --git setup.py setup.py
index f9a2866..d1c2c1c 100644
--- setup.py
+++ setup.py
@@ -41,18 +41,6 @@ if any(['intel' in arg for arg in sys.argv]) or 'intel' in get_default_compiler(
 		if os.path.exists(path):
 			library_dirs += [path]
 
-elif sys.platform == 'darwin':
-	# clang-specific options
-	include_dirs = []
-	library_dirs = []
-	libraries = []
-	extra_compile_args = ['-std=c++0x', '-stdlib=libc++']
-	extra_link_args = []
-
-	os.environ['CC'] = 'clang++'
-	os.environ['CXX'] = 'clang++'
-	os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.7'
-
 else:
 	# gcc-specific options
 	include_dirs = []
