diff --git setup.py setup.py
index 82bdacd..22c7044 100755
--- setup.py
+++ setup.py
@@ -50,7 +50,6 @@
 	libraries = []
 	extra_compile_args = [
 		'-std=c++0x',
-		'-stdlib=libc++',
 		'-Wno-deprecated-register',
 		'-Wno-#warnings']
 	extra_link_args = []
@@ -59,6 +58,7 @@
 		os.environ['CC'] = 'clang++'
 	if 'CXX' not in os.environ:
 		os.environ['CXX'] = 'clang++'
+		extra_compile_args.append('-stdlib=libc++')
 	if 'MACOSX_DEPLOYMENT_TARGET' not in os.environ:
 		os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.7'
 
