diff --git code/cmt/python/tests/univariate_test.py code/cmt/python/tests/univariate_test.py
index 0059e85..bedaa2d 100644
--- code/cmt/python/tests/univariate_test.py
+++ code/cmt/python/tests/univariate_test.py
@@ -89,6 +89,9 @@ class Tests(unittest.TestCase):
 
 
 
+        @unittest.skip("Skipping test as it has a numerical error."
+                       "See issue ( https://github.com/lucastheis/cmt/issues/15 )."
+        )
 	def test_binomial_sample(self):
 		T = 1000
 
