diff --git a/gloo/CMakeLists.txt b/gloo/CMakeLists.txt
index d5e6a1a..49dd8f8 100644
--- a/gloo/CMakeLists.txt
+++ b/gloo/CMakeLists.txt
@@ -153,7 +153,7 @@ if(USE_ROCM)
   target_link_libraries(gloo_hip gloo)
 endif()
 if(USE_LIBUV)
-  target_link_libraries(gloo PRIVATE uv_a)
+  target_link_libraries(gloo PRIVATE uv)
 endif()
 if(USE_TCP_OPENSSL_LINK)
   target_compile_definitions(gloo PRIVATE USE_TCP_OPENSSL_LINK)
