diff --git a/gloo/CMakeLists.txt b/gloo/CMakeLists.txt
index d5e6a1a..9c1ad5d 100644
--- a/gloo/CMakeLists.txt
+++ b/gloo/CMakeLists.txt
@@ -77,7 +77,7 @@ endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
   set(GLOO_HAVE_TRANSPORT_TCP 1)
   if(${USE_TCP_OPENSSL_LINK} OR ${USE_TCP_OPENSSL_LOAD})
-    find_package(OpenSSL 1.1 REQUIRED EXACT)
+    find_package(OpenSSL 1.1...<4.0 REQUIRED)
     set(GLOO_HAVE_TRANSPORT_TCP_TLS 1)
   else()
     set(GLOO_HAVE_TRANSPORT_TCP_TLS 0)
