--- build.sh.orig     2024-08-20 10:40:17.860801638 -0500
+++ build.sh          2024-08-20 10:40:08.708639616 -0500
@@ -52,49 +52,7 @@

-if [ "x$CC" = "x" ]; then
-    CC=gcc
-    which ${CROSS_COMPILE}cc >/dev/null 2>&1 && CC=cc
-fi
-if which ${CROSS_COMPILE}${CC} >/dev/null 2>&1; then
-    CC=${CROSS_COMPILE}${CC}
-fi
-if [ "x$CROSS_COMPILE" = "x" ]; then
-    CROSS_COMPILE=`echo $CC |
-                   awk '{ print substr($1,0,match($1,"-(g?cc|clang)$")) }' 2>/dev/null`
-    # fix up android prefix...
-    CROSS_COMPILE=`echo $CROSS_COMPILE |
-                   awk '{ off=match($1,"-android[0-9]+-");
-                          if (off) { printf "%sandroid-\n",substr($1,0,off) }
-                          else     { print $1 } }'`
-fi
-
 predefs=`(${CC} ${CFLAGS} -dM -E -x c /dev/null || true) 2>/dev/null`

-if [ -z "${CROSS_COMPILE}${AR}" ] && echo ${predefs} | grep -q clang; then
-    search_dirs=`${CC} -print-search-dirs  | awk -F= '/^programs:/{print$2}' | \
-                 (sed -E -e 's/([a-z]):\\\/\/\1\//gi' -e 'y/\\\;/\/:/' 2>/dev/null || true)`
-    if [ -n "$search_dirs" ] && \
-       env PATH="$search_dirs:$PATH" which llvm-ar > /dev/null 2>&1; then
-        PATH="$search_dirs:$PATH"
-        AR=llvm-ar
-    fi
-fi
-AR=${AR:-${CROSS_COMPILE}ar}
-
-if echo ${predefs} | grep -q x86_64; then
-    if (grep -q -e '^flags.*\badx\b' /proc/cpuinfo) 2>/dev/null; then
-        cflags="-D__ADX__ $cflags"
-    fi
-fi
-if echo ${predefs} | grep -q __AVX__; then
-    cflags="$cflags -mno-avx" # avoid costly transitions
-fi
-if echo ${predefs} | grep -q 'x86_64\|aarch64'; then :; else
-    cflags="$cflags -D__BLST_NO_ASM__"
-fi
+cflags=$(powershell.exe -File ${RECIPE_DIR}/helpers/update_win_cflags.ps1 -predefs "$predefs")

 CFLAGS="$CFLAGS $cflags"
-TMPDIR=${TMPDIR:-/tmp}
-
-rm -f libblst.a
-trap '[ $? -ne 0 ] && rm -f libblst.a; rm -f *.o ${TMPDIR}/*.blst.$$' 0

