From 6a05bafdbc525ea72800ea43356aa26b056dd45a Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 23 Jul 2022 23:39:09 -0600
Subject: [PATCH] Python3

---
 ib/opt/dispatcher.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ib/opt/dispatcher.py b/ib/opt/dispatcher.py
index c8c5080..8b35231 100644
--- a/ib/opt/dispatcher.py
+++ b/ib/opt/dispatcher.py
@@ -5,7 +5,7 @@
 # Defines Dispatcher class to send messages to registered listeners.
 #
 ##
-from Queue import Queue, Empty
+from queue import Queue, Empty
 
 from ib.lib import maybeName, logger
 from ib.opt import message
