diff --git setup.py setup.py
index f7fc386..cd64e11 100644
--- setup.py
+++ setup.py
@@ -3,36 +3,33 @@ import os
 from distutils.core import setup, Extension
 import distutils.util
 import numpy
+import numpy.distutils.system_info
 
 # includes numpy : package numpy.distutils , numpy.get_include()
 # python setup.py build --inplace
 # python setup.py install --prefix=dist, 
-incs = ['.'] + map(lambda x: os.path.join('spams',x),[ 'linalg', 'prox', 'decomp', 'dictLearn']) + [numpy.get_include()] + ['/usr/include/python2.7/']
+incs = ['.'] + map(lambda x: os.path.join('spams',x),[ 'linalg', 'prox', 'decomp', 'dictLearn']) + [numpy.get_include()] + numpy.distutils.system_info.blas_info().get_include_dirs()
 
 osname = distutils.util.get_platform()
-cc_flags = ['-fPIC', '-fopenmp']
-link_flags = ['-fopenmp', '-s' ]
-libs = ['stdc++', 'blas', 'lapack' ]
-libdirs = []
 
-if osname.startswith("macosx"):
-    cc_flags = ['-fPIC', '-fopenmp','-m32']
-    link_flags = ['-m32', '-framework', 'Python']
+cc_flags = ['-fPIC', '-fopenmp', '-Wunused-variable', '-m64']
+for _ in numpy.__config__.blas_opt_info.get("extra_compile_args", []):
+    if _ not in cc_flags:
+        cc_flags.append(_)
+for _ in numpy.__config__.lapack_opt_info.get("extra_compile_args", []):
+    if _ not in cc_flags:
+        cc_flags.append(_)
 
-if osname.startswith("win32"):
-    cc_flags = ['-fPIC', '-fopenmp','-DWIN32']
-    link_flags = ['-fopenmp', '-mwindows']
-    path = os.environ['PATH']
-    os.environ['PATH'] = 'C:/MinGW/bin;' + path
-    libs = ['stdc++', 'Rblas', 'Rlapack' ]
-    libdirs = ['C:/Program Files/R/R-2.15.1/bin/i386']
-
-if osname.startswith("win-amd64"):
-    cc_flags = ['-openmp', '-EHsc', '-DWIN32', '-DCYGWIN', '-DWINDOWS', '-I','C:/Program Files (x86)/Microsoft Visual Studio 9.0/VC/include']
-    link_flags = []
-    libs = [ 'Rblas', 'Rlapack' ]
-    libdirs = ['C:/Program Files (x86)/Microsoft Visual Studio 9.0/VC/lib/amd64','C:/Program Files/R/R-2.15.1/bin/x64']
+link_flags = ['-fopenmp']
+for _ in numpy.__config__.blas_opt_info.get("extra_link_args", []):
+    if _ not in link_flags:
+        link_flags.append(_)
+for _ in numpy.__config__.lapack_opt_info.get("extra_link_args", []):
+    if _ not in link_flags:
+        link_flags.append(_)
 
+libs = ['stdc++', 'blas', 'lapack', 'gomp']
+libdirs = numpy.distutils.system_info.blas_info().get_lib_dirs()
 
 ##path = os.environ['PATH']; print "XX OS %s, path %s" %(osname,path)
 
