From a2d70f4df7a568bedec4a452b371bd8a594809cd Mon Sep 17 00:00:00 2001
From: Toni G <toni.giorgino@gmail.com>
Date: Mon, 3 Dec 2018 19:27:04 +0100
Subject: [PATCH] use python kernel in prefix

---
 python/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/Makefile b/python/Makefile
index 0bfbf9f4f..6e168262a 100644
--- a/python/Makefile
+++ b/python/Makefile
@@ -24,7 +24,7 @@ all:
           plumed_program_name=plumed \
           plumed_version=$(VERSION) \
           plumed_include_dir=../src/wrapper \
-          plumed_default_kernel="$$PWD/../src/lib/libplumedKernel.$(SOEXT)" \
+          plumed_default_kernel="$$PREFIX/lib/libplumedKernel.$(SOEXT)" \
           $(python_bin) setup.py build_ext -i
 
 else
-- 
2.17.1

