From 95ee980043db8cc01b65b17c30a99bfab5996dc9 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 11 Apr 2020 22:59:14 +0200
Subject: [PATCH] remove (obsolete) if-guards that confuse MSVC

---
 blist/_blist.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/blist/_blist.c b/blist/_blist.c
index f7fcb60..fee7b1a 100644
--- a/blist/_blist.c
+++ b/blist/_blist.c
@@ -5986,11 +5986,7 @@ py_blist_ass_subscript(PyObject *oself, PyObject *item, PyObject *value)
 
                 ext_mark(self, 0, DIRTY);
 
-#if PY_MAJOR_VERSION < 3 || PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION < 2
-                if (PySlice_GetIndicesEx((PySliceObject*)item, self->n,
-#else
                 if (PySlice_GetIndicesEx(item, self->n,
-#endif
                                          &start, &stop,&step,&slicelength)<0)
                         return _int(-1);
 
@@ -6955,11 +6951,7 @@ py_blist_subscript(PyObject *oself, PyObject *item)
                 PyBList* result;
                 PyObject* it;
 
-#if PY_MAJOR_VERSION < 3 || PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION < 2
-                if (PySlice_GetIndicesEx((PySliceObject*)item, self->n,
-#else
                 if (PySlice_GetIndicesEx(item, self->n,
-#endif
                                          &start, &stop,&step,&slicelength)<0) {
                         return _ob(NULL);
                 }
-- 
2.26.0.windows.1

