diff --git a/dispatch/dispatch.h b/dispatch/dispatch.h
index 0c7bdd4..821901b 100644
--- a/dispatch/dispatch.h
+++ b/dispatch/dispatch.h
@@ -39,7 +39,7 @@
 #include <stdarg.h>
 #include <string.h>
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
-#include <unistd.h>
+#include <dispatch/unistd.h>
 #endif
 #include <fcntl.h>
 #if defined(_WIN32)
diff --git a/dispatch/unistd.h b/dispatch/unistd.h
new file mode 100644
index 0000000..7e33916
--- /dev/null
+++ b/dispatch/unistd.h
@@ -0,0 +1,16 @@
+/* unistd.h wrapper that fixes compatibility issue of clang with OLDer glibc.
+ * This code snippet is taken from:
+ *   https://github.com/mackyle/blocksruntime/blob/master/testprefix.h
+ *
+ * More information about the issue can be found at:
+ *   https://mackyle.github.io/blocksruntime/#glibc
+ *
+ * -- chawyehsu 2024/03/14
+ */
+#ifdef __linux__
+#undef __block
+#endif
+#include <unistd.h>
+#ifdef __linux__
+#define __block __attribute__((__blocks__(byref)))
+#endif
diff --git a/private/private.h b/private/private.h
index b87f5dc..063d5b4 100644
--- a/private/private.h
+++ b/private/private.h
@@ -44,7 +44,7 @@
 #include <mach/message.h>
 #endif
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
-#include <unistd.h>
+#include <dispatch/unistd.h>
 #endif
 #if !defined(_WIN32)
 #include <pthread.h>
diff --git a/src/internal.h b/src/internal.h
index 4df1a29..cd896d3 100644
--- a/src/internal.h
+++ b/src/internal.h
@@ -310,7 +310,7 @@ upcast(dispatch_object_t dou)
 #include <stdlib.h>
 #include <string.h>
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
-#include <unistd.h>
+#include <dispatch/unistd.h>
 #endif
 #if defined(_WIN32)
 #include <io.h>
diff --git a/src/queue.c b/src/queue.c
index efafe92..676b0b2 100644
--- a/src/queue.c
+++ b/src/queue.c
@@ -7286,7 +7286,7 @@ libdispatch_init(void)
 
 #if DISPATCH_USE_THREAD_LOCAL_STORAGE
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
-#include <unistd.h>
+#include <dispatch/unistd.h>
 #endif
 #if !defined(_WIN32)
 #include <sys/syscall.h>
diff --git a/src/shims/lock.h b/src/shims/lock.h
index 6bf825a..fd0d75e 100644
--- a/src/shims/lock.h
+++ b/src/shims/lock.h
@@ -59,7 +59,7 @@ _dispatch_lock_owner(dispatch_lock lock_value)
 #elif defined(__linux__)
 
 #include <linux/futex.h>
-#include <unistd.h>
+#include <dispatch/unistd.h>
 #include <sys/syscall.h>   /* For SYS_xxx definitions */
 
 typedef uint32_t dispatch_tid;
