--- tensorstore/internal/oauth2/oauth_utils.cc.orig	2022-07-12 12:56:31.000000000 +0200
+++ tensorstore/internal/oauth2/oauth_utils.cc	2022-07-12 12:56:40.000000000 +0200
@@ -25,7 +25,7 @@
 #include "absl/strings/str_cat.h"
 #include "absl/time/time.h"
 #include <openssl/bio.h>     // IWYU pragma: keep
-#include <openssl/digest.h>  // IWYU pragma: keep
+// #include <openssl/digest.h>  // IWYU pragma: keep
 #include <openssl/evp.h>     // IWYU pragma: keep
 #include <openssl/pem.h>     // IWYU pragma: keep
 #include <openssl/rsa.h>     // IWYU pragma: keep
