--- /dev/null	2022-07-14 12:02:41.000000000 +0200
+++ third_party/com_github_pybind_pybind11/system.BUILD.bazel	2022-07-14 11:58:44.000000000 +0200
@@ -0,0 +1,16 @@
+load("@com_google_tensorstore//:utils.bzl", "cc_library_with_strip_include_prefix")
+
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])
+
+exports_files(["LICENSE"])
+
+cc_library_with_strip_include_prefix(
+    name = "pybind11",
+    hdrs = glob(include = ["$(CB_PREFIX)/include/pybind11/**/*.h"]),
+    strip_include_prefix = "$(CB_PREFIX)/include",
+    deps = [
+        "@local_config_python//:python_headers",
+    ],
+)
