--- python/tensorstore/pybind11.bzl	2022-07-14 11:28:31.000000000 +0200
+++ python/tensorstore/pybind11.bzl.mod	2022-07-14 11:28:09.000000000 +0200
@@ -82,6 +82,10 @@
         if name.endswith(".so"):
             deps += [":" + linker_script_name]
             linkopts += ["-Wl,--version-script", "$(location :" + linker_script_name + ")"]
+        # MacOS needs this flag for _tensorstore.so
+        if name.startswith("_tensorstore"):
+            linkopts += ["-lpython$(PY_VER)"]
+
         native.cc_binary(
             name = cc_binary_name,
             linkshared = True,
