--- third_party/nasm/system.BUILD.bazel 2022-07-11 17:56:42.000000000 +0200
+++ third_party/nasm/system.BUILD.bazel 2022-07-11 18:17:02.000000000 +0200
@@ -1,4 +1,11 @@
+genrule(
+    name = "lnnasmlink",
+    outs = ["nasmlink"],
+    cmd = "ln -s $$CB_BUILD_PREFIX/bin/nasm $@",
+)
+
 sh_binary(
     name = "nasm",
-    srcs = ["nasm"],
+    srcs = ["nasmlink"],
+    visibility = ["//visibility:public"],
 )