From d483506b7e913755aaeaf8e5f465af49ac774a22 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Wed, 2 Apr 2025 14:55:46 -0700
Subject: [PATCH] Remove ninja from install requirements

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index ff0dd8e..6325b23 100644
--- a/setup.py
+++ b/setup.py
@@ -113,7 +113,6 @@ setup(
     download_url=f"{URL}/archive/gsplat-{__version__}.tar.gz",
     python_requires=">=3.7",
     install_requires=[
-        "ninja",
         "numpy",
         "jaxtyping",
         "rich>=12",
-- 
2.20.1.windows.1

