From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Thu, 13 Sep 2018 20:59:21 +0200
Subject: Mark PCLBase as nogil to fix compilation

---
 pcl/pcl_defs.pxd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pcl/pcl_defs.pxd b/pcl/pcl_defs.pxd
index 881f707..1117ee6 100644
--- a/pcl/pcl_defs.pxd
+++ b/pcl/pcl_defs.pxd
@@ -509,7 +509,7 @@ ctypedef shared_ptr[vector[int]] IndicesPtr_t;
 # pcl_base.h
 # template <typename PointT>
 # class PCLBase
-cdef extern from "pcl/pcl_base.h" namespace "pcl":
+cdef extern from "pcl/pcl_base.h" namespace "pcl" nogil:
     cdef cppclass PCLBase[PointT]:
         PCLBase ()
         # PCLBase (const PCLBase& base)
