From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Sun, 23 Oct 2022 16:43:21 +0200
Subject: Use distutils instead of setuptools for Cython compatibility

Closes: #1022498
---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 94e838e..6c8d7c5 100644
--- a/setup.py
+++ b/setup.py
@@ -5,7 +5,6 @@ from Cython.Distutils import build_ext
 from distutils.core import setup
 from distutils.extension import Extension
 from Cython.Build import cythonize
-from setuptools import setup, find_packages, Extension
 
 import subprocess
 import numpy
