From: Tobias Fischer <info@tobiasfischer.info>
Date: Fri, 31 Mar 2023 18:54:49 +0200
Subject: Update for VTK 9.2
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 9c20d1a..94e838e 100644
--- a/setup.py
+++ b/setup.py
@@ -546,7 +546,7 @@ else:
     # ext_args['include_dirs'].append('/usr/include/vtk')
     # ext_args['include_dirs'].append('/usr/local/include/vtk')
     # pcl 1.7(Ubuntu)
-    ext_args['include_dirs'].append('/usr/include/vtk-9.1')
+    ext_args['include_dirs'].append('/usr/include/vtk-9.2')
     ext_args['library_dirs'].append('/usr/lib')
     # ext_args['libraries'].append('libvtk*.so')
     # pcl 1.8.1(MacOSX)
