--- setup.py	2016-01-17 14:01:36.000000000 +0100
+++ setup.py	2018-05-03 19:38:13.436323207 +0200
@@ -120,9 +120,6 @@
                         'Doc':  ["Sphinx>=1.1"],
                         'Tests':  ["guidata>=1.7.0"],
                         },
-      entry_points={'gui_scripts':
-                    ['PythonQwt-tests-py%d = qwt.tests:run [Tests]'\
-                     % sys.version_info.major,]},
       author = "Pierre Raybaut",
       author_email = 'pierre.raybaut@gmail.com',
       url = 'https://github.com/PierreRaybaut/%s' % LIBNAME,
