From 04a54f895d5aeec8015334a888cd73f910580239 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 4 Mar 2023 15:41:08 +1100
Subject: [PATCH] minimal changes to make bolt build standalone

---
 bolt/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/bolt/CMakeLists.txt b/bolt/CMakeLists.txt
index a97878cd3c..b1e0bd0aab 100644
--- a/bolt/CMakeLists.txt
+++ b/bolt/CMakeLists.txt
@@ -1,5 +1,14 @@
 include(ExternalProject)
 
+cmake_minimum_required(VERSION 3.13.4)
+
+# Check for a standalone build and configure as appropriate from
+# there.
+if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  message("Building Flang as a standalone project.")
+  project(Bolt)
+endif()
+
 set(BOLT_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 set(BOLT_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
 set(CMAKE_CXX_STANDARD 14)
-- 
2.38.1.windows.1

