From f2ee4893b93c254d97fd51fe6f6ee605934cf6a9 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Tue, 4 Aug 2020 21:06:30 -0500
Subject: [PATCH 1/2] pass through QEMU_LD_PREFIX & SDKROOT

---
 llvm/utils/lit/lit/TestingConfig.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/llvm/utils/lit/lit/TestingConfig.py b/llvm/utils/lit/lit/TestingConfig.py
index 76fd66502009..3009e921b621 100644
--- a/llvm/utils/lit/lit/TestingConfig.py
+++ b/llvm/utils/lit/lit/TestingConfig.py
@@ -25,6 +25,8 @@ class TestingConfig(object):
             "LD_LIBRARY_PATH",
             "SYSTEMROOT",
             "TERM",
+            "QEMU_LD_PREFIX",
+            "SDKROOT",
             "CLANG",
             "LLDB",
             "LD_PRELOAD",
