From aad066540c383da4525bd02cfd084cc72d753b4d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 4 Mar 2023 15:41:08 +1100
Subject: [PATCH] minimal changes to make bolt build standalone

---
 bolt/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bolt/CMakeLists.txt b/bolt/CMakeLists.txt
index a97878cd3c..acff1d924e 100644
--- a/bolt/CMakeLists.txt
+++ b/bolt/CMakeLists.txt
@@ -1,5 +1,8 @@
 include(ExternalProject)
 
+cmake_minimum_required(VERSION 3.13.4)
+project(Bolt)
+
 set(BOLT_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 set(BOLT_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
 set(CMAKE_CXX_STANDARD 14)
-- 
2.38.1.windows.1

