From 03e49260a6bb072fd880f8d34fb97d456b19ee64 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 4 Mar 2023 17:28:21 +1100
Subject: [PATCH 2/2] module path code from lld

---
 bolt/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bolt/CMakeLists.txt b/bolt/CMakeLists.txt
index ece7589df21a..cae0b6760b8b 100644
--- a/bolt/CMakeLists.txt
+++ b/bolt/CMakeLists.txt
@@ -1,6 +1,9 @@
 cmake_minimum_required(VERSION 3.13.4)
 project(Bolt)
 
+find_package(LLVM REQUIRED HINTS "${LLVM_CMAKE_DIR}")
+list(APPEND CMAKE_MODULE_PATH "${LLVM_DIR}")
+
 include(AddLLVM)
 include(ExternalProject)
 
