From decd9c824e7f7b86b70034f953515cd9fc2549d0 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 4 Mar 2023 17:04:37 +1100
Subject: [PATCH 2/2] module path code from libcxx

---
 bolt/CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/bolt/CMakeLists.txt b/bolt/CMakeLists.txt
index b4d6862931..235fa0f9c0 100644
--- a/bolt/CMakeLists.txt
+++ b/bolt/CMakeLists.txt
@@ -1,6 +1,14 @@
 cmake_minimum_required(VERSION 3.13.4)
 project(Bolt)
 
+set(LLVM_COMMON_CMAKE_UTILS "${CMAKE_CURRENT_SOURCE_DIR}/../cmake")
+
+# Add path for custom modules
+list(INSERT CMAKE_MODULE_PATH 0
+  "${LLVM_COMMON_CMAKE_UTILS}"
+  "${LLVM_COMMON_CMAKE_UTILS}/Modules"
+)
+
 include(AddLLVM)
 include(ExternalProject)
 
-- 
2.38.1.windows.1

