diff --git a/src/detection/localip/localip_linux.c b/src/detection/localip/localip_linux.c
index f9cc343d..524d63e3 100644
--- a/src/detection/localip/localip_linux.c
+++ b/src/detection/localip/localip_linux.c
@@ -5,7 +5,7 @@

 #include <string.h>
 #include <ctype.h>
-#include <net/if.h>
+// #include <net/if.h>
 #include <ifaddrs.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
@@ -16,6 +16,8 @@
 #include <linux/ethtool.h>
 #include <linux/sockios.h>
 #include <linux/if.h>
+#else
+#include <net/if.h>
 #endif

 #if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || defined(__NetBSD__) || defined(__HAIKU__)
