diff --git a/src/cmd/go/script_test.go b/src/cmd/go/script_test.go
index c56c1fd..dd916df 100644
--- a/src/cmd/go/script_test.go
+++ b/src/cmd/go/script_test.go
@@ -84,6 +84,13 @@ type backgroundCmd struct {
 
 var extraEnvKeys = []string{
 	"SYSTEMROOT", // must be preserved on Windows to find DLLs; golang.org/issue/25210
+	"CC",
+	"CXX",
+	"CGO_CFLAGS",
+	"CGO_CPPFLAGS",
+	"CGO_CXXFLAGS",
+	"CGO_FFLAGS",
+	"CGO_LDFLAGS",
 }
 
 // setup sets up the test execution temporary directory and environment.
