diff --git a/metatensor-core/cmake/tempdir.cmake b/metatensor-core/cmake/tempdir.cmake
index b302c4c6..47103c25 100644
--- a/metatensor-core/cmake/tempdir.cmake
+++ b/metatensor-core/cmake/tempdir.cmake
@@ -22,6 +22,7 @@ function(get_tempdir _outvar_)
         )
 
         if(_status_ EQUAL 0)
+            file(MAKE_DIRECTORY ${_output_})
             set(${_outvar_} ${_output_} PARENT_SCOPE)
             return()
         endif()
