--- a/tests/client/indexer/grpc/test_indexer_grpc_explorer_api.py
+++ b/tests/client/indexer/grpc/test_indexer_grpc_explorer_api.py
@@ -542,7 +542,7 @@ class TestIndexerGrpcExplorerApi:
             ],
         }

-        assert result_validators == expected_validators
+        assert sorted(result_validators) == sorted(expected_validators)

     @pytest.mark.asyncio
     async def test_fetch_validator(
@@ -625,7 +625,7 @@ class TestIndexerGrpcExplorerApi:
             },
         }

-        assert result_validator == expected_validator
+        assert sorted(result_validator) == sorted(expected_validator)

     @pytest.mark.asyncio
     async def test_fetch_validator_uptime(
@@ -1332,7 +1332,7 @@ class TestIndexerGrpcExplorerApi:
             },
         }

-        assert result_wasm_contracts == expected_wasm_contracts
+        assert sorted(result_wasm_contracts) == sorted(expected_wasm_contracts)

     @pytest.mark.asyncio
     async def test_fetch_wasm_contract_by_address(
@@ -1394,7 +1394,7 @@ class TestIndexerGrpcExplorerApi:
             "proposalId": str(wasm_contract.proposal_id),
         }

-        assert result_wasm_contract == expected_wasm_contract
+        assert sorted(result_wasm_contract) == sorted(expected_wasm_contract)

     @pytest.mark.asyncio
     async def test_fetch_cw20_balance(
