--- a/tests/test_async_client.py
+++ b/tests/test_async_client.py
@@ -44,7 +44,6 @@ class TestAsyncClient:
     async def test_sync_timeout_height_logs_exception(self, caplog):
         client = AsyncClient(
             network=Network.local(),
-            insecure=False,
         )

         with caplog.at_level(logging.DEBUG):
@@ -63,7 +62,6 @@ class TestAsyncClient:
     async def test_get_account_logs_exception(self, caplog):
         client = AsyncClient(
             network=Network.local(),
-            insecure=False,
         )

         with caplog.at_level(logging.DEBUG):
@@ -102,7 +100,6 @@ class TestAsyncClient:

         client = AsyncClient(
             network=Network.local(),
-            insecure=False,
         )

         client.exchange_spot_api._stub = spot_servicer
@@ -164,7 +161,6 @@ class TestAsyncClient:

         client = AsyncClient(
             network=Network.local(),
-            insecure=False,
         )

         client.bank_api._stub = bank_servicer
