From 340ee894912ee1c439bdd14637fcc63945050813 Mon Sep 17 00:00:00 2001
From: rd235 <rd235@users.noreply.github.com>
Date: Fri, 10 Jan 2014 15:04:08 +0000
Subject: [PATCH 16/62] wirefilter -N missing in getopt's optstring

---
 src/wirefilter.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/wirefilter.c b/src/wirefilter.c
index 12b823349..a9d7fd059 100644
--- a/src/wirefilter.c
+++ b/src/wirefilter.c
@@ -1467,7 +1467,7 @@ int main(int argc,char *argv[])
 
 	while(1) {
 		int c;
-		c = GETOPT_LONG (argc, argv, "hl:n:d:M:D:m:b:s:c:v:L:f:",
+		c = GETOPT_LONG (argc, argv, "hNl:n:d:M:D:m:b:s:c:v:L:f:",
 				long_options, &option_index);
 		if (c<0)
 			break;
-- 
2.35.0

