From 45486d501e5dc7e8614f243f76a86b26d18e5d48 Mon Sep 17 00:00:00 2001
From: Ludovico Gardenghi <lu@dovi.co>
Date: Sun, 12 Oct 2014 19:22:54 +0000
Subject: [PATCH 19/62] Add format string to fprintf calls.

---
 src/common/cmdparse.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/common/cmdparse.c b/src/common/cmdparse.c
index 05186d76a..db349c725 100644
--- a/src/common/cmdparse.c
+++ b/src/common/cmdparse.c
@@ -284,13 +284,13 @@ int utm_run(struct utm *utm, struct utm_buf *buf, int fd, int argc, char **argv,
 								int i;
 								for (i=0;i<argc;i++) {
 									if (i) fprintf(mf," ");
-									fprintf(mf,argv[i]);
+									fprintf(mf, "%s", argv[i]);
 								}
 							} else {
 								int num=atoi(t);
 								while (*t >='0' && *t <= '9') t++;
 								if (num < argc) 
-									fprintf(mf,argv[num]);
+									fprintf(mf, "%s", argv[num]);
 							}
 						} else
 							fprintf(mf,"%c",*t);
-- 
2.35.0

