From 75d7ad71066e740e660405b19b65aad90653ed9a Mon Sep 17 00:00:00 2001
From: Ludovico Gardenghi <lu@dovi.co>
Date: Sun, 12 Oct 2014 20:34:33 +0000
Subject: [PATCH 22/62] Add missing headers in vde_router Makefile.am.

---
 src/vde_router/Makefile.am | 26 ++++++++++++++++++++++----
 1 file changed, 22 insertions(+), 4 deletions(-)

diff --git a/src/vde_router/Makefile.am b/src/vde_router/Makefile.am
index c544ca339..d87238875 100644
--- a/src/vde_router/Makefile.am
+++ b/src/vde_router/Makefile.am
@@ -11,9 +11,27 @@ if ENABLE_PROFILE
 endif
 
 bin_PROGRAMS = vde_router
-vde_router_SOURCES = rbtree.h  vde_headers.h  vder_arp.h  vder_datalink.h  vder_icmp.h \
-    vde_router.h  vder_packet.h	vder_queue.h rbtree.c  vder_arp.c  vder_datalink.c \
-    vder_icmp.c  vde_router.c  vder_packet.c  vder_queue.c vder_udp.c vder_dhcp.c \
-	vder_olsr.c
+vde_router_SOURCES = \
+	rbtree.c \
+	rbtree.h \
+	vde_headers.h \
+	vde_router.c \
+	vde_router.h \
+	vder_arp.c \
+	vder_arp.h \
+	vder_datalink.c \
+	vder_datalink.h \
+	vder_dhcp.c \
+	vder_dhcp.h \
+	vder_icmp.c \
+	vder_icmp.h \
+	vder_olsr.c \
+	vder_olsr.h \
+	vder_packet.c \
+	vder_packet.h \
+	vder_queue.c \
+	vder_queue.h \
+	vder_udp.c \
+	vder_udp.h
 
 vde_router_LDADD = $(top_builddir)/src/common/libvdecommon.la $(top_builddir)/src/lib/libvdeplug.la -lpthread
-- 
2.35.0

