From 229c2058ceb3cd0532715fe0229fd2189cbfab85 Mon Sep 17 00:00:00 2001
From: Ludovico Gardenghi <lu@dovi.co>
Date: Sun, 12 Oct 2014 20:49:53 +0000
Subject: [PATCH 23/62] Add missing format string in snprintf.

---
 src/vde_router/vde_router.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vde_router/vde_router.c b/src/vde_router/vde_router.c
index 722d1988d..3f7887d91 100644
--- a/src/vde_router/vde_router.c
+++ b/src/vde_router/vde_router.c
@@ -1357,7 +1357,7 @@ static int newmgmtconn(int fd,struct pollfd *pfd,int nfds)
 		return nfds;
 	}
 	if (nfds < MAXCONN) {
-		snprintf(buf,MAXCMD,header);
+		snprintf(buf,MAXCMD,"%s",header);
 		write(new,buf,strlen(buf));
 		write(new,prompt,strlen(prompt));
 		pfd[nfds].fd=new;
-- 
2.35.0

