From e0a9750e5cb41b352ee610e4588c40a0692c11cc Mon Sep 17 00:00:00 2001
From: Ludovico Gardenghi <lu@dovi.co>
Date: Sun, 12 Oct 2014 20:55:52 +0000
Subject: [PATCH 24/62] Add missing source files in vde_vxlan Makefile.am.

---
 src/vde_vxlan/Makefile.am | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/vde_vxlan/Makefile.am b/src/vde_vxlan/Makefile.am
index 436294db7..00e63af5d 100644
--- a/src/vde_vxlan/Makefile.am
+++ b/src/vde_vxlan/Makefile.am
@@ -9,6 +9,15 @@ if ENABLE_PROFILE
 endif
 
 bin_PROGRAMS = vde_vxlan
-vde_vxlan_SOURCES = vde_vxlan.c vxlan_hash.c log.c plug.c vxlan.c
+vde_vxlan_SOURCES = \
+	log.c \
+	log.h \
+	plug.c \
+	plug.h \
+	vde_vxlan.c \
+	vxlan.c \
+	vxlan.h \
+	vxlan_hash.c \
+	vxlan_hash.h
 vde_vxlan_LDADD = $(top_builddir)/src/common/libvdecommon.la \
 		  $(top_builddir)/src/lib/libvdeplug.la
-- 
2.35.0

