From 01b5f79554e61c3906e1f805bcad0edfab39f7d6 Mon Sep 17 00:00:00 2001
From: Ludovico Gardenghi <lu@dovi.co>
Date: Wed, 12 Nov 2014 14:23:16 +0000
Subject: [PATCH 26/62] vde_autolink: Add missing null entry in getopt_long
 array.

---
 src/vde_autolink.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/vde_autolink.c b/src/vde_autolink.c
index 594e1fc1f..64d699604 100644
--- a/src/vde_autolink.c
+++ b/src/vde_autolink.c
@@ -1329,6 +1329,7 @@ int main(int argc,char **argv)
 		{"mgmtmode",	1, 0, MGMTMODEARG},
 		{"sock",	1, 0, 's'},
 		{"switchmgmt",	1, 0, 'S'},
+		{NULL,		0, 0, 0},
 	};
 	progname=basename(argv[0]);
 
-- 
2.35.0

