From ddf49f6e94d0214bd7f7fb497b96ad03bb460f03 Mon Sep 17 00:00:00 2001
From: Jordi Guillaumes Pons <jguillaumes@gmail.com>
Date: Thu, 30 Nov 2017 14:17:20 +0100
Subject: [PATCH 29/62] - Remove debugging macros

---
 src/vde_router/vde_headers.h | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/vde_router/vde_headers.h b/src/vde_router/vde_headers.h
index db32f2a26..370e24f60 100644
--- a/src/vde_router/vde_headers.h
+++ b/src/vde_router/vde_headers.h
@@ -22,15 +22,6 @@
 #define PROTO_TCP 6
 #define PROTO_UDP 17
 
-#define XSTR(x) STR(x)
-#define STR(x) #x
-
-#ifdef VDE_DARWIN
-#pragma message "Darwin!"
-#else
-#pragma message "No Darwin!"
-#endif
-
 #if defined(VDE_FREEBSD) || defined(VDE_DARWIN)
 struct iphdr
 {
-- 
2.35.0

