From 6703aa55b146358b998ce6955ad1c75321099a4c Mon Sep 17 00:00:00 2001
From: Renzo Davoli <renzo@cs.unibo.it>
Date: Sat, 12 Jan 2019 16:52:54 +0100
Subject: [PATCH 30/62] vde_switch: min packet length enforced

---
 src/vde_switch/port.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/vde_switch/port.c b/src/vde_switch/port.c
index 2ab19e75f..4d3df1168 100644
--- a/src/vde_switch/port.c
+++ b/src/vde_switch/port.c
@@ -583,6 +583,12 @@ void handle_in_packet(struct endpoint *ep,  struct packet *packet, int len)
 	int vlan,tagged;
 	int port=ep->port;
 
+	//DG minimum length of a packet is 60 bytes plus trailing CRC
+	if (__builtin_expect(len < 60, 0)) {
+		memset(packet->data+len,0,60-len);
+		len=60;
+	}
+
 	if(PACKETFILTER(PKTFILTIN,port,packet,len)) {
 
 #ifdef PORTCOUNTERS
-- 
2.35.0

