From 5fc7b66a3d2a0f77066a571d0286cda0c0c22674 Mon Sep 17 00:00:00 2001
From: Alexander Borkowski <a.borkowski@ontolinguistics.org>
Date: Mon, 15 Jul 2019 08:00:45 +0200
Subject: [PATCH 33/62] Fix description for "vlan/delport" command.

---
 src/vde_switch/port.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vde_switch/port.c b/src/vde_switch/port.c
index 4d3df1168..016fbf163 100644
--- a/src/vde_switch/port.c
+++ b/src/vde_switch/port.c
@@ -1370,7 +1370,7 @@ static struct comlist cl[]={
 	{"vlan/create","N","create the VLAN with tag N",vlancreate,INTARG},
 	{"vlan/remove","N","remove the VLAN with tag N",vlanremove,INTARG},
 	{"vlan/addport","N PORT","add port to the vlan N (tagged)",vlanaddport,STRARG},
-	{"vlan/delport","N PORT","add port to the vlan N (tagged)",vlandelport,STRARG},
+	{"vlan/delport","N PORT","delete port from the vlan N (tagged)",vlandelport,STRARG},
 	{"vlan/print","[N]","print the list of defined vlan",vlanprint,STRARG|WITHFILE},
 	{"vlan/allprint","[N]","print the list of defined vlan (including inactive port)",vlanprintall,STRARG|WITHFILE},
 };
-- 
2.35.0

