From 3886085fbbee6d8963aff160a6cc51e515681ab0 Mon Sep 17 00:00:00 2001
From: Ryan Schmidt <git@ryandesign.com>
Date: Sun, 6 Oct 2019 05:40:06 -0500
Subject: [PATCH 37/62] Fix typo in error message

---
 src/slirpvde/slirpvde.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/slirpvde/slirpvde.c b/src/slirpvde/slirpvde.c
index fb4ae9144..8b9584c7c 100644
--- a/src/slirpvde/slirpvde.c
+++ b/src/slirpvde/slirpvde.c
@@ -290,7 +290,7 @@ static void parse_redir_locx(char *buff)
 		*path++=0;
 		tcp2unix_add(port,path);
 	} else 
-		fprintf(stderr,"Error: tcp2unix redirection sytax error -x port:path e.g. -x 6000:/tmp/.X11-unix/X0\r\n");
+		fprintf(stderr,"Error: tcp2unix redirection syntax error -x port:path e.g. -x 6000:/tmp/.X11-unix/X0\r\n");
 }
 
 static void do_redir_tcp(struct redir_tcp_udp *head, int quiet)
-- 
2.35.0

