From a9b2ea5c1c83df98b94569e3703c250b21b9afb6 Mon Sep 17 00:00:00 2001
From: Daniele Lacamera <dan@danielinux.net>
Date: Sun, 6 Oct 2019 11:52:21 +0200
Subject: [PATCH 38/62] Added missing include (github issue #10)

---
 src/vde_cryptcab/cryptcab.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/vde_cryptcab/cryptcab.h b/src/vde_cryptcab/cryptcab.h
index d76d08124..fe9f42b76 100644
--- a/src/vde_cryptcab/cryptcab.h
+++ b/src/vde_cryptcab/cryptcab.h
@@ -36,6 +36,7 @@
 #include <config.h>
 #include <vde.h>
 #include <vdecommon.h>
+#include <limits.h>
 
 
 #define PORTNO 7667
-- 
2.35.0

