From cbfa8db9eda4513df640eb97b1378716ddc80ab4 Mon Sep 17 00:00:00 2001
From: Daniele Lacamera <dan@danielinux.net>
Date: Sun, 6 Oct 2019 11:55:12 +0200
Subject: [PATCH 39/62] Fixed typo in include guard (github issue #7)

---
 src/vde_l3/vde_l3.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/vde_l3/vde_l3.h b/src/vde_l3/vde_l3.h
index 2cfe5bb16..64d48b0f7 100644
--- a/src/vde_l3/vde_l3.h
+++ b/src/vde_l3/vde_l3.h
@@ -1,5 +1,5 @@
-#ifndef _VDE_L3_H_
-#define _VDE_L3_H__
+#ifndef VDE_L3_H
+#define VDE_L3_H
 /* pfifo.c */
 int pfifo_enqueue(struct vde_buff *vdb, struct vde_iface *vif);
 int pfifo_dequeue(struct vde_iface *vif);
@@ -49,4 +49,4 @@ uint32_t ascii2ip(char *c);
 uint32_t valid_nm(uint32_t nm);
 uint32_t ascii2nm(char *c);
 int ip_input(struct vde_buff *vdb);
-#endif /* _VDE_L3_H__ */
+#endif /* VDE_L3_H */
-- 
2.35.0

