From bdb3a2905bb77d92cf22f4e2125f33489a64a42c Mon Sep 17 00:00:00 2001
From: Daniele Lacamera <dan@danielinux.net>
Date: Sun, 6 Oct 2019 11:56:28 +0200
Subject: [PATCH 40/62] removed unused variables (github issue #8)

---
 src/vde_cryptcab/cryptcab.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/vde_cryptcab/cryptcab.c b/src/vde_cryptcab/cryptcab.c
index d362347d8..842a1cf1e 100644
--- a/src/vde_cryptcab/cryptcab.c
+++ b/src/vde_cryptcab/cryptcab.c
@@ -91,8 +91,6 @@ isvalid_crc32(unsigned char *block, int len)
 
 int data_encrypt_decrypt(unsigned char *src, unsigned char *dst, int len, unsigned char *key, unsigned char *iv)
 {
-	int tlen, olen;
-
 	if (encryption_disabled){
 		memcpy(dst,src,len);
 		return len;
-- 
2.35.0

