From 1f792282b3779965fb9ebcb7c5ed5e36550cf9bc Mon Sep 17 00:00:00 2001
From: Daniele Lacamera <dan@danielinux.net>
Date: Sun, 6 Oct 2019 11:57:49 +0200
Subject: [PATCH 41/62] Fixed missing return value (github issue #9)

---
 src/vde_cryptcab/cryptcab.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/vde_cryptcab/cryptcab.c b/src/vde_cryptcab/cryptcab.c
index 842a1cf1e..f0c3a0dda 100644
--- a/src/vde_cryptcab/cryptcab.c
+++ b/src/vde_cryptcab/cryptcab.c
@@ -99,6 +99,7 @@ int data_encrypt_decrypt(unsigned char *src, unsigned char *dst, int len, unsign
 	wc_Chacha_SetIV(&ctx, iv, CHACHA_IV_BYTES);
 	if (wc_Chacha_Process(&ctx, dst, src, len) == 0)
 		return len;
+    return -1;
 }
 
 
-- 
2.35.0

