From 42272e81b2bc49e760d5d1bb5fbf725a974cb5ae Mon Sep 17 00:00:00 2001
From: Renzo Davoli <renzo@cs.unibo.it>
Date: Sun, 5 Sep 2021 22:08:40 +0200
Subject: [PATCH 56/62] use -module for libraries only (closes PR #24)

---
 src/vde_router/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vde_router/Makefile.am b/src/vde_router/Makefile.am
index d87238875..d05525938 100644
--- a/src/vde_router/Makefile.am
+++ b/src/vde_router/Makefile.am
@@ -1,7 +1,7 @@
 
 moddir = $(pkglibdir)/vde_router
 
-AM_LDFLAGS = -module -avoid-version -export-dynamic
+AM_LDFLAGS = -avoid-version -export-dynamic
 AM_LIBTOOLFLAGS = --tag=disable-static
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-- 
2.35.0

