From 8c1ccaa9b8baa1e7c8208f22a544ea09560774ef Mon Sep 17 00:00:00 2001
From: Renzo Davoli <renzo@cs.unibo.it>
Date: Tue, 7 Sep 2021 12:28:16 +0200
Subject: [PATCH 57/62] vdetaplib man page fix typos

---
 man/vdetaplib.1.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/man/vdetaplib.1.in b/man/vdetaplib.1.in
index 55b0b3dd2..a3a363008 100644
--- a/man/vdetaplib.1.in
+++ b/man/vdetaplib.1.in
@@ -18,7 +18,7 @@ To use it, the libvdetab.so library must be preloaded
 .B 
 export LD_PRELOAD=%PKGLIBDIR%/libvdetap.so
 .RE
-(csh, tchs syntax):
+(csh, tcsh syntax):
 .RS
 .br
 .B
@@ -34,7 +34,7 @@ variable VDEALLTAP to the vde socket.
 .B
 export VDEALLTAP=/tmp/vde.ctl
 .RE
-(csh, tchs syntax):
+(csh, tcsh syntax):
 .RS
 .br
 .B
@@ -56,7 +56,7 @@ export tap1=/tmp/myvde.ctl
 .B
 export ppc=/tmp/ppc.ctl
 .RE
-(csh, tchs syntax):
+(csh, tcsh syntax):
 .RS
 .br
 .B
@@ -85,7 +85,7 @@ export tap0_group=vde-net
 .B
 export tap0_mode=0660
 .RE
-(csh, tchs, syntax):
+(csh, tcsh, syntax):
 .RS
 .br
 .B
-- 
2.35.0

