diff --git a/mdsdcl/mdsdcl_show_version.c.in b/mdsdcl/mdsdcl_show_version.c.in
index 60302ba98..35e667c98 100644
--- a/mdsdcl/mdsdcl_show_version.c.in
+++ b/mdsdcl/mdsdcl_show_version.c.in
@@ -2,7 +2,6 @@
 #include <config.h>
 #include <string.h>
 #include <stdio.h>
-#include <malloc.h>
 
 EXPORT int mdsdcl_show_version(void *ctx __attribute__ ((unused)), char **error __attribute__ ((unused)), char **output)
 {
