diff --git a/mdsdcl/Makefile.in b/mdsdcl/Makefile.in
index 9772b5c7c..22e002161 100644
--- a/mdsdcl/Makefile.in
+++ b/mdsdcl/Makefile.in
@@ -41,13 +41,13 @@ install: $(libdir) $(bindir)
 		@RANLIB@ $@
 
 mdsdclThreadSafe.o: mdsdclThreadSafe.c
-	$(COMPILE.c) $(CFLAGS) -c -I/usr/include/libxml2 $^
+	$(COMPILE.c) $(CFLAGS) -c $(XML_CPPFLAGS) $^
 
 cmdExecute.o: cmdExecute.c
-	$(COMPILE.c) $(CFLAGS) -c -I/usr/include/libxml2 $^
+	$(COMPILE.c) $(CFLAGS) -c $(XML_CPPFLAGS) $^
 
 cmdHelp.o: cmdHelp.c
-	$(COMPILE.c) $(CFLAGS) -c -I/usr/include/libxml2 $^
+	$(COMPILE.c) $(CFLAGS) -c $(XML_CPPFLAGS) $^
 
 cmdParseLex.o: cmdParseLex.c
 	$(COMPILE.c) $(CFLAGS) -Wno-unused-parameter -c $^
