diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index ad2015f3..f79faf73 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -114,7 +114,7 @@ drjit_test(nested nested.cpp)
 drjit_test(sh sh.cpp)
 # drjit_test(special special.cpp
 # drjit_test(sphere sphere.cpp
-drjit_test(struct struct.cpp)
+# drjit_test(struct struct.cpp)
 drjit_test(trig trig.cpp)
 # drjit_test(vector vector.cpp
 
