diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index ad2015f3..da0296b5 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -115,7 +115,9 @@ drjit_test(sh sh.cpp)
 # drjit_test(special special.cpp
 # drjit_test(sphere sphere.cpp
 # drjit_test(struct struct.cpp)
-drjit_test(trig trig.cpp)
+if (NOT APPLE)
+  drjit_test(trig trig.cpp)
+endif()
 # drjit_test(vector vector.cpp
 
 # if (DRJIT_ENABLE_JIT)
