diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index ad2015f3..e3b9a41e 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -53,6 +53,25 @@ function(drjit_test NAME)
   target_compile_options(${NAME}_none PRIVATE ${DRJIT_NONE_FLAGS})
   target_link_libraries(${NAME}_none drjit)
 
+  include(CheckCXXCompilerFlag)
+  check_cxx_compiler_flag("${DRJIT_AVX512_FLAGS}" COMPILER_SUPPORTS_AVX512)
+
+  include(CheckCXXSourceRuns)
+  # Test program to check AVX512 support
+  set(CMAKE_REQUIRED_FLAGS "${DRJIT_AVX512_FLAGS}")
+  check_cxx_source_runs("
+  #include <immintrin.h>
+  int main() {
+      __m512i i = _mm512_set1_epi32(1);
+      __m512i j = _mm512_set1_epi32(2);
+      __m512i k = _mm512_add_epi32(i, j);
+      __m512 f = _mm512_set1_ps(1.0f);
+      __m512 g = _mm512_set1_ps(2.0f);
+      __m512 h = _mm512_add_ps(f, g);
+      return 0;
+  }
+  " AVX512_SUPPORTED)
+
   if (CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
     add_executable(${NAME}_neon ${ARGN} ${DRJIT_HEADERS})
     target_compile_options(${NAME}_neon PRIVATE ${DRJIT_NEON_FLAGS})
@@ -82,7 +101,7 @@ function(drjit_test NAME)
     set_tests_properties(${NAME}_avx2_test PROPERTIES LABELS "avx2")
     target_link_libraries(${NAME}_avx2 drjit)
 
-    if (NOT MSVC)
+    if (AVX512_SUPPORTED AND NOT APPLE AND NOT MSVC)
       add_executable(${NAME}_avx512 ${ARGN} ${DRJIT_HEADERS})
       target_compile_options(${NAME}_avx512 PRIVATE ${DRJIT_AVX512_FLAGS})
       set_target_properties(${NAME}_avx512 PROPERTIES FOLDER ${NAME})
