From 9ba727eb08b411f6a198b131a98e0f456fd27258 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Tue, 25 Dec 2018 04:21:33 +0000
Subject: [PATCH] Look for python in byobu prefix

---
 usr/lib/byobu/include/constants | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/usr/lib/byobu/include/constants b/usr/lib/byobu/include/constants
index f9c2368f..83a9c9a1 100755
--- a/usr/lib/byobu/include/constants
+++ b/usr/lib/byobu/include/constants
@@ -59,13 +59,16 @@ eval $BYOBU_TEST ulimit >/dev/null 2>&1 && export BYOBU_ULIMIT="ulimit" || expor
 
 # Find a suitable python interpreter, if undefined
 if [ -z "$BYOBU_PYTHON" ]; then
-	if python3 -c "import snack" >/dev/null 2>&1; then
+	if $BYOBU_PREFIX/bin/python -c "import snack" >/dev/null 2>&1; then
+		export BYOBU_PYTHON="$BYOBU_PREFIX/bin/python"
+	elif python3 -c "import snack" >/dev/null 2>&1; then
 		export BYOBU_PYTHON="python3"
 	elif python2 -c "import snack" >/dev/null 2>&1; then
 		export BYOBU_PYTHON="python2"
 	elif python -c "import snack" >/dev/null 2>&1; then
 		export BYOBU_PYTHON="python"
 	fi
+
 fi
 
 export BYOBU_WINDOW_NAME="-"
-- 
2.19.2

