diff --git a/include/RMF/internal/SharedDataData.h b/include/RMF/internal/SharedDataData.h
index 9586e380..0b6ee6cd 100644
--- a/include/RMF/internal/SharedDataData.h
+++ b/include/RMF/internal/SharedDataData.h
@@ -44,7 +44,7 @@ struct TypeData : RMF_SMALL_UNORDERED_MAP<ID<Traits>, KeyData<Traits> > {
     P::operator=(o);
     return *this;
   }
-  void swap(TypeData<Traits>& o) { std::swap<P>(*this, o); }
+  void swap(TypeData<Traits>& o) { P::swap(o); }
 };
 
 #define RMF_SHARED_DATA_TYPE_PARENT(Traits, UCName) , public TypeData<Traits>
