diff --git a/CMakeLists.txt b/CMakeLists.txt
index d621cc9..64aa1da 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,9 +118,8 @@ set_target_properties(${PROJECT_NAME} PROPERTIES
                             SOVERSION ${PROJECT_VERSION_MAJOR})
 
 if(WIN32)
-  # Because SOVERSION has no effect to file naming on Windows
   set_target_properties(${PROJECT_NAME} PROPERTIES
-    RUNTIME_OUTPUT_NAME ${PROJECT_NAME}-${PROJECT_VERSION_MAJOR})
+    RUNTIME_OUTPUT_NAME ${PROJECT_NAME})
 endif()
 
 target_include_directories(${PROJECT_NAME}
