From 379dc75237791fd31cfa6a724f32632a75806de2 Mon Sep 17 00:00:00 2001
From: danielnachun <daniel.nachun@gmail.com>
Date: Mon, 17 Jul 2023 23:17:32 -0700
Subject: [PATCH] add def

---
 include/gccdiag/compilation_database.hpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/gccdiag/compilation_database.hpp b/include/gccdiag/compilation_database.hpp
index c6d714c..ab0fb0f 100644
--- a/include/gccdiag/compilation_database.hpp
+++ b/include/gccdiag/compilation_database.hpp
@@ -12,6 +12,18 @@ using fs::path;
 using std::string;
 using std::vector;
 
+#if (__GNUG__ && __GNUG__ < 12)
+namespace std {
+template <>
+struct hash<path> {
+    size_t operator()(const path &p) const
+    {
+        return std::filesystem::hash_value(p);
+    }
+};
+} // namespace std
+#endif
+
 namespace gccdiag {
 struct CompilationDatabaseEntry {
     path build_dir;
-- 
2.41.0

