diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d785be..edcf192 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,7 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
 
-add_subdirectory(third_party)
+# add_subdirectory(third_party)
 
 # The Brunsli library definition.
 include(brunsli.cmake)
diff --git a/brunsli.cmake b/brunsli.cmake
index 1e87527..9a974d6 100644
--- a/brunsli.cmake
+++ b/brunsli.cmake
@@ -80,7 +80,18 @@ target_link_libraries(brunslidec-c PRIVATE brunslidec-static)
 add_library(brunslienc-c SHARED
   c/enc/encode.cc
 )
+target_link_libraries(brunslidec-c PRIVATE
+  brotlienc
+  brotlidec
+  brotlicommon
+)
+
 target_link_libraries(brunslienc-c PRIVATE brunslienc-static)
+target_link_libraries(brunslienc-c PRIVATE
+  brotlienc
+  brotlidec
+  brotlicommon
+)
 list(APPEND BRUNSLI_LIBRARIES brunslidec-c brunslienc-c)
 endif()  # BRUNSLI_EMSCRIPTEN
 
@@ -98,10 +109,16 @@ if(NOT BRUNSLI_EMSCRIPTEN)
 add_executable(cbrunsli c/tools/cbrunsli.cc)
 target_link_libraries(cbrunsli PRIVATE
   brunslienc-static
+  brotlienc
+  brotlidec
+  brotlicommon
 )
 add_executable(dbrunsli c/tools/dbrunsli.cc)
 target_link_libraries(dbrunsli PRIVATE
   brunslidec-static
+  brotlienc
+  brotlidec
+  brotlicommon
 )
 else()  # BRUNSLI_EMSCRIPTEN
 set(WASM_MODULES brunslicodec-wasm brunslidec-wasm brunslienc-wasm)
