diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d785be..edcf192 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,7 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
 
-add_subdirectory(third_party)
+# add_subdirectory(third_party)
 
 # The Brunsli library definition.
 include(brunsli.cmake)
diff --git a/brunsli.cmake b/brunsli.cmake
index 1e87527..8639fb4 100644
--- a/brunsli.cmake
+++ b/brunsli.cmake
@@ -57,7 +57,8 @@ add_library(brunslidec-static STATIC
   ${BRUNSLI_DEC_HEADERS}
 )
 target_link_libraries(brunslidec-static PRIVATE
-  brotlidec-static
+  brotlidec
+  brotlicommon
   brunslicommon-static
 )
 
@@ -66,7 +67,8 @@ add_library(brunslienc-static STATIC
   ${BRUNSLI_ENC_HEADERS}
 )
 target_link_libraries(brunslienc-static PRIVATE
-  brotlienc-static
+  brotlienc
+  brotlicommon
   brunslicommon-static
 )
 
@@ -80,7 +82,16 @@ target_link_libraries(brunslidec-c PRIVATE brunslidec-static)
 add_library(brunslienc-c SHARED
   c/enc/encode.cc
 )
+target_link_libraries(brunslidec-c PRIVATE
+  brotlidec
+  brotlicommon
+)
+
 target_link_libraries(brunslienc-c PRIVATE brunslienc-static)
+target_link_libraries(brunslienc-c PRIVATE
+  brotlienc
+  brotlicommon
+)
 list(APPEND BRUNSLI_LIBRARIES brunslidec-c brunslienc-c)
 endif()  # BRUNSLI_EMSCRIPTEN
 
@@ -98,10 +109,14 @@ if(NOT BRUNSLI_EMSCRIPTEN)
 add_executable(cbrunsli c/tools/cbrunsli.cc)
 target_link_libraries(cbrunsli PRIVATE
   brunslienc-static
+  brotlienc
+  brotlicommon
 )
 add_executable(dbrunsli c/tools/dbrunsli.cc)
 target_link_libraries(dbrunsli PRIVATE
   brunslidec-static
+  brotlidec
+  brotlicommon
 )
 else()  # BRUNSLI_EMSCRIPTEN
 set(WASM_MODULES brunslicodec-wasm brunslidec-wasm brunslienc-wasm)
@@ -138,6 +153,11 @@ if(NOT BRUNSLI_EMSCRIPTEN)
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   )
 
+  install(
+    TARGETS cbrunsli dbrunsli
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+  )
+
   install(
     DIRECTORY ${BRUNSLI_INCLUDE_DIRS}/brunsli
     DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
