diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca53bde..bac0621 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,7 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
 
-add_subdirectory(third_party)
+# add_subdirectory(third_party)
 
 # The Brunsli library definition.
 include(brunsli.cmake)
diff --git a/brunsli.cmake b/brunsli.cmake
index c8d4b8b..adcb1da 100644
--- a/brunsli.cmake
+++ b/brunsli.cmake
@@ -97,10 +97,16 @@ endforeach()
 add_executable(cbrunsli c/tools/cbrunsli.cc)
 target_link_libraries(cbrunsli PRIVATE
   brunslienc-static
+  brotlienc
+  brotlidec
+  brotlicommon
 )
 add_executable(dbrunsli c/tools/dbrunsli.cc)
 target_link_libraries(dbrunsli PRIVATE
   brunslidec-static
+  brotlienc
+  brotlidec
+  brotlicommon
 )
 if(BRUNSLI_EMSCRIPTEN)
 set(WASM_MODULES brunslicodec-wasm brunslidec-wasm brunslienc-wasm)
