From 750ca97d9a7065da1958f6ea0747f20ba2a7a980 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Sun, 10 Oct 2021 17:05:04 +0200
Subject: [PATCH] Comment out integration tests

---
 test/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 0b421b73..30e5fe16 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -46,7 +46,7 @@ ExternalProject_Add(
 
 add_subdirectory(benchmark)
 add_subdirectory(plugins)
-add_subdirectory(integration)
+#add_subdirectory(integration)
 add_subdirectory(performance)
 add_subdirectory(regression)
 add_subdirectory(static_assert)
