--- a/test/CMakeLists.txt	2021-06-14 08:29:09.563903942 +1000
+++ b/test/CMakeLists.txt	2021-06-14 08:27:09.987901000 +1000
@@ -26,23 +26,23 @@
 #============================================================================
 # Do a fake install of ign-physics in order to test the examples.
 #============================================================================
-set(FAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/fake/install")
+# set(FAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/fake/install")
 
-file(MAKE_DIRECTORY ${FAKE_INSTALL_PREFIX})
+# file(MAKE_DIRECTORY ${FAKE_INSTALL_PREFIX})
 
-include(ExternalProject)
-ExternalProject_Add(
-  FAKE_INSTALL
+# include(ExternalProject)
+# ExternalProject_Add(
+#   FAKE_INSTALL
 
-  SOURCE_DIR "${CMAKE_SOURCE_DIR}"
-  EXCLUDE_FROM_ALL 1
-  LOG_CONFIGURE 1
-  LOG_BUILD 1
-  LOG_INSTALL 1
-  CMAKE_ARGS
-    "-DBUILD_TESTING=OFF"
-    "-DCMAKE_INSTALL_PREFIX=${FAKE_INSTALL_PREFIX}"
-)
+#   SOURCE_DIR "${CMAKE_SOURCE_DIR}"
+#   EXCLUDE_FROM_ALL 1
+#   LOG_CONFIGURE 1
+#   LOG_BUILD 1
+#   LOG_INSTALL 1
+#   CMAKE_ARGS
+#     "-DBUILD_TESTING=OFF"
+#     "-DCMAKE_INSTALL_PREFIX=${FAKE_INSTALL_PREFIX}"
+# )
 
 add_subdirectory(benchmark)

--- a/test/integration/CMakeLists.txt	2021-06-14 08:32:06.839908269 +1000
+++ b/test/integration/CMakeLists.txt	2021-06-14 08:32:31.523908871 +1000
@@ -38,10 +38,10 @@
   endforeach()
 endif()
 
-if(TARGET INTEGRATION_ExamplesBuild_TEST)
-  add_dependencies(INTEGRATION_ExamplesBuild_TEST FAKE_INSTALL)
-  set(_env_vars)
-  list(APPEND _env_vars "CMAKE_PREFIX_PATH=${FAKE_INSTALL_PREFIX}:$ENV{CMAKE_PREFIX_PATH}")
-  set_tests_properties(INTEGRATION_ExamplesBuild_TEST PROPERTIES
-    ENVIRONMENT "${_env_vars}")
-endif()
+# if(TARGET INTEGRATION_ExamplesBuild_TEST)
+#   add_dependencies(INTEGRATION_ExamplesBuild_TEST FAKE_INSTALL)
+#   set(_env_vars)
+#   list(APPEND _env_vars "CMAKE_PREFIX_PATH=${FAKE_INSTALL_PREFIX}:$ENV{CMAKE_PREFIX_PATH}")
+#   set_tests_properties(INTEGRATION_ExamplesBuild_TEST PROPERTIES
+#     ENVIRONMENT "${_env_vars}")
+# endif()
