From 1246650df9a8e5228ec6f962eb11d5a06dbaff5d Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sat, 3 Oct 2020 13:47:57 -0400
Subject: [PATCH] Move pytest-runner to tests-require

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 83a2052..ea960f1 100644
--- a/setup.py
+++ b/setup.py
@@ -31,8 +31,7 @@ if use_cython:
 
 setup(
     ext_modules = extensions,
-    setup_requires=["pytest-runner"],
-    tests_require=["pytest", "hypothesis"],
+    tests_require=["pytest", "hypothesis", "pytest-runner"],
     name="pytiff",
     version=_version.__version__,
     packages=["pytiff", "pytiff._version"],
-- 
2.26.2

