From: Will Shanks <wshaos@posteo.net>
Date: Wed, 30 Nov 2022 23:43:28 -0500
Subject: [PATCH] drop setup_requires

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 2a9866c..3714d4b 100644
--- a/setup.py
+++ b/setup.py
@@ -131,7 +131,6 @@ setup(
     ],
     python_requires=">=3.7",
     install_requires=requirements,
-    setup_requires=setup_requirements,
     include_package_data=True,
     extras_require=extras_requirements,
     cmake_args=cmake_args,
-- 
2.38.1

