From 6bed2ead4e6670439d125294c036de84f387efc1 Mon Sep 17 00:00:00 2001
From: Will Shanks <willshanks@us.ibm.com>
Date: Sun, 18 Dec 2022 13:22:38 -0500
Subject: [PATCH] Drop setup_requires

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index aa5099ca..147c7ca8 100644
--- a/setup.py
+++ b/setup.py
@@ -132,7 +132,6 @@ setup(
     ],
     python_requires=">=3.7",
     install_requires=requirements,
-    setup_requires=setup_requirements,
     include_package_data=False,
     package_data={"qiskit_aer": ["VERSION.txt"], "qiskit_aer.library": ["*.csv"]},
     extras_require=extras_requirements,
-- 
2.38.1

