diff --git a/atlab/src/multipole.f90 b/atlab/src/multipole.f90
index 1dea68b..3d752f2 100644
--- a/atlab/src/multipole.f90
+++ b/atlab/src/multipole.f90
@@ -1,4 +1,4 @@
-module multipole
+module multipole_atlab
   use f_precisions, only: dp => f_double
   use f_enums
   implicit none
@@ -634,4 +634,4 @@ contains
 
     deallocate(nboxes)
   end subroutine multipole_centers_to_gaussian_density
-end module multipole
+end module multipole_atlab
