From a7be1a05270bfb01b5af4a7e41c267abd759ce34 Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Thu, 9 Feb 2023 12:01:44 +0100
Subject: [PATCH] ldraw-library-path

---
 qt/qtmain.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/qt/qtmain.cpp b/qt/qtmain.cpp
index db9d884c..21bf5b5a 100644
--- a/qt/qtmain.cpp
+++ b/qt/qtmain.cpp
@@ -198,7 +198,8 @@ int main(int argc, char *argv[])
 #endif
 
 #ifdef LC_LDRAW_LIBRARY_PATH
-	LibraryPaths += qMakePair(QString::fromLatin1(LC_LDRAW_LIBRARY_PATH), false);
+       volatile const char* LDRAW_PATH = LC_LDRAW_LIBRARY_PATH; std::string path; for (volatile const char* it = LDRAW_PATH ; *it != 0; ++it) { path.push_back(*it); }; 
+	LibraryPaths += qMakePair(QString::fromStdString(path), false);
 #endif
 	
 	setlocale(LC_NUMERIC, "C");
-- 
2.30.1 (Apple Git-130)

