From d95471a13922ea36d1210897151570a5938a955c Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Mon, 6 Apr 2026 18:13:07 -0600
Subject: [PATCH] fix: Remove unused dev_requirements

---
 setup.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/setup.py b/setup.py
index 905ca96..6f36974 100644
--- a/setup.py
+++ b/setup.py
@@ -14,12 +14,6 @@ with open(os.path.join(here, "README.rst"), encoding="utf-8") as f:
 with open(os.path.join(here, "requirements.txt"), encoding="utf-8") as f:
     requirements = f.read().splitlines()
 
-with open(os.path.join(here, "devtools/dev-requirements.txt"), encoding="utf-8") as f:
-    dev_requirements = f.read().splitlines()
-dev_requirements = [
-    foo for foo in dev_requirements if not (foo.startswith("#") or foo.startswith("-r"))
-]
-
 setup(
     name="desc-opt",
     version=versioneer.get_version(),
-- 
2.53.0

