From 84abbc36f1f3e61292e2010475b8ee8c2325006a Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 23 Jan 2026 17:13:46 +1100
Subject: [PATCH] remove cython as a runtime dependency

---
 requirements.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index 2e3e590..93e4d1a 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,4 @@
 attrs
-cython>=3.0.0
 numpy>=1.23.0
 protobuf>=4.25
 pyparsing
