From 92948a9a2fd03b370fcda29bc6f4ecce19b9d801 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Sun, 14 Nov 2021 09:28:58 -0600
Subject: [PATCH] do not use .value() which is not available until macOS 10.13

---
 libflatsurf/src/flow_connection.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libflatsurf/src/flow_connection.cc b/libflatsurf/src/flow_connection.cc
index 1610bb90..c500e61e 100644
--- a/libflatsurf/src/flow_connection.cc
+++ b/libflatsurf/src/flow_connection.cc
@@ -99,7 +99,9 @@ FlowConnection<Surface> FlowConnection<Surface>::operator-() const {
 
   LIBFLATSURF_ASSERT(self->negative->self->kind == static_cast<typename ImplementationOf<FlowConnection>::Kind>(-static_cast<int>(self->kind)), "Negative of connection is of unexpected type.");
 
-  return self->negative.value();
+  LIBFLATSURF_ASSERT(self->negative.has_value(), "Negative of flow connection must be computable.");
+
+  return *self->negative;
 }
 
 template <typename Surface>
-- 
2.33.1

