From 15b5dedad7d3f8787e3a29876704451e4517dbc2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Sun, 14 Nov 2021 16:39:15 -0600
Subject: [PATCH] work around problems with {} constructors on clang

---
 libflatsurf/test/path.test.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libflatsurf/test/path.test.cc b/libflatsurf/test/path.test.cc
index c7fe4acf..915b9333 100644
--- a/libflatsurf/test/path.test.cc
+++ b/libflatsurf/test/path.test.cc
@@ -98,7 +98,7 @@ TEMPLATE_TEST_CASE("Short Paths", "[path]", (long long), (mpq_class), (renf_elem
     });
     CAPTURE(path);
 
-    REQUIRE(path.tighten() == Path<Surface>(-SaddleConnection{*surface, surface->previousInFace(face)}));
+    REQUIRE(path.tighten() == Path<Surface>(-SaddleConnection<Surface>(*surface, surface->previousInFace(face))));
   }
 }
 
@@ -119,7 +119,7 @@ TEMPLATE_TEST_CASE("Random Paths", "[path]", (long long), (mpq_class), (renf_ele
     auto last = SaddleConnection{*surface, initial};
     auto path = Path{last};
     for (int i = 1; i < length; i++) {
-      const auto connections = SaddleConnections{*surface}.source(Vertex::source(last.target(), *surface)).sample();
+      const auto connections = SaddleConnections<Surface>(*surface).source(Vertex::source(last.target(), *surface)).sample();
       for (const auto& next : connections) {
         CAPTURE(last, next);
         if ((-last).angle(next) != 0)
-- 
2.33.1

