From 8674a6a3a40063f173c1535405e74407e1708e3a Mon Sep 17 00:00:00 2001
From: "Lucas A. M. Magalhaes" <lucas.magalhaes@ess.eu>
Date: Fri, 25 Jul 2025 11:12:04 +0200
Subject: [PATCH] Add environment variable for softIocPVX.dbd

---
 qsrv/softMain.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/qsrv/softMain.cpp b/qsrv/softMain.cpp
index aa8b0bb..5df5d32 100644
--- a/qsrv/softMain.cpp
+++ b/qsrv/softMain.cpp
@@ -50,6 +50,8 @@ extern "C" int softIocPVX_registerRecordDeviceDriver(struct dbBase *pdbbase);
 #define DBD_FILE EPICS_BASE OSI_PATH_SEPARATOR DBD_BASE
 #define EXIT_FILE EPICS_BASE OSI_PATH_SEPARATOR EXIT_BASE
 
+#define PVXS_PATH_ENV "PVXS"
+
 namespace {
 
 bool verbose = false;
@@ -161,6 +163,12 @@ int main(int argc, char *argv[])
         }
 #endif
 
+        // If DBD env variable exists overwrite dbd_file
+        if (const char* dbd_env = std::getenv(PVXS_PATH_ENV)) {
+            dbd_file = std::string(dbd_env) + DBD_BASE;
+            exit_file = std::string(dbd_env) + EXIT_BASE;
+        }
+
         int opt;
 
         while ((opt = getopt(argc, argv, "hVa:D:d:m:Ssx:vG:")) != -1) {
-- 
2.49.0

