--- qtwebengine/src/3rdparty/chromium/third_party/libvpx/libvpx.gyp.old	2016-06-07 10:35:03.451245241 -0500
+++ qtwebengine/src/3rdparty/chromium/third_party/libvpx/libvpx.gyp	2016-06-07 10:35:39.578363468 -0500
@@ -153,7 +153,7 @@
                 'libvpx_intrinsics_sse4_1',
                 # Currently no avx intrinsic functions
                 #'libvpx_intrinsics_avx',
-                'libvpx_intrinsics_avx2',
+                #'libvpx_intrinsics_avx2',
               ],
             }],
             ['target_arch=="arm64"', {
@@ -176,7 +176,7 @@
                     'libvpx_intrinsics_sse4_1',
                     # Currently no avx intrinsic functions
                     #'libvpx_intrinsics_avx',
-                    'libvpx_intrinsics_avx2',
+                    #'libvpx_intrinsics_avx2',
                   ],
                 }],
               ],
