diff --git qtwebengine/src/3rdparty/chromium/build/common.gypi qtwebengine/src/3rdparty/chromium/build/common.gypi
index e912790..a1269ca 100644
--- qtwebengine/src/3rdparty/chromium/build/common.gypi
+++ qtwebengine/src/3rdparty/chromium/build/common.gypi
@@ -5676,6 +5676,11 @@
           # the RTM release in the /W4 set.
           4456, 4457, 4458, 4459,
 
+		  # TODO(brucedawson): http://crbug.com/593448 4334 is a 'suspicious
+          # shift' warning and 4595 is an 'illegal inline operator new' warning
+          # Both are new in VS 2015 Update 2 and can safely be deferred for now.
+          4334, 4595,
+
           # Further added for MSVC 2015 support of Qt WebEngine
           4302, # 'reinterpret_cast': truncation from 'BYTE *' to 'unsigned int'
           4311, # 'reinterpret_cast': pointer truncation from 'BYTE *' to 'unsigned int'
diff --git qtwebengine/src/3rdparty/chromium/build/config/compiler/BUILD.gn qtwebengine/src/3rdparty/chromium/build/config/compiler/BUILD.gn
index 92bb838..8e9c919 100644
--- qtwebengine/src/3rdparty/chromium/build/config/compiler/BUILD.gn
+++ qtwebengine/src/3rdparty/chromium/build/config/compiler/BUILD.gn
@@ -671,6 +671,15 @@ config("runtime_library") {
     ]
 
     lib_dirs += [ "$android_libcpp_root/libs/$android_app_abi" ]
+	if (visual_studio_version == "2015") {
+      cflags += [
+        # TODO(brucedawson): http://crbug.com/593448 4334 is a 'suspicious
+        # shift' warning and 4595 is an 'illegal inline operator new' warning
+        # Both are new in VS 2015 Update 2 and can safely be deferred for now.
+        "/wd4334",
+        "/wd4595",
+      ]
+    }
 
     if (component_mode == "shared_library") {
       android_libcpp_library = "c++_shared"
