diff --git qtwebengine/src/3rdparty/chromium/build/common.gypi qtwebengine/src/3rdparty/chromium/build/common.gypi
index 9439773..dccfbc6 100644
--- qtwebengine/src/3rdparty/chromium/build/common.gypi
+++ qtwebengine/src/3rdparty/chromium/build/common.gypi
@@ -3330,17 +3330,17 @@
             'MinimumRequiredVersion': '5.02',  # Server 2003.
             'TargetMachine': '17', # x86 - 64
             'AdditionalLibraryDirectories!':
-              ['<(windows_sdk_path)/Lib/win8/um/x86'],
+              ['<(windows_sdk_path)/Lib/winv6.3/um/x86'],
             'AdditionalLibraryDirectories':
-              ['<(windows_sdk_path)/Lib/win8/um/x64'],
+              ['<(windows_sdk_path)/Lib/winv6.3/um/x64'],
             # Doesn't exist x64 SDK. Should use oleaut32 in any case.
             'IgnoreDefaultLibraryNames': [ 'olepro32.lib' ],
           },
           'VCLibrarianTool': {
             'AdditionalLibraryDirectories!':
-              ['<(windows_sdk_path)/Lib/win8/um/x86'],
+              ['<(windows_sdk_path)/Lib/winv6.3/um/x86'],
             'AdditionalLibraryDirectories':
-              ['<(windows_sdk_path)/Lib/win8/um/x64'],
+              ['<(windows_sdk_path)/Lib/winv6.3/um/x64'],
             'TargetMachine': '17', # x64
           },
         },
@@ -5698,7 +5698,7 @@
           'VCLibrarianTool': {
             'AdditionalOptions': ['/ignore:4221'],
             'AdditionalLibraryDirectories': [
-              '<(windows_sdk_path)/Lib/win8/um/x86',
+              '<(windows_sdk_path)/Lib/winv6.3/um/x86',
             ],
           },
           'VCLinkerTool': {
@@ -5715,7 +5715,7 @@
               'shlwapi.lib',
             ],
             'AdditionalLibraryDirectories': [
-              '<(windows_sdk_path)/Lib/win8/um/x86',
+              '<(windows_sdk_path)/Lib/winv6.3/um/x86',
             ],
             'GenerateDebugInformation': 'true',
             'MapFileName': '$(OutDir)\\$(TargetName).map',
