From 08235e9af2ed833bc3dab46c92f04b6825932dc4 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 17 Jan 2022 11:21:31 -0700
Subject: [PATCH] Update setup.cfg

---
 setup.cfg | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index e18fdf9..ad6b584 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -21,6 +21,5 @@ include_package_data = True
 python_requires = >=3.6
 install_requires =
     ryvencore ==0.3.1.*
-    PySide2
     QtPy
     waiting
