From eb8009491a273717201774e5f3e48db9a2116e5c Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 15 Jan 2022 22:01:42 -0700
Subject: [PATCH] Update setup.cfg

---
 setup.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index edc2e37..17ef442 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -20,7 +20,7 @@ packages = find:
 include_package_data = True
 python_requires = >=3.6
 install_requires = 
-	ryvencore ==0.3.1
+	ryvencore ==0.3.1.*
 	PySide2
 	QtPy
 	waiting
