From 052fb3b0f132cb2bbad9c8ec94f713dc6bb2959e Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 15 Jan 2022 22:47:21 -0700
Subject: [PATCH] Update setup.cfg

---
 setup.cfg | 1 +
 1 file changed, 1 insertion(+), 0 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index edc2e37..2e903a8 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -20,8 +20,7 @@ packages = find:
 include_package_data = True
 python_requires = >=3.6
 install_requires = 
 ryvencore ==0.3.1.*
-	PySide2
 	QtPy
 	waiting
 
