From 7d14b6451d3f63a949fee9c09d349aec6edd193e Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Tue, 10 May 2016 16:23:04 +0200
Subject: [PATCH] Allows building with or without -std=c++11 on OSX 10.9

---
 src/common/context.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/context.hpp b/src/common/context.hpp
index e50cd74..5263da1 100644
--- src/common/context.hpp
+++ src/common/context.hpp
@@ -32,7 +32,7 @@
 #include <vector>
 #include <fstream>
 
-#if (__cplusplus > 199711L) || defined(HAVE_CPP11) || defined(_MSC_VER)
+#if (__cplusplus > 201103L) || defined(HAVE_CPP11) || defined(_MSC_VER)
   #include <unordered_map>
   #define UM_NS std
 #else
