From 1c49b2e9bbd405666df847d521a89cb61acc13f4 Mon Sep 17 00:00:00 2001
From: Kevin Murray <kdmfoss@gmail.com>
Date: Wed, 30 May 2018 16:16:47 +1000
Subject: [PATCH] Add cstddef header to zbar.h for NULL

---
 include/zbar.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/zbar.h b/include/zbar.h
index 0ab1d62..b65b88a 100644
--- a/include/zbar.h
+++ b/include/zbar.h
@@ -1298,6 +1298,7 @@ extern zbar_color_t zbar_scanner_get_color(const zbar_scanner_t *scanner);
     }
 }
 
+# include <cstddef>  // For NULL, which is not defined on C++>=11, but used throughout
 # include "zbar/Exception.h"
 # include "zbar/Decoder.h"
 # include "zbar/Scanner.h"
-- 
2.17.0

