diff --git a/libmamba/CMakeLists.txt b/libmamba/CMakeLists.txt
index c6dcc3a37..977d60421 100644
--- a/libmamba/CMakeLists.txt
+++ b/libmamba/CMakeLists.txt
@@ -289,6 +289,9 @@ macro(libmamba_create_target target_name linkage deps_linkage output_name)
 
                 libarchive.a
 
+                libsolv.a
+                libsolvext.a
+
                 libiconv.a
                 libbz2.a
                 liblz4.a
@@ -297,9 +300,6 @@ macro(libmamba_create_target target_name linkage deps_linkage output_name)
                 liblzma.a
                 liblzo2.a
 
-                libsolv.a
-                libsolvext.a
-
                 libnghttp2.a
 
                 libyaml-cpp.a
