diff --git a/setup.py b/setup.py
index fa61a34..a31a6fb 100644
--- a/setup.py
+++ b/setup.py
@@ -18,7 +18,7 @@ default_lib_dir = (
 
 INCLUDE_DIRS = [str(Path(default_lib_dir) / "include"), pybind11.get_include()]
 
-USE_PYTHON_CASADI = False if system() == "Windows" else True
+USE_PYTHON_CASADI = False if (system() == "Windows" or system() == "Linux") else True
 
 # ---------- set environment variables for vcpkg on Windows ----------------------------
 
@@ -41,10 +41,7 @@ def set_vcpkg_environment_variables():
 
 
 def get_casadi_python_include_dir():
-    import casadi
-
-    casadi_path = Path(casadi.__file__).parent
-    include_dir = casadi_path / "include"
+    include_dir = Path(os.environ["PREFIX"]) / "include" / "casadi"
     assert include_dir.exists(), f"CasADi include directory not found at {include_dir}"
     return str(include_dir)
 
