From 359eb3fb8d350b4971db2fbef87d56ca5827eaf8 Mon Sep 17 00:00:00 2001
From: Pradyot Ranjan <99216956+pradyotRanjan@users.noreply.github.com>
Date: Sat, 11 Jan 2025 16:09:10 +0530
Subject: [PATCH] Adding pybind11 to pyproject

Signed-off-by: Pradyot Ranjan <99216956+pradyotRanjan@users.noreply.github.com>
---
 pyproject.toml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pyproject.toml b/pyproject.toml
index da35f93..d8aa0dc 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,6 +2,7 @@
 requires = [
     "setuptools>=64",
     "wheel",
+    "pybind11",
     # On Windows, use the CasADi vcpkg registry and CMake bundled from MSVC
     "casadi>=3.6.7; platform_system!='Windows'",
     # Note: the version of CasADi as a build-time dependency should be matched
-- 
2.41.0

