diff --git a/setup.py b/setup.py
index 022a883..f2db077 100644
--- a/setup.py
+++ b/setup.py
@@ -37,20 +37,11 @@ def set_vcpkg_environment_variables():
     )
 
 
-# ---------- find Python CasADi's include dirs (for Linux and macOS) -------------------
-
-
-def get_casadi_python_include_dir():
-    import casadi
-
-    casadi_path = Path(casadi.__file__).parent
-    include_dir = casadi_path / "include"
-    assert include_dir.exists(), f"CasADi include directory not found at {include_dir}"
-    return str(include_dir)
-
+# ---------- get CasADi's include dirs on conda-forge -------------------
 
 if USE_PYTHON_CASADI:
-    casadi_include = get_casadi_python_include_dir()
+    casadi_include = str(Path(os.environ["PREFIX"]) / "include" / "casadi")
+    assert Path(casadi_include).exists(), f"CasADi include directory not found at {casadi_include}"
     INCLUDE_DIRS.append(casadi_include)
     print(f"Adding CasADi include directory: {casadi_include}")

