diff --git a/src/pybammsolvers/idaklu_source/IDAKLUSolverOpenMP.hpp b/src/pybammsolvers/idaklu_source/IDAKLUSolverOpenMP.hpp
index c3ec1ff..41c385a 100644
--- a/src/pybammsolvers/idaklu_source/IDAKLUSolverOpenMP.hpp
+++ b/src/pybammsolvers/idaklu_source/IDAKLUSolverOpenMP.hpp
@@ -8,7 +8,6 @@ using std::vector;
 
 #include "Options.hpp"
 #include "Solution.hpp"
-#include "sundials_legacy_wrapper.hpp"
 
 /**
  * @brief Abstract solver class based on OpenMP vectors
