diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,3 +31,9 @@ if (NOT MSVC)
 endif ()
 
+if (DEFINED ENV{LIBRARY_PREFIX})
+    list(APPEND CMAKE_PREFIX_PATH "$ENV{LIBRARY_PREFIX}")
+    set(BLA_VENDOR OpenBLAS)
+endif ()
+message(STATUS "[pybammsolvers] LIBRARY_PREFIX=$ENV{LIBRARY_PREFIX} BLA_VENDOR=${BLA_VENDOR} CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}")
+
 find_package(pybind11 CONFIG REQUIRED)
