diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -211,14 +211,8 @@ target_include_directories(idaklu PRIVATE ${SUNDIALS_INCLUDE_DIR})
 target_link_libraries(idaklu PRIVATE ${SUNDIALS_LIBRARIES})
 
 # link suitesparse
-# if using vcpkg, use config mode to
-# find suitesparse. Otherwise, use FindSuiteSparse module
-if (DEFINED VCPKG_ROOT_DIR)
-    find_package(SuiteSparse CONFIG REQUIRED)
-else ()
-    find_package(SuiteSparse REQUIRED)
-    message("SuiteSparse found in ${SuiteSparse_INCLUDE_DIRS}: ${SuiteSparse_LIBRARIES}")
-endif ()
+find_package(SuiteSparse REQUIRED)
+message("SuiteSparse found in ${SuiteSparse_INCLUDE_DIRS}: ${SuiteSparse_LIBRARIES}")
 include_directories(${SuiteSparse_INCLUDE_DIRS})
 target_link_libraries(idaklu PRIVATE ${SuiteSparse_LIBRARIES})
