diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd43f39..f399a84 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,17 +92,13 @@ if (NOT DEFINED USE_PYTHON_CASADI)
 endif ()
 
 if (${USE_PYTHON_CASADI})
-    execute_process(
-            COMMAND "${PYTHON_EXECUTABLE}" -c
-            "import os; import sysconfig; print(os.path.join(sysconfig.get_path('purelib'), 'casadi', 'cmake'))"
-            OUTPUT_VARIABLE CASADI_DIR
-            OUTPUT_STRIP_TRAILING_WHITESPACE)
+    set(CASADI_DIR "$ENV{PREFIX}/lib/cmake/casadi" CACHE PATH "Path to casadi CMake package from Python casadi installation")
 
-    if (CASADI_DIR)
+    if (EXISTS "${CASADI_DIR}")
         file(TO_CMAKE_PATH ${CASADI_DIR} CASADI_DIR)
         message("Found Python casadi path: ${CASADI_DIR}")
     else ()
-        message(FATAL_ERROR "Did not find casadi path")
+        message(FATAL_ERROR "Did not find casadi path ${CASADI_DIR}.")
     endif ()
 
     message("Trying to link against Python casadi package in ${CASADI_DIR}")
