diff -ruN apache-airflow-providers-postgres-1.0.1/apache_airflow_providers_postgres.egg-info/requires.txt apache-airflow-providers-postgres-1.0.1-patch/apache_airflow_providers_postgres.egg-info/requires.txt
--- apache_airflow_providers_postgres.egg-info/requires.txt	2021-02-08 21:37:18.000000000 +0100
+++ apache_airflow_providers_postgres.egg-info/requires.txt	2021-02-21 23:09:09.388749931 +0100
@@ -1,5 +1,5 @@
 apache-airflow>=2.0.0
-psycopg2-binary>=2.7.4
+psycopg2>=2.7.4

 [amazon]
 apache-airflow-providers-amazon
diff -ruN apache-airflow-providers-postgres-1.0.1/setup.py apache-airflow-providers-postgres-1.0.1-patch/setup.py
--- setup.py	2021-02-08 21:37:14.000000000 +0100
+++ setup.py	2021-02-21 23:17:28.853438102 +0100
@@ -57,7 +57,7 @@
         ),
         zip_safe=False,
         include_package_data=True,
-        install_requires=['apache-airflow>=2.0.0', 'psycopg2-binary>=2.7.4'],
+        install_requires=['apache-airflow>=2.0.0', 'psycopg2>=2.7.4'],
         setup_requires=['setuptools', 'wheel'],
         extras_require={'amazon': ['apache-airflow-providers-amazon']},
         classifiers=[
