diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64a11d9..560f3ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,9 +44,9 @@ if (${MSVC})
   # using Visual Studio C++
 
   # this ensures that the min/max macros of minwindef.h are not used
-  add_definitions(-DNOMINMAX /W3)
+  add_compile_options(-DNOMINMAX /W3)
 
-  add_definitions(/W3)
+  add_compile_options(/W3)
   
   # add_definitions(-D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS)
   # set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
