diff -urN src/extra/trio/Makefile.win src.new/extra/trio/Makefile.win
--- src/extra/trio/Makefile.win	2011-10-31 23:05:05.000000000 +0000
+++ src.new/extra/trio/Makefile.win	2016-03-30 13:13:06.013926400 +0100
@@ -2,7 +2,7 @@
 include ../../gnuwin32/MkRules
 
 # triop.h says this does not work, but useful to debug wchar conversions.
-trio-CPPFLAGS=-DTRIO_FEATURE_WIDECHAR=1
+trio-CPPFLAGS=-DTRIO_FEATURE_WIDECHAR=1 -I../../include -I.
 
 # trionan.c and triostr.c are included by trio.o
 OBJS = compat.o trio.o
