--- configure.ac.old	2016-06-14 23:07:04.000000000 +0100
+++ configure.ac	2016-06-14 23:07:11.000000000 +0100
@@ -678,7 +678,7 @@
 R_LD_LIBRARY_PATH_save=${R_LD_LIBRARY_PATH}
 R_LD_LIBRARY_PATH=
 case "${host_os}" in
-  darwin*)
+  some_other_platform*)
     ## Darwin provides a full path in the ID of each library such 
     ## that the linker can add library's path to the binary at link time.
     ## This allows the dyld to find libraries even without xx_LIBRARY_PATH.
--- configure.old	2016-06-14 23:03:38.000000000 +0100
+++ configure	2016-06-14 23:04:50.000000000 +0100
@@ -22373,7 +22373,7 @@
 R_LD_LIBRARY_PATH_save=${R_LD_LIBRARY_PATH}
 R_LD_LIBRARY_PATH=
 case "${host_os}" in
-  darwin*)
+  some_other_platform*)
     ## Darwin provides a full path in the ID of each library such
     ## that the linker can add library's path to the binary at link time.
     ## This allows the dyld to find libraries even without xx_LIBRARY_PATH.
