diff -urN R-3.3.2.orig/configure R-3.3.2/configure
--- R-3.3.2.orig/configure	2016-12-06 14:52:40.000000000 +0000
+++ R-3.3.2/configure	2016-12-06 14:52:55.000000000 +0000
@@ -34410,8 +34410,8 @@
        modlist="${modlist} cairo-svg"
        r_cairo_svg=yes
     fi
-      if "${PKGCONF}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKGCONF}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -34433,7 +34433,7 @@
 
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
-#include <cairo-xlib.h>
+#include <cairo.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
@@ -34497,8 +34497,8 @@
          modlist="${modlist} cairo-svg"
          r_cairo_svg=yes
       fi
-      if "${PKGCONF}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKGCONF}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -34529,7 +34529,6 @@
 /* end confdefs.h.  */
 
 #include <cairo.h>
-#include <cairo-xlib.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
diff -urN R-3.3.2.orig/m4/cairo.m4 R-3.3.2/m4/cairo.m4
--- R-3.3.2.orig/m4/cairo.m4	2016-12-06 14:50:35.000000000 +0000
+++ R-3.3.2/m4/cairo.m4	2016-12-06 14:52:55.000000000 +0000
@@ -52,8 +52,8 @@
        modlist="${modlist} cairo-svg"
        r_cairo_svg=yes
     fi
-      if "${PKGCONF}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKGCONF}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -70,7 +70,7 @@
                     [AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
-#include <cairo-xlib.h>
+#include <cairo.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
@@ -116,8 +116,8 @@
          modlist="${modlist} cairo-svg"
          r_cairo_svg=yes
       fi
-      if "${PKGCONF}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKGCONF}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -143,7 +143,6 @@
 		     [r_cv_cairo_works], 
                      [AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <cairo.h>
-#include <cairo-xlib.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
