diff --git a/qdCocoa.m b/qdCocoa.m
index 954b17f..41ab6f7 100644
--- src/library/grDevices/src/qdCocoa.m
+++ src/library/grDevices/src/qdCocoa.m
@@ -127,7 +127,7 @@ static QuartzFunctions_t *qf;
 	    menuItem = [[NSMenuItem alloc] initWithTitle:@"Close Window" action:@selector(performClose:) keyEquivalent:@"w"]; [menu addItem:menuItem]; [menuItem release];
 	    menuItem = [[NSMenuItem alloc] initWithTitle:@"Save" action:@selector(saveDocument:) keyEquivalent:@"s"]; [menu addItem:menuItem]; [menuItem release];
 	    [menu addItem:[NSMenuItem separatorItem]];
-	    menuItem = [[NSMenuItem alloc] initWithTitle:@"Page Setup…" action:@selector(runPageLayout:) keyEquivalent:@"P"]; [menu addItem:menuItem]; [menuItem release];
+	    menuItem = [[NSMenuItem alloc] initWithTitle:@"Page Setup..." action:@selector(runPageLayout:) keyEquivalent:@"P"]; [menu addItem:menuItem]; [menuItem release];
 	    menuItem = [[NSMenuItem alloc] initWithTitle:@"Print" action:@selector(printDocument:) keyEquivalent:@"p"]; [menu addItem:menuItem]; [menuItem release];   
 	    
             menuItem = [[NSMenuItem alloc] initWithTitle:[menu title] action:nil keyEquivalent:@""]; /* the "Quartz" item in the main menu */
