Index: src/modules/lapack/Makefile.in
===================================================================
--- src/modules/lapack/Makefile.in	(revision 1003)
+++ src/modules/lapack/Makefile.in	(revision 1004)
@@ -45,7 +45,7 @@
 ## needed being loaded into R.bin/libR.so, and that need not be the
 ## case for a static libfortran or if a package is compiled under a
 ## different compiler.
-Rlapack_la_LIBADD =  $(FLIBS_IN_SO) @DYLIB_UNDEFINED_ALLOWED_FALSE@ $(LIBR)
+Rlapack_la_LIBADD =  @LAPACK_LDFLAGS@ $(FLIBS_IN_SO) @DYLIB_UNDEFINED_ALLOWED_FALSE@ $(LIBR)
 
 ALL_CFLAGS = $(ALL_CFLAGS_LO)
 ALL_FFLAGS = $(ALL_FFLAGS_LO)
