From a64ede9cf4b3a4c78e026d28df3fdbc591dd5340 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 19 Jun 2024 11:56:47 +0200
Subject: [PATCH] Rename ambiguous _iter_driver and pair_contained

Signed-off-by: Silvio Traversaro <silvio@traversaro.it>
---
 include/zenohcxx/api.hxx | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/zenohcxx/api.hxx b/include/zenohcxx/api.hxx
index c2c341a..8ce2ae2 100644
--- a/include/zenohcxx/api.hxx
+++ b/include/zenohcxx/api.hxx
@@ -709,8 +709,8 @@ struct AttachmentView : public Copyable<::z_attachment_t> {
     AttachmentView(const AttachmentView::IterDriver& _iter_driver)
         : Copyable({static_cast<const void*>(&_iter_driver),
                     [](const void* data, z_attachment_iter_body_t body, void* ctx) -> int8_t {
-                        const IterDriver* _iter_driver = static_cast<const IterDriver*>(data);
-                        return (*_iter_driver)([&body, &ctx](const BytesView& key, const BytesView& value) {
+                        const IterDriver* _iter_driver_ptr = static_cast<const IterDriver*>(data);
+                        return (*_iter_driver_ptr)([&body, &ctx](const BytesView& key, const BytesView& value) {
                             return body(key, value, ctx);
                         });
                     }}) {}
@@ -722,8 +722,8 @@ struct AttachmentView : public Copyable<::z_attachment_t> {
     AttachmentView(const T& pair_container)
         : Copyable({static_cast<const void*>(&pair_container),
                     [](const void* data, z_attachment_iter_body_t body, void* ctx) -> int8_t {
-                        const T* pair_container = static_cast<const T*>(data);
-                        for (const auto& it : *pair_container) {
+                        const T* pair_container_ptr = static_cast<const T*>(data);
+                        for (const auto& it : *pair_container_ptr) {
                             int8_t ret = body(BytesView(it.first), BytesView(it.second), ctx);
                             if (ret) {
                                 return ret;
