From b648c05f0ebfb540f32df4e4b2102847600cdd8f Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Mon, 23 Sep 2024 22:24:50 +0200
Subject: [PATCH] Disable tests, examples and docs

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddc12340..0bf197fc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,7 +80,7 @@ endif()
 #
 if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
 	add_subdirectory(install)
-	add_subdirectory(tests)
-	add_subdirectory(examples)
-	add_subdirectory(docs)
+	#add_subdirectory(tests)
+	#add_subdirectory(examples)
+	#add_subdirectory(docs)
 endif()
