diff --git a/casatools/setup.py b/casatools/setup.py
index cd8944556f..05ffa8c97c 100644
--- a/casatools/setup.py
+++ b/casatools/setup.py
@@ -208,7 +208,7 @@ class XmlCMakeBuildExt(build_ext):
         subprocess.check_call(['cmake', '--build', '.'])
 
         # Copy GCC libs on Macos. 
-        if (sys.platform == 'darwin'):
+        if False: # conda-forge: no
             if (os.environ.get('FC') != None):
                 fortran_compiler = os.environ.get('FC')
             else:
@@ -234,7 +234,7 @@ class XmlCMakeBuildExt(build_ext):
                         sys.exit("\tGLIBC_PRIVATE cleanup failed...")
 
             #Convert /opt/local/lib to @rpath on macOS
-            if (sys.platform == 'darwin'):
+            if False: # conda-forge: no
                 casac_lib_dir = extdir + "/casatools/__casac__/lib"
                 from subprocess import call
                 status=call("for f in `ls`; do if [[ $(otool -l $f | grep \"path /opt/local/lib\" | grep -v  \"/opt/local/lib/.*gcc\") ]];then install_name_tool -rpath /opt/local/lib  @loader_path:@loader_path/lib:@rpath $f ;fi; done",cwd=casac_lib_dir,shell=True)
