diff --git a/src/functions/GaussPoly.h b/src/functions/GaussPoly.h
index 088089c..15909de 100644
--- a/src/functions/GaussPoly.h
+++ b/src/functions/GaussPoly.h
@@ -63,8 +63,8 @@ public:
 
     double calcSquareNorm() override;
 
-    double evalfCore(const Coord<D> &r) const;
-    double evalf(double r, int dim) const;
+    double evalfCore(const Coord<D> &r) const override;
+    double evalf(double r, int dim) const override;
 
     double calcOverlap(GaussFunc<D> &b) override;
     double calcOverlap(GaussPoly<D> &b) override;
diff --git a/src/trees/FunctionTree.h b/src/trees/FunctionTree.h
index ee3db15..f6cc875 100644
--- a/src/trees/FunctionTree.h
+++ b/src/trees/FunctionTree.h
@@ -58,7 +58,7 @@ public:
     void clear();
 
     double integrate() const;
-    double evalf(const Coord<D> &r) const;
+    double evalf(const Coord<D> &r) const override;
 
     void getEndValues(Eigen::VectorXd &data);
     void setEndValues(Eigen::VectorXd &data);
