diff --git a/src/treebuilders/grid.cpp b/src/treebuilders/grid.cpp
index 529e28d..5dfddef 100644
--- a/src/treebuilders/grid.cpp
+++ b/src/treebuilders/grid.cpp
@@ -32,6 +32,7 @@
 #include "WaveletAdaptor.h"
 #include "add.h"
 #include "functions/GaussExp.h"
+#include "functions/Gaussian.h"
 #include "functions/function_utils.h"
 #include "utils/Printer.h"
 
@@ -300,6 +301,9 @@ template <int D> int refine_grid(FunctionTree<D> &out, FunctionTree<D> &inp) {
 template void build_grid<1>(FunctionTree<1> &out, const GaussExp<1> &inp, int maxIter);
 template void build_grid<2>(FunctionTree<2> &out, const GaussExp<2> &inp, int maxIter);
 template void build_grid<3>(FunctionTree<3> &out, const GaussExp<3> &inp, int maxIter);
+template void build_grid<1>(FunctionTree<1> &out, const Gaussian<1> &inp, int maxIter);
+template void build_grid<2>(FunctionTree<2> &out, const Gaussian<2> &inp, int maxIter);
+template void build_grid<3>(FunctionTree<3> &out, const Gaussian<3> &inp, int maxIter);
 template void build_grid<1>(FunctionTree<1> &out, const RepresentableFunction<1> &inp, int maxIter);
 template void build_grid<2>(FunctionTree<2> &out, const RepresentableFunction<2> &inp, int maxIter);
 template void build_grid<3>(FunctionTree<3> &out, const RepresentableFunction<3> &inp, int maxIter);
