diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6b7712f..5d8399e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -2,9 +2,6 @@ add_library(mrcpp "")
 
 add_dependencies(mrcpp mrcpp-info)
 
-# TODO Set compiler options at this level, rather than globally
-#target_compile_options(mrcpp )
-
 target_compile_definitions(mrcpp
   INTERFACE
     $<INSTALL_INTERFACE:USING_MRCPP>
@@ -37,7 +34,6 @@ add_subdirectory(treebuilders)
 add_subdirectory(trees)
 add_subdirectory(utils)
 
-# FIXME Fix versioning of DSO
 set_target_properties(mrcpp
   PROPERTIES
     VERSION ${MRCPP_VERSION_MAJOR}
@@ -48,6 +44,13 @@ set_target_properties(mrcpp
     PUBLIC_HEADER "${mrcpp_headers}"
   )
 
+if(ENABLE_TESTS AND APPLE)
+  set_target_properties(mrcpp
+    PROPERTIES
+      LINK_FLAGS "-undefined dynamic_lookup"
+    )
+endif()
+
 install(
   TARGETS
     mrcpp
@@ -65,7 +68,6 @@ install(
 
 # <<<  Export Config  >>>
 include(CMakePackageConfigHelpers)
-# FIXME Fix versioning
 write_basic_package_version_file(
   ${PROJECT_BINARY_DIR}/${CMAKECONFIG_INSTALL_DIR}/MRCPPConfigVersion.cmake
   VERSION ${MRCPP_VERSION}
