From 4f4323256dddcc3a32b98a7ab164543a4628b4ce Mon Sep 17 00:00:00 2001
From: Marko Petric <marko.petric@cern.ch>
Date: Thu, 3 Sep 2020 11:45:38 +0200
Subject: [PATCH] Set search order for python on macOS to search for system
 python last

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7c8537d66..f24c55032 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,6 +113,7 @@ dd4hep_configure_output( OUTPUT "${PROJECT_BINARY_DIR}" INSTALL "${CMAKE_INSTALL
 ########################
 
 # Configure Python
+set(Python_FIND_FRAMEWORK LAST)
 find_package(Python COMPONENTS Development)
 set(DD4HEP_PYTHON_INSTALL_DIR lib/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages)
 
