From abc853751bd83b9ec844f346af867e56b9b37108 Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Fri, 19 Jun 2020 17:25:24 -0400
Subject: [PATCH] Fix CMake include of GetPrerequisites.

---
 cmake/GetDependencies.cmake.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/GetDependencies.cmake.in b/cmake/GetDependencies.cmake.in
index 0d25f670..cfb13230 100644
--- a/cmake/GetDependencies.cmake.in
+++ b/cmake/GetDependencies.cmake.in
@@ -11,7 +11,7 @@ set(CMAKE_SYSTEM_LIBRARY_PATH @CMAKE_SYSTEM_LIBRARY_PATH@)
 set(CMAKE_FIND_ROOT_PATH @CMAKE_FIND_ROOT_PATH@)
 set(CODEC2_DLL ${CMAKE_BINARY_DIR}/src/libcodec2.dll)
 
-include(${CMAKE_SOURCE_DIR}/cmake/GetPrerequisites.cmake)
+include(GetPrerequisites)
 get_prerequisites(${CODEC2_DLL} _deps 1 0 "" "")
 foreach(_runtime ${_deps})
     message("Looking for ${_runtime}")
-- 
2.20.1

