diff --git a/mkdocs_exclude/plugin.py b/mkdocs_exclude/plugin.py
index 391a94e..e6016a8 100644
--- a/mkdocs_exclude/plugin.py
+++ b/mkdocs_exclude/plugin.py
@@ -6,12 +6,14 @@ import mkdocs
 import mkdocs.plugins
 import mkdocs.structure.files
 
+from mkdocs.config.config_options import Type
+
 class Exclude(mkdocs.plugins.BasePlugin):
     """A mkdocs plugin that removes all matching files from the input list."""
 
     config_scheme = (
-        ('glob', mkdocs.config.config_options.Type((str, list), default=None)),
-        ('regex', mkdocs.config.config_options.Type((str, list), default=None)),
+        ('glob', Type((str, list), default=None)),
+        ('regex', Type((str, list), default=None)),
     )
 
     def on_files(self, files, config):
