--- Makefile.in.orig	2017-05-11 21:05:18.000000000 -0600
+++ Makefile.in	2017-05-11 21:05:33.000000000 -0600
@@ -383,15 +383,13 @@
              missing mkinstalldirs \
              compile
 
-SUBDIRS = bindings bitmaps \
+SUBDIRS = bitmaps \
              config \
              localized \
              lib \
              include \
              tools \
-             clients \
-             doc \
-             demos
+             clients
 
 AUTOMAKE_OPTIONS = 1.4
 ACLOCAL_AMFLAGS = -I .
