diff --git a/demos/programs/workspace/xrmLib.c b/demos/programs/workspace/xrmLib.c
index e3f56bd..d056e03 100644
--- a/demos/programs/workspace/xrmLib.c
+++ b/demos/programs/workspace/xrmLib.c
@@ -30,7 +30,7 @@ static char rcsid[] = "$XConsortium: xrmLib.c /main/6 1995/07/14 10:01:41 drk $"
 #endif

 #include <stdio.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include <Xm/Xm.h>
 #include "wsm.h"
 #include "wsmDebug.h"
