diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -17,8 +17,7 @@ setup(name='rl-games',
       long_description_content_type="text/markdown",
       url="https://github.com/Denys88/rl_games",
       #packages=[package for package in find_packages() if package.startswith('rl_games')],
-      packages = ['.','rl_games','docs'],
-      package_data={'rl_games':['*','*/*','*/*/*'],'docs':['*','*/*','*/*/*'],},
+      packages = ['.','rl_games'],
       version='1.6.1',
       author='Denys Makoviichuk, Viktor Makoviichuk',
       author_email='trrrrr97@gmail.com, victor.makoviychuk@gmail.com',
@@ -32,7 +31,7 @@ setup(name='rl-games',
             "Programming Language :: Python :: 3.10"
       ],
       #packages=["rlg"],
-      include_package_data=True,
+      include_package_data=False,
       install_requires=[
             # this setup is only for pytorch
             # 
