diff --git a/build.sh b/build.sh
index e49c132e5..a035474bc 100755
--- a/build.sh
+++ b/build.sh
@@ -412,35 +412,8 @@ if [ "$PYTHON" != "" ] && [ "$PYTHON" != "-" ]; then
 
   # Get Python installation files
   if [ "$PYTHONLIBFILE" = "" ]; then
-    PYTHONLIBFILENAME=`$PYTHON -c "import sysconfig; print(sysconfig.get_config_vars('LDLIBRARY')[0])" 2>/dev/null`
-    if [ "$PYTHONLIBFILENAME" = "" ]; then
-      PYTHONLIBFILENAME=`$PYTHON -c "import sysconfig; print(sysconfig.get_config_vars('LIBRARY')[0])" 2>/dev/null`
-    fi
-    if [ "$PYTHONLIBFILENAME" = "" ]; then
-      PYTHONLIBFILENAME=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_vars('LDLIBRARY')[0])" 2>/dev/null`
-    fi
-    if [ "$PYTHONLIBFILENAME" = "" ]; then
-      PYTHONLIBFILENAME=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_vars('LIBRARY')[0])" 2>/dev/null`
-    fi
-    if [ "$PYTHONLIBFILENAME" = "" ]; then
-      echo "*** WARNING: Could not get Python library name"
-    else
-      PYTHONLIBFILEPATH=`$PYTHON -c "import sysconfig; print(sysconfig.get_config_vars('LIBDIR')[0])" 2>/dev/null`
-      if [ "$PYTHONLIBFILEPATH" = "" ]; then
-        PYTHONLIBFILEPATH=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_vars('LIBDIR')[0])" 2>/dev/null`
-      fi
-      if [ "$PYTHONLIBFILEPATH" = "" ]; then
-        echo "*** WARNING: Could not get Python library path"
-      else
-        PYTHONLIBFILE="$PYTHONLIBFILEPATH/$PYTHONLIBFILENAME"
-        if [ ! -f "$PYTHONLIBFILE" ]; then
-          echo "    INFO: Python library not in default path, trying to use MULTIARCH"
-          PYTHONMULTIARCH=`$PYTHON -c "import sysconfig; print(sysconfig.get_config_vars('MULTIARCH')[0])" 2>/dev/null`
-          PYTHONLIBFILE="$PYTHONLIBFILEPATH/$PYTHONMULTIARCH/$PYTHONLIBFILENAME"
-        fi
-        echo "    Python library found: $PYTHONLIBFILE"
-      fi
-    fi
+    PYTHONLIBFILE=-lpython`$PYTHON -c "import sysconfig; print(sysconfig.get_config_vars('LDVERSION')[0])" 2>/dev/null`
+    echo "    Python library found: $PYTHONLIBFILE"
   fi
 
   if [ "$PYTHONINCLUDE" = "" ]; then
@@ -537,8 +510,8 @@ if [ "$PYTHONINCLUDE" != "" ]; then
     echo "*** ERROR: wrong -pyinc path: $PYTHONINCLUDE/Python does not exist or is not a file"
     exit 1
   fi
-  if [ ! -f "$PYTHONLIBFILE" ]; then
-    echo "*** ERROR: wrong -pylib path: $PYTHONLIBFILE does not exist or not a file"
+  if [[ ! ${PYTHONLIBFILE} == -lpython3* ]]; then
+    echo "*** ERROR: wrong -pylib path: $PYTHONLIBFILE does not look like a python library name"
     exit 1
   fi
 else
