diff --git a/setup.py b/setup.py
index cc6ac9e..cfc8d98 100644
--- a/setup.py
+++ b/setup.py
@@ -1,19 +1,11 @@
 import setuptools
 
-with open("README.md", "r") as fh:
-    long_description = fh.read()
-
-with open("requirements.txt", 'r') as dependencies:
-    requirements = [pkg.strip() for pkg in dependencies]
-
 setuptools.setup(
     name="zeus-mcmc",
     version="2.3.0",
     author="Minas Karamanis",
     author_email="minaskar@gmail.com",
     description="zeus: Lightning Fast MCMC",
-    long_description=long_description,
-    long_description_content_type="text/markdown",
     url="https://github.com/minaskar/zeus",
     packages=setuptools.find_packages(),
     classifiers=[
@@ -24,6 +16,5 @@ setuptools.setup(
         'Topic :: Scientific/Engineering',
         'Topic :: Scientific/Engineering :: Mathematics',
     ],
-    install_requires=requirements,
     python_requires='>=3.6',
 )

