diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 96ca496..945100a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -40,7 +40,7 @@ if (JPEG_LIBRARY)
     target_link_libraries       (leptonica ${JPEG_LIBRARY})
 endif()
 if (JP2K_FOUND)
-    target_include_directories  (leptonica PUBLIC ${JP2K_INCLUDE_DIR})
+    target_include_directories  (leptonica PUBLIC ${JP2K_INCLUDE_DIRS})
     target_link_libraries       (leptonica ${JP2K_LIBRARIES})
 endif()
 if (PNG_LIBRARY)
@@ -52,7 +52,7 @@ if (TIFF_LIBRARY)
     target_link_libraries       (leptonica ${TIFF_LIBRARY})
 endif()
 if (WEBP_FOUND)
-    target_include_directories  (leptonica PUBLIC ${WEBP_INCLUDE_DIR})
+    target_include_directories  (leptonica PUBLIC ${WEBP_INCLUDE_DIRS})
     target_link_libraries       (leptonica ${WEBP_LIBRARIES})
 endif()
 if (ZLIB_LIBRARY)
