diff --git a/setup.py b/setup.py
index f51d154..3d7e65e 100644
--- a/setup.py
+++ b/setup.py
@@ -1,10 +1,10 @@
 import os
 
 from setuptools import setup, find_packages
-from pip.req import parse_requirements
-
-install_reqs = parse_requirements('requirements.txt', session=False)
-test_reqs = parse_requirements('requirements_test.txt', session=False)
+try: # for pip >= 10
+    from pip._internal.req import parse_requirements
+except ImportError: # for pip <= 9.0.3
+    from pip.req import parse_requirements
 
 version = '0.6.1'
 
@@ -28,10 +28,6 @@ setup(
     packages=find_packages(),
     url='https://github.com/capless/warrant',
     license='Apache License 2.0',
-    install_requires=[str(ir.req) for ir in install_reqs],
-    extras_require={
-        'test': [str(ir.req) for ir in test_reqs]
-    },
     include_package_data=True,
     zip_safe=True,
 
