
---
 setup.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 4afab33..a494b24 100644
--- a/setup.py
+++ b/setup.py
@@ -22,9 +22,12 @@ from astropy_helpers.git_helpers import get_git_devstr
 from astropy_helpers.version_helpers import generate_version_py
 
 # Get some values from the setup.cfg
-from distutils import config
+try:
+    from ConfigParser import ConfigParser
+except ImportError:
+    from configparser import ConfigParser
 
-conf = config.ConfigParser()
+conf = ConfigParser()
 # Make it case sensitive
 conf.optionxform = str
 
-- 
