--- tools/ign_TEST.cc
+++ tools/ign_TEST.cc
@@ -17,6 +17,7 @@
 
 #include <iostream>
 #include <string>
+#include <stdio.h>
 #include <ignition/msgs.hh>
 
 #include "gtest/gtest.h"
@@ -25,6 +26,11 @@
 
 using namespace ignition;
 
+#ifdef _WIN32
+#define popen _popen
+#define pclose _pclose
+#endif
+
 static std::string g_partition;
 static std::string g_topicCBStr;
 static const std::string g_ignVersion("--force-version " +
