From a57b745e5bf91abf042f93dfa0e5f4a626c0875a Mon Sep 17 00:00:00 2001
From: Matus Novak <email@matusnovak.com>
Date: Wed, 25 Nov 2020 12:12:54 +0100
Subject: [PATCH] Do not install nlohmann json if it is not embedded (#177)

---
 CMakeLists.txt | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a5b2abf..87691d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -161,11 +161,13 @@ if(INJA_INSTALL)
     DESTINATION ${INJA_INSTALL_INCLUDE_DIR}
     FILES_MATCHING PATTERN "*.hpp"
   )
-  install(
-    DIRECTORY third_party/include/nlohmann
-    DESTINATION ${INJA_INSTALL_INCLUDE_DIR}
-    FILES_MATCHING PATTERN "*.hpp"
-  )
+  if(INJA_USE_EMBEDDED_JSON)
+    install(
+      DIRECTORY third_party/include/nlohmann
+      DESTINATION ${INJA_INSTALL_INCLUDE_DIR}
+      FILES_MATCHING PATTERN "*.hpp"
+    )
+  endif()
   install(
     FILES
       "${CMAKE_CURRENT_BINARY_DIR}/${INJA_CONFIG_PATH}/injaConfig.cmake"
-- 
2.17.1 (Apple Git-112)

