From 57a50c34ceee7a5e096e724f71c3086d8c61e8ad Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 26 Feb 2024 09:03:54 +0100
Subject: [PATCH] Update Makefile

---
 src/Makefile | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index c9f7eb5..bb7c99d 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,14 +1,10 @@
-F90=gfortran
-#F90=ifort
-DEPS = 
-FLAGS=
 OBJ = preclib.o vars.o read_input.o read_dat.o sigma_real.o sum_rule.o sigma_real_dc.o kg4vasp.o
 
 %.o: %.f90 $(DEPS)
-	$(F90) -c -o $@ $< $(FLAGS)
+	$(F90) -c -o $@ $< $(FFLAGS)
 
 kg4vasp.x: $(OBJ)
-	$(F90) -o $@ $^ $(FLAGS)
+	$(F90) -o $@ $^ $(FFLAGS)
 
 
 PHONY: clean
