From fe972242ff792c9261ee43cf0b9af07b2f720499 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sat, 18 Jul 2026 23:04:59 +0200
Subject: [PATCH] fix: link pthread into pacparser for glibc < 2.34

---
 src/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index da867b8..b9cd8c1 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -92,7 +92,7 @@ pacparser.o: pacparser.c pac_utils.h pacparser.h
 	touch pymod/pacparser_o_buildstamp
 
 $(LIBRARY): pacparser.o quickjs/libquickjs.a
-	$(MKSHLIB) $(MAINT_CFLAGS) $(CFLAGS) $(LDFLAGS) $(LIB_OPTS) -o $(LIBRARY) pacparser.o quickjs/libquickjs.a -lm
+	$(MKSHLIB) $(MAINT_CFLAGS) $(CFLAGS) $(LDFLAGS) $(LIB_OPTS) -o $(LIBRARY) pacparser.o quickjs/libquickjs.a -lm -lpthread
 
 libpacparser.a: pacparser.o quickjs/libquickjs.a
 	cp quickjs/libquickjs.a libpacparser.a
@@ -102,7 +102,7 @@ $(LIBRARY_LINK): $(LIBRARY)
 	ln -sf $(LIBRARY) $(LIBRARY_LINK)
 
 pactester: pactester.c pacparser.h libpacparser.a
-	$(CC) $(MAINT_CFLAGS) $(CFLAGS) $(LDFLAGS) pactester.c libpacparser.a -o pactester -lm -L. -I.
+	$(CC) $(MAINT_CFLAGS) $(CFLAGS) $(LDFLAGS) pactester.c libpacparser.a -o pactester -lm -lpthread -L. -I.
 
 testpactester: pactester $(LIBRARY_LINK)
 	echo "Running tests for pactester."
