diff --git a/fdep/fortran_dependencies.mk b/fdep/fortran_dependencies.mk
index 7e1eaff..1a55047 100644
--- a/fdep/fortran_dependencies.mk
+++ b/fdep/fortran_dependencies.mk
@@ -99,7 +99,7 @@ endif
 # $1 program
 define program_dependencies
 	$(_f90_only_verbose){ $(foreach argument,$(_$p_use_mods) $(_$p_def_mods) $(foreach l,$(call recursive_lib_deps,$p),$(_$l_use_mods) $(_$l_def_mods)),echo $(argument); ) true; } | \
-	$(top_srcdir)/fdep/fortran_dependencies.pl $p >> $@ || { rm $@; exit 1; }
+	perl -w $(top_srcdir)/fdep/fortran_dependencies.pl $p >> $@ || { rm $@; exit 1; }

 endef

