From 7b26c01bbe1586afc19e123ea6d1bb1afd747069 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 13 Nov 2024 19:09:58 +0100
Subject: [PATCH 1/2] Add BUILD_WITH_DEFAULT_MSVC_RUNTIME_LIBRARY option

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 899d057..c2db1a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,6 +38,8 @@ option(BUILD_MCMODELLARGE "Build the library linked with mcmodel=large flag, use
 option(BUILD_LICCHECK_PRINT "Build libraries with licence check flag" OFF)
 option(BUILD_DYNRT_LIBS "Build the libraries linked against the dynamic runtime (windows only)" OFF)
 option(BUILD_MT_LIBS "Build the multithreaded library" OFF)
+option(BUILD_WITH_DEFAULT_MSVC_RUNTIME_LIBRARY "If ON, do not manually set MSVC_RUNTIME_LIBRARY property" OFF)
+mark_as_advanced(BUILD_WITH_DEFAULT_MSVC_RUNTIME_LIBRARY)
 option(BUILD_CPP "Build the cpp interface" OFF)
 option(BUILD_F2C "Build the f2c library" OFF)
 if (${CMAKE_VERSION} VERSION_LESS "3.9.0")
@@ -346,7 +348,7 @@ macro(createSingleASL name sourcedir sources)
   if(NOT WIN32)
     target_link_libraries(${name} PRIVATE m)
   endif()
-  if(MSVC)
+  if(MSVC AND NOT BUILD_WITH_DEFAULT_MSVC_RUNTIME_LIBRARY)
     if(${name} MATCHES "dynrt")
 	set_property(TARGET ${name} PROPERTY
             MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")

From 3e1987ac6eb3ccb017f5ec348a9b6fcc9ccb7826 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 13 Nov 2024 19:32:25 +0100
Subject: [PATCH 2/2] Update CMakeLists.txt

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2db1a5..f7891ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -356,6 +356,8 @@ macro(createSingleASL name sourcedir sources)
 	set_property(TARGET ${name} PROPERTY
 	    MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
     endif()
+  endif()
+  if(MSVC)
     target_compile_options(${name} PRIVATE 
       /wd4013 /wd4018 /wd4101 /wd4244 /wd4273 /wd4267 /wd4996)
   else()
