From 2c0d77baf0bc0232ea4d6d05e46c1432aab89b3a Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 13 Nov 2024 19:53:48 +0100
Subject: [PATCH] Install library in default location

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 899d057..521e7b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -464,7 +464,7 @@ if(NOT ASL_SKIP_INSTALL)
   install(FILES ${ASL2_HEADERS} ${ASL2_SOURCE_DIR}/opcode.hd ${ASL2_SOURCE_DIR}/r_opn.hd
                 ${GENERATED_INCLUDE_DIR}/stdio1.h ${GENERATED_INCLUDE_DIR}/arith.h
                 DESTINATION include/asl2  COMPONENT asl)
-  install(TARGETS asl asl2 EXPORT ampl-asl-config DESTINATION lib COMPONENT asl)
+  install(TARGETS asl asl2 EXPORT ampl-asl-config COMPONENT asl)
   if(BUILD_MT_LIBS)
     install(TARGETS asl-mt asl2-mt EXPORT ampl-asl-config DESTINATION lib COMPONENT asl)
   endif()
