From f3ec0330fe684ba4ce6cd130fb156ee140cafb60 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Wed, 3 Apr 2024 18:15:19 -0500
Subject: [PATCH] patch Makefile

---
 Makefile | 16 ++++------------
 1 file changed, 4 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index 95ccdc3..fd46e01 100644
--- a/Makefile
+++ b/Makefile
@@ -4,10 +4,6 @@
 
 default: build/debug/mujs build/debug/mujs-pp
 
-CFLAGS = -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter
-
-OPTIM = -O3
-
 prefix = /usr/local
 bindir = $(prefix)/bin
 incdir = $(prefix)/include
@@ -25,10 +21,6 @@ else
   SO = so
 endif
 
-ifeq ($(shell uname),FreeBSD)
-  CFLAGS += -I/usr/local/include -L/usr/local/lib
-endif
-
 HDRS = mujs.h jsi.h regexp.h utf.h astnames.h opnames.h
 
 ifneq ($(HAVE_READLINE),no)
@@ -97,16 +89,16 @@ build/debug/mujs-pp: pp.c build/debug/libmujs.o
 
 build/release/libmujs.$(SO): one.c $(SRCS) $(HDRS)
 	@mkdir -p $(@D)
-	$(CC) $(CFLAGS) $(OPTIM) -fPIC -shared -o $@ one.c -lm
+	$(CC) $(CFLAGS) -fPIC -shared -o $@ one.c -lm
 build/release/libmujs.o: one.c $(SRCS) $(HDRS)
 	@mkdir -p $(@D)
-	$(CC) $(CFLAGS) $(OPTIM) -c -o $@ one.c
+	$(CC) $(CFLAGS) -c -o $@ one.c
 build/release/libmujs.a: build/release/libmujs.o
 	$(AR) cr $@ $^
 build/release/mujs: main.c build/release/libmujs.o
-	$(CC) $(CFLAGS) $(OPTIM) -o $@ $^ -lm $(READLINE_CFLAGS) $(READLINE_LIBS)
+	$(CC) $(CFLAGS) -o $@ $^ -lm $(READLINE_CFLAGS) $(READLINE_LIBS)
 build/release/mujs-pp: pp.c build/release/libmujs.o
-	$(CC) $(CFLAGS) $(OPTIM) -o $@ $^ -lm
+	$(CC) $(CFLAGS) -o $@ $^ -lm
 
 build/release/mujs.pc:
 	@mkdir -p $(@D)
