From 056e5dfbebbea6e4599a77e411419a5bf40e31cd Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Wed, 3 Apr 2024 18:33:53 -0500
Subject: [PATCH] Update Makefile

---
 Makefile | 20 ++++++--------------
 1 file changed, 6 insertions(+), 14 deletions(-)

diff --git a/Makefile b/Makefile
index 95ccdc3..49dd8e4 100644
--- a/Makefile
+++ b/Makefile
@@ -4,10 +4,6 @@
 
 default: build/debug/mujs build/debug/mujs-pp
 
-CFLAGS = -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter
-
-OPTIM = -O3
-
 prefix = /usr/local
 bindir = $(prefix)/bin
 incdir = $(prefix)/include
@@ -25,10 +21,6 @@ else
   SO = so
 endif
 
-ifeq ($(shell uname),FreeBSD)
-  CFLAGS += -I/usr/local/include -L/usr/local/lib
-endif
-
 HDRS = mujs.h jsi.h regexp.h utf.h astnames.h opnames.h
 
 ifneq ($(HAVE_READLINE),no)
@@ -80,7 +72,7 @@ utfdata.h: genucd.py UnicodeData.txt
 
 build/sanitize/mujs: main.c one.c $(SRCS) $(HDRS)
 	@mkdir -p $(@D)
-	$(CC) $(CFLAGS) -g -fsanitize=address -fno-omit-frame-pointer -o $@ main.c one.c -lm $(READLINE_CFLAGS) $(READLINE_LIBS)
+	$(CC) $(CFLAGS) -g -fsanitize=address -fno-omit-frame-pointer -o $@ main.c one.c -L$(libdir) -lm $(READLINE_CFLAGS) $(READLINE_LIBS)
 
 build/debug/libmujs.$(SO): one.c $(SRCS) $(HDRS)
 	@mkdir -p $(@D)
@@ -91,22 +83,22 @@ build/debug/libmujs.o: one.c $(SRCS) $(HDRS)
 build/debug/libmujs.a: build/debug/libmujs.o
 	$(AR) cr $@ $^
 build/debug/mujs: main.c build/debug/libmujs.o
-	$(CC) $(CFLAGS) -g -o $@ $^ -lm $(READLINE_CFLAGS) $(READLINE_LIBS)
+	$(CC) $(CFLAGS) -g -o $@ $^ -L$(libdir) -lm $(READLINE_CFLAGS) $(READLINE_LIBS)
 build/debug/mujs-pp: pp.c build/debug/libmujs.o
 	$(CC) $(CFLAGS) -g -o $@ $^ -lm
 
 build/release/libmujs.$(SO): one.c $(SRCS) $(HDRS)
 	@mkdir -p $(@D)
-	$(CC) $(CFLAGS) $(OPTIM) -fPIC -shared -o $@ one.c -lm
+	$(CC) $(CFLAGS) -fPIC -shared -o $@ one.c -lm
 build/release/libmujs.o: one.c $(SRCS) $(HDRS)
 	@mkdir -p $(@D)
-	$(CC) $(CFLAGS) $(OPTIM) -c -o $@ one.c
+	$(CC) $(CFLAGS) -c -o $@ one.c
 build/release/libmujs.a: build/release/libmujs.o
 	$(AR) cr $@ $^
 build/release/mujs: main.c build/release/libmujs.o
-	$(CC) $(CFLAGS) $(OPTIM) -o $@ $^ -lm $(READLINE_CFLAGS) $(READLINE_LIBS)
+	$(CC) $(CFLAGS) -o $@ $^ -L$(libdir) -lm $(READLINE_CFLAGS) $(READLINE_LIBS)
 build/release/mujs-pp: pp.c build/release/libmujs.o
-	$(CC) $(CFLAGS) $(OPTIM) -o $@ $^ -lm
+	$(CC) $(CFLAGS) -o $@ $^ -lm
 
 build/release/mujs.pc:
 	@mkdir -p $(@D)
