--- setup.py.orig  2024-07-19 14:52:16.008520276 -0500
+++ setup.py       2024-07-19 14:54:03.834496484 -0500
@@ -8,13 +8,2 @@

-class CustomBuild(build_ext):
-    def run(self):
-        if system() == "Windows":
-            try:
-                check_call(["blst\\build.bat"])
-            except Exception:
-                pass
-        check_call(["make", "-C", "src", "blst"])
-        super().run()
-
-
 def main():
@@ -24,2 +13,14 @@

+    so = os.getenv('BLST_MAJOR_VERSION')
+    h_prefix = os.getenv('PREFIX')
+    h_prefix = os.path.join(h_prefix, "Library") if system() == "Windows" else h_prefix
+    b_prefix = os.getenv('BUILD_PREFIX')
+    b_prefix = os.path.join(b_prefix, "Library") if system() == "Windows" else b_prefix
+
+    blst = f"blst-{so}.lib" if system() == "Windows" else None
+    blst = f"libblst.so.{so}" if system() == "Linux" else blst
+    blst = f"libblst.{so}.dylib" if system() == "Darwin" else blst
+    print(f"Using {blst} as the blst library ({system()}).")
+    blst = "blst" if blst is None else blst
+
     setup(
@@ -38,5 +38,6 @@
                 sources=["bindings/python/ckzg_wrap.c", "src/ckzg.c"],
-                include_dirs=["inc", "src"],
-                library_dirs=["lib"],
-                libraries=["blst"]
+                include_dirs=["inc", "src", os.path.join(b_prefix, "include"), os.path.join(h_prefix, "include")],
+                library_dirs=["lib", os.path.join(b_prefix, "lib"), os.path.join(h_prefix, "lib")],
+                extra_objects=[os.path.join(h_prefix, "lib", blst)],
+                # libraries=[blst],
             )
@@ -44,3 +44,3 @@
         cmdclass={
-            "build_ext": CustomBuild,
+            "build_ext": build_ext,
         }
