From 467d4fd86bb6cc214bcd0cee8efd810744766d34 Mon Sep 17 00:00:00 2001
From: Olivier Roussel <olivier.roussel@inria.fr>
Date: Wed, 7 Jun 2023 14:47:34 +0200
Subject: [PATCH] Disable AVX and AVX2 for Simd embedded lib.

---
 3rdparty/simdlib/Simd/SimdConfig.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/3rdparty/simdlib/Simd/SimdConfig.h b/3rdparty/simdlib/Simd/SimdConfig.h
index 776368f06..4f2880526 100755
--- a/3rdparty/simdlib/Simd/SimdConfig.h
+++ b/3rdparty/simdlib/Simd/SimdConfig.h
@@ -26,9 +26,9 @@
 
 //#define SIMD_SSE41_DISABLE
 
-//#define SIMD_AVX_DISABLE
+#define SIMD_AVX_DISABLE
 
-//#define SIMD_AVX2_DISABLE
+#define SIMD_AVX2_DISABLE
 
 //#define SIMD_NEON_DISABLE
 
-- 
2.34.1

