diff --git a/pckg/pygpc_extensions_cuda/src/pygpc_extensions_cuda.cpp b/pckg/pygpc_extensions_cuda/src/pygpc_extensions_cuda.cpp
index 76ce2e7..95ebaf2 100755
--- a/pckg/pygpc_extensions_cuda/src/pygpc_extensions_cuda.cpp
+++ b/pckg/pygpc_extensions_cuda/src/pygpc_extensions_cuda.cpp
@@ -28,19 +28,19 @@ static PyObject* create_gpc_matrix_cuda(PyObject* self, PyObject* args)
     result = PyArray_FROM_OTF(py_result, NPY_DOUBLE,
         NPY_ARRAY_OUT_ARRAY);
 
-    npy_intp* ptr_dim_arguments = PyArray_DIMS(arguments);
+    npy_intp* ptr_dim_arguments = PyArray_DIMS((PyArrayObject*)arguments);
     npy_intp n_arguments = ptr_dim_arguments[0];
     npy_intp n_dim = ptr_dim_arguments[1];
-    double* ptr_arguments = (double*)PyArray_DATA(arguments);
+    double* ptr_arguments = (double*)PyArray_DATA((PyArrayObject*)arguments);
 
-    npy_intp* ptr_dim_result = PyArray_DIMS(result);
+    npy_intp* ptr_dim_result = PyArray_DIMS((PyArrayObject*)result);
     npy_intp n_basis = ptr_dim_result[1];
     npy_intp n_grad = ptr_dim_result[2];
-    double* ptr_result = (double*)PyArray_DATA(result);
+    double* ptr_result = (double*)PyArray_DATA((PyArrayObject*)result);
 
-    npy_intp* ptr_dim_coeffs = PyArray_DIMS(coeffs);
+    npy_intp* ptr_dim_coeffs = PyArray_DIMS((PyArrayObject*)coeffs);
     npy_intp n_coeffs = ptr_dim_coeffs[0];
-    double* ptr_coeffs = (double*)PyArray_DATA(coeffs);
+    double* ptr_coeffs = (double*)PyArray_DATA((PyArrayObject*)coeffs);
 
     create_gpc_matrix_wrapper(ptr_arguments,
         ptr_coeffs, ptr_result, n_arguments, n_dim, n_basis, n_grad, n_coeffs);
@@ -78,21 +78,21 @@ static PyObject* get_approximation_cuda(PyObject* self, PyObject* args)
     result = PyArray_FROM_OTF(py_result, NPY_DOUBLE,
         NPY_ARRAY_OUT_ARRAY);
 
-    npy_intp* ptr_dim_arguments = PyArray_DIMS(arguments);
+    npy_intp* ptr_dim_arguments = PyArray_DIMS((PyArrayObject*)arguments);
     npy_intp n_arguments = ptr_dim_arguments[0];
     npy_intp n_dim = ptr_dim_arguments[1];
-    double* ptr_arguments = (double*)PyArray_DATA(arguments);
+    double* ptr_arguments = (double*)PyArray_DATA((PyArrayObject*)arguments);
 
-    npy_intp* ptr_dim_poly_coeffs = PyArray_DIMS(poly_coeffs);
+    npy_intp* ptr_dim_poly_coeffs = PyArray_DIMS((PyArrayObject*)poly_coeffs);
     npy_intp n_poly_coeffs = ptr_dim_poly_coeffs[0];
-    double* ptr_poly_coeffs = (double*)PyArray_DATA(poly_coeffs);
+    double* ptr_poly_coeffs = (double*)PyArray_DATA((PyArrayObject*)poly_coeffs);
 
-    npy_intp* ptr_dim_gpc_coeffs = PyArray_DIMS(gpc_coeffs);
+    npy_intp* ptr_dim_gpc_coeffs = PyArray_DIMS((PyArrayObject*)gpc_coeffs);
     npy_intp n_basis = ptr_dim_gpc_coeffs[0];
     npy_intp n_gpc_coeffs = ptr_dim_gpc_coeffs[1];
-    double* ptr_gpc_coeffs = (double*)PyArray_DATA(gpc_coeffs);
+    double* ptr_gpc_coeffs = (double*)PyArray_DATA((PyArrayObject*)gpc_coeffs);
 
-    double* ptr_result = (double*)PyArray_DATA(result);
+    double* ptr_result = (double*)PyArray_DATA((PyArrayObject*)result);
     
     get_approximation_wrapper(ptr_arguments, ptr_poly_coeffs, ptr_gpc_coeffs,
         ptr_result, n_arguments, n_dim, n_basis, n_poly_coeffs, n_gpc_coeffs);
