diff --git a/src/sysfiles.c b/src/sysfiles.c
index 4a81595..cb75df9 100644
--- a/src/sysfiles.c
+++ b/src/sysfiles.c
@@ -141,10 +141,10 @@ libGAP_Int libGAP_SyFindOrLinkGapRootFile (
     libGAP_Int                 found_gap = 0;
     libGAP_Int                 found_dyn = 0;
     libGAP_Int                 found_sta = 0;
-    libGAP_Char                tmpbuffer[256];
+    libGAP_Char                tmpbuffer[512];
     libGAP_Char *              tmp;
-    libGAP_Char                libGAP_module[256];
-    libGAP_Char                name[256];
+    libGAP_Char                libGAP_module[512];
+    libGAP_Char                name[512];
     libGAP_StructInitInfo *    info_dyn = 0;
     libGAP_StructInitInfo *    info_sta = 0;
     libGAP_Int                 k;
@@ -809,7 +809,7 @@ libGAP_Char * libGAP_SyWinCmd (
 **  stores buffers for the relatively few files that need them.
 */
 
-libGAP_SYS_SY_BUF libGAP_syBuf [256];
+libGAP_SYS_SY_BUF libGAP_syBuf [512];
 
 libGAP_SYS_SY_BUFFER libGAP_syBuffers [ 32];
 
@@ -1773,7 +1773,7 @@ void libGAP_syEchos (
 **  'SyFputs' is called to put the  <line>  to the file identified  by <fid>.
 */
 libGAP_UInt   libGAP_syNrchar;                        /* nr of chars already on the line */
-libGAP_Char   libGAP_syPrompt [256];                  /* characters already on the line   */
+libGAP_Char   libGAP_syPrompt [512];                  /* characters already on the line   */
 
 
 
@@ -3913,9 +3913,9 @@ libGAP_Char * libGAP_SyFindGapRootFile ( const libGAP_Char * filename, libGAP_Ch
     for ( k=0;  k<sizeof(libGAP_SyGapRootPaths)/sizeof(libGAP_SyGapRootPaths[0]);  k++ ) {
         if ( libGAP_SyGapRootPaths[k][0] ) {
             result[0] = '\0';
-            if (libGAP_strlcpy( result, libGAP_SyGapRootPaths[k], 256 ) >= 256)
+            if (libGAP_strlcpy( result, libGAP_SyGapRootPaths[k], 512 ) >= 512)
                 continue;
-            if (libGAP_strlcat( result, filename, 256 ) >= 256)
+            if (libGAP_strlcat( result, filename, 512 ) >= 512)
             	continue;
             if ( libGAP_SyIsReadableFile(result) == 0 ) {
                 return result;
diff --git a/src/sysfiles.h b/src/sysfiles.h
index 6648256..bd05e33 100644
--- a/src/sysfiles.h
+++ b/src/sysfiles.h
@@ -47,7 +47,7 @@
 */
 
 typedef union {
-  libGAP_Char pathname[256];
+  libGAP_Char pathname[512];
   libGAP_StructInitInfo * module_info;
 } libGAP_TypGRF_Data;
 
@@ -155,7 +155,7 @@ typedef struct {
   libGAP_UInt buflen;
 } libGAP_SYS_SY_BUFFER;
 
-extern libGAP_SYS_SY_BUF libGAP_syBuf [256];
+extern libGAP_SYS_SY_BUF libGAP_syBuf [512];
 
 extern libGAP_SYS_SY_BUFFER libGAP_syBuffers[32];
 
diff --git a/src/system.c b/src/system.c
index cd04706..8f1b149 100644
--- a/src/system.c
+++ b/src/system.c
@@ -167,7 +167,7 @@ libGAP_Int libGAP_SyCheckCRCCompiledModule;
 **
 *V  SyCompileInput  . . . . . . . . . . . . . . . . . .  from this input file
 */
-libGAP_Char libGAP_SyCompileInput [256];
+libGAP_Char libGAP_SyCompileInput [512];
 
 
 /****************************************************************************
@@ -181,20 +181,20 @@ libGAP_Char * libGAP_SyCompileMagic1;
 **
 *V  SyCompileName . . . . . . . . . . . . . . . . . . . . . .  with this name
 */
-libGAP_Char libGAP_SyCompileName [256];
+libGAP_Char libGAP_SyCompileName [512];
 
 
 /****************************************************************************
 **
 *V  SyCompileOutput . . . . . . . . . . . . . . . . . . into this output file
 */
-libGAP_Char libGAP_SyCompileOutput [256];
+libGAP_Char libGAP_SyCompileOutput [512];
 
 /****************************************************************************
 **
 *V  SyCompileOutput . . . . . . . . . . . . . . . . . . into this output file
 */
-libGAP_Char libGAP_SyCompileOptions [256] = {'\0'};
+libGAP_Char libGAP_SyCompileOptions [512] = {'\0'};
 
 
 /****************************************************************************
@@ -252,7 +252,7 @@ libGAP_Int libGAP_IgnoreGapRC;
 *V  SyUserHome . . . . . . . . . . . . .  path of users home (it is exists)
 */
 libGAP_Int libGAP_SyHasUserHome;
-libGAP_Char libGAP_SyUserHome [256];
+libGAP_Char libGAP_SyUserHome [512];
 
 /****************************************************************************
 **
@@ -420,7 +420,7 @@ libGAP_Int libGAP_SyStorMin;
 **
 *V  SySystemInitFile  . . . . . . . . . . .  name of the system "init.g" file
 */
-libGAP_Char libGAP_SySystemInitFile [256];
+libGAP_Char libGAP_SySystemInitFile [512];
 
 
 /****************************************************************************
diff --git a/src/system.h b/src/system.h
index 403fe6d..6405677 100644
--- a/src/system.h
+++ b/src/system.h
@@ -267,7 +267,7 @@ extern libGAP_Int libGAP_SyCheckCRCCompiledModule;
 **
 *V  SyCompileInput  . . . . . . . . . . . . . . . . . .  from this input file
 */
-extern libGAP_Char libGAP_SyCompileInput [256];
+extern libGAP_Char libGAP_SyCompileInput [512];
 
 
 /****************************************************************************
@@ -281,20 +281,20 @@ extern libGAP_Char * libGAP_SyCompileMagic1;
 **
 *V  SyCompileName . . . . . . . . . . . . . . . . . . . . . .  with this name
 */
-extern libGAP_Char libGAP_SyCompileName [256];
+extern libGAP_Char libGAP_SyCompileName [512];
 
 
 /****************************************************************************
 **
 *V  SyCompileOutput . . . . . . . . . . . . . . . . . . into this output file
 */
-extern libGAP_Char libGAP_SyCompileOutput [256];
+extern libGAP_Char libGAP_SyCompileOutput [512];
 
 /****************************************************************************
 **
 *V  SyCompileOptions . . . . . . . . . . . . . . . . . with these options
 */
-extern libGAP_Char libGAP_SyCompileOptions [256];
+extern libGAP_Char libGAP_SyCompileOptions [512];
 
 
 /****************************************************************************
@@ -376,7 +376,7 @@ extern libGAP_Char libGAP_SyGapRCFilename [512];
 *V  SyUserHome . . . . . . . . . . . . .  path of users home (it is exists)
 */
 extern libGAP_Int libGAP_SyHasUserHome;
-extern libGAP_Char libGAP_SyUserHome [256];
+extern libGAP_Char libGAP_SyUserHome [512];
 
 
 /****************************************************************************
@@ -544,7 +544,7 @@ extern libGAP_Int libGAP_SyStorMin;
 **
 *V  SySystemInitFile  . . . . . . . . . . .  name of the system "init.g" file
 */
-extern libGAP_Char libGAP_SySystemInitFile [256];
+extern libGAP_Char libGAP_SySystemInitFile [512];
 
 
 /****************************************************************************
