From af623efe51a425cd00fd57306920216a68cdad05 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 15 Oct 2023 23:23:43 -0400
Subject: [PATCH] Get package version and description from conda build

---
 utils/update_build_version.py | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/utils/update_build_version.py b/utils/update_build_version.py
index 5a78ada2..b85a21ae 100755
--- a/utils/update_build_version.py
+++ b/utils/update_build_version.py
@@ -143,13 +143,9 @@ def main():
     changes_file_path = os.path.realpath(sys.argv[1])
     output_file_path = sys.argv[2]
 
-    success, version = deduce_software_version(changes_file_path)
-    if not success:
-      logging.error("Could not deduce latest release version from {}.".format(changes_file_path))
-      sys.exit(1)
+    version = os.environ.get('PKG_VERSION', 'unknown')
+    description = os.environ.get('PKG_VERSION', 'unknown') + ", " + os.environ.get('PKG_BUILDNUM', 'unkown')
 
-    repo_path = os.path.dirname(changes_file_path)
-    description = describe(repo_path)
     content = OUTPUT_FORMAT.format(version_tag=version, description=description)
 
     # Escape file content.
-- 
2.34.1

