From 62590422ced5d115a692f4e9b8127c7fa6c1cf0f Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Thu, 30 Dec 2021 02:22:52 -0500
Subject: [PATCH 2/3] prefix / datadir

---
 Build | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/Build b/Build
index d1d36e399..e73c05bc9 100755
--- a/Build
+++ b/Build
@@ -22,9 +22,10 @@ mydir=`pwd` # make absolute
 
 # allow override of install destination.
 if test -z "$TL_INSTALL_DEST"; then
-  H=`pwd`
-  test -d inst || mkdir -p inst/texmf  # avoid configure warnings
-  TL_INSTALL_DEST=$H/inst
+  H=$PREFIX
+  mkdir -p $PREFIX/texmf  # avoid configure warnings
+  mkdir -p $PREFIX/share
+  TL_INSTALL_DEST=$PREFIX
 fi
 export TL_INSTALL_DEST
 
@@ -124,7 +125,7 @@ cd $TL_WORKDIR || exit 1
 
   eval $TL_BUILD_ENV $TL_CONFIGURE \
         --prefix=$TL_INSTALL_DEST \
-        --datadir=$TL_INSTALL_DEST \
+        --datadir=$TL_INSTALL_DEST/share \
         $TL_CONF_BANNER \
         $TL_CONF_MISSING \
         $TL_CONF_LARGEFILE \
-- 
2.34.1

