From f0fd47fe7efee64e28ab0c817650b8efd7d04a54 Mon Sep 17 00:00:00 2001
From: ngam <67342040+ngam@users.noreply.github.com>
Date: Thu, 30 Dec 2021 02:26:10 -0500
Subject: [PATCH 3/3] add banner

---
 Build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Build b/Build
index e73c05bc9..21606d5e0 100755
--- a/Build
+++ b/Build
@@ -77,7 +77,7 @@ fi
 
 # allow for changing the banner identification, e.g.,
 # --with-banner-add='/SomeDistro'; see the build doc.
-: ${TL_CONF_BANNER=}; export TL_CONF_BANNER
+: ${TL_CONF_BANNER=--with-banner-add='A conda-forge release'}; export TL_CONF_BANNER
 
 # default to supporting large files as much as possible;
 # see comments at --disable-largefile in README.config.
-- 
2.34.1

