diff --git a/python/setup.py b/python/setup.py
index bc2eb4c..f500354 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -162,7 +162,11 @@ try:
         url='https://github.com/apache/spark/tree/master/python',
         packages=['pyspark',
                   'pyspark.mllib',
+                  'pyspark.mllib.linalg',
+                  'pyspark.mllib.stat',
                   'pyspark.ml',
+                  'pyspark.ml.linalg',
+                  'pyspark.ml.param',
                   'pyspark.sql',
                   'pyspark.streaming',
                   'pyspark.bin',

