diff --git a/tests/test_generate.py b/tests/test_generate.py
index aa88114..fadf73e 100644
--- a/tests/test_generate.py
+++ b/tests/test_generate.py
@@ -16,7 +16,7 @@ def test_generate_hide_empty_columns():
     schema = Car.model_json_schema()
     markdown = generate(schema, hide_empty_columns=True)
 
-    with open("tests/model_no-empty-columns.md", "r") as f:
+    with open("tests/model_no-empty-columns.md", "r", encoding="utf-8") as f:
         expected_markdown = f.read()
 
     assert markdown == expected_markdown
@@ -26,7 +26,7 @@ def test_generate_custom_title_no_footer():
     schema = Car.model_json_schema()
     markdown = generate(schema, title="Car (custom title)", footer=False)
 
-    with open("tests/model_custom-title_no-footer.md", "r") as f:
+    with open("tests/model_custom-title_no-footer.md", "r", encoding="utf-8") as f:
         expected_markdown = f.read()
 
     assert markdown == expected_markdown
diff --git a/tests/test_schema_examples.py b/tests/test_schema_examples.py
index aa45a76..487b954 100644
--- a/tests/test_schema_examples.py
+++ b/tests/test_schema_examples.py
@@ -32,10 +32,10 @@ def get_test_cases():
 
 @pytest.mark.parametrize("json_path, md_path, kwargs", get_test_cases())
 def test_schema_examples(json_path, md_path, kwargs):
-    with open(json_path, "r") as f:
+    with open(json_path, "r", encoding="utf-8") as f:
         schema = json.load(f)
 
-    with open(md_path, "r") as f:
+    with open(md_path, "r", encoding="utf-8") as f:
         expected_markdown = f.read()
 
     markdown = jsonschema_markdown.generate(schema, **kwargs)
