From 8cf5766d72b7af5185b7a8097c2483cd04bbd3ce Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Tue, 16 Nov 2021 16:06:35 +0100
Subject: [PATCH] Remove Werror

---
 bin/format | 2 +-
 setup.py   | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/bin/format b/bin/format
index 78cf5a8..c1b9464 100755
--- a/bin/format
+++ b/bin/format
@@ -80,7 +80,7 @@ def fmt(args, files) -> int:
 
 
 def check(args, files) -> int:
-    cmd = (*BASE_CMD, "--dry-run", "-Werror")
+    cmd = (*BASE_CMD, "--dry-run")
 
     needs_reformatted_ct = 0
 
diff --git a/setup.py b/setup.py
index 6b63015..bb2016c 100644
--- a/setup.py
+++ b/setup.py
@@ -75,12 +75,12 @@ class BuildExt(build_ext):
 
     c_opts = {
         "msvc": ["/EHsc", "/O2", "/W4", "/WX", "/std:c++17", "-D_USE_MATH_DEFINES"],
-        "unix": ["-O3", "-fPIC", "-shared", "-fopenmp", "-Wall", "-Wextra", "-Werror"],
+        "unix": ["-O3", "-fPIC", "-shared", "-fopenmp", "-Wall", "-Wextra"],
     }
 
     l_opts = {
         "msvc": ["/WX"],
-        "unix": ["-O3", "-fPIC", "-shared", "-fopenmp", "-Wall", "-Wextra", "-Werror"],
+        "unix": ["-O3", "-fPIC", "-shared", "-fopenmp", "-Wall", "-Wextra"],
     }
     if platform.system() == "Linux" and platform.machine() == "x86_64":
         c_opts["unix"].append("-mavx")
-- 
2.33.1

