diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1..2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -145,7 +145,9 @@ if( NOT ${Boost_USE_AUTO_LINK} )
 	add_definitions( "-DBOOST_ALL_NO_LIB" )
 endif()

-option( Boost_USE_STATIC_LIBS "boost use dynamic libraries" OFF )
+if(NOT DEFINED Boost_USE_STATIC_LIBS)
+	option( Boost_USE_STATIC_LIBS "boost use dynamic libraries" OFF )
+endif()
 if( Boost_USE_STATIC_LIBS )
 	message( STATUS "Boost_USE_STATIC_LIBS=ON" )
 	add_definitions( "-DBOOST_THREAD_USE_LIB" )
