diff --git a/src/primitive3d/Cylinder.h b/src/primitive3d/Cylinder.h
index 3a034057..0f2ba119 100644
--- a/src/primitive3d/Cylinder.h
+++ b/src/primitive3d/Cylinder.h
@@ -7,6 +7,7 @@
 
 #include "SFCGAL/Kernel.h"
 #include "SFCGAL/export.h"
+#include "SFCGAL/Coordinate.h"
 #include <CGAL/Polyhedron_3.h>
 #include <CGAL/Surface_mesh.h>
 #include <optional>
diff --git a/src/primitive3d/Sphere.h b/src/primitive3d/Sphere.h
index 054acced..acf6c63d 100644
--- a/src/primitive3d/Sphere.h
+++ b/src/primitive3d/Sphere.h
@@ -14,6 +14,7 @@
 #include "SFCGAL/Kernel.h"
 #include "SFCGAL/export.h"
 #include "SFCGAL/numeric.h"
+#include "SFCGAL/Coordinate.h"
 
 namespace SFCGAL {
 
