diff --git a/src/Geometry.h b/src/Geometry.h
index e50da9c1..a58c6674 100644
--- a/src/Geometry.h
+++ b/src/Geometry.h
@@ -7,6 +7,7 @@
 #define SFCGAL_GEOMETRY_H_
 
 #include "SFCGAL/config.h"
+#include "SFCGAL/Coordinate.h"
 
 #include <boost/endian/conversion.hpp>
 #include <boost/shared_ptr.hpp>
diff --git a/src/Polygon.cpp b/src/Polygon.cpp
index d5fc6d7e..4b9ce79b 100644
--- a/src/Polygon.cpp
+++ b/src/Polygon.cpp
@@ -3,6 +3,7 @@
 // Copyright (c) 2024-2025, SFCGAL team.
 // SPDX-License-Identifier: LGPL-2.0-or-later
 
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/Polygon.h"
 #include "SFCGAL/GeometryVisitor.h"
 
diff --git a/src/Polygon.h b/src/Polygon.h
index dd1abb6e..383922ce 100644
--- a/src/Polygon.h
+++ b/src/Polygon.h
@@ -15,6 +15,7 @@
 #include <CGAL/Polygon_with_holes_2.h>
 
 #include "SFCGAL/Kernel.h"
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/LineString.h"
 #include "SFCGAL/Surface.h"
 
diff --git a/src/Segment.cpp b/src/Segment.cpp
index eed22556..5cb07f85 100644
--- a/src/Segment.cpp
+++ b/src/Segment.cpp
@@ -1,6 +1,7 @@
 // Copyright (c) 2025-2025, SFCGAL Team.
 // SPDX-License-Identifier: LGPL-2.0-or-later
 
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/Segment.h"
 #include "SFCGAL/Exception.h"
 #include "SFCGAL/LineString.h"
diff --git a/src/Segment.h b/src/Segment.h
index 36542221..f9299c8e 100644
--- a/src/Segment.h
+++ b/src/Segment.h
@@ -6,6 +6,7 @@
 
 #include "SFCGAL/Kernel.h"
 #include "SFCGAL/Point.h"
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/export.h"
 #include "SFCGAL/numeric.h"
 #include <type_traits>
diff --git a/src/algorithm/buffer3D.cpp b/src/algorithm/buffer3D.cpp
index 10ad3d55..e14d9181 100644
--- a/src/algorithm/buffer3D.cpp
+++ b/src/algorithm/buffer3D.cpp
@@ -4,7 +4,6 @@
 // SPDX-License-Identifier: LGPL-2.0-or-later
 
 #include "SFCGAL/algorithm/buffer3D.h"
-#include "SFCGAL/Kernel.h"
 #include "SFCGAL/PolyhedralSurface.h"
 #include "SFCGAL/algorithm/minkowskiSum3D.h"
 #include "SFCGAL/algorithm/union.h"
@@ -12,6 +11,9 @@
 #include "SFCGAL/numeric.h"
 #include "SFCGAL/primitive3d/Cylinder.h"
 #include "SFCGAL/primitive3d/Sphere.h"
+#include "SFCGAL/Kernel.h"
+#include "SFCGAL/export.h"
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/triangulate/triangulatePolygon.h"
 #include <CGAL/Nef_polyhedron_3.h>
 #include <CGAL/Polygon_mesh_processing/corefinement.h>
diff --git a/src/algorithm/buffer3D.h b/src/algorithm/buffer3D.h
index 8500196c..d3bff900 100644
--- a/src/algorithm/buffer3D.h
+++ b/src/algorithm/buffer3D.h
@@ -5,7 +5,10 @@
 #ifndef SFCGAL_BUFFER3D_ALGORITHM
 #define SFCGAL_BUFFER3D_ALGORITHM
 
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/Geometry.h"
+#include "SFCGAL/Kernel.h"
+#include "SFCGAL/export.h"
 #include "SFCGAL/LineString.h"
 #include "SFCGAL/Point.h"
 #include "SFCGAL/PolyhedralSurface.h"
diff --git a/src/algorithm/translate.h b/src/algorithm/translate.h
index 87507a5c..8e9437f7 100644
--- a/src/algorithm/translate.h
+++ b/src/algorithm/translate.h
@@ -7,7 +7,7 @@
 #define SFCGAL_TRANSLATE_H_
 
 #include "SFCGAL/export.h"
-
+#include "SFCGAL/Geometry.h"
 #include "SFCGAL/Kernel.h"
 
 namespace SFCGAL {
diff --git a/src/capi/sfcgal_c.cpp b/src/capi/sfcgal_c.cpp
index 6bcb085d..7420a449 100644
--- a/src/capi/sfcgal_c.cpp
+++ b/src/capi/sfcgal_c.cpp
@@ -3,6 +3,9 @@
 // Copyright (c) 2024-2025, SFCGAL team.
 // SPDX-License-Identifier: LGPL-2.0-or-later
 
+#include "SFCGAL/Kernel.h"
+#include "SFCGAL/export.h"
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/Geometry.h"
 #include "SFCGAL/GeometryCollection.h"
 #include "SFCGAL/LineString.h"
diff --git a/src/capi/sfcgal_c.h b/src/capi/sfcgal_c.h
index bf476b4b..54382720 100644
--- a/src/capi/sfcgal_c.h
+++ b/src/capi/sfcgal_c.h
@@ -6,6 +6,7 @@
 #ifndef SFCGAL_CAPI_H_
 #define SFCGAL_CAPI_H_
 
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/config.h"
 #include <stdbool.h>
 #include <stddef.h>
diff --git a/src/detail/transform/AffineTransform2.cpp b/src/detail/transform/AffineTransform2.cpp
index 597ab727..f04d45dc 100644
--- a/src/detail/transform/AffineTransform2.cpp
+++ b/src/detail/transform/AffineTransform2.cpp
@@ -5,6 +5,7 @@
 
 #include "SFCGAL/detail/transform/AffineTransform2.h"
 
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/Point.h"
 
 #include <utility>
diff --git a/src/detail/transform/AffineTransform2.h b/src/detail/transform/AffineTransform2.h
index 63196ae5..73a779a2 100644
--- a/src/detail/transform/AffineTransform2.h
+++ b/src/detail/transform/AffineTransform2.h
@@ -6,6 +6,7 @@
 #ifndef SFCGAL_TRANSFORM_AFFINETRANSFORM2_H_
 #define SFCGAL_TRANSFORM_AFFINETRANSFORM2_H_
 
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/Kernel.h"
 #include "SFCGAL/Transform.h"
 #include "SFCGAL/config.h"
diff --git a/src/detail/transform/AffineTransform3.cpp b/src/detail/transform/AffineTransform3.cpp
index 2785a69e..2e7cb4fa 100644
--- a/src/detail/transform/AffineTransform3.cpp
+++ b/src/detail/transform/AffineTransform3.cpp
@@ -9,6 +9,7 @@
 #include <CGAL/Aff_transformation_3.h>
 
 #include "SFCGAL/GeometryCollection.h"
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/LineString.h"
 #include "SFCGAL/MultiLineString.h"
 #include "SFCGAL/MultiPoint.h"
diff --git a/src/detail/transform/AffineTransform3.h b/src/detail/transform/AffineTransform3.h
index 92ada7d7..6109516a 100644
--- a/src/detail/transform/AffineTransform3.h
+++ b/src/detail/transform/AffineTransform3.h
@@ -6,6 +6,7 @@
 #ifndef SFCGAL_TRANSFORM_AFFINETRANSFORM3_H_
 #define SFCGAL_TRANSFORM_AFFINETRANSFORM3_H_
 
+#include "SFCGAL/Coordinate.h"
 #include "SFCGAL/Kernel.h"
 #include "SFCGAL/Transform.h"
 #include "SFCGAL/config.h"
diff --git a/src/primitive3d/Cylinder.h b/src/primitive3d/Cylinder.h
index 3a034057..0f2ba119 100644
--- a/src/primitive3d/Cylinder.h
+++ b/src/primitive3d/Cylinder.h
@@ -7,6 +7,7 @@
 
 #include "SFCGAL/Kernel.h"
 #include "SFCGAL/export.h"
+#include "SFCGAL/Coordinate.h"
 #include <CGAL/Polyhedron_3.h>
 #include <CGAL/Surface_mesh.h>
 #include <optional>
diff --git a/src/primitive3d/Sphere.h b/src/primitive3d/Sphere.h
index 054acced..acf6c63d 100644
--- a/src/primitive3d/Sphere.h
+++ b/src/primitive3d/Sphere.h
@@ -14,6 +14,7 @@
 #include "SFCGAL/Kernel.h"
 #include "SFCGAL/export.h"
 #include "SFCGAL/numeric.h"
+#include "SFCGAL/Coordinate.h"
 
 namespace SFCGAL {
 
