diff --git a/pyproject.toml b/pyproject.toml
index 71c6c77..f578b3a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,28 +19,28 @@ classifiers=[
 Homepage = "https://github.com/usda-ars-ussl/rosetta-soil"
 
 [tool.hatch.build]
-packages = ["src/rosetta"]
+packages = ["rosetta"]
 
 [tool.hatch.build.targets.wheel]
 include = [
-  "src/rosetta/data/rose1_mod2_0.npz",
-  "src/rosetta/data/rose1_mod2_1.npz",
-  "src/rosetta/data/rose1_mod3_0.npz",
-  "src/rosetta/data/rose1_mod3_1.npz",
-  "src/rosetta/data/rose1_mod4_0.npz",
-  "src/rosetta/data/rose1_mod4_1.npz",
-  "src/rosetta/data/rose1_mod5_0.npz",
-  "src/rosetta/data/rose1_mod5_1.npz",
-  "src/rosetta/data/rose1_unsat_k.npz",
-  "src/rosetta/data/rose3_mod2_0.npz",
-  "src/rosetta/data/rose3_mod3_0.npz",
-  "src/rosetta/data/rose3_mod4_0.npz",
-  "src/rosetta/data/rose3_mod5_0.npz",
+  "rosetta/data/rose1_mod2_0.npz",
+  "rosetta/data/rose1_mod2_1.npz",
+  "rosetta/data/rose1_mod3_0.npz",
+  "rosetta/data/rose1_mod3_1.npz",
+  "rosetta/data/rose1_mod4_0.npz",
+  "rosetta/data/rose1_mod4_1.npz",
+  "rosetta/data/rose1_mod5_0.npz",
+  "rosetta/data/rose1_mod5_1.npz",
+  "rosetta/data/rose1_unsat_k.npz",
+  "rosetta/data/rose3_mod2_0.npz",
+  "rosetta/data/rose3_mod3_0.npz",
+  "rosetta/data/rose3_mod4_0.npz",
+  "rosetta/data/rose3_mod5_0.npz",
 ]
 
 [tool.hatch.version]
 source = "vcs"
 
 [tool.hatch.build.hooks.version]
-path = "src/rosetta/_version.py"
+path = "rosetta/_version.py"
 
