diff -Naur auto/lib/pcre/conf.bak auto/lib/pcre/conf
--- auto/lib/pcre/conf.bak      2016-11-05 23:40:26.000000000 -0500
+++ auto/lib/pcre/conf  2016-11-06 00:00:31.000000000 -0500
@@ -110,6 +110,22 @@

         if [ $ngx_found = no ]; then

+            # Conda
+
+            ngx_feature="PCRE library in $PREFIX/lib"
+            ngx_feature_path="$PREFIX/include"
+
+            if [ $NGX_RPATH = YES ]; then
+                ngx_feature_libs="-R$PREFIX/lib -L$PREFIX/lib -lpcre"
+            else
+                ngx_feature_libs="-L$PREFIX/lib -lpcre"
+            fi
+
+            . auto/feature
+        fi
+
+        if [ $ngx_found = no ]; then
+
             # FreeBSD port

             ngx_feature="PCRE library in /usr/local/"
