Subject: [PATCH] do not use pybind11 subdir
---
Index: TopologicPythonBindings/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/TopologicPythonBindings/CMakeLists.txt b/TopologicPythonBindings/CMakeLists.txt
--- a/TopologicPythonBindings/CMakeLists.txt	(revision a7169281189d3cd03ab6369462742e4ba3e71ad5)
+++ b/TopologicPythonBindings/CMakeLists.txt	(date 1731830659295)
@@ -2,12 +2,18 @@
 
 project(TopologicPythonBindings CXX)
 set(PROJECT_NAMESPACE Topologic)
+option(USE_CONDA_PYBIND11 "Use pybind11 installed via conda" OFF)
 
-
-# Import pybind11
-set(PYBIND11_FINDPYTHON ON)
-add_subdirectory(pybind11)
-
+if(USE_CONDA_PYBIND11)
+    set(CMAKE_PREFIX_PATH ${CONDA_PREFIX})
+    # Find pybind11 package installed via conda
+    set(PYBIND11_FINDPYTHON ON) # Make sure pybind11 uses Python
+    find_package(pybind11 REQUIRED) # This will look for pybind11 installed via conda
+else ()
+    # Import pybind11 from submodule
+    set(PYBIND11_FINDPYTHON ON)
+    add_subdirectory(pybind11)
+endif()
 
 # Source groups
 set(SRC_FILES 
