diff --git a/route/build/Makefile b/route/build/Makefile
index 70f29f1..ff75cc6 100644
--- a/route/build/Makefile
+++ b/route/build/Makefile
@@ -1,55 +1,55 @@
-#========================================================================
-# Makefile to compile the routing model
-#========================================================================
+##========================================================================
+## Makefile to compile the routing model
+##========================================================================
+##
+##
+##
+##========================================================================
+## PART 0: User configure part
+##========================================================================
+## Define fortran compiler - gfortran, ifort or pgf90
+#FC  =
 #
+## Define the compiler exe
+#FC_EXE =
 #
+## Define core directory below which everything resides
+## parent directory of the 'build' directory
+## do not put space at the end of path
+#F_MASTER =
+#
+## Define the NetCDF libraries and path to include files
+#ifeq "$(FC)" "gfortran"
+# NCDF_PATH = /opt/local
+# LIBNETCDF = -L$(NCDF_PATH)/lib -lnetcdff
+# INCNETCDF = -I$(NCDF_PATH)/include
+#endif
+#ifeq "$(FC)" "ifort"
+# NCDF_PATH = /opt/netcdf4-intel
+# LIBNETCDF = -L$(NCDF_PATH)/lib -lnetcdff
+# INCNETCDF = -I$(NCDF_PATH)/include
+#endif
+#ifeq "$(FC)" "pgf90"
+# NCDF_PATH = /opt/netcdf4-pgf90
+# LIBNETCDF = -L$(NCDF_PATH)/lib -lnetcdff -lnetcdf
+# INCNETCDF = -I$(NCDF_PATH)/include
+#endif
+#
+## Define flags
+#ifeq "$(FC)" "gfortran"
+# FLAGS = -p -g -Wall -ffree-line-length-none -fmax-errors=0 -fbacktrace -fcheck=bounds
+#endif
+#ifeq "$(FC)" "ifort"
+# FLAGS1 = -warn all -noerror_limit -FR -O0 -auto -WB -traceback -g -fltconsistency
+## FLAGS = -static -debug -warn all -check all,noarg_temp_created -FR -O0 -auto -WB -traceback -g -fltconsistency -fpe0
+# FLAGS = -warn all -check all,noarg_temp_created -FR -O2 -auto -WB -traceback -g -fltconsistency -fpe0
+## FLAGS = -debug -warn all -check all,noarg_temp_created -FR -O0 -auto -WB -traceback -g -fltconsistency -fpe0
+#endif
+#ifeq "$(FC)" "pgf90"
+# FLAGS1 = -Bstatic -Mbackslash -g -Mchkptr -Mchkstk -Mpgicoff -Minform=inform -Ktrap=divz,inv -Mprof=lines,time
+# FLAGS = -Mbackslash -g -Mchkptr -Mchkstk -Mpgicoff -Minform=inform -Ktrap=divz,inv
+#endif
 #
-#========================================================================
-# PART 0: User configure part 
-#========================================================================
-# Define fortran compiler - gfortran, ifort or pgf90
-FC  =
-
-# Define the compiler exe
-FC_EXE =
-
-# Define core directory below which everything resides
-# parent directory of the 'build' directory
-# do not put space at the end of path
-F_MASTER =
-
-# Define the NetCDF libraries and path to include files
-ifeq "$(FC)" "gfortran"
- NCDF_PATH = /opt/local
- LIBNETCDF = -L$(NCDF_PATH)/lib -lnetcdff
- INCNETCDF = -I$(NCDF_PATH)/include
-endif
-ifeq "$(FC)" "ifort"
- NCDF_PATH = /opt/netcdf4-intel
- LIBNETCDF = -L$(NCDF_PATH)/lib -lnetcdff
- INCNETCDF = -I$(NCDF_PATH)/include
-endif
-ifeq "$(FC)" "pgf90"
- NCDF_PATH = /opt/netcdf4-pgf90
- LIBNETCDF = -L$(NCDF_PATH)/lib -lnetcdff -lnetcdf
- INCNETCDF = -I$(NCDF_PATH)/include
-endif
-
-# Define flags
-ifeq "$(FC)" "gfortran"
- FLAGS = -p -g -Wall -ffree-line-length-none -fmax-errors=0 -fbacktrace -fcheck=bounds
-endif
-ifeq "$(FC)" "ifort"
- FLAGS1 = -warn all -noerror_limit -FR -O0 -auto -WB -traceback -g -fltconsistency
-# FLAGS = -static -debug -warn all -check all,noarg_temp_created -FR -O0 -auto -WB -traceback -g -fltconsistency -fpe0
- FLAGS = -warn all -check all,noarg_temp_created -FR -O2 -auto -WB -traceback -g -fltconsistency -fpe0
-# FLAGS = -debug -warn all -check all,noarg_temp_created -FR -O0 -auto -WB -traceback -g -fltconsistency -fpe0
-endif
-ifeq "$(FC)" "pgf90"
- FLAGS1 = -Bstatic -Mbackslash -g -Mchkptr -Mchkstk -Mpgicoff -Minform=inform -Ktrap=divz,inv -Mprof=lines,time
- FLAGS = -Mbackslash -g -Mchkptr -Mchkstk -Mpgicoff -Minform=inform -Ktrap=divz,inv
-endif
-
 #========================================================================
 # PART 1: Define directory paths
 #========================================================================
