diff --git a/MANIFEST.in b/MANIFEST.in
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -10,10 +10,6 @@ include pyroomacoustics/data/sofa/mit_kemar_normal_pinna.sofa
 include pyroomacoustics/libroom_src/*.h
 include pyroomacoustics/libroom_src/*.hpp
 include pyroomacoustics/libroom_src/*.cpp
-
-# The compiled extension code rely on Eigen, which is included
-include pyroomacoustics/libroom_src/ext/eigen/COPYING.*
-graft pyroomacoustics/libroom_src/ext/eigen/Eigen
 
 include pyproject.toml
 include requirements.txt
