diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -67,7 +67,10 @@ ext_modules = [
             libroom_src_dir,
             str(get_pybind_include()),
             str(get_pybind_include(user=True)),
-            os.path.join(libroom_src_dir, "ext/eigen"),
+            os.environ.get(
+                "EIGEN_INCLUDE_DIR",
+                os.path.join(libroom_src_dir, "ext/eigen"),
+            ),
         ],
         language="c++",
         extra_compile_args=[
