diff --git a/.cmake/ConfigISCE.cmake b/.cmake/ConfigISCE.cmake
index 9b6d0376..ca0af8d6 100644
--- a/.cmake/ConfigISCE.cmake
+++ b/.cmake/ConfigISCE.cmake
@@ -49,7 +49,7 @@ endfunction()
 
 ##Make sure that a reasonable version of Python is installed
 function(CheckISCEPython)
-    find_package(Python 3.6 COMPONENTS Interpreter Development)
+    find_package(Python 3.6 REQUIRED COMPONENTS Interpreter Development NumPy)
 endfunction()
 
 ##Check for GDAL installation
diff --git a/CMakeLists.txt b/CMakeLists.txt
index abf9871c..e252deaa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 
 # CMP0074 (CMake 3.12+)
 # find_package() uses <PackageName>_ROOT variables.
diff --git a/python/extensions/cython/isce/CMakeLists.txt b/python/extensions/cython/isce/CMakeLists.txt
index ace02ff6..3176a859 100644
--- a/python/extensions/cython/isce/CMakeLists.txt
+++ b/python/extensions/cython/isce/CMakeLists.txt
@@ -1,5 +1,3 @@
-find_package(Numpy REQUIRED)
-
 set_source_files_properties(
     isceextension.pyx
     PROPERTIES CYTHON_IS_CXX TRUE
@@ -34,12 +32,7 @@ set_target_properties(
     LIBRARY_OUTPUT_DIRECTORY ${ISCE_BUILDPACKAGESDIR}/isce3/extensions
 )
 
-target_include_directories(
-        isceextension PUBLIC
-        ${PYTHON_NUMPY_INCLUDE_DIR}
-)
-
-target_link_libraries(isceextension PRIVATE ${LISCE})
+target_link_libraries(isceextension PUBLIC ${LISCE} Python::NumPy)
 
 install(TARGETS isceextension
         DESTINATION ${ISCE_PACKAGESDIR}/isce3/extensions
diff --git a/python/extensions/cython/isce/cuda/CMakeLists.txt b/python/extensions/cython/isce/cuda/CMakeLists.txt
index 8416a466..f3642fe2 100644
--- a/python/extensions/cython/isce/cuda/CMakeLists.txt
+++ b/python/extensions/cython/isce/cuda/CMakeLists.txt
@@ -13,12 +13,11 @@ set_target_properties(iscecudaextension PROPERTIES
 
 target_include_directories(
         iscecudaextension PUBLIC
-        ${PYTHON_NUMPY_INCLUDE_DIR}
         ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
 )
 
 include(IsceCuda)
-target_link_libraries(iscecudaextension PUBLIC ${LISCECUDA} ${LISCE} m)
+target_link_libraries(iscecudaextension PUBLIC ${LISCECUDA} ${LISCE} m Python::NumPy)
 
 install(
     TARGETS iscecudaextension
