From 1029b950a24ad7ed993b998abaa49323fc6da3b0 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Fri, 13 Feb 2026 01:10:31 -0700
Subject: [PATCH] 1.4.0-cmake-config

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d39d85a..5b2e249 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -231,6 +231,6 @@ endif()
 
 # Install the CMake description of the project.
 install( EXPORT KLFitterTargets
-   FILE KLFitterConfig-targets.cmake
-   DESTINATION cmake
+   FILE KLFitterConfig.cmake
+   DESTINATION lib/cmake/KLFitter
    NAMESPACE "KLFitter::" )
-- 
2.52.0

