--- setup.py	2021-02-28 14:40:26.926057592 -0500
+++ setup.py.new	2021-02-28 14:40:59.966182924 -0500
@@ -19,7 +19,7 @@
     maintainer_email = "walker.ab@gmail.com",
     url          = "http://github.com/AndrewWalker/cymbal",
     license      = "MIT",
-    packages     = find_packages(), 
+    packages     = find_packages(exclude=["tests*"]), 
     classifiers  = [
         'Development Status :: 2 - Pre-Alpha',
         'Intended Audience :: Developers',
