--- configure.orig	2019-11-13 16:27:47.959300832 -0500
+++ configure	2019-11-13 16:27:16.051504825 -0500
@@ -234,7 +234,7 @@
 : ${CC:=gcc}
 
 # Test for a supported C++ compiler.
-if $($CC --version | grep -is -e gcc > /dev/null)
+if $($CC --version | grep -is -e gnu-cc > /dev/null)
 then
 	echo "Found gcc version $($CC -dumpversion)"
 	if $CC -dumpversion | awk '{split($0, a, "."); if ((lshift(a[1],16) + lshift(a[2],8) + a[3]) < lshift(4,16) + lshift(8,8) + 1) exit 0; else exit 1;}'
