diff --git CMakeLists.txt CMakeLists.txt
index c7791812..767d4199 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -29,7 +29,11 @@ set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
 
 if (UNIX)
 	set(CMAKE_LIBRARY_PATH ${CMAKE_LIBRARY_PATH} /usr/local/lib)
-	link_directories(${CMAKE_LIBRARY_PATH})
+	if(DEFINED ENV{CONDA_PREFIX})
+		link_directories(${CMAKE_LIBRARY_PATH} $ENV{CONDA_PREFIX}/lib )
+	else (DEFINED ENV{CONDA_PREFIX})
+		link_directories(${CMAKE_LIBRARY_PATH} )
+	endif (DEFINED ENV{CONDA_PREFIX})
 endif (UNIX)
 
 # create variables with various bundles of source files
