--- a/src/cherab/imas/datasets/_fetchers.py
+++ b/src/cherab/imas/datasets/_fetchers.py
@@ -31,7 +31,7 @@ def fetch_data(dataset_name: str, data_fetcher=data_fetcher, show_progress=True)
         progress = None

     downloader = pooch.DOIDownloader(
-        headers={"User-Agent": f"CHERAB-IMAS {__version__}"}, progressbar=progress
+        headers={"User-Agent": f"CHERAB-IMAS {__version__}"}, progressbar=progress, timeout=60
     )

     #  The "fetch" method returns the full path to the downloaded data file.
