diff --git a/src/napari_indices/_widget.py b/src/napari_indices/_widget.py
index 3ef9700..2e9f40a 100644
--- a/src/napari_indices/_widget.py
+++ b/src/napari_indices/_widget.py
@@ -21,7 +21,7 @@ from napari.utils.notifications import show_info
 #from napari_plugin_engine import napari_hook_implementation
 from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
 
-from PyQt5 import QtCore, QtWidgets
+from qtpy import QtCore, QtWidgets
 from matplotlib.figure import Figure
 from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg, NavigationToolbar2QT as NavigationToolbar
 from matplotlib.figure import Figure
