From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Mon, 11 Sep 2023 14:41:00 -0400
Subject: [PATCH] miri_fm: Don't define round() for MSVC, use version provided
 with C11

---
 src/miri_fm.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/miri_fm.c b/src/miri_fm.c
index 51465df..7bd5e31 100644
--- a/src/miri_fm.c
+++ b/src/miri_fm.c
@@ -61,9 +61,6 @@
 #include <io.h>
 #include "getopt/getopt.h"
 #define usleep(x) Sleep(x/1000)
-#ifdef _MSC_VER
-#define round(x) (x > 0.0 ? floor(x + 0.5): ceil(x - 0.5))
-#endif
 #define _USE_MATH_DEFINES
 #endif
 
-- 
2.40.0

