From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Mon, 11 Sep 2023 15:14:16 -0400
Subject: [PATCH] Rename mirisdr.dll to mirisdr-4.dll on Windows

---
 src/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ece0157..13e60b7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -33,6 +33,10 @@ set_target_properties(mirisdr_shared PROPERTIES OUTPUT_NAME mirisdr)
 set_target_properties(mirisdr_shared PROPERTIES SOVERSION ${MAJOR_VERSION})
 set_target_properties(mirisdr_shared PROPERTIES VERSION ${LIBVER})
 
+if(MSVC)
+    set_target_properties(mirisdr_shared PROPERTIES RUNTIME_OUTPUT_NAME "mirisdr-${MAJOR_VERSION}")
+endif()
+
 add_library(mirisdr_static STATIC
     libmirisdr.c
 )
-- 
2.40.0

