diff --git a/sources/pyside2/cmake/Macros/PySideModules.cmake b/sources/pyside2/cmake/Macros/PySideModules.cmake
index 14707f9..bfac636 100644
--- a/sources/pyside2/cmake/Macros/PySideModules.cmake
+++ b/sources/pyside2/cmake/Macros/PySideModules.cmake
@@ -167,7 +167,7 @@ macro(create_pyside_module)
     if (WIN32)
         set(ld_prefix_var_name "PATH")
     elseif(APPLE)
-        set(ld_prefix_var_name "DYLD_LIBRARY_PATH")
+        set(ld_prefix_var_name "DYLD_FALLBACK_LIBRARY_PATH")
     else()
         set(ld_prefix_var_name "LD_LIBRARY_PATH")
     endif()
