From 1fef984fed04491b03a784d040fd67d8393cfbaf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Patrick=20Sodr=C3=A9?= <psodre@gmail.com>
Date: Mon, 18 May 2020 18:35:26 -0400
Subject: [PATCH] Work-a-round the idea of parsing requirements.txt  in
 setup.py

---
 setup.py | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/setup.py b/setup.py
index bf87053..14bfe66 100644
--- a/setup.py
+++ b/setup.py
@@ -8,13 +8,6 @@ try: # for pip >= 10
 except ImportError: # for pip <= 9.0.3
     from pip.req import parse_requirements
 
-# noinspection PyProtectedMember
-try:
-    # noinspection PyProtectedMember,PyProtectedMember
-    from pip._internal.download import PipSession
-except:
-    from pip.download import PipSession
-
 VERSION = open(os.path.join('angreal', 'VERSION')).read().strip()
 py_version_tag = '-%s.%s'.format(sys.version_info[:2])
 
@@ -30,8 +23,8 @@ def read_requirements(f):
     :return:
     """
 
-    install_reqs = parse_requirements(f, session=PipSession())
-    reqs = [str(ir.req) for ir in install_reqs]
+    install_reqs = parse_requirements(f, session=None)
+    reqs = [str(ir.requirement) for ir in install_reqs]
     return reqs
 
 
@@ -44,7 +37,7 @@ setuptools.setup(
     author_email='dylan.storey@gmail.com',
     license='GPLv3',
     packages=setuptools.find_packages(exclude=['tests*']),
-    install_requires=read_requirements('requirements/requirements.txt'),
+    install_requires=read_requirements('requirements.txt'),
     zip_safe=False,
     version=VERSION,
     entry_points={
@@ -57,6 +50,6 @@ setuptools.setup(
     tests_require=['nose'],
     test_suite='nose.collector',
     extras_requires={
-        'dev': read_requirements('requirements/dev.txt')
+        'dev': read_requirements('dev.txt')
     }
 )
-- 
2.26.2

