--- a/build.zig	2026-09-11 04:17:39.000000000 +0200
+++ b/build.zig	2026-09-15 06:38:33.519703192 +0200
@@ -66,9 +66,9 @@
             .linux => {
                 if (arch == .aarch64) {
                     try args.appendSlice(gpa, "clang_base_path=\"/usr/lib/llvm-23\"\n");
-                    try args.appendSlice(gpa, "clang_use_chrome_plugins=false\n");
-                    try args.appendSlice(gpa, "treat_warnings_as_errors=false\n");
                 }
+                try args.appendSlice(gpa, "clang_use_chrome_plugins=false\n");
+                try args.appendSlice(gpa, "treat_warnings_as_errors=false\n");
             },
             else => {},
         }
