diff --git a/setup.py b/setup.py
index abfaa3f..e909c20 100755
--- a/setup.py
+++ b/setup.py
@@ -66,5 +66,5 @@ setup(
     packages=find_packages(exclude=EXCLUDE_FROM_PACKAGES),
     install_requires=install_requires,
     classifiers=classifiers,
-    entry_points={'console_scripts': ['slugify=slugify.slugify:main']},
+    entry_points={'console_scripts': ['slugify=slugify.cli:main']},
 )
diff --git a/slugify/cli.py b/slugify/cli.py
new file mode 100644
index 0000000..0fee3de
--- /dev/null
+++ b/slugify/cli.py
@@ -0,0 +1,9 @@
+import sys
+from slugify import slugify
+
+def main(): # pragma: no cover
+    if len(sys.argv) < 2:
+        print("Usage %s TEXT TO SLUGIFY" % sys.argv[0])
+    else:
+        text = ' '.join(sys.argv[1:])
+        print(slugify(text))
diff --git a/slugify/slugify.py b/slugify/slugify.py
index af0c609..6ad766e 100644
--- a/slugify/slugify.py
+++ b/slugify/slugify.py
@@ -155,11 +155,3 @@ def slugify(text, entities=True, decimal=True, hexadecimal=True, max_length=0, w
         text = text.replace(DEFAULT_SEPARATOR, separator)
 
     return text
-
-
-def main(): # pragma: no cover
-    if len(sys.argv) < 2:
-        print("Usage %s TEXT TO SLUGIFY" % sys.argv[0])
-    else:
-        text = ' '.join(sys.argv[1:])
-        print(slugify(text))
