diff --git a/externals/pacparser/src/makeHook.sh b/externals/pacparser/src/makeHook.sh
index 55b1a95..5699448 100755
--- a/externals/pacparser/src/makeHook.sh
+++ b/externals/pacparser/src/makeHook.sh
@@ -4,10 +4,8 @@ set -e
 
 static_result_dir=src/static
 
+# Use the compilers provided by the environment (conda-forge toolchain)
 FIX_COMP=""
-if [ x"$(uname)" = x"Darwin" ]; then
-  FIX_COMP="CC=/usr/bin/clang CXX=/usr/bin/clang++"
-fi
 
 FIX_PYTHON="PYTHON=python3"
 if ! python3 -V >/dev/null 2>&1; then
