diff --git a/CMakeLists.txt b/CMakeLists.txt
index a35af8f..eb3e4f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -362,11 +362,17 @@ endif()
 
 ### Linking ###
 
-if(CARLSIM_TEST)
+if(CARLSIM_TEST AND (CMAKE_C_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
     target_link_libraries(carlsim
         PUBLIC
         gcov
     )
+elseif(CARLSIM_TEST AND (CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"))
+	add_compile_options(-fprofile-instr-generate -fcoverage-mapping)
+	target_link_libraries(carlsim
+		PUBLIC
+		-fprofile-instr-generate
+	)
 endif()
 
 if(NOT CARLSIM_NO_CUDA)
diff --git a/carlsim/test/CMakeLists.txt b/carlsim/test/CMakeLists.txt
index 55da81f..dbc581f 100644
--- a/carlsim/test/CMakeLists.txt
+++ b/carlsim/test/CMakeLists.txt
@@ -57,9 +57,21 @@ endif()
             ${GTEST_LIBRARIES}
             #if(CARLSIM_GH_ACTIONS)
             GTest::gtest
-            gcov
             #endif()
     )
+    if(CMAKE_C_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        target_link_libraries(carlsim-tests
+            PRIVATE
+            gcov
+        )
+    elseif(CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        add_compile_options(-fprofile-instr-generate -fcoverage-mapping)
+        target_link_libraries(carlsim-tests
+            PRIVATE
+            -fprofile-instr-generate
+        )
+    endif()
+
 
     #if(CARLSIM_GH_ACTIONS)
     #target_link_libraries(carlsim-tests
diff --git a/carlsim/test6/CMakeLists.txt b/carlsim/test6/CMakeLists.txt
index 5042e33..c0c1262 100644
--- a/carlsim/test6/CMakeLists.txt
+++ b/carlsim/test6/CMakeLists.txt
@@ -54,9 +54,20 @@ PRIVATE
     ${GTEST_LIBRARIES}
     #if(CARLSIM_GH_ACTIONS)
     GTest::gtest
-    gcov
     #endif()
 )
+if(CMAKE_C_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    target_link_libraries(carlsim-tests
+        PRIVATE
+        gcov
+    )
+elseif(CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+	add_compile_options(-fprofile-instr-generate -fcoverage-mapping)
+	target_link_libraries(carlsim-tests
+        PRIVATE
+		-fprofile-instr-generate
+	)
+endif()
 
 #if(CARLSIM_GH_ACTIONS)
 #    target_link_libraries(carlsim-tests6
