From 0f0f42fa908ac3812cee91645102c440f8f542c7 Mon Sep 17 00:00:00 2001
From: Unal Ege Gaznepoglu <uenal.ege.gaznepoglu@iis.fraunhofer.de>
Date: Tue, 9 Jan 2024 10:11:36 +0100
Subject: [PATCH] fix compiler for rfilter1

---
 src/rfilter1/makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rfilter1/makefile.in b/src/rfilter1/makefile.in
index 5fac0f3..e85db92 100644
--- a/src/rfilter1/makefile.in
+++ b/src/rfilter1/makefile.in
@@ -18,7 +18,7 @@ bindir = $(PREFIX)/bin
 
 rfilter1: rfilter1.c
 	@echo "On cygwin, it is possible this compilation will fail, please see/adapt the rfilter1/makefile OPTIONS entry"
-	gcc $(OPTIONS) -o rfilter1 rfilter1.c
+	$(CC) $(OPTIONS) -o rfilter1 rfilter1.c
 
 setTests:
 	./rfilter1 ../test_suite/test.arb2004.glm < ../test_suite/test.arb2004.txt > ../test_suite/test.arb2004.txt.rfilter1
-- 
2.34.1

