diff --git a/polyglot/detect/base.py b/polyglot/detect/base.py
index 0f9422d..a63b510 100644
--- a/polyglot/detect/base.py
+++ b/polyglot/detect/base.py
@@ -9,7 +9,7 @@ import logging
 
 
 from icu import Locale
-import pycld2 as cld2
+import cld2
 
 logger = logging.getLogger(__name__)
 
diff --git a/requirements.txt b/requirements.txt
index 996fc36..7b68b1e 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,7 @@
+icu
 wheel>=0.23.0
 PyICU>=1.8
-pycld2>=0.3
+cld2-cffi>=0.1.4
 six>=1.7.3
 futures>=2.1.6
 morfessor>=2.0.2a1
diff --git a/rtd_requirements.txt b/rtd_requirements.txt
index 2dc1ddd..972d0d7 100644
--- a/rtd_requirements.txt
+++ b/rtd_requirements.txt
@@ -1,5 +1,5 @@
 wheel>=0.23.0
-pycld2>=0.20
+cld2-cffi>=0.1.4
 six>=1.7.3
 futures>=2.1.6
 sphinxcontrib-napoleon>=0.2.8
diff --git a/setup.py b/setup.py
index 4933225..ac8e645 100755
--- a/setup.py
+++ b/setup.py
@@ -28,7 +28,9 @@ test_requirements = [
 
 setup(
     name='polyglot',
-    version='16.07.04',
+    #version='16.07.04',
+    version_format='{tag}.dev{commitcount}+{gitsha}',
+    setup_requires=['setuptools-git-version'],
     description='Polyglot is a natural language pipeline that supports massive multilingual applications.',
     long_description=readme + '\n\n' + history,
     author='Rami Al-Rfou',
