From 41deb6530ee27d19e50e6d6698ee36e972d70f17 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Wed, 6 Oct 2021 18:10:22 +0200
Subject: [PATCH 1/2] Do not install python deps

---
 cmake/python.cmake | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/cmake/python.cmake b/cmake/python.cmake
index a19d2d2ab3..6db987915d 100644
--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -39,11 +39,7 @@ function(search_python_module MODULE_NAME)
   if(${_RESULT} STREQUAL "0")
     message(STATUS "Found python module: ${MODULE_NAME} (found version \"${MODULE_VERSION}\")")
   else()
-    message(WARNING "Can't find python module \"${MODULE_NAME}\", install it using pip...")
-    execute_process(
-      COMMAND ${Python3_EXECUTABLE} -m pip install --user ${MODULE_NAME}
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      )
+        message(FATAL_ERROR "Can't find python module \"${MODULE_NAME}\", please install it using your system package manager.")
   endif()
 endfunction()

--
2.30.1 (Apple Git-130)
