From 7bdcc0fcfaca2af660285ff90b2b157b65fa3adf Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Sun, 26 Sep 2021 22:22:39 +0200
Subject: [PATCH] Do not build deps with wrappers

---
 CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7a6be9deca..0cdd4ac2a7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,11 +151,7 @@ endif()
 
 # By default all dependencies are NOT built (i.e. BUILD_DEPS=OFF),
 # BUT if building any wrappers (Python, Java or .Net) then BUILD_DEPS=ON.
-if(BUILD_PYTHON OR BUILD_JAVA OR BUILD_DOTNET)
-  option(BUILD_DEPS "Build all dependencies" ON)
-else()
-  option(BUILD_DEPS "Build all dependencies" OFF)
-endif()
+option(BUILD_DEPS "Build all dependencies" OFF)
 message(STATUS "Build all dependencies: ${BUILD_DEPS}")
 # Install built dependencies if any,
 option(INSTALL_BUILD_DEPS "Install build all dependencies" ON)
-- 
2.33.0

