From 8e3e8021dfe79157d719418c04b5ed8120f3af18 Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Wed, 6 Oct 2021 21:09:59 +0200
Subject: [PATCH 2/2] Change wheel to sdist

---
 cmake/python.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/python.cmake b/cmake/python.cmake
index a19d2d2ab3..669212dd05 100644
--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -178,7 +178,7 @@ add_custom_target(python_package ALL
   COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:pywraprcpsp> ${PYTHON_PROJECT}/scheduling
   COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:sorted_interval_list> ${PYTHON_PROJECT}/util
   #COMMAND ${Python3_EXECUTABLE} setup.py bdist_egg bdist_wheel
-  COMMAND ${Python3_EXECUTABLE} setup.py bdist_wheel
+  COMMAND ${Python3_EXECUTABLE} setup.py sdist
   BYPRODUCTS
     python/${PYTHON_PROJECT}
     python/build
-- 
2.33.0

