From e97676e2d865d6454a088c3556d9434c98e56e5c Mon Sep 17 00:00:00 2001
From: BastianZim <10774221+BastianZim@users.noreply.github.com>
Date: Mon, 18 Oct 2021 22:00:20 +0200
Subject: [PATCH] Don't install

---
 cmake/python.cmake | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/cmake/python.cmake b/cmake/python.cmake
index a19d2d2ab3..feac574fee 100644
--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -189,13 +189,6 @@ add_custom_target(python_package ALL
   )
 add_dependencies(python_package ortools::ortools Py${PROJECT_NAME}_proto)
 
-# Install rules
-configure_file(
-  ${PROJECT_SOURCE_DIR}/cmake/python-install.cmake.in
-  ${PROJECT_BINARY_DIR}/python/python-install.cmake
-  @ONLY)
-install(SCRIPT ${PROJECT_BINARY_DIR}/python/python-install.cmake)
-
 # Test
 if(BUILD_TESTING)
   # Look for python module venv
-- 
2.33.0

