diff --git a/cmake/cpp.cmake b/cmake/cpp.cmake
index 9d4f9f26c..fb47fb1db 100644
--- a/cmake/cpp.cmake
+++ b/cmake/cpp.cmake
@@ -80,12 +80,12 @@ add_library(${PROJECT_NAME} "")
 
 if(CMAKE_VERSION VERSION_LESS "3.8.2")
   set_target_properties(${PROJECT_NAME} PROPERTIES
-    CXX_STANDARD 11
+    CXX_STANDARD 17
     CXX_STANDARD_REQUIRED ON
     CXX_EXTENSIONS OFF
   )
 else()
-  target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
+  target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_17)
 endif()
 
 if(NOT APPLE)
@@ -134,7 +134,7 @@ target_compile_definitions(${PROJECT_NAME} PUBLIC __WIN32__)
 endif()
 target_compile_definitions(${PROJECT_NAME}
   PUBLIC USE_BOP USE_GLOP USE_CBC USE_CLP)
-target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
+target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_17)
 add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
 # Generate Protobuf cpp sources
@@ -174,7 +174,7 @@ endforeach()
 #add_library(${PROJECT_NAME}_proto STATIC ${PROTO_SRCS} ${PROTO_HDRS})
 add_library(${PROJECT_NAME}_proto OBJECT ${PROTO_SRCS} ${PROTO_HDRS})
 set_target_properties(${PROJECT_NAME}_proto PROPERTIES POSITION_INDEPENDENT_CODE ON)
-set_target_properties(${PROJECT_NAME}_proto PROPERTIES CXX_STANDARD 11)
+set_target_properties(${PROJECT_NAME}_proto PROPERTIES CXX_STANDARD 17)
 set_target_properties(${PROJECT_NAME}_proto PROPERTIES CXX_STANDARD_REQUIRED ON)
 set_target_properties(${PROJECT_NAME}_proto PROPERTIES CXX_EXTENSIONS OFF)
 target_include_directories(${PROJECT_NAME}_proto PRIVATE
diff --git a/cmake/dependencies/CMakeLists.txt b/cmake/dependencies/CMakeLists.txt
index ae26db9db..78a33ff15 100644
--- a/cmake/dependencies/CMakeLists.txt
+++ b/cmake/dependencies/CMakeLists.txt
@@ -55,6 +55,11 @@ if(BUILD_absl)
       "https://github.com/abseil/abseil-cpp.git"
     TAG
       "8ba96a8"
+    CMAKE_ARGS
+      -DCMAKE_CXX_COMPILER=$ENV{CXX}
+      -DCMAKE_CXX_STANDARD=17
+      -DCMAKE_CXX_FLAGS=$ENV{CXXFLAGS}
+      -ELDFLAGS=$ENV{LDFLAGS}
  )
 endif()
 
diff --git a/cmake/sample/CMakeLists.txt b/cmake/sample/CMakeLists.txt
index 97dfe663c..a3dad9325 100644
--- a/cmake/sample/CMakeLists.txt
+++ b/cmake/sample/CMakeLists.txt
@@ -5,7 +5,7 @@ include(CTest)
 find_package(ortools CONFIG REQUIRED)
 
 add_executable(sample main.cpp)
-target_compile_features(sample PUBLIC cxx_std_11)
+target_compile_features(sample PUBLIC cxx_std_17)
 set_target_properties(sample PROPERTIES VERSION ${PROJECT_VERSION})
 target_link_libraries(sample PRIVATE ortools::ortools)
 
diff --git a/examples/cpp/CMakeLists.txt b/examples/cpp/CMakeLists.txt
index c864621d8..8d592ca94 100644
--- a/examples/cpp/CMakeLists.txt
+++ b/examples/cpp/CMakeLists.txt
@@ -49,7 +49,7 @@ foreach(EXECUTABLE
   weighted_tardiness_sat)
   add_executable(${EXECUTABLE} ${EXECUTABLE}.cc)
   target_include_directories(${EXECUTABLE} PUBLIC ${PARENT_SOURCE_DIR})
-  target_compile_features(${EXECUTABLE} PRIVATE cxx_std_11)
+  target_compile_features(${EXECUTABLE} PRIVATE cxx_std_17)
   target_link_libraries(${EXECUTABLE} PRIVATE ortools::ortools)
   install(TARGETS ${EXECUTABLE}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
diff --git a/ortools/algorithms/CMakeLists.txt b/ortools/algorithms/CMakeLists.txt
index 867668a61..b5333d754 100644
--- a/ortools/algorithms/CMakeLists.txt
+++ b/ortools/algorithms/CMakeLists.txt
@@ -10,7 +10,7 @@ set(NAME ${PROJECT_NAME}_algorithms)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/base/CMakeLists.txt b/ortools/base/CMakeLists.txt
index 3af8148ec..dea5f039a 100644
--- a/ortools/base/CMakeLists.txt
+++ b/ortools/base/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_base)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/bop/CMakeLists.txt b/ortools/bop/CMakeLists.txt
index b604d8980..77f936594 100644
--- a/ortools/bop/CMakeLists.txt
+++ b/ortools/bop/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_bop)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/constraint_solver/CMakeLists.txt b/ortools/constraint_solver/CMakeLists.txt
index 8ac574ffb..086443017 100644
--- a/ortools/constraint_solver/CMakeLists.txt
+++ b/ortools/constraint_solver/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_constraint_solver)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/data/CMakeLists.txt b/ortools/data/CMakeLists.txt
index f9f794ede..1edfbdf6c 100644
--- a/ortools/data/CMakeLists.txt
+++ b/ortools/data/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_data)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/glop/CMakeLists.txt b/ortools/glop/CMakeLists.txt
index 0e7cb813e..3e06b6c9b 100644
--- a/ortools/glop/CMakeLists.txt
+++ b/ortools/glop/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_glop)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/graph/CMakeLists.txt b/ortools/graph/CMakeLists.txt
index 5ec7072f0..9e9064afd 100644
--- a/ortools/graph/CMakeLists.txt
+++ b/ortools/graph/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_graph)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/linear_solver/CMakeLists.txt b/ortools/linear_solver/CMakeLists.txt
index 35f8239fc..94a985039 100644
--- a/ortools/linear_solver/CMakeLists.txt
+++ b/ortools/linear_solver/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_linear_solver)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/lp_data/CMakeLists.txt b/ortools/lp_data/CMakeLists.txt
index c4f930c85..5ffa9826e 100644
--- a/ortools/lp_data/CMakeLists.txt
+++ b/ortools/lp_data/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_lp_data)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/port/CMakeLists.txt b/ortools/port/CMakeLists.txt
index 87407ef4f..885c97c4a 100644
--- a/ortools/port/CMakeLists.txt
+++ b/ortools/port/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_port)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/sat/CMakeLists.txt b/ortools/sat/CMakeLists.txt
index 7ddcca8b3..1887620b8 100644
--- a/ortools/sat/CMakeLists.txt
+++ b/ortools/sat/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_sat)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON
diff --git a/ortools/util/CMakeLists.txt b/ortools/util/CMakeLists.txt
index 1d155aaa7..666e7b0c4 100644
--- a/ortools/util/CMakeLists.txt
+++ b/ortools/util/CMakeLists.txt
@@ -5,7 +5,7 @@ set(NAME ${PROJECT_NAME}_util)
 #add_library(${NAME} STATIC ${_SRCS})
 add_library(${NAME} OBJECT ${_SRCS})
 set_target_properties(${NAME} PROPERTIES
-  CXX_STANDARD 11
+  CXX_STANDARD 17
   CXX_STANDARD_REQUIRED ON
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON