diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a34d0e9..6494ecc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -21,6 +21,21 @@ message(STATUS "ROOT libraries: ${ROOT_LIBRARIES}")
 list(APPEND CMAKE_PREFIX_PATH $ENV{ROOTSYS})
 include(${ROOT_USE_FILE})

+# When cross-compiling without an emulator, force ROOT to
+# use `rootcling` as found on the PATH, not as related to the
+# ROOT libraries.
+#
+# This presumes that the relevant environment has an appropriate
+# version of rootcling installed for the _build_ platform as
+# separately to that of the host platform.
+if (CMAKE_CROSSCOMPILING AND NOT CMAKE_CROSSCOMPILING_EMULATOR)
+  set_target_properties(
+    ROOT::rootcling
+    PROPERTIES
+    IMPORTED_LOCATION_NOCONFIG "${ROOT_rootcling_CMD}"
+  )
+endif()
+
 # -- C libraries ------------

 add_subdirectory(Time)
