diff --git tests/test__utils.py tests/test__utils.py
index 89925ef..140e4db 100644
--- tests/test__utils.py
+++ tests/test__utils.py
@@ -318,7 +318,7 @@ def test__get_iterations(expected, iterations):
     "expected, a",
     [
         (numpy.bool8, False),
-        (numpy.int64, 2),
+        (numpy.int_, 2),
         (numpy.float64, 3.1),
         (numpy.complex128, 1 + 2j),
         (numpy.int16, numpy.int16(6)),
