--- cmake/Dependencies.common.cmake     2024-09-17 18:40:11.362178814 -0500
+++ cmake/Dependencies.common.cmake     2024-09-17 21:45:55.315367330 -0500
@@ -292,10 +292,7 @@
     )
     FetchContent_Populate(nvimgcodec_headers)
     set(nvimgcodec_SEARCH_PATH "${nvimgcodec_headers_SOURCE_DIR}/${CUDA_VERSION_MAJOR}/include")
-    find_path(nvimgcodec_INCLUDE_DIR nvimgcodec.h PATHS "${nvimgcodec_SEARCH_PATH}")
-    if (${nvimgcodec_INCLUDE_DIR} STREQUAL "nvimgcodec_INCLUDE_DIR-NOTFOUND")
-      message(FATAL_ERROR "nvimgcodec not found in ${nvimgcodec_SEARCH_PATH} - something went wrong with the download")
-    endif()
+    set(nvimgcodec_INCLUDE_DIR "${nvimgcodec_SEARCH_PATH}")
     message(STATUS "Using nvimgcodec_INCLUDE_DIR=${nvimgcodec_INCLUDE_DIR}")
     include_directories(SYSTEM ${nvimgcodec_INCLUDE_DIR})
 

