--- cmake/Dependencies.cmake.bak        2024-08-26 16:51:04.317397950 -0500
+++ cmake/Dependencies.cmake    2024-08-26 16:51:58.412583315 -0500
@@ -16,8 +16,7 @@
 # CUDA Toolkit libraries (including NVJPEG)
 ##################################################################
 
-CUDA_find_library(CUDART_LIB cudart_static)
-list(APPEND DALI_EXCLUDES libcudart_static.a)
+CUDA_find_library(CUDART_LIB cudart)
 
 # For NVJPEG
 if (BUILD_NVJPEG)
@@ -31,7 +30,6 @@
   # load using dlopen or link statically here
   if (NOT WITH_DYNAMIC_NVJPEG)
     list(APPEND DALI_LIBS ${NVJPEG_LIBRARY})
-    list(APPEND DALI_EXCLUDES libnvjpeg_static.a)
   endif (NOT WITH_DYNAMIC_NVJPEG)
 
   add_definitions(-DDALI_USE_NVJPEG)
@@ -46,34 +44,29 @@
 endif()
 
 if (BUILD_NVJPEG2K)
-  CUDA_find_library(NVJPEG2K_LIBRARY nvjpeg2k_static)
+  CUDA_find_library(NVJPEG2K_LIBRARY nvjpeg2)
   if (${NVJPEG2K_LIBRARY} STREQUAL "NVJPEG2K_LIBRARY-NOTFOUND")
     message(WARNING "nvJPEG2k not found - disabled")
     set(BUILD_NVJPEG2K OFF CACHE BOOL INTERNAL)
     set(BUILD_NVJPEG2K OFF)
   else()
     list(APPEND DALI_LIBS ${NVJPEG2K_LIBRARY})
-    list(APPEND DALI_EXCLUDES libnvjpeg2k_static.a)
   endif()
 endif ()
 
 # NVIDIA NPP library
 if (NOT WITH_DYNAMIC_NPP)
-  CUDA_find_library(CUDA_nppicc_LIBRARY nppicc_static)
-  CUDA_find_library(CUDA_nppig_LIBRARY nppig_static)
-  CUDA_find_library(CUDA_nppc_LIBRARY nppc_static)
+  CUDA_find_library(CUDA_nppicc_LIBRARY nppicc)
+  CUDA_find_library(CUDA_nppig_LIBRARY nppig)
+  CUDA_find_library(CUDA_nppc_LIBRARY nppc)
   list(APPEND DALI_LIBS ${CUDA_nppicc_LIBRARY})
-  list(APPEND DALI_EXCLUDES libnppicc_static.a)
   list(APPEND DALI_LIBS ${CUDA_nppig_LIBRARY})
-  list(APPEND DALI_EXCLUDES libnppig_static.a)
   list(APPEND DALI_LIBS ${CUDA_nppc_LIBRARY})
-  list(APPEND DALI_EXCLUDES libnppc_static.a)
 endif ()
 
 # cuFFT library
 if (NOT WITH_DYNAMIC_CUFFT)
-  CUDA_find_library(CUDA_cufft_LIBRARY cufft_static)
-  list(APPEND DALI_EXCLUDES libcufft_static.a)
+  CUDA_find_library(CUDA_cufft_LIBRARY cufft)
 endif ()
 
 # CULIBOS needed when using static CUDA libs

