diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c3bb5f6..a27c3b7d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,7 +151,7 @@ if (BUILD_DALI_NODEPS)
   set(BUILD_OPENCV OFF)
   set(BUILD_PROTOBUF OFF)
 else()
-  set(BUILD_OPENCV ON)
+  set(BUILD_OPENCV OFF)
   set(BUILD_PROTOBUF ON)
 endif()
 
diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 01f3d354..17e8c6ce 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -16,8 +16,7 @@
 # CUDA Toolkit libraries (including NVJPEG)
 ##################################################################
 
-CUDA_find_library(CUDART_LIB cudart_static)
-list(APPEND DALI_EXCLUDES libcudart_static.a)
+CUDA_find_library(CUDART_LIB cudart)
 
 # For NVJPEG
 if (BUILD_NVJPEG)
@@ -31,7 +30,6 @@ if (BUILD_NVJPEG)
   # load using dlopen or link statically here
   if (NOT WITH_DYNAMIC_NVJPEG)
     list(APPEND DALI_LIBS ${NVJPEG_LIBRARY})
-    list(APPEND DALI_EXCLUDES libnvjpeg_static.a)
   endif (NOT WITH_DYNAMIC_NVJPEG)
 
   add_definitions(-DDALI_USE_NVJPEG)
@@ -46,34 +44,29 @@ if (BUILD_NVJPEG)
 endif()
 
 if (BUILD_NVJPEG2K)
-  CUDA_find_library(NVJPEG2K_LIBRARY nvjpeg2k_static)
+  CUDA_find_library(NVJPEG2K_LIBRARY nvjpeg2)
   if (${NVJPEG2K_LIBRARY} STREQUAL "NVJPEG2K_LIBRARY-NOTFOUND")
     message(WARNING "nvJPEG2k not found - disabled")
     set(BUILD_NVJPEG2K OFF CACHE BOOL INTERNAL)
     set(BUILD_NVJPEG2K OFF)
   else()
     list(APPEND DALI_LIBS ${NVJPEG2K_LIBRARY})
-    list(APPEND DALI_EXCLUDES libnvjpeg2k_static.a)
   endif()
 endif ()
 
 # NVIDIA NPP library
 if (NOT WITH_DYNAMIC_NPP)
-  CUDA_find_library(CUDA_nppicc_LIBRARY nppicc_static)
-  CUDA_find_library(CUDA_nppig_LIBRARY nppig_static)
-  CUDA_find_library(CUDA_nppc_LIBRARY nppc_static)
+  CUDA_find_library(CUDA_nppicc_LIBRARY nppicc)
+  CUDA_find_library(CUDA_nppig_LIBRARY nppig)
+  CUDA_find_library(CUDA_nppc_LIBRARY nppc)
   list(APPEND DALI_LIBS ${CUDA_nppicc_LIBRARY})
-  list(APPEND DALI_EXCLUDES libnppicc_static.a)
   list(APPEND DALI_LIBS ${CUDA_nppig_LIBRARY})
-  list(APPEND DALI_EXCLUDES libnppig_static.a)
   list(APPEND DALI_LIBS ${CUDA_nppc_LIBRARY})
-  list(APPEND DALI_EXCLUDES libnppc_static.a)
 endif ()
 
 # cuFFT library
 if (NOT WITH_DYNAMIC_CUFFT)
-  CUDA_find_library(CUDA_cufft_LIBRARY cufft_static)
-  list(APPEND DALI_EXCLUDES libcufft_static.a)
+  CUDA_find_library(CUDA_cufft_LIBRARY cufft)
 endif ()
 
 # CULIBOS needed when using static CUDA libs
diff --git a/cmake/Dependencies.common.cmake b/cmake/Dependencies.common.cmake
index 8dd91fd5..453cf4b8 100644
--- a/cmake/Dependencies.common.cmake
+++ b/cmake/Dependencies.common.cmake
@@ -83,7 +83,7 @@ endif()
 ##################################################################
 if (BUILD_PYTHON)
   set(PYBIND11_CPP_STANDARD -std=c++14)
-  check_and_add_cmake_submodule(${PROJECT_SOURCE_DIR}/third_party/pybind11)
+  #check_and_add_cmake_submodule(${PROJECT_SOURCE_DIR}/third_party/pybind11)
 endif()
 
 ##################################################################
