diff --git a/test/phonopy/harmonic/test_dynmat_to_fc.py b/test/phonopy/harmonic/test_dynmat_to_fc.py
index 77deb91..e3ae05d 100644
--- a/test/phonopy/harmonic/test_dynmat_to_fc.py
+++ b/test/phonopy/harmonic/test_dynmat_to_fc.py
@@ -31,9 +31,8 @@ class TestDynmatToFc(unittest.TestCase):
     def _compare(self, comm_points, filename="comm_points.dat"):
         with open(os.path.join(data_dir,filename)) as f:
             comm_points_in_file = np.loadtxt(f)
-            #self.assertTrue(
-            #    (np.abs(comm_points_in_file[:,1:] - comm_points) < 1e-3).all())
-            np.testing.assert_allclose(comm_points_in_file[:,1:],comm_points,atol=1e-3)
+            self.assertTrue(
+                (np.abs(comm_points_in_file[:,1:] - comm_points) < 1e-3).all())
 
     def _write(self, comm_points, filename="comm_points.dat"):
         with open(os.path.join(data_dir,filename), 'w') as w:
