diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8631a5d..28ff86e 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.25)
-if(APPLE)
+if(APPLE AND NOT DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
     set(CMAKE_OSX_DEPLOYMENT_TARGET "11.0" CACHE STRING
-        "Minimum macOS version" FORCE)
+        "Minimum macOS version")
 endif()
 file(READ "${CMAKE_CURRENT_SOURCE_DIR}/VERSION" GEL_VERSION)
 string(STRIP "${GEL_VERSION}" GEL_VERSION)
@@ -26,8 +26,8 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set(CMAKE_BUILD_TYPE Release)
 endif()
 
-if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "Build architectures for MacOS" FORCE)
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT DEFINED CMAKE_OSX_ARCHITECTURES)
+    set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "Build architectures for MacOS")
 endif ()
 
 option(Use_GLGraphics "Compile the OpenGL Viewer" ON)
