diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7577f8f..06727ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,9 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
     include_directories(./src/GEL)
     add_definitions(-DGLEW_STATIC)
 endif ()
-if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND
+    NOT DEFINED CMAKE_OSX_ARCHITECTURES AND
+    "$ENV{CMAKE_OSX_ARCHITECTURES}" STREQUAL "")
     set(CMAKE_OSX_ARCHITECTURES "x86_64;arm64" CACHE STRING "Build architectures for MacOS" FORCE)
 endif ()
 
