diff --git a/src/GEL/HMesh/face_loop.cpp b/src/GEL/HMesh/face_loop.cpp
index b228c7e..5b8a1ed 100644
--- a/src/GEL/HMesh/face_loop.cpp
+++ b/src/GEL/HMesh/face_loop.cpp
@@ -13,7 +13,6 @@
 
 //#include "LogMap.h"
 #include <GEL/CGLA/CGLA.h>
-#include <GEL/GLGraphics/ManifoldRenderer.h>
 #include <GEL/Geometry/RGrid.h>
 #include <GEL/Geometry/GridAlgorithm.h>
 #include <GEL/Geometry/KDTree.h>
@@ -25,7 +24,6 @@ using namespace Util;
 using namespace std;
 using namespace CGLA;
 using namespace Geometry;
-using namespace GLGraphics;
 using namespace HMesh;
 
 void create_hex(Manifold& m, vector<Vec3d>& pts, double scale) {
@@ -787,7 +785,6 @@ Geometry::AMGraph3D face_loop_skeleton(HMesh::Manifold& m) {
                     Walker w = m.walker(h);
                     FaceID f = w.face();
                     touched[f] = 1;
-                    DebugRenderer::face_colors[f] = Vec3f(1.0-l.cylindricity,0.3,l.cylindricity);
                     VertexID v0 = w.vertex();
                     VertexID v1 = w.opp().vertex();
                     graph.pos[n0] += m.pos(v0);
