diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7577f8f..efb7fa3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,7 +92,7 @@ if (Use_GLGraphics)
     endif ()
     target_link_libraries(PyGEL PRIVATE GEL glfw ${GLFW_LIBRARIES})
 else ()
-    list(REMOVE_ITEM PYG_SRC_LIST "./src/PyGEL/Viewer.cpp")
+    list(FILTER PYG_SRC_LIST EXCLUDE REGEX ".*/src/PyGEL/Viewer\\.cpp$")
     add_library(PyGEL SHARED ${PYG_SRC_LIST})
     target_include_directories(PyGEL PRIVATE ./src)
     target_link_libraries(PyGEL PRIVATE GEL)
