diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -16,5 +16,10 @@
 libs_data = []
-libs = glob('build/*.dylib')+glob('build/*.so*')+glob('build/*.dll',recursive=True)
+libs = sorted(set(
+    glob('build/*.dylib')
+    + glob('build/*.so*')
+    + glob('build/*.dll')
+    + glob('build/**/*.dll', recursive=True)
+))
 for lib_file in libs:
     _,fn = path.split(lib_file)
     dst = "build/pygel3d/"+fn
