--- uwsgiconfig.py
+++ uwsgiconfig.py
@@ -875,8 +875,7 @@ class uConf(object):
                 self.cflags.append('-DNO_EXECINFO')
                 self.cflags.append('-DOLD_REALPATH')
             self.cflags.append('-mmacosx-version-min=10.5')
-            if GCC in ('clang',):
-                self.libs.remove('-rdynamic')
+            self.libs.remove('-rdynamic')

         if uwsgi_os.startswith('CYGWIN'):
             self.libs.remove('-rdynamic')
