diff --git a/plugins/php/uwsgiplugin.py b/plugins/php/uwsgiplugin.py
index 963e11a..8657eb7 100644
--- plugins/php/uwsgiplugin.py
+++ plugins/php/uwsgiplugin.py
@@ -25,6 +25,6 @@ LIBS = [os.popen(PHPPATH + ' --libs').read().rstrip(), '-lphp' + php_version]

 phplibdir = os.environ.get('UWSGICONFIG_PHPLIBDIR')
 if phplibdir:
-    LIBS.append('-Wl,-rpath=%s' % phplibdir)
+    LIBS.append('-Wl,-rpath,%s' % phplibdir)

 GCC_LIST = ['php_plugin', 'session']


diff --git a/plugins/python/uwsgiplugin.py b/plugins/python/uwsgiplugin.py
index 2a2f18c..1b33972 100644
--- plugins/python/uwsgiplugin.py
+++ plugins/python/uwsgiplugin.py
@@ -76,7 +76,7 @@ if 'UWSGI_PYTHON_NOLIB' not in os.environ:
             libdir = "%s/lib" % sysconfig.PREFIX

         LDFLAGS.append("-L%s" % libdir)
-        LDFLAGS.append("-Wl,-rpath=%s" % libdir)
+        LDFLAGS.append("-Wl,-rpath,%s" % libdir)

         os.environ['LD_RUN_PATH'] = "%s" % libdir
