diff --git a/setup.py b/setup.py
index b4802ef..a12af1f 100644
--- a/setup.py
+++ b/setup.py
@@ -32,8 +32,8 @@ here = os.path.abspath(os.path.dirname(__file__))
 
 with io.open(os.path.join(here, "README.md"), encoding="utf-8") as f:
     long_description = "\n" + f.read()
-with io.open(os.path.join(here, "CHANGELOG.md"), encoding="utf-8") as f:
-    long_description = long_description + "\n\n" + f.read()
+
+
 
 # Load the package's __version__.py module as a dictionary.
 about = {}
