From 88c67a15996e9b9cf1633852d9401a5636634360 Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Fri, 30 Oct 2020 18:57:52 +0100
Subject: [PATCH] add virtual desctructor on base class

---
 include/hpp/statistics/bin.hh | 1 +
 1 file changed, 1 insertion(+)

diff --git include/hpp/statistics/bin.hh include/hpp/statistics/bin.hh
index ac6421f..ebaf7d3 100644
--- include/hpp/statistics/bin.hh
+++ include/hpp/statistics/bin.hh
@@ -67,6 +67,7 @@ namespace hpp {
       protected:
         /// Constructor
         Bin () : freq_ (0) {}
+        virtual ~Bin() {}
 
       private:
         /// The number of occurence.
-- 
2.17.1

