diff --git a/planarity/cplanarity.pxd b/planarity/cplanarity.pxd
index 9d641b6..492cfb4 100644
--- a/planarity/cplanarity.pxd
+++ b/planarity/cplanarity.pxd
@@ -1,5 +1,5 @@
 """Interface for Boyer's (C) planarity algorithms."""
-cdef extern from "src/graphStructures.h":
+cdef extern from "planarity/graphStructures.h":
     ctypedef struct baseGraphStructure:
         pass
     ctypedef baseGraphStructure * graphP
@@ -18,7 +18,7 @@ cdef extern from "src/graphStructures.h":
     cdef int gp_GetNextArc(graphP theGraph, int v)
     cdef int gp_GetDirection(graphP theGraph, int v)
 
-cdef extern from "src/graph.h":
+cdef extern from "planarity/graph.h":
     cdef int OK, NOTOK, NULL 
     cdef int EMBEDFLAGS_PLANAR, NONEMBEDDABLE, EMBEDFLAGS_DRAWPLANAR
     cdef int WRITE_ADJLIST
@@ -33,12 +33,12 @@ cdef extern from "src/graph.h":
     cdef void gp_SortVertices(graphP theGraph)
 
 
-cdef extern from "src/graphDrawPlanar.h":
+cdef extern from "planarity/graphDrawPlanar.h":
     cdef char * _RenderToString(graphP theEmbedding)
     cdef int gp_AttachDrawPlanar(graphP theGraph)
 
 
-cdef extern from "src/graphDrawPlanar.private.h":
+cdef extern from "planarity/graphDrawPlanar.private.h":
     ctypedef struct DrawPlanar_VertexInfo:
        int pos
        int start
@@ -55,6 +55,6 @@ cdef extern from "src/graphDrawPlanar.private.h":
         DrawPlanar_EdgeRecP E
         DrawPlanar_VertexInfoP VI
 
-cdef extern from "src/graphExtensions.h":
+cdef extern from "planarity/graphExtensions.h":
     cdef void * gp_GetExtension(graphP theGraph, int moduleID)
     cdef int gp_FindExtension(graphP theGraph, int moduleID, void *pContext)
diff --git a/setup.py b/setup.py
index d92e42e..20d52e5 100644
--- a/setup.py
+++ b/setup.py
@@ -40,11 +40,11 @@ classifiers = [
 ext = '.pyx' if USE_CYTHON else '.c'
 
 sourcefiles = ['planarity/planarity'+ext]
-sourcefiles.extend(glob("planarity/src/*.c"))
 
 extensions = [Extension("planarity.planarity",
                         sourcefiles,
-                        include_dirs=['planarity/src/'],
+                        include_dirs=[os.getenv("CONDA_PREFIX", "/usr")+"/include"],
+                        libraries=['planarity'],
                         )]
 
 if USE_CYTHON:
