diff --git a/planarity/cplanarity.pxd b/planarity/cplanarity.pxd
index 9d641b6..492cfb4 100644
--- a/planarity/cplanarity.pxd
+++ b/planarity/cplanarity.pxd
@@ -1,5 +1,5 @@
 """Interface for Boyer's (C) planarity algorithms."""
-cdef extern from "src/graphStructures.h":
+cdef extern from "planarity/graphStructures.h":
     ctypedef struct baseGraphStructure:
         pass
     ctypedef baseGraphStructure * graphP
@@ -18,7 +18,7 @@ cdef extern from "src/graphStructures.h":
     cdef int gp_GetNextArc(graphP theGraph, int v)
     cdef int gp_GetDirection(graphP theGraph, int v)
 
-cdef extern from "src/graph.h":
+cdef extern from "planarity/graph.h":
     cdef int OK, NOTOK, NULL 
     cdef int EMBEDFLAGS_PLANAR, NONEMBEDDABLE, EMBEDFLAGS_DRAWPLANAR
     cdef int WRITE_ADJLIST
@@ -33,12 +33,12 @@ cdef extern from "src/graph.h":
     cdef void gp_SortVertices(graphP theGraph)
 
 
-cdef extern from "src/graphDrawPlanar.h":
+cdef extern from "planarity/graphDrawPlanar.h":
     cdef char * _RenderToString(graphP theEmbedding)
     cdef int gp_AttachDrawPlanar(graphP theGraph)
 
 
-cdef extern from "src/graphDrawPlanar.private.h":
+cdef extern from "planarity/graphDrawPlanar.private.h":
     ctypedef struct DrawPlanar_VertexInfo:
        int pos
        int start
@@ -55,6 +55,6 @@ cdef extern from "src/graphDrawPlanar.private.h":
         DrawPlanar_EdgeRecP E
         DrawPlanar_VertexInfoP VI
 
-cdef extern from "src/graphExtensions.h":
+cdef extern from "planarity/graphExtensions.h":
     cdef void * gp_GetExtension(graphP theGraph, int moduleID)
     cdef int gp_FindExtension(graphP theGraph, int moduleID, void *pContext)
diff --git a/setup.py b/setup.py
index 6a1f641..e7f1a43 100644
--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,7 @@ classifiers = [
         'Topic :: Scientific/Engineering :: Physics']
 
 sourcefiles = ['planarity/planarity.pyx']
-sourcefiles.extend(glob("planarity/src/*.c"))
 
 setup(
     name= 'planarity',
@@ -48,7 +47,8 @@ setup(
     cmdclass = {'build_ext': build_ext},
     ext_modules = [Extension("planarity.planarity",
                              sourcefiles,
-                             include_dirs=['planarity/src/'],
+                             include_dirs=[os.path.join(os.getenv("LIBRARY_PREFIX", os.getenv("CONDA_PREFIX", "/usr")), "include")],
+                             library_dirs=[os.path.join(os.getenv("LIBRARY_PREFIX", os.getenv("CONDA_PREFIX", "/usr")), "lib")],
+                             libraries=['planarity'],
                              )],
     version          = '0.4.1',
     url = 'https://github.com/hagberg/planarity/',
