From f1293e87a55cd759e2993a8a2a10a65e2b025450 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 9 Jan 2022 10:48:19 +1100
Subject: [PATCH] avoid wrongly falling back for source builds of
 fribidi/harfbuzz

---
 meson.build | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/meson.build b/meson.build
index 9af9d21..09a8e4a 100644
--- a/meson.build
+++ b/meson.build
@@ -42,23 +42,10 @@ if not freetype.found()
                                            'zlib=disabled', 'harfbuzz=disabled'])
 endif
 
-harfbuzz = dependency('harfbuzz', version : '>= 1.7.2',
-                      fallback : ['harfbuzz', 'libharfbuzz_dep'],
-                      default_options : ['freetype=enabled',
-                                         'glib=disabled', 'gobject=disabled',
-                                         'cairo=disabled', 'icu=disabled',
-                                         'tests=disabled'])
+harfbuzz = dependency('harfbuzz', version : '>= 1.7.2', required : true)
 
-fribidi = dependency('', required : false)
+fribidi = dependency('fribidi', required : true)
 sheenbidi = dependency('', required : false)
-if get_option('sheenbidi')
-  sheenbidi = dependency('sheenbidi',
-                         fallback : ['sheenbidi', 'sheenbidi_dep'])
-else
-  fribidi = dependency('fribidi',
-                       fallback : ['fribidi', 'libfribidi_dep'],
-                       default_options : ['docs=false', 'tests=false'])
-endif
 
 deps = [freetype, harfbuzz, fribidi, sheenbidi]
 
-- 
2.32.0.windows.2

