From b8fb3483d95d1e264d04a37a79ab2116d3195a00 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 28 May 2026 22:33:13 +1100
Subject: [PATCH] undef NDEBUG for tests

---
 tests/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/meson.build b/tests/meson.build
index 5f720a1..92a13af 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -6,6 +6,8 @@ raqm_test = executable(
     include_directories: include_directories(['../src']),
     link_with: libraqm_test,
     override_options: ['b_ndebug=false'],
+    c_args: ['-UNDEBUG'],
+    cpp_args: ['-UNDEBUG'],
     dependencies: deps,
     install: false,
 )
