--- a/bootstrap-generic
+++ b/bootstrap-generic
@@ -22 +22 @@
-    -e "s/@ASSEMBLY_VERSION@/$ASSEMBLY_VERSION/" $srcdir/configure.in.in > $srcdir/configure.in
+    -e "s/@ASSEMBLY_VERSION@/$ASSEMBLY_VERSION/" $srcdir/configure.in.in > $srcdir/configure.ac
@@ -51 +51 @@
-(grep "^AM_PROG_LIBTOOL" $srcdir/configure.in >/dev/null) && {
+(grep "^AM_PROG_LIBTOOL" $srcdir/configure.ac >/dev/null) && {
@@ -104,2 +104,2 @@
-
-if grep "^AM_PROG_LIBTOOL" configure.in >/dev/null; then
+autoupdate
+if grep "^AM_PROG_LIBTOOL" configure.ac >/dev/null; then
@@ -123 +123 @@
-if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
+if grep "^AC_CONFIG_HEADERS" configure.ac >/dev/null; then
--- a/configure.in.in
+++ b/configure.in.in
@@ -6,0 +6,1 @@
+AC_CONFIG_MACRO_DIRS([m4])
