From 15abc347783872a7d5510645e822daabaf16ed58 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 12 Aug 2021 19:39:51 +0200
Subject: [PATCH] redirect tomlplusplus-include to global library path

---
 include/pytomlpp/pytomlpp.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/pytomlpp/pytomlpp.hpp b/include/pytomlpp/pytomlpp.hpp
index 96e5b5c..40d4144 100644
--- a/include/pytomlpp/pytomlpp.hpp
+++ b/include/pytomlpp/pytomlpp.hpp
@@ -29,7 +29,7 @@
 #endif
 #include <pybind11/pybind11.h>
 #include <sstream>
-#include <tomlplusplus/include/toml++/toml.h>
+#include <toml++/toml.h>
 
 // namespace and type forward declarations
 namespace py = pybind11;
-- 
2.31.1.windows.1

