diff -ru scripts/zf_install.bak scripts/zf_install
--- scripts/zf_install.bak	2024-10-22 16:53:51.407414781 -0500
+++ scripts/zf_install	2024-10-22 16:54:11.447059514 -0500
@@ -63,11 +63,7 @@
 }
 
 lib64_path() {
-  if [ -e /lib/x86_64-linux-gnu/libc.so.6 ]; then
-      echo "$(usr_path)/lib/x86_64-linux-gnu"
-  else
-      echo "$(usr_path)/lib64"
-  fi
+  echo "$(usr_path)/lib"
 }
 
 package_name() {
