From f4f7c3ce8990676f5027b915eeddad5e8c631486 Mon Sep 17 00:00:00 2001
From: Alexey <rudy@05bit.com>
Date: Mon, 17 Mar 2014 15:30:10 +0300
Subject: [PATCH] fix: import path conflict in Python 3

---
 easywatch/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/easywatch/__init__.py b/easywatch/__init__.py
index cfe1ae3..11facf9 100644
--- a/easywatch/__init__.py
+++ b/easywatch/__init__.py
@@ -1 +1 @@
-from easywatch import watch
+from .easywatch import watch
