--- CMakeLists.txt.orig      2024-07-05 14:24:49.031916858 -0500
+++ CMakeLists.txt           2024-07-05 14:25:32.952609960 -0500
@@ -192,2 +192,25 @@
        add_library(${LIB_NAME_MAIN} SHARED ${SOURCES})
+       set_target_properties(${LIB_NAME_MAIN} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+       if (WIN32)
+         set_target_properties(${LIB_NAME_MAIN} PROPERTIES OUTPUT_NAME "${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}")
+         set(MAJOR_DLL_PATH "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}")
+         add_custom_command(
+           TARGET ${LIB_NAME_MAIN} POST_BUILD
+           COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                   "${MAJOR_DLL_PATH}.dll"
+                   "${MAJOR_DLL_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+           COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                   "${MAJOR_DLL_PATH}.dll"
+                   "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.dll"
+           COMMENT "Copying ${LIB_NAME_C}.${PROJECT_VERSION_MAJOR}.dll to versioned filenames"
+         )
+         add_custom_target(CopyVersionedCDLL ALL
+           DEPENDS ${LIB_NAME_MAIN}
+         )
+         install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.dll"
+                       "${MAJOR_DLL_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+                 DESTINATION bin
+                 COMPONENT Runtime
+         )
+       endif ()
 else ()
