--- CMakeLists.txt.orig      2024-07-05 14:24:49.031916858 -0500
+++ CMakeLists.txt           2024-07-05 14:25:32.952609960 -0500
@@ -192,2 +192,29 @@
        add_library(${LIB_NAME_MAIN} SHARED ${SOURCES})
+       set_target_properties(${LIB_NAME_MAIN} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+       if (WIN32)
+         set_target_properties(${LIB_NAME_MAIN} PROPERTIES OUTPUT_NAME "${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}")
+         set(BASE_OUTPUT_PATH "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}")
+         add_custom_command(
+           OUTPUT "${BASE_OUTPUT_PATH}.dll"
+                  "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+           COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.dll"
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+           COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.dll"
+                   "${BASE_OUTPUT_PATH}.dll"
+           DEPENDS ${LIB_NAME_MAIN}
+           COMMENT "Copying ${LIB_NAME_MAIN}.3.dll to versioned filenames"
+         )
+         add_custom_target(CopyVersionedDLL ALL
+           DEPENDS "${BASE_OUTPUT_PATH}.dll"
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.dll"
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+         )
+         install(FILES "${BASE_OUTPUT_PATH}.dll"
+                       "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+                 DESTINATION bin
+                 COMPONENT Runtime
+         )
+       endif ()
 else ()
