--- CMakeLists.txt.orig      2024-07-05 14:24:49.031916858 -0500
+++ CMakeLists.txt           2024-07-05 14:25:32.952609960 -0500
@@ -192,2 +192,28 @@
        add_library(${LIB_NAME_MAIN} SHARED ${SOURCES})
+        set_target_properties(${LIB_NAME_MAIN} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+       if(WIN32)
+         # For Windows, append version information to the library name
+         set_target_properties(${LIB_NAME_MAIN} PROPERTIES
+           VERSION ${PROJECT_VERSION}
+           SOVERSION ${PROJECT_VERSION_MAJOR}
+           OUTPUT_NAME "${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}"
+         )
+         add_custom_command(
+           OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.dll"
+           COMMAND ${CMAKE_COMMAND} -E copy
+                   "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+                   "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.dll"
+           DEPENDS ${LIB_NAME_MAIN}
+         )
+         add_custom_command(
+           OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.dll"
+           COMMAND ${CMAKE_COMMAND} -E copy
+                   "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+                   "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}.dll"
+           DEPENDS ${LIB_NAME_MAIN}
+         )
+         add_custom_target(CopyNonVersionedLibrary ALL
+           DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.dll" "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_MAIN}.${PROJECT_VERSION_MAJOR}.dll"
+         )
+endif()
 else ()
