--- cmake/target_c.cmake.orig  2024-07-05 14:57:30.395099755 -0500
+++ cmake/target_c.cmake       2024-07-05 14:58:11.635870965 -0500
@@ -13,2 +13,28 @@
        add_library(${LIB_NAME_C} SHARED ${SOURCES} ${C_SOURCES} ${C_HEADERS})
+        set_target_properties(${LIB_NAME_C} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+       if (WIN32)
+         set(BASE_OUTPUT_PATH "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME_C}")
+         add_custom_command(
+           OUTPUT "${BASE_OUTPUT_PATH}.3.dll"
+                  "${BASE_OUTPUT_PATH}.3.0.0.dll"
+           COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                   "${BASE_OUTPUT_PATH}.dll"
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+           COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                   "${BASE_OUTPUT_PATH}.dll"
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.dll"
+           DEPENDS ${LIB_NAME_C}
+           COMMENT "Copying ${LIB_NAME_C}.dll to versioned filenames"
+         )
+         add_custom_target(CopyVersionedCDLL ALL
+           DEPENDS "${BASE_OUTPUT_PATH}.dll"
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.dll"
+                   "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+         )
+         install(FILES "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.dll"
+                       "${BASE_OUTPUT_PATH}.${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.dll"
+                 DESTINATION bin
+                 COMPONENT Runtime
+         )
+       endif ()
 else ()
