diff --git a/SRC/makefile b/SRC/makefile
index 0e7b346..8c00a1c 100644
--- a/SRC/makefile
+++ b/SRC/makefile
@@ -194,9 +194,8 @@ single_double_complex: $(SCDOBJS)
 	rm -f *_lapack_single_double_complex.o
 
 single_double_complex_dcomplex: $(SCDZOBJS)
-	rm -f ../lapack95.a
-	ar cr ../lapack95.a $(SCDZOBJS)
-	ranlib ../lapack95.a
+	rm -f ../lapack95$(SHLIB_EXT)
+	$(FC) -shared -o $(LAPACK95) -Wl,-all_load $(SCDZOBJS) -Wl,-noall_load $(LIBS)
 	rm -fr ../lapack95_modules
 	mkdir ../lapack95_modules
 	'cp' *.mod ../lapack95_modules/
