diff --git a/setup.py b/setup.py
index bd8064c..4edddaa 100644
--- a/setup.py
+++ b/setup.py
@@ -19,7 +19,7 @@ import numpy as np
 
 extensions = "deodr/differentiable_renderer_cython.pyx"
 
-my_modules = cythonize(extensions, annotate=True, language="c++")
+my_modules = cythonize(extensions, annotate=True, language="c++", include_path=[np.get_include()])
 
 libname = "deodr"
 
@@ -34,7 +34,7 @@ setup(
     packages=find_packages(),
     package_data={"deodr": ["*.pyx", "*.pxd", "data/*.*", "data/**/*.*"]},
     ext_modules=my_modules,  # additional source file(s)),
-    include_dirs=[np.get_include()],
+    data_files=[('C++', ['C++/DifferentiableRenderer.h'])],
     setup_requires=["numpy", "scipy", "cython"],
     install_requires=["numpy", "scipy"],
 )
