diff --git a/pyproject.toml b/pyproject.toml
index ad67101..5f62ea4 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,8 +2,8 @@
 name = "KDEpy"
 version = "1.1.4"
 dependencies = [
-    "numpy>=1.14.2",
-    "scipy>=1.0.1",
+    "numpy>=1.14.2,<2.0",
+    "scipy>=1.0.1,<2.0",
     "matplotlib>=2.2.2",
 ]
 description = "Kernel Density Estimation in Python."
@@ -40,5 +40,5 @@ lint = [
 ]
 
 [build-system]
-requires = ["setuptools>=45", "wheel", "cython>=0.29", "numpy>=1.14.2"]
+requires = ["setuptools>=45", "wheel", "cython>=0.29,<1.0.0", "numpy>=1.14.2,<2.0"]
 build-backend = "setuptools.build_meta"
