--- a/a/cpp/bosdyn/client/service_client/service_client.h
+++ b/b/cpp/bosdyn/client/service_client/service_client.h
@@ -140,7 +140,7 @@ class ServiceClient {
         std::promise<Result<PromiseResultType>> result_promise, const RPCParameters& parameters) {
         BOSDYN_ASSERT_PRECONDITION(m_message_pump != nullptr,
                                    "Message pump cannot be null for request type %s",
-                                   Request::GetDescriptor()->full_name().c_str());
+                                   std::string(Request::descriptor()->full_name()));
         // The one_time pointer is deleted by MessagePump::Update after the callback function
         // returns
@@ -231,7 +231,7 @@ class ServiceClient {
         auto ret = one_time.get();
         m_message_pump->AddCall(std::move(one_time));
         ret->Start(std::move(requests), header, rpc_call, callback, std::move(promise),
-                   Request::descriptor()->full_name());
+                   std::string(Request::descriptor()->full_name()));
         return ret;
     }
@@ -300,7 +300,7 @@ class ServiceClient {
         auto ret = one_time.get();
         m_message_pump->AddCall(std::move(one_time));
         ret->Start(std::move(chunks), request.header(), rpc_call, callback, std::move(promise),
-                   Request::descriptor()->full_name());
+                   std::string(Request::descriptor()->full_name()));
         return ret;
     }
@@ -380,7 +380,7 @@ class ServiceClient {
         auto ret = one_time.get();
         m_message_pump->AddCall(std::move(one_time));
         ret->Start(std::move(wrapper_chunks), request.header(), rpc_call, callback,
-                   std::move(promise), Request::descriptor()->full_name());
+                   std::move(promise), std::string(Request::descriptor()->full_name()));
         return ret;
     }
@@ -657,7 +657,7 @@ class ServiceClient {
         auto ret = one_time.get();
         m_message_pump->AddCall(std::move(one_time));
         ret->Start(std::move(chunks), request.header(), rpc_call, callback, std::move(promise),
-                   Request::descriptor()->full_name());
+                   std::string(Request::descriptor()->full_name()));
         return ret;
     }
diff --git a/a/cpp/bosdyn/common/common_header_handling.cpp b/b/cpp/bosdyn/common/common_header_handling.cpp
index 95eca39..63f979b 100644
--- a/a/cpp/bosdyn/common/common_header_handling.cpp
+++ b/b/cpp/bosdyn/common/common_header_handling.cpp
@@ -37,7 +37,7 @@ void PrepareResponseHeader(const ::bosdyn::api::RequestHeader& request_header,
 void StripLargeByteFields(::google::protobuf::Message* proto_message) {
     // Operates on Request and Response messages.
-    const std::string& message_type = proto_message->GetTypeName();
+    const auto message_type = std::string(proto_message->GetTypeName());
     const auto& whitelist_map = ::bosdyn::common::strip_messages::kWhitelistedBytesFieldsMap;
     auto strip_message_func = whitelist_map.find(message_type);
     if (strip_message_func != whitelist_map.end()) {
