
--- a/cpp/bosdyn/client/service_client/service_client.h
+++ b/cpp/bosdyn/client/service_client/service_client.h
@@ -155,9 +155,11 @@

         RPCParameters parameters_to_use = CombineRPCParameters(parameters);
         SetLoggingControl(parameters_to_use.logging_control, &request);
-        one_time->context()->set_deadline(std::chrono::system_clock::now() +
-                                          CONVERT_DURATION_FOR_GRPC(parameters_to_use.timeout));

+        const std::chrono::system_clock::time_point deadline =
+            std::chrono::system_clock::now() + CONVERT_DURATION_FOR_GRPC(parameters_to_use.timeout);
+        one_time->context()->set_deadline(deadline);
+
         auto status = m_request_processor_chain.Process(one_time->context(),
                                                         request.mutable_header(), &request);
         if (!status) {
@@ -405,8 +407,9 @@
             return nullptr;
         }

-        one_time->context()->set_deadline(std::chrono::system_clock::now() +
-                                          CONVERT_DURATION_FOR_GRPC(timeout));
+        const std::chrono::system_clock::time_point deadline =
+            std::chrono::system_clock::now() + CONVERT_DURATION_FOR_GRPC(timeout);
+        one_time->context()->set_deadline(deadline);

         return one_time;
     }
@@ -457,9 +460,11 @@
         }

         RPCParameters parameters_to_use = CombineRPCParameters(parameters);
-        one_time->context()->set_deadline(std::chrono::system_clock::now() +
-                                          CONVERT_DURATION_FOR_GRPC(parameters_to_use.timeout));

+        const std::chrono::system_clock::time_point deadline =
+            std::chrono::system_clock::now() + CONVERT_DURATION_FOR_GRPC(parameters_to_use.timeout);
+        one_time->context()->set_deadline(deadline);
+
         SetLoggingControl(parameters_to_use.logging_control, &request_to_stream);
         auto processor_status = m_request_processor_chain.Process(
             one_time->context(), request_to_stream.mutable_header(), &request_to_stream);
@@ -522,9 +527,11 @@

         RPCParameters parameters_to_use = CombineRPCParameters(parameters);
         SetLoggingControl(parameters_to_use.logging_control, &request);
-        one_time->context()->set_deadline(std::chrono::system_clock::now() +
-                                          CONVERT_DURATION_FOR_GRPC(parameters_to_use.timeout));

+        const std::chrono::system_clock::time_point deadline =
+            std::chrono::system_clock::now() + CONVERT_DURATION_FOR_GRPC(parameters_to_use.timeout);
+        one_time->context()->set_deadline(deadline);
+
         auto status = m_request_processor_chain.Process(one_time->context(),
                                                         request.mutable_header(), &request);
         if (!status) {
@@ -682,8 +689,9 @@
             return nullptr;
         }

-        one_time->context()->set_deadline(std::chrono::system_clock::now() +
-                                          CONVERT_DURATION_FOR_GRPC(timeout));
+        const std::chrono::system_clock::time_point deadline =
+            std::chrono::system_clock::now() + CONVERT_DURATION_FOR_GRPC(timeout);
+        one_time->context()->set_deadline(deadline);

         return one_time;
     }
