--- a/cpp/bosdyn/client/service_client/service_client.h
+++ b/cpp/bosdyn/client/service_client/service_client.h
@@ -155,8 +155,9 @@

         RPCParameters parameters_to_use = CombineRPCParameters(parameters);
         SetLoggingControl(parameters_to_use.logging_control, &request);
-        one_time->context()->set_deadline(std::chrono::system_clock::now() +
-                                          CONVERT_DURATION_FOR_GRPC(parameters_to_use.timeout));
+        const std::chrono::system_clock::time_point deadline =
+            std::chrono::system_clock::now() + CONVERT_DURATION_FOR_GRPC(parameters_to_use.timeout);
+        one_time->context()->set_deadline(deadline);

         auto status = m_request_processor_chain.Process(one_time->context(),
                                                         request.mutable_header(), &request);
