--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -67,8 +67,8 @@
   )

   if (BUILD_SHARED_LIBS)
-    add_library(bosdyn_api SHARED $<TARGET_OBJECTS:bosdyn_api_obj>)
-    target_link_libraries(bosdyn_api INTERFACE bosdyn_api_obj)
+    add_library(bosdyn_api SHARED)
+    target_link_libraries(bosdyn_api PUBLIC bosdyn_api_obj gRPC::grpc gRPC::grpc++ ${PROTOBUF_LIBRARIES})
     install(
       TARGETS bosdyn_api
       EXPORT ${PROJECT_NAME}Targets
