--- a/cpp/examples/joint_control/joint_api_helper.cpp
+++ b/cpp/examples/joint_control/joint_api_helper.cpp
@@ -118,7 +118,7 @@
             // Want to send next command no sooner than `dt` after the previous one
             auto now = ::bosdyn::common::NowNsec();
             auto this_dt_time_d_ns = std::max(previous_dt_time_ns + dt, now);
-            auto sleep_time_ns = std::max(0L, this_dt_time_d_ns - now);
+            auto sleep_time_ns = std::max(0LL, this_dt_time_d_ns - now);

             // Sleep in [us]
             usleep(sleep_time_ns * 1e-3);
