--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -41,6 +41,10 @@
 file(GLOB_RECURSE bosdyn_protos_files CONFIGURE_DEPENDS
     RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "${API_protos_PATH}/*.proto")

+if (MSVC)
+  list(FILTER bosdyn_protos_files EXCLUDE REGEX "choreography|mission|autowalk")
+endif()
+
 if (bosdyn_protos_files)
   add_library(bosdyn_api_obj OBJECT ${bosdyn_protos_files})
   set_property(TARGET bosdyn_api_obj PROPERTY POSITION_INDEPENDENT_CODE 1)
@@ -155,6 +159,11 @@

 ### CLIENT LIBRARY ###
 file(GLOB_RECURSE bosdyn_client_SRC CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/bosdyn/*")
+
+if (MSVC)
+  list(FILTER bosdyn_client_SRC EXCLUDE REGEX "choreography|mission|autowalk")
+endif()
+
 if (BUILD_SHARED_LIBS)
   add_library (bosdyn_client SHARED ${bosdyn_client_SRC})
   set_property(TARGET bosdyn_client PROPERTY POSITION_INDEPENDENT_CODE 1)
