From 4966246b43c850c3244ee33b4fd03a0145b8570c Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 4 Oct 2021 17:45:49 -0600
Subject: [PATCH] Update Makefile

---
 feff/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/feff/Makefile b/feff/Makefile
index f6163fd..f33462c 100644
--- a/feff/Makefile
+++ b/feff/Makefile
@@ -1,7 +1,7 @@
 #note this needs a somewhat recent version of gfortran
 #version 4.1 is known not to work
-FC=gfortran
-FFLAGS=-O2
+#FC=gfortran
+#FFLAGS=-O2
 
 #for ifort
 #FC=ifort
